diff -up libmsn-4.2/msn/CMakeLists.txt.xmlParser_h libmsn-4.2/msn/CMakeLists.txt
--- libmsn-4.2/msn/CMakeLists.txt.xmlParser_h	2010-01-05 06:07:08.000000000 -0600
+++ libmsn-4.2/msn/CMakeLists.txt	2011-11-18 17:00:37.758298054 -0600
@@ -37,7 +37,7 @@ set(msn_HEADERS
     buddy.h
     passport.h
     sstream_fix.h
-    soap.h
+    soap.h xmlParser.h
     p2p.h
     msnobject.h
     libmsn_export.h)
