--- PyQt-x11-gpl-4.7/configure.py~	2010-03-04 04:54:35.000000000 +0100
+++ PyQt-x11-gpl-4.7/configure.py	2010-03-04 04:56:46.000000000 +0100
@@ -362,7 +364,7 @@
         check_module("QtWebKit", "qwebpage.h", "new QWebPage()")
         check_module("QtXml", "qdom.h", "new QDomDocument()")
         check_module("QtXmlPatterns", "qxmlname.h", "new QXmlName()")
-        check_module("phonon", "Phonon/VideoWidget",
+        check_module("phonon", "phonon/videowidget.h",
                 "new Phonon::VideoWidget()")
         check_module("QtAssistant", "qassistantclient.h",
                 "new QAssistantClient(\"foo\")", extra_lib_dirs=ass_lib_dirs,