--- sip/QtCore/qglobal.sip~	2010-09-20 15:10:28.000000000 +0200
+++ sip/QtCore/qglobal.sip	2010-09-30 15:00:13.804577344 +0200
@@ -312,12 +312,12 @@ public:
     // Qt.Alignment class.
     QFlags operator|(int f);
 %MethodCode
-        sipRes = new QFlags(*a0 | (ENUM(a1)));
+        sipRes = new QFlags(*a0 | a1);
 %End
 
     QFlags operator^(int f);
 %MethodCode
-        sipRes = new QFlags(*a0 ^ (ENUM(a1)));
+        sipRes = new QFlags(*a0 ^ a1);
 %End
 
     // These are necessary to prevent Python comparing object IDs.
