diff -Nur kdelibs-3.5.10.orig/configure kdelibs-3.5.10/configure
--- kdelibs-3.5.10.orig/configure	2008-08-20 18:06:05.000000000 +0200
+++ kdelibs-3.5.10/configure	2009-05-29 20:19:40.049141853 +0200
@@ -50392,7 +50392,7 @@
 
 #include <asm/unistd.h>
 #define _S390_BITOPS_H
-#include <linux/inotify.h>
+#include <sys/inotify.h>
 
 int
 main ()
diff -Nur kdelibs-3.5.10.orig/configure.in kdelibs-3.5.10/configure.in
--- kdelibs-3.5.10.orig/configure.in	2008-08-20 18:05:03.000000000 +0200
+++ kdelibs-3.5.10/configure.in	2009-05-29 20:19:40.069141952 +0200
@@ -1703,7 +1703,7 @@
   [
 #include <asm/unistd.h>
 #define _S390_BITOPS_H
-#include <linux/inotify.h>
+#include <sys/inotify.h>
   ],
   [
 #ifndef IN_ALL_EVENTS
diff -Nur kdelibs-3.5.10.orig/kio/kio/configure.in.in kdelibs-3.5.10/kio/kio/configure.in.in
--- kdelibs-3.5.10.orig/kio/kio/configure.in.in	2006-07-22 10:16:37.000000000 +0200
+++ kdelibs-3.5.10/kio/kio/configure.in.in	2009-05-29 20:19:40.092475716 +0200
@@ -124,7 +124,7 @@
   [
 #include <asm/unistd.h>
 #define _S390_BITOPS_H
-#include <linux/inotify.h>
+#include <sys/inotify.h>
   ],
   [
 #ifndef IN_ALL_EVENTS
diff -Nur kdelibs-3.5.10.orig/kio/kio/kdirwatch.cpp kdelibs-3.5.10/kio/kio/kdirwatch.cpp
--- kdelibs-3.5.10.orig/kio/kio/kdirwatch.cpp	2006-07-22 10:16:37.000000000 +0200
+++ kdelibs-3.5.10/kio/kio/kdirwatch.cpp	2009-05-29 20:21:43.978299306 +0200
@@ -70,9 +70,10 @@
 #include <sys/syscall.h>
 #include <linux/types.h>
 // Linux kernel headers are documented to not compile
-#define _S390_BITOPS_H
-#include <linux/inotify.h>
+/*#define _S390_BITOPS_H*/
+#include <sys/inotify.h>
 
+/*
 static inline int inotify_init (void)
 {
   return syscall (__NR_inotify_init);
@@ -87,6 +88,7 @@
 {
   return syscall (__NR_inotify_rm_watch, fd, wd);
 }
+*/
 
 #ifndef  IN_ONLYDIR
 #define  IN_ONLYDIR 0x01000000 
