Index: plugins/node.d.linux/df_inode.in
===================================================================
--- plugins/node.d.linux/df_inode.in	(revision 4638)
+++ plugins/node.d.linux/df_inode.in	(working copy)
@@ -14,7 +14,7 @@
 The plugin excludes per default the following special, read-only or
 dynamically allocating file systems from graphing:
 
-  none unknown iso9660 squashfs udf romfs ramfs
+  none unknown iso9660 squashfs udf romfs ramfs vfat debugfs simfs nilfs2 rootfs
 
 To change this set the environment variable "exclude" with a list of
 space separated fs types.  The environment variables "warning" and
@@ -24,7 +24,7 @@
 This configuration snipplet is an example with the defaults:
 
   [df_inode]
-    env.exclude none unknown iso9660 squashfs udf romfs ramfs debugfs
+    env.exclude none unknown iso9660 squashfs udf romfs ramfs vfat debugfs simfs nilfs2 rootfs
     env.warning 92
     env.critical 98
 
@@ -87,7 +87,7 @@
 # For these devices use the mount point, the device is useless
 my %usemntpt = ( tmpfs => 1, none => 1, udev => 1 );
 
-my $exclude = $ENV{'exclude'} || 'none unknown iso9660 squashfs udf romfs ramfs vfat debugfs simfs';
+my $exclude = $ENV{'exclude'} || 'none unknown iso9660 squashfs udf romfs ramfs vfat debugfs simfs nilfs2 rootfs';
 my $dfopts  = "-P -l -i ".join(' -x ',('',split('\s+',$exclude)));
 
 my $mode = ($ARGV[0] or "print");
