diff -ur munin-2.0.1.orig/master/_bin/munin-cgi-graph.in munin-2.0.1/master/_bin/munin-cgi-graph.in
--- munin-2.0.1.orig/master/_bin/munin-cgi-graph.in	2012-06-16 11:05:43.000000000 +0000
+++ munin-2.0.1/master/_bin/munin-cgi-graph.in	2012-07-01 22:37:11.166756888 +0000
@@ -27,6 +27,9 @@
 use strict;
 use warnings;
 use IO::Handle;
+BEGIN {
+    $Date::Manip::Backend = 'DM5';
+}
 use Date::Manip;
 use POSIX qw(strftime locale_h);
 use CGI::Fast qw(:cgi);
diff -ur munin-2.0.1.orig/master/_bin/munin-graph.in munin-2.0.1/master/_bin/munin-graph.in
--- munin-2.0.1.orig/master/_bin/munin-graph.in	2012-06-16 11:05:43.000000000 +0000
+++ munin-2.0.1/master/_bin/munin-graph.in	2012-07-01 22:37:34.549926610 +0000
@@ -27,6 +27,9 @@
 use strict;
 use warnings;
 use IO::Handle;
+BEGIN {
+    $Date::Manip::Backend = 'DM5';
+}
 use Date::Manip;
 use POSIX qw(strftime);
 use Time::HiRes qw(gettimeofday tv_interval);
