--- phonon-4.6.0/includes/CMakeLists.txt~	2013-02-25 18:10:04.769629747 +0000
+++ phonon-4.6.0/includes/CMakeLists.txt	2013-02-25 18:10:19.366188789 +0000
@@ -51,7 +51,7 @@
     phonon/VolumeFaderEffect
     phonon/VolumeFaderInterface
     phonon/VolumeSlider
-DESTINATION ${INCLUDE_INSTALL_DIR}/phonon COMPONENT Devel)
+DESTINATION ${INCLUDE_INSTALL_DIR}/qt4/phonon COMPONENT Devel)
 
 
 install( FILES
