diff -ruN pm-utils-0.99.4.orig/pm/functions pm-utils-0.99.4/pm/functions
--- pm-utils-0.99.4.orig/pm/functions	2007-05-15 14:55:58.000000000 +0000
+++ pm-utils-0.99.4/pm/functions	2007-12-26 13:17:15.000000000 +0000
@@ -52,7 +52,7 @@
 	rm -f /var/run/pm-suspend
 	chvt 1
 	chvt $VT
-	openvt -- sh -c "usleep $1 ; rm -f /.suspended >/dev/null 2>&1 0<&1" >/dev/null 2>&1 0<&1 &
+	openvt -- sh -c "sleep $1 ; rm -f /.suspended >/dev/null 2>&1 0<&1" >/dev/null 2>&1 0<&1 &
 }
 
 find_sleepd_files()
@@ -153,7 +153,7 @@
 
 	run_hooks "$2" reverse
 
-	remove_suspend_lock 200
+	remove_suspend_lock 0.0002
 
 	return 0
 }
diff -ruN pm-utils-0.99.4.orig/pm/hooks/99video pm-utils-0.99.4/pm/hooks/99video
--- pm-utils-0.99.4.orig/pm/hooks/99video	2007-06-18 20:39:53.000000000 +0000
+++ pm-utils-0.99.4/pm/hooks/99video	2007-12-26 13:17:53.000000000 +0000
@@ -47,7 +47,7 @@
 	# We might need to do one or many of these quirks
 	if [ "${DISPLAY_QUIRK_VBE_POST}" == "true" ]; then
 		vbe post
-		usleep 100000
+		usleep 0.1
 	fi
 	if [ "${DISPLAY_QUIRK_VBESTATE_RESTORE}" == "true" ]; then
 		vbe vbestate restore < /var/run/vbestate
