from upstream git

diff --git a/src/Clients/MainApp/FSpot/GroupSelector.cs b/src/Clients/MainApp/FSpot/GroupSelector.cs
index b1e4993..63edd84 100644
--- a/src/Clients/MainApp/FSpot/GroupSelector.cs
+++ b/src/Clients/MainApp/FSpot/GroupSelector.cs
@@ -467,7 +467,7 @@ namespace FSpot {
 				bounds.Height = selector.background.Height;
 				bounds.Y = selector.background.Y;
 				bounds.X = selector.BoxX (item);
-				bounds.Width = Math.Max (selector.BoxX (item + 1) - bounds.X, 1);
+				bounds.Width = Math.Max (selector.BoxX (item + 1) - selector.BoxX (item), 1);
 
 				if (item < 0 || item > selector.box_counts.Length - 1)
 					return;
