--- extensions/SmugMugExport/Makefile.in.orig	2008-06-01 15:30:29.000000000 +0000
+++ extensions/SmugMugExport/Makefile.in	2008-06-01 15:30:53.000000000 +0000
@@ -314,7 +314,7 @@
 	-r:$(DIR_TAO_EXTENSIONLOADER)/Tao.OpenGl.ExtensionLoader.dll
 
 CSC_FLAGS = -debug
-CSC_LIB = gmcs $(CSC_FLAGS) -target:library
+CSC_LIB = gmcs -target:library
 MONO_DEBUGFLAGS = --debug
 RUNTIME = mono $(MONO_DEBUGFLAGS)
 PLUGIN_NAME = SmugMugExport
@@ -702,7 +702,7 @@
 
 $(PLUGIN_ASSEMBLY): $(PLUGIN_SOURCES) $(PLUGIN_MANIFEST)
 	$(MAKE) -C $(SUBDIRS)
-	$(CSC_LIB) -out:$@ $(PLUGIN_SOURCES) $(REFS) $(PKGS) $(ASSEMBLIES) $(RESOURCES)
+	$(CSC_LIB) -out:$@ $(PLUGIN_SOURCES) $(REFS) $(PKGS) $(ASSEMBLIES) $(RESOURCES) -g
 
 install-data-hook:
 	rm -f $(plugindir)/$(PLUGIN_NAME).addin.xml
