--- geeqie-1.7.1/Makefile.am	2022-01-17 15:26:31.000000000 +0100
+++ geeqie-1.7.1/Makefile.am.new	2022-01-17 15:31:28.796386390 +0100
@@ -10,9 +10,9 @@
 readmedir = @readmedir@
 
 if HAVE_MARKDOWN
-readme_DATA = README.md COPYING ChangeLog TODO README.lirc AUTHORS README.html ChangeLog.html
+readme_DATA = README.md COPYING ChangeLog.gqview TODO README.lirc AUTHORS README.html ChangeLog.html
 else
-readme_DATA = README.md COPYING ChangeLog TODO README.lirc AUTHORS ChangeLog.html
+readme_DATA = README.md COPYING ChangeLog.gqview TODO README.lirc AUTHORS
 endif
 
 desktopdir = $(datadir)/applications
