Index: tw.color.c
===================================================================
RCS file: /p/tcsh/cvsroot/tcsh/tw.color.c,v
retrieving revision 1.24
diff -u -u -r1.24 tw.color.c
--- tw.color.c	2 Mar 2006 18:46:45 -0000	1.24
+++ tw.color.c	17 Oct 2008 19:03:56 -0000
@@ -85,6 +85,7 @@
     VAR(NOS, "tw", ""),		/* Sticky and other writable dir (+t,o+w) */
     VAR(NOS, "ow", ""),		/* Other writable dir (o+w) but not sticky */
     VAR(NOS, "st", ""),		/* Sticky dir (+t) but not other writable */
+    VAR(NOS, "rs", "0"),	/* Reset to normal color */
 };
 
 enum FileType {
