From 1e335af70f29d1a1e9c132338aa35b8971934441 Mon Sep 17 00:00:00 2001
From: Dave Reisner <dreisner@archlinux.org>
Date: Fri, 19 Apr 2013 16:31:25 -0400
Subject: [PATCH] nss-myhostname: ensure that glibc's assert is used

---
 src/nss-myhostname/nss-myhostname.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/nss-myhostname/nss-myhostname.c b/src/nss-myhostname/nss-myhostname.c
index 16ccb3e..8699098 100644
--- a/src/nss-myhostname/nss-myhostname.c
+++ b/src/nss-myhostname/nss-myhostname.c
@@ -25,7 +25,6 @@
 #include <netdb.h>
 #include <errno.h>
 #include <string.h>
-#include <assert.h>
 #include <unistd.h>
 #include <net/if.h>
 #include <stdlib.h>
@@ -35,6 +34,11 @@
 #include "macro.h"
 #include "util.h"
 
+/* Ensure that glibc's assert is used. We cannot use assert from macro.h, as
+ * libnss_myhostname will be linked into arbitrary programs which will, in turn
+ * attempt to write to the journal via log_dispatch() */
+#include <assert.h>
+
 /* We use 127.0.0.2 as IPv4 address. This has the advantage over
  * 127.0.0.1 that it can be translated back to the local hostname. For
  * IPv6 we use ::1 which unfortunately will not translate back to the
-- 
1.8.2.1

