From a0d4d089903e53fb7aaa758354f69c00eab4e93a Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <jan.steffens@gmail.com>
Date: Tue, 10 Sep 2019 18:08:25 +0000
Subject: [PATCH 2/8] meson: Install p11-kit-client module

---
 p11-kit/meson.build | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/p11-kit/meson.build b/p11-kit/meson.build
index f410ee7..beb0fb2 100644
--- a/p11-kit/meson.build
+++ b/p11-kit/meson.build
@@ -86,7 +86,9 @@ if host_system != 'windows'
                 'client.c', 'client-init.c',
                 name_prefix: '',
                 include_directories: [configinc, commoninc],
-                link_with: [libp11_kit_internal])
+                link_with: [libp11_kit_internal],
+                install: true,
+                install_dir: p11_module_path)
 endif
 
 fixtures_dir = meson.current_source_dir() / 'fixtures'
-- 
2.23.0

