From 20f3c418a1577e0fa48a8e9a734bd6d5278a1ca9 Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <jan.steffens@gmail.com>
Date: Tue, 10 Sep 2019 18:08:25 +0000
Subject: [PATCH 6/8] meson: Install p11-kit-server units

---
 p11-kit/meson.build | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/p11-kit/meson.build b/p11-kit/meson.build
index 18b3d26..48a3c75 100644
--- a/p11-kit/meson.build
+++ b/p11-kit/meson.build
@@ -176,6 +176,17 @@ executable('p11-kit-server-testable',
            dependencies: [libp11_tool_dep] + libsystemd_deps + libffi_deps + dlopen_deps,
            link_with: libp11_kit_testable)
 
+if with_systemd
+  p11_kit_server_service_conf = configuration_data({
+    'bindir': prefix / bindir
+  })
+  p11_kit_server_service = configure_file(input: 'p11-kit-server.service.in',
+                                          output: '@BASENAME@',
+                                          configuration: p11_kit_server_service_conf)
+  install_data([p11_kit_server_service, 'p11-kit-server.socket'],
+               install_dir: systemduserunitdir)
+endif
+
 # Check if compilation succeeds with CRYPTOKI_GNU=1
 
 pkcs11_gnu_headers = [
-- 
2.23.0

