--- bzip2-1.0.7/Makefile-libbz2_so.orig	2019-06-29 09:08:21.186324313 +0000
+++ bzip2-1.0.7/Makefile-libbz2_so	2019-06-29 09:09:37.397240774 +0000
@@ -35,7 +35,7 @@
       bzlib.o
 
 all: $(OBJS)
-	$(CC) -shared -Wl,-soname -Wl,libbz2.so.1.0 -o libbz2.so.1.0.7 $(OBJS)
+	$(CC) -shared -Wl,-soname -Wl,libbz2.so.1.0 -o libbz2.so.1.0.7 $(OBJS) $(LDFLAGS)
 	$(CC) $(CFLAGS) -o bzip2-shared bzip2.c libbz2.so.1.0.7
 	rm -f libbz2.so.1.0
 	ln -s libbz2.so.1.0.7 libbz2.so.1.0
