diff -Naur file-5.05-old//magic/Magdir/archive file-5.05/magic/Magdir/archive
--- file-5.05-old//magic/Magdir/archive	2011-01-08 06:24:25.000000000 +1000
+++ file-5.05/magic/Magdir/archive	2011-01-19 14:37:47.766673895 +1000
@@ -688,6 +688,7 @@
 >>4	byte		0x0b		\b, at least v1.1 to extract
 >>0x161	string		WINZIP		\b, WinZIP self-extracting
 >>4	byte		0x14		\b, at least v2.0 to extract
+>>4	byte		0x2d		\b, at least v3.0 to extract
 
 # Zoo archiver
 20	lelong		0xfdc4a7dc	Zoo archive data
