diff -up flex-2.5.35/flex.skl\~ flex-2.5.35/flex.skl
--- flex-2.5.35/flex.skl~	2010-07-13 17:18:43.000000000 +0200
+++ flex-2.5.35/flex.skl	2010-07-13 17:23:49.000000000 +0200
@@ -960,6 +960,22 @@ m4_ifdef( [[M4_YY_NO_SET_LINENO]],,
 void yyset_lineno M4_YY_PARAMS( int line_number M4_YY_PROTO_LAST_ARG );
 ]])
 
+m4_ifdef( [[M4_YY_REENTRANT]],
+[[
+m4_ifdef( [[M4_YY_NO_GET_COLUMN]],,
+[[
+int yyget_column  M4_YY_PARAMS( M4_YY_PROTO_ONLY_ARG );
+]])
+]])
+
+m4_ifdef( [[M4_YY_REENTRANT]],
+[[
+m4_ifdef( [[M4_YY_NO_SET_COLUMN]],,
+[[
+void yyset_column M4_YY_PARAMS( int column_no M4_YY_PROTO_LAST_ARG );
+]])
+]])
+
 %if-bison-bridge
 m4_ifdef( [[M4_YY_NO_GET_LVAL]],,
 [[

Diff finished.  Tue Jul 13 17:27:50 2010
--- flex-2.5.35/skel.c~	2010-07-14 13:15:42.000000000 +0200
+++ flex-2.5.35/skel.c	2010-07-14 13:16:05.000000000 +0200
@@ -1027,6 +1027,22 @@
   "void yyset_lineno M4_YY_PARAMS( int line_number M4_YY_PROTO_LAST_ARG );",
   "]])",
   "",
+  "m4_ifdef( [[M4_YY_REENTRANT]],",
+  "[[",
+  "m4_ifdef( [[M4_YY_NO_GET_COLUMN]],,",
+  "[[",
+  "int yyget_column  M4_YY_PARAMS( M4_YY_PROTO_ONLY_ARG );",
+  "]])",
+  "]])",
+  "",
+  "m4_ifdef( [[M4_YY_REENTRANT]],",
+  "[[",
+  "m4_ifdef( [[M4_YY_NO_SET_COLUMN]],,",
+  "[[",
+  "void yyset_column M4_YY_PARAMS( int column_no M4_YY_PROTO_LAST_ARG );",
+  "]])",
+  "]])",
+  "",
   "%if-bison-bridge",
   "m4_ifdef( [[M4_YY_NO_GET_LVAL]],,",
   "[[",
