diff --git a/utils.c b/utils.c
index e9cb3a82f..f867e5a7f 100644
--- a/utils.c
+++ b/utils.c
@@ -2484,11 +2484,6 @@ const char *subvol_strip_mountpoint(const char *mnt, const char *full_path)
 	if (!len)
 		return full_path;
 
-	if ((strncmp(mnt, full_path, len) != 0) || (full_path[len] != '/')) {
-		error("not on mount point: %s", mnt);
-		exit(1);
-	}
-
 	if (mnt[len - 1] != '/')
 		len += 1;
 
