diff -Naur gpm-1.20.1-orig/src/lib/liblow.c gpm-1.20.1/src/lib/liblow.c
--- gpm-1.20.1-orig/src/lib/liblow.c	2002-12-24 14:57:16.000000000 -0800
+++ gpm-1.20.1/src/lib/liblow.c	2005-12-05 00:14:05.000000000 -0800
@@ -198,6 +198,11 @@
    struct winsize win;
    Gpm_Stst *new = NULL;
    char* sock_name = 0;
+   /* don't bother continuing if the control socket isn't there */
+   struct stat sbuf;
+   if(stat(GPM_NODE_CTL, &sbuf)) {
+      return -1;
+   }
 
    option.consolename = NULL;
 
@@ -369,7 +374,7 @@
 
   /*....................................... Error: free all memory */
    err:
-   gpm_report(GPM_PR_ERR,"Oh, oh, it's an error! possibly I die! ");
+   /*gpm_report(GPM_PR_ERR,"Oh, oh, it's an error! possibly I die! ");*/
    do {
       new=gpm_stack->next;
       free(gpm_stack);
