Description: Add missing include to fix FTBFS with eglibc 2.17
Origin: vendor, http://patches.ubuntu.com/j/jfsutils/jfsutils_1.1.15-2ubuntu1.patch
Bug-Debian: http://bugs.debian.org/701433
Last-Update: 2013-07-05

Index: b/fscklog/extract.c
===================================================================
--- a/fscklog/extract.c	2006-06-05 19:31:40.000000000 +0000
+++ b/fscklog/extract.c	2013-04-02 07:13:08.737654963 +0000
@@ -28,6 +28,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <unistd.h>
+#include <sys/types.h>
 
 #include "devices.h"
 #include "diskmap.h"
