From 0348f788d9dc1ac35ff07d2a15acc9da109ca60c Mon Sep 17 00:00:00 2001
From: Richard Hughes <richard@hughsie.com>
Date: Mon, 14 Dec 2009 12:35:14 +0000
Subject: Fix segfault in the history code. Fixes fd#25632

---
diff --git a/src/dkp-history.c b/src/dkp-history.c
index 23bf4c2..b851642 100644
--- a/src/dkp-history.c
+++ b/src/dkp-history.c
@@ -295,6 +295,12 @@ dkp_history_get_profile_data (DkpHistory *history, gboolean charging)
 
 		/* round to the nearest int */
 		bin = rint (obj->value);
+
+		/* ensure bin is in range */
+		if (bin >= data->len)
+			bin = data->len - 1;
+
+		/* different */
 		if (oldbin != bin) {
 			oldbin = bin;
 			if (obj_old != NULL) {
--
cgit v0.8.3-6-g21f6
