--- doc/de/index.docbook~	2006-02-11 14:44:04.000000000 +0100
+++ doc/de/index.docbook	2006-06-10 10:23:51.000000000 +0200
@@ -110,7 +110,7 @@
 <userinput><command>oooqs2</command></userinput> in dem Ausführen-Dialog des Desktops eingeben.
 Diesen Dialog können Sie erreichen, in dem Sie die Tastenkombination
 <keycombo action="simul">&Alt;<keycap>F2</keycap></keycombo> betätigen oder indem Sie
-<guimenuitem>Befehl ausführen...</guiitem> im <guimenu>K-Menü</guimenu> wählen.
+<guimenuitem>Befehl ausführen...</guimenuitem> im <guimenu>K-Menü</guimenu> wählen.
 </para>
 </chapter>
 
@@ -341,7 +341,7 @@
 Wenn Sie Einstellungen des Applets global speichern möchten, so dass
 diese für andere Benutzer des Systems verwandt werden, kopieren oder verschieben
 Sie ihre persönliche Einstellungsdatei nach
-<filename>$<envvar>KDEDIR</envvar>/share/config/oooqs2rc</filename>.
+<filename>$<envar>KDEDIR</envar>/share/config/oooqs2rc</filename>.
 </para>
 </answer>
 </qandaentry>
