diff -Nur gst-plugins-ugly-0.10.13.orig/ext/x264/gstx264enc.c gst-plugins-ugly-0.10.13/ext/x264/gstx264enc.c
--- gst-plugins-ugly-0.10.13.orig/ext/x264/gstx264enc.c	2010-01-10 22:43:15.000000000 +0200
+++ gst-plugins-ugly-0.10.13/ext/x264/gstx264enc.c	2010-01-10 22:42:42.000000000 +0200
@@ -582,7 +582,7 @@
   encoder->x264param.analyse.i_noise_reduction = encoder->noise_reduction;
   encoder->x264param.i_frame_reference = encoder->ref;
   encoder->x264param.i_bframe = encoder->bframes;
-  encoder->x264param.b_bframe_pyramid = encoder->b_pyramid;
+  encoder->x264param.i_bframe_pyramid = encoder->b_pyramid;
 #if X264_BUILD < 63
   encoder->x264param.b_bframe_adaptive = encoder->b_adapt;
 #else
