--- licq-1.8.2/src/licq.cpp.orig	2016-05-12 15:27:35.927500283 +0000
+++ licq-1.8.2/src/licq.cpp	2016-05-12 15:28:01.107415744 +0000
@@ -568,18 +568,18 @@
         if (!licqConf.get(szKey, pluginName))
           continue;
 
-        bool loaded = LoadPlugin(pluginName, argc, argv);
+        bool loaded= (bool)LoadPlugin(pluginName, argc, argv);
 
         // Make upgrade from 1.3.x and older easier by automatically switching from kde/qt-gui to kde4/qt4-gui
         if (!loaded && pluginName == "kde-gui")
         {
           gLog.warning(tr("Plugin kde-gui is no longer available, trying to load kde4-gui instead."));
-          loaded = LoadPlugin("kde4-gui", argc, argv);
+          loaded= (bool)LoadPlugin("kde4-gui", argc, argv);
         }
         if (!loaded && (pluginName == "qt-gui" || pluginName == "kde-gui"))
         {
           gLog.warning(tr("Plugin %s is no longer available, trying to load qt4-gui instead."), pluginName.c_str());
-          loaded = LoadPlugin("qt4-gui", argc, argv);
+          loaded= (bool)LoadPlugin("qt4-gui", argc, argv);
         }
 
         if (!loaded)
