diff -Naur ivtv-utils-1.3.0-orig/test/Makefile ivtv-utils-1.3.0/test/Makefile
--- ivtv-utils-1.3.0-orig/test/Makefile	2008-12-17 01:08:23.000000000 +0100
+++ ivtv-utils-1.3.0/test/Makefile	2008-12-17 01:08:51.000000000 +0100
@@ -1,5 +1,4 @@
 INSTALLDIR = /usr/local/bin
-HEADERDIR = /usr/include/linux
 
 EXES = vbi wss vbi-passthrough vbi-detect ivtv-pcm-tester \
 	mpeg-read-test mpeg-freq-test mpeg-read-sleep-test \
diff -Naur ivtv-utils-1.3.0-orig/utils/Makefile ivtv-utils-1.3.0/utils/Makefile
--- ivtv-utils-1.3.0-orig/utils/Makefile	2008-12-17 01:08:23.000000000 +0100
+++ ivtv-utils-1.3.0/utils/Makefile	2008-12-17 01:09:21.000000000 +0100
@@ -1,6 +1,5 @@
-PREFIX = /usr/local
+PREFIX = /usr
 BINDIR = $(PREFIX)/bin
-HDRDIR = /usr/include/linux
 
 X86  := $(shell if echo - | $(CC) -E -dM - | egrep -e '__i386__|__x86_64__' >/dev/null; \
 	then echo y; else echo n; fi)
@@ -13,9 +12,6 @@
 
 BIN := $(EXES) ivtv-tune/ivtv-tune cx25840ctl/cx25840ctl
 
-
-HEADERS := linux/ivtv.h linux/ivtvfb.h
-
 CFLAGS = -D_GNU_SOURCE -O2 -Wall -g -I.
 CXXFLAGS = $(CFLAGS)
 
@@ -42,8 +38,6 @@
 	$(CC) -lpthread -o $@ $^
 
 install: all
-	install -d $(DESTDIR)/$(HDRDIR)
-	install -m 0644 $(HEADERS) $(DESTDIR)/$(HDRDIR)
 	install -d $(DESTDIR)/$(BINDIR)
 	install -m 0755 $(BIN) $(DESTDIR)/$(BINDIR)
 
diff -Naur ivtv-utils-1.3.0-orig/utils/linux/ivtv.h ivtv-utils-1.3.0/utils/linux/ivtv.h
--- ivtv-utils-1.3.0-orig/utils/linux/ivtv.h	2008-12-17 01:08:23.000000000 +0100
+++ ivtv-utils-1.3.0/utils/linux/ivtv.h	1970-01-01 01:00:00.000000000 +0100
@@ -1,67 +0,0 @@
-/*
-    Public ivtv API header
-    Copyright (C) 2003-2004  Kevin Thayer <nufan_wfk at yahoo.com>
-    Copyright (C) 2004-2007  Hans Verkuil <hverkuil@xs4all.nl>
-
-    This program is free software; you can redistribute it and/or modify
-    it under the terms of the GNU General Public License as published by
-    the Free Software Foundation; either version 2 of the License, or
-    (at your option) any later version.
-
-    This program is distributed in the hope that it will be useful,
-    but WITHOUT ANY WARRANTY; without even the implied warranty of
-    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-    GNU General Public License for more details.
-
-    You should have received a copy of the GNU General Public License
-    along with this program; if not, write to the Free Software
-    Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
- */
-
-#ifndef _LINUX_IVTV_H
-#define _LINUX_IVTV_H
-
-#include <linux/types.h>
-
-/* ivtv knows several distinct output modes: MPEG streaming,
-   YUV streaming, YUV updates through user DMA and the passthrough
-   mode.
-
-   In order to clearly tell the driver that we are in user DMA
-   YUV mode you need to call IVTV_IOC_DMA_FRAME with y_source == NULL
-   first (althrough if you don't then the first time
-   DMA_FRAME is called the mode switch is done automatically).
-
-   When you close the file handle the user DMA mode is exited again.
-
-   While in one mode, you cannot use another mode (EBUSY is returned).
-
-   All this means that if you want to change the YUV interlacing
-   for the user DMA YUV mode you first need to do call IVTV_IOC_DMA_FRAME
-   with y_source == NULL before you can set the correct format using
-   VIDIOC_S_FMT.
-
-   Eventually all this should be replaced with a proper V4L2 API,
-   but for now we have to do it this way. */
-
-struct ivtv_dma_frame {
-	enum v4l2_buf_type type; /* V4L2_BUF_TYPE_VIDEO_OUTPUT */
-	__u32 pixelformat;	 /* 0 == same as destination */
-	void *y_source;   /* if NULL and type == V4L2_BUF_TYPE_VIDEO_OUTPUT,
-				    then just switch to user DMA YUV output mode */
-	void *uv_source;  /* Unused for RGB pixelformats */
-	struct v4l2_rect src;
-	struct v4l2_rect dst;
-	__u32 src_width;
-	__u32 src_height;
-};
-
-#define IVTV_IOC_DMA_FRAME  _IOW ('V', BASE_VIDIOC_PRIVATE+0, struct ivtv_dma_frame)
-
-/* These are the VBI types as they appear in the embedded VBI private packets. */
-#define IVTV_SLICED_TYPE_TELETEXT_B     (1)
-#define IVTV_SLICED_TYPE_CAPTION_525    (4)
-#define IVTV_SLICED_TYPE_WSS_625        (5)
-#define IVTV_SLICED_TYPE_VPS            (7)
-
-#endif /* _LINUX_IVTV_H */
diff -Naur ivtv-utils-1.3.0-orig/utils/linux/ivtvfb.h ivtv-utils-1.3.0/utils/linux/ivtvfb.h
--- ivtv-utils-1.3.0-orig/utils/linux/ivtvfb.h	2008-12-17 01:08:23.000000000 +0100
+++ ivtv-utils-1.3.0/utils/linux/ivtvfb.h	1970-01-01 01:00:00.000000000 +0100
@@ -1,37 +0,0 @@
-/*
-    On Screen Display cx23415 Framebuffer driver
-
-    Copyright (C) 2006, 2007  Ian Armstrong <ian@iarmst.demon.co.uk>
-
-    This program is free software; you can redistribute it and/or modify
-    it under the terms of the GNU General Public License as published by
-    the Free Software Foundation; either version 2 of the License, or
-    (at your option) any later version.
-
-    This program is distributed in the hope that it will be useful,
-    but WITHOUT ANY WARRANTY; without even the implied warranty of
-    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-    GNU General Public License for more details.
-
-    You should have received a copy of the GNU General Public License
-    along with this program; if not, write to the Free Software
-    Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
- */
-
-#ifndef __LINUX_IVTVFB_H__
-#define __LINUX_IVTVFB_H__
-
-#include <linux/types.h>
-
-/* Framebuffer external API */
-
-struct ivtvfb_dma_frame {
-	void __user *source;
-	unsigned long dest_offset;
-	int count;
-};
-
-#define IVTVFB_IOC_DMA_FRAME 	_IOW('V', BASE_VIDIOC_PRIVATE+0, struct ivtvfb_dma_frame)
-#define FBIO_WAITFORVSYNC	_IOW('F', 0x20, u_int32_t)
-
-#endif
