diff --git a/cmake-modules/KDEPython.cmake b/cmake-modules/KDEPython.cmake
index 8209c98c..b687a230 100644
--- a/cmake-modules/KDEPython.cmake
+++ b/cmake-modules/KDEPython.cmake
@@ -108,13 +108,13 @@ MACRO(KDE_ADD_PYTHON_EXECUTABLE _pyname _exename)
     set(TARGET ${CMAKE_INSTALL_PREFIX}/${DATA_INSTALL_DIR}/${PROJECT_NAME}/${_pyname})
     set(LINK_NAME ${CMAKE_INSTALL_PREFIX}/${BIN_INSTALL_DIR}/${_exename})
 
-    GET_FILENAME_COMPONENT(abs_link_name $ENV{DESTDIR}/${LINK_NAME} ABSOLUTE)
-    GET_FILENAME_COMPONENT(link_path $ENV{DESTDIR}/${LINK_NAME} PATH)
+    GET_FILENAME_COMPONENT(abs_link_name ${LINK_NAME} ABSOLUTE)
+    GET_FILENAME_COMPONENT(link_path ${LINK_NAME} PATH)
     GET_FILENAME_COMPONENT(abs_link_path ${link_path} ABSOLUTE)
-    FILE(MAKE_DIRECTORY ${abs_link_path})
     GET_FILENAME_COMPONENT(abs_target ${TARGET} ABSOLUTE)
 
-    INSTALL(CODE "EXECUTE_PROCESS(COMMAND ${CMAKE_COMMAND} -E create_symlink ${abs_target} ${abs_link_name})")
-    INSTALL(CODE "EXECUTE_PROCESS(COMMAND chmod a+x $ENV{DESTDIR}/${abs_target})")
+    INSTALL(CODE "EXECUTE_PROCESS(COMMAND ${CMAKE_COMMAND} -E make_directory \$ENV\{DESTDIR\}${abs_link_path})")
+    INSTALL(CODE "EXECUTE_PROCESS(COMMAND ${CMAKE_COMMAND} -E create_symlink ${abs_target} \$ENV\{DESTDIR\}${abs_link_name})")
+    INSTALL(CODE "EXECUTE_PROCESS(COMMAND chmod a+x \$ENV\{DESTDIR\}/${abs_target})")
 
 ENDMACRO(KDE_ADD_PYTHON_EXECUTABLE)
