diff --git i/examples/alffplay.cpp w/examples/alffplay.cpp
index b9739e7b..806ef10c 100644
--- i/examples/alffplay.cpp
+++ w/examples/alffplay.cpp
@@ -1804,7 +1804,7 @@ int MovieState::streamComponentOpen(unsigned int stream_index)
     if(avcodec_parameters_to_context(avctx.get(), mFormatCtx->streams[stream_index]->codecpar))
         return -1;
 
-    AVCodec *codec{avcodec_find_decoder(avctx->codec_id)};
+    const AVCodec *codec{avcodec_find_decoder(avctx->codec_id)};
     if(!codec || avcodec_open2(avctx.get(), codec, nullptr) < 0)
     {
         std::cerr<< "Unsupported codec: "<<avcodec_get_name(avctx->codec_id)
