diff -upr smplayer-14.9.0.6994.orig/src/qtsingleapplication/qtlocalpeer.cpp smplayer-14.9.0.6994/src/qtsingleapplication/qtlocalpeer.cpp
--- smplayer-14.9.0.6994.orig/src/qtsingleapplication/qtlocalpeer.cpp	2013-08-30 03:53:57.000000000 +0300
+++ smplayer-14.9.0.6994/src/qtsingleapplication/qtlocalpeer.cpp	2015-08-28 11:37:18.435899495 +0300
@@ -41,6 +41,7 @@
 
 #include "qtlocalpeer.h"
 #include <QCoreApplication>
+#include <QDataStream>
 #include <QTime>
 
 #if defined(Q_OS_WIN)
