--- Makefile.old	2008-03-24 07:02:35.000000000 -0500
+++ Makefile	2008-03-24 07:02:53.000000000 -0500
@@ -458,7 +458,7 @@
 #
 lib_LTLIBRARIES = libarchive.la
 man_MANS = $(libarchive_man_MANS) $(bsdtar_man_MANS) $(bsdcpio_man_MANS)
-BUILT_SOURCES = libarchive/test/list.h tar/test/list.h cpio/test/list.h $(libarchive_test_REFFILES) $(bsdcpio_test_REFFILES)
+BUILT_SOURCES = libarchive/test/list.h tar/test/list.h cpio/test/list.h
 TESTS_ENVIRONMENT = $(libarchive_TESTS_ENVIRONMENT) $(bsdtar_TESTS_ENVIRONMENT) $(bsdcpio_TESTS_ENVIRONMENT)
 # Always build and test both bsdtar and bsdcpio as part of 'distcheck'
 DISTCHECK_CONFIGURE_FLAGS = --enable-bsdtar --enable-bsdcpio
