Index: libarchive/archive_write_set_format_pax.c
===================================================================
--- libarchive/archive_write_set_format_pax.c	(revision 4094)
+++ libarchive/archive_write_set_format_pax.c	(working copy)
@@ -1647,7 +1647,7 @@
 			return (total);
 
 		p = ((const unsigned char *)buff) + total;
-		ws = s;
+		ws = s - total;
 		if (ws > pax->sparse_list->remaining)
 			ws = pax->sparse_list->remaining;
 
