diff -Nur orage-4.7.5.16.orig/src/reminder.c orage-4.7.5.16/src/reminder.c
--- orage-4.7.5.16.orig/src/reminder.c	2010-03-04 00:34:55.000000000 -0700
+++ orage-4.7.5.16/src/reminder.c	2010-11-05 13:32:22.317539758 -0600
@@ -549,12 +549,8 @@
     strncpy(heading,  _("Reminder "), 199);
     if (alarm->title)
         g_strlcat(heading, alarm->title, 50);
-    n = notify_notification_new(heading, alarm->description, NULL, NULL);
+    n = notify_notification_new(heading, alarm->description, NULL);
     alarm->active_alarm->active_notify = n;
-    if (g_par.trayIcon 
-    && gtk_status_icon_is_embedded((GtkStatusIcon *)g_par.trayIcon))
-        notify_notification_attach_to_status_icon(n
-                , (GtkStatusIcon *)g_par.trayIcon);
 
     if (alarm->notify_timeout == -1)
         notify_notification_set_timeout(n, NOTIFY_EXPIRES_NEVER);
