diff -u -r totem-3.20.1/src/backend/bacon-video-widget.c totem-3.20.1-i686/src/backend/bacon-video-widget.c
--- totem-3.20.1/src/backend/bacon-video-widget.c	2016-03-22 17:45:32.000000000 +0100
+++ totem-3.20.1-i686/src/backend/bacon-video-widget.c	2016-04-13 22:31:35.477858516 +0200
@@ -2334,7 +2334,8 @@
     if (!gst_toc_entry_get_start_stop_times (entry, &start, &stop)) {
       GST_DEBUG ("Chapter #%d (couldn't get times)", i);
     } else {
-      GST_DEBUG ("Chapter #%d (start: %li stop: %li)", i, start, stop);
+      GST_DEBUG ("Chapter #%d (start: %" G_GINT64_FORMAT " stop: %" G_GINT64_FORMAT ")",
+          i, start, stop);
     }
   }
 
