--- dictionaries/prj/build.lst	2008-08-15 10:41:08.000000000 +0000
+++ dictionaries/prj/build.lst.new	2008-09-17 16:11:22.000000000 +0000
@@ -4,10 +4,10 @@
 di      dictionaries\cs_CZ      nmake   -       all     di_cs_CZ NULL
 di      dictionaries\da_DK      nmake   -       all     di_da_DK NULL
 di      dictionaries\de_AT      nmake   -       all     di_de_AT NULL
-di      dictionaries\de_CH      nmake   -       all     di_de_CH NULL
+di      dictionaries\de_CH      nmake   -       all     di_de_CH di_de_AT NULL
 di      dictionaries\de_DE      nmake   -       all     di_de_DE di_de_AT di_de_CH NULL
 di      dictionaries\en_US      nmake   -       all     di_en_US NULL
-di      dictionaries\en_ZA      nmake   -       all     di_en_ZA NULL
+di      dictionaries\en_ZA      nmake   -       all     di_en_ZA di_en_US NULL
 di      dictionaries\en_GB      nmake   -       all     di_en_GB di_en_US di_en_ZA NULL
 di      dictionaries\es_ES      nmake   -       all     di_es_ES NULL
 di      dictionaries\et_EE      nmake   -       all     di_et_EE NULL
@@ -28,3 +28,4 @@
 di      dictionaries\sw_TZ      nmake   -       all     di_sw_TZ NULL
 di      dictionaries\vi         nmake   -       all     di_vi NULL
 di      dictionaries\zu_ZA      nmake   -       all     di_zu_ZA NULL
+
