diff -Naur old/gutils/gimagereadgif.c new/gutils/gimagereadgif.c
--- old/gutils/gimagereadgif.c	2014-01-01 05:56:11.000000000 -1000
+++ new/gutils/gimagereadgif.c	2014-05-27 07:04:34.930338020 -1000
@@ -170,13 +170,13 @@
 
     if ( DGifSlurp(gif)!=GIF_OK ) {
 	fprintf(stderr,"Bad input file \"%s\"\n",filename );
-	DGifCloseFile(gif);
+	DGifCloseFile(gif, NULL);
 	return( NULL );
     }
 
     /* Process each image so that it/they can be imported into FF. */
     if ( (images=(GImage **) malloc(gif->ImageCount*sizeof(GImage *)))==NULL ) {
-	DGifCloseFile(gif);
+	DGifCloseFile(gif, NULL);
 	NoMoreMemMessage();
 	return( NULL );
     }
@@ -185,7 +185,7 @@
 	if ( (images[i]=ProcessSavedImage(gif,&gif->SavedImages[i],il))==NULL ) {
 	    while ( --i>=0 ) free(images[i]);
 	    free(images);
-	    DGifCloseFile(gif);
+	    DGifCloseFile(gif, NULL);
 	    return( NULL );
 	}
     }
@@ -195,7 +195,7 @@
 	ret = images[0];
     else
 	ret = GImageCreateAnimation(images,gif->ImageCount);
-    DGifCloseFile(gif);
+    DGifCloseFile(gif, NULL);
     free(images);
     return( ret );
 }
