From 437e08e9303dfe186d14b1cd3b2e9226c61feb4e Mon Sep 17 00:00:00 2001
From: Chris Lamb <lamby@debian.org>
Date: Sun, 11 Oct 2009 15:44:44 +0000
Subject: Execute "javascript:" URIs instead of browsing for them.

Signed-off-by: Xan Lopez <xan@gnome.org>
---
diff --git a/embed/ephy-embed-utils.c b/embed/ephy-embed-utils.c
index 0405de7..d937d2e 100644
--- a/embed/ephy-embed-utils.c
+++ b/embed/ephy-embed-utils.c
@@ -91,6 +91,7 @@ ephy_embed_utils_address_has_web_scheme (const char *address)
 			  g_str_has_prefix (address, "file:") ||
 			  g_str_has_prefix (address, "data:") ||
 			  g_str_has_prefix (address, "about:") ||
+			  g_str_has_prefix (address, "javascript:") ||
 			  g_str_has_prefix (address, "gopher:"));
 
 	return has_web_scheme;
diff --git a/embed/ephy-web-view.c b/embed/ephy-web-view.c
index acf3e5f..0d3be0d 100644
--- a/embed/ephy-web-view.c
+++ b/embed/ephy-web-view.c
@@ -1118,7 +1118,10 @@ ephy_web_view_load_url (EphyWebView *view,
   } else
     effective_url = ephy_embed_utils_normalize_address (url);
 
-  webkit_web_view_open (WEBKIT_WEB_VIEW (view), effective_url);
+  if (g_str_has_prefix (effective_url, "javascript:"))
+    webkit_web_view_execute_script (WEBKIT_WEB_VIEW (view), effective_url);
+  else
+    webkit_web_view_open (WEBKIT_WEB_VIEW (view), effective_url);
 
   if (soup_uri)
     soup_uri_free (soup_uri);
--
cgit v0.8.2
