diff -urNad trunk~/modules/ldapbackend/ldapbackend.cc trunk/modules/ldapbackend/ldapbackend.cc
--- trunk~/modules/ldapbackend/ldapbackend.cc	2006-03-16 21:18:02.000000000 +0100
+++ trunk/modules/ldapbackend/ldapbackend.cc	2006-10-01 15:28:47.017756302 +0200
@@ -281,7 +281,7 @@
 
 	filter = strbind( ":target:", filter, getArg( "filter-lookup" ) );
 
-	stringtok( parts, qesc, "." );
+	stringtok( parts, toLower( qname ), "." );
 	for( i = parts.rbegin(); i != parts.rend(); i++ )
 	{
 		dn = "dc=" + *i + "," + dn;
