Index: mozilla/editor/libeditor/text/nsTextEditRules.cpp
===================================================================
RCS file: /cvsroot/mozilla/editor/libeditor/text/nsTextEditRules.cpp,v
retrieving revision 1.195
diff -d -u -p -r1.195 nsTextEditRules.cpp
--- mozilla/editor/libeditor/text/nsTextEditRules.cpp	16 Jun 2005 13:10:56 -0000	1.195
+++ mozilla/editor/libeditor/text/nsTextEditRules.cpp	4 Oct 2006 16:02:19 -0000
@@ -1390,13 +1390,13 @@ nsTextEditRules::EchoInsertionToPWBuff(P
   // manage the password buffer
   mPasswordText.Insert(*aOutString, aStart);
 
-  // change the output to '*' only
+  // change the output to 'U+2022' only
   PRInt32 length = aOutString->Length();
   PRInt32 i;
   aOutString->Truncate();
   for (i=0; i<length; i++)
   {
-    aOutString->Append(PRUnichar('*'));
+    aOutString->Append(PRUnichar(0x2022));
   }
 
   return NS_OK;
