diff -Nur vdpau-video-0.7.2.orig/configure.ac vdpau-video-0.7.2/configure.ac
--- vdpau-video-0.7.2.orig/configure.ac	2010-10-08 00:09:53.000000000 -0700
+++ vdpau-video-0.7.2/configure.ac	2010-10-08 11:34:00.536686684 -0700
@@ -203,7 +203,7 @@
     #  error "VA-API version >= 0.30.0-sds$LIBVA_GLX_SDS_VERSION_0_30 is required"
     # endif
     #elif VA_MAJOR_VERSION == 0 && VA_MINOR_VERSION == 31
-    # if !defined(VA_SDS_VERSION) || (VA_SDS_VERSION < $LIBVA_GLX_SDS_VERSION_0_31)
+    # if !VA_CHECK_VERSION(0,31,0)
     #  error "VA-API version >= 0.31.0-sds$LIBVA_GLX_SDS_VERSION_0_31 is required"
     # endif
     #else
