diff --git a/src/lib/corelib/tools/tools.pri b/src/lib/corelib/tools/tools.pri
index 9db30664f..bb3a55f12 100644
--- a/src/lib/corelib/tools/tools.pri
+++ b/src/lib/corelib/tools/tools.pri
@@ -131,6 +131,7 @@ osx {
         $$PWD/qbs_export.h \
         $$PWD/settings.h \
         $$PWD/settingsmodel.h \
+        $$PWD/settingsrepresentation.h \
         $$PWD/setupprojectparameters.h \
         $$PWD/toolchains.h \
         $$PWD/version.h
