--- beagle-0.2.1/beagled/IndexHelper/IndexHelper.cs.spew	2006-02-24 01:19:59.000000000 -0500
+++ beagle-0.2.1/beagled/IndexHelper/IndexHelper.cs	2006-02-24 01:43:33.000000000 -0500
@@ -59,16 +59,13 @@
 			bool run_by_hand = (Environment.GetEnvironmentVariable ("BEAGLE_RUN_HELPER_BY_HAND") != null);
 			bool log_in_fg = (Environment.GetEnvironmentVariable ("BEAGLE_LOG_IN_THE_FOREGROUND_PLEASE") != null);
 			
-			// FIXME: We always turn on full debugging output!  We are still
-			// debugging this code, after all...
-			//bool debug = (Environment.GetEnvironmentVariable ("BEAGLE_DEBUG_FLAG_IS_SET") != null);
+			bool debug = (Environment.GetEnvironmentVariable ("BEAGLE_DEBUG_FLAG_IS_SET") != null);
 
 			last_activity = DateTime.Now;
 
 			Log.Initialize (PathFinder.LogDir,
 					"IndexHelper",
-					//debug ? LogLevel.Debug : LogLevel.Warn,
-					LogLevel.Debug,
+					debug ? LogLevel.Debug : LogLevel.Error,
 					run_by_hand || log_in_fg);
 
 			Application.InitCheck ("IndexHelper", ref args);
--- beagle-0.3.2/beagled/BeagleDaemon.cs	2008-01-03 19:06:36.000000000 +0100
+++ beagle-0.3.2/beagled/BeagleDaemon.cs_new	2008-01-03 19:13:44.000000000 +0100
@@ -455,7 +455,7 @@
 			// debugging this code, after all...
 			// arg_debug ? LogLevel.Debug : LogLevel.Warn
 			
-			Log.Initialize (PathFinder.LogDir, "Beagle", LogLevel.Debug, arg_fg);
+			Log.Initialize (PathFinder.LogDir, "Beagle", arg_debug ? LogLevel.Debug : LogLevel.Error, arg_fg);
 			Log.Always ("Starting Beagle Daemon (version {0})", ExternalStringsHack.Version);
 			Log.Always ("Running on {0}", SystemInformation.MonoRuntimeVersion);
 			Log.Always ("Command Line: {0}",
