--- progs/xdemos/Makefile.dont-libglut-me-harder-ok-thx-bye	2006-07-25 05:27:36.000000000 -0400
+++ progs/xdemos/Makefile	2006-07-25 05:33:57.000000000 -0400
@@ -49,8 +49,13 @@
 	-rm -f $(PROGS)
 	-rm -f *.o *~
 
-
 # special cases
+glxgears:
+	$(CC) $(CFLAGS) glxgears.c -lGL -o $@
+
+glxinfo:
+	$(CC) $(CFLAGS) -I$(INCDIR) glxinfo.c -L$(TOP)/lib -L$(TOP)/lib64 -lGL -lGLU -o $@
+
 pbinfo: pbinfo.o pbutil.o
 	$(CC) $(CFLAGS) pbinfo.o pbutil.o $(APP_LIB_DEPS) -o $@
 
