diff -u -r deluge-06ee112/deluge/httpdownloader.py deluge-06ee112.twisted/deluge/httpdownloader.py
--- deluge-06ee112/deluge/httpdownloader.py	2013-08-07 03:52:59.000000000 +0200
+++ deluge-06ee112.twisted/deluge/httpdownloader.py	2013-08-25 09:44:55.895379041 +0200
@@ -196,12 +196,12 @@
     if hasattr(client, '_parse'):
         scheme, host, port, path = client._parse(url)
     else:
-        from twisted.web import _URI
+        from twisted.web.client import _URI
         uri = _URI.fromBytes(url)
         scheme = uri.scheme
         host = uri.host
         port = uri.port
-        path = uri.originFrom
+        path = uri.path
         
     factory = HTTPDownloader(url, filename, callback, headers, force_filename, allow_compression)
     if scheme == "https":
