diff --git a/core/libs/video/qtav/codec/video/VideoDecoderFFmpeg.cpp b/core/libs/video/qtav/codec/video/VideoDecoderFFmpeg.cpp
index a57e5b73dd..eba04111c7 100644
--- a/core/libs/video/qtav/codec/video/VideoDecoderFFmpeg.cpp
+++ b/core/libs/video/qtav/codec/video/VideoDecoderFFmpeg.cpp
@@ -99,10 +99,13 @@ public:
 
     enum MotionVectorVisFlag
     {
-        No = 0, // default
+        No = 0 // default
+#   if LIBAVCODEC_VERSION_MAJOR < 60
+        ,
         PF = FF_DEBUG_VIS_MV_P_FOR,
         BF = FF_DEBUG_VIS_MV_B_FOR,
         BB = FF_DEBUG_VIS_MV_B_BACK
+#   endif
     };
     Q_DECLARE_FLAGS(MotionVectorVisFlags, MotionVectorVisFlag)
     Q_ENUM(MotionVectorVisFlag)
diff --git a/core/libs/video/qtav/codec/video/VideoDecoderFFmpegHW.cpp b/core/libs/video/qtav/codec/video/VideoDecoderFFmpegHW.cpp
index d0a4313762..dedbb43a7f 100644
--- a/core/libs/video/qtav/codec/video/VideoDecoderFFmpegHW.cpp
+++ b/core/libs/video/qtav/codec/video/VideoDecoderFFmpegHW.cpp
@@ -149,7 +149,9 @@ bool VideoDecoderFFmpegHWPrivate::prepare()
 {
     // From vlc begin
 
+#   if LIBAVCODEC_VERSION_MAJOR < 60
     codec_ctx->thread_safe_callbacks = true;
+#   endif
     codec_ctx->thread_count          = threads;
 
 #ifdef _MSC_VER                  // krazy:exclude=cpp
