diff --git a/CMakeLists.txt b/CMakeLists.txt
index 415512d7..99bfd039 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -9,6 +9,12 @@ set(CMAKE_MODULE_PATH ${ECM_MODULE_PATH})
 include(ECMInstallIcons)
 
 
+include(KDEInstallDirs)
+include(KDECMakeSettings)
+include(KDEFrameworkCompilerSettings NO_POLICY_SCOPE)
+include(ECMAddAppIcon)
+include(ECMQtDeclareLoggingCategory)
+
 include(ECMSetupVersion)
 include(ECMAddTests)
 
@@ -20,13 +26,6 @@ include(CheckFunctionExists)
 include(ECMGeneratePriFile)
 
 
-include(KDEInstallDirs)
-include(KDECMakeSettings)
-include(KDEFrameworkCompilerSettings NO_POLICY_SCOPE)
-include(ECMAddAppIcon)
-include(ECMQtDeclareLoggingCategory)
-
-
 # Do NOT add quote
 set(KDEPIM_DEV_VERSION beta)
 
