diff --git a/src/definitions.h b/src/definitions.h
index c82b64efc..10cf52d29 100644
--- a/src/definitions.h
+++ b/src/definitions.h
@@ -252,10 +252,12 @@ QDebug operator<<(QDebug qd, const ItemInfo &info);
 
 // we provide hash function for qstring and QPersistentModelIndex
 namespace std {
+#if (QT_VERSION < QT_VERSION_CHECK(5, 14, 0))
 template <> struct hash<QString>
 {
     std::size_t operator()(const QString &k) const { return qHash(k); }
 };
+#endif
 template <> struct hash<QPersistentModelIndex>
 {
     std::size_t operator()(const QPersistentModelIndex &k) const { return qHash(k); }
