diff --git a/src/apps/marble-kde/CMakeLists.txt b/src/apps/marble-kde/CMakeLists.txt
index 8548aac73..e764163ee 100644
--- a/src/apps/marble-kde/CMakeLists.txt
+++ b/src/apps/marble-kde/CMakeLists.txt
@@ -2,7 +2,7 @@ if(NOT ECM_FOUND)
     return()
 endif()
 
-macro_optional_find_package(KF5 ${REQUIRED_KF5_MIN_VERSION} QUIET COMPONENTS Crash NewStuff Parts CoreAddons I18n)
+macro_optional_find_package(KF5 ${REQUIRED_KF5_MIN_VERSION} QUIET COMPONENTS Crash Parts CoreAddons I18n)
 
 if(NOT KF5_FOUND)
     return()
@@ -22,7 +22,6 @@ set_target_properties(marble_part PROPERTIES PREFIX "${CMAKE_SHARED_LIBRARY_PREF
 target_link_libraries(marble_part
     marbleui
     marblewidget
-    KF5::NewStuff
     KF5::Parts
     KF5::CoreAddons
     KF5::I18n
@@ -31,11 +30,6 @@ target_link_libraries(marble_part
 install(TARGETS marble_part DESTINATION ${KDE_INSTALL_PLUGINDIR})
 install(FILES marble_part.desktop DESTINATION ${KDE_INSTALL_KSERVICES5DIR})
 install(FILES marble_part.rc marbleui.rc DESTINATION ${KDE_INSTALL_KXMLGUI5DIR}/marble)
-if(KF5NewStuff_VERSION VERSION_GREATER "5.56.0")
-    install(FILES marble.knsrc DESTINATION ${KDE_INSTALL_KNSRCDIR})
-else()
-    install(FILES marble.knsrc DESTINATION ${KDE_INSTALL_CONFDIR})
-endif()
 
 ki18n_wrap_ui(marble_kde_SRCS ${marble_UI})
 kconfig_add_kcfg_files(marble_kde_SRCS settings.kcfgc)
@@ -55,7 +49,6 @@ target_link_libraries (
     marbleui
     marblewidget
     astro
-    KF5::NewStuff
     KF5::Parts
     KF5::Crash
     KF5::CoreAddons
diff --git a/src/apps/marble-kde/marble.knsrc b/src/apps/marble-kde/marble.knsrc
deleted file mode 100644
index b695f4521..000000000
--- a/src/apps/marble-kde/marble.knsrc
+++ /dev/null
@@ -1,5 +0,0 @@
-[KNewStuff3]
-ProvidersUrl=http://edu.kde.org/marble/newstuff/providers-4.5.xml
-Categories=Marble Map
-XdgTargetDir=marble/maps
-Uncompress=archive
