diff --git a/CMakeLists.txt b/CMakeLists.txt
index 49403562..73308392 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -47,7 +47,6 @@ find_package(KF5 ${KF5_MIN_VERSION} MODULE REQUIRED COMPONENTS
   NewStuff
   Sonnet
   TextWidgets
-  KCMUtils
   XmlGui
   Notifications
 )
@@ -99,7 +98,6 @@ set( parley_LINK_LIBS
   KF5::NewStuffWidgets
   KF5::SonnetCore
   KF5::SonnetUi
-  KF5::KCMUtils
   KF5::XmlGui
   KF5::Completion
   KF5::ConfigWidgets
