diff --git a/src/KCMTelepathyAccounts/CMakeLists.txt b/src/KCMTelepathyAccounts/CMakeLists.txt
index 1fe7c09..34bc5e0 100644
--- a/src/KCMTelepathyAccounts/CMakeLists.txt
+++ b/src/KCMTelepathyAccounts/CMakeLists.txt
@@ -79,7 +79,7 @@ target_link_libraries (ktpaccountskcminternal
                        KF5::ItemViews
                        KF5::IconThemes
                        KF5::Service
-                       ${TELEPATHY_QT5_LIBRARIES}
+                       TelepathyQt5::Core
 )
 
 
