diff -up kopete-15.12.2/plugins/history2/history2logger.h.than kopete-15.12.2/plugins/history2/history2logger.h
--- kopete-15.12.2/plugins/history2/history2logger.h.than	2016-02-16 23:43:48.245204675 +0100
+++ kopete-15.12.2/plugins/history2/history2logger.h	2016-02-16 23:46:49.042349833 +0100
@@ -94,7 +94,7 @@ public:
 	 * from Kopete::Contact @param c in the given @param sens
 	 */
 	QList<Kopete::Message> readMessages(int lines,
-	                                    int offset=0, const Kopete::MetaContact *c=false, bool reverseOrder=true);
+	                                    int offset=0, const Kopete::MetaContact *c=0, bool reverseOrder=true);
 
 	/**
 	 * Same as the following, but for one date. I did'nt reuse the above function
