diff -up qyoto-4.9.3/cmake/CMakeDetermineCSharpCompiler.cmake.cmake_2_8_10 qyoto-4.9.3/cmake/CMakeDetermineCSharpCompiler.cmake
--- qyoto-4.9.3/cmake/CMakeDetermineCSharpCompiler.cmake.cmake_2_8_10	2012-08-13 04:00:53.000000000 -0500
+++ qyoto-4.9.3/cmake/CMakeDetermineCSharpCompiler.cmake	2012-11-06 15:31:18.719822193 -0600
@@ -84,6 +84,9 @@ endif (CMAKE_CSharp_COMPILER)
 get_filename_component(script_dir ${CMAKE_CURRENT_LIST_FILE} PATH)
 
 # configure variables set in this file for fast reload later on
-configure_file(${script_dir}/CMakeCSharpCompiler.cmake.in
-  ${CMAKE_BINARY_DIR}${CMAKE_FILES_DIRECTORY}/CMakeCSharpCompiler.cmake IMMEDIATE @ONLY)
+ if(NOT CMAKE_PLATFORM_INFO_DIR) # pre-2.8.10
+   set(CMAKE_PLATFORM_INFO_DIR ${CMAKE_BINARY_DIR}${CMAKE_FILES_DIRECTORY})
+ endif()
+ configure_file(${script_dir}/CMakeCSharpCompiler.cmake.in
+   ${CMAKE_PLATFORM_INFO_DIR}/CMakeCSharpCompiler.cmake IMMEDIATE @ONLY)
 set(CMAKE_CSharp_COMPILER_ENV_VAR "CSC")
