diff -Naur isapnptools-1.26.orig/src/isapnp_main.l isapnptools-1.26/src/isapnp_main.l
--- isapnptools-1.26.orig/src/isapnp_main.l	2005-07-21 11:27:39.000000000 +0200
+++ isapnptools-1.26/src/isapnp_main.l	2005-07-21 11:27:45.000000000 +0200
@@ -363,7 +363,7 @@
 	return 0;
 }
 
-static unsigned char serial_identifier[NUM_CARDS+1][IDENT_LEN];
+unsigned char serial_identifier[NUM_CARDS+1][IDENT_LEN];
 static char *boardid[NUM_CARDS+1];
 static unsigned long serno[NUM_CARDS+1];
 
