diff -Nur -x '*.orig' -x '*~' gksu-2.0.0/configure.ac gksu-2.0.0.new/configure.ac
--- gksu-2.0.0/configure.ac	2008-04-04 20:04:14.000000000 +0200
+++ gksu-2.0.0.new/configure.ac	2008-04-04 20:05:12.000000000 +0200
@@ -66,7 +66,7 @@
 AM_CONDITIONAL(ENABLE_NAUTILUS_EXTENSION, test x$enable_nautilus_extension = xyes)
 
 if test x$enable_nautilus_extension = xyes; then
-   PKG_CHECK_MODULES(NAUTILUS, [libgksu2 >= 2.0.0, libnautilus-extension])
+   PKG_CHECK_MODULES(NAUTILUS, [libgksu2 >= 2.0.0, libnautilus-extension, gnome-vfs-2.0])
 fi
 
 AC_CONFIG_FILES([
