--- xorg-server-1.2.0/randr/randr.c.jx	2007-01-22 22:13:16.000000000 -0500
+++ xorg-server-1.2.0/randr/randr.c	2007-02-23 10:23:13.000000000 -0500
@@ -472,8 +472,8 @@
     rep.type = X_Reply;
     rep.length = 0;
     rep.sequenceNumber = client->sequence;
-    rep.majorVersion = RANDR_MAJOR;
-    rep.minorVersion = RANDR_MINOR;
+    rep.majorVersion = 1;
+    rep.minorVersion = 1;
     if (client->swapped) {
     	swaps(&rep.sequenceNumber, n);
     	swapl(&rep.length, n);
