Index: /trunk/plugins/node.d.linux/cpu.in
===================================================================
--- node/node.d.linux/cpu.in (revision 2293)
+++ node/node.d.linux/cpu.in (revision 2352)
@@ -94,5 +93,4 @@
 	echo 'system.label system'
 	echo 'system.draw AREA'
-	echo "system.max $MAX"
 	echo 'system.min 0'
 	echo 'system.type DERIVE'
@@ -101,5 +99,4 @@
 	echo 'user.draw STACK'
 	echo 'user.min 0'
-	echo "user.max $MAX"
 	echo 'user.type DERIVE'
 	echo 'user.info CPU time spent by normal programs and daemons'
@@ -107,5 +104,4 @@
 	echo 'nice.draw STACK'
 	echo 'nice.min 0'
-	echo "nice.max $MAX"
 	echo 'nice.type DERIVE'
 	echo 'nice.info CPU time spent by nice(1)d programs'
@@ -113,5 +109,4 @@
 	echo 'idle.draw STACK'
 	echo 'idle.min 0'
-	echo "idle.max $MAX"
 	echo 'idle.type DERIVE'
 	echo 'idle.info Idle CPU time'
@@ -133,5 +128,4 @@
 		echo 'iowait.draw STACK'
 		echo 'iowait.min 0'
-		echo "iowait.max $MAX"
 		echo 'iowait.type DERIVE'
 		echo 'iowait.info CPU time spent waiting for I/O operations to finish when there is nothing else to do.'
@@ -139,5 +133,4 @@
 		echo 'irq.draw STACK'
 		echo 'irq.min 0'
-		echo "irq.max $MAX"
 		echo 'irq.type DERIVE'
 		echo 'irq.info CPU time spent handling interrupts'
@@ -145,5 +138,4 @@
 		echo 'softirq.draw STACK'
 		echo 'softirq.min 0'
-		echo "softirq.max $MAX"
 		echo 'softirq.type DERIVE'
 		echo 'softirq.info CPU time spent handling "batched" interrupts'
