diff -up OpenCV-2.2.0/modules/highgui/src/cap_v4l.cpp.nov4l1 OpenCV-2.2.0/modules/highgui/src/cap_v4l.cpp
--- OpenCV-2.2.0/modules/highgui/src/cap_v4l.cpp.nov4l1	2011-02-02 16:55:22.844244001 +0100
+++ OpenCV-2.2.0/modules/highgui/src/cap_v4l.cpp	2011-02-02 16:53:22.224244002 +0100
@@ -214,7 +214,9 @@ make & enjoy!
 #include <sys/types.h>
 #include <sys/mman.h>
 
+#ifdef HAVE_CAMV4L
 #include <linux/videodev.h>
+#endif
 
 #include <string.h>
 #include <stdlib.h>
diff -up OpenCV-2.2.0/modules/highgui/src/cap.cpp.nov4l1 OpenCV-2.2.0/modules/highgui/src/cap.cpp
--- OpenCV-2.2.0/modules/highgui/src/cap.cpp.nov4l1	2011-02-04 15:10:51.461243999 +0100
+++ OpenCV-2.2.0/modules/highgui/src/cap.cpp	2011-02-04 15:11:22.040244001 +0100
@@ -171,7 +171,7 @@ CV_IMPL CvCapture * cvCreateCameraCaptur
             if (capture)
                 return capture;
         #endif
-        #if defined (HAVE_CAMV4L) || defined (HAVE_CAMV4L2)
+        #if defined (HAVE_CAMV4L)
             capture = cvCreateCameraCapture_V4L (index);
             if (capture)
                 return capture;
