--- farstream-0.1.2/gst/fsrtpconference/fs-rtp-session.c.orig	2012-03-22 15:23:10.000000000 -0500
+++ farstream-0.1.2/gst/fsrtpconference/fs-rtp-session.c	2014-08-08 11:59:55.206979402 -0500
@@ -3624,6 +3624,9 @@
 
   if (!codecbin)
   {
+    g_set_error (error, FS_ERROR, FS_ERROR_CONSTRUCTION,
+        "Could create codec bin for: " FS_CODEC_FORMAT,
+        FS_CODEC_ARGS (send_codec_copy));
     fs_codec_destroy (send_codec_copy);
     fs_codec_destroy (codec_copy);
     fs_codec_list_destroy (codecs);
@@ -3690,6 +3693,9 @@
 
   if (!g_value_get_boolean (&link_rv))
   {
+    g_set_error (error, FS_ERROR, FS_ERROR_CONSTRUCTION,
+        "Could not get boolean value for link"
+        " the main pad for: " FS_CODEC_FORMAT, FS_CODEC_ARGS (send_codec_copy));
     gst_iterator_free (iter);
     goto error;
   }
@@ -3709,8 +3715,12 @@
 
   gst_iterator_free (iter);
 
-  if (!g_value_get_boolean (&link_rv))
+  if (!g_value_get_boolean (&link_rv)) {
+    g_set_error (error, FS_ERROR, FS_ERROR_CONSTRUCTION,
+        "Could not get boolean value for link"
+        " the main pad for: " FS_CODEC_FORMAT, FS_CODEC_ARGS (send_codec_copy));
     goto error;
+  }
 
   gst_element_set_locked_state (codecbin, FALSE);
 
