From e349a7db681b643beaf8174a32b071292abd980e Mon Sep 17 00:00:00 2001
From: Olivier Crête <olivier.crete@collabora.com>
Date: Thu, 27 Feb 2014 23:00:13 +0000
Subject: raw: Fix crash where the stream would try to contact its session before its been set

---
(limited to 'gst/fsrawconference')

diff --git a/gst/fsrawconference/fs-raw-stream.c b/gst/fsrawconference/fs-raw-stream.c
index fe5326b..eb8db74 100644
--- a/gst/fsrawconference/fs-raw-stream.c
+++ b/gst/fsrawconference/fs-raw-stream.c
@@ -436,7 +436,8 @@ fs_raw_stream_set_property (GObject *object,
               (direction & FS_DIRECTION_SEND) ? TRUE : FALSE, NULL);
           g_object_unref (st);
         }
-        fs_raw_session_update_direction (self->priv->session, direction);
+        if (self->priv->session)
+          fs_raw_session_update_direction (self->priv->session, direction);
         if (conference)
           GST_OBJECT_LOCK (conference);
       }
--
cgit v0.9.0.3
