--- kdeadmin-4.5.80/ksystemlog/src/modes/system/systemConfiguration.h~	2010-11-24 08:52:42.826666681 +0100
+++ kdeadmin-4.5.80/ksystemlog/src/modes/system/systemConfiguration.h	2010-11-24 08:52:51.140000014 +0100
@@ -39,7 +39,7 @@
 		SystemConfiguration() :
 			GenericLogModeConfiguration(
 					QLatin1String( SYSTEM_LOG_MODE_ID ),
-					QStringList() << QLatin1String( "/var/log/syslog" ),
+					QStringList() << QLatin1String( "/var/log/messages.log" ),
 					QList<int>() << INFORMATION_LOG_LEVEL_ID
 			) {
 
