diff -Naur vigra1.6.0-orig/src/impex/png.cxx vigra1.6.0/src/impex/png.cxx
--- vigra1.6.0-orig/src/impex/png.cxx	2010-01-20 19:46:57.000000000 -0500
+++ vigra1.6.0/src/impex/png.cxx	2010-01-20 19:47:55.000000000 -0500
@@ -270,8 +270,8 @@
         // expand gray values to at least one byte size
         if ( color_type == PNG_COLOR_TYPE_GRAY && bit_depth < 8 ) {
             if (setjmp(png->jmpbuf))
-                vigra_postcondition( false,png_error_message.insert(0, "error in png_set_gray_1_2_4_to_8(): ").c_str());
-            png_set_gray_1_2_4_to_8(png);
+                vigra_postcondition( false,png_error_message.insert(0, "error in png_set_expand_gray_1_2_4_to_8(): ").c_str());
+            png_set_expand_gray_1_2_4_to_8(png);
             bit_depth = 8;
         }
 
