diff --git i/data/meson.build w/data/meson.build
index 781dc9e..c18c617 100644
--- i/data/meson.build
+++ w/data/meson.build
@@ -1,4 +1,4 @@
-foreach size: [ '16x16', '22x22', '24x24', '32x32', '48x48' ]
+foreach size: [ '16x16', '22x22', '24x24', '32x32', '48x48', '256x256' ]
     install_data (join_paths ('icons', 'hicolor', size, 'gnome-mines.png'),
                   install_dir: join_paths (datadir, 'icons', 'hicolor', size, 'apps'))
 endforeach
