diff --git i/data/meson.build w/data/meson.build
index 1b3585c00bd34553..f862c7a877419409 100644
--- i/data/meson.build
+++ w/data/meson.build
@@ -1,9 +1,9 @@
-foreach size: [ '16x16', '22x22', '24x24', '32x32', '48x48' ]
+foreach size: [ '16x16', '22x22', '24x24', '32x32', '48x48', '256x256' ]
     install_data (join_paths ('icons', 'hicolor', size, 'gnome-mines.png'),
                   install_dir: join_paths (datadir, 'icons', 'hicolor', size, 'apps'))
 endforeach
 install_data (join_paths ('icons', 'hicolor', 'scalable', 'gnome-mines-symbolic.svg'),
-              install_dir: join_paths (datadir, 'icons', 'hicolor', size, 'apps'))
+              install_dir: join_paths (datadir, 'icons', 'hicolor', 'scalable', 'apps'))
 install_data ('flag-symbolic.svg',
               install_dir: join_paths (datadir, 'icons', 'hicolor', '16x16', 'actions'))
 
