diff -ruN -x .gitignore xf86-input-citron-9a4d7854273a34c1a32f46fb52f2f8c3a227cd12/Makefile.am xf86-input-citron-32dbc726871bf7e3a7535d788f169f8e11c981fd/Makefile.am
--- xf86-input-citron-9a4d7854273a34c1a32f46fb52f2f8c3a227cd12/Makefile.am	2007-09-09 15:09:45.000000000 +0000
+++ xf86-input-citron-32dbc726871bf7e3a7535d788f169f8e11c981fd/Makefile.am	2009-02-02 20:32:11.000000000 +0000
@@ -20,11 +20,13 @@
 
 AUTOMAKE_OPTIONS = foreign
 SUBDIRS = src man
-
-CLEANFILES = ChangeLog
 EXTRA_DIST = ChangeLog
+EXTRA_DIST = ChangeLog
+MAINTAINERCLEANFILES = ChangeLog
 
 .PHONY: ChangeLog
 
 ChangeLog:
-	(GIT_DIR=$(top_srcdir)/.git git-log > .changelog.tmp && mv .changelog.tmp ChangeLog; rm -f .changelog.tmp) || (touch ChangeLog; echo 'git directory not found: installing possibly empty changelog.' >&2)
+	$(CHANGELOG_CMD)
+
+dist-hook: ChangeLog
diff -ruN -x .gitignore xf86-input-citron-9a4d7854273a34c1a32f46fb52f2f8c3a227cd12/README xf86-input-citron-32dbc726871bf7e3a7535d788f169f8e11c981fd/README
--- xf86-input-citron-9a4d7854273a34c1a32f46fb52f2f8c3a227cd12/README	1970-01-01 00:00:00.000000000 +0000
+++ xf86-input-citron-32dbc726871bf7e3a7535d788f169f8e11c981fd/README	2009-02-02 20:32:11.000000000 +0000
@@ -0,0 +1,19 @@
+
+Please submit bugs & patches to the Xorg bugzilla:
+
+        https://bugs.freedesktop.org/enter_bug.cgi?product=xorg
+
+All questions regarding this software should be directed at the
+Xorg mailing list:
+
+        http://lists.freedesktop.org/mailman/listinfo/xorg
+
+The master development code repository can be found at:
+
+        git://anongit.freedesktop.org/git/xorg/driver/xf86-input-citron
+
+        http://cgit.freedesktop.org/xorg/driver/xf86-input-citron
+
+For more information on the git code manager, see:
+
+        http://wiki.x.org/wiki/GitPage
diff -ruN -x .gitignore xf86-input-citron-9a4d7854273a34c1a32f46fb52f2f8c3a227cd12/configure.ac xf86-input-citron-32dbc726871bf7e3a7535d788f169f8e11c981fd/configure.ac
--- xf86-input-citron-9a4d7854273a34c1a32f46fb52f2f8c3a227cd12/configure.ac	2007-09-09 15:09:45.000000000 +0000
+++ xf86-input-citron-32dbc726871bf7e3a7535d788f169f8e11c981fd/configure.ac	2009-02-02 20:32:11.000000000 +0000
@@ -32,15 +32,16 @@
 
 AM_MAINTAINER_MODE
 
-DRIVER_NAME=citron
-AC_SUBST([DRIVER_NAME])
-
+# Require xorg-macros: XORG_CWARNFLAGS, XORG_CHANGELOG
+m4_ifndef([XORG_MACROS_VERSION], [AC_FATAL([must install xorg-macros 1.2 or later before running autoconf/autogen])])
+XORG_MACROS_VERSION(1.2)
 AM_CONFIG_HEADER([config.h])
 
 # Checks for programs.
 AC_DISABLE_STATIC
 AC_PROG_LIBTOOL
 AC_PROG_CC
+XORG_CWARNFLAGS
 
 AH_TOP([#include "xorg-server.h"])
 
@@ -77,16 +78,18 @@
 # Checks for pkg-config packages
 PKG_CHECK_MODULES(XORG, [xorg-server >= 1.0.99.901] xproto $REQUIRED_MODULES)
 sdkdir=$(pkg-config --variable=sdkdir xorg-server)
-
-CFLAGS="$CFLAGS $XORG_CFLAGS "' -I$(top_srcdir)/src'
-AC_SUBST([CFLAGS])
+XORG_CFLAGS="$CWARNFLAGS $XORG_CFLAGS"
 
 # Checks for libraries.
 
 # Checks for header files.
 AC_HEADER_STDC
 
+DRIVER_NAME=citron
+AC_SUBST([DRIVER_NAME])
+
 XORG_MANPAGE_SECTIONS
 XORG_RELEASE_VERSION
+XORG_CHANGELOG
 
 AC_OUTPUT([Makefile src/Makefile man/Makefile])
diff -ruN -x .gitignore xf86-input-citron-9a4d7854273a34c1a32f46fb52f2f8c3a227cd12/man/citron.man xf86-input-citron-32dbc726871bf7e3a7535d788f169f8e11c981fd/man/citron.man
--- xf86-input-citron-9a4d7854273a34c1a32f46fb52f2f8c3a227cd12/man/citron.man	2007-09-09 15:09:45.000000000 +0000
+++ xf86-input-citron-32dbc726871bf7e3a7535d788f169f8e11c981fd/man/citron.man	2009-02-02 20:32:11.000000000 +0000
@@ -903,6 +903,6 @@
 "60" 
 
 .SH "SEE ALSO"
-__xservername__(__appmansuffix__), __xconfigfile__(__filemansuffix__), xorgconfig(1), Xserver(1), X(__miscmansuffix__).
+__xservername__(__appmansuffix__), __xconfigfile__(__filemansuffix__), Xserver(1), X(__miscmansuffix__).
 .SH "AUTHORS"
 2000\-2005 \- written  by  Citron GmbH (support@citron.de)
diff -ruN -x .gitignore xf86-input-citron-9a4d7854273a34c1a32f46fb52f2f8c3a227cd12/src/Makefile.am xf86-input-citron-32dbc726871bf7e3a7535d788f169f8e11c981fd/src/Makefile.am
--- xf86-input-citron-9a4d7854273a34c1a32f46fb52f2f8c3a227cd12/src/Makefile.am	2007-09-09 15:09:45.000000000 +0000
+++ xf86-input-citron-32dbc726871bf7e3a7535d788f169f8e11c981fd/src/Makefile.am	2009-02-02 20:32:11.000000000 +0000
@@ -24,6 +24,8 @@
 # -avoid-version prevents gratuitous .0.0.0 version numbers on the end
 # _ladir passes a dummy rpath to libtool so the thing will actually link
 # TODO: -nostdlib/-Bstatic/-lgcc platform magic, not installing the .a, etc.
+AM_CFLAGS = $(XORG_CFLAGS)
+
 @DRIVER_NAME@_drv_la_LTLIBRARIES = @DRIVER_NAME@_drv.la
 @DRIVER_NAME@_drv_la_LDFLAGS = -module -avoid-version
 @DRIVER_NAME@_drv_ladir = @inputdir@
diff -ruN -x .gitignore xf86-input-citron-9a4d7854273a34c1a32f46fb52f2f8c3a227cd12/src/citron.c xf86-input-citron-32dbc726871bf7e3a7535d788f169f8e11c981fd/src/citron.c
--- xf86-input-citron-9a4d7854273a34c1a32f46fb52f2f8c3a227cd12/src/citron.c	2007-09-09 15:09:45.000000000 +0000
+++ xf86-input-citron-32dbc726871bf7e3a7535d788f169f8e11c981fd/src/citron.c	2009-02-02 20:32:11.000000000 +0000
@@ -305,7 +305,7 @@
 	MODULEVENDORSTRING,			/* vendor specific string */
 	MODINFOSTRING1,				
 	MODINFOSTRING2,
-	XF86_VERSION_CURRENT,		/* Current XFree version */
+	XORG_VERSION_CURRENT,		/* Current Xorg version */
 	CITOUCH_VERSION_MAJOR,		/* Module-specific major version */
 	CITOUCH_VERSION_MINOR,		/* Module-specific minor version */
 	CITOUCH_VERSION_PATCH,		/* Module-specific patch level */
@@ -1368,7 +1368,10 @@
 	 * screen to fit one meter.
 	 * Device may reports touch pressure on the 3rd axis.
 	 */
-	if (InitValuatorClassDeviceStruct (dev, 2, xf86GetMotionEvents,
+	if (InitValuatorClassDeviceStruct (dev, 2,
+#if GET_ABI_MAJOR(ABI_XINPUT_VERSION) < 3
+                    xf86GetMotionEvents,
+#endif
 									local->history_size, Absolute) == FALSE)
 	{
 		ErrorF ("%sUnable to allocate Citron touchscreen ValuatorClassDeviceStruct\n", CI_ERROR);
@@ -2058,12 +2061,14 @@
 		DBG(6, ErrorF("%s\treporting mode = %s\n", CI_INFO, mode==TS_Raw?"raw":"scaled"));
 		return (Success);
 	}
+#if GET_ABI_MAJOR(ABI_XINPUT_VERSION) == 0
 	else if ((mode == SendCoreEvents) || (mode == DontSendCoreEvents))
 	{
 		xf86XInputSetSendCoreEvents (local, (mode == SendCoreEvents));
 		DBG(6, ErrorF("%s\tmode = %sSend Core Events\n", CI_INFO, mode==DontSendCoreEvents?"Don\'t ":""));
 		return (Success);
 	}
+#endif
 #ifdef CIT_MODE_EXT
 	else if (mode == ClickMode_Enter)
 	{
@@ -2466,6 +2471,7 @@
 	if(priv->beep == 0)
 		return;
 
+#if GET_ABI_MAJOR(ABI_XINPUT_VERSION) == 0
 	/* ring release bell */
 	if(press == 0)
 
@@ -2492,6 +2498,7 @@
 	else
 	/* ring press bell */
 		xf86SoundKbdBell(priv->press_vol,priv->press_pitch, priv->press_dur);
+#endif
 
 	DBG(7, ErrorF("%scit_Beep called - %s\n", CI_INFO, (press == 0) ? "release" : "press"));
 #endif
@@ -2508,6 +2515,7 @@
 	if(priv->beepkey == 0)
 		return;
 
+#if GET_ABI_MAJOR(ABI_XINPUT_VERSION) == 0
 	/* ring release bell */
 	if(press <= 0)
 
@@ -2532,6 +2540,7 @@
 	else
 	/* ring press bell */
 		xf86SoundKbdBell(priv->presskey_vol,priv->presskey_pitch, priv->presskey_dur);
+#endif
 
 	DBG(7, ErrorF("%scit_BeepKey called - %s\n", CI_INFO, (press == 0) ? "release" : "press"));
 #endif
