diff -u -Nura soprano-2.3.70.20100119/cmake/modules/FindRedland.cmake soprano-2.3.70.20100119.new/cmake/modules/FindRedland.cmake
--- soprano-2.3.70.20100119/cmake/modules/FindRedland.cmake	2010-01-19 09:53:38.000000000 +0100
+++ soprano-2.3.70.20100119.new/cmake/modules/FindRedland.cmake	2010-01-20 07:10:36.735873069 +0100
@@ -106,7 +106,7 @@
       ARGS ${_REDLAND_STORAGE_LIBS}
     )
     if(NOT "${_TEST_EXITCODE}" EQUAL 0)
-      set(_REDLAND_VERSION_OK)
+#      set(_REDLAND_VERSION_OK)
       message(STATUS "${_OUTPUT}")
       message(STATUS "Redland with broken NEEDED section detected, disabling")
     endif(NOT "${_TEST_EXITCODE}" EQUAL 0)
