From 47e1a496c9a7124c28d420afe202e5bbb01ce02e Mon Sep 17 00:00:00 2001
From: Thomas Dziedzic <gostrc@gmail.com>
Date: Sat, 13 Oct 2012 20:37:09 -0700
Subject: [PATCH] disable setting jobs due to cabal 1.16.0 bug shipped with
 ghc 7.6.1

---
 Distribution/Client/Config.hs | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/Distribution/Client/Config.hs b/Distribution/Client/Config.hs
index 93e5a52..ecbc6f7 100644
--- a/Distribution/Client/Config.hs
+++ b/Distribution/Client/Config.hs
@@ -202,8 +202,7 @@ initialSavedConfig = do
     },
     savedInstallFlags    = mempty {
       installSummaryFile = [toPathTemplate (logsDir </> "build.log")],
-      installBuildReports= toFlag AnonymousReports,
-      installNumJobs     = toFlag (Just numberOfProcessors)
+      installBuildReports= toFlag AnonymousReports
     }
   }
 
-- 
1.7.12.3

