--- libpst-0.6.41/configure.in.old	2009-08-23 15:08:48.000000000 +0200
+++ libpst-0.6.41/configure.in	2009-08-23 15:09:33.000000000 +0200
@@ -134,7 +134,7 @@
 AC_HEADER_DIRENT
 AC_HEADER_STDC
 AC_CHECK_HEADERS([ctype.h dirent.h errno.h fcntl.h inttypes.h limits.h regex.h semaphore.h signal.h stdarg.h stdint.h stdio.h stdlib.h string.h sys/param.h sys/shm.h sys/stat.h sys/types.h time.h unistd.h wchar.h])
-AC_SEARCH_LIBS([sem_init],rt)
+AC_SEARCH_LIBS([sem_init],pthread)
 
 
 # Checks for typedefs, structures, and compiler characteristics.
