--- tc/Makefile.old	2010-06-14 08:29:49.257421437 +0200
+++ tc/Makefile	2010-06-14 08:30:07.157419977 +0200
@@ -99,18 +99,11 @@
 	$(AR) rcs $@ $(TCLIB)
 
 install: all
-	echo mkdir -p $(MODDESTDIR)
-	echo install -m 0755 tc $(DESTDIR)$(SBINDIR)
-	for i in $(TCSO); \
-	do echo install -m 755 $$i $(MODDESTDIR); \
-	done
-	if [ ! -f $(MODDESTDIR)/m_ipt.so ]; then \
-	if [ -f $(MODDESTDIR)/m_xt.so ]; \
-		then ln -s m_xt.so $(MODDESTDIR)/m_ipt.so ; \
-	elif [ -f $(MODDESTDIR)/m_xt_old.so ]; \
-		then ln -s m_xt_old.so $(MODDESTDIR)/m_ipt.so ; \
-	fi; \
-	fi
+	mkdir -p $(DESTDIR)$(LIBDIR)/tc
+	install -m 0755 tc $(DESTDIR)$(SBINDIR)
+	for i in $(TCSO); \
+	do install -m 755 $$i $(DESTDIR)$(LIBDIR)/tc; \
+	done
 
 clean:
 	rm -f $(TCOBJ) $(TCLIB) libtc.a tc *.so emp_ematch.yacc.h; \
