diff -Naur file-5.05.ori/magic/Magdir/python file-5.05/magic/Magdir/python
--- file-5.05.ori/magic/Magdir/python	2010-12-31 10:15:28.000000000 -0800
+++ file-5.05/magic/Magdir/python	2011-02-24 07:07:36.030004467 -0800
@@ -20,6 +20,7 @@
 0	belong		0x03f30d0a	python 2.7 byte-compiled
 0	belong		0x3b0c0d0a	python 3.0 byte-compiled
 0	belong		0x4f0c0d0a	python 3.1 byte-compiled
+0	belong		0x6c0c0d0a	python 3.2 byte-compiled
 
 0	search/1/w	#!\ /usr/bin/python	Python script text executable
 !:mime text/x-python
