From 6e3e20a80ba38832413696452d297c4765f14ffd Mon Sep 17 00:00:00 2001
From: Christoph Reiter <reiter.christoph@gmail.com>
Date: Sun, 27 May 2018 10:51:54 +0200
Subject: [PATCH] Disable a new pycodestyle warning

---
 setup.cfg | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.cfg b/setup.cfg
index 68d10b7..96f7216 100644
--- a/setup.cfg
+++ b/setup.cfg
@@ -6,6 +6,6 @@ omit=
     mutagen/_senf/*
 
 [flake8]
-ignore=E128,W601,E402,E731,W503,E741,E305,E121,E124
+ignore=E128,W601,E402,E731,W503,E741,E305,E121,E124,W504
 builtins=cmp,unicode,long,xrange,basestring
 exclude=
