diff --git a/configure.in b/configure.in
index e952cc7..e5f86cc 100644
--- a/configure.in
+++ b/configure.in
@@ -868,22 +868,6 @@ if test "$GCC" = yes; then
     for oflag in -fno-fast-math; do
 	RUBY_TRY_CFLAGS($oflag, [RUBY_APPEND_OPTION(optflags, $oflag)])
     done
-    AS_CASE(["$target"],
-	[*-darwin*], [
-	    # doesn't seem necessary on Mac OS X
-	],
-	[[i[4-6]86*]], [
-	    RUBY_TRY_CFLAGS(-msse2 -mfpmath=sse, [
-		RUBY_APPEND_OPTION(XCFLAGS, -msse2 -mfpmath=sse)
-	    ])
-            AS_CASE(["$XCFLAGS"],
-                [[*-msse2*]], [
-                    RUBY_TRY_CFLAGS(-mstackrealign, [
-                        RUBY_APPEND_OPTION(XCFLAGS, -mstackrealign)
-                    ])
-                ])
-	]
-    )
 fi
 
 AC_ARG_WITH(opt-dir,
