diff -u -r rtkit-0.11/rtkit-daemon.service.in rtkit-0.11-sd205/rtkit-daemon.service.in
--- rtkit-0.11/rtkit-daemon.service.in	2012-05-15 15:25:40.000000000 +0200
+++ rtkit-0.11-sd205/rtkit-daemon.service.in	2013-07-25 10:27:37.790884664 +0200
@@ -24,12 +24,7 @@
 BusName=org.freedesktop.RealtimeKit1
 NotifyAccess=main
 CapabilityBoundingSet=CAP_SYS_NICE CAP_DAC_READ_SEARCH CAP_SYS_PTRACE CAP_SYS_CHROOT CAP_SETGID CAP_SETUID
-PrivateTmp=yes
 PrivateNetwork=yes
 
-# Work around the fact that the Linux currently doesn't assign any RT
-# budget to CPU control groups that have none configured explicitly
-ControlGroup=cpu:/
-
 [Install]
 WantedBy=graphical.target
