Subject: [PATCH 1/1] insmod, modprobe: Don't hardcode path to modinfo (Alioth: #313569)

--- a/completions/insmod
+++ b/completions/insmod
@@ -9,8 +9,8 @@ _insmod()
 
     if [[ $cword -gt 1 && "${words[cword-1]}" != -* ]]; then
         # do module parameter completion
-        COMPREPLY=( $( compgen -W "$( /sbin/modinfo -p ${words[1]} | \
-            cut -d: -f1 )" -- "$cur" ) )
+        COMPREPLY=( $( compgen -W "$( PATH="$PATH:/sbin" modinfo \
+            -p ${words[1]} 2>/dev/null | cut -d: -f1 )" -- "$cur" ) )
     else
         _modules $(uname -r)
     fi
