diff -Naur iptraf-3.0.0-minus/src/othptab.c iptraf-3.0.0/src/othptab.c
--- iptraf-3.0.0-minus/src/othptab.c	2006-09-27 03:30:33.000000000 +0200
+++ iptraf-3.0.0/src/othptab.c	2006-09-27 03:34:24.000000000 +0200
@@ -17,6 +17,7 @@
 ***/
 
 #include <asm/types.h>
+#include <sys/types.h>
 #include <linux/if_ether.h>
 #include <winops.h>
 #include "arphdr.h"
diff -Naur iptraf-3.0.0-minus/src/tr.c iptraf-3.0.0/src/tr.c
--- iptraf-3.0.0-minus/src/tr.c	2005-09-13 08:42:54.000000000 +0200
+++ iptraf-3.0.0/src/tr.c	2006-09-27 03:32:54.000000000 +0200
@@ -7,8 +7,8 @@
  */
 
 #include <asm/types.h>
-#include <linux/if_tr.h>
 #include <netinet/in.h>
+#include <linux/if_tr.h>
 
 unsigned int get_tr_ip_offset(unsigned char *pkt)
 {

