--- sapi/apache2handler/config.m4.orig	2015-12-08 19:42:18.000000000 +0100
+++ sapi/apache2handler/config.m4	2015-12-15 07:26:17.866096130 +0100
@@ -122,7 +122,6 @@
       PHP_BUILD_THREAD_SAFE
     fi
   else
-    APACHE_THREADED_MPM=`$APXS_HTTPD -V | grep 'threaded:.*yes'`
     if test -n "$APACHE_THREADED_MPM"; then
       PHP_BUILD_THREAD_SAFE
     fi
--- configure.orig	2015-12-08 19:42:18.000000000 +0100
+++ configure	2015-12-15 07:26:18.146090579 +0100
@@ -6875,7 +6875,6 @@
 
     fi
   else
-    APACHE_THREADED_MPM=`$APXS_HTTPD -V | grep 'threaded:.*yes'`
     if test -n "$APACHE_THREADED_MPM"; then
 
   enable_maintainer_zts=yes
