diff -Nur comm-1.9.1.orig//mozilla/gfx/ots/src/os2.cc comm-1.9.1/mozilla/gfx/ots/src/os2.cc
--- comm-1.9.1.orig//mozilla/gfx/ots/src/os2.cc	2011-04-20 22:57:49.000000000 -0700
+++ comm-1.9.1/mozilla/gfx/ots/src/os2.cc	2011-04-29 09:14:58.929150027 -0700
@@ -2,6 +2,8 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+#include <cstddef>
+
 #include "os2.h"
 
 #include "head.h"
