? security/manager/ssl/src/nsNSSCallbacks.cpp.bck
Index: security/manager/ssl/src/nsNSSCallbacks.cpp
===================================================================
RCS file: /cvsroot/mozilla/security/manager/ssl/src/nsNSSCallbacks.cpp,v
retrieving revision 1.37
diff -u -r1.37 nsNSSCallbacks.cpp
--- security/manager/ssl/src/nsNSSCallbacks.cpp	25 Apr 2004 15:02:36 -0000	1.37
+++ security/manager/ssl/src/nsNSSCallbacks.cpp	12 Feb 2005 18:35:40 -0000
@@ -206,7 +206,7 @@
   }
   
   if (NS_SUCCEEDED(rv) && value) {
-    char* str = ToNewCString(nsDependentString(password));
+    char* str = ToNewUTF8String(nsDependentString(password));
     Recycle(password);
     return str;
   }
