diff -Nur comm-1.9.1.orig/config/autoconf.mk.in comm-1.9.1/config/autoconf.mk.in
--- comm-1.9.1.orig/config/autoconf.mk.in	2009-12-06 17:37:33.000000000 +0200
+++ comm-1.9.1/config/autoconf.mk.in	2009-12-16 16:40:05.477454406 +0200
@@ -54,7 +54,7 @@
 MOZ_BUILD_APP = @MOZ_BUILD_APP@
 MOZ_APP_NAME	= @MOZ_APP_NAME@
 MOZ_APP_DISPLAYNAME = @MOZ_APP_DISPLAYNAME@
-MOZ_APP_VERSION = @MOZ_APP_VERSION@
+MOZ_APP_VERSION="2.0"
 MOZ_APP_VERSION_TXT = @MOZ_APP_VERSION_TXT@
 
 MOZ_PKG_SPECIAL = @MOZ_PKG_SPECIAL@
