diff -ruN libtunepimp-0.5.3.orig/lib/fileio.cpp libtunepimp-0.5.3/lib/fileio.cpp
--- libtunepimp-0.5.3.orig/lib/fileio.cpp	2009-05-28 11:49:43.000000000 -0700
+++ libtunepimp-0.5.3/lib/fileio.cpp	2009-05-28 11:59:05.000000000 -0700
@@ -124,7 +124,8 @@
 
 void tmktempname(const char *path, char *newPath, int newPathLen)
 {
-    char *ptr, *temp;
+    const char *ptr;
+    char *temp;
 
     temp = (char *)malloc(strlen(path) + 32);
     ptr = strrchr(path, dirSepChar);
