diff -Nur taglib-1.7.orig/taglib/ape/apeproperties.cpp taglib-1.7/taglib/ape/apeproperties.cpp
--- taglib-1.7.orig/taglib/ape/apeproperties.cpp	2011-03-12 00:15:38.000000000 +1100
+++ taglib-1.7/taglib/ape/apeproperties.cpp	2011-07-29 21:19:45.984345060 +1000
@@ -137,7 +137,7 @@
   long ID3v2OriginalSize = 0;
   bool hasID3v2 = false;
   if(ID3v2Location >= 0) {
-    ID3v2::Tag tag(d->file, ID3v2Location, 0);
+    ID3v2::Tag tag(d->file, ID3v2Location);
     ID3v2OriginalSize = tag.header()->completeTagSize();
     if(tag.header()->tagSize() > 0)
       hasID3v2 = true;
