--- hal-0.5.11/hal.conf.in	2008-05-27 16:09:25.000000000 +0200
+++ hal-0.5.11/hal.conf.in	2008-05-27 16:21:28.000000000 +0200
@@ -37,9 +37,27 @@
 	   receive_sender="org.freedesktop.Hal"/>
   </policy>
 
-  <!-- Default policy for the exported interfaces; if PolicyKit is not used
-       for access control you will need to modify this -->
+  <!-- Default policy for the exported interfaces -->
   <policy context="default">
+    <deny send_interface="org.freedesktop.Hal.Device.SystemPowerManagement"/>
+    <deny send_interface="org.freedesktop.Hal.Device.VideoAdapterPM"/>
+    <deny send_interface="org.freedesktop.Hal.Device.LaptopPanel"/>
+    <deny send_interface="org.freedesktop.Hal.Device.Volume"/>
+    <deny send_interface="org.freedesktop.Hal.Device.Volume.Crypto"/>
+  </policy>
+
+  <policy group="power">
+    <allow send_interface="org.freedesktop.Hal.Device.SystemPowerManagement"/>
+    <allow send_interface="org.freedesktop.Hal.Device.LaptopPanel"/>
+  </policy>
+
+  <policy group="storage">
+    <allow send_interface="org.freedesktop.Hal.Device.Volume"/>
+    <allow send_interface="org.freedesktop.Hal.Device.Volume.Crypto"/>
+  </policy>
+
+  <!-- You can change this to a more suitable user, or make per-group -->
+  <policy user="0">
     <allow send_interface="org.freedesktop.Hal.Device.SystemPowerManagement"/>
     <allow send_interface="org.freedesktop.Hal.Device.VideoAdapterPM"/>
     <allow send_interface="org.freedesktop.Hal.Device.LaptopPanel"/>
