diff --git a/refind/main.c b/refind/main.c
index 9558655..3f54b4f 100644
--- a/refind/main.c
+++ b/refind/main.c
@@ -58,13 +58,13 @@
 
 #define MACOSX_LOADER_PATH      L"System\\Library\\CoreServices\\boot.efi"
 #if defined (EFIX64)
-#define SHELL_NAMES             L"\\EFI\\tools\\shell.efi,\\shellx64.efi"
+#define SHELL_NAMES             L"\\EFI\\tools\\shellx64.efi,\\EFI\\tools\\shell.efi,\\shellx64.efi"
 #define DRIVER_DIRS             L"drivers,drivers_x64"
 #elif defined (EFI32)
-#define SHELL_NAMES             L"\\EFI\\tools\\shell.efi,\\shellia32.efi"
+#define SHELL_NAMES             L"\\EFI\\tools\\shellia32.efi,\\EFI\\tools\\shell.efi,\\shellia32.efi"
 #define DRIVER_DIRS             L"drivers,drivers_ia32"
 #else
-#define SHELL_NAMES             L"\\EFI\\tools\\shell.efi"
+#define SHELL_NAMES             L"\\EFI\\tools\\shell.efi,\\shell.efi"
 #define DRIVER_DIRS             L"drivers"
 #endif
 
