diff --git a/refind/main.c b/refind/main.c
index 298a6d2..3e92f2e 100644
--- a/refind/main.c
+++ b/refind/main.c
@@ -57,13 +57,13 @@
 
 #define MACOSX_LOADER_PATH      L"System\\Library\\CoreServices\\boot.efi"
 #if defined (EFIX64)
-#define SHELL_NAMES             L"\\EFI\\tools\\shell.efi,\\shellx64.efi"
+#define SHELL_NAMES             L"\\EFI\\tools\\shellx64.efi,\\EFI\\tools\\shell.efi,\\shellx64.efi"
 #define DRIVER_DIRS             L"drivers,drivers_x64"
 #elif defined (EFI32)
-#define SHELL_NAMES             L"\\EFI\\tools\\shell.efi,\\shellia32.efi"
+#define SHELL_NAMES             L"\\EFI\\tools\\shellia32.efi,\\EFI\\tools\\shell.efi,\\shellia32.efi"
 #define DRIVER_DIRS             L"drivers,drivers_ia32"
 #else
-#define SHELL_NAMES             L"\\EFI\\tools\\shell.efi"
+#define SHELL_NAMES             L"\\EFI\\tools\\shell.efi,\\shell.efi"
 #endif
 
 // Filename patterns that identify EFI boot loaders. Note that a single case (either L"*.efi" or
