Index: eqtexsvg.py
===================================================================
--- share/extensions/eqtexsvg.py	(revision 16441)
+++ share/extensions/eqtexsvg.py	(revision 18440)
@@ -49,16 +49,18 @@
     doc_sizeW = max(doc_width,doc_height)
 
     def clone_and_rewrite(self, node_in):
-        if node_in.tag != 'svg':
-            node_out = inkex.etree.Element(inkex.addNS(node_in.tag,'svg'))
+        in_tag = node_in.tag.rsplit('}',1)[-1]
+        if in_tag != 'svg':
+            node_out = inkex.etree.Element(inkex.addNS(in_tag,'svg'))
             for name in node_in.attrib:
                 node_out.set(name, node_in.attrib[name])
         else:
             node_out = inkex.etree.Element(inkex.addNS('g','svg'))
         for c in node_in.iterchildren():
-            if c.tag in ('g', 'path', 'polyline', 'polygon'):
+            c_tag = c.tag.rsplit('}',1)[-1]
+            if c_tag in ('g', 'path', 'polyline', 'polygon'):
                 child = clone_and_rewrite(self, c)
-                if c.tag == 'g':
+                if c_tag == 'g':
                     child.set('transform','matrix('+str(doc_sizeH/700.)+',0,0,'+str(-doc_sizeH/700.)+','+str(-doc_sizeH*0.25)+','+str(doc_sizeW*0.75)+')')
                 node_out.append(child)
 
@@ -86,6 +88,7 @@
         dvi_file = os.path.join(base_dir, "eq.dvi")
         svg_file = os.path.join(base_dir, "eq.svg")
         out_file = os.path.join(base_dir, "eq.out")
+        err_file = os.path.join(base_dir, "eq.err")
 
         def clean():
             os.remove(latex_file)
@@ -95,11 +98,13 @@
             os.remove(dvi_file)
             os.remove(svg_file)
             os.remove(out_file)
+            if os.path.exists(err_file):
+                os.remove(err_file)
             os.rmdir(base_dir)
 
         create_equation_tex(latex_file, self.options.formula)
-        #os.system('cd ' + base_dir)
-        os.system('latex -output-directory=' + base_dir + ' -halt-on-error ' + latex_file + ' > ' + out_file)
+        os.system('latex "-output-directory=%s" -halt-on-error "%s" > "%s"' \
+                  % (base_dir, latex_file, out_file))
 	try:
 	    os.stat(dvi_file)
 	except OSError:
@@ -108,9 +113,23 @@
             print >>sys.stderr, "temporary files were left in:", base_dir
             sys.exit(1)
 
-        os.system('dvips -q -f -E -D 600 -y 5000 -o ' + ps_file + ' ' + dvi_file)
-        #os.system('cd ' + base_dir)
-        os.system('pstoedit -f plot-svg -dt -ssp ' + ps_file + ' ' + svg_file + '> ' + out_file)
+        os.system('dvips -q -f -E -D 600 -y 5000 -o "%s" "%s"' % (ps_file, dvi_file))
+        # cd to base_dir is necessary, because pstoedit writes
+        # temporary files to cwd and needs write permissions
+        separator = ';'
+        if os.name == 'nt':
+            separator = '&&'
+        os.system('cd "%s" %s pstoedit -f plot-svg -dt -ssp "%s" "%s" > "%s" 2> "%s"' \
+                  % (base_dir, separator, ps_file, svg_file, out_file, err_file))
+
+        # forward errors to stderr but skip pstoedit header
+        if os.path.exists(err_file):
+            err_stream = open(err_file, 'r')
+            for line in err_stream:
+                if not line.startswith('pstoedit: version'):
+                    sys.stderr.write(line + '\n')
+            err_stream.close()
+ 
         svg_open(self, svg_file)
 
         clean()
