diff -U 0 -r src.orig/Makefile src/Makefile
--- src.orig/Makefile	2003-09-03 00:00:31.900099744 -0400
+++ src/Makefile	2003-09-03 00:01:07.776645680 -0400
@@ -88 +88 @@
-	| sed s}HOME}"`head -1 home`"}g \
+	| sed s}HOME}"`head -n 1 home`"}g \
@@ -393 +393 @@
-	| sed s}HOME}"`head -1 home`"}g \
+	| sed s}HOME}"`head -n 1 home`"}g \
diff -U 0 -r src.orig/print-cc.sh src/print-cc.sh
--- src.orig/print-cc.sh	2003-09-03 00:00:31.902099440 -0400
+++ src/print-cc.sh	2003-09-03 00:01:07.777645528 -0400
@@ -1 +1 @@
-cc="`head -1 conf-cc`"
+cc="`head -n 1 conf-cc`"
diff -U 0 -r src.orig/print-ld.sh src/print-ld.sh
--- src.orig/print-ld.sh	2003-09-03 00:00:31.902099440 -0400
+++ src/print-ld.sh	2003-09-03 00:01:07.777645528 -0400
@@ -1 +1 @@
-ld="`head -1 conf-ld`"
+ld="`head -n 1 conf-ld`"
