--- erts/emulator/hipe/hipe_x86_signal.c	2008-06-12 14:17:48.831373565 +0200
+++ erts/emulator/hipe/hipe_x86_signal_ok.c	2008-06-12 14:43:05.132064804 +0200
@@ -34,7 +34,7 @@
 #endif
 #include "hipe_signal.h"
 
-#if __GLIBC__ == 2 && (__GLIBC_MINOR__ >= 3 && __GLIBC_MINOR__ <= 7)
+#if __GLIBC__ == 2 && __GLIBC_MINOR__ >= 3
 /* See comment below for glibc 2.2. */
 #ifndef __USE_GNU
 #define __USE_GNU		/* to un-hide RTLD_NEXT */
