diff -ur otp_src_R12B-4.orig/erts/emulator/hipe/hipe_x86_signal.c otp_src_R12B-4/erts/emulator/hipe/hipe_x86_signal.c
--- otp_src_R12B-4.orig/erts/emulator/hipe/hipe_x86_signal.c	2008-09-01 14:51:25.000000000 +0200
+++ otp_src_R12B-4/erts/emulator/hipe/hipe_x86_signal.c	2008-10-29 09:42:04.000000000 +0100
@@ -33,8 +33,7 @@
 #include "erl_alloc.h"
 #endif
 #include "hipe_signal.h"
-
-#if __GLIBC__ == 2 && (__GLIBC_MINOR__ >= 3)
+#if __GLIBC__ == 2 && (__GLIBC_MINOR__ >= 3 && __GLIBC_MINOR__ <= 7)
 /* See comment below for glibc 2.2. */
 #ifndef __USE_GNU
 #define __USE_GNU		/* to un-hide RTLD_NEXT */
