--- gnome-vfs-2.15.1/modules/inotify-helper.c.use-sys-inotify	2006-05-17 13:27:10.000000000 -0400
+++ gnome-vfs-2.15.1/modules/inotify-helper.c	2006-05-17 13:27:51.000000000 -0400
@@ -28,19 +28,11 @@
 #include <time.h>
 #include <string.h>
 #include <sys/ioctl.h>
-/* Just include the local header to stop all the pain */
-#include "local_inotify.h"
-#if 0
 #ifdef HAVE_SYS_INOTIFY_H
-/* We don't actually include the libc header, because there has been
- * problems with libc versions that was built without inotify support.
- * Instead we use the local version.
- */
-#include "local_inotify.h"
+#include <sys/inotify.h>
 #elif defined (HAVE_LINUX_INOTIFY_H)
 #include <linux/inotify.h>
 #endif
-#endif
 #include <libgnomevfs/gnome-vfs-module-shared.h>
 #include <libgnomevfs/gnome-vfs-utils.h>
 #include "inotify-helper.h"
--- gnome-vfs-2.15.1/modules/inotify-kernel.c.use-sys-inotify	2006-05-17 13:27:10.000000000 -0400
+++ gnome-vfs-2.15.1/modules/inotify-kernel.c	2006-05-17 13:28:33.000000000 -0400
@@ -24,22 +24,12 @@
 #include <glib.h>
 #include "inotify-kernel.h"
 
-/* Just include the local headers to stop all the pain */
-#include "local_inotify.h"
-#include "local_inotify_syscalls.h"
-#if 0
 #ifdef HAVE_SYS_INOTIFY_H
-/* We don't actually include the libc header, because there has been
- * problems with libc versions that was built without inotify support.
- * Instead we use the local version.
- */
-#include "local_inotify.h"
-#include "local_inotify_syscalls.h"
+#include <sys/inotify.h>
 #elif defined (HAVE_LINUX_INOTIFY_H)
 #include <linux/inotify.h>
 #include "local_inotify_syscalls.h"
 #endif
-#endif
 
 /* Timings for pairing MOVED_TO / MOVED_FROM events */
 #define PROCESS_EVENTS_TIME 125 /* milliseconds */
--- gnome-vfs-2.15.1/modules/inotify-path.c.use-sys-inotify	2006-05-17 13:27:10.000000000 -0400
+++ gnome-vfs-2.15.1/modules/inotify-path.c	2006-05-17 13:50:50.000000000 -0400
@@ -28,18 +28,14 @@
 /* Don't put conflicting kernel types in the global namespace: */
 #define __KERNEL_STRICT_NAMES
 
-#include "local_inotify.h"
-#if 0
 #ifdef HAVE_SYS_INOTIFY_H
-/* We don't actually include the libc header, because there has been
- * problems with libc versions that was built without inotify support.
- * Instead we use the local version.
- */
-#include "local_inotify.h"
+#include <sys/inotify.h>
+#ifndef IN_ONLYDIR
+#define IN_ONLYDIR              0x01000000 
+#endif
 #elif defined (HAVE_LINUX_INOTIFY_H)
 #include <linux/inotify.h>
 #endif
-#endif
 #include <string.h>
 #include <glib.h>
 #include <libgnomevfs/gnome-vfs-module-shared.h>
