diff -urN abuse_sdl-0.7.0/src/sdlport/setup.cpp abuse_sdl-0.7.0-new/src/sdlport/setup.cpp
--- abuse_sdl-0.7.0/src/sdlport/setup.cpp	2002-12-14 21:00:32.000000000 -0800
+++ abuse_sdl-0.7.0-new/src/sdlport/setup.cpp	2003-10-19 21:23:33.000000000 -0700
@@ -72,7 +72,7 @@
 	if( (fd = fopen( rcfile, "w" )) != NULL )
 	{
 		fputs( "; Abuse-SDL Configuration file\n\n", fd );
-		fputs( "; Location of the datafiles\ndatadir=/usr/local/share/games/abuse\n\n", fd );
+		fputs( "; Location of the datafiles\ndatadir=/usr/share/games/abuse\n\n", fd );
 		fputs( "; Startup fullscreen\nfullscreen=0\n\n", fd );
 		fputs( "; Use DoubleBuffering\ndoublebuf=0\n\n", fd );
 		fputs( "; Use mono audio only\nmono=0\n\n", fd );
