--- speech_tools/config/project.mak.orig	2007-03-13 15:27:53.000000000 -0400
+++ speech_tools/config/project.mak	2007-03-13 15:29:12.000000000 -0400
@@ -99,8 +99,9 @@
 PROJECT_LIBRARY_DIR_eststring = $(TOP)/lib
 PROJECT_LIBRARY_DIR_estjava = $(TOP)/lib
 
-PROJECT_LIBRARY_USES_estbase = eststring
-
+PROJECT_LIBRARY_USES_estbase = eststring m
+PROJECT_LIBRARY_USES_estools = estbase m ncurses
+PROJECT_LIBRARY_USES_eststring = m
 PROJECT_LIBRARY_USES_estjava = estbase eststring
 
 PROJECT_LIBRARY_VERSION_estools = $(PROJECT_VERSION).1

--- speech_tools/config/rules/defaults.mak.orig	2007-03-16 12:39:12.000000000 -0400
+++ speech_tools/config/rules/defaults.mak	2007-03-16 12:39:19.000000000 -0400
@@ -69,15 +69,21 @@
 	MADE_FROM_ABOVE:=$(N)
 endif
 
+ifneq ($(SHARED),0)
+    LIBTYPE=so
+else
+    LIBTYPE=a   
+endif
+
 ifndef PROJECT_LIBDEPS
-    PROJECT_LIBDEPS = $(foreach l,$(PROJECT_LIBRARIES),$(PROJECT_LIBRARY_DIR_$(l))/lib$(l).a)
+    PROJECT_LIBDEPS = $(foreach l,$(PROJECT_LIBRARIES),$(PROJECT_LIBRARY_DIR_$(l))/lib$(l).$(LIBTYPE))
 endif
 ifndef PROJECT_LIBS
     PROJECT_LIBS = $(foreach l,$(PROJECT_LIBRARIES),-L$(PROJECT_LIBRARY_DIR_$(l)) -l$(l))
 endif
 
 ifndef REQUIRED_LIBDEPS
-    REQUIRED_LIBDEPS = $(foreach l,$(REQUIRED_LIBRARIES),$(REQUIRED_LIBRARY_DIR_$(l))/lib$(l).a)
+    REQUIRED_LIBDEPS = $(foreach l,$(REQUIRED_LIBRARIES),$(REQUIRED_LIBRARY_DIR_$(l))/lib$(l).$(LIBTYPE))
 endif
 ifndef REQUIRED_LIBS
     REQUIRED_LIBS = $(foreach l,$(REQUIRED_LIBRARIES),-L$(REQUIRED_LIBRARY_DIR_$(l)) -l$(l))
--- festival/config/project.mak.orig	2007-03-16 13:01:40.000000000 -0400
+++ festival/config/project.mak	2007-03-16 13:02:24.000000000 -0400
@@ -84,6 +84,8 @@
 PROJECT_LIBRARY_DIR_Festival = $(TOP)/src/lib
 PROJECT_DEFAULT_LIBRARY = Festival
 
+PROJECT_LIBRARY_VERSION_Festival = $(PROJECT_VERSION).0
+
 # Libraries used from other projects
 
 REQUIRED_LIBRARIES = estools estbase eststring
