diff -u -r gnome-control-center-3.10.1/panels/background/Makefile.am gnome-control-center-3.10.1-buildfix/panels/background/Makefile.am
--- gnome-control-center-3.10.1/panels/background/Makefile.am	2013-10-11 13:19:22.000000000 +0200
+++ gnome-control-center-3.10.1-buildfix/panels/background/Makefile.am	2013-10-16 00:59:02.966707377 +0200
@@ -52,11 +52,6 @@
 libbackground_la_LIBADD += $(SOCIALWEB_LIBS)
 endif
 
-noinst_PROGRAMS = test-chooser-dialog
-test_chooser_dialog_SOURCES = test-chooser-dialog.c
-test_chooser_dialog_LDADD = libbackground-chooser.la
-test_chooser_dialog_LDFLAGS = $(PANEL_LIBS) $(BACKGROUND_PANEL_LIBS)
-
 resource_files = $(shell glib-compile-resources --sourcedir=$(srcdir) --generate-dependencies $(srcdir)/background.gresource.xml)
 cc-background-resources.c: background.gresource.xml $(resource_files)
 	$(AM_V_GEN) glib-compile-resources --target=$@ --sourcedir=$(srcdir) --generate-source --c-name cc_background $<
