diff -ru nfs-utils-1.0.10.orig/tools/Makefile.am nfs-utils-1.0.10/tools/Makefile.am
--- nfs-utils-1.0.10.orig/tools/Makefile.am	2006-08-07 06:40:50.000000000 +0000
+++ nfs-utils-1.0.10/tools/Makefile.am	2006-12-31 14:01:57.000000000 +0000
@@ -1,5 +1,5 @@
 ## Process this file with automake to produce Makefile.in
 
-SUBDIRS = getiversion getkversion locktest rpcdebug rpcgen nlmtest
+SUBDIRS = getiversion locktest rpcdebug rpcgen nlmtest
 
 MAINTAINERCLEANFILES = Makefile.in
diff -ru nfs-utils-1.0.10.orig/tools/Makefile.in nfs-utils-1.0.10/tools/Makefile.in
--- nfs-utils-1.0.10.orig/tools/Makefile.in	2006-08-07 06:43:43.000000000 +0000
+++ nfs-utils-1.0.10/tools/Makefile.in	2006-12-31 14:02:07.000000000 +0000
@@ -207,7 +207,7 @@
 statedir = @statedir@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
-SUBDIRS = getiversion getkversion locktest rpcdebug rpcgen nlmtest
+SUBDIRS = getiversion locktest rpcdebug rpcgen nlmtest
 MAINTAINERCLEANFILES = Makefile.in
 all: all-recursive
 
