diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6ab34dc..818197a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -189,12 +189,8 @@ if (ENABLE_LLVM_BACKEND)
             message(ERROR "Unknown processor, unable to set LLVM codegen targets")
         endif()
 
-        llvm_map_components_to_libnames(LLVM_LIB
-            Interpreter
-            MCJIT
-            ObjCARCOpts
-            Passes
-            ${SeExpr2_PLATFORM_CODEGEN_LIBS}
+        set(LLVM_LIB
+            LLVM
         )
 
         message(STATUS "LLVM_LIB = ${LLVM_LIB}")
