--- bridges/source/cpp_uno/gcc3_linux_intel/except.cxx
+++ bridges/source/cpp_uno/gcc3_linux_intel/except.cxx
@@ -46,6 +46,7 @@
 #include <typelib/typedescription.hxx>
 #include <uno/any2.h>
 
+#include <string.h>
 #include "share.hxx"
 
 
--- filter/source/pdfimport/wrapper/wrapper.cxx
+++ filter/source/pdfimport/wrapper/wrapper.cxx
@@ -72,6 +72,7 @@
 #include <hash_map>
 
 #include "rtl/bootstrap.h"
+#include "string.h"
 
 #ifndef PDFI_IMPL_IDENTIFIER
 # error define implementation name for pdfi extension, please!
