diff -Nur davfs2-1.4.7.orig/configure.ac davfs2-1.4.7/configure.ac
--- davfs2-1.4.7.orig/configure.ac	2012-07-19 13:38:47.000000000 +0200
+++ davfs2-1.4.7/configure.ac	2014-01-13 23:00:15.021471570 +0100
@@ -36,7 +36,7 @@
 # Checks for libraries.
 AM_GNU_GETTEXT_VERSION(0.14.4)
 AM_GNU_GETTEXT([external])
-NE_REQUIRE_VERSIONS([0], [25 26 27 28 29])
+NE_REQUIRE_VERSIONS([0], [25 26 27 28 29 30])
 DAV_CHECK_NEON
 
 # Checks for header files.
