Index: makefile.mk
===================================================================
RCS file: /cvs/api/bean/native/unix/makefile.mk,v
retrieving revision 1.5
diff -u -u -r1.5 makefile.mk
--- bean/native/unix/makefile.mk	15 Mar 2005 09:07:46 -0000	1.5
+++ bean/native/unix/makefile.mk	22 Mar 2005 14:06:03 -0000
@@ -41,6 +41,7 @@
 .ENDIF
 
 NO_SHL1DESCRIPTION=TRUE
+SHL1NOCHECK="TRUE"
 
 # --- Targets ------------------------------------------------------
 