--- desktop/source/deployment/gui/dp_gui_updatedialog.cxx
+++ desktop/source/deployment/gui/dp_gui_updatedialog.cxx
@@ -493,7 +493,7 @@
         package->getUpdateInformationURLs());
     if (urls.getLength() == 0) {
         map->insert(
-            std::pair< Map::key_type const, Map::data_type >(
+            Map::value_type(
                 id, Entry(package, packageManager, package->getVersion())));
     } else {
         css::uno::Sequence< css::uno::Reference< css::xml::dom::XElement > >
