--- cppu/inc/uno/lbnames.h.old	2010-02-02 12:49:41.000000000 +0100
+++ cppu/inc/uno/lbnames.h	2010-03-05 17:13:22.000000000 +0100
@@ -62,12 +62,12 @@ provoking error here, because PP ignores
 #error "Tested gcc 2 versions are 2.91 and 2.95.  Patch uno/lbnames.h to try your gcc 2 version."
 #elif (__GNUC__ == 3 && __GNUC_MINOR__ <= 4)
 #define __CPPU_ENV gcc3
-#elif (__GNUC__ == 4 && __GNUC_MINOR__ <= 4)
+#elif (__GNUC__ == 4 && __GNUC_MINOR__ <= 5)
 #define __CPPU_ENV gcc3
 #elif (__GNUC__ == 3)
 #error "Tested gcc 3 version is <= 3.4.  Patch uno/lbnames.h to try your gcc 3 version."
 #else
-#error "Supported gcc majors are 2 , 3 and 4 <= 4.4.  Unsupported gcc major version."
+#error "Supported gcc majors are 2 , 3 and 4 <= 4.5.  Unsupported gcc major version."
 #endif /* defined (__GNUC__) */
 #endif /* defined (_MSC_VER) */
 
