Index: officecfg/registry/data/org/openoffice/Office/Jobs.xcu
===================================================================
RCS file: /cvs/util/officecfg/registry/data/org/openoffice/Office/Jobs.xcu,v
retrieving revision 1.7
diff -u -p -u -r1.7 Jobs.xcu
--- officecfg/registry/data/org/openoffice/Office/Jobs.xcu	15 Nov 2004 15:05:43 -0000	1.7
+++ officecfg/registry/data/org/openoffice/Office/Jobs.xcu	10 Dec 2004 09:37:59 -0000
@@ -69,14 +69,5 @@
     </node>
   </node>
   <node oor:name="Events">
-    <node oor:name="onFirstVisibleTask" oor:op="fuse">
-      <node oor:name="JobList">
-        <node oor:name="RegistrationRequest" oor:op="replace">
-            <prop oor:name="UserTime">
-                <value>2003-01-01T00:00:00+00:01</value>
-            </prop>
-        </node>
-      </node>
-    </node>
   </node>
 </oor:component-data>
