Index: source/ucp/webdav/NeonSession.cxx
===================================================================
RCS file: /cvs/ucb/ucb/source/ucp/webdav/NeonSession.cxx,v
retrieving revision 1.49
diff -u -r1.49 NeonSession.cxx
--- openoffice.org.orig/ucb/source/ucp/webdav/NeonSession.cxx	18 Jul 2007 07:55:41 -0000	1.49
+++ openoffice.org/ucb/source/ucp/webdav/NeonSession.cxx	23 Aug 2007 09:48:56 -0000
@@ -364,32 +364,6 @@
 }
 
 // -------------------------------------------------------------------
-extern "C" void NeonSession_ProgressNotify( void *,
-                                            off_t,
-                                            off_t )
-{
-    // progress: bytes read so far
-    // total:    total bytes to read, -1 -> total count not known
-}
-
-// -------------------------------------------------------------------
-extern "C" void NeonSession_StatusNotify( void *,
-                                          ne_conn_status,
-                                          const char * )
-{
-#if 0
-    typedef enum {
-        ne_conn_namelookup, /* lookup up hostname (info = hostname) */
-        ne_conn_connecting, /* connecting to host (info = hostname) */
-        ne_conn_connected, /* connected to host (info = hostname) */
-        ne_conn_secure /* connection now secure (info = crypto level) */
-    } ne_conn_status;
-#endif
-
-    // info: hostname
-}
-
-// -------------------------------------------------------------------
 extern "C" void NeonSession_PreSendRequest( ne_request * req,
                                             void * userdata,
                                             ne_buffer * headers )
@@ -578,13 +552,6 @@
                                 NeonUri::makeConnectionEndPointString(
                                     m_aHostName, m_nPort ) );
 
-        // Set a progress callback for the session.
-        ne_set_progress( m_pHttpSession, NeonSession_ProgressNotify, this );
-
-        // Set a status notification callback for the session, to report
-        // connection status.
-        ne_set_status( m_pHttpSession, NeonSession_StatusNotify, this );
-
         // Add hooks (i.e. for adding additional headers to the request)
 
 #if 0
diff -ru config_office/configure.in config_office/configure.in
--- openoffice.org.orig/config_office/configure.in	2007-08-23 11:49:35.000000000 +0100
+++ openoffice.org/config_office/configure.in	2007-08-23 11:50:28.000000000 +0100
@@ -4019,8 +4019,7 @@
 if test -n "$with_system_neon" -o -n "$with_system_libs" && \
 	test "$with_system_neon" != "no"; then
         AC_MSG_RESULT([external])
-        PKG_CHECK_MODULES(NEON, neon >= 0.26.0)
-        PKG_CHECK_MODULES(NEON, neon < 0.27.0, , AC_MSG_ERROR([you need neon 0.26.x for system-neon]))
+        PKG_CHECK_MODULES(NEON, neon >= 0.26.0, , AC_MSG_ERROR([you need neon >= 0.26.x for system-neon]))
         NEON_VERSION="`$PKG_CONFIG --modversion neon | $SED 's/\.//g'`"
         NEON_CFLAGS="$NEON_CFLAGS -DSYSTEM_NEON -DUSE_DAV_LOCKS=1"
         SYSTEM_NEON=YES

