diff -up lftp-4.3.8/lib/stdio.in.h.gets lftp-4.3.8/lib/stdio.in.h
--- lftp-4.3.8/lib/stdio.in.h.gets	2012-07-22 15:07:15.101115146 -0400
+++ lftp-4.3.8/lib/stdio.in.h	2012-07-22 15:08:09.777098512 -0400
@@ -702,10 +702,12 @@ _GL_WARN_ON_USE (getline, "getline is un
 /* It is very rare that the developer ever has full control of stdin,
    so any use of gets warrants an unconditional warning; besides, C11
    removed it.  */
+#if defined gets
 #undef gets
 #if HAVE_RAW_DECL_GETS
 _GL_WARN_ON_USE (gets, "gets is a security hole - use fgets instead");
 #endif
+#endif // if defined gets
 
 
 #if @GNULIB_OBSTACK_PRINTF@ || @GNULIB_OBSTACK_PRINTF_POSIX@
