diff -u -r libmbim-1.0.0/cli/mbimcli-basic-connect.c libmbim-1.0.0-format/cli/mbimcli-basic-connect.c
--- libmbim-1.0.0/cli/mbimcli-basic-connect.c	2013-05-22 16:20:29.000000000 +0200
+++ libmbim-1.0.0-format/cli/mbimcli-basic-connect.c	2013-05-30 05:53:49.269777442 +0200
@@ -909,8 +909,8 @@
              "\t         Network error: '%s'\n"
              "\t  Packet service state: '%s'\n"
              "\tAvailable data classes: '%s'\n"
-             "\t          Uplink speed: '%lu bps'\n"
-             "\t        Downlink speed: '%lu bps'\n",
+             "\t          Uplink speed: '%" G_GUINT64_FORMAT " bps'\n"
+             "\t        Downlink speed: '%" G_GUINT64_FORMAT " bps'\n",
              mbim_device_get_path_display (device),
              VALIDATE_UNKNOWN (mbim_nw_error_get_string (nw_error)),
              VALIDATE_UNKNOWN (mbim_packet_service_state_get_string (packet_service_state)),
