diff -ur gnumeric-1.7.12/plugins/fn-tsa/functions.c gnumeric-new/plugins/fn-tsa/functions.c
--- gnumeric-1.7.12/plugins/fn-tsa/functions.c	2007-09-01 20:07:39.000000000 -0500
+++ gnumeric-new/plugins/fn-tsa/functions.c	2007-10-21 23:36:39.000000000 -0500
@@ -45,7 +45,7 @@
 #include <expr-impl.h>
 #include <func-builtin.h>
 #include <gnm-i18n.h>
-#include <goffice/utils/go-cspline.h>
+#include <goffice/math/go-cspline.h>
 #include <gnm-plugin.h>
 #include <tools/analysis-tools.h>
 
diff -ur gnumeric-1.7.12/src/gnm-so-filled.c gnumeric-new/src/gnm-so-filled.c
--- gnumeric-1.7.12/src/gnm-so-filled.c	2007-09-01 20:07:07.000000000 -0500
+++ gnumeric-new/src/gnm-so-filled.c	2007-10-21 23:22:29.000000000 -0500
@@ -272,7 +272,7 @@
 		cairo_close_path (cairo);
 	}
 	/* Fill the shape */
-	pat = gog_style_create_cairo_pattern (style, width, height, &obj);
+	pat = gog_style_create_cairo_pattern (style, cairo);
 	if (pat) {
 		cairo_set_source (cairo, pat);
 		cairo_fill_preserve (cairo);
diff -ur gnumeric-1.7.12/src/graph.c gnumeric-new/src/graph.c
--- gnumeric-1.7.12/src/graph.c	2007-09-01 20:07:07.000000000 -0500
+++ gnumeric-new/src/graph.c	2007-10-21 23:18:48.000000000 -0500
@@ -36,7 +36,7 @@
 #include "ranges.h"
 #include "parse-util.h"
 #include <goffice/data/go-data-impl.h>
-#include <goffice/utils/go-math.h>
+#include <goffice/math/go-math.h>
 
 #include <gsf/gsf-impl-utils.h>
 #include <string.h>
diff -ur gnumeric-1.7.12/src/mathfunc.c gnumeric-new/src/mathfunc.c
--- gnumeric-1.7.12/src/mathfunc.c	2007-09-01 20:07:07.000000000 -0500
+++ gnumeric-new/src/mathfunc.c	2007-10-21 23:19:46.000000000 -0500
@@ -62,7 +62,7 @@
 #include <unistd.h>
 #include <locale.h>
 #include <string.h>
-#include <goffice/utils/go-math.h>
+#include <goffice/math/go-math.h>
 #include <glib/gstdio.h>
 
 #if defined (HAVE_IEEEFP_H) || defined (HAVE_IEEE754_H)
diff -ur gnumeric-1.7.12/src/numbers.h gnumeric-new/src/numbers.h
--- gnumeric-1.7.12/src/numbers.h	2007-09-01 20:07:07.000000000 -0500
+++ gnumeric-new/src/numbers.h	2007-10-21 23:11:32.000000000 -0500
@@ -2,7 +2,7 @@
 #define GNUMERIC_NUMBERS_H
 
 #include <gnumeric-config.h>
-#include <goffice/utils/go-math.h>
+#include <goffice/math/go-math.h>
 
 #ifdef HAVE_IEEEFP_H
 #include <ieeefp.h>
diff -ur gnumeric-1.7.12/src/rangefunc.h gnumeric-new/src/rangefunc.h
--- gnumeric-1.7.12/src/rangefunc.h	2007-09-01 20:07:07.000000000 -0500
+++ gnumeric-new/src/rangefunc.h	2007-10-21 23:16:41.000000000 -0500
@@ -2,7 +2,7 @@
 #define GNUMERIC_RANGEFUNC_H
 
 #include "numbers.h"
-#include <goffice/utils/go-rangefunc.h>
+#include <goffice/math/go-rangefunc.h>
 
 #ifdef WITH_LONG_DOUBLE
 #	define gnm_range_sum go_range_suml
diff -ur gnumeric-1.7.12/src/regression.h gnumeric-new/src/regression.h
--- gnumeric-1.7.12/src/regression.h	2007-09-01 20:07:07.000000000 -0500
+++ gnumeric-new/src/regression.h	2007-10-21 23:12:50.000000000 -0500
@@ -2,7 +2,7 @@
 #define GNUMERIC_REGRESSION_H
 
 #include "numbers.h"
-#include <goffice/utils/go-regression.h>
+#include <goffice/math/go-regression.h>
 
 #ifdef WITH_LONG_DOUBLE
 #	define gnm_regression_stat_t go_regression_stat_tl
diff -ur gnumeric-1.7.12/src/sheet-object-graph.c gnumeric-new/src/sheet-object-graph.c
--- gnumeric-1.7.12/src/sheet-object-graph.c	2007-09-01 20:07:07.000000000 -0500
+++ gnumeric-new/src/sheet-object-graph.c	2007-10-21 23:32:21.000000000 -0500
@@ -389,7 +389,7 @@
 			  double width, double height)
 {
 #ifdef GOFFICE_WITH_CAIRO
-	gog_graph_render_to_cairo_vector
+	gog_graph_render_to_cairo
 		(SHEET_OBJECT_GRAPH (so)->graph, data, width, height);
 #endif
 }
@@ -610,7 +610,7 @@
 	if (sog->renderer != NULL)
 		g_object_set (sog->renderer, "model", graph, NULL);
 	else
-		sog->renderer = gog_renderer_new_for_pixbuf (sog->graph);
+		sog->renderer = gog_renderer_new (sog->graph);
 }
 
 static void
