--- patch-sane.sh~	2004-11-21 14:28:11.000000000 +0100
+++ patch-sane.sh	2007-04-15 17:23:40.000000000 +0200
@@ -94,9 +94,9 @@
 
 # Adds a line with 'hp_rts88xx' to dll.conf
 echo -n "Modifying backend/dll.conf..."
-if [ -z $(grep hp_rts88xx $sane/backend/dll.conf) ]
+if [ -z $(grep hp_rts88xx $sane/backend/dll.conf.in) ]
     then
-    echo "hp_rts88xx" >> $sane/backend/dll.conf
+    echo "hp_rts88xx" >> $sane/backend/dll.conf.in
 fi
 evaluate_retval
 
