diff --git a/systemd/gpsdctl@.service b/systemd/gpsdctl@.service
index 58c5176..3d1590d 100644
--- a/systemd/gpsdctl@.service
+++ b/systemd/gpsdctl@.service
@@ -6,8 +6,9 @@ After=dev-%i.device
 
 [Service]
 Type=oneshot
-Environment="GPSD_SOCKET=/var/run/gpsd.sock"
-EnvironmentFile=-/etc/default/gpsd
+EnvironmentFile=/etc/gpsd.conf
 RemainAfterExit=yes
-ExecStart=/bin/sh -c "[ \"$USBAUTO\" = true ] && /usr/sbin/gpsdctl add /dev/%I || :"
-ExecStop=/bin/sh -c "[ \"$USBAUTO\" = true ] && /usr/sbin/gpsdctl remove /dev/%I || :"
+
+ExecStartPre=/usr/bin/chmod a+rw /dev/%I
+ExecStart=/usr/bin/gpsdctl add /dev/%I
+ExecStop=/usr/bin/gpsdctl remove /dev/%I
