--- solenv/bin/modules/installer/control.pm
+++ solenv/bin/modules/installer/control.pm
@@ -714,7 +714,8 @@
 		
 	if ( $installer::globals::patch ) {	$installer::globals::addsystemintegration = 0; }
 	if ( $installer::globals::languagepack ) { $installer::globals::addsystemintegration = 0; }
-
+if ( $installer::globals::packageformat eq "native" ) { 
+	$installer::globals::addsystemintegration = 0; }
 	my $infoline = "Value of \$installer::globals::addsystemintegration: $installer::globals::addsystemintegration\n";
 	push( @installer::globals::globallogfileinfo, $infoline);
 }
