--- desktop/source/migration/wizard.cxx
+++ desktop/source/migration/wizard.cxx
@@ -121,6 +121,7 @@
     ,m_bLicenseWasAccepted(sal_False)
     ,m_bAutomaticUpdChk(sal_True)
 {
+	return;
     // ---
     // FreeResource();
 //	enableState(STATE_USER, sal_False);
@@ -381,7 +382,7 @@
 
 short FirstStartWizard::Execute()
 {
-    return svt::RoadmapWizard::Execute();
+    return sal_True || svt::RoadmapWizard::Execute();
 }
 
 
