Index: makefile.pmk
===================================================================
--- helpcontent2/makefile.pmk	(revision 268394)
+++ helpcontent2/makefile.pmk	(working copy)
@@ -44,7 +44,7 @@
 
 $(COMMONMISC)$/$(TARGET).done : $(HLANGXHPFILES)
 .IF "$(WITH_LANG)"!=""
-	@$(WRAPCMD) helpex -QQ -p helpcontent2 -r $(PRJ) -i @$(mktmp $(uniq $(foreach,i,$? $(!eq,$(i:f),$(i:f:s/.xhp//) $(i:f) $(XHPFILES))))) -x $(COMMONMISC) -y $(SHELL_PACKAGE) -l all -lf $(aux_alllangiso:t",") -m localize.sdf && $(TOUCH) $@
+	@$(AUGMENT_LIBRARY_PATH) $(WRAPCMD) helpex -QQ -p helpcontent2 -r $(PRJ) -i @$(mktmp $(uniq $(foreach,i,$? $(!eq,$(i:f),$(i:f:s/.xhp//) $(i:f) $(XHPFILES))))) -x $(COMMONMISC) -y $(SHELL_PACKAGE) -l all -lf $(aux_alllangiso:t",") -m localize.sdf && $(TOUCH) $@
 .ELSE			# "$(WITH_LANG)"!=""
 .IF "$(USE_SHELL)"=="4nt"
 # it's getting real ugly...
