diff --git a/dictionaries/prj/build.lst b/dictionaries/prj/build.lst
--- a/dictionaries/prj/build.lst
+++ b/dictionaries/prj/build.lst
@@ -1,4 +1,4 @@
-di      dictionaries : soltools MYTHES:mythes NULL
+di      dictionaries : soltools mythes NULL
 di	dictionaries					usr1 - all di_mkout NULL
 di      dictionaries\af_ZA      nmake   -       all     di_af_ZA NULL
 di      dictionaries\ca         nmake   -       all     di_ca NULL
diff --git a/mythes/makefile.mk b/mythes/makefile.mk
--- a/mythes/makefile.mk
+++ b/mythes/makefile.mk
@@ -47,6 +47,8 @@
 .IF "$(GUI)"=="UNX"
 CONFIGURE_DIR=$(BUILD_DIR)
 
+.IF "$(SYSTEM_MYTHES)" != "YES"
+
 .IF "$(SYSTEM_HUNSPELL)" != "YES"
 HUNSPELL_CFLAGS +:= -I$(SOLARINCDIR)$/hunspell
 HUNSPELL_LIBS +:= -L$(SOLARLIBDIR) -lhunspell-1.2
@@ -74,14 +76,9 @@
 CONFIGURE_FLAGS+=CFLAGS='$(LCL_CONFIGURE_CFLAGS)'
 .ENDIF
 
-.IF "$(SYSTEM_MYTHES)" == "YES"
-@all:
-	echo "Nothing to do here."
-.ELSE
 BUILD_ACTION=make
 OUT2INC += mythes.hxx
 .ENDIF
-
 .ENDIF # "$(GUI)"=="UNX"
 
 
