Modified: cws/configuretoplevel/configure.in
==============================================================================
--- config_office/configure.in	Mon Dec  8 16:39:46 2008	(r265012)
+++ config_office/configure.in	Mon Dec  8 16:51:28 2008	(r265013)
@@ -4028,7 +4028,7 @@
 
     if $PKG_CONFIG --exists libxul ; then
       PKG_CHECK_MODULES( MOZILLAXPCOM, libxul )
-      MOZ_INC=`$PKG_CONFIG --variable=includedir libxul`
+      MOZ_INC=`$PKG_CONFIG --variable=includedir libxul`/stable
       MOZ_LIB=`$PKG_CONFIG --variable=libdir libxul`
       if ! test -e "$MOZ_LIB/libxul.so"; then
          MOZ_LIB=`$PKG_CONFIG --variable=sdkdir libxul`

Index: makefile.mk
===================================================================
--- extensions/source/plugin/base/makefile.mk	(Revision 264632)
+++ extensions/source/plugin/base/makefile.mk	(Arbeitskopie)
@@ -37,10 +37,19 @@
 
 .INCLUDE :  ..$/util$/makefile.pmk
 
+.IF "$(SYSTEM_MOZILLA)" == "YES"
+INCPRE+=-I$(MOZ_INC) -I$(MOZ_INC)$/plugin
+.ELSE
 INCPRE+=-I$(SOLARINCDIR)$/mozilla$/plugin
+.ENDIF
 .IF "$(SOLAR_JAVA)" != ""
+.IF "$(SYSTEM_MOZILLA)" == "YES"
+INCPRE+=-I$(MOZ_INC)$/java
+INCPRE+=-I$(MOZ_INC) -I$(MOZ_INC)$/nspr
+.ELSE
 INCPRE+=-I$(SOLARINCDIR)$/mozilla$/java
 INCPRE+=-I$(SOLARINCDIR)$/mozilla$/nspr
+.ENDIF
 CDEFS+=-DOJI
 .ENDIF
 
Index: makefile.mk
===================================================================
--- extensions/source/plugin/unx/makefile.mk	(Revision 264632)
+++ extensions/source/plugin/unx/makefile.mk	(Arbeitskopie)
@@ -45,10 +45,19 @@
 
 # --- Files --------------------------------------------------------
 
+.IF "$(SYSTEM_MOZILLA)" == "YES"
+INCPRE+=-I$(MOZ_INC) -I$(MOZ_INC)/plugin
+.ELSE
 INCPRE+=-I$(SOLARINCDIR)$/mozilla$/plugin
+.ENDIF
 .IF "$(SOLAR_JAVA)" != ""
+.IF "$(SYSTEM_MOZILLA)" == "YES"
+INCPRE+=-I$(MOZ_INC)$/java
+INCPRE+=-I$(MOZ_INC) -I$(MOZ_INC)$/nspr
+.ELSE
 INCPRE+=-I$(SOLARINCDIR)$/mozilla$/java
 INCPRE+=-I$(SOLARINCDIR)$/mozilla$/nspr
+.ENDIF
 CDEFS+=-DOJI
 .ENDIF
 

