diff -r 743094d52f7e shell/source/backends/gconfbe/gconfaccess.cxx
--- a/shell/source/backends/gconfbe/gconfaccess.cxx	Wed Mar 10 08:14:34 2010 +0100
+++ b/shell/source/backends/gconfbe/gconfaccess.cxx	Wed Mar 10 09:50:41 2010 +0100
@@ -448,7 +448,6 @@
 #ifdef ENABLE_LOCKDOWN
         case SETTING_AUTO_SAVE:
         {
-            GConfClient* aClient = GconfBackend::getGconfClient();
             GConfValue* aGconfValue = gconf_client_get( aClient, GCONF_AUTO_SAVE_KEY, NULL );
 
             if( ( aGconfValue != NULL ) )

