diff --git i/tests/dbusmock/accounts_service.py w/tests/dbusmock/accounts_service.py
index 81692b2..eabe12c 100644
--- i/tests/dbusmock/accounts_service.py
+++ w/tests/dbusmock/accounts_service.py
@@ -170,8 +170,9 @@ def CreateUser(self, name, fullname, account_type):
 
     self.users_auto_uids += 1
 
-    return self.AddUser(self.users_auto_uids, name, DEFAULT_USER_PASSWORD, {
-        'RealName': fullname, 'AccountType': account_type})
+    return self.AddUser(self.users_auto_uids, name, DEFAULT_USER_PASSWORD,
+        dbus.Dictionary({'RealName': fullname, 'AccountType': account_type},
+            signature='sv'))
 
 
 @dbus.service.method(MAIN_IFACE, in_signature='xb')
