From 6a0d9896165998ef415c7169dcd8fe7eb2ec41cf Mon Sep 17 00:00:00 2001
From: Thiago Macieira <thiago.macieira@intel.com>
Date: Thu, 27 Feb 2020 08:08:25 -0800
Subject: [PATCH] QMimeXMLProvider: add missing semi-colon in the #else case

Fixes: QTBUG-82547
Change-Id: Ia8b65350cd5d49debca9fffd15f74e22c0536805
---

diff --git a/src/corelib/mimetypes/qmimeprovider_p.h b/src/corelib/mimetypes/qmimeprovider_p.h
index 0629df8..c4e712b 100644
--- a/src/corelib/mimetypes/qmimeprovider_p.h
+++ b/src/corelib/mimetypes/qmimeprovider_p.h
@@ -140,7 +140,7 @@
     enum : bool { InternalDatabaseAvailable = false };
     QMimeXMLProvider(QMimeDatabasePrivate *db, InternalDatabaseEnum)
         : QMimeProviderBase(db, QString())
-    { Q_UNREACHABLE() };
+    { Q_UNREACHABLE(); };
 #endif
     QMimeXMLProvider(QMimeDatabasePrivate *db, const QString &directory);
     ~QMimeXMLProvider();
