diff -u -r wpewebkit-2.34.1/Source/PlatformWPE.cmake wpewebkit-2.34.1-doc/Source/PlatformWPE.cmake
--- wpewebkit-2.34.1/Source/PlatformWPE.cmake	2021-09-16 12:43:07.000000000 +0000
+++ wpewebkit-2.34.1-doc/Source/PlatformWPE.cmake	2021-11-03 20:19:45.981812827 +0000
@@ -10,10 +10,10 @@
 )
 
 if (ENABLE_GTKDOC)
-    install(DIRECTORY ${CMAKE_BINARY_DIR}/Documentation/wpe-${WPE_API_VERSION}/html/wpe-${WPE_API_VERSION}
-            DESTINATION "${CMAKE_INSTALL_DATADIR}/gtk-doc/html"
+    install(DIRECTORY ${CMAKE_BINARY_DIR}/Documentation/wpe-${WPE_API_DOC_VERSION}/html/wpe-${WPE_API_DOC_VERSION}
+        DESTINATION "${CMAKE_INSTALL_DATADIR}/gtk-doc/html"
     )
-    install(DIRECTORY ${CMAKE_BINARY_DIR}/Documentation/wpe-webextensions-${WPE_API_VERSION}/html/wpe-webextensions-${WPE_API_VERSION}
+    install(DIRECTORY ${CMAKE_BINARY_DIR}/Documentation/wpe-webextensions-${WPE_API_DOC_VERSION}/html/wpe-webextensions-${WPE_API_DOC_VERSION}
         DESTINATION "${CMAKE_INSTALL_DATADIR}/gtk-doc/html"
     )
 endif ()
