diff -Nur icedtea-web-1-0-b3647620fa41.orig/Makefile.am icedtea-web-1-0-b3647620fa41/Makefile.am
--- icedtea-web-1-0-b3647620fa41.orig/Makefile.am	2011-01-07 05:08:41.000000000 -0800
+++ icedtea-web-1-0-b3647620fa41/Makefile.am	2011-01-13 13:25:41.170004189 -0800
@@ -114,8 +114,8 @@
 	${INSTALL_PROGRAM} $(NETX_DIR)/launcher/controlpanel/itweb-settings $(DESTDIR)$(bindir)
 
 install-data-local:
-	${mkinstalldirs} -d $(DESTDIR)$(prefix)/man/man1
-	${INSTALL_DATA} $(NETX_SRCDIR)/javaws.1 $(DESTDIR)$(prefix)/man/man1
+	${mkinstalldirs} -d $(DESTDIR)$(mandir)/man1
+	${INSTALL_DATA} $(NETX_SRCDIR)/javaws.1 $(DESTDIR)$(mandir)/man1
 	${INSTALL_DATA} $(NETX_RESOURCE_DIR)/about.jnlp $(DESTDIR)$(prefix)/jre/lib
 if ENABLE_DOCS
 	${mkinstalldirs} $(DESTDIR)$(htmldir)
