diff -ur Intel-536.orig/coredrv/lock_lin.h Intel-536/coredrv/lock_lin.h
--- Intel-536.orig/coredrv/lock_lin.h	2005-07-26 17:59:33.000000000 +0200
+++ Intel-536/coredrv/lock_lin.h	2006-07-31 18:23:28.000000000 +0200
@@ -89,44 +89,44 @@
 
 
 
-extern void linux_MT_AcquireLock_LAPMrx_que(void);
-extern void linux_MT_ReleaseLock_LAPMrx_que(void);
-extern void linux_MT_AcquireLock_LAPMtx_que(void);
-extern void linux_MT_ReleaseLock_LAPMtx_que(void);
-extern void linux_MT_AcquireLock_LAPMemptied_que(void);
-extern void linux_MT_ReleaseLock_LAPMemptied_que(void);
-extern void linux_MT_AcquireLock_PortQIn(void);
-extern void linux_MT_ReleaseLock_PortQIn(void);
-extern void linux_MT_AcquireLock_PortQOut(void);
-extern void linux_MT_ReleaseLock_PortQOut(void);
-extern void linux_MT_AcquireLock_dte_rx_buffer(void);
-extern void linux_MT_ReleaseLock_dte_rx_buffer(void);
-extern void linux_MT_AcquireLock_dte_tx_buffer(void);
-extern void linux_MT_ReleaseLock_dte_tx_buffer(void);
-extern void linux_MT_AcquireLock_acu_rx_buffer(void);
-extern void linux_MT_ReleaseLock_acu_rx_buffer(void);
-extern void linux_MT_AcquireLock_acu_tx_buffer(void);
-extern void linux_MT_ReleaseLock_acu_tx_buffer(void);
-extern void linux_MT_AcquireLock_dce_rx_buffer(void);
-extern void linux_MT_ReleaseLock_dce_rx_buffer(void);
-extern void linux_MT_AcquireLock_dce_tx_buffer(void);
-extern void linux_MT_ReleaseLock_dce_tx_buffer(void);
-extern void linux_MT_AcquireLock_received_que(void);
-extern void linux_MT_ReleaseLock_received_que(void);
-extern void linux_MT_AcquireLock_emptied_que(void);
-extern void linux_MT_ReleaseLock_emptied_que(void);
-extern void linux_MT_AcquireLock_transmit_que(void);
-extern void linux_MT_ReleaseLock_transmit_que(void);
-extern void linux_MT_AcquireLock_clm_bg(void);
-extern void linux_MT_ReleaseLock_clm_bg(void);
-extern void linux_MT_AcquireLock_timer_task(void);
-extern void linux_MT_ReleaseLock_timer_task(void);
-extern void linux_MT_AcquireLock_mt_dsp(void);
-extern void linux_MT_ReleaseLock_mt_dsp(void);
-extern void linux_AcquireLock_data_to_user(void);
-extern void linux_ReleaseLock_data_to_user(void);
-extern void linux_AcquireLock_rts1(void);
-extern void linux_ReleaseLock_rts1(void);
+extern asmlinkage void linux_MT_AcquireLock_LAPMrx_que(void);
+extern asmlinkage void linux_MT_ReleaseLock_LAPMrx_que(void);
+extern asmlinkage void linux_MT_AcquireLock_LAPMtx_que(void);
+extern asmlinkage void linux_MT_ReleaseLock_LAPMtx_que(void);
+extern asmlinkage void linux_MT_AcquireLock_LAPMemptied_que(void);
+extern asmlinkage void linux_MT_ReleaseLock_LAPMemptied_que(void);
+extern asmlinkage void linux_MT_AcquireLock_PortQIn(void);
+extern asmlinkage void linux_MT_ReleaseLock_PortQIn(void);
+extern asmlinkage void linux_MT_AcquireLock_PortQOut(void);
+extern asmlinkage void linux_MT_ReleaseLock_PortQOut(void);
+extern asmlinkage void linux_MT_AcquireLock_dte_rx_buffer(void);
+extern asmlinkage void linux_MT_ReleaseLock_dte_rx_buffer(void);
+extern asmlinkage void linux_MT_AcquireLock_dte_tx_buffer(void);
+extern asmlinkage void linux_MT_ReleaseLock_dte_tx_buffer(void);
+extern asmlinkage void linux_MT_AcquireLock_acu_rx_buffer(void);
+extern asmlinkage void linux_MT_ReleaseLock_acu_rx_buffer(void);
+extern asmlinkage void linux_MT_AcquireLock_acu_tx_buffer(void);
+extern asmlinkage void linux_MT_ReleaseLock_acu_tx_buffer(void);
+extern asmlinkage void linux_MT_AcquireLock_dce_rx_buffer(void);
+extern asmlinkage void linux_MT_ReleaseLock_dce_rx_buffer(void);
+extern asmlinkage void linux_MT_AcquireLock_dce_tx_buffer(void);
+extern asmlinkage void linux_MT_ReleaseLock_dce_tx_buffer(void);
+extern asmlinkage void linux_MT_AcquireLock_received_que(void);
+extern asmlinkage void linux_MT_ReleaseLock_received_que(void);
+extern asmlinkage void linux_MT_AcquireLock_emptied_que(void);
+extern asmlinkage void linux_MT_ReleaseLock_emptied_que(void);
+extern asmlinkage void linux_MT_AcquireLock_transmit_que(void);
+extern asmlinkage void linux_MT_ReleaseLock_transmit_que(void);
+extern asmlinkage void linux_MT_AcquireLock_clm_bg(void);
+extern asmlinkage void linux_MT_ReleaseLock_clm_bg(void);
+extern asmlinkage void linux_MT_AcquireLock_timer_task(void);
+extern asmlinkage void linux_MT_ReleaseLock_timer_task(void);
+extern asmlinkage void linux_MT_AcquireLock_mt_dsp(void);
+extern asmlinkage void linux_MT_ReleaseLock_mt_dsp(void);
+extern asmlinkage void linux_AcquireLock_data_to_user(void);
+extern asmlinkage void linux_ReleaseLock_data_to_user(void);
+extern asmlinkage void linux_AcquireLock_rts1(void);
+extern asmlinkage void linux_ReleaseLock_rts1(void);
 
 #if(1)
 #define  MT_AcquireLock_mt_dsp()          linux_MT_AcquireLock_mt_dsp();
