--- coredrv/coredrv.c~	2006-03-29 11:45:23.000000000 +0200
+++ coredrv/coredrv.c	2006-03-29 12:03:20.000000000 +0200
@@ -48,6 +48,7 @@
 
 #ifdef CONFIG_PM
 #include <linux/pm.h>
+#include <linux/pm_legacy.h>
 #endif
 
 #include "uart.h"
@@ -71,7 +72,7 @@
 
 #include"softcore.h"
 int softcore_init_struct(struct softcore_struct*);
-EXPORT_SYMBOL_NOVERS(softcore_init_struct);
+EXPORT_SYMBOL(softcore_init_struct);
 int open(void);
 void close(void);
 void stop(void);
