diff -Naur subversion-1.6.17.ori/subversion/bindings/swig/perl/native/Makefile.PL.in subversion-1.6.17/subversion/bindings/swig/perl/native/Makefile.PL.in
--- subversion-1.6.17.ori/subversion/bindings/swig/perl/native/Makefile.PL.in	2010-11-24 20:42:16.000000000 +0000
+++ subversion-1.6.17/subversion/bindings/swig/perl/native/Makefile.PL.in	2011-07-01 20:16:16.520892074 +0000
@@ -43,7 +43,7 @@
 my %config = (
     ABSTRACT => 'Perl bindings for Subversion',
     DEFINE => $cppflags,
-    CCFLAGS => $cflags,
+    CCFLAGS => $Config{ccflags},
     INC  => join(' ',$apr_cflags, $apu_cflags, 
                  " -I$swig_srcdir/perl/libsvn_swig_perl",
                  " -I$svnlib_srcdir/include",
