--- Xaw3d/XawI18n.h.orig	1996-10-15 14:41:26.000000000 +0000
+++ Xaw3d/XawI18n.h	2005-12-21 13:21:57.000000000 +0000
@@ -26,10 +26,11 @@
 in this Software without prior written authorization from the X Consortium.
 
 ********************************************************/
-
+#if 0
 #ifdef HAS_WCTYPE_H
 #include <wctype.h>
 #include <widec.h>
+#endif
 #define wcslen(c) wslen(c)
 #define wcscpy(d,s) wscpy(d,s)
 #define wcsncpy(d,s,l) wsncpy(d,s,l)
--- Xaw3d/Imakefile.orig	2005-12-21 13:20:04.000000000 +0000
+++ Xaw3d/Imakefile	2005-12-21 13:20:09.000000000 +0000
@@ -39,9 +39,9 @@
 
        LINTLIBS = $(LINTXLIB) $(LINTXTOOL)
 
-#  EXTRA_DEFINES = -DARROW_SCROLLBAR -DUSEGRAY
+  EXTRA_DEFINES = -DARROW_SCROLLBAR -DUSEGRAY
 
-#  EXTRA_INCLUDES = -I.
+  EXTRA_INCLUDES = -I.
 
 HEADERS = \
 	AllWidgets.h \
