--- cmake/modules/FindQCA2.cmake.orig	2014-12-02 22:38:40.129259069 +0100
+++ cmake/modules/FindQCA2.cmake	2014-12-02 23:05:45.986414768 +0100
@@ -17,13 +17,13 @@
 
 if (NOT WIN32)
   find_package(PkgConfig QUIET)
-  pkg_check_modules(PC_QCA2 QUIET qca2)
+  pkg_search_module(PC_QCA2 QUIET qca2-qt5 qca2)
   set(QCA2_DEFINITIONS ${PC_QCA2_CFLAGS_OTHER})
 endif (NOT WIN32)
 
 find_library(QCA2_LIBRARIES
              WIN32_DEBUG_POSTFIX d
-             NAMES qca
+             NAMES qca-qt5 qca
              HINTS ${PC_QCA2_LIBDIR} ${PC_QCA2_LIBRARY_DIRS}
 )
 
