From 1282d9e3756d38362a5dc34c41ba0dd6c5c3ff71 Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <jan.steffens@gmail.com>
Date: Fri, 17 Feb 2017 23:38:48 +0100
Subject: [PATCH] meson: Fix library name of optical-media plugin

---
 src/optical-media/meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/optical-media/meson.build b/src/optical-media/meson.build
index aaf6a68786853aec..5a8845e4aecf37bb 100644
--- a/src/optical-media/meson.build
+++ b/src/optical-media/meson.build
@@ -13,7 +13,7 @@ optical_media_sources = [
 configure_file(output: 'config.h',
     configuration: cdata)
 
-shared_library('grlopticalmedia',
+shared_library('grloptical-media',
     sources: optical_media_sources,
     install: true,
     install_dir: pluginsdir,
-- 
2.11.1

