--- src/wavpack_tag.c
+++ src/wavpack_tag.c
@@ -82,15 +82,15 @@ gboolean Wavpack_Tag_Write_File (FILE *file_in, gchar *filename_in, vcedit_state
  */
 gboolean Wavpack_Tag_Read_File_Tag (gchar *filename, File_Tag *FileTag)
 {
-    if (!filename || !FileTag)
-        return FALSE;
-
     WavpackContext *wpc;
     gchar *field, *field2;
     guint length;
 
     int open_flags = OPEN_TAGS;
 
+    if (!filename || !FileTag)
+        return FALSE;
+
     wpc = WavpackOpenFileInput(filename, NULL, open_flags, 0);
 
     if ( wpc == NULL ) {
@@ -271,9 +271,6 @@ gboolean Wavpack_Tag_Read_File_Tag (gchar *filename, File_Tag *FileTag)
 
 gboolean Wavpack_Tag_Write_File_Tag (ET_File *ETFile)
 {
-    if (!ETFile || !ETFile->FileTag)
-        return FALSE;
-
     WavpackContext *wpc;
 
     gchar    *filename = ((File_Name *)((GList *)ETFile->FileNameCur)->data)->value;
@@ -282,6 +279,9 @@ gboolean Wavpack_Tag_Write_File_Tag (ET_File *ETFile)
 
     int open_flags = OPEN_EDIT_TAGS;
 
+    if (!ETFile || !ETFile->FileTag)
+        return FALSE;
+
     wpc = WavpackOpenFileInput(filename, NULL, open_flags, 0);
 
     if ( wpc == NULL ) {
--
cgit v0.9.1
