diff -ur breeze-icons-5.16.0.orig/icons/CMakeLists.txt breeze-icons-5.16.0/icons/CMakeLists.txt
--- breeze-icons-5.16.0.orig/icons/CMakeLists.txt	2015-11-08 12:58:09.000000000 +0100
+++ breeze-icons-5.16.0/icons/CMakeLists.txt	2015-11-22 10:37:17.223501606 +0100
@@ -11,7 +11,7 @@
 
 ########### install files ###############
 
-set( breeze_icon_dirs  actions apps categories devices emblems emotes mimetypes places status )
+set( breeze_icon_dirs  actions applets apps categories devices emblems emotes mimetypes places status )
 
 
 set(BREEZE_INSTALL_DIR ${ICON_INSTALL_DIR}/breeze)
diff -ur breeze-icons-5.16.0.orig/icons-dark/CMakeLists.txt breeze-icons-5.16.0/icons-dark/CMakeLists.txt
--- breeze-icons-5.16.0.orig/icons-dark/CMakeLists.txt	2015-11-08 12:58:09.000000000 +0100
+++ breeze-icons-5.16.0/icons-dark/CMakeLists.txt	2015-11-22 10:37:34.356830327 +0100
@@ -11,7 +11,7 @@
 
 ########### install files ###############
 
-set( breeze_icon_dark_dirs  actions apps categories devices emblems emotes mimetypes places status)
+set( breeze_icon_dark_dirs  actions applets apps categories devices emblems emotes mimetypes places status)
 
 
 set(BREEZE_INSTALL_DIR ${ICON_INSTALL_DIR}/breeze-dark)
