From b653d6ffe9ec25c203fc0cbbbdd72e9b42546283 Mon Sep 17 00:00:00 2001
From: Thomas Dziedzic <gostrc@gmail.com>
Date: Sat, 10 Aug 2013 10:50:16 -0700
Subject: [PATCH] remove db support

---
 ext/dbm/extconf.rb | 8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

diff --git a/ext/dbm/extconf.rb b/ext/dbm/extconf.rb
index 32a9446..2412404 100644
--- a/ext/dbm/extconf.rb
+++ b/ext/dbm/extconf.rb
@@ -5,17 +5,11 @@ dir_config("dbm")
 if dblib = with_config("dbm-type", nil)
   dblib = dblib.split(/[ ,]+/)
 else
-  dblib = %w(libc db db2 db1 db5 db4 db3 gdbm_compat gdbm qdbm)
+  dblib = %w(libc gdbm_compat gdbm qdbm)
 end
 
 headers = {
   "libc" => ["ndbm.h"], # 4.3BSD original ndbm, Berkeley DB 1 in 4.4BSD libc.
-  "db" => ["db.h"],
-  "db1" => ["db1/ndbm.h", "db1.h", "ndbm.h"],
-  "db2" => ["db2/db.h", "db2.h", "db.h"],
-  "db3" => ["db3/db.h", "db3.h", "db.h"],
-  "db4" => ["db4/db.h", "db4.h", "db.h"],
-  "db5" => ["db5/db.h", "db5.h", "db.h"],
   "gdbm_compat" => ["gdbm-ndbm.h", "gdbm/ndbm.h", "ndbm.h"], # GDBM since 1.8.1
   "gdbm" => ["gdbm-ndbm.h", "gdbm/ndbm.h", "ndbm.h"], # GDBM until 1.8.0
   "qdbm" => ["qdbm/relic.h", "relic.h"],
-- 
1.8.3.4

