diff -ur flac-1.1.2-orig/configure flac-1.1.2/configure
--- flac-1.1.2-orig/configure	2005-07-24 17:58:19.180907080 +0100
+++ flac-1.1.2/configure	2005-07-24 17:58:59.139832400 +0100
@@ -3462,7 +3462,7 @@
 # This must be Linux ELF.
 linux-gnu*)
   case $host_cpu in
-  alpha* | hppa* | i*86 | powerpc* | sparc* | ia64* | s390* )
+  alpha* | hppa* | i*86 | powerpc* | sparc* | ia64* | s390* | x86_64* )
     lt_cv_deplibs_check_method=pass_all ;;
   *)
     # glibc up to 2.1.1 does not perform some relocations on ARM
