diff -u -r apricots-0.2.6/apricots/apricots.cpp apricots-0.2.6-freealut/apricots/apricots.cpp
--- apricots-0.2.6/apricots/apricots.cpp	2003-08-05 22:01:16.000000000 +0000
+++ apricots-0.2.6-freealut/apricots/apricots.cpp	2019-09-21 11:16:31.723765376 +0000
@@ -16,6 +16,7 @@
 int main(int, char**){
 #endif
 
+atexit(SDL_Quit);
   // Initialize data
   gamedata g;
   init_data(g);
diff -u -r apricots-0.2.6/apricots/sampleio.cpp apricots-0.2.6-freealut/apricots/sampleio.cpp
--- apricots-0.2.6/apricots/sampleio.cpp	2003-08-05 23:21:22.000000000 +0000
+++ apricots-0.2.6-freealut/apricots/sampleio.cpp	2019-09-21 11:17:03.164207665 +0000
@@ -13,6 +13,9 @@
 //   Code cleanup
 
 #include "sampleio.h"
+#include <AL/al.h>
+#include <AL/alc.h>
+#include <AL/alut.h>
 
 // Empty Noaudio sound routines
 #if AP_AUDIO==AUDIO_NOAUDIO
@@ -74,12 +77,9 @@
   alListenerfv(AL_ORIENTATION, front );
 
   // Load in samples
-  ALvoid* data = malloc(5 * (512 * 3) * 1024);
   alGenBuffers(numsamples, samples);
 
   for (int i = 0; i < numsamples; i++){
-    ALsizei freq;
-    ALboolean fileok;
     // Evil OpenAL portability fix done here
 #ifdef _WIN32
     ALenum format;
@@ -87,15 +87,12 @@
     alutLoadWAVFile(filenames[i],&format,&data,&filelen,&freq,&trash);
     fileok = (alGetError() == AL_NO_ERROR);
 #else
-    ALsizei format;
-    ALsizei trash;
-    fileok = alutLoadWAV(filenames[i],&data,&format,&filelen,&trash,&freq);
+	samples[i] = alutCreateBufferFromFile(filenames[i]);
 #endif
-    if (!fileok){
+    if (samples[i] == AL_NONE){
       cerr << "sampleio: could not open " << filenames[i] << endl;
       exit(1);
     }         
-    alBufferData(samples[i], format, data, filelen, freq);
   }
 
   // Generate Sources
@@ -107,7 +104,6 @@
     alSourcefv(sources[j], AL_ORIENTATION, back );
   }
   
-  free(data);
   
 }
 
diff -u -r apricots-0.2.6/configure.in apricots-0.2.6-freealut/configure.in
--- apricots-0.2.6/configure.in	2003-08-13 01:11:25.000000000 +0000
+++ apricots-0.2.6-freealut/configure.in	2019-09-21 11:16:31.723765376 +0000
@@ -100,7 +100,8 @@
 
 dnl Check for OpenAL
 AC_CHECK_HEADER(AL/al.h,[CXXFLAGS="$CXXFLAGS -DAP_AUDIO_OPENAL"])
-AC_CHECK_LIB( openal, alutInit)
+AC_CHECK_LIB( openal, alGetError)
+AC_CHECK_LIB( alut, alutInit)
 
 KDE_CHECK_EXTRA_LIBS
 all_libraries="$all_libraries $USER_LDFLAGS"
