--- branches/KDE/4.5/kdebase/workspace/kstyles/oxygen/oxygenstyle.cpp	2010/10/02 17:21:35	1181905
+++ branches/KDE/4.5/kdebase/workspace/kstyles/oxygen/oxygenstyle.cpp	2010/10/02 17:23:29	1181906
@@ -1450,7 +1450,9 @@
             case ProgressBar::BusyIndicator:
             {
 
-                if( r.width() < 4 || r.height() < 4 ) return true;
+                if( orientation == Qt::Horizontal && r.width() < 4 ) return true;
+                else if( orientation == Qt::Vertical && r.height() < 4 ) return true;
+
                 QPixmap pixmap( _helper.progressBarIndicator( pal, rect ) );
                 p->drawPixmap( rect.adjusted(-1, -2, 0, 0).topLeft(), pixmap );
                 return true;
