Index: kdm/config.def
===================================================================
--- kdm/config.def	(revision 1216141)
+++ kdm/config.def	(revision 1216142)
@@ -2680,7 +2680,7 @@
 Type: string
 Default: ""
 User: greeter
-Instance: */KDMDATA "/themes/oxygen-air"
+Instance: */KDMDATA "/themes/horos"
 Comment: &
 Description:
  The theme to use for the greeter. Can point to either a directory or an XML
