--- kdebase-workspace-4.6.2/kdm/config.def~	2011-04-08 00:29:48.291941154 -0700
+++ kdebase-workspace-4.6.2/kdm/config.def	2011-04-08 00:32:04.738607619 -0700
@@ -38,7 +38,7 @@
 # define HALT_CMD	"/usr/sbin/shutdown -h now"
 # define REBOOT_CMD	"/usr/sbin/shutdown -r now"
 #elif defined(BSD) || defined(__linux__)
-# define HALT_CMD	"/sbin/shutdown -p now"
+# define HALT_CMD	"/sbin/shutdown -h now"
 # define REBOOT_CMD	"/sbin/shutdown -r now"
 #elif defined(__SVR4)
 # define HALT_CMD	"/usr/sbin/halt"
