--- lyx-2.2.3/src/EnchantChecker.cpp.orig	2017-11-23 11:19:54.636524228 +0000
+++ lyx-2.2.3/src/EnchantChecker.cpp	2017-11-23 11:21:06.583432311 +0000
@@ -68,12 +68,12 @@
 
 enchant::Dict * EnchantChecker::Private::addSpeller(string const & lang)
 {
-	enchant::Broker * instance = enchant::Broker::instance();
+	enchant::Broker instance;
 	Speller m;
 
 	try {
 		LYXERR(Debug::FILES, "request enchant speller for language " << lang);
-		m.speller = instance->request_dict(lang);
+		m.speller = instance.request_dict(lang);
 	}
 	catch (enchant::Exception & e) {
 		// FIXME error handling?
@@ -186,8 +186,8 @@
 {
 	if (!lang)
 		return false;
-	enchant::Broker * instance = enchant::Broker::instance();
-	return (instance->dict_exists(lang->code()));
+	enchant::Broker instance;
+	return (instance.dict_exists(lang->code()));
 }
 
 
