--- script.c.old	2006-09-30 11:12:32.000000000 -0400
+++ script.c	2006-09-30 11:12:38.000000000 -0400
@@ -262,17 +262,23 @@
     /* add only new files in the global dir */
     if((dir = opendir(global_dir))) {
         while((ent = readdir(dir))) {
-	    if(ent->d_name[0] != '.') {
-		for(l = filenames; l; l = g_list_next(l)) {
-		    if(strcmp((char *) l->data, ent->d_name) == 0)
-			goto dont_add;
-		}
-		files = g_list_append(files,
-				      g_strconcat(global_dir, "/",
-						  ent->d_name, NULL));
-	    dont_add:
-	    }
-	}
+            if(ent->d_name[0] != '.') {
+                char dont_add = 0;
+                for(l = filenames; l; l = g_list_next(l)) {
+                    if(strcmp((char *) l->data, ent->d_name) == 0)
+                    {
+                        dont_add = 1;
+                        break;
+                    }
+                }
+                if (!dont_add)
+                {
+                    files = g_list_append(files,
+                                      g_strconcat(global_dir, "/",
+                                                  ent->d_name, NULL));
+                }
+            }
+        }
         closedir(dir);
     }
     /* clean up */
