Index: muttlib.c
===================================================================
--- muttlib.c	(revision 6128:eff500f87889)
+++ muttlib.c	(revision 6161:1a4c43138685)
@@ -1961,5 +1961,6 @@
   char *p = safe_strdup (src);
   int rc = mutt_convert_string (&p, Charset, "utf-8", 0);
-  strfcpy (dest, rc == 0 ? p : src, dlen);
+  /* `src' may be NULL, such as when called from the pop3 driver. */
+  strfcpy (dest, (rc == 0) ? NONULL(p) : NONULL(src), dlen);
   FREE (&p);
 }
