--- misc/build/cppunit-1.12.1/src/cppunit/Makefile.am	2008-02-20 06:36:38.000000000 +0100
+++ misc/build/cppunit-1.12.1/src/cppunit/Makefile.am.new	2010-06-19 16:48:11.181542138 +0200
@@ -63,5 +63,11 @@

 libcppunit_la_LDFLAGS= \
  -no-undefined -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) \
- -release $(LT_RELEASE)
+ -release $(LT_RELEASE) @LIBADD_DL@

+TESTS = t_link
+
+check_PROGRAMS = t_link
+
+t_link_SOURCES = t_link.cpp
+t_link_LDADD = libcppunit.la
