Index: source/plugin/base/makefile.mk
===================================================================
RCS file: /cvs/util/extensions/source/plugin/base/makefile.mk,v
retrieving revision 1.14.64.1
retrieving revision 1.14.64.1.2.1
diff -u -r1.14.64.1 -r1.14.64.1.2.1
--- extensions/source/plugin/base/makefile.mk	19 Nov 2008 15:01:51 -0000	1.14.64.1
+++ extensions/source/plugin/base/makefile.mk	9 Dec 2008 16:08:38 -0000	1.14.64.1.2.1
@@ -55,11 +55,6 @@
 CFLAGSCXX+=$(OBJCXXFLAGS)
 .ENDIF  # "$(GUIBASE)"=="aqua"
 
-.IF "$(GUIBASE)" == "unx" && "$(ENABLE_GTK)" == "TRUE"
-PKGCONFIG_MODULES=gtk+-2.0
-.INCLUDE : pkg_config.mk
-.ENDIF
-
 SLOFILES=		\
 				$(SLO)$/plctrl.obj		\
 				$(SLO)$/service.obj		\
Index: source/plugin/unx/makefile.mk
===================================================================
RCS file: /cvs/util/extensions/source/plugin/unx/makefile.mk,v
retrieving revision 1.15.64.1
retrieving revision 1.15.64.1.2.1
diff -u -r1.15.64.1 -r1.15.64.1.2.1
--- extensions/source/plugin/unx/makefile.mk	19 Nov 2008 15:02:32 -0000	1.15.64.1
+++ extensions/source/plugin/unx/makefile.mk	9 Dec 2008 16:08:38 -0000	1.15.64.1.2.1
@@ -58,12 +58,6 @@
 CDEFS+=-DDISABLE_XAW
 .ENDIF
 
-.IF "$(ENABLE_GTK)" == "TRUE"
-CDEFS+=-DENABLE_GTK
-PKGCONFIG_MODULES=gtk+-2.0 gthread-2.0
-.INCLUDE : pkg_config.mk
-.ENDIF
-
 SLOFILES=\
 	$(SLO)$/nppapi.obj		\
 	$(SLO)$/sysplug.obj		\

