--- connectivity/source/drivers/mozab/bootstrap/makefile.mk	2008-12-08 11:18:53.481875000 +0100
+++ connectivity/source/drivers/mozab/bootstrap/makefile.mk	2008-12-08 11:33:38.028750000 +0100
@@ -40,13 +40,6 @@
 PRJNAME=connectivity
 TARGET=mozbootstrap
 
-.IF ( "$(SYSTEM_MOZILLA)" == "YES" && "$(WITH_MOZILLA)" == "YES") || "$(WITH_MOZILLA)" == "NO"
-all: 
-	@echo "		Not building the mozabsrc stuff in OpenOffice.org build"
-	@echo "		dependency to Mozilla developer snapshots not feasable at the moment"
-	@echo "         see http://bugzilla.mozilla.org/show_bug.cgi?id=135137"
-.ENDIF
-
 USE_DEFFILE=TRUE
 ENABLE_EXCEPTIONS=TRUE
 
--- scp2/source/ooo/file_library_ooo.scp     (Revision 264861)
+++ scp2/source/ooo/file_library_ooo.scp     (Arbeitskopie)
@@ -940,8 +940,6 @@
 
 #else
 
-#ifndef WITHOUT_MOZILLA
-
 File gid_File_Lib_Mozbootstrap
     TXT_FILE_BODY;
     Styles = (PACKED,UNO_COMPONENT,PATCH);
@@ -956,8 +954,6 @@
 
 #endif
 
-#endif
-
 STD_UNO_LIB_FILE(gid_File_Lib_Wpft,wpft)
 
 STD_UNO_LIB_FILE(gid_File_Lib_Msworks,msworks)
