diff -urN ooo_DEV300_m77_src.orig/postprocess/checkdeliver/checkdeliver.pl ooo_DEV300_m77_src/postprocess/checkdeliver/checkdeliver.pl
--- ooo_DEV300_m77_src.orig/postprocess/checkdeliver/checkdeliver.pl	2010-04-22 18:05:56.000000000 +0200
+++ ooo_DEV300_m77_src/postprocess/checkdeliver/checkdeliver.pl	2010-04-22 18:06:13.000000000 +0200
@@ -48,7 +48,7 @@
 my $platform         = '';
 my $milestoneext     = '';
 my $local_env        = 0;
-my $source_config    = SourceConfig -> new();
+my $source_config    = SourceConfig -> new( $ENV{SOLARSRC} );
 my @exceptionmodlist = (
                         "postprocess",
                         "instset.*native",
