*** soffice.orig    2007-10-27 16:40:34.000000000 +0200
--- soffice    2007-10-27 16:41:14.000000000 +0200
***************
*** 34,39 ****
--- 34,44 ----
  #
  #*************************************************************************
  
+ if [ -z $SAL_USE_VCLPLUGIN ]; then
+         export SAL_USE_VCLPLUGIN=gtk
+ fi
+ # export SAL_ENABLE_NATIVE_XFONTS=0
+ 
  #
  # STAR_PROFILE_LOCKING_DISABLED=1
  # export STAR_PROFILE_LOCKING_DISABLED
