--- opera.orig	2008-10-22 19:37:21.000000000 +0200
+++ opera	2008-10-22 19:38:13.000000000 +0200
@@ -30,7 +30,7 @@
 done
 test -z "$toset" || die Missing parameter for $arg
 
-test "${_QTSTYLE_FROM_CMDLINE_}" = 'NO' && test -f "$HOME/.qt/qtrc" && _OPERA_QTSTYLE=`sed -n 's/^style=//p' $HOME/.qt/qtrc` && test "$_OPERA_QTSTYLE" && set -- "$@" -style "$_OPERA_QTSTYLE"
+test "${_QTSTYLE_FROM_CMDLINE_}" = 'NO' && test -f "$HOME/.config/Trolltech.conf" && _OPERA_QTSTYLE=`sed -n 's/^style=//p' $HOME/.config/Trolltech.conf` && test "$_OPERA_QTSTYLE" && set -- "$@" -style "$_OPERA_QTSTYLE"
 
 
 # Add ending path divider "/" if missing, so execution won't fail
