--- NetworkManager-0.7.999/src/NetworkManagerPolicy.c.orig	2010-01-30 19:43:15.000000000 +0100
+++ NetworkManager-0.7.999/src/NetworkManagerPolicy.c	2010-01-30 19:45:02.000000000 +0100
@@ -431,7 +431,7 @@
 		g_free (configured_hostname);
 		return;
 	}
-
+#if 0
 	/* Try automatically determined hostname from the best device's IP config */
 	if (!best)
 		best = get_best_device (policy->manager, &best_req);
@@ -475,6 +475,7 @@
 		/* Fall back to 'localhost.localdomain' */
 		set_system_hostname (NULL, "error starting hostname thread");
 	}
+#endif
 }
 
 static void
