diff -up gtk+-2.17.11/po/Makefile.in.in.fix-install gtk+-2.17.11/po/Makefile.in.in
--- gtk+-2.17.11/po/Makefile.in.in.fix-install	2009-09-05 01:19:45.303824009 -0400
+++ gtk+-2.17.11/po/Makefile.in.in	2009-09-05 01:19:14.632573806 -0400
@@ -126,7 +126,7 @@ install-data: install-data-@USE_NLS@
 install-data-no: all
 install-data-yes: all
 	$(mkdir_p) $(DESTDIR)$(datadir); \
-	@catalogs='$(CATALOGS)'; \
+	catalogs='$(CATALOGS)'; \
 	for cat in $$catalogs; do \
 	  cat=`basename $$cat`; \
 	  case "$$cat" in \
diff -up gtk+-2.17.11/po-properties/Makefile.in.in.fix-install gtk+-2.17.11/po-properties/Makefile.in.in
--- gtk+-2.17.11/po-properties/Makefile.in.in.fix-install	2009-09-05 01:19:54.543583934 -0400
+++ gtk+-2.17.11/po-properties/Makefile.in.in	2009-09-05 01:20:16.427830666 -0400
@@ -107,7 +107,7 @@ install-data: install-data-@USE_NLS@
 install-data-no: all
 install-data-yes: all
 	$(mkdir_p) $(DESTDIR)$(datadir); \
-	@catalogs='$(CATALOGS)'; \
+	catalogs='$(CATALOGS)'; \
 	for cat in $$catalogs; do \
 	  cat=`basename $$cat`; \
 	  case "$$cat" in \
