diff -u -r alsa-plugins-1.0.25/a52/pcm_a52.c alsa-plugins-1.0.25-ffmpeg/a52/pcm_a52.c
--- alsa-plugins-1.0.25/a52/pcm_a52.c	2012-01-25 08:57:07.000000000 +0100
+++ alsa-plugins-1.0.25-ffmpeg/a52/pcm_a52.c	2012-06-09 00:42:52.177219012 +0200
@@ -702,7 +702,6 @@
 	rec->channels = channels;
 	rec->format = format;
 
-	avcodec_init();
 	avcodec_register_all();
 
 	rec->codec = avcodec_find_encoder_by_name("ac3_fixed");
