diff --git a/com/aelitis/azureus/core/update/impl/AzureusRestarterImpl.java b/com/aelitis/azureus/core/update/impl/AzureusRestarterImpl.java
index 735cbf7..ce8108f 100644
--- a/com/aelitis/azureus/core/update/impl/AzureusRestarterImpl.java
+++ b/com/aelitis/azureus/core/update/impl/AzureusRestarterImpl.java
@@ -29,8 +29,6 @@ import org.gudy.azureus2.core3.logging.*;
 import org.gudy.azureus2.core3.util.*;
 import org.gudy.azureus2.platform.PlatformManager;
 import org.gudy.azureus2.platform.PlatformManagerFactory;
-import org.gudy.azureus2.platform.win32.access.AEWin32Access;
-import org.gudy.azureus2.platform.win32.access.AEWin32Manager;
 
 import com.aelitis.azureus.core.AzureusCore;
 import com.aelitis.azureus.core.AzureusCoreFactory;
@@ -212,27 +210,6 @@ AzureusRestarterImpl
 	}
 	  
 	
-	private boolean
-	win32NativeRestart(
-		PrintWriter	log,
-		String		exec )
-	{
-	    try{
-	    		// we need to spawn without inheriting handles
-	    	
-	    	PlatformManager pm = PlatformManagerFactory.getPlatformManager();
-	    	
-	    	pm.createProcess( exec, false );
-	    
-	    	return( true );
-	    	
-	    }catch(Throwable e) {
-	        e.printStackTrace(log);
-	        
-	        return( false );
-	    }
-	}
-	
 
 	private String getExeUpdater(PrintWriter log) {
 		try {
@@ -294,7 +271,7 @@ AzureusRestarterImpl
 
 		return null;
 	}
-
+/*
   private boolean restartViaEXE(PrintWriter log,
   		String exeUpdater,
       String[]  properties,
@@ -353,7 +330,7 @@ AzureusRestarterImpl
 							SystemProperties.getApplicationPath(), AEWin32Access.SW_NORMAL);
 				}
 			}
-
+*/
 			/*
 			 * Some results:
 			 * 0: OOM
@@ -370,7 +347,6 @@ AzureusRestarterImpl
 			 * 31: No Association
 			 * 32: DLL Not found
 			 * >32: OK!
-			 */
 			log.println("   -> " + result);
 
 			if (result <= 32) {
@@ -430,7 +406,7 @@ AzureusRestarterImpl
 		return true;
 	}
   
-
+*/
   // ****************** This code is copied into Restarter / Updater so make changes there too !!!
   
   
@@ -444,20 +420,9 @@ AzureusRestarterImpl
     String[]  parameters,
     boolean update_only) 
   {
-    if(Constants.isOSX){
-    	
-    	restartAzureus_OSX(log,mainClass,properties,parameters);
-    	
-    }else if( Constants.isUnix ){
-    	
     	restartAzureus_Unix(log,mainClass,properties,parameters);
-      
-    }else{
-    	
-    	restartAzureus_win32(log,mainClass,properties,parameters,update_only);
-    }
   }
-  
+/*  
   private void 
   restartAzureus_win32(
       PrintWriter log,
@@ -496,7 +461,7 @@ AzureusRestarterImpl
 		}
 	}
   
-
+*/
 	private boolean
 	javaSpawn(
 		PrintWriter log, 
@@ -520,30 +485,6 @@ AzureusRestarterImpl
 		}
 	}
 
-	private void 
-  restartAzureus_OSX(
-      PrintWriter log,
-    String mainClass,
-    String[]  properties,
-    String[] parameters) 
-  {
-
-     String exec = "\"" + JAVA_EXEC_DIR + "java\" " + getClassPath() + getLibraryPath();
-  	 
-     for (int i=0;i<properties.length;i++){
-    	 exec += properties[i] + " ";
-     }
-    
-     exec += mainClass ;
-    
-     for(int i = 0 ; i < parameters.length ; i++) {
-    	 exec += " \"" + parameters[i] + "\"";
-     }
-
-     runExternalCommandViaUnixShell( log, exec );
-  }
-  
-  
   
   private void 
   restartAzureus_Unix(
