--- trunk/app.cpp	2012/02/21 23:57:28	211
+++ trunk/app.cpp	2012/04/16 23:28:34	212
@@ -332,6 +332,15 @@
             Login();
         }
     }
+
+    // Set NumLock
+    string numlock = cfg->getOption("numlock");
+    if (numlock == "on") {
+        NumLock::setOn(Dpy);
+    } else if (numlock == "off") {
+        NumLock::setOff(Dpy);
+    }
+    
     // Start looping
     int panelclosed = 1;
     Panel::ActionType Action;
@@ -971,13 +980,6 @@
         break;
     }
 
-    string numlock = cfg->getOption("numlock");
-    if (numlock == "on") {
-        NumLock::setOn(Dpy);
-    } else if (numlock == "off") {
-        NumLock::setOff(Dpy);
-    }
-    
     delete args;
 
     serverStarted = true;
--- trunk/numlock.cpp	2012/02/21 23:57:28	211
+++ trunk/numlock.cpp	2012/04/16 23:28:34	212
@@ -68,9 +68,9 @@
         return;
 
     if( flag == true )
-        XkbLockModifiers ( dpy, XkbUseCoreKbd, mask, 0);
-    else
         XkbLockModifiers ( dpy, XkbUseCoreKbd, mask, mask);
+    else
+        XkbLockModifiers ( dpy, XkbUseCoreKbd, mask, 0);
 }
 
 void NumLock::setOn(Display *dpy) {
