Index: /trunk/program/src/rrd_format.h
===================================================================
--- /trunk/program/src/rrd_format.h	(revision 2042)
+++ /trunk/program/src/rrd_format.h	(revision 2069)
@@ -36,5 +36,5 @@
 #define RRD_VERSION   "0004"
 #define RRD_VERSION3  "0003"
-#define FLOAT_COOKIE  8.642135E130
+#define FLOAT_COOKIE  ((double)8.642135E130)
 
 typedef union unival {
