diff --git a/Make.defaults b/Make.defaults
index 38da180..aa1f4de 100644
--- a/Make.defaults
+++ b/Make.defaults
@@ -54,9 +54,9 @@ GCCNEWENOUGH := $(shell ([ $(GCCVERSION) -gt "4" ] || ([ $(GCCVERSION) -eq "4" ]
 
 CPPFLAGS   = -DCONFIG_$(ARCH)
 ifeq ($(GCCNEWENOUGH),1)
-CPPFLAGS   += -DGNU_EFI_USE_MS_ABI -maccumulate-outgoing-args
+CPPFLAGS   += -maccumulate-outgoing-args
 endif
-CFLAGS	   = $(ARCH3264) -O2 -fpic -Wall -fshort-wchar -fno-strict-aliasing -fno-merge-constants
+CFLAGS	   = $(ARCH3264) -O2 -fpic -Wall -fshort-wchar -fno-strict-aliasing -fno-stack-protector -fno-merge-constants
 ASFLAGS    = $(ARCH3264)
 LDFLAGS	   = -nostdlib
 INSTALL	   = install
