diff -Naur old/doc/screen.texinfo new/doc/screen.texinfo
--- old/doc/screen.texinfo	2017-02-25 05:35:35.128808129 -1000
+++ new/doc/screen.texinfo	2017-03-01 23:22:32.078134723 -1000
@@ -5796,8 +5796,8 @@
 ============
 
 @example
-     Thomas Renninger <treen@suse.com>,
-     Axel Beckert <abe@deuxchevaux.org>,
+     Thomas Renninger <treen@@suse.com>,
+     Axel Beckert <abe@@deuxchevaux.org>,
      Ken Beal <kbeal@@amber.ssd.csd.harris.com>,
      Rudolf Koenig <rfkoenig@@informatik.uni-erlangen.de>,
      Toerless Eckert <eckert@@informatik.uni-erlangen.de>,
