diff --git a/src/dbus/mpris2/MediaPlayer2.cpp b/src/dbus/mpris2/MediaPlayer2.cpp
index cbe1a2a..a1d1bda 100644
--- a/src/dbus/mpris2/MediaPlayer2.cpp
+++ b/src/dbus/mpris2/MediaPlayer2.cpp
@@ -94,11 +94,7 @@ QString MediaPlayer2::Identity() const
 
 QString MediaPlayer2::DesktopEntry() const
 {
-    // Amarok desktop file is installed in $prefix/share/applications/kde4/
-    // rather than in $prefix/share/applications. The standard way to
-    // represent this dir is with a "kde4-" prefix. See:
-    // http://standards.freedesktop.org/menu-spec/1.0/go01.html#term-desktop-file-id
-    return QLatin1String("kde4-amarok");
+    return QLatin1String("amarok");
 }
 
 QStringList MediaPlayer2::SupportedUriSchemes() const

