diff -Nur claws-mail-extra-plugins-3.7.6-orig/notification_plugin-0.25/src/notification_popup.c claws-mail-extra-plugins-3.7.6/notification_plugin-0.25/src/notification_popup.c
--- claws-mail-extra-plugins-3.7.6-orig/notification_plugin-0.25/src/notification_popup.c	2009-04-26 03:05:05.000000000 +0200
+++ claws-mail-extra-plugins-3.7.6/notification_plugin-0.25/src/notification_popup.c	2010-11-12 13:39:02.363909583 +0100
@@ -339,7 +339,7 @@
     break;
   }
 
-  ppopup->notification = notify_notification_new(summary, utf8_str, NULL, NULL);
+  ppopup->notification = notify_notification_new(summary, utf8_str, NULL);
   g_free(utf8_str);
   if(ppopup->notification == NULL) {
     debug_print("Notification Plugin: Failed to create a new "
diff -Nur claws-mail-extra-plugins-3.7.6-orig/notification_plugin-0.25/src/notification_trayicon.c claws-mail-extra-plugins-3.7.6/notification_plugin-0.25/src/notification_trayicon.c
--- claws-mail-extra-plugins-3.7.6-orig/notification_plugin-0.25/src/notification_trayicon.c	2010-03-25 09:56:34.000000000 +0100
+++ claws-mail-extra-plugins-3.7.6/notification_plugin-0.25/src/notification_trayicon.c	2010-11-12 13:38:30.809340907 +0100
@@ -589,8 +589,7 @@
   summary  = notification_trayicon_popup_assemble_summary();
   utf8_str = notification_trayicon_popup_assemble_body(msginfo);
 
-  popup.notification = notify_notification_new(summary, utf8_str, NULL, NULL);
-  notify_notification_attach_to_status_icon(popup.notification, trayicon);
+  popup.notification = notify_notification_new(summary, utf8_str, NULL);
 
   g_free(summary);
   g_free(utf8_str);
