--- cups-filters/ppd/testppd.c	2020-11-16 22:40:25.000000000 +0100
+++ cups-filters/ppd/testppd.c.new	2020-11-16 22:44:11.495836665 +0100
@@ -455,7 +455,7 @@
 
     fputs("ppdOpenFile(test.ppd): ", stdout);
 
-    if ((ppd = ppdOpenFileWithLocalization("test.ppd", PPD_LOCALIZATION_ALL)) != NULL)
+    if ((ppd = ppdOpenFileWithLocalization("ppd/test.ppd", PPD_LOCALIZATION_ALL)) != NULL)
       puts("PASS");
     else
     {
--- cups-filters/ppd/testppd.c	2020-11-16 22:45:54.000000000 +0100
+++ cups-filters/ppd/testppd.c.new	2020-11-16 22:54:32.592990707 +0100
@@ -911,9 +911,9 @@
     * Test new constraints...
     */
 
-    fputs("ppdOpenFile(test2.ppd): ", stdout);
+    fputs("ppdOpenFile(ppd/test2.ppd): ", stdout);
 
-    if ((ppd = ppdOpenFile("test2.ppd")) != NULL)
+    if ((ppd = ppdOpenFile("ppd/test2.ppd")) != NULL)
       puts("PASS");
     else
     {
