Index: kdepim/korganizer/views/agendaview/koagendaview.cpp
===================================================================
--- kdepim/korganizer/views/agendaview/koagendaview.cpp	(Revision 933656)
+++ kdepim/korganizer/views/agendaview/koagendaview.cpp	(Revision 933657)
@@ -1544,8 +1544,12 @@
   KConfigGroup group = config->group( "Views" );
 
   QList<int> sizes = group.readEntry( "Separator AgendaView", QList<int>() );
-  mSplitterAgenda->setSizes( sizes );
 
+  // the size depends on the number of plugins used
+  if ( sizes.count() >= 2 ) {
+    mSplitterAgenda->setSizes( sizes );
+  }
+
   updateConfig();
 }
 
