--- branches/KDE/4.4/kdepim/korganizer/urihandler.cpp	2010/01/06 22:46:33	1070858
+++ branches/KDE/4.4/kdepim/korganizer/urihandler.cpp	2010/05/03 21:56:54	1122444
@@ -78,8 +78,11 @@
       */
       QString iconPath =
         KIconLoader::global()->iconPath( "view-pim-contacts", KIconLoader::SizeSmall );
+#if 0
       QString tmpStr = "kaddressbook --editor-only --uid ";
       tmpStr += KShell::quoteArg( uri.mid( ::qstrlen( KDEPIMPROTOCOL_CONTACT ) ) );
+#endif
+      QString tmpStr = "kaddressbook";
       KRun::runCommand( tmpStr, "KAddressBook", iconPath, 0 );
       return true;
     }
