diff -Naur distribute-0.6.16.ori/distribute.egg-info/entry_points.txt distribute-0.6.16/distribute.egg-info/entry_points.txt
--- distribute-0.6.16.ori/distribute.egg-info/entry_points.txt	2011-04-30 06:11:02.000000000 -0700
+++ distribute-0.6.16/distribute.egg-info/entry_points.txt	2011-05-03 10:35:31.027759028 -0700
@@ -33,6 +33,7 @@
 [console_scripts]
 easy_install = setuptools.command.easy_install:main
 easy_install-2.7 = setuptools.command.easy_install:main
+easy_install-3.2 = setuptools.command.easy_install:main
 
 [setuptools.file_finders]
 svn_cvs = setuptools.command.sdist:_default_revctrl
diff -Naur distribute-0.6.16.ori/distribute_setup.py distribute-0.6.16/distribute_setup.py
--- distribute-0.6.16.ori/distribute_setup.py	2011-03-18 07:32:22.000000000 -0700
+++ distribute-0.6.16/distribute_setup.py	2011-05-03 10:36:38.231094059 -0700
@@ -299,8 +299,7 @@
         log.warn('Could not find the install location')
         return
     pyver = '%s.%s' % (sys.version_info[0], sys.version_info[1])
-    setuptools_file = 'setuptools-%s-py%s.egg-info' % \
-            (SETUPTOOLS_FAKED_VERSION, pyver)
+    setuptools_file = 'setuptools-%s.egg-info' % SETUPTOOLS_FAKED_VERSION
     pkg_info = os.path.join(placeholder, setuptools_file)
     if os.path.exists(pkg_info):
         log.warn('%s already exists', pkg_info)
diff -Naur distribute-0.6.16.ori/setup.py distribute-0.6.16/setup.py
--- distribute-0.6.16.ori/setup.py	2011-03-18 07:32:22.000000000 -0700
+++ distribute-0.6.16/setup.py	2011-05-03 10:36:38.231094059 -0700
@@ -96,6 +96,8 @@
                 f.close()
 
 
+console_scripts = ["easy_install-%s = setuptools.command.easy_install:main" % sys.version[:3]]
+
 # if we are installing Distribute using "python setup.py install"
 # we need to get setuptools out of the way
 def _easy_install_marker():
@@ -182,11 +184,7 @@
             "dependency_links.txt = setuptools.command.egg_info:overwrite_arg",
         ],
 
-        "console_scripts": [
-             "easy_install = setuptools.command.easy_install:main",
-             "easy_install-%s = setuptools.command.easy_install:main"
-                % sys.version[:3]
-        ],
+        "console_scripts": console_scripts,
 
         "setuptools.file_finders":
             ["svn_cvs = setuptools.command.sdist:_default_revctrl"],
