diff -Naur distribute-0.6.32.ori/distribute.egg-info/entry_points.txt distribute-0.6.32/distribute.egg-info/entry_points.txt
--- distribute-0.6.32.ori/distribute.egg-info/entry_points.txt	2012-11-26 12:18:19.000000000 -0500
+++ distribute-0.6.32/distribute.egg-info/entry_points.txt	2012-12-16 16:52:39.351735173 -0500
@@ -11,6 +11,8 @@
 install_egg_info = setuptools.command.install_egg_info:install_egg_info
 alias = setuptools.command.alias:alias
 easy_install = setuptools.command.easy_install:easy_install
+easy_install-2.7 = setuptools.command.easy_install:main
+easy_install-3.2 = setuptools.command.easy_install:main  
 install_scripts = setuptools.command.install_scripts:install_scripts
 bdist_wininst = setuptools.command.bdist_wininst:bdist_wininst
 bdist_egg = setuptools.command.bdist_egg:bdist_egg
diff -Naur distribute-0.6.32.ori/distribute_setup.py distribute-0.6.32/distribute_setup.py
--- distribute-0.6.32.ori/distribute_setup.py	2012-11-26 09:08:58.000000000 -0500
+++ distribute-0.6.32/distribute_setup.py	2012-12-16 16:52:39.351735173 -0500
@@ -314,8 +314,7 @@
         log.warn('Could not find the install location')
         return
     pyver = '%s.%s' % (sys.version_info[0], sys.version_info[1])
-    setuptools_file = 'setuptools-%s-py%s.egg-info' % \
-            (SETUPTOOLS_FAKED_VERSION, pyver)
+    setuptools_file = 'setuptools-%s.egg-info' % SETUPTOOLS_FAKED_VERSION
     pkg_info = os.path.join(placeholder, setuptools_file)
     if os.path.exists(pkg_info):
         log.warn('%s already exists', pkg_info)
diff -Naur distribute-0.6.32.ori/setup.py distribute-0.6.32/setup.py
--- distribute-0.6.32.ori/setup.py	2012-11-26 09:14:07.000000000 -0500
+++ distribute-0.6.32/setup.py	2012-12-16 16:52:39.351735173 -0500
@@ -108,6 +108,7 @@
             finally:
                 f.close()
 
+console_scripts = ["easy_install-%s = setuptools.command.easy_install:main" % sys.version[:3]]
 
 # if we are installing Distribute using "python setup.py install"
 # we need to get setuptools out of the way
