--- src/qp_fswrap.cpp.orig	2006-08-10 18:43:36.000000000 +0200
+++ src/qp_fswrap.cpp	2006-08-10 18:43:58.000000000 +0200
@@ -1002,7 +1002,11 @@
 /*---FAT32 WRAPPER---------------------------------------------------------------*/
 QString QP_FSFat32::_get_label(PedPartition *part) 
 {
-    char buffer[PED_SECTOR_SIZE];
+#ifdef PED_SECTOR_SIZE // PED_SECTOR_SIZE is gone in parted 1.7.x
+    char *buffer=new char[PED_SECTOR_SIZE];
+#else
+    char *buffer=new char[part->disk->dev->sector_size];
+#endif
     char label[12];
     
     if (!QP_FSWrap::read_sector(part, 0, 1, buffer)) 
