diff -Nur evolution-sharp-0.21.1.orig/evolution/evolution-sharp.dll.config.in evolution-sharp-0.21.1/evolution/evolution-sharp.dll.config.in
--- evolution-sharp-0.21.1.orig/evolution/evolution-sharp.dll.config.in	2009-05-26 00:25:06.000000000 -0700
+++ evolution-sharp-0.21.1/evolution/evolution-sharp.dll.config.in	2011-05-04 12:14:58.087467727 -0700
@@ -1,7 +1,7 @@
 <configuration>
-  <dllmap dll="ebook" target="libebook@EDS_VERSION@@LIB_PREFIX@.@EBOOK_SOVERSION@@LIB_SUFFIX@"/>
-  <dllmap dll="ecal" target="libecal@EDS_VERSION@@LIB_PREFIX@.@ECAL_SOVERSION@@LIB_SUFFIX@"/>
-  <dllmap dll="edataserver" target="libedataserver@EDS_VERSION@@LIB_PREFIX@.@EDS_SOVERSION@@LIB_SUFFIX@"/>
-  <dllmap dll="ebackend" target="libebackend@EDS_VERSION@@LIB_PREFIX@.@EBACKEND_SOVERSION@@LIB_SUFFIX@"/>
-  <dllmap dll="edatacal" target="libedata-cal@EDS_VERSION@@LIB_PREFIX@.@EDATACAL_SOVERSION@@LIB_SUFFIX@"/>
+  <dllmap dll="ebook" target="libebook@EDS_VERSION@@LIB_PREFIX@.10@LIB_SUFFIX@"/>
+  <dllmap dll="ecal" target="libecal@EDS_VERSION@@LIB_PREFIX@.8@LIB_SUFFIX@"/>
+  <dllmap dll="edataserver" target="libedataserver@EDS_VERSION@@LIB_PREFIX@.14@LIB_SUFFIX@"/>
+  <dllmap dll="ebackend" target="libebackend@EDS_VERSION@@LIB_PREFIX@.1@LIB_SUFFIX@"/>
+  <dllmap dll="edatacal" target="libedata-cal@EDS_VERSION@@LIB_PREFIX@.11@LIB_SUFFIX@"/>
 </configuration>
