diff -Nur evolution-sharp-0.21.1.orig/evolution/evolution-sharp.dll.config.in evolution-sharp-0.21.1/evolution/evolution-sharp.dll.config.in
--- evolution-sharp-0.21.1.orig/evolution/evolution-sharp.dll.config.in	2009-05-26 10:25:06.000000000 +0300
+++ evolution-sharp-0.21.1/evolution/evolution-sharp.dll.config.in	2010-06-22 01:15:32.451348727 +0300
@@ -1,7 +1,7 @@
 <configuration>
   <dllmap dll="ebook" target="libebook@EDS_VERSION@@LIB_PREFIX@.@EBOOK_SOVERSION@@LIB_SUFFIX@"/>
   <dllmap dll="ecal" target="libecal@EDS_VERSION@@LIB_PREFIX@.@ECAL_SOVERSION@@LIB_SUFFIX@"/>
-  <dllmap dll="edataserver" target="libedataserver@EDS_VERSION@@LIB_PREFIX@.@EDS_SOVERSION@@LIB_SUFFIX@"/>
+  <dllmap dll="edataserver" target="libedataserver@EDS_VERSION@@LIB_PREFIX@.13@LIB_SUFFIX@"/>
   <dllmap dll="ebackend" target="libebackend@EDS_VERSION@@LIB_PREFIX@.@EBACKEND_SOVERSION@@LIB_SUFFIX@"/>
   <dllmap dll="edatacal" target="libedata-cal@EDS_VERSION@@LIB_PREFIX@.@EDATACAL_SOVERSION@@LIB_SUFFIX@"/>
 </configuration>
