--- a/manedit/makefile_append.ini	2008-10-11 16:00:00.000000000 -0700
+++ b/manedit/makefile_append.ini	2009-11-20 15:33:47.000000000 -0800
@@ -39,9 +39,9 @@
 
 manpage:
 	@echo "Generating manual page $(BIN).1.bz2"
-	@-$(SED) 's|%%PREFIX%%|$(PREFIX)|g ; \
-                  s|%%LOCALBASE%%|${LOCALBASE}|g ; \
-                  s|%%X11BASE%%|${X11BASE}|g' $(BIN).1 > $(BIN).1.out
+	@-$(SED) -e 's|%%PREFIX%%|$(PREFIX)|g ;' \
+                 -e 's|%%LOCALBASE%%|${LOCALBASE}|g ;' \
+                 -e 's|%%X11BASE%%|${X11BASE}|g' $(BIN).1 > $(BIN).1.out
 	@-$(BZIP2) $(BIN).1.out --stdout > $(BIN).1.bz2
 
 modules: $(OBJ_C) $(OBJ_CPP)
