diff --git i/tests/layouts/valid-20.layout w/tests/layouts/valid-20.layout
index 8217b28b..6e81507f 100644
--- i/tests/layouts/valid-20.layout
+++ w/tests/layouts/valid-20.layout
@@ -140,23 +140,23 @@
                 "glyph" : 244,
                 "width" : 19456,
                 "x-offset" : 14764,
-                "y-offset" : 3844,
+                "y-offset" : 3845,
                 "is-cluster-start" : true,
                 "log-cluster" : 0
               },
               {
                 "glyph" : 272,
                 "width" : 19456,
                 "x-offset" : 14764,
                 "y-offset" : 4280,
                 "is-cluster-start" : true,
                 "log-cluster" : 1
               },
               {
                 "glyph" : 273,
                 "width" : 19456,
                 "x-offset" : 14764,
-                "y-offset" : 3506,
+                "y-offset" : 3507,
                 "is-cluster-start" : true,
                 "log-cluster" : 2
               }
