diff -up ConsoleKit-0.4.1/data/org.freedesktop.ConsoleKit.service.in.nodaemon ConsoleKit-0.4.1/data/org.freedesktop.ConsoleKit.service.in
--- ConsoleKit-0.4.1/data/org.freedesktop.ConsoleKit.service.in.nodaemon	2009-12-15 10:56:39.839818556 -0500
+++ ConsoleKit-0.4.1/data/org.freedesktop.ConsoleKit.service.in	2009-12-15 10:56:50.177792048 -0500
@@ -1,4 +1,4 @@
 [D-BUS Service]
 Name=org.freedesktop.ConsoleKit
-Exec=@sbindir@/console-kit-daemon
+Exec=@sbindir@/console-kit-daemon --no-daemon
 User=root
