--- kdeadmin-4.6.0/ksystemlog/src/modes/system/systemConfiguration.h~	2011-01-27 14:24:29.840000005 +0100
+++ kdeadmin-4.6.0/ksystemlog/src/modes/system/systemConfiguration.h	2011-01-27 14:24:53.803333335 +0100
@@ -39,7 +39,7 @@
 		SystemConfiguration() :
 			GenericLogModeConfiguration(
 					QLatin1String( SYSTEM_LOG_MODE_ID ),
-					QStringList() << QLatin1String( "/var/log/syslog" ),
+					QStringList() << QLatin1String( "/var/log/messages.log" ),
 					QList<int>() << INFORMATION_LOG_LEVEL_ID
 			) {
 
