From 8c337455533e9a57102bd590dc5bfee111110241 Mon Sep 17 00:00:00 2001
From: Dominique Leuenberger <dimstar@opensuse.org>
Date: Thu, 5 Feb 2015 13:50:55 +0100
Subject: build: detect vala 0.28 branch

https://bugzilla.gnome.org/show_bug.cgi?id=744041

diff --git a/configure.ac b/configure.ac
index d109e47..ec62cf6 100644
--- a/configure.ac
+++ b/configure.ac
@@ -329,7 +329,7 @@ if test "$user_disabled_vala" = 1; then
 else
         AC_MSG_RESULT(no)
 	AC_MSG_CHECKING(if libvala is available)
-	m4_foreach([VERSION], [[0.18], [0.20], [0.22], [0.24], [0.26]],
+	m4_foreach([VERSION], [[0.18], [0.20], [0.22], [0.24], [0.26], [0.28]],
 		[PKG_CHECK_EXISTS([ libvala-VERSION ],
 			 [ valaver="-VERSION" ])
 			 ])
-- 
cgit v0.10.2

