Index: pyside-setup-opensource-src-5.15.2/build_scripts/config.py
===================================================================
--- pyside-setup-opensource-src-5.15.2.orig/build_scripts/config.py
+++ pyside-setup-opensource-src-5.15.2/build_scripts/config.py
@@ -93,6 +93,7 @@ class Config(object):
             'Programming Language :: Python :: 3.7',
             'Programming Language :: Python :: 3.8',
             'Programming Language :: Python :: 3.9',
+            'Programming Language :: Python :: 3.10',
         ]
 
         self.setup_script_dir = None
@@ -135,7 +136,7 @@ class Config(object):
         setup_kwargs['zip_safe'] = False
         setup_kwargs['cmdclass'] = cmd_class_dict
         setup_kwargs['version'] = package_version
-        setup_kwargs['python_requires'] = ">=2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*, !=3.4.*, <3.10"
+        setup_kwargs['python_requires'] = ">=2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*, !=3.4.*, <3.11"
 
         if quiet:
             # Tells distutils / setuptools to be quiet, and only print warnings or errors.
