diff -up qtwebkit-2.2.2-source/Source/WebKit.pri.no_Werror qtwebkit-2.2.2-source/Source/WebKit.pri
--- qtwebkit-2.2.2-source/Source/WebKit.pri.no_Werror	2012-05-22 09:30:37.000000000 -0500
+++ qtwebkit-2.2.2-source/Source/WebKit.pri	2012-05-22 09:51:42.126610618 -0500
@@ -102,7 +102,7 @@ CONFIG -= warn_on
 
 # Treat warnings as errors on x86/Linux/GCC
 linux-g++* {
-    !CONFIG(standalone_package):isEqual(QT_ARCH,x86_64)|isEqual(QT_ARCH,i386): QMAKE_CXXFLAGS += -Werror
+    #!CONFIG(standalone_package):isEqual(QT_ARCH,x86_64)|isEqual(QT_ARCH,i386): QMAKE_CXXFLAGS += -Werror
 
     greaterThan(QT_GCC_MAJOR_VERSION, 3):greaterThan(QT_GCC_MINOR_VERSION, 5) {
         if (!contains(QMAKE_CXXFLAGS, -std=c++0x) && !contains(QMAKE_CXXFLAGS, -std=gnu++0x)) {

diff -up webkit-qtwebkit/Source/common.pri.me webkit-qtwebkit/Source/common.pri
--- webkit-qtwebkit/Source/common.pri.me	2012-01-24 14:51:49.000000000 +0100
+++ webkit-qtwebkit/Source/common.pri	2012-01-24 14:52:01.000000000 +0100
@@ -3,12 +3,12 @@
 contains(JAVASCRIPTCORE_JIT,yes): DEFINES+=ENABLE_JIT=1
 contains(JAVASCRIPTCORE_JIT,no): DEFINES+=ENABLE_JIT=0
 
-linux-g++ {
-isEmpty($$(SBOX_DPKG_INST_ARCH)):exists(/usr/bin/ld.gold) {
-    message(Using gold linker)
-    QMAKE_LFLAGS+=-fuse-ld=gold
-}
-}
+#linux-g++ {
+#isEmpty($$(SBOX_DPKG_INST_ARCH)):exists(/usr/bin/ld.gold) {
+#    message(Using gold linker)
+#    QMAKE_LFLAGS+=-fuse-ld=gold
+#}
+#}
 
 # We use this flag on production branches
 # See https://bugs.webkit.org/show_bug.cgi?id=60824
diff -up webkit-qtwebkit/Source/JavaScriptCore/runtime/JSGlobalObject.h.me webkit-qtwebkit/Source/JavaScriptCore/runtime/JSGlobalObject.h
