--- cmake-2.8.10.2/Modules/FindQt4.cmake~	2013-02-24 12:13:41.172266220 +0000
+++ cmake-2.8.10.2/Modules/FindQt4.cmake	2013-02-24 12:20:11.229349625 +0000
@@ -497,7 +497,7 @@
 # check for qmake
 # Debian uses qmake-qt4
 # macports' Qt uses qmake-mac
-find_program(QT_QMAKE_EXECUTABLE NAMES qmake qmake4 qmake-qt4 qmake-mac
+find_program(QT_QMAKE_EXECUTABLE NAMES qmake4 qmake-qt4 qmake-mac
   PATHS
     ENV QTDIR
     "[HKEY_CURRENT_USER\\Software\\Trolltech\\Versions\\${qt_install_version};InstallDir]"
