--- a/doc/Makefile.am	2011-02-05 17:52:48.000000000 +0100
+++ b/doc/Makefile.am	2012-08-06 14:33:37.000000000 +0200
@@ -48,14 +48,7 @@
 $(srcdir)/vice.txt:	vice.texi
 	$(MAKEINFO) -o $(srcdir)/vice.txt --no-headers $(srcdir)/vice.texi
 
-if BUILD_PDF
-$(srcdir)/vice.pdf:	vice.texi
-	$(TEXI2DVI) -q --clean --pdf -o $(srcdir)/vice.pdf $(srcdir)/vice.texi
-
-PDF_DEPS=$(srcdir)/vice.pdf
-else
 PDF_DEPS=
-endif
 
 if BUILD_AMIGAGUIDE
 $(srcdir)/vice.guide:	vice.texi
@@ -99,4 +92,4 @@
 IPF_DEPS=
 endif
 
-INFO_DEPS = $(srcdir)/vice.txt $(srcdir)/vice.info $(PDF_DEPS) $(GUIDE_DEPS) $(HLP_DEPS) $(CHM_DEPS) $(IPF_DEPS)
+INFO_DEPS = $(srcdir)/vice.info $(PDF_DEPS) $(GUIDE_DEPS) $(HLP_DEPS) $(CHM_DEPS) $(IPF_DEPS)
