--- sonata/info.py	2011-07-05 22:32:08.452556016 -0500
+++ sonata/info.py	2011-07-05 22:45:50.849311466 -0500
@@ -393,9 +393,9 @@
 				if content.startswith("#REDIRECT [["):
 					addr = "http://lyricwiki.org/index.php?title=%s&action=edit" % urllib.quote(content.split("[[")[1].split("]]")[0])
 					content = urllib.urlopen(addr).read()
-				lyrics = content.split("&lt;lyrics&gt;")[1].split("&lt;/lyrics&gt;")[0]
-				if lyrics.strip() != "&lt;!-- PUT LYRICS HERE (and delete this entire line) --&gt;":
-					lyrics = misc.unescape_html(lyrics)
+				content = misc.unescape_html(content)
+				lyrics = content.split("<lyrics>")[1].split("</lyrics>")[0].strip()
+				if lyrics.strip() != "<!-- PUT LYRICS HERE (and delete this entire line) -->":
 					lyrics = misc.wiki_to_html(lyrics)
 					lyrics = lyrics.decode("utf-8")
 # Save lyrics to file:
