From bfd81de1a4871f2b16d8d4d2630d8dcfb5e45491 Mon Sep 17 00:00:00 2001
Message-Id: <bfd81de1a4871f2b16d8d4d2630d8dcfb5e45491.1505481573.git.jan.steffens@gmail.com>
From: "Jan Alexander Steffens (heftig)" <jan.steffens@gmail.com>
Date: Fri, 15 Sep 2017 15:17:34 +0200
Subject: [PATCH 1/2] meson: Remove duplicate and/or unused defines from sendto

---
 sendto/meson.build | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/sendto/meson.build b/sendto/meson.build
index bf4f81652a126047..cc683117a16b91e3 100644
--- a/sendto/meson.build
+++ b/sendto/meson.build
@@ -5,21 +5,14 @@ deps = [
   dependency('gio-2.0')
 ]
 
-cflags = [
-  '-DDATADIR="@0@"'.format(gnomebt_datadir),
-  '-DICONDIR="@0@"'.format(gnomebt_icondir),
-  '-DLOCALEDIR="@0@"'.format(gnomebt_localedir)
-]
-
 executable(
   name,
   'main.c',
   include_directories: [
     top_inc,
     lib_inc
   ],
   dependencies: deps,
-  c_args: cflags,
   link_with: libgnome_bluetooth,
   install: true,
   install_dir: gnomebt_bindir
-- 
2.14.1

