--- applet/notify.c
+++ applet/notify.c
@@ -31,6 +31,10 @@
 #include <libnotify/notify.h>
 #include "notify.h"
 
+#ifndef NOTIFY_CHECK_VERSION
+#define NOTIFY_CHECK_VERSION(x,y,z) 0
+#endif
+
 static GtkStatusIcon *statusicon = NULL;
 static char *icon_name = NULL;
 static char *tooltip = NULL;
@@ -71,7 +75,12 @@
 		notify_notification_close(notify, NULL);
 	}
 
-	notify = notify_notification_new(summary, message, icon_name, NULL);
+	notify = notify_notification_new(summary, message, icon_name
+#if NOTIFY_CHECK_VERSION (0, 7, 0)
+	);
+#else
+	, NULL);
+#endif
 
 	notify_notification_set_timeout(notify, timeout);
 
