diff -Nur kdelibs-3.5.10.orig/kioslave/ftp/ftp.cc kdelibs-3.5.10/kioslave/ftp/ftp.cc
--- kdelibs-3.5.10.orig/kioslave/ftp/ftp.cc	2008-08-19 20:18:12.000000000 +0200
+++ kdelibs-3.5.10/kioslave/ftp/ftp.cc	2009-05-29 20:40:07.618209710 +0200
@@ -876,7 +876,7 @@
   // The usual answer is '227 Entering Passive Mode. (160,39,200,55,6,245)'
   // but anonftpd gives '227 =160,39,200,55,6,245'
   int i[6];
-  char *start = strchr(ftpResponse(3), '(');
+  const char *start = strchr(ftpResponse(3), '(');
   if ( !start )
     start = strchr(ftpResponse(3), '=');
   if ( !start ||
@@ -931,7 +931,7 @@
     return ERR_INTERNAL;
   }
 
-  char *start = strchr(ftpResponse(3), '|');
+  const char *start = strchr(ftpResponse(3), '|');
   if ( !start || sscanf(start, "|||%d|", &portnum) != 1)
     return ERR_INTERNAL;
 
