--- trunk/KDE/kdenetwork/kget/core/kget.cpp	2009/09/14 12:52:57	1023303
+++ trunk/KDE/kdenetwork/kget/core/kget.cpp	2009/09/14 13:01:00	1023304
@@ -234,8 +234,11 @@
 
     if(m_transferTreeModel->findTransferByDestination(destUrl) != 0 || (destUrl.isLocalFile() && QFile::exists(destUrl.path()))) {
         KIO::RenameDialog dlg( m_mainWindow, i18n("Rename transfer"), srcUrl,
-                               destUrl, KIO::M_MULTI);
-        if (dlg.exec() == KIO::R_RENAME)
+                               destUrl, KIO::M_OVERWRITE);
+                               
+        int result = dlg.exec();
+        
+        if ( result == KIO::R_RENAME || result == KIO::R_OVERWRITE )
             destUrl = dlg.newDestUrl();
         else
             return 0;
