--- kdenetwork-4.9.3/kopete/cmake/modules/FindLibOTR.cmake~	2012-11-06 10:51:56.566038215 +0000
+++ kdenetwork-4.9.3/kopete/cmake/modules/FindLibOTR.cmake	2012-11-06 10:53:38.861993686 +0000
@@ -13,13 +13,13 @@
     SET(LIBOTR_FIND_QUIETLY TRUE)
 ENDIF (LIBOTR_INCLUDE_DIR AND LIBOTR_LIBRARY)
 
-FIND_PATH(LIBOTR_INCLUDE_DIR libotr/version.h)
+FIND_PATH(LIBOTR_INCLUDE_DIR libotr3/version.h)
 
-FIND_LIBRARY(LIBOTR_LIBRARY NAMES otr libotr)
+FIND_LIBRARY(LIBOTR_LIBRARY NAMES otr3 libotr3)
 
-# Determine version information from libotr/version.h
+# Determine version information from libotr3/version.h
 IF( LIBOTR_INCLUDE_DIR AND LIBOTR_LIBRARY )
-  EXECUTE_PROCESS(COMMAND grep "OTRL_VERSION" "${LIBOTR_INCLUDE_DIR}/libotr/version.h" OUTPUT_VARIABLE output)
+  EXECUTE_PROCESS(COMMAND grep "OTRL_VERSION" "${LIBOTR_INCLUDE_DIR}/libotr3/version.h" OUTPUT_VARIABLE output)
   STRING(REGEX MATCH "OTRL_VERSION \"[0-9]+\\.[0-9]+\\.[0-9]+" LIBOTR_VERSION "${output}")
   STRING(REGEX REPLACE "^OTRL_VERSION \"" "" LIBOTR_VERSION "${LIBOTR_VERSION}")
   # Check if version is at least 3.2.0
