diff --git i/plugins/power/meson.build w/plugins/power/meson.build
index 69e619f2..014bb6da 100644
--- i/plugins/power/meson.build
+++ w/plugins/power/meson.build
@@ -76,7 +76,6 @@ if host_is_linux
   )
 
   i18n.merge_file(
-    policy,
     input: policy_in,
     output: policy,
     po_dir: po_dir,
diff --git i/plugins/wacom/meson.build w/plugins/wacom/meson.build
index d8dd0ec7..c00323d4 100644
--- i/plugins/wacom/meson.build
+++ w/plugins/wacom/meson.build
@@ -7,7 +7,6 @@ policy_in = configure_file(
 )
 
 i18n.merge_file(
-  policy,
   input: policy_in,
   output: policy,
   po_dir: po_dir,
