 --- configure   2 Mar 2005 16:57:17 -0000   1.1395
+++ configure   3 Mar 2005 17:50:31 -0000
@@ -7042,7 +7042,7 @@

 fi

-if test "$GNU_CC"; then
+if /bin/false; then
   echo $ac_n "checking for visibility(hidden) attribute""... $ac_c" 1>&6
 echo "configure:7048: checking for visibility(hidden) attribute" >&5
 if eval "test \"`echo '$''{'ac_cv_visibility_hidden'+set}'`\" = set"; then

diff -u -r1.1.2.1 gcc_hidden.h
--- nsprpub/config/gcc_hidden.h 15 Dec 2004 05:52:39 -0000  1.1.2.1
+++ nsprpub/config/gcc_hidden.h 3 Mar 2005 17:51:06 -0000
@@ -1,2 +1,3 @@
-/* Begin all files as hidden visibility */
+/* Begin all files as hidden visibility
 #pragma GCC visibility push(hidden)
+*/ 
