diff -Nur comm-1.9.1.orig/mozilla/modules/libpr0n/encoders/png/nsPNGEncoder.cpp comm-1.9.1/mozilla/modules/libpr0n/encoders/png/nsPNGEncoder.cpp
--- comm-1.9.1.orig/mozilla/modules/libpr0n/encoders/png/nsPNGEncoder.cpp	2009-12-05 03:56:19.000000000 +0200
+++ comm-1.9.1/mozilla/modules/libpr0n/encoders/png/nsPNGEncoder.cpp	2010-01-17 18:34:32.000000000 +0200
@@ -135,7 +135,7 @@
 
   // initialize
   mPNG = png_create_write_struct(PNG_LIBPNG_VER_STRING,
-                                 png_voidp_NULL,
+                                 NULL,
                                  ErrorCallback,
                                  ErrorCallback);
   if (! mPNG)
