diff -Nur wine-1.1.37.old/dlls/windowscodecs/Makefile.in wine-1.1.37/dlls/windowscodecs/Makefile.in
--- wine-1.1.37.old/dlls/windowscodecs/Makefile.in	2010-01-22 18:54:11.000000000 +0200
+++ wine-1.1.37/dlls/windowscodecs/Makefile.in	2010-02-03 18:45:58.000000000 +0200
@@ -4,7 +4,7 @@
 VPATH     = @srcdir@
 MODULE    = windowscodecs.dll
 IMPORTLIB = windowscodecs
-IMPORTS   = uuid ole32 advapi32 kernel32
+IMPORTS   = uuid ole32 advapi32 kernel32 png
 EXTRAINCL = @PNGINCL@
 
 C_SRCS = \
diff -Nur wine-1.1.37.old/dlls/windowscodecs/pngformat.c wine-1.1.37/dlls/windowscodecs/pngformat.c
--- wine-1.1.37.old/dlls/windowscodecs/pngformat.c	2010-01-22 18:54:11.000000000 +0200
+++ wine-1.1.37/dlls/windowscodecs/pngformat.c	2010-02-03 18:48:53.000000000 +0200
@@ -59,7 +59,7 @@
 MAKE_FUNCPTR(png_get_tRNS);
 MAKE_FUNCPTR(png_set_bgr);
 MAKE_FUNCPTR(png_set_filler);
-MAKE_FUNCPTR(png_set_gray_1_2_4_to_8);
+MAKE_FUNCPTR(png_set_expand_gray_1_2_4_to_8);
 MAKE_FUNCPTR(png_set_gray_to_rgb);
 MAKE_FUNCPTR(png_set_IHDR);
 MAKE_FUNCPTR(png_set_pHYs);
@@ -100,7 +100,7 @@
         LOAD_FUNCPTR(png_get_tRNS);
         LOAD_FUNCPTR(png_set_bgr);
         LOAD_FUNCPTR(png_set_filler);
-        LOAD_FUNCPTR(png_set_gray_1_2_4_to_8);
+        LOAD_FUNCPTR(png_set_expand_gray_1_2_4_to_8);
         LOAD_FUNCPTR(png_set_gray_to_rgb);
         LOAD_FUNCPTR(png_set_IHDR);
         LOAD_FUNCPTR(png_set_pHYs);
@@ -282,7 +282,7 @@
         {
             if (bit_depth < 8)
             {
-                ppng_set_gray_1_2_4_to_8(This->png_ptr);
+                ppng_set_expand_gray_1_2_4_to_8(This->png_ptr);
                 bit_depth = 8;
             }
             ppng_set_gray_to_rgb(This->png_ptr);
diff -Nur wine-1.1.37.old/programs/winemenubuilder/Makefile.in wine-1.1.37/programs/winemenubuilder/Makefile.in
--- wine-1.1.37.old/programs/winemenubuilder/Makefile.in	2010-01-22 18:54:11.000000000 +0200
+++ wine-1.1.37/programs/winemenubuilder/Makefile.in	2010-02-03 18:46:21.000000000 +0200
@@ -5,7 +5,7 @@
 VPATH     = @srcdir@
 MODULE    = winemenubuilder.exe
 APPMODE   = -mwindows
-IMPORTS   = uuid shell32 shlwapi ole32 user32 advapi32 kernel32
+IMPORTS   = uuid shell32 shlwapi ole32 user32 advapi32 kernel32 png
 EXTRAINCL = @PNGINCL@
 
 C_SRCS = \
