diff -upr amarok-2.5.0.orig/src/musicbrainz/MusicDNSAudioDecoder.cpp amarok-2.5.0/src/musicbrainz/MusicDNSAudioDecoder.cpp
--- amarok-2.5.0.orig/src/musicbrainz/MusicDNSAudioDecoder.cpp	2012-06-07 16:12:22.000000000 +0300
+++ amarok-2.5.0/src/musicbrainz/MusicDNSAudioDecoder.cpp	2012-06-07 16:14:56.000000000 +0300
@@ -136,7 +136,7 @@ MusicDNSAudioDecoder::run()
     foreach( Meta::TrackPtr track, m_tracks )
     {
         //TODO replace with "avformat_open_input" since av_open_input_file is deprecated
-        if( av_open_input_file( &pFormatCtx, ( const char * )track->playableUrl().toLocalFile().toAscii(), NULL, 0, NULL ) )
+        if( avformat_open_input( &pFormatCtx, ( const char * )track->playableUrl().toLocalFile().toAscii(), NULL, NULL ) )
         {
             warning() << QLatin1String( "Unable to open input file: " ) + track->playableUrl().toLocalFile();
             continue;
@@ -259,4 +259,4 @@ MusicDNSAudioDecoder::run()
 }
 
 
-#include "MusicDNSAudioDecoder.moc"
\ No newline at end of file
+#include "MusicDNSAudioDecoder.moc"
