--- usr/bin/miro.old	2009-02-22 21:49:26.332356738 +0100
+++ usr/bin/miro	2009-02-22 21:50:39.169010435 +0100
@@ -21,7 +21,7 @@
         exit;
     fi
 
-    LD_LIBRARY_PATH=/usr/lib/xulrunner-1.9 $GDB -ex 'set breakpoint pending on' -ex 'break gdk_x_error' -ex 'run' --args $PYTHON ./miro.real --sync "$@"
+    LD_LIBRARY_PATH=/usr/lib/xulrunner-1.9 $GDB -ex 'set breakpoint pending on' -ex 'break gdk_x_error' -ex 'run' --args $PYTHON ./miro.real --set xine-driver=opengl --sync "$@"
 else
-    LD_LIBRARY_PATH=/usr/lib/xulrunner-1.9 miro.real "$@"
+    LD_LIBRARY_PATH=/usr/lib/xulrunner-1.9 miro.real --set xine-driver=opengl "$@"
 fi
