--- usr/bin/miro.old	2009-04-14 19:40:09.578785938 +0200
+++ usr/bin/miro	2009-04-14 19:40:34.445435006 +0200
@@ -23,5 +23,5 @@
 
     LD_LIBRARY_PATH=/usr/lib/xulrunner-1.9 $GDB -ex 'set breakpoint pending on' -ex 'run' --args $PYTHON ./miro.real --sync "$@"
 else
-    LD_LIBRARY_PATH=/usr/lib/xulrunner-1.9 miro.real "$@"
+    MOZILLA_FIVE_HOME="/usr/lib/xulrunner-1.9/" LD_LIBRARY_PATH=/usr/lib/xulrunner-1.9 miro.real "$@"
 fi
