From 174a51819570a975cc61e642f7f26cb4bc2f9c57 Mon Sep 17 00:00:00 2001
From: Diego Escalante Urrelo <descalante@igalia.com>
Date: Mon, 01 Nov 2010 05:56:51 +0000
Subject: tests: DBUS_LIBS is needed for libephymain.la

---
diff --git a/tests/Makefile.am b/tests/Makefile.am
index bd2c378..b973214 100644
--- a/tests/Makefile.am
+++ b/tests/Makefile.am
@@ -23,6 +23,7 @@ LDADD = \
 	$(top_builddir)/lib/widgets/libephywidgets.la \
 	$(top_builddir)/lib/libephymisc.la \
 	$(top_builddir)/lib/egg/libegg.la \
+	$(DBUS_LIBS) \
 	$(DEPENDENCIES_LIBS) 
 
 if ENABLE_NSS
--
cgit v0.8.3.4
