Change temp file location from /usr/tmp to /tmp.

diff -Naur mpage-2.5.4-orig/file.c mpage-2.5.4/file.c
--- mpage-2.5.4-orig/file.c	2004-05-30 15:34:15.000000000 -0400
+++ mpage-2.5.4/file.c	2005-11-02 09:56:00.000000000 -0500
@@ -168,7 +168,7 @@
      * but the tmpfilename
      */
      
-    (void) strcpy(tmpfile, "/usr/tmp/mpage-stdin-XXXXXX");
+    (void) strcpy(tmpfile, "/tmp/mpage-stdin-XXXXXX");
     if ( (tmpfd = mkstemp(tmpfile)) == -1) {
         fprintf(stderr, "%s: cannot create temporary file", MPAGE);
         perror(MPAGE);
@@ -209,7 +209,7 @@
          * a temporary file; this temporary file will then
          * be used as input to the do_doc routine
          */
-        (void)strcpy(tmpfile, "/usr/tmp/mpageXXXXXX");
+        (void)strcpy(tmpfile, "/tmp/mpageXXXXXX");
         if ( (tmpfd = mkstemp(tmpfile)) == -1) {
             fprintf(stderr, "%s: cannot create temporary file", MPAGE);
             perror(MPAGE);
@@ -280,7 +280,7 @@
 	/*
 	 * store the input to the temporary file to guess encoding correctly
 	 */
-        (void)strcpy(tmpfile, "/usr/tmp/mpageXXXXXX");
+        (void)strcpy(tmpfile, "/tmp/mpageXXXXXX");
         if ( (tmpfd = mkstemp(tmpfile)) == -1) {
             fprintf(stderr, "%s: cannot create temporary file", MPAGE);
 	    tmpfile[0] = 0;
diff -Naur mpage-2.5.4-orig/mpage.1.in mpage-2.5.4/mpage.1.in
--- mpage-2.5.4-orig/mpage.1.in	2004-05-30 15:41:09.000000000 -0400
+++ mpage-2.5.4/mpage.1.in	2005-11-02 09:55:36.000000000 -0500
@@ -549,7 +549,7 @@
 environment variables.
 
 .SH FILES
-/usr/tmp/mpageXXXXXX
+/tmp/mpageXXXXXX
 .br
 PREFIX/share/mpage
 
