diff -ruN seahorse-plugins-2.29.90.orig/agent/seahorse-agent-cache.c seahorse-plugins-2.29.90/agent/seahorse-agent-cache.c
--- seahorse-plugins-2.29.90.orig/agent/seahorse-agent-cache.c	2010-01-25 03:06:50.000000000 +0100
+++ seahorse-plugins-2.29.90/agent/seahorse-agent-cache.c	2010-02-14 12:31:01.000000000 +0100
@@ -236,6 +236,7 @@
     if (g_cache == NULL) {
         g_cache =
             g_hash_table_new_full (g_str_hash, g_str_equal, NULL, destroy_cache_item);
+        gpgme_check_version (NULL);
 
         err = gpgme_engine_check_version (proto);
         g_return_if_fail (GPG_IS_OK (err));
diff -ruN seahorse-plugins-2.29.90.orig/libseahorse/seahorse-pgp-operation.c seahorse-plugins-2.29.90/libseahorse/seahorse-pgp-operation.c
--- seahorse-plugins-2.29.90.orig/libseahorse/seahorse-pgp-operation.c	2010-01-25 03:06:50.000000000 +0100
+++ seahorse-plugins-2.29.90/libseahorse/seahorse-pgp-operation.c	2010-02-14 12:31:25.000000000 +0100
@@ -322,6 +322,8 @@
     gpgme_protocol_t proto = GPGME_PROTOCOL_OpenPGP;
     gpgme_error_t err;
     gpgme_ctx_t ctx;
+
+    gpgme_check_version (NULL);
  
     err = gpgme_engine_check_version (proto);
     g_return_if_fail (GPG_IS_OK (err));
diff -ruN seahorse-plugins-2.29.90.orig/plugins/nautilus/seahorse-tool.c seahorse-plugins-2.29.90/plugins/nautilus/seahorse-tool.c
--- seahorse-plugins-2.29.90.orig/plugins/nautilus/seahorse-tool.c	2010-01-25 03:06:50.000000000 +0100
+++ seahorse-plugins-2.29.90/plugins/nautilus/seahorse-tool.c	2010-02-14 12:32:44.000000000 +0100
@@ -171,6 +171,7 @@
         
         if (recips) {
             
+            gpgme_check_version (NULL);
             gerr = gpgme_new (&ctx);
             g_return_val_if_fail (GPG_IS_OK (gerr), NULL);
         
@@ -328,6 +329,8 @@
         id = cryptui_keyset_key_raw_keyid (keyset, signer);
         g_free (signer);
         
+        gpgme_check_version (NULL);
+        
         gerr = gpgme_new (&ctx);
         g_return_val_if_fail (GPG_IS_OK (gerr), NULL);
         
