diff -ruN seahorse-plugins-2.28.0/plugins/nautilus/seahorse-tool.c seahorse-plugins-2.28.0.patched/plugins/nautilus/seahorse-tool.c
--- seahorse-plugins-2.28.0/plugins/nautilus/seahorse-tool.c	2009-09-03 13:20:15.000000000 +0200
+++ seahorse-plugins-2.28.0.patched/plugins/nautilus/seahorse-tool.c	2009-10-13 09:03:34.000000000 +0200
@@ -171,6 +172,7 @@
         
         if (recips) {
             
+            gpgme_check_version (NULL);
             gerr = gpgme_new (&ctx);
             g_return_val_if_fail (GPG_IS_OK (gerr), NULL);
         
