From 4a974f2686d2fafdcda4a180b0483a7b17fd2d71 Mon Sep 17 00:00:00 2001
From: Marty Jack <martyj@linux.local>
Date: Thu, 22 Jul 2010 19:46:13 -0400
Subject: [PATCH 3/3] Fix build issue with symbol "alarm" showing up on F14 (Bug3033293)

---
 src/plugins/batt/batt.c |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/plugins/batt/batt.c b/src/plugins/batt/batt.c
index 288231f..05c0deb 100644
--- a/src/plugins/batt/batt.c
+++ b/src/plugins/batt/batt.c
@@ -95,7 +95,7 @@ typedef struct {
 typedef struct {
     char *command;
     sem_t *lock;
-} alarm;
+} Alarm;
 
 static void destructor(Plugin *p);
 static void update_display(lx_battery *lx_b, gboolean repaint);
@@ -103,7 +103,7 @@ static void update_display(lx_battery *lx_b, gboolean repaint);
 /* alarmProcess takes the address of a dynamically allocated alarm struct (which
    it must free). It ensures that alarm commands do not run concurrently. */
 static void * alarmProcess(void *arg) {
-    alarm *a = (alarm *) arg;
+    Alarm *a = (Alarm *) arg;
 
     sem_wait(a->lock);
     system(a->command);
@@ -157,7 +157,7 @@ void update_display(lx_battery *lx_b, gboolean repaint) {
 	/* Run the alarm command if it isn't already running */
 	if (alarmCanRun) {
 	    
-	    alarm *a = (alarm *) malloc(sizeof(alarm));
+	    Alarm *a = (Alarm *) malloc(sizeof(Alarm));
 	    a->command = lx_b->alarmCommand;
 	    a->lock = &(lx_b->alarmProcessLock);
 	    
-- 
1.7.1.1

