=== modified file 'configure'
--- configure	2009-03-15 08:35:38 +0000
+++ configure	2009-03-15 13:13:54 +0000
@@ -1677,8 +1677,8 @@
                           db, gdbm, ndbm)
   --with-config-file=CF   use config file CF [CF=SYSCONFDIR/man_db.conf]
   --with-sections=SECTIONS
-                          use manual page sections SECTIONS [1 n l 8 3 2 5 4 9
-                          6 7]
+                          use manual page sections SECTIONS [1 n l 8 3 0 2 5 4
+                          9 6 7]
   --with-browser=BROWSER  use BROWSER as default web browser
   --with-pager=PAGER      use PAGER as default pager
   --with-nroff=NROFF      use NROFF as roff formatter for character devices

=== modified file 'configure.ac'
--- configure.ac	2009-03-14 23:12:00 +0000
+++ configure.ac	2009-03-15 13:13:54 +0000
@@ -85,7 +85,7 @@
 config_file_basename=${withval##*/}
 config_file_dirname=`AS_DIRNAME(["$config_file"])`
 AC_ARG_WITH(sections,
-[AS_HELP_STRING([--with-sections=SECTIONS], [use manual page sections SECTIONS @<:@1 n l 8 3 2 5 4 9 6 7@:>@])],
+[AS_HELP_STRING([--with-sections=SECTIONS], [use manual page sections SECTIONS @<:@1 n l 8 3 0 2 5 4 9 6 7@:>@])],
 	[if test "$withval" = "yes" || test "$withval" = "no"
 	 then
 		AC_MSG_ERROR(--with-sections requires an argument)

=== modified file 'man/man5/manpath.man5'
--- man/man5/manpath.man5	2008-12-30 12:58:00 +0000
+++ man/man5/manpath.man5	2009-03-15 13:13:54 +0000
@@ -122,7 +122,7 @@
 .PP
 .RS
 .nf
-.if !'po4a'hide' SECTION 1 n l 8 3 2 5 4 9 6 7
+.if !'po4a'hide' SECTION 1 n l 8 3 0 2 5 4 9 6 7
 .fi
 .RE
 .PP

=== modified file 'src/man_db.conf.in'
--- src/man_db.conf.in	2008-12-07 17:43:50 +0000
+++ src/man_db.conf.in	2009-03-15 13:13:54 +0000
@@ -99,7 +99,7 @@
 #DEFINE		apropos_regex_grep_flags	-iE
 #---------------------------------------------------------
 # Section names. Manual sections will be searched in the order listed here;
-# the default is 1, n, l, 8, 3, 2, 5, 4, 9, 6, 7. Multiple SECTION
+# the default is 1, n, l, 8, 3, 0, 2, 5, 4, 9, 6, 7. Multiple SECTION
 # directives may be given for clarity, and will be concatenated together in
 # the expected way.
 # If a particular extension is not in this list (say, 1mh), it will be

