diff -Naur xine-ui-0.99.5-orig/src/xitk/Imlib-light/load.c xine-ui-0.99.5/src/xitk/Imlib-light/load.c
--- xine-ui-0.99.5-orig/src/xitk/Imlib-light/load.c	2010-01-19 02:46:21.000000000 -0500
+++ xine-ui-0.99.5/src/xitk/Imlib-light/load.c	2010-01-19 02:49:22.000000000 -0500
@@ -193,7 +193,7 @@
     return 0;
   fread(buf, 1, 8, f);
   rewind(f);
-  return (int)png_check_sig(buf, 8);
+  return (int)png_sig_cmp(buf, 0, 8)==0;
 }
 
 ImlibImage * Imlib_load_image(ImlibData * id, char *file) {
