diff -Naur libpng-1.4.0-orig/pngconf.h libpng-1.4.0/pngconf.h
--- libpng-1.4.0-orig/pngconf.h	2010-01-27 20:38:27.000000000 -0500
+++ libpng-1.4.0/pngconf.h	2010-01-27 20:44:44.000000000 -0500
@@ -497,11 +497,9 @@
 #  ifndef PNG_NO_READ_INVERT
 #    define PNG_READ_INVERT_SUPPORTED
 #  endif
-#if 0 /* removed from libpng-1.4.0 */
 #  ifndef PNG_NO_READ_DITHER
 #    define PNG_READ_DITHER_SUPPORTED
 #  endif
-#endif /* 0 */
 #  ifndef PNG_NO_READ_BACKGROUND
 #    define PNG_READ_BACKGROUND_SUPPORTED
 #  endif
