--- drivers/amrmo_init.c~	2010-11-01 09:59:31.253338565 +0100
+++ drivers/amrmo_init.c	2010-11-01 10:02:56.413334268 +0100
@@ -430,11 +430,11 @@
 
 
 
-static int amrmo_ioctl(struct inode *inode, struct file *file, unsigned int cmd, unsigned long arg)
+static long amrmo_unlocked_ioctl(struct file *file, unsigned int cmd, unsigned long arg)
 {
 	struct amrmo_struct *amrmo = (struct amrmo_struct *)file->private_data;
         unsigned long flags;
-	int ret;
+	long ret;
 	unsigned stat;
 	AMRMO_DBG("amrmo_ioctl: cmd %x, arg %lx...\n",cmd,arg);
         switch (cmd) 
@@ -502,7 +502,7 @@
         .read =    amrmo_read,
         .write =   amrmo_write,
         .poll =    amrmo_poll,
-        .ioctl =   amrmo_ioctl,
+        .unlocked_ioctl =   amrmo_unlocked_ioctl,
         .open =    amrmo_open,
         .release = amrmo_release,
 };
