Index: settings.mk
===================================================================
RCS file: /cvs/tools/solenv/inc/settings.mk,v
retrieving revision 1.236
diff -u -r1.236 settings.mk
--- solenv/inc/settings.mk	17 Jul 2008 13:37:59 -0000	1.236
+++ solenv/inc/settings.mk	24 Jul 2008 17:44:53 -0000
@@ -622,7 +622,10 @@
 	@$(MKOUT) $(ROUT)
 	@echo $(EMQ)# > $@
 
+# don't need/want output trees in solenv!!!
+.IF "$(PRJNAME)"!="solenv"
 .INCLUDE : $(OUT)$/inc$/myworld.mk
+.ENDIF			# "$(PRJNAME)"!="solenv"
 
 .IF "$(common_build)"!=""
 $(LOCAL_COMMON_OUT)$/inc$/%world.mk :
