Index: scp2/source/ooo/makefile.mk
===================================================================
--- scp2/source/ooo/makefile.mk	(Revision 277420)
+++ scp2/source/ooo/makefile.mk	(Arbeitskopie)
@@ -212,6 +212,9 @@
 .IF "$(DISABLE_NEON)" == "TRUE"
 SCPDEFS+=-DDISABLE_NEON
 .ENDIF
+.IF "$(SYSTEM_NEON)" == "YES"
+SCPDEFS+=-DSYSTEM_NEON
+.ENDIF
 
 # if yes or unset (neon not used) -> do not install openssl library!
 .IF $(SYSTEM_OPENSSL) != "YES"
Index: source/ooo/file_library_ooo.scp
===================================================================
--- scp2/source/ooo/file_library_ooo.scp	(Revision 277420)
+++ scp2/source/ooo/file_library_ooo.scp	(Arbeitskopie)
@@ -1547,8 +1547,10 @@
 
 #ifndef DISABLE_NEON
 SPECIAL_UNO_LIB_FILE(gid_File_Lib_Ucpdav1,ucpdav1)
+#ifndef SYSTEM_NEON
 SPECIAL_LIB_FILE(gid_File_Lib_Neon,neon)
 #endif
+#endif
 
 SPECIAL_UNO_LIB_FILE_PATCH(gid_File_Lib_Ucpfile1,ucpfile1)
 

