--- drivers/gpu/drm/nouveau/nv04_dac.c	2009-10-20 17:54:19.454045092 +0200
+++ drivers/gpu/drm/nouveau/nv04_dac.c.block3	2009-10-20 18:01:44.891084375 +0200
@@ -223,6 +223,20 @@
 		saved_rtest_ctrl, temp, saved_gpio_ext = 0, routput;
 	int head, present = 0;
 
+nv_wr32(dev, 0x68252c, 0x100); /*363389.094484   write32 #1 NV11_PRAMDAC.OUTPUT_OR_2_3 <- {00000000000000000000000}, OUTPUT_SELECT_CRTC1 = 0x1, {0000000}, OUTPUT_DAC_ENABLE = 0x0*/ 
+nv_wr32(dev, 0x68252c, 0x100); /*363391.094548   write32 #1 NV11_PRAMDAC.OUTPUT_OR_2_3 <- {00000000000000000000000}, OUTPUT_SELECT_CRTC1 = 0x1, {0000000}, OUTPUT_DAC_ENABLE = 0x0*/ 
+nv_wr32(dev, 0x68252c, 0x101); /*363392.095582   write32 #1 NV11_PRAMDAC.OUTPUT_OR_2_3 <- {00000000000000000000000}, OUTPUT_SELECT_CRTC1 = 0x1, {0000000}, OUTPUT_DAC_ENABLE = 0x1*/ 
+nv_wr32(dev, 0x68252c, 0x101); /*363394.095650   write32 #1 NV11_PRAMDAC.OUTPUT_OR_2_3 <- {00000000000000000000000}, OUTPUT_SELECT_CRTC1 = 0x1, {0000000}, OUTPUT_DAC_ENABLE = 0x1*/ 
+nv_wr32(dev, 0x68252c, 0x100); /*363396.095714   write32 #1 NV11_PRAMDAC.OUTPUT_OR_2_3 <- {00000000000000000000000}, OUTPUT_SELECT_CRTC1 = 0x1, {0000000}, OUTPUT_DAC_ENABLE = 0x0*/ 
+nv_wr32(dev, 0x68252c, 0x101); /*363397.161792   write32 #1 NV11_PRAMDAC.OUTPUT_OR_2_3 <- {00000000000000000000000}, OUTPUT_SELECT_CRTC1 = 0x1, {0000000}, OUTPUT_DAC_ENABLE = 0x1*/ 
+nv_wr32(dev, 0x682610, 0x94050140); /*363399.161860   write32 #1 NV11_PRAMDAC[1].TEST_DATA <- 0x94050140*/ 
+nv_wr32(dev, 0x682610, 0x94050140); /*363400.161892   write32 #1 NV11_PRAMDAC[1].TEST_DATA <- 0x94050140*/ 
+nv_wr32(dev, 0x682608, 0xf0101000); /*363402.161956   write32 #1 NV11_PRAMDAC.TEST_CONTROL_OR_2_3 <- 0xf0101000*/ 
+nv_wr32(dev, 0x682608, 0x100000); /*363405.163058   write32 #1 NV11_PRAMDAC.TEST_CONTROL_OR_2_3 <- 0x00100000*/ 
+nv_wr32(dev, 0x68252c, 0x100); /*363406.163090   write32 #1 NV11_PRAMDAC.OUTPUT_OR_2_3 <- {00000000000000000000000}, OUTPUT_SELECT_CRTC1 = 0x1, {0000000}, OUTPUT_DAC_ENABLE = 0x0*/ 
+
+msleep(30);
+
 #define RGB_TEST_DATA(r, g, b) (r << 0 | g << 10 | b << 20)
 	if (dcb->type == OUTPUT_TV) {
 		testval = RGB_TEST_DATA(0xa0, 0xa0, 0xa0);
@@ -236,6 +250,8 @@
 			testval = dev_priv->vbios->dactestval;
 	}
 
+	testval = 0x94050140;
+
 	saved_rtest_ctrl = NVReadRAMDAC(dev, 0, NV_PRAMDAC_TEST_CONTROL + regoffset);
 	NVWriteRAMDAC(dev, 0, NV_PRAMDAC_TEST_CONTROL + regoffset,
 		      saved_rtest_ctrl & ~NV_PRAMDAC_TEST_CONTROL_PWRDWN_DAC_OFF);
@@ -251,8 +267,10 @@
 	if (dev_priv->chipset >= 0x34) {
 		saved_gpio_ext = NVReadCRTC(dev, 0, NV_PCRTC_GPIO_EXT);
 
+#if 0
 		NVWriteCRTC(dev, 0, NV_PCRTC_GPIO_EXT, (saved_gpio_ext & ~(3 << 20)) |
 			    (dcb->type == OUTPUT_TV ? (1 << 20) : 0));
+#endif
 	}
 
 	msleep(4);
@@ -267,13 +285,14 @@
 #endif
 	/* nv driver and nv31 use 0xfffffeee, nv34 and 6600 use 0xfffffece */
 	routput = (saved_routput & 0xfffffece) | head << 8;
-
+#if 0
 	if (nv_arch(dev) >= NV_40) {
 		if (dcb->type == OUTPUT_TV)
 			routput |= 1 << 20;
 		else
 			routput &= ~(1 << 20);
 	}
+#endif
 
 	NVWriteRAMDAC(dev, 0, NV_PRAMDAC_DACCLK + regoffset, routput);
 	msleep(1);
