--- calendar/gui/alarm-notify/alarm-queue.c
+++ calendar/gui/alarm-notify/alarm-queue.c
@@ -40,6 +40,9 @@
 
 #ifdef HAVE_LIBNOTIFY
 #include <libnotify/notify.h>
+#ifndef NOTIFY_CHECK_VERSION
+#define NOTIFY_CHECK_VERSION(x,y,z) 0
+#endif
 #endif
 
 #include "alarm.h"
@@ -1606,7 +1609,12 @@
 			body = g_strdup_printf ("%s %s", start_str, time_str);
 	}
 
-	n = notify_notification_new (summary, body, "stock_appointment-reminder", NULL);
+	n = notify_notification_new (summary, body, "stock_appointment-reminder"
+#if NOTIFY_CHECK_VERSION (0, 7, 0)
+	);
+#else
+	, NULL);
+#endif
 	if (!notify_notification_show(n, NULL))
 	    g_warning ("Could not send notification to daemon\n");
 
--- plugins/mail-notification/mail-notification.c
+++ plugins/mail-notification/mail-notification.c
@@ -47,6 +47,9 @@
 
 #ifdef HAVE_LIBNOTIFY
 #include <libnotify/notify.h>
+#ifndef NOTIFY_CHECK_VERSION
+#define NOTIFY_CHECK_VERSION(x,y,z) 0
+#endif
 #endif
 
 #define GCONF_KEY_ROOT			"/apps/evolution/eplugin/mail-notification/"
@@ -564,9 +567,14 @@
 
 			notify  = notify_notification_new (
 				_("New email"), safetext,
-				"mail-unread", NULL);
+				"mail-unread"
+#if NOTIFY_CHECK_VERSION (0, 7, 0)
+				);
+#else
+				, NULL);
 			notify_notification_attach_to_status_icon (
 				notify, status_icon);
+#endif
 
 			/* Check if actions are supported */
 			if (can_support_actions ()) {
--- plugins/publish-calendar/publish-calendar.c
+++ plugins/publish-calendar/publish-calendar.c
@@ -46,6 +46,9 @@
 
 #ifdef HAVE_LIBNOTIFY
 #include <libnotify/notify.h>
+#ifndef NOTIFY_CHECK_VERSION
+#define NOTIFY_CHECK_VERSION(x,y,z) 0
+#endif
 #endif
 
 static GtkListStore *store = NULL;
@@ -150,8 +153,13 @@
 				return;
 			}
 
-			notify  = notify_notification_new (_("Calendar Publishing"), actual_msg->str, stock_name, NULL);
+			notify  = notify_notification_new (_("Calendar Publishing"), actual_msg->str, stock_name
+#if NOTIFY_CHECK_VERSION (0, 7, 0)
+			);
+#else
+			, NULL);
 			notify_notification_attach_to_status_icon (notify, status_icon);
+#endif
 			notify_notification_set_urgency (notify, NOTIFY_URGENCY_NORMAL);
 			notify_notification_set_timeout (notify, NOTIFY_EXPIRES_DEFAULT);
 			g_timeout_add (500, show_notify_cb, NULL);
