--- QtCurve-KDE3-1.6.0/CMakeLists.txt~	2010-09-14 16:19:36.543333337 +0200
+++ QtCurve-KDE3-1.6.0/CMakeLists.txt	2010-09-14 16:22:18.160000003 +0200
@@ -44,6 +44,12 @@
     endif (QT_PLUGINS_DIR MATCHES NOTFOUND)
 endif (NOT QTC_QT_ONLY)
 
+# Comment the kde3/kde4 prefix detection as a whole as it wants to run
+# kde-config and kde4-config, which in turn can create stuff in ~/.kde.
+# This cannot work on read-only homes.
+# So, the "fallback" prefixes ("/usr" for both) will be used.
+if (BRAINDEAD_DETECTION)
+
 if (NOT QTC_QT_ONLY)
 exec_program(kde-config
              ARGS --prefix --expandvars
@@ -71,6 +77,8 @@
                  OUTPUT_VARIABLE KDE4PREFIX)
 endif(KDE4_KDECONFIG_EXECUTABLE)
 
+endif (BRAINDEAD_DETECTION)
+
 if(NOT KDE4PREFIX)
     set(KDE4PREFIX ${KDE3PREFIX})
 endif(NOT KDE4PREFIX)
