--- officecfg/registry/data/org/openoffice/Office/Common.xcu	2010-07-17 11:04:59.000000000 +0000
+++ officecfg/registry/data/org/openoffice/Office/Common.xcu.new	2010-07-18 09:09:58.031564593 +0000
@@ -576,7 +576,7 @@
   </node>
   <node oor:name="Misc">
     <prop oor:name="UseSystemFileDialog">
-      <value install:module="UseOOoFileDialogs">false</value>
+      <value install:module="UseOOoFileDialogs">true</value>
       <value install:module="gconflockdown" oor:external=
          "com.sun.star.configuration.backend.GconfBackend UseSystemFileDialog"/>
     </prop>
