diff --git i/data/systemd/meson.build w/data/systemd/meson.build
index a42f7179..dec3f95d 100644
--- i/data/systemd/meson.build
+++ w/data/systemd/meson.build
@@ -2,10 +2,14 @@ configure_file(
   input: 'org.gnome.GPaste.Ui.systemd.in',
   configuration: conf,
   output: 'org.gnome.GPaste.Ui.service',
+  install: get_option('systemd'),
+  install_dir: userunit_dir,
 )
 
 configure_file(
   input: 'org.gnome.GPaste.systemd.in',
   configuration: conf,
   output: 'org.gnome.GPaste.service',
-)
\ No newline at end of file
+  install: get_option('systemd'),
+  install_dir: userunit_dir,
+)
diff --git i/src/libgpaste/meson.build w/src/libgpaste/meson.build
index 1a1b0ca5..68a116cc 100644
--- i/src/libgpaste/meson.build
+++ w/src/libgpaste/meson.build
@@ -143,14 +143,17 @@ libgpaste_inc = include_directories(
   'util',
 )
 
+libgpaste_symbols_file = meson.current_source_dir() / 'libgpaste.sym'
+
 libgpaste = library(
   'gpaste',
   sources: libgpaste_sources,
   version: gpaste_soversion,
   dependencies: libgpaste_deps,
   install: true,
   install_dir: get_option('libdir'),
-   include_directories : libgpaste_inc,
+  include_directories : libgpaste_inc,
+  link_args: ['-Wl,--version-script=' + libgpaste_symbols_file],
 )
 
 libgpaste_internal_dep = declare_dependency(
