From 073d5b0b392781bf4a6aa7f9e2dbe2ae51caed2c Mon Sep 17 00:00:00 2001
From: Adam Jackson <ajax@redhat.com>
Date: Tue, 25 Sep 2012 08:54:38 -0400
Subject: [PATCH 1/4] Remove mibstore.h

Signed-off-by: Adam Jackson <ajax@redhat.com>
---
 src/glint_driver.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/src/glint_driver.c b/src/glint_driver.c
index b6d20a9..aa78516 100644
--- a/src/glint_driver.c
+++ b/src/glint_driver.c
@@ -52,8 +52,6 @@
 #include "compiler.h"
 #include "mipointer.h"
 
-#include "mibstore.h"
-
 #include "pm3_regs.h"
 #include "glint_regs.h"
 #include "IBM.h"
@@ -2904,7 +2902,6 @@ GLINTScreenInit(SCREEN_INIT_ARGS_DECL)
         }
     }
 
-    miInitializeBackingStore(pScreen);
     xf86SetBackingStore(pScreen);
     xf86SetSilkenMouse(pScreen);
 
-- 
2.10.2


From b54f27f0de0558880b2146a4315a99907eb72249 Mon Sep 17 00:00:00 2001
From: Adam Jackson <ajax@redhat.com>
Date: Mon, 11 Aug 2014 12:29:28 -0400
Subject: [PATCH 2/4] Pass the pci device through to fbdevHWInit

Signed-off-by: Adam Jackson <ajax@redhat.com>
---
 src/glint_driver.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/src/glint_driver.c b/src/glint_driver.c
index aa78516..019a2f1 100644
--- a/src/glint_driver.c
+++ b/src/glint_driver.c
@@ -944,7 +944,9 @@ GLINTPreInit(ScrnInfoPtr pScrn, int flags)
     if (FBDevProbed) {
 	int default_depth, fbbpp;
 	
-	if (!fbdevHWInit(pScrn,NULL,xf86FindOptionValue(pGlint->pEnt->device->options,"fbdev"))) {
+	if (!fbdevHWInit(pScrn, pGlint->PciInfo,
+			 xf86FindOptionValue(pGlint->pEnt->device->options,
+					     "fbdev"))) {
 		xf86DrvMsg(pScrn->scrnIndex, X_ERROR, "fbdevHWInit failed!\n");	
 		return FALSE;
 	}
@@ -1076,7 +1078,9 @@ GLINTPreInit(ScrnInfoPtr pScrn, int flags)
 		return FALSE;
 	}
 
-	if (!fbdevHWInit(pScrn,NULL,xf86FindOptionValue(pGlint->pEnt->device->options,"fbdev")))
+	if (!fbdevHWInit(pScrn, pGlint->PciInfo,
+			 xf86FindOptionValue(pGlint->pEnt->device->options,
+					     "fbdev")))
 	{
 		xf86DrvMsg(pScrn->scrnIndex, X_ERROR, "fbdevHWInit failed!\n");
 		return FALSE;
-- 
2.10.2


From 5b22c69898330d2e321596f7061b91362c0e6216 Mon Sep 17 00:00:00 2001
From: Michael Lorenz <macallan@NetBSD.org>
Date: Thu, 19 Feb 2015 15:26:46 +0100
Subject: [PATCH 3/4] Fix I2C for DDC2.

When doing DDC2 on Permedia make sure to clear the USE_MONID bit
in the DDCdata register so the i2c bits actually do something.

Signed-off-by: Thomas Klausner <wiz@NetBSD.org>
---
 src/pm2_dac.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/pm2_dac.c b/src/pm2_dac.c
index 85093ff..fc4bfea 100644
--- a/src/pm2_dac.c
+++ b/src/pm2_dac.c
@@ -458,7 +458,8 @@ Permedia2I2CPutBits(I2CBusPtr b, int scl, int sda)
 {
     GLINTPtr pGlint = (GLINTPtr) b->DriverPrivate.ptr;
     int r = (pGlint->DDCBus == b) ? PMDDCData : VSSerialBusControl;
-    CARD32 v = GLINT_READ_REG(r) & ~(ClkOut | DataOut);
+
+    CARD32 v = 0;
 
     if (scl > 0) v |= ClkOut;
     if (sda > 0) v |= DataOut;
-- 
2.10.2


From 2eb4d03d2280deec4d6f1b06aaa16477a641af9a Mon Sep 17 00:00:00 2001
From: Adam Jackson <ajax@redhat.com>
Date: Tue, 19 Jul 2016 10:03:56 -0400
Subject: [PATCH 4/4] Adapt Block/WakeupHandler signature for ABI 23

Signed-off-by: Adam Jackson <ajax@redhat.com>
---
 src/compat-api.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/compat-api.h b/src/compat-api.h
index 6bc946f..89976e4 100644
--- a/src/compat-api.h
+++ b/src/compat-api.h
@@ -75,8 +75,13 @@
 
 #define SCREEN_INIT_ARGS_DECL ScreenPtr pScreen, int argc, char **argv
 
+#if ABI_VIDEODRV_VERSION >= SET_ABI_VERSION(23, 0)
+#define BLOCKHANDLER_ARGS_DECL ScreenPtr arg, pointer pTimeout
+#define BLOCKHANDLER_ARGS arg, pTimeout
+#else
 #define BLOCKHANDLER_ARGS_DECL ScreenPtr arg, pointer pTimeout, pointer pReadmask
 #define BLOCKHANDLER_ARGS arg, pTimeout, pReadmask
+#endif
 
 #define CLOSE_SCREEN_ARGS_DECL ScreenPtr pScreen
 #define CLOSE_SCREEN_ARGS pScreen
-- 
2.10.2

