From 225e00a14a9200f11253775a7c7778c1c29c8b23 Mon Sep 17 00:00:00 2001
From: Kevin Bowling <kevin.bowling@kev009.com>
Date: Thu, 8 Jul 2021 22:05:12 -0700
Subject: [PATCH 3/3] pkgconfig: meson build should match autotools requires

Signed-off-by: Laurent Carlier <lordheavym@gmail.com>
---
 meson.build | 1 +
 1 file changed, 1 insertion(+)

diff --git a/meson.build b/meson.build
index cffa7f8..536fbef 100644
--- a/meson.build
+++ b/meson.build
@@ -34,4 +34,5 @@ pkgconf.generate(
   libglu,
   name : 'glu',
   description : 'Mesa OpenGL Utility Library',
+  requires: dep_gl
 )
-- 
2.34.1

