diff --git i/src/gcal-utils.c w/src/gcal-utils.c
index 9846a59b..7f96a40d 100644
--- i/src/gcal-utils.c
+++ w/src/gcal-utils.c
@@ -154,6 +154,9 @@ icaltime_to_datetime (const icaltimetype  *date)
   GDateTime *dt;
   GTimeZone *tz;
 
+  if (!date)
+    return NULL;
+
   tz = date->zone ? g_time_zone_new (icaltime_get_tzid (*date)) : g_time_zone_new_utc ();
   dt = g_date_time_new (tz,
                         date->year,
