diff --git a/grub-extras/lua/grub_lib.c b/grub-extras/lua/grub_lib.c
index 9014320..055d620 100644
--- a/grub-extras/lua/grub_lib.c
+++ b/grub-extras/lua/grub_lib.c
@@ -466,7 +466,7 @@ grub_lua_add_menu (lua_State *state)
       if (! p)
 	return push_result (state);
 
-      grub_normal_add_menu_entry (n, args, NULL, NULL, NULL, NULL, p, 0);
+      grub_normal_add_menu_entry (n, args, NULL, NULL, NULL, NULL, NULL, p, 0);
     }
   else
     {
