diff -ur grub-1.98.orig/util/grub.d/00_header.in grub-1.98/util/grub.d/00_header.in
--- grub-1.98.orig/util/grub.d/00_header.in	2010-03-11 00:36:05.766730374 -0400
+++ grub-1.98/util/grub.d/00_header.in	2010-03-11 00:40:49.046203888 -0400
@@ -150,7 +150,7 @@
   fi
   cat << EOF
 if sleep$verbose --interruptible ${GRUB_HIDDEN_TIMEOUT} ; then
-  set timeout=${GRUB_TIMEOUT}
+  set timeout=0
 fi
 EOF
 else
