Index: WINGs/Examples/colorpick.c
===================================================================
--- WINGs/Examples/colorpick.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WINGs/Examples/colorpick.c	2013-01-18 16:57:47.955049409 +0100
@@ -1,4 +1,8 @@
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <stdlib.h>
 #include <stdio.h>
 #include <WINGs/WINGs.h>
Index: WINGs/Examples/fontl.c
===================================================================
--- WINGs/Examples/fontl.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WINGs/Examples/fontl.c	2013-01-18 16:57:47.955049409 +0100
@@ -18,6 +18,10 @@
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <stdlib.h>
 #include <stdio.h>
 #include <stdint.h>
Index: WINGs/Examples/puzzle.c
===================================================================
--- WINGs/Examples/puzzle.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WINGs/Examples/puzzle.c	2013-01-18 16:57:47.955049409 +0100
@@ -1,4 +1,8 @@
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <stdlib.h>
 #include <stdio.h>
 #include <WINGs/WINGs.h>
Index: WINGs/Extras/test.c
===================================================================
--- WINGs/Extras/test.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WINGs/Extras/test.c	2013-01-18 16:57:47.956049439 +0100
@@ -1,4 +1,8 @@
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <WINGs/WINGs.h>
 #include <stdio.h>
 #include <stdint.h>
Index: WINGs/Extras/wtabledelegates.c
===================================================================
--- WINGs/Extras/wtabledelegates.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WINGs/Extras/wtabledelegates.c	2013-01-18 16:57:47.956049439 +0100
@@ -1,4 +1,8 @@
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <stdint.h>
 #include <WINGs/WINGsP.h>
 
Index: WINGs/Extras/wtableview.c
===================================================================
--- WINGs/Extras/wtableview.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WINGs/Extras/wtableview.c	2013-01-18 16:57:47.956049439 +0100
@@ -1,4 +1,8 @@
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <WINGs/WINGsP.h>
 #include <X11/cursorfont.h>
 #include <stdint.h>
Index: WINGs/Tests/mywidget.c
===================================================================
--- WINGs/Tests/mywidget.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WINGs/Tests/mywidget.c	2013-01-18 16:57:47.956049439 +0100
@@ -13,6 +13,10 @@
  *
  *
  */
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <WINGs/WINGsP.h>
 
 /*
Index: WINGs/Tests/testmywidget.c
===================================================================
--- WINGs/Tests/testmywidget.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WINGs/Tests/testmywidget.c	2013-01-18 16:57:47.956049439 +0100
@@ -1,4 +1,8 @@
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <WINGs/WINGs.h>
 #include <stdlib.h>
 
Index: WINGs/Tests/wmfile.c
===================================================================
--- WINGs/Tests/wmfile.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WINGs/Tests/wmfile.c	2013-01-18 16:57:47.957049469 +0100
@@ -11,6 +11,10 @@
  -----------------------------------------------------------------------
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <WINGs/WINGs.h>
 
 #include <unistd.h>
Index: WINGs/Tests/wmquery.c
===================================================================
--- WINGs/Tests/wmquery.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WINGs/Tests/wmquery.c	2013-01-18 16:57:47.957049469 +0100
@@ -3,6 +3,10 @@
  * Author: Len Trigg <trigg@cs.waikato.ac.nz>
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <WINGs/WINGs.h>
 
 #include <unistd.h>
Index: WINGs/Tests/wtest.c
===================================================================
--- WINGs/Tests/wtest.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WINGs/Tests/wtest.c	2013-01-18 16:57:47.957049469 +0100
@@ -2,6 +2,10 @@
  * WINGs test application
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <WINGs/WINGs.h>
 
 #include <stdio.h>
Index: WINGs/array.c
===================================================================
--- WINGs/array.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WINGs/array.c	2013-01-18 16:57:47.957049469 +0100
@@ -8,6 +8,10 @@
  * proper credit is always appreciated :)
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <stdlib.h>
 #include <string.h>
 
Index: WINGs/bagtree.c
===================================================================
--- WINGs/bagtree.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WINGs/bagtree.c	2013-01-18 16:57:47.957049469 +0100
@@ -1,4 +1,8 @@
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <stdlib.h>
 #include <string.h>
 
Index: WINGs/configuration.c
===================================================================
--- WINGs/configuration.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WINGs/configuration.c	2013-01-18 16:57:47.958049499 +0100
@@ -1,4 +1,8 @@
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "WINGsP.h"
 #include "wconfig.h"
 
Index: WINGs/data.c
===================================================================
--- WINGs/data.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WINGs/data.c	2013-01-18 16:57:47.958049499 +0100
@@ -18,6 +18,10 @@
  *  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <string.h>
 #include "WUtil.h"
 
Index: WINGs/dragcommon.c
===================================================================
--- WINGs/dragcommon.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WINGs/dragcommon.c	2013-01-18 16:57:47.958049499 +0100
@@ -1,4 +1,8 @@
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "WINGsP.h"
 
 #define XDND_SOURCE_VERSION(dragInfo) dragInfo->protocolVersion
Index: WINGs/dragdestination.c
===================================================================
--- WINGs/dragdestination.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WINGs/dragdestination.c	2013-01-18 16:57:47.958049499 +0100
@@ -1,4 +1,8 @@
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "WINGsP.h"
 #include <X11/Xatom.h>
 
Index: WINGs/dragsource.c
===================================================================
--- WINGs/dragsource.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WINGs/dragsource.c	2013-01-18 16:57:47.959049529 +0100
@@ -1,4 +1,8 @@
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "wconfig.h"
 #include "WINGsP.h"
 
Index: WINGs/error.c
===================================================================
--- WINGs/error.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WINGs/error.c	2013-01-18 16:57:47.959049529 +0100
@@ -18,6 +18,10 @@
  *  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "wconfig.h"
 
 #include <stdarg.h>
Index: WINGs/findfile.c
===================================================================
--- WINGs/findfile.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WINGs/findfile.c	2013-01-18 16:57:47.959049529 +0100
@@ -18,6 +18,10 @@
  *  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "wconfig.h"
 
 #include "WUtil.h"
Index: WINGs/handlers.c
===================================================================
--- WINGs/handlers.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WINGs/handlers.c	2013-01-18 16:57:47.959049529 +0100
@@ -3,6 +3,10 @@
  * WINGs internal handlers: timer, idle and input handlers
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "wconfig.h"
 #include "WINGsP.h"
 
Index: WINGs/hashtable.c
===================================================================
--- WINGs/hashtable.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WINGs/hashtable.c	2013-01-18 16:57:47.959049529 +0100
@@ -1,4 +1,8 @@
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <sys/types.h>
 #include <string.h>
 #include <stdlib.h>
Index: WINGs/memory.c
===================================================================
--- WINGs/memory.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WINGs/memory.c	2013-01-18 16:57:47.959049529 +0100
@@ -18,6 +18,10 @@
  *  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "wconfig.h"
 #include "WUtil.h"
 
Index: WINGs/misc.c
===================================================================
--- WINGs/misc.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WINGs/misc.c	2013-01-18 16:57:47.960049559 +0100
@@ -1,6 +1,10 @@
 
 /* Miscelaneous helper functions */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "WINGsP.h"
 
 WMRange wmkrange(int start, int count)
Index: WINGs/notification.c
===================================================================
--- WINGs/notification.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WINGs/notification.c	2013-01-18 16:57:47.960049559 +0100
@@ -1,4 +1,8 @@
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <stdlib.h>
 #include <assert.h>
 #include <stdio.h>
Index: WINGs/proplist.c
===================================================================
--- WINGs/proplist.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WINGs/proplist.c	2013-01-18 16:57:47.960049559 +0100
@@ -1,4 +1,8 @@
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <sys/types.h>
 #include <sys/stat.h>
 
Index: WINGs/selection.c
===================================================================
--- WINGs/selection.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WINGs/selection.c	2013-01-18 16:57:47.960049559 +0100
@@ -1,4 +1,8 @@
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <stdlib.h>
 
 #include <X11/Xatom.h>
Index: WINGs/string.c
===================================================================
--- WINGs/string.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WINGs/string.c	2013-01-18 16:57:47.961049589 +0100
@@ -6,6 +6,10 @@
 #   undef _XOPEN_SOURCE
 #endif
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "wconfig.h"
 
 #include <string.h>
Index: WINGs/tree.c
===================================================================
--- WINGs/tree.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WINGs/tree.c	2013-01-18 16:57:47.961049589 +0100
@@ -1,4 +1,8 @@
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <string.h>
 
 #include "WUtil.h"
Index: WINGs/userdefaults.c
===================================================================
--- WINGs/userdefaults.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WINGs/userdefaults.c	2013-01-18 16:57:47.961049589 +0100
@@ -1,4 +1,8 @@
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <stdlib.h>
 #include <string.h>
 #include <strings.h>
Index: WINGs/usleep.c
===================================================================
--- WINGs/usleep.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WINGs/usleep.c	2013-01-18 16:57:47.961049589 +0100
@@ -1,4 +1,8 @@
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <errno.h>
 #include <time.h>
 
Index: WINGs/wapplication.c
===================================================================
--- WINGs/wapplication.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WINGs/wapplication.c	2013-01-18 16:57:47.961049589 +0100
@@ -1,4 +1,8 @@
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <unistd.h>
 #include <X11/Xlocale.h>
 
Index: WINGs/wappresource.c
===================================================================
--- WINGs/wappresource.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WINGs/wappresource.c	2013-01-18 16:57:47.961049589 +0100
@@ -1,4 +1,8 @@
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <unistd.h>
 
 #include "WINGsP.h"
Index: WINGs/wballoon.c
===================================================================
--- WINGs/wballoon.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WINGs/wballoon.c	2013-01-18 16:57:47.962049620 +0100
@@ -1,4 +1,8 @@
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "wconfig.h"
 #include "WINGsP.h"
 
Index: WINGs/wbox.c
===================================================================
--- WINGs/wbox.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WINGs/wbox.c	2013-01-18 16:57:47.962049620 +0100
@@ -1,4 +1,8 @@
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "WINGsP.h"
 
 typedef struct {
Index: WINGs/wbrowser.c
===================================================================
--- WINGs/wbrowser.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WINGs/wbrowser.c	2013-01-18 16:57:47.962049620 +0100
@@ -1,4 +1,8 @@
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "WINGsP.h"
 #include <math.h>		/* for : double rint (double) */
 
Index: WINGs/wbutton.c
===================================================================
--- WINGs/wbutton.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WINGs/wbutton.c	2013-01-18 16:57:47.962049620 +0100
@@ -1,4 +1,8 @@
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "WINGsP.h"
 
 typedef struct W_Button {
Index: WINGs/wcolor.c
===================================================================
--- WINGs/wcolor.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WINGs/wcolor.c	2013-01-18 16:57:47.962049620 +0100
@@ -1,4 +1,8 @@
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "WINGsP.h"
 
 #include "wconfig.h"
Index: WINGs/wcolorpanel.c
===================================================================
--- WINGs/wcolorpanel.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WINGs/wcolorpanel.c	2013-01-18 16:57:47.963049651 +0100
@@ -23,6 +23,10 @@
  * 	-	Resizing
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "wconfig.h"
 #include "WINGsP.h"
 #include "rgb.h"
Index: WINGs/wcolorwell.c
===================================================================
--- WINGs/wcolorwell.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WINGs/wcolorwell.c	2013-01-18 16:57:47.964049682 +0100
@@ -1,4 +1,8 @@
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "WINGsP.h"
 
 #define XDND_COLOR_DATA_TYPE "application/X-color"
Index: WINGs/wevent.c
===================================================================
--- WINGs/wevent.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WINGs/wevent.c	2013-01-18 16:57:47.964049682 +0100
@@ -3,6 +3,10 @@
  * This event handling stuff was inspired on Tk.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "WINGsP.h"
 
 /* table to map event types to event masks */
Index: WINGs/wfilepanel.c
===================================================================
--- WINGs/wfilepanel.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WINGs/wfilepanel.c	2013-01-18 16:57:47.964049682 +0100
@@ -1,4 +1,8 @@
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <sys/types.h>
 #include <sys/stat.h>
 
Index: WINGs/wfont.c
===================================================================
--- WINGs/wfont.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WINGs/wfont.c	2013-01-18 16:57:47.964049682 +0100
@@ -1,4 +1,8 @@
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <stdlib.h>
 
 #include "wconfig.h"
Index: WINGs/wfontpanel.c
===================================================================
--- WINGs/wfontpanel.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WINGs/wfontpanel.c	2013-01-18 16:57:47.964049682 +0100
@@ -1,4 +1,8 @@
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "WINGsP.h"
 #include "WUtil.h"
 #include "wconfig.h"
Index: WINGs/wframe.c
===================================================================
--- WINGs/wframe.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WINGs/wframe.c	2013-01-18 16:57:47.965049713 +0100
@@ -1,4 +1,8 @@
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "WINGsP.h"
 
 typedef struct W_Frame {
Index: WINGs/widgets.c
===================================================================
--- WINGs/widgets.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WINGs/widgets.c	2013-01-18 16:57:47.965049713 +0100
@@ -1,4 +1,8 @@
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "WINGsP.h"
 #include "wconfig.h"
 
Index: WINGs/winputmethod.c
===================================================================
--- WINGs/winputmethod.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WINGs/winputmethod.c	2013-01-18 16:57:47.965049713 +0100
@@ -1,4 +1,8 @@
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <X11/Xlib.h>
 
 #include "WINGsP.h"
Index: WINGs/wlabel.c
===================================================================
--- WINGs/wlabel.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WINGs/wlabel.c	2013-01-18 16:57:47.965049713 +0100
@@ -1,4 +1,8 @@
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "WINGsP.h"
 
 typedef struct W_Label {
Index: WINGs/wlist.c
===================================================================
--- WINGs/wlist.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WINGs/wlist.c	2013-01-18 16:57:47.965049713 +0100
@@ -1,4 +1,8 @@
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "WINGsP.h"
 
 char *WMListDidScrollNotification = "WMListDidScrollNotification";
Index: WINGs/wmenuitem.c
===================================================================
--- WINGs/wmenuitem.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WINGs/wmenuitem.c	2013-01-18 16:57:47.966049743 +0100
@@ -1,4 +1,8 @@
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "WINGsP.h"
 
 typedef struct W_MenuItem {
Index: WINGs/wmisc.c
===================================================================
--- WINGs/wmisc.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WINGs/wmisc.c	2013-01-18 16:57:47.966049743 +0100
@@ -1,4 +1,8 @@
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "WINGsP.h"
 
 #include <wraster.h>
Index: WINGs/wpanel.c
===================================================================
--- WINGs/wpanel.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WINGs/wpanel.c	2013-01-18 16:57:47.966049743 +0100
@@ -1,4 +1,8 @@
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "WINGsP.h"
 
 #include <X11/keysym.h>
Index: WINGs/wpixmap.c
===================================================================
--- WINGs/wpixmap.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WINGs/wpixmap.c	2013-01-18 16:57:47.966049743 +0100
@@ -1,4 +1,8 @@
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "WINGsP.h"
 
 #include <wraster.h>
Index: WINGs/wpopupbutton.c
===================================================================
--- WINGs/wpopupbutton.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WINGs/wpopupbutton.c	2013-01-18 16:57:47.966049743 +0100
@@ -1,4 +1,8 @@
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "WINGsP.h"
 
 typedef struct W_PopUpButton {
Index: WINGs/wprogressindicator.c
===================================================================
--- WINGs/wprogressindicator.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WINGs/wprogressindicator.c	2013-01-18 16:57:47.967049773 +0100
@@ -5,6 +5,10 @@
  * - centralized drawing into one pain function
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "WINGsP.h"
 
 typedef struct W_ProgressIndicator {
Index: WINGs/wruler.c
===================================================================
--- WINGs/wruler.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WINGs/wruler.c	2013-01-18 16:57:47.967049773 +0100
@@ -18,6 +18,10 @@
  *  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "WINGsP.h"
 #include "wconfig.h"
 
Index: WINGs/wscroller.c
===================================================================
--- WINGs/wscroller.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WINGs/wscroller.c	2013-01-18 16:57:47.967049773 +0100
@@ -7,6 +7,10 @@
 #   undef _XOPEN_SOURCE
 #endif
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "WINGsP.h"
 
 #include <math.h>
Index: WINGs/wscrollview.c
===================================================================
--- WINGs/wscrollview.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WINGs/wscrollview.c	2013-01-18 16:57:47.967049773 +0100
@@ -1,4 +1,8 @@
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "WINGsP.h"
 
 typedef struct W_ScrollView {
Index: WINGs/wslider.c
===================================================================
--- WINGs/wslider.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WINGs/wslider.c	2013-01-18 16:57:47.968049802 +0100
@@ -1,4 +1,8 @@
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "WINGsP.h"
 
 #undef STRICT_NEXT_BEHAVIOUR
Index: WINGs/wsplitview.c
===================================================================
--- WINGs/wsplitview.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WINGs/wsplitview.c	2013-01-18 16:57:47.968049802 +0100
@@ -1,4 +1,8 @@
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "WINGsP.h"
 
 /*
Index: WINGs/wtabview.c
===================================================================
--- WINGs/wtabview.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WINGs/wtabview.c	2013-01-18 16:57:47.968049802 +0100
@@ -1,4 +1,8 @@
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "WINGsP.h"
 
 typedef struct W_TabView {
Index: WINGs/wtext.c
===================================================================
--- WINGs/wtext.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WINGs/wtext.c	2013-01-18 16:57:47.969049831 +0100
@@ -1,6 +1,10 @@
 
 /*  WINGs WMText: multi-line/font/color/graphic text widget,  by Nwanua. */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "WINGsP.h"
 #include <ctype.h>
 #include <X11/keysym.h>
Index: WINGs/wtextfield.c
===================================================================
--- WINGs/wtextfield.c.orig	2013-01-09 22:42:49.000000000 +0100
+++ WINGs/wtextfield.c	2013-01-18 16:57:47.969049831 +0100
@@ -1,4 +1,8 @@
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "WINGsP.h"
 #include "wconfig.h"
 
Index: WINGs/wutil.c
===================================================================
--- WINGs/wutil.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WINGs/wutil.c	2013-01-18 16:57:47.969049831 +0100
@@ -3,6 +3,10 @@
  * Handle events for non-GUI based applications
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "WINGsP.h"
 
 void WHandleEvents()
Index: WINGs/wview.c
===================================================================
--- WINGs/wview.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WINGs/wview.c	2013-01-18 16:57:47.970049862 +0100
@@ -1,4 +1,8 @@
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "WINGsP.h"
 
 #include <X11/Xresource.h>
Index: WINGs/wwindow.c
===================================================================
--- WINGs/wwindow.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WINGs/wwindow.c	2013-01-18 16:57:47.970049862 +0100
@@ -1,4 +1,8 @@
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <X11/Xmd.h>
 
 #include "WINGsP.h"
Index: WPrefs.app/Appearance.c
===================================================================
--- WPrefs.app/Appearance.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WPrefs.app/Appearance.c	2013-01-18 16:57:47.970049862 +0100
@@ -19,6 +19,10 @@
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "WPrefs.h"
 
 #include <unistd.h>
Index: WPrefs.app/Configurations.c
===================================================================
--- WPrefs.app/Configurations.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WPrefs.app/Configurations.c	2013-01-18 16:57:47.971049894 +0100
@@ -19,6 +19,10 @@
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "WPrefs.h"
 
 typedef struct _Panel {
Index: WPrefs.app/Expert.c
===================================================================
--- WPrefs.app/Expert.c.orig	2013-01-09 22:42:49.000000000 +0100
+++ WPrefs.app/Expert.c	2013-01-18 16:57:47.971049894 +0100
@@ -19,6 +19,10 @@
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "WPrefs.h"
 
 /* This structure containts the list of all the check-buttons to display in the
Index: WPrefs.app/Focus.c
===================================================================
--- WPrefs.app/Focus.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WPrefs.app/Focus.c	2013-01-18 16:57:47.971049894 +0100
@@ -19,6 +19,10 @@
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "WPrefs.h"
 
 typedef struct _Panel {
Index: WPrefs.app/FontSimple.c
===================================================================
--- WPrefs.app/FontSimple.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WPrefs.app/FontSimple.c	2013-01-18 16:57:47.971049894 +0100
@@ -19,6 +19,10 @@
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "WPrefs.h"
 #include <unistd.h>
 #include <fontconfig/fontconfig.h>
Index: WPrefs.app/Icons.c
===================================================================
--- WPrefs.app/Icons.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WPrefs.app/Icons.c	2013-01-18 16:57:47.971049894 +0100
@@ -19,6 +19,10 @@
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "WPrefs.h"
 
 typedef struct _Panel {
Index: WPrefs.app/KeyboardSettings.c
===================================================================
--- WPrefs.app/KeyboardSettings.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WPrefs.app/KeyboardSettings.c	2013-01-18 16:57:47.972049925 +0100
@@ -19,6 +19,10 @@
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "WPrefs.h"
 
 typedef struct _Panel {
Index: WPrefs.app/Menu.c
===================================================================
--- WPrefs.app/Menu.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WPrefs.app/Menu.c	2013-01-18 16:57:47.972049925 +0100
@@ -19,6 +19,10 @@
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "WPrefs.h"
 #include <assert.h>
 #include <ctype.h>
Index: WPrefs.app/MenuPreferences.c
===================================================================
--- WPrefs.app/MenuPreferences.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WPrefs.app/MenuPreferences.c	2013-01-18 16:57:47.972049925 +0100
@@ -19,6 +19,10 @@
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "WPrefs.h"
 
 typedef struct _Panel {
Index: WPrefs.app/MouseSettings.c
===================================================================
--- WPrefs.app/MouseSettings.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WPrefs.app/MouseSettings.c	2013-01-18 16:57:47.972049925 +0100
@@ -20,6 +20,10 @@
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "WPrefs.h"
 
 #include <X11/Xutil.h>
Index: WPrefs.app/Paths.c
===================================================================
--- WPrefs.app/Paths.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WPrefs.app/Paths.c	2013-01-18 16:57:47.973049956 +0100
@@ -19,6 +19,10 @@
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "WPrefs.h"
 #include <unistd.h>
 #include <assert.h>
Index: WPrefs.app/Preferences.c
===================================================================
--- WPrefs.app/Preferences.c.orig	2013-01-09 22:42:49.000000000 +0100
+++ WPrefs.app/Preferences.c	2013-01-18 16:57:47.973049956 +0100
@@ -19,6 +19,10 @@
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "WPrefs.h"
 
 typedef struct _Panel {
Index: WPrefs.app/TexturePanel.c
===================================================================
--- WPrefs.app/TexturePanel.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WPrefs.app/TexturePanel.c	2013-01-18 16:57:47.973049956 +0100
@@ -20,6 +20,10 @@
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
Index: WPrefs.app/Themes.c
===================================================================
--- WPrefs.app/Themes.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WPrefs.app/Themes.c	2013-01-18 16:57:47.973049956 +0100
@@ -19,6 +19,10 @@
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "WPrefs.h"
 
 #include <unistd.h>
Index: WPrefs.app/WPrefs.c
===================================================================
--- WPrefs.app/WPrefs.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WPrefs.app/WPrefs.c	2013-01-18 16:57:47.974049987 +0100
@@ -19,6 +19,10 @@
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "WPrefs.h"
 #include <assert.h>
 
Index: WPrefs.app/WindowHandling.c
===================================================================
--- WPrefs.app/WindowHandling.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WPrefs.app/WindowHandling.c	2013-01-18 16:57:47.974049987 +0100
@@ -19,6 +19,10 @@
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "WPrefs.h"
 
 typedef struct _Panel {
Index: WPrefs.app/Workspace.c
===================================================================
--- WPrefs.app/Workspace.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WPrefs.app/Workspace.c	2013-01-18 16:57:47.974049987 +0100
@@ -19,6 +19,10 @@
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "WPrefs.h"
 
 typedef struct _Panel {
Index: WPrefs.app/double.c
===================================================================
--- WPrefs.app/double.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WPrefs.app/double.c	2013-01-18 16:57:47.974049987 +0100
@@ -4,6 +4,10 @@
  *
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <WINGs/WINGsP.h>
 
 #include "double.h"
Index: WPrefs.app/editmenu.c
===================================================================
--- WPrefs.app/editmenu.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WPrefs.app/editmenu.c	2013-01-18 16:57:47.975050017 +0100
@@ -19,6 +19,10 @@
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <WINGs/WINGsP.h>
 #include <WINGs/WUtil.h>
 #include <stdlib.h>
Index: WPrefs.app/main.c
===================================================================
--- WPrefs.app/main.c.orig	2013-01-09 22:42:38.000000000 +0100
+++ WPrefs.app/main.c	2013-01-18 16:57:47.975050017 +0100
@@ -18,6 +18,10 @@
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "WPrefs.h"
 
 #include <assert.h>
Index: WPrefs.app/xmodifier.c
===================================================================
--- WPrefs.app/xmodifier.c.orig	2013-01-09 22:42:39.000000000 +0100
+++ WPrefs.app/xmodifier.c	2013-01-18 16:57:47.975050017 +0100
@@ -31,6 +31,10 @@ Perpetrator: Sudish Joseph <sj@eng.minds
  Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
  Boston, MA 02110-1301 USA. */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <string.h>
 #include <strings.h>
 #include <X11/Xlib.h>
Index: src/actions.c
===================================================================
--- src/actions.c.orig	2013-01-09 22:42:49.000000000 +0100
+++ src/actions.c	2013-01-18 16:57:47.975050017 +0100
@@ -20,6 +20,10 @@
  *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "wconfig.h"
 
 #include <X11/Xlib.h>
Index: src/appicon.c
===================================================================
--- src/appicon.c.orig	2013-01-09 22:42:49.000000000 +0100
+++ src/appicon.c	2013-01-18 16:57:47.976050047 +0100
@@ -20,6 +20,10 @@
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "wconfig.h"
 
 #include <X11/Xlib.h>
Index: src/application.c
===================================================================
--- src/application.c.orig	2013-01-09 22:42:49.000000000 +0100
+++ src/application.c	2013-01-18 16:57:47.976050047 +0100
@@ -18,6 +18,10 @@
  *  with this program; if not, write to the Free Software Foundation, Inc.,
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "wconfig.h"
 
 #include <X11/Xlib.h>
Index: src/appmenu.c
===================================================================
--- src/appmenu.c.orig	2013-01-09 22:42:39.000000000 +0100
+++ src/appmenu.c	2013-01-18 16:57:47.976050047 +0100
@@ -19,6 +19,10 @@
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "wconfig.h"
 
 #include <X11/Xlib.h>
Index: src/balloon.c
===================================================================
--- src/balloon.c.orig	2013-01-09 22:42:39.000000000 +0100
+++ src/balloon.c	2013-01-18 16:57:47.976050047 +0100
@@ -18,6 +18,10 @@
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "wconfig.h"
 
 #ifdef BALLOON_TEXT
Index: src/client.c
===================================================================
--- src/client.c.orig	2013-01-09 22:42:49.000000000 +0100
+++ src/client.c	2013-01-18 16:57:47.977050077 +0100
@@ -17,6 +17,10 @@
  *  with this program; if not, write to the Free Software Foundation, Inc.,
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "wconfig.h"
 
 #include <X11/Xlib.h>
Index: src/colormap.c
===================================================================
--- src/colormap.c.orig	2013-01-09 22:42:39.000000000 +0100
+++ src/colormap.c	2013-01-18 16:57:47.977050077 +0100
@@ -23,6 +23,10 @@
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "wconfig.h"
 
 #include "WindowMaker.h"
Index: src/cycling.c
===================================================================
--- src/cycling.c.orig	2013-01-09 22:42:49.000000000 +0100
+++ src/cycling.c	2013-01-18 16:57:47.977050077 +0100
@@ -19,6 +19,10 @@
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "wconfig.h"
 
 #include <stdlib.h>
Index: src/defaults.c
===================================================================
--- src/defaults.c.orig	2013-01-09 22:42:49.000000000 +0100
+++ src/defaults.c	2013-01-18 16:57:47.978050108 +0100
@@ -20,6 +20,10 @@
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "wconfig.h"
 
 #include <stdio.h>
Index: src/dialog.c
===================================================================
--- src/dialog.c.orig	2013-01-09 22:42:39.000000000 +0100
+++ src/dialog.c	2013-01-18 16:57:47.978050108 +0100
@@ -20,6 +20,10 @@
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "wconfig.h"
 
 #include <X11/Xlib.h>
Index: src/dock.c
===================================================================
--- src/dock.c.orig	2013-01-09 22:42:49.000000000 +0100
+++ src/dock.c	2013-01-18 16:57:47.979050138 +0100
@@ -20,6 +20,10 @@
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "wconfig.h"
 
 #include <X11/Xlib.h>
Index: src/dockedapp.c
===================================================================
--- src/dockedapp.c.orig	2013-01-09 22:42:39.000000000 +0100
+++ src/dockedapp.c	2013-01-18 16:57:47.979050138 +0100
@@ -19,6 +19,10 @@
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "wconfig.h"
 
 #include <X11/Xlib.h>
Index: src/event.c
===================================================================
--- src/event.c.orig	2013-01-09 22:42:39.000000000 +0100
+++ src/event.c	2013-01-18 16:57:47.980050168 +0100
@@ -19,6 +19,10 @@
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "wconfig.h"
 
 #ifdef HAVE_INOTIFY
Index: src/framewin.c
===================================================================
--- src/framewin.c.orig	2013-01-09 22:42:39.000000000 +0100
+++ src/framewin.c	2013-01-18 16:57:47.980050168 +0100
@@ -18,6 +18,10 @@
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "wconfig.h"
 
 #include <X11/Xlib.h>
Index: src/geomview.c
===================================================================
--- src/geomview.c.orig	2013-01-09 22:42:39.000000000 +0100
+++ src/geomview.c	2013-01-18 16:57:47.980050168 +0100
@@ -1,4 +1,8 @@
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <WINGs/WINGsP.h>
 
 typedef struct W_GeometryView {
Index: src/icon.c
===================================================================
--- src/icon.c.orig	2013-01-09 22:42:49.000000000 +0100
+++ src/icon.c	2013-01-18 16:57:47.980050168 +0100
@@ -19,6 +19,10 @@
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "wconfig.h"
 
 #include <X11/Xlib.h>
Index: src/main.c
===================================================================
--- src/main.c.orig	2013-01-09 22:42:49.000000000 +0100
+++ src/main.c	2013-01-18 16:57:47.981050198 +0100
@@ -18,6 +18,10 @@
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "wconfig.h"
 
 #ifdef HAVE_INOTIFY
Index: src/menu.c
===================================================================
--- src/menu.c.orig	2013-01-09 22:42:39.000000000 +0100
+++ src/menu.c	2013-01-18 16:57:47.981050198 +0100
@@ -20,6 +20,10 @@
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "wconfig.h"
 
 #include <X11/Xlib.h>
Index: src/misc.c
===================================================================
--- src/misc.c.orig	2013-01-09 22:42:39.000000000 +0100
+++ src/misc.c	2013-01-18 16:57:47.981050198 +0100
@@ -17,6 +17,10 @@
  *  with this program; if not, write to the Free Software Foundation, Inc.,
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "wconfig.h"
 
 #include <X11/Xlib.h>
Index: src/monitor.c
===================================================================
--- src/monitor.c.orig	2013-01-09 22:42:39.000000000 +0100
+++ src/monitor.c	2013-01-18 16:57:47.982050228 +0100
@@ -19,6 +19,10 @@
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "wconfig.h"
 #include <unistd.h>
 #include <stdlib.h>
Index: src/motif.c
===================================================================
--- src/motif.c.orig	2013-01-09 22:42:39.000000000 +0100
+++ src/motif.c	2013-01-18 16:57:47.982050228 +0100
@@ -19,6 +19,10 @@
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "wconfig.h"
 
 #ifdef MWM_HINTS
Index: src/moveres.c
===================================================================
--- src/moveres.c.orig	2013-01-09 22:42:49.000000000 +0100
+++ src/moveres.c	2013-01-18 16:57:47.982050228 +0100
@@ -18,6 +18,10 @@
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "wconfig.h"
 
 #include <X11/Xlib.h>
Index: src/osdep_bsd.c
===================================================================
--- src/osdep_bsd.c.orig	2013-01-09 22:42:39.000000000 +0100
+++ src/osdep_bsd.c	2013-01-18 16:57:47.983050258 +0100
@@ -11,6 +11,10 @@
 #else /* OPENBSD || NETBSD */
 #   include <sys/param.h>
 #endif
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <sys/sysctl.h>
 
 #include <assert.h>
Index: src/osdep_darwin.c
===================================================================
--- src/osdep_darwin.c.orig	2013-01-09 22:42:39.000000000 +0100
+++ src/osdep_darwin.c	2013-01-18 16:57:47.983050258 +0100
@@ -1,4 +1,8 @@
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <sys/types.h>
 #include <sys/sysctl.h>
 
Index: src/osdep_linux.c
===================================================================
--- src/osdep_linux.c.orig	2013-01-09 22:42:39.000000000 +0100
+++ src/osdep_linux.c	2013-01-18 16:57:47.983050258 +0100
@@ -1,4 +1,8 @@
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <sys/types.h>
 #include <sys/stat.h>
 
Index: src/osdep_stub.c
===================================================================
--- src/osdep_stub.c.orig	2013-01-09 22:42:39.000000000 +0100
+++ src/osdep_stub.c	2013-01-18 16:57:47.983050258 +0100
@@ -1,4 +1,8 @@
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <sys/utsname.h>
 
 #include <WINGs/WUtil.h>
Index: src/pixmap.c
===================================================================
--- src/pixmap.c.orig	2013-01-09 22:42:39.000000000 +0100
+++ src/pixmap.c	2013-01-18 16:57:47.983050258 +0100
@@ -18,6 +18,10 @@
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "wconfig.h"
 
 #include <X11/Xlib.h>
Index: src/placement.c
===================================================================
--- src/placement.c.orig	2013-01-09 22:42:39.000000000 +0100
+++ src/placement.c	2013-01-18 16:57:47.983050258 +0100
@@ -19,6 +19,10 @@
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "wconfig.h"
 
 #include <X11/Xlib.h>
Index: src/properties.c
===================================================================
--- src/properties.c.orig	2013-01-09 22:42:39.000000000 +0100
+++ src/properties.c	2013-01-18 16:57:47.983050258 +0100
@@ -18,6 +18,10 @@
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "wconfig.h"
 
 #include <X11/Xlib.h>
Index: src/resources.c
===================================================================
--- src/resources.c.orig	2013-01-09 22:42:39.000000000 +0100
+++ src/resources.c	2013-01-18 16:57:47.984050288 +0100
@@ -19,6 +19,10 @@
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "wconfig.h"
 
 #include <stdio.h>
Index: src/rootmenu.c
===================================================================
--- src/rootmenu.c.orig	2013-01-09 22:42:39.000000000 +0100
+++ src/rootmenu.c	2013-01-18 16:57:47.984050288 +0100
@@ -20,6 +20,10 @@
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "wconfig.h"
 
 #include <stdio.h>
Index: src/screen.c
===================================================================
--- src/screen.c.orig	2013-01-09 22:42:49.000000000 +0100
+++ src/screen.c	2013-01-18 16:57:47.984050288 +0100
@@ -19,6 +19,10 @@
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "wconfig.h"
 
 #include <stdio.h>
Index: src/session.c
===================================================================
--- src/session.c.orig	2013-01-09 22:42:39.000000000 +0100
+++ src/session.c	2013-01-18 16:57:47.985050319 +0100
@@ -57,6 +57,10 @@
  * if the Die message is received, exit.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "wconfig.h"
 
 #include <X11/Xlib.h>
Index: src/shutdown.c
===================================================================
--- src/shutdown.c.orig	2013-01-09 22:42:39.000000000 +0100
+++ src/shutdown.c	2013-01-18 16:57:47.985050319 +0100
@@ -18,6 +18,10 @@
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "wconfig.h"
 
 #include <stdlib.h>
Index: src/stacking.c
===================================================================
--- src/stacking.c.orig	2013-01-09 22:42:39.000000000 +0100
+++ src/stacking.c	2013-01-18 16:57:47.985050319 +0100
@@ -19,6 +19,10 @@
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "wconfig.h"
 
 #include <stdio.h>
Index: src/startup.c
===================================================================
--- src/startup.c.orig	2013-01-09 22:42:39.000000000 +0100
+++ src/startup.c	2013-01-18 16:57:47.985050319 +0100
@@ -19,6 +19,10 @@
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "wconfig.h"
 
 #include <stdio.h>
Index: src/superfluous.c
===================================================================
--- src/superfluous.c.orig	2013-01-09 22:42:39.000000000 +0100
+++ src/superfluous.c	2013-01-18 16:57:47.985050319 +0100
@@ -19,6 +19,10 @@
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "wconfig.h"
 
 #include <X11/Xlib.h>
Index: src/switchmenu.c
===================================================================
--- src/switchmenu.c.orig	2013-01-09 22:42:39.000000000 +0100
+++ src/switchmenu.c	2013-01-18 16:57:47.986050350 +0100
@@ -19,6 +19,10 @@
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "wconfig.h"
 
 #include <stdio.h>
Index: src/switchpanel.c
===================================================================
--- src/switchpanel.c.orig	2013-01-09 22:42:49.000000000 +0100
+++ src/switchpanel.c	2013-01-18 16:57:47.986050350 +0100
@@ -18,6 +18,10 @@
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "wconfig.h"
 
 #include <stdlib.h>
Index: src/texture.c
===================================================================
--- src/texture.c.orig	2013-01-09 22:42:39.000000000 +0100
+++ src/texture.c	2013-01-18 16:57:47.986050350 +0100
@@ -18,6 +18,10 @@
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "wconfig.h"
 
 #include <X11/Xlib.h>
Index: src/usermenu.c
===================================================================
--- src/usermenu.c.orig	2013-01-09 22:42:39.000000000 +0100
+++ src/usermenu.c	2013-01-18 16:57:47.986050350 +0100
@@ -56,6 +56,10 @@
  *
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "wconfig.h"
 
 #ifdef USER_MENU
Index: src/wcore.c
===================================================================
--- src/wcore.c.orig	2013-01-09 22:42:39.000000000 +0100
+++ src/wcore.c	2013-01-18 16:57:47.986050350 +0100
@@ -18,6 +18,10 @@
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "wconfig.h"
 
 #include <X11/Xlib.h>
Index: src/wdefaults.c
===================================================================
--- src/wdefaults.c.orig	2013-01-09 22:42:39.000000000 +0100
+++ src/wdefaults.c	2013-01-18 16:57:47.987050381 +0100
@@ -19,6 +19,10 @@
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "wconfig.h"
 
 #include <stdio.h>
Index: src/window.c
===================================================================
--- src/window.c.orig	2013-01-09 22:42:49.000000000 +0100
+++ src/window.c	2013-01-18 16:57:47.987050381 +0100
@@ -19,6 +19,10 @@
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "wconfig.h"
 
 #include <X11/Xlib.h>
Index: src/winmenu.c
===================================================================
--- src/winmenu.c.orig	2013-01-09 22:42:39.000000000 +0100
+++ src/winmenu.c	2013-01-18 16:57:47.988050412 +0100
@@ -19,6 +19,10 @@
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "wconfig.h"
 
 #include <stdio.h>
Index: src/winspector.c
===================================================================
--- src/winspector.c.orig	2013-01-09 22:42:49.000000000 +0100
+++ src/winspector.c	2013-01-18 16:57:47.988050412 +0100
@@ -20,6 +20,10 @@
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "wconfig.h"
 
 #include <X11/Xlib.h>
Index: src/wmspec.c
===================================================================
--- src/wmspec.c.orig	2013-01-09 22:42:49.000000000 +0100
+++ src/wmspec.c	2013-01-18 16:57:47.989050442 +0100
@@ -28,6 +28,10 @@
  * return to be Success is not enough. -Dan
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "wconfig.h"
 
 #include <X11/Xlib.h>
Index: src/workspace.c
===================================================================
--- src/workspace.c.orig	2013-01-09 22:42:39.000000000 +0100
+++ src/workspace.c	2013-01-18 16:57:47.989050442 +0100
@@ -18,6 +18,10 @@
  *  with this program; if not, write to the Free Software Foundation, Inc.,
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "wconfig.h"
 
 #include <X11/Xlib.h>
Index: src/xdnd.c
===================================================================
--- src/xdnd.c.orig	2013-01-09 22:42:39.000000000 +0100
+++ src/xdnd.c	2013-01-18 16:57:47.989050442 +0100
@@ -1,5 +1,9 @@
 /* Many part of code are ripped of an example from JX's site */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "wconfig.h"
 
 #ifdef XDND
Index: src/xinerama.c
===================================================================
--- src/xinerama.c.orig	2013-01-09 22:42:39.000000000 +0100
+++ src/xinerama.c	2013-01-18 16:57:47.989050442 +0100
@@ -18,6 +18,10 @@
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "wconfig.h"
 
 #include "xinerama.h"
Index: src/xmodifier.c
===================================================================
--- src/xmodifier.c.orig	2013-01-09 22:42:39.000000000 +0100
+++ src/xmodifier.c	2013-01-18 16:57:47.989050442 +0100
@@ -27,6 +27,10 @@ Perpetrator: Sudish Joseph <sj@eng.minds
  Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
  Boston, MA 02110-1301 USA. */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "wconfig.h"
 
 #include <string.h>
Index: src/xutil.c
===================================================================
--- src/xutil.c.orig	2013-01-09 22:42:39.000000000 +0100
+++ src/xutil.c	2013-01-18 16:57:47.990050472 +0100
@@ -18,6 +18,10 @@
  *  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <X11/Xlib.h>
 #include <stdlib.h>
 #include <X11/Xutil.h>
Index: test/notest.c
===================================================================
--- test/notest.c.orig	2013-01-09 22:42:39.000000000 +0100
+++ test/notest.c	2013-01-18 16:57:47.990050472 +0100
@@ -3,6 +3,10 @@
  * TODO: remake
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <stdio.h>
 #include <X11/Xlib.h>
 #include <X11/Xutil.h>
Index: test/wtest.c
===================================================================
--- test/wtest.c.orig	2013-01-09 22:42:39.000000000 +0100
+++ test/wtest.c	2013-01-18 16:57:47.990050472 +0100
@@ -8,6 +8,10 @@
  * TODO: remake
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <stdio.h>
 #include <stdlib.h>
 #include <X11/Xlib.h>
Index: util/convertfonts.c
===================================================================
--- util/convertfonts.c.orig	2013-01-09 22:42:39.000000000 +0100
+++ util/convertfonts.c	2013-01-18 16:57:47.990050472 +0100
@@ -23,6 +23,10 @@
 #define _GNU_SOURCE		/* getopt_long */
 #endif
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <sys/stat.h>
 
 #include <getopt.h>
Index: util/fontconv.c
===================================================================
--- util/fontconv.c.orig	2013-01-09 22:42:39.000000000 +0100
+++ util/fontconv.c	2013-01-18 16:57:47.990050472 +0100
@@ -1,4 +1,8 @@
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
Index: util/geticonset.c
===================================================================
--- util/geticonset.c.orig	2013-01-09 22:42:39.000000000 +0100
+++ util/geticonset.c	2013-01-18 16:57:47.990050472 +0100
@@ -23,6 +23,10 @@
 #define _GNU_SOURCE		/* getopt_long */
 #endif
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <getopt.h>
 #include <stdio.h>
 #include <stdlib.h>
Index: util/getstyle.c
===================================================================
--- util/getstyle.c.orig	2013-01-09 22:42:39.000000000 +0100
+++ util/getstyle.c	2013-01-18 16:57:47.991050502 +0100
@@ -23,6 +23,10 @@
 #define _GNU_SOURCE		/* getopt_long */
 #endif
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <sys/types.h>
 #include <sys/stat.h>
 
Index: util/seticons.c
===================================================================
--- util/seticons.c.orig	2013-01-09 22:42:39.000000000 +0100
+++ util/seticons.c	2013-01-18 16:57:47.991050502 +0100
@@ -23,6 +23,10 @@
 #define _GNU_SOURCE		/* getopt_long */
 #endif
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <getopt.h>
 #include <stdio.h>
 #include <stdlib.h>
Index: util/setstyle.c
===================================================================
--- util/setstyle.c.orig	2013-01-09 22:42:39.000000000 +0100
+++ util/setstyle.c	2013-01-18 16:57:47.991050502 +0100
@@ -23,6 +23,10 @@
 #define _GNU_SOURCE		/* getopt_long */
 #endif
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <sys/stat.h>
 
 #include <getopt.h>
Index: util/wdread.c
===================================================================
--- util/wdread.c.orig	2013-01-09 22:42:39.000000000 +0100
+++ util/wdread.c	2013-01-18 16:57:47.991050502 +0100
@@ -28,6 +28,10 @@
  * WindowMaker defaults DB reader
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <getopt.h>
 #include <limits.h>
 #include <stdio.h>
Index: util/wdwrite.c
===================================================================
--- util/wdwrite.c.orig	2013-01-09 22:42:39.000000000 +0100
+++ util/wdwrite.c	2013-01-18 16:57:47.991050502 +0100
@@ -28,6 +28,10 @@
  */
 
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <getopt.h>
 #include <limits.h>
 #include <stdio.h>
Index: util/wmagnify.c
===================================================================
--- util/wmagnify.c.orig	2013-01-09 22:42:39.000000000 +0100
+++ util/wmagnify.c	2013-01-18 16:57:47.991050502 +0100
@@ -6,6 +6,10 @@
  * This program is in the Public Domain.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <X11/Xproto.h>
 
 #include <WINGs/WINGs.h>
Index: util/wmgenmenu.c
===================================================================
--- util/wmgenmenu.c.orig	2013-01-09 22:42:39.000000000 +0100
+++ util/wmgenmenu.c	2013-01-18 16:57:47.992050532 +0100
@@ -4,6 +4,10 @@
 #define _GNU_SOURCE		/* getopt_long */
 #endif
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <ctype.h>
 #include <getopt.h>
 #include <limits.h>
Index: util/wmmenugen.c
===================================================================
--- util/wmmenugen.c.orig	2013-01-09 22:42:39.000000000 +0100
+++ util/wmmenugen.c	2013-01-18 16:57:47.992050532 +0100
@@ -18,6 +18,10 @@
  * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <sys/types.h>
 #include <sys/stat.h>
 
Index: util/wmmenugen_misc.c
===================================================================
--- util/wmmenugen_misc.c.orig	2013-01-09 22:42:39.000000000 +0100
+++ util/wmmenugen_misc.c	2013-01-18 16:57:47.992050532 +0100
@@ -20,6 +20,10 @@
  * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <libgen.h>
 #include <stdlib.h>
 #include <string.h>
Index: util/wmmenugen_parse_wmconfig.c
===================================================================
--- util/wmmenugen_parse_wmconfig.c.orig	2013-01-09 22:42:39.000000000 +0100
+++ util/wmmenugen_parse_wmconfig.c	2013-01-18 16:57:47.992050532 +0100
@@ -20,6 +20,10 @@
  * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <sys/types.h>
 #include <sys/stat.h>
 
Index: util/wmmenugen_parse_xdg.c
===================================================================
--- util/wmmenugen_parse_xdg.c.orig	2013-01-09 22:42:39.000000000 +0100
+++ util/wmmenugen_parse_xdg.c	2013-01-18 16:57:47.992050532 +0100
@@ -36,6 +36,10 @@
  * Basic validation of the .desktop file is done.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <sys/types.h>
 #include <sys/stat.h>
 
Index: util/wxcopy.c
===================================================================
--- util/wxcopy.c.orig	2013-01-09 22:42:39.000000000 +0100
+++ util/wxcopy.c	2013-01-18 16:57:47.993050562 +0100
@@ -17,6 +17,10 @@
  *  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
Index: wmlib/app.c
===================================================================
--- wmlib/app.c.orig	2013-01-09 22:42:39.000000000 +0100
+++ wmlib/app.c	2013-01-18 16:57:47.993050562 +0100
@@ -19,6 +19,10 @@
  *  Software Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <X11/Xlib.h>
 #include <stdlib.h>
 #include <string.h>
Index: wmlib/command.c
===================================================================
--- wmlib/command.c.orig	2013-01-09 22:42:39.000000000 +0100
+++ wmlib/command.c	2013-01-18 16:57:47.993050562 +0100
@@ -19,6 +19,10 @@
  *  Software Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <X11/Xlib.h>
 #include <stdlib.h>
 
Index: wmlib/event.c
===================================================================
--- wmlib/event.c.orig	2013-01-09 22:42:39.000000000 +0100
+++ wmlib/event.c	2013-01-18 16:57:47.993050562 +0100
@@ -19,6 +19,10 @@
  *  Software Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <stdlib.h>
 #include <stdio.h>
 #include <X11/Xlib.h>
Index: wmlib/menu.c
===================================================================
--- wmlib/menu.c.orig	2013-01-09 22:42:39.000000000 +0100
+++ wmlib/menu.c	2013-01-18 16:57:47.993050562 +0100
@@ -19,6 +19,10 @@
  *  Software Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <stdlib.h>
 #include <stdio.h>
 #include <string.h>
Index: wrlib/alpha_combine.c
===================================================================
--- wrlib/alpha_combine.c.orig	2013-01-09 22:42:39.000000000 +0100
+++ wrlib/alpha_combine.c	2013-01-18 16:57:47.993050562 +0100
@@ -17,6 +17,10 @@
  *  Software Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "wraster.h"
 
 void RCombineAlpha(unsigned char *d, unsigned char *s, int s_has_alpha,
Index: wrlib/tests/testdraw.c
===================================================================
--- wrlib/tests/testdraw.c.orig	2013-01-09 22:42:39.000000000 +0100
+++ wrlib/tests/testdraw.c	2013-01-18 16:57:47.994050592 +0100
@@ -1,4 +1,8 @@
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <X11/Xlib.h>
 #include "wraster.h"
 #include <stdio.h>
Index: wrlib/tests/testgrad.c
===================================================================
--- wrlib/tests/testgrad.c.orig	2013-01-09 22:42:39.000000000 +0100
+++ wrlib/tests/testgrad.c	2013-01-18 16:57:47.994050592 +0100
@@ -1,4 +1,8 @@
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <X11/Xlib.h>
 #include "wraster.h"
 #include <stdio.h>
Index: wrlib/tests/testrot.c
===================================================================
--- wrlib/tests/testrot.c.orig	2013-01-09 22:42:39.000000000 +0100
+++ wrlib/tests/testrot.c	2013-01-18 16:57:47.994050592 +0100
@@ -1,4 +1,8 @@
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <X11/Xlib.h>
 #include "wraster.h"
 #include <stdlib.h>
Index: wrlib/tests/view.c
===================================================================
--- wrlib/tests/view.c.orig	2013-01-09 22:42:39.000000000 +0100
+++ wrlib/tests/view.c	2013-01-18 16:57:47.994050592 +0100
@@ -1,3 +1,7 @@
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <X11/Xlib.h>
 #include "wraster.h"
 #include <stdlib.h>
Index: configure.ac
===================================================================
--- configure.ac.orig	2013-01-09 22:42:49.000000000 +0100
+++ configure.ac	2013-01-18 16:57:47.994050592 +0100
@@ -67,8 +67,9 @@ dnl AC_CANONICAL_HOST -- already done by
 
 dnl Checks for programs.
 dnl ===================
-AC_ISC_POSIX
-AC_PROG_CC
+AC_PROG_CC_STDC
+AC_USE_SYSTEM_EXTENSIONS
+AC_SYS_LARGEFILE
 AC_PROG_LN_S
 AC_PROG_GCC_TRADITIONAL
 AC_PROG_LIBTOOL
Index: wrlib/gif.c
===================================================================
--- wrlib/gif.c.orig	2013-01-09 22:42:39.000000000 +0100
+++ wrlib/gif.c	2013-01-18 16:57:47.994050592 +0100
@@ -50,17 +50,24 @@ RImage *RLoadGIF(char *file, int index)
 	unsigned char rmap[256];
 	unsigned char gmap[256];
 	unsigned char bmap[256];
+    int dgiferr;
 
 	if (index < 0)
 		index = 0;
 
 	/* default error message */
 	RErrorCode = RERR_BADINDEX;
-
+#if defined(GIFLIB_MAJOR) && (GIFLIB_MAJOR >= 5)
+    gif = DGifOpenFileName(file, &dgiferr);
+#else
 	gif = DGifOpenFileName(file);
-
+#endif
 	if (!gif) {
+#if defined(GIFLIB_MAJOR) && (GIFLIB_MAJOR >= 5)
+        switch (dgiferr) {
+#else
 		switch (GifLastError()) {
+#endif
 		case D_GIF_ERR_OPEN_FAILED:
 			RErrorCode = RERR_OPEN;
 			break;
@@ -197,7 +204,11 @@ RImage *RLoadGIF(char *file, int index)
 	/* yuck! */
 	goto did_not_get_any_errors;
  giferr:
+#if defined(GIFLIB_MAJOR) && (GIFLIB_MAJOR >= 5)
+    switch (gif->Error) {
+#else
 	switch (GifLastError()) {
+#endif
 	case D_GIF_ERR_OPEN_FAILED:
 		RErrorCode = RERR_OPEN;
 		break;
