diff -up ypserv-2.27/conf_post.h.qdbm ypserv-2.27/conf_post.h
--- ypserv-2.27/conf_post.h.qdbm	2012-02-01 14:18:01.780846425 +0100
+++ ypserv-2.27/conf_post.h	2012-02-01 14:18:01.780846425 +0100
@@ -0,0 +1,10 @@
+/* Define common preprocessor symbol for HAVE_LIBGDBM and HAVE_LIBQDBM, since
+ * libraries gdbm and qdbm have same symbol names. */
+#undef HAVE_COMPAT_LIBGDBM
+#ifdef HAVE_LIBGDBM
+#define HAVE_COMPAT_LIBGDBM
+#endif
+#ifdef HAVE_LIBQDBM
+#define HAVE_COMPAT_LIBGDBM
+#endif
+
