--- qtiplot.pro	2008-09-30 15:16:15.807594378 +0200
+++ qtiplot.pro.archlinux	2008-09-30 15:19:19.170958588 +0200
@@ -17,7 +17,7 @@
 # Uncomment the following line if you want to perform a custom installation using
 # the *.path variables defined bellow.
 ######################################################################################
-#CONFIG          += CustomInstall
+CONFIG          += CustomInstall
 
 CONFIG          += release
 #CONFIG          += debug
@@ -27,13 +27,11 @@
 #!!! Warning: You must modify these paths according to your computer settings
 #############################################################################
 
-INCLUDEPATH       += ../3rdparty/muparser/include
-INCLUDEPATH       += ../3rdparty/qwtplot3d/include
-INCLUDEPATH       += ../3rdparty/qwt/src
+INCLUDEPATH       += /usr/include/qwt
+INCLUDEPATH       += /usr/include/qwtplot3d
+INCLUDEPATH       += /usr/include/boost
 INCLUDEPATH       += ../3rdparty/liborigin
-INCLUDEPATH       += ../3rdparty/gsl/include
-INCLUDEPATH       += ../3rdparty/zlib123/include
-INCLUDEPATH       += ../3rdparty/boost_1_36_0
+INCLUDEPATH       += /usr/include
 
 ##################### 3rd PARTY LIBRARIES SECTION ###########################
 #!!! Warning: You must modify these paths according to your computer settings
@@ -42,17 +40,19 @@
 ##################### Linux (Mac OS X) ######################################
 
 # statically link against libraries in 3rdparty
-unix:LIBS         += ../3rdparty/muparser/lib/libmuparser.a
-unix:LIBS         += ../3rdparty/qwtplot3d/lib/libqwtplot3d.a
-unix:LIBS         += ../3rdparty/qwt/lib/libqwt.a
-unix:LIBS         += ../3rdparty/gsl/lib/libgsl.a
-unix:LIBS         += ../3rdparty/gsl/lib/libgslcblas.a
+#unix:LIBS         += ../3rdparty/muparser/lib/libmuparser.a
+#unix:LIBS         += ../3rdparty/qwtplot3d/lib/libqwtplot3d.a
+#unix:LIBS         += ../3rdparty/qwt/lib/libqwt.a
+#unix:LIBS         += ../3rdparty/gsl/lib/libgsl.a
+#unix:LIBS         += ../3rdparty/gsl/lib/libgslcblas.a
 
 # dynamically link against dependencies if they are installed system-wide
-#unix:LIBS         += -lmuparser
-#unix:LIBS         += -lqwtplot3d
-#unix:LIBS         += -lqwt
-#unix:LIBS         += -lgsl -lgslcblas
+unix:LIBS         += -lmuparser
+unix:LIBS         += -lqwtplot3d
+unix:LIBS         += -lqwt
+unix:LIBS         += -lgsl -lgslcblas
+#unix:LIBS         += -lorigin
+unix:LIBS	  += -lz
 
 ##################### Windows ###############################################
 
@@ -569,4 +569,4 @@
 			 $${SIP_DIR}/sipqtiConvolution.cpp \
 			 $${SIP_DIR}/sipqtiDeconvolution.cpp \
 }
-###############################################################
+###############################################################
