--- hal-0.5.7/hal.conf.in.orig	2006-03-03 20:29:50.000000000 +0000
+++ hal-0.5.7/hal.conf.in	2006-03-03 20:31:26.000000000 +0000
@@ -33,10 +33,12 @@
     <deny send_interface="org.freedesktop.Hal.Device.Volume.Crypto"/>
   </policy>
 
-  <!-- This will not work if pam_console support is not enabled -->
-  <policy at_console="true">
+  <policy group="power">
     <allow send_interface="org.freedesktop.Hal.Device.SystemPowerManagement"/>
     <allow send_interface="org.freedesktop.Hal.Device.LaptopPanel"/>
+  </policy>
+
+  <policy group="storage">
     <allow send_interface="org.freedesktop.Hal.Device.Volume"/>
     <allow send_interface="org.freedesktop.Hal.Device.Volume.Crypto"/>
   </policy>
