--- fdi/policy/10osvendor/20-storage-methods.fdi.orig	2007-10-28 19:40:22.000000000 +0000
+++ fdi/policy/10osvendor/20-storage-methods.fdi	2007-10-28 19:40:58.000000000 +0000
@@ -240,6 +240,8 @@
 	  <append key="volume.mount.valid_options" type="strlist">uid=</append>
 	  <append key="volume.mount.valid_options" type="strlist">gid=</append>
 	  <append key="volume.mount.valid_options" type="strlist">umask=</append>
+	  <append key="volume.mount.valid_options" type="strlist">dmask=</append>
+	  <append key="volume.mount.valid_options" type="strlist">fmask=</append>
 	  <append key="volume.mount.valid_options" type="strlist">locale=</append>
 	  <append key="volume.mount.valid_options" type="strlist">utf8</append>
 	</match>
