From 536337dda1fa4bd30decc0e89f944603dc97fad9 Mon Sep 17 00:00:00 2001
From: Rico Tzschichholz <ricotz@ubuntu.com>
Date: Thu, 6 Feb 2020 08:46:54 +0100
Subject: [PATCH 3/3] build: Allow building with vala 0.48

---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 3526a16e6..200526c2d 100644
--- a/configure.ac
+++ b/configure.ac
@@ -338,7 +338,7 @@ if test "$user_disabled_vala" = 1; then
 else
         AC_MSG_RESULT(no)
 	AC_MSG_CHECKING(if libvala is available)
-	m4_foreach_w([VERSION], [0.46 0.44 0.42 0.40 0.38 0.36 0.34 0.32],
+	m4_foreach_w([VERSION], [0.48 0.46 0.44 0.42 0.40 0.38 0.36 0.34 0.32],
 		[PKG_CHECK_EXISTS([ libvala-VERSION ],
 			 [ valaver="-VERSION" ])
 			 ])
-- 
2.25.1

