Index: src/p80211/p80211netdev.c
===================================================================
--- src/p80211/p80211netdev.c	(revision 1840)
+++ src/p80211/p80211netdev.c	(working copy)
@@ -83,6 +83,11 @@
 #include <net/iw_handler.h>
 #endif
 
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,24) )
+#include <net/net_namespace.h>
+#define proc_net init_net.proc_net
+#endif 
+
 /*================================================================*/
 /* Project Includes */
 
@@ -938,7 +943,11 @@
 		dev->wireless_handlers = &p80211wext_handler_def;
 #endif
 #endif
-			
+
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,24) )
+		dev->nd_net = &init_net;
+#endif
+
 #if (LINUX_VERSION_CODE < KERNEL_VERSION(2,3,38) )
 		dev->tbusy = 1;
 		dev->start = 0;
Index: src/prism2/driver/prism2_plx.c
===================================================================
--- src/prism2/driver/prism2_plx.c	(revision 1840)
+++ src/prism2/driver/prism2_plx.c	(working copy)
@@ -228,7 +228,9 @@
 	/* Register the wlandev, this gets us a name and registers the
 	 * linux netdevice.
 	 */
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(2,6,24))
 	SET_MODULE_OWNER(wlandev->netdev);
+#endif
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,0))
        SET_NETDEV_DEV(wlandev->netdev, &(pdev->dev));
 #endif
Index: src/prism2/driver/prism2_cs.c
===================================================================
--- src/prism2/driver/prism2_cs.c	(revision 1840)
+++ src/prism2/driver/prism2_cs.c	(working copy)
@@ -517,7 +517,9 @@
 #endif
 
 	/* Register the network device and get assigned a name */
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(2,6,24))
 	SET_MODULE_OWNER(wlandev->netdev);
+#endif
 	SET_NETDEV_DEV(wlandev->netdev,  &handle_to_dev(pdev));
 	if (register_wlandev(wlandev) != 0) {
 		WLAN_LOG_NOTICE("prism2sta_cs: register_wlandev() failed.\n");
@@ -1080,7 +1082,9 @@
 	link->dev = &hw->node;
 
 	/* Register the network device and get assigned a name */
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(2,6,24))
 	SET_MODULE_OWNER(wlandev->netdev);
+#endif
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,11) )
 	SET_NETDEV_DEV(wlandev->netdev,  &handle_to_dev(link->handle));
 #endif
Index: src/prism2/driver/prism2_usb.c
===================================================================
--- src/prism2/driver/prism2_usb.c	(revision 1840)
+++ src/prism2/driver/prism2_usb.c	(working copy)
@@ -132,7 +132,9 @@
 	/* Register the wlandev, this gets us a name and registers the
 	 * linux netdevice.
 	 */
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(2,6,24))
 	SET_MODULE_OWNER(wlandev->netdev);
+#endif
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,0))
 	SET_NETDEV_DEV(wlandev->netdev, &(interface->dev));
 #endif
Index: src/prism2/driver/prism2_pci.c
===================================================================
--- src/prism2/driver/prism2_pci.c	(revision 1840)
+++ src/prism2/driver/prism2_pci.c	(working copy)
@@ -119,7 +119,9 @@
 	/* Register the wlandev, this gets us a name and registers the
 	 * linux netdevice.
 	 */
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(2,6,24))
 	SET_MODULE_OWNER(wlandev->netdev);
+#endif
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,0))
        SET_NETDEV_DEV(wlandev->netdev, &(pdev->dev));
 #endif
