diff -ur gtkam-0.1.12/help/C/gtkam.xml gtkam-0.1.12-new/help/C/gtkam.xml
--- gtkam-0.1.12/help/C/gtkam.xml	2004-04-27 03:38:53.000000000 -0700
+++ gtkam-0.1.12-new/help/C/gtkam.xml	2004-07-14 23:19:19.000000000 -0700
@@ -183,7 +183,7 @@
     		then press <keycap>Return</keycap>:</para>
     		<para> 
 			<command>gtkam</command> 
-			<!--FIXME comand line --switches -->
+			<!--FIXME comand line switches -->
     		</para> 
     		</listitem>
     	</varlistentry>
@@ -203,7 +203,7 @@
 		fileref="figures/gtkam_start_window.png" format="PNG"/> 
 	    </imageobject>
 	    <textobject> 
-	      <phrase>The &app; main window. On the left is the camera tree. The file list is on the right.
+	      <phrase>The &app; main window. On the left is the camera tree. The file list is on the right.</phrase>
 	    </textobject> 
 	  </mediaobject> 
 	</screenshot> 
diff -ur gtkam-0.1.12/omf-install/Makefile.am gtkam-0.1.12-new/omf-install/Makefile.am
--- gtkam-0.1.12/omf-install/Makefile.am	2004-01-21 02:55:11.000000000 -0800
+++ gtkam-0.1.12-new/omf-install/Makefile.am	2004-07-14 23:19:41.000000000 -0700
@@ -24,7 +24,7 @@
 	-for file in $(srcdir)/*.omf; do \
 		$(INSTALL_DATA) $(srcdir)/$$file $(DESTDIR)$(omf_dest_dir); \
 	done
-	-scrollkeeper-update -p $(scrollkeeper_localstate_dir) 
+	-/bin/true -p $(scrollkeeper_localstate_dir) 
 
 uninstall-local:
 	-for file in $(srcdir)/*.omf; do \
@@ -32,7 +32,7 @@
 		rm -f $(omf_dest_dir)/$$basefile; \
 	done
 	-rmdir $(omf_dest_dir)
-	-scrollkeeper-update -p $(scrollkeeper_localstate_dir)
+	-/bin/true -p $(scrollkeeper_localstate_dir)
 
 clean-local:
 	-for file in $(omffile); do \
diff -ur gtkam-0.1.12/omf-install/Makefile.in gtkam-0.1.12-new/omf-install/Makefile.in
--- gtkam-0.1.12/omf-install/Makefile.in	2004-05-21 18:28:56.000000000 -0700
+++ gtkam-0.1.12-new/omf-install/Makefile.in	2004-07-14 23:19:58.000000000 -0700
@@ -232,7 +232,7 @@
 	-for file in $(srcdir)/*.omf; do \
 		$(INSTALL_DATA) $(srcdir)/$$file $(DESTDIR)$(omf_dest_dir); \
 	done
-	-scrollkeeper-update -p $(scrollkeeper_localstate_dir) 
+	-/bin/true -p $(scrollkeeper_localstate_dir) 
 
 uninstall-local:
 	-for file in $(srcdir)/*.omf; do \
@@ -240,7 +240,7 @@
 		rm -f $(omf_dest_dir)/$$basefile; \
 	done
 	-rmdir $(omf_dest_dir)
-	-scrollkeeper-update -p $(scrollkeeper_localstate_dir)
+	-/bin/true -p $(scrollkeeper_localstate_dir)
 
 clean-local:
 	-for file in $(omffile); do \
