diff -Nur notification-daemon-0.4.0.orig/src/capplet/notification-properties.c notification-daemon-0.4.0/src/capplet/notification-properties.c
--- notification-daemon-0.4.0.orig/src/capplet/notification-properties.c	2008-11-20 02:46:16.000000000 -0800
+++ notification-daemon-0.4.0/src/capplet/notification-properties.c	2011-03-07 03:23:07.293334003 -0800
@@ -28,6 +28,10 @@
 #include <string.h>
 #include <libnotify/notify.h>
 
+#ifndef NOTIFY_CHECK_VERSION
+#define NOTIFY_CHECK_VERSION(x,y,z) 0
+#endif
+
 #include "../daemon/stack.h"
 
 #define GCONF_KEY_DAEMON         "/apps/notification-daemon"
@@ -434,8 +438,12 @@
 
 	dialog->preview = notify_notification_new(_("Notification Test"),
 											  _("Just a test"),
-											  "gnome-util",
-											  NULL);
+											  "gnome-util"
+#if NOTIFY_CHECK_VERSION (0, 7, 0)
+                                                                                          );
+#else
+                                                                                           , NULL);
+#endif
 
 	if (!notify_notification_show(dialog->preview, &error))
 	{
