--- officecfg/registry/data/org/openoffice/Office/Common.xcu
+++ officecfg/registry/data/org/openoffice/Office/Common.xcu
@@ -554,8 +554,8 @@
     </node>
   </node>
   <node oor:name="Misc">
-    <prop oor:name="UseSystemFileDialog"  install:module="UseOOoFileDialogs">
-      <value>false</value>
+    <prop oor:name="UseSystemFileDialog"  install:module="UseOOoFileDialogs"> <!-- safe -->
+      <value>true</value>
     </prop>
   </node>
   <node oor:name="Save">
