--- connectivity/source/drivers/hsqldb/HDriver.cxx
+++ connectivity/source/drivers/hsqldb/HDriver.cxx
@@ -241,7 +241,11 @@
 				aConvertedProperties[nPos].Name = ::rtl::OUString(RTL_CONSTASCII_USTRINGPARAM("JavaDriverClass"));
 				aConvertedProperties[nPos++].Value <<= ::rtl::OUString(RTL_CONSTASCII_USTRINGPARAM("org.hsqldb.jdbcDriver"));
 				aConvertedProperties[nPos].Name = ::rtl::OUString(RTL_CONSTASCII_USTRINGPARAM("JavaDriverClassPath"));
+#ifdef SYSTEM_HSQLDB
+                aConvertedProperties[nPos++].Value <<= (::rtl::OUString(RTL_CONSTASCII_USTRINGPARAM(HSQLDB_JAR)) + ::rtl::OUString(RTL_CONSTASCII_USTRINGPARAM(" vnd.sun.star.expand:$ORIGIN/classes/sdbc_hsqldb.jar")));
+#else
 				aConvertedProperties[nPos++].Value <<= ::rtl::OUString(RTL_CONSTASCII_USTRINGPARAM("vnd.sun.star.expand:$ORIGIN/classes/hsqldb.jar vnd.sun.star.expand:$ORIGIN/classes/sdbc_hsqldb.jar"));
+#endif
 				aConvertedProperties[nPos].Name = ::rtl::OUString(RTL_CONSTASCII_USTRINGPARAM("IsAutoRetrievingEnabled"));
 				aConvertedProperties[nPos++].Value <<= sal_True;
 				aConvertedProperties[nPos].Name = ::rtl::OUString(RTL_CONSTASCII_USTRINGPARAM("AutoRetrievingStatement"));
--- connectivity/source/drivers/hsqldb/makefile.mk
+++ connectivity/source/drivers/hsqldb/makefile.mk
@@ -50,6 +50,9 @@
 	@echo "Not building jurt because Java is disabled"
 .ENDIF
 
+.IF "$(SYSTEM_HSQLDB)" == "YES"
+CDEFS+=-DSYSTEM_HSQLDB -DHSQLDB_JAR=\""file://$(HSQLDB_JAR)"\"
+.ENDIF
 # --- Resources ---------------------------------
 
 SRS1NAME=$(TARGET)
