Index: sal/osl/unx/file.cxx
===================================================================
RCS file: /cvs/porting/sal/osl/unx/file.cxx,v
retrieving revision 1.8
diff -u -r1.8 file.cxx
--- sal/osl/unx/file.cxx	31 May 2005 17:07:41 -0000	1.8
+++ sal/osl/unx/file.cxx	14 Jun 2005 15:42:45 -0000
@@ -474,6 +474,8 @@
     *pItem = (oslDirectoryItem) oslDirectoryItemImpl_CreateNew( ustrFilePath, true, pEntry->d_type );
 #else
     /* use path as directory item */
+	if (*pItem)
+		rtl_uString_release( (rtl_uString *) *pItem );
     *pItem = (oslDirectoryItem) ustrFilePath;
 #endif
 
