--- sal/cpprt/makefile.mk.old	2010-02-02 12:49:41.000000000 +0100
+++ sal/cpprt/makefile.mk	2010-03-04 18:06:37.000000000 +0100
@@ -43,6 +43,11 @@ LIBTARGET=NO
 CFLAGS+= $(LFS_CFLAGS)
 CXXFLAGS+= $(LFS_CFLAGS)
 
+.IF "$(CCNUMVER)" >= "000400050000" && "$(CCNUMVER)" < "000400060000"
+CFLAGS+= -fno-ipa-sra
+CXXFLAGS+= -fno-ipa-sra
+.ENDIF 
+
 # --- Files --------------------------------------------------------
 
 SLOFILES =	\
