--- openoffice.org.orig/svtools/source/numbers/numhead.cxx	2006-09-17 17:20:40.000000000 +0200
+++ openoffice.org/svtools/source/numbers/numhead.cxx	2007-02-22 13:05:26.000000000 +0100
@@ -236,7 +236,7 @@ ImpSvNumMultipleWriteHeader::~ImpSvNumMu
 	{
 		nDataSize = nDataEnd - nDataPos;
 		ULONG nPos = rStream.Tell();
-		rStream.Seek(nDataPos-sizeof(ULONG));
+		rStream.Seek(nDataPos-sizeof(sal_uInt32));
 		rStream << nDataSize;								// Groesse am Anfang eintragen
 		rStream.Seek(nPos);
 	}
--- openoffice.org.orig/extensions/source/packer/packer2.hxx	2006-03-16 14:00:21.000000000 +0100
+++ openoffice.org/extensions/source/packer/packer2.hxx	2007-02-22 19:04:50.000000000 +0100
@@ -93,7 +93,7 @@ class INetURLObject;
 \* ******************************************************************/
 
 const sal_uInt32 PACK_ID = ((('S' << 8 | 'Z' ) << 8 | 'I' ) << 8 | 'P' );
-const USHORT HEADER_SIZE = 2*sizeof( ULONG ) + 3*sizeof( USHORT );
+const USHORT HEADER_SIZE = 2*sizeof( sal_uInt32 ) + 3*sizeof( USHORT );
 
 class LocalFileHeader
 {
--- openoffice.org.orig/svx/source/items/flditem.cxx	2006-10-12 14:53:54.000000000 +0200
+++ openoffice.org/svx/source/items/flditem.cxx	2007-02-22 19:06:03.000000000 +0100
@@ -402,7 +402,7 @@ void SvxURLField::Load( SvPersistStream 
 {
 	USHORT nFormat;
 	sal_uInt32 nFrameMarker, nCharSetMarker;
-	long nUlongSize = (long)sizeof(ULONG);
+	long nUlongSize = (long)sizeof(sal_uInt32);
 	String aTmpURL;
 
 	rStm >> nFormat;
--- openoffice.org.orig/svtools/source/items1/poolio.cxx	2006-10-12 17:22:39.000000000 +0200
+++ openoffice.org/svtools/source/items1/poolio.cxx	2007-02-22 19:09:40.000000000 +0100
@@ -788,7 +788,7 @@ SvStream &SfxItemPool::Load1_Impl(SvStre
 	if ( pImp->nMajorVer > 1 || pImp->nMinorVer >= 3 )
 	{
 		// Version-Map finden (letztes ULONG der Size-Table gibt Pos an)
-		rStream.Seek( nEndOfSizes - sizeof(ULONG) );
+		rStream.Seek( nEndOfSizes - sizeof(sal_uInt32) );
 		sal_uInt32 nVersionMapPos;
 		rStream >> nVersionMapPos;
 		rStream.Seek( nVersionMapPos );
--- openoffice.org.orig/svtools/source/numbers/numhead.cxx	2007-02-22 13:59:11.000000000 +0100
+++ openoffice.org/svtools/source/numbers/numhead.cxx	2007-02-22 19:09:23.000000000 +0100
@@ -106,7 +106,7 @@ SvNumWriteHeader::~SvNumWriteHeader()
 	if ( nPos - nDataPos != nDataSize )				// Default getroffen?
 	{
 		nDataSize = nPos - nDataPos;
-		rStream.Seek(nDataPos - sizeof(ULONG));
+		rStream.Seek(nDataPos - sizeof(sal_uInt32));
 		rStream << nDataSize;						// Groesse am Anfang eintragen
 		rStream.Seek(nPos);
 	}
--- openoffice.org.orig/binfilter/bf_sc/source/core/tool/sc_rechead.cxx	2006-10-27 16:37:48.000000000 +0200
+++ openoffice.org/binfilter/bf_sc/source/core/tool/sc_rechead.cxx	2007-02-22 19:07:47.000000000 +0100
@@ -98,7 +98,7 @@ namespace binfilter {
 /*N*/ 	if ( nPos - nDataPos != nDataSize )				// Default getroffen?
 /*N*/ 	{
 /*N*/ 		nDataSize = nPos - nDataPos;
-/*N*/ 		rStream.Seek(nDataPos - sizeof(ULONG));
+/*N*/ 		rStream.Seek(nDataPos - sizeof(sal_uInt32));
 /*N*/ 		rStream << nDataSize;						// Groesse am Anfang eintragen
 /*N*/ 		rStream.Seek(nPos);
 /*N*/ 	}
@@ -214,7 +214,7 @@ namespace binfilter {
 /*N*/ 	{
 /*N*/ 		nDataSize = nDataEnd - nDataPos;
 /*N*/ 		ULONG nPos = rStream.Tell();
-/*N*/ 		rStream.Seek(nDataPos-sizeof(ULONG));
+/*N*/ 		rStream.Seek(nDataPos-sizeof(sal_uInt32));
 /*N*/ 		rStream << nDataSize;								// Groesse am Anfang eintragen
 /*N*/ 		rStream.Seek(nPos);
 /*N*/ 	}
--- openoffice.org.orig/binfilter/bf_svx/source/items/svx_flditem.cxx	2006-10-27 23:15:22.000000000 +0200
+++ openoffice.org/binfilter/bf_svx/source/items/svx_flditem.cxx	2007-02-22 19:06:25.000000000 +0100
@@ -403,7 +403,7 @@ namespace binfilter {
 /*N*/ {
 /*N*/ 	USHORT nFormat;
 /*N*/ 	sal_uInt32 nFrameMarker, nCharSetMarker;
-/*N*/ 	long nUlongSize = (long)sizeof(ULONG);
+/*N*/ 	long nUlongSize = (long)sizeof(sal_uInt32);
 /*N*/ 	String aTmpURL;
 /*N*/ 
 /*N*/ 	rStm >> nFormat;
--- openoffice.org.orig/connectivity/source/inc/dbase/DTable.hxx	2007-02-23 15:02:12.000000000 +0100
+++ openoffice.org/connectivity/source/inc/dbase/DTable.hxx	2007-02-23 15:02:38.000000000 +0100
@@ -92,7 +92,7 @@ namespace connectivity
 			struct DBFColumn {                       /* Feldbezeichner               */
 								BYTE    db_fnm[11];                     /* Feldname                     */
 								BYTE    db_typ;                         /* Feldtyp                      */
-								ULONG   db_adr;                         /* Feldadresse                  */
+								UINT32  db_adr;                         /* Feldadresse                  */
 								BYTE    db_flng;                        /* Feldlaenge                   */
 								BYTE    db_dez;                         /* Dezimalstellen fuer N        */
 								BYTE    db_frei2[14];                   /* reserviert                   */
--- openoffice.org.orig/basctl/inc/basidesh.hxx	2007-01-24 18:24:39.000000000 +0100
+++ openoffice.org/basctl/inc/basidesh.hxx	2007-02-19 18:12:19.000000000 +0100
@@ -200,7 +200,7 @@ public:
 	void				ExecuteBasic( SfxRequest& rReq );
 	void				ExecuteDialog( SfxRequest& rReq );
 
-	virtual	BOOL		HasUIFeature( ULONG nFeature );
+	virtual	sal_Bool	HasUIFeature( sal_uInt32 nFeature );
 
 	long				CallBasicErrorHdl( StarBASIC* pBasic );
 	long				CallBasicBreakHdl( StarBASIC* pBasic );
--- openoffice.org.orig/basctl/source/basicide/basides1.cxx	2007-01-02 16:49:14.000000000 +0100
+++ openoffice.org/basctl/source/basicide/basides1.cxx	2007-02-19 18:12:44.000000000 +0100
@@ -1113,15 +1113,15 @@ void __EXPORT BasicIDEShell::GetState(Sf
 		pCurWin->GetState( rSet );
 }
 
-BOOL BasicIDEShell::HasUIFeature( ULONG nFeature )
+sal_Bool BasicIDEShell::HasUIFeature( sal_uInt32 nFeature )
 {
-    BOOL bResult = FALSE;
+    sal_Bool bResult = sal_False;
 
     if ( (nFeature & BASICIDE_UI_FEATURE_SHOW_BROWSER) == BASICIDE_UI_FEATURE_SHOW_BROWSER )
     {
         // fade out (in) property browser in module (dialog) windows
 	    if ( pCurWin && pCurWin->IsA( TYPE( DialogWindow ) ) && !pCurWin->IsReadOnly() )
-            bResult = TRUE;
+            bResult = sal_True;
     }
 
     return bResult;
--- openoffice.org.orig/binfilter/bf_sfx2/source/control/sfx2_shell.cxx	2006-10-27 21:10:22.000000000 +0200
+++ openoffice.org/binfilter/bf_sfx2/source/control/sfx2_shell.cxx	2007-02-19 18:26:16.000000000 +0100
@@ -1383,9 +1383,9 @@ DBG_BF_ASSERT(0, "STRIP"); //STRIP001 	(
 /*N*/ 	return SFX_APP()->GetSlotPool( GetFrame() );
 /*N*/ }
 
-/*N*/ BOOL SfxShell::HasUIFeature( ULONG )
+/*N*/ sal_Bool SfxShell::HasUIFeature( sal_uInt32 )
 /*N*/ {
-/*N*/ 	return FALSE;
+/*N*/ 	return sal_False;
 /*N*/ }
 
 /*N*/ long DispatcherUpdate_Impl( void* pObj, void* pArg )
--- openoffice.org.orig/binfilter/bf_sch/source/ui/view/sch_viewshe4.cxx	2006-10-27 20:00:56.000000000 +0200
+++ openoffice.org/binfilter/bf_sch/source/ui/view/sch_viewshe4.cxx	2007-02-19 18:27:49.000000000 +0100
@@ -442,28 +442,28 @@ namespace binfilter {
 
 
 
-/** Returns TRUE if the corresponding feature is available,
+/** Returns sal_True if the corresponding feature is available,
     eg it returns only true if the chart is three dimensional
     and the 3d effect flyer is to be displayed
 */
-/*N*/ BOOL SchViewShell::HasUIFeature( ULONG nFeature )
+/*N*/ sal_Bool SchViewShell::HasUIFeature( sal_uInt32 nFeature )
 /*N*/ {
 /*N*/     switch( nFeature )
 /*N*/     {
 /*N*/         case SCH_UI_FEATURE_THREE_D:
 /*N*/             if( pDocSh )
 /*N*/                 return pDocSh->GetDoc().IsReal3D();
-/*?*/             return FALSE;
+/*?*/             return sal_False;
 /*N*/ 
 /*N*/         case SCH_UI_FEATURE_OWN_DATA:
 /*N*/             // ToDo: If chart is inside Calc/Writer using external data, FALSE
 /*N*/             // should be returned. It currently works in a different way (hack)
-/*N*/             return TRUE;
+/*N*/             return sal_True;
 /*N*/ 
 /*?*/         default:
 /*?*/             DBG_ERRORFILE( "Unknown UI Feature requested!" );
-/*?*/             return FALSE;
 /*N*/     }
+/*?*/     return sal_False;
 /*N*/ }
 
 
--- openoffice.org.orig/binfilter/bf_sw/source/ui/inc/view.hxx	2006-10-28 03:00:02.000000000 +0200
+++ openoffice.org/binfilter/bf_sw/source/ui/inc/view.hxx	2007-02-19 18:31:21.000000000 +0100
@@ -365,7 +365,7 @@ public:
 
 	SfxDispatcher 	&GetDispatcher();
 
-	virtual	BOOL			HasUIFeature( sal_uInt32 nFeature );
+	virtual	sal_Bool		HasUIFeature( sal_uInt32 nFeature );
 	virtual	void			ShowCursor( FASTBOOL bOn = TRUE );
 
 	ShellModes				GetShellMode();
--- openoffice.org.orig/binfilter/inc/bf_sfx2/shell.hxx	2006-10-28 05:09:03.000000000 +0200
+++ openoffice.org/binfilter/inc/bf_sfx2/shell.hxx	2007-02-19 18:23:50.000000000 +0100
@@ -235,7 +235,7 @@ public:
 	SfxDispatcher*				GetDispatcher() const;
 	SfxViewFrame*				GetFrame() const;
 	ResMgr* 					GetResMgr() const;
-	virtual	BOOL				HasUIFeature( ULONG nFeature );
+	virtual	sal_Bool			HasUIFeature( sal_uInt32 nFeature );
 	void						UIFeatureChanged();
 
 	// Items
--- openoffice.org.orig/binfilter/inc/bf_sch/viewshel.hxx	2006-10-28 04:50:58.000000000 +0200
+++ openoffice.org/binfilter/inc/bf_sch/viewshel.hxx	2007-02-19 18:23:41.000000000 +0100
@@ -174,7 +174,7 @@ public:
 	
 	// drag and drop
 	
-	virtual	BOOL HasUIFeature( ULONG nFeature );
+	virtual	sal_Bool HasUIFeature( sal_uInt32 nFeature );
 
     // SfxListener
 	virtual void		Notify( SfxBroadcaster& rBC, const SfxHint& rHint );
--- openoffice.org.orig/sfx2/inc/objface.hxx	2006-10-12 17:44:51.000000000 +0200
+++ openoffice.org/sfx2/inc/objface.hxx	2007-02-19 17:48:55.000000000 +0100
@@ -107,19 +107,19 @@ public:
 	const SfxInterface*		GetRealInterfaceForSlot( const SfxSlot* ) const;
 
 	void					RegisterObjectBar( USHORT, const ResId&, const String* pST=0 );
-	void					RegisterObjectBar( USHORT, const ResId&, ULONG nFeature, const String* pST=0 );
+	void					RegisterObjectBar( USHORT, const ResId&, sal_uInt32 nFeature, const String* pST=0 );
 	void					RegisterChildWindow( USHORT, BOOL bContext, const String* pST=0 );
-	void					RegisterChildWindow( USHORT, BOOL bContext, ULONG nFeature, const String* pST=0 );
+	void					RegisterChildWindow( USHORT, BOOL bContext, sal_uInt32 nFeature, const String* pST=0 );
 	void					RegisterStatusBar( const ResId& );
 	const ResId&			GetObjectBarResId( USHORT nNo ) const;
 	USHORT                  GetObjectBarPos( USHORT nNo ) const;
-	ULONG                   GetObjectBarFeature( USHORT nNo ) const;
+	sal_uInt32              GetObjectBarFeature( USHORT nNo ) const;
 	USHORT					GetObjectBarCount() const;
 	void                    SetObjectBarPos( USHORT nPos, USHORT nId );
 	const String*           GetObjectBarName( USHORT nNo ) const;
 	BOOL					IsObjectBarVisible( USHORT nNo) const;
-	ULONG					GetChildWindowFeature( USHORT nNo ) const;
-	ULONG					GetChildWindowId( USHORT nNo ) const;
+	sal_uInt32				GetChildWindowFeature( USHORT nNo ) const;
+	sal_uInt32				GetChildWindowId( USHORT nNo ) const;
 	USHORT					GetChildWindowCount() const;
 	void					RegisterPopupMenu( const ResId& );
 	const ResId&			GetPopupMenuResId() const;
@@ -164,7 +164,7 @@ public:
                     SfxIFConfig_Impl();
 			        ~SfxIFConfig_Impl();
 	BOOL    		Store(SvStream&);
-    void            RegisterObjectBar( USHORT, const ResId&, ULONG nFeature, const String* pST=0 );
+    void            RegisterObjectBar( USHORT, const ResId&, sal_uInt32 nFeature, const String* pST=0 );
     USHORT          GetType();
 };
 #endif
--- openoffice.org.orig/sfx2/inc/shell.hxx	2006-06-20 00:04:31.000000000 +0200
+++ openoffice.org/sfx2/inc/shell.hxx	2007-02-19 18:34:07.000000000 +0100
@@ -255,7 +255,7 @@ public:
 	SfxDispatcher*				GetDispatcher() const;
 	SfxViewFrame*				GetFrame() const;
 	ResMgr* 					GetResMgr() const;
-	virtual	BOOL				HasUIFeature( ULONG nFeature );
+	virtual	sal_Bool			HasUIFeature( sal_uInt32 nFeature );
 	void						UIFeatureChanged();
 
 	// Items
--- openoffice.org.orig/sfx2/source/appl/workwin.cxx	2007-02-16 15:29:45.000000000 +0100
+++ openoffice.org/sfx2/source/appl/workwin.cxx	2007-02-19 12:23:40.000000000 +0100
@@ -2142,7 +2142,7 @@ void SfxWorkWindow::ConfigChild_Impl(Sfx
 
 //--------------------------------------------------------------------
 
-void SfxWorkWindow::SetChildWindowVisible_Impl( ULONG lId, BOOL bEnabled, USHORT nMode )
+void SfxWorkWindow::SetChildWindowVisible_Impl( sal_uInt32 lId, BOOL bEnabled, USHORT nMode )
 {
 	USHORT nInter = (USHORT) ( lId >> 16 );
 	USHORT nId = (USHORT) ( lId & 0xFFFF );
--- openoffice.org.orig/sfx2/source/control/dispatch.cxx	2006-11-01 19:26:15.000000000 +0100
+++ openoffice.org/sfx2/source/control/dispatch.cxx	2007-02-19 17:12:02.000000000 +0100
@@ -1829,7 +1829,7 @@ long SfxDispatcher::Update_Impl( sal_Boo
 
 sal_uInt32 SfxDispatcher::_Update_Impl( sal_Bool bUIActive, sal_Bool bIsMDIApp, sal_Bool bIsIPOwner, SfxWorkWindow *pTaskWin )
 {
-	sal_uInt32 nHelpId = 0L;
+	sal_uInt32 nHelpId = 0;
 	SFX_APP();
     SfxWorkWindow *pWorkWin = pImp->pFrame->GetFrame()->GetWorkWindow_Impl();
 	sal_Bool bIsActive = sal_False;
--- openoffice.org.orig/sfx2/source/control/objface.cxx	2006-10-12 17:51:28.000000000 +0200
+++ openoffice.org/sfx2/source/control/objface.cxx	2007-02-19 17:51:36.000000000 +0100
@@ -88,9 +88,9 @@ struct SfxObjectUI_Impl
 	BOOL	bVisible;
 	BOOL	bContext;
 	String* pName;
-	ULONG	nFeature;
+	sal_uInt32	nFeature;
 
-	SfxObjectUI_Impl(USHORT n, const ResId& rResId, BOOL bVis, ULONG nFeat) :
+	SfxObjectUI_Impl(USHORT n, const ResId& rResId, BOOL bVis, sal_uInt32 nFeat) :
 		nPos(n),
 		aResId(rResId.GetId(), rResId.GetResMgr()),
 		bVisible(bVis),
@@ -140,7 +140,7 @@ struct SfxInterface_Impl
 	}
 };
 
-static SfxObjectUI_Impl* CreateObjectBarUI_Impl( USHORT nPos, const ResId& rResId, ULONG nFeature, const String *pStr );
+static SfxObjectUI_Impl* CreateObjectBarUI_Impl( USHORT nPos, const ResId& rResId, sal_uInt32 nFeature, const String *pStr );
 
 //====================================================================
 
@@ -432,14 +432,14 @@ void SfxInterface::RegisterObjectBar( US
 }
 
 
-void SfxInterface::RegisterObjectBar( USHORT nPos, const ResId& rResId, ULONG nFeature, const String *pStr )
+void SfxInterface::RegisterObjectBar( USHORT nPos, const ResId& rResId, sal_uInt32 nFeature, const String *pStr )
 {
     SfxObjectUI_Impl* pUI = CreateObjectBarUI_Impl( nPos, rResId, nFeature, pStr );
     if ( pUI )
         pImpData->pObjectBars->Append(pUI);
 }
 
-SfxObjectUI_Impl* CreateObjectBarUI_Impl( USHORT nPos, const ResId& rResId, ULONG nFeature, const String *pStr )
+SfxObjectUI_Impl* CreateObjectBarUI_Impl( USHORT nPos, const ResId& rResId, sal_uInt32 nFeature, const String *pStr )
 {
 	if ((nPos & SFX_VISIBILITY_MASK) == 0)
         nPos |= SFX_VISIBILITY_STANDARD;
@@ -524,7 +524,7 @@ void SfxInterface::RegisterChildWindow(U
 	RegisterChildWindow( nId, bContext, 0UL, pChildWinName );
 }
 
-void SfxInterface::RegisterChildWindow(USHORT nId, BOOL bContext, ULONG nFeature, const String*)
+void SfxInterface::RegisterChildWindow(USHORT nId, BOOL bContext, sal_uInt32 nFeature, const String*)
 {
 	SfxObjectUI_Impl* pUI = new SfxObjectUI_Impl(0, nId, TRUE, nFeature);
 	pUI->bContext = bContext;
@@ -537,7 +537,7 @@ void SfxInterface::RegisterStatusBar(con
 }
 
 
-ULONG SfxInterface::GetChildWindowId (USHORT nNo) const
+sal_uInt32 SfxInterface::GetChildWindowId (USHORT nNo) const
 {
 	if ( pGenoType )
 	{
@@ -554,13 +554,13 @@ ULONG SfxInterface::GetChildWindowId (US
 	USHORT nCWCount = pImpData->pChildWindows->Count();
 	DBG_ASSERT( nNo<nCWCount,"ChildWindow ist unbekannt!" );
 #endif
-	ULONG nRet = (ULONG) (*pImpData->pChildWindows)[nNo]->aResId.GetId();
+	sal_uInt32 nRet = (*pImpData->pChildWindows)[nNo]->aResId.GetId();
 	if ( (*pImpData->pChildWindows)[nNo]->bContext )
-		nRet += ( (ULONG) nClassId ) << 16;
+		nRet += sal_uInt32( nClassId ) << 16;
 	return nRet;
 }
 
-ULONG SfxInterface::GetChildWindowFeature (USHORT nNo) const
+sal_uInt32 SfxInterface::GetChildWindowFeature (USHORT nNo) const
 {
 	if ( pGenoType )
 	{
@@ -577,8 +577,7 @@ ULONG SfxInterface::GetChildWindowFeatur
 	USHORT nCWCount = pImpData->pChildWindows->Count();
 	DBG_ASSERT( nNo<nCWCount,"ChildWindow ist unbekannt!" );
 #endif
-	ULONG nRet = (ULONG) (*pImpData->pChildWindows)[nNo]->nFeature;
-	return nRet;
+	return (*pImpData->pChildWindows)[nNo]->nFeature;
 }
 
 //--------------------------------------------------------------------
@@ -630,7 +629,7 @@ const String* SfxInterface::GetObjectBar
 	return (*pImpData->pObjectBars)[nNo]->pName;
 }
 
-ULONG SfxInterface::GetObjectBarFeature ( USHORT nNo ) const
+sal_uInt32 SfxInterface::GetObjectBarFeature ( USHORT nNo ) const
 {
 	BOOL bGenoType = (pGenoType != 0 && !pGenoType->HasName());
 	if ( bGenoType )
--- openoffice.org.orig/sfx2/source/control/shell.cxx	2006-10-12 17:51:54.000000000 +0200
+++ openoffice.org/sfx2/source/control/shell.cxx	2007-02-19 18:14:36.000000000 +0100
@@ -1293,9 +1294,9 @@ SfxObjectShell* SfxShell::GetObjectShell
 
 //--------------------------------------------------------------------
 
-BOOL SfxShell::HasUIFeature( ULONG )
+sal_Bool SfxShell::HasUIFeature( sal_uInt32 )
 {
-	return FALSE;
+	return sal_False;
 }
 
 long DispatcherUpdate_Impl( void*, void* pArg )
--- openoffice.org.orig/sfx2/source/inc/workwin.hxx	2007-02-16 15:29:39.000000000 +0100
+++ openoffice.org/sfx2/source/inc/workwin.hxx	2007-02-16 15:30:39.000000000 +0100
@@ -171,7 +171,7 @@ struct SfxChildWin_Impl
 	BOOL							bEnable;
 	BOOL							bDisabled;
 
-	SfxChildWin_Impl( ULONG nID ) :
+	SfxChildWin_Impl( sal_uInt32 nID ) :
 		nSaveId((USHORT) (nID & 0xFFFF) ),
 		nInterfaceId((USHORT) (nID >> 16)),
 		nId(nSaveId),
@@ -375,7 +375,7 @@ public:
 	// Methoden f"ur ChildWindows
 	void            		UpdateChildWindows_Impl();
 	void					ResetChildWindows_Impl();
-	void					SetChildWindowVisible_Impl( ULONG, BOOL, USHORT );
+	void					SetChildWindowVisible_Impl( sal_uInt32, BOOL, USHORT );
     void                    ToggleChildWindow_Impl(USHORT,BOOL);
 	BOOL					HasChildWindow_Impl(USHORT);
 	BOOL					KnowsChildWindow_Impl(USHORT);
--- openoffice.org.orig/sch/inc/viewshel.hxx	2007-01-24 18:14:14.000000000 +0100
+++ openoffice.org/sch/inc/viewshel.hxx	2007-02-19 18:22:46.000000000 +0100
@@ -199,7 +199,7 @@ public:
 	virtual sal_Int8 AcceptDrop( const AcceptDropEvent& rEvt, SchWindow* pWin );
 	virtual sal_Int8 ExecuteDrop( const ExecuteDropEvent& rEvt, SchWindow* pWin );
 	
-	virtual	BOOL HasUIFeature( ULONG nFeature );
+	virtual	sal_Bool HasUIFeature( sal_uInt32 nFeature );
 
     // SfxListener
 	virtual void		Notify( SfxBroadcaster& rBC, const SfxHint& rHint );
--- openoffice.org.orig/sch/source/ui/view/viewshe4.cxx	2007-01-24 18:14:41.000000000 +0100
+++ openoffice.org/sch/source/ui/view/viewshe4.cxx	2007-02-19 18:23:17.000000000 +0100
@@ -673,24 +673,24 @@ USHORT SchViewShell::PrepareClose(BOOL b
     eg it returns only true if the chart is three dimensional
     and the 3d effect flyer is to be displayed
 */
-BOOL SchViewShell::HasUIFeature( ULONG nFeature )
+sal_Bool SchViewShell::HasUIFeature( sal_uInt32 nFeature )
 {
     switch( nFeature )
     {
         case SCH_UI_FEATURE_THREE_D:
             if( pDocSh )
                 return pDocSh->GetDoc().IsReal3D();
-            return FALSE;
+            return sal_False;
 
         case SCH_UI_FEATURE_OWN_DATA:
             // ToDo: If chart is inside Calc/Writer using external data, FALSE
             // should be returned. It currently works in a different way (hack)
-            return TRUE;
+            return sal_True;
 
         default:
             DBG_ERRORFILE( "Unknown UI Feature requested!" );
-            return FALSE;
     }
+    return sal_False;
 }
 
 void SchViewShell::AllowSelectionChangeBroadcasts( sal_Bool bAllow )
--- openoffice.org.orig/sw/source/ui/inc/view.hxx	2007-01-24 18:20:23.000000000 +0100
+++ openoffice.org/sw/source/ui/inc/view.hxx	2007-02-19 18:35:01.000000000 +0100
@@ -435,7 +435,7 @@ public:
 
 	void                    GotFocus() const;
 	virtual SdrView*		GetDrawView() const;
-	virtual	BOOL			HasUIFeature( sal_uInt32 nFeature );
+	virtual	sal_Bool		HasUIFeature( sal_uInt32 nFeature );
 	virtual	void			ShowCursor( FASTBOOL bOn = TRUE );
 	virtual ErrCode			DoVerb( long nVerb );
 
--- openoffice.org/tools/inc/gen.hxx	2005-09-09 15:41:09.000000000 +0200
+++ openoffice.org.orig/tools/inc/gen.hxx	2007-02-26 17:08:21.000000000 +0100
@@ -254,8 +254,9 @@ inline Size::Size( long nWidth, long nHe
 // - Range -
 // ---------
 
-#define RANGE_MIN	((long)0x80000000L)
-#define RANGE_MAX	0x7FFFFFFFL
+#include <limits.h>
+#define RANGE_MIN	LONG_MIN
+#define RANGE_MAX	LONG_MAX
 
 class Range : public Pair
 {
@@ -302,8 +302,8 @@ inline void Range::Justify()
 // - Selection -
 // -------------
 
-#define SELECTION_MIN	((long)0x80000000L)
-#define SELECTION_MAX	0x7FFFFFFFL
+#define SELECTION_MIN	LONG_MIN
+#define SELECTION_MAX	LONG_MAX
 
 class Selection : public Pair
 {
Index: inc/editbrowsebox.hxx
===================================================================
RCS file: /cvs/util/svtools/inc/editbrowsebox.hxx,v
retrieving revision 1.17
diff -u -r1.17 editbrowsebox.hxx
--- openoffice.org.orig/svtools/inc/editbrowsebox.hxx	19 Jun 2006 20:15:24 -0000	1.17
+++ openoffice.org/svtools/inc/editbrowsebox.hxx	12 Mar 2007 09:08:49 -0000
@@ -528,7 +528,7 @@
 		CellControllerRef		 aController,
 								 aOldController;
 
-		sal_uInt32	nStartEvent, nEndEvent, nCellModifiedEvent;		// event ids
+		ULONG	nStartEvent, nEndEvent, nCellModifiedEvent;		// event ids
 		Window*	m_pFocusWhileRequest;
 			// In ActivateCell, we grab the focus asynchronously, but if between requesting activation
 			// and the asynchornous event the focus has changed, we won't grab it for ourself.
diff -pur extensions/source/xmlextract/xmxtrct.cxx extensions/source/xmlextract/xmxtrct.cxx
--- openoffice.org.orig/extensions/source/xmlextract/xmxtrct.cxx	2006-09-16 15:31:04.000000000 +0200
+++ openoffice.org/extensions/source/xmlextract/xmxtrct.cxx	2007-03-08 20:26:23.000000000 +0100
@@ -59,10 +59,10 @@ public:
 									XMXLockBytes( const REF( NMSP_IO::XInputStream )& rxIStm );
 	virtual							~XMXLockBytes();
 
-	virtual ErrCode					ReadAt( sal_uInt32 nPos, void* pBuffer, sal_uInt32 nCount, sal_uInt32* pRead ) const;
-	virtual ErrCode					WriteAt( sal_uInt32 nPos, const void* pBuffer, sal_uInt32 nCount, sal_uInt32* pWritten );
+	virtual ErrCode					ReadAt( sal_Size nPos, void* pBuffer, sal_Size nCount, sal_Size* pRead ) const;
+	virtual ErrCode					WriteAt( sal_Size nPos, const void* pBuffer, sal_Size nCount, sal_Size* pWritten );
 	virtual ErrCode					Flush() const;
-	virtual ErrCode					SetSize( sal_uInt32 nSize );
+	virtual ErrCode					SetSize( sal_Size nSize );
 	virtual ErrCode					Stat( SvLockBytesStat*, SvLockBytesStatFlag ) const;
 };
 
@@ -101,9 +101,9 @@ XMXLockBytes::~XMXLockBytes()
 
 // ------------------------------------------------------------------------
 
-ErrCode XMXLockBytes::ReadAt( sal_uInt32 nPos, void* pBuffer, sal_uInt32 nCount, sal_uInt32* pRead ) const
+ErrCode XMXLockBytes::ReadAt( sal_Size nPos, void* pBuffer, sal_Size nCount, sal_Size* pRead ) const
 {
-	const sal_uInt32	nSeqLen = maSeq.getLength();
+	const sal_Size		nSeqLen = maSeq.getLength();
 	ErrCode				nErr = ERRCODE_NONE;
 
 	if( nPos < nSeqLen )
@@ -122,7 +122,7 @@ ErrCode XMXLockBytes::ReadAt( sal_uInt32
 
 // ------------------------------------------------------------------------
 
-ErrCode XMXLockBytes::WriteAt( sal_uInt32 nPos, const void* pBuffer, sal_uInt32 nCount, sal_uInt32* pWritten )
+ErrCode XMXLockBytes::WriteAt( sal_Size nPos, const void* pBuffer, sal_Size nCount, sal_Size* pWritten )
 {
 	return ERRCODE_IO_CANTWRITE;
 }
@@ -136,7 +136,7 @@ ErrCode XMXLockBytes::Flush() const
 
 // ------------------------------------------------------------------------
 
-ErrCode XMXLockBytes::SetSize( sal_uInt32 nSize )
+ErrCode XMXLockBytes::SetSize( sal_Size nSize )
 {
 	return ERRCODE_IO_CANTWRITE;
 }
diff -pur forms/source/component/imgprod.cxx forms/source/component/imgprod.cxx
--- openoffice.org.orig/forms/source/component/imgprod.cxx	2006-10-12 13:13:26.000000000 +0200
+++ openoffice.org/forms/source/component/imgprod.cxx	2007-03-08 20:30:00.000000000 +0100
@@ -73,10 +73,10 @@ public:
 						ImgProdLockBytes( ::com::sun::star::uno::Reference< ::com::sun::star::io::XInputStream > & rStreamRef );
 	virtual				~ImgProdLockBytes();
 
-	virtual ErrCode		ReadAt( sal_uInt32 nPos, void* pBuffer, sal_uInt32 nCount, sal_Size* pRead ) const;
-	virtual ErrCode		WriteAt( sal_uInt32 nPos, const void* pBuffer, sal_uInt32 nCount, sal_Size* pWritten );
+	virtual ErrCode		ReadAt( sal_Size nPos, void* pBuffer, sal_Size nCount, sal_Size* pRead ) const;
+	virtual ErrCode		WriteAt( sal_Size nPos, const void* pBuffer, sal_Size nCount, sal_Size* pWritten );
 	virtual ErrCode		Flush() const;
-	virtual ErrCode		SetSize( sal_uInt32 nSize );
+	virtual ErrCode		SetSize( sal_Size nSize );
 	virtual ErrCode		Stat( SvLockBytesStat*, SvLockBytesStatFlag ) const;
 };
 
@@ -122,7 +122,7 @@ ImgProdLockBytes::~ImgProdLockBytes()
 
 // ------------------------------------------------------------------------
 
-ErrCode ImgProdLockBytes::ReadAt( sal_uInt32 nPos, void* pBuffer, sal_uInt32 nCount, sal_Size* pRead ) const
+ErrCode ImgProdLockBytes::ReadAt( sal_Size nPos, void* pBuffer, sal_Size nCount, sal_Size* pRead ) const
 {
 	if( GetStream() )
 	{
@@ -133,7 +133,7 @@ ErrCode ImgProdLockBytes::ReadAt( sal_uI
 	}
 	else
 	{
-		const sal_uInt32 nSeqLen = maSeq.getLength();
+		const sal_Size nSeqLen = maSeq.getLength();
 		ErrCode nErr = ERRCODE_NONE;
 
 		if( nPos < nSeqLen )
@@ -153,7 +153,7 @@ ErrCode ImgProdLockBytes::ReadAt( sal_uI
 
 // ------------------------------------------------------------------------
 
-ErrCode ImgProdLockBytes::WriteAt( sal_uInt32 nPos, const void* pBuffer, sal_uInt32 nCount, sal_Size* pWritten )
+ErrCode ImgProdLockBytes::WriteAt( sal_Size nPos, const void* pBuffer, sal_Size nCount, sal_Size* pWritten )
 {
 	if( GetStream() )
 		return SvLockBytes::WriteAt( nPos, pBuffer, nCount, pWritten );
@@ -173,7 +173,7 @@ ErrCode ImgProdLockBytes::Flush() const
 
 // ------------------------------------------------------------------------
 
-ErrCode ImgProdLockBytes::SetSize( sal_uInt32 nSize )
+ErrCode ImgProdLockBytes::SetSize( sal_Size nSize )
 {
 	if( GetStream() )
 		return SvLockBytes::SetSize( nSize );
diff -pur sc/source/ui/Accessibility/AccessibleDocumentPagePreview.cxx sc/source/ui/Accessibility/AccessibleDocumentPagePreview.cxx
--- openoffice.org.orig/sc/source/ui/Accessibility/AccessibleDocumentPagePreview.cxx	2006-12-19 14:25:59.000000000 +0100
+++ openoffice.org/sc/source/ui/Accessibility/AccessibleDocumentPagePreview.cxx	2007-03-09 11:04:22.000000000 +0100
@@ -1769,7 +1769,7 @@ sal_Int32 SAL_CALL ScAccessibleDocumentP
 	return nRet;
 }
 
-uno::Reference<XAccessible> SAL_CALL ScAccessibleDocumentPagePreview::getAccessibleChild(long nIndex)
+uno::Reference<XAccessible> SAL_CALL ScAccessibleDocumentPagePreview::getAccessibleChild(sal_Int32 nIndex)
 			    throw (uno::RuntimeException, lang::IndexOutOfBoundsException)
 {
 	ScUnoGuard aGuard;
diff -pur sc/source/ui/inc/AccessibleDocumentPagePreview.hxx sc/source/ui/inc/AccessibleDocumentPagePreview.hxx
--- openoffice.org.orig/sc/source/ui/inc/AccessibleDocumentPagePreview.hxx	2006-03-31 11:31:54.000000000 +0200
+++ openoffice.org/sc/source/ui/inc/AccessibleDocumentPagePreview.hxx	2007-03-09 11:03:38.000000000 +0100
@@ -85,7 +85,7 @@ public:
 
     ///	Return the specified child or NULL if index is invalid.
     virtual ::com::sun::star::uno::Reference< ::com::sun::star::accessibility::XAccessible> SAL_CALL
-    	getAccessibleChild(long nIndex)
+    	getAccessibleChild(sal_Int32 nIndex)
         throw (::com::sun::star::uno::RuntimeException,
 				::com::sun::star::lang::IndexOutOfBoundsException);
 
diff -pur sc/source/ui/inc/csvruler.hxx sc/source/ui/inc/csvruler.hxx
--- openoffice.org.orig/sc/source/ui/inc/csvruler.hxx	2007-03-12 13:55:20.000000000 +0100
+++ openoffice.org/sc/source/ui/inc/csvruler.hxx	2007-03-12 13:56:09.000000000 +0100
@@ -92,9 +92,9 @@ public:
 public:
     /** Sets position and size of the ruler. The height is calculated internally. */
     virtual void                SetPosSizePixel(
-                                    sal_Int32 nX, sal_Int32 nY,
-                                    sal_Int32 nWidth, sal_Int32 nHeight,
-                                    sal_uInt16 nFlags = WINDOW_POSSIZE_ALL );
+                                    long nX, long nY,
+                                    long nWidth, long nHeight,
+                                    USHORT nFlags = WINDOW_POSSIZE_ALL );
 
     /** Apply current layout data to the ruler. */
     void                        ApplyLayout( const ScCsvLayoutData& rOldData );
diff -pur sc/source/ui/dbgui/csvruler.cxx sc/source/ui/dbgui/csvruler.cxx
--- openoffice.org.orig/sc/source/ui/dbgui/csvruler.cxx	2007-03-12 13:59:42.000000000 +0100
+++ openoffice.org/sc/source/ui/dbgui/csvruler.cxx	2007-03-12 14:00:12.000000000 +0100
@@ -140,7 +140,7 @@ ScCsvRuler::~ScCsvRuler()
 // common ruler handling ------------------------------------------------------
 
 void ScCsvRuler::SetPosSizePixel(
-        sal_Int32 nX, sal_Int32 nY, sal_Int32 nWidth, sal_Int32 nHeight, sal_uInt16 nFlags )
+        long nX, long nY, long nWidth, long nHeight, USHORT nFlags )
 {
     if( nFlags & WINDOW_POSSIZE_HEIGHT )
         nHeight = GetTextHeight() + mnSplitSize + 2;
--- openoffice.org.orig/sc/source/ui/dbgui/csvtablebox.cxx	2007-03-12 14:45:29.000000000 +0100
+++ openoffice.org/sc/source/ui/dbgui/csvtablebox.cxx	2007-03-12 14:45:33.000000000 +0100
@@ -148,10 +148,10 @@ void ScCsvTableBox::InitControls()
 {
     maGrid.UpdateLayoutData();
 
-    sal_Int32 nScrollBarSize = GetSettings().GetStyleSettings().GetScrollBarSize();
+    long nScrollBarSize = GetSettings().GetStyleSettings().GetScrollBarSize();
     Size aWinSize = CalcOutputSize( GetSizePixel() );
-    sal_Int32 nDataWidth = aWinSize.Width() - nScrollBarSize;
-    sal_Int32 nDataHeight = aWinSize.Height() - nScrollBarSize;
+    long nDataWidth = aWinSize.Width() - nScrollBarSize;
+    long nDataHeight = aWinSize.Height() - nScrollBarSize;
 
     maData.mnWinWidth = nDataWidth;
     maData.mnWinHeight = nDataHeight;
diff -pur sd/source/ui/accessibility/AccessibleDocumentViewBase.cxx sd/source/ui/accessibility/AccessibleDocumentViewBase.cxx
--- openoffice.org.orig/sd/source/ui/accessibility/AccessibleDocumentViewBase.cxx	2006-12-12 17:47:49.000000000 +0100
+++ openoffice.org/sd/source/ui/accessibility/AccessibleDocumentViewBase.cxx	2007-03-09 16:22:48.000000000 +0100
@@ -311,7 +311,7 @@ sal_Int32 SAL_CALL
 
 
 Reference<XAccessible> SAL_CALL
-    AccessibleDocumentViewBase::getAccessibleChild (long nIndex)
+    AccessibleDocumentViewBase::getAccessibleChild (sal_Int32 nIndex)
     throw (uno::RuntimeException)
 {
     ::osl::MutexGuard aGuard (maMutex);
diff -pur sd/source/ui/accessibility/AccessibleDrawDocumentView.cxx sd/source/ui/accessibility/AccessibleDrawDocumentView.cxx
--- openoffice.org.orig/sd/source/ui/accessibility/AccessibleDrawDocumentView.cxx	2006-11-14 15:23:19.000000000 +0100
+++ openoffice.org/sd/source/ui/accessibility/AccessibleDrawDocumentView.cxx	2007-03-09 16:23:05.000000000 +0100
@@ -276,7 +276,7 @@ sal_Int32 SAL_CALL
 
 
 uno::Reference<XAccessible> SAL_CALL
-    AccessibleDrawDocumentView::getAccessibleChild (long nIndex)
+    AccessibleDrawDocumentView::getAccessibleChild (sal_Int32 nIndex)
     throw (::com::sun::star::uno::RuntimeException)
 {
     ::osl::ClearableMutexGuard aGuard (maMutex);
diff -pur sd/source/ui/accessibility/AccessibleOutlineView.cxx sd/source/ui/accessibility/AccessibleOutlineView.cxx
--- openoffice.org.orig/sd/source/ui/accessibility/AccessibleOutlineView.cxx	2006-12-12 17:48:28.000000000 +0100
+++ openoffice.org/sd/source/ui/accessibility/AccessibleOutlineView.cxx	2007-03-09 16:23:27.000000000 +0100
@@ -192,7 +192,7 @@ sal_Int32 SAL_CALL
 
 
 uno::Reference<XAccessible> SAL_CALL
-    AccessibleOutlineView::getAccessibleChild (long nIndex)
+    AccessibleOutlineView::getAccessibleChild (sal_Int32 nIndex)
     throw (::com::sun::star::uno::RuntimeException)
 {
     // Forward request to children manager.
diff -pur sd/source/ui/accessibility/AccessiblePageShape.cxx sd/source/ui/accessibility/AccessiblePageShape.cxx
--- openoffice.org.orig/sd/source/ui/accessibility/AccessiblePageShape.cxx	2006-12-12 17:48:43.000000000 +0100
+++ openoffice.org/sd/source/ui/accessibility/AccessiblePageShape.cxx	2007-03-09 16:23:49.000000000 +0100
@@ -125,7 +125,7 @@ sal_Int32 SAL_CALL
     an exception for a wrong index.
 */
 uno::Reference<XAccessible> SAL_CALL 
-    AccessiblePageShape::getAccessibleChild(long )
+    AccessiblePageShape::getAccessibleChild( sal_Int32 )
     throw (::com::sun::star::uno::RuntimeException)
 {
     throw lang::IndexOutOfBoundsException (
diff -pur sd/source/ui/inc/AccessibleDocumentViewBase.hxx sd/source/ui/inc/AccessibleDocumentViewBase.hxx
--- openoffice.org.orig/sd/source/ui/inc/AccessibleDocumentViewBase.hxx	2005-09-09 06:57:44.000000000 +0200
+++ openoffice.org/sd/source/ui/inc/AccessibleDocumentViewBase.hxx	2007-03-09 16:20:53.000000000 +0100
@@ -189,7 +189,7 @@ public:
     */
     virtual ::com::sun::star::uno::Reference<
         ::com::sun::star::accessibility::XAccessible> SAL_CALL
-        getAccessibleChild (long nIndex)
+        getAccessibleChild (sal_Int32 nIndex)
         throw (::com::sun::star::uno::RuntimeException);
 
 
diff -pur sd/source/ui/inc/AccessibleDrawDocumentView.hxx sd/source/ui/inc/AccessibleDrawDocumentView.hxx
--- openoffice.org.orig/sd/source/ui/inc/AccessibleDrawDocumentView.hxx	2005-09-09 06:58:01.000000000 +0200
+++ openoffice.org/sd/source/ui/inc/AccessibleDrawDocumentView.hxx	2007-03-09 16:24:04.000000000 +0100
@@ -84,7 +84,7 @@ public:
         throw (::com::sun::star::uno::RuntimeException);
 
     virtual ::com::sun::star::uno::Reference< ::com::sun::star::accessibility::XAccessible> SAL_CALL
-        getAccessibleChild (long nIndex)
+        getAccessibleChild (sal_Int32 nIndex)
         throw (::com::sun::star::uno::RuntimeException);
 
 
diff -pur sd/source/ui/inc/AccessibleOutlineView.hxx sd/source/ui/inc/AccessibleOutlineView.hxx
--- openoffice.org.orig/sd/source/ui/inc/AccessibleOutlineView.hxx	2006-12-12 18:30:24.000000000 +0100
+++ openoffice.org/sd/source/ui/inc/AccessibleOutlineView.hxx	2007-03-09 16:24:14.000000000 +0100
@@ -90,7 +90,7 @@ public:
         throw (::com::sun::star::uno::RuntimeException);
 
     virtual ::com::sun::star::uno::Reference< ::com::sun::star::accessibility::XAccessible> SAL_CALL
-        getAccessibleChild (long nIndex)
+        getAccessibleChild (sal_Int32 nIndex)
         throw (::com::sun::star::uno::RuntimeException);
 
     //=====  XAccessibleEventBroadcaster  ========================================
diff -pur sd/source/ui/inc/AccessiblePageShape.hxx sd/source/ui/inc/AccessiblePageShape.hxx
--- openoffice.org.orig/sd/source/ui/inc/AccessiblePageShape.hxx	2006-12-12 18:30:39.000000000 +0100
+++ openoffice.org/sd/source/ui/inc/AccessiblePageShape.hxx	2007-03-09 16:24:24.000000000 +0100
@@ -128,7 +128,7 @@ public:
     */
     virtual ::com::sun::star::uno::Reference<
             ::com::sun::star::accessibility::XAccessible> SAL_CALL 
-    	getAccessibleChild (long nIndex)
+    	getAccessibleChild (sal_Int32 nIndex)
         throw (::com::sun::star::uno::RuntimeException);
 
 
diff -pur sd/source/ui/toolpanel/SimpleLayoutWrapper.hxx sd/source/ui/toolpanel/SimpleLayoutWrapper.hxx
--- openoffice.org.orig/sd/source/ui/toolpanel/SimpleLayoutWrapper.hxx	2007-03-12 14:07:00.000000000 +0100
+++ openoffice.org/sd/source/ui/toolpanel/SimpleLayoutWrapper.hxx	2007-03-12 14:08:11.000000000 +0100
@@ -59,8 +59,8 @@ public:
     virtual ~SimpleLayoutWrapper (void);
 
     virtual Size GetPreferredSize (void);
-    virtual int GetPreferredWidth (int nHeight);
-    virtual int GetPreferredHeight (int nWidth);
+    virtual sal_Int32 GetPreferredWidth (sal_Int32 nHeight);
+    virtual sal_Int32 GetPreferredHeight (sal_Int32 nWidth);
     virtual bool IsResizable (void);
     virtual ::Window* GetWindow (void);
 
diff -pur sd/source/ui/toolpanel/SimpleLayoutWrapper.cxx sd/source/ui/toolpanel/SimpleLayoutWrapper.cxx
--- openoffice.org.orig/sd/source/ui/toolpanel/SimpleLayoutWrapper.cxx	2007-03-12 14:11:01.000000000 +0100
+++ openoffice.org/sd/source/ui/toolpanel/SimpleLayoutWrapper.cxx	2007-03-12 14:11:51.000000000 +0100
@@ -72,7 +72,7 @@ Size SimpleLayoutWrapper::GetPreferredSi
 
 
 
-int SimpleLayoutWrapper::GetPreferredWidth (int )
+sal_Int32 SimpleLayoutWrapper::GetPreferredWidth (sal_Int32)
 {
     return mnPreferredWidth;
 }
@@ -80,7 +80,7 @@ int SimpleLayoutWrapper::GetPreferredWid
 
 
 
-int SimpleLayoutWrapper::GetPreferredHeight (int )
+sal_Int32 SimpleLayoutWrapper::GetPreferredHeight (sal_Int32)
 {
     return mnPreferredHeight;
 }
diff -pur svx/inc/fmgridif.hxx svx/inc/fmgridif.hxx
--- openoffice.org.orig/svx/inc/fmgridif.hxx	2007-03-12 12:19:17.000000000 +0100
+++ openoffice.org/svx/inc/fmgridif.hxx	2007-03-12 12:19:57.000000000 +0100
@@ -288,7 +288,7 @@ public:
 	virtual void SAL_CALL setDesignMode(sal_Bool bOn) throw(::com::sun::star::uno::RuntimeException);
 
 // ::com::sun::star::awt::XView
-	virtual void SAL_CALL draw( long x, long y ) throw(::com::sun::star::uno::RuntimeException);
+	virtual void SAL_CALL draw( sal_Int32 x, sal_Int32 y ) throw(::com::sun::star::uno::RuntimeException);
 
 // ::com::sun::star::form::XBoundComponent
 	virtual void SAL_CALL addUpdateListener(const ::com::sun::star::uno::Reference< ::com::sun::star::form::XUpdateListener >& l) throw(::com::sun::star::uno::RuntimeException);
@@ -527,7 +527,7 @@ public:
 	void columnHidden(DbGridColumn* pColumn);
 
 // ::com::sun::star::awt::XView
-	virtual void SAL_CALL draw( long x, long y ) throw(::com::sun::star::uno::RuntimeException);
+	virtual void SAL_CALL draw( sal_Int32 x, sal_Int32 y ) throw(::com::sun::star::uno::RuntimeException);
 
 // ::com::sun::star::frame::XDispatchProvider
     virtual ::com::sun::star::uno::Reference< ::com::sun::star::frame::XDispatch >  SAL_CALL queryDispatch(const ::com::sun::star::util::URL& aURL, const ::rtl::OUString& aTargetFrameName, sal_Int32 nSearchFlags) throw(::com::sun::star::uno::RuntimeException);
diff -pur svx/inc/fmpage.hxx svx/inc/fmpage.hxx
--- openoffice.org.orig/svx/inc/fmpage.hxx	2006-11-14 13:37:34.000000000 +0100
+++ openoffice.org/svx/inc/fmpage.hxx	2007-03-09 16:15:03.000000000 +0100
@@ -58,12 +58,7 @@ class SdrView;
 class SfxJSArray;
 class HelpEvent;
 
-class SVX_DLLPUBLIC FmFormPage :
-#if SUPD < 396
-	public VCDrawPage
-#else
-	public SdrPage
-#endif
+class SVX_DLLPUBLIC FmFormPage : public SdrPage
 {
 	friend class FmFormObj;
 	FmFormPageImpl*     m_pImpl;
@@ -77,21 +72,19 @@ public:
 	FmFormPage(const FmFormPage& rPage);
 	~FmFormPage();
 
+	using SdrPage::NbcInsertObject;
+	using SdrPage::NbcRemoveObject;
+	using SdrPage::NbcReplaceObject;
+	using SdrPage::ReplaceObject;
+
 	virtual void  	SetModel(SdrModel* pNewModel);
 
 	virtual SdrPage* Clone() const;
-    using SdrPage::Clone;
 
-	virtual void	NbcInsertObject(SdrObject* pObj, sal_uInt32 nPos=CONTAINER_APPEND,
-									const SdrInsertReason* pReason=NULL);
-	virtual void	InsertObject(SdrObject* pObj, sal_uInt32 nPos=CONTAINER_APPEND,
+	virtual void	InsertObject(SdrObject* pObj, ULONG nPos = CONTAINER_APPEND,
 									const SdrInsertReason* pReason=NULL);
 
-	virtual SdrObject* NbcRemoveObject(sal_uInt32 nObjNum);
-	virtual SdrObject* RemoveObject(sal_uInt32 nObjNum);
-
-	virtual SdrObject* NbcReplaceObject(SdrObject* pNewObj, sal_uInt32 nObjNum);
-	virtual SdrObject* ReplaceObject(SdrObject* pNewObj, sal_uInt32 nObjNum);
+	virtual SdrObject* RemoveObject(ULONG nObjNum);
 
 	// Zugriff auf alle Formulare
 	const ::com::sun::star::uno::Reference< ::com::sun::star::container::XNameContainer>& GetForms( bool _bForceCreate = true ) const;
diff -pur svx/source/fmcomp/fmgridif.cxx svx/source/fmcomp/fmgridif.cxx
--- openoffice.org.orig/svx/source/fmcomp/fmgridif.cxx	2007-03-12 12:20:43.000000000 +0100
+++ openoffice.org/svx/source/fmcomp/fmgridif.cxx	2007-03-12 12:21:22.000000000 +0100
@@ -781,7 +781,7 @@ void SAL_CALL FmXGridControl::removeModi
 }
 
 //------------------------------------------------------------------------------
-void SAL_CALL FmXGridControl::draw( long x, long y ) throw( RuntimeException )
+void SAL_CALL FmXGridControl::draw( sal_Int32 x, sal_Int32 y ) throw( RuntimeException )
 {
     ::vos::OGuard aGuard( Application::GetSolarMutex() );
 	m_bInDraw = sal_True;
@@ -2507,7 +2507,7 @@ void FmXGridPeer::columnHidden(DbGridCol
 }
 
 //------------------------------------------------------------------------------
-void FmXGridPeer::draw( long x, long y ) throw( RuntimeException )
+void FmXGridPeer::draw( sal_Int32 x, sal_Int32 y ) throw( RuntimeException )
 {
 	FmGridControl* pGrid = (FmGridControl*) GetWindow();
 	sal_Int32 nOldFlags = pGrid->GetBrowserFlags();
diff -pur svx/source/form/fmpage.cxx svx/source/form/fmpage.cxx
--- openoffice.org.orig/svx/source/form/fmpage.cxx	2006-09-17 07:07:59.000000000 +0200
+++ openoffice.org/svx/source/form/fmpage.cxx	2007-03-09 16:09:40.000000000 +0100
@@ -209,15 +209,7 @@ SdrPage* FmFormPage::Clone() const
 }
 
 //------------------------------------------------------------------
-void FmFormPage::NbcInsertObject(SdrObject* pObj,
-								 sal_uInt32 nPos,
-								 const SdrInsertReason* pReason)
-{
-	SdrPage::NbcInsertObject(pObj, nPos, pReason);
-}
-
-//------------------------------------------------------------------
-void FmFormPage::InsertObject(SdrObject* pObj, sal_uInt32 nPos,
+void FmFormPage::InsertObject(SdrObject* pObj, ULONG nPos,
 							  const SdrInsertReason* pReason)
 {
 	SdrPage::InsertObject( pObj, nPos, pReason );
@@ -308,13 +300,7 @@ sal_Bool FmFormPage::RequestHelp( Window
 }
 
 //------------------------------------------------------------------
-SdrObject* FmFormPage::NbcRemoveObject(sal_uInt32 nObjNum)
-{
-	return SdrPage::NbcRemoveObject(nObjNum);
-}
-
-//------------------------------------------------------------------
-SdrObject* FmFormPage::RemoveObject(sal_uInt32 nObjNum)
+SdrObject* FmFormPage::RemoveObject(ULONG nObjNum)
 {
 	SdrObject* pObj = SdrPage::RemoveObject(nObjNum);
 #ifndef SVX_LIGHT
@@ -323,16 +309,3 @@ SdrObject* FmFormPage::RemoveObject(sal_
 #endif
 	return pObj;
 }
-
-//------------------------------------------------------------------
-SdrObject* FmFormPage::NbcReplaceObject(SdrObject* pNewObj, sal_uInt32 nObjNum)
-{
-	return SdrPage::NbcReplaceObject(pNewObj, nObjNum);
-}
-
-//------------------------------------------------------------------
-SdrObject* FmFormPage::ReplaceObject(SdrObject* pNewObj, sal_uInt32 nObjNum)
-{
-	return SdrPage::ReplaceObject(pNewObj, nObjNum);
-}
-
diff -pur sw/source/ui/inc/glossary.hxx sw/source/ui/inc/glossary.hxx
--- openoffice.org.orig/sw/source/ui/inc/glossary.hxx	2006-08-14 19:41:51.000000000 +0200
+++ openoffice.org/sw/source/ui/inc/glossary.hxx	2007-03-09 16:39:41.000000000 +0100
@@ -115,12 +115,12 @@ class SwGlTreeListBox : public SvTreeLis
 	virtual sal_Bool    NotifyMoving(   SvLBoxEntry*  pTarget,
 									SvLBoxEntry*  pEntry,
 									SvLBoxEntry*& rpNewParent,
-									sal_uInt32&        rNewChildPos
+									ULONG&        rNewChildPos
 								);
 	virtual sal_Bool	NotifyCopying(	SvLBoxEntry*  pTarget,
 									SvLBoxEntry*  pEntry,
 									SvLBoxEntry*& rpNewParent,
-									sal_uInt32&		  rNewChildPos);
+									ULONG&		  rNewChildPos);
 public:
 	SwGlTreeListBox(Window* pParent, const ResId& rResId);
 
diff -pur sw/source/ui/misc/glossary.cxx sw/source/ui/misc/glossary.cxx
--- openoffice.org.orig/sw/source/ui/misc/glossary.cxx	2006-09-17 01:07:03.000000000 +0200
+++ openoffice.org/sw/source/ui/misc/glossary.cxx	2007-03-09 16:40:55.000000000 +0100
@@ -1176,7 +1176,7 @@ sal_Bool    SwGlTreeListBox::NotifyAccep
 sal_Bool  SwGlTreeListBox::NotifyMoving(   SvLBoxEntry*  pTarget,
 									SvLBoxEntry*  pEntry,
 									SvLBoxEntry*& rpNewParent,
-									sal_uInt32&        rNewChildPos
+									ULONG&        rNewChildPos
 								)
 {
 	pDragEntry = 0;
@@ -1224,7 +1224,7 @@ sal_Bool  SwGlTreeListBox::NotifyMoving(
 sal_Bool  SwGlTreeListBox::NotifyCopying(   SvLBoxEntry*  pTarget,
 									SvLBoxEntry*  pEntry,
 									SvLBoxEntry*& rpNewParent,
-									sal_uInt32&        rNewChildPos
+									ULONG&        rNewChildPos
 								)
 {
 	pDragEntry = 0;
diff -pur tools/inc/contnr.hxx tools/inc/contnr.hxx
--- openoffice.org.orig/tools/inc/contnr.hxx	2006-06-19 15:23:55.000000000 +0200
+++ openoffice.org/tools/inc/contnr.hxx	2007-03-09 13:59:30.000000000 +0100
@@ -43,6 +43,8 @@
 #include <solar.h>
 #endif
 
+#include <limits.h>
+
 class CBlock;
 
 // -------------
@@ -52,8 +54,8 @@ class CBlock;
 // Maximale Blockgroesse
 #define CONTAINER_MAXBLOCKSIZE      ((USHORT)0x3FF0)
 
-#define CONTAINER_APPEND            ((ULONG)0xFFFFFFFF)
-#define CONTAINER_ENTRY_NOTFOUND    ((ULONG)0xFFFFFFFF)
+#define CONTAINER_APPEND            ULONG_MAX
+#define CONTAINER_ENTRY_NOTFOUND    ULONG_MAX
 
 class TOOLS_DLLPUBLIC Container
 {
diff -pur binfilter/bf_forms/source/component/forms_imgprod.cxx binfilter/bf_forms/source/component/forms_imgprod.cxx
--- openoffice.org.orig/binfilter/bf_forms/source/component/forms_imgprod.cxx	2007-01-02 17:46:09.000000000 +0100
+++ openoffice.org/binfilter/bf_forms/source/component/forms_imgprod.cxx	2007-03-09 16:46:57.000000000 +0100
@@ -70,10 +70,10 @@ public:
 						ImgProdLockBytes( ::com::sun::star::uno::Reference< ::com::sun::star::io::XInputStream > & rStreamRef );
 	virtual				~ImgProdLockBytes();
 
-	virtual ErrCode		ReadAt( sal_uInt32 nPos, void* pBuffer, sal_uInt32 nCount, sal_Size* pRead ) const;
-	virtual ErrCode		WriteAt( sal_uInt32 nPos, const void* pBuffer, sal_uInt32 nCount, sal_Size* pWritten );
+	virtual ErrCode		ReadAt( sal_Size nPos, void* pBuffer, sal_Size nCount, sal_Size* pRead ) const;
+	virtual ErrCode		WriteAt( sal_Size nPos, const void* pBuffer, sal_Size nCount, sal_Size* pWritten );
 	virtual ErrCode		Flush() const;
-	virtual ErrCode		SetSize( sal_uInt32 nSize );
+	virtual ErrCode		SetSize( sal_Size nSize );
 	virtual ErrCode		Stat( SvLockBytesStat*, SvLockBytesStatFlag ) const;
 };
 
@@ -119,7 +119,7 @@ ImgProdLockBytes::~ImgProdLockBytes()
 
 // ------------------------------------------------------------------------
 
-ErrCode ImgProdLockBytes::ReadAt( sal_uInt32 nPos, void* pBuffer, sal_uInt32 nCount, sal_Size* pRead ) const
+ErrCode ImgProdLockBytes::ReadAt( sal_Size nPos, void* pBuffer, sal_Size nCount, sal_Size* pRead ) const
 {
 	if( GetStream() )
 	{
@@ -130,7 +130,7 @@ ErrCode ImgProdLockBytes::ReadAt( sal_uI
 	}
 	else
 	{
-		const sal_uInt32 nSeqLen = maSeq.getLength();
+		const sal_Size nSeqLen = maSeq.getLength();
 		ErrCode nErr = ERRCODE_NONE;
 
 		if( nPos < nSeqLen )
@@ -150,7 +150,7 @@ ErrCode ImgProdLockBytes::ReadAt( sal_uI
 
 // ------------------------------------------------------------------------
 
-ErrCode ImgProdLockBytes::WriteAt( sal_uInt32 nPos, const void* pBuffer, sal_uInt32 nCount, sal_Size* pWritten )
+ErrCode ImgProdLockBytes::WriteAt( sal_Size nPos, const void* pBuffer, sal_Size nCount, sal_Size* pWritten )
 {
 	if( GetStream() )
 		return SvLockBytes::WriteAt( nPos, pBuffer, nCount, pWritten );
@@ -170,7 +170,7 @@ ErrCode ImgProdLockBytes::Flush() const
 
 // ------------------------------------------------------------------------
 
-ErrCode ImgProdLockBytes::SetSize( sal_uInt32 nSize )
+ErrCode ImgProdLockBytes::SetSize( sal_Size nSize )
 {
 	if( GetStream() )
 		return SvLockBytes::SetSize( nSize );
diff -pur binfilter/bf_svx/source/form/svx_fmpage.cxx binfilter/bf_svx/source/form/svx_fmpage.cxx
--- openoffice.org.orig/binfilter/bf_svx/source/form/svx_fmpage.cxx	2006-10-27 22:56:08.000000000 +0200
+++ openoffice.org/binfilter/bf_svx/source/form/svx_fmpage.cxx	2007-03-09 16:42:39.000000000 +0100
@@ -194,15 +194,7 @@ namespace binfilter {
 /*N*/ }
 
 //------------------------------------------------------------------
-/*N*/ void FmFormPage::NbcInsertObject(SdrObject* pObj,
-/*N*/ 								 sal_uInt32 nPos,
-/*N*/ 								 const SdrInsertReason* pReason)
-/*N*/ {
-/*N*/ 	SdrPage::NbcInsertObject(pObj, nPos, pReason);
-/*N*/ }
-
-//------------------------------------------------------------------
-/*N*/ void FmFormPage::InsertObject(SdrObject* pObj, sal_uInt32 nPos,
+/*N*/ void FmFormPage::InsertObject(SdrObject* pObj, ULONG nPos,
 /*N*/ 							  const SdrInsertReason* pReason)
 /*N*/ {
 /*N*/ 	SdrPage::InsertObject( pObj, nPos, pReason );
@@ -212,12 +204,6 @@ namespace binfilter {
 /*N*/ #endif
 /*N*/ }
 
-#ifndef SVX_LIGHT
-//------------------------------------------------------------------
-
-//------------------------------------------------------------------
-#endif
-
 //------------------------------------------------------------------
 /*N*/ const ::com::sun::star::uno::Reference< ::com::sun::star::container::XNameContainer > & FmFormPage::GetForms() const
 /*N*/ {
@@ -230,11 +216,7 @@ namespace binfilter {
 /*N*/ }
 
 //------------------------------------------------------------------
-
-//------------------------------------------------------------------
-
-//------------------------------------------------------------------
-/*N*/ SdrObject* FmFormPage::RemoveObject(sal_uInt32 nObjNum)
+/*N*/ SdrObject* FmFormPage::RemoveObject(ULONG nObjNum)
 /*N*/ {
 /*N*/ 	SdrObject* pObj = SdrPage::RemoveObject(nObjNum);
 /*N*/ #ifndef SVX_LIGHT
@@ -244,21 +226,4 @@ namespace binfilter {
 /*N*/ 	return pObj;
 /*N*/ }
 
-//------------------------------------------------------------------
-
-//------------------------------------------------------------------
-/*N*/ SdrObject* FmFormPage::ReplaceObject(SdrObject* pNewObj, sal_uInt32 nObjNum)
-/*N*/ {
-/*N*/ 	return SdrPage::ReplaceObject(pNewObj, nObjNum);
-/*N*/ }
-
-
-//------------------------------------------------------------------
-
-
-
-
-
-
-
 }
diff -pur binfilter/inc/bf_svx/fmpage.hxx binfilter/inc/bf_svx/fmpage.hxx
--- openoffice.org.orig/binfilter/inc/bf_svx/fmpage.hxx	2006-10-28 05:33:18.000000000 +0200
+++ openoffice.org/binfilter/inc/bf_svx/fmpage.hxx	2007-03-09 16:33:05.000000000 +0100
@@ -56,12 +56,7 @@ class FmFormPageImpl;	// haelt die Liste
 class SfxJSArray;
 class HelpEvent;
 
-class FmFormPage :
-#if SUPD < 396
-	public VCDrawPage
-#else
-	public SdrPage
-#endif
+class FmFormPage : public SdrPage
 {
 	friend class FmFormObj;
 	FmFormPageImpl* pImpl;
@@ -72,20 +67,19 @@ public:
 	FmFormPage(FmFormModel& rModel,StarBASIC*, FASTBOOL bMasterPage=sal_False);
 	~FmFormPage();
 
+	using SdrPage::NbcInsertObject;
+	using SdrPage::ReplaceObject;
+
 	virtual void  	WriteData(SvStream& rOut) const;
 	virtual void  	ReadData(const SdrIOHeader& rHead, SvStream& rIn);
 	virtual void  	SetModel(SdrModel* pNewModel);
 
 	virtual SdrPage* Clone() const;
 
-	virtual void	NbcInsertObject(SdrObject* pObj, sal_uInt32 nPos=CONTAINER_APPEND,
+	virtual void	InsertObject(SdrObject* pObj, ULONG nPos = CONTAINER_APPEND,
 									const SdrInsertReason* pReason=NULL);
-	virtual void	InsertObject(SdrObject* pObj, sal_uInt32 nPos=CONTAINER_APPEND,
-									const SdrInsertReason* pReason=NULL);
-
-	virtual SdrObject* RemoveObject(sal_uInt32 nObjNum);
 
-	virtual SdrObject* ReplaceObject(SdrObject* pNewObj, sal_uInt32 nObjNum);
+	virtual SdrObject* RemoveObject(ULONG nObjNum);
 
 #ifndef SVX_LIGHT
 	/**	Insert _pClone into the page.
