diff -u -r a/kernel/conftest.sh b/kernel/conftest.sh
--- a/kernel/conftest.sh	2018-10-26 21:12:23.587468332 +0200
+++ b/kernel/conftest.sh	2018-10-26 21:15:22.541274337 +0200
@@ -2363,17 +2363,17 @@
 
         ;;
 
-        drm_mode_connector_list_update_has_merge_type_bits_arg)
+        drm_connector_list_update_has_merge_type_bits_arg)
             #
-            # Detect if drm_mode_connector_list_update() has a
+            # Detect if drm_connector_list_update() has a
             # 'merge_type_bits' second argument.  This argument was
             # remove by:
             #   2015-12-03  6af3e6561243f167dabc03f732d27ff5365cd4a4
             #
             CODE="
             #include <drm/drmP.h>
-            void conftest_drm_mode_connector_list_update_has_merge_type_bits_arg(void) {
-                drm_mode_connector_list_update(
+            void conftest_drm_connector_list_update_has_merge_type_bits_arg(void) {
+                drm_connector_list_update(
                     NULL,  /* struct drm_connector *connector */
                     true); /* bool merge_type_bits */
             }"
diff -u -r a/kernel/nvidia-drm/nvidia-drm-connector.c b/kernel/nvidia-drm/nvidia-drm-connector.c
--- a/kernel/nvidia-drm/nvidia-drm-connector.c	2018-08-22 02:55:29.000000000 +0200
+++ b/kernel/nvidia-drm/nvidia-drm-connector.c	2018-10-26 21:14:14.891095557 +0200
@@ -226,7 +226,7 @@
 
 
     if (nv_connector->edid != NULL) {
-        drm_mode_connector_update_edid_property(
+        drm_connector_update_edid_property(
             connector, nv_connector->edid);
     }
 
diff -u -r a/kernel/nvidia-drm/nvidia-drm-encoder.c b/kernel/nvidia-drm/nvidia-drm-encoder.c
--- a/kernel/nvidia-drm/nvidia-drm-encoder.c	2018-08-22 02:55:29.000000000 +0200
+++ b/kernel/nvidia-drm/nvidia-drm-encoder.c	2018-10-26 21:13:46.644354274 +0200
@@ -216,7 +216,7 @@
 
     /* Attach encoder and connector */
 
-    ret = drm_mode_connector_attach_encoder(connector, encoder);
+    ret = drm_connector_attach_encoder(connector, encoder);
 
     if (ret != 0) {
         NV_DRM_DEV_LOG_ERR(
diff -u -r a/kernel/nvidia-drm/nvidia-drm.Kbuild b/kernel/nvidia-drm/nvidia-drm.Kbuild
--- a/kernel/nvidia-drm/nvidia-drm.Kbuild	2018-08-21 23:45:43.000000000 +0200
+++ b/kernel/nvidia-drm/nvidia-drm.Kbuild	2018-10-26 21:15:39.911320257 +0200
@@ -74,7 +74,7 @@
 NV_CONFTEST_TYPE_COMPILE_TESTS += drm_driver_has_set_busid
 NV_CONFTEST_TYPE_COMPILE_TESTS += drm_crtc_state_has_connectors_changed
 NV_CONFTEST_TYPE_COMPILE_TESTS += drm_init_function_args
-NV_CONFTEST_TYPE_COMPILE_TESTS += drm_mode_connector_list_update_has_merge_type_bits_arg
+NV_CONFTEST_TYPE_COMPILE_TESTS += drm_connector_list_update_has_merge_type_bits_arg
 NV_CONFTEST_TYPE_COMPILE_TESTS += drm_helper_mode_fill_fb_struct
 NV_CONFTEST_TYPE_COMPILE_TESTS += drm_master_drop_has_from_release_arg
 NV_CONFTEST_TYPE_COMPILE_TESTS += drm_driver_unload_has_int_return_type
