diff -rupN a/fltk-config.in b/fltk-config.in
--- a/fltk-config.in	2011-07-19 06:49:30.000000000 +0200
+++ b/fltk-config.in	2013-12-26 07:50:58.393155344 +0100
@@ -357,26 +357,26 @@ if test "$echo_ldstaticflags" = "yes"; t
 fi
 
 if test "$echo_libs" = "yes"; then
-    USELIBS="$libdir/libfltk.a"
+    USELIBS="$libdir/libfltk.so"
 
     if test x$use_forms = xyes; then
-        USELIBS="$libdir/libfltk_forms.a $USELIBS"
+        USELIBS="$libdir/libfltk_forms.so $USELIBS"
     fi
 
     if test x$use_gl = xyes; then
-        USELIBS="$libdir/libfltk_gl.a $USELIBS"
+        USELIBS="$libdir/libfltk_gl.so $USELIBS"
     fi
 
     if test x$use_cairo = xyes; then
-        USELIBS="$libdir/libfltk_cairo.a $USELIBS"
+        USELIBS="$libdir/libfltk_cairo.so $USELIBS"
     fi
 
     if test x$use_images = xyes; then
-        USELIBS="$libdir/libfltk_images.a $USELIBS"
+        USELIBS="$libdir/libfltk_images.so $USELIBS"
 
         for lib in fltk_jpeg fltk_png fltk_z; do
-            if test -f $libdir/lib$lib.a; then
-                USELIBS="$libdir/lib$lib.a $USELIBS"
+            if test -f $libdir/lib$lib.so; then
+                USELIBS="$libdir/lib$lib.so $USELIBS"
             fi
 	done
     fi
