--- doc/Makefile.am	(revision 49837)
+++ doc/Makefile.am	(working copy)
@@ -123,6 +123,7 @@ 
 	$(AM_V_POD2MAN)$(POD2MAN)                       \
 	--center="The Wireshark Network Analyzer"       \
 	--release=$(VERSION)                            \
+	--utf8                                          \
 	$< > $@
 
 .pod.html: ../config.h $(top_srcdir)/docbook/ws.css
--- doc/wireshark.pod.template	(revision 49837)
+++ doc/wireshark.pod.template	(working copy)
@@ -1,4 +1,9 @@ 
+=begin man
+
+=encoding utf8
+
+=end man

 =head1 NAME
 
 wireshark - Interactively dump and analyze network traffic
