Index: dasher-4.11/Src/Gtk2/game_mode_helper.cpp
===================================================================
--- dasher-4.11.orig/Src/Gtk2/game_mode_helper.cpp	2009-12-18 13:40:29.000000000 -0500
+++ dasher-4.11/Src/Gtk2/game_mode_helper.cpp	2011-11-11 12:33:12.349741593 -0500
@@ -324,6 +324,7 @@
 				   GTK_DIALOG_MODAL,
 				   GTK_MESSAGE_INFO,
 				   GTK_BUTTONS_OK,
+                                   "%s",
 				   message);
   gtk_dialog_run (GTK_DIALOG (pDialog));
   gtk_widget_destroy (pDialog);
