--- cmake/config.h.cmake.orig	2010-10-29 12:51:05.000000000 -0400
+++ cmake/config.h.cmake	2010-10-29 12:51:48.000000000 -0400
@@ -76,9 +76,7 @@
 /* use Nvwa memory leak detector */
 #cmakedefine FIND_LEAKS
 
-#if defined(OSS_SUPPORT) || defined (USE_ARTS) || defined(USE_SDL) || defined(__APPLE__) || defined(__WIN32) || defined(ALSA_SUPPORT)
 #define HAVE_AUDIO
-#endif
 
 #define HAVE_ENCODER
 
