diff -urdH avidemux_2.4.1-orig/avidemux/ADM_libraries/ADM_mplex/inputstrm.cpp avidemux_2.4.1/avidemux/ADM_libraries/ADM_mplex/inputstrm.cpp
--- avidemux_2.4.1-orig/avidemux/ADM_libraries/ADM_mplex/inputstrm.cpp	2008-03-11 20:55:19.000000000 +0300
+++ avidemux_2.4.1/avidemux/ADM_libraries/ADM_mplex/inputstrm.cpp	2008-03-12 07:15:32.000000000 +0300
@@ -28,6 +28,8 @@
 #include "inputstrm.hpp"
 #include "multiplexor.hpp"
 
+#include <climits>
+
 MuxStream::MuxStream() : init(false) 
 {
 }
diff -urdH avidemux_2.4.1-orig/avidemux/ADM_libraries/ADM_mplex/multiplexor.cpp avidemux_2.4.1/avidemux/ADM_libraries/ADM_mplex/multiplexor.cpp
--- avidemux_2.4.1-orig/avidemux/ADM_libraries/ADM_mplex/multiplexor.cpp	2008-03-11 20:55:19.000000000 +0300
+++ avidemux_2.4.1/avidemux/ADM_libraries/ADM_mplex/multiplexor.cpp	2008-03-12 07:15:32.000000000 +0300
@@ -35,7 +35,7 @@
 #include "zalphastrm.hpp"
 #endif
 #include "multiplexor.hpp"
-
+#include <cstring>
 
 /****************
  *
diff -urdH avidemux_2.4.1-orig/avidemux/ADM_libraries/ADM_mplex/padstrm.cpp avidemux_2.4.1/avidemux/ADM_libraries/ADM_mplex/padstrm.cpp
--- avidemux_2.4.1-orig/avidemux/ADM_libraries/ADM_mplex/padstrm.cpp	2008-03-11 20:55:19.000000000 +0300
+++ avidemux_2.4.1/avidemux/ADM_libraries/ADM_mplex/padstrm.cpp	2008-03-12 07:15:32.000000000 +0300
@@ -26,6 +26,7 @@
 
 #include "padstrm.hpp"
 
+#include <cstring>
 
 
 //
diff -urdH avidemux_2.4.1-orig/avidemux/ADM_script/DirectorySearch.h avidemux_2.4.1/avidemux/ADM_script/DirectorySearch.h
--- avidemux_2.4.1-orig/avidemux/ADM_script/DirectorySearch.h	2008-03-11 20:55:19.000000000 +0300
+++ avidemux_2.4.1/avidemux/ADM_script/DirectorySearch.h	2008-03-12 07:15:32.000000000 +0300
@@ -23,6 +23,7 @@
 #include <dirent.h>
 #endif
 #include "StdFile.h"
+#include <cstring>
 
 // create a non file bit for unix
 #define _A_NONFILE	0x03
diff -urdH avidemux_2.4.1-orig/avidemux/ADM_userInterfaces/ADM_GTK/ADM_toolkit_gtk/ADM_jogshuttle.cpp avidemux_2.4.1/avidemux/ADM_userInterfaces/ADM_GTK/ADM_toolkit_gtk/ADM_jogshuttle.cpp
--- avidemux_2.4.1-orig/avidemux/ADM_userInterfaces/ADM_GTK/ADM_toolkit_gtk/ADM_jogshuttle.cpp	2008-03-11 20:55:19.000000000 +0300
+++ avidemux_2.4.1/avidemux/ADM_userInterfaces/ADM_GTK/ADM_toolkit_gtk/ADM_jogshuttle.cpp	2008-03-12 07:15:32.000000000 +0300
@@ -41,6 +41,8 @@
 #include "config.h"
 
 #ifdef USE_JOG
+#include <cstdlib>
+#include <cstring>
 #include <string>
 using std::string;
 #include <iostream>
diff -urdH avidemux_2.4.1-orig/avidemux/ADM_userInterfaces/ADM_GTK/ADM_toolkit_gtk/mediactrl.h avidemux_2.4.1/avidemux/ADM_userInterfaces/ADM_GTK/ADM_toolkit_gtk/mediactrl.h
--- avidemux_2.4.1-orig/avidemux/ADM_userInterfaces/ADM_GTK/ADM_toolkit_gtk/mediactrl.h	2008-03-11 20:55:19.000000000 +0300
+++ avidemux_2.4.1/avidemux/ADM_userInterfaces/ADM_GTK/ADM_toolkit_gtk/mediactrl.h	2008-03-12 07:15:32.000000000 +0300
@@ -97,6 +97,8 @@
 extern "C" {
 #endif
 
+#include <time.h>
+
 struct media_ctrl_device;
 
 struct media_ctrl_key
diff -urdH avidemux_2.4.1-orig/avidemux/ADM_videoFilter/ADM_guiResize.cpp avidemux_2.4.1/avidemux/ADM_videoFilter/ADM_guiResize.cpp
--- avidemux_2.4.1-orig/avidemux/ADM_videoFilter/ADM_guiResize.cpp	2008-03-11 20:55:19.000000000 +0300
+++ avidemux_2.4.1/avidemux/ADM_videoFilter/ADM_guiResize.cpp	2008-03-12 07:15:32.000000000 +0300
@@ -41,7 +41,7 @@
 #include <ADM_assert.h>
 
 
-static int getResizeParams(uint32_t * w, uint32_t * h, uint32_t * algo,uint32_t w,uint32_t h,uint32_t fps);
+static int getResizeParams(uint32_t * w, uint32_t * h, uint32_t * algo, uint32_t ow, uint32_t oh, uint32_t fps);
 
 
 extern uint8_t DIA_resize(uint32_t *width,uint32_t *height,uint32_t *algo,uint32_t originalw, uint32_t originalh,uint32_t fps1000);
diff -urdH avidemux_2.4.1-orig/avidemux/gui_action.cpp avidemux_2.4.1/avidemux/gui_action.cpp
--- avidemux_2.4.1-orig/avidemux/gui_action.cpp	2008-03-11 20:55:19.000000000 +0300
+++ avidemux_2.4.1/avidemux/gui_action.cpp	2008-03-12 07:28:27.000000000 +0300
@@ -1,4 +1,5 @@
-#include <stdio.h>
+#include <cstdio>
+#include <cstring>
 #include <vector>
 using std::vector;
 #include <utility>
