diff -p -up avidemux_2.5.0/avidemux/ADM_userInterfaces/ADM_GTK/ADM_dialog/DIA_properties.cpp.format-strings avidemux_2.5.0/avidemux/ADM_userInterfaces/ADM_GTK/ADM_dialog/DIA_properties.cpp
--- avidemux_2.5.0/avidemux/ADM_userInterfaces/ADM_GTK/ADM_dialog/DIA_properties.cpp.format-strings	2008-05-17 19:45:51.000000000 +0200
+++ avidemux_2.5.0/avidemux/ADM_userInterfaces/ADM_GTK/ADM_dialog/DIA_properties.cpp	2009-07-09 21:24:29.000000000 +0200
@@ -84,10 +84,10 @@ void DIA_properties( void )
               switch (wavinfo->channels)
                 {
                 case 1:
-                    sprintf(text, QT_TR_NOOP("Mono"));
+		  sprintf(text, "%s", QT_TR_NOOP("Mono"));
                     break;
                 case 2:
-                    sprintf(text, QT_TR_NOOP("Stereo"));
+		  sprintf(text, "%s", QT_TR_NOOP("Stereo"));
                     break;
                 default:
                     sprintf(text, "%d",wavinfo->channels);
diff -p -up avidemux_2.5.0/avidemux/ADM_userInterfaces/ADM_QT4/ADM_dialog/Q_props.cpp.format-strings avidemux_2.5.0/avidemux/ADM_userInterfaces/ADM_QT4/ADM_dialog/Q_props.cpp
--- avidemux_2.5.0/avidemux/ADM_userInterfaces/ADM_QT4/ADM_dialog/Q_props.cpp.format-strings	2009-02-05 21:04:06.000000000 +0100
+++ avidemux_2.5.0/avidemux/ADM_userInterfaces/ADM_QT4/ADM_dialog/Q_props.cpp	2009-07-09 21:24:29.000000000 +0200
@@ -76,10 +76,10 @@ propWindow::propWindow(QWidget *parent) 
               switch (wavinfo->channels)
                 {
                 case 1:
-                    sprintf(text, QT_TR_NOOP("Mono"));
+		  sprintf(text, "%s", QT_TR_NOOP("Mono"));
                     break;
                 case 2:
-                    sprintf(text, QT_TR_NOOP("Stereo"));
+                    sprintf(text, "%s", QT_TR_NOOP("Stereo"));
                     break;
                 default:
                     sprintf(text, "%d",wavinfo->channels);
diff -p -up avidemux_2.5.0/avidemux/gtk_gui.cpp.format-strings avidemux_2.5.0/avidemux/gtk_gui.cpp
--- avidemux_2.5.0/avidemux/gtk_gui.cpp.format-strings	2009-04-20 17:58:54.000000000 +0200
+++ avidemux_2.5.0/avidemux/gtk_gui.cpp	2009-07-09 21:24:29.000000000 +0200
@@ -1352,7 +1352,7 @@ A_loadAC3 (char *name)
   if (ac3->open (path) == 0)
     {
       GUI_Error_HIG (QT_TR_NOOP("Failed to open the file"), QT_TR_NOOP("Not a WAV file?"));
-      printf (QT_TR_NOOP("WAV open file failed..."));
+      printf ("%s", QT_TR_NOOP("WAV open file failed..."));
       delete ac3;
 	  delete [] path;
       return 0;
