diff -up gnome-python-extras-2.19.1/docs/gtkmozembed/html/class-gtkmozembed.html.old gnome-python-extras-2.19.1/docs/gtkmozembed/html/class-gtkmozembed.html
diff -up gnome-python-extras-2.19.1/docs/gtkmozembed/html/pygtkmozembed.devhelp.old gnome-python-extras-2.19.1/docs/gtkmozembed/html/pygtkmozembed.devhelp
diff -up gnome-python-extras-2.19.1/docs/gtkmozembed/gtkmozembed-gtkmozembed.xml.old gnome-python-extras-2.19.1/docs/gtkmozembed/gtkmozembed-gtkmozembed.xml
diff -up gnome-python-extras-2.19.1/gtkmozembed/gtkmozembed.override.old gnome-python-extras-2.19.1/gtkmozembed/gtkmozembed.override
diff -up gnome-python-extras-2.19.1/gtkmozembed/gtkmozembed.defs.old gnome-python-extras-2.19.1/gtkmozembed/gtkmozembed.defs
--- gnome-python-extras-2.19.1/gtkmozembed/gtkmozembed.defs.old	2007-06-05 16:52:30.000000000 +0200
+++ gnome-python-extras-2.19.1/gtkmozembed/gtkmozembed.defs	2007-11-27 12:54:42.000000000 +0100
@@ -123,6 +123,14 @@
   )
 )
 
+(define-function set_path
+  (c-name "gtk_moz_embed_set_path")
+  (return-type "none")
+  (parameters
+    '("char*" "aPath")
+  )
+)
+
 (define-function gtk_moz_embed_set_profile_path
   (c-name "gtk_moz_embed_set_profile_path_deprecated")
   (return-type "none")
