From 8d35b36baa8204b68c9520ec92327e34c1d7c580 Mon Sep 17 00:00:00 2001
From: Ritesh Khadgaray <khadgaray@gmail.com>
Date: Sun, 04 Jul 2010 09:52:07 +0000
Subject: remove deprecated GTK+ symbols, GTK_WIDGET_SET_FLAGS.

---
diff --git a/src/fe-gnome/xtext.c b/src/fe-gnome/xtext.c
index 92562a3..51503f8 100644
--- a/src/fe-gnome/xtext.c
+++ b/src/fe-gnome/xtext.c
@@ -963,7 +963,7 @@ gtk_xtext_realize (GtkWidget * widget)
 	GdkColor col;
 	GdkColormap *cmap;
 
-	GTK_WIDGET_SET_FLAGS (widget, GTK_REALIZED);
+	gtk_widget_set_realized (widget, TRUE);
 	xtext = GTK_XTEXT (widget);
 
 	attributes.x = widget->allocation.x;
--
cgit v0.8.3.1

