--- driver/ntoskernel.h (revision 2701)
+++ driver/ntoskernel.h (working copy)
@@ -875,7 +875,7 @@
 #define u64_low_32(x) ((u32)x)
 #define u64_high_32(x) ((u32)(x >> 32))

-#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,31)
+//#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,31)
 static inline u64 cmpxchg8b(volatile u64 *ptr, u64 old, u64 new)
 {
        u64 prev;
@@ -887,7 +887,7 @@
                : "A" (old), "b" (u64_low_32(new)), "c" (u64_high_32(new)));
        return prev;
 }
-#endif
+//#endif

 /* slist routines below update slist atomically - no need for
  * spinlocks */
