Index: /branches/release-0-23-fixes/mythplugins/mythnetvision/mythnetvision/netsearch.cpp
===================================================================
--- /branches/release-0-23-fixes/mythplugins/mythnetvision/mythnetvision/netsearch.cpp	(revision 24265)
+++ /branches/release-0-23-fixes/mythplugins/mythnetvision/mythnetvision/netsearch.cpp	(revision 26391)
@@ -45,7 +45,7 @@
       m_progress(NULL),              m_busyPopup(NULL),
       m_okPopup(NULL),               m_popupStack(),
-      m_netSearch(),                 m_currentSearch(NULL),
-      m_currentGrabber(0),           m_currentCmd(NULL),
-      m_currentDownload(NULL),       m_pagenum(0),
+      m_netSearch(),                 m_currentSearch(QString()),
+      m_currentGrabber(0),           m_currentCmd(QString()),
+      m_currentDownload(QString()),  m_pagenum(0),
       m_lock(QMutex::Recursive)
 {
Index: /branches/release-0-23-fixes/mythplugins/mythvideo/mythvideo/videodlg.h
===================================================================
--- /branches/release-0-23-fixes/mythplugins/mythvideo/mythvideo/videodlg.h	(revision 22916)
+++ /branches/release-0-23-fixes/mythplugins/mythvideo/mythvideo/videodlg.h	(revision 26391)
@@ -151,5 +151,5 @@
     QString GetCoverImage(MythGenericTree *node);
     QString GetFirstImage(MythGenericTree *node, QString type,
-                          QString gpnode = NULL, int levels = 0);
+                          QString gpnode = QString(), int levels = 0);
     QString GetImageFromFolder(Metadata *metadata);
     QString GetScreenshot(MythGenericTree *node);
