From 916a48ffcb9f1c259a0fc778a2de1be729423092 Mon Sep 17 00:00:00 2001
From: Alec Leamas <leamas.alec@gmail.com>
Date: Tue, 25 Oct 2016 21:29:27 +0200
Subject: [PATCH 06/10] lirc.pc: Fix bad library specification (#236).

---
 lirc.pc.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lirc.pc.in b/lirc.pc.in
index 802f00e..5ed7c1c 100644
--- a/lirc.pc.in
+++ b/lirc.pc.in
@@ -10,5 +10,5 @@ includedir=@includedir@
 pkgdatadir=@datadir@/lirc
 plugindir=${libdir}/lirc/plugins
 
-Libs: -L${libdir} -llirc
+Libs: -L${libdir} -llirc_client
 Cflags: -I${includedir}
-- 
2.5.5

