From 0463f56357d1dd223a4d2882dbc460a0a7a7e9ad Mon Sep 17 00:00:00 2001
From: Alec Leamas <leamas.alec@gmail.com>
Date: Sat, 29 Oct 2016 11:06:50 +0200
Subject: [PATCH 07/10] Build: Use HAVE_UINPUT=1 to force building uinput code
 (#238).

---
 configure.ac | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index cde9813..08e2c8e 100644
--- a/configure.ac
+++ b/configure.ac
@@ -33,7 +33,7 @@ if test x$XSLTPROC != "xyes"; then
     AC_MSG_ERROR([
       xsltproc is required to build. It usually comes with libxml])
 fi
-AM_CONDITIONAL(HAVE_UINPUT, test -e /dev/uinput)
+AM_CONDITIONAL(HAVE_UINPUT, test -n "$HAVE_UINPUT" -o -e /dev/uinput)
 AC_CHECK_PROG([MAN2HTML],[man2html],[yes],[no])
 AM_CONDITIONAL(HAVE_MAN2HTML, test x$MAN2HTML = xyes)
 AC_CHECK_PROG([DOXYGEN],[doxygen],[yes],[no])
@@ -446,6 +446,11 @@ AM_CONDITIONAL(NEED_PYTHON3, [test "x$enable_python3_fix" = "xyes"])
 AC_ARG_VAR(HAVE_WORKING_POLL,[
 	   On cross-compile: Target has a working poll(2) implementation])
 
+AC_ARG_VAR(HAVE_UINPUT,[
+  If not-empty, assume uinput is available despite missing /dev/uinput
+])
+
+
 suffix=$(echo "$VERSION" | sed 's/.*-//')
 AM_CONDITIONAL(DEVEL, [test x$suffix = xdevel])
 
-- 
2.5.5

