diff --git a/ChangeLog_Keshav b/ChangeLog_Keshav
new file mode 100644
index 0000000..0eafd65
--- /dev/null
+++ b/ChangeLog_Keshav
@@ -0,0 +1,12 @@
+2012-01-04  Keshav P R  <the.ridikulus.rat@gmail.com>
+
+	Fixes for automake 1.11.2
+
+	* conf/Makefile.common: Change pkglib_SCRIPTS to pkgdata_SCRIPTS.
+	* conf/Makefile.common: Change pkglib_DATA to pkgdata_DATA.
+	* Makefile.am: Likewise.
+	* gentpl.py: Likewise.
+	* util/grub-mkstandalone.in: Likewise.
+	* util/grub-mknetdir.in: Likewise.
+	* util/grub-mkrescue.in: Likewise.
+	* util/grub-mkstandalone.in: Likewise.
diff --git a/Makefile.am b/Makefile.am
index c5f486e..395b0dd 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -101,8 +101,8 @@ CLEANFILES += widthspec.h
 # Install config.h into platformdir
 platform_HEADERS = config.h
 
-pkglib_DATA += grub-mkconfig_lib
-pkglib_DATA += update-grub_lib
+pkgdata_DATA += grub-mkconfig_lib
+pkgdata_DATA += update-grub_lib
 
 
 if COND_i386_coreboot
diff --git a/conf/Makefile.common b/conf/Makefile.common
index 751188b..bbd59cc 100644
--- a/conf/Makefile.common
+++ b/conf/Makefile.common
@@ -137,7 +137,7 @@ KERNEL_HEADER_FILES =
 
 man_MANS =
 noinst_DATA =
-pkglib_DATA =
+pkgdata_DATA =
 bin_SCRIPTS =
 sbin_SCRIPTS =
 bin_PROGRAMS =
@@ -147,7 +147,7 @@ check_SCRIPTS =
 grubconf_DATA =
 check_PROGRAMS =
 noinst_SCRIPTS =
-pkglib_SCRIPTS =
+pkgdata_SCRIPTS =
 noinst_PROGRAMS =
 grubconf_SCRIPTS =
 noinst_LIBRARIES =
diff --git a/gentpl.py b/gentpl.py
index 3008b80..a935f4d 100644
--- a/gentpl.py
+++ b/gentpl.py
@@ -512,7 +512,7 @@ def script(platform):
     r += "[+ ENDIF +]"
 
     r += rule("[+ name +]", platform_sources(platform) + " $(top_builddir)/config.status", """
-$(top_builddir)/config.status --file=-:$< | sed -e 's,@pkglib_DATA@,$(pkglib_DATA),g' > $@
+$(top_builddir)/config.status --file=-:$< | sed -e 's,@pkgdata_DATA@,$(pkgdata_DATA),g' > $@
 chmod a+x [+ name +]
 """)
 
diff --git a/util/grub-mknetdir.in b/util/grub-mknetdir.in
index e5a2172..7f6a36d 100644
--- a/util/grub-mknetdir.in
+++ b/util/grub-mknetdir.in
@@ -30,7 +30,7 @@ PACKAGE_VERSION=@PACKAGE_VERSION@
 host_os=@host_os@
 localedir=@datadir@/locale
 datarootdir=@datarootdir@
-pkglib_DATA="moddep.lst command.lst fs.lst partmap.lst parttool.lst handler.lst video.lst crypto.lst terminal.lst"
+pkgdata_DATA="moddep.lst command.lst fs.lst partmap.lst parttool.lst handler.lst video.lst crypto.lst terminal.lst"
 
 self=`basename $0`
 
@@ -183,7 +183,7 @@ process_input_dir ()
             cp -f "$file" "$grubdir/"
         fi
     done
-    for file in ${pkglib_DATA}; do
+    for file in ${pkgdata_DATA}; do
 	if test -f "${input_dir}/${file}"; then
             cp -f "${input_dir}/${file}" "$grubdir/"
 	fi
diff --git a/util/grub-mkrescue.in b/util/grub-mkrescue.in
index eff7708..f6b96d4 100644
--- a/util/grub-mkrescue.in
+++ b/util/grub-mkrescue.in
@@ -27,7 +27,7 @@ libdir=@libdir@
 PACKAGE_NAME=@PACKAGE_NAME@
 PACKAGE_TARNAME=@PACKAGE_TARNAME@
 PACKAGE_VERSION=@PACKAGE_VERSION@
-pkglib_DATA="moddep.lst command.lst fs.lst partmap.lst parttool.lst handler.lst video.lst crypto.lst terminal.lst"
+pkgdata_DATA="moddep.lst command.lst fs.lst partmap.lst parttool.lst handler.lst video.lst crypto.lst terminal.lst"
 
 self=`basename $0`
 
@@ -163,7 +163,7 @@ process_input_dir ()
             cp -f "$file" ${iso9660_dir}/boot/grub/${platform}/
         fi
     done
-    for file in ${pkglib_DATA}; do
+    for file in ${pkgdata_DATA}; do
 	if test -f "${input_dir}/${file}"; then
             cp -f "${input_dir}/${file}" ${iso9660_dir}/boot/grub/${platform}/
 	fi
diff --git a/util/grub-mkstandalone.in b/util/grub-mkstandalone.in
index b0dbf9b..92b7306 100644
--- a/util/grub-mkstandalone.in
+++ b/util/grub-mkstandalone.in
@@ -27,7 +27,7 @@ libdir=@libdir@
 PACKAGE_NAME=@PACKAGE_NAME@
 PACKAGE_TARNAME=@PACKAGE_TARNAME@
 PACKAGE_VERSION=@PACKAGE_VERSION@
-pkglib_DATA="moddep.lst command.lst fs.lst partmap.lst parttool.lst handler.lst video.lst crypto.lst terminal.lst"
+pkgdata_DATA="moddep.lst command.lst fs.lst partmap.lst parttool.lst handler.lst video.lst crypto.lst terminal.lst"
 
 self=`basename $0`
 
@@ -163,7 +163,7 @@ for file in "${source_directory}/"*.mod "${source_directory}/"efiemu32.o "${sour
     fi
 done
 
-for file in ${pkglib_DATA}; do
+for file in ${pkgdata_DATA}; do
     if test -f "${source_directory}/${file}"; then
         cp -f "${source_directory}/${file}" "${memdisk_dir}"/boot/grub/
     fi
