diff -u -r kernel/nvidia-drm/nvidia-drm-fence.c kernel-patched/nvidia-drm/nvidia-drm-fence.c
--- kernel/nvidia-drm/nvidia-drm-fence.c	2016-08-09 01:43:19.000000000 +0200
+++ kernel-patched/nvidia-drm/nvidia-drm-fence.c	2016-08-25 14:31:34.419655564 +0200
@@ -325,7 +325,7 @@
     struct nvidia_drm_gem_object *nv_gem;
 };
 
-static void nvidia_drm_gem_prime_soft_fence_event
+static void NVKMS_KAPI_CALL nvidia_drm_gem_prime_soft_fence_event
 (
     void *dataPtr,
     NvU32 dataU32
