From 1b8d3b9210c1d2201bb01a748d748f5d17e9eb4f Mon Sep 17 00:00:00 2001
From: Matthias Clasen <mclasen@redhat.com>
Date: Mon, 30 Sep 2013 16:50:34 +0000
Subject: Actually look in /usr/share/gtk-2.0 for default rc files

I managed to forget the 'share' component of the path.
---
diff --git a/gtk/gtkrc.c b/gtk/gtkrc.c
index b1f68e2..50b2563 100644
--- a/gtk/gtkrc.c
+++ b/gtk/gtkrc.c
@@ -525,7 +525,7 @@ gtk_rc_add_initial_default_files (void)
       const gchar * const *config_dirs;
       const gchar *config_dir;
 
-      str = g_build_filename (GTK_DATA_PREFIX, "gtk-2.0", "gtkrc", NULL);
+      str = g_build_filename (GTK_DATA_PREFIX, "share", "gtk-2.0", "gtkrc", NULL);
       gtk_rc_add_default_file (str);
       g_free (str);
 
--
cgit v0.9.2
