diff -u -r nss-3.31/nss/build.sh nss-3.31-libpkix/nss/build.sh
--- nss-3.31/nss/build.sh	2017-06-08 12:53:01.000000000 +0200
+++ nss-3.31-libpkix/nss/build.sh	2017-06-09 19:11:21.746133040 +0200
@@ -52,7 +52,7 @@
 no_local_nspr=0
 armhf=0
 
-gyp_params=(--depth="$cwd" --generator-output=".")
+gyp_params=(--depth="$cwd" --generator-output="." -Ddisable_libpkix=0)
 nspr_params=()
 ninja_params=()
 
