diff -aur old/display.h new/display.h
--- old/display.h	2014-04-16 11:38:41.000000000 -1000
+++ new/display.h	2014-04-17 16:20:36.992994397 -1000
@@ -73,7 +73,7 @@
   struct win *d_other;		/* pointer to other window */
   int   d_nonblock;		/* -1 don't block if obufmax reached */
 				/* >0: block after nonblock secs */
-  char  d_termname[20 + 1];	/* $TERM */
+  char  d_termname[40 + 1];	/* $TERM */
   char	*d_tentry;		/* buffer for tgetstr */
   char	d_tcinited;		/* termcap inited flag */
   int	d_width, d_height;	/* width/height of the screen */
diff -aur old/screen.h new/screen.h
--- old/screen.h	2014-04-16 11:38:41.000000000 -1000
+++ new/screen.h	2014-04-17 16:20:36.992994397 -1000
@@ -214,7 +214,7 @@
 	  char preselect[20];
 	  int esc;		/* his new escape character unless -1 */
 	  int meta_esc;		/* his new meta esc character unless -1 */
-	  char envterm[20 + 1];	/* terminal type */
+	  char envterm[40 + 1];	/* terminal type */
 	  int encoding;		/* encoding of display */
 	  int detachfirst;      /* whether to detach remote sessions first */
 	}
