From 27a8c9677a95b8de67c91f983b25691f0864c194 Mon Sep 17 00:00:00 2001
From: Amadeusz Sławiński <amade@asmblr.net>
Date: Sun, 21 Jun 2015 20:34:29 +0000
Subject: Fix off by one error when resizing windows

Bug: 45321

Signed-off-by: Amadeusz Sławiński <amade@asmblr.net>
---
diff --git a/src/resize.c b/src/resize.c
index d48be5f..7dfd77c 100644
--- a/src/resize.c
+++ b/src/resize.c
@@ -538,7 +538,7 @@ int wi;
 	  wi = MAXWIDTH;
   if (wi <= maxwidth)
     return;
-  maxwidth = wi;
+  maxwidth = wi + 1;
   debug1("New maxwidth: %d\n", maxwidth);
   blank = (unsigned char *)xrealloc((char *)blank, maxwidth);
   null = (unsigned char *)xrealloc((char *)null, maxwidth);
--
cgit v0.9.0.2
