From 4c3fb4070214ed28b1208aab55d7f5e3bcc9608f Mon Sep 17 00:00:00 2001
From: Allan Sandfeld Jensen <allan.jensen@qt.io>
Date: Mon, 28 Mar 2022 11:30:14 +0200
Subject: [PATCH] Support system ICU>=70

Fixes: QTBUG-100495
Change-Id: I243308ffd4ea274e1c2c9e3c24b9a17355abac7c
---

diff --git a/configure.cmake b/configure.cmake
index 7d92341..b77a7a8 100644
--- a/configure.cmake
+++ b/configure.cmake
@@ -41,6 +41,7 @@
     pkg_check_modules(ZLIB zlib)
     pkg_check_modules(RE2 re2 IMPORTED_TARGET)
     pkg_check_modules(ICU icu-uc>=68 icu-i18n>=68)
+    pkg_check_modules(ICU70P icu-uc>=70 icu-i18n>=70)
     pkg_check_modules(WEBP libwebp libwebpmux libwebpdemux)
     pkg_check_modules(LCMS2 lcms2)
     pkg_check_modules(FREETYPE freetype2 IMPORTED_TARGET)
@@ -276,6 +277,10 @@
     AUTODETECT FALSE
     CONDITION ICU_FOUND
 )
+qt_feature("webengine-system-icu70plus" PRIVATE
+    LABEL "icu>=70"
+    CONDITION ICU70P_FOUND AND QT_FEATURE_webengine_system_icu
+)
 qt_feature("webengine-system-libwebp" PRIVATE
     LABEL "libwebp, libwebpmux and libwebpdemux"
     CONDITION UNIX AND WEBP_FOUND
@@ -565,6 +570,7 @@
     qt_configure_add_summary_section(NAME "Optional system libraries")
     qt_configure_add_summary_entry(ARGS "webengine-system-re2")
     qt_configure_add_summary_entry(ARGS "webengine-system-icu")
+    qt_configure_add_summary_entry(ARGS "webengine-system-icu70plus")
     qt_configure_add_summary_entry(ARGS "webengine-system-libwebp")
     qt_configure_add_summary_entry(ARGS "webengine-system-opus")
     qt_configure_add_summary_entry(ARGS "webengine-system-ffmpeg")
diff --git a/src/core/CMakeLists.txt b/src/core/CMakeLists.txt
index e745207..1f8ee8d 100644
--- a/src/core/CMakeLists.txt
+++ b/src/core/CMakeLists.txt
@@ -402,6 +402,10 @@
                 CONDITION QT_FEATURE_webengine_system_pulseaudio
             )
             extend_gn_list(gnArgArg
+                ARGS use_system_icu70plus
+                CONDITION QT_FEATURE_webengine_system_icu70plus
+            )
+            extend_gn_list(gnArgArg
                 ARGS ozone_platform_x11 use_xkbcommon
                 CONDITION QT_FEATURE_webengine_ozone_x11
             )
diff --git a/src/3rdparty/chromium/third_party/icu/BUILD.gn b/src/3rdparty/chromium/third_party/icu/BUILD.gn
index d5b1361..2ef484e 100644
--- a/src/3rdparty/chromium/third_party/icu/BUILD.gn
+++ b/src/3rdparty/chromium/third_party/icu/BUILD.gn
@@ -11,6 +11,7 @@
 
 declare_args() {
   use_system_icu = false
+  use_system_icu70plus = false
 }
 
 if (is_android) {
@@ -411,10 +412,12 @@
     "ICU_UTIL_DATA_IMPL=ICU_UTIL_DATA_STATIC",
   ]
 
-  if (is_win) {
-    defines += [ "UCHAR_TYPE=wchar_t" ]
-  } else {
-    defines += [ "UCHAR_TYPE=uint16_t" ]
+  if (!use_system_icu70plus) {
+    if (is_win) {
+      defines += [ "UCHAR_TYPE=wchar_t" ]
+    } else {
+      defines += [ "UCHAR_TYPE=uint16_t" ]
+    }
   }
 }
 
