diff -ruN a/tests/Makefile.am b/tests/Makefile.am
--- a/tests/Makefile.am	2014-11-14 19:11:13.000000000 +0100
+++ b/tests/Makefile.am	2023-03-02 19:49:59.269974344 +0100
@@ -37,7 +37,7 @@
 # Used to make N-triples output consistent
 BASE_URI=http://librdf.org/raptor/tests/
 
-SUBDIRS = rdfxml ntriples ntriples-2013 nquads-2013 turtle mkr turtle-2013 trig grddl rdfa rdfa11 json feeds
+SUBDIRS = ntriples ntriples-2013 nquads-2013 mkr turtle-2013 trig grddl rdfa rdfa11 json
 
 
 $(top_builddir)/src/libraptor2.la:
diff -ruN a/tests/Makefile.in b/tests/Makefile.in
--- a/tests/Makefile.in	2023-03-01 18:58:10.000000000 +0100
+++ b/tests/Makefile.in	2023-03-02 19:50:09.006659970 +0100
@@ -618,7 +618,7 @@
 
 # Used to make N-triples output consistent
 BASE_URI = http://librdf.org/raptor/tests/
-SUBDIRS = rdfxml ntriples ntriples-2013 nquads-2013 turtle mkr turtle-2013 trig grddl rdfa rdfa11 json feeds
+SUBDIRS = ntriples ntriples-2013 nquads-2013 mkr turtle-2013 trig grddl rdfa rdfa11 json
 all: all-recursive
 
 .SUFFIXES:
