--- src/scripting/qti.sip	2009-01-28 01:09:59.409564097 +0100
+++ src/scripting/qti.sip.new	2009-01-28 01:08:07.552907918 +0100
@@ -74,8 +74,6 @@
   MdiSubWindow(const MdiSubWindow&);
 };
 
-typedef QList<MdiSubWindow*> MDIWindowList;
-
 class Table: MdiSubWindow
 {
 %TypeHeaderCode
@@ -1072,7 +1070,7 @@
     Graph3D* plotParametricSurface(const QString&, const QString&, const QString&,
             double, double, double, double, int=40, int=40, bool=true, bool=true) /PyName=plot3D/;
 
-  MDIWindowList windowsList() /PyName=windows/;
+  QList<MdiSubWindow*> windowsList() /PyName=windows/;
 
   // folders
   Folder *activeFolder() /NoDerived/;
@@ -1867,7 +1865,7 @@
 #include "src/core/Folder.h"
 %End
 public:
-  MDIWindowList windowsList() /PyName=windows/;
+  QList<MdiSubWindow*> windowsList() /PyName=windows/;
 // TODO: implement signal Folder::nameChanged and make it update the project explorer; adjust renaming from GUI accordingly
 //  void setFolderName(const QString&) /PyName=setName/;
   QString name();
@@ -1887,4 +1885,4 @@
 
 private:
   Folder(const Folder&);
-};
+};
