--- qtiplot.pro.old	2009-02-03 11:25:02.575512255 +0100
+++ qtiplot.pro	2009-02-03 11:24:53.717052708 +0100
@@ -20,7 +20,7 @@
 win32:LIBS 		  += ../3rdparty/libpng/libpng.a
 
 # Uncomment the following line if you want to perform a custom installation using the *.path variables defined bellow.
-#CONFIG          += CustomInstall
+CONFIG          += CustomInstall
 
 CONFIG          += release
 #CONFIG          += debug
@@ -30,13 +30,12 @@
 #!!! Warning: You must modify these paths according to your computer settings
 #############################################################################
 
-INCLUDEPATH       += ../3rdparty/muparser/include
-INCLUDEPATH       += ../3rdparty/qwtplot3d/include
-INCLUDEPATH       += ../3rdparty/qwt/src
+INCLUDEPATH       += /usr/include
+INCLUDEPATH       += /usr/include/qwt
+INCLUDEPATH       += /usr/include/gsl
+INCLUDEPATH       += /usr/include/boost
 INCLUDEPATH       += ../3rdparty/liborigin
-INCLUDEPATH       += ../3rdparty/gsl/include
-INCLUDEPATH       += ../3rdparty/zlib123/include
-INCLUDEPATH       += ../3rdparty/boost_1_36_0
+INCLUDEPATH       += ../3rdparty/qwtplot3d/include
 
 ##################### 3rd PARTY LIBRARIES SECTION ###########################
 #!!! Warning: You must modify these paths according to your computer settings
@@ -45,17 +44,18 @@
 ##################### Linux (Mac OS X) ######################################
 
 # statically link against libraries in 3rdparty
-unix:LIBS         += ../3rdparty/muparser/lib/libmuparser.a
-unix:LIBS         += ../3rdparty/qwt/lib/libqwt.a
-unix:LIBS         += ../3rdparty/gsl/lib/libgsl.a
-unix:LIBS         += ../3rdparty/gsl/lib/libgslcblas.a
-unix:LIBS         += /usr/local/lib/libboost_date_time-gcc41-mt.a
-unix:LIBS         += /usr/local/lib/libboost_thread-gcc41-mt.a
+#unix:LIBS         += ../3rdparty/muparser/lib/libmuparser.a
+#unix:LIBS         += ../3rdparty/qwt/lib/libqwt.a
+#unix:LIBS         += ../3rdparty/gsl/lib/libgsl.a
+#unix:LIBS         += ../3rdparty/gsl/lib/libgslcblas.a
+#unix:LIBS         += /usr/local/lib/libboost_date_time-gcc41-mt.a
+#unix:LIBS         += /usr/local/lib/libboost_thread-gcc41-mt.a
 
 # dynamically link against dependencies if they are installed system-wide
 #unix:LIBS         += -lmuparser
 #unix:LIBS         += -lqwt
 #unix:LIBS         += -lgsl -lgslcblas
+unix:LIBS	   += -lmuparser -lqwt $$system(pkg-config --libs gsl) -lz -lboost_date_time -lboost_thread-mt
 
 ##################### Windows ###############################################
 
@@ -186,7 +186,7 @@
   						qtiUtil.py \
   						qti_wordlist.txt \
 
-  	unix: pythonconfig.path = /usr/local/qtiplot
+  	unix: pythonconfig.path = /usr/share/qtiplot
   	win32: pythonconfig.path = $$INSTALLBASE
   	DEFINES += PYTHON_CONFIG_PATH="\\\"$$replace(pythonconfig.path," ","\ ")\\\"
   }
