=== modified file 'build.conf.example'
--- build.conf.example	2010-02-14 15:07:01 +0000
+++ build.conf.example	2010-02-14 15:16:08 +0000
@@ -118,6 +118,14 @@
 EMF_ENGINE_LIBS = $$QTI_ROOT/3rdparty/EmfEngine/libEmfEngine.a
 
 ##########################################################
+## liborigin - keep commented if you wish to use local copy
+#  http://sourceforge.net/projects/liborigin/
+##########################################################
+
+# LIB_ORIGIN_INCLUDEPATH = /usr/include/liborigin2
+# LIB_ORIGIN_LIBS = -lorigin2
+
+##########################################################
 ## python - only used if python is needed
 ##########################################################
 

=== modified file 'qtiplot/qtiplot.pro'
--- qtiplot/qtiplot.pro	2010-02-14 15:07:01 +0000
+++ qtiplot/qtiplot.pro	2010-02-14 15:13:58 +0000
@@ -9,7 +9,6 @@
 #############################################################################
 
 # local copy included
-INCLUDEPATH       += ../3rdparty/liborigin
 INCLUDEPATH       += ../3rdparty/zlib
 
 # configurable
@@ -18,6 +17,7 @@
 INCLUDEPATH       += $$QWT3D_INCLUDEPATH
 INCLUDEPATH       += $$GSL_INCLUDEPATH
 INCLUDEPATH       += $$BOOST_INCLUDEPATH
+INCLUDEPATH       += $$LIB_ORIGIN_INCLUDEPATH
 
 # configurable libs
 LIBS         += $$MUPARSER_LIBS
@@ -26,6 +26,7 @@
 LIBS         += $$QTEXENGINE_LIBS
 LIBS         += $$GSL_LIBS
 LIBS         += $$BOOST_LIBS
+LIBS         += $$LIB_ORIGIN_LIBS
 
 #############################################################################
 ###################### BASIC PROJECT PROPERTIES #############################
@@ -154,6 +155,10 @@
 	SOURCES     += ../3rdparty/QTeXEngine/src/QTeXPaintDevice.cpp
 }
 
+isEmpty(LIB_ORIGIN_LIBS) {
+	INCLUDEPATH       += ../3rdparty/liborigin
+}
+
 ###############################################################
 ##################### Scripting: PYTHON + SIP + PyQT ##########
 ###############################################################

=== modified file 'qtiplot/src/origin/origin.pri'
--- qtiplot/src/origin/origin.pri	2009-10-18 01:48:15 +0000
+++ qtiplot/src/origin/origin.pri	2010-02-14 15:11:15 +0000
@@ -4,14 +4,17 @@
 INCLUDEPATH += src/origin/
 
 HEADERS += src/origin/importOPJ.h
-HEADERS += ../3rdparty/liborigin/OriginObj.h
-HEADERS += ../3rdparty/liborigin/OriginFile.h
-HEADERS += ../3rdparty/liborigin/OriginParser.h
-HEADERS += ../3rdparty/liborigin/OriginDefaultParser.h
-HEADERS += ../3rdparty/liborigin/Origin750Parser.h
-
 SOURCES += src/origin/importOPJ.cpp
-SOURCES += ../3rdparty/liborigin/OriginFile.cpp
-SOURCES += ../3rdparty/liborigin/OriginParser.cpp
-SOURCES += ../3rdparty/liborigin/OriginDefaultParser.cpp
-SOURCES += ../3rdparty/liborigin/Origin750Parser.cpp
+
+isEmpty(LIB_ORIGIN_LIBS) {
+	HEADERS += ../3rdparty/liborigin/OriginObj.h
+	HEADERS += ../3rdparty/liborigin/OriginFile.h
+	HEADERS += ../3rdparty/liborigin/OriginParser.h
+	HEADERS += ../3rdparty/liborigin/OriginDefaultParser.h
+	HEADERS += ../3rdparty/liborigin/Origin750Parser.h
+
+	SOURCES += ../3rdparty/liborigin/OriginFile.cpp
+	SOURCES += ../3rdparty/liborigin/OriginParser.cpp
+	SOURCES += ../3rdparty/liborigin/OriginDefaultParser.cpp
+	SOURCES += ../3rdparty/liborigin/Origin750Parser.cpp
+}



