diff --git a/qtiplot/qtiplot.pro b/qtiplot/qtiplot.pro
index 95a5f0c..72e99ab 100755
--- a/qtiplot/qtiplot.pro
+++ b/qtiplot/qtiplot.pro
@@ -205,18 +205,14 @@ contains(SCRIPTING_LANGS, Python) {
 !isEmpty(TAMUANOVA_LIBS) {
 	DEFINES += HAVE_TAMUANOVA
 	INCLUDEPATH += $$TAMUANOVA_INCLUDEPATH
-	#LIBS        += $$TAMUANOVA_LIBS
-
-	HEADERS += $$TAMUANOVA_INCLUDEPATH/tamu_anova.h
-	SOURCES += $$TAMUANOVA_INCLUDEPATH/anova_1.c
-	SOURCES += $$TAMUANOVA_INCLUDEPATH/anova_2.c
+	LIBS        += $$TAMUANOVA_LIBS
 }
 
 ###############################################################
 
 # At the very end: add global include- and lib path
-#unix:INCLUDEPATH += $$SYS_INCLUDEPATH
-#unix:LIBS += $$SYS_LIBS
+unix:INCLUDEPATH += $$SYS_INCLUDEPATH
+unix:LIBS += $$SYS_LIBS
 
 ###############################################################
 ############### Building QtiPlot as a browser plugin ##########
