diff -Naur qtiplot-0.9.7.3/qtiplot/src/plot3D/Graph3D.cpp qtiplot-0.9.7.3.new/qtiplot/src/plot3D/Graph3D.cpp
--- src/plot3D/Graph3D.cpp	2008-10-20 06:32:35.000000000 -0400
+++ src/plot3D/Graph3D.cpp	2008-10-24 11:04:56.000000000 -0400
@@ -777,7 +777,7 @@
 void Graph3D::setNumbersFont(const QFont& font)
 {
 	sp->coordinates()->setNumberFont (font);
-	sp->legend()->axis()->setNumberFont (font);
+	//sp->legend()->axis()->setNumberFont (font);
 	sp->makeCurrent();
 	sp->updateGL();
 }
@@ -1570,7 +1570,7 @@
 void Graph3D::setNumbersColor(const QColor& numColor)
 {
 	if(numCol != numColor){
-		sp->legend()->axis()->setNumberColor(Qt2GL(numColor));
+		//sp->legend()->axis()->setNumberColor(Qt2GL(numColor));
 		sp->coordinates()->setNumberColor(Qt2GL(numColor));
 		numCol = numColor;
 	}
