--- qtiplot/src/scripting/scripting.pri.bak	2009-09-09 07:18:51.000000000 +0800
+++ qtiplot/src/scripting/scripting.pri	2009-11-08 12:45:03.000000000 +0800
@@ -80,7 +80,9 @@
              $${SIP_DIR}/sipqtiPythonScript.cpp\
              $${SIP_DIR}/sipqtiPythonScripting.cpp\
              $${SIP_DIR}/sipqtiFolder.cpp\
-             $${SIP_DIR}/sipqtiQList.cpp\
+             $${SIP_DIR}/sipqtiQList0101Folder.cpp \
+             $${SIP_DIR}/sipqtiQList0101Graph.cpp \
+             $${SIP_DIR}/sipqtiQList0101MdiSubWindow.cpp \
              $${SIP_DIR}/sipqtiFit.cpp \
              $${SIP_DIR}/sipqtiExponentialFit.cpp \
              $${SIP_DIR}/sipqtiTwoExpFit.cpp \
