diff -u -r qtbase-opensource-src-5.9.1/src/platformsupport/fontdatabases/fontconfig/qfontconfigdatabase.cpp qtbase-opensource-src-5.9.1-emoji/src/platformsupport/fontdatabases/fontconfig/qfontconfigdatabase.cpp
--- qtbase-opensource-src-5.9.1/src/platformsupport/fontdatabases/fontconfig/qfontconfigdatabase.cpp	2017-06-28 11:54:29.000000000 +0200
+++ qtbase-opensource-src-5.9.1-emoji/src/platformsupport/fontdatabases/fontconfig/qfontconfigdatabase.cpp	2017-09-15 12:19:03.581896442 +0200
@@ -749,6 +749,9 @@
         FcPatternAddWeak(pattern, FC_FAMILY, value, FcTrue);
     }
 
+    value.u.s = (const FcChar8 *)"emoji";
+    FcPatternAddWeak(pattern, FC_FAMILY, value, FcTrue);
+
     FcConfigSubstitute(0, pattern, FcMatchPattern);
     FcDefaultSubstitute(pattern);
 
