From 269de2e0bf5011072da2f40f4f2d4023fad696b9 Mon Sep 17 00:00:00 2001
From: Lucas De Marchi <lucas.demarchi@profusion.mobi>
Date: Tue, 7 Feb 2012 09:48:59 -0200
Subject: [PATCH] libkmod-module: probe: Fix ignore-loaded flag not being
 applied

---
 TODO                     |    3 +++
 libkmod/libkmod-module.c |    3 ++-
 2 files changed, 5 insertions(+), 1 deletions(-)

diff --git a/libkmod/libkmod-module.c b/libkmod/libkmod-module.c
index 0af3e2e..b5eb7c9 100644
--- a/libkmod/libkmod-module.c
+++ b/libkmod/libkmod-module.c
@@ -1132,7 +1132,8 @@ KMOD_EXPORT int kmod_module_probe_insert_module(struct kmod_module *mod,
 	if (mod == NULL)
 		return -ENOENT;
 
-	if (module_is_inkernel(mod)) {
+	if (!(flags & KMOD_PROBE_IGNORE_LOADED)
+					&& module_is_inkernel(mod)) {
 		if (flags & KMOD_PROBE_FAIL_ON_LOADED)
 			return -EEXIST;
 		else
-- 
1.7.9

