--- a/Makefile	2014-12-13 13:22:59.470287957 +0100
+++ b/Makefile	2014-12-13 16:48:35.988030397 +0100
@@ -93,8 +93,8 @@
 ifneq (,$(filter $(shell uname),Linux Darwin GNU/kFreeBSD GNU))
 
 install:
-	@cd $(LZ4DIR); sudo $(MAKE) -e install
-	@cd $(PRGDIR); sudo $(MAKE) -e install
+	@cd $(LZ4DIR); $(MAKE) -e install
+	@cd $(PRGDIR); $(MAKE) -e install
 
 uninstall:
 	@cd $(LZ4DIR); $(MAKE) uninstall
