diff -Naur pkg-config-0.25-old//pkg.m4 pkg-config-0.25/pkg.m4
--- pkg-config-0.25-old//pkg.m4	2009-12-07 06:39:02.000000000 +1000
+++ pkg-config-0.25/pkg.m4	2010-07-15 14:44:37.466942978 +1000
@@ -135,7 +135,7 @@
 Consider adjusting the PKG_CONFIG_PATH environment variable if you
 installed software in a non-standard prefix.
 
-_PKG_TEXT])dnl
+_PKG_TEXT])[]dnl
         ])
 elif test $pkg_failed = untried; then
      	AC_MSG_RESULT([no])
