Index: glib/gkeyfile.c
===================================================================
--- glib/gkeyfile.c	(revision 5714)
+++ glib/gkeyfile.c	(working copy)
@@ -3639,9 +3639,9 @@
 {
   gchar *value_utf8;
 
-  if (strcmp (value, "true") == 0 || strcmp (value, "1") == 0)
+  if (g_ascii_strcasecmp (value, "true") == 0 || strcmp (value, "1") == 0)
     return TRUE;
-  else if (strcmp (value, "false") == 0 || strcmp (value, "0") == 0)
+  else if (g_ascii_strcasecmp (value, "false") == 0 || strcmp (value, "0") == 0)
     return FALSE;
 
   value_utf8 = _g_utf8_make_valid (value);
