Index: net.c
===================================================================
--- net.c (revision 9a55b4faafd865e491200640133177269c4d91fd)
+++ net.c (revision f6c275e9610384bb326db341a221b142d26fc5f6)
@@ -273,4 +273,6 @@
 	ifs = ifl = NULL;
 	for (ifa = ifaddrs; ifa; ifa = ifa->ifa_next) {
+		if (ifa->ifa_addr == NULL) /* should not happen */
+			continue;
 #ifdef AF_LINK
 		if (ifa->ifa_addr->sa_family != AF_LINK)
@@ -373,4 +375,6 @@
 	retval = 0;
 	for (ifa = ifaddrs; ifa; ifa = ifa->ifa_next) {
+		if (ifa->ifa_addr == NULL) /* should not happen */
+			continue;
 		if (ifa->ifa_addr->sa_family != AF_INET ||
 		    strcmp(ifa->ifa_name, ifname) != 0)
