diff -r fc3dd3be67e9 plugins/sudoers/sudoers.c
--- a/plugins/sudoers/sudoers.c	Sat Mar 05 15:34:30 2011 -0500
+++ b/plugins/sudoers/sudoers.c	Sun Mar 06 15:35:39 2011 -0500
@@ -1172,6 +1172,11 @@
 		SET(flags, MODE_IGNORE_TICKET);
 	    continue;
 	}
+	if (MATCHES(*cur, "noninteractive=")) {
+	    if (atobool(*cur + sizeof("noninteractive=") - 1) == TRUE)
+		SET(flags, MODE_NONINTERACTIVE);
+	    continue;
+	}
 	if (MATCHES(*cur, "sudoedit=")) {
 	    if (atobool(*cur + sizeof("sudoedit=") - 1) == TRUE)
 		SET(flags, MODE_EDIT);
