--- kde-workspace-4.9.2/kdm/config.def~	2012-10-17 13:58:07.236399517 +0000
+++ kde-workspace-4.9.2/kdm/config.def	2012-10-17 14:01:35.181535177 +0000
@@ -38,8 +38,8 @@
 # define HALT_CMD	"/usr/sbin/shutdown -h now"
 # define REBOOT_CMD	"/usr/sbin/shutdown -r now"
 #elif defined(__linux__)
-# define HALT_CMD	"/sbin/shutdown -h -P now"
-# define REBOOT_CMD	"/sbin/shutdown -r now"
+# define HALT_CMD	"/usr/bin/systemctl poweroff"
+# define REBOOT_CMD	"/usr/bin/systemctl reboot"
 #elif defined(BSD)
 # define HALT_CMD	"/sbin/shutdown -p now"
 # define REBOOT_CMD	"/sbin/shutdown -r now"
