diff -Naur a/src/fileman.cc b/src/fileman.cc
--- a/src/fileman.cc	2006-01-29 02:55:32.000000000 +0300
+++ b/src/fileman.cc	2007-04-08 19:25:50.000000000 +0400
@@ -237,12 +237,14 @@
 	{
 		DIR
 			*dir2;
+  	struct stat
+	   	buffy;
+
 
 		PTH_YIELD;
 
-		if ( (dir2 = opendir(entries[i])) ) /* path is a dir */
+		if (!stat(entries[i], &buffy) && S_ISDIR(buffy.st_mode) )
 		{
-			closedir(dir2);
 			dirs[dircount] = (char *)malloc((strlen(entries[i]) + 2) *
 				sizeof(char));
 			strcpy(dirs[dircount], entries[i]);
@@ -369,10 +371,11 @@
 	
 	DIR
 		*dir2;
+	struct stat
+	   	buffy;
 
-	if ( (dir2 = opendir(getItem(index))) ) /* path is a dir */
+	if (!stat(getItem(index), &buffy) && S_ISDIR(buffy.st_mode) )
 	{
-		closedir(dir2);
 		return 1;
 	}
 	
