diff --git i/src/evinceview.js w/src/evinceview.js
index 60c314c5..8f01128a 100644
--- i/src/evinceview.js
+++ w/src/evinceview.js
@@ -315,9 +315,9 @@ var EvinceView = new Lang.Class({
             // the following invocation to work.
             let evDoc = this._model.get_document();
             evDoc.find_text();
-        } catch (e if e instanceof TypeError) {
-            canFind = false;
         } catch (e) {
+            if (e instanceof TypeError)
+                canFind = false;
         }
 
         this.getAction('find').enabled = (this.hasPages && canFind);
diff --git i/src/places.js w/src/places.js
index 220cd311..0bebbde0 100644
--- i/src/places.js
+++ w/src/places.js
@@ -60,7 +60,6 @@ var PlacesDialog = new Lang.Class({
             // This is a hack to find out if evDoc implements the
             // EvDocument.DocumentLinks interface or not.
             docHasLinks = evDoc.has_document_links();
-        } catch (e if e instanceof TypeError) {
         } catch (e) {
         }
 
