diff -Naur linux-atm-2.4.1-orig/src/arpd/atmarp.c linux-atm-2.4.1/src/arpd/atmarp.c
--- linux-atm-2.4.1-orig/src/arpd/atmarp.c	2001-09-03 18:41:05.000000000 +0000
+++ linux-atm-2.4.1/src/arpd/atmarp.c	2007-03-16 10:53:18.000000000 +0000
@@ -19,6 +19,7 @@
 #include <net/if_arp.h>

 #include <stdint.h>
+#include <linux/types.h>
 #include <linux/atmarp.h>
 #include <linux/atmclip.h>

diff -Naur linux-atm-2.4.1-orig/src/arpd/arp.c linux-atm-2.4.1/src/arpd/arp.c
--- linux-atm-2.4.1-orig/src/arpd/arp.c 2001-09-03 18:41:05.000000000 +0000
+++ linux-atm-2.4.1/src/arpd/arp.c      2007-03-16 11:02:54.000000000 +0000
@@ -16,6 +16,7 @@
 #include <sys/socket.h> /* for linux/if_arp.h */
 #include <netinet/in.h> /* for ntohs, etc. */
 #define _LINUX_NETDEVICE_H /* very crude hack for glibc2 */
+#include <linux/types.h>
 #include <linux/if_arp.h>
 #include <linux/if_ether.h>
 #include <atm.h>

diff -Naur linux-atm-2.4.1-orig/src/arpd/io.c linux-atm-2.4.1/src/arpd/io.c
--- linux-atm-2.4.1-orig/src/arpd/io.c  2001-09-03 18:41:05.000000000 +0000
+++ linux-atm-2.4.1/src/arpd/io.c       2007-03-16 11:07:11.000000000 +0000
@@ -19,6 +19,7 @@
 #include <net/if.h>
 #include <netinet/in.h>
 #include <atm.h>
+#include <linux/types.h>
 #include <linux/atmclip.h> /* for CLIP_DEFAULT_IDLETIMER */
 #include <linux/atmarp.h>
 #define _LINUX_NETDEVICE_H /* glibc2 */

diff -Naur linux-atm-2.4.1-orig/src/arpd/itf.c linux-atm-2.4.1/src/arpd/itf.c
--- linux-atm-2.4.1-orig/src/arpd/itf.c 2001-09-03 18:41:05.000000000 +0000
+++ linux-atm-2.4.1/src/arpd/itf.c      2007-03-16 11:08:34.000000000 +0000
@@ -13,6 +13,7 @@
 #include <linux/atmclip.h>
 #include <sys/socket.h>
 #define _LINUX_NETDEVICE_H /* glibc2 */
+#include <linux/types.h>
 #include <linux/if_arp.h>

 #include "atmd.h"

diff -Naur linux-atm-2.4.1-orig/src/mpoad/io.c linux-atm-2.4.1/src/mpoad/io.c
--- linux-atm-2.4.1-orig/src/mpoad/io.c 2001-09-03 18:41:06.000000000 +0000
+++ linux-atm-2.4.1/src/mpoad/io.c      2007-03-16 11:10:19.000000000 +0000
@@ -19,6 +19,7 @@
 _syscall3(int,poll,struct pollfd *,ufds,unsigned int,nfds,int,timeout);
 #endif
 #include <atm.h>
+#include <linux/types.h>
 #include <linux/atmioc.h>
 #include <linux/atmmpc.h>
 #include <sys/types.h>

diff -Naur linux-atm-2.4.1-orig/src/mpoad/k_interf.c linux-atm-2.4.1/src/mpoad/k_interf.c
--- linux-atm-2.4.1-orig/src/mpoad/k_interf.c   2001-09-03 18:41:06.000000000 +0000
+++ linux-atm-2.4.1/src/mpoad/k_interf.c        2007-03-16 11:12:00.000000000 +0000
@@ -9,6 +9,7 @@
 #include <sys/param.h> /* for OPEN_MAX   */
 #include <stdint.h>
 #include <netinet/in.h> /* for ntohl()    */
+#include <linux/types.h>
 #include <linux/atmmpc.h>
 #include <atm.h>
 #include "k_interf.h"

diff -Naur linux-atm-2.4.1-orig/src/mpoad/main.c linux-atm-2.4.1/src/mpoad/main.c
--- linux-atm-2.4.1-orig/src/mpoad/main.c       2001-09-03 18:41:06.000000000 +0000
+++ linux-atm-2.4.1/src/mpoad/main.c    2007-03-16 11:13:08.000000000 +0000
@@ -9,6 +9,7 @@
 #include <time.h>
 #include <sys/ioctl.h>
 #include <atm.h>
+#include <linux/types.h>
 #include <linux/atmdev.h>
 #include <linux/atmmpc.h>
 #include <sys/socket.h>

diff -Naur linux-atm-2.4.1-orig/src/mpoad/p_factory.c linux-atm-2.4.1/src/mpoad/p_factory.c
--- linux-atm-2.4.1-orig/src/mpoad/p_factory.c  2001-09-03 18:41:06.000000000 +0000
+++ linux-atm-2.4.1/src/mpoad/p_factory.c       2007-03-16 11:14:29.000000000 +0000
@@ -5,6 +5,7 @@
 #include <sys/types.h>
 #include <stdint.h>
 #include <netinet/in.h>
+#include <linux/types.h>
 #include <linux/atmmpc.h>
 #include <sys/socket.h>
 #include <unistd.h>

diff -Naur linux-atm-2.4.1-orig/src/mpoad/p_recogn.c linux-atm-2.4.1/src/mpoad/p_recogn.c
--- linux-atm-2.4.1-orig/src/mpoad/p_recogn.c   2001-09-03 18:41:06.000000000 +0000
+++ linux-atm-2.4.1/src/mpoad/p_recogn.c        2007-03-16 11:15:09.000000000 +0000
@@ -7,6 +7,7 @@
 #include <stdlib.h>
 #include <sys/time.h>
 #include <atm.h>
+#include <linux/types.h>
 #include <linux/atmmpc.h>
 #include <netinet/in.h>
 #include <limits.h> /* For UINT_MAX */

diff -Naur linux-atm-2.4.1-orig/src/mpoad/lecs.c linux-atm-2.4.1/src/mpoad/lecs.c
--- linux-atm-2.4.1-orig/src/mpoad/lecs.c       2001-09-03 18:41:06.000000000 +0000
+++ linux-atm-2.4.1/src/mpoad/lecs.c    2007-03-16 11:15:40.000000000 +0000
@@ -11,6 +11,7 @@
 #include <errno.h>
 #include <atm.h>
 #include <atmsap.h>
+#include <linux/types.h>
 #include <linux/atmmpc.h>   /* for MPOA Device type TLV */
 #include "lecs.h"
 #include "k_interf.h"

