diff -u Newt-1.08/newtlib/newt.c Newt-1.08/newtlib/newt.c
--- Newt-1.08/newtlib/newt.c	2006-04-09 15:02:27.000000000 +0200
+++ Newt-1.08/newtlib/newt.c	2006-04-09 14:50:12.000000000 +0200
@@ -13,7 +13,7 @@
 
 struct Window {
     int height, width, top, left;
-    short * buffer;
+    SLsmg_Char_Type * buffer;
     char * title;
 };
 
@@ -375,7 +375,7 @@
     currentWindow->height = height;
     currentWindow->title = title ? strdup(title) : NULL;
 
-    currentWindow->buffer = malloc(sizeof(short) * (width + 3) * (height + 3));
+    currentWindow->buffer = malloc(sizeof(SLsmg_Char_Type) * (width + 3) * (height + 3));
 
     row = top - 1;
     col = left - 1;
