diff -urp tk8.5.3-old/generic/tkBind.c tk8.5.3/generic/tkBind.c
--- tk8.5.3-old/generic/tkBind.c	2007-12-14 01:24:13.000000000 +1000
+++ tk8.5.3/generic/tkBind.c	2008-07-01 16:25:16.000000000 +1000
@@ -574,6 +574,9 @@ static int flagArray[TK_LASTEVENT] = {
    /* ColormapNotify */		COLORMAP,
    /* ClientMessage */		0,
    /* MappingNotify */		0,
+#ifdef GenericEvent
+   /* GenericEvent */		0,
+#endif 
    /* VirtualEvent */		VIRTUAL,
    /* Activate */		ACTIVATE,
    /* Deactivate */		ACTIVATE,
