diff -Nur bzr.debian.0.9.4.patchupdate/src/NetworkManager.conf bzr.debian.0.9.4.patchupdate.new/src/NetworkManager.conf
--- bzr.debian.0.9.4.patchupdate/src/NetworkManager.conf	2007-06-25 17:14:46.000000000 +0200
+++ bzr.debian.0.9.4.patchupdate.new/src/NetworkManager.conf	2007-06-25 17:15:51.000000000 +0200
@@ -8,6 +8,10 @@
                 <allow send_destination="org.freedesktop.NetworkManager"/>
                 <allow send_interface="org.freedesktop.NetworkManager"/>
         </policy>
+        <policy user="hal">
+                <allow send_destination="org.freedesktop.NetworkManager"/>
+                <allow send_interface="org.freedesktop.NetworkManager"/>
+        </policy>
         <policy at_console="true">
                 <allow send_destination="org.freedesktop.NetworkManager"/>
                 <allow send_interface="org.freedesktop.NetworkManager"/>
