--- initscript/Arch/networkmanager-dispatcher.in	2007-04-18 18:13:08.000000000 +0000
+++ initscript/Arch/networkmanager-dispatcher.in	2007-08-19 12:22:50.000000000 +0000
@@ -25,7 +25,7 @@
 		if [ ! -z "$PID" -o $? -gt 0 ]; then
 			stat_fail
 		else
-			add_daemon NetworkManagerDispatcher
+			add_daemon networkmanager-dispatcher
 			stat_done
 		fi
 		;;
@@ -35,7 +35,7 @@
 		if [ $? -gt 0 ]; then
 			stat_fail
 		else
-			rm_daemon NetworkManagerDispatcher
+			rm_daemon networkmanager-dispatcher
 			stat_done
 		fi
 		;;
--- initscript/Arch/networkmanager.in	2007-04-18 18:13:08.000000000 +0000
+++ initscript/Arch/networkmanager.in	2007-08-19 12:24:56.000000000 +0000
@@ -19,6 +19,8 @@
 PID=`pidof -o %PPID $NETWORKMANAGER_BIN`
 case "$1" in
 	start)
+	        ck_daemon dbus && /etc/rc.d/dbus start
+		ck_daemon dhcdbd && /etc/rc.d/dhcdbd start
 		stat_busy "Starting NetworkManager"
 		if [ -z "$PID" ]; then
 			$NETWORKMANAGER_BIN
@@ -26,7 +28,7 @@
 		if [ ! -z "$PID" -o $? -gt 0 ]; then
 			stat_fail
 		else
-			add_daemon NetworkManager
+			add_daemon networkmanager
 			stat_done
 		fi
 		;;
@@ -36,7 +38,7 @@
 		if [ $? -gt 0 ]; then
 			stat_fail
 		else
-			rm_daemon NetworkManager
+			rm_daemon networkmanager
 			stat_done
 		fi
 		;;
@@ -45,8 +47,22 @@
 		sleep 1
 		$0 start
 		;;
+	sleep)
+		/usr/bin/dbus-send --system \
+		--dest=org.freedesktop.NetworkManager \
+		--type=method_call \
+		/org/freedesktop/NetworkManager \
+		org.freedesktop.NetworkManager.sleep
+		;;
+	wake)
+		/usr/bin/dbus-send --system \
+		--dest=org.freedesktop.NetworkManager \
+		--type=method_call \
+		/org/freedesktop/NetworkManager \
+		org.freedesktop.NetworkManager.wake
+		;;
 	*)
-		echo "usage: $0 {start|stop|restart}"
+		echo "usage: $0 {start|stop|restart|sleep|wake}"
 		;;
 esac
 exit 0
