diff -u gitg-3.14.0/Gitg-1.0.gir gitg-3.14.0-webkit/Gitg-1.0.gir
--- gitg-3.14.0/Gitg-1.0.gir	2014-08-20 11:26:30.000000000 +0200
+++ gitg-3.14.0-webkit/Gitg-1.0.gir	2014-10-08 19:49:50.285616042 +0200
@@ -10,7 +10,7 @@
 <include name="cairo" version="1.0"/>
 <include name="Gdk" version="3.0"/>
 <include name="GdkPixbuf" version="2.0"/>
-<include name="WebKit2" version="3.0"/>
+<include name="WebKit2" version="4.0"/>
 <include name="Soup" version="2.4"/>
 <package name="libgitg/libgitg-1.0"/>
 <c:include name="libgitg/libgitg.h"/>
