diff -wbBur linphone-3.4.3/mediastreamer2/src/Makefile.am linphone-3.4.3.my/mediastreamer2/src/Makefile.am
--- linphone-3.4.3/mediastreamer2/src/Makefile.am	2011-02-09 08:43:11.000000000 +0000
+++ linphone-3.4.3.my/mediastreamer2/src/Makefile.am	2011-04-01 14:54:04.000000000 +0000
@@ -117,7 +117,7 @@
 endif
 
 if BUILD_V4L
-libmediastreamer_la_SOURCES+=msv4l.c msv4l2.c
+libmediastreamer_la_SOURCES+=msv4l2.c
 endif
 
 if BUILD_WIN32
diff -wbBur linphone-3.4.3/mediastreamer2/src/Makefile.in linphone-3.4.3.my/mediastreamer2/src/Makefile.in
--- linphone-3.4.3/mediastreamer2/src/Makefile.in	2011-03-28 18:34:07.000000000 +0000
+++ linphone-3.4.3.my/mediastreamer2/src/Makefile.in	2011-04-01 14:55:00.000000000 +0000
@@ -54,7 +54,7 @@
 @BUILD_MACAQSND_TRUE@am__append_12 = aqsnd.c
 @BUILD_PULSEAUDIO_TRUE@am__append_13 = pulseaudio.c
 @BUILD_MACOSX_TRUE@@BUILD_VIDEO_TRUE@am__append_14 = msv4m.m
-@BUILD_V4L_TRUE@@BUILD_VIDEO_TRUE@am__append_15 = msv4l.c msv4l2.c
+@BUILD_V4L_TRUE@@BUILD_VIDEO_TRUE@am__append_15 = msv4l2.c
 @BUILD_VIDEO_TRUE@@BUILD_WIN32_TRUE@am__append_16 = msdscap-mingw.cc drawdib-display.c
 @BUILD_THEORA_TRUE@@BUILD_VIDEO_TRUE@am__append_17 = theora.c
 @BUILD_FFMPEG_TRUE@@BUILD_VIDEO_TRUE@am__append_18 = videoenc.c \
@@ -135,7 +135,7 @@
 	audiostream.c msspeex.c speexec.c gsm.c winsnd3.c \
 	msfileplayer_win.c msfilerec_win.c msfileplayer.c msfilerec.c \
 	msresample.c alsa.c oss.c arts.c pasnd.c macsnd.c msiounit.c \
-	aqsnd.c pulseaudio.c msv4m.m msv4l.c msv4l2.c msdscap-mingw.cc \
+	aqsnd.c pulseaudio.c msv4m.m msv4l2.c msdscap-mingw.cc \
 	drawdib-display.c theora.c videoenc.c videodec.c pixconv.c \
 	sizeconv.c nowebcam.c nowebcam.h swscale.h ffmpeg-priv.h \
 	h264dec.c jpegwriter.c videoout.c x11video.c rfc2429.h \
@@ -156,7 +156,7 @@
 @BUILD_MACAQSND_TRUE@am__objects_12 = aqsnd.lo
 @BUILD_PULSEAUDIO_TRUE@am__objects_13 = pulseaudio.lo
 @BUILD_MACOSX_TRUE@@BUILD_VIDEO_TRUE@am__objects_14 = msv4m.lo
-@BUILD_V4L_TRUE@@BUILD_VIDEO_TRUE@am__objects_15 = msv4l.lo msv4l2.lo
+@BUILD_V4L_TRUE@@BUILD_VIDEO_TRUE@am__objects_15 = msv4l2.lo
 @BUILD_VIDEO_TRUE@@BUILD_WIN32_TRUE@am__objects_16 = msdscap-mingw.lo \
 @BUILD_VIDEO_TRUE@@BUILD_WIN32_TRUE@	drawdib-display.lo
 @BUILD_THEORA_TRUE@@BUILD_VIDEO_TRUE@am__objects_17 = theora.lo
@@ -600,7 +600,6 @@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/mssndcard.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/msspeex.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/msticker.Plo@am__quote@
-@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/msv4l.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/msv4l2.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/msv4m.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/msvideo.Plo@am__quote@
