diff -ur lirc-0.8.2/drivers/lirc_i2c/lirc_i2c.c lirc-0.8.2.new/drivers/lirc_i2c/lirc_i2c.c
--- lirc-0.8.2/drivers/lirc_i2c/lirc_i2c.c	2007-02-13 00:45:15.000000000 -0600
+++ lirc-0.8.2.new/drivers/lirc_i2c/lirc_i2c.c	2007-07-23 09:47:51.582717400 -0500
@@ -420,7 +420,7 @@
 		   Hauppauge cards, but the data flow is different, so we need
 		   to deal with it by its own.
 		 */
-		strcpy(ir->c.name,"Hauppauge IR (PVR150)");
+		strcpy(ir->c.name,"Hauppauge PVR150");
 		ir->l.code_length = 13;
 		ir->l.add_to_buf=add_to_buf_haup_pvr150;
 		break;
