--- acpid-1.0.4/ud_socket.c	2005-04-26 10:26:29.000000000 +0000
+++ acpid-1.0.4.az/ud_socket.c	2005-04-26 10:27:09.000000000 +0000
@@ -60,7 +60,7 @@
 		struct sockaddr_un cliaddr;
 		int len = sizeof(struct sockaddr_un);
 
-		newsock = accept(listenfd, (struct sockaddr *)&cliaddr, &len);
+		newsock = accept(listenfd, (struct sockaddr *)&cliaddr, (socklen_t *)&len);
 		if (newsock < 0) {
 			if (errno == EINTR) {
 				continue; /* signal */
@@ -71,7 +71,7 @@
 
 		if (cred) {
 			len = sizeof(struct ucred);
-			getsockopt(newsock, SOL_SOCKET, SO_PEERCRED,cred,&len);
+			getsockopt(newsock, SOL_SOCKET, SO_PEERCRED,cred,(socklen_t *)&len);
 		}
 
 		return newsock;
