diff -Naur krb5-1.9.1.ori/src/krb5-config.in krb5-1.9.1/src/krb5-config.in
--- krb5-1.9.1.ori/src/krb5-config.in	2010-01-19 13:44:57.000000000 -0500
+++ krb5-1.9.1/src/krb5-config.in	2011-09-26 18:27:09.018487087 -0400
@@ -186,7 +186,7 @@
 	    -e 's#\$(RPATH_FLAG)#'"$RPATH_FLAG"'#' \
 	    -e 's#\$(LDFLAGS)#'"$LDFLAGS"'#' \
 	    -e 's#\$(PTHREAD_CFLAGS)#'"$PTHREAD_CFLAGS"'#' \
-	    -e 's#\$(CFLAGS)#'"$CFLAGS"'#'`
+	    -e 's#\$(CFLAGS)##'`
 
     if test $library = 'kdb'; then
 	lib_flags="$lib_flags -lkdb5 $KDB5_DB_LIB"
@@ -214,9 +214,13 @@
     fi
 
     if test $library = 'krb5'; then
-	lib_flags="$lib_flags -lkrb5 -lk5crypto -lcom_err $GEN_LIB $LIBS $DL_LIB"
+	lib_flags="$lib_flags -lkrb5 -lk5crypto -lcom_err"
     fi
 
+    # If we ever support a flag to generate output suitable for static
+    # linking, we would output "-lkrb5support $GEN_LIB $LIBS $DL_LIB"
+    # here.
+
     echo $lib_flags
 fi
 
