--- lib-src/soundtouch/source/SoundTouch/Makefile.am.old	2006-11-02 13:30:00.000000000 +0900
+++ lib-src/soundtouch/source/SoundTouch/Makefile.am	2006-11-02 13:30:12.000000000 +0900
@@ -35,7 +35,7 @@
 
 # ??? test for -fcheck-new in configure.ac
 # other compiler flags to add
-AM_CXXFLAGS=-O3 -msse -fcheck-new -I../../include
+AM_CXXFLAGS=-fcheck-new -I../../include
 
 # other linking flags to add
 #libSoundTouch_la_LIBADD=
