--- src/FFmpeg.cpp.original	2010-04-14 14:59:48.000000000 +0200
+++ src/FFmpeg.cpp	2010-04-14 15:25:52.136755669 +0200
@@ -341,7 +341,7 @@
          }
          // Otherwize, resort to extension matching if available
          else if (fmt1->extensions) {
-            if (FFmpegLibsInst->match_ext(filename, fmt1->extensions)) {
+            if (FFmpegLibsInst->av_match_ext(filename, fmt1->extensions)) {
                score = 50;
             }
          }
@@ -814,7 +814,7 @@
    INITDYN(avformat,av_open_input_file);
    INITDYN(avformat,av_open_input_stream);
    INITDYN(avformat,get_buffer);
-   INITDYN(avformat,match_ext);
+   INITDYN(avformat,av_match_ext);
 
 #if FFMPEG_STABLE
    INITDYN(avformat,av_init_packet);
--- src/FFmpeg.h.original	2010-04-14 14:59:48.000000000 +0200
+++ src/FFmpeg.h	2010-04-14 15:26:39.932590241 +0200
@@ -222,7 +222,7 @@
    AVStream*         (*av_new_stream)                 (AVFormatContext *s, int id);
    AVFormatContext*  (*av_alloc_format_context)       (void);
    AVOutputFormat*   (*guess_format)                  (const char *short_name, const char *filename, const char *mime_type);
-   int               (*match_ext)                     (const char *filename, const char *extensions);
+   int               (*av_match_ext)                     (const char *filename, const char *extensions);
    int               (*av_write_trailer)              (AVFormatContext *s);
    int               (*av_interleaved_write_frame)    (AVFormatContext *s, AVPacket *pkt);
    int               (*av_write_frame)                (AVFormatContext *s, AVPacket *pkt);
