diff -Naur audacity-src-1.3.2-beta.orig/lib-src/soundtouch/include/STTypes.h audacity-src-1.3.2-beta/lib-src/soundtouch/include/STTypes.h
--- audacity-src-1.3.2-beta.orig/lib-src/soundtouch/include/STTypes.h	2006-10-29 08:06:01.000000000 +0900
+++ audacity-src-1.3.2-beta/lib-src/soundtouch/include/STTypes.h	2006-11-17 01:27:15.000000000 +0900
@@ -92,7 +92,6 @@
     /// However, if you're having difficulties getting the optimized routines
     /// compiled with your compler (e.g. some gcc compiler versions may be picky),
     /// you may wish to disable the optimizations to make the library compile.
-    #define ALLOW_OPTIMIZATIONS     1
 
 
     // If defined, allows the SIMD-optimized routines to take minor shortcuts
