--- Python-2.4.4/setup.py.orig	2007-01-14 12:50:46.000000000 -0600
+++ Python-2.4.4/setup.py	2007-01-14 12:54:02.000000000 -0600
@@ -685,12 +685,12 @@
                 if self.compiler.find_library_file(lib_dirs, 'ndbm'):
                     ndbm_libs = ['ndbm']
                 else:
-                    ndbm_libs = []
+                    ndbm_libs = ['gdbm', 'gdbm_compat']
                 exts.append( Extension('dbm', ['dbmmodule.c'],
                                        define_macros=[('HAVE_NDBM_H',None)],
                                        libraries = ndbm_libs ) )
             elif (self.compiler.find_library_file(lib_dirs, 'gdbm')
-                  and find_file("gdbm/ndbm.h", inc_dirs, []) is not None):
+                  and find_file("ndbm.h", inc_dirs, []) is not None):
                 exts.append( Extension('dbm', ['dbmmodule.c'],
                                        define_macros=[('HAVE_GDBM_NDBM_H',None)],
                                        libraries = ['gdbm'] ) )
