diff -ruN mysql-5.0.51.orig/Makefile.in mysql-5.0.51/Makefile.in
--- mysql-5.0.51.orig/Makefile.in	2007-12-07 12:59:21.000000000 +0000
+++ mysql-5.0.51/Makefile.in	2007-12-07 13:03:21.000000000 +0000
@@ -376,7 +376,7 @@
 SUBDIRS = . include @docs_dirs@ @zlib_dir@ @yassl_dir@ \
 			@readline_topdir@ sql-common scripts \
 			@thread_dirs@ pstack \
-			@sql_union_dirs@ @man_dirs@ tests \
+			strings mysys dbug extra regex bdb innobase myisam myisammrg heap vio sql @man_dirs@ tests \
 			netware @libmysqld_dirs@ \
 			@bench_dirs@ support-files @tools_dirs@
 
diff -ruN mysql-5.0.51.orig/include/Makefile.in mysql-5.0.51/include/Makefile.in
--- mysql-5.0.51.orig/include/Makefile.in	2007-12-07 12:59:20.000000000 +0000
+++ mysql-5.0.51/include/Makefile.in	2007-12-07 13:00:12.000000000 +0000
@@ -557,7 +557,6 @@
 	  test -z "$$dir" || $(mkdir_p) "$$dir"; \
 	done
 install: $(BUILT_SOURCES)
-	$(MAKE) $(AM_MAKEFLAGS) install-am
 install-exec: install-exec-am
 install-data: install-data-am
 uninstall: uninstall-am
diff -ruN mysql-5.0.51.orig/tests/Makefile.in mysql-5.0.51/tests/Makefile.in
--- mysql-5.0.51.orig/tests/Makefile.in	2007-12-07 12:59:20.000000000 +0000
+++ mysql-5.0.51/tests/Makefile.in	2007-12-07 13:01:51.000000000 +0000
@@ -96,7 +96,7 @@
 am_mysql_client_test_OBJECTS = mysql_client_test.$(OBJEXT) \
 	my_memmem.$(OBJEXT)
 mysql_client_test_OBJECTS = $(am_mysql_client_test_OBJECTS)
-am__DEPENDENCIES_1 = $(top_builddir)/libmysql/libmysqlclient.la
+am__DEPENDENCIES_1 = /usr/lib/mysql/libmysqlclient.so
 am__DEPENDENCIES_2 =
 mysql_client_test_DEPENDENCIES = $(am__DEPENDENCIES_1) \
 	$(am__DEPENDENCIES_2)
@@ -106,7 +106,7 @@
 thread_test_SOURCES = thread_test.c
 thread_test_OBJECTS = thread_test.$(OBJEXT)
 thread_test_LDADD = $(LDADD)
-thread_test_DEPENDENCIES = $(top_builddir)/libmysql/libmysqlclient.la
+thread_test_DEPENDENCIES = /usr/lib/mysql/libmysqlclient.so
 DEFAULT_INCLUDES = -I. -I$(srcdir) -I$(top_builddir)/include
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__depfiles_maybe = depfiles
@@ -405,7 +405,7 @@
 			$(openssl_includes)
 
 LDADD = @CLIENT_EXTRA_LDFLAGS@ \
-                        $(top_builddir)/libmysql/libmysqlclient.la
+                        /usr/lib/mysql/libmysqlclient.so
 
 mysql_client_test_LDADD = $(LDADD) $(CXXLDFLAGS)
 mysql_client_test_SOURCES = mysql_client_test.c\
diff -ruN mysql-5.0.51.orig/tools/Makefile.in mysql-5.0.51/tools/Makefile.in
--- mysql-5.0.51.orig/tools/Makefile.in	2007-12-07 12:59:19.000000000 +0000
+++ mysql-5.0.51/tools/Makefile.in	2007-12-07 13:02:32.000000000 +0000
@@ -369,7 +369,7 @@
 		$(openssl_includes)
 
 LDADD = @CLIENT_EXTRA_LDFLAGS@ \
-		$(top_builddir)/libmysql_r/libmysqlclient_r.la \
+		/usr/lib/mysql/libmysqlclient_r.so \
 		@openssl_libs@ @yassl_libs@ @ZLIB_LIBS@
 
 mysqltestmanager_SOURCES = mysqlmanager.c
