--- kdelibs-4.3.80/kded/applications.menu~	2009-12-12 15:53:10.989993958 +0100
+++ kdelibs-4.3.80/kded/applications.menu	2009-12-12 15:55:47.653312211 +0100
@@ -15,6 +15,7 @@
 		<Menuname>More</Menuname>
 	</DefaultLayout>
 	<Layout>
+		<Menuname>Archlinux</Menuname>
 		<Menuname>Internet</Menuname>
 		<Menuname>Office</Menuname>
 		<Menuname>Graphics</Menuname>
@@ -30,6 +31,13 @@
 	</Layout>
 
 	<Menu>
+		<Name>Archlinux</Name>                                                                                                   
+		<Directory>Archlinux.directory</Directory>                                                                               
+		<Include>                                                                                                                
+			<Category>Archlinux</Category>                                                                                           
+		</Include>
+	</Menu>
+	<Menu>
 		<Name>LostFound</Name>
 		<Directory>kde-unknown.directory</Directory>
 		<OnlyUnallocated/>
