diff -aur old/bin/Makefile.in new/bin/Makefile.in
--- old/bin/Makefile.in	2009-10-05 05:07:08.000000000 -0700
+++ new/bin/Makefile.in	2011-08-28 19:16:17.245495043 -0700
@@ -19,8 +19,7 @@
 VPATH =		@srcdir@
 top_srcdir =	@top_srcdir@
 
-SUBDIRS =	named rndc dig dnssec tests tools nsupdate \
-		check confgen @PKCS11_TOOLS@
+SUBDIRS =	dig
 TARGETS =
 
 @BIND9_MAKE_RULES@
diff -aur old/lib/Makefile.in new/lib/Makefile.in
--- old/lib/Makefile.in	2007-06-19 16:47:13.000000000 -0700
+++ new/lib/Makefile.in	2011-08-28 19:17:50.828688599 -0700
@@ -23,7 +23,7 @@
 # Attempt to disable parallel processing.
 .NOTPARALLEL:
 .NO_PARALLEL:
-SUBDIRS =	isc isccc dns isccfg bind9 lwres tests
+SUBDIRS =	isc dns isccfg bind9 lwres
 TARGETS =
 
 @BIND9_MAKE_RULES@
