Index: /trunk/opencv/modules/core/include/opencv2/core/core.hpp
===================================================================
--- /trunk/opencv/modules/core/include/opencv2/core/core.hpp	(revision 4804)
+++ /trunk/opencv/modules/core/include/opencv2/core/core.hpp	(revision 4861)
@@ -56,4 +56,5 @@
 #include <algorithm>
 #include <cmath>
+#include <cstddef>
 #include <complex>
 #include <map>
