diff -Naur OpenCV-2.0.0-orig/src/highgui/grfmt_png.cpp OpenCV-2.0.0/src/highgui/grfmt_png.cpp
--- OpenCV-2.0.0-orig/src/highgui/grfmt_png.cpp	2010-01-21 01:08:46.000000000 -0500
+++ OpenCV-2.0.0/src/highgui/grfmt_png.cpp	2010-01-21 01:13:11.000000000 -0500
@@ -223,7 +223,7 @@
                 png_set_palette_to_rgb( png_ptr );
 
             if( m_color_type == PNG_COLOR_TYPE_GRAY && m_bit_depth < 8 )
-                png_set_gray_1_2_4_to_8( png_ptr );
+                png_set_expand_gray_1_2_4_to_8( png_ptr );
 
             if( CV_MAT_CN(m_type) > 1 && color )
                 png_set_bgr( png_ptr ); // convert RGB to BGR
