--- gcc/intl.c.orig	2006-10-18 19:09:47.000000000 +0000
+++ gcc/intl.c	2006-10-18 19:16:47.000000000 +0000
@@ -39,8 +39,8 @@
   setlocale (LC_ALL, "");
 #endif
 
-  (void) bindtextdomain ("gcc", LOCALEDIR);
-  (void) textdomain ("gcc");
+  (void) bindtextdomain (PACKAGE, LOCALEDIR);
+  (void) textdomain (PACKAGE);
 }
 
 #endif
--- gcc/Makefile.in.orig	2006-10-18 19:21:58.000000000 +0000
+++ gcc/Makefile.in	2006-10-18 19:22:12.000000000 +0000
@@ -2172,6 +2172,7 @@
 intl.o: intl.c $(CONFIG_H) system.h intl.h Makefile
 	$(CC) $(ALL_CFLAGS) $(ALL_CPPFLAGS) $(INCLUDES) \
 	  -DLOCALEDIR=\"$(localedir)\" \
+	  -DPACKAGE=\"$(PACKAGE)\" \
 	  -c $(srcdir)/intl.c $(OUTPUT_OPTION)
 
 $(top_builddir)/intl/libintl.a: intl.all
@@ -2479,6 +2480,7 @@
 # Remake the info files.
 
 doc: $(BUILD_INFO) $(GENERATED_MANPAGES) gccbug
+iv := -$(shell echo $(version) | awk -F. '{printf "%s.%s",$$1,$$2}')
 
 INFOFILES = doc/cpp.info doc/gcc.info doc/gccint.info \
             doc/gccinstall.info doc/cppinternals.info
@@ -3694,7 +3696,7 @@
 GMSGFMT = @GMSGFMT@
 MSGMERGE = msgmerge
 
-PACKAGE = @PACKAGE@
+PACKAGE = @PACKAGE@$(iv)
 CATALOGS = @CATALOGS@
 
 .PHONY: build- install- build-po install-po update-po
