--- doc/asn2deb.pod.orig	2013-06-06 21:30:28.314150100 +0200
+++ doc/asn2deb.pod	2013-06-06 21:31:59.448850746 +0200
@@ -5,8 +5,21 @@
 
 =head1 SYNOPSIS
 
-B<asn2deb> [-a I<ASN.1 file>] [--asn=I<ASN.1 file>] [B<-d >I<opts>] [B<--dbopts=>I<opts>] [B<-e >I<address>]
-[B<--email=>I<address>] [B<-h>] [B<--help>] [B<-n >I<name>] [B<--name=>I<name>] [B<-p>] [B<--preserve>] [B<-v>] [B<--version>]
+B<asn2deb>
+S<[ B<-a> E<lt>ASN.1 fileE<gt> ]>
+S<[ B<--asn>=E<lt>ASN.1 fileE<gt> ]>
+S<[ B<-d> E<lt>optsE<gt> ]>
+S<[ B<--dbopts>=E<lt>optsE<gt> ]>
+S<[ B<-e> E<lt>addressE<gt> ]>
+S<[ B<--email>=E<lt>addressE<gt> ]>
+S<[ B<-h> ]>
+S<[ B<--help> ]>
+S<[ B<-n> E<lt>nameE<gt> ]>
+S<[ B<--name>=E<lt>nameE<gt> ]>
+S<[ B<-p> ]>
+S<[ B<--preserve> ]>
+S< [B<-v> ]>
+S<[ B<--version> ]>
 
 =head1 DESCRIPTION
 
@@ -19,31 +32,31 @@
 
 =over 4
 
-=item B<-a >I<ASN.1 file> B<--asn=>I<ASN.1 file>
+=item -a E<lt>ASN.1 fileE<gt> --asn=E<lt>ASN.1 fileE<gt>
 
 ASN.1 file to use (mandatory).
 
-=item B<-d >I<opts> B<--dbopts=>I<opts>
+=item -d E<lt>optsE<gt> --dbopts=E<lt>optsE<gt>
 
 options for dpkg-buildpackage.
 
-=item B<-e >I<address> B<--email=>I<address>
+=item -e E<lt>addressE<gt> --email=E<lt>addressE<gt>
 
 use e-mail address.
 
-=item B<-h> B<--help>
+=item -h --help
 
 print help and exit.
 
-=item B<-n >I<name> B<--name=>I<name>
+=item -n E<lt>nameE<gt> --name=E<lt>nameE<gt>
 
 use user name.
 
-=item B<-p> B<--preserve>
+=item -p --preserve
 
 do not overwrite files.
 
-=item B<-v> B<--version>
+=item -v --version
 
 print version and exit.
 
@@ -68,8 +81,8 @@
 
 =head1 AUTHOR
 
-B<W. Borgert> Author. 
+B<W. Borgert> Author.
 
 =head1 COPYRIGHT
 
-Copyright (C) 2003, 2005 W. Borger
\ No newline at end of file
+Copyright (C) 2003, 2005 W. Borger
