--- nxdialog~	2011-02-01 09:05:28.993333438 +0100
+++ nxdialog	2011-02-01 09:05:29.010000106 +0100
@@ -85,29 +85,6 @@
 then
 	dialog_interface="xdialog"
 	DIALOG=/usr/bin/Xdialog # just in case that we have no good path
-elif [ -x /usr/bin/dialog ]
-then
-	#JJK: Added 'dialog_interface=dialog' option because Xdialog not standard
-	#JJK: on some distros such as Fedora and xmessage won't handle long
-	#JJK: lists of ppd files while the combination of 'dialog' and 'xterm'
-	#JJK: should be present on most setups.
-
-	dialog_interface="dialog"
-	DIALOG=/usr/bin/dialog
-	if [ -z "$NXCLIENT_FIRST_TIME" ]
-	then
-		# Run only once in case of subdialogs. Capture result in tempfile
-		TMPFILE=$(mktemp /tmp/nxclient.XXXXX)
-		export NXCLIENT_FIRST_TIME=1
-
-		xterm -geometry 120x24+100+100 +sb -title "NXclient" -e \
-		       /bin/bash -c "$DIALOG --infobox 'Please wait...' 3 25; $0 $INPUTS | tee $TMPFILE" || exit 1
-
-		#Need to recover the last line output (and not remove non-printing chars, because --stdout is used)
-		tail -1 $TMPFILE 
-		rm -f $TMPFILE
-		exit 0
-	fi
 else
 	dialog_interface="xmessage"
 	xmessage=$(which xmessage 2>/dev/null)
