diff -Naur kde-i18n-ru-3.5.9.orig/data/kdeedu/kturtle/Makefile.in kde-i18n-ru-3.5.9/data/kdeedu/kturtle/Makefile.in
--- kde-i18n-ru-3.5.9.orig/data/kdeedu/kturtle/Makefile.in	2008-02-13 15:26:23.000000000 +0100
+++ kde-i18n-ru-3.5.9/data/kdeedu/kturtle/Makefile.in	2008-02-23 17:18:30.000000000 +0100
@@ -215,7 +215,7 @@
 	echo ' cd $(top_srcdir) && $(AUTOMAKE) --gnu  data/kdeedu/kturtle/Makefile'; \
 	cd $(top_srcdir) && \
 	  $(AUTOMAKE) --gnu  data/kdeedu/kturtle/Makefile
-	cd $(top_srcdir) && perl ../scripts/admin/am_edit -p../scripts/admin data/kdeedu/kturtle/Makefile.in
+#	cd $(top_srcdir) && perl ../scripts/admin/am_edit -p../scripts/admin data/kdeedu/kturtle/Makefile.in
 .PRECIOUS: Makefile
 Makefile: $(srcdir)/Makefile.in $(top_builddir)/config.status
 	@case '$?' in \
@@ -444,7 +444,7 @@
 	echo ' cd $(top_srcdir) && $(AUTOMAKE) --gnu  data/kdeedu/kturtle/Makefile'; \
 	cd $(top_srcdir) && \
 	  $(AUTOMAKE) --gnu  data/kdeedu/kturtle/Makefile
-	cd $(top_srcdir) && perl ../scripts/admin/am_edit -p../scripts/admin data/kdeedu/kturtle/Makefile.in
+#	cd $(top_srcdir) && perl ../scripts/admin/am_edit -p../scripts/admin data/kdeedu/kturtle/Makefile.in
 
 
 #>+ 21
diff -Naur kde-i18n-ru-3.5.9.orig/data/kdeedu/kturtle/Makefile.in kde-i18n-ru-3.5.9/data/kdeedu/kturtle/Makefile.in
--- kde-i18n-ru-3.5.9.orig/data/kdeedu/kturtle/Makefile.in	2008-02-23 17:35:21.000000000 +0100
+++ kde-i18n-ru-3.5.9/data/kdeedu/kturtle/Makefile.in	2008-02-23 17:38:50.000000000 +0100
@@ -178,7 +178,7 @@
 xdg_directorydir = @xdg_directorydir@
 xdg_menudir = @xdg_menudir@
 txt_DATA = curly.logo square.logo triangle.logo \
-flower.logo arrow.logo example2.logo example3.logo
+flower.logo arrow.logo 
 
 txtdir = $(kde_datadir)/kturtle/examples/ru
 xml_DATA = logohighlightstyle.ru.xml
