diff --git a/src/recognition-opencv-lbph/facerec_borrowed.cpp b/src/recognition-opencv-lbph/facerec_borrowed.cpp
index 3c37ce2..b21df1c 100644
--- a/src/recognition-opencv-lbph/facerec_borrowed.cpp
+++ b/src/recognition-opencv-lbph/facerec_borrowed.cpp
@@ -377,7 +377,7 @@ void LBPHFaceRecognizer::train(InputArrayOfArrays _in_src, InputArray _inm_label
     }
 }
 
-#if OPENCV_TEST_VERSION(3,1,0)
+#if OPENCV_TEST_VERSION(3,2,0)
 void LBPHFaceRecognizer::predict(InputArray _src, int &minClass, double &minDist) const
 #else
 void LBPHFaceRecognizer::predict(cv::InputArray _src, cv::Ptr<cv::face::PredictCollector> collector, const int state) const
@@ -400,7 +400,7 @@ void LBPHFaceRecognizer::predict(cv::InputArray _src, cv::Ptr<cv::face::PredictC
                                       m_grid_y,                                                          /* grid size y                 */
                                       true                                                               /* normed histograms           */
                                      );
-#if OPENCV_TEST_VERSION(3,1,0)
+#if OPENCV_TEST_VERSION(3,2,0)
     minDist      = DBL_MAX;
     minClass     = -1;
 #else
@@ -416,7 +416,7 @@ void LBPHFaceRecognizer::predict(cv::InputArray _src, cv::Ptr<cv::face::PredictC
         {
             double dist = compareHist(m_histograms[sampleIdx], query, CV_COMP_CHISQR);
 
-#if OPENCV_TEST_VERSION(3,1,0)
+#if OPENCV_TEST_VERSION(3,2,0)
             if((dist < minDist) && (dist < m_threshold))
             {
                 minDist  = dist;
@@ -463,7 +463,7 @@ void LBPHFaceRecognizer::predict(cv::InputArray _src, cv::Ptr<cv::face::PredictC
             double mean = sum / it->second.size();
             s          += QString::fromLatin1("%1: %2 - ").arg(it->first).arg(mean);
 
-#if OPENCV_TEST_VERSION(3,1,0)
+#if OPENCV_TEST_VERSION(3,2,0)
             if((mean < minDist) && (mean < m_threshold))
             {
                 minDist = mean;
@@ -505,7 +505,7 @@ void LBPHFaceRecognizer::predict(cv::InputArray _src, cv::Ptr<cv::face::PredictC
             scoreMap[it->second]++;
         }
 
-#if OPENCV_TEST_VERSION(3,1,0)
+#if OPENCV_TEST_VERSION(3,2,0)
         minDist = 0;
 #endif
         QString s = QString::fromLatin1("Nearest Neighbor score: ");
@@ -515,7 +515,7 @@ void LBPHFaceRecognizer::predict(cv::InputArray _src, cv::Ptr<cv::face::PredictC
             double score = double(it->second) / countMap.at(it->first);
             s           += QString::fromLatin1("%1/%2 %3  ").arg(it->second).arg(countMap.at(it->first)).arg(score);
 
-#if OPENCV_TEST_VERSION(3,1,0)
+#if OPENCV_TEST_VERSION(3,2,0)
             if (score > minDist)
             {
                 minDist  = score;
@@ -534,7 +534,7 @@ void LBPHFaceRecognizer::predict(cv::InputArray _src, cv::Ptr<cv::face::PredictC
     }
 }
 
-#if OPENCV_TEST_VERSION(3,1,0)
+#if OPENCV_TEST_VERSION(3,2,0)
 int LBPHFaceRecognizer::predict(InputArray _src) const
 {
     int    label;
diff --git a/src/recognition-opencv-lbph/facerec_borrowed.h b/src/recognition-opencv-lbph/facerec_borrowed.h
index f197d22..0f13d55 100644
--- a/src/recognition-opencv-lbph/facerec_borrowed.h
+++ b/src/recognition-opencv-lbph/facerec_borrowed.h
@@ -126,7 +126,7 @@ public:
     void update(cv::InputArrayOfArrays src, cv::InputArray labels);
 
 
-#if OPENCV_TEST_VERSION(3,1,0)
+#if OPENCV_TEST_VERSION(3,2,0)
     /**
      * Predicts the label of a query image in src.
      */
