--- ati-powermode.sh.old	2007-06-02 23:29:13.000000000 -0400
+++ ati-powermode.sh	2007-06-02 23:33:06.000000000 -0400
@@ -69,7 +69,9 @@
     echo "high power"
     if [ ${has_pplib} -eq 1 ]; then
         su $user -c '/usr/bin/aticonfig --pplib-cmd="notify psrc ac"'
     else
-        su $user -c "/usr/bin/aticonfig --set-powerstate=3 "
+        newlevel=$(aticonfig --lsp | grep ":" | tail -1 | cut -d':' -f 1 | sed -e "s/\*//g" -e "s/ *//g")
+        [ -z "$newlevel" ] && newlevel=3
+        su $user -c "/usr/bin/aticonfig --set-powerstate=$newlevel --effective=now"
     fi
 fi
