=== modified file 'bitlbee.conf'
--- old/bitlbee.conf	2006-01-04 11:16:58 +0000
+++ new/bitlbee.conf	2007-02-22 00:53:52 +0000
@@ -19,6 +19,13 @@
 ##
 # RunMode = Inetd
 
+## User:
+## 
+## If BitlBee is run as root as a Daemon or ForkDaemon, it can drop root
+## privileges, and change to specified user.
+##
+# User = bitlbee
+
 ## DaemonPort/DaemonInterface:
 ##
 ## For daemon mode, you can specify on what interface and port the daemon

=== modified file 'conf.c'
--- old/conf.c	2007-02-18 23:47:17 +0000
+++ new/conf.c	2007-02-22 01:36:34 +0000
@@ -66,6 +66,7 @@
 	conf->motdfile = g_strdup( ETCDIR "/motd.txt" );
 	conf->ping_interval = 180;
 	conf->ping_timeout = 300;
+	conf->user = "";
 	proxytype = 0;
 	
 	i = conf_loadini( conf, CONF_FILE );
@@ -79,7 +80,7 @@
 		fprintf( stderr, "Warning: Unable to read configuration file `%s'.\n", CONF_FILE );
 	}
 	
-	while( argc > 0 && ( opt = getopt( argc, argv, "i:p:P:nvIDFc:d:hR:" ) ) >= 0 )
+	while( argc > 0 && ( opt = getopt( argc, argv, "i:p:P:nvIDFc:d:hR:u:" ) ) >= 0 )
 	/*     ^^^^ Just to make sure we skip this step from the REHASH handler. */
 	{
 		if( opt == 'i' )
@@ -139,6 +140,7 @@
 			        "  -I  Classic/InetD mode. (Default)\n"
 			        "  -D  Daemon mode. (Still EXPERIMENTAL!)\n"
 			        "  -F  Forking daemon. (one process per client)\n"
+				"  -u  Run daemon as specified user.\n"
 			        "  -P  Specify PID-file (not for inetd mode)\n"
 			        "  -i  Specify the interface (by IP address) to listen on.\n"
 			        "      (Default: 0.0.0.0 (any interface))\n"
@@ -158,6 +160,10 @@
 			   mode anyway!) */
 			ipc_master_set_statefile( optarg );
 		}
+		else if(opt == 'u' )
+		{
+			conf->user = g_strdup( optarg );
+		}
 	}
 	
 	if( conf->configdir[strlen(conf->configdir)-1] != '/' )
@@ -293,6 +299,10 @@
 				
 				g_free( url );
 			}
+			else if( g_strcasecmp( ini->key, "user" ) == 0 )
+			{
+				conf->user = g_strdup( ini->value );
+			}
 			else
 			{
 				fprintf( stderr, "Error: Unknown setting `%s` in configuration file.\n", ini->key );

=== modified file 'conf.h'
--- old/conf.h	2006-02-02 13:21:44 +0000
+++ new/conf.h	2007-02-22 00:51:18 +0000
@@ -48,6 +48,7 @@
 	char **migrate_storage;
 	int ping_interval;
 	int ping_timeout;
+	char *user;
 } conf_t;
 
 conf_t *conf_load( int argc, char *argv[] );

=== modified file 'unix.c'
--- old/unix.c	2006-06-25 17:07:25 +0000
+++ new/unix.c	2007-02-22 01:46:15 +0000
@@ -33,6 +33,7 @@
 #include <unistd.h>
 #include <sys/time.h>
 #include <sys/wait.h>
+#include <pwd.h>
 
 global_t global;	/* Against global namespace pollution */
 
@@ -59,6 +60,19 @@
 	if( global.conf == NULL )
 		return( 1 );
 	
+	if( ( global.conf->runmode == RUNMODE_DAEMON || 
+		global.conf->runmode == RUNMODE_FORKDAEMON ) 
+		&& ( !getuid() || !geteuid() ) )
+	{
+		struct passwd *pw = NULL;
+		pw = getpwnam(global.conf->user);
+		if (pw)
+		{
+			setgid(pw->pw_gid);
+			setuid(pw->pw_uid);
+		}
+	}
+
 	if( global.conf->runmode == RUNMODE_INETD )
 	{
 		i = bitlbee_inetd_init();

