diff -Naur dmenu-3.2/config.mk dmenu-3.2.patch/config.mk
--- dmenu-3.2/config.mk	2007-05-30 06:21:59.000000000 -0400
+++ dmenu-3.2.patch/config.mk	2007-06-05 21:54:26.000000000 -0400
@@ -4,11 +4,11 @@
 # Customize below to fit your system
 
 # paths
-PREFIX = /usr/local
+PREFIX = /usr
 MANPREFIX = ${PREFIX}/share/man
 
-X11INC = /usr/X11R6/include
-X11LIB = /usr/X11R6/lib
+X11INC = /usr/include/X11
+X11LIB = /usr/lib/X11
 
 # includes and libs
 INCS = -I. -I/usr/include -I${X11INC}
