From 3c223e8f7b03e2d7f8c31faeeeeb37030c461176 Mon Sep 17 00:00:00 2001
From: Alan Coopersmith <alan.coopersmith@sun.com>
Date: Sat, 10 Jan 2009 00:39:07 +0000
Subject: Remove xorgconfig & xorgcfg from See Also list in man page

---
diff --git a/man/vmware.man b/man/vmware.man
index 65f0d94..fd12717 100644
--- a/man/vmware.man
+++ b/man/vmware.man
@@ -38,6 +38,6 @@ and will not be overridden at runtime.
 The format is "Width1xHeight1+Xoffset1+Yoffset1;Width2xHeight2+Xoffset2+Yoffset2"
 and so on. Negative offsets are not supported.
 .SH "SEE ALSO"
-__xservername__(__appmansuffix__), __xconfigfile__(__filemansuffix__), xorgconfig(__appmansuffix__), Xserver(__appmansuffix__), X(__miscmansuffix__)
+__xservername__(__appmansuffix__), __xconfigfile__(__filemansuffix__), Xserver(__appmansuffix__), X(__miscmansuffix__)
 .SH AUTHORS
 Copyright (c) 1999-2007 VMware, Inc.
--
cgit v0.8.2
