# Description: Fix location of external programs.
diff -Nur -x '*.orig' -x '*~' gdm-2.27.90/daemon/gdm-server.c gdm-2.27.90.new/daemon/gdm-server.c
--- gdm-2.27.90/daemon/gdm-server.c	2009-08-25 05:31:18.000000000 +1000
+++ gdm-2.27.90.new/daemon/gdm-server.c	2009-08-25 14:35:12.000000000 +1000
@@ -127,7 +127,7 @@
         g_return_val_if_fail (GDM_IS_SERVER (server), NULL);
 
         error = NULL;
-        command = g_strdup_printf (LIBEXECDIR "/ck-get-x11-display-device --display %s",
+        command = g_strdup_printf ("/usr/lib/ConsoleKit/ck-get-x11-display-device --display %s",
                                    server->priv->display_name);
 
         g_debug ("GdmServer: Running helper %s", command);
diff -Nur -x '*.orig' -x '*~' gdm-2.27.90/data/greeter-autostart/gnome-settings-daemon.desktop.in.in gdm-2.27.90.new/data/greeter-autostart/gnome-settings-daemon.desktop.in.in
--- gdm-2.27.90/data/greeter-autostart/gnome-settings-daemon.desktop.in.in	2009-08-25 05:31:18.000000000 +1000
+++ gdm-2.27.90.new/data/greeter-autostart/gnome-settings-daemon.desktop.in.in	2009-08-25 14:35:12.000000000 +1000
@@ -1,8 +1,8 @@
 [Desktop Entry]
 Type=Application
 _Name=GNOME Settings Daemon
-TryExec=@LIBEXECDIR@/gnome-settings-daemon
-Exec=@LIBEXECDIR@/gnome-settings-daemon --gconf-prefix=/apps/gdm/simple-greeter/settings-manager-plugins
+TryExec=/usr/bin/gnome-settings-daemon
+Exec=/usr/bin/gnome-settings-daemon --gconf-prefix=/apps/gdm/simple-greeter/settings-manager-plugins
 OnlyShowIn=GNOME;
 X-GNOME-Autostart-Phase=Initialization
 X-GNOME-Autostart-Notify=true
diff -Nur -x '*.orig' -x '*~' gdm-2.27.90/gui/simple-chooser/gdm-chooser-session.c gdm-2.27.90.new/gui/simple-chooser/gdm-chooser-session.c
--- gdm-2.27.90/gui/simple-chooser/gdm-chooser-session.c	2009-08-25 05:31:18.000000000 +1000
+++ gdm-2.27.90.new/gui/simple-chooser/gdm-chooser-session.c	2009-08-25 14:35:12.000000000 +1000
@@ -131,7 +131,7 @@
         ret = FALSE;
 
         error = NULL;
-        g_spawn_command_line_async (LIBEXECDIR "/gnome-settings-daemon --gconf-prefix=/apps/gdm/simple-chooser/settings-manager-plugins", &error);
+        g_spawn_command_line_async ("/usr/bin/gnome-settings-daemon --gconf-prefix=/apps/gdm/simple-chooser/settings-manager-plugins", &error);
         if (error != NULL) {
                 g_warning ("Error starting settings daemon: %s", error->message);
                 g_error_free (error);
--- gdm-2.28.0/data/greeter-autostart/polkit-gnome-authentication-agent-1.desktop.in.orig	2009-09-29 10:07:17.000000000 +0200
+++ gdm-2.28.0/data/greeter-autostart/polkit-gnome-authentication-agent-1.desktop.in	2009-09-29 10:07:29.000000000 +0200
@@ -1,5 +1,5 @@
 [Desktop Entry]
 Type=Application
 _Name=PolicyKit Authentication Agent
-Exec=/usr/libexec/polkit-gnome-authentication-agent-1
+Exec=/usr/lib/polkit-gnome/polkit-gnome-authentication-agent-1
 OnlyShowIn=GNOME;
