diff -rupN a/src/gifcodec.c b/src/gifcodec.c
--- a/src/gifcodec.c	2011-12-02 18:23:12.000000000 +0100
+++ b/src/gifcodec.c	2014-06-01 13:20:17.027203704 +0200
@@ -581,7 +581,7 @@ gdip_load_gif_image (void *stream, GpIma
 	}
 
 	FreeExtensionMono(&global_extensions);
-	DGifCloseFile (gif);
+	DGifCloseFile (gif, NULL);
 
 	*image = result;
 	return Ok;
@@ -597,7 +597,7 @@ error:
 
 	if (gif != NULL) {
 		FreeExtensionMono (&global_extensions);
-		DGifCloseFile (gif);
+		DGifCloseFile (gif, NULL);
 	}
 
 	*image = NULL;
@@ -923,7 +923,7 @@ gdip_save_gif_image (void *stream, GpIma
 		}
 	}
 
-	EGifCloseFile (fp);	
+	EGifCloseFile (fp, NULL);	
 	
 	return Ok;
 
