--- a/src/core/k9tools.cpp	2009-05-27 20:15:20.000000000 -0700
+++ b/src/core/k9tools.cpp	2009-05-27 20:15:20.000000000 -0700
@@ -11,6 +11,7 @@
 //
 #include "k9tools.h"
 #include <kstandarddirs.h>
+#include <stdint.h>
 #include <qdir.h>
 #include <QComboBox>
 #ifdef __FreeBSD__
--- a/src/core/k9dvdbackupinterface.h	2009-05-27 20:51:26.000000000 -0700
+++ b/src/core/k9dvdbackupinterface.h	2009-05-27 20:51:26.000000000 -0700
@@ -19,6 +19,7 @@
  ***************************************************************************/
 #ifndef K9DVDBACKUPINTERFACE_H
 #define K9DVDBACKUPINTERFACE_H
+#include <stdint.h>
 
 #include <QObject>
 class k9DVDBackupInterface : public QObject
