diff -uNr acx-20070101/common.c acx-20070101-patched/common.c
--- acx-20070101/common.c	2007-01-10 21:15:27.000000000 +0000
+++ acx-20070101-patched/common.c	2007-08-22 22:55:52.000000000 +0100
@@ -2753,8 +2753,11 @@
 
 	skb->dev = adev->ndev;
 	skb->dev->last_rx = jiffies;
-
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,22)
+	skb_reset_mac_header(skb);
+#else
 	skb->mac.raw = skb->data;
+#endif
 	skb->ip_summed = CHECKSUM_NONE;
 	skb->pkt_type = PACKET_OTHERHOST;
 	skb->protocol = htons(ETH_P_80211_RAW);
diff -uNr acx-20070101/pci.c acx-20070101-patched/pci.c
--- acx-20070101/pci.c	2007-01-10 12:27:16.000000000 +0000
+++ acx-20070101-patched/pci.c	2007-08-22 22:55:52.000000000 +0100
@@ -2104,7 +2104,11 @@
 /* TODO: pci_set_power_state(pdev, PCI_D0); ? */
 
 	/* request shared IRQ handler */
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,21)	
+	if (request_irq(ndev->irq, acxpci_i_interrupt, IRQF_SHARED, ndev->name, ndev)) {
+#else
 	if (request_irq(ndev->irq, acxpci_i_interrupt, SA_SHIRQ, ndev->name, ndev)) {
+#endif		
 		printk("%s: request_irq FAILED\n", ndev->name);
 		result = -EAGAIN;
 		goto done;
