diff -urNad trunk~/pdns/dnspacket.cc trunk/pdns/dnspacket.cc
--- trunk~/pdns/dnspacket.cc	2006-03-16 21:18:00.000000000 +0100
+++ trunk/pdns/dnspacket.cc	2006-11-09 21:42:48.216274631 +0100
@@ -540,9 +540,9 @@
   
   string piece4=makeSoaHostmasterPiece(soadata.hostmaster);
 
-  char piece5[20];
-  
-  uint32_t *i_p=(uint32_t *)piece5;
+  uint32_t piece5[5];
+
+  uint32_t *i_p=piece5;
   
   uint32_t soaoffset=0;
   if(soadata.serial && (soaoffset=arg().asNum("soa-serial-offset")))
@@ -561,7 +561,7 @@
   stringbuffer.append(p,10);
   stringbuffer+=piece3;
   stringbuffer+=piece4;
-  stringbuffer.append(piece5,20);
+  stringbuffer.append((char*)piece5,20);
   if(place==DNSResourceRecord::ANSWER)
     d.ancount++;
   else
