diff -urNad trunk~/pdns/backends/bind/bindbackend2.cc trunk/pdns/backends/bind/bindbackend2.cc
--- trunk~/pdns/backends/bind/bindbackend2.cc	2005-11-01 13:10:30.000000000 +0100
+++ trunk/pdns/backends/bind/bindbackend2.cc	2006-10-29 10:58:24.439566203 +0100
@@ -49,6 +49,8 @@
     on start of query, we find the best zone to answer from
 */
 
+static Bind2Backend *us;
+
 Bind2Backend::name_id_map_t Bind2Backend::s_name_id_map;
 
 // this map contains BB2DomainInfo structs, each of which contains a *pointer* to domain data
@@ -147,7 +149,7 @@
     throw DBException("Unable to commit (rename to: '"+s_id_zone_map[d_transaction_id].d_filename+"') AXFRed zone: "+stringerror());
 
 
-  queueReload(&s_id_zone_map[d_transaction_id]);
+  us->queueReload(&s_id_zone_map[d_transaction_id]);
 
   d_transaction_id=0;
 
@@ -296,8 +298,6 @@
   return ret;
 }
 
-static Bind2Backend *us;
-
 static void InsertionCallback(unsigned int domain_id, const string &domain, const string &qtype, const string &content, int ttl, int prio)
 {
   us->insert(domain_id, domain, qtype, content, ttl, prio);
@@ -431,7 +431,7 @@
 
 void Bind2Backend::rediscover(string *status)
 {
-  loadConfig(status);
+  us->loadConfig(status);
 }
 
 void Bind2Backend::loadConfig(string* status)
@@ -695,7 +695,7 @@
     
     if(!bbd.current()) {
       L<<Logger::Warning<<"Zone '"<<bbd.d_name<<"' ("<<bbd.d_filename<<") needs reloading"<<endl;
-      queueReload(&bbd);
+      us->queueReload(&bbd);
       d_handle.d_records=s_id_zone_map[iditer->second].d_records; // give it a *fresh* copy
     }
   }
