diff -urN sfs-0.7.2/crypt/random_prime.C sfs-0.7.2-new/crypt/random_prime.C
--- sfs-0.7.2/crypt/random_prime.C	2001-01-13 11:46:11.000000000 -0800
+++ sfs-0.7.2-new/crypt/random_prime.C	2004-03-02 22:39:11.000000000 -0800
@@ -336,7 +336,7 @@
   bigint t1, t2;
   for (;;) {
     next_weak ();
-    if (!tmp || (fermat2_test (tmp, &t1, &t2) == 2 && tmp.probab_prime (iter)))
+    if (!tmp || (fermat2_test (tmp, &t1, &t2) && tmp.probab_prime (iter)))
       return tmp;
   }
 }
