--- src/systeminfo.cpp	2012-10-04 02:53:03.000000000 +0200
+++ src/systeminfo.cpp	2012-10-04 07:18:45.111088591 +0200
@@ -114,7 +114,7 @@
 		{ LinuxSlackware,	OsAppendFile,	"/etc/slackware-version",	"Slackware Linux"	},
 		{ LinuxPLD,		OsUseFile,	"/etc/pld-release",		"PLD Linux"		},
 		{ LinuxAurox,		OsUseName,	"/etc/aurox-release",		"Aurox Linux"		},
-		{ LinuxArch,		OsUseFile,	"/etc/arch-release",		"Arch Linux"		},
+		{ LinuxArch,		OsUseName,	"/etc/arch-release",		"Arch Linux"		},
 		{ LinuxLFS,		OsAppendFile,	"/etc/lfs-release",		"LFS Linux"		},
 
 		// untested
