diff -upr Thunar-1.4.0.orig/thunar/thunar-gio-extensions.c Thunar-1.4.0/thunar/thunar-gio-extensions.c
--- Thunar-1.4.0.orig/thunar/thunar-gio-extensions.c	2012-05-26 07:03:39.000000000 +0300
+++ Thunar-1.4.0/thunar/thunar-gio-extensions.c	2012-05-26 07:05:06.000000000 +0300
@@ -550,6 +550,11 @@ thunar_g_volume_is_present (GVolume *vol
       has_media = g_drive_has_media (drive);
       g_object_unref (drive);
     }
+  else
+    {
+      /* XXX: hack to show loop devices with gvfs 1.12 / udisks2 */
+      has_media = g_volume_can_mount (volume);
+    }
 
   mount = g_volume_get_mount (volume);
   if (mount != NULL)
