--- snes9x-1.5-src.orig/unix/opengl.cpp	2006-06-30 13:34:57.000000000 +0000
+++ snes9x-1.5-src/unix/opengl.cpp	2006-09-01 08:48:39.000000000 +0000
@@ -214,7 +214,7 @@
 
 	// Strip dots from the version string
 	char *ptr;
-	while (ptr = strchr (ver, '.'))
+	while ((ptr = strchr (ver, '.')))
 	    memmove (ptr, ptr + 1, strlen (ptr + 1) + 1);
 	
 	// Pad the string with zeros to 4 digits
@@ -381,7 +381,7 @@
                 int i = 0;
                 for (uint32 y = 0; y < pheight; y++)
                 {
-                    uint16 *ptr = (uint16 *) (GFX.Screen + (y + startl) * GFX.Pitch2) + startx;
+                    uint16 *ptr = (uint16 *) (GFX.Screen + (y + startl) * GFX.Pitch) + startx;
                     for (uint32 x = 0; x < pwidth; x++)
                     {
                         int color = *ptr++;
@@ -399,7 +399,7 @@
 		for (uint32 y = 0; y < pheight; y++)
 		{
 		    memmove (basetexbuffer + sizeof (uint16) * texture_size * y,
-			     (GFX.Screen + (y + startl) * GFX.Pitch2) + startx,
+			     (GFX.Screen + (y + startl) * GFX.Pitch) + startx,
 			     sizeof (uint16) * texture_size);
 		}
 		data = basetexbuffer;
