# Author: Sébastien Luttringer <seblu@seblu.net>
--- a/Makefile	2011-12-07 19:23:51.000000000 +0100
+++ b/Makefile	2013-05-13 00:48:19.620000000 +0200
@@ -46,9 +46,6 @@
 	gzip -c -9 dkms.8 > $(MAN)/dkms.8.gz
 	chmod 0644 $(MAN)/dkms.8.gz
 	touch --reference=dkms.8 $(MAN)/dkms.8.gz
-	mkdir   -p -m 0755 $(KCONF)/prerm.d $(KCONF)/postinst.d
-	install -p -m 0755 kernel_prerm.d_dkms  $(KCONF)/prerm.d/dkms
-	install -p -m 0755 kernel_postinst.d_dkms $(KCONF)/postinst.d/dkms
 
 DOCFILES=sample.spec sample.conf AUTHORS COPYING README.dkms sample-suse-9-mkkmp.spec sample-suse-10-mkkmp.spec
 
