--- extra-cmake-modules-5.30.0/find-modules/FindPythonModuleGeneration.cmake.0  2017-01-11 21:36:42.219762963 +0000
+++ extra-cmake-modules-5.30.0/find-modules/FindPythonModuleGeneration.cmake    2017-01-11 21:37:48.932898438 +0000
@@ -409,7 +409,6 @@
     foreach(pyversion ${_pyversions})
         file(MAKE_DIRECTORY
             "${CMAKE_CURRENT_BINARY_DIR}/py${pyversion}/${pythonnamespace_value}")
-        execute_process(COMMAND "${CMAKE_COMMAND}" -E touch "${CMAKE_CURRENT_BINARY_DIR}/py${pyversion}/${pythonnamespace_value}/__init__.py")
 
         add_library(Py${pyversion}KF5${modulename_value} MODULE
           "${CMAKE_CURRENT_BINARY_DIR}/pybuild/${pythonnamespace_value}/${modulename_value}/unified${modulename_value}.cpp"

