--- extra-cmake-modules-5.29.0/find-modules/FindPythonModuleGeneration.cmake.0	2017-01-06 09:25:44.345565087 +0000
+++ extra-cmake-modules-5.29.0/find-modules/FindPythonModuleGeneration.cmake	2017-01-06 09:29:03.881879316 +0000
@@ -192,7 +192,7 @@
   while(NOT libclang_LIBRARY AND NOT _LIBCLANG_FIND_VERSION EQUAL _LIBCLANG_MIN_MAJOR_VERSION)
     math(EXPR _LIBCLANG_FIND_VERSION "${_LIBCLANG_FIND_VERSION} - 1")
     set(_GPB_CLANG_SUFFIX ${_LIBCLANG_FIND_VERSION}.0)
-    find_library(libclang_LIBRARY clang-${_LIBCLANG_FIND_VERSION}.0)
+    find_library(libclang_LIBRARY NAMES clang-${_LIBCLANG_FIND_VERSION}.0 clang)
   endwhile()
 
   if (NOT libclang_LIBRARY)

