diff -aur namcap-3.2.8/namcap-tags namcap-3.2.8.new/namcap-tags
--- namcap-3.2.8/namcap-tags	2018-07-07 03:52:57.000000000 +0200
+++ namcap-3.2.8.new/namcap-tags	2018-08-04 17:43:17.776311307 +0200
@@ -98,6 +98,7 @@
 dconf-needed-for-glib-schemas :: needed for glib schemas
 glib2-needed-for-gio-modules :: needed for GIO modules
 hicolor-icon-theme-needed-for-hicolor-dir :: needed for hicolor theme hierarchy
+desktop-file-utils-needed :: needed for update-desktop-database
 kdebase-runtime-needed %s :: needed for programs %s
 java-runtime-needed %s :: found class files %s
 libraries-needed %s %s :: libraries %s needed in files %s
