From 4ece4901d13b9fa590a538cc2133374d3c17df6f Mon Sep 17 00:00:00 2001
From: Jelle van der Waa <jelle@vdwaa.nl>
Date: Tue, 11 Sep 2018 18:21:39 +0200
Subject: Ignore .so for no PIE check

Signed-off-by: Kyle Keen <keenerd@gmail.com>
---
 Namcap/rules/elffiles.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Namcap/rules/elffiles.py b/Namcap/rules/elffiles.py
index 6cef680..4ad1e66 100644
--- a/Namcap/rules/elffiles.py
+++ b/Namcap/rules/elffiles.py
@@ -228,6 +228,8 @@ class NoPIERule(TarballRule):
 		for entry in tar:
 			if not entry.isfile():
 				continue
+			if '.so' in entry.name:
+				continue
 			fp = tar.extractfile(entry)
 			if not is_elf(fp):
 				continue
-- 
cgit v1.2.1-1-g437b

