diff -baur hydrogen-0.9.5.old/Sconstruct hydrogen-0.9.5/Sconstruct
--- hydrogen-0.9.5.old/Sconstruct	2011-08-11 18:00:07.889112348 +0800
+++ hydrogen-0.9.5/Sconstruct	2011-08-11 18:00:34.945632437 +0800
@@ -83,6 +83,7 @@
 	includes.append( './' )
 	includes.append( 'gui/src/' )
 	includes.append( '3rdparty/install/include' )
+	includes.append( '/usr/include/raptor2' )
 
 	if sys.platform == "darwin":
 		ldflags.append( '-L/opt/local/lib' )
