diff -aur old/help/Makefile.am new/help/Makefile.am
--- old/help/Makefile.am	2011-01-03 17:14:20.000000000 +0100
+++ new/help/Makefile.am	2011-01-13 21:42:06.046672489 +0100
@@ -104,7 +104,7 @@
 else
 stardict-install-html:
 	for lc in $(STARDICT_ALL_LINGUAS); do \
-		install_dir=$(STARDICT_HTML_INSTALL_DIR)/$${lc}; \
+		install_dir=$(DESTDIR)$(STARDICT_HTML_INSTALL_DIR)/$${lc}; \
 		htmldir=$(srcdir)/$${lc}/html; \
 		for file in `find $${htmldir} -type f -print`; do \
 			relfile=$${file#$${htmldir}/}; \
@@ -126,7 +126,7 @@
 else
 stardict-uninstall-html:
 	for lc in $(STARDICT_ALL_LINGUAS); do \
-		install_dir=$(STARDICT_HTML_INSTALL_DIR)/$${lc}; \
+		install_dir=$(DESTDIR)$(STARDICT_HTML_INSTALL_DIR)/$${lc}; \
 		rm -rf $${install_dir}; \
 	done;
 endif
