diff -up texinfo-6.0/info/Makefile.in.orig texinfo-6.0/info/Makefile.in
--- texinfo-6.0/info/Makefile.in.orig	2015-07-14 13:17:44.298882856 +0200
+++ texinfo-6.0/info/Makefile.in	2015-07-14 13:18:13.082004108 +0200
@@ -1695,7 +1695,6 @@ TESTS = \
 	t/end-of-line.sh \
 	t/goal-column.sh \
 	t/star-note-non-whitespace.sh \
-	t/c-u-m-x-scroll-forward.sh \
 	t/last-no-history.sh \
 	t/split-footnotes.sh \
 	t/window-split-dir.sh \
