diff -up SuiteSparse/SuiteSparse_config/SuiteSparse_config.c.math SuiteSparse/SuiteSparse_config/SuiteSparse_config.c
--- SuiteSparse/SuiteSparse_config/SuiteSparse_config.c.math	2014-03-21 13:15:22.000000000 -0600
+++ SuiteSparse/SuiteSparse_config/SuiteSparse_config.c	2014-09-15 16:53:50.243297000 -0600
@@ -9,6 +9,7 @@
  * Author: Timothy A. Davis.
  */
 
+#include <math.h>
 #include <stdlib.h>
 
 #ifndef NPRINT
diff -up SuiteSparse/SuiteSparse_config/SuiteSparse_config.h.math SuiteSparse/SuiteSparse_config/SuiteSparse_config.h
--- SuiteSparse/SuiteSparse_config/SuiteSparse_config.h.math	2014-07-16 15:35:20.000000000 -0600
+++ SuiteSparse/SuiteSparse_config/SuiteSparse_config.h	2014-09-15 16:23:31.802267503 -0600
@@ -44,7 +44,6 @@ extern "C" {
 
 #include <limits.h>
 #include <stdlib.h>
-#include <math.h>
 
 /* ========================================================================== */
 /* === SuiteSparse_long ===================================================== */
