diff -aur gnome-media-2.32.0/gnome-volume-control/src/gvc-channel-bar.c gnome-media-2.32.0.new/gnome-volume-control/src/gvc-channel-bar.c
--- gnome-media-2.32.0/gnome-volume-control/src/gvc-channel-bar.c	2010-09-28 15:46:15.000000000 +0100
+++ gnome-media-2.32.0.new/gnome-volume-control/src/gvc-channel-bar.c	2010-12-31 15:14:23.964994328 +0000
@@ -449,7 +449,7 @@
                 /* Switch side scroll to vertical */
                 if (direction == GDK_SCROLL_RIGHT)
                         direction = GDK_SCROLL_UP;
-                else if (GDK_SCROLL_LEFT)
+                else if (direction == GDK_SCROLL_LEFT)
                         direction = GDK_SCROLL_DOWN;
         }
 
