--- Linux-PAM-1.1.1/modules/pam_userdb/pam_userdb.c.ark	2010-04-15 13:41:11.521290322 +0200
+++ Linux-PAM-1.1.1/modules/pam_userdb/pam_userdb.c	2010-04-15 13:41:35.923290703 +0200
@@ -29,7 +29,8 @@
 # include <ndbm.h>
 #else
 # ifdef HAVE_DB_H
-#  define DB_DBM_HSEARCH    1 /* use the dbm interface */
+#  define DB_DBM_HSEARCH    1 /* use the dbm interface for db <= 4.8 */
+#  define HAVE_DBM 1 /* use the dbm interface for db >= 5.0 */
 #  include <db.h>
 # else
 #  error "failed to find a libdb or equivalent"
