diff --git a/gst/fsrtpconference/fs-rtp-discover-codecs.c b/gst/fsrtpconference/fs-rtp-discover-codecs.c
index d128e84..3521580 100644
--- a/gst/fsrtpconference/fs-rtp-discover-codecs.c
+++ b/gst/fsrtpconference/fs-rtp-discover-codecs.c
@@ -1110,11 +1110,8 @@ compare_media_caps (gconstpointer a, gconstpointer b)
 }
 
 static gint
-compare_rtp_caps (gconstpointer a, gconstpointer b)
+compare_rtp_caps (CodecCap *element, GstCaps *c_caps)
 {
-  CodecCap *element = (CodecCap *)a;
-  GstCaps *c_caps = (GstCaps *)b;
-
   return !gst_caps_can_intersect (element->rtp_caps, c_caps);
 }
 
@@ -1226,7 +1223,7 @@ create_codec_cap_list (GstElementFactory *factory,
         if (rtp_caps) {
           if (entry->rtp_caps) {
             GstCaps *new_rtp_caps;
-            new_rtp_caps = gst_caps_union (rtp_caps, entry->rtp_caps);
+            new_rtp_caps = gst_caps_intersect (rtp_caps, entry->rtp_caps);
             gst_caps_unref (entry->rtp_caps);
             entry->rtp_caps = new_rtp_caps;
           } else {
