--- Makefile.in.orig	2006-01-23 08:51:14.000000000 +0000
+++ Makefile.in	2006-01-23 08:51:29.000000000 +0000
@@ -582,7 +582,7 @@
 
 info-am:
 
-install-data-am: install-extra_langDATA install-pkgconfigDATA
+install-data-am: install-pkgconfigDATA
 
 install-exec-am:
 
@@ -610,7 +610,7 @@
 
 ps-am:
 
-uninstall-am: uninstall-extra_langDATA uninstall-info-am \
+uninstall-am: uninstall-info-am \
 	uninstall-pkgconfigDATA
 
 uninstall-info: uninstall-info-recursive
