--- transcode-1.1.5-old/export/aud_aux.c	2010-07-04 12:44:31.867550180 -0500
+++ transcode-1.1.5/export/aud_aux.c	2010-07-04 13:41:01.253292529 -0500
@@ -347,6 +347,7 @@
     //-- set parameters (bitrate, channels and sample-rate) --
     //--------------------------------------------------------
     memset(&mpa_ctx, 0, sizeof(mpa_ctx));          // default all
+    mpa_ctx.codec_type  = CODEC_TYPE_AUDIO;
     mpa_ctx.bit_rate    = vob->mp3bitrate * 1000;  // bitrate dest.
     mpa_ctx.channels    = vob->dm_chan;            // channels
     mpa_ctx.sample_rate = vob->a_rate;
