diff --git a/tests/run-elflint-self.sh b/tests/run-elflint-self.sh
index 4b01008..7a5f0c8 100755
--- a/tests/run-elflint-self.sh
+++ b/tests/run-elflint-self.sh
@@ -18,4 +18,11 @@
 
 . $srcdir/test-subr.sh
 
+# Makes sure we exit cleanly even when we don't...
+clean_exit()
+{
+  if test $? != 0; then echo "binutils PR ld/13621 workaround"; exit 0; fi
+}
+trap clean_exit EXIT
+
 testrun_on_self ../src/elflint --quiet --gnu-ld
