Help parallel compilation
patch by Sebastien Fabbro 
--- scripts/Makefile.am.orig	2011-06-09 22:27:22.688268651 +0100
+++ scripts/Makefile.am	2011-02-08 10:00:51.000000000 +0000
@@ -316,7 +316,7 @@
 
 gethelp_SOURCES = gethelp.cc
 
-.DOCSTRINGS: $(FCN_FILES) $(GEN_FCN_FILES) mkdoc $(gethelp_SOURCES) Makefile
+.DOCSTRINGS: $(FCN_FILES) $(GEN_FCN_FILES) mkdoc $(gethelp_SOURCES) Makefile gethelp
 	@$(MAKE) $(AM_MAKEFLAGS) gethelp$(BUILD_EXEEXT)
 	if [ "x$(srcdir)" != "x." ] && [ -f $(srcdir)/DOCSTRINGS ] && [ ! -f DOCSTRINGS ]; then \
 		cp $(srcdir)/DOCSTRINGS DOCSTRINGS; \
