diff -up octave-3.4.0/liboctave/oct-alloc.h.gcc46 octave-3.4.0/liboctave/oct-alloc.h
--- octave-3.4.0/liboctave/oct-alloc.h.gcc46	2011-02-08 03:00:51.000000000 -0700
+++ octave-3.4.0/liboctave/oct-alloc.h	2011-02-08 09:41:23.984081687 -0700
@@ -23,6 +23,8 @@ along with Octave; see the file COPYING.
 #if !defined (octave_oct_alloc_h)
 #define octave_oct_alloc_h 1
 
+#include <cstddef>
+
 class
 OCTAVE_API
 octave_allocator
diff -up octave-3.4.0/src/pr-output.cc.gcc46 octave-3.4.0/src/pr-output.cc
--- octave-3.4.0/src/pr-output.cc.gcc46	2011-02-08 03:00:52.000000000 -0700
+++ octave-3.4.0/src/pr-output.cc	2011-02-08 09:55:16.149662744 -0700
@@ -3024,7 +3024,7 @@ abs (T x)
 }
 
 #define INSTANTIATE_ABS(T) \
-  template /* static */ inline T abs (T)
+  template /* static */ T abs (T)
 
 INSTANTIATE_ABS(signed char);
 INSTANTIATE_ABS(short);
