diff --git a/knotify/config/knotifyconfigactionswidget.cpp b/knotify/config/knotifyconfigactionswidget.cpp
index 88d18b6..0e76658 100644
--- a/knotify/config/knotifyconfigactionswidget.cpp
+++ b/knotify/config/knotifyconfigactionswidget.cpp
@@ -128,7 +128,7 @@ void KNotifyConfigActionsWidget::slotPlay(  )
 	KUrl soundURL = m_ui.Sound_select->url();
 	if ( soundURL.isRelative() )
 	{
-		QString soundString = soundURL.toLocalFile();
+		QString soundString = m_ui.Sound_select->text();
 		// we need a way to get the application name in order to ba able to do this :
 		/*QString search = QString("%1/sounds/%2").arg(config->appname).arg(soundFile);
 		search = KGlobal::mainComponent().dirs()->findResource("data", search);
