--- kdelibs/kdecore/localization/kencodingdetector.cpp	2009/03/11 16:32:16	938347
+++ kdelibs/kdecore/localization/kencodingdetector.cpp	2009/03/11 16:37:50	938348
@@ -1052,10 +1052,8 @@
     }
 
     if (len<20)
-    {
-        setEncoding("",DefaultEncoding);
         return false;
-    }
+
 #ifdef DECODE_DEBUG
     kDebug( 6005 ) << "KEncodingDetector: using heuristics (" << strlen(data) << ")";
 #endif
@@ -1110,10 +1108,9 @@
             // huh. somethings broken in this code ### FIXME
             //enc = 0; //Reset invalid codec we tried, so we get back to latin1 fallback.
             break;
-        }
+    }
 
-        setEncoding("",DefaultEncoding);
-        return true;
+    return true;
 }
 
 
