--- kdecore/kiconloader.cpp.old	2005-09-10 20:32:02.000000000 +0000
+++ kdecore/kiconloader.cpp	2005-09-10 20:34:21.000000000 +0000
@@ -263,6 +263,6 @@
     dirs += d->mpDirs->resourceDirs("icon");
     dirs += d->mpDirs->resourceDirs("pixmap");
     dirs += d->mpDirs->resourceDirs("xdgdata-icon");
-    dirs += "/usr/share/pixmaps";
+    dirs += d->mpDirs->resourceDirs("xdgdata-pixmap");
     for (QStringList::ConstIterator it = dirs.begin(); it != dirs.end(); ++it)
 	d->mpDirs->addResourceDir("appicon", *it);
 
--- kdecore/kstandarddirs.cpp.old	2005-09-10 20:42:16.000000000 +0000
+++ kdecore/kstandarddirs.cpp	2005-09-10 20:42:21.000000000 +0000
@@ -98,9 +98,9 @@
 			      "data", "locale", "services", "mime",
 			      "servicetypes", "config", "exe",
 			      "wallpaper", "lib", "pixmap", "templates",
-			      "module", "qtplugins",
-			      "xdgdata-apps", "xdgdata-dirs", "xdgconf-menu",
+			      "module", "qtplugins", "xdgdata-icon",
+			      "xdgdata-pixmap", "xdgdata-apps", 
+			      "xdgdata-dirs", "xdgconf-menu",
-			      "xdgdata-icon",
 			      "kcfg", "emoticons", 0 };
 
 static int tokenize( QStringList& token, const QString& str,
@@ -1055,6 +1056,8 @@
         return "applications/";
     if (!strcmp(type, "xdgdata-icon"))
         return "icons/";
+    if (!strcmp(type, "xdgdata-pixmap"))
+	return "pixmaps/";
     if (!strcmp(type, "xdgdata-dirs"))
         return "desktop-directories/";
     if (!strcmp(type, "xdgconf-menu"))
