--- zlib-1.2.3.4/Makefile.in	2009-12-24 16:09:43.000000000 +0100
+++ zlib-1.2.3.4/Makefile.in	2009-12-25 17:19:13.594735916 +0100
@@ -34,7 +34,7 @@
 SHAREDLIB=libz.so
 SHAREDLIBV=libz.so.1.2.3.4
 SHAREDLIBM=libz.so.1
-LIBS=$(STATICLIB) $(SHAREDLIB)
+LIBS=$(STATICLIB) $(SHAREDLIB) $(SHAREDLIBV)
 
 AR=ar rc
 RANLIB=ranlib
@@ -179,6 +179,7 @@
 	chmod 644 $(DESTDIR)$(man3dir)/zlib.3
 	cp zlib.pc $(DESTDIR)$(pkgconfigdir)
 	chmod 644 $(DESTDIR)$(pkgconfigdir)/zlib.pc
+	chmod 644 $(DESTDIR)$(libdir)/libz.a
 # The ranlib in install is needed on NeXTSTEP which checks file times
 # ldconfig is for Linux
 
