diff -ur audacious-plugins-1.5.1.orig/src/amidi-plug/i_configure.c audacious-plugins-1.5.1/src/amidi-plug/i_configure.c
--- audacious-plugins-1.5.1.orig/src/amidi-plug/i_configure.c	2008-05-24 00:44:19.000000000 +0200
+++ audacious-plugins-1.5.1/src/amidi-plug/i_configure.c	2008-08-18 12:06:08.000000000 +0200
@@ -29,6 +29,7 @@
 #include "i_configure-dummy.h"
 #include "i_utils.h"
 #include <audacious/auddrct.h>
+#include <audacious/plugin.h>
 
 
 amidiplug_cfg_backend_t * amidiplug_cfg_backend;
