diff --git i/docs/reference/libbrasero-burn/libbrasero-burn-docs.sgml w/docs/reference/libbrasero-burn/libbrasero-burn-docs.sgml
index 612491f1..873a5497 100644
--- i/docs/reference/libbrasero-burn/libbrasero-burn-docs.sgml
+++ w/docs/reference/libbrasero-burn/libbrasero-burn-docs.sgml
@@ -39,10 +39,6 @@
     <xi:include href="xml/brasero-blank-dialog.xml"/>
 
 
-  </chapter>
-  <chapter id="object-tree">
-    <title>Object Hierarchy</title>
-     <xi:include href="xml/tree_index.sgml"/>
   </chapter>
   <index id="api-index-full">
     <title>API Index</title>
diff --git i/docs/reference/libbrasero-media/libbrasero-media-docs.sgml w/docs/reference/libbrasero-media/libbrasero-media-docs.sgml
index c7ec48c0..a89de72b 100644
--- i/docs/reference/libbrasero-media/libbrasero-media-docs.sgml
+++ w/docs/reference/libbrasero-media/libbrasero-media-docs.sgml
@@ -24,10 +24,6 @@
     <xi:include href="xml/brasero-drive-selection.xml"/>
 
   </chapter>
-  <chapter id="object-tree">
-    <title>Object Hierarchy</title>
-     <xi:include href="xml/tree_index.sgml"/>
-  </chapter>
   <index id="api-index-full">
     <title>API Index</title>
     <xi:include href="xml/api-index-full.xml"><xi:fallback /></xi:include>
