commit c90b87e021709be78853354bd7776acfdd1c95d9
Author: Wayne Davison <wayned@samba.org>
Date:   Fri Jan 4 21:43:50 2019 -0800

    Avoid a failed test if dirs report 1 hlink (e.g. WSL weirdness).

diff --git a/testsuite/xattrs.test b/testsuite/xattrs.test
index f7d9a6df..86165e0c 100644
--- a/testsuite/xattrs.test
+++ b/testsuite/xattrs.test
@@ -163,7 +163,7 @@ checkit "$RSYNC -aiX $XFILT $dashH --fake-super --link-dest=../chk . ../to" "$ch
 cd "$todir"
 xls $dirs $files | diff $diffopt "$scratchdir/xattrs.txt" -
 
-sed -n -e '/^[^ ][^ ]*  *[^ ][^ ]*  *[^ ][^ ]*  *1 /p' "$scratchdir/ls-to" >"$scratchdir/ls-diff-all"
+sed -n -e '/^[^d ][^ ]*  *[^ ][^ ]*  *[^ ][^ ]*  *1 /p' "$scratchdir/ls-to" >"$scratchdir/ls-diff-all"
 fgrep -v './file1' "$scratchdir/ls-diff-all" >"$scratchdir/ls-diff" || :
 if [ -s "$scratchdir/ls-diff" ]; then
     echo "Missing hard links on:"
