From c30f02063e548c20d1021ec42cb958a7ef0c25af Mon Sep 17 00:00:00 2001
From: Simo Sorce <simo@redhat.com>
Date: Wed, 7 Jan 2015 15:59:53 -0500
Subject: [PATCH] Fix error in compiling without SELinux

Fixes: #131

Signed-off-by: Simo Sorce <simo@redhat.com>
---
 proxy/src/gp_selinux.h | 19 +++++++++----------
 1 file changed, 9 insertions(+), 10 deletions(-)

diff --git a/proxy/src/gp_selinux.h b/proxy/src/gp_selinux.h
index 693a12421823e0e6808cf19b8835b68411c106c7..f53da5a7bc1c986278722c9da0a680cbec8b2ef4 100644
--- a/src/gp_selinux.h
+++ b/src/gp_selinux.h
@@ -48,9 +48,9 @@
 #define SELINUX_CTX void *
 #define SEC_CTX void *
 
-void *SELINUX_context_new(const char *str) { return NULL; }
-#define SELINUX_context_free(x) (x) = NULL;
-const char *SELINUX_context_dummy_get(void *) { return NULL; }
+#define SELINUX_context_new(x) NULL
+#define SELINUX_context_free(x) (x) = NULL
+#define SELINUX_context_dummy_get(x) "<SELinux not compiled in>"
 #define SELINUX_context_str SELINUX_context_dummy_get
 #define SELINUX_context_type_get SELINUX_context_dummy_get
 #define SELINUX_context_user_get SELINUX_context_dummy_get
@@ -58,13 +58,12 @@ const char *SELINUX_context_dummy_get(void *) { return NULL; }
 #define SELINUX_context_range_get SELINUX_context_dummy_get
 
 #include <errno.h>
-int SELINUX_getpeercon(int fd, SEC_CTX *con)
-{
-    *con = NULL;
-    errno = ENOTSUP;
-    return -1;
-}
-#define SELINUX_freecon(x) (x) = NULL;
+#define SELINUX_getpeercon(x, y) -1; do { \
+    *(y) = NULL; \
+    errno = ENOTSUP; \
+} while(0)
+
+#define SELINUX_freecon(x) (x) = NULL
 
 #endif /* done HAVE_SELINUX */
 
-- 
2.1.0


