From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jan Tojnar <jtojnar@gmail.com>
Date: Fri, 1 Oct 2021 23:11:26 +0200
Subject: [PATCH] gir: Fix build with WebkitGTK 4.1

---
 devhelp/meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/devhelp/meson.build b/devhelp/meson.build
index 010e563f8996..0afdbe760b1d 100644
--- a/devhelp/meson.build
+++ b/devhelp/meson.build
@@ -126,7 +126,7 @@ libdevhelp_gir = GNOME.generate_gir(
   header: 'devhelp/devhelp.h',
   identifier_prefix: 'Dh',
   include_directories: ROOT_INCLUDE_DIR,
-  includes: ['Gtk-3.0', 'WebKit2-4.0'],
+  includes: ['Gtk-3.0', 'WebKit2-' + webkit_abi],
   install: true,
   namespace: 'Devhelp',
   nsversion: LIBDEVHELP_API_VERSION_FULL,
