From f6b4103e237cea8c924c307ccc8f1881921d90d0 Mon Sep 17 00:00:00 2001
From: Kevin Bowling <kevin.bowling@kev009.com>
Date: Thu, 8 Jul 2021 21:51:34 -0700
Subject: [PATCH 2/3] pkgconfig: Drop unneeded lines from autotools build

Signed-off-by: Laurent Carlier <lordheavym@gmail.com>
---
 glu.pc.in | 2 --
 1 file changed, 2 deletions(-)

diff --git a/glu.pc.in b/glu.pc.in
index 8bb957a..9f1425c 100644
--- a/glu.pc.in
+++ b/glu.pc.in
@@ -1,5 +1,4 @@
 prefix=@prefix@
-exec_prefix=${prefix}
 libdir=@libdir@
 includedir=@includedir@
 
@@ -8,5 +7,4 @@ Description: Mesa OpenGL Utility library
 Requires: @GLU_REQUIRES@
 Version: @VERSION@
 Libs: -L${libdir} -lGLU
-Libs.private: -lm
 Cflags: -I${includedir}
-- 
2.34.1

