From b93c2581e5f98b4826e7c645d873efa746e269d4 Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <jan.steffens@gmail.com>
Date: Tue, 1 Oct 2019 13:41:52 +0000
Subject: [PATCH 2/3] config: Don't warn when wifi URL unset

---
 src/gclue-config.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/gclue-config.c b/src/gclue-config.c
index df4adcc..f09e897 100644
--- a/src/gclue-config.c
+++ b/src/gclue-config.c
@@ -232,7 +232,7 @@ load_wifi_config (GClueConfig *config)
                                                 "url",
                                                 &error);
         if (error != NULL) {
-                g_warning ("%s", error->message);
+                g_debug ("No wifi URL: %s", error->message);
                 g_clear_error (&error);
                 priv->wifi_url = g_strdup (DEFAULT_WIFI_URL);
         }
-- 
2.23.0

