From db01ee6d453cd2f6bfcbb3420ec1a48b2d37631c Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <jan.steffens@gmail.com>
Date: Mon, 29 Aug 2016 08:43:10 +0200
Subject: [PATCH 2/3] Enable infinality subpixel hinting

---
 include/freetype/config/ftoption.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/freetype/config/ftoption.h b/include/freetype/config/ftoption.h
index 12f47a82e..1d4469a99 100644
--- a/include/freetype/config/ftoption.h
+++ b/include/freetype/config/ftoption.h
@@ -658,8 +658,8 @@ FT_BEGIN_HEADER
    * https://www.microsoft.com/typography/cleartype/truetypecleartype.aspx
    */
 /* #define TT_CONFIG_OPTION_SUBPIXEL_HINTING  1         */
-#define TT_CONFIG_OPTION_SUBPIXEL_HINTING  2
-/* #define TT_CONFIG_OPTION_SUBPIXEL_HINTING  ( 1 | 2 ) */
+/* #define TT_CONFIG_OPTION_SUBPIXEL_HINTING  2         */
+#define TT_CONFIG_OPTION_SUBPIXEL_HINTING  ( 1 | 2 )
 
 
   /**************************************************************************
-- 
2.22.0

