From ef10d555fa4b1c27a0cfd17e00331db3643828b0 Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <jan.steffens@gmail.com>
Date: Tue, 1 Oct 2019 13:42:11 +0000
Subject: [PATCH 3/3] config: Don't crash when submission-url unset

---
 src/gclue-config.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/gclue-config.c b/src/gclue-config.c
index f09e897..eab0b6f 100644
--- a/src/gclue-config.c
+++ b/src/gclue-config.c
@@ -255,7 +255,7 @@ load_wifi_config (GClueConfig *config)
                                                        &error);
         if (error != NULL) {
                 g_debug ("No wifi submission URL: %s", error->message);
-                g_error_free (error);
+                g_clear_error (&error);
                 priv->wifi_submit_url = g_strdup (DEFAULT_WIFI_SUBMIT_URL);
         }
 
-- 
2.23.0

