From 33c40d127877179e0cba2f6595816377bb6bcda1 Mon Sep 17 00:00:00 2001
From: Alec Leamas <leamas.alec@gmail.com>
Date: Tue, 25 Oct 2016 10:28:14 +0200
Subject: [PATCH 05/10] lib: curl_poll.h: Ensure build on unconfiguredclients.

---
 lib/curl_poll.h | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/lib/curl_poll.h b/lib/curl_poll.h
index 6144c42..af25381 100644
--- a/lib/curl_poll.h
+++ b/lib/curl_poll.h
@@ -21,7 +21,9 @@
  * KIND, either express or implied.
  *
  ***************************************************************************/
+#ifdef HAVE_CONFIG_H
 #include "config.h"
+#endif
 
 #ifdef __cplusplus
 extern "C" {
@@ -29,7 +31,7 @@ extern "C" {
 
 #ifdef HAVE_SYS_POLL_H
 #include <sys/poll.h>
-#elif defined(HAVE_POLL_H)
+#else
 #include <poll.h>
 #endif
 
-- 
2.5.5

