--- xserver-xorg-video-intel-2.2.0+git20080107/src/i830_quirks.c.orig	2008-01-31 15:31:33.000000000 -0800
+++ xserver-xorg-video-intel-2.2.0+git20080107/src/i830_quirks.c	2008-01-31 15:32:35.000000000 -0800
@@ -68,6 +68,8 @@ static i830_quirk i830_quirk_list[] = {
     { PCI_CHIP_I915_GM, 0x1028, 0x01a3, quirk_ignore_tv },
     /* Dell XPS 1330 */
     { PCI_CHIP_I965_GM, 0x1028, 0x0209, quirk_ignore_tv },
+    /* Dell Inspiron 1535 */
+    { PCI_CHIP_I965_GM, 0x1028, 0x0254, quirk_ignore_tv },
 
     /* Lenovo X60s has no TV output */
     { PCI_CHIP_I945_GM, 0x17aa, 0x201a, quirk_ignore_tv },
