--- xinit-1.4.2/startx.cpp	2022-12-04 21:29:02.886416839 +0100
+++ xinit-1.4.2/startx.cpp.new	2022-12-04 21:34:36.943389951 +0100
@@ -272,7 +272,7 @@
     dummy=0
 
     XCOMM create a file with auth information for the server. ':0' is a dummy.
-    xserverauthfile=$HOME/.serverauth.$$
+    xserverauthfile=`mktemp -p /tmp serverauth.XXXXXXXXXX`
     trap "rm -f '$xserverauthfile'" HUP INT QUIT ILL TRAP BUS TERM
     xauth -q -f "$xserverauthfile" << EOF
 add :$dummy . $mcookie
