From 7b9eb1528d7092ebe62c0a816cd3d14d4407463b Mon Sep 17 00:00:00 2001
From: Taegil Bae <esrevinu@gmail.com>
Date: Mon, 22 Oct 2018 02:50:05 +0000
Subject: [PATCH 1/2] Set RPATH for libnm_device_plugin_bluetooth.so

---
 src/devices/bluetooth/meson.build | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/devices/bluetooth/meson.build b/src/devices/bluetooth/meson.build
index 019f04f6b..628a3bc80 100644
--- a/src/devices/bluetooth/meson.build
+++ b/src/devices/bluetooth/meson.build
@@ -26,7 +26,8 @@ libnm_device_plugin_bluetooth = shared_module(
   link_args: ldflags_linker_script_devices,
   link_depends: linker_script_devices,
   install: true,
-  install_dir: nm_plugindir
+  install_dir: nm_plugindir,
+  install_rpath: nm_plugindir
 )
 
 core_plugins += libnm_device_plugin_bluetooth
-- 
2.18.1


From f7c5745d81205ce3675bee55b0ab1e814a63c817 Mon Sep 17 00:00:00 2001
From: Taegil Bae <esrevinu@gmail.com>
Date: Mon, 22 Oct 2018 02:52:06 +0000
Subject: [PATCH 2/2] Set RPATH for libnm_device_plugin_wwan.so

---
 src/devices/wwan/meson.build | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/devices/wwan/meson.build b/src/devices/wwan/meson.build
index 2c0ea5a10..5fe6e433e 100644
--- a/src/devices/wwan/meson.build
+++ b/src/devices/wwan/meson.build
@@ -54,7 +54,8 @@ libnm_device_plugin_wwan = shared_module(
   link_args: ldflags_linker_script_devices,
   link_depends: linker_script_devices,
   install: true,
-  install_dir: nm_plugindir
+  install_dir: nm_plugindir,
+  install_rpath: nm_plugindir
 )
 
 core_plugins += libnm_device_plugin_wwan
-- 
2.18.1

