diff -ruN mesa-c490349825e2c5a11a5f598fb52a6f92597f694e/docs/relnotes-7.4.2.html mesa-656cd707db3699079d3a1b886d82e6f04afb1065/docs/relnotes-7.4.2.html
--- mesa-c490349825e2c5a11a5f598fb52a6f92597f694e/docs/relnotes-7.4.2.html	1970-01-01 01:00:00.000000000 +0100
+++ mesa-656cd707db3699079d3a1b886d82e6f04afb1065/docs/relnotes-7.4.2.html	2009-04-28 18:58:13.000000000 +0200
@@ -0,0 +1,62 @@
+<HTML>
+
+<TITLE>Mesa Release Notes</TITLE>
+
+<head><link rel="stylesheet" type="text/css" href="mesa.css"></head>
+
+<BODY>
+
+<body bgcolor="#eeeeee">
+
+<H1>Mesa 7.4.2 Release Notes / TBD</H1>
+
+<p>
+Mesa 7.4.2 is a stable development release fixing bugs since the 7.4.1 release.
+</p>
+<p>
+Mesa 7.4.2 implements the OpenGL 2.1 API, but the version reported by
+glGetString(GL_VERSION) depends on the particular driver being used.
+Some drivers don't support all the features required in OpenGL 2.1.
+</p>
+<p>
+See the <a href="install.html">Compiling/Installing page</a> for prerequisites
+for DRI ardware acceleration.
+</p>
+
+
+<h2>MD5 checksums</h2>
+<pre>
+tbd
+</pre>
+
+
+<h2>Bug fixes</h2>
+<ul>
+<li>Fixed segfault when rendering to front buffer with DRI 1.
+<li>Fixed swrast texture rectangle bug when wrap mode = GL_CLAMP_TO_BORDER and
+    filter mode = GL_LINEAR.  (bug 21461)
+</ul>
+
+
+
+<h2>Driver Status</h2>
+
+<pre>
+Driver			Status
+----------------------	----------------------
+DRI drivers		varies with the driver
+XMesa/GLX (on Xlib)	implements OpenGL 2.1
+OSMesa (off-screen)	implements OpenGL 2.1
+Windows/Win32		implements OpenGL 2.1
+Glide (3dfx Voodoo1/2)	implements OpenGL 1.3
+SVGA			unsupported
+Wind River UGL		unsupported
+DJGPP			unsupported
+GGI			unsupported
+BeOS			unsupported
+Allegro			unsupported
+D3D			unsupported
+</pre>
+
+</body>
+</html>
diff -ruN mesa-c490349825e2c5a11a5f598fb52a6f92597f694e/docs/relnotes.html mesa-656cd707db3699079d3a1b886d82e6f04afb1065/docs/relnotes.html
--- mesa-c490349825e2c5a11a5f598fb52a6f92597f694e/docs/relnotes.html	2009-04-18 19:12:33.000000000 +0200
+++ mesa-656cd707db3699079d3a1b886d82e6f04afb1065/docs/relnotes.html	2009-04-28 18:58:13.000000000 +0200
@@ -20,6 +20,7 @@
 </p>
 
 <UL>
+<LI><A HREF="relnotes-7.4.2.html">7.4.2 release notes</A>
 <LI><A HREF="relnotes-7.4.1.html">7.4.1 release notes</A>
 <LI><A HREF="relnotes-7.4.html">7.4 release notes</A>
 <LI><A HREF="relnotes-7.3.html">7.3 release notes</A>
diff -ruN mesa-c490349825e2c5a11a5f598fb52a6f92597f694e/progs/xdemos/glxcontexts.c mesa-656cd707db3699079d3a1b886d82e6f04afb1065/progs/xdemos/glxcontexts.c
--- mesa-c490349825e2c5a11a5f598fb52a6f92597f694e/progs/xdemos/glxcontexts.c	2009-04-18 19:12:33.000000000 +0200
+++ mesa-656cd707db3699079d3a1b886d82e6f04afb1065/progs/xdemos/glxcontexts.c	2009-04-28 18:58:13.000000000 +0200
@@ -385,6 +385,10 @@
    } else
       do_draw();
 
+   glDeleteLists(gear1, 1);
+   glDeleteLists(gear2, 1);
+   glDeleteLists(gear3, 1);
+
    glXSwapBuffers(dpy, win);
    glXDestroyContext(dpy, ctx);
 }
diff -ruN mesa-c490349825e2c5a11a5f598fb52a6f92597f694e/src/mesa/drivers/dri/i915/intel_context.c mesa-656cd707db3699079d3a1b886d82e6f04afb1065/src/mesa/drivers/dri/i915/intel_context.c
--- mesa-c490349825e2c5a11a5f598fb52a6f92597f694e/src/mesa/drivers/dri/i915/intel_context.c	2009-04-18 19:12:33.000000000 +0200
+++ mesa-656cd707db3699079d3a1b886d82e6f04afb1065/src/mesa/drivers/dri/i915/intel_context.c	2009-04-28 18:58:13.000000000 +0200
@@ -534,7 +534,8 @@
    if ((ctx->DrawBuffer->Name == 0) && intel->front_buffer_dirty) {
       __DRIscreen *const screen = intel->intelScreen->driScrnPriv;
 
-      if ((screen->dri2.loader->base.version >= 2)
+      if (screen->dri2.loader
+          && (screen->dri2.loader->base.version >= 2)
 	  && (screen->dri2.loader->flushFrontBuffer != NULL)) {
 	 (*screen->dri2.loader->flushFrontBuffer)(intel->driDrawable,
 						  intel->driDrawable->loaderPrivate);
diff -ruN mesa-c490349825e2c5a11a5f598fb52a6f92597f694e/src/mesa/drivers/dri/i915/intel_pixel.c mesa-656cd707db3699079d3a1b886d82e6f04afb1065/src/mesa/drivers/dri/i915/intel_pixel.c
--- mesa-c490349825e2c5a11a5f598fb52a6f92597f694e/src/mesa/drivers/dri/i915/intel_pixel.c	2009-04-18 19:12:33.000000000 +0200
+++ mesa-656cd707db3699079d3a1b886d82e6f04afb1065/src/mesa/drivers/dri/i915/intel_pixel.c	2009-04-28 18:58:13.000000000 +0200
@@ -183,7 +183,7 @@
    intel->meta.saved_vp_height = ctx->Viewport.Height;
    intel->meta.saved_matrix_mode = ctx->Transform.MatrixMode;
 
-   /*   _mesa_Viewport(0, 0, ctx->DrawBuffer->Width, ctx->DrawBuffer->Height);*/
+   _mesa_Viewport(0, 0, ctx->DrawBuffer->Width, ctx->DrawBuffer->Height);
 
    _mesa_MatrixMode(GL_PROJECTION);
    _mesa_PushMatrix();
@@ -205,8 +205,8 @@
 
    _mesa_MatrixMode(intel->meta.saved_matrix_mode);
 
-   /*   _mesa_Viewport(intel->meta.saved_vp_x, intel->meta.saved_vp_y,
-	intel->meta.saved_vp_width, intel->meta.saved_vp_height);*/
+   _mesa_Viewport(intel->meta.saved_vp_x, intel->meta.saved_vp_y,
+		  intel->meta.saved_vp_width, intel->meta.saved_vp_height);
 }
 
 /**
diff -ruN mesa-c490349825e2c5a11a5f598fb52a6f92597f694e/src/mesa/drivers/dri/i965/brw_wm_surface_state.c mesa-656cd707db3699079d3a1b886d82e6f04afb1065/src/mesa/drivers/dri/i965/brw_wm_surface_state.c
--- mesa-c490349825e2c5a11a5f598fb52a6f92597f694e/src/mesa/drivers/dri/i965/brw_wm_surface_state.c	2009-04-18 19:12:33.000000000 +0200
+++ mesa-656cd707db3699079d3a1b886d82e6f04afb1065/src/mesa/drivers/dri/i965/brw_wm_surface_state.c	2009-04-28 18:58:13.000000000 +0200
@@ -310,6 +310,7 @@
       GLubyte color_mask[4];
       GLboolean color_blend;
       uint32_t tiling;
+      uint32_t draw_offset;
    } key;
 
    memset(&key, 0, sizeof(key));
@@ -326,6 +327,7 @@
       key.width = region->pitch; /* XXX: not really! */
       key.height = region->height;
       key.cpp = region->cpp;
+      key.draw_offset = region->draw_offset; /* cur 3d or cube face offset */
    } else {
       key.surface_type = BRW_SURFACE_NULL;
       key.surface_format = BRW_SURFACEFORMAT_B8G8R8A8_UNORM;
@@ -333,6 +335,7 @@
       key.width = 1;
       key.height = 1;
       key.cpp = 4;
+      key.draw_offset = 0;
    }
    memcpy(key.color_mask, ctx->Color.ColorMask,
 	  sizeof(key.color_mask));
@@ -354,8 +357,9 @@
 
       surf.ss0.surface_format = key.surface_format;
       surf.ss0.surface_type = key.surface_type;
+      surf.ss1.base_addr =  key.draw_offset;
       if (region_bo != NULL)
-	 surf.ss1.base_addr = region_bo->offset; /* reloc */
+	 surf.ss1.base_addr += region_bo->offset; /* reloc */
 
       surf.ss2.width = key.width - 1;
       surf.ss2.height = key.height - 1;
@@ -380,12 +384,12 @@
 	  * them both.  We might be able to figure out from other state
 	  * a more restrictive relocation to emit.
 	  */
-	 dri_bo_emit_reloc(brw->wm.surf_bo[unit],
-			   I915_GEM_DOMAIN_RENDER,
-			   I915_GEM_DOMAIN_RENDER,
-			   0,
-			   offsetof(struct brw_surface_state, ss1),
-			   region_bo);
+	 drm_intel_bo_emit_reloc(brw->wm.surf_bo[unit],
+				 offsetof(struct brw_surface_state, ss1),
+				 region_bo,
+				 key.draw_offset,
+				 I915_GEM_DOMAIN_RENDER,
+				 I915_GEM_DOMAIN_RENDER);
       }
    }
 }
diff -ruN mesa-c490349825e2c5a11a5f598fb52a6f92597f694e/src/mesa/drivers/dri/i965/intel_context.c mesa-656cd707db3699079d3a1b886d82e6f04afb1065/src/mesa/drivers/dri/i965/intel_context.c
--- mesa-c490349825e2c5a11a5f598fb52a6f92597f694e/src/mesa/drivers/dri/i965/intel_context.c	2009-04-18 19:12:33.000000000 +0200
+++ mesa-656cd707db3699079d3a1b886d82e6f04afb1065/src/mesa/drivers/dri/i965/intel_context.c	2009-04-28 18:58:13.000000000 +0200
@@ -534,7 +534,8 @@
    if ((ctx->DrawBuffer->Name == 0) && intel->front_buffer_dirty) {
       __DRIscreen *const screen = intel->intelScreen->driScrnPriv;
 
-      if ((screen->dri2.loader->base.version >= 2)
+      if (screen->dri2.loader
+          && (screen->dri2.loader->base.version >= 2)
 	  && (screen->dri2.loader->flushFrontBuffer != NULL)) {
 	 (*screen->dri2.loader->flushFrontBuffer)(intel->driDrawable,
 						  intel->driDrawable->loaderPrivate);
diff -ruN mesa-c490349825e2c5a11a5f598fb52a6f92597f694e/src/mesa/drivers/dri/i965/intel_pixel.c mesa-656cd707db3699079d3a1b886d82e6f04afb1065/src/mesa/drivers/dri/i965/intel_pixel.c
--- mesa-c490349825e2c5a11a5f598fb52a6f92597f694e/src/mesa/drivers/dri/i965/intel_pixel.c	2009-04-18 19:12:33.000000000 +0200
+++ mesa-656cd707db3699079d3a1b886d82e6f04afb1065/src/mesa/drivers/dri/i965/intel_pixel.c	2009-04-28 18:58:13.000000000 +0200
@@ -183,7 +183,7 @@
    intel->meta.saved_vp_height = ctx->Viewport.Height;
    intel->meta.saved_matrix_mode = ctx->Transform.MatrixMode;
 
-   /*   _mesa_Viewport(0, 0, ctx->DrawBuffer->Width, ctx->DrawBuffer->Height);*/
+   _mesa_Viewport(0, 0, ctx->DrawBuffer->Width, ctx->DrawBuffer->Height);
 
    _mesa_MatrixMode(GL_PROJECTION);
    _mesa_PushMatrix();
@@ -205,8 +205,8 @@
 
    _mesa_MatrixMode(intel->meta.saved_matrix_mode);
 
-   /*   _mesa_Viewport(intel->meta.saved_vp_x, intel->meta.saved_vp_y,
-	intel->meta.saved_vp_width, intel->meta.saved_vp_height);*/
+   _mesa_Viewport(intel->meta.saved_vp_x, intel->meta.saved_vp_y,
+		  intel->meta.saved_vp_width, intel->meta.saved_vp_height);
 }
 
 /**
diff -ruN mesa-c490349825e2c5a11a5f598fb52a6f92597f694e/src/mesa/drivers/dri/intel/intel_context.c mesa-656cd707db3699079d3a1b886d82e6f04afb1065/src/mesa/drivers/dri/intel/intel_context.c
--- mesa-c490349825e2c5a11a5f598fb52a6f92597f694e/src/mesa/drivers/dri/intel/intel_context.c	2009-04-18 19:12:33.000000000 +0200
+++ mesa-656cd707db3699079d3a1b886d82e6f04afb1065/src/mesa/drivers/dri/intel/intel_context.c	2009-04-28 18:58:13.000000000 +0200
@@ -534,7 +534,8 @@
    if ((ctx->DrawBuffer->Name == 0) && intel->front_buffer_dirty) {
       __DRIscreen *const screen = intel->intelScreen->driScrnPriv;
 
-      if ((screen->dri2.loader->base.version >= 2)
+      if (screen->dri2.loader
+          && (screen->dri2.loader->base.version >= 2)
 	  && (screen->dri2.loader->flushFrontBuffer != NULL)) {
 	 (*screen->dri2.loader->flushFrontBuffer)(intel->driDrawable,
 						  intel->driDrawable->loaderPrivate);
diff -ruN mesa-c490349825e2c5a11a5f598fb52a6f92597f694e/src/mesa/drivers/dri/intel/intel_pixel.c mesa-656cd707db3699079d3a1b886d82e6f04afb1065/src/mesa/drivers/dri/intel/intel_pixel.c
--- mesa-c490349825e2c5a11a5f598fb52a6f92597f694e/src/mesa/drivers/dri/intel/intel_pixel.c	2009-04-18 19:12:33.000000000 +0200
+++ mesa-656cd707db3699079d3a1b886d82e6f04afb1065/src/mesa/drivers/dri/intel/intel_pixel.c	2009-04-28 18:58:13.000000000 +0200
@@ -183,7 +183,7 @@
    intel->meta.saved_vp_height = ctx->Viewport.Height;
    intel->meta.saved_matrix_mode = ctx->Transform.MatrixMode;
 
-   /*   _mesa_Viewport(0, 0, ctx->DrawBuffer->Width, ctx->DrawBuffer->Height);*/
+   _mesa_Viewport(0, 0, ctx->DrawBuffer->Width, ctx->DrawBuffer->Height);
 
    _mesa_MatrixMode(GL_PROJECTION);
    _mesa_PushMatrix();
@@ -205,8 +205,8 @@
 
    _mesa_MatrixMode(intel->meta.saved_matrix_mode);
 
-   /*   _mesa_Viewport(intel->meta.saved_vp_x, intel->meta.saved_vp_y,
-	intel->meta.saved_vp_width, intel->meta.saved_vp_height);*/
+   _mesa_Viewport(intel->meta.saved_vp_x, intel->meta.saved_vp_y,
+		  intel->meta.saved_vp_width, intel->meta.saved_vp_height);
 }
 
 /**
diff -ruN mesa-c490349825e2c5a11a5f598fb52a6f92597f694e/src/mesa/drivers/dri/radeon/radeon_screen.c mesa-656cd707db3699079d3a1b886d82e6f04afb1065/src/mesa/drivers/dri/radeon/radeon_screen.c
--- mesa-c490349825e2c5a11a5f598fb52a6f92597f694e/src/mesa/drivers/dri/radeon/radeon_screen.c	2009-04-18 19:12:33.000000000 +0200
+++ mesa-656cd707db3699079d3a1b886d82e6f04afb1065/src/mesa/drivers/dri/radeon/radeon_screen.c	2009-04-28 18:58:13.000000000 +0200
@@ -554,11 +554,8 @@
       screen->chip_family = CHIP_FAMILY_RS300;
       break;
 
-      /* 9500 with 1 pipe verified by: Reid Linnemann <lreid@cs.okstate.edu> */
+
    case PCI_CHIP_R300_AD:
-      screen->chip_family = CHIP_FAMILY_RV350;
-      screen->chip_flags = RADEON_CHIPSET_TCL;
-      break;
    case PCI_CHIP_R300_AE:
    case PCI_CHIP_R300_AF:
    case PCI_CHIP_R300_AG:
@@ -886,6 +883,18 @@
        } else {
 	   screen->num_gb_pipes = temp;
        }
+
+       /* pipe overrides */
+       switch (dri_priv->deviceID) {
+       case PCI_CHIP_R300_AD: /* 9500 with 1 quadpipe verified by: Reid Linnemann <lreid@cs.okstate.edu> */
+       case PCI_CHIP_RV410_5E4C: /* RV410 SE only have 1 quadpipe */
+       case PCI_CHIP_RV410_5E4F: /* RV410 SE only have 1 quadpipe */
+	   screen->num_gb_pipes = 1;
+	   break;
+       default:
+	   break;
+       }
+
    }
 
    if ( sPriv->drm_version.minor >= 10 ) {
diff -ruN mesa-c490349825e2c5a11a5f598fb52a6f92597f694e/src/mesa/shader/arbprogparse.c mesa-656cd707db3699079d3a1b886d82e6f04afb1065/src/mesa/shader/arbprogparse.c
--- mesa-c490349825e2c5a11a5f598fb52a6f92597f694e/src/mesa/shader/arbprogparse.c	2009-04-18 19:12:33.000000000 +0200
+++ mesa-656cd707db3699079d3a1b886d82e6f04afb1065/src/mesa/shader/arbprogparse.c	2009-04-28 18:58:13.000000000 +0200
@@ -1496,10 +1496,16 @@
    curr = vc_head;
    while (curr) {
       if (curr->type == vt_attrib) {
-         if (curr->attrib_is_generic)
-            genericAttrib[ curr->attrib_binding ] = GL_TRUE;
-         else
+         if (curr->attrib_is_generic) {
+            GLuint attr = (curr->attrib_binding == 0)
+               ? 0 : (curr->attrib_binding - VERT_ATTRIB_GENERIC0);
+            assert(attr < MAX_VERTEX_PROGRAM_ATTRIBS);
+            genericAttrib[attr] = GL_TRUE;
+         }
+         else {
+            assert(curr->attrib_binding < MAX_VERTEX_PROGRAM_ATTRIBS);
             explicitAttrib[ curr->attrib_binding ] = GL_TRUE;
+         }
       }
 
       curr = curr->next;
diff -ruN mesa-c490349825e2c5a11a5f598fb52a6f92597f694e/src/mesa/swrast/s_texfilter.c mesa-656cd707db3699079d3a1b886d82e6f04afb1065/src/mesa/swrast/s_texfilter.c
--- mesa-c490349825e2c5a11a5f598fb52a6f92597f694e/src/mesa/swrast/s_texfilter.c	2009-04-18 19:12:33.000000000 +0200
+++ mesa-656cd707db3699079d3a1b886d82e6f04afb1065/src/mesa/swrast/s_texfilter.c	2009-04-28 18:58:13.000000000 +0200
@@ -2061,6 +2061,7 @@
       fcol -= 0.5F;
       i0 = IFLOOR(fcol);
       i1 = i0 + 1;
+      break;
    default:
       _mesa_problem(NULL, "bad wrapMode in clamp_rect_coord_linear");
       i0 = i1 = 0;
