Index: a2ps-4.13/configure.in
===================================================================
--- a2ps-4.13.orig/configure.in
+++ a2ps-4.13/configure.in
@@ -158,8 +158,9 @@ if test $ac_cv_func_fnmatch_works = no; 
 fi
 
 # Check for C99 va_copy
-AC_CACHE_CHECK([
-  AC_TRY_LINK([#include <stdarg.h>],
+AC_CACHE_CHECK([if compiler supports C99 va_copy],
+  [ac_cv_va_copy],
+  [AC_TRY_LINK([#include <stdarg.h>],
    [
 	va_list va1, va2;
 
