diff -Naur abiword-2.4.2.orig/abi/src/wp/ap/unix/ap_UnixDialog_Spell.cpp abiword-2.4.2/abi/src/wp/ap/unix/ap_UnixDialog_Spell.cpp
--- abiword-2.4.2.orig/abi/src/wp/ap/unix/ap_UnixDialog_Spell.cpp	2005-02-19 05:50:22.000000000 +0100
+++ abiword-2.4.2/abi/src/wp/ap/unix/ap_UnixDialog_Spell.cpp	2006-03-22 23:03:40.000000000 +0100
@@ -206,10 +206,13 @@
             // show word in main window
             makeWordVisible();
      
+			gpointer inst = gtk_tree_view_get_selection (GTK_TREE_VIEW (m_lvSuggestions));
+			g_signal_handler_block (inst, m_listHandlerID);
             // update dialog with new misspelled word info/suggestions
             _updateWindow();
-     
-            // run into the GTK event loop for this window
+			g_signal_handler_unblock (inst, m_listHandlerID);
+
+			// run into the GTK event loop for this window
 	    gint response = abiRunModalDialog (GTK_DIALOG(mainWindow), false);
 	    UT_DEBUGMSG (("ROB: response='%d'\n", response));
             switch(response) {
@@ -303,10 +306,6 @@
 					  "row-activated", 
 					  G_CALLBACK (AP_UnixDialog_Spell__onSuggestionDblClicked), 
 					  (gpointer)this);
-	m_listHandlerID = g_signal_connect (gtk_tree_view_get_selection (GTK_TREE_VIEW (m_lvSuggestions)), 
-					  "changed",
-					  G_CALLBACK (AP_UnixDialog_Spell__onSuggestionSelected), 
-					  (gpointer)this);
     m_replaceHandlerID = g_signal_connect (G_OBJECT(m_eChange), 
 					   "changed",
 					   G_CALLBACK (AP_UnixDialog_Spell__onSuggestionChanged),
@@ -336,6 +335,10 @@
 	GtkTreeViewColumn *column = gtk_tree_view_get_column (GTK_TREE_VIEW (m_lvSuggestions), 0);
 	gtk_tree_view_column_set_sort_column_id (column, COLUMN_SUGGESTION);
 
+	m_listHandlerID = g_signal_connect (gtk_tree_view_get_selection (GTK_TREE_VIEW (m_lvSuggestions)), 
+				  "changed",
+				  G_CALLBACK (AP_UnixDialog_Spell__onSuggestionSelected), 
+				  (gpointer)this);
 
     gtk_widget_show_all (m_wDialog);
     return m_wDialog;
@@ -433,6 +436,11 @@
 								COLUMN_NUMBER, i,
 								-1);
 	    }
+		// put the first suggestion in the entry
+		suggest = (gchar*) _convertToMB((UT_UCSChar*)m_Suggestions->getNthItem(0));
+        g_signal_handler_block(G_OBJECT(m_eChange), m_replaceHandlerID);
+        gtk_entry_set_text(GTK_ENTRY(m_eChange), suggest);
+        g_signal_handler_unblock(G_OBJECT(m_eChange), m_replaceHandlerID);      
     }
 
 	gtk_tree_view_set_model (GTK_TREE_VIEW (m_lvSuggestions), model);
