From d6bb94e9c05072fc791fb5045f5b1ab22d11d916 Mon Sep 17 00:00:00 2001
From: Sjoerd Simons <sjoerd.simons@collabora.co.uk>
Date: Mon, 19 Apr 2010 13:47:26 +0000
Subject: Make sure the service files are created correctly

---
diff --git a/daemon/Makefile.am b/daemon/Makefile.am
index 4c95b5a..9edb13f 100644
--- a/daemon/Makefile.am
+++ b/daemon/Makefile.am
@@ -68,7 +68,7 @@ service_DATA     = $(service_in_files:.service.in=.service)
 service_PATH     = $(VALGRIND_RUN)$(bindir)
 
 # Rule to make the service file with bindir expanded
-$(service_DATA): $(service_in_files) Makefile
+%.service: %.service.in Makefile
 	@sed -e "s|\@PATH\@|$(service_PATH)|" $< > $@
 
 desktop_in_files = \
--
cgit v0.8.3.1
