From b36d6beeb2fa430210c8a2aa19f772bd75325e42 Mon Sep 17 00:00:00 2001
From: Rob Bradford <rob@linux.intel.com>
Date: Wed, 14 Apr 2010 18:12:41 +0000
Subject: Add dbus activation service file for org.freedesktop.secrets

Fixes: https://bugzilla.gnome.org/show_bug.cgi?id=611002
---
diff --git a/daemon/Makefile.am b/daemon/Makefile.am
index be96523..4c95b5a 100644
--- a/daemon/Makefile.am
+++ b/daemon/Makefile.am
@@ -54,14 +54,16 @@ gnome_keyring_daemon_LDADD = \
 
 EXTRA_DIST = \
 	org.gnome.keyring.service.in \
+	org.freedesktop.secrets.service.in \
 	$(desktop_in_files)
 
 CLEANFILES = \
 	org.gnome.keyring.service \
+	org.freedesktop.service \
 	$(desktop_DATA)
 
 servicedir       = $(DBUS_SERVICES_DIR)
-service_in_files = org.gnome.keyring.service.in
+service_in_files = org.gnome.keyring.service.in org.freedesktop.secrets.service.in
 service_DATA     = $(service_in_files:.service.in=.service)
 service_PATH     = $(VALGRIND_RUN)$(bindir)
 
diff --git a/daemon/org.freedesktop.secrets.service.in b/daemon/org.freedesktop.secrets.service.in
new file mode 100644
index 0000000..d4db713
--- a/dev/null
+++ b/daemon/org.freedesktop.secrets.service.in
@@ -0,0 +1,3 @@
+[D-BUS Service]
+Name=org.freedesktop.secrets
+Exec=@PATH@/gnome-keyring-daemon --start --foreground --components=secrets
--
cgit v0.8.3.1
