From 37c5abe588d1ad63dc600cd9b9fd4d362f973a23 Mon Sep 17 00:00:00 2001
From: Dave Beckett <dave@dajobe.org>
Date: Mon, 11 Jul 2011 09:56:04 -0700
Subject: [PATCH] Allow = instead of := in TRiG

---
 src/turtle_lexer.l |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/turtle_lexer.l b/src/turtle_lexer.l
index c2f249b..e775771 100644
--- a/src/turtle_lexer.l
+++ b/src/turtle_lexer.l
@@ -345,7 +345,7 @@ EXPONENT [eE][+-]?[0-9]+
                   yyterminate();  }
 
 
-{QUOTEDURI}[\ \t\v\n]*(":-")?[\ \t\v\n]*"{"   {
+{QUOTEDURI}[\ \t\v\n]*("=")?[\ \t\v\n]*"{"   {
                   raptor_stringbuffer* sb;
                   unsigned char* uri_string;
 
@@ -377,11 +377,11 @@ EXPONENT [eE][+-]?[0-9]+
                     TURTLE_LEXER_OOM();
                 return GRAPH_NAME_LEFT_CURLY; }
 
-{QNAME}[\ \t\v\n]*(":-")?[\ \t\v\n]*"{"   {
+{QNAME}[\ \t\v\n]*("=")?[\ \t\v\n]*"{"   {
                while(1) {
                   int c = yytext[yyleng - 1];
                   if(c == '{' || c == ' ' || c=='\t' || c == '\v' || c == '\n' ||
-                     c == ':' || c == '-') {
+                     c == '=') {
                     yyleng--;
                   } else
                     break;
-- 
1.7.6

