From a8f405b8f90d752c3609b9e918dc949c4efadf87 Mon Sep 17 00:00:00 2001
From: Takashi Iwai <tiwai@suse.de>
Date: Thu, 25 Apr 2013 09:55:49 +0200
Subject: [PATCH] Add definition of __inline__ for non-GCC

Signed-off-by: Takashi Iwai <tiwai@suse.de>
---
 include/asoundlib-head.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/include/asoundlib-head.h b/include/asoundlib-head.h
index 6edbab0..71b5c29 100644
--- a/include/asoundlib-head.h
+++ b/include/asoundlib-head.h
@@ -40,6 +40,10 @@
 #include <errno.h>
 #include <stdarg.h>
 
+#ifndef __GNUC__
+#define __inline__ inline
+#endif
+
 #include <alsa/asoundef.h>
 #include <alsa/version.h>
 #include <alsa/global.h>
-- 
1.7.11.7

