From 437e64ffb5f19d2f088339954686e794cc276319 Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Mon, 23 Sep 2013 18:27:17 +0000
Subject: main: Fix typo in action name

See: https://bugzilla.gnome.org/show_bug.cgi?id=708647
---
diff --git a/src/totem.c b/src/totem.c
index bf82c04..12a0e5b 100644
--- a/src/totem.c
+++ b/src/totem.c
@@ -107,7 +107,7 @@ app_init (Totem *totem, char **argv)
 	set_rtl_icon_name (totem, "next-chapter", "media-skip-forward");
 	set_rtl_icon_name (totem, "previous-chapter", "media-skip-backward");
 	set_rtl_icon_name (totem, "skip-forward", "media-seek-forward");
-	set_rtl_icon_name (totem, "skip-backward", "media-seek-backward");
+	set_rtl_icon_name (totem, "skip-backwards", "media-seek-backward");
 
 	totem->win = GTK_WIDGET (gtk_builder_get_object (totem->xml, "totem_main_window"));
 	g_object_bind_property (G_OBJECT (totem), "fullscreen",
--
cgit v0.9.2
