diff --git a/avidemux/ADM_inputs/ADM_h263/ADM_mp4.cpp b/avidemux/ADM_inputs/ADM_h263/ADM_mp4.cpp
index 9d11e5d..8f6ac7f 100644
--- a/avidemux/ADM_inputs/ADM_h263/ADM_mp4.cpp
+++ b/avidemux/ADM_inputs/ADM_h263/ADM_mp4.cpp
@@ -70,8 +70,8 @@ cb cb_decode[]={
 
 };
 
-#define ONEOPT(x) {if(parser->read1bit()) printf("\t"x"\n");			}
-#define NOT_ONEOPT(x) {if(!parser->read1bit()) printf("\t"x"\n");			}
+#define ONEOPT(x) {if(parser->read1bit()) printf("\t" x "\n");			}
+#define NOT_ONEOPT(x) {if(!parser->read1bit()) printf("\t" x "\n");			}
 #define TWOPT(x,y) {if(parser->read1bit()) printf(x);	else printf(y);		}
 
 //_________________________________
diff --git a/avidemux/ADM_inputs/ADM_ogm/ADM_ogmAudio.cpp b/avidemux/ADM_inputs/ADM_ogm/ADM_ogmAudio.cpp
index 944d0bf..3763dd0 100644
--- a/avidemux/ADM_inputs/ADM_ogm/ADM_ogmAudio.cpp
+++ b/avidemux/ADM_inputs/ADM_ogm/ADM_ogmAudio.cpp
@@ -402,7 +402,7 @@ static WAVHeader hdr;
 			// Now we forward till the next header is > value
 			while(_demuxer->readHeaderOfType(_currentTrack->audioTrack,&cursize,&flags,&f))
 			{
-				if(f>val || abs(f-val)<CLOSE_ENOUGH)
+				if(f>val || abs(static_cast<int64_t>(f-val))<CLOSE_ENOUGH)
 				{
 					aprintf("Wanted %llu",val);
 					aprintf(" got %llu\n",f);					
diff --git a/plugins/ADM_videoEncoder/ADM_vidEnc_mpeg2enc/mpeg2enc/find_best_one_pel.cc b/plugins/ADM_videoEncoder/ADM_vidEnc_mpeg2enc/mpeg2enc/find_best_one_pel.cc
index 43d40cc..e855d4f 100644
--- a/plugins/ADM_videoEncoder/ADM_vidEnc_mpeg2enc/mpeg2enc/find_best_one_pel.cc
+++ b/plugins/ADM_videoEncoder/ADM_vidEnc_mpeg2enc/mpeg2enc/find_best_one_pel.cc
@@ -30,7 +30,7 @@ void find_best_one_pel_mmxe( me_result_set *sub22set,
                 int x;
 		matchrec = sub22set->mests[k];
 		orgblk = org + (i0+matchrec.x)+rowstride*(j0+matchrec.y);
-		penalty = (abs(matchrec.x) + abs(matchrec.y))<<3;
+		penalty = (abs(static_cast<int32_t>(matchrec.x)) + abs(static_cast<int32_t>(matchrec.y)))<<3;
 		
 		/* Get SAD for macroblocks: 	orgblk,orgblk(+1,0),
 		   orgblk(0,+1), and orgblk(+1,+1)
diff --git a/plugins/ADM_videoFilters/Telecide/ADM_vidDecTelecide.cpp b/plugins/ADM_videoFilters/Telecide/ADM_vidDecTelecide.cpp
index c0cabbe..3a8f699 100644
--- a/plugins/ADM_videoFilters/Telecide/ADM_vidDecTelecide.cpp
+++ b/plugins/ADM_videoFilters/Telecide/ADM_vidDecTelecide.cpp
@@ -690,7 +690,7 @@ uint8_t *finalpU,*finalpV;
                         {
                                 // The chosen frame doesn't match the prediction.
                                 if (predicted_metric == 0) mismatch = 0.0;
-                                else mismatch = (100.0*abs(predicted_metric - lowest))/predicted_metric;
+                                else mismatch = (100.0*abs(static_cast<int>(predicted_metric - lowest)))/predicted_metric;
                                 if (mismatch < gthresh)
                                 {
                                         // It's close enough, so use the predicted one.
@@ -732,7 +732,7 @@ uint8_t *finalpU,*finalpV;
                                         {
                                                 // The chosen frame doesn't match the prediction.
                                                 if (predicted_metric == 0) mismatch = 0.0;
-                                                else mismatch = (100.0*abs(predicted_metric - lowest))/predicted_metric;
+                                                else mismatch = (100.0*abs(static_cast<int>(predicted_metric - lowest)))/predicted_metric;
                                                 if ((int) mismatch <= gthresh)
                                                 {
                                                         // It's close enough, so use the predicted one.
