diff --git i/src/devices/bluetooth/meson.build w/src/devices/bluetooth/meson.build
index a77ad60d4..1505f2820 100644
--- i/src/devices/bluetooth/meson.build
+++ w/src/devices/bluetooth/meson.build
@@ -40,8 +40,9 @@ if enable_tests
 
   executable(
     test_unit,
+    sources,
     'tests/' + test_unit + '.c',
-    dependencies: libnetwork_manager_test_dep,
+    dependencies: deps + [libnetwork_manager_test_dep],
     c_args: test_c_flags,
   )
 endif
