--- a/speed.c	2016-11-22 22:43:25.336155948 +0000
+++ b/speed.c	2016-11-22 22:43:38.266122647 +0000
@@ -457,7 +457,6 @@
     double telapse;
     time_t temp;
     struct  timeval tv;
-    struct  timezone tz;
 
     temp = time(0);
 
@@ -478,7 +477,7 @@
         telapse = difftime( start, temp );
     }
 */
-        gettimeofday (&tv, &tz);
+        gettimeofday (&tv, NULL);
 
 	startsec = tv.tv_sec;
         startmilli = tv.tv_usec/1000;
@@ -489,12 +488,11 @@
 int gettime(double *elapse, float *elapscpu, int *status)
 {
         struct  timeval tv;
-        struct  timezone tz;
 	int stopmilli;
 	long stopsec;
 
 
-        gettimeofday (&tv, &tz);
+        gettimeofday (&tv, NULL);
     ecpu = clock();
     finish = time(0);
 
@@ -509,3 +507,4 @@
 
     return( *status );
 }
+
