--- a/w3mimg/x11/x11_w3mimg.c	2004-11-08 12:14:06.000000000 -0500
+++ b/w3mimg/x11/x11_w3mimg.c	2015-03-12 23:23:44.964555734 -0400
@@ -214,9 +214,12 @@
 	Pixmap p;
 	GC gc;
 	XImage *i;
+	XWindowAttributes attr;
+
+	XGetWindowAttributes(xi->display, xi->window, &attr);
 
 	p = XCreatePixmap(xi->display, xi->window, 1, 1,
-			  DefaultDepth(xi->display, 0));
+			  attr.depth);
 	gc = XCreateGC(xi->display, xi->window, 0, NULL);
 	if (!p || !gc)
 	    exit(1);		/* XXX */
@@ -328,6 +331,7 @@
     ImlibImage *im;
 #elif defined(USE_IMLIB2)
     Imlib_Image im;
+    XWindowAttributes attr;
 #elif defined(USE_GDKPIXBUF)
     GdkPixbufAnimation *animation;
     int j, iw, ih, n, frame_num, delay = -1, max_anim;
@@ -374,15 +378,16 @@
 	w = imlib_image_get_width();
     if (h <= 0)
 	h = imlib_image_get_height();
+    XGetWindowAttributes(xi->display, xi->window, &attr);
     img->pixmap = (void *)XCreatePixmap(xi->display, xi->parent, w, h,
-					DefaultDepth(xi->display, 0));
+					attr.depth);
     if (!img->pixmap)
 	return 0;
     XSetForeground(xi->display, xi->imageGC, xi->background_pixel);
     XFillRectangle(xi->display, (Pixmap) img->pixmap, xi->imageGC, 0, 0, w, h);
     imlib_context_set_display(xi->display);
-    imlib_context_set_visual(DefaultVisual(xi->display, 0));
-    imlib_context_set_colormap(DefaultColormap(xi->display, 0));
+    imlib_context_set_visual(attr.visual);
+    imlib_context_set_colormap(attr.colormap);
     imlib_context_set_drawable((Drawable) img->pixmap);
     imlib_render_image_on_drawable_at_size(0, 0, w, h);
     imlib_free_image();
