diff -urp coreutils-6.10-orig/src/Makefile.am coreutils-6.10/src/Makefile.am
--- coreutils-6.10-orig/src/Makefile.am	2008-01-11 11:34:22.000000000 +0100
+++ coreutils-6.10/src/Makefile.am	2008-01-23 14:52:26.000000000 +0100
@@ -170,7 +170,7 @@ install_su =						\
     echo " chown root $(installed_su)";			\
     chown root $(installed_su);				\
     echo " chmod $(setuid_root_mode) $(installed_su)";	\
-    chmod $(setuid_root_mode) $(installed_su)		\
+    chmod $(setuid_root_mode) $(installed_su);		\
   else							\
     :;							\
   fi
diff -urp coreutils-6.10-orig/gnulib-tests/test-getaddrinfo.c coreutils-6.10/gnulib-tests/test-getaddrinfo.c
--- coreutils-6.10-orig/gnulib-tests/test-getaddrinfo.c	2007-11-17 07:59:42.000000000 +0100
+++ coreutils-6.10/gnulib-tests/test-getaddrinfo.c	2008-01-25 19:19:43.000000000 +0100
@@ -61,6 +61,10 @@ int simple (char *host, char *service)
       if (res == EAI_SERVICE)
 	return 0;
 
+      /* Do not fail this test for temporary name resolution errors. */
+      if (res == EAI_AGAIN)
+         return 0;
+
       return 1;
     }
 
