diff -Nur leafpad-0.8.18.orig/src/file.c leafpad-0.8.18/src/file.c
--- leafpad-0.8.18.orig/src/file.c	2010-12-17 22:01:35.000000000 -0800
+++ leafpad-0.8.18/src/file.c	2010-12-21 09:26:00.236670201 -0800
@@ -223,7 +223,7 @@
 			GTK_MESSAGE_ERROR, _("Can't open file to write"));
 		return -1;
 	}
-	if (fwrite(str, 1, wbytes, fp) != wbytes) {
+	if (fwrite(cstr, 1, wbytes, fp) != wbytes) {
 		run_dialog_message(gtk_widget_get_toplevel(view),
 			GTK_MESSAGE_ERROR, _("Can't write file"));
 		return -1;
