diff -Nur NetworkManager-0.7.998.old/src/NetworkManagerPolicy.c NetworkManager-0.7.998/src/NetworkManagerPolicy.c
--- NetworkManager-0.7.998.old/src/NetworkManagerPolicy.c	2010-01-03 03:04:56.000000000 +0200
+++ NetworkManager-0.7.998/src/NetworkManagerPolicy.c	2010-01-08 15:21:28.000000000 +0200
@@ -391,7 +391,7 @@
 
 		/* No valid IP4 config (!!); fall back to localhost.localdomain */
 		msg = g_strdup_printf ("address lookup failed: %d", thread->ret);
-		set_system_hostname (NULL, msg);
+		//set_system_hostname (NULL, msg);
 		g_free (msg);
 	} else
 		set_system_hostname (thread->hostname, "from address lookup");
@@ -438,7 +438,7 @@
 
 	if (!best) {
 		/* No best device; fall back to localhost.localdomain */
-		set_system_hostname (NULL, "no default device");
+		//set_system_hostname (NULL, "no default device");
 		return;
 	}
 
@@ -462,7 +462,7 @@
 	    || (nm_ip4_config_get_num_nameservers (ip4_config) == 0)
 	    || (nm_ip4_config_get_num_addresses (ip4_config) == 0)) {
 		/* No valid IP4 config (!!); fall back to localhost.localdomain */
-		set_system_hostname (NULL, "no IPv4 config");
+		//set_system_hostname (NULL, "no IPv4 config");
 		return;
 	}
 
@@ -473,7 +473,7 @@
 	policy->lookup = lookup_thread_new (nm_ip4_address_get_address (addr), lookup_callback, policy);
 	if (!policy->lookup) {
 		/* Fall back to 'localhost.localdomain' */
-		set_system_hostname (NULL, "error starting hostname thread");
+		//set_system_hostname (NULL, "error starting hostname thread");
 	}
 }
 
