From ae84810db576a2e9237b926023e983f4f527fd22 Mon Sep 17 00:00:00 2001
From: Colin Walters <walters@verbum.org>
Date: Wed, 08 Sep 2010 16:58:20 +0000
Subject: gdk: Update gdkkeysyms.h

In preparation for adding a namespace prefix for introspection.
---
diff --git a/gdk/gdkkeysyms.h b/gdk/gdkkeysyms.h
index b4da5de..dfcea16 100644
--- a/gdk/gdkkeysyms.h
+++ b/gdk/gdkkeysyms.h
@@ -2195,6 +2195,7 @@
 #define GDK_Blue 0x1008ffa6
 #define GDK_Suspend 0x1008ffa7
 #define GDK_Hibernate 0x1008ffa8
+#define GDK_TouchpadToggle 0x1008ffa9
 #define GDK_Switch_VT_1 0x1008fe01
 #define GDK_Switch_VT_2 0x1008fe02
 #define GDK_Switch_VT_3 0x1008fe03
--
cgit v0.8.3.1
