diff --git a/libempathy-gtk/empathy-spell.c b/libempathy-gtk/empathy-spell.c
index 27f27f902..bb0159fd8 100644
--- a/libempathy-gtk/empathy-spell.c
+++ b/libempathy-gtk/empathy-spell.c
@@ -424,7 +424,7 @@ empathy_spell_add_to_dictionary (const gchar *code,
 	if (lang == NULL)
 		return;
 
-	enchant_dict_add_to_pwl (lang->speller, word, strlen (word));
+	enchant_dict_add (lang->speller, word, strlen (word));
 }
 
 #else /* not HAVE_ENCHANT */
