--- freecell-solver-4.20.0/CMakeLists.txt.orig	2018-07-25 06:54:13.736780097 +0000
+++ freecell-solver-4.20.0/CMakeLists.txt	2018-07-25 06:55:31.873652725 +0000
@@ -232,13 +232,6 @@
 SET (patsolve_dir "${pats_dir}/patsolve")
 SET (p_dir "${patsolve_dir}")
 
-SET (patsolve_modules
-    "param.c"
-    "${p_dir}/pat.c"
-    "${p_dir}/patsolve.c"
-    "${p_dir}/tree.c"
-    )
-
 FIND_PROGRAM(_GPERF "gperf")
 
 IF ("${_GPERF}" STREQUAL "_GPERF-NOTFOUND")
@@ -311,6 +304,12 @@
 IF (NOT "${FCS_DISABLE_PATSOLVE}")
     SET (pp_dir "${CMAKE_CURRENT_SOURCE_DIR}/${p_dir}")
     EXECUTE_PROCESS(COMMAND "python3" "${pp_dir}/param.py" "${pp_dir}/param.dat")
+    SET (patsolve_modules
+    "param.c"
+    "${p_dir}/pat.c"
+    "${p_dir}/patsolve.c"
+    "${p_dir}/tree.c"
+    )
     add_lib_mods(${patsolve_modules})
 ENDIF ()
 IF (NOT "${FCS_USE_PRECOMPILED_CMD_LINE_THEME}")
