--- src/net/base/mime_util.cc	2010-05-28 16:39:20.000000000 +0200
+++ src/net/base/mime_util.cc	2010-05-28 17:14:08.047881145 +0200
@@ -189,7 +189,6 @@
   "audio/ogg",
   "application/ogg",
 
-#if defined(GOOGLE_CHROME_BUILD)
   // MPEG-4.
   "video/mp4",
   "video/x-m4v",
@@ -200,7 +199,6 @@
   "audio/mp3",
   "audio/x-mp3",
   "audio/mpeg",
-#endif
 };
 
 // List of supported codecs when passed in with <source type="...">.
@@ -208,10 +206,8 @@
 // Refer to http://wiki.whatwg.org/wiki/Video_type_parameters#Browser_Support
 // for more information.
 static const char* const supported_media_codecs[] = {
-#if defined(GOOGLE_CHROME_BUILD)
   "avc1",
   "mp4a",
-#endif
   "theora",
   "vorbis",
 };
