--- file-5.07/magic/Magdir/archive.old	2011-04-23 17:02:48.000000000 +0200
+++ file-5.07/magic/Magdir/archive	2011-05-15 10:14:44.509271579 +0200
@@ -654,6 +654,10 @@
 >>>>78	string	-template		Template
 !:mime	application/vnd.oasis.opendocument.image-template
 
+>26 byte x Zip archive data
+
+
+
 # StarView Metafile
 # From Pierre Ducroquet <pinaraf@pinaraf.info>
 0	string	VCLMTF	StarView MetaFile
