diff -b -u -r mozilla.orig/widget/src/gtk2/nsWindow.cpp mozilla/widget/src/gtk2/nsWindow.cpp
--- mozilla.orig/widget/src/gtk2/nsWindow.cpp   2005-07-01 12:29:41.000000000 +0800
+++ mozilla/widget/src/gtk2/nsWindow.cpp        2005-09-07 18:31:10.000000000 +0800
@@ -4405,6 +4405,9 @@
         delete[] textEvent.rangeArray;
     }

+    if (aLen == 0)
+        return;
+
     gint x1, y1, x2, y2;
     GtkWidget *widget =
         get_gtk_widget_for_gdk_window(this->mDrawingarea->inner_window);
