diff --git a/sources/pyside6/PySide6/__init__.py.in b/sources/pyside6/PySide6/__init__.py.in
index d8439985d..5788ca57d 100644
--- a/sources/pyside6/PySide6/__init__.py.in
+++ b/sources/pyside6/PySide6/__init__.py.in
@@ -123,7 +123,7 @@ def _find_all_qt_modules():
     location = Path(__file__).resolve().parent
 
     # Note: We should _not_ call this function while still building, but use the existing value!
-    in_build = location.parents[1].name == "build"
+    in_build = location.parents[2].name == "build"
     if in_build:
         return __all__
 
