=== modified file 'libdb/db_lookup.c'
--- libdb/db_lookup.c	2008-12-25 22:13:50 +0000
+++ libdb/db_lookup.c	2009-06-21 01:29:52 +0000
@@ -323,11 +323,15 @@
 		if (!info->name)
 			info->name = xstrdup (page);
 		if (!(flags & MATCH_CASE) || STREQ (info->name, page)) {
-			if (section == NULL ||
-			    STRNEQ (section, info->ext,
-				    flags & EXACT ? strlen (info->ext) :
-						    strlen (section))) {
+			if (section == NULL)
 				return info;
+			if (flags & EXACT) {
+				if (STREQ (section, info->ext))
+					return info;
+			} else {
+				if (STRNEQ (section, info->ext,
+					    strlen (section)))
+					return info;
 			}
 		}
 		free_mandata_struct (info);
@@ -357,11 +361,16 @@
 			if ((flags & MATCH_CASE) && !STREQ (names[i], page))
 				continue;
 
-			if (section != NULL &&
-			    !STRNEQ (section, ext[i],
-				     flags & EXACT ? strlen (ext[i])
-						   : strlen (section)))
-				continue;
+			if (section != NULL) {
+				if (flags & EXACT) {
+					if (!STREQ (section, ext[i]))
+						continue;
+				} else {
+					if (!STRNEQ (section, ext[i],
+						     strlen (section)))
+						continue;
+				}
+			}
 
 			/* So the key is suitable ... */
 			key = make_multi_key (names[i], ext[i]);

