From aab140d868c1ebedbb04bbd65deddf94dc21787b Mon Sep 17 00:00:00 2001
From: Jannis Pohlmann <jannis@xfce.org>
Date: Mon, 26 Sep 2011 17:28:51 +0000
Subject: Fix generation of video images by starting the pipeline (bug #7996).

Patch provided by Sam Thursfield<ssssam@gmail.com>.
---
diff --git a/NEWS b/NEWS
index 0ff1c1c..48e3896 100644
--- a/NEWS
+++ b/NEWS
@@ -1,3 +1,7 @@
+0.1.xx
+======
+- Fix generation of video images by starting the pipeline (bug #7996).
+
 0.1.22
 ======
 - Add coding style documentation.
diff --git a/plugins/gst-thumbnailer/gst-helper.c b/plugins/gst-thumbnailer/gst-helper.c
index b8a29ab..dead043 100644
--- a/plugins/gst-thumbnailer/gst-helper.c
+++ b/plugins/gst-thumbnailer/gst-helper.c
@@ -154,6 +154,8 @@ gst_helper_convert_buffer_to_pixbuf (GstBuffer    *buffer,
 
   bus = gst_element_get_bus (GST_ELEMENT (pipeline));
 
+  gst_element_set_state (GST_ELEMENT (pipeline), GST_STATE_PLAYING);
+
   i = 0;
   msg = NULL;
   while (msg == NULL && i < 5)
--
cgit 
