diff --git i/tests/layouts/valid-20.layout w/tests/layouts/valid-20.layout
index 93b8a1b1..db692b73 100644
--- i/tests/layouts/valid-20.layout
+++ w/tests/layouts/valid-20.layout
@@ -140,23 +140,23 @@
                 "glyph" : 244,
                 "width" : 15360,
                 "x-offset" : 14764,
-                "y-offset" : 3844,
+                "y-offset" : 3845,
                 "is-cluster-start" : true,
                 "log-cluster" : 0
               },
               {
                 "glyph" : 272,
                 "width" : 15360,
                 "x-offset" : 14764,
                 "y-offset" : 4280,
                 "is-cluster-start" : true,
                 "log-cluster" : 1
               },
               {
                 "glyph" : 273,
                 "width" : 15360,
                 "x-offset" : 14764,
-                "y-offset" : 3506,
+                "y-offset" : 3507,
                 "is-cluster-start" : true,
                 "log-cluster" : 2
               }
diff --git i/tests/layouts/vertical.layout w/tests/layouts/vertical.layout
index e8619bfb..a7f65068 100644
--- i/tests/layouts/vertical.layout
+++ w/tests/layouts/vertical.layout
@@ -128,24 +128,24 @@
               {
                 "glyph" : 1,
                 "width" : 32768,
-                "x-offset" : 28835,
+                "x-offset" : 28836,
                 "y-offset" : 9945,
                 "is-cluster-start" : true,
                 "log-cluster" : 0
               },
               {
                 "glyph" : 2,
                 "width" : 32768,
-                "x-offset" : 28835,
+                "x-offset" : 28836,
                 "y-offset" : 10764,
                 "is-cluster-start" : true,
                 "log-cluster" : 1
               },
               {
                 "glyph" : 3,
                 "width" : 32768,
-                "x-offset" : 28835,
-                "y-offset" : 10452,
+                "x-offset" : 28836,
+                "y-offset" : 10453,
                 "is-cluster-start" : true,
                 "log-cluster" : 2
               }
@@ -179,24 +179,24 @@
               {
                 "glyph" : 4,
                 "width" : 32768,
-                "x-offset" : 28835,
-                "y-offset" : 16383,
+                "x-offset" : 28836,
+                "y-offset" : 16384,
                 "is-cluster-start" : true,
                 "log-cluster" : 0
               },
               {
                 "glyph" : 5,
                 "width" : 32768,
-                "x-offset" : 28835,
-                "y-offset" : 16383,
+                "x-offset" : 28836,
+                "y-offset" : 16384,
                 "is-cluster-start" : true,
                 "log-cluster" : 3
               },
               {
                 "glyph" : 7,
                 "width" : 32768,
-                "x-offset" : 28835,
-                "y-offset" : 16383,
+                "x-offset" : 28836,
+                "y-offset" : 16384,
                 "is-cluster-start" : true,
                 "log-cluster" : 6
               }
@@ -230,24 +230,24 @@
               {
                 "glyph" : 1,
                 "width" : 32768,
-                "x-offset" : 28835,
+                "x-offset" : 28836,
                 "y-offset" : 9945,
                 "is-cluster-start" : true,
                 "log-cluster" : 0
               },
               {
                 "glyph" : 2,
                 "width" : 32768,
-                "x-offset" : 28835,
+                "x-offset" : 28836,
                 "y-offset" : 10764,
                 "is-cluster-start" : true,
                 "log-cluster" : 1
               },
               {
                 "glyph" : 3,
                 "width" : 32768,
-                "x-offset" : 28835,
-                "y-offset" : 10452,
+                "x-offset" : 28836,
+                "y-offset" : 10453,
                 "is-cluster-start" : true,
                 "log-cluster" : 2
               }
