--- lprint-1.1.0/Makefile.in	2022-01-04 13:28:46.000000000 +0100
+++ lprint-1.1.0/Makefile.in.new	2022-01-04 13:30:11.355829642 +0100
@@ -131,9 +131,9 @@
 	    $(INSTALL) -d -m 755 $(BUILDROOT)/Library/LaunchDaemons; \
 	    $(INSTALL) -c -m 644 org.msweet.lprint.plist $(BUILDROOT)/Library/LaunchDaemons; \
 	else \
-	    echo "Installing systemd service to $(BUILDROOT)$(sysconfdir)/systemd/system..."; \
-	    $(INSTALL) -d -m 755 $(BUILDROOT)$(sysconfdir)/systemd/system; \
-	    $(INSTALL) -c -m 644 lprint.service $(BUILDROOT)$(sysconfdir)/systemd/system; \
+	    echo "Installing systemd service to $(BUILDROOT)$(libdir)/systemd/system..."; \
+	    $(INSTALL) -d -m 755 $(BUILDROOT)$(libdir)/systemd/system; \
+	    $(INSTALL) -c -m 644 lprint.service $(BUILDROOT)$(libdir)/systemd/system; \
 	fi
 
 
