diff --git i/meson.build w/meson.build
index 5885b9f4..771adf17 100644
--- i/meson.build
+++ w/meson.build
@@ -7,7 +7,7 @@ sound_juicer_libdir = join_paths(sound_juicer_prefix, get_option('libdir'))
 sound_juicer_libexecdir = join_paths(sound_juicer_prefix, get_option('libexecdir'))
 sound_juicer_localedir = join_paths(sound_juicer_prefix, get_option('localedir'))
 
-sound_juicer_docdir = join_paths(sound_juicer_prefix, 'doc', meson.project_name())
+sound_juicer_docdir = join_paths(sound_juicer_datadir, 'doc', meson.project_name())
 
 gnome = import('gnome')
 i18n = import('i18n')
