diff --git a/find-modules/FindPythonModuleGeneration.cmake b/find-modules/FindPythonModuleGeneration.cmake
index 083b767..5baefd3 100644
--- a/find-modules/FindPythonModuleGeneration.cmake
+++ b/find-modules/FindPythonModuleGeneration.cmake
@@ -484,14 +484,9 @@ headers = sipAPI${GPB_MODULENAME}
         set_property(TARGET Py${pyversion}KF5${GPB_MODULENAME} PROPERTY
             OUTPUT_NAME "${GPB_MODULENAME}")
 
-        if (GPB_SIP_DEPENDS MATCHES PyKF5)
-          set(_kf5_python_prefix ${KDE_INSTALL_FULL_PYTHON${pyversion}DIR})
-        else()
-          set(_kf5_python_prefix ${CMAKE_BINARY_DIR}/py${pyversion})
-        endif()
         add_test(NAME Py${pyversion}Test${GPB_MODULENAME} COMMAND
             ${GPB_PYTHON${pyversion}_COMMAND} "${CMAKE_SOURCE_DIR}/autotests/pythontest.py"
-            ${_kf5_python_prefix}
+            ${CMAKE_BINARY_DIR}/py${pyversion}
         )
 
         install(DIRECTORY ${CMAKE_BINARY_DIR}/py${pyversion}/${GPB_PYTHONNAMESPACE}
