diff -ru PyQt6-6.0.1.dev2101052155/sip/QtQuick/qsggeometry.sip PyQt6-6.0.1.dev2101060956/sip/QtQuick/qsggeometry.sip
--- PyQt6-6.0.1.dev2101052155/sip/QtQuick/qsggeometry.sip       2021-01-06 01:30:33.889339000 +0100
+++ PyQt6-6.0.1.dev2101060956/sip/QtQuick/qsggeometry.sip       2021-01-07 01:30:34.746270000 +0100
@@ -27,28 +27,6 @@
 %End
 
 public:
-// Convenient primitives and drawing modes.
-enum /NoScope/
-{
-    GL_BYTE,
-%If (PyQt_OpenGL_ES2)
-    GL_DOUBLE,
-%End
-    GL_FLOAT,
-    GL_INT
-};
-
-enum /NoScope/
-{
-    GL_POINTS,
-    GL_LINES,
-    GL_LINE_LOOP,
-    GL_LINE_STRIP,
-    GL_TRIANGLES,
-    GL_TRIANGLE_STRIP,
-    GL_TRIANGLE_FAN
-};
-
     struct Attribute
     {
 %TypeHeaderCode
@@ -125,21 +103,19 @@
             
                         switch (attr->type)
                         {
-                        case GL_BYTE:
+                        case QSGGeometry::ByteType:
                             size = sizeof (qint8);
                             break;
             
-            #if !defined(SIP_FEATURE_PyQt_OpenGL_ES2) && GL_DOUBLE != GL_FLOAT
-                        case GL_DOUBLE:
+                        case QSGGeometry::DoubleType:
                             size = sizeof (double);
                             break;
-            #endif
             
-                        case GL_FLOAT:
+                        case QSGGeometry::FloatType:
                             size = sizeof (float);
                             break;
             
-                        case GL_INT:
+                        case QSGGeometry::IntType:
                             size = sizeof (qint32);
                             break;
             
