diff --git a/symbols/in b/symbols/in
index 3ca78a20664d803d92e87315b6372909f31594e8..b3a5ca9ef613876b668a521a8f464596ab835d9e 100644
--- a/symbols/in
+++ b/symbols/in
@@ -2326,7 +2326,7 @@ xkb_symbols "marathi" {
 	key <AB09> { [ period, U0964, U0965, U093d ] };
 	key <AB10> { [ U092f, question, slash, question ] };
 };
-=======
+
 // EXTRAS:
 
 //           Modi is an ancient Indian script that is used to write texts in Marathi, Hindi and Sanskrit.
