diff -Nur gdm-3.1.90.orig/daemon/gdm-server.c gdm-3.1.90/daemon/gdm-server.c
--- gdm-3.1.90.orig/daemon/gdm-server.c	2011-09-02 13:13:16.132989960 +0000
+++ gdm-3.1.90/daemon/gdm-server.c	2011-09-02 13:18:17.495996840 +0000
@@ -131,7 +131,7 @@
         g_return_val_if_fail (GDM_IS_SERVER (server), NULL);
 
         error = NULL;
-        command = g_strdup_printf (LIBEXECDIR "/ck-get-x11-display-device --display %s",
+        command = g_strdup_printf ("/usr/lib/ConsoleKit/ck-get-x11-display-device --display %s",
                                    server->priv->display_name);
 
         g_debug ("GdmServer: Running helper %s", command);
diff -Nur gdm-3.1.90.orig/gui/simple-chooser/gdm-chooser-session.c gdm-3.1.90/gui/simple-chooser/gdm-chooser-session.c
--- gdm-3.1.90.orig/gui/simple-chooser/gdm-chooser-session.c	2011-09-02 13:13:16.136323327 +0000
+++ gdm-3.1.90/gui/simple-chooser/gdm-chooser-session.c	2011-09-02 13:22:54.168703021 +0000
@@ -131,7 +131,7 @@
         ret = FALSE;
 
         error = NULL;
-        g_spawn_command_line_async (LIBEXECDIR "/gnome-settings-daemon", &error);
+        g_spawn_command_line_async ("/usr/lib/gnome-settings-daemon/gnome-settings-daemon", &error);
         if (error != NULL) {
                 g_warning ("Error starting settings daemon: %s", error->message);
                 g_error_free (error);
