diff --git i/gtkdoc/highlight.py w/gtkdoc/highlight.py
index 8f6e470..82e11e8 100644
--- i/gtkdoc/highlight.py
+++ w/gtkdoc/highlight.py
@@ -47,6 +47,6 @@ def highlight_code(code, lang='c'):
 
 
 def append_style_defs(css_file_name):
-    os.chmod(css_file_name, stat.S_IWRITE)
+    os.chmod(css_file_name, 0o644)
     with open(css_file_name, 'at', newline='\n', encoding='utf-8') as css:
         css.write(HTML_FORMATTER.get_style_defs())
