diff --git a/fc-solve/source/CMakeLists.txt b/fc-solve/source/CMakeLists.txt
index 94a3f3fc0..fb37caecd 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -273,7 +273,7 @@ ADD_CUSTOM_COMMAND(
     DEPENDS "${is_parent_gen}"
 )
 # Execute it right away to avoid missing files/includes
-EXECUTE_PROCESS(COMMAND "${PERL_EXECUTABLE}" ${is_parent_args} RESULT_VARIABLE _retcode)
+EXECUTE_PROCESS(COMMAND "${PERL_EXECUTABLE}" ${is_parent_args} WORKING_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}" RESULT_VARIABLE _retcode)
 
 IF (NOT ("${_retcode}" STREQUAL "0"))
     MESSAGE(FATAL_ERROR "${PERL_EXECUTABLE} ${is_parent_args} failed!")
@@ -308,6 +308,7 @@ ADD_CUSTOM_COMMAND(
         "${cl_enum_h}"
     COMMAND ${_cmd}
     DEPENDS "${CL_INC_SCRIPT}"
+    WORKING_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}"
 )
 
 # Execute it right away to avoid missing files/includes
@@ -329,6 +330,7 @@ ADD_CUSTOM_COMMAND(
         "${CMAKE_CURRENT_BINARY_DIR}/${MOVE_FUNCS_H}"
     COMMAND "${PERL_EXECUTABLE}" "${MOVE_FUNCS_GEN}"
     DEPENDS "${MOVE_FUNCS_GEN}" "${MOVE_FUNCS_GEN_MOD}"
+    WORKING_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}"
 )
 
 ADD_CUSTOM_TARGET(
@@ -374,7 +376,10 @@ ENDIF ()
 IF (NOT "${FCS_DISABLE_PATSOLVE}")
     SET (pp_dir "${CMAKE_CURRENT_SOURCE_DIR}/${p_dir}")
     SET (cmd "python3" "${pp_dir}/param.py" "${pp_dir}/param.dat")
-    EXECUTE_PROCESS(COMMAND ${cmd} RESULT_VARIABLE _retcode)
+    EXECUTE_PROCESS(
+        COMMAND ${cmd} RESULT_VARIABLE _retcode
+        WORKING_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}"
+    )
     IF (NOT ("${_retcode}" STREQUAL "0"))
         MESSAGE(FATAL_ERROR "${cmd} failed!")
     ENDIF ()
