--- xinit-1.4.0/startx.cpp	2018-03-11 17:49:26.995313610 +0100
+++ xinit-1.4.0/startx.cpp.new	2018-03-11 17:49:11.334908458 +0100
@@ -296,9 +296,9 @@
     XCOMM now add the same credentials to the client authority file
     XCOMM if '$displayname' already exists do not overwrite it as another
     XCOMM server may need it. Add them to the '$xserverauthfile' instead.
-    for displayname in $authdisplay $hostname$authdisplay; do
+    for displayname in $authdisplay $hostname/unix$authdisplay; do
         authcookie=`XAUTH list "$displayname" @@
-        | sed -n "s/.*$displayname[[:space:]*].*[[:space:]*]//p"` 2>/dev/null;
+        | sed -n "s/.*$displayname\/unix$authdisplay[[:space:]*].*[[:space:]*]//p"` 2>/dev/null;
         if [ "z${authcookie}" = "z" ] ; then
             XAUTH -q << EOF 
 add $displayname . $mcookie
