diff -Naur cdparanoia-III-alpha9.8-orig/Makefile.in cdparanoia-III-alpha9.8/Makefile.in
--- cdparanoia-III-alpha9.8-orig/Makefile.in	2001-03-27 14:46:58.000000000 -0800
+++ cdparanoia-III-alpha9.8/Makefile.in	2004-07-11 08:22:24.000000000 -0700
@@ -8,7 +8,7 @@
 srcdir=@srcdir@
 
 @SET_MAKE@
-FLAGS=@TYPESIZES@ @CFLAGS@
+FLAGS=@TYPESIZES@ @CFLAGS@ -Iinterface
 OPT=@OPT@ $(FLAGS)
 DEBUG=@DEBUG@ $(FLAGS)
 CC=@CC@
diff -Naur cdparanoia-III-alpha9.8-orig/interface/utils.h cdparanoia-III-alpha9.8/interface/utils.h
--- cdparanoia-III-alpha9.8-orig/interface/utils.h	2000-04-19 15:41:04.000000000 -0700
+++ cdparanoia-III-alpha9.8/interface/utils.h	2004-07-11 08:22:48.000000000 -0700
@@ -112,6 +112,7 @@
       break;
     case CDDA_MESSAGE_FORGETIT:
     default:
+      ;
     }
   }
 }
@@ -127,6 +128,7 @@
       break;
     case CDDA_MESSAGE_FORGETIT:
     default:
+      ;
     }
   }
 }
@@ -169,6 +171,7 @@
       break;
     case CDDA_MESSAGE_FORGETIT:
     default:
+      ;
     }
   }
   if(malloced)free(buffer);
@@ -205,6 +208,7 @@
       break;
     case CDDA_MESSAGE_FORGETIT:
     default:
+      ;
     }
   }
   if(malloced)free(buffer);
diff -Naur cdparanoia-III-alpha9.8-orig/paranoia/Makefile.in cdparanoia-III-alpha9.8/paranoia/Makefile.in
--- cdparanoia-III-alpha9.8-orig/paranoia/Makefile.in	1999-12-13 20:28:04.000000000 -0800
+++ cdparanoia-III-alpha9.8/paranoia/Makefile.in	2004-07-11 08:22:36.000000000 -0700
@@ -8,7 +8,7 @@
 srcdir=@srcdir@
 
 @SET_MAKE@
-FLAGS=@TYPESIZES@ @CFLAGS@
+FLAGS=@TYPESIZES@ @CFLAGS@ -I../interface
 OPT=@OPT@ $(FLAGS)
 DEBUG=@DEBUG@ $(FLAGS)
 CC=@CC@
