--- pycups-1.9.49/Makefile~	2010-05-19 17:56:30.437732869 +0200
+++ pycups-1.9.49/Makefile	2010-05-19 17:57:10.201877902 +0200
@@ -9,7 +9,7 @@
 	COPYING NEWS README TODO ChangeLog
 
 cups.so: $(SOURCES)
-	CFLAGS=-DVERSION=\\\"$(VERSION)\\\" python setup.py build
+	python setup.py build
 	mv build/lib*/$@ .
 
 doc:	cups.so
--- pycups-1.9.49/setup.py~	2010-05-19 17:57:19.441043994 +0200
+++ pycups-1.9.49/setup.py	2010-05-19 17:57:55.434385149 +0200
@@ -3,4 +3,5 @@
        ext_modules=[Extension("cups",
                               ["cupsmodule.c", "cupsconnection.c",
                                "cupsppd.c", "cupsipp.c"],
-                              libraries=["cups"])])
+                              libraries=["cups"],
+                              define_macros=[("VERSION", '"1.9.49"')])])
