--- gdb-7.0/gdb/configure.ac~	2010-07-08 14:54:43.770583270 +0200
+++ gdb-7.0/gdb/configure.ac	2010-07-08 14:56:17.539947910 +0200
@@ -881,6 +881,12 @@
   LIBS="$LIBS ${python_libs}"
   have_libpython=no
   if test "${have_libpython}" = no; then
+    AC_TRY_LIBPYTHON(python2.7, have_libpython)
+    if test "${HAVE_LIBPYTHON2_7}" = yes; then
+      AC_DEFINE(HAVE_LIBPYTHON2_7, 1, [Define if Python 2.7 is being used.])
+    fi
+  fi
+  if test "${have_libpython}" = no; then
     AC_TRY_LIBPYTHON(python2.6, have_libpython)
     if test "${HAVE_LIBPYTHON2_6}" = yes; then
       AC_DEFINE(HAVE_LIBPYTHON2_6, 1, [Define if Python 2.6 is being used.])
--- gdb-7.0/gdb/python/python-internal.h~	2010-07-08 14:54:43.136858331 +0200
+++ gdb-7.0/gdb/python/python-internal.h	2010-07-08 14:56:56.739736159 +0200
@@ -45,6 +45,9 @@
 #elif HAVE_LIBPYTHON2_6
 #include "python2.6/Python.h"
 #include "python2.6/frameobject.h"
+#elif HAVE_LIBPYTHON2_7
+#include "python2.7/Python.h"
+#include "python2.7/frameobject.h"
 #else
 #error "Unable to find usable Python.h"
 #endif
