--- tomboy-0.12.1/Tomboy/Addins/PrintNotes/gedit-print.c	2008-09-22 14:11:18.000000000 +0200
+++ tomboy-0.12.1.patched/Tomboy/Addins/PrintNotes/gedit-print.c	2008-10-11 23:27:29.000000000 +0200
@@ -48,7 +48,6 @@
 #include <gtk/gtkimage.h>
 #include <gtk/gtkwindow.h>
 #include <libgnomeprintui/gnome-print-dialog.h>
-#include <libgnomeprintui/gnome-print-job-preview.h>
 
 #include "gedit-print.h"
 #include "gtksourceprintjob.h"
