From 178c810b7797ff2fca0440e519ddf4dbede589a1 Mon Sep 17 00:00:00 2001
From: Giovanni Campagna <gcampagna@src.gnome.org>
Date: Sun, 12 May 2013 16:46:19 +0000
Subject: gettext: fix typo

https://bugzilla.gnome.org/show_bug.cgi?id=700336
---
diff --git a/modules/gettext.js b/modules/gettext.js
index 8f73c13..60991e1 100644
--- a/modules/gettext.js
+++ b/modules/gettext.js
@@ -46,7 +46,7 @@ function gettext(msgid) {
     return GLib.dgettext(null, msgid);
 }
 function dgettext(domain, msgid) {
-    return GLib.dgetext(domain, msgid);
+    return GLib.dgettext(domain, msgid);
 }
 function dcgettext(domain, msgid, category) {
     return GLib.dcgettext(domain, msgid, category);
--
cgit v0.9.2
