--- plugins/collab/backends/service/xp/tls_tunnel.cpp	2014-11-04 02:12:14.000000000 +0100
+++ plugins/collab/backends/service/xp/tls_tunnel.cpp.new	2015-04-17 17:23:16.297151397 +0200
@@ -306,9 +306,6 @@
 	disconnect_(transport_ptr, session_ptr, local_socket_ptr, remote_socket_ptr);		
 }
 
-static const int PRIORITIES[] = { GNUTLS_KX_ANON_DH, GNUTLS_KX_RSA, GNUTLS_KX_DHE_DSS, GNUTLS_KX_DHE_RSA, 0 };
-static const int CIPHERS[] = { GNUTLS_CIPHER_AES_256_CBC, GNUTLS_CIPHER_AES_128_CBC, GNUTLS_CIPHER_3DES_CBC, GNUTLS_CIPHER_ARCFOUR_128, 0 };
-
 // FIXME: this clientproxy can only handle 1 SSL connection at the same time
 ClientProxy::ClientProxy(const std::string& connect_address, unsigned short connect_port, 
 		const std::string& ca_file, bool check_hostname)
@@ -400,8 +397,6 @@
 	// setup session
 	return_val_if_neg(gnutls_init(session_ptr.get(), GNUTLS_CLIENT), session_ptr_t());
 	return_val_if_neg(gnutls_set_default_priority(*session_ptr), session_ptr_t());
-	return_val_if_neg(gnutls_kx_set_priority(*session_ptr,PRIORITIES), session_ptr_t());
-	return_val_if_neg(gnutls_cipher_set_priority(*session_ptr,CIPHERS), session_ptr_t());
 	return_val_if_neg(gnutls_credentials_set(*session_ptr, GNUTLS_CRD_CERTIFICATE, x509cred), session_ptr_t());
 
 	// setup transport
