From 4d07d80b662913de3935829cd783b5b6f38dccd4 Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <heftig@archlinux.org>
Date: Wed, 3 Jun 2020 23:53:04 +0200
Subject: [PATCH 1/7] install gpg-zip

---
 tools/Makefile.am | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/tools/Makefile.am b/tools/Makefile.am
index 0c828a7bd..d9aa7d010 100644
--- a/tools/Makefile.am
+++ b/tools/Makefile.am
@@ -35,8 +35,8 @@ AM_CFLAGS = $(LIBGCRYPT_CFLAGS) $(GPG_ERROR_CFLAGS) $(LIBASSUAN_CFLAGS)
 sbin_SCRIPTS = addgnupghome applygnupgdefaults
 
 if HAVE_USTAR
-# bin_SCRIPTS += gpg-zip
-noinst_SCRIPTS = gpg-zip
+bin_SCRIPTS = gpg-zip
+# noinst_SCRIPTS = gpg-zip
 endif
 
 if BUILD_SYMCRYPTRUN
-- 
2.27.0

