From c17f20b949d2e80e596ce21ecd944db80aaa80b1 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?G=C3=BCnther=20Deschner?= <gdeschner@redhat.com>
Date: Wed, 29 Jan 2014 17:59:03 +0100
Subject: [PATCH] Fix potential segfault in gssi_inquire_context().
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Günther Deschner <gdeschner@redhat.com>
Reviewed-by: Simo Sorce <simo@redhat.com>
---
 proxy/src/mechglue/gpp_context.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/proxy/src/mechglue/gpp_context.c b/proxy/src/mechglue/gpp_context.c
index 6010724..bb16a93 100644
--- a/proxy/src/mechglue/gpp_context.c
+++ b/proxy/src/mechglue/gpp_context.c
@@ -223,7 +223,7 @@ OM_uint32 gssi_inquire_context(OM_uint32 *minor_status,
         maj = gss_inquire_context(&min,
                                   ctx_handle->local,
                                   s_name ? &s_name->local : NULL,
-                                  s_name ? &t_name->local : NULL,
+                                  t_name ? &t_name->local : NULL,
                                   lifetime_rec,
                                   &mech_oid,
                                   ctx_flags,
@@ -233,7 +233,7 @@ OM_uint32 gssi_inquire_context(OM_uint32 *minor_status,
         maj = gpm_inquire_context(&min,
                                   ctx_handle->remote,
                                   s_name ? &s_name->remote : NULL,
-                                  s_name ? &t_name->remote : NULL,
+                                  t_name ? &t_name->remote : NULL,
                                   lifetime_rec,
                                   &mech_oid,
                                   ctx_flags,
-- 
1.8.5.3


