From 36ba0c99c1780ca9f23f220ced70a76b97ce8260 Mon Sep 17 00:00:00 2001
From: Paolo Bacchilega <paobac@src.gnome.org>
Date: Fri, 27 Sep 2013 15:40:33 +0000
Subject: Fixed broken window layout with gtk+ 3.10

[bug #708800]
---
diff --git a/gthumb/gth-window.c b/gthumb/gth-window.c
index a2960bc..a2cce99 100644
--- a/gthumb/gth-window.c
+++ b/gthumb/gth-window.c
@@ -324,6 +324,7 @@ gth_window_attach (GthWindow     *window,
 		return;
 	}
 
+	gtk_widget_set_vexpand (child, FALSE);
 	gtk_grid_attach (GTK_GRID (window->priv->grid),
 			  child,
 			  0, position,
@@ -362,6 +363,7 @@ gth_window_attach_content (GthWindow *window,
 
 	_gtk_container_remove_children (GTK_CONTAINER (window->priv->contents[page]), NULL, NULL);
 	gtk_widget_set_hexpand (child, TRUE);
+	gtk_widget_set_vexpand (child, TRUE);
 	gtk_container_add (GTK_CONTAINER (window->priv->contents[page]), child);
 }
 
--
cgit v0.9.2
