diff -Naur libarchive-2.6.0.orig/libarchive/archive_read_support_compression_gzip.c libarchive-2.6.0/libarchive/archive_read_support_compression_gzip.c
--- libarchive-2.6.0.orig/libarchive/archive_read_support_compression_gzip.c	2008-12-28 14:08:31.000000000 -0600
+++ libarchive-2.6.0/libarchive/archive_read_support_compression_gzip.c	2009-01-02 20:32:14.000000000 -0600
@@ -432,6 +432,7 @@
 				}
 				/* Restart header parser with the next block. */
 				state->header_state = state->header_done = 0;
+				state->eof = 1;
 				/* FALL THROUGH */
 			case Z_OK: /* Decompressor made some progress. */
 				/* If we filled our buffer, update stats and return. */
