diff --git i/js/misc/weather.js w/js/misc/weather.js
index aaec36cf4..3b09c28c1 100644
--- i/js/misc/weather.js
+++ w/js/misc/weather.js
@@ -183,10 +183,8 @@ var WeatherClient = class {
 
     _onGClueLocationChanged() {
         let geoLocation = this._gclueService.location;
-        let location = GWeather.Location.new_detached(geoLocation.description,
-                                                      null,
-                                                      geoLocation.latitude,
-                                                      geoLocation.longitude);
+        let location = this._world.find_nearest_city(geoLocation.latitude,
+                                                     geoLocation.longitude);
         this._setLocation(location);
     }
 
