--- phonon-4.6.0/includes/CMakeLists.txt~	2013-02-25 18:42:31.962003363 +0000
+++ phonon-4.6.0/includes/CMakeLists.txt	2013-02-25 18:42:44.505244982 +0000
@@ -51,7 +51,7 @@
     phonon/VolumeFaderEffect
     phonon/VolumeFaderInterface
     phonon/VolumeSlider
-DESTINATION ${INCLUDE_INSTALL_DIR}/phonon COMPONENT Devel)
+DESTINATION ${INCLUDE_INSTALL_DIR}/qt4/phonon COMPONENT Devel)
 
 
 install( FILES
--- phonon-4.6.0/phonon/CMakeLists.txt~	2013-02-25 18:42:52.168522284 +0000
+++ phonon-4.6.0/phonon/CMakeLists.txt	2013-02-25 18:43:12.821704615 +0000
@@ -155,7 +155,7 @@
    volumefaderinterface.h
    volumeslider.h
    ${CMAKE_CURRENT_BINARY_DIR}/phononnamespace.h
-   DESTINATION ${INCLUDE_INSTALL_DIR}/phonon COMPONENT Devel)
+   DESTINATION ${INCLUDE_INSTALL_DIR}/qt4/phonon COMPONENT Devel)
 
 if (QT_QTDBUS_FOUND AND NOT PHONON_NO_DBUS)
     install(FILES org.kde.Phonon.AudioOutput.xml  DESTINATION  ${DBUS_INTERFACES_INSTALL_DIR})
