--- phonon-4.6.0/includes/CMakeLists.txt~	2013-02-25 18:50:09.158434991 +0000
+++ phonon-4.6.0/includes/CMakeLists.txt	2013-02-25 18:50:23.251657541 +0000
@@ -51,7 +51,7 @@
     phonon/VolumeFaderEffect
     phonon/VolumeFaderInterface
     phonon/VolumeSlider
-DESTINATION ${INCLUDE_INSTALL_DIR}/phonon COMPONENT Devel)
+DESTINATION ${INCLUDE_INSTALL_DIR}/qt4/phonon COMPONENT Devel)
 
 
 install( FILES
--- phonon-4.6.0/phonon/CMakeLists.txt~	2013-02-25 18:50:28.991612426 +0000
+++ phonon-4.6.0/phonon/CMakeLists.txt	2013-02-25 18:52:56.570453620 +0000
@@ -155,7 +155,7 @@
    volumefaderinterface.h
    volumeslider.h
    ${CMAKE_CURRENT_BINARY_DIR}/phononnamespace.h
-   DESTINATION ${INCLUDE_INSTALL_DIR}/phonon COMPONENT Devel)
+   DESTINATION ${INCLUDE_INSTALL_DIR}/qt4/phonon COMPONENT Devel)
 
 if (QT_QTDBUS_FOUND AND NOT PHONON_NO_DBUS)
     install(FILES org.kde.Phonon.AudioOutput.xml  DESTINATION  ${DBUS_INTERFACES_INSTALL_DIR})
--- phonon-4.6.0/phonon/experimental/CMakeLists.txt~	2013-02-25 18:51:08.734633479 +0000
+++ phonon-4.6.0/phonon/experimental/CMakeLists.txt	2013-02-25 18:51:23.997846924 +0000
@@ -55,4 +55,4 @@
     videowidget.h
     visualization.h
     globalconfig.h
-    DESTINATION ${INCLUDE_INSTALL_DIR}/phonon/experimental COMPONENT Devel)
+    DESTINATION ${INCLUDE_INSTALL_DIR}/qt4/phonon/experimental COMPONENT Devel)
