diff --git a/test/with_dummyserver/test_https.py b/test/with_dummyserver/test_https.py
index b322c75c..a54e70e6 100644
--- a/test/with_dummyserver/test_https.py
+++ b/test/with_dummyserver/test_https.py
@@ -155,6 +155,7 @@ class TestHTTPS(HTTPSDummyServerTestCase):
                     # Python 3.7.4+
                     or "WSAECONNRESET" in str(e)  # Windows
                     or "EPIPE" in str(e)  # macOS
+                    or "ECONNRESET" in str(e)
                 ):
                     raise