diff -up ./plugin/icedteanp/IcedTeaNPPlugin.cc.sav ./plugin/icedteanp/IcedTeaNPPlugin.cc
--- ./plugin/icedteanp/IcedTeaNPPlugin.cc.sav	2011-11-08 17:27:29.046520865 -0500
+++ ./plugin/icedteanp/IcedTeaNPPlugin.cc	2011-11-08 17:27:40.469611421 -0500
@@ -2262,7 +2262,7 @@ NP_Initialize (NPNetscapeFuncs* browserT
 
 // Returns a string describing the MIME type that this plugin
 // handles.
-char*
+const char*
 NP_GetMIMEDescription ()
 {
   PLUGIN_DEBUG ("NP_GetMIMEDescription\n");
