From 034da0871ad78629f9d0df618be494dbb19157e4 Mon Sep 17 00:00:00 2001
From: Debarshi Ray <debarshir@gnome.org>
Date: Tue, 18 Jun 2013 10:35:42 +0000
Subject: imap-smtp: Prevent error messages from expanding the dialog

---
diff --git a/src/goabackend/goaimapsmtpprovider.c b/src/goabackend/goaimapsmtpprovider.c
index d10a9d8..841327d 100644
--- a/src/goabackend/goaimapsmtpprovider.c
+++ b/src/goabackend/goaimapsmtpprovider.c
@@ -669,6 +669,7 @@ create_account_details_ui (GoaProvider    *provider,
 
   data->cluebar_label = gtk_label_new ("");
   gtk_label_set_line_wrap (GTK_LABEL (data->cluebar_label), TRUE);
+  gtk_label_set_max_width_chars (GTK_LABEL (data->cluebar_label), 36);
   gtk_container_add (GTK_CONTAINER (gtk_info_bar_get_content_area (GTK_INFO_BAR (data->cluebar))),
                      data->cluebar_label);
 
--
cgit v0.9.2
