From 5ddbfecdcc1cb300072572775fafbec9084f422d Mon Sep 17 00:00:00 2001
From: Sean Davis <smd.seandavis@gmail.com>
Date: Sun, 1 Mar 2020 07:35:07 -0500
Subject: [PATCH] Improve contrast in weather report, similar to GTK2 + Adwaita
 (bug #15806)

---
 panel-plugin/weather-summary.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/panel-plugin/weather-summary.c b/panel-plugin/weather-summary.c
index e66abf0..afce939 100644
--- a/panel-plugin/weather-summary.c
+++ b/panel-plugin/weather-summary.c
@@ -947,13 +947,17 @@ make_forecast(plugin_data *data)
     GtkCssProvider *provider = gtk_css_provider_new ();
     gchar *css_string;
 
-    css_string = g_strdup (".forecast-cell.lightbg { background-color: rgba(0, 0, 0, 0.2); }"
-                           ".forecast-cell.darkbg { background-color: rgba(0, 0, 0, 0.4); }");
+    GtkStyleContext *ctx;
+
+    css_string = g_strdup (".forecast-cell.lightbg { background-color: rgba(0, 0, 0, 0.05); }"
+                           ".forecast-cell.darkbg { background-color: rgba(0, 0, 0, 0.6); }");
 
     gtk_css_provider_load_from_data (provider, css_string, -1, NULL);
     gtk_style_context_add_provider_for_screen (screen, GTK_STYLE_PROVIDER (provider), GTK_STYLE_PROVIDER_PRIORITY_APPLICATION);
 
     grid = gtk_grid_new ();
+    ctx = gtk_widget_get_style_context (GTK_WIDGET (grid));
+    gtk_style_context_add_class (ctx, "background");
 
     gtk_grid_set_row_spacing(GTK_GRID (grid), 0);
     gtk_grid_set_column_spacing(GTK_GRID (grid), 0);
-- 
GitLab

