diff -aur avahi-0.6.27/avahi-gobject/AvahiCore-0.6.gir avahi-0.6.27-introspection/avahi-gobject/AvahiCore-0.6.gir
--- avahi-0.6.27/avahi-gobject/AvahiCore-0.6.gir	2010-06-27 02:38:43.018291433 +0200
+++ avahi-0.6.27-introspection/avahi-gobject/AvahiCore-0.6.gir	2010-09-30 16:16:41.260000339 +0200
@@ -1,10 +1,12 @@
 <?xml version="1.0"?>
-<repository version="1.0"
+<repository version="1.2"
             xmlns="http://www.gtk.org/introspection/core/1.0"
             xmlns:c="http://www.gtk.org/introspection/c/1.0"
             xmlns:glib="http://www.gtk.org/introspection/glib/1.0">
   <namespace name="AvahiCore" version="0.6" shared-library="avahi-core">
-    <alias name="IfIndex" target="int" c:type="AvahiIfIndex"/>
+    <alias name="IfIndex" target="int">
+      <type name="AvahiIfIndex" c:type="AvahiIfIndex"/>
+    </alias>
     <enumeration name="Protocol" c:type="AvahiProtocol">
       <member name="inet"
               value="0"
diff -aur avahi-0.6.27/configure.ac avahi-0.6.27-introspection/configure.ac
--- avahi-0.6.27/configure.ac	2010-07-13 05:05:05.464755077 +0200
+++ avahi-0.6.27-introspection/configure.ac	2010-09-30 16:17:06.103333559 +0200
@@ -472,7 +472,7 @@
 #
 # Introspection support.
 #
-GOBJECT_INTROSPECTION_CHECK([0.6.7])
+GOBJECT_INTROSPECTION_CHECK([0.9.5])
 
 #
 # Check for Qt 3
