diff --git a/cmake/modules/FindNetworkManager.cmake b/cmake/modules/FindNetworkManager.cmake
index ced2017c..a82d43da 100644
--- a/cmake/modules/FindNetworkManager.cmake
+++ b/cmake/modules/FindNetworkManager.cmake
@@ -14,29 +14,27 @@
 # For details see the accompanying COPYING-CMAKE-SCRIPTS file.
 
 
-IF (NETWORKMANAGER_INCLUDE_DIRS AND NM-UTIL_INCLUDE_DIRS)
+IF (NETWORKMANAGER_INCLUDE_DIRS)
    # in cache already
    SET(NetworkManager_FIND_QUIETLY TRUE)
-ENDIF (NETWORKMANAGER_INCLUDE_DIRS AND NM-UTIL_INCLUDE_DIRS)
+ENDIF (NETWORKMANAGER_INCLUDE_DIRS)
 
 IF (NOT WIN32)
    # use pkg-config to get the directories and then use these values
    # in the FIND_PATH() and FIND_LIBRARY() calls
    find_package(PkgConfig)
-   PKG_SEARCH_MODULE( NETWORKMANAGER NetworkManager )
-   PKG_SEARCH_MODULE( NM-UTIL libnm-util )
+   PKG_SEARCH_MODULE( NETWORKMANAGER libnm )
 ENDIF (NOT WIN32)
 
-IF (NETWORKMANAGER_FOUND AND NM-UTIL_FOUND)
+IF (NETWORKMANAGER_FOUND)
    IF (NOT NetworkManager_FIND_QUIETLY)
       MESSAGE(STATUS "Found NetworkManager ${NETWORKMANAGER_VERSION}: ${NETWORKMANAGER_LIBRARY_DIRS}")
-      MESSAGE(STATUS "Found libnm-util: ${NM-UTIL_LIBRARY_DIRS}")
    ENDIF (NOT NetworkManager_FIND_QUIETLY)
-ELSE (NETWORKMANAGER_FOUND AND NM-UTIL_FOUND)
+ELSE (NETWORKMANAGER_FOUND)
    IF (NetworkManager_FIND_REQUIRED)
-      MESSAGE(FATAL_ERROR "Could NOT find NetworkManager or libnm-util, check FindPkgConfig output above!")
+      MESSAGE(FATAL_ERROR "Could NOT find NetworkManager, check FindPkgConfig output above!")
    ENDIF (NetworkManager_FIND_REQUIRED)
-ENDIF (NETWORKMANAGER_FOUND AND NM-UTIL_FOUND)
+ENDIF (NETWORKMANAGER_FOUND)
 
-MARK_AS_ADVANCED(NETWORKMANAGER_INCLUDE_DIRS NM-UTIL_INCLUDE_DIRS)
+MARK_AS_ADVANCED(NETWORKMANAGER_INCLUDE_DIRS)
 
diff --git a/src/solid-networkstatus/kded/CMakeLists.txt b/src/solid-networkstatus/kded/CMakeLists.txt
index 9fa30651..e8d33d74 100644
--- a/src/solid-networkstatus/kded/CMakeLists.txt
+++ b/src/solid-networkstatus/kded/CMakeLists.txt
@@ -16,12 +16,10 @@ set_package_properties(NetworkManager PROPERTIES DESCRIPTION "The NetworkManager
                       )
 
 if(NETWORKMANAGER_FOUND)
-    if(NM_0_7)
-        message(STATUS "Adding in-process NetworkManager service-wart to kded module")
-        set(kded_networkstatus_PART_SRCS ${kded_networkstatus_PART_SRCS} networkmanagerstatus.cpp)
-        include_directories(${NETWORKMANAGER_INCLUDE_DIRS})
-        add_definitions(-DNM_BACKEND_ENABLED)
-    endif()
+    message(STATUS "Adding in-process NetworkManager service-wart to kded module")
+    set(kded_networkstatus_PART_SRCS ${kded_networkstatus_PART_SRCS} networkmanagerstatus.cpp)
+    include_directories(${NETWORKMANAGER_INCLUDE_DIRS})
+    add_definitions(-DNM_BACKEND_ENABLED)
 endif()
 
 
diff --git a/src/solid-networkstatus/kded/networkmanagerstatus.h b/src/solid-networkstatus/kded/networkmanagerstatus.h
index bbe0879a..2104abd8 100644
--- a/src/solid-networkstatus/kded/networkmanagerstatus.h
+++ b/src/solid-networkstatus/kded/networkmanagerstatus.h
@@ -38,7 +38,7 @@ public:
     /* reimp */ bool isSupported() const;
     /* reimp */ QString serviceName() const;
 
-private slots:
+private Q_SLOTS:
     void nmStateChanged( uint nmState );
 
 private:
