diff -Naur kismet-2007-01-R1b-old/ringbuf.cc kismet-2007-01-R1b/ringbuf.cc
--- kismet-2007-01-R1b-old/ringbuf.cc	2005-12-12 05:14:39.000000000 +1000
+++ kismet-2007-01-R1b/ringbuf.cc	2008-08-16 22:49:23.000000000 +1000
@@ -17,6 +17,7 @@
 */
 
 #include "ringbuf.h"
+#include <cstring>
 
 RingBuffer::RingBuffer(int in_size) {
     ring_len = in_size;
diff -Naur kismet-2007-01-R1b-old/util.cc kismet-2007-01-R1b/util.cc
--- kismet-2007-01-R1b-old/util.cc	2006-08-29 06:37:29.000000000 +1000
+++ kismet-2007-01-R1b/util.cc	2008-08-16 22:47:57.000000000 +1000
@@ -22,6 +22,7 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <fcntl.h>
+#include <cstring>
 
 // We need this to make uclibc happy since they don't even have rintf...
 #ifndef rintf
