diff --git a/cmake/modules/FindPyQt5.py b/cmake/modules/FindPyQt5.py
index 5849f40868..a42ba6c624 100644
--- a/cmake/modules/FindPyQt5.py
+++ b/cmake/modules/FindPyQt5.py
@@ -3,7 +3,7 @@
 # For details see the accompanying COPYING-CMAKE-SCRIPTS file.
 
 import PyQt5.Qt
-import sys
+import sys, site
 import os.path
 
 print("pyqt_version:%06.0x" % PyQt5.Qt.PYQT_VERSION)
@@ -30,7 +30,7 @@ except ValueError:
     pass
 
 # FIXME This next line is just a little bit too crude.
-pyqt_sip_dir = os.path.join(sys.prefix, "share", "sip", "PyQt5")
+pyqt_sip_dir = os.path.join(site.getsitepackages()[0], "PyQt5", "bindings")
 print("pyqt_sip_dir:%s" % pyqt_sip_dir)
 
 print("pyqt_sip_flags:%s" % PyQt5.Qt.PYQT_CONFIGURATION["sip_flags"])
