diff --git a/CMakeLists.txt b/CMakeLists.txt
index f88105d9dd..87b7f8e006 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -854,13 +854,19 @@ set_package_properties(Poppler PROPERTIES
 ##
 ## Test for quazip
 ##
-find_package(QuaZip 0.6)
-set_package_properties(QuaZip PROPERTIES
-    DESCRIPTION "A library for reading and writing zip files"
-    URL "https://stachenov.github.io/quazip/"
-    TYPE REQUIRED
-    PURPOSE "Needed for reading and writing KRA and ORA files"
-)
+find_package(QuaZip-Qt5)
+if(QuaZip-Qt5_FOUND)
+    set(QUAZIP_INCLUDE_DIRS QuaZip::QuaZip)
+    set(QUAZIP_LIBRARIES QuaZip::QuaZip)
+else()
+    find_package(QuaZip 0.6)
+    set_package_properties(QuaZip PROPERTIES
+        DESCRIPTION "A library for reading and writing zip files"
+        URL "https://stachenov.github.io/quazip/"
+        TYPE REQUIRED
+        PURPOSE "Needed for reading and writing KRA and ORA files"
+    )
+endif()
 
 # FIXME: better way to do this?
 list (APPEND ANDROID_EXTRA_LIBS ${QUAZIP_LIBRARIES}
