diff -u -r libpcap-0.6.2/Makefile.in libpcap-shared/Makefile.in
--- libpcap-0.6.2/Makefile.in	Thu Jan 18 05:06:24 2001
+++ libpcap-shared/Makefile.in	Fri Jul 20 16:26:05 2001
@@ -22,6 +22,10 @@
 #
 # Various configurable paths (remember to edit Makefile.in, not Makefile)
 #
+MAJOR = @MAJOR@
+MINOR = @MINOR@
+SUBMINOR = @SUBMINOR@
+VERSION = @VERSION@
 
 # Top level hierarchy
 prefix = @prefix@
@@ -47,7 +51,7 @@
 DEFS = @DEFS@
 
 # Standard CFLAGS
-CFLAGS = $(CCOPT) $(INCLS) $(DEFS)
+CFLAGS = $(CCOPT) $(INCLS) $(DEFS) -fPIC -DPIC
 
 INSTALL = @INSTALL@
 INSTALL_PROGRAM = @INSTALL_PROGRAM@
@@ -91,9 +95,14 @@
 TAGFILES = \
 	$(SRC) $(HDR) $(TAGHDR)
 
-CLEANFILES = $(OBJ) libpcap.a $(GENSRC) $(GENHDR) lex.yy.c
 
-all: libpcap.a
+
+CLEANFILES = $(OBJ) libpcap.a $(GENSRC) $(GENHDR) lex.yy.c libpcap.so.$(VERSION)
+
+all: libpcap.a libpcap.so.$(VERSION)
+
+libpcap.so.$(VERSION):
+	gcc -shared -Wl,-soname -Wl,libpcap.so.$(VERSION) -o libpcap.so.$(VERSION) $(OBJ) -lc
 
 libpcap.a: $(OBJ)
 	@rm -f $@
@@ -139,6 +148,10 @@
 	[ -d $(DESTDIR)$(libdir) ] || \
 	    (mkdir -p $(DESTDIR)$(libdir); chmod 755 $(DESTDIR)$(libdir))
 	$(INSTALL_DATA) libpcap.a $(DESTDIR)$(libdir)/libpcap.a
+	$(INSTALL_DATA) libpcap.so.$(VERSION)  $(DESTDIR)$(libdir)/libpcap.so.$(VERSION)
+	ln -s libpcap.so.$(VERSION) $(DESTDIR)$(libdir)/libpcap.so
+	ln -s libpcap.so.$(VERSION) $(DESTDIR)$(libdir)/libpcap.so.$(MAJOR)
+	ln -s libpcap.so.$(VERSION) $(DESTDIR)$(libdir)/libpcap.so.$(MAJOR).$(MINOR)
 	$(RANLIB) $(DESTDIR)$(libdir)/libpcap.a
 	[ -d $(DESTDIR)$(includedir) ] || \
 	    (mkdir -p $(DESTDIR)$(includedir); chmod 755 $(DESTDIR)$(includedir))
