From 7a31aae32f3638ec99f432795f5c721b60e67268 Mon Sep 17 00:00:00 2001
From: Frédéric Péters <fpeters@0d.be>
Date: Fri, 09 Oct 2009 18:07:20 +0000
Subject: Pass mailto: URI to gtk_show_uri()

(Part of) Bug #584537

Signed-off-by: Xan Lopez <xan@gnome.org>
---
diff --git a/embed/ephy-embed-utils.c b/embed/ephy-embed-utils.c
index d937d2e..eb7c434 100644
--- a/embed/ephy-embed-utils.c
+++ b/embed/ephy-embed-utils.c
@@ -92,6 +92,7 @@ ephy_embed_utils_address_has_web_scheme (const char *address)
 			  g_str_has_prefix (address, "data:") ||
 			  g_str_has_prefix (address, "about:") ||
 			  g_str_has_prefix (address, "javascript:") ||
+			  g_str_has_prefix (address, "mailto:") ||
 			  g_str_has_prefix (address, "gopher:"));
 
 	return has_web_scheme;
diff --git a/src/ephy-window.c b/src/ephy-window.c
index 59a06c7..61cc17c 100644
--- a/src/ephy-window.c
+++ b/src/ephy-window.c
@@ -2529,10 +2529,18 @@ policy_decision_required_cb (WebKitWebView *web_view,
 	WebKitWebNavigationReason reason;
 	gint button;
 	gint state;
+	const char *uri;
 
 	reason = webkit_web_navigation_action_get_reason (action);
 	button = webkit_web_navigation_action_get_button (action);
 	state = webkit_web_navigation_action_get_modifier_state (action);
+	uri = webkit_network_request_get_uri (request);
+
+	if (g_str_has_prefix (uri, "mailto:")) {
+		webkit_web_policy_decision_ignore (decision);
+		gtk_show_uri (NULL, uri, GDK_CURRENT_TIME, NULL);
+		return TRUE;
+	}
 
 	/* Open in new tab for middle click or ctrl+click */
 	if (reason == WEBKIT_WEB_NAVIGATION_REASON_LINK_CLICKED &&
--
cgit v0.8.2
