diff --git a/man/Makefile.am b/man/Makefile.am
index 7d45968..4d36ec4 100644
--- a/man/Makefile.am
+++ b/man/Makefile.am
@@ -33,6 +33,9 @@ CLEANFILES = $(appman_DATA)
 
 SUFFIXES = .$(APP_MAN_SUFFIX) .man
 
+# taken from manpages.am in xserver
+MAN_SUBSTS += 	-e 's|__datadir__|$(datadir)|g'
+
 # String replacements in MAN_SUBSTS now come from xorg-macros.m4 via configure
 .man.$(APP_MAN_SUFFIX):
 	$(AM_V_GEN)$(SED) $(MAN_SUBSTS) < $< > $@
diff --git a/man/twm.man b/man/twm.man
index cc8bc5f..d1dbaa0 100644
--- a/man/twm.man
+++ b/man/twm.man
@@ -132,7 +132,7 @@ differing visual types.
 .B "$HOME/.twmrc"
 This is the usual name for an individual user's startup file.
 .TP 8
-.B __projectroot__/lib/X11/twm/system.twmrc
+.B __datadir__/X11/twm/system.twmrc
 If neither of the preceding files are found, \fItwm\fP will look in this
 file for a
 default configuration.  This is often tailored by the site administrator to
