Index: desktop-menu-dentry.c
===================================================================
--- modules/menu/desktop-menu-dentry.c	(revision 26634)
+++ modules/menu/desktop-menu-dentry.c	(revision 26635)
@@ -173,6 +173,7 @@
         if(cmpname && g_ascii_strcasecmp(name, cmpname) < 0)
             break;
     }
+    g_list_free(items);
     
     gtk_menu_shell_insert(menu_shell, mi, i);
     
