--- a/src/common/mm_io.cpp	2012-07-08 20:06:02.000000000 +0200
+++ b/src/common/mm_io.cpp	2012-07-21 11:59:59.000000000 +0200
@@ -147,7 +147,7 @@
 void
 mm_file_io_c::prepare_path(const std::string &path) {
   boost::filesystem::path directory = boost::filesystem::path(path).parent_path();
-  if (boost::filesystem::exists(directory))
+  if (directory.empty() || boost::filesystem::exists(directory))
     return;
 
   boost::system::error_code error_code;
