diff -up mtr-0.83/configure.in.glib2 mtr-0.83/configure.in
--- mtr-0.83/configure.in.glib2	2013-02-18 20:10:00.767417356 +0100
+++ mtr-0.83/configure.in	2013-02-18 20:10:53.549513018 +0100
@@ -59,6 +59,8 @@ else
 	GTK_OBJ=""
 fi
 
+PKG_CHECK_MODULES([GLIB2], [glib-2.0])
+
 AC_CHECK_FUNC(socket, , 
   AC_CHECK_LIB(socket, socket, , AC_MSG_ERROR(No socket library found)))
 
diff -up mtr-0.83/Makefile.am.glib2 mtr-0.83/Makefile.am
--- mtr-0.83/Makefile.am.glib2	2013-02-18 20:13:54.635841512 +0100
+++ mtr-0.83/Makefile.am	2013-02-18 20:16:17.877101679 +0100
@@ -19,8 +19,9 @@ mtr_SOURCES = mtr.c \
               mtr-gtk.h
 EXTRA_mtr_SOURCES = curses.c \
                     gtk.c
+AM_CFLAGS = $(GLIB2_CFLAGS)
 mtr_DEPENDENCIES = $(GTK_OBJ) $(CURSES_OBJ)
-mtr_LDFLAGS = $(GTK_OBJ) $(CURSES_OBJ)
+mtr_LDFLAGS = $(GTK_OBJ) $(GLIB2_LIBS) $(CURSES_OBJ)
 
 EXTRA_DIST = SECURITY mtr.8 Makefile Makefile.dist
 distclean-local:
