--- ext/mysqlnd/mysqlnd.c	2009-11-19 14:02:01.934775092 +0100
+++ ext/mysqlnd/mysqlnd.c	2009-11-19 14:08:04.598072581 +0100
@@ -547,7 +547,7 @@
 #ifndef PHP_WIN32
 	if (host_len == sizeof("localhost") - 1 && !strncasecmp(host, "localhost", host_len)) {
 		if (!socket) {
-			socket = "/tmp/mysql.sock";
+			socket = "/var/run/mysqld/mysqld.sock";
 		}
 		transport_len = spprintf(&transport, 0, "unix://%s", socket);
 		unix_socket = TRUE;
--- ext/pdo_mysql/pdo_mysql.c	2009-10-14 15:51:25.000000000 +0200
+++ ext/pdo_mysql/pdo_mysql.c	2009-11-19 22:52:13.460628605 +0100
@@ -40,7 +40,7 @@
 
 #ifndef PHP_WIN32
 # ifndef PDO_MYSQL_UNIX_ADDR
-#  define PDO_MYSQL_UNIX_ADDR  "/tmp/mysql.sock"
+#  define PDO_MYSQL_UNIX_ADDR  "/var/run/mysqld/mysqld.sock"
 # endif
 #endif
 
--- ext/pdo_mysql/pdo_mysql.c	2009-10-14 16:51:25.000000000 +0300
+++ ext/pdo_mysql/pdo_mysql.c	2009-11-20 14:08:04.000000000 +0200
@@ -49,7 +49,7 @@ ZEND_DECLARE_MODULE_GLOBALS(pdo_mysql);
 */
 PHP_INI_BEGIN()
 #ifndef PHP_WIN32
-	STD_PHP_INI_ENTRY("pdo_mysql.default_socket", PDO_MYSQL_UNIX_ADDR, PHP_INI_SYSTEM, OnUpdateString, default_socket, zend_pdo_mysql_globals, pdo_mysql_globals)
+	STD_PHP_INI_ENTRY("pdo_mysql.default_socket", PDO_MYSQL_UNIX_ADDR, PHP_INI_SYSTEM, OnUpdateStringUnempty, default_socket, zend_pdo_mysql_globals, pdo_mysql_globals)
 #endif
 #if PDO_DBG_ENABLED
 	STD_PHP_INI_ENTRY("pdo_mysql.debug",	NULL, PHP_INI_SYSTEM, OnUpdateString, debug, zend_pdo_mysql_globals, pdo_mysql_globals)
