Index: src/trusted/service_runtime/linux/nacl_bootstrap.gyp
diff --git a/src/trusted/service_runtime/linux/nacl_bootstrap.gyp b/src/trusted/service_runtime/linux/nacl_bootstrap.gyp
index 66d95ca4dc98653219c965b899dbd5ad35d2960c..083e3fdd24ff4562d2037f8d17a06540e05357b8 100644
--- a/src/trusted/service_runtime/linux/nacl_bootstrap.gyp
+++ b/src/trusted/service_runtime/linux/nacl_bootstrap.gyp
@@ -84,6 +84,11 @@
         '-fasan',
         '-faddress-sanitizer',
         '-w',
+        # We filter these out because release_extra_cflags or another
+        # such thing might be adding them in, and those options wind up
+        # coming after the -fno-stack-protector we added above.
+        '-fstack-protector',
+        '-fstack-protector-all',
       ],
       'conditions': [
         ['clang==1', {
