--- libmailwatch-core/mailwatch-net-conn.c.orig	2012-04-29 05:38:20.015564589 +0300
+++ libmailwatch-core/mailwatch-net-conn.c	2012-04-29 05:39:01.943290639 +0300
@@ -621,8 +621,10 @@ xfce_mailwatch_net_conn_make_secure(Xfce
                            net_conn->gt_creds);
     gnutls_transport_set_ptr(net_conn->gt_session,
                              (gnutls_transport_ptr_t)net_conn->fd);
+#if GNUTLS_VERSION_NUMBER < 0x020c00
     if(fcntl(net_conn->fd, F_GETFL) & O_NONBLOCK)
         gnutls_transport_set_lowat(net_conn->gt_session, 0);
+#endif
     
     if(!xfce_mailwatch_net_conn_tls_handshake(net_conn, error)) {
 #if 0
