--- openssl-0.9.8m/apps/CA.pl.in	2006-04-28 02:28:51.000000000 +0200
+++ openssl-0.9.8m/apps/CA.pl.in	2010-02-26 17:34:10.776962626 +0100
@@ -53,7 +53,7 @@
 $X509="$openssl x509";
 $PKCS12="$openssl pkcs12";
 
-$CATOP="./demoCA";
+$CATOP="/etc/ssl";
 $CAKEY="cakey.pem";
 $CAREQ="careq.pem";
 $CACERT="cacert.pem";
--- openssl-0.9.8m/apps/CA.sh	2009-10-15 19:28:02.000000000 +0200
+++ openssl-0.9.8m/apps/CA.sh	2010-02-26 17:33:21.897801410 +0100
@@ -68,7 +68,7 @@
 X509="$OPENSSL x509"
 PKCS12="openssl pkcs12"
 
-if [ -z "$CATOP" ] ; then CATOP=./demoCA ; fi
+if [ -z "$CATOP" ] ; then CATOP=/etc/ssl ; fi
 CAKEY=./cakey.pem
 CAREQ=./careq.pem
 CACERT=./cacert.pem
--- openssl-0.9.8m/apps/openssl.cnf	2005-09-16 14:20:24.000000000 +0200
+++ openssl-0.9.8m/apps/openssl.cnf	2010-02-26 17:32:31.516971476 +0100
@@ -34,7 +34,7 @@
 ####################################################################
 [ CA_default ]
 
-dir		= ./demoCA		# Where everything is kept
+dir		= /etc/ssl		# Where everything is kept
 certs		= $dir/certs		# Where the issued certs are kept
 crl_dir		= $dir/crl		# Where the issued crl are kept
 database	= $dir/index.txt	# database index file.
