diff -Naur old/configure.ac new/configure.ac
--- old/configure.ac	2013-11-23 07:08:24.000000000 -1000
+++ new/configure.ac	2013-11-26 17:05:23.108828275 -1000
@@ -749,6 +749,7 @@
 if test x$enable_cdio_paranoia = xyes; then
 	AC_DEFINE([ENABLE_CDIO_PARANOIA], 1,
 		[Define to enable libcdio_paranoia support])
+	AC_CHECK_HEADERS(cdio/paranoia/paranoia.h)
 fi
 
 AM_CONDITIONAL(ENABLE_CDIO_PARANOIA, test x$enable_cdio_paranoia = xyes)
diff -Naur old/src/input/CdioParanoiaInputPlugin.cxx new/src/input/CdioParanoiaInputPlugin.cxx
--- old/src/input/CdioParanoiaInputPlugin.cxx	2013-11-18 12:50:46.000000000 -1000
+++ new/src/input/CdioParanoiaInputPlugin.cxx	2013-11-26 17:05:35.152212566 -1000
@@ -41,7 +41,12 @@
 #include <glib.h>
 #include <assert.h>
 
+#ifdef HAVE_CDIO_PARANOIA_PARANOIA_H
+#include <cdio/parannoia/paranoia.h>
+#else
 #include <cdio/paranoia.h>
+#endif
+
 #include <cdio/cd_types.h>
 
 struct CdioParanoiaInputStream {
