diff --git a/CMakeLists.txt b/CMakeLists.txt
index 726436b..8e3a5c2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -75,7 +75,7 @@ include(CTest)
 
 configure_file("libcerf.pc.in" "libcerf.pc" @ONLY)
 install(FILES "${CMAKE_CURRENT_BINARY_DIR}/libcerf.pc"
-    DESTINATION "${CMAKE_INSTALL_PREFIX}/pkgconfig/")
+    DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig/")
 
 add_subdirectory(lib)
 add_subdirectory(test)
