diff --git c/src/pip/_vendor/__init__.py w/src/pip/_vendor/__init__.py
index 3843cb099..39579aa96 100644
--- c/src/pip/_vendor/__init__.py
+++ w/src/pip/_vendor/__init__.py
@@ -106,6 +106,16 @@ if DEBUNDLED:
     vendored("requests.packages.urllib3.util.timeout")
     vendored("requests.packages.urllib3.util.url")
     vendored("resolvelib")
+    vendored("rich")
+    vendored("rich.console")
+    vendored("rich.highlighter")
+    vendored("rich.logging")
+    vendored("rich.markup")
+    vendored("rich.progress")
+    vendored("rich.segment")
+    vendored("rich.style")
+    vendored("rich.text")
+    vendored("rich.traceback")
     vendored("tenacity")
     vendored("tomli")
     vendored("urllib3")
