 embed/ephy-embed-shell.c | 1 +
 meson.build              | 1 +
 2 files changed, 2 insertions(+)

diff --git c/embed/ephy-embed-shell.c i/embed/ephy-embed-shell.c
index b4837f834..1b5012bae 100644
--- c/embed/ephy-embed-shell.c
+++ i/embed/ephy-embed-shell.c
@@ -910,6 +910,7 @@ ephy_embed_shell_create_web_context (EphyEmbedShell *shell)
   g_free (cache_dir);
 
   priv->web_context = webkit_web_context_new_with_website_data_manager (manager);
+  webkit_web_context_set_additional_plugins_directory (priv->web_context, EPHY_PLUGINS_DIR);
   g_object_unref (manager);
 }
 
diff --git c/meson.build i/meson.build
index a4a8a36b7..0c98f6c36 100644
--- c/meson.build
+++ i/meson.build
@@ -25,6 +25,7 @@ conf = configuration_data()
 conf.set_quoted('BUILD_ROOT', meson.build_root())
 conf.set_quoted('DISTRIBUTOR_NAME', get_option('distributor_name'))
 conf.set_quoted('EPHY_WEB_EXTENSIONS_DIR', webextensionsdir)
+conf.set_quoted('EPHY_PLUGINS_DIR', join_paths(pkglibdir, 'plugins'))
 conf.set_quoted('GETTEXT_PACKAGE', meson.project_name())
 conf.set_quoted('ISO_CODES_PREFIX', iso_codes_prefix)
 conf.set_quoted('LOCALEDIR', localedir)
