diff -ur poppler-0.6.1.org/poppler/Makefile.in poppler-0.6.1.new/poppler/Makefile.in
--- poppler-0.6.1.org/poppler/Makefile.in	2007-10-12 05:48:31.000000000 -0600
+++ poppler-0.6.1.new/poppler/Makefile.in	2007-10-17 12:19:20.000000000 -0600
@@ -140,6 +140,7 @@
 	$(am__libpoppler_la_SOURCES_DIST)
 am__poppler_include_HEADERS_DIST = SplashOutputDev.h Annot.h Array.h \
 	BaseFile.h BuiltinFont.h BuiltinFontTables.h Catalog.h \
+	CairoFontEngine.h CairoOutputDev.h \
 	CharCodeToUnicode.h CMap.h Decrypt.h Dict.h Error.h \
 	FontEncodingTables.h FontInfo.h Form.h Function.cc Function.h \
 	Gfx.h GfxFont.h GfxState.h GlobalParams.h JArithmeticDecoder.h \
@@ -418,6 +419,8 @@
 @ENABLE_XPDF_HEADERS_TRUE@	BuiltinFont.h		\
 @ENABLE_XPDF_HEADERS_TRUE@	BuiltinFontTables.h	\
 @ENABLE_XPDF_HEADERS_TRUE@	Catalog.h		\
+@ENABLE_XPDF_HEADERS_TRUE@	CairoFontEngine.h	\
+@ENABLE_XPDF_HEADERS_TRUE@	CairoOutputDev.h	\
 @ENABLE_XPDF_HEADERS_TRUE@	CharCodeToUnicode.h	\
 @ENABLE_XPDF_HEADERS_TRUE@	CMap.h			\
 @ENABLE_XPDF_HEADERS_TRUE@	Decrypt.h		\
