--- Lib/idlelib/WidgetRedirector.py.orig	2008-04-23 23:39:57.000000000 -0700
+++ Lib/idlelib/WidgetRedirector.py	2008-04-23 23:46:55.000000000 -0700
@@ -53,7 +53,13 @@
             if m:
                 return m(*args)
             else:
-                return self.tk.call((self.orig, cmd) + args)
+                #return self.tk.call((self.orig, cmd) + args)
+                r = self.tk.call((self.orig, cmd) + args)
+                if isinstance(r, tuple):
+                    # convert to string ourselves so we get a Tcl list
+                    # that can be converted back into a tuple
+                    r = ' '.join(map(str, r))
+                return r
         except TclError:
             return ""
 
