diff --git i/CMakeLists.txt w/CMakeLists.txt
index 98642e1..b4295fd 100644
--- i/CMakeLists.txt
+++ w/CMakeLists.txt
@@ -188,7 +188,7 @@ install(FILES dwarves.h dwarves_emit.h dwarves_reorganize.h
 install(FILES man-pages/pahole.1 DESTINATION ${CMAKE_INSTALL_PREFIX}/share/man/man1/)
 if(Python3_FOUND)
 	install(PROGRAMS ostra/ostra-cg DESTINATION ${CMAKE_INSTALL_PREFIX}/bin)
-	install(FILES ostra/python/ostra.py DESTINATION ${CMAKE_INSTALL_PREFIX}/share/dwarves/runtime/python)
+	install(FILES ostra/python/ostra.py DESTINATION ${Python3_SITELIB})
 endif()
 install(PROGRAMS btfdiff fullcircle DESTINATION ${CMAKE_INSTALL_PREFIX}/bin)
 install(FILES lib/Makefile lib/ctracer_relay.c lib/ctracer_relay.h lib/linux.blacklist.cu
