diff -u -r inkscape-0.92.5/share/extensions/uniconv-ext.py inkscape-0.92.5-python2/share/extensions/uniconv-ext.py
--- inkscape-0.92.5/share/extensions/uniconv-ext.py	2020-04-09 21:37:16.000000000 +0000
+++ inkscape-0.92.5-python2/share/extensions/uniconv-ext.py	2020-04-13 13:46:07.038597950 +0000
@@ -58,7 +58,7 @@
                      'https://sk1project.net/modules.php?name=Products&product=uniconvertor&op=download\n'+\
                      'and install into your Inkscape\'s Python location\n'))
         sys.exit(1)
-    cmd = 'python -c "import uniconvertor; uniconvertor.uniconv_run()"'
+    cmd = 'python2 -c "import uniconvertor; uniconvertor.uniconv_run()"'
 
 run((cmd+' "%s" "%%s"') % sys.argv[1].replace("%","%%"), "UniConvertor")
 
diff -u -r inkscape-0.92.5/share/extensions/uniconv_output.py inkscape-0.92.5-python2/share/extensions/uniconv_output.py
--- inkscape-0.92.5/share/extensions/uniconv_output.py	2020-04-09 21:37:16.000000000 +0000
+++ inkscape-0.92.5-python2/share/extensions/uniconv_output.py	2020-04-13 13:46:14.225378534 +0000
@@ -126,7 +126,7 @@
                          'https://sk1project.net/modules.php?name=Products&product=uniconvertor&op=download\n'+\
                          'and install into your Inkscape\'s Python location\n'))
             sys.exit(1)
-        cmd = 'python -c "import uniconvertor; uniconvertor.uniconv_run();"'
+        cmd = 'python2 -c "import uniconvertor; uniconvertor.uniconv_run();"'
 
     return cmd
 
diff -u -r inkscape-0.92.5/share/filters/CMakeLists.txt inkscape-0.92.5-python2/share/filters/CMakeLists.txt
--- inkscape-0.92.5/share/filters/CMakeLists.txt	2020-04-09 21:37:16.000000000 +0000
+++ inkscape-0.92.5-python2/share/filters/CMakeLists.txt	2020-04-13 13:52:42.971451750 +0000
@@ -1,6 +1,6 @@
 add_custom_command(
   OUTPUT ${CMAKE_CURRENT_SOURCE_DIR}/filters.svg.h
-  COMMAND python ${CMAKE_CURRENT_SOURCE_DIR}/i18n.py ${CMAKE_CURRENT_SOURCE_DIR}/filters.svg > ${CMAKE_CURRENT_SOURCE_DIR}/filters.svg.h
+  COMMAND python2 ${CMAKE_CURRENT_SOURCE_DIR}/i18n.py ${CMAKE_CURRENT_SOURCE_DIR}/filters.svg > ${CMAKE_CURRENT_SOURCE_DIR}/filters.svg.h
   MAIN_DEPENDENCY ${CMAKE_CURRENT_SOURCE_DIR}/filters.svg
   DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/i18n.py
 )
diff -u -r inkscape-0.92.5/share/palettes/CMakeLists.txt inkscape-0.92.5-python2/share/palettes/CMakeLists.txt
--- inkscape-0.92.5/share/palettes/CMakeLists.txt	2020-04-09 21:37:16.000000000 +0000
+++ inkscape-0.92.5-python2/share/palettes/CMakeLists.txt	2020-04-13 13:52:27.677882059 +0000
@@ -2,7 +2,7 @@
 
 add_custom_command(
   OUTPUT ${CMAKE_CURRENT_SOURCE_DIR}/palettes.h
-  COMMAND python ${CMAKE_CURRENT_SOURCE_DIR}/i18n.py ${I18N_FILES} > ${CMAKE_CURRENT_SOURCE_DIR}/palettes.h
+  COMMAND python2 ${CMAKE_CURRENT_SOURCE_DIR}/i18n.py ${I18N_FILES} > ${CMAKE_CURRENT_SOURCE_DIR}/palettes.h
   WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
   DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/i18n.py ${I18N_FILES}
 )
diff -u -r inkscape-0.92.5/share/patterns/CMakeLists.txt inkscape-0.92.5-python2/share/patterns/CMakeLists.txt
--- inkscape-0.92.5/share/patterns/CMakeLists.txt	2020-04-09 21:37:16.000000000 +0000
+++ inkscape-0.92.5-python2/share/patterns/CMakeLists.txt	2020-04-13 13:52:30.824597371 +0000
@@ -1,6 +1,6 @@
 add_custom_command(
   OUTPUT ${CMAKE_CURRENT_SOURCE_DIR}/patterns.svg.h
-  COMMAND python ${CMAKE_CURRENT_SOURCE_DIR}/i18n.py ${CMAKE_CURRENT_SOURCE_DIR}/patterns.svg > ${CMAKE_CURRENT_SOURCE_DIR}/patterns.svg.h
+  COMMAND python2 ${CMAKE_CURRENT_SOURCE_DIR}/i18n.py ${CMAKE_CURRENT_SOURCE_DIR}/patterns.svg > ${CMAKE_CURRENT_SOURCE_DIR}/patterns.svg.h
   MAIN_DEPENDENCY ${CMAKE_CURRENT_SOURCE_DIR}/patterns.svg
   DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/i18n.py
 )
diff -u -r inkscape-0.92.5/share/symbols/CMakeLists.txt inkscape-0.92.5-python2/share/symbols/CMakeLists.txt
--- inkscape-0.92.5/share/symbols/CMakeLists.txt	2020-04-09 21:37:16.000000000 +0000
+++ inkscape-0.92.5-python2/share/symbols/CMakeLists.txt	2020-04-13 13:52:33.297968934 +0000
@@ -2,7 +2,7 @@
 
 add_custom_command(
   OUTPUT ${CMAKE_CURRENT_SOURCE_DIR}/symbols.h
-  COMMAND python ${CMAKE_CURRENT_SOURCE_DIR}/i18n.py ${_FILES} > ${CMAKE_CURRENT_SOURCE_DIR}/symbols.h
+  COMMAND python2 ${CMAKE_CURRENT_SOURCE_DIR}/i18n.py ${_FILES} > ${CMAKE_CURRENT_SOURCE_DIR}/symbols.h
   WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
   DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/i18n.py ${_FILES}
 )
diff -u -r inkscape-0.92.5/share/templates/CMakeLists.txt inkscape-0.92.5-python2/share/templates/CMakeLists.txt
--- inkscape-0.92.5/share/templates/CMakeLists.txt	2020-04-09 21:37:16.000000000 +0000
+++ inkscape-0.92.5-python2/share/templates/CMakeLists.txt	2020-04-13 13:52:37.261363522 +0000
@@ -2,7 +2,7 @@
 
 add_custom_command(
   OUTPUT ${CMAKE_CURRENT_SOURCE_DIR}/templates.h
-  COMMAND python ${CMAKE_CURRENT_SOURCE_DIR}/i18n.py ${_FILES} > ${CMAKE_CURRENT_SOURCE_DIR}/templates.h
+  COMMAND python2 ${CMAKE_CURRENT_SOURCE_DIR}/i18n.py ${_FILES} > ${CMAKE_CURRENT_SOURCE_DIR}/templates.h
   WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
   DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/i18n.py ${_FILES}
 )
diff -u -r inkscape-0.92.5/src/extension/implementation/script.cpp inkscape-0.92.5-python2/src/extension/implementation/script.cpp
--- inkscape-0.92.5/src/extension/implementation/script.cpp	2020-04-13 13:44:57.737495376 +0000
+++ inkscape-0.92.5-python2/src/extension/implementation/script.cpp	2020-04-13 13:45:24.331252716 +0000
@@ -88,7 +88,7 @@
         {"python", "python-interpreter", "pythonw" },
 #else
         {"perl",   "perl-interpreter",   "perl"   },
-        {"python", "python-interpreter", "python" },
+        {"python", "python-interpreter", "python2" },
 #endif
         {"ruby",   "ruby-interpreter",   "ruby"   },
         {"shell",  "shell-interpreter",  "sh"     },
diff -u -r inkscape-0.92.5/src/main.cpp inkscape-0.92.5-python2/src/main.cpp
--- inkscape-0.92.5/src/main.cpp	2020-04-09 21:37:16.000000000 +0000
+++ inkscape-0.92.5-python2/src/main.cpp	2020-04-13 13:46:56.519380781 +0000
@@ -569,11 +569,11 @@
     gchar const *path = g_getenv("PATH");
     gchar const *pythonpath = g_getenv("PYTHONPATH");
 
-    gchar *python = g_build_filename(exe, "python", NULL);
-    gchar *scripts = g_build_filename(exe, "python", "Scripts", NULL);
-    gchar *perl = g_build_filename(exe, "python", NULL);
-    gchar *pythonlib = g_build_filename(exe, "python", "Lib", NULL);
-    gchar *pythondll = g_build_filename(exe, "python", "DLLs", NULL);
+    gchar *python = g_build_filename(exe, "python2", NULL);
+    gchar *scripts = g_build_filename(exe, "python2", "Scripts", NULL);
+    gchar *perl = g_build_filename(exe, "python2", NULL);
+    gchar *pythonlib = g_build_filename(exe, "python2", "Lib", NULL);
+    gchar *pythondll = g_build_filename(exe, "python2", "DLLs", NULL);
 
     // Python 2.x needs short paths in PYTHONPATH.
     // Otherwise it doesn't work when Inkscape is installed in Unicode directories.
@@ -605,7 +605,7 @@
     printf("PATH = %s\n\n", g_getenv("PATH"));
     printf("PYTHONPATH = %s\n\n", g_getenv("PYTHONPATH"));
 
-    gchar *p = g_find_program_in_path("python");
+    gchar *p = g_find_program_in_path("python2");
     if (p) {
         printf("python in %s\n\n", p);
         g_free(p);
