diff -aur old/bin/Makefile.in new/bin/Makefile.in
--- old/bin/Makefile.in	2009-10-05 05:07:08.000000000 -0700
+++ new/bin/Makefile.in	2011-08-28 19:14:17.025674400 -0700
@@ -19,7 +19,7 @@
 VPATH =		@srcdir@
 top_srcdir =	@top_srcdir@
 
-SUBDIRS =	named rndc dig dnssec tests tools nsupdate \
+SUBDIRS =	named rndc dnssec tests tools nsupdate \
 		check confgen @PYTHON_TOOLS@ @PKCS11_TOOLS@
 TARGETS =
 
