diff --git a/symbols/fr b/symbols/fr
index 9c89db8..3898b1c 100644
--- a/symbols/fr
+++ b/symbols/fr
@@ -36,8 +36,6 @@ xkb_symbols "basic" {
     key <AB09>	{ [     colon,      slash, periodcentered,   division ]	};
     key <AB10>	{ [    exclam,    section, dead_belowdot, dead_abovedot ] };
 
-    include "kpdl(comma)"
-
     include "level3(ralt_switch)"
 };
 
diff --git a/symbols/it b/symbols/it
index 8a66a30..2673cf9 100644
--- a/symbols/it
+++ b/symbols/it
@@ -33,8 +33,6 @@ xkb_symbols "basic" {
 
     key <LSGT>	{ [      less,    greater,guillemotleft,guillemotright] };
 
-    include "kpdl(comma)"
-
     include "level3(ralt_switch)"
 };
 
diff --git a/symbols/nl b/symbols/nl
index 1dde405..76cb018 100644
--- a/symbols/nl
+++ b/symbols/nl
@@ -49,8 +49,6 @@ xkb_symbols "basic" {
 
     key <LSGT>	{ [bracketright, bracketleft,       bar,    brokenbar ]	};
 
-    include "kpdl(comma)"
-
     include "level3(ralt_switch)"
 };
 
