From e96f5349f905d5005bd29c5a4786c10fb86d1453 Mon Sep 17 00:00:00 2001
From: Richard Hughes <richard@hughsie.com>
Date: Tue, 29 Oct 2013 12:52:47 +0000
Subject: trivial: post release version bump


diff --git a/RELEASE b/RELEASE
index 0f5abf2..f87444e 100644
--- a/RELEASE
+++ b/RELEASE
@@ -2,9 +2,9 @@ UPower Release Notes
 
 1. Write NEWS entries for UPower in the same format as usual.
 
-git shortlog UPOWER_0_9_22.. | grep -i -v trivial | grep -v Merge > NEWS.new
+git shortlog UPOWER_0_99_0.. | grep -i -v trivial | grep -v Merge > NEWS.new
 
-Version 0.99.0
+Version 0.99.1
 ~~~~~~~~~~~~~~
 Released: 2013-xx-xx
 
@@ -13,8 +13,8 @@ Bugfixes:
 
 2. Commit changes to git:
 
-git commit -a -m "Released UPower 0.99.0"
-git tag -s -f -m "Released UPower 0.99.0" UPOWER_0_99_0
+git commit -a -m "Released UPower 0.99.1"
+git tag -s -f -m "Released UPower 0.99.1" UPOWER_0_99_1
 <wait>
 git push --tags
 git push
diff --git a/configure.ac b/configure.ac
index af41082..db25851 100644
--- a/configure.ac
+++ b/configure.ac
@@ -2,7 +2,7 @@ dnl Process this file with autoconf to produce a configure script.
 
 m4_define([up_major_version], [0])
 m4_define([up_minor_version], [99])
-m4_define([up_micro_version], [0])
+m4_define([up_micro_version], [1])
 m4_define([up_version],
           [up_major_version.up_minor_version.up_micro_version])
 
-- 
cgit v0.10.2

