--- prog/detect/sensors-detect.orig	2008-11-08 02:15:07.000000000 -0500
+++ prog/detect/sensors-detect	2008-11-08 02:35:27.000000000 -0500
@@ -5871,7 +5871,7 @@
     exit -1;
   }
 
-  if (-x "/sbin/service" && -f "/etc/init.d/lm_sensors" &&
+  if (-x "/sbin/service" && -f "/etc/rc.d/lm_sensors" &&
       -f "/var/lock/subsys/lm_sensors") {
     system("/sbin/service", "lm_sensors", "stop");
   }
@@ -6084,21 +6084,21 @@
     }
   }
 
-  my $have_sysconfig = -d '/etc/sysconfig';
-  printf "Do you want to \%s /etc/sysconfig/lm_sensors? (\%s): ",
-         (-e '/etc/sysconfig/lm_sensors' ? 'overwrite' : 'generate'),
+  my $have_sysconfig = -d '/etc/conf.d';
+  printf "Do you want to \%s /etc/conf.d/lm_sensors? (\%s): ",
+         (-e '/etc/conf.d/lm_sensors' ? 'overwrite' : 'generate'),
          ($have_sysconfig ? 'YES/no' : 'yes/NO');
   $_ = <STDIN>;
   if (($have_sysconfig and not m/^\s*n/i) or m/^\s*y/i) {
     unless ($have_sysconfig) {
-      mkdir('/etc/sysconfig', 0777)
-        or die "Sorry, can't create /etc/sysconfig ($!)";
+      mkdir('/etc/conf.d', 0777)
+        or die "Sorry, can't create /etc/conf.d ($!)";
     }
-    open(local *SYSCONFIG, ">/etc/sysconfig/lm_sensors")
-      or die "Sorry, can't create /etc/sysconfig/lm_sensors ($!)";
+    open(local *SYSCONFIG, ">/etc/conf.d/lm_sensors")
+      or die "Sorry, can't create /etc/conf.d/lm_sensors ($!)";
     print SYSCONFIG <<'EOT';
-#    /etc/sysconfig/lm_sensors - Defines modules loaded by
-#                                /etc/init.d/lm_sensors
+#    /etc/conf.d/lm_sensors - Defines modules loaded by
+#                                /etc/rc.d/lm_sensors
 #    Copyright (c) 1998 - 2001  Frodo Looijaard <frodol@dds.nl>
 #
 #    This program is free software; you can redistribute it and/or modify
@@ -6120,8 +6120,8 @@
 # See also the lm_sensors homepage at:
 #     http://www.lm-sensors.org/
 #
-# This file is used by /etc/init.d/lm_sensors and defines the modules to
-# be loaded/unloaded. This file is sourced into /etc/init.d/lm_sensors.
+# This file is used by /etc/rc.d/lm_sensors and defines the modules to
+# be loaded/unloaded. This file is sourced into /etc/rc.d/lm_sensors.
 #
 # The format of this file is a shell script that simply defines the modules
 # in order as normal variables with the special names:
@@ -6143,13 +6143,13 @@
     print SYSCONFIG $sysconfig;
     close(SYSCONFIG);
 
-    print "Copy prog/init/lm_sensors.init to /etc/init.d/lm_sensors\n".
+    print "Copy prog/init/lm_sensors.init to /etc/rc.d/lm_sensors\n".
           "for initialization at boot time.\n"
-      unless -f "/etc/init.d/lm_sensors";
+      unless -f "/etc/rc.d/lm_sensors";
 
-    if (-x "/sbin/insserv" && -f "/etc/init.d/lm_sensors") {
-      system("/sbin/insserv", "/etc/init.d/lm_sensors");
-    } elsif (-x "/sbin/chkconfig" && -f "/etc/init.d/lm_sensors") {
+    if (-x "/sbin/insserv" && -f "/etc/rc.d/lm_sensors") {
+      system("/sbin/insserv", "/etc/rc.d/lm_sensors");
+    } elsif (-x "/sbin/chkconfig" && -f "/etc/rc.d/lm_sensors") {
       system("/sbin/chkconfig", "lm_sensors", "on");
       if (-x "/sbin/service") {
         system("/sbin/service", "lm_sensors", "start");
