From 88df18345069c295e68d319606a11b7727b2bd4b Mon Sep 17 00:00:00 2001
From: Maxim Ermilov <zaspire@rambler.ru>
Date: Tue, 24 May 2011 22:46:47 +0000
Subject: shell-recorder: missing XFree

https://bugzilla.gnome.org/show_bug.cgi?id=650934
---
diff --git a/src/shell-recorder.c b/src/shell-recorder.c
index 6555187..d297923 100644
--- a/src/shell-recorder.c
+++ b/src/shell-recorder.c
@@ -412,6 +412,8 @@ recorder_fetch_cursor_image (ShellRecorder *recorder)
       *(guint32 *)(data + i * stride + 4 * j) = cursor_image->pixels[i * cursor_image->width + j];
 
   cairo_surface_mark_dirty (recorder->cursor_image);
+
+  XFree (cursor_image);
 }
 
 /* Overlay the cursor image on the frame. We draw the cursor image
--
cgit v0.9
