--- source/texk/web2c/pdftexdir/pdftosrc.cc.orig	2018-04-23 05:56:51.757082063 +0000
+++ source/texk/web2c/pdftexdir/pdftosrc.cc	2018-04-23 06:00:05.458749056 +0000
@@ -68,7 +68,7 @@
     Stream *s;
     Object srcStream, srcName, catalogDict;
     FILE *outfile;
-    char *outname;
+    const char *outname;
     int objnum = 0, objgen = 0;
     bool extract_xref_table = false;
     int c;
--- source/texk/web2c/luatexdir/lua/lepdflib.cc.orig	2018-04-23 06:02:00.090127690 +0000
+++ source/texk/web2c/luatexdir/lua/lepdflib.cc	2018-04-23 06:14:14.535401600 +0000
@@ -516,7 +516,7 @@
 #define m_poppler_get_GOOSTRING(in, function)                  \
 static int m_##in##_##function(lua_State * L)                  \
 {                                                              \
-    GooString *gs;                                             \
+    const GooString *gs;                                             \
     udstruct *uin;                                             \
     uin = (udstruct *) luaL_checkudata(L, 1, M_##in);          \
     if (uin->pd != NULL && uin->pd->pc != uin->pc)             \
@@ -1650,7 +1650,7 @@
 
 static int m_Object_getString(lua_State * L)
 {
-    GooString *gs;
+    const GooString *gs;
     udstruct *uin;
     uin = (udstruct *) luaL_checkudata(L, 1, M_Object);
     if (uin->pd != NULL && uin->pd->pc != uin->pc)
--- source/texk/web2c/luatexdir/image/pdftoepdf.w.orig	2018-04-23 06:16:08.003740279 +0000
+++ source/texk/web2c/luatexdir/image/pdftoepdf.w	2018-04-23 06:25:27.372390750 +0000
@@ -357,7 +357,7 @@
     pdf->cave = true;
 }
 
-static void copyString(PDF pdf, GooString * string)
+static void copyString(PDF pdf, const GooString * string)
 {
     char *p;
     unsigned char c;
@@ -389,7 +389,7 @@
     pdf->cave = true;
 }
 
-static void copyName(PDF pdf, char *s)
+static void copyName(PDF pdf, const char *s)
 {
     pdf_out(pdf, '/');
     for (; *s != 0; s++) {
