--- a/makefile.unix	2009-05-04 10:31:57.000000000 -0700
+++ b/makefile.unix	2009-05-04 10:31:57.000000000 -0700
@@ -49,11 +49,11 @@
 #DESTDIR=/usr
 
 # Solaris using CC
-CXX=CC
-CXXFLAGS=-fast -erroff=wvarhidemem
-DEFINES=-D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE
-STRIP=strip
-DESTDIR=/usr
+#CXX=CC
+#CXXFLAGS=-fast -erroff=wvarhidemem
+#DEFINES=-D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE
+#STRIP=strip
+#DESTDIR=/usr
 
 # Solaris using GCC (optimized for UltraSPARC 1 CPU)
 #CXX=g++
