diff -aur asp.old/asp.in asp/asp.in
--- asp.old/asp.in	2020-07-22 00:44:08.805544581 +0200
+++ asp/asp.in	2020-07-22 00:45:13.875934993 +0200
@@ -132,7 +132,7 @@
   if [[ ! -f $ASPROOT/.asp ]]; then
     git init -q "$ASPROOT" || return 1
     for remote in "${ARCH_GIT_REPOS[@]}"; do
-      git remote add "$remote" "https://git.archlinux.org/svntogit/$remote.git" || return 1
+      git remote add "$remote" "https:///github.com/archlinux/svntogit-$remote.git" || return 1
     done
 
     touch "$ASPROOT/.asp" || return 1
@@ -300,7 +300,7 @@
   esac
 
   for remote in "${ARCH_GIT_REPOS[@]}"; do
-    git remote set-url "$remote" "$1://git.archlinux.org/svntogit/$remote.git"
+    git remote set-url "$remote" "$1:///github.com/archlinux/svntogit-$remote.git"
   done
 }
 
diff -aur asp.old/man/asp.1.txt asp/man/asp.1.txt
--- asp.old/man/asp.1.txt	2020-07-22 00:44:08.805544581 +0200
+++ asp/man/asp.1.txt	2020-07-22 00:46:13.996296652 +0200
@@ -16,7 +16,7 @@
 -----------
 Manage the version-controlled sources for the build scripts used to create Arch
 Linux packages. This program provides a thin wrapper over the svntogit
-repositories hosted at https://git.archlinux.org. It aims to provide a
+repositories hosted at https://github.com/archlinux. It aims to provide a
 replacement for abs which favors a sparse checkout.
 
 Commands
