Index: tomboy/configure.in
===================================================================
--- tomboy.orig/configure.in	2014-01-09 10:41:48.213622716 +0000
+++ tomboy/configure.in	2014-01-09 10:41:48.209622716 +0000
@@ -175,7 +175,7 @@
 #
 # Check for dbus-sharp
 #
-PKG_CHECK_MODULES(DBUS, dbus-sharp-1.0 >= 0.4 dbus-sharp-glib-1.0 >= 0.3)
+PKG_CHECK_MODULES(DBUS, dbus-sharp-2.0 >= 0.4 dbus-sharp-glib-2.0 >= 0.3)
 AC_SUBST(DBUS_LIBS)
 
 #
Index: tomboy/Tomboy/RemoteControlProxy.cs
===================================================================
--- tomboy.orig/Tomboy/RemoteControlProxy.cs	2013-10-27 23:42:50.649722383 +0000
+++ tomboy/Tomboy/RemoteControlProxy.cs	2014-01-09 10:42:13.273621654 +0000
@@ -55,8 +55,7 @@
 				return null;
 
 			RemoteControl remote_control = new RemoteControl (manager);
-			Bus.Session.Register (Namespace,
-			                      new ObjectPath (Path),
+			Bus.Session.Register (new ObjectPath (Path),
 			                      remote_control);
 			return remote_control;
 #else
