=== modified file 'libzeitgeist/where-clause.vala'
--- libzeitgeist/where-clause.vala	2013-02-16 22:35:31 +0000
+++ libzeitgeist/where-clause.vala	2014-03-17 19:57:49 +0000
@@ -211,7 +211,11 @@
          */
         private T[] generic_array_to_unowned_array<T> (GenericArray<T> gptrarr)
         {
+#if VALA_0_24
+            long[] pointers = new long[gptrarr.length];
+#else
             long[] pointers = new long[gptrarr.length + 1];
+#endif
             Memory.copy(pointers, ((PtrArray *) gptrarr)->pdata,
                 (gptrarr.length) * sizeof (void *));
             return (T[]) pointers;

