diff --git a/src/plugins/platforms/xcb/qxcbwindow.cpp b/src/plugins/platforms/xcb/qxcbwindow.cpp
index 9382488b74..d87713db40 100644
--- a/src/plugins/platforms/xcb/qxcbwindow.cpp
+++ b/src/plugins/platforms/xcb/qxcbwindow.cpp
@@ -846,6 +846,7 @@ void QXcbWindow::doFocusIn()
 void QXcbWindow::doFocusOut()
 {
     connection()->setFocusWindow(nullptr);
+    QWindowSystemInterface::handleWindowActivated(nullptr, Qt::ActiveWindowFocusReason);
     relayFocusToModalWindow();
     // Do not set the active window to nullptr if there is a FocusIn coming.
     connection()->focusInTimer().start(400);
