--- a/modules/video_output/x11/xcommon.c
+++ b/modules/video_output/x11/xcommon.c
@@ -2179,21 +2179,24 @@ static void ToggleFullScreen ( vout_thread_t *p_vout )
 
         if( p_vout->p_sys->b_net_wm_state_fullscreen )
         {
-            XClientMessageEvent event;
-
-            memset( &event, 0, sizeof( XClientMessageEvent ) );
-
-            event.type = ClientMessage;
-            event.message_type = p_vout->p_sys->net_wm_state;
-            event.display = p_vout->p_sys->p_display;
-            event.window = p_vout->p_sys->p_win->base_window;
-            event.format = 32;
-            event.data.l[ 0 ] = 1; /* set property */
-            event.data.l[ 1 ] = p_vout->p_sys->net_wm_state_fullscreen;
+            XClientMessageEvent event = {
+                .type = ClientMessage,
+                .window = p_vout->p_sys->p_win->base_window,
+                .message_type = p_vout->p_sys->net_wm_state,
+                .format = 32,
+                .data = {
+                    .l = {
+                        1, /* set property */
+                        p_vout->p_sys->net_wm_state_fullscreen,
+                        0,
+                        1,
+                    },
+                },
+            };
 
             XSendEvent( p_vout->p_sys->p_display,
                         DefaultRootWindow( p_vout->p_sys->p_display ),
-                        False, SubstructureRedirectMask,
+                        False, SubstructureNotifyMask|SubstructureRedirectMask,
                         (XEvent*)&event );
         }
 
--- a/modules/video_output/x11/xcommon.c
+++ b/modules/video_output/x11/xcommon.c
@@ -2169,14 +2169,13 @@ static void ToggleFullScreen ( vout_thread_t *p_vout )
                                      p_vout->p_sys->p_win->base_window,
                                      CWOverrideRedirect,
                                      &attributes);
-
-            /* Make sure the change is effective */
-            XReparentWindow( p_vout->p_sys->p_display,
-                             p_vout->p_sys->p_win->base_window,
-                             DefaultRootWindow( p_vout->p_sys->p_display ),
-                             0, 0 );
         }
 
+        /* Make sure the change is effective */
+        XReparentWindow( p_vout->p_sys->p_display,
+                         p_vout->p_sys->p_win->base_window,
+                         DefaultRootWindow( p_vout->p_sys->p_display ), 0, 0 );
+
         if( p_vout->p_sys->b_net_wm_state_fullscreen )
         {
             XClientMessageEvent event = {
@@ -2215,11 +2214,6 @@ static void ToggleFullScreen ( vout_thread_t *p_vout )
  * as really the wm should be deciding if, on fullscreening of a window
  * the focus should go there or not, so let the wm decided */
 #define APPFOCUS 0
-        /* Make sure the change is effective */
-        XReparentWindow( p_vout->p_sys->p_display,
-                         p_vout->p_sys->p_win->base_window,
-                         DefaultRootWindow( p_vout->p_sys->p_display ),
-                         0, 0 );
 
 #ifdef HAVE_XINERAMA
         if( XineramaQueryExtension( p_vout->p_sys->p_display, &i_d1, &i_d2 ) &&

