diff -NarU 3 vte-0.16.0.orig/python/vte.defs vte-0.16.0/python/vte.defs
--- vte-0.16.0.orig/python/vte.defs	2007-02-26 22:05:02.000000000 +0100
+++ vte-0.16.0/python/vte.defs	2007-03-14 20:44:55.000000000 +0100
@@ -7,6 +7,14 @@
   (gtype-id "VTE_TYPE_TERMINAL")
 )
 
+(define-object Reaper
+  (in-module "Vte")
+  (parent "GObject")
+  (c-name "VteReaper")
+  (gtype-id "VTE_TYPE_REAPER")
+)
+
+
 ;; Enumerations and flags ...
 
 (define-enum TerminalEraseBinding
@@ -701,4 +709,14 @@
   (return-type "const-char*")
 )
 
+;; From ../src/reaper.h
+(define-function vte_reaper_get_type
+  (c-name "vte_reaper_get_type")
+  (return-type "GtkType")
+)
+
+(define-function reaper_get
+  (c-name "vte_reaper_get")
+  (return-type "VteReaper*")
+)
 
diff -NarU 3 vte-0.16.0.orig/python/vte.override vte-0.16.0/python/vte.override
--- vte-0.16.0.orig/python/vte.override	2007-02-26 22:05:02.000000000 +0100
+++ vte-0.16.0/python/vte.override	2007-03-14 20:46:12.000000000 +0100
@@ -7,12 +7,14 @@
 #include <pygobject.h>
 #include <gtk/gtk.h>
 #include "../src/vte.h"
+#include "../src/reaper.h"
 %%
 modulename vte
 %%
 import gtk.gdk.Pixbuf as PyGdkPixbuf_Type
 import gtk.MenuShell as PyGtkMenuShell_Type
 import gtk.Widget as PyGtkWidget_Type
+import gobject.GObject as PyGObject_Type
 %%
 override vte_terminal_feed kwargs
 static PyObject *
diff -NarU 3 vte-0.16.0.orig/python/vtemodule.c vte-0.16.0/python/vtemodule.c
--- vte-0.16.0.orig/python/vtemodule.c	2007-02-26 22:05:02.000000000 +0100
+++ vte-0.16.0/python/vtemodule.c	2007-03-14 20:47:12.000000000 +0100
@@ -25,6 +25,7 @@
 #include <pygobject.h>
 #include <pygtk/pygtk.h>
 #include "../src/vte.h"
+#include "../src/reaper.h"
 
 extern void pyvte_register_classes(PyObject * d);
 extern PyMethodDef pyvte_functions[];
