--- xorg-server-1.2.0/hw/kdrive/Makefile.am.xephyr	2007-01-22 22:13:14.000000000 -0500
+++ xorg-server-1.2.0/hw/kdrive/Makefile.am	2007-01-23 12:28:32.000000000 -0500
@@ -1,16 +1,3 @@
-if KDRIVEVESA
-VESA_SUBDIRS = vesa ati chips epson i810 mach64 mga neomagic nvidia pm2 r128 \
-               smi via
-endif
-
-if KDRIVEFBDEV
-FBDEV_SUBDIRS = fbdev
-endif
-
-if XSDLSERVER
-XSDL_SUBDIRS = sdl
-endif
-
 if XEPHYR
 XEPHYR_SUBDIRS = ephyr
 endif
@@ -22,11 +9,7 @@
 SUBDIRS =			\
 	src			\
 	$(LINUX_SUBDIRS)	\
-	$(XSDL_SUBDIRS)		\
-	$(FBDEV_SUBDIRS)	\
-	$(VESA_SUBDIRS)		\
-	$(XEPHYR_SUBDIRS)       \
-	fake
+	$(XEPHYR_SUBDIRS)
 
 DIST_SUBDIRS = vesa ati chips epson i810 mach64 mga neomagic nvidia pm2 r128 \
                smi via fbdev sdl ephyr src linux fake sis300
