diff -Naur nmap-4.68/zenmap/setup.py nmap-4.68-new/zenmap/setup.py
--- nmap-4.68/zenmap/setup.py	2008-06-06 02:55:01.000000000 -0300
+++ nmap-4.68-new/zenmap/setup.py	2008-07-09 00:57:19.000000000 -0300
@@ -44,7 +44,7 @@
 # These are created after a "install" or "py2exe" command
 # These directories are relative to the installation or dist directory
 data_dir = os.path.join('share', APP_NAME)
-pixmaps_dir = os.path.join(data_dir, 'pixmaps')
+pixmaps_dir = os.path.join('share', 'pixmaps', APP_NAME)
 locale_dir = os.path.join(data_dir, 'locale')
 config_dir = os.path.join(data_dir, 'config')
 docs_dir = os.path.join(data_dir, 'docs')
diff -Naur nmap-4.68/zenmap/zenmapCore/Paths.py nmap-4.68-new/zenmap/zenmapCore/Paths.py
--- nmap-4.68/zenmap/zenmapCore/Paths.py	2008-06-06 02:32:19.000000000 -0300
+++ nmap-4.68-new/zenmap/zenmapCore/Paths.py	2008-07-09 00:59:22.000000000 -0300
@@ -60,7 +60,7 @@
 CONFIG_DIR = join(prefix, "share", APP_NAME, "config")
 LOCALE_DIR = join(prefix, "share", APP_NAME, "locale")
 MISC_DIR = join(prefix, "share", APP_NAME, "misc")
-PIXMAPS_DIR = join(prefix, "share", "zenmap", "pixmaps")
+PIXMAPS_DIR = join(prefix, "share", "zenmap", "pixmaps", APP_NAME)
 DOCS_DIR = join(prefix, "share", APP_NAME, "docs")
 
 def get_extra_executable_search_paths():
