diff -ruN a/source/Makefile b/source/Makefile
--- a/source/Makefile	2022-07-14 17:04:29.000000000 +0200
+++ b/source/Makefile	2022-07-19 21:33:40.627248973 +0200
@@ -31,7 +31,6 @@
 
 CPPFLAGS += -DVERSION=\"$(VERSION)\" -D_REENTRANT -D_POSIX_PTHREAD_SEMANTICS
 CXXFLAGS += -O2 -Wall -fPIC
-CXXFLAGS += -march=native
 LDLFAGS +=
 LDLIBS +=
 
@@ -46,15 +45,15 @@
 
 $(ZITA-ALSA-PCMI_MIN): $(ZITA-ALSA-PCMI_O)
 	$(CXX) -shared $(LDFLAGS) -Wl,-soname,$(ZITA-ALSA-PCMI_MAJ) -o $(ZITA-ALSA-PCMI_MIN) $(ZITA-ALSA-PCMI_O) $(ZITA-ALSA-PCMI_DEP)
+	ln -sf $(ZITA-ALSA-PCMI_MIN) $(ZITA-ALSA-PCMI_SO)
+	ln -sf $(ZITA-ALSA-PCMI_MIN) $(ZITA-ALSA-PCMI_MAJ)
 
 
 install:	$(ZITA-ALSA-PCMI_MIN)
-	install -d $(DESTDIR)$(INCDIR)
-	install -d $(DESTDIR)$(LIBDIR)
-	install -m 644 $(ZITA-ALSA-PCMI_H) $(DESTDIR)$(INCDIR)
-	install -m 755 $(ZITA-ALSA-PCMI_MIN) $(DESTDIR)$(LIBDIR)
-	ldconfig
-	ln -sf $(ZITA-ALSA-PCMI_MIN) $(DESTDIR)$(LIBDIR)/$(ZITA-ALSA-PCMI_SO)
+	install -vDm 644 $(ZITA-ALSA-PCMI_H) -t $(DESTDIR)$(INCDIR)
+	install -vDm 755 $(ZITA-ALSA-PCMI_MIN) -t $(DESTDIR)$(LIBDIR)
+	ln -fsv $(ZITA-ALSA-PCMI_MIN) $(DESTDIR)$(LIBDIR)/$(ZITA-ALSA-PCMI_SO)
+	ln -fsv $(ZITA-ALSA-PCMI_MIN) $(DESTDIR)$(LIBDIR)/$(ZITA-ALSA-PCMI_MAJ)
 
 uninstall:
 	rm -rf $(DESTDIR)$(INCDIR)/$(ZITA-ALSA-PCMI_H)
