diff -ruN a/apps/Makefile b/apps/Makefile
--- a/apps/Makefile	2022-07-14 09:07:32.000000000 +0200
+++ b/apps/Makefile	2022-07-19 21:12:39.787546862 +0200
@@ -25,7 +25,8 @@
 
 VERSION = "0.3.2"
 CPPFLAGS +=  -MMD -MP -DVERSION=\"$(VERSION)\"
-CXXFLAGS += -O2 -Wall
+CXXFLAGS += -O2 -Wall -I ../source
+LDFLAGS += -L ../source
 
 
 all:	alsa_loopback alsa_delay
@@ -48,14 +49,11 @@
 
 
 install:	all
-	install -d $(BINDIR)
-	install -m 755 alsa_loopback  $(DESTDIR)$(BINDIR)
-	install -m 755 alsa_delay     $(DESTDIR)$(BINDIR)
+	install -vDm 755 alsa_loopback alsa_delay -t $(DESTDIR)$(BINDIR)
 
 
 uninstall:
-	rm -f $(DESTDIR)$(BINDIR)/alsa_loopback
-	rm -f $(DESTDIR)$(BINDIR)/bin/alsa_delay
+	rm -f $(DESTDIR)$(BINDIR)/{alsa_delay,alsa_loopback}
 
 
 clean:
