diff --git a/app/dialogs/about-dialog.c b/app/dialogs/about-dialog.c
index d767d830f0..4352f021ca 100644
--- a/app/dialogs/about-dialog.c
+++ b/app/dialogs/about-dialog.c
@@ -163,7 +163,9 @@ about_dialog_create (GimpCoreConfig *config)
 #ifdef GIMP_UNSTABLE
           about_dialog_add_unstable_message (children->data);
 #endif /* GIMP_UNSTABLE */
+#ifdef CHECK_UPDATE
           about_dialog_add_update (&dialog, config);
+#endif /* CHECK_UPDATE */
         }
       else
         g_warning ("%s: ooops, no box in this container?", G_STRLOC);
