From cf8b629ebf30d6aa57dd405ce450fb0f33136626 Mon Sep 17 00:00:00 2001
From: Peter Hutterer <peter.hutterer@who-t.net>
Date: Mon, 13 Jul 2015 08:30:43 +1000
Subject: [PATCH xf86-input-libinput] Fedora: use unaccelerated vmask on input
 ABI > 21

Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>
---
 src/xf86libinput.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/libinput.c b/src/libinput.c
index a06e44f..5eb5adf 100644
--- a/src/xf86libinput.c
+++ b/src/xf86libinput.c
@@ -47,7 +47,7 @@
 #define XI86_SERVER_FD 0x20
 #endif

-#if GET_ABI_MAJOR(ABI_XINPUT_VERSION) * 1000 + GET_ABI_MINOR(ABI_XINPUT_VERSION) > 22000
+#if GET_ABI_MAJOR(ABI_XINPUT_VERSION) * 1000 + GET_ABI_MINOR(ABI_XINPUT_VERSION) > 21000
 #define HAVE_VMASK_UNACCEL 1
 #else
 #undef HAVE_VMASK_UNACCEL
--
2.4.3

