From e1877fb7b84237bb40c88436e0a56aa999801595 Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <jan.steffens@gmail.com>
Date: Tue, 10 Sep 2019 18:08:25 +0000
Subject: [PATCH 4/8] meson: Don't prefix p11_user_config

---
 p11-kit/meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/p11-kit/meson.build b/p11-kit/meson.build
index e2d807b..18b3d26 100644
--- a/p11-kit/meson.build
+++ b/p11-kit/meson.build
@@ -300,7 +300,7 @@ pkg.generate(libraries: libp11_kit,
 
 pkcs11_conf_example_conf = configuration_data({
   'p11_system_config_file': prefix / p11_system_config_file,
-  'p11_user_config': prefix / p11_user_config
+  'p11_user_config': p11_user_config
 })
 pkcs11_conf_example = configure_file(input: 'pkcs11.conf.example.in',
                                      output: '@BASENAME@',
-- 
2.23.0

