diff -upr xfce4-session-4.12.1.orig/scripts/xinitrc.in.in xfce4-session-4.12.1/scripts/xinitrc.in.in
--- xfce4-session-4.12.1.orig/scripts/xinitrc.in.in	2018-12-30 05:05:04.201829621 +0200
+++ xfce4-session-4.12.1/scripts/xinitrc.in.in	2018-12-30 05:08:45.200127486 +0200
@@ -95,6 +95,14 @@ if [ -d /etc/X11/xinit/xinitrc.d ]; then
     unset f
 fi
 
+# unless DBUS_SESSION_BUS_ADDRESS is set elsewhere
+if [ -z "$DBUS_SESSION_BUS_ADDRESS" ]; then
+    # use the user session bus if available
+    if [ -S "$XDG_RUNTIME_DIR/bus" -a -O "$XDG_RUNTIME_DIR/bus" ]; then
+        export DBUS_SESSION_BUS_ADDRESS="unix:path=$XDG_RUNTIME_DIR/bus"
+    fi
+fi
+
 # run xfce4-session if installed
 if which xfce4-session >/dev/null 2>&1; then
 
