--- xserver-xorg-video-intel-2.2.0.90-patched/src/i830_quirks.c.orig	2008-02-14 18:05:37.000000000 -0800
+++ xserver-xorg-video-intel-2.2.0.90-patched/src/i830_quirks.c	2008-02-14 18:06:25.000000000 -0800
@@ -78,6 +78,8 @@ static i830_quirk i830_quirk_list[] = {
     { PCI_CHIP_I965_GM, 0x1028, 0x0209, quirk_ignore_tv },
     /* Dell Inspiron 1535 */
     { PCI_CHIP_I965_GM, 0x1028, 0x0254, quirk_ignore_tv },
+    /* Dell Inspiron 1735 */
+    { PCI_CHIP_I965_GM, 0x1028, 0x0256, quirk_ignore_tv },
 
     /* Lenovo X60s has no TV output */
     { PCI_CHIP_I945_GM, 0x17aa, 0x201a, quirk_ignore_tv },
