diff -Nur gdm-2.30.2.orig/daemon/gdm-server.c gdm-2.30.2/daemon/gdm-server.c
--- gdm-2.30.2.orig/daemon/gdm-server.c	2010-04-26 20:08:42.000000000 +0300
+++ gdm-2.30.2/daemon/gdm-server.c	2010-04-29 22:55:49.631648355 +0300
@@ -127,7 +127,7 @@
         g_return_val_if_fail (GDM_IS_SERVER (server), NULL);
 
         error = NULL;
-        command = g_strdup_printf (LIBEXECDIR "/ck-get-x11-display-device --display %s",
+        command = g_strdup_printf ("/usr/lib/ConsoleKit/ck-get-x11-display-device --display %s",
                                    server->priv->display_name);
 
         g_debug ("GdmServer: Running helper %s", command);
diff -Nur gdm-2.30.2.orig/data/greeter-autostart/gnome-settings-daemon.desktop.in.in gdm-2.30.2/data/greeter-autostart/gnome-settings-daemon.desktop.in.in
--- gdm-2.30.2.orig/data/greeter-autostart/gnome-settings-daemon.desktop.in.in	2010-03-17 23:27:16.000000000 +0200
+++ gdm-2.30.2/data/greeter-autostart/gnome-settings-daemon.desktop.in.in	2010-04-29 22:53:54.984878068 +0300
@@ -1,8 +1,8 @@
 [Desktop Entry]
 Type=Application
 _Name=GNOME Settings Daemon
-TryExec=@LIBEXECDIR@/gnome-settings-daemon
-Exec=@LIBEXECDIR@/gnome-settings-daemon --gconf-prefix=/apps/gdm/simple-greeter/settings-manager-plugins
+TryExec=/usr/bin/gnome-settings-daemon
+Exec=/usr/bin/gnome-settings-daemon --gconf-prefix=/apps/gdm/simple-greeter/settings-manager-plugins
 OnlyShowIn=GNOME;
 X-GNOME-Autostart-Phase=Initialization
 X-GNOME-Autostart-Notify=true
diff -Nur gdm-2.30.2.orig/data/greeter-autostart/polkit-gnome-authentication-agent-1.desktop.in.in gdm-2.30.2/data/greeter-autostart/polkit-gnome-authentication-agent-1.desktop.in.in
--- gdm-2.30.2.orig/data/greeter-autostart/polkit-gnome-authentication-agent-1.desktop.in.in	2010-03-26 15:28:03.000000000 +0200
+++ gdm-2.30.2/data/greeter-autostart/polkit-gnome-authentication-agent-1.desktop.in.in	2010-04-29 22:54:24.638396328 +0300
@@ -1,5 +1,5 @@
 [Desktop Entry]
 Type=Application
 _Name=PolicyKit Authentication Agent
-Exec=@LIBEXECDIR@/polkit-gnome-authentication-agent-1
+Exec=/usr/lib/polkit-gnome/polkit-gnome-authentication-agent-1
 OnlyShowIn=GNOME;
diff -Nur gdm-2.30.2.orig/gui/simple-chooser/gdm-chooser-session.c gdm-2.30.2/gui/simple-chooser/gdm-chooser-session.c
--- gdm-2.30.2.orig/gui/simple-chooser/gdm-chooser-session.c	2010-03-17 23:27:16.000000000 +0200
+++ gdm-2.30.2/gui/simple-chooser/gdm-chooser-session.c	2010-04-29 22:57:08.691626507 +0300
@@ -131,7 +131,7 @@
         ret = FALSE;
 
         error = NULL;
-        g_spawn_command_line_async (LIBEXECDIR "/gnome-settings-daemon --gconf-prefix=/apps/gdm/simple-chooser/settings-manager-plugins", &error);
+        g_spawn_command_line_async ("/usr/bin/gnome-settings-daemon --gconf-prefix=/apps/gdm/simple-chooser/settings-manager-plugins", &error);
         if (error != NULL) {
                 g_warning ("Error starting settings daemon: %s", error->message);
                 g_error_free (error);
