diff -Nur gdm-2.91.91.orig/daemon/gdm-server.c gdm-2.91.91/daemon/gdm-server.c
--- gdm-2.91.91.orig/daemon/gdm-server.c	2011-02-17 19:27:53.000000000 -0800
+++ gdm-2.91.91/daemon/gdm-server.c	2011-02-27 16:56:51.283347074 -0800
@@ -131,7 +131,7 @@
         g_return_val_if_fail (GDM_IS_SERVER (server), NULL);
 
         error = NULL;
-        command = g_strdup_printf (LIBEXECDIR "/ck-get-x11-display-device --display %s",
+        command = g_strdup_printf ("/usr/lib/ConsoleKit/ck-get-x11-display-device --display %s",
                                    server->priv->display_name);
 
         g_debug ("GdmServer: Running helper %s", command);
diff -Nur gdm-2.91.91.orig/data/greeter-autostart/gnome-settings-daemon.desktop.in.in gdm-2.91.91/data/greeter-autostart/gnome-settings-daemon.desktop.in.in
--- gdm-2.91.91.orig/data/greeter-autostart/gnome-settings-daemon.desktop.in.in	2011-02-17 19:27:53.000000000 -0800
+++ gdm-2.91.91/data/greeter-autostart/gnome-settings-daemon.desktop.in.in	2011-02-26 08:43:28.700009251 -0800
@@ -1,8 +1,8 @@
 [Desktop Entry]
 Type=Application
 _Name=GNOME Settings Daemon
-TryExec=@LIBEXECDIR@/gnome-settings-daemon
-Exec=@LIBEXECDIR@/gnome-settings-daemon
+TryExec=/usr/bin/gnome-settings-daemon
+Exec=/usr/bin/gnome-settings-daemon
 OnlyShowIn=GNOME;
 X-GNOME-Autostart-Phase=Initialization
 X-GNOME-Autostart-Notify=true
diff -Nur gdm-2.91.91.orig/data/greeter-autostart/polkit-gnome-authentication-agent-1.desktop.in.in gdm-2.91.91/data/greeter-autostart/polkit-gnome-authentication-agent-1.desktop.in.in
--- gdm-2.91.91.orig/data/greeter-autostart/polkit-gnome-authentication-agent-1.desktop.in.in	2011-02-17 19:27:53.000000000 -0800
+++ gdm-2.91.91/data/greeter-autostart/polkit-gnome-authentication-agent-1.desktop.in.in	2011-02-26 08:44:06.810009153 -0800
@@ -1,5 +1,5 @@
 [Desktop Entry]
 Type=Application
 _Name=PolicyKit Authentication Agent
-Exec=@LIBEXECDIR@/polkit-gnome-authentication-agent-1
+Exec=/usr/lib/polkit-gnome/polkit-gnome-authentication-agent-1
 OnlyShowIn=GNOME;
diff -Nur gdm-2.91.91.orig/gui/simple-chooser/gdm-chooser-session.c gdm-2.91.91/gui/simple-chooser/gdm-chooser-session.c
--- gdm-2.91.91.orig/gui/simple-chooser/gdm-chooser-session.c	2011-02-17 19:27:53.000000000 -0800
+++ gdm-2.91.91/gui/simple-chooser/gdm-chooser-session.c	2011-02-26 08:45:01.946675918 -0800
@@ -131,7 +131,7 @@
         ret = FALSE;
 
         error = NULL;
-        g_spawn_command_line_async (LIBEXECDIR "/gnome-settings-daemon", &error);
+        g_spawn_command_line_async ("/usr/bin/gnome-settings-daemon", &error);
         if (error != NULL) {
                 g_warning ("Error starting settings daemon: %s", error->message);
                 g_error_free (error);
