--- src/mkfs.fat.c.old	2013-06-09 12:30:43.626408604 -0400
+++ src/mkfs.fat.c	2013-06-09 12:30:28.306756934 -0400
@@ -743,10 +743,10 @@
 	 * differently: The jump code is only 2 bytes (and m68k machine code
 	 * :-), then 6 bytes filler (ignored), then 3 byte serial number. */
 	bs.boot_jump[2] = 'm';
-	strcpy((char *)bs.system_id, "kdosf");
+	memcpy((char *)bs.system_id, "kdosf", strlen("kdosf"));
     }
     else
-	strcpy((char *)bs.system_id, "mkfs.fat");
+	memcpy((char *)bs.system_id, "mkfs.fat", strlen("mkfs.fat"));
     if (sectors_per_cluster)
 	bs.cluster_size = (char)sectors_per_cluster;
     if (size_fat == 32) {
