diff -aur a2ps-4.14.old/etc/Makefile.am a2ps-4.14/etc/Makefile.am
--- a2ps-4.14.old/etc/Makefile.am	2020-05-15 16:22:46.236954254 +0200
+++ a2ps-4.14/etc/Makefile.am	2020-05-15 16:58:42.382348094 +0200
@@ -44,5 +44,5 @@
 
 # Building a time stamp to know the version.
 README: README.in Makefile
-	sed "s!@date@!`date`!g;s!@version@!$(VERSION)!g"	\
+	sed "s!@date@!$(shell date --utc --date="@${SOURCE_DATE_EPOCH:-$(date +%s)}")!g;s!@version@!$(VERSION)!g"	\
 	$(srcdir)/README.in > README
diff -aur a2ps-4.14.old/etc/Makefile.in a2ps-4.14/etc/Makefile.in
--- a2ps-4.14.old/etc/Makefile.in	2020-05-15 16:22:46.236954254 +0200
+++ a2ps-4.14/etc/Makefile.in	2020-05-15 16:59:00.132720058 +0200
@@ -486,7 +486,7 @@
 
 # Building a time stamp to know the version.
 README: README.in Makefile
-	sed "s!@date@!`date`!g;s!@version@!$(VERSION)!g"	\
+	sed "s!@date@!$(shell date --utc --date="@${SOURCE_DATE_EPOCH:-$(date +%s)}")!g;s!@version@!$(VERSION)!g"	\
 	$(srcdir)/README.in > README
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
