diff -ruN a/Authorization/Polkit/Makefile.in b/Authorization/Polkit/Makefile.in
--- a/Authorization/Polkit/Makefile.in	2020-12-11 23:51:58.000000000 +0100
+++ b/Authorization/Polkit/Makefile.in	2021-01-20 23:49:22.918774464 +0100
@@ -35,7 +35,7 @@
 	$(INSTALL_DATA) $(SRC_DIR)/actions $(POLKIT_ACTIONS_DIRECTORY)/$(POLKIT_PACKAGE_NAME).$(POLKIT_ACTIONS_EXTENSION)
 
 install-rules-directory:
-	$(INSTALL_DIRECTORY) $(POLKIT_RULES_DIRECTORY)
+	install -vdm 750 $(POLKIT_RULES_DIRECTORY)
 
 install-rules: install-rules-directory
 	$(INSTALL_DATA) $(SRC_DIR)/rules $(POLKIT_RULES_DIRECTORY)/$(POLKIT_PACKAGE_NAME).$(POLKIT_RULES_EXTENSION)
