Index: gpgkeys_curl.c
===================================================================
--- keyserver/gpgkeys_curl.c	(revision 4741)
+++ keyserver/gpgkeys_curl.c	(working copy)
@@ -286,7 +286,7 @@
 
   if(follow_redirects)
     {
-      curl_easy_setopt(curl,CURLOPT_FOLLOWLOCATION,1);
+      curl_easy_setopt(curl,CURLOPT_FOLLOWLOCATION,1L);
       if(follow_redirects>0)
 	curl_easy_setopt(curl,CURLOPT_MAXREDIRS,follow_redirects);
     }
@@ -298,10 +298,10 @@
     {
       fprintf(console,"gpgkeys: curl version = %s\n",curl_version());
       curl_easy_setopt(curl,CURLOPT_STDERR,console);
-      curl_easy_setopt(curl,CURLOPT_VERBOSE,1);
+      curl_easy_setopt(curl,CURLOPT_VERBOSE,1L);
     }
 
-  curl_easy_setopt(curl,CURLOPT_SSL_VERIFYPEER,opt->flags.check_cert);
+  curl_easy_setopt(curl,CURLOPT_SSL_VERIFYPEER,(long)opt->flags.check_cert);
   curl_easy_setopt(curl,CURLOPT_CAINFO,opt->ca_cert_file);
 
   if(proxy)
Index: gpgkeys_hkp.c
===================================================================
--- keyserver/gpgkeys_hkp.c	(revision 4741)
+++ keyserver/gpgkeys_hkp.c	(working copy)
@@ -202,9 +202,9 @@
     fprintf(console,"gpgkeys: HTTP URL is `%s'\n",request);
 
   curl_easy_setopt(curl,CURLOPT_URL,request);
-  curl_easy_setopt(curl,CURLOPT_POST,1);
+  curl_easy_setopt(curl,CURLOPT_POST,1L);
   curl_easy_setopt(curl,CURLOPT_POSTFIELDS,key);
-  curl_easy_setopt(curl,CURLOPT_FAILONERROR,1);
+  curl_easy_setopt(curl,CURLOPT_FAILONERROR,1L);
 
   res=curl_easy_perform(curl);
   if(res!=0)
@@ -661,7 +661,7 @@
     {
       fprintf(console,"gpgkeys: curl version = %s\n",curl_version());
       curl_easy_setopt(curl,CURLOPT_STDERR,console);
-      curl_easy_setopt(curl,CURLOPT_VERBOSE,1);
+      curl_easy_setopt(curl,CURLOPT_VERBOSE,1L);
     }
 
   if(proxy)