Index: lib/Driver/Tools.cpp
===================================================================
--- lib/Driver/Tools.cpp	(revision 123373)
+++ lib/Driver/Tools.cpp	(working copy)
@@ -3306,7 +3306,7 @@
     else if (ToolChain.getArch() == llvm::Triple::arm)
       CmdArgs.push_back("/lib/ld-linux.so.3");
     else
-      CmdArgs.push_back("/lib64/ld-linux-x86-64.so.2");
+      CmdArgs.push_back("/lib/ld-linux-x86-64.so.2");
   }
 
   CmdArgs.push_back("-o");
Index: lib/Driver/ToolChains.cpp
===================================================================
--- lib/Driver/ToolChains.cpp	(revision 123373)
+++ lib/Driver/ToolChains.cpp	(working copy)
@@ -1317,18 +1317,10 @@
   if (Arch == llvm::Triple::x86)
     Suffix64 = "/64";
 
-  std::string Lib32 = "lib";
-
-  bool Exists;
-  if (!llvm::sys::fs::exists("/lib32", Exists) && Exists)
-    Lib32 = "lib32";
-
+  std::string Lib32 = "lib32";
   std::string Lib64 = "lib";
-  bool Symlink;
-  if (!llvm::sys::fs::exists("/lib64", Exists) && Exists &&
-      (llvm::sys::fs::is_symlink("/lib64", Symlink) || !Symlink))
-    Lib64 = "lib64";
 
+  bool Exists;
   std::string GccTriple = "";
   if (Arch == llvm::Triple::arm) {
     if (!llvm::sys::fs::exists("/usr/lib/gcc/arm-linux-gnueabi", Exists) &&
