From d3132f65594e8ad27b74643786a786cd3e60c715 Mon Sep 17 00:00:00 2001
From: Christos Zoulas <christos@zoulas.com>
Date: Tue, 8 Oct 2019 20:25:13 +0000
Subject: [PATCH] PR/112: connesc: move mime magic higher so that it always
 takes effect.

---
 magic/Magdir/compress | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/magic/Magdir/compress b/magic/Magdir/compress
index 600d216b7..c700b5a45 100644
--- a/magic/Magdir/compress
+++ b/magic/Magdir/compress
@@ -58,11 +58,11 @@
 >>>13	string		09			\b, version 9
 # other gzipped binary like gzipped tar, VirtualBox extension package,...
 >>10	default		x		gzip compressed data
+!:mime	application/gzip
 >>>0	use	gzip-info
 # size of the original (uncompressed) input data modulo 2^32
 >>>-4	ulelong		x		\b, original size modulo 2^32 %u
 # gzipped TAR or VirtualBox extension package
-!:mime	application/gzip
 #!:mime	application/x-compressed-tar
 #!:mime	application/x-virtualbox-vbox-extpack
 # https://www.w3.org/TR/SVG/mimereg.html
