--- setup.py	2008-02-23 20:58:52.000000000 +0000
+++ setup.py.new	2008-02-23 21:04:18.000000000 +0000
@@ -608,7 +608,7 @@
         # a release.  Most open source OSes come with one or more
         # versions of BerkeleyDB already installed.
 
-        max_db_ver = (4, 5)
+        max_db_ver = (4, 7)
         # NOTE: while the _bsddb.c code links against BerkeleyDB 4.6.x
         # we leave that version disabled by default as it has proven to be
         # quite a buggy library release on many platforms.
