--- discover-5.10.5/discover/CMakeLists.txt.orig	2017-08-22 17:30:12.209293604 +0000
+++ discover-5.10.5/discover/CMakeLists.txt	2017-08-22 17:30:25.102651653 +0000
@@ -36,7 +36,7 @@
 
 # Standard desktop file accepts local files as input.
 set(DesktopNoDisplay "false")
-set(DesktopMimeType "application/vnd.debian.binary-package;application/x-rpm;")
+set(DesktopMimeType "")
 set(DesktopExec "plasma-discover %F")
 configure_file(org.kde.discover.desktop.cmake ${CMAKE_CURRENT_BINARY_DIR}/org.kde.discover.desktop)
 install(PROGRAMS ${CMAKE_CURRENT_BINARY_DIR}/org.kde.discover.desktop DESTINATION ${XDG_APPS_INSTALL_DIR} )
