diff -Naur old/cmd/dotty/dotty_layout.lefty new/cmd/dotty/dotty_layout.lefty
--- old/cmd/dotty/dotty_layout.lefty	2013-09-06 15:07:52.000000000 -1000
+++ new/cmd/dotty/dotty_layout.lefty	2013-10-22 15:23:50.153028328 -1000
@@ -5,7 +5,7 @@
     local fd;
 
     if (~dotty.lservers[lserver] | tablesize (dotty.lservers[lserver]) == 0) {
-        if (~((fd = openio ('pipe', lserver, 'r+', '%e -Txdot')) >= 0)) {
+        if (~((fd = openio ('pipe', lserver, 'r+', '%e -Txdot1.2')) >= 0)) {
             dotty.message (0, concat ('cannot start ', lserver));
             return null;
         }
@@ -438,6 +438,8 @@
             }
         } else if (t[i] == 'I') {
             i = i + 7;
+        } else if (t[i] == 't') {
+            i = i + 2;
         } else {
             dotty.message (0, concat ('draw language parser error: ', t[i]));
             return null;
