--- src/build/common.gypi	2009-12-17 01:57:20.000000000 +0100
+++ src/build/common.gypi	2009-12-17 02:37:05.663568345 +0100
@@ -652,15 +652,6 @@
             # used during computation does not change depending on how the
             # compiler optimized the code, since the value is always kept
             # in its specified precision.
-            'conditions': [
-              ['branding=="Chromium"', {
-                'cflags': [
-                  '-march=pentium4',
-                  '-msse2',
-                  '-mfpmath=sse',
-                ],
-              }],
-            ],
             'cflags': [
               '-m32',
             ],
