diff --git a/autoconf/emacsbrand b/autoconf/emacsbrand
index 77dfb65c..2884ec45 100755
--- a/autoconf/emacsbrand
+++ b/autoconf/emacsbrand
@@ -35,6 +35,7 @@ done
 #*    The autodetection ELisp expression                               */
 #*---------------------------------------------------------------------*/
 expr="(cond \
+        ((string-match \"GNU Emacs 28\" (emacs-version)) (kill-emacs 28)) \
         ((string-match \"GNU Emacs 27\" (emacs-version)) (kill-emacs 27)) \
         ((string-match \"GNU Emacs 26\" (emacs-version)) (kill-emacs 26)) \
         ((string-match \"GNU Emacs 25\" (emacs-version)) (kill-emacs 25)) \
@@ -74,6 +75,10 @@ else
               else
                 if [ "$version" = "27" ]; then
                     echo "emacs27"
+                else
+                  if [ "$version" = "28" ]; then
+                      echo "emacs28"
+                  fi
                 fi
               fi
             fi    
diff --git a/bmacs/Makefile b/bmacs/Makefile
index 3c64876e..37932158 100644
--- a/bmacs/Makefile
+++ b/bmacs/Makefile
@@ -116,7 +116,7 @@ POPULATION	= Makefile README $(EMACS_SOURCES) $(OLD_SOURCES) \
 #*---------------------------------------------------------------------*/
 all: $(EMACSBRAND)
 
-emacs22 emacs23 emacs24 emacs25 emacs26 emacs27:
+emacs22 emacs23 emacs24 emacs25 emacs26 emacs27 emacs28:
 	(expr=load-path; \
           for p in . $(LOADPATH); do \
              expr="(cons \"../$$p\" (cons \"$$p\" $$expr))"; \
@@ -183,6 +183,7 @@ install.emacs24: doinstall
 install.emacs25: doinstall
 install.emacs26: doinstall
 install.emacs27: doinstall
+install.emacs28: doinstall
 install.xemacs: doinstall
 
 install.generic:
