--- festival/src/modules/clustergen/HTS_vocoder_me.cc.omv~	2017-12-28 22:12:57.990128926 +0100
+++ festival/src/modules/clustergen/HTS_vocoder_me.cc	2017-12-28 22:16:12.636790019 +0100
@@ -91,6 +91,9 @@
 HTS_VOCODER_ME_C_START;
 
 #include "./HTS_vocoder_me.h"
+#define HTS_Vocoder_initialize HTS_Vocoder_initialize_nodupes
+#define HTS_Vocoder_synthesize HTS_Vocoder_synthesize_nodupes
+#define HTS_Vocoder_clear HTS_Vocoder_clear_nodupes
 #include "../hts_engine/HTS_vocoder.c"
 
 /* HTS_Vocoder_initialize_me: initialize vocoder (mixed excitation) */
