--- file-5.07-orig/magic/Magdir/archive	2011-05-14 04:28:25.918516697 +0300
+++ file-5.07/magic/Magdir/archive	2011-05-14 04:29:14.351587647 +0300
@@ -565,6 +565,9 @@
 0	string		PK\x07\x08PK\x03\x04	Zip multi-volume archive data, at least PKZIP v2.50 to extract
 !:mime	application/zip
 
+# Zip archiver (Archlinux bugfix: detect generic zip archives)
+0	string		PK\003\004	Zip archive data
+
 # Zip archives (Greg Roelofs, c/o zip-bugs@wkuvx1.wku.edu)
 0	string		PK\003\004
 
