From 22d7b61791c382088a6c0df5dce3a15405d6c495 Mon Sep 17 00:00:00 2001
From: Chris Wilson <chris@chris-wilson.co.uk>
Date: Tue, 04 Jan 2011 12:32:01 +0000
Subject: i965: Fix off-by-one in assert

Signed-off-by: Chris Wilson <chris@chris-wilson.co.uk>
---
diff --git a/src/i965_render.c b/src/i965_render.c
index ca5e6b6..b1ac6ba 100644
--- a/src/i965_render.c
+++ b/src/i965_render.c
@@ -1730,8 +1730,7 @@ static void i965_bind_surfaces(struct intel_screen_private *intel)
 {
 	uint32_t *binding_table;
 
-	assert(intel->surface_used + 4 * sizeof(struct brw_surface_state_padded) <
-	       sizeof(intel->surface_data));
+	assert(intel->surface_used + 4 * sizeof(struct brw_surface_state_padded) <= sizeof(intel->surface_data));
 
 	binding_table = (uint32_t*) (intel->surface_data + intel->surface_used);
 	intel->surface_table = intel->surface_used;
--
cgit v0.8.3-6-g21f6
