diff -Nur gdm-3.3.92.orig/gui/simple-chooser/gdm-chooser-session.c gdm-3.3.92/gui/simple-chooser/gdm-chooser-session.c
--- gdm-3.3.92.orig/gui/simple-chooser/gdm-chooser-session.c	2012-03-19 22:30:05.304823391 +0000
+++ gdm-3.3.92/gui/simple-chooser/gdm-chooser-session.c	2012-03-19 22:30:17.004732251 +0000
@@ -131,7 +131,7 @@
         ret = FALSE;
 
         error = NULL;
-        g_spawn_command_line_async (LIBEXECDIR "/gnome-settings-daemon", &error);
+        g_spawn_command_line_async ("/usr/lib/gnome-settings-daemon/gnome-settings-daemon", &error);
         if (error != NULL) {
                 g_warning ("Error starting settings daemon: %s", error->message);
                 g_error_free (error);
