--- src/fr-process.c.orig	2008-11-30 20:10:50.000000000 +0100
+++ src/fr-process.c	2008-11-30 20:11:54.000000000 +0100
@@ -981,7 +981,7 @@
 	if (command_is_sticky (process, process->priv->current_command))
 		allow_sticky_processes_only (process, emit_signal);
 
-	else if (process->term_on_stop)
+	else if (process->term_on_stop && process->priv->command_pid > 0)
 		kill (process->priv->command_pid, SIGTERM);
 
 	else {
