From f81a43f2c4ed5e5083e3a850bc7164d11a910358 Mon Sep 17 00:00:00 2001
From: Eugen Dedu <Eugen.Dedu@pu-pm.univ-fcomte.fr>
Date: Sat, 24 Mar 2012 09:17:40 +0000
Subject: Fix compile error with gcc 4.7

---
diff --git a/lib/engine/account/bank-impl.h b/lib/engine/account/bank-impl.h
index e16848e..b0a8775 100644
--- a/lib/engine/account/bank-impl.h
+++ b/lib/engine/account/bank-impl.h
@@ -203,7 +203,7 @@ template<typename AccountType>
 void
 Ekiga::BankImpl<AccountType>::add_account (boost::shared_ptr<AccountType> account)
 {
-  add_object (account);
+  this->add_object (account);
 
   account->questions.connect (boost::ref (questions));
 }
@@ -213,7 +213,7 @@ template<typename AccountType>
 void
 Ekiga::BankImpl<AccountType>::remove_account (boost::shared_ptr<AccountType> account)
 {
-  remove_object (account);
+  this->remove_object (account);
 }
 
 #endif
diff --git a/lib/engine/addressbook/book-impl.h b/lib/engine/addressbook/book-impl.h
index 711d8d9..4fea7d5 100644
--- a/lib/engine/addressbook/book-impl.h
+++ b/lib/engine/addressbook/book-impl.h
@@ -204,7 +204,7 @@ void
 Ekiga::BookImpl<ContactType>::add_contact (boost::shared_ptr<ContactType> contact)
 {
   contact->questions.connect (boost::ref (questions));
-  add_object (contact);
+  this->add_object (contact);
 }
 
 
@@ -212,7 +212,7 @@ template<typename ContactType>
 void
 Ekiga::BookImpl<ContactType>::remove_contact (boost::shared_ptr<ContactType> contact)
 {
-  remove_object (contact);
+  this->remove_object (contact);
 }
 
 #endif
diff --git a/lib/engine/addressbook/source-impl.h b/lib/engine/addressbook/source-impl.h
index a30cd03..630236f 100644
--- a/lib/engine/addressbook/source-impl.h
+++ b/lib/engine/addressbook/source-impl.h
@@ -206,7 +206,7 @@ template<typename BookType>
 void
 Ekiga::SourceImpl<BookType>::add_book (boost::shared_ptr<BookType> book)
 {
-  add_object (book);
+  this->add_object (book);
 
   add_connection (book, book->contact_added.connect (boost::bind (boost::ref (contact_added), book, _1)));
 
diff --git a/lib/engine/presence/cluster-impl.h b/lib/engine/presence/cluster-impl.h
index 245a7c6..92074a8 100644
--- a/lib/engine/presence/cluster-impl.h
+++ b/lib/engine/presence/cluster-impl.h
@@ -147,7 +147,7 @@ Ekiga::ClusterImpl<HeapType>::add_heap (boost::shared_ptr<HeapType> heap)
 
   add_connection (heap, heap->questions.connect (boost::ref (questions)));
 
-  add_object (heap);
+  this->add_object (heap);
 }
 
 template<typename HeapType>
diff --git a/lib/engine/presence/heap-impl.h b/lib/engine/presence/heap-impl.h
index d63c491..15fe86f 100644
--- a/lib/engine/presence/heap-impl.h
+++ b/lib/engine/presence/heap-impl.h
@@ -160,14 +160,14 @@ Ekiga::HeapImpl<PresentityType>::add_presentity (boost::shared_ptr<PresentityTyp
 {
   presentity->questions.connect (boost::ref (questions));
 
-  add_object (presentity);
+  this->add_object (presentity);
 }
 
 template<typename PresentityType>
 void
 Ekiga::HeapImpl<PresentityType>::remove_presentity (boost::shared_ptr<PresentityType> presentity)
 {
-  remove_object (presentity);
+  this->remove_object (presentity);
 }
 
 #endif
--
cgit v0.9.0.2
