diff -Nur comm-release.orig/mozilla/ipc/chromium/src/base/file_util_linux.cc comm-release/mozilla/ipc/chromium/src/base/file_util_linux.cc
--- comm-release.orig/mozilla/ipc/chromium/src/base/file_util_linux.cc	2012-04-24 23:15:57.347221336 +0000
+++ comm-release/mozilla/ipc/chromium/src/base/file_util_linux.cc	2012-04-24 23:18:34.116046872 +0000
@@ -5,6 +5,9 @@
 #include "base/file_util.h"
 
 #include <fcntl.h>
+#if defined(ANDROID) || defined(OS_POSIX)
+#include <unistd.h>
+#endif
 
 #include <string>
 #include <vector>
diff -Nur comm-release.orig/mozilla/ipc/chromium/src/base/message_pump_libevent.cc comm-release/mozilla/ipc/chromium/src/base/message_pump_libevent.cc
--- comm-release.orig/mozilla/ipc/chromium/src/base/message_pump_libevent.cc	2012-04-24 23:15:57.353887955 +0000
+++ comm-release/mozilla/ipc/chromium/src/base/message_pump_libevent.cc	2012-04-24 23:18:34.116046872 +0000
@@ -6,6 +6,9 @@
 
 #include <errno.h>
 #include <fcntl.h>
+#if defined(ANDROID) || defined(OS_POSIX)
+#include <unistd.h>
+#endif
 
 #include "eintr_wrapper.h"
 #include "base/logging.h"
diff -Nur comm-release.orig/mozilla/ipc/chromium/src/base/time_posix.cc comm-release/mozilla/ipc/chromium/src/base/time_posix.cc
--- comm-release.orig/mozilla/ipc/chromium/src/base/time_posix.cc	2012-04-24 23:15:57.350554645 +0000
+++ comm-release/mozilla/ipc/chromium/src/base/time_posix.cc	2012-04-24 23:18:34.116046872 +0000
@@ -13,6 +13,9 @@
 #else
 #include <time.h>
 #endif
+#if defined(ANDROID) || defined(OS_POSIX)
+#include <unistd.h>
+#endif
 
 #include <limits>
 
