diff -Nur telepathy-qt-0.9.1.orig/TelepathyQt/client-registrar.cpp telepathy-qt-0.9.1/TelepathyQt/client-registrar.cpp
--- telepathy-qt-0.9.1.orig/TelepathyQt/client-registrar.cpp	2012-04-01 17:44:00.013466055 +0000
+++ telepathy-qt-0.9.1/TelepathyQt/client-registrar.cpp	2012-04-01 17:51:00.233486476 +0000
@@ -39,6 +39,8 @@
 #include <TelepathyQt/MethodInvocationContext>
 #include <TelepathyQt/PendingComposite>
 #include <TelepathyQt/PendingReady>
+ 
+#include <stdint.h>
 
 namespace Tp
 {
diff -Nur telepathy-qt-0.9.1.orig/TelepathyQt/stream-tube-client.cpp telepathy-qt-0.9.1/TelepathyQt/stream-tube-client.cpp
--- telepathy-qt-0.9.1.orig/TelepathyQt/stream-tube-client.cpp	2012-04-01 17:44:00.006799442 +0000
+++ telepathy-qt-0.9.1/TelepathyQt/stream-tube-client.cpp	2012-04-01 17:51:00.233486476 +0000
@@ -38,6 +38,8 @@
 #include <QAbstractSocket>
 #include <QHash>
 
+#include <stdint.h>
+
 namespace Tp
 {
 
diff -Nur telepathy-qt-0.9.1.orig/TelepathyQt/stream-tube-server.cpp telepathy-qt-0.9.1/TelepathyQt/stream-tube-server.cpp
--- telepathy-qt-0.9.1.orig/TelepathyQt/stream-tube-server.cpp	2012-04-01 17:44:00.003466134 +0000
+++ telepathy-qt-0.9.1/TelepathyQt/stream-tube-server.cpp	2012-04-01 17:51:00.233486476 +0000
@@ -38,6 +38,8 @@
 #include <TelepathyQt/OutgoingStreamTubeChannel>
 #include <TelepathyQt/StreamTubeChannel>
 
+#include <stdint.h>
+
 namespace Tp
 {
 
