diff -u -r libmypaint-1.3.0/configure.ac libmypaint-1.3.0-gegl/configure.ac
--- libmypaint-1.3.0/configure.ac	2018-05-13 16:49:44.997077642 +0200
+++ libmypaint-1.3.0-gegl/configure.ac	2018-05-13 16:49:36.223744476 +0200
@@ -45,7 +45,7 @@
 
 ## Dependencies ##
 
-m4_define([gegl_required_version], [0.3])
+m4_define([gegl_required_version], [0.4])
 m4_define([introspection_required_version], [1.32.0])
 
 AC_INIT([libmypaint],
@@ -336,7 +336,7 @@
 )
 
 if eval "test x$enable_gegl = xyes"; then
-  PKG_CHECK_MODULES(GEGL, gegl-0.3 >= gegl_required_version)
+  PKG_CHECK_MODULES(GEGL, gegl-0.4 >= gegl_required_version)
 fi
 AM_CONDITIONAL(ENABLE_GEGL, test "x$enable_gegl" = "xyes")
 
diff -u -r libmypaint-1.3.0/gegl/libmypaint-gegl.pc.in libmypaint-1.3.0-gegl/gegl/libmypaint-gegl.pc.in
--- libmypaint-1.3.0/gegl/libmypaint-gegl.pc.in	2016-12-01 23:47:14.000000000 +0100
+++ libmypaint-1.3.0-gegl/gegl/libmypaint-gegl.pc.in	2018-05-13 16:50:16.620410376 +0200
@@ -6,6 +6,6 @@
 Name: libmypaint
 Description: MyPaint brush engine library, with GEGL integration.
 Version: @LIBMYPAINT_VERSION@
-Requires: gegl-0.3 libmypaint
+Requires: gegl-0.4 libmypaint
 Cflags: -I${includedir}/libmypaint-gegl
 Libs: -L${libdir} -lmypaint-gegl
diff -u -r libmypaint-1.3.0/gegl/Makefile.am libmypaint-1.3.0-gegl/gegl/Makefile.am
--- libmypaint-1.3.0/gegl/Makefile.am	2018-05-13 16:48:53.437078622 +0200
+++ libmypaint-1.3.0-gegl/gegl/Makefile.am	2018-05-13 16:49:58.523744053 +0200
@@ -11,7 +11,7 @@
 INTROSPECTION_GIRS =
 INTROSPECTION_SCANNER_ARGS = \
     --warn-all \
-    --pkg="gegl-0.3" \
+    --pkg="gegl-0.4" \
     --pkg="glib-2.0" \
     --namespace="MyPaintGegl" \
     --nsversion="$(LIBMYPAINT_MAJOR_VERSION).$(LIBMYPAINT_MINOR_VERSION)" \
@@ -38,7 +38,7 @@
 	mypaint-gegl-surface.c
 
 MyPaintGegl-@LIBMYPAINT_MAJOR_VERSION@.@LIBMYPAINT_MINOR_VERSION@.gir: libmypaint-gegl-@LIBMYPAINT_API_PLATFORM_VERSION@.la Makefile
-MyPaintGegl_@LIBMYPAINT_MAJOR_VERSION@_@LIBMYPAINT_MINOR_VERSION@_gir_INCLUDES = GObject-2.0 MyPaint-$(LIBMYPAINT_MAJOR_VERSION).$(LIBMYPAINT_MINOR_VERSION) Gegl-0.3
+MyPaintGegl_@LIBMYPAINT_MAJOR_VERSION@_@LIBMYPAINT_MINOR_VERSION@_gir_INCLUDES = GObject-2.0 MyPaint-$(LIBMYPAINT_MAJOR_VERSION).$(LIBMYPAINT_MINOR_VERSION) Gegl-0.4
 MyPaintGegl_@LIBMYPAINT_MAJOR_VERSION@_@LIBMYPAINT_MINOR_VERSION@_gir_CFLAGS = $(AM_CFLAGS) $(AM_CPPFLAGS) -I. -I..
 MyPaintGegl_@LIBMYPAINT_MAJOR_VERSION@_@LIBMYPAINT_MINOR_VERSION@_gir_LIBS = libmypaint-gegl-@LIBMYPAINT_API_PLATFORM_VERSION@.la ../libmypaint-@LIBMYPAINT_API_PLATFORM_VERSION@.la
 MyPaintGegl_@LIBMYPAINT_MAJOR_VERSION@_@LIBMYPAINT_MINOR_VERSION@_gir_FILES = $(introspection_sources)
