diff --git a/posix/confstr.c b/posix/confstr.c
index 3c9566d..3ba9e5a 100644
--- a/posix/confstr.c
+++ b/posix/confstr.c
@@ -104,7 +104,7 @@ confstr (name, buf, len)
 	  }
 #endif
 	restenvs[string_len++] = '\0';
-	string = restenvs;
+	string = strdup(restenvs);
       }
       break;
 
@@ -167,7 +167,7 @@ confstr (name, buf, len)
 	  }
 #endif
 	restenvs[string_len++] = '\0';
-	string = restenvs;
+	string = strdup(restenvs);
       }
       break;
 
@@ -230,7 +230,7 @@ confstr (name, buf, len)
 	  }
 #endif
 	restenvs[string_len++] = '\0';
-	string = restenvs;
+	string = strdup(restenvs);
       }
       break;
 
