diff -Nur ifplugd-0.28.orig/src/ifplugd.c ifplugd-0.28/src/ifplugd.c
--- ifplugd-0.28.orig/src/ifplugd.c	2005-06-04 21:21:51.000000000 +0200
+++ ifplugd-0.28/src/ifplugd.c	2012-08-29 21:51:53.137644163 +0200
@@ -447,7 +447,7 @@
             goto finish;
 
     if (send_retval && daemonize && wait_on_fork) {
-        char c = status == IFSTATUS_UP ? 2 : (status == IFSTATUS_DOWN ? 3 : 1);
+        char c = status == IFSTATUS_UP ? 1 : (status == IFSTATUS_DOWN ? 2 : 3);
         daemon_retval_send(c);
         send_retval = 0;
     }
