http://bugs.gentoo.org/show_bug.cgi?id=305749

diff -Naur inkscape-0.47-orig/src/eraser-context.cpp inkscape-0.47/src/eraser-context.cpp
--- inkscape-0.47-orig/src/eraser-context.cpp	2010-02-18 08:45:58.000000000 -0500
+++ inkscape-0.47/src/eraser-context.cpp	2010-02-18 08:46:26.000000000 -0500
@@ -749,7 +749,7 @@
                 if ( eraserMode ) {
                     toWorkOn = sp_document_partial_items_in_box(sp_desktop_document(desktop), desktop->dkey, bounds);
                 } else {
-                    Inkscape::Rubberband::Rubberband *r = Inkscape::Rubberband::get(desktop);
+                    Inkscape::Rubberband *r = Inkscape::Rubberband::get(desktop);
                     toWorkOn = sp_document_items_at_points(sp_desktop_document(desktop), desktop->dkey, r->getPoints());
                 }
                 toWorkOn = g_slist_remove( toWorkOn, acid );
diff -Naur inkscape-0.47-orig/src/extension/internal/filter/filter-file.cpp inkscape-0.47/src/extension/internal/filter/filter-file.cpp
--- inkscape-0.47-orig/src/extension/internal/filter/filter-file.cpp	2010-02-18 08:45:58.000000000 -0500
+++ inkscape-0.47/src/extension/internal/filter/filter-file.cpp	2010-02-18 09:07:54.000000000 -0500
@@ -161,7 +161,7 @@
 	mywriter writer;
 	sp_repr_write_stream(node, writer, 0, FALSE, g_quark_from_static_string("svg"), 0, 0);
 
-    Inkscape::Extension::build_from_mem(xml_str, new Filter::Filter(g_strdup(writer.c_str())));
+    Inkscape::Extension::build_from_mem(xml_str, new Filter(g_strdup(writer.c_str())));
 	g_free(xml_str);
     return;
 }
diff -Naur inkscape-0.47-orig/src/extension/internal/filter/filter.cpp inkscape-0.47/src/extension/internal/filter/filter.cpp
--- inkscape-0.47-orig/src/extension/internal/filter/filter.cpp	2010-02-18 08:45:58.000000000 -0500
+++ inkscape-0.47/src/extension/internal/filter/filter.cpp	2010-02-18 09:21:06.000000000 -0500
@@ -217,7 +217,7 @@
                 "<menu-tip>%s</menu-tip>\n"
             "</effect>\n"
         "</inkscape-extension>\n", name, id, submenu, tip);
-    Inkscape::Extension::build_from_mem(xml_str, new Filter::Filter(filter));
+    Inkscape::Extension::build_from_mem(xml_str, new Filter(filter));
 	g_free(xml_str);
     return;
 }
diff -Naur inkscape-0.47-orig/src/flood-context.cpp inkscape-0.47/src/flood-context.cpp
--- inkscape-0.47-orig/src/flood-context.cpp	2010-02-18 08:45:58.000000000 -0500
+++ inkscape-0.47/src/flood-context.cpp	2010-02-18 08:48:16.000000000 -0500
@@ -900,7 +900,7 @@
     if (is_point_fill) {
         fill_points.push_back(Geom::Point(event->button.x, event->button.y));
     } else {
-        Inkscape::Rubberband::Rubberband *r = Inkscape::Rubberband::get(desktop);
+        Inkscape::Rubberband *r = Inkscape::Rubberband::get(desktop);
         fill_points = r->getPoints();
     }
 
@@ -1206,7 +1206,7 @@
 
     case GDK_BUTTON_RELEASE:
         if (event->button.button == 1 && !event_context->space_panning) {
-            Inkscape::Rubberband::Rubberband *r = Inkscape::Rubberband::get(desktop);
+            Inkscape::Rubberband *r = Inkscape::Rubberband::get(desktop);
             if (r->is_started()) {
                 // set "busy" cursor
                 desktop->setWaitingCursor();
diff -Naur inkscape-0.47-orig/src/gradient-context.cpp inkscape-0.47/src/gradient-context.cpp
--- inkscape-0.47-orig/src/gradient-context.cpp	2010-02-18 08:45:58.000000000 -0500
+++ inkscape-0.47/src/gradient-context.cpp	2010-02-18 08:47:33.000000000 -0500
@@ -641,7 +641,7 @@
                 if (!event_context->within_tolerance) {
                     // we've been dragging, either do nothing (grdrag handles that),
                     // or rubberband-select if we have rubberband
-                    Inkscape::Rubberband::Rubberband *r = Inkscape::Rubberband::get(desktop);
+                    Inkscape::Rubberband *r = Inkscape::Rubberband::get(desktop);
                     if (r->is_started() && !event_context->within_tolerance) {
                         // this was a rubberband drag
                         if (r->getMode() == RUBBERBAND_MODE_RECT) {
diff -Naur inkscape-0.47-orig/src/select-context.cpp inkscape-0.47/src/select-context.cpp
--- inkscape-0.47-orig/src/select-context.cpp	2010-02-18 08:45:58.000000000 -0500
+++ inkscape-0.47/src/select-context.cpp	2010-02-18 08:50:11.000000000 -0500
@@ -602,7 +602,7 @@
                     }
                     sc->item = NULL;
                 } else {
-                    Inkscape::Rubberband::Rubberband *r = Inkscape::Rubberband::get(desktop);
+                    Inkscape::Rubberband *r = Inkscape::Rubberband::get(desktop);
                     if (r->is_started() && !within_tolerance) {
                         // this was a rubberband drag
                         GSList *items = NULL;
