diff --git a/plugins/sftp/mounter.cpp b/plugins/sftp/mounter.cpp
index fca9907c..f673c9f7 100644
--- a/plugins/sftp/mounter.cpp
+++ b/plugins/sftp/mounter.cpp
@@ -145,6 +145,7 @@ void Mounter::onPackageReceived(const NetworkPacket& np)
         << QStringLiteral("-o") << QStringLiteral("reconnect")
         << QStringLiteral("-o") << QStringLiteral("ServerAliveInterval=30")
         << QStringLiteral("-o") << QStringLiteral("password_stdin")
+        << QStringLiteral("-o") << QStringLiteral("KexAlgorithms=+diffie-hellman-group14-sha1")
         ;
 
     m_proc->setProgram(program, arguments);
