diff -ruN a/Make.Rules b/Make.Rules
--- a/Make.Rules	2021-08-16 04:05:04.000000000 +0200
+++ b/Make.Rules	2021-08-18 17:50:42.028791561 +0200
@@ -21,6 +21,14 @@
 lib=$(shell ldd /usr/bin/ld|egrep "ld-linux|ld.so"|cut -d/ -f2)
 endif
 
+ifndef sbin
+sbin=sbin
+endif
+
+ifdef sbindir
+sbin=$(sbindir)
+endif
+
 ifdef prefix
 exec_prefix=$(prefix)
 lib_prefix=$(exec_prefix)
@@ -37,7 +45,7 @@
 # Target directories
 
 MANDIR=$(man_prefix)/man
-SBINDIR=$(exec_prefix)/sbin
+SBINDIR=$(exec_prefix)/$(sbin)
 INCDIR=$(inc_prefix)/include
 LIBDIR=$(lib_prefix)/$(lib)
 PKGCONFIGDIR=$(LIBDIR)/pkgconfig
@@ -79,7 +87,7 @@
 
 SYSTEM_HEADERS = /usr/include
 INCS=$(topdir)/libcap/include/sys/capability.h
-CFLAGS += -Dlinux $(WARNINGS) $(DEBUG)
+CFLAGS += $(CPPFLAGS) -Dlinux $(WARNINGS) $(DEBUG)
 INDENT := $(shell if [ -n "$$(which indent 2>/dev/null)" ]; then echo "| indent -kr" ; fi)
 
 # SHARED tracks whether or not the SHARED libraries (libcap.so,
