diff -Nur comm-release.orig/mozilla/configure.in comm-release/mozilla/configure.in
--- comm-release.orig/mozilla/configure.in	2012-02-05 18:02:17.376830377 +0000
+++ comm-release/mozilla/configure.in	2012-02-05 18:03:25.246274275 +0000
@@ -5629,7 +5629,7 @@
             dnl v0.9.6 one to check for.
             AC_TRY_COMPILE([
                 #include <vpx/vpx_decoder.h>
-                #if !defined(VPX_CODEC_USE_INPUT_PARTITION)
+                #if !defined(VPX_CODEC_USE_INPUT_FRAGMENTS)
                     #error "test failed."
                 #endif
                 ],
