--- a/Makefile.in
+++ b/Makefile.in
@@ -383,7 +383,7 @@
 @WITH_NCURSES_TRUE@slabtop_SOURCES = slabtop.c $(top_srcdir)/lib/strutils.c
 @WITH_NCURSES_TRUE@slabtop_LDADD = @NCURSES_LIBS@
 @WITH_NCURSES_TRUE@watch_SOURCES = watch.c $(top_srcdir)/lib/strutils.c
-@WITH_NCURSES_TRUE@watch_LDADD = @NCURSES_LIBS@
+@WITH_NCURSES_TRUE@watch_LDADD = @WATCH_NCURSES_LIBS@
 kill_SOURCES = skill.c $(top_srcdir)/lib/strutils.c
 skill_SOURCES = skill.c $(top_srcdir)/lib/strutils.c
 snice_SOURCES = skill.c $(top_srcdir)/lib/strutils.c
