diff --git a/src/server/storage/dbconfigmysql.cpp b/src/server/storage/dbconfigmysql.cpp
index 8c5a52628..35fffd865 100644
--- a/src/server/storage/dbconfigmysql.cpp
+++ b/src/server/storage/dbconfigmysql.cpp
@@ -390,12 +390,6 @@ bool DbConfigMysql::startInternalServer()
             }
         }
 
-        // clear mysql ib_logfile's in case innodb_log_file_size option changed in last confUpdate
-        if (confUpdate) {
-            QFile(dataDir + QDir::separator() + QLatin1String("ib_logfile0")).remove();
-            QFile(dataDir + QDir::separator() + QLatin1String("ib_logfile1")).remove();
-        }
-
         qCDebug(AKONADISERVER_LOG) << "Executing:" << mMysqldPath << arguments.join(QLatin1Char(' '));
         mDatabaseProcess = new QProcess;
         mDatabaseProcess->start(mMysqldPath, arguments);
