diff --git i/dbus/dbus-internals.c w/dbus/dbus-internals.c
index ab498b15..cf86f4ae 100644
--- i/dbus/dbus-internals.c
+++ w/dbus/dbus-internals.c
@@ -193,7 +193,7 @@ static dbus_bool_t warn_initted = FALSE;
  * threads, the worst that will happen is that warnings get their default
  * fatal/non-fatal nature. */
 static dbus_bool_t fatal_warnings = FALSE;
-static dbus_bool_t fatal_warnings_on_check_failed = TRUE;
+static dbus_bool_t fatal_warnings_on_check_failed = FALSE;
 
 static void
 init_warnings(void)
