diff --git i/test/automated/displayless/meson.build w/test/automated/displayless/meson.build
index b154cb191..9d77b509d 100644
--- i/test/automated/displayless/meson.build
+++ w/test/automated/displayless/meson.build
@@ -35,11 +35,7 @@ tests = [
   ]]
 ]
 
-tracker_tests = [
-  ['test-nautilus-search-engine-tracker', [
-    'test-nautilus-search-engine-tracker.c',
-  ]],
-]
+tracker_tests = []
 
 foreach t: tests
   test(
