From efb6a7c0db5c40207bf78bde9d0f9b32ac667e9b Mon Sep 17 00:00:00 2001
From: Jack Gurulian <jack.gurulian@gmail.com>
Date: Tue, 15 Oct 2013 10:54:47 +0000
Subject: Correctly use AlbumArtCache.get_media_title in notifications

---
diff --git a/gnomemusic/notification.py b/gnomemusic/notification.py
index 2c8485e..5775db1 100644
--- a/gnomemusic/notification.py
+++ b/gnomemusic/notification.py
@@ -78,7 +78,7 @@ class NotificationManager:
             album = item.get_string(Grl.METADATA_KEY_ALBUM)\
                 or _("Unknown Album")
 
-            self._notification.update(item.get_title(),
+            self._notification.update(AlbumArtCache.get_media_title(item),
                                       # TRANSLATORS: by refers to the artist, from to the album
                                       _("by %s, from %s") % ('<b>' + artist + '</b>',
                                                              '<i>' + album + '</i>'),
--
cgit v0.9.2
