--- httpd-2.4.25/support/ab.c.orig	2017-05-21 21:59:17.131193359 +0000
+++ httpd-2.4.25/support/ab.c	2017-05-21 22:01:34.704322361 +0000
@@ -2496,6 +2496,8 @@
         heartbeatres = 0;
 
 #ifdef USE_SSL
+    SSL_load_error_strings();
+    SSL_library_init();
 #ifdef RSAREF
     R_malloc_init();
 #else
@@ -2505,8 +2507,6 @@
     OPENSSL_malloc_init();
 #endif
 #endif
-    SSL_load_error_strings();
-    SSL_library_init();
     bio_out=BIO_new_fp(stdout,BIO_NOCLOSE);
     bio_err=BIO_new_fp(stderr,BIO_NOCLOSE);
 
