diff -r c6ca4f24c8d1 auth/pam.c
--- auth/pam.c	Wed Jun 02 10:15:09 2010 -0400
+++ auth/pam.c	Fri Jun 18 16:26:28 2010 -0400
@@ -102,14 +102,6 @@
 	log_error(USE_ERRNO|NO_EXIT|NO_MAIL, "unable to initialize PAM");
 	return(AUTH_FATAL);
     }
-
-    /*
-     * Set PAM_RUSER to the invoking user (the "from" user).
-     * We set PAM_RHOST to avoid a bug in Solaris 7 and below.
-     */
-    (void) pam_set_item(pamh, PAM_RUSER, user_name);
-    (void) pam_set_item(pamh, PAM_RHOST, user_host);
-
     /*
      * Some versions of pam_lastlog have a bug that
      * will cause a crash if PAM_TTY is not set so if
@@ -208,10 +200,13 @@
 	pam_init(pw, NULL, NULL);
 
     /*
-     * Update PAM_USER to reference the user we are running the command
-     * as, as opposed to the user we authenticated as.
+     * Set PAM_USER to the user we are changing *to* and
+     * set PAM_RUSER to the user we are coming *from*.
+     * We set PAM_RHOST to avoid a bug in Solaris 7 and below.
      */
     (void) pam_set_item(pamh, PAM_USER, pw->pw_name);
+    (void) pam_set_item(pamh, PAM_RUSER, user_name);
+    (void) pam_set_item(pamh, PAM_RHOST, user_host);
 
     /*
      * Set credentials (may include resource limits, device ownership, etc).
