Index: src/libs/zbxcrypto/tls.c
===================================================================
--- src/libs/zbxcrypto/tls.c	(revision 85017)
+++ src/libs/zbxcrypto/tls.c	(working copy)
@@ -127,6 +127,7 @@
 #define TLS_client_method				TLSv1_2_client_method
 #define SSL_CTX_get_ciphers(ciphers)			((ciphers)->cipher_list)
 #define SSL_CTX_set_min_proto_version(ctx, TLSv)	1
+#define SSL_CTX_set_max_proto_version(ctx, TLSv)	1
 
 static int	OPENSSL_init_ssl(int opts, void *settings)
 {
@@ -3131,8 +3132,11 @@
 		if (NULL == (ctx_cert = SSL_CTX_new(method)))
 			goto out_method;
 
-		if (1 != SSL_CTX_set_min_proto_version(ctx_cert, TLS1_2_VERSION))
+		if (1 != SSL_CTX_set_min_proto_version(ctx_cert, TLS1_2_VERSION) ||
+				1 != SSL_CTX_set_max_proto_version(ctx_cert, TLS1_2_VERSION))
+		{
 			goto out_method;
+		}
 	}
 
 	/* Create context for PSK-only authentication. PSK can come from configuration file (in proxy, agentd) */
@@ -3142,8 +3146,11 @@
 		if (NULL == (ctx_psk = SSL_CTX_new(method)))
 			goto out_method;
 
-		if (1 != SSL_CTX_set_min_proto_version(ctx_psk, TLS1_2_VERSION))
+		if (1 != SSL_CTX_set_min_proto_version(ctx_psk, TLS1_2_VERSION) ||
+				1 != SSL_CTX_set_max_proto_version(ctx_psk, TLS1_2_VERSION))
+		{
 			goto out_method;
+		}
 	}
 
 	/* Sometimes we need to be ready for both certificate and PSK whichever comes in. Set up a universal context */
@@ -3153,8 +3160,11 @@
 		if (NULL == (ctx_all = SSL_CTX_new(method)))
 			goto out_method;
 
-		if (1 != SSL_CTX_set_min_proto_version(ctx_all, TLS1_2_VERSION))
+		if (1 != SSL_CTX_set_min_proto_version(ctx_all, TLS1_2_VERSION) ||
+				1 != SSL_CTX_set_max_proto_version(ctx_all, TLS1_2_VERSION))
+		{
 			goto out_method;
+		}
 	}
 
 	/* 'TLSCAFile' parameter (in zabbix_server.conf, zabbix_proxy.conf, zabbix_agentd.conf) */
