diff -u -r postgresql-15.1.orig/src/include/pg_config_manual.h postgresql-15.1/src/include/pg_config_manual.h
--- postgresql-15.1.orig/src/include/pg_config_manual.h	2022-02-07 21:15:36.000000000 +0000
+++ postgresql-15.1/src/include/pg_config_manual.h	2022-04-19 18:54:54.716761530 +0000
@@ -207,7 +207,7 @@
  * support them yet.
  */
 #ifndef WIN32
-#define DEFAULT_PGSOCKET_DIR  "/tmp"
+#define DEFAULT_PGSOCKET_DIR  "/run/postgresql"
 #else
 #define DEFAULT_PGSOCKET_DIR ""
 #endif
