diff -Nur ppp-2.4.6.orig/chat/Makefile.linux ppp-2.4.6/chat/Makefile.linux
--- ppp-2.4.6.orig/chat/Makefile.linux	2014-01-02 05:42:08.000000000 +0100
+++ ppp-2.4.6/chat/Makefile.linux	2014-02-24 09:00:16.666577906 +0100
@@ -3,7 +3,7 @@
 COPTS=@CFLAGS@
 
 DESTDIR = $(INSTROOT)@DESTDIR@
-BINDIR = $(DESTDIR)/sbin
+BINDIR = $(DESTDIR)/bin
 MANDIR = $(DESTDIR)/share/man/man8
 
 CDEF1=	-DTERMIOS			# Use the termios structure
diff -Nur ppp-2.4.6.orig/linux/Makefile.top ppp-2.4.6/linux/Makefile.top
--- ppp-2.4.6.orig/linux/Makefile.top	2014-01-02 05:42:08.000000000 +0100
+++ ppp-2.4.6/linux/Makefile.top	2014-02-24 09:00:16.743242620 +0100
@@ -1,7 +1,7 @@
 # PPP top-level Makefile for Linux.
 
 DESTDIR = $(INSTROOT)@DESTDIR@
-BINDIR = $(DESTDIR)/sbin
+BINDIR = $(DESTDIR)/bin
 INCDIR = $(DESTDIR)/include
 MANDIR = $(DESTDIR)/share/man
 ETCDIR = $(INSTROOT)@SYSCONF@/ppp
diff -Nur ppp-2.4.6.orig/pppd/Makefile.linux ppp-2.4.6/pppd/Makefile.linux
--- ppp-2.4.6.orig/pppd/Makefile.linux	2014-01-02 05:42:08.000000000 +0100
+++ ppp-2.4.6/pppd/Makefile.linux	2014-02-24 09:00:16.743242620 +0100
@@ -9,7 +9,7 @@
 
 # Default installation locations
 DESTDIR = $(INSTROOT)@DESTDIR@
-BINDIR = $(DESTDIR)/sbin
+BINDIR = $(DESTDIR)/bin
 MANDIR = $(DESTDIR)/share/man/man8
 INCDIR = $(DESTDIR)/include
 
diff -Nur ppp-2.4.6.orig/pppd/plugins/Makefile.linux ppp-2.4.6/pppd/plugins/Makefile.linux
--- ppp-2.4.6.orig/pppd/plugins/Makefile.linux	2014-01-02 05:42:08.000000000 +0100
+++ ppp-2.4.6/pppd/plugins/Makefile.linux	2014-02-24 09:00:16.779908379 +0100
@@ -3,7 +3,7 @@
 COPTS=@CFLAGS@
 
 DESTDIR = $(INSTROOT)@DESTDIR@
-BINDIR = $(DESTDIR)/sbin
+BINDIR = $(DESTDIR)/bin
 MANDIR = $(DESTDIR)/share/man/man8
 LIBDIR = $(DESTDIR)/lib/pppd/$(VERSION)
 
diff -Nur ppp-2.4.6.orig/pppd/plugins/pppoe/Makefile.linux ppp-2.4.6/pppd/plugins/pppoe/Makefile.linux
--- ppp-2.4.6.orig/pppd/plugins/pppoe/Makefile.linux	2014-01-02 05:42:08.000000000 +0100
+++ ppp-2.4.6/pppd/plugins/pppoe/Makefile.linux	2014-02-24 09:00:16.809907637 +0100
@@ -17,7 +17,7 @@
 COPTS=@CFLAGS@
 
 DESTDIR = $(INSTROOT)@DESTDIR@
-BINDIR = $(DESTDIR)/sbin
+BINDIR = $(DESTDIR)/bin
 LIBDIR = $(DESTDIR)/lib/pppd/$(PPPDVERSION)
 
 PPPDVERSION = $(shell awk -F '"' '/VERSION/ { print $$2; }' ../../patchlevel.h)
diff -Nur ppp-2.4.6.orig/pppdump/Makefile.linux ppp-2.4.6/pppdump/Makefile.linux
--- ppp-2.4.6.orig/pppdump/Makefile.linux	2014-01-02 05:42:08.000000000 +0100
+++ ppp-2.4.6/pppdump/Makefile.linux	2014-02-24 09:00:16.809907637 +0100
@@ -3,7 +3,7 @@
 COPTS=@CFLAGS@
 
 DESTDIR = $(INSTROOT)@DESTDIR@
-BINDIR = $(DESTDIR)/sbin
+BINDIR = $(DESTDIR)/bin
 MANDIR = $(DESTDIR)/share/man/man8
 
 CFLAGS = $(COPTS) -I../include/net
diff -Nur ppp-2.4.6.orig/pppstats/Makefile.linux ppp-2.4.6/pppstats/Makefile.linux
--- ppp-2.4.6.orig/pppstats/Makefile.linux	2014-01-02 05:42:08.000000000 +0100
+++ ppp-2.4.6/pppstats/Makefile.linux	2014-02-24 09:00:16.809907637 +0100
@@ -6,7 +6,7 @@
 COPTS=@CFLAGS@
 
 DESTDIR = $(INSTROOT)@DESTDIR@
-BINDIR = $(DESTDIR)/sbin
+BINDIR = $(DESTDIR)/bin
 MANDIR = $(DESTDIR)/share/man/man8
 
 PPPSTATSRCS = pppstats.c
 
