--- PyQt5-5.13.2/project.py.orig	2019-12-13 19:02:20.618573711 +0000
+++ PyQt5-5.13.2/project.py	2019-12-13 19:02:31.142055299 +0000
@@ -306,10 +306,6 @@
         target_plugin_dir = os.path.join(
                 builder.qt_configuration['QT_INSTALL_PLUGINS'], target_subdir)
 
-        if not os.access(target_plugin_dir, os.W_OK):
-            self.progress("The {0} plugin was disabled because you do not have write permission on '{1}'.".format(user_name, target_plugin_dir))
-            return
-
         # Create the buildable and add it to the builder.
         buildable = Buildable(self, name)
         self.buildables.append(buildable)
