--- capture_freaklabs_zigbee/Makefile.in	2019-10-10 10:49:00.494256269 +0200
+++ capture_freaklabs_zigbee/Makefile.in	2019-10-10 10:52:49.374152190 +0200
@@ -10,7 +10,7 @@
 	sed -i -E 's/^import kismet_/from . import kismet_/' $@
 
 install:
-	$(PYTHON) setup.py install --root="/$(DESTDIR)" --prefix="$(prefix)"
+	$(PYTHON) setup.py install --root="/$(DESTDIR)" --prefix="$(prefix)" -O1

 clean:
 	@-$(PYTHON) setup.py clean
--- capture_sdr_rtl433/Makefile.in	2019-10-10 10:49:00.507590313 +0200
+++ capture_sdr_rtl433/Makefile.in	2019-10-10 10:52:28.069331801 +0200
@@ -11,7 +11,7 @@
 	sed -i -E 's/^import kismet_/from . import kismet_/' $@
 
 install:
-	$(PYTHON) setup.py install --root="/$(DESTDIR)" --prefix="$(prefix)"
+	$(PYTHON) setup.py install --root="/$(DESTDIR)" --prefix="$(prefix)" -O1

 clean:
 	@-$(PYTHON) setup.py clean
--- capture_sdr_rtladsb/Makefile.in	2019-10-10 10:49:00.510923825 +0200
+++ capture_sdr_rtladsb/Makefile.in	2019-10-10 10:52:32.542260533 +0200
@@ -11,7 +11,7 @@
 	sed -i -E 's/^import kismet_/from . import kismet_/' $@
 
 install:
-	$(PYTHON) setup.py install --root="/$(DESTDIR)" --prefix="$(prefix)"
+	$(PYTHON) setup.py install --root="/$(DESTDIR)" --prefix="$(prefix)" -O1

 clean:
 	@-$(PYTHON) setup.py clean
--- capture_sdr_rtlamr/Makefile.in	2019-10-10 10:49:00.514257335 +0200
+++ capture_sdr_rtlamr/Makefile.in	2019-10-10 10:52:35.978622527 +0200
@@ -11,7 +11,7 @@
 	sed -i -E 's/^import kismet_/from . import kismet_/' $@
 
 install:
-	$(PYTHON) setup.py install --root="/$(DESTDIR)" --prefix="$(prefix)"
+	$(PYTHON) setup.py install --root="/$(DESTDIR)" --prefix="$(prefix)" -O1

 clean:
 	@-$(PYTHON) setup.py clean

