diff --git i/etc/login.defs w/etc/login.defs
index 114dbcd9..1315e546 100644
--- i/etc/login.defs
+++ w/etc/login.defs
@@ -3,6 +3,8 @@
 #
 #	$Id$
 #
+# This file is adapted for the use on Arch Linux.
+# Options overlapping with tooling from util-linux are commented.
 
 #
 # Delay in seconds before being allowed another attempt after a login failure
@@ -14,7 +16,7 @@ FAIL_DELAY		3
 #
 # Enable logging and display of /var/log/faillog login(1) failure info.
 #
-FAILLOG_ENAB		yes
+# FAILLOG_ENAB		yes
 
 #
 # Enable display of unknown usernames when login(1) failures are recorded.
@@ -24,12 +26,12 @@ LOG_UNKFAIL_ENAB	no
 #
 # Enable logging of successful logins
 #
-LOG_OK_LOGINS		no
+# LOG_OK_LOGINS		no
 
 #
 # Enable logging and display of /var/log/lastlog login(1) time info.
 #
-LASTLOG_ENAB		yes
+# LASTLOG_ENAB		yes
 
 #
 # Limit the highest user ID number for which the lastlog entries should
@@ -46,28 +48,28 @@ LASTLOG_ENAB		yes
 # Disable if the shell startup files already check for mail
 # ("mailx -e" or equivalent).
 #
-MAIL_CHECK_ENAB		yes
+# MAIL_CHECK_ENAB		yes
 
 #
 # Enable additional checks upon password changes.
 #
-OBSCURE_CHECKS_ENAB	yes
+# OBSCURE_CHECKS_ENAB	yes
 
 #
 # Enable checking of time restrictions specified in /etc/porttime.
 #
-PORTTIME_CHECKS_ENAB	yes
+# PORTTIME_CHECKS_ENAB	yes
 
 #
 # Enable setting of ulimit, umask, and niceness from passwd(5) gecos field.
 #
-QUOTAS_ENAB		yes
+# QUOTAS_ENAB		yes
 
 #
 # Enable "syslog" logging of su(1) activity - in addition to sulog file logging.
 # SYSLOG_SG_ENAB does the same for newgrp(1) and sg(1).
 #
-SYSLOG_SU_ENAB		yes
+# SYSLOG_SU_ENAB		yes
 SYSLOG_SG_ENAB		yes
 
 #
@@ -87,7 +89,7 @@ CONSOLE		/etc/securetty
 # If defined, ":" delimited list of "message of the day" files to
 # be displayed upon login.
 #
-MOTD_FILE	/etc/motd
+MOTD_FILE
 #MOTD_FILE	/etc/motd:/usr/lib/news/news-motd
 
 #
@@ -105,14 +107,14 @@ MOTD_FILE	/etc/motd
 # If defined, login(1) failures will be logged here in a utmp format.
 # last(1), when invoked as lastb(1), will read /var/log/btmp, so...
 #
-FTMP_FILE	/var/log/btmp
+# FTMP_FILE	/var/log/btmp
 
 #
 # If defined, name of file whose presence will inhibit non-root
 # logins.  The content of this file should be a message indicating
 # why logins are inhibited.
 #
-NOLOGINS_FILE	/etc/nologin
+# NOLOGINS_FILE	/etc/nologin
 
 #
 # If defined, the command name to display when running "su -".  For
@@ -120,7 +122,7 @@ NOLOGINS_FILE	/etc/nologin
 # command as "-su".  If not defined, then ps(1) will display the
 # name of the shell actually being run, e.g. something like "-sh".
 #
-SU_NAME		su
+# SU_NAME		su
 
 #
 # *REQUIRED*
@@ -150,7 +152,7 @@ HUSHLOGIN_FILE	.hushlogin
 # If defined, an HZ environment parameter spec.
 #
 # for Linux/x86
-ENV_HZ		HZ=100
+# ENV_HZ		HZ=100
 # For Linux/Alpha...
 #ENV_HZ		HZ=1024
 
@@ -158,8 +160,8 @@ ENV_HZ		HZ=100
 # *REQUIRED*  The default PATH settings, for superuser and normal users.
 #
 # (they are minimal, add the rest in the shell startup files)
-ENV_SUPATH	PATH=/sbin:/bin:/usr/sbin:/usr/bin
-ENV_PATH	PATH=/bin:/usr/bin
+ENV_SUPATH	PATH=/usr/local/sbin:/usr/local/bin:/usr/bin
+ENV_PATH	PATH=/usr/local/sbin:/usr/local/bin:/usr/bin
 
 #
 # Terminal permissions
@@ -188,8 +190,8 @@ TTYPERM		0600
 #
 # Prefix these values with "0" to get octal, "0x" to get hexadecimal.
 #
-ERASECHAR	0177
-KILLCHAR	025
+# ERASECHAR	0177
+# KILLCHAR	025
 #ULIMIT		2097152
 
 # Default initial "umask" value used by login(1) on non-PAM enabled systems.
@@ -216,7 +218,7 @@ UMASK		022
 #
 PASS_MAX_DAYS	99999
 PASS_MIN_DAYS	0
-PASS_MIN_LEN	5
+# PASS_MIN_LEN	5
 PASS_WARN_AGE	7
 
 #
@@ -225,12 +227,12 @@ PASS_WARN_AGE	7
 # to uid 0 accounts.  If the group doesn't exist or is empty, no one
 # will be able to "su" to uid 0.
 #
-SU_WHEEL_ONLY	no
+# SU_WHEEL_ONLY	no
 
 #
 # If compiled with cracklib support, sets the path to the dictionaries
 #
-CRACKLIB_DICTPATH	/var/cache/cracklib/cracklib_dict
+# CRACKLIB_DICTPATH	/var/cache/cracklib/cracklib_dict
 
 #
 # Min/max values for automatic uid selection in useradd(8)
@@ -238,7 +240,7 @@ CRACKLIB_DICTPATH	/var/cache/cracklib/cracklib_dict
 UID_MIN			 1000
 UID_MAX			60000
 # System accounts
-SYS_UID_MIN		  101
+SYS_UID_MIN		  500
 SYS_UID_MAX		  999
 # Extra per user uids
 SUB_UID_MIN		   100000
@@ -251,7 +253,7 @@ SUB_UID_COUNT		    65536
 GID_MIN			 1000
 GID_MAX			60000
 # System accounts
-SYS_GID_MIN		  101
+SYS_GID_MIN		  500
 SYS_GID_MAX		  999
 # Extra per user group ids
 SUB_GID_MIN		   100000
@@ -271,12 +273,12 @@ LOGIN_TIMEOUT		60
 #
 # Maximum number of attempts to change password if rejected (too easy)
 #
-PASS_CHANGE_TRIES	5
+# PASS_CHANGE_TRIES	5
 
 #
 # Warn about weak passwords (but still allow them) if you are root.
 #
-PASS_ALWAYS_WARN	yes
+# PASS_ALWAYS_WARN	yes
 
 #
 # Number of significant characters in the password for crypt().
@@ -288,7 +290,7 @@ PASS_ALWAYS_WARN	yes
 #
 # Require password before chfn(1)/chsh(1) can make any changes.
 #
-CHFN_AUTH		yes
+# CHFN_AUTH		yes
 
 #
 # Which fields may be changed by regular users using chfn(1) - use
@@ -334,7 +336,7 @@ CHFN_RESTRICT		rwh
 # Note: If you use PAM, it is recommended to use a value consistent with
 # the PAM modules configuration.
 #
-#ENCRYPT_METHOD DES
+ENCRYPT_METHOD SHA512
 
 #
 # Only works if ENCRYPT_METHOD is set to SHA256 or SHA512.
@@ -410,7 +412,7 @@ NONEXISTENT	/nonexistent
 # If this file exists and is readable, login environment will be
 # read from it.  Every line should be in the form name=value.
 #
-ENVIRON_FILE	/etc/environment
+# ENVIRON_FILE	/etc/environment
 
 #
 # If defined, this command is run when removing a user.
@@ -465,7 +467,7 @@ USERGROUPS_ENAB yes
 # Set to "yes" to prevent for all accounts
 # Set to "superuser" to prevent for UID 0 / root (default)
 # Set to "no" to not prevent for any account (dangerous, historical default)
-PREVENT_NO_AUTH superuser
+# PREVENT_NO_AUTH superuser
 
 #
 # Select the HMAC cryptography algorithm.
