diff -upr xfce4-session-4.14.0.orig/scripts/xinitrc.in.in xfce4-session-4.14.0/scripts/xinitrc.in.in
--- xfce4-session-4.14.0.orig/scripts/xinitrc.in.in	2019-08-11 23:11:06.000000000 +0300
+++ xfce4-session-4.14.0/scripts/xinitrc.in.in	2019-08-12 03:28:44.464707715 +0300
@@ -83,6 +83,13 @@ if command -v systemctl >/dev/null 2>&1
   dbus-update-activation-environment --systemd XAUTHLOCALHOSTNAME=$XAUTHLOCALHOSTNAME
 fi
 
+# source system xinitrc scripts
+if [ -d /etc/X11/xinit/xinitrc.d ]; then
+  for f in /etc/X11/xinit/xinitrc.d/?*.sh; do
+    [ -x "$f" ] && . "$f"
+  done
+  unset f
+fi
 
 # check if we start xfce4-session with ck-launch-session. this is only
 # required for starting from a console, not a login manager
