diff -rupN tomcat-native.orig/native/buildconf tomcat-native/native/buildconf
--- tomcat-native.orig/native/buildconf	2020-09-20 14:44:45.114714616 +0200
+++ tomcat-native/native/buildconf	2020-09-20 14:44:58.368172703 +0200
@@ -55,7 +55,7 @@ if [ ! -d "$apr_src_dir/build" ]; then
 fi
 
 # Remove some files, then copy them from apr source tree
-for file in apr_common.m4 find_apr.m4 install.sh config.guess config.sub
+for file in apr_common.m4 find_apr.m4 install.sh
 do
   if [ ! -f "$apr_src_dir/build/$file" ]; then
     echo "File '$apr_src_dir/build/$file' missing - wrong apr source directory?"
