diff -u -r gnome-disk-utility-3.8.2/src/disks/gducreatediskimagedialog.c gnome-disk-utility-3.8.2-typefixes/src/disks/gducreatediskimagedialog.c
--- gnome-disk-utility-3.8.2/src/disks/gducreatediskimagedialog.c	2013-05-14 07:15:48.000000000 +0200
+++ gnome-disk-utility-3.8.2-typefixes/src/disks/gducreatediskimagedialog.c	2013-05-29 03:17:37.201963964 +0200
@@ -610,7 +610,7 @@
                                   error))
     {
       g_prefix_error (error,
-                      "Error writing %" G_GUINT64_FORMAT " bytes to offset %" G_GUINT64_FORMAT ": ",
+                      "Error writing %" G_GSIZE_FORMAT " bytes to offset %" G_GUINT64_FORMAT ": ",
                       num_bytes_to_write,
                       offset);
       goto out;
diff -u -r gnome-disk-utility-3.8.2/src/disks/gducreatefilesystemwidget.c gnome-disk-utility-3.8.2-typefixes/src/disks/gducreatefilesystemwidget.c
--- gnome-disk-utility-3.8.2/src/disks/gducreatefilesystemwidget.c	2013-05-14 07:15:48.000000000 +0200
+++ gnome-disk-utility-3.8.2-typefixes/src/disks/gducreatefilesystemwidget.c	2013-05-29 03:18:30.224935009 +0200
@@ -429,7 +429,7 @@
   if (widget->drive != NULL && udisks_drive_get_removable (widget->drive))
     {
       /* default FAT for flash and disks/media smaller than 20G (assumed to be flash cards) */
-      if (is_flash (widget->drive) || udisks_drive_get_size (widget->drive) < 20L * 1000L*1000L*1000L)
+      if (is_flash (widget->drive) || udisks_drive_get_size (widget->drive) < (guint64)(20ULL * 1000ULL*1000ULL*1000ULL))
         {
           gtk_combo_box_set_active_id (GTK_COMBO_BOX (widget->type_combobox), "vfat");
         }
diff -u -r gnome-disk-utility-3.8.2/src/disks/gduformatdiskdialog.c gnome-disk-utility-3.8.2-typefixes/src/disks/gduformatdiskdialog.c
--- gnome-disk-utility-3.8.2/src/disks/gduformatdiskdialog.c	2013-05-14 07:15:48.000000000 +0200
+++ gnome-disk-utility-3.8.2-typefixes/src/disks/gduformatdiskdialog.c	2013-05-29 03:17:37.338629696 +0200
@@ -293,7 +293,7 @@
   /* Default to MBR for removable drives < 2TB... GPT otherwise */
   if (data->drive != NULL &&
       udisks_drive_get_removable (data->drive) &&
-      udisks_drive_get_size (data->drive) < 2 * 1000L*1000L*1000L*1000L)
+      udisks_drive_get_size (data->drive) < (guint64)(2ULL * 1000ULL*1000ULL*1000ULL*1000ULL))
     {
       gtk_combo_box_set_active_id (GTK_COMBO_BOX (data->type_combobox), "dos");
     }
diff -u -r gnome-disk-utility-3.8.2/src/disks/gdupartitiondialog.c gnome-disk-utility-3.8.2-typefixes/src/disks/gdupartitiondialog.c
--- gnome-disk-utility-3.8.2/src/disks/gdupartitiondialog.c	2013-05-14 07:15:48.000000000 +0200
+++ gnome-disk-utility-3.8.2-typefixes/src/disks/gdupartitiondialog.c	2013-05-29 03:17:37.338629696 +0200
@@ -91,11 +91,11 @@
       if (gtk_toggle_button_get_active (GTK_TOGGLE_BUTTON (data->bootable_checkbutton)))
         flags |= (1UL<<2);
       if (gtk_toggle_button_get_active (GTK_TOGGLE_BUTTON (data->readonly_checkbutton)))
-        flags |= (1UL<<60);
+        flags |= (1ULL<<60);
       if (gtk_toggle_button_get_active (GTK_TOGGLE_BUTTON (data->hidden_checkbutton)))
-        flags |= (1UL<<62);
+        flags |= (1ULL<<62);
       if (gtk_toggle_button_get_active (GTK_TOGGLE_BUTTON (data->do_not_automount_checkbutton)))
-        flags |= (1UL<<63);
+        flags |= (1ULL<<63);
     }
   else if (g_strcmp0 (data->partition_table_type, "dos") == 0)
     {
@@ -245,9 +245,9 @@
       flags = udisks_partition_get_flags (data->partition);
       gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (data->system_checkbutton),           (flags & (1UL<< 0)) != 0);
       gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (data->bootable_checkbutton),         (flags & (1UL<< 2)) != 0);
-      gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (data->readonly_checkbutton),         (flags & (1UL<<60)) != 0);
-      gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (data->hidden_checkbutton),           (flags & (1UL<<62)) != 0);
-      gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (data->do_not_automount_checkbutton), (flags & (1UL<<63)) != 0);
+      gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (data->readonly_checkbutton),         (flags & (1ULL<<60)) != 0);
+      gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (data->hidden_checkbutton),           (flags & (1ULL<<62)) != 0);
+      gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (data->do_not_automount_checkbutton), (flags & (1ULL<<63)) != 0);
     }
   else if (g_strcmp0 (data->partition_table_type, "dos") == 0)
     {
diff -u -r gnome-disk-utility-3.8.2/src/disks/gdurestorediskimagedialog.c gnome-disk-utility-3.8.2-typefixes/src/disks/gdurestorediskimagedialog.c
--- gnome-disk-utility-3.8.2/src/disks/gdurestorediskimagedialog.c	2013-05-14 07:15:48.000000000 +0200
+++ gnome-disk-utility-3.8.2-typefixes/src/disks/gdurestorediskimagedialog.c	2013-05-29 03:17:37.485295361 +0200
@@ -587,7 +587,7 @@
                                     &error))
         {
           g_prefix_error (&error,
-                          "Error reading %" G_GUINT64_FORMAT " bytes from offset %" G_GUINT64_FORMAT ": ",
+                          "Error reading %" G_GSIZE_FORMAT " bytes from offset %" G_GUINT64_FORMAT ": ",
                           num_bytes_to_read,
                           num_bytes_completed);
           goto out;
@@ -595,7 +595,7 @@
       if (num_bytes_read != num_bytes_to_read)
         {
           g_set_error (&error, G_IO_ERROR, G_IO_ERROR_FAILED,
-                       "Requested %" G_GUINT64_FORMAT " bytes from offset %" G_GUINT64_FORMAT " but only read %" G_GUINT64_FORMAT " bytes",
+                       "Requested %" G_GSIZE_FORMAT " bytes from offset %" G_GUINT64_FORMAT " but only read %" G_GSIZE_FORMAT " bytes",
                        num_bytes_read,
                        num_bytes_completed,
                        num_bytes_to_read);
@@ -610,7 +610,7 @@
             goto copy_write_again;
 
           g_set_error (&error, G_IO_ERROR, G_IO_ERROR_FAILED,
-                       "Error writing %" G_GUINT64_FORMAT " bytes to offset %" G_GUINT64_FORMAT ": %m",
+                       "Error writing %" G_GSIZE_FORMAT " bytes to offset %" G_GUINT64_FORMAT ": %m",
                        num_bytes_read,
                        num_bytes_completed);
           goto out;
