--- a/src/core/main.c	2012-09-20 23:02:54.000000000 -0400
+++ b/src/core/main.c	2012-09-20 23:03:13.000000000 -0400
@@ -1455,11 +1455,7 @@
 
         /* Set up PATH unless it is already set */
         setenv("PATH",
-#ifdef HAVE_SPLIT_USR
                "/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
-#else
-               "/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin",
-#endif
                arg_running_as == SYSTEMD_SYSTEM);
 
         if (arg_running_as == SYSTEMD_SYSTEM) {
