--- miro.real	2008-10-24 19:49:17.000000000 +0200
+++ miro.real.new	2008-11-12 08:51:17.649458321 +0100
@@ -6,8 +6,12 @@
 if xlibhelper.XInitThreads() == 0:
     print "WARNING: XInitThreads() failed!"
 
-import os.path
+import miro.feedparser
+import miro.storedatabase
 import sys
+sys.modules['feedparser'] = miro.feedparser
+sys.modules['storedatabase'] = miro.storedatabase
+import os.path
 import optparse
 
 import pygtk
