diff -Naur fitPlugins/exp_saturation/exp_saturation.pro fitPlugins.new/exp_saturation/exp_saturation.pro
--- fitPlugins/exp_saturation/exp_saturation.pro	2008-11-05 00:06:49.884616891 +0100
+++ fitPlugins.new/exp_saturation/exp_saturation.pro	2008-11-05 00:08:51.259644717 +0100
@@ -12,12 +12,13 @@
 DESTDIR           = ../
  
 # statically link against GSL in 3rdparty
-INCLUDEPATH += ../../3rdparty/gsl/include/
-LIBS        += ../../3rdparty/gsl/lib/libgsl.a
-LIBS        += ../../3rdparty/gsl/lib/libgslcblas.a
+#INCLUDEPATH += ../../3rdparty/gsl/include/
+#LIBS        += ../../3rdparty/gsl/lib/libgsl.a
+#LIBS        += ../../3rdparty/gsl/lib/libgslcblas.a
 
 #dynamically link against GSL installed system-wide
-#unix:LIBS += -L /usr/lib$${libsuff} -lgsl -lgslcblas
+INCLUDEPATH += /usr/include/gsl
+unix:LIBS += $$system(pkg-config --libs gsl)
 
 target.path=/usr/lib$${libsuff}/qtiplot/plugins
 INSTALLS += target
diff -Naur fitPlugins/explin/explin.pro fitPlugins.new/explin/explin.pro
--- fitPlugins/explin/explin.pro	2008-10-13 09:43:31.000000000 +0200
+++ fitPlugins.new/explin/explin.pro	2008-11-05 00:09:13.118018704 +0100
@@ -12,12 +12,13 @@
 DESTDIR           = ../
  
 # statically link against GSL in 3rdparty
-INCLUDEPATH += ../../3rdparty/gsl/include/
-LIBS        += ../../3rdparty/gsl/lib/libgsl.a
-LIBS        += ../../3rdparty/gsl/lib/libgslcblas.a
+#INCLUDEPATH += ../../3rdparty/gsl/include/
+#LIBS        += ../../3rdparty/gsl/lib/libgsl.a
+#LIBS        += ../../3rdparty/gsl/lib/libgslcblas.a
 
 #dynamically link against GSL installed system-wide
-#unix:LIBS += -L /usr/lib$${libsuff} -lgsl -lgslcblas
+INCLUDEPATH += /usr/include/gsl
+unix:LIBS += $$system(pkg-config --libs gsl)
 
 target.path=/usr/lib$${libsuff}/qtiplot/plugins
 INSTALLS += target
diff -Naur fitPlugins/fitRational0/fitRational0.pro fitPlugins.new/fitRational0/fitRational0.pro
--- fitPlugins/fitRational0/fitRational0.pro	2008-10-13 18:14:52.000000000 +0200
+++ fitPlugins.new/fitRational0/fitRational0.pro	2008-11-05 00:09:28.755220241 +0100
@@ -12,14 +12,15 @@
 DESTDIR           = ../
  
 # statically link against GSL in 3rdparty
-INCLUDEPATH += ../../3rdparty/gsl/include/
-LIBS        += ../../3rdparty/gsl/lib/libgsl.a
-LIBS        += ../../3rdparty/gsl/lib/libgslcblas.a
+#INCLUDEPATH += ../../3rdparty/gsl/include/
+#LIBS        += ../../3rdparty/gsl/lib/libgsl.a
+#LIBS        += ../../3rdparty/gsl/lib/libgslcblas.a
 
 #dynamically link against GSL installed system-wide
-#unix:LIBS += -L /usr/lib$${libsuff} -lgsl -lgslcblas
+INCLUDEPATH += /usr/include/gsl
+unix:LIBS += $$system(pkg-config --libs gsl)
 
 target.path=/usr/lib$${libsuff}/qtiplot/plugins
 INSTALLS += target
 
-SOURCES += fitRational0.cpp
+SOURCES += fitRational0.cpp
diff -Naur fitPlugins/fitRational1/fitRational1.pro fitPlugins.new/fitRational1/fitRational1.pro
--- fitPlugins/fitRational1/fitRational1.pro	2007-10-29 00:44:06.000000000 +0100
+++ fitPlugins.new/fitRational1/fitRational1.pro	2008-11-05 00:09:42.154619945 +0100
@@ -12,12 +12,13 @@
 DESTDIR           = ../
  
 # statically link against GSL in 3rdparty
-INCLUDEPATH += ../../3rdparty/gsl/include/
-LIBS        += ../../3rdparty/gsl/lib/libgsl.a
-LIBS        += ../../3rdparty/gsl/lib/libgslcblas.a
+#INCLUDEPATH += ../../3rdparty/gsl/include/
+#LIBS        += ../../3rdparty/gsl/lib/libgsl.a
+#LIBS        += ../../3rdparty/gsl/lib/libgslcblas.a
 
 #dynamically link against GSL installed system-wide
-#unix:LIBS += -L /usr/lib$${libsuff} -lgsl -lgslcblas
+INCLUDEPATH += /usr/include/gsl
+unix:LIBS += $$system(pkg-config --libs gsl)
 
 target.path=/usr/lib$${libsuff}/qtiplot/plugins
 INSTALLS += target
