# fix gcc-4.3 missing headers 

diff -Naur qtiplot-0.9.7.2/3rdparty/liborigin/OriginObj.h qtiplot-0.9.7.2.new/3rdparty/liborigin/OriginObj.h
--- ../3rdparty/liborigin/OriginObj.h	2008-09-21 13:22:33.000000000 -0400
+++ ../3rdparty/liborigin/OriginObj.h	2008-09-25 10:07:27.000000000 -0400
@@ -33,6 +33,7 @@
 
 #include <string>
 #include <vector>
+#include <cstring>
 #include "boost/variant.hpp"
 #include "boost/date_time/posix_time/ptime.hpp"
 

