--- qtiplot.pro	2008-11-05 11:32:26.973838219 +0100
+++ qtiplot.pro.new	2008-11-05 11:32:54.793839850 +0100
@@ -23,7 +23,7 @@
 win32:LIBS 		  += ../3rdparty/libpng/libpng.a
 
 # Uncomment the following line if you want to perform a custom installation using the *.path variables defined bellow.
-#CONFIG          += CustomInstall
+CONFIG          += CustomInstall
 
 CONFIG          += release
 #CONFIG          += debug
@@ -33,13 +33,12 @@
 #!!! Warning: You must modify these paths according to your computer settings
 #############################################################################
 
-INCLUDEPATH       += ../3rdparty/muparser/include
-INCLUDEPATH       += ../3rdparty/qwtplot3d/include
-INCLUDEPATH       += ../3rdparty/qwt/src
+INCLUDEPATH       += /usr/include
+INCLUDEPATH       += /usr/include/qwtplot3d
+INCLUDEPATH       += /usr/include/qwt
 INCLUDEPATH       += ../3rdparty/liborigin
-INCLUDEPATH       += ../3rdparty/gsl/include
-INCLUDEPATH       += ../3rdparty/zlib123/include
-INCLUDEPATH       += ../3rdparty/boost_1_36_0
+INCLUDEPATH       += /usr/include/gsl
+INCLUDEPATH       += /usr/include/boost
 
 ##################### 3rd PARTY LIBRARIES SECTION ###########################
 #!!! Warning: You must modify these paths according to your computer settings
@@ -48,18 +47,18 @@
 ##################### Linux (Mac OS X) ######################################
 
 # statically link against libraries in 3rdparty
-unix:LIBS         += ../3rdparty/muparser/lib/libmuparser.a
-unix:LIBS         += ../3rdparty/qwt/lib/libqwt.a
-unix:LIBS         += ../3rdparty/gsl/lib/libgsl.a
-unix:LIBS         += ../3rdparty/gsl/lib/libgslcblas.a
-unix:LIBS         += /usr/local/lib/libboost_date_time-gcc41-mt.a
-unix:LIBS         += /usr/local/lib/libboost_thread-gcc41-mt.a
+#unix:LIBS         += ../3rdparty/muparser/lib/libmuparser.a
+#unix:LIBS         += ../3rdparty/qwt/lib/libqwt.a
+#unix:LIBS         += ../3rdparty/gsl/lib/libgsl.a
+#unix:LIBS         += ../3rdparty/gsl/lib/libgslcblas.a
+#unix:LIBS         += /usr/local/lib/libboost_date_time-gcc41-mt.a
+#unix:LIBS         += /usr/local/lib/libboost_thread-gcc41-mt.a
 
 # dynamically link against dependencies if they are installed system-wide
 #unix:LIBS         += -lmuparser
 #unix:LIBS         += -lqwt
 #unix:LIBS         += -lgsl -lgslcblas
-
+unix:LIBS         += -lmuparser -lqwtplot3d -lqwt $$system(pkg-config --libs gsl) -lz -lboost_date_time -lboost_thread-mt
 ##################### Windows ###############################################
 
 win32:LIBS        += ../3rdparty/muparser/lib/libmuparser.a
@@ -184,7 +183,7 @@
 ################# Default Modules #############################
 ###############################################################
 
-include(../3rdparty/qwtplot3d/qwtplot3d.pri)
+#include(../3rdparty/qwtplot3d/qwtplot3d.pri)
 include(src/analysis/analysis.pri)
 include(src/core/core.pri)
 include(src/lib/libqti.pri)
@@ -242,4 +241,4 @@
   }
 }
 
-###############################################################
+###############################################################
