=== modified file 'build.conf.example'
--- build.conf.example	2010-04-22 18:57:40 +0000
+++ build.conf.example	2010-04-22 18:58:06 +0000
@@ -118,6 +118,14 @@
 EMF_ENGINE_LIBS = $$QTI_ROOT/3rdparty/EmfEngine/libEmfEngine.a
 
 ##########################################################
+## liborigin - keep commented if you wish to use local copy
+#  http://sourceforge.net/projects/liborigin/
+##########################################################
+
+# LIB_ORIGIN_INCLUDEPATH = /usr/include/liborigin2
+# LIB_ORIGIN_LIBS = -lorigin2
+
+##########################################################
 ## python - only used if python is needed
 ##########################################################
 

=== modified file 'qtiplot/qtiplot.pro'
--- qtiplot/qtiplot.pro	2010-04-22 18:57:40 +0000
+++ qtiplot/qtiplot.pro	2010-04-22 19:08:57 +0000
@@ -16,6 +16,7 @@
 INCLUDEPATH       += $$QWT_INCLUDEPATH
 INCLUDEPATH       += $$QWT3D_INCLUDEPATH
 INCLUDEPATH       += $$GSL_INCLUDEPATH
+INCLUDEPATH       += $$LIB_ORIGIN_INCLUDEPATH
 
 # configurable libs
 LIBS         += $$MUPARSER_LIBS
@@ -23,6 +24,7 @@
 LIBS         += $$QWT3D_LIBS
 LIBS         += $$QTEXENGINE_LIBS
 LIBS         += $$GSL_LIBS
+LIBS         += $$LIB_ORIGIN_LIBS
 
 #############################################################################
 ###################### BASIC PROJECT PROPERTIES #############################
@@ -150,6 +152,10 @@
 	SOURCES     += ../3rdparty/QTeXEngine/src/QTeXPaintDevice.cpp
 }
 
+isEmpty(LIB_ORIGIN_LIBS) {
+	INCLUDEPATH       += ../3rdparty/liborigin
+}
+
 ###############################################################
 ##################### Scripting: PYTHON + SIP + PyQT ##########
 ###############################################################

=== modified file 'qtiplot/src/origin/origin.pri'
--- qtiplot/src/origin/origin.pri	2010-04-22 17:58:49 +0000
+++ qtiplot/src/origin/origin.pri	2010-04-22 19:08:15 +0000
@@ -4,24 +4,28 @@
 INCLUDEPATH += src/origin/
 
 HEADERS += src/origin/importOPJ.h
-HEADERS += ../3rdparty/liborigin/OriginObj.h
-HEADERS += ../3rdparty/liborigin/OriginFile.h
-HEADERS += ../3rdparty/liborigin/OriginParser.h
-HEADERS += ../3rdparty/liborigin/OriginDefaultParser.h
-HEADERS += ../3rdparty/liborigin/Origin600Parser.h
-HEADERS += ../3rdparty/liborigin/Origin610Parser.h
-HEADERS += ../3rdparty/liborigin/Origin700Parser.h
-HEADERS += ../3rdparty/liborigin/Origin750Parser.h
-HEADERS += ../3rdparty/liborigin/Origin800Parser.h
-HEADERS += ../3rdparty/liborigin/Origin810Parser.h
 
 SOURCES += src/origin/importOPJ.cpp
-SOURCES += ../3rdparty/liborigin/OriginFile.cpp
-SOURCES += ../3rdparty/liborigin/OriginParser.cpp
-SOURCES += ../3rdparty/liborigin/OriginDefaultParser.cpp
-SOURCES += ../3rdparty/liborigin/Origin600Parser.cpp
-SOURCES += ../3rdparty/liborigin/Origin610Parser.cpp
-SOURCES += ../3rdparty/liborigin/Origin700Parser.cpp
-SOURCES += ../3rdparty/liborigin/Origin750Parser.cpp
-SOURCES += ../3rdparty/liborigin/Origin800Parser.cpp
-SOURCES += ../3rdparty/liborigin/Origin810Parser.cpp
+
+isEmpty(LIB_ORIGIN_LIBS) {
+	HEADERS += ../3rdparty/liborigin/OriginObj.h
+	HEADERS += ../3rdparty/liborigin/OriginFile.h
+	HEADERS += ../3rdparty/liborigin/OriginParser.h
+	HEADERS += ../3rdparty/liborigin/OriginDefaultParser.h
+	HEADERS += ../3rdparty/liborigin/Origin600Parser.h
+	HEADERS += ../3rdparty/liborigin/Origin610Parser.h
+	HEADERS += ../3rdparty/liborigin/Origin700Parser.h
+	HEADERS += ../3rdparty/liborigin/Origin750Parser.h
+	HEADERS += ../3rdparty/liborigin/Origin800Parser.h
+	HEADERS += ../3rdparty/liborigin/Origin810Parser.h
+
+	SOURCES += ../3rdparty/liborigin/OriginFile.cpp
+	SOURCES += ../3rdparty/liborigin/OriginParser.cpp
+	SOURCES += ../3rdparty/liborigin/OriginDefaultParser.cpp
+	SOURCES += ../3rdparty/liborigin/Origin600Parser.cpp
+	SOURCES += ../3rdparty/liborigin/Origin610Parser.cpp
+	SOURCES += ../3rdparty/liborigin/Origin700Parser.cpp
+	SOURCES += ../3rdparty/liborigin/Origin750Parser.cpp
+	SOURCES += ../3rdparty/liborigin/Origin800Parser.cpp
+	SOURCES += ../3rdparty/liborigin/Origin810Parser.cpp
+}
