From 88e8a8bbaf026aa10225880001ab7ca1c392168a Mon Sep 17 00:00:00 2001
From: Ben Skeggs <bskeggs@redhat.com>
Date: Tue, 16 Feb 2010 05:42:00 +0000
Subject: nouveau: bump MAX_PUSH to 512

---
diff --git a/include/drm/nouveau_drm.h b/include/drm/nouveau_drm.h
index af13e46..a6a9f4a 100644
--- a/include/drm/nouveau_drm.h
+++ b/include/drm/nouveau_drm.h
@@ -140,7 +140,7 @@ struct drm_nouveau_gem_pushbuf_reloc {
 	uint32_t tor;
 };
 
-#define NOUVEAU_GEM_MAX_PUSH 64
+#define NOUVEAU_GEM_MAX_PUSH 512
 struct drm_nouveau_gem_pushbuf_push {
 	uint32_t bo_index;
 	uint32_t pad;
--
cgit v0.8.3-6-g21f6
