diff -urN cdw-0.2.4/src/cddb.c cdw-0.2.4-patched/src/cddb.c
--- cdw-0.2.4/src/cddb.c	2005-11-07 04:18:38.000000000 -0500
+++ cdw-0.2.4-patched/src/cddb.c	2005-11-11 23:58:34.000000000 -0500
@@ -905,7 +905,7 @@
 		    file_list[c].cd_name=(char *)malloc(strlen(row[1])+1);
 		    sprintf(file_list[c].cd_name, "%s", row[1]);
 		    sprintf(line, "%s", row[2]);
-		    (char *)linestr=strtok((char *)line, (char *)"/");
+		    linestr=strtok((char *)line, (char *)"/");
 		    if (dir_level>2)
 			for (i=1; i<=(dir_level-2); i++){
 			    linestr=strtok(NULL, "/");
