--- build.xml	2012-09-14 18:22:48.000000000 +0000
+++ build.xml.patched	2013-01-09 10:19:28.414079935 +0000
@@ -46,9 +46,10 @@
 				<include name="**/*.rsc"/>
 				<include name="META-INF/**"/>
 				<include name="**/*.properties"/>
+				<include name="**/*.props"/>
 				<exclude name="META-INF/eclipse.inf"/>
 			</fileset>
 		</zip>
 		<delete dir="${output}" />
 	</target>
-</project>
\ No newline at end of file
+</project>
