diff -ruN a/src/pip/_vendor/__init__.py b/src/pip/_vendor/__init__.py
--- a/src/pip/_vendor/__init__.py	2022-05-31 13:05:52.000000000 +0200
+++ b/src/pip/_vendor/__init__.py	2022-06-27 13:03:22.909131639 +0200
@@ -14,7 +14,7 @@
 # Downstream redistributors which have debundled our dependencies should also
 # patch this value to be true. This will trigger the additional patching
 # to cause things like "six" to be available as pip.
-DEBUNDLED = False
+DEBUNDLED = True
 
 # By default, look in this directory for a bunch of .whl files which we will
 # add to the beginning of sys.path before attempting to import anything. This
