diff -upr wstools-0.4.10.orig/wstools/Utility.py wstools-0.4.10/wstools/Utility.py
--- wstools-0.4.10.orig/wstools/Utility.py	2020-04-15 17:15:13.000000000 +0300
+++ wstools-0.4.10/wstools/Utility.py	2021-12-07 13:46:43.544654058 +0200
@@ -33,7 +33,7 @@ try:
     from UserDict import DictMixin  # noqa
 except ImportError:
     from collections import UserDict
-    from collections import MutableMapping as DictMixin  # noqa
+    from collections.abc import MutableMapping as DictMixin  # noqa
 
 from .TimeoutSocket import TimeoutSocket, TimeoutError  # noqa
 
