--- libjava/classpath/tools/Makefile.am.orig	2007-03-17 18:43:46.000000000 +0100
+++ libjava/classpath/tools/Makefile.am	2007-03-17 18:44:19.000000000 +0100
@@ -204,13 +204,13 @@
 ## END GCJ LOCAL
 ## First add classpath tools stuff.
 	(cd classes; \
-	if test "$(ZIP)" != ""; then $(ZIP) -r ../$(TOOLS_ZIP) .; fi; \
-	if test "$(FASTJAR)" != ""; then $(FASTJAR) cf ../$(TOOLS_ZIP) .; fi; \
+	if test "$(ZIP)" != ""; then $(ZIP) -r ../$(TOOLS_ZIP) *; fi; \
+	if test "$(FASTJAR)" != ""; then $(FASTJAR) cf ../$(TOOLS_ZIP) *; fi; \
 	cd ..)
 ## Now add ASM classes.
 	(cd asm; \
-	if test "$(ZIP)" != ""; then $(ZIP) -u -r ../$(TOOLS_ZIP) .; fi; \
-	if test "$(FASTJAR)" != ""; then $(FASTJAR) uf ../$(TOOLS_ZIP) .; fi; \
+	if test "$(ZIP)" != ""; then $(ZIP) -u -r ../$(TOOLS_ZIP) *; fi; \
+	if test "$(FASTJAR)" != ""; then $(FASTJAR) uf ../$(TOOLS_ZIP) *; fi; \
 	cd ..)
 	rm -rf asm classes classes.lst asm.lst
 
--- libjava/classpath/tools/Makefile.in.orig	2007-03-17 18:43:55.000000000 +0100
+++ libjava/classpath/tools/Makefile.in	2007-03-17 18:44:44.000000000 +0100
@@ -1121,12 +1121,12 @@
 	cp -pR $(srcdir)/asm .
 	cp -pR $(srcdir)/classes .
 	(cd classes; \
-	if test "$(ZIP)" != ""; then $(ZIP) -r ../$(TOOLS_ZIP) .; fi; \
-	if test "$(FASTJAR)" != ""; then $(FASTJAR) cf ../$(TOOLS_ZIP) .; fi; \
+	if test "$(ZIP)" != ""; then $(ZIP) -r ../$(TOOLS_ZIP) *; fi; \
+	if test "$(FASTJAR)" != ""; then $(FASTJAR) cf ../$(TOOLS_ZIP) *; fi; \
 	cd ..)
 	(cd asm; \
-	if test "$(ZIP)" != ""; then $(ZIP) -u -r ../$(TOOLS_ZIP) .; fi; \
-	if test "$(FASTJAR)" != ""; then $(FASTJAR) uf ../$(TOOLS_ZIP) .; fi; \
+	if test "$(ZIP)" != ""; then $(ZIP) -u -r ../$(TOOLS_ZIP) *; fi; \
+	if test "$(FASTJAR)" != ""; then $(FASTJAR) uf ../$(TOOLS_ZIP) *; fi; \
 	cd ..)
 	rm -rf asm classes classes.lst asm.lst
 
