--- old/fig2dev/dev/genpstex.c	2005-12-19 02:57:40.000000000 +0100
+++ new/fig2dev/dev/genpstex.c	2009-03-16 16:18:06.000000000 +0100
@@ -85,7 +85,7 @@
 F_compound	*objects;
 {
 	/* Put PostScript Image if any*/
-fprintf(stderr,"here pstex_file[0] = %d\n",pstex_file[0]);
+/*fprintf(stderr,"here pstex_file[0] = %d\n",pstex_file[0]);*/
         if (pstex_file[0] != '\0') {
 		fprintf(tfp, "\\begin{picture}(0,0)%%\n");
 /* newer includegraphics directive suggested by Stephen Harker 1/13/99 */
@@ -98,7 +98,7 @@
 #else
 		fprintf(tfp, "\\special{psfile=%s}%%\n",pstex_file);
 #endif
-fprintf(stderr,"end picture\n");
+/*fprintf(stderr,"end picture\n");*/
 		fprintf(tfp, "\\end{picture}%%\n");
 	}
         genlatex_start(objects);
