Index: trunk/Source/WebCore/html/HTMLCanvasElement.cpp
===================================================================
--- trunk/Source/WebCore/html/HTMLCanvasElement.cpp	(revision 139541)
+++ trunk/Source/WebCore/html/HTMLCanvasElement.cpp	(revision 139742)
@@ -190,5 +190,5 @@
     Settings* settings = document()->settings();
     if (settings && settings->webGLEnabled()
-#if !PLATFORM(CHROMIUM) && !PLATFORM(GTK) && !PLATFORM(EFL)
+#if !PLATFORM(CHROMIUM) && !PLATFORM(GTK) && !PLATFORM(EFL) && !PLATFORM(QT)
         && settings->acceleratedCompositingEnabled()
 #endif
