Index: gnome-alsamixer-0.9.7~cvs.20060916.ds.1/src/gam-main.c
===================================================================
--- gnome-alsamixer-0.9.7~cvs.20060916.ds.1.orig/src/gam-main.c	2005-07-06 17:14:57.000000000 +0200
+++ gnome-alsamixer-0.9.7~cvs.20060916.ds.1/src/gam-main.c	2011-10-26 02:47:47.000000000 +0200
@@ -22,7 +22,7 @@
 #include <config.h>
 #endif
 
-#include <gnome.h>
+#include <libgnomeui/gnome-ui-init.h>
 #include <glib/gi18n.h>
 
 #include "gam-app.h"
Index: gnome-alsamixer-0.9.7~cvs.20060916.ds.1/src/gam-app.c
===================================================================
--- gnome-alsamixer-0.9.7~cvs.20060916.ds.1.orig/src/gam-app.c	2011-10-26 02:47:21.000000000 +0200
+++ gnome-alsamixer-0.9.7~cvs.20060916.ds.1/src/gam-app.c	2011-10-26 02:47:21.000000000 +0200
@@ -27,7 +27,7 @@
 #include <gtk/gtkmenuitem.h>
 #include <glib/gi18n.h>
 #include <libgnomeui/gnome-about.h>
-#include <libgnomeui/gnome-app-helper.h>
+/*#include <libgnomeui/gnome-app-helper.h>*/
 #include <libgnomeui/gnome-stock-icons.h>
 
 #include "gam-app.h"
