--- nx-libs-3.5.0.17/nx-X11/programs/Xserver/hw/nxagent/Keyboard.c	2013-01-04 13:25:49.000000000 +0100
+++ nx-libs-3.5.0.17/nx-X11/programs/Xserver/hw/nxagent/Keyboard.c.new	2013-03-15 18:24:54.084216888 +0100
@@ -140,7 +140,7 @@
 #define XKB_CONFIG_FILE      "/etc/nxagent/nxagent.keyboard"
 #endif
 #ifndef XKB_DFLT_RULES_FILE
-#define XKB_DFLT_RULES_FILE  "xfree86"
+#define XKB_DFLT_RULES_FILE  "base"
 #endif
 #ifndef XKB_ALTS_RULES_FILE
 #define XKB_ALTS_RULES_FILE  "xorg"
