diff -u -r gnome-session-3.4.2/tools/gnome-session-check-accelerated.c gnome-session-3.4.2-timeout/tools/gnome-session-check-accelerated.c
--- gnome-session-3.4.2/tools/gnome-session-check-accelerated.c	2011-03-22 21:31:43.000000000 +0100
+++ gnome-session-3.4.2-timeout/tools/gnome-session-check-accelerated.c	2012-06-04 11:06:35.911690067 +0200
@@ -30,7 +30,7 @@
 #include <X11/Xatom.h>
 
 /* Wait up to this long for a running check to finish */
-#define PROPERTY_CHANGE_TIMEOUT 5000
+#define PROPERTY_CHANGE_TIMEOUT 10000
 
 /* Values used for the _GNOME_SESSION_ACCELERATED root window property */
 #define NO_ACCEL            0
