diff --git a/src/kbuildsycoca/kbuildsycoca.cpp b/src/kbuildsycoca/kbuildsycoca.cpp
index 827ac96..6d34235 100644
--- a/src/kbuildsycoca/kbuildsycoca.cpp
+++ b/src/kbuildsycoca/kbuildsycoca.cpp
@@ -515,7 +515,7 @@ void KBuildSycoca::save(QDataStream *str)
     (*str) << qint32(0); // No more factories.
     // Write XDG_DATA_DIRS
     (*str) << QStandardPaths::standardLocations(QStandardPaths::GenericDataLocation).join(QString(QLatin1Char(':')));
-    (*str) << (quint32)newTimestamp / 1000; // TODO just newTimestamp when using a new filename
+    (*str) << (quint32)(newTimestamp / 1000); // TODO just newTimestamp when using a new filename
     (*str) << QLocale().bcp47Name();
     // This makes it possible to trigger a ksycoca update for all users (KIOSK feature)
     (*str) << calcResourceHash(QStringLiteral("kservices5"), QStringLiteral("update_ksycoca"));
diff --git a/src/sycoca/ksycoca.cpp b/src/sycoca/ksycoca.cpp
index 81c875d..44984fa 100644
--- a/src/sycoca/ksycoca.cpp
+++ b/src/sycoca/ksycoca.cpp
@@ -505,7 +505,7 @@ KSycoca::KSycocaHeader KSycocaPrivate::readSycocaHeader()
     KSycocaUtilsPrivate::read(*str, header.prefixes);
     quint32 oldTimeStamp; // compat code --> TODO remove this when switching to a different fileName
     *str >> oldTimeStamp;
-    header.timeStamp = oldTimeStamp * 1000;
+    header.timeStamp = static_cast<quint64>(oldTimeStamp) * 1000;
     KSycocaUtilsPrivate::read(*str, header.language);
     *str >> header.updateSignature;
     KSycocaUtilsPrivate::read(*str, allResourceDirs);

