diff --git a/CMakeLists.txt b/CMakeLists.txt
index adc513cbe6..947a041ab2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1053,6 +1053,9 @@ list (APPEND ANDROID_EXTRA_LIBS ${LCMS2_LIBRARIES})
 ## Test for xsimd
 ##
 find_package(xsimd 8.1.0)
+if(NOT xsimd_FOUND)
+    find_package(xsimd 9)
+endif()
 set_package_properties(xsimd PROPERTIES
     DESCRIPTION "C++ wrappers for SIMD intrinsics"
     URL "https://github.com/xtensor-stack/xsimd"
