--- tzdata.orig/southamerica	2009-10-16 13:23:49.000000000 +0000
+++ tzdata/southamerica	2009-10-16 13:24:14.000000000 +0000
@@ -215,9 +215,14 @@
 # http://www.jujuy.gov.ar/index2/partes_prensa/18_10_08/235-181008.doc
 # </a>
 
+# Margarita Manterola:
+# Same as always, Argentina's government just decided to change its mind
+# regarding DST, 2 days in advance.
+# http://www.lanacion.com.ar/nota.asp?nota_id=1186963&pid=7542694&toi=6275
+
 Rule	Arg	2007	only	-	Dec	30	0:00	1:00	S
-Rule	Arg	2008	max	-	Mar	Sun>=15	0:00	0	-
-Rule	Arg	2008	max	-	Oct	Sun>=15	0:00	1:00	S
+Rule	Arg	2008	2009	-	Mar	Sun>=15	0:00	0	-
+Rule	Arg	2008	only	-	Oct	Sun>=15	0:00	1:00	S
  
 # From Mariano Absatz (2004-05-21):
 # Today it was officially published that the Province of Mendoza is changing
