diff -ur nxcomp.orig/Pgn.cpp nxcomp/Pgn.cpp
--- nxcomp.orig/Pgn.cpp	2010-03-01 19:18:59.000000000 +0200
+++ nxcomp/Pgn.cpp	2011-09-13 16:35:12.000000000 +0300
@@ -414,7 +414,7 @@
 
   png_read_info(pngPtr, infoPtr);
 
-  if (infoPtr -> color_type == PNG_COLOR_TYPE_PALETTE)
+  if (png_get_color_type(pngPtr, infoPtr) == PNG_COLOR_TYPE_PALETTE)
   {
     png_set_expand(pngPtr);
   }
@@ -565,7 +565,7 @@
 
   png_read_info( pngPtr, infoPtr ) ;
 
-  if (infoPtr -> color_type == PNG_COLOR_TYPE_PALETTE)
+  if (png_get_color_type(pngPtr, infoPtr) == PNG_COLOR_TYPE_PALETTE)
   {
     png_set_expand(pngPtr);
   }
@@ -709,7 +709,7 @@
   png_read_info(pngPtr, infoPtr) ;
 
 
-  if (infoPtr -> color_type == PNG_COLOR_TYPE_PALETTE)
+  if (png_get_color_type(pngPtr, infoPtr) == PNG_COLOR_TYPE_PALETTE)
   {
     png_set_expand(pngPtr);
   }
