From 7b62f2cb1e363ac4a0c907856c983b195997ccd2 Mon Sep 17 00:00:00 2001
Message-Id: <7b62f2cb1e363ac4a0c907856c983b195997ccd2.1505028215.git.jan.steffens@gmail.com>
From: "Jan Alexander Steffens (heftig)" <jan.steffens@gmail.com>
Date: Sun, 10 Sep 2017 09:23:13 +0200
Subject: [PATCH] meson: Actually use the soversion

---
 atspi/meson.build | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/atspi/meson.build b/atspi/meson.build
index 1ec3ecd63e6b5fac..5aeba49ba9c27079 100644
--- a/atspi/meson.build
+++ b/atspi/meson.build
@@ -68,6 +68,8 @@ atspi_enums = gnome.mkenums('atspi-enum-types',
 atspi_enum_h = atspi_enums[1]
 
 atspi = shared_library('atspi', atspi_sources + atspi_enums,
+                       version: soversion.split('.')[0],
+                       soversion: soversion,
                        include_directories: [ root_inc, registryd_inc ],
                        dependencies: [ libdbus_dep, gobject_dep, dbind_dep ],
                        install: true)
-- 
2.14.1

