diff -Nur -x '*.orig' -x '*~' gksu-2.0.0/nautilus-gksu/Makefile.am gksu-2.0.0.new/nautilus-gksu/Makefile.am
--- gksu-2.0.0/nautilus-gksu/Makefile.am	2006-09-15 10:45:10.000000000 +0800
+++ gksu-2.0.0.new/nautilus-gksu/Makefile.am	2008-04-12 15:45:47.000000000 +0800
@@ -2,7 +2,7 @@
 INCLUDES = ${NAUTILUS_CFLAGS}
 AM_CPPFLAGS = -DLOCALEDIR=\"$(datadir)/locale\" -DDATA_DIR=\"$(datadir)\" -DPREFIX=\"$(prefix)\"
 
-nautilus_extensiondir=$(libdir)/nautilus/extensions-1.0
+nautilus_extensiondir=$(libdir)/nautilus/extensions-2.0
 nautilus_extension_LTLIBRARIES = libnautilus-gksu.la
 libnautilus_gksu_la_SOURCES = libnautilus-gksu.c
 libnautilus_gksu_la_LDFLAGS = LDFLAGS = -module -avoid-version -no-undefined
diff -Nur -x '*.orig' -x '*~' gksu-2.0.0/nautilus-gksu/Makefile.in gksu-2.0.0.new/nautilus-gksu/Makefile.in
--- gksu-2.0.0/nautilus-gksu/Makefile.in	2006-09-24 04:23:10.000000000 +0800
+++ gksu-2.0.0.new/nautilus-gksu/Makefile.in	2008-04-12 15:45:53.000000000 +0800
@@ -241,7 +241,7 @@
 AM_CFLAGS = -g -O2 -Wall
 INCLUDES = ${NAUTILUS_CFLAGS}
 AM_CPPFLAGS = -DLOCALEDIR=\"$(datadir)/locale\" -DDATA_DIR=\"$(datadir)\" -DPREFIX=\"$(prefix)\"
-nautilus_extensiondir = $(libdir)/nautilus/extensions-1.0
+nautilus_extensiondir = $(libdir)/nautilus/extensions-2.0
 nautilus_extension_LTLIBRARIES = libnautilus-gksu.la
 libnautilus_gksu_la_SOURCES = libnautilus-gksu.c
 libnautilus_gksu_la_LDFLAGS = LDFLAGS = -module -avoid-version -no-undefined
