From 784c5168d1a62618e53619fd0a979363d596775a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Fridrich=20=C5=A0trba?= <fridrich.strba@bluewin.ch>
Date: Sun, 3 Jul 2016 21:43:25 +0200
Subject: [PATCH] Disable werror by default since gperf code will not build
 with gcc 6

---
 configure.ac | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index 06e9afb..6e00ed5 100644
--- a/configure.ac
+++ b/configure.ac
@@ -223,9 +223,9 @@ AS_IF([test "x$platform_win32" = "xyes"],
 # Check for cflags
 # ================
 AC_ARG_ENABLE([werror],
-    [AS_HELP_STRING([--disable-werror], [Treat all warnings as errors, usefull for development])],
+    [AS_HELP_STRING([--enable-werror], [Treat all warnings as errors, usefull for development])],
     [enable_werror="$enableval"],
-    [enable_werror=yes]
+    [enable_werror=no]
 )
 AS_IF([test x"$enable_werror" != "xno"], [
     CFLAGS="$CFLAGS -Werror"
-- 
2.10.2

