diff -Naur old/src/ccurl_thread.cc new/src/ccurl_thread.cc
--- old/src/ccurl_thread.cc	2015-06-21 14:49:06.000000000 -1000
+++ new/src/ccurl_thread.cc	2015-07-11 09:11:49.053040998 -1000
@@ -119,9 +119,7 @@
 			headers.h = curl_slist_append(headers.h, ("If-None-Match: " + etag).c_str());
 			etag.clear();
 		}
-		if (headers.h) {
-			curl_easy_setopt(curl, CURLOPT_HTTPHEADER, headers.h);
-		}
+		curl_easy_setopt(curl, CURLOPT_HTTPHEADER, headers.h);
 
 		res = curl_easy_perform(curl);
 		if (res == CURLE_OK) {
