--- configure	2013-07-03 08:16:15.000000000 +0200
+++ configure	2013-08-07 19:19:16.797273766 +0200
@@ -42409,7 +42409,7 @@
 /* end confdefs.h.  */
 
 #include "$THIS_INCLUDE"
-#if DB_VERSION_MAJOR == 4 || (4 == 4 && DB_VERSION_MAJOR == 5)
+#if DB_VERSION_MAJOR == 4 || (4 == 4 && DB_VERSION_MAJOR >= 5)
             yes
 #endif
 
@@ -42785,7 +42785,7 @@
 /* end confdefs.h.  */
 
 #include "$THIS_INCLUDE"
-#if DB_VERSION_MAJOR == 3 || (3 == 4 && DB_VERSION_MAJOR == 5)
+#if DB_VERSION_MAJOR == 3 || (3 == 4 && DB_VERSION_MAJOR >= 5)
             yes
 #endif
 
@@ -43161,7 +43161,7 @@
 /* end confdefs.h.  */
 
 #include "$THIS_INCLUDE"
-#if DB_VERSION_MAJOR == 2 || (2 == 4 && DB_VERSION_MAJOR == 5)
+#if DB_VERSION_MAJOR == 2 || (2 == 4 && DB_VERSION_MAJOR >= 5)
             yes
 #endif
 
--- ext/dba/config.m4	2013-07-03 08:10:53.000000000 +0200
+++ ext/dba/config.m4	2013-08-07 19:18:55.774448348 +0200
@@ -245,7 +245,7 @@
         ],[
           AC_EGREP_CPP(yes,[
 #include "$THIS_INCLUDE"
-#if DB_VERSION_MAJOR == $1 || ($1 == 4 && DB_VERSION_MAJOR == 5)
+#if DB_VERSION_MAJOR == $1 || ($1 == 4 && DB_VERSION_MAJOR >= 5)
             yes
 #endif
           ],[
--- ext/dba/dba_db4.c	2013-07-03 08:10:53.000000000 +0200
+++ ext/dba/dba_db4.c	2013-08-07 19:19:04.827562638 +0200
@@ -44,7 +44,7 @@
 {
 	TSRMLS_FETCH();
 
-#if (DB_VERSION_MAJOR == 5 || (DB_VERSION_MAJOR == 4 && DB_VERSION_MINOR == 8))
+#if (DB_VERSION_MAJOR >= 5 || (DB_VERSION_MAJOR == 4 && DB_VERSION_MINOR == 8))
 /* Bug 51086, Berkeley DB 4.8.26 */
 /* This code suppresses a BDB 4.8+ error message, thus keeping PHP test compatibility */
 	{
--- ext/dba/tests/dba_db4_handlers.phpt	2013-07-03 08:10:53.000000000 +0200
+++ ext/dba/tests/dba_db4_handlers.phpt	2013-08-07 20:39:34.640194224 +0200
@@ -47,4 +47,4 @@
 Test 1
 Success: db4 enabled
 Test 2 - full info
-.*Berkeley DB (4|5).*
+.*Berkeley DB (4|5|6).*
