From f15c376d251dd49b2202005ffed157f7d32c4393 Mon Sep 17 00:00:00 2001
From: Kohei Yoshida <kohei.yoshida@gmail.com>
Date: Tue, 7 Jul 2015 22:19:33 -0400
Subject: [PATCH] Use the master branch of mdds.

---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 512a5dd..c1901e3 100644
--- a/configure.ac
+++ b/configure.ac
@@ -82,7 +82,7 @@ else
 fi
 
 # Check for mdds.
-PKG_CHECK_MODULES([MDDS],[mdds >= 0.12.0])
+PKG_CHECK_MODULES([MDDS],[mdds-1.0 >= 0.13.0])
 
 # Check for python.
 AS_IF([test "x$enable_python" != "xno"], [
--
libgit2 0.23.3

