diff --git a/giscanner/sourcescanner.py b/giscanner/sourcescanner.py
index acfc048..ae3d29d 100644
--- a/giscanner/sourcescanner.py
+++ b/giscanner/sourcescanner.py
@@ -196,7 +196,7 @@ class SourceSymbol(object):
 class SourceScanner(object):
 
     def __init__(self):
-        with LibtoolImporter:
+        with LibtoolImporter(None, None):
             from giscanner._giscanner import SourceScanner
         self._scanner = SourceScanner()
         self._filenames = []
diff --git a/giscanner/xmlwriter.py b/giscanner/xmlwriter.py
index 0db286b..c10bcc6 100644
--- a/giscanner/xmlwriter.py
+++ b/giscanner/xmlwriter.py
@@ -67,7 +67,7 @@ def collect_attributes(tag_name, attributes, self_indent,
     return attr_value
 
 
-with LibtoolImporter:
+with LibtoolImporter(None, None):
     from giscanner._giscanner import collect_attributes
 

