--- configure.orig	2010-09-01 15:42:35.343947508 +0200
+++ configure	2010-09-01 15:43:43.816857993 +0200
@@ -2975,9 +2975,7 @@
                         if test -z "$available_patch" ; then
                                 available_patch=0
                         fi
-                        if test $available_major -ne $required_major \
-                                -o $available_minor -ne $required_minor \
-                                -o $available_patch -lt $required_patch ; then
+                        if test $available_major -lt $required_major ; then
                                 { $as_echo "$as_me:$LINENO: WARNING: SWIG version >= 1.3.21 is required.  You have $swig_version.  You should look at http://www.swig.org" >&5
 $as_echo "$as_me: WARNING: SWIG version >= 1.3.21 is required.  You have $swig_version.  You should look at http://www.swig.org" >&2;}
                                 SWIG=false
