diff -Nur avidemux_2.5.4.orig/plugins/ADM_videoEncoder/ADM_vidEnc_x264/x264Options.cpp avidemux_2.5.4/plugins/ADM_videoEncoder/ADM_vidEnc_x264/x264Options.cpp
--- avidemux_2.5.4.orig/plugins/ADM_videoEncoder/ADM_vidEnc_x264/x264Options.cpp	2010-10-10 06:45:42.000000000 -0700
+++ avidemux_2.5.4/plugins/ADM_videoEncoder/ADM_vidEnc_x264/x264Options.cpp	2011-01-16 07:16:48.283341102 -0800
@@ -1386,7 +1386,7 @@
 		case X264_WEIGHTP_NONE:
 			strcpy((char*)xmlBuffer, "none");
 			break;
-		case X264_WEIGHTP_BLIND:
+		case X264_WEIGHTP_SIMPLE:
 			strcpy((char*)xmlBuffer, "blind");
 			break;
 		case X264_WEIGHTP_SMART:
@@ -1896,7 +1896,7 @@
 				int weightedPredPFrames = X264_WEIGHTP_NONE;
 
 				if (strcmp(content, "blind") == 0)
-					weightedPredPFrames = X264_WEIGHTP_BLIND;
+					weightedPredPFrames = X264_WEIGHTP_SIMPLE;
 				else if (strcmp(content, "smart") == 0)
 					weightedPredPFrames = X264_WEIGHTP_SMART;
 
