--- xfce4-sensors-plugin-1.0.0/panel-plugin/Makefile.am	2011-01-14 15:40:24.065537723 +0000
+++ xfce4-sensors-plugin-1.0.0/panel-plugin/Makefile.am.new	2011-01-14 15:41:48.248960397 +0000
@@ -20,6 +20,7 @@
 
 xfce4_sensors_plugin_CFLAGS = \
 	@LIBXFCE4PANEL_CFLAGS@										   \
+	@LIBXFCEGUI4_CFLAGS@										\
 	@PLATFORM_CFLAGS@												\
 	-DPACKAGE_LOCALE_DIR=\"$(localedir)\"							\
 	@LIBSENSORS_CFLAGS@
@@ -30,6 +31,7 @@
 xfce4_sensors_plugin_LDFLAGS = 								\
 	$(top_builddir)/lib/libxfce4sensors.la								\
 	@LIBXFCE4PANEL_LIBS@											 \
+	@LIBXFCEGUI4_LIBS@											\
 	@SOLLIBS@														\
 	@LIBSENSORS_LDFLAGS@											 \
 	@LIBM@
--- xfce4-sensors-plugin-1.0.0/src/Makefile.am	2009-11-06 11:02:11.000000000 +0000
+++ xfce4-sensors-plugin-1.0.0/src/Makefile.am.new	2011-01-14 16:19:09.777636193 +0000
@@ -27,6 +27,7 @@
 	-DPACKAGE_LOCALE_DIR=\"$(localedir)\" \
 	@LIBSENSORS_CFLAGS@	\
 	@LIBXFCE4UTIL_CFLAGS@ \
+	@LIBXFCE4PANEL_CFLAGS@ \
 	@LIBXFCEGUI4_CFLAGS@
 
 # @GTK_CFLAGS@ 	\
@@ -41,6 +42,7 @@
 	$(top_builddir)/lib/libxfce4sensors.la \
 	@LIBSENSORS_LDFLAGS@ \
 	@LIBXFCE4UTIL_LIBS@	\
+	@LIBXFCE4PANEL_LIBS@ \
 	@LIBXFCEGUI4_LIBS@ 
 
 # @GTK_LIBS@ \
