Index: src/exp-gfx.c
===================================================================
RCS file: /cvsroot/zapping/vbi/src/exp-gfx.c,v
retrieving revision 1.16
diff -u -B -r1.16 exp-gfx.c
--- src/exp-gfx.c	24 Feb 2008 14:17:47 -0000	1.16
+++ src/exp-gfx.c	15 Feb 2011 20:03:58 -0000
@@ -1672,11 +1672,11 @@
 	char title[80];
 	unsigned int i;
 
-	if (setjmp (png_ptr->jmpbuf))
+	if (setjmp (png_jmpbuf(png_ptr)))
 		return FALSE;
 
 	png_set_write_fn (png_ptr,
-			  (voidp) gfx,
+			  (png_voidp) gfx,
 			  write_data,
 			  flush_data);
 
