diff -uNr kshutdown-3.0beta2.orig/src/kshutdown.cpp kshutdown-3.0beta2/src/kshutdown.cpp
--- kshutdown-3.0beta2.orig/src/kshutdown.cpp	2011-12-29 22:18:45.000000000 -0500
+++ kshutdown-3.0beta2/src/kshutdown.cpp	2011-12-30 19:09:08.128451522 -0500
@@ -886,13 +886,13 @@
 	#ifdef KS_NATIVE_KDE
 	if (
 // TODO: check if logout is available
-		(m_kdeShutDownAvailable || (m_type == U_SHUTDOWN_TYPE_LOGOUT)) &&
+		(m_kdeShutDownAvailable || (m_type == U_SHUTDOWN_TYPE_LOGOUT))) 
 		KWorkSpace::requestShutDown(
 			KWorkSpace::ShutdownConfirmNo,
 			m_type,
 			KWorkSpace::ShutdownModeForceNow
-		)
-	)
+		);
+
 		return true;
 	#endif // KS_NATIVE_KDE
 	
