diff --git i/include/freetype/config/ftoption.h w/include/freetype/config/ftoption.h
index d162780..0883c69 100644
--- i/include/freetype/config/ftoption.h
+++ w/include/freetype/config/ftoption.h
@@ -765,7 +765,7 @@ FT_BEGIN_HEADER
   /* switch between the two engines using the `hinting-engine' property of */
   /* the cff driver module.                                                */
   /*                                                                       */
-#define CFF_CONFIG_OPTION_OLD_ENGINE
+/* #define CFF_CONFIG_OPTION_OLD_ENGINE */
 
 
   /*************************************************************************/
