--- tools/hal-storage-mount.c.orig	2008-04-21 01:48:27.000000000 +0200
+++ tools/hal-storage-mount.c	2008-04-21 01:47:44.000000000 +0200
@@ -769,13 +769,13 @@
 
 	if (pol_is_fixed) {
 		if (pol_change_uid) {
-			action = NULL; /* "hal-storage-mount-fixed-extra-options"; TODO: rethink */
+			action = "hal-storage-mount-fixed-extra-options"; /*TODO: rethink */
 		} else {
 			action = "org.freedesktop.hal.storage.mount-fixed";
 		}
 	} else {
 		if (pol_change_uid) {
-			action = NULL; /* "hal-storage-mount-removable-extra-options"; TODO: rethink "extra-options" */
+			action = "hal-storage-mount-removable-extra-options"; /*TODO: rethink "extra-options" */
 		} else {
 			action = "org.freedesktop.hal.storage.mount-removable";
 		}
