diff -baur hydrogen-0.9.6.orig/CMakeLists.txt hydrogen-0.9.6/CMakeLists.txt
--- hydrogen-0.9.6.orig/CMakeLists.txt	2014-08-11 15:10:25.960752930 +0600
+++ hydrogen-0.9.6/CMakeLists.txt	2014-08-11 17:01:48.001219876 +0600
@@ -193,6 +193,9 @@
 FIND_HELPER(PULSEAUDIO pulseaudio pulse/pulseaudio.h pulse)
 FIND_HELPER(LASH lash-1.0 lash/lash.h lash)
 FIND_HELPER(LRDF lrdf lrdf.h lrdf)
+IF(LRDF_FOUND)
+    include_directories(${LRDF_INCLUDE_DIRS})
+ENDIF(LRDF_FOUND)
 FIND_HELPER(RUBBERBAND rubberband RubberBandStretcher.h rubberband)
 FIND_HELPER(CPPUNIT cppunit cppunit/TestCase.h cppunit)
 # CHECK FOR Qt4
