Index: rdesktop.c
===================================================================
--- rdesktop.c	(revision 1730)
+++ rdesktop.c	(revision 1731)
@@ -87,7 +87,7 @@
 RD_BOOL g_sendmotion = True;
 RD_BOOL g_bitmap_cache = True;
 RD_BOOL g_bitmap_cache_persist_enable = False;
-RD_BOOL g_bitmap_cache_precache = True;
+RD_BOOL g_bitmap_cache_precache = False;
 RD_BOOL g_use_ctrl = True;
 RD_BOOL g_encryption = True;
 RD_BOOL g_encryption_initial = True;
