From 4c0e17d7fc78525007833a10524e88d523123a40 Mon Sep 17 00:00:00 2001
From: Andre Miranda <andre.miranda@liferay.com>
Date: Wed, 5 Dec 2018 13:27:51 -0300
Subject: [PATCH] Expand scroll window of file operation progress dialog (Bug
 #14946)

---
 thunar/thunar-progress-dialog.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/thunar/thunar-progress-dialog.c b/thunar/thunar-progress-dialog.c
index 7d227c214..f4de6097f 100644
--- a/thunar/thunar-progress-dialog.c
+++ b/thunar/thunar-progress-dialog.c
@@ -368,6 +368,7 @@ thunar_progress_dialog_add_job (ThunarProgressDialog *dialog,
       dialog->scrollwin = gtk_scrolled_window_new (NULL, NULL);
       gtk_scrolled_window_set_policy (GTK_SCROLLED_WINDOW (dialog->scrollwin),
                                       GTK_POLICY_NEVER, GTK_POLICY_AUTOMATIC);
+      gtk_widget_set_vexpand (dialog->scrollwin, TRUE);
       gtk_container_add (GTK_CONTAINER (dialog->vbox), dialog->scrollwin);
       gtk_widget_show (dialog->scrollwin);
 
