Index: avidemux_2.5.0/plugins/ADM_audioEncoders/lavcodec/audioencoder_lavcodec.cpp
===================================================================
--- avidemux_2.5.0/plugins/ADM_audioEncoders/lavcodec/audioencoder_lavcodec.cpp
+++ avidemux_2.5.0/plugins/ADM_audioEncoders/lavcodec/audioencoder_lavcodec.cpp	2009-07-10 18:53:47.000000000 +0300
@@ -19,7 +19,6 @@
 #include "ADM_default.h"
 #include "DIA_factory.h"
 #include "DIA_coreToolkit.h"
-#include <lame/lame.h>
 #include "audioencoder.h"
 #include "audioencoderInternal.h"
 //
