diff -ur avidemux_2.4.orig/avidemux/ADM_userInterfaces/ADM_QT4/ui_support.cpp avidemux_2.4/avidemux/ADM_userInterfaces/ADM_QT4/ui_support.cpp
--- avidemux_2.4.orig/avidemux/ADM_userInterfaces/ADM_QT4/ui_support.cpp	2007-10-16 21:23:21.000000000 +0300
+++ avidemux_2.4/avidemux/ADM_userInterfaces/ADM_QT4/ui_support.cpp	2007-12-28 20:35:30.000000000 +0200
@@ -53,7 +53,7 @@
 {
 	printf("\n[Locale] Locale: %s\n", QLocale::system().name().toUtf8().constData());
 
-	QString appdir = QCoreApplication::applicationDirPath() + "/i18n/";
+	QString appdir = "/usr/share/avidemux/i18n/";
 
 	loadTranslation(&qtTranslator, appdir + "qt_" + QLocale::system().name());
 	loadTranslation(&avidemuxTranslator, appdir + "avidemux_" + QLocale::system().name());
diff -ur avidemux_2.4.orig/cmake/Ts.cmake avidemux_2.4/cmake/Ts.cmake
--- avidemux_2.4.orig/cmake/Ts.cmake	2007-10-14 14:58:28.000000000 +0300
+++ avidemux_2.4/cmake/Ts.cmake	2007-12-28 20:36:05.000000000 +0200
@@ -63,7 +63,7 @@
                 
             SET(qm_files ${qm_files} ${_outXml} ${_out})
 
-			INSTALL(FILES ${_out} DESTINATION "${CMAKE_INSTALL_PREFIX}/bin/i18n")
+			INSTALL(FILES ${_out} DESTINATION "${CMAKE_INSTALL_PREFIX}/share/avidemux/i18n")
         ENDFOREACH(ts_input ${ts_files})
 
         SET(${_sources} ${${_sources}} ${qm_files})
@@ -94,7 +94,7 @@
                 
             SET(qm_files ${qm_files} ${_out})
 
-			INSTALL(FILES ${_out} DESTINATION "${CMAKE_INSTALL_PREFIX}/bin/i18n")
+			INSTALL(FILES ${_out} DESTINATION "${CMAKE_INSTALL_PREFIX}/share/avidemux/i18n")
         ENDFOREACH(ts_input ${ts_files})
 
         SET(${_sources} ${${_sources}} ${qm_files})
