Index: avidemux_2.5.0/plugins/ADM_audioDecoders/ADM_ad_ac3/ADM_liba52/CMakeLists.txt
===================================================================
--- avidemux_2.5.0/plugins/ADM_audioDecoders/ADM_ad_ac3/ADM_liba52/CMakeLists.txt
+++ avidemux_2.5.0/plugins/ADM_audioDecoders/ADM_ad_ac3/ADM_liba52/CMakeLists.txt	2009-07-10 20:24:52.000000000 +0300
@@ -5,9 +5,7 @@
 
 ADD_LIBRARY(${ADM_LIB} STATIC ${${ADM_LIB}_SRCS})
 
-IF (WIN32 OR APPLE)
-	TARGET_LINK_LIBRARIES(${ADM_LIB} ADM_core)
-ENDIF (WIN32 OR APPLE)
+TARGET_LINK_LIBRARIES(${ADM_LIB} ADM_core)
 
 IF (UNIX)
 	ADD_TARGET_CFLAGS(${ADM_LIB} -fPIC)
Index: avidemux_2.5.0/plugins/ADM_audioDevices/Alsa/CMakeLists.txt
===================================================================
--- avidemux_2.5.0/plugins/ADM_audioDevices/Alsa/CMakeLists.txt
+++ avidemux_2.5.0/plugins/ADM_audioDevices/Alsa/CMakeLists.txt	2009-07-10 20:46:43.000000000 +0300
@@ -4,7 +4,7 @@
 SET(ADM_av_alsa_SRCS ADM_deviceALSA.cpp)
 
 ADD_LIBRARY(ADM_av_alsa SHARED ${ADM_av_alsa_SRCS})
-TARGET_LINK_LIBRARIES(ADM_av_alsa ${ASOUND_LIBRARY})
+TARGET_LINK_LIBRARIES(ADM_av_alsa ADM_core ADM_coreAudio ${ASOUND_LIBRARY})
 
 INIT_AUDIO_DEVICE(ADM_av_alsa)
 INSTALL_AUDIODEVICE(ADM_av_alsa)
Index: avidemux_2.5.0/plugins/ADM_audioDevices/Arts/CMakeLists.txt
===================================================================
--- avidemux_2.5.0/plugins/ADM_audioDevices/Arts/CMakeLists.txt
+++ avidemux_2.5.0/plugins/ADM_audioDevices/Arts/CMakeLists.txt	2009-07-10 21:23:13.000000000 +0300
@@ -4,7 +4,7 @@
 SET(ADM_av_arts_SRCS ADM_deviceArts.cpp)
 
 ADD_LIBRARY(ADM_av_arts SHARED ${ADM_av_arts_SRCS})
-TARGET_LINK_LIBRARIES(ADM_av_arts ${ARTS_LDFLAGS})
+TARGET_LINK_LIBRARIES(ADM_av_arts ADM_coreAudio ${ARTS_LDFLAGS})
 ADD_DEFINITIONS(${ARTS_CFLAGS})
 INIT_AUDIO_DEVICE(ADM_av_arts)
 INSTALL_AUDIODEVICE(ADM_av_arts)
Index: avidemux_2.5.0/plugins/ADM_audioDevices/Oss/CMakeLists.txt
===================================================================
--- avidemux_2.5.0/plugins/ADM_audioDevices/Oss/CMakeLists.txt
+++ avidemux_2.5.0/plugins/ADM_audioDevices/Oss/CMakeLists.txt	2009-07-10 21:14:29.000000000 +0300
@@ -4,6 +4,7 @@
 SET(ADM_av_oss_SRCS ADM_deviceoss.cpp)
 
 ADD_LIBRARY(ADM_av_oss SHARED ${ADM_av_oss_SRCS})
+TARGET_LINK_LIBRARIES(ADM_av_oss ADM_coreAudio)
 include_directories(${OSS_INCLUDE_DIR})
 INIT_AUDIO_DEVICE(ADM_av_oss)
 INSTALL_AUDIODEVICE(ADM_av_oss)
Index: avidemux_2.5.0/avidemux/ADM_libraries/ADM_smjs/CMakeLists.txt
===================================================================
--- avidemux_2.5.0/avidemux/ADM_libraries/ADM_smjs/CMakeLists.txt
+++ avidemux_2.5.0/avidemux/ADM_libraries/ADM_smjs/CMakeLists.txt	2009-07-10 19:15:45.000000000 +0300
@@ -14,5 +14,6 @@
 ENDIF (MINGW)
 
 ADD_TARGET_CFLAGS(${ADM_LIB} "-DNDEBUG -DTRIMMED -DEXPORT_JS_API -DJS_USE_SAFE_ARENA -D_IEEE_LIBM")
+TARGET_LINK_LIBRARIES(${ADM_LIB} m)
 
 INSTALL(TARGETS ${ADM_LIB} RUNTIME DESTINATION ${BIN_DIR}  LIBRARY DESTINATION lib  ARCHIVE DESTINATION lib)
\ Ei rivinvaihtoa tiedoston lopussa
Index: avidemux_2.5.0/avidemux/ADM_userInterfaces/ADM_render/CMakeLists.txt
===================================================================
--- avidemux_2.5.0/avidemux/ADM_userInterfaces/ADM_render/CMakeLists.txt
+++ avidemux_2.5.0/avidemux/ADM_userInterfaces/ADM_render/CMakeLists.txt	2009-07-10 19:50:46.000000000 +0300
@@ -27,6 +27,10 @@
 		TARGET_LINK_LIBRARIES(${ADM_LIB}_gtk ${SDL_LIBRARY})
 	ENDIF (USE_SDL)
 
+	IF (USE_XV)
+		TARGET_LINK_LIBRARIES(${ADM_LIB}_gtk ${XVIDEO_LIBRARY_DIR} X11 Xext)
+	ENDIF (USE_XV)
+
 	INSTALL(TARGETS ${ADM_LIB}_gtk RUNTIME DESTINATION ${BIN_DIR}  LIBRARY DESTINATION lib  ARCHIVE DESTINATION lib)
 ENDIF (ADM_UI_GTK)
 
@@ -35,6 +39,10 @@
 		TARGET_LINK_LIBRARIES(${ADM_LIB}_qt4 ${SDL_LIBRARY})
 	ENDIF (USE_SDL)
 
+	IF (USE_XV)
+		TARGET_LINK_LIBRARIES(${ADM_LIB}_qt4 ${XVIDEO_LIBRARY_DIR} X11 Xext)
+	ENDIF (USE_XV)
+
 	INSTALL(TARGETS ${ADM_LIB}_qt4 RUNTIME DESTINATION ${BIN_DIR}  LIBRARY DESTINATION lib  ARCHIVE DESTINATION lib)
 ENDIF (ADM_UI_QT4)
 
Index: avidemux_2.5.0/avidemux/ADM_core/src/CMakeLists.txt
===================================================================
--- avidemux_2.5.0/avidemux/ADM_core/src/CMakeLists.txt
+++ avidemux_2.5.0/avidemux/ADM_core/src/CMakeLists.txt	2009-07-10 19:06:53.000000000 +0300
@@ -5,6 +5,7 @@
 
 ADD_LIBRARY(ADM_core SHARED ${ADM_core_SRCS})
 ADD_TARGET_DEFINITIONS(ADM_core ADM_DEBUG FIND_LEAKS)
+TARGET_LINK_LIBRARIES(ADM_core dl)
 
 IF (MINGW)
 	TARGET_LINK_LIBRARIES(ADM_core "${PTHREAD_LIBRARIES}" imagehlp ws2_32)
