diff -urN NetworkManager-orig/gnome/applet/nm-applet.conf NetworkManager-policy/gnome/applet/nm-applet.conf
--- NetworkManager-orig/gnome/applet/nm-applet.conf	2005-04-27 20:05:02.000000000 +0200
+++ NetworkManager-policy/gnome/applet/nm-applet.conf	2006-02-11 19:04:03.000000000 +0100
@@ -8,7 +8,7 @@
 		<allow send_destination="org.freedesktop.NetworkManagerInfo"/>
                 <allow send_interface="org.freedesktop.NetworkManagerInfo"/>
 	</policy>
-	<policy at_console="true">
+	<policy group="network">
 		<allow own="org.freedesktop.NetworkManagerInfo"/>
 
 		<allow send_destination="org.freedesktop.NetworkManagerInfo"/>
