diff -Naur docbookx412-orig/Makefile docbookx412/Makefile
--- docbookx412-orig/Makefile	2002-02-16 11:08:29.000000000 -0800
+++ docbookx412/Makefile	2002-08-24 12:03:31.000000000 -0700
@@ -1,7 +1,7 @@
 SHELL=/bin/sh
 VERSION=4.1.2-9
-DESTDIR=/usr/share/xml/docbook/xml-dtd-$(VERSION)
-CATALOG=/usr/share/xml/docbook/catalog
+DESTDIR=$(ROOT)/usr/share/xml/docbook/xml-dtd-$(VERSION)
+CATALOG=$(ROOT)/usr/share/xml/docbook/catalog
 OPENJADEVER=1.3
 
 all:
@@ -24,7 +24,7 @@
 	#
         #
 	# DocBook XML V4.1.2
-	@./buildDocBookCatalog
+	@./buildDocBookCatalog $(ROOT)
 
 uninstall:
 	#
diff -Naur docbookx412-orig/buildDocBookCatalog docbookx412/buildDocBookCatalog
--- docbookx412-orig/buildDocBookCatalog	2002-02-17 09:32:45.000000000 -0800
+++ docbookx412/buildDocBookCatalog	2002-08-24 12:15:23.000000000 -0700
@@ -28,16 +28,20 @@
 # Modifications made to Daniel Veillard's catalog build script at 
 # http://xmlsoft.org/buildDocBookCatalog.
 
-if [ ! -d /etc/xml -a -w /etc ]
+DESTDIR=$1
+echo "DESTDIR is $DESTDIR"
+mkdir $DESTDIR/etc
+
+if [ ! -d $DESTDIR/etc/xml -a -w $DESTDIR/etc ]
 then
     echo Directory /etc/xml missing creating it
-    mkdir /etc/xml
-    chmod 755 /etc/xml
+    mkdir $DESTDIR/etc/xml
+    chmod 755 $DESTDIR/etc/xml
 fi
-if [ -w /etc/xml ]
+if [ -w $DESTDIR/etc/xml ]
 then
-    ROOTCATALOG=/etc/xml/catalog
-    CATALOG=/etc/xml/docbook
+    ROOTCATALOG=$DESTDIR/etc/xml/catalog
+    CATALOG=$DESTDIR/etc/xml/docbook
 else
     echo Unable to update root catalog /etc/xml/catalog
     echo Using $HOME/xmlcatalog as the root catalog
@@ -69,16 +73,7 @@
 #
 # find the prefix for DocBook DTD
 #
-found=`find /usr/share/xml -name docbookx.dtd`
-if [ "$found" = "" ] ; then
-    found=`find $HOME -name docbookx.dtd`
-fi
-if [ "$found" = "" ] ; then
-    found=`find /usr/local -name docbookx.dtd`
-fi
-if [ "$found" = "" ] ; then
-    found=`find /usr/share/sgml -name docbookx.dtd`
-fi
+found=`find $DESTDIR/usr/share/xml -name docbookx.dtd`
 if [ "$found" = "" ] ; then
     echo could not locate docbookx.dtd for DocBook XML
     exit 1
@@ -105,49 +100,52 @@
 #    exit 2
 #fi
 
+fakedocbookdir=`echo $docbookdir | sed "s|^$DESTDIR||"`
+FAKECATALOG=`echo $CATALOG | sed "s|^$DESTDIR||"`
+
 xmlcatalog --noout --add "public" \
     "-//OASIS//ELEMENTS DocBook XML Information Pool V4.1.2//EN" \
-    "file://$docbookdir/dbpoolx.mod" $CATALOG
+    "file://$fakedocbookdir/dbpoolx.mod" $CATALOG
 xmlcatalog --noout --add "public" \
     "-//OASIS//DTD DocBook XML V4.1.2//EN" \
-    "file://$docbookdir/docbookx.dtd" $CATALOG
+    "file://$fakedocbookdir/docbookx.dtd" $CATALOG
 xmlcatalog --noout --add "public" \
     "-//OASIS//ENTITIES DocBook XML Character Entities V4.1.2//EN" \
-    "file://$docbookdir/dbcentx.mod" $CATALOG
+    "file://$fakedocbookdir/dbcentx.mod" $CATALOG
 xmlcatalog --noout --add "public" \
     "-//OASIS//ENTITIES DocBook XML Notations V4.1.2//EN" \
-    "file://$docbookdir/dbnotnx.mod" $CATALOG
+    "file://$fakedocbookdir/dbnotnx.mod" $CATALOG
 xmlcatalog --noout --add "public" \
     "-//OASIS//ENTITIES DocBook XML Additional General Entities V4.1.2//EN" \
-    "file://$docbookdir/dbgenent.mod" $CATALOG
+    "file://$fakedocbookdir/dbgenent.mod" $CATALOG
 xmlcatalog --noout --add "public" \
     "-//OASIS//ELEMENTS DocBook XML Document Hierarchy V4.1.2//EN" \
-    "file://$docbookdir/dbhierx.mod" $CATALOG
+    "file://$fakedocbookdir/dbhierx.mod" $CATALOG
 xmlcatalog --noout --add "public" \
     "-//OASIS//DTD XML Exchange Table Model 19990315//EN" \
-    "file://$docbookdir/soextblx.dtd" $CATALOG
+    "file://$fakedocbookdir/soextblx.dtd" $CATALOG
 xmlcatalog --noout --add "public" \
     "-//OASIS//DTD DocBook XML CALS Table Model V4.1.2//EN" \
-    "file://$docbookdir/calstblx.dtd" $CATALOG
+    "file://$fakedocbookdir/calstblx.dtd" $CATALOG
 xmlcatalog --noout --add "rewriteSystem" \
     "http://www.oasis-open.org/docbook/xml/4.1.2" \
-    "file://$docbookdir" $CATALOG
+    "file://$fakedocbookdir" $CATALOG
 xmlcatalog --noout --add "rewriteURI" \
     "http://www.oasis-open.org/docbook/xml/4.1.2" \
-    "file://$docbookdir" $CATALOG
+    "file://$fakedocbookdir" $CATALOG
 
 xmlcatalog --noout --add "delegatePublic" \
     "-//OASIS//ENTITIES DocBook XML" \
-    "file://$CATALOG" $ROOTCATALOG
+    "file://$FAKECATALOG" $ROOTCATALOG
 xmlcatalog --noout --add "delegatePublic" \
     "-//OASIS//DTD DocBook XML" \
-    "file://$CATALOG" $ROOTCATALOG
+    "file://$FAKECATALOG" $ROOTCATALOG
 xmlcatalog --noout --add "delegateSystem" \
     "http://www.oasis-open.org/docbook/" \
-    "file://$CATALOG" $ROOTCATALOG
+    "file://$FAKECATALOG" $ROOTCATALOG
 xmlcatalog --noout --add "delegateURI" \
     "http://www.oasis-open.org/docbook/" \
-    "file://$CATALOG" $ROOTCATALOG
+    "file://$FAKECATALOG" $ROOTCATALOG
 
 #
 # find the prefix for ISO DocBook entities
@@ -187,66 +185,68 @@
 isodir=`dirname $entxml`
 echo Found ISO DocBook entities in $isodir
 
+fakeisodir=`echo $isodir | sed "s|^$DESTDIR||"`
+
 xmlcatalog --noout --add "public" \
     "ISO 8879:1986//ENTITIES Publishing//EN" \
-    "file://$isodir/iso-pub.ent" $CATALOG
+    "file://$fakeisodir/iso-pub.ent" $CATALOG
 xmlcatalog --noout --add "public" \
     "ISO 8879:1986//ENTITIES Greek Letters//EN" \
-    "file://$isodir/iso-grk1.ent" $CATALOG
+    "file://$fakeisodir/iso-grk1.ent" $CATALOG
 xmlcatalog --noout --add "public" \
     "ISO 8879:1986//ENTITIES Box and Line Drawing//EN" \
-    "file://$isodir/iso-box.ent" $CATALOG
+    "file://$fakeisodir/iso-box.ent" $CATALOG
 xmlcatalog --noout --add "public" \
     "ISO 8879:1986//ENTITIES Greek Symbols//EN" \
-    "file://$isodir/iso-grk3.ent" $CATALOG
+    "file://$fakeisodir/iso-grk3.ent" $CATALOG
 xmlcatalog --noout --add "public" \
     "ISO 8879:1986//ENTITIES Added Math Symbols: Negated Relations//EN" \
-    "file://$isodir/iso-amsn.ent" $CATALOG
+    "file://$fakeisodir/iso-amsn.ent" $CATALOG
 xmlcatalog --noout --add "public" \
     "ISO 8879:1986//ENTITIES Numeric and Special Graphic//EN" \
-    "file://$isodir/iso-num.ent" $CATALOG
+    "file://$fakeisodir/iso-num.ent" $CATALOG
 xmlcatalog --noout --add "public" \
     "ISO 8879:1986//ENTITIES Alternative Greek Symbols//EN" \
-    "file://$isodir/iso-grk4.ent" $CATALOG
+    "file://$fakeisodir/iso-grk4.ent" $CATALOG
 xmlcatalog --noout --add "public" \
     "ISO 8879:1986//ENTITIES Diacritical Marks//EN" \
-    "file://$isodir/iso-dia.ent" $CATALOG
+    "file://$fakeisodir/iso-dia.ent" $CATALOG
 xmlcatalog --noout --add "public" \
     "ISO 8879:1986//ENTITIES Monotoniko Greek//EN" \
-    "file://$isodir/iso-grk2.ent" $CATALOG
+    "file://$fakeisodir/iso-grk2.ent" $CATALOG
 xmlcatalog --noout --add "public" \
     "ISO 8879:1986//ENTITIES Added Math Symbols: Arrow Relations//EN" \
-    "file://$isodir/iso-amsa.ent" $CATALOG
+    "file://$fakeisodir/iso-amsa.ent" $CATALOG
 xmlcatalog --noout --add "public" \
     "ISO 8879:1986//ENTITIES Added Math Symbols: Ordinary//EN" \
-    "file://$isodir/iso-amso.ent" $CATALOG
+    "file://$fakeisodir/iso-amso.ent" $CATALOG
 xmlcatalog --noout --add "public" \
     "ISO 8879:1986//ENTITIES Russian Cyrillic//EN" \
-    "file://$isodir/iso-cyr1.ent" $CATALOG
+    "file://$fakeisodir/iso-cyr1.ent" $CATALOG
 xmlcatalog --noout --add "public" \
     "ISO 8879:1986//ENTITIES General Technical//EN" \
-    "file://$isodir/iso-tech.ent" $CATALOG
+    "file://$fakeisodir/iso-tech.ent" $CATALOG
 xmlcatalog --noout --add "public" \
     "ISO 8879:1986//ENTITIES Added Math Symbols: Delimiters//EN" \
-    "file://$isodir/iso-amsc.ent" $CATALOG
+    "file://$fakeisodir/iso-amsc.ent" $CATALOG
 xmlcatalog --noout --add "public" \
     "ISO 8879:1986//ENTITIES Added Latin 1//EN" \
-    "file://$isodir/iso-lat1.ent" $CATALOG
+    "file://$fakeisodir/iso-lat1.ent" $CATALOG
 xmlcatalog --noout --add "public" \
     "ISO 8879:1986//ENTITIES Added Math Symbols: Binary Operators//EN" \
-    "file://$isodir/iso-amsb.ent" $CATALOG
+    "file://$fakeisodir/iso-amsb.ent" $CATALOG
 xmlcatalog --noout --add "public" \
     "ISO 8879:1986//ENTITIES Added Latin 2//EN" \
-    "file://$isodir/iso-lat2.ent" $CATALOG
+    "file://$fakeisodir/iso-lat2.ent" $CATALOG
 xmlcatalog --noout --add "public" \
     "ISO 8879:1986//ENTITIES Added Math Symbols: Relations//EN" \
-    "file://$isodir/iso-amsr.ent" $CATALOG
+    "file://$fakeisodir/iso-amsr.ent" $CATALOG
 xmlcatalog --noout --add "public" \
     "ISO 8879:1986//ENTITIES Non-Russian Cyrillic//EN" \
-    "file://$isodir/iso-cyr2.ent" $CATALOG
+    "file://$fakeisodir/iso-cyr2.ent" $CATALOG
 
 xmlcatalog --noout --add "delegatePublic" \
     "ISO 8879:1986" \
-    "file://$CATALOG" $ROOTCATALOG
+    "file://$FAKECATALOG" $ROOTCATALOG
 #
 #
