--- lib/modules/fglrx/build_mod/firegl_public.c	2008-08-18 11:45:42.000000000 +0000
+++ lib/modules/fglrx/build_mod/firegl_public.c.new	2008-08-22 16:25:10.643033742 +0000
@@ -2668,7 +2668,7 @@
 {
 /*Some kernel developer removed the export of symbol "flush_tlb_page" on 2.6.25 x86_64 SMP kernel.
   Define a simple version here.*/
-#if defined(__x86_64__) && defined(__SMP__) && (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,25)) 
+#if defined(__x86_64__) && (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,25)) 
     on_each_cpu(KCL_flush_tlb_one, &va, 1, 1);
 #else
     flush_tlb_page(vma, va);
