diff -up ghostscript-8.71/lib/pdf2dsc.ps.pdf2dsc ghostscript-8.71/lib/pdf2dsc.ps
--- ghostscript-8.71/lib/pdf2dsc.ps.pdf2dsc	2010-02-20 15:50:51.287734970 +0000
+++ ghostscript-8.71/lib/pdf2dsc.ps	2010-02-20 15:51:02.902609964 +0000
@@ -116,7 +116,7 @@ systemdict /.setsafe known { .setsafe } 
    DSCfile PDFname write==only
    ( \(r\) file { DELAYSAFER { .setsafe } if } stopped pop\n) puts
    ( pdfopen begin\n) puts
-   ( copy_trailer_attrs\n) puts
+   ( process_trailer_attrs\n) puts
    (%%EndSetup\n) puts
 
    /.hasPageLabels false def % see "Page Labels" in the PDF Reference
