diff --git i/CMakeLists.txt w/CMakeLists.txt
index 4140574..4bfe743 100644
--- i/CMakeLists.txt
+++ w/CMakeLists.txt
@@ -187,6 +187,7 @@ install(FILES dwarves.h dwarves_emit.h dwarves_reorganize.h
 install(FILES man-pages/pahole.1 DESTINATION ${CMAKE_INSTALL_PREFIX}/share/man/man1/)
 install(PROGRAMS ostra/ostra-cg DESTINATION ${CMAKE_INSTALL_PREFIX}/bin)
 install(PROGRAMS btfdiff fullcircle DESTINATION ${CMAKE_INSTALL_PREFIX}/bin)
-install(FILES ostra/python/ostra.py DESTINATION ${CMAKE_INSTALL_PREFIX}/share/dwarves/runtime/python)
+find_package(Python3)
+install(FILES ostra/python/ostra.py DESTINATION ${Python3_SITELIB})
 install(FILES lib/Makefile lib/ctracer_relay.c lib/ctracer_relay.h lib/linux.blacklist.cu
 	DESTINATION ${CMAKE_INSTALL_PREFIX}/share/dwarves/runtime)
