diff -up m4-1.4.17/build-aux/update-copyright.orig m4-1.4.17/build-aux/update-copyright
--- m4-1.4.17/build-aux/update-copyright.orig	2015-07-13 11:25:19.502749690 +0200
+++ m4-1.4.17/build-aux/update-copyright	2015-07-13 11:26:20.445044297 +0200
@@ -124,7 +124,7 @@ use strict;
 use warnings;
 
 my $copyright_re = 'Copyright';
-my $circle_c_re = '(?:\([cC]\)|@copyright{}|&copy;)';
+my $circle_c_re = '(?:\([cC]\)|@copyright\{}|\\\\\(co|&copy;)';
 my $holder = $ENV{UPDATE_COPYRIGHT_HOLDER};
 $holder ||= 'Free Software Foundation, Inc.';
 my $prefix_max = 5;
