diff -Naur kismet-2007-10-R1-old/kismet_wrapper.cc kismet-2007-10-R1/kismet_wrapper.cc
--- kismet-2007-10-R1-old/kismet_wrapper.cc	2007-10-07 07:50:03.000000000 +1000
+++ kismet-2007-10-R1/kismet_wrapper.cc	2008-08-17 15:51:19.000000000 +1000
@@ -25,6 +25,7 @@
 #include <sys/wait.h>
 #include <fcntl.h>
 #include <unistd.h>
+#include <cstring>
 
 #include "config.h"
 
diff -Naur kismet-2007-10-R1-old/ringbuf.cc kismet-2007-10-R1/ringbuf.cc
--- kismet-2007-10-R1-old/ringbuf.cc	2005-12-12 05:14:39.000000000 +1000
+++ kismet-2007-10-R1/ringbuf.cc	2008-08-17 15:50:43.000000000 +1000
@@ -17,6 +17,7 @@
 */
 
 #include "ringbuf.h"
+#include <cstring>
 
 RingBuffer::RingBuffer(int in_size) {
     ring_len = in_size;
diff -Naur kismet-2007-10-R1-old/util.cc kismet-2007-10-R1/util.cc
--- kismet-2007-10-R1-old/util.cc	2006-08-29 06:37:29.000000000 +1000
+++ kismet-2007-10-R1/util.cc	2008-08-17 15:51:03.000000000 +1000
@@ -22,6 +22,7 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <fcntl.h>
+#include <cstring>
 
 // We need this to make uclibc happy since they don't even have rintf...
 #ifndef rintf
