--- media/media.gyp.orig	2012-02-02 10:03:41.000000000 +0100
+++ media/media.gyp	2012-02-02 10:04:09.000000000 +0100
@@ -467,14 +467,6 @@
         [ 'os_posix == 1 and OS != "mac" and OS != "android"', {
           'cflags': [
             '-msse2',
-            '-msse3',
-            '-mssse3',
-          ],
-        }],
-        [ 'OS == "openbsd"', {
-          # OpenBSD's gcc (4.2.1) does not support -mssse3
-          'cflags!': [
-            '-mssse3',
           ],
         }],
         [ 'OS == "mac"', {
