diff -aur old//Makefile.am new//Makefile.am
--- old//Makefile.am	2006-12-21 23:22:37.000000000 +0100
+++ new//Makefile.am	2010-11-25 18:15:29.253376150 +0100
@@ -16,7 +16,7 @@
 	$(CC) -c $(DEFS) $(ourdefs) $(CFLAGS) $(CPPFLAGS) $(LDFLAGS) $<
 
 install-exec-hook:
-	chmod u+s $(libexecdir)/rssh_chroot_helper
+	chmod u+s $(DESTDIR)$(libexecdir)/rssh_chroot_helper
 
 rpm:	dist
 	rpmbuild -ta --sign $(base).tar.gz
diff -aur old//Makefile.in new//Makefile.in
--- old//Makefile.in	2010-08-01 15:59:54.000000000 +0200
+++ new//Makefile.in	2010-11-25 18:15:29.253376150 +0100
@@ -830,7 +830,7 @@
 	$(CC) -c $(DEFS) $(ourdefs) $(CFLAGS) $(CPPFLAGS) $(LDFLAGS) $<
 
 install-exec-hook:
-	chmod u+s $(libexecdir)/rssh_chroot_helper
+	chmod u+s $(DESTDIR)$(libexecdir)/rssh_chroot_helper
 
 rpm:	dist
 	rpmbuild -ta --sign $(base).tar.gz
