diff -Naur plt-4.2.3-orig/src/wxcommon/wxJPEG.cxx plt-4.2.3/src/wxcommon/wxJPEG.cxx
--- plt-4.2.3-orig/src/wxcommon/wxJPEG.cxx	2010-01-18 00:27:47.000000000 -0500
+++ plt-4.2.3/src/wxcommon/wxJPEG.cxx	2010-01-18 00:33:49.000000000 -0500
@@ -726,7 +726,12 @@
        png_set_strip_16(png_ptr);
 
      /* Expand grayscale images to the full 8 bits from 1, 2, or 4 bits/pixel */
+#if PNG_LIBPNG_VER < 10400
      png_set_gray_1_2_4_to_8(png_ptr);
+#else
+     png_set_expand_gray_1_2_4_to_8(png_ptr);
+#endif
+
    }
 
    /* Set the background color to draw transparent and alpha images over.
