--- ufraw-0.11/configure.ac.exiv2	2007-03-06 15:02:40.000000000 -0600
+++ ufraw-0.11/configure.ac	2007-04-25 13:09:06.000000000 -0500
@@ -185,12 +185,17 @@
 AC_MSG_RESULT($with_exiv2)
 have_exiv2=no
 if test $with_exiv2 = yes; then
+PKG_CHECK_MODULES(EXIV2, exiv2 >= 0.14,
+ [ have_exiv2=yes
+   AC_DEFINE(HAVE_EXIV2, 1, have exiv2)],
+ [ 
   AC_CHECK_PROGS(EXIV2_CONFIG, exiv2-config, , $PATH)
   if test -n "$EXIV2_CONFIG"; then
     have_exiv2=yes
     AC_DEFINE(HAVE_EXIV2, 1, have exiv2)
     EXIV2_LIBS=`$EXIV2_CONFIG --libs`
   fi
+ ])
 fi
 
 UFRAW_CFLAGS="$UFRAW_CFLAGS $EXIF_CFLAGS $GIMP_CFLAGS $GTK_CFLAGS $GLIB_CFLAGS $LCMS_CFLAGS"
--- ufraw-0.11/ufraw_exiv2.cc.exiv2	2007-02-08 15:28:41.000000000 -0600
+++ ufraw-0.11/ufraw_exiv2.cc	2007-04-25 13:10:16.000000000 -0500
@@ -201,7 +201,7 @@
 }
 catch (Exiv2::AnyError& e) {
     std::cerr.rdbuf(savecerr);
-    ufraw_message(UFRAW_SET_WARNING, "%s\n", e.what().c_str());
+    ufraw_message(UFRAW_SET_WARNING, "%s\n", e.what());
     return UFRAW_ERROR;
 }
 
