Index: ffserver.c
===================================================================
--- ffserver.c	(revision 23065)
+++ ffserver.c	(working copy)
@@ -2232,7 +2232,7 @@
     switch(c->state) {
     case HTTPSTATE_SEND_DATA_HEADER:
         memset(&c->fmt_ctx, 0, sizeof(c->fmt_ctx));
-        av_metadata_set2(&c->fmt_ctx.metadata, "author"   , c->stream->author   , 0);
+        av_metadata_set2(&c->fmt_ctx.metadata, "artist"   , c->stream->author   , 0);
         av_metadata_set2(&c->fmt_ctx.metadata, "comment"  , c->stream->comment  , 0);
         av_metadata_set2(&c->fmt_ctx.metadata, "copyright", c->stream->copyright, 0);
         av_metadata_set2(&c->fmt_ctx.metadata, "title"    , c->stream->title    , 0);
Index: libavformat/rm.c
===================================================================
--- libavformat/rm.c	(revision 23065)
+++ libavformat/rm.c	(working copy)
@@ -23,7 +23,7 @@
 
 const char * const ff_rm_metadata[4] = {
     "title",
-    "author",
+    "artist",
     "copyright",
     "comment"
 };
Index: libavformat/mov.c
===================================================================
--- libavformat/mov.c	(revision 23065)
+++ libavformat/mov.c	(working copy)
@@ -144,7 +144,7 @@
     switch (atom.type) {
     case MKTAG(0xa9,'n','a','m'): key = "title";     break;
     case MKTAG(0xa9,'a','u','t'):
-    case MKTAG(0xa9,'A','R','T'): key = "author";    break;
+    case MKTAG(0xa9,'A','R','T'): key = "artist";    break;
     case MKTAG(0xa9,'w','r','t'): key = "composer";  break;
     case MKTAG( 'c','p','r','t'):
     case MKTAG(0xa9,'c','p','y'): key = "copyright"; break;
Index: libavformat/vqf.c
===================================================================
--- libavformat/vqf.c	(revision 23065)
+++ libavformat/vqf.c	(working copy)
@@ -113,7 +113,7 @@
             add_metadata(s, "copyright", len, header_size);
             break;
         case MKTAG('A','U','T','H'):
-            add_metadata(s, "author"   , len, header_size);
+            add_metadata(s, "artist"   , len, header_size);
             break;
         case MKTAG('A','L','B','M'):
             add_metadata(s, "album"    , len, header_size);
Index: libavformat/rpl.c
===================================================================
--- libavformat/rpl.c	(revision 23065)
+++ libavformat/rpl.c	(working copy)
@@ -135,7 +135,7 @@
     error |= read_line(pb, line, sizeof(line));      // date/copyright
     av_metadata_set2(&s->metadata, "copyright", line, 0);
     error |= read_line(pb, line, sizeof(line));      // author and other
-    av_metadata_set2(&s->metadata, "author"   , line, 0);
+    av_metadata_set2(&s->metadata, "artist"   , line, 0);
 
     // video headers
     vst = av_new_stream(s, 0);
Index: libavformat/mp3.c
===================================================================
--- libavformat/mp3.c	(revision 23065)
+++ libavformat/mp3.c	(working copy)
@@ -215,7 +215,7 @@
     buf[1] = 'A';
     buf[2] = 'G';
     count += id3v1_set_string(s, "title",   buf +  3, 30);
-    count += id3v1_set_string(s, "author",  buf + 33, 30);
+    count += id3v1_set_string(s, "artist",  buf + 33, 30);
     count += id3v1_set_string(s, "album",   buf + 63, 30);
     count += id3v1_set_string(s, "date",    buf + 93,  4);
     count += id3v1_set_string(s, "comment", buf + 97, 30);
Index: libavformat/movenc.c
===================================================================
--- libavformat/movenc.c	(revision 23065)
+++ libavformat/movenc.c	(working copy)
@@ -1444,7 +1444,7 @@
     put_be32(pb, 0); /* size */
     put_tag(pb, "ilst");
     mov_write_string_metadata(s, pb, "\251nam", "title"    , 1);
-    mov_write_string_metadata(s, pb, "\251ART", "author"   , 1);
+    mov_write_string_metadata(s, pb, "\251ART", "artist"   , 1);
     mov_write_string_metadata(s, pb, "aART", "album_artist", 1);
     mov_write_string_metadata(s, pb, "\251wrt", "composer" , 1);
     mov_write_string_metadata(s, pb, "\251alb", "album"    , 1);
@@ -1572,7 +1572,7 @@
 
         if (mov->mode & MODE_3GP) {
             mov_write_3gp_udta_tag(pb_buf, s, "titl", "title");
-            mov_write_3gp_udta_tag(pb_buf, s, "auth", "author");
+            mov_write_3gp_udta_tag(pb_buf, s, "auth", "artist");
             mov_write_3gp_udta_tag(pb_buf, s, "gnre", "genre");
             mov_write_3gp_udta_tag(pb_buf, s, "dscp", "comment");
             mov_write_3gp_udta_tag(pb_buf, s, "albm", "album");
@@ -1580,7 +1580,7 @@
             mov_write_3gp_udta_tag(pb_buf, s, "yrrc", "date");
         } else if (mov->mode == MODE_MOV) { // the title field breaks gtkpod with mp4 and my suspicion is that stuff is not valid in mp4
             mov_write_string_metadata(s, pb_buf, "\251nam", "title"      , 0);
-            mov_write_string_metadata(s, pb_buf, "\251aut", "author"     , 0);
+            mov_write_string_metadata(s, pb_buf, "\251aut", "artist"     , 0);
             mov_write_string_metadata(s, pb_buf, "\251alb", "album"      , 0);
             mov_write_string_metadata(s, pb_buf, "\251day", "date"       , 0);
             mov_write_string_tag(pb_buf, "\251enc", LIBAVFORMAT_IDENT, 0, 0);
Index: libavformat/aiffdec.c
===================================================================
--- libavformat/aiffdec.c	(revision 23065)
+++ libavformat/aiffdec.c	(working copy)
@@ -223,7 +223,7 @@
             get_meta(s, "title"    , size);
             break;
         case MKTAG('A', 'U', 'T', 'H'):     /* Author chunk */
-            get_meta(s, "author"   , size);
+            get_meta(s, "artist"   , size);
             break;
         case MKTAG('(', 'c', ')', ' '):     /* Copyright chunk */
             get_meta(s, "copyright", size);
