diff -aur old/configure new/configure
--- old/configure	2011-02-10 20:34:48.153348697 +0100
+++ new/configure	2011-02-10 20:35:09.623348816 +0100
@@ -174,7 +174,7 @@
 
 
 QT4_IM='N'
-QT4_IM_DIR=/usr/$LIB/qt4/plugins/inputmethods
+QT4_IM_DIR=/usr/$LIB/qt/plugins/inputmethods
 echo "testing if you have qt4 support"
 if [ -d  $QT4_IM_DIR ]; then
   QT4_IM='Y'
diff -aur old/qt4-im/Makefile new/qt4-im/Makefile
--- old/qt4-im/Makefile	2011-02-10 20:34:23.236682152 +0100
+++ new/qt4-im/Makefile	2011-02-10 20:35:09.623348816 +0100
@@ -1,5 +1,5 @@
 include ../config.mak
-QT=qt4
+QT=qt
 QTIM=$(QT)/plugins/inputmethods
 IMMODULES=$(libdir)/$(QTIM)
 
@@ -15,7 +15,7 @@
 	$(CXX) -E $(CFLAGS) $(INCS) $< > $@
 
 moc_gcin-qt.cpp:	gcin-qt.h
-	/usr/lib/$(QT)/bin/moc $< -o moc_gcin-qt.cpp
+	/usr/bin/moc $< -o moc_gcin-qt.cpp
 
 im-gcin.so: $(OBJS)
 	export LD_RUN_PATH=$(gcinlibdir) ;\
