diff -aur old/configure new/configure
--- old/configure	2011-11-07 06:51:12.000000000 +1100
+++ new/configure	2011-11-25 14:02:10.819256960 +1100
@@ -188,7 +188,7 @@
 
 
 QT4_IM='N'
-QT4_IM_DIR=/usr/$LIB/qt4/plugins/inputmethods
+QT4_IM_DIR=/usr/$LIB/qt/plugins/inputmethods
 echo "testing if you have qt4 support"
 pkg-config --libs QtCore QtGui > /dev/null 2> /dev/null
 if [ $? = 0 ]; then
diff -aur old/qt4-im/Makefile new/qt4-im/Makefile
--- old/qt4-im/Makefile	2011-06-08 23:03:50.000000000 +1000
+++ new/qt4-im/Makefile	2011-11-25 14:05:28.844110772 +1100
@@ -1,5 +1,5 @@
 include ../config.mak
-QT=qt4
+QT=qt
 QTIM=$(QT)/plugins/inputmethods
 IMMODULES=$(libdir)/$(QTIM)
 
@@ -15,7 +15,7 @@
 	$(CXX) -E $(CFLAGS) $(INCS) $< > $@
 
 moc_gcin-qt.cpp:	gcin-qt.h
-	/usr/$(LIB)/$(QT)/bin/moc $< -o moc_gcin-qt.cpp
+	/usr/bin/moc $< -o moc_gcin-qt.cpp
 
 im-gcin.so: $(OBJS)
 	export LD_RUN_PATH=$(gcinlibdir) ;\
