--- tools/qdoc3/qdoc3.pro       2009-06-25 17:34:42.000000000 +0200
+++ tools/qdoc3/qdoc3.pro.new   2009-06-25 17:35:42.000000000 +0200
@@ -8,6 +8,11 @@
     CONFIG -= build_all
     CONFIG += release
 }
+
+TARGET= qdoc3
+target.path = $$[QT_INSTALL_BINS]
+INSTALLS += target
+
 mac:CONFIG -= app_bundle
 HEADERS += apigenerator.h \
            archiveextractor.h \
diff -up qt-x11-opensource-src-4.5.0/tools/tools.pro.qdoc3 qt-x11-opensource-src-4.5.0/tools/tools.pro
--- tools/tools.pro.qdoc3	2009-02-25 15:09:27.000000000 -0600
+++ tools/tools.pro	2009-03-04 09:44:18.000000000 -0600
@@ -13,6 +13,7 @@ no-png {
          SUBDIRS += designer
      }
      SUBDIRS     += linguist
+     SUBDIRS     += qdoc3
      wince*: SUBDIRS = qtestlib designer
      unix:!mac:!embedded:contains(QT_CONFIG, qt3support):SUBDIRS += qtconfig
      win32:!wince*:!contains(QT_EDITION, OpenSource|Console):SUBDIRS += activeqt
