--- nxclient~	2005-08-02 19:49:08.000000000 +0200
+++ nxclient	2006-07-03 15:14:43.000000000 +0200
@@ -21,7 +21,7 @@
 
 TEMP=`getopt -a -o d: --long local,noautokill,dialog:,caption:,message:,display:,printer: -n $(basename $0) -- "$@"`
 
-if [ $? != 0 ] ; then echo "Terminating..." >&2 ; exit 1 ; fi
+if [ $? = 0 ] ; then echo "Terminating..." >&2 ; exit 1 ; fi
 
 # Note the quotes around `$TEMP': they are essential!
 eval set -- "$TEMP"
