--- nxnode.old	2007-04-20 19:59:07.000000000 +0200
+++ nxnode	2007-10-09 08:10:50.000000000 +0200
@@ -300,7 +300,7 @@
 	
 	if [ "$virtualdesktop" = "1" -a "$type" = "unix-application" -a "$DEFAULT_X_WM" != "" -a -x "$(find_app $DEFAULT_X_WM)" ]
 	then
-		DISPLAY=unix:$display $DEFAULT_X_WM >>"$USER_FAKE_HOME/.nx/C-$sess_id/session" 2>&1 &
+		DISPLAY=:$display $DEFAULT_X_WM >>"$USER_FAKE_HOME/.nx/C-$sess_id/session" 2>&1 &
 		NODE_WM_PID=$!
 	fi
 
@@ -308,7 +308,7 @@
 	# Startup the application
 	#
 	
-	DISPLAY=unix:$display $NODE_APPLICATION >>"$USER_FAKE_HOME/.nx/C-$sess_id/session" 2>&1 &
+	DISPLAY=:$display $NODE_APPLICATION >>"$USER_FAKE_HOME/.nx/C-$sess_id/session" 2>&1 &
 	NODE_APP_PID=$!
 	
 	mkdir -p "$USER_FAKE_HOME/.nx/C-$sess_id/pids/"
@@ -1087,13 +1087,13 @@
 
 $COMMAND_XAUTH >/dev/null 2>&1 <<EOF
 add localhost:$display MIT-MAGIC-COOKIE-1 $cookie
-add unix:$display MIT-MAGIC-COOKIE-1 $cookie
+add :$display MIT-MAGIC-COOKIE-1 $cookie
 exit
 EOF
 
 $COMMAND_XAUTH -f "$USER_FAKE_HOME/.nx/C-$sess_id/authority" >/dev/null 2>&1 <<EOF
 add localhost:$display MIT-MAGIC-COOKIE-1 $cookie
-add unix:$display MIT-MAGIC-COOKIE-1 $cookie
+add :$display MIT-MAGIC-COOKIE-1 $cookie
 exit
 EOF
 
@@ -1101,7 +1101,7 @@
 
 cat << EOF >$USER_FAKE_HOME/.nx/C-$sess_id/scripts/authority
 remove localhost:$display
-remove unix:$display
+remove :$display
 exit
 EOF
 
@@ -1126,7 +1126,7 @@
 then
 	sess_id="$SERVER_NAME-$display-$uniqueid"
 	# go into background immediately
-	NXSESSIONID="$sess_id" DISPLAY=unix:$display "$NODE_AUTOSTART" "$1" >/dev/null 2>&1 &
+	NXSESSIONID="$sess_id" DISPLAY=:$display "$NODE_AUTOSTART" "$1" >/dev/null 2>&1 &
 	# dont't wait for this child!
 	disown $!
 fi
