--- speech_tools/ling_class/item_feats.cc	2006/06/27 11:43:49	1.1
+++ speech_tools/ling_class/item_feats.cc	2006/06/27 11:44:25
@@ -78,7 +78,7 @@ EST_String get_featname(const EST_Item_f
    EST_String name = EST_FeatureFunctionContext::global->get_featfunc_name(func, found);
 
    if (!found)
-	EST_error("featfunc 0x%x has no name", (int)func);
+	EST_error("featfunc 0x%p has no name", func);
 
     return name;
 }
