--- koffice-1.6.2/kexi/examples/build_kexi_file.sh.orig	2007-03-06 03:30:23.000000000 +0000
+++ koffice-1.6.2/kexi/examples/build_kexi_file.sh	2007-03-06 03:31:56.000000000 +0000
@@ -9,7 +9,7 @@
 
 KEXISQL3PATH=../3rdparty/kexisql3/src/.libs/
 PATH=$PATH:$KEXISQL3PATH
-LD_LIBRARY_PATH=$KEXISQL3PATH:$KEXISQL3PATH
+export LD_LIBRARY_PATH=$KEXISQL3PATH:$KEXISQL3PATH
 which ksqlite > /dev/null || exit 1
 
 [ $# -lt 1 ] && echo "Missing .sql filename." && exit 1
