diff --git a/libdiscover/backends/FlatpakBackend/FlatpakFetchDataJob.h b/libdiscover/backends/FlatpakBackend/FlatpakFetchDataJob.h
index b9aaf80e..3e01eb7b 100644
--- a/libdiscover/backends/FlatpakBackend/FlatpakFetchDataJob.h
+++ b/libdiscover/backends/FlatpakBackend/FlatpakFetchDataJob.h
@@ -8,10 +8,8 @@
 #define FLATPAKFETCHDATAJOB_H
 
 #include <QByteArray>
-extern "C" {
 #include <flatpak.h>
 #include <glib.h>
-}
 
 class FlatpakResource;
 
diff --git a/libdiscover/backends/FlatpakBackend/FlatpakJobTransaction.cpp b/libdiscover/backends/FlatpakBackend/FlatpakJobTransaction.cpp
index 9499c566..37adb046 100644
--- a/libdiscover/backends/FlatpakBackend/FlatpakJobTransaction.cpp
+++ b/libdiscover/backends/FlatpakBackend/FlatpakJobTransaction.cpp
@@ -13,9 +13,6 @@
 #include <QDebug>
 #include <QTimer>
 
-extern "C" {
-}
-
 FlatpakJobTransaction::FlatpakJobTransaction(FlatpakResource *app, Role role, bool delayStart)
     : Transaction(app->backend(), app, role, {})
     , m_app(app)
diff --git a/libdiscover/backends/FlatpakBackend/FlatpakJobTransaction.h b/libdiscover/backends/FlatpakBackend/FlatpakJobTransaction.h
index ea2aafaa..bcfbc31e 100644
--- a/libdiscover/backends/FlatpakBackend/FlatpakJobTransaction.h
+++ b/libdiscover/backends/FlatpakBackend/FlatpakJobTransaction.h
@@ -11,11 +11,9 @@
 #include <Transaction/Transaction.h>
 #include <QPointer>
 
-extern "C" {
 #include <flatpak.h>
 #include <gio/gio.h>
 #include <glib.h>
-}
 
 class FlatpakResource;
 class FlatpakTransactionThread;
diff --git a/libdiscover/backends/FlatpakBackend/FlatpakTransactionThread.h b/libdiscover/backends/FlatpakBackend/FlatpakTransactionThread.h
index b8eef827..f1c21997 100644
--- a/libdiscover/backends/FlatpakBackend/FlatpakTransactionThread.h
+++ b/libdiscover/backends/FlatpakBackend/FlatpakTransactionThread.h
@@ -7,11 +7,9 @@
 #ifndef FLATPAKTRANSACTIONTHREAD_H
 #define FLATPAKTRANSACTIONTHREAD_H
 
-extern "C" {
 #include <flatpak.h>
 #include <gio/gio.h>
 #include <glib.h>
-}
 
 #include <Transaction/Transaction.h>
 #include <QThread>
