diff --git a/src/modules/plus/producer_count.c b/src/modules/plus/producer_count.c
index 9ef4fbcf..0126b134 100644
--- a/src/modules/plus/producer_count.c
+++ b/src/modules/plus/producer_count.c
@@ -638,8 +638,9 @@ mlt_producer producer_count_init( mlt_profile profile, mlt_service_type type, co
 		mlt_properties_set( properties, "sound", "none" );
 		mlt_properties_set( properties, "background", "clock" );
 		mlt_properties_set( properties, "drop", "0" );
+		mlt_properties_clear( properties, "resource" );
 		// Let the arg specify the producer to use with the factory, e.g. loader-nogl
-		if (arg && strcmp(arg, "") && strcmp(arg, "<producer>"))
+		if (arg && strcmp(arg, ""))
 			mlt_properties_set(properties, FACTORY_PRODUCER, arg);
 
 		// Callback registration
