diff --git a/src/modules/opencv/configure b/src/modules/opencv/configure
index 36851f67..f8084301 100755
--- a/src/modules/opencv/configure
+++ b/src/modules/opencv/configure
@@ -26,18 +26,18 @@ else
 		exit 0
 	fi
 
-    pkg-config --atleast-version=3.1.0 'opencv'
+    pkg-config --atleast-version=3.1.0 'opencv4'
 	if [ $? -eq 0 ]
 	then
-                result=`pkg-config --libs opencv | grep "opencv_tracking"`
+                result=`pkg-config --libs opencv4 | grep "opencv_tracking"`
                 if [ -z "$result" ]
                 then
                         echo "- OpenCV tracking contrib module NOT found, disabling OpenCV modules"
                         touch ../disable-opencv
                         exit 0
                 else
-                        echo "CFLAGS += $(pkg-config --cflags opencv)" >> config.mak
-                        echo "LDFLAGS += $(pkg-config --libs opencv)" >> config.mak
+                        echo "CFLAGS += $(pkg-config --cflags opencv4)" >> config.mak
+                        echo "LDFLAGS += $(pkg-config --libs opencv4)" >> config.mak
                 fi
 	else
 	    echo "- OpenCV >= 3.1.0 NOT found: disabling"
diff --git a/src/modules/opencv/filter_opencv_tracker.cpp b/src/modules/opencv/filter_opencv_tracker.cpp
index d49862fc..2a176ef8 100644
--- a/src/modules/opencv/filter_opencv_tracker.cpp
+++ b/src/modules/opencv/filter_opencv_tracker.cpp
@@ -110,7 +110,7 @@ static void analyze( mlt_filter filter, cv::Mat cvFrame, private_data* data, int
         {
 		// Build tracker
 		data->algo = mlt_properties_get( filter_properties, "algo" );
-#if CV_VERSION_MAJOR == 3 && CV_VERSION_MINOR >= 3
+#if (CV_VERSION_MAJOR == 3 && CV_VERSION_MINOR >= 3) || CV_VERSION_MAJOR >= 4
 		if ( !data->algo || *data->algo == '\0' || !strcmp(data->algo, "KCF" ) )
 		{
 			data->tracker = cv::TrackerKCF::create();
