--- sc/source/ui/vba/vbaapplication.cxx
+++ sc/source/ui/vba/vbaapplication.cxx
@@ -293,7 +293,7 @@
 ScVbaApplication::setStatusBar( const uno::Any& _statusbar ) throw (uno::RuntimeException)
 {
     rtl::OUString sText;
-    sal_Bool bDefault;
+    sal_Bool bDefault = sal_False;
 	uno::Reference< frame::XModel > xModel( getCurrentDocument(), uno::UNO_QUERY_THROW );
     uno::Reference< task::XStatusIndicatorSupplier > xStatusIndicatorSupplier( xModel->getCurrentController(), uno::UNO_QUERY_THROW );
     uno::Reference< task::XStatusIndicator > xStatusIndicator( xStatusIndicatorSupplier->getStatusIndicator(), uno::UNO_QUERY_THROW );
@@ -426,7 +426,7 @@
     
     if( Scroll.hasValue() )
     {
-        sal_Bool aScroll;
+        sal_Bool aScroll = sal_False;
         if( Scroll >>= aScroll )
         {
             bScroll = aScroll;
--- sc/source/ui/vba/vbahelper.hxx
+++ sc/source/ui/vba/vbahelper.hxx
@@ -38,8 +38,8 @@
 #include <com/sun/star/frame/XModel.hpp>
 #include <docsh.hxx>
 
-#define css com::sun::star
-#define oo org::openoffice
+namespace css = ::com::sun::star;
+
 namespace org 
 {
 	namespace openoffice 
@@ -69,6 +69,8 @@
 	}
 }
 
+namespace oo = org::openoffice;
+
 #ifdef DEBUG
 #  define SC_VBA_FIXME(a) OSL_TRACE( a )
 #  define SC_VBA_STUB() SC_VBA_FIXME(( "%s - stubbed\n", __FUNCTION__ ))
--- sc/source/ui/vba/vbarange.cxx
+++ sc/source/ui/vba/vbarange.cxx
@@ -487,7 +487,7 @@
 	{
 		case  uno::TypeClass_BOOLEAN:
 		{
-			sal_Bool bState;
+			sal_Bool bState = sal_False;
 			if ( aValue >>= bState 	 )
 			{
 				uno::Reference< table::XCellRange > xRange( xCell, uno::UNO_QUERY_THROW );
@@ -2318,7 +2318,7 @@
 		return;
 	}
 
-	sal_Bool bHidden;
+	sal_Bool bHidden = sal_False;
 	if ( !(_hidden >>= bHidden) )
 		throw uno::RuntimeException( ::rtl::OUString( RTL_CONSTASCII_USTRINGPARAM( "Failed to extract param for Hidden property" ) ), uno::Reference< uno::XInterface >() ); 
 
--- sc/source/ui/vba/vbaworksheets.cxx
+++ sc/source/ui/vba/vbaworksheets.cxx
@@ -227,7 +227,7 @@
 void SAL_CALL 
 ScVbaWorksheets::setVisible( const uno::Any& _visible ) throw (uno::RuntimeException)
 {
-	sal_Bool bState;
+	sal_Bool bState = sal_False;
 	if ( _visible >>= bState )
 	{
 		uno::Reference< container::XEnumeration > xEnum( createEnumeration(), uno::UNO_QUERY_THROW );
--- sc/util/makefile.mk
+++ sc/util/makefile.mk
@@ -266,6 +266,7 @@
 
 SHL9STDLIBS= \
 		$(CPPUHELPERLIB) \
+		$(VCLLIB) \
 		$(CPPULIB) \
 		$(COMPHELPERLIB) \
 		$(SVLIB) \
@@ -282,6 +283,12 @@
 SHL9DEPN=$(SHL1TARGETN) $(SHL8TARGETN)
 SHL9LIBS=$(SLB)$/$(TARGET_VBA).lib
 
+.IF "$(GUI)"=="UNX" || "$(GUI)"=="MAC"
+    LIBCOMPNAME=$(LOCAL_COMMON_OUT)$/lib/lib$(SHL9TARGET)$(DLLPOST)
+.ELSE
+    LIBCOMPNAME=$(COMMONBIN)$/$(SHL9TARGET)$(DLLPOST)
+.ENDIF
+
 .ENDIF
  
 
