diff -urN BitTorrent-5.0.0/bittorrent-curses.py BitTorrent-5.0.0.new/bittorrent-curses.py
--- BitTorrent-5.0.0/bittorrent-curses.py	2006-10-26 23:42:56.000000000 +0200
+++ BitTorrent-5.0.0.new/bittorrent-curses.py	2006-11-07 23:44:19.000000000 +0100
@@ -42,7 +42,7 @@
 from BitTorrent import BTFailure, UserFailure
 from BitTorrent import version
 from BTL import GetTorrent
-from BitTorrent.RawServer_twisted import RawServer, task
+from BitTorrent.RawServer_twisted import RawServer
 from BTL.ConvertedMetainfo import ConvertedMetainfo
 from BTL.yielddefer import launch_coroutine, _wrap_task
 from BitTorrent import inject_main_logfile
@@ -498,12 +498,12 @@
             if config['save_in']:
                 raise BTFailure(_("You cannot specify both --save_as and "
                                   "--save_in."))
-            saveas,bad = platform.encode_for_filesystem(config['save_as'])
+            saveas,bad = encode_for_filesystem(config['save_as'])
             if bad:
                 raise BTFailure(_("Invalid path encoding."))
             savein = os.path.dirname(os.path.abspath(saveas))
         elif config['save_in']:
-            savein,bad = platform.encode_for_filesystem(config['save_in'])
+            savein,bad = encode_for_filesystem(config['save_in'])
             if bad:
                 raise BTFailure(_("Invalid path encoding."))
             saveas = os.path.join(savein,torrent_name)
@@ -511,14 +511,14 @@
             saveas = torrent_name
         if config['save_incomplete_in']:
             save_incomplete_in,bad = \
-                platform.encode_for_filesystem(config['save_incomplete_in'])
+                encode_for_filesystem(config['save_incomplete_in'])
             if bad:
                 raise BTFailure(_("Invalid path encoding."))
             save_incomplete_as = os.path.join(save_incomplete_in,torrent_name)
         else:
             save_incomplete_as = os.path.join(savein,torrent_name)
     
-        data_dir,bad = platform.encode_for_filesystem(config['data_dir'])
+        data_dir,bad = encode_for_filesystem(config['data_dir'])
         if bad:
             raise BTFailure(_("Invalid path encoding."))
 
diff -urN BitTorrent-5.0.0/launchmany-console.py BitTorrent-5.0.0.new/launchmany-console.py
--- BitTorrent-5.0.0/launchmany-console.py	2006-10-26 23:44:14.000000000 +0200
+++ BitTorrent-5.0.0.new/launchmany-console.py	2006-11-07 23:44:19.000000000 +0100
@@ -80,10 +80,10 @@
         if args:
             torrent_dir = args[0]
             config['torrent_dir'] = \
-                platform.decode_from_filesystem(torrent_dir)
+                decode_from_filesystem(torrent_dir)
         else:
             torrent_dir = config['torrent_dir']
-            torrent_dir,bad = platform.encode_for_filesystem(torrent_dir)
+            torrent_dir,bad = encode_for_filesystem(torrent_dir)
             if bad:
               raise BTFailure(_("Warning: ")+config['torrent_dir']+
                               _(" is not a directory"))
diff -urN BitTorrent-5.0.0/launchmany-curses.py BitTorrent-5.0.0.new/launchmany-curses.py
--- BitTorrent-5.0.0/launchmany-curses.py	2006-10-26 23:44:14.000000000 +0200
+++ BitTorrent-5.0.0.new/launchmany-curses.py	2006-11-07 23:44:19.000000000 +0100
@@ -316,10 +316,10 @@
         if args:
             torrent_dir = args[0]
             config['torrent_dir'] = \
-                platform.decode_from_filesystem(torrent_dir)    
+                decode_from_filesystem(torrent_dir)    
         else:
             torrent_dir = config['torrent_dir']
-            torrent_dir,bad = platform.encode_for_filesystem(torrent_dir)
+            torrent_dir,bad = encode_for_filesystem(torrent_dir)
             if bad:
               raise BTFailure(_("Warning: ")+config['torrent_dir']+
                               _(" is not a directory"))
