Index: xine-ui-0.99.6/src/xitk/Imlib-light/load.c
===================================================================
--- xine-ui-0.99.6.orig/src/xitk/Imlib-light/load.c
+++ xine-ui-0.99.6/src/xitk/Imlib-light/load.c
@@ -61,7 +61,7 @@ unsigned char *_LoadPNG(ImlibData * id,
       png_destroy_read_struct(&png_ptr, NULL, NULL);
       return NULL;
     }
-  if (info_ptr->color_type == PNG_COLOR_TYPE_RGB_ALPHA)
+  if (png_get_color_type(png_ptr, info_ptr) == PNG_COLOR_TYPE_RGB_ALPHA)
     {
       png_destroy_read_struct(&png_ptr, &info_ptr, NULL);
       return NULL;
