diff -ru martian-full-20080407.orig/kmodule/kmartian.h martian-full-20080407/kmodule/kmartian.h
--- martian-full-20080407.orig/kmodule/kmartian.h	2006-12-01 13:13:01.000000000 +1100
+++ martian-full-20080407/kmodule/kmartian.h	2008-06-07 13:43:11.000000000 +1000
@@ -11,6 +11,11 @@
 #include "fifo.h"
 #include "mfifo.h"
 
+#ifndef __stringify
+#define __stringify_nomacro(x)	#x
+#define __stringify(x)		__stringify_nomacro (x)
+#endif
+
 #define MPREFIX			__stringify (KBUILD_MODNAME)
 
 #define MINFO(fmt...)		printk (KERN_INFO MPREFIX ": " fmt)
