--- yelp-2.20.0/configure.in.old	2007-12-02 18:55:15.000000000 +0900
+++ yelp-2.20.0/configure.in	2007-12-02 18:55:47.000000000 +0900
@@ -133,7 +133,7 @@
 dnl beagle
 dnl ******
 
-BEAGLE_MODULES="libbeagle-0.0 >= 0.2.4"
+BEAGLE_MODULES="libbeagle-1.0 >= 0.3.0"
 
 AC_ARG_WITH([search],
 	[AC_HELP_STRING([--with-search=basic|beagle|auto],
--- yelp-2.20.0/src/yelp-search-parser.c.old	2007-12-02 20:05:27.000000000 +0900
+++ yelp-2.20.0/src/yelp-search-parser.c	2007-12-02 20:06:03.000000000 +0900
@@ -552,7 +552,7 @@
 
 	beagle_query_set_max_hits (query, 10000);
 	beagle_query_add_text (query, parser->search_terms);
-	beagle_query_add_source (query, "documentation");
+//	beagle_query_add_source (query, "documentation");
 
 	parser->hits = g_ptr_array_new ();
 
