--- azureus/com/aelitis/azureus/core/clientmessageservice/secure/impl/SecureMessageServiceClientHelper.java.orig	2007-01-20 18:23:49.000000000 +0100
+++ azureus/com/aelitis/azureus/core/clientmessageservice/secure/impl/SecureMessageServiceClientHelper.java	2007-01-20 18:25:29.000000000 +0100
@@ -36,7 +36,7 @@
 import org.bouncycastle.crypto.encodings.PKCS1Encoding;
 import org.bouncycastle.crypto.engines.RSAEngine;
 import org.bouncycastle.crypto.params.ParametersWithRandom;
-import org.bouncycastle.jce.provider.RSAUtil;
+import org.bouncycastle.jce.provider.DSAUtil;
 import org.gudy.azureus2.core3.util.Debug;
 import org.gudy.azureus2.plugins.utils.StaticUtilities;
 
@@ -97,7 +97,7 @@
 				
 				PKCS1Encoding	padded_eng = new PKCS1Encoding( eng );
 				
-	            CipherParameters param = RSAUtil.generatePublicKeyParameter(public_key);
+	            CipherParameters param = DSAUtil.generatePublicKeyParameter(public_key);
 	            
 	            param = new ParametersWithRandom(param, new SecureRandom());
 	            
