--- org/gudy/azureus2/core3/internat/MessageText.java	2007-02-12 18:10:08.000000000 +0100
+++ org/gudy/azureus2/core3/internat/MessageText.java	2007-07-01 20:28:57.000000000 +0200
@@ -30,6 +30,7 @@
 import java.util.jar.JarFile;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
+import java.security.PrivilegedAction;
 
 import org.gudy.azureus2.core3.config.COConfigurationManager;
 import org.gudy.azureus2.core3.logging.LogAlert;
@@ -39,8 +40,6 @@
 import org.gudy.azureus2.core3.util.FileUtil;
 import org.gudy.azureus2.core3.util.SystemProperties;
 
-import sun.security.action.GetPropertyAction;
-
 /**
  * @author Arbeiten
  * 
@@ -685,10 +684,18 @@
   	// a way to call this code explicitly..
     String language, region, country, variant;
     language = (String) AccessController.doPrivileged(
-                    new GetPropertyAction("user.language", "en"));
+		    new PrivilegedAction() {
+			    public Object run() {
+				    return System.getProperty("user.language", "en");
+			    }
+		    });
     // for compatibility, check for old user.region property
     region = (String) AccessController.doPrivileged(
-                    new GetPropertyAction("user.region"));
+		    new PrivilegedAction() {
+			    public Object run() {
+				    return System.getProperty("user.region");
+			    }
+		    });
     if (region != null) {
         // region can be of form country, country_variant, or _variant
         int i = region.indexOf('_');
@@ -701,9 +708,17 @@
         }
     } else {
         country = (String) AccessController.doPrivileged(
-                        new GetPropertyAction("user.country", ""));
+			new PrivilegedAction() {
+				public Object run() {
+					return System.getProperty("user.country", "");
+				}
+			});
         variant = (String) AccessController.doPrivileged(
-                        new GetPropertyAction("user.variant", ""));
+			new PrivilegedAction() {
+				public Object run() {
+					return System.getProperty("user.variant", "");
+				}
+			});
     }
     changeLocale(new Locale(language, country, variant));
     COConfigurationManager.removeParameter("locale");
