--- driver.c.old	2012-05-21 09:23:33.979381345 +0000
+++ driver.c	2012-05-21 09:23:44.479210439 +0000
@@ -23,7 +23,6 @@
 #include <asm/io.h>
 #include <asm/irq.h>
 #include <asm/atomic.h>
-#include <asm/system.h>
 #include <linux/version.h>
 #include <linux/ioport.h>
 #include <linux/sched.h>
--- fcpcmcia_cs.c.old	2012-05-21 09:38:50.941025365 +0000
+++ fcpcmcia_cs.c	2012-05-21 09:39:00.300870334 +0000
@@ -24,7 +24,6 @@
 #include <linux/serial.h>
 #include <linux/major.h>
 #include <asm/io.h>
-#include <asm/system.h>
 
 #include <pcmcia/cistpl.h>
 #include <pcmcia/ciscode.h>
