From 1e48d0a6404c6660bbf3f456e5ec48bed8fe7b80 Mon Sep 17 00:00:00 2001
From: Tim Waugh <twaugh@redhat.com>
Date: Tue, 23 Oct 2012 13:55:12 +0000
Subject: Fixed typo in systemd config file.

---
diff --git a/udev/configure-printer@.service.in b/udev/configure-printer@.service.in
index 8a20967..5898c79 100644
--- a/udev/configure-printer@.service.in
+++ b/udev/configure-printer@.service.in
@@ -4,4 +4,4 @@ Requires=cups.socket
 After=cups.socket
 
 [Service]
-ExecStart=@udevhelperdir@/configure-printer add "%i"
+ExecStart=@udevhelperdir@/udev-configure-printer add "%i"
--
cgit v0.9.1
