--- system-config-printer-1.3.5/cupshelpers/cupshelpers.py	2011-08-01 10:47:11.614519332 +0100
+++ system-config-printer-1.3.5/cupshelpers/cupshelpers.py	2011-08-02 15:45:33.807623959 +0100
@@ -21,7 +21,6 @@
 
 import cups, pprint, os, tempfile, re, string
 import locale
-import packagekit.client, packagekit.enums
 from . import _debugprint
 
 class Printer:
@@ -829,24 +828,7 @@
     missing executables
     """
     executables = missingExecutables(ppd)
-    packages = []
-    if executables:
-        unresolved_executables = []
-        client = packagekit.client.PackageKitClient ()
-        for executable in executables:
-            if not executable.startswith ("/"):
-                executable = "/usr/bin/" + executable
-
-            result = client.search_file ([executable],
-                                         packagekit.enums.FILTER_NOT_INSTALLED)
-            if result:
-                packages.extend (set (map (lambda x: x.name, result)))
-            else:
-                unresolved_executables.append (executable)
-
-        executables = unresolved_executables
-        
-    return (packages, executables)
+    return ([], executables)
 
 def _main():
     c = cups.Connection()
