Index: /trunk/edje/src/bin/edje_cc_out.c
===================================================================
--- /trunk/edje/src/bin/edje_cc_out.c	(revision 76448)
+++ /trunk/edje/src/bin/edje_cc_out.c	(revision 76449)
@@ -1882,25 +1882,30 @@
              for (i = 0 ; i < pc->programs.fnmatch_count ; i++)
                {
-                  if (!strcmp(pl->u.ep->name, pc->programs.fnmatch[i]->name))
+                  if (pl->u.ep->name && pc->programs.fnmatch[i]->name &&
+                      !strcmp(pl->u.ep->name, pc->programs.fnmatch[i]->name))
                     data_queue_anonymous_lookup(pc, pc->programs.fnmatch[i], dest);
                }
              for (i = 0 ; i < pc->programs.strcmp_count ; i++)
                {
-                  if (!strcmp(pl->u.ep->name, pc->programs.strcmp[i]->name))
+                  if (pl->u.ep->name && pc->programs.strcmp[i]->name &&
+                      !strcmp(pl->u.ep->name, pc->programs.strcmp[i]->name))
                     data_queue_anonymous_lookup(pc, pc->programs.strcmp[i], dest);
                }
              for (i = 0 ; i < pc->programs.strncmp_count ; i++)
                {
-                  if (!strcmp(pl->u.ep->name, pc->programs.strncmp[i]->name))
+                  if (pl->u.ep->name && pc->programs.strncmp[i]->name &&
+                      !strcmp(pl->u.ep->name, pc->programs.strncmp[i]->name))
                     data_queue_anonymous_lookup(pc, pc->programs.strncmp[i], dest);
                }
              for (i = 0 ; i < pc->programs.strrncmp_count ; i++)
                {
-                  if (!strcmp(pl->u.ep->name, pc->programs.strrncmp[i]->name))
+                  if (pl->u.ep->name && pc->programs.strrncmp[i]->name &&
+                      !strcmp(pl->u.ep->name, pc->programs.strrncmp[i]->name))
                     data_queue_anonymous_lookup(pc, pc->programs.strrncmp[i], dest);
                }
              for (i = 0 ; i < pc->programs.nocmp_count ; i++)
                {
-                  if (!strcmp(pl->u.ep->name, pc->programs.nocmp[i]->name))
+                  if (pl->u.ep->name && pc->programs.nocmp[i]->name &&
+                      !strcmp(pl->u.ep->name, pc->programs.nocmp[i]->name))
                     data_queue_anonymous_lookup(pc, pc->programs.nocmp[i], dest);
                }
@@ -2450,4 +2455,5 @@
 {
    Code_Lookup *cl;
+
    cl = mem_alloc(SZ(Code_Lookup));
    cl->ptr = ptr;
