From cb93a9b513b5814b18594de7a3b67bb225fb9c34 Mon Sep 17 00:00:00 2001
From: Jesse Adkins <jesserayadkins@gmail.com>
Date: Tue, 28 Sep 2010 20:29:51 +0000
Subject: Purge cvs tags.

Signed-off-by: Jesse Adkins <jesserayadkins@gmail.com>
Signed-off-by: Alan Coopersmith <alan.coopersmith@oracle.com>
---
diff --git a/man/nv.man b/man/nv.man
index 228f1da..c679803 100644
--- a/man/nv.man
+++ b/man/nv.man
@@ -1,4 +1,3 @@
-.\" $XFree86: xc/programs/Xserver/hw/xfree86/drivers/nv/nv.man,v 1.30 2006/06/16 00:19:32 mvojkovi Exp $ 
 .\" shorthand for double quote that works everywhere.
 .ds q \N'34'
 .TH NV __drivermansuffix__ __vendorversion__
--
cgit v0.9.0.2-2-gbebe
From d7727e5307c5c3fd0dbc51394893385cb3b01101 Mon Sep 17 00:00:00 2001
From: Alan Coopersmith <alan.coopersmith@oracle.com>
Date: Sat, 30 Oct 2010 16:39:39 +0000
Subject: Sun's copyrights now belong to Oracle

Signed-off-by: Alan Coopersmith <alan.coopersmith@oracle.com>
---
diff --git a/man/Makefile.am b/man/Makefile.am
index b3688ce..e39d1fc 100644
--- a/man/Makefile.am
+++ b/man/Makefile.am
@@ -1,5 +1,5 @@
 #
-# Copyright 2005 Sun Microsystems, Inc.  All rights reserved.
+# Copyright (c) 2010, Oracle and/or its affiliates. All rights reserved.
 #
 # Permission is hereby granted, free of charge, to any person obtaining a
 # copy of this software and associated documentation files (the "Software"),
@@ -19,7 +19,7 @@
 # LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
 # FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
 # DEALINGS IN THE SOFTWARE.
-# 
+#
 
 drivermandir = $(DRIVER_MAN_DIR)
 
--
cgit v0.9.0.2-2-gbebe
From 95108089e22d33723488baba22467f2f3c4f1bf6 Mon Sep 17 00:00:00 2001
From: Alan Coopersmith <alan.coopersmith@oracle.com>
Date: Sat, 30 Oct 2010 16:53:33 +0000
Subject: Correct copyright date in previous commit

Oops, pasted in the new template and forgot to restore the original date

Signed-off-by: Alan Coopersmith <alan.coopersmith@oracle.com>
---
diff --git a/man/Makefile.am b/man/Makefile.am
index e39d1fc..1ea26b3 100644
--- a/man/Makefile.am
+++ b/man/Makefile.am
@@ -1,5 +1,5 @@
 #
-# Copyright (c) 2010, Oracle and/or its affiliates. All rights reserved.
+# Copyright (c) 2005, Oracle and/or its affiliates. All rights reserved.
 #
 # Permission is hereby granted, free of charge, to any person obtaining a
 # copy of this software and associated documentation files (the "Software"),
--
cgit v0.9.0.2-2-gbebe
From 2b17d1ac579ee426f631f4c62c9f0cf617a06e51 Mon Sep 17 00:00:00 2001
From: Cyril Brulebois <kibi@debian.org>
Date: Wed, 02 Mar 2011 01:09:10 +0000
Subject: Fix compiler warning.

Get rid of this with CFLAGS="-Wall -Werror":
|   CC     g80_display.lo
| cc1: warnings being treated as errors
| g80_display.c: In function ‘G80CrtcSetPClk’:
| g80_display.c:216: error: unused variable ‘i’

Signed-off-by: Cyril Brulebois <kibi@debian.org>
Signed-off-by: Aaron Plattner <aplattner@nvidia.com>
Reviewed-by: Aaron Plattner <aplattner@nvidia.com>
Tested-by: Aaron Plattner <aplattner@nvidia.com>
---
diff --git a/src/g80_display.c b/src/g80_display.c
index ae946a8..84e731c 100644
--- a/src/g80_display.c
+++ b/src/g80_display.c
@@ -213,7 +213,7 @@ G80CrtcSetPClk(xf86CrtcPtr crtc)
     if(pNv->architecture <= 0xa0 ||
        pNv->architecture == 0xaa ||
        pNv->architecture == 0xac) {
-        int lo_n, lo_m, hi_n, hi_m, p, i;
+        int lo_n, lo_m, hi_n, hi_m, p;
         CARD32 lo = pNv->reg[(0x00614104+headOff)/4];
         CARD32 hi = pNv->reg[(0x00614108+headOff)/4];
 
--
cgit v0.9.0.2-2-gbebe
From 4b03459a835ee43d940054cbdaefa7c330bff385 Mon Sep 17 00:00:00 2001
From: Nicolas Kaiser <nikai@nikai.net>
Date: Mon, 25 Apr 2011 11:26:33 +0000
Subject: remove duplicated includes

Remove duplicated includes of guarded headers.

Signed-off-by: Nicolas Kaiser <nikai@nikai.net>
Reviewed-by: Aaron Plattner <aplattner@nvidia.com>
Signed-off-by: Aaron Plattner <aplattner@nvidia.com>
---
diff --git a/src/g80_type.h b/src/g80_type.h
index 9bb07a4..270e383 100644
--- a/src/g80_type.h
+++ b/src/g80_type.h
@@ -5,7 +5,6 @@
 #include <xf86Cursor.h>
 #include <xf86DDC.h>
 #include <xf86Crtc.h>
-#include <xf86int10.h>
 
 #define G80_NUM_I2C_PORTS 10
 
diff --git a/src/riva_include.h b/src/riva_include.h
index 04c35a6..16fd40d 100644
--- a/src/riva_include.h
+++ b/src/riva_include.h
@@ -50,7 +50,6 @@
 #include "vgaHW.h"
 
 #include "xf86Cursor.h"
-#include "xf86DDC.h"
 
 #include "region.h"
 
--
cgit v0.9.0.2-2-gbebe
From 5b7f07f8171dd5ca4f13d7da25d7d91140b036e1 Mon Sep 17 00:00:00 2001
From: Adam Jackson <ajax@redhat.com>
Date: Mon, 19 Dec 2011 21:55:30 +0000
Subject: Make failure to XAA non-fatal

Fall back to NoAccel on G80 since there's no shadowfb support there,
otherwise fall to shadowfb.

Signed-off-by: Adam Jackson <ajax@redhat.com>
---
diff --git a/src/g80_driver.c b/src/g80_driver.c
index 551abe3..35e5d7e 100644
--- a/src/g80_driver.c
+++ b/src/g80_driver.c
@@ -429,10 +429,10 @@ G80PreInit(ScrnInfoPtr pScrn, int flags)
     if(!pNv->NoAccel) {
         switch(pNv->AccelMethod) {
         case XAA:
-            if(!xf86LoadSubModule(pScrn, "xaa")) goto fail;
+            if(!xf86LoadSubModule(pScrn, "xaa")) pNv->NoAccel = 1;
             break;
         case EXA:
-            if(!xf86LoadSubModule(pScrn, "exa")) goto fail;
+            if(!xf86LoadSubModule(pScrn, "exa")) pNv->NoAccel = 1;
             break;
         }
     }
diff --git a/src/nv_driver.c b/src/nv_driver.c
index 3031f8b..195299a 100644
--- a/src/nv_driver.c
+++ b/src/nv_driver.c
@@ -2048,9 +2048,9 @@ NVPreInit(ScrnInfoPtr pScrn, int flags)
     /* Load XAA if needed */
     if (!pNv->NoAccel) {
 	if (!xf86LoadSubModule(pScrn, "xaa")) {
-	    xf86FreeInt10(pNv->pInt);
-	    NVFreeRec(pScrn);
-	    return FALSE;
+	    xf86DrvMsg(pScrn->scrnIndex, X_INFO, "Falling back to shadwwfb\n");
+	    pNv->NoAccel = 1;
+	    pNv->ShadowFB = 1;
 	}
     }
 
diff --git a/src/riva_driver.c b/src/riva_driver.c
index e33a7f3..f0fcab4 100644
--- a/src/riva_driver.c
+++ b/src/riva_driver.c
@@ -769,9 +769,9 @@ RivaPreInit(ScrnInfoPtr pScrn, int flags)
     /* Load XAA if needed */
     if (!pRiva->NoAccel) {
 	if (!xf86LoadSubModule(pScrn, "xaa")) {
-	    xf86FreeInt10(pRiva->pInt);
-	    RivaFreeRec(pScrn);
-	    return FALSE;
+	    xf86DrvMsg(pScrn->scrnIndex, X_INFO, "Falling back to shadowfb\n");
+	    pRiva->NoAccel = 1;
+	    pRiva->ShadowFB = 1;
 	}
     }
 
--
cgit v0.9.0.2-2-gbebe
From b3d61828837661c9b383378a7d6006752659108b Mon Sep 17 00:00:00 2001
From: Adam Jackson <ajax@redhat.com>
Date: Mon, 19 Dec 2011 21:57:40 +0000
Subject: Fix for new vgahw ABI

Signed-off-by: Adam Jackson <ajax@redhat.com>
---
diff --git a/src/nv_driver.c b/src/nv_driver.c
index 195299a..527ae5f 100644
--- a/src/nv_driver.c
+++ b/src/nv_driver.c
@@ -1543,6 +1543,7 @@ NVPreInit(ScrnInfoPtr pScrn, int flags)
 	xf86FreeInt10(pNv->pInt);
 	return FALSE;
     }
+    vgaHWSetStdFuncs(VGAHWPTR(pScrn));
     
     /* We use a programmable clock */
     pScrn->progClock = TRUE;
diff --git a/src/riva_driver.c b/src/riva_driver.c
index f0fcab4..b52f476 100644
--- a/src/riva_driver.c
+++ b/src/riva_driver.c
@@ -484,6 +484,7 @@ RivaPreInit(ScrnInfoPtr pScrn, int flags)
 	xf86FreeInt10(pRiva->pInt);
 	return FALSE;
     }
+    vgaHWSetStdFuncs(VGAHWPTR(pScrn));
     
     /* We use a programmable clock */
     pScrn->progClock = TRUE;
--
cgit v0.9.0.2-2-gbebe
