diff -ru gst-plugins-good-0.10.17/configure gst-plugins-good-0.10.17.png14/configure
--- gst-plugins-good-0.10.17/configure	2009-11-16 23:40:59.000000000 +0100
+++ gst-plugins-good-0.10.17.png14/configure	2010-01-17 14:18:30.000000000 +0100
@@ -30924,7 +30924,7 @@
   HAVE_LIBPNG=no
 
 
-  which="libpng12"
+  which="libpng14"
     required="no"
 
 
diff -ru gst-plugins-good-0.10.17/configure.ac gst-plugins-good-0.10.17.png14/configure.ac
--- gst-plugins-good-0.10.17/configure.ac	2009-11-16 23:38:14.000000000 +0100
+++ gst-plugins-good-0.10.17.png14/configure.ac	2010-01-17 14:18:22.000000000 +0100
@@ -749,7 +749,7 @@
 dnl *** libpng ***
 translit(dnm, m, l) AM_CONDITIONAL(USE_LIBPNG, true)
 AG_GST_CHECK_FEATURE(LIBPNG, [Portable Network Graphics library], png, [
-  AG_GST_PKG_CHECK_MODULES(LIBPNG, libpng12)
+  AG_GST_PKG_CHECK_MODULES(LIBPNG, libpng14)
 ])
 
 dnl *** pulseaudio ***
