diff -Naur vips-7.13.0-orig/libsrcCC/VDisplay.cc vips-7.13.0/libsrcCC/VDisplay.cc
--- vips-7.13.0-orig/libsrcCC/VDisplay.cc	2008-04-13 02:47:27.000000000 -0400
+++ vips-7.13.0/libsrcCC/VDisplay.cc	2008-04-13 02:48:01.000000000 -0400
@@ -32,6 +32,7 @@
 #include <vips/intl.h>
 
 #include <cstdlib>
+#include <cstring>
 
 #include <vips/vips.h>
 #include <vips/vipscpp.h>
diff -Naur vips-7.13.0-orig/libsrcCC/VImage.cc vips-7.13.0/libsrcCC/VImage.cc
--- vips-7.13.0-orig/libsrcCC/VImage.cc	2008-04-13 02:47:27.000000000 -0400
+++ vips-7.13.0/libsrcCC/VImage.cc	2008-04-13 02:48:01.000000000 -0400
@@ -33,6 +33,7 @@
 
 #include <cstdlib>
 #include <cstdio>
+#include <cstring>
 
 #include <vips/vips.h>
 #include <vips/vipscpp.h>
