--- qt-x11-opensource-src-4.5.0/tools/linguist/lrelease/main.cpp.orig	2009-03-24 13:42:02.000000000 +0100
+++ qt-x11-opensource-src-4.5.0/tools/linguist/lrelease/main.cpp	2009-03-24 13:47:21.000000000 +0100
@@ -102,7 +102,7 @@
                 if (!msg.comment().isEmpty())
                     qWarning("\n* Comment: %s", qPrintable(msg.comment()));
             }
-            ok = false;
+            //ok = false;
         }
     }
     return ok;
