diff -uBbr qt-x11-free-3.3.5-old/src/widgets/qlistview.cpp
qt-x11-free-3.3.5/src/widgets/qlistview.cpp
--- qt-x11-free-3.3.5-old/src/widgets/qlistview.cpp 2005-09-02
05:43:15.000000000 -0700
+++ qt-x11-free-3.3.5/src/widgets/qlistview.cpp 2005-11-04
04:07:40.000000000 -0800
@@ -3231,6 +3231,7 @@
     d->focusItem = 0;
     d->selectAnchor = 0;
     d->pressedItem = 0;
+    d->highlighted = 0;

     // if it's down its downness makes no sense, so undown it
     d->buttonDown = FALSE; 
