--- sapi/apache2handler/config.m4.orig	2015-12-16 11:41:55.000000000 +0100
+++ sapi/apache2handler/config.m4	2016-01-01 18:56:23.530338559 +0100
@@ -122,7 +122,6 @@
       PHP_BUILD_THREAD_SAFE
     fi
   else
-    APACHE_THREADED_MPM=`$APXS_HTTPD -V | grep 'threaded:.*yes'`
     if test -n "$APACHE_THREADED_MPM"; then
       PHP_BUILD_THREAD_SAFE
     fi
--- configure.orig	2015-12-16 11:41:54.000000000 +0100
+++ configure	2016-01-01 18:56:23.793665417 +0100
@@ -6875,7 +6875,6 @@
 
     fi
   else
-    APACHE_THREADED_MPM=`$APXS_HTTPD -V | grep 'threaded:.*yes'`
     if test -n "$APACHE_THREADED_MPM"; then
 
   enable_maintainer_zts=yes
