From 2046eaeaaabbc58326066ee37f53cc97ebf679bb Mon Sep 17 00:00:00 2001
From: Petr Mladek <pmladek@suse.cz>
Date: Mon, 18 Jul 2011 19:47:39 +0200
Subject: [PATCH 4/4] distro-install-desktop-integration: put $PREFIXDIR/bin/soffice into file list

---
 bin/distro-install-desktop-integration |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/bin/distro-install-desktop-integration b/bin/distro-install-desktop-integration
index 9e3b499..77ccf3a 100755
--- a/bin/distro-install-desktop-integration
+++ b/bin/distro-install-desktop-integration
@@ -161,6 +161,7 @@ add_wrapper unopkg         unopkg  ""            "unopkg"      ""
 #       For example, the Novell package mark this symlink as %ghost
 #	and update it in %post and %postun
 ln -sf $INSTALLDIR/program/soffice $DESTDIR$PREFIXDIR/bin/soffice
+test -f $DESTDIR/gid_Module_Root_Brand && echo "$PREFIXDIR/bin/soffice" >>$DESTDIR/gid_Module_Root_Brand
 
 # create bash completion
 mkdir -p $DESTDIR/etc/bash_completion.d
-- 
1.7.3.4

