diff -r 4ab9e11cb892 connectivity/qa/makefile.mk
--- a/connectivity/qa/makefile.mk	Thu Mar 03 15:22:23 2011 +0100
+++ b/connectivity/qa/makefile.mk	Fri Mar 04 09:51:48 2011 +0100
@@ -35,9 +35,16 @@
 
 #----- compile .java files -----------------------------------------
 
-JARFILES        = ridl.jar unoil.jar jurt.jar juh.jar java_uno.jar OOoRunner.jar hsqldb.jar
 JAVAFILES       :=  $(shell @$(FIND) complex -name "*.java")
 
+JARFILES        = ridl.jar unoil.jar jurt.jar juh.jar java_uno.jar OOoRunner.jar
+
+.IF "$(SYSTEM_HSQLDB)" == "YES"
+EXTRAJARFILES = $(HSQLDB_JAR)
+.ELSE
+JARFILES += hsqldb.jar
+.ENDIF
+
 #----- make a jar from compiled files ------------------------------
 
 JARCLASSDIRS = $(PACKAGE)

