From ea898caa151fb8e94b6bdb00b1008b6da98fddad Mon Sep 17 00:00:00 2001
From: Katarina Machalkova <kmachalkova@suse.cz>
Date: Thu, 07 Apr 2011 11:11:00 +0000
Subject: Removed duplicate code block (mis-merge, prolly)

This fixes build with --enable-kde4
---
diff --git a/vcl/unx/kde4/KDEXLib.cxx b/vcl/unx/kde4/KDEXLib.cxx
index 7757014..ca435c8 100644
--- a/vcl/unx/kde4/KDEXLib.cxx
+++ b/vcl/unx/kde4/KDEXLib.cxx
@@ -219,16 +219,6 @@ static bool qt_event_filter( void* m )
     return false;
 }
 
-static bool ( *old_qt_event_filter )( void* );
-static bool qt_event_filter( void* m )
-{
-    if( old_qt_event_filter != NULL && old_qt_event_filter( m ))
-        return true;
-    if( SalKDEDisplay::self() && SalKDEDisplay::self()->checkDirectInputEvent( static_cast< XEvent* >( m )))
-        return true;
-    return false;
-}
-
 void KDEXLib::setupEventLoop()
 {
     old_qt_event_filter = QAbstractEventDispatcher::instance()->setEventFilter( qt_event_filter );
--
cgit v0.8.3-6-g21f6
