From 5dbb3b7b920045156e998ea2f0a93a870b93f09c Mon Sep 17 00:00:00 2001
From: Katarina Machalkova <kmachalkova@suse.cz>
Date: Thu, 07 Apr 2011 11:28:50 +0000
Subject: Fixed more build breakages with --enable-kde4

---
diff --git a/vcl/source/gdi/gdimtf.cxx b/vcl/source/gdi/gdimtf.cxx
index 4809b1f..5c787bb 100644
--- a/vcl/source/gdi/gdimtf.cxx
+++ b/vcl/source/gdi/gdimtf.cxx
@@ -439,8 +439,6 @@ void GDIMetaFile::Play( GDIMetaFile& rMtf, sal_uLong nPos )
 
         rMtf.UseCanvas( rMtf.GetUseCanvas() || bUseCanvas );
 
-        rMtf.UseCanvas( rMtf.GetUseCanvas() || bUseCanvas );
-
         if( nPos > nObjCount )
             nPos = nObjCount;
 
diff --git a/vcl/unx/kde4/KDESalGraphics.cxx b/vcl/unx/kde4/KDESalGraphics.cxx
index 6cf466d..3b9a13c 100644
--- a/vcl/unx/kde4/KDESalGraphics.cxx
+++ b/vcl/unx/kde4/KDESalGraphics.cxx
@@ -433,8 +433,6 @@ sal_Bool KDESalGraphics::drawNativeControl( ControlType type, ControlPart part,
         const int width = kapp->style()->pixelMetric(QStyle::PM_ToolBarHandleExtent);
         QRect rect( 0, 0, width, widgetRect.height());
         clipRegion = new QRegion( widgetRect.x(), widgetRect.y(), width, widgetRect.height());
-        XRectangle xRect = { widgetRect.x(), widgetRect.y(), width, widgetRect.height() };
-        XUnionRectWithRegion( &xRect, pTempClipRegion, pTempClipRegion );
         
         QStyleOption option;
         option.state = QStyle::State_Horizontal;
--
cgit v0.8.3-6-g21f6
