--- build/solenv/bin/ooinstall	2011-04-21 21:06:44.000000000 +0200
+++ build/solenv/bin/ooinstall	2011-05-03 16:59:30.501750167 +0200
@@ -29,8 +29,10 @@
 	$help = 1;
     } else {
 	# Cwd::realpath does not work if the path does not exist
-	mkpath($arg) unless -d $arg;
-	$path = Cwd::realpath( $arg );
+	mkpath($ENV{DESTDIR} . $arg) unless -d $ENV{DESTDIR} . $arg;
+	$path = Cwd::realpath( $ENV{DESTDIR} . $arg );
+	my $destdir = Cwd::realpath( $ENV{DESTDIR} );
+	$path =~ s|$destdir||;
     }
 }
 
