--- trunk/plugins/profiler/gprof-call-graph-view.c	2007/09/10 06:43:02	3114
+++ trunk/plugins/profiler/gprof-call-graph-view.c	2007/09/23 11:03:16	3153
@@ -22,6 +22,7 @@
  */
 
 #include "gprof-call-graph-view.h"
+#include <glib/gi18n-lib.h>
 
 struct _GProfCallGraphViewPriv
 {
--- trunk/plugins/profiler/gprof-flat-profile-view.c	2007/09/10 06:43:02	3114
+++ trunk/plugins/profiler/gprof-flat-profile-view.c	2007/09/23 11:03:16	3153
@@ -22,6 +22,7 @@
  */
 
 #include "gprof-flat-profile-view.h"
+#include <glib/gi18n-lib.h>
 
 struct _GProfFlatProfileViewPriv
 {
--- trunk/plugins/profiler/gprof-function-call-tree-view.c	2007/09/10 06:43:02	3114
+++ trunk/plugins/profiler/gprof-function-call-tree-view.c	2007/09/23 11:03:16	3153
@@ -22,6 +22,7 @@
  */
 
 #include "gprof-function-call-tree-view.h"
+#include <glib/gi18n-lib.h>
 
 struct _GProfFunctionCallTreeViewPriv
 {
