From 61d9d0d58d53214bdedd2220fb3522c5440a6231 Mon Sep 17 00:00:00 2001
From: Igor <f2404@yandex.ru>
Date: Thu, 15 Mar 2018 09:04:04 -0400
Subject: Fix handling of tabs reordering

Fix bug #14278
---
 terminal/terminal-window.c | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/terminal/terminal-window.c b/terminal/terminal-window.c
index eca35ba..68daa03 100644
--- a/terminal/terminal-window.c
+++ b/terminal/terminal-window.c
@@ -1105,10 +1105,8 @@ terminal_window_notebook_page_reordered (GtkNotebook     *notebook,
                                          guint            page_num,
                                          TerminalWindow  *window)
 {
-
-  /* update actions in the window */
-  terminal_window_update_actions (window);
-
+  /* Regenerate the "Go" menu */
+  terminal_window_rebuild_tabs_menu (window);
 }
 
 
-- 
cgit v1.1

