--- src/Makefile.in	2006-02-05 21:01:03.000000000 +0100
+++ src/Makefile.in.new	2006-02-05 21:03:19.000000000 +0100
@@ -134,10 +134,10 @@
 	$(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS)
 LTCOMPILE = $(LIBTOOL) --tag=CC --mode=compile $(CC) $(DEFS) \
 	$(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) \
-	$(AM_CFLAGS) $(CFLAGS)
+	$(AM_CFLAGS) $(CFLAGS) -fPIC
 CCLD = $(CC)
 LINK = $(LIBTOOL) --tag=CC --mode=link $(CCLD) $(AM_CFLAGS) $(CFLAGS) \
-	$(AM_LDFLAGS) $(LDFLAGS) -o $@
+	$(AM_LDFLAGS) $(LDFLAGS) -fPIC -o $@
 SOURCES = $(libm17n_X_la_SOURCES) $(libm17n_core_la_SOURCES) \
 	$(libm17n_gd_la_SOURCES) $(libm17n_gui_la_SOURCES) \
 	$(libm17n_la_SOURCES) $(linkcore_SOURCES) $(linkgui_SOURCES) \
