diff -u -Nwr ca-certificates/sbin/update-ca-certificates ca-certificates.confd/sbin/update-ca-certificates
--- ca-certificates/sbin/update-ca-certificates	2014-03-13 13:43:00.000000000 +0100
+++ ca-certificates.confd/sbin/update-ca-certificates	2014-08-24 13:36:19.264068119 +0200
@@ -37,7 +37,7 @@
   shift
 done
 
-CERTSCONF=/etc/ca-certificates.conf
+CERTSCONFS=/etc/ca-certificates/conf.d/*.conf
 CERTSDIR=/usr/share/ca-certificates
 LOCALCERTSDIR=/usr/local/share/ca-certificates
 CERTBUNDLE=ca-certificates.crt
@@ -101,6 +101,7 @@
 
 echo -n "Updating certificates in $ETCCERTSDIR... "
 
+for CERTSCONF in $CERTSCONFS; do
 # Handle certificates that should be removed.  This is an explicit act
 # by prefixing lines in the configuration files with exclamation marks (!).
 sed -n -e '/^$/d' -e 's/^!//p' $CERTSCONF | while read crt
@@ -117,6 +118,7 @@
   fi
   add "$CERTSDIR/$crt"
 done
+done
 
 # Now process certificate authorities installed by the local system
 # administrator.
