--- a/error.cc	2007-07-06 12:04:24.000000000 -0700
+++ b/error.cc	2009-05-02 12:56:26.000000000 -0700
@@ -25,6 +25,7 @@
 
 #include <errno.h>
 #include <string.h>
+#include <stdio.h>
 
 // C++ includes
 
--- a/smiltime.cc	2008-11-25 19:41:39.000000000 -0800
+++ b/smiltime.cc	2009-05-02 12:57:51.000000000 -0700
@@ -30,6 +30,7 @@
 
 #include <stdlib.h>
 #include <math.h>
+#include <stdio.h>
 
 #include "stringutils.h"
 
