From 502af4fcca53c81025446bb73fa29570abb34feb Mon Sep 17 00:00:00 2001
From: Alec Leamas <leamas.alec@gmail.com>
Date: Thu, 11 Dec 2014 10:28:47 +0100
Subject: [PATCH 05/13] lirc_options.conf: Add effective-user to template.

---
 lirc_options.conf | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lirc_options.conf b/lirc_options.conf
index 9dc672a..9346de8 100644
--- a/lirc_options.conf
+++ b/lirc_options.conf
@@ -12,6 +12,7 @@ plugindir       = /usr/lib/lirc/plugins
 permission      = 666
 allow-simulate  = No
 repeat-max      = 600
+#effective-user =
 #listen         = [address:]port
 #connect        = host[:port]
 #debug          = 6
-- 
2.2.2

