--- a/configure	2011-08-06 12:50:26.000000000 +0200
+++ b/configure	2011-08-06 13:01:56.000000000 +0200
@@ -15732,7 +15732,7 @@
 if ${ac_cv_search_dn_expand+:} false; then :
 
 else
-  ac_cv_search_dn_expand=no
+  ac_cv_search_dn_expand='-lresolv'
 fi
 rm conftest.$ac_ext
 LIBS=$ac_func_search_save_LIBS
@@ -15790,14 +15790,14 @@
  ac_ec_dns=yes
 
 else
-   if false; then
+   if true; then
   HAVE_DN_EXPAND_TRUE=
   HAVE_DN_EXPAND_FALSE='#'
 else
   HAVE_DN_EXPAND_TRUE='#'
   HAVE_DN_EXPAND_FALSE=
 fi
- ac_ec_dns=no
+ ac_ec_dns=yes
 fi
 
 
--- a/src/ec_log.c	2004-09-30 18:01:45.000000000 +0200
+++ b/src/ec_log.c	2011-08-06 13:04:30.000000000 +0200
@@ -190,7 +190,7 @@
       if (fd->cfd == NULL)
          SEMIFATAL_ERROR("%s", gzerror(fd->cfd, &zerr));
    } else {
-      fd->fd = open(filename, O_CREAT | O_TRUNC | O_RDWR | O_BINARY);
+      fd->fd = open(filename, O_CREAT | O_TRUNC | O_RDWR | O_BINARY, 0666);
       if (fd->fd == -1)
          SEMIFATAL_ERROR("Can't create %s: %s", filename, strerror(errno));
    }
