diff -Nur NetworkManager-0.8.2.orig/src/nm-policy.c NetworkManager-0.8.2/src/nm-policy.c
--- NetworkManager-0.8.2.orig/src/nm-policy.c	2010-11-03 12:58:42.000000000 -0700
+++ NetworkManager-0.8.2/src/nm-policy.c	2010-11-23 06:22:32.436707081 -0800
@@ -342,6 +342,7 @@
 		return;
 	}
 
+#if 0
 	/* Try automatically determined hostname from the best device's IP config */
 	if (!best4)
 		best4 = get_best_ip4_device (policy->manager, &best_req4);
@@ -396,6 +397,7 @@
 		}
 	}
 
+#endif
 	/* If no automatically-configured hostname, try using the hostname from
 	 * when NM started up.
 	 */
