diff -Nur NetworkManager-0.8.0.997.orig/src/nm-policy.c NetworkManager-0.8.0.997/src/nm-policy.c
--- NetworkManager-0.8.0.997.orig/src/nm-policy.c	2010-05-05 10:38:52.000000000 +0300
+++ NetworkManager-0.8.0.997/src/nm-policy.c	2010-05-10 16:15:02.995948368 +0300
@@ -286,7 +286,7 @@
 		g_free (configured_hostname);
 		return;
 	}
-
+#if 0
 	/* Try automatically determined hostname from the best device's IP config */
 	if (!best)
 		best = get_best_ip4_device (policy->manager, &best_req);
@@ -298,6 +298,7 @@
 		_set_hostname (policy->orig_hostname, "no default device");
 		return;
 	}
+#endif
 
 	/* Grab a hostname out of the device's DHCP4 config */
 	dhcp4_config = nm_device_get_dhcp4_config (best);
