diff -ruN Mesa-7.2.orig/src/mesa/drivers/dri/intel/intel_context.c Mesa-7.2/src/mesa/drivers/dri/intel/intel_context.c
--- Mesa-7.2.orig/src/mesa/drivers/dri/intel/intel_context.c	2008-09-13 20:25:44.000000000 +0000
+++ Mesa-7.2/src/mesa/drivers/dri/intel/intel_context.c	2008-12-21 21:27:45.000000000 +0000
@@ -490,9 +490,6 @@
    if (intel->bufmgr == NULL) {
       if (ttm_disable) {
 	 fprintf(stderr, "TTM buffer manager disabled.  Using classic.\n");
-      } else {
-	 fprintf(stderr, "Failed to initialize TTM buffer manager.  "
-		 "Falling back to classic.\n");
       }
 
       if (intelScreen->tex.size == 0) {
