diff -Nur lilo-22.8.orig/geometry.c lilo-22.8/geometry.c
--- lilo-22.8.orig/geometry.c	2006-12-17 04:46:17.000000000 +0100
+++ lilo-22.8/geometry.c	2009-11-07 03:26:02.815256390 +0100
@@ -901,42 +901,15 @@
 	    DEVICE dev;
 	    struct dm_task *dmt;
 	    void *next = NULL;
-	    char dmdev[PATH_MAX+1];
 	    char buf[PATH_MAX+1];
-	    char *slash;
 	    int result;
 
 	    dev_open(&dev, device, -1);
-	    strncpy(dmdev, dev.name, PATH_MAX);
-	    dmdev[PATH_MAX] = 0;
-	    do {
-		memset(buf, 0, PATH_MAX + 1);
-		if ((result = readlink(dmdev, buf, PATH_MAX)) < 0 && errno != EINVAL)
-		    die("device-mapper: readlink(\"%s\") failed with: %s",buf,
-			strerror(errno));
-		if (result >= 0) {
-		    if (buf[0] != '/' && (slash = strrchr(dmdev, '/')) != NULL)
-			slash++;
-		    else
-			slash = dmdev;
-		    strncpy(slash, buf, PATH_MAX - (slash-dmdev));
-		}
-		if (realpath(dmdev, buf) == NULL)
-		    die("device-mapper: realpath(\"%s\") failed with: %s",dmdev,
-			strerror(errno));
-		strncpy(dmdev, buf, PATH_MAX);
-	    } while (result >= 0);
-	    dmdev[PATH_MAX] = 0;
 
 	    if (!(dmt = dm_task_create(DM_DEVICE_TABLE)))
 		die("device-mapper: dm_task_create(DM_DEVICE_TABLE) failed");
-	    slash = strrchr(dmdev, '/');
-		if (slash)
-	    slash++;
-		else
-	    slash = dmdev;
-	    if (!dm_task_set_name(dmt, slash))
-		die("device-mapper: dm_task_set_name(\"%s\") failed",dmdev);
+	    if (!dm_task_set_name(dmt, dev.name))
+		die("device-mapper: dm_task_set_name(\"%s\") failed",dev.name);
 	    if (!dm_task_run(dmt))
 		die("device-mapper: dm_task_run(DM_DEVICE_TABLE) failed");
 
