diff -Nur udev-103.orig/extras/volume_id/lib/linux_swap.c udev-103/extras/volume_id/lib/linux_swap.c
--- udev-103.orig/extras/volume_id/lib/linux_swap.c	2006-10-20 14:43:35.000000000 +0200
+++ udev-103/extras/volume_id/lib/linux_swap.c	2006-10-29 17:15:19.000000000 +0100
@@ -73,6 +73,11 @@
 				strcpy(id->type_version, "ulsuspend");
 				goto found_label;
 			}
+			if (memcmp(buf, "z", 1) == 0 || memcmp(buf, "Z", 1) == 0) {
+				id->type = "suspend";
+				strcpy(id->type_version, "suspend2");
+				goto found_label;
+			}
 	}
 	return -1;
 
