Index: src/roster_window.py
===================================================================
--- src/roster_window.py	(revision 13002)
+++ src/roster_window.py	(revision 13032)
@@ -2323,5 +2323,5 @@
                 gajim.config.set('roster_x-position', x)
                 gajim.config.set('roster_y-position', y)
-                self.window.hide()
+            self.window.hide()
         elif gajim.config.get('quit_on_roster_x_button'):
             self.on_quit_request()
