diff -Nur gajim-0.13.1.old/src/common/passwords.py gajim-0.13.1/src/common/passwords.py
--- gajim-0.13.1.old/src/common/passwords.py	2009-10-29 12:15:17.000000000 +0200
+++ gajim-0.13.1/src/common/passwords.py	2009-12-07 13:59:42.179454769 +0200
@@ -122,6 +122,8 @@
 		user = gajim.config.get_per('accounts', account_name, 'name')
 		display_name = _('XMPP account %s@%s') % (user, server)
 		attributes1 = dict(server=str(server), user=str(user), protocol='xmpp')
+		if password is None:
+			password = str()
 		try:
 			auth_token = gnomekeyring.item_create_sync(
 				self.keyring, gnomekeyring.ITEM_NETWORK_PASSWORD,
