--- canvas/source/cairo/cairo_canvashelper.hxx
+++ canvas/source/cairo/cairo_canvashelper.hxx
@@ -331,7 +331,8 @@
 					Operation aOperation,
 					::cairo::Cairo* pCairo,
 					const ::com::sun::star::uno::Sequence< ::com::sun::star::rendering::Texture >* pTextures=NULL,
-					SpriteCanvas* pDevice=NULL );
+					SpriteCanvas* pDevice=NULL, 
+					::com::sun::star::rendering::FillRule eFillrule=::com::sun::star::rendering::FillRule_EVEN_ODD);
     void doOperation( Operation aOperation,
 		      ::cairo::Cairo* pCairo,
 		      sal_uInt32 nPolygonIndex=0,
