diff -ruN pmount-0.9.13/src/pmount-hal.c pmount-0.9.13-new/src/pmount-hal.c
--- pmount-0.9.13/src/pmount-hal.c	2006-08-15 22:31:02.000000000 +0200
+++ pmount-0.9.13-new/src/pmount-hal.c	2007-02-05 12:01:09.000000000 +0100
@@ -361,7 +361,6 @@
     /* shut down hal connection */
     libhal_ctx_shutdown( hal_ctx, &error );
     libhal_ctx_free( hal_ctx );
-    dbus_connection_close( dbus_conn );
     dbus_connection_unref( dbus_conn );
 
     /* go */
