diff --git a/core/app/utils/digikam_opencv.h.cmake.in b/core/app/utils/digikam_opencv.h.cmake.in
index 9195aad6e7..b3d6de1f86 100644
--- a/core/app/utils/libopencv.h.cmake.in
+++ b/core/app/utils/libopencv.h.cmake.in
@@ -54,24 +54,36 @@
 #   include <opencv2/legacy/compat.hpp>
 #   include <opencv/cvaux.h>
 #   include <opencv2/imgproc/imgproc.hpp>
-#else
+#elif OPENCV_TEST_VERSION(4,0,0)
 #   include <opencv/cv.h>
 #   include <opencv/cvaux.h>
 #   include <opencv/cxcore.h>
 #   include <opencv2/imgproc.hpp>
+#else
+#   include <opencv2/imgproc.hpp>
+#   include <opencv2/core/core_c.h>
+#   include <opencv2/core/types_c.h>
+#   include <opencv2/imgproc/types_c.h>
 #endif
 
 #if OPENCV_TEST_VERSION(3,0,0)
 #   include <opencv2/core/core.hpp>
 #   include <opencv2/core/internal.hpp>
 #   include <opencv2/contrib/contrib.hpp>
-#else
+#elif OPENCV_TEST_VERSION(4,0,0)
 #   include <opencv2/objdetect.hpp>
 #   include <opencv2/core.hpp>
 #   include <opencv2/core/utility.hpp>
 #   include <opencv2/core/persistence.hpp>
 #   include <opencv2/imgcodecs.hpp>
 #   include <opencv2/imgcodecs/imgcodecs_c.h>
+#else
+#   include <opencv2/objdetect.hpp>
+#   include <opencv2/core.hpp>
+#   include <opencv2/core/utility.hpp>
+#   include <opencv2/core/persistence.hpp>
+#   include <opencv2/imgcodecs.hpp>
+#   include <opencv2/imgcodecs/legacy/constants_c.h>
 #endif
 
 // for old-style code
diff --git a/core/libs/facesengine/detection/opencvfacedetector.cpp b/core/libs/facesengine/detection/opencvfacedetector.cpp
index 2a28276d09..639540a964 100644
--- a/core/libs/facesengine/detection/opencvfacedetector.cpp
+++ b/core/libs/facesengine/detection/opencvfacedetector.cpp
@@ -358,7 +358,7 @@ void OpenCVFaceDetector::updateParameters(const cv::Size& /*scaledSize*/, const
      * unless in we want very high sensitivity at low speed
      */
     if (d->sensitivityVsSpecificity > 0.1 || d->speedVsAccuracy < 0.9)
-        d->primaryParams.flags = CV_HAAR_DO_CANNY_PRUNING;
+        d->primaryParams.flags = cv::CASCADE_DO_CANNY_PRUNING;
     else
         d->primaryParams.flags = 0;
 
