diff --git a/core/dplugins/dimg/imagemagick/dimgimagemagickplugin.cpp b/core/dplugins/dimg/imagemagick/dimgimagemagickplugin.cpp
index 1858b4d27b..20ef01be73 100644
--- a/core/dplugins/dimg/imagemagick/dimgimagemagickplugin.cpp
+++ b/core/dplugins/dimg/imagemagick/dimgimagemagickplugin.cpp
@@ -125,7 +125,7 @@ QMap<QString, QString> DImgImageMagickPlugin::extraAboutData() const
     QString mimes = typeMimes();
 
     QMap<QString, QString> map;
-    ExceptionInfo ex;
+    ExceptionInfo ex = *AcquireExceptionInfo();
     size_t n                  = 0;
     const MagickInfo** inflst = GetMagickInfoList("*", &n, &ex);
 
@@ -219,7 +219,7 @@ int DImgImageMagickPlugin::canRead(const QFileInfo& fileInfo, bool magic) const
 int DImgImageMagickPlugin::canWrite(const QString& format) const
 {
     QStringList formats;
-    ExceptionInfo ex;
+    ExceptionInfo ex = *AcquireExceptionInfo();
     size_t n                  = 0;
     const MagickInfo** inflst = GetMagickInfoList("*", &n, &ex);
 
@@ -266,7 +266,7 @@ DImgLoader* DImgImageMagickPlugin::loader(DImg* const image, const DRawDecoding&
 QStringList DImgImageMagickPlugin::decoderFormats() const
 {
     QStringList formats;
-    ExceptionInfo ex;
+    ExceptionInfo ex = *AcquireExceptionInfo();
     size_t n                  = 0;
     const MagickInfo** inflst = GetMagickInfoList("*", &n, &ex);
 
diff --git a/core/tests/dimg/magickloader.cpp b/core/tests/dimg/magickloader.cpp
index 2526779047..73f2abec0e 100644
--- a/core/tests/dimg/magickloader.cpp
+++ b/core/tests/dimg/magickloader.cpp
@@ -127,7 +127,7 @@ int main(int argc, char** argv)
 {
     MagickCoreGenesis((char*)NULL, MagickFalse);
 
-    ExceptionInfo ex;
+    ExceptionInfo ex = *AcquireExceptionInfo();
     size_t n                  = 0;
     const MagickInfo** inflst = GetMagickInfoList("*", &n, &ex);
 
