diff -Nur guile-1.8.3.orig/libguile/guile.c guile-1.8.3/libguile/guile.c
--- guile-1.8.3.orig/libguile/guile.c	2007-05-09 22:22:03.000000000 +0200
+++ guile-1.8.3/libguile/guile.c	2008-03-04 15:54:19.000000000 +0100
@@ -68,8 +68,8 @@
 {
 #if !defined (__MINGW32__)
   /* libtool automagically inserts this variable into your executable... */
-  extern const lt_dlsymlist lt_preloaded_symbols[];
-  lt_dlpreload_default (lt_preloaded_symbols);
+  /*extern const lt_dlsymlist lt_preloaded_symbols[];
+  lt_dlpreload_default (lt_preloaded_symbols);*/
 #endif
   scm_boot_guile (argc, argv, inner_main, 0);
   return 0; /* never reached */
