--- trunk/libindi/CMakeLists.txt	2010/11/20 11:45:07	397
+++ trunk/libindi/CMakeLists.txt	2010/12/01 08:36:20	412
@@ -164,7 +164,7 @@
 ############# INDI Client Library ################
 ##################################################
 add_library(indiclient STATIC ${indibase_SRCS})
-target_link_libraries(indiclient indi)
+target_link_libraries(indiclient indi pthread)
 install(TARGETS indiclient ARCHIVE DESTINATION lib${LIB_POSTFIX})
 
 #####################################
