http://bugzilla.xfce.org/show_bug.cgi?id=3439
Index: events.c
===================================================================
--- src/events.c	(revision 26059)
+++ src/events.c	(revision 26060)
@@ -1430,7 +1430,7 @@
            stealing prevention is not activated, otherwise we just set the "demands attention"
            flag...
          */
-        if ((ev->value_mask & CWStackMode) && (wc.stack_mode == Above) && (wc.sibling == None))
+        if ((ev->value_mask & CWStackMode) && (wc.stack_mode == Above) && (wc.sibling == None) && !(c->type & WINDOW_TYPE_DONT_FOCUS))
         {
             Client *last_raised;
 
