--- ipp-usb-0.9.11/systemd-udev/ipp-usb.service	2020-08-21 00:52:44.000000000 +0200
+++ ipp-usb-0.9.11/systemd-udev/ipp-usb.service.new	2020-08-22 22:02:38.798950382 +0200
@@ -1,9 +1,12 @@
 [Unit]
 Description=Daemon for IPP over USB printer support
 Documentation=man:ipp-usb(8)
-After=cups.service avahi-daemon.service
+After=org.cups.cupsd.service avahi-daemon.service
 Wants=avahi-daemon.service
 
 [Service]
 Type=oneshot
-ExecStart=/sbin/ipp-usb udev
+ExecStart=/usr/bin/ipp-usb udev
+
+[Install]
+WantedBy=multi-user.target
