diff -Nur webkit-1.7.91.orig/Tools/GNUmakefile.am webkit-1.7.91/Tools/GNUmakefile.am
--- webkit-1.7.91.orig/Tools/GNUmakefile.am	2012-03-08 12:37:34.221447643 +0000
+++ webkit-1.7.91/Tools/GNUmakefile.am	2012-03-08 12:53:28.713523074 +0000
@@ -326,7 +326,7 @@
 	  fi; \
 	fi
 endif
-	$(srcdir)/Tools/gtk/generate-gtkdoc --rebase --virtual-root=$${DESTDIR}
+	$(AM_V_GEN)$(PYTHON) $(srcdir)/Tools/gtk/generate-gtkdoc --rebase --virtual-root=$${DESTDIR}
 
 uninstall-local:
 	@DOC_MODULE_VERSION=`cat ./Documentation/webkitgtk/version.xml`; \
