diff -Naur qtiplot-0.9.8.8.orig/3rdparty/qwtplot3d/include/qwt3d_io_gl2ps.h qtiplot-0.9.8.8/3rdparty/qwtplot3d/include/qwt3d_io_gl2ps.h
--- qtiplot-0.9.8.8.orig/3rdparty/qwtplot3d/include/qwt3d_io_gl2ps.h	2012-01-29 17:42:01.000000000 -0500
+++ qtiplot-0.9.8.8/3rdparty/qwtplot3d/include/qwt3d_io_gl2ps.h	2012-01-29 17:58:30.000000000 -0500
@@ -2,12 +2,7 @@
 #define qwt3d_io_gl2ps_h__2004_05_07_01_16_begin_guarded_code
 
 #include <time.h>
-
-#if QT_VERSION < 0x040000
-#include <qgl.h>
-#else
-#include <QtOpenGL/qgl.h>
-#endif
+#include <QtOpenGL>
 
 #include "qwt3d_types.h"
 #include "qwt3d_io.h"
diff -Naur qtiplot-0.9.8.8.orig/3rdparty/qwtplot3d/include/qwt3d_openglhelper.h qtiplot-0.9.8.8/3rdparty/qwtplot3d/include/qwt3d_openglhelper.h
--- qtiplot-0.9.8.8.orig/3rdparty/qwtplot3d/include/qwt3d_openglhelper.h	2012-01-29 17:42:01.000000000 -0500
+++ qtiplot-0.9.8.8/3rdparty/qwtplot3d/include/qwt3d_openglhelper.h	2012-01-29 17:57:21.000000000 -0500
@@ -2,11 +2,8 @@
 #define __openglhelper_2003_06_06_15_49__
 
 #include "qglobal.h"
-#if QT_VERSION < 0x040000
-#include <qgl.h>
-#else
-#include <QtOpenGL/qgl.h>
-#endif
+#include <QtOpenGL>
+#include <GL/glu.h>
 
 namespace Qwt3D
 {
