diff -Nur clutter-gtk-0.10.8.orig/configure.ac clutter-gtk-0.10.8/configure.ac
--- clutter-gtk-0.10.8.orig/configure.ac	2012-01-21 14:52:17.627228994 +0000
+++ clutter-gtk-0.10.8/configure.ac	2012-01-21 15:04:37.317308303 +0000
@@ -57,6 +57,9 @@
 LT_PREREQ([2.2.6])
 LT_INIT([disable-static])
 
+AC_CHECK_LIBM
+AC_SUBST(LIBM)
+
 m4_define([clutter_req_version], [1.2.0])
 m4_define([gtk_req_version], [2.19.5])
 m4_define([flavour_default], [x11])
diff -Nur clutter-gtk-0.10.8.orig/examples/Makefile.am clutter-gtk-0.10.8/examples/Makefile.am
--- clutter-gtk-0.10.8.orig/examples/Makefile.am	2012-01-21 14:52:17.600561077 +0000
+++ clutter-gtk-0.10.8/examples/Makefile.am	2012-01-21 15:00:04.549770032 +0000
@@ -22,18 +22,18 @@
 
 gtk_clutter_test_SOURCES = gtk-clutter-test.c
 gtk_clutter_test_DEPENDENCIES = $(common_deps)
-gtk_clutter_test_LDADD = $(common_ldadd)
+gtk_clutter_test_LDADD = $(common_ldadd) $(LIBM)
 
 gtk_clutter_events_SOURCES = gtk-clutter-events.c
 gtk_clutter_events_DEPENDENCIES = $(common_deps)
-gtk_clutter_events_LDADD = $(common_ldadd)
+gtk_clutter_events_LDADD = $(common_ldadd) $(LIBM)
 
 gtk_clutter_multistage_SOURCES = gtk-clutter-multistage.c
 gtk_clutter_multistage_DEPENDENCIES = $(common_deps)
-gtk_clutter_multistage_LDADD = $(common_ldadd)
+gtk_clutter_multistage_LDADD = $(common_ldadd) $(LIBM)
 
 gtk_clutter_viewport_SOURCES = gtk-clutter-viewport.c
 gtk_clutter_viewport_DEPENDENCIES = $(common_deps)
-gtk_clutter_viewport_LDADD = $(common_ldadd)
+gtk_clutter_viewport_LDADD = $(common_ldadd) $(LIBM)
 
 EXTRA_DIST = redhand.png
