--- Miro-1.2.7/portable/libtorrent/src/torrent_info.cpp.old	2008-09-21 13:47:00.122974031 +0200
+++ Miro-1.2.7/portable/libtorrent/src/torrent_info.cpp	2008-09-21 13:48:54.812965713 +0200
@@ -352,7 +352,7 @@
 		fs::path tmp = m_name;
 		if (tmp.is_complete()) throw std::runtime_error("torrent contains "
 			"a file with an absolute path: '" + m_name + "'");
-		if (tmp.has_branch_path()) throw std::runtime_error(
+		if (!tmp.branch_path().empty()) throw std::runtime_error(
 			"torrent contains name with directories: '" + m_name + "'");
 	
 		// extract file list
@@ -556,7 +556,7 @@
 	{
 //		TORRENT_ASSERT(file.begin() != file.end());
 
-		if (!file.has_branch_path())
+		if (file.branch_path().empty())
 		{
 			// you have already added at least one file with a
 			// path to the file (branch_path), which means that

--- Miro-1.2.7/portable/libtorrent/include/libtorrent/disk_io_thread.hpp.old	2008-09-21 13:57:05.433330334 +0200
+++ Miro-1.2.7/portable/libtorrent/include/libtorrent/disk_io_thread.hpp	2008-09-21 13:57:29.060304305 +0200
@@ -41,6 +41,7 @@
 #include <boost/bind.hpp>
 #include <boost/pool/pool.hpp>
 #include <boost/noncopyable.hpp>
+#include <boost/thread/condition.hpp>
 #include "libtorrent/config.hpp"
 
 namespace libtorrent

