--- tools.old	2011-05-20 11:35:02.621304102 +0200
+++ tools.c	2011-05-20 11:36:23.022819535 +0200
@@ -110,7 +110,7 @@
 #define	FENCE2_OK(h,m)	(*(unsigned *)(((char *) m)+(h)->size)==FENCE_TAG)
 
 static unsigned		alloc_count	= 0;
-static spinlock_t	track_lock	= SPIN_LOCK_UNLOCKED;
+static DEFINE_SPINLOCK(track_lock);
 
 #if !defined (NDEBUG) && defined (LOG_TIMER)
 static struct timeval	zero_time;
--- tools.old	2011-05-20 11:46:38.706462621 +0200
+++ tools.c	2011-05-20 11:46:27.123401621 +0200
@@ -529,7 +529,7 @@
 		ERROR("Could not allocate lock structure!!!\n");
 		return 0;
 	}
-	tmp->lock = SPIN_LOCK_UNLOCKED;
+        spin_lock_init(&tmp->lock);
 	*plock = tmp;
 	return 1;
 } /* lock_init */
