--- openjdk/jdk/make/common/shared/Defs-utils.gmk.old	2011-08-06 15:46:01.000000000 +0200
+++ openjdk/jdk/make/common/shared/Defs-utils.gmk	2011-08-06 15:59:26.000000000 +0200
@@ -91,10 +91,10 @@
 CHMOD          = $(UTILS_COMMAND_PATH)chmod
 CMP            = $(UTILS_USR_BIN_PATH)cmp
 COMM           = $(UTILS_USR_BIN_PATH)comm
-COMPRESS       = $(UTILS_USR_BIN_PATH)compress
+COMPRESS       = $(UTILS_COMMAND_PATH)compress
 CP             = $(UTILS_COMMAND_PATH)cp
 CPIO           = $(UTILS_COMMAND_PATH)cpio
-CUT            = $(UTILS_USR_BIN_PATH)cut
+CUT            = $(UTILS_COMMAND_PATH)cut
 DATE           = $(UTILS_COMMAND_PATH)date
 DF             = $(UTILS_COMMAND_PATH)df
 DIFF           = $(UTILS_USR_BIN_PATH)diff
@@ -141,7 +141,7 @@
 TAR            = $(UTILS_COMMAND_PATH)tar
 TEST           = $(UTILS_USR_BIN_PATH)test
 TOUCH          = $(UTILS_COMMAND_PATH)touch
-TR             = $(UTILS_USR_BIN_PATH)tr
+TR             = $(UTILS_COMMAND_PATH)tr
 TRUE           = $(UTILS_COMMAND_PATH)true
 UNAME          = $(UTILS_COMMAND_PATH)uname
 UNIQ           = $(UTILS_USR_BIN_PATH)uniq
