--- a/Makefile	2011-06-06 11:10:28.000000000 +0200
+++ b/Makefile	2012-02-27 18:20:11.000000000 +0100
@@ -382,7 +382,7 @@
 
 $(TARGET): $(OBJECTS)
 	@echo "linking binary '$(TARGET)'"
-	@$(CC) $(LDFLAGS) $(OBJECTS) -o $(TARGET) -Wl,--as-needed $(lLIBS)
+	@$(CC) $(LDFLAGS) $(OBJECTS) -o $(TARGET) -lm -ldl -Wl,--as-needed $(lLIBS)
 ifeq ($(STRIP), 1)
 ifneq ($(DEBUG), 1)
 	@echo "stripping binary '$(TARGET)'"
