diff --git a/applets/clock/calendar-window.c b/applets/clock/calendar-window.c
index 932ebda..bfad14e 100644
--- a/applets/clock/calendar-window.c
+++ b/applets/clock/calendar-window.c
@@ -450,6 +450,8 @@ is_appointment (GtkTreeModel *model,
 	gtk_tree_model_get (model, iter, APPOINTMENT_COLUMN_URI, &uri, -1);
 	if (uri)
 		return (g_ascii_strcasecmp (uri, "file")   == 0 ||
+			g_ascii_strcasecmp (uri, "local")   == 0 ||
+			g_ascii_strcasecmp (uri, "mapi")   == 0 ||
 			g_ascii_strcasecmp (uri, "webcal") == 0 ||
 			g_ascii_strcasecmp (uri, "caldav") == 0 ||
 			g_ascii_strcasecmp (uri, "exchange")  == 0 ||
