--- src/base/FileHelper.cpp.orig	2008-03-30 12:13:24.000000000 +0000
+++ src/base/FileHelper.cpp	2008-03-30 12:13:38.000000000 +0000
@@ -32,6 +32,7 @@
 #include <map>
 #include <iostream>
 #include <fstream>
+#include <cstring>
 
 using namespace std;
 
--- src/base/XMLHelper.cpp.orig	2008-03-30 12:14:36.000000000 +0000
+++ src/base/XMLHelper.cpp	2008-03-30 12:14:49.000000000 +0000
@@ -23,6 +23,7 @@
 
 #include <libxml/parserInternals.h>
 #include <iostream>
+#include <cstring>
 
 using namespace std;
 
--- src/base/MemHelper.cpp.orig	2008-03-30 12:15:46.000000000 +0000
+++ src/base/MemHelper.cpp	2008-03-30 12:16:07.000000000 +0000
@@ -37,6 +37,7 @@
 #else
 #include <sstream>
 #endif
+#include <cstdlib>
 
 namespace avg {
 
--- src/graphics/Bitmap.cpp.orig	2008-03-30 12:16:54.000000000 +0000
+++ src/graphics/Bitmap.cpp	2008-03-30 12:17:08.000000000 +0000
@@ -35,6 +35,7 @@
 #include <iostream>
 #include <iomanip>
 #include <stdlib.h>
+#include <cstring>
 
 using namespace Magick;
 using namespace std;
--- src/graphics/Filterflip.cpp.orig	2008-03-30 12:17:44.000000000 +0000
+++ src/graphics/Filterflip.cpp	2008-03-30 12:17:59.000000000 +0000
@@ -20,6 +20,7 @@
 //
 
 #include "Filterflip.h"
+#include <cstring>
 
 namespace avg {
 
--- src/graphics/FilterHighpass.cpp.orig	2008-03-30 12:18:44.000000000 +0000
+++ src/graphics/FilterHighpass.cpp	2008-03-30 12:18:57.000000000 +0000
@@ -26,6 +26,7 @@
 
 #include <iostream>
 #include <sstream>
+#include <cstring>
 
 using namespace std;
 
--- src/graphics/FilterFastBandpass.cpp.orig	2008-03-30 12:21:00.000000000 +0000
+++ src/graphics/FilterFastBandpass.cpp	2008-03-30 12:21:17.000000000 +0000
@@ -27,6 +27,7 @@
 
 #include <iostream>
 #include <sstream>
+#include <cstring>
 
 using namespace std;
 
--- src/graphics/testgraphics.cpp.orig	2008-03-30 12:22:17.000000000 +0000
+++ src/graphics/testgraphics.cpp	2008-03-30 12:22:28.000000000 +0000
@@ -46,6 +46,7 @@
 #include <iostream>
 #include <stdio.h>
 #include <stdlib.h>
+#include <cstring>
 
 using namespace avg;
 using namespace std;
--- src/imaging/TrackerConfig.cpp.orig	2008-03-30 12:23:16.000000000 +0000
+++ src/imaging/TrackerConfig.cpp	2008-03-30 12:23:29.000000000 +0000
@@ -32,6 +32,7 @@
 #include <libxml/xmlstring.h>
 
 #include <sstream>
+#include <cstring>
 
 using namespace std;
 
--- src/imaging/V4LCamera.cpp.orig	2008-03-30 12:49:22.000000000 +0000
+++ src/imaging/V4LCamera.cpp	2008-03-30 12:49:37.000000000 +0000
@@ -39,6 +39,7 @@
 #include <sys/time.h>
 #include <unistd.h>
 #include <iostream>
+#include <cstring>
 
 #define CLEAR(x) memset (&(x), 0, sizeof (x))
 
--- src/imaging/DeDistort.cpp.orig	2008-03-30 12:48:49.000000000 +0000
+++ src/imaging/DeDistort.cpp	2008-03-30 12:49:13.000000000 +0000
@@ -27,6 +27,7 @@
 
 #include <iostream>
 #include <math.h>
+#include <cstring>
 
 const double sqrt3 = sqrt(3.);
 
--- src/video/FFMpegDemuxer.cpp.orig	2008-03-30 12:50:35.000000000 +0000
+++ src/video/FFMpegDemuxer.cpp	2008-03-30 12:50:48.000000000 +0000
@@ -25,6 +25,7 @@
 
 #include <iostream>
 #include <assert.h>
+#include <cstring>
 
 using namespace std;
 
--- src/parport/ParPort.cpp.orig	2008-03-30 12:52:22.000000000 +0000
+++ src/parport/ParPort.cpp	2008-03-30 12:52:36.000000000 +0000
@@ -39,6 +39,7 @@
 #include <iostream>
 #include <sstream>
 #include <cerrno>
+#include <cstring>
 
 using namespace std;
 
