--- src/libavg-0.5.9/src/player/OGLTile.cpp	2006-08-11 12:16:15.000000000 +0200
+++ src/libavg-0.5.9/src/player/OGLTile.cpp.new	2006-08-12 22:28:46.000000000 +0200
@@ -196,7 +196,7 @@
     unsigned char * pStartPos = (unsigned char *)
             (Extent.tl.y*stride*bpp + Extent.tl.x*bpp);
     if (MemoryMode == OGL) {
-        pStartPos += (unsigned int)(pBmp->getPixels());
+        pStartPos += (unsigned long)(pBmp->getPixels());
     }
     {
         ScopeTimer Timer(TexSubImageProfilingZone);
