--- rstart-0.99.1/Makefile.am.installation-location-fixes	2005-11-23 06:28:35.000000000 -0500
+++ rstart-1.0.0/Makefile.am	2005-11-23 06:31:44.000000000 -0500
@@ -26,7 +26,7 @@
 
 rstartd_real_CFLAGS = $(RSTART_CFLAGS)			\
 		-DSERVERNAME=\"rstart\"			\
-		-DDEFAULT_CONFIG=\"$(libdir)/X11/rstart/config\"
+		-DDEFAULT_CONFIG=\"$(datadir)/X11/rstart/config\"
 
 rstartd_real_LDADD = $(RSTART_LIBS)
 
@@ -50,9 +50,10 @@
 
 # config data
 
-configdir = $(libdir)/X11/rstart
+configdir = $(sysconfdir)/X11/rstart
 config_DATA = config
 
+# The -DLIBDIR=$(configdir) here just looks "wrong".. we'll see...
 config: config.cpp
 	$(RAWCPP) $(RAWCPPFLAGS) -DPACKAGEname=rstart -DLIBDIR=$(configdir) \
 	-DENVPREFIX=RSTART < ${srcdir}/config.cpp | $(CPP_SED_MAGIC) > $@
@@ -99,7 +100,7 @@
 # FIXME: Someone who cares about rstart may want to look into sanitizing
 # the stuff below.
 
-DATA_DIR = $(libdir)/X11/rstart
+DATA_DIR = $(datadir)/X11/rstart
 
 install-data-hook:
 	for name in `find ${srcdir}/commands -print | grep -v CVS` ; do	\
