--- transcode-1.0.1.orig/filter/filter_compare.c
+++ transcode-1.0.1/filter/filter_compare.c
@@ -180,6 +180,7 @@
 			if (!(compare[instance]->results = fopen(results_name, "w")))
 			{
 				perror("could not open file for writing");
+				return (-1);
 			}
 			
 			InitializeMagick("");
--- transcode-1.0.1.orig/filter/filter_logo.c	2005-10-21 22:18:49.000000000 +0200
+++ transcode-1.0.1/filter/filter_logo.c	2005-10-21 22:19:52.000000000 +0200
@@ -183,17 +183,18 @@
 
 	if (optstr_get (options, "help", "") >= 0)
 	    help_optstr();
-    }
 
-    if (verbose > 1) {
-	printf (" Logo renderer Settings:\n");
-	printf ("              file = %s\n", mfd->file);
-	printf ("            posdef = %d\n", mfd->pos);
-	printf ("               pos = %dx%d\n", mfd->posx, mfd->posy);
-	printf ("             range = %u-%u\n", mfd->start, mfd->end);
-	printf ("              flip = %d\n", mfd->flip);
-	printf ("       ignoredelay = %d\n", mfd->ignoredelay);
-	printf ("           rgbswap = %d\n", mfd->rgbswap);
+    	if (verbose > 1) {
+		printf (" Logo renderer Settings:\n");
+		printf ("              file = %s\n", mfd->file);
+		printf ("            posdef = %d\n", mfd->pos);
+		printf ("               pos = %dx%d\n", mfd->posx, mfd->posy);
+		printf ("             range = %u-%u\n", mfd->start, mfd->end);
+		printf ("              flip = %d\n", mfd->flip);
+		printf ("       ignoredelay = %d\n", mfd->ignoredelay);
+		printf ("           rgbswap = %d\n", mfd->rgbswap);
+    	}
+    return (-1);
     }
 
     InitializeMagick("");
