--- gscanbus.c	2001-07-11 11:53:51.000000000 +0100
+++ gscanbus.c.plasmaroo	2003-10-28 21:45:18.000000000 +0000
@@ -109,14 +109,8 @@
 
 static GdkPixmap *pixmap = NULL;
 
-const char not_compatible[] = "\
-This libraw1394 does not work with your version of Linux. You need a different
-version that matches your kernel (see kernel help text for the raw1394 option to
-find out which is the correct version).\n";
-
-const char not_loaded[] = "\
-This probably means that you don't have raw1394 support in the kernel or that
-you haven't loaded the raw1394 module.\n";
+const char not_compatible[] = "This libraw1394 does not work with your version of Linux. You need a different version that matches your kernel (see kernel help text for the raw1394 option to find out which is the correct version).\n";
+const char not_loaded[] = "This probably means that you don't have raw1394 support in the kernel or that you haven't loaded the raw1394 module.\n";
 
 /*---------------------------------------------------------------------------
  * Drawing routines
