--- build/src/mn-evolution-server.c.orig	2008-09-30 15:30:07.756216978 +0200
+++ build/src/mn-evolution-server.c	2008-09-30 15:30:49.950220711 +0200
@@ -27,6 +27,7 @@
 #include <stdio.h>
 #include <libintl.h>
 #include <gobject/gvaluecollector.h>
+#include <libedataserver/eds-version.h>
 #include <camel/camel-folder.h>
 #include <mail/em-folder-view.h>
 #include <mail/em-format.h>
@@ -595,7 +596,12 @@
 
 	for (i = 0; i < summary->len; i++)
 	  {
+#if EDS_CHECK_VERSION(2,23,5)
+	    char *uid = summary->pdata[i];
+	    CamelMessageInfo *info = camel_folder_get_message_info(folder, uid);
+#else
 	    CamelMessageInfo *info = summary->pdata[i];
+#endif
 
 	    if ((camel_message_info_flags(info) & CAMEL_MESSAGE_SEEN) == 0)
 	      g_ptr_array_add(*ret, self_camel_message_info_to_dbus_struct(info));
--- build/src/mn-evolution-folder-tree-server.c.evolution	2010-02-01 10:00:12.128683776 -0500
+++ build/src/mn-evolution-folder-tree-server.c	2010-02-01 10:02:32.809697243 -0500
@@ -26,5 +26,8 @@
 #include <dbus/dbus.h>
+#include <libedataserver/eds-version.h>
+#if !EDS_CHECK_VERSION(2,29,0)
 #include <mail/mail-component.h>
+#endif
 #include <mail/em-folder-tree.h>
 #include "mn-evolution-plugin.h"
 #include "mn-evolution.h"
@@ -396,7 +399,11 @@ mn_evolution_folder_tree_server_construc
 	
+#if EDS_CHECK_VERSION(2,29,0)
+    selfp->tree = em_folder_tree_new();
+#else
     EMFolderTreeModel *model;

     model = mail_component_peek_tree_model(mail_component_peek());
     selfp->tree = em_folder_tree_new_with_model(model);
+#endif
 
     selfp->plug = gtk_plug_new((GdkNativeWindow) selfp->id);
--- build/src/mn-evolution-server.c.evolution	2010-02-01 10:06:36.484698060 -0500
+++ build/src/mn-evolution-server.c	2010-02-01 10:09:01.946682081 -0500
@@ -28,9 +28,14 @@
 #include <gobject/gvaluecollector.h>
 #include <libedataserver/eds-version.h>
 #include <camel/camel-folder.h>
+#if EDS_CHECK_VERSION(2,29,0)
+#include <shell/e-shell.h>
+#include <mail/e-mail-browser.h>
+#else
 #include <mail/em-folder-view.h>
 #include <mail/em-format.h>
 #include <mail/em-message-browser.h>
+#endif
 #include <mail/em-utils.h>
 #include <mail/mail-session.h>
 #include <mail/mail-tools.h>
@@ -568,6 +573,19 @@ mn_evolution_server_open_message (MNEvol
     folder = self_lookup_folder(folder_uri, err);
     if (folder)
       {
+#if EDS_CHECK_VERSION(2,29,0)
+	EShell *shell;
+	EShellBackend *shell_backend;
+	GtkWidget *browser;
+
+	shell = e_shell_get_default ();
+	shell_backend = e_shell_get_backend_by_name (shell, "mail");
+
+	browser = e_mail_browser_new (shell_backend);
+	e_mail_reader_set_folder (E_MAIL_READER (browser), folder, folder_uri);
+	e_mail_reader_set_message (E_MAIL_READER (browser), message_uid);
+	gtk_widget_show (browser);
+#else
 	GtkWidget *browser;
 
 	/* modelled after Evolution's handleuri_got_folder() */
@@ -578,6 +596,7 @@ mn_evolution_server_open_message (MNEvol
 	em_folder_view_set_folder((EMFolderView *) browser, folder, folder_uri);
 	em_folder_view_set_message((EMFolderView *) browser, message_uid, FALSE);
 	gtk_widget_show(((EMMessageBrowser *) browser)->window);
+#endif
 
 	camel_object_unref(folder);
       }
--- src/mn-evolution-plugin.c.evolution	2008-05-22 11:45:35.000000000 -0400
+++ src/mn-evolution-plugin.c	2010-02-01 11:52:06.141664757 -0500
@@ -204,7 +204,7 @@ connect_to_session_bus (void)
 }
 
 int
-e_plugin_lib_enable (EPluginLib *ep, int enable)
+e_plugin_lib_enable (EPlugin *ep, int enable)
 {
   static gboolean enabled = FALSE;
   GError *err = NULL;
