--- gpm-1.20.1/src/report.c.nodebug	2002-12-24 17:57:16.000000000 -0500
+++ gpm-1.20.1/src/report.c	2003-07-02 15:07:19.000000000 -0400
@@ -126,6 +126,7 @@
       /******************** RUNNING *****************/
       case GPM_RUN_DAEMON:
          switch(stat) {
+#if 0
             case GPM_STAT_INFO:
 #ifdef HAVE_VSYSLOG
                syslog(LOG_INFO | LOG_USER, GPM_STRING_INFO);
@@ -165,6 +166,7 @@
                   fclose(console);
                }
                break;
+#endif
 
             case GPM_STAT_OOPS:
 #ifdef HAVE_VSYSLOG
@@ -183,6 +185,7 @@
       /******************** DEBUGGING and CLIENT *****************/
       case GPM_RUN_DEBUG:
          switch(stat) {
+#if 0
             case GPM_STAT_INFO:
                console = stdout;
                fprintf(console,GPM_STRING_INFO); break;
@@ -195,13 +198,15 @@
             case GPM_STAT_DEBUG:
                console = stderr;
                fprintf(console,GPM_STRING_DEBUG); break;
+#endif
             case GPM_STAT_OOPS:
                console = stderr;
                fprintf(console,GPM_STRING_OOPS); break;
+               vfprintf(console,text,ap);
+               fprintf(console,"\n");
+               break;
          }
 
-         vfprintf(console,text,ap);
-         fprintf(console,"\n");
          
          if(stat == GPM_STAT_OOPS) exit(1);
 
--- gpm-1.20.1/src/lib/liblow.c.orig	2003-07-02 15:24:40.000000000 -0400
+++ gpm-1.20.1/src/lib/liblow.c	2003-07-02 15:24:44.000000000 -0400
@@ -201,7 +201,9 @@
 
    option.consolename = NULL;
 
+#if 0   
    gpm_report(GPM_PR_DEBUG,"VC: %d",flag);
+#endif
 
    /*....................................... First of all, check xterm */
 
