diff -Naur gpm-1.20.0-orig/doc/Makefile.in gpm-1.20.0/doc/Makefile.in
--- gpm-1.20.0-orig/doc/Makefile.in	2002-02-23 07:42:23.000000000 -0800
+++ gpm-1.20.0/doc/Makefile.in	2002-08-20 13:06:46.000000000 -0700
@@ -44,11 +44,11 @@
 
 # PS (dvips)
 %.ps: %.dvi
-	if [ $(DVIPS) != "no" ]; then $(DVIPS) -f $< > $@; fi
+#	if [ $(DVIPS) != "no" ]; then $(DVIPS) -f $< > $@; fi
 
 # INFO (makeinfo)
 %.info: %.texinfo
-	if [ "$(MAKEINFO)" != "no" ]; then $(MAKEINFO) $< -o $@; fi
+#	if [ "$(MAKEINFO)" != "no" ]; then $(MAKEINFO) $< -o $@; fi
 
 # TXT (-)
 %.txt: %.info
--- gpm-1.20.1.old/src/lib/liblow.c	2002-12-25 03:57:16.000000000 +0500
+++ gpm-1.20.1/src/lib/liblow.c	2003-05-02 21:51:37.000000000 +0600
@@ -199,7 +199,8 @@
    Gpm_Stst *new = NULL;
    char* sock_name = 0;
 
-   option.consolename = NULL;
+   if (!checked_con)
+      option.consolename = NULL;
 
    gpm_report(GPM_PR_DEBUG,"VC: %d",flag);
 
