diff --git a/lldb/docs/conf.py b/lldb/docs/conf.py
index b9b94672cbde..d318762e7440 100644
--- a/lldb/docs/conf.py
+++ b/lldb/docs/conf.py
@@ -37,8 +37,7 @@ automodapi_toctreedirnm = 'python_api'
 
 # Add any Sphinx extension module names here, as strings. They can be extensions
 # coming with Sphinx (named 'sphinx.ext.*') or your custom ones.
-extensions = ['sphinx.ext.todo', 'sphinx.ext.mathjax', 'sphinx.ext.intersphinx',
-              'sphinx_automodapi.automodapi']
+extensions = ['sphinx.ext.todo', 'sphinx.ext.mathjax', 'sphinx.ext.intersphinx']
 
 # Add any paths that contain templates here, relative to this directory.
 templates_path = ['_templates']
