
diff -r 59decc957b39 -r d20059f7d3a1 src/pop3/pop3-commands.c
--- a/src/pop3/pop3-commands.c	Fri Feb 07 15:03:33 2014 -0500
+++ b/src/pop3/pop3-commands.c	Thu Feb 13 05:33:13 2014 +0200
@@ -253,7 +253,7 @@
 		for (msgnum = 0; msgnum < client->messages_count; msgnum++) {
 			bit = 1 << (msgnum % CHAR_BIT);
 			if ((client->deleted_bitmask[msgnum / CHAR_BIT] & bit) != 0)
-				seq_range_array_add(&deleted_msgs, client->msgnum_to_seq_map[msgnum]);
+				seq_range_array_add(&deleted_msgs, msgnum_to_seq(client, msgnum));
 		}
 	}
 	t_array_init(&seen_msgs, 8);
@@ -261,7 +261,7 @@
 		for (msgnum = 0; msgnum < client->messages_count; msgnum++) {
 			bit = 1 << (msgnum % CHAR_BIT);
 			if ((client->seen_bitmask[msgnum / CHAR_BIT] & bit) != 0)
-				seq_range_array_add(&seen_msgs, client->msgnum_to_seq_map[msgnum]);
+				seq_range_array_add(&seen_msgs, msgnum_to_seq(client, msgnum));
 		}
 	}
 
@@ -826,7 +826,7 @@
 				      client->messages_count+1);
 	for (msgnum = 0; msgnum < client->messages_count; msgnum++) {
 		client->message_uidls[msgnum] =
-			seq_uidls[client->msgnum_to_seq_map[msgnum]];
+			seq_uidls[msgnum_to_seq(client, msgnum)];
 	}
 	i_free(seq_uidls);
 }





