diff --git a/rc.sysinit b/rc.sysinit
index eb49e2b..8f9727b 100755
--- a/rc.sysinit
+++ b/rc.sysinit
@@ -34,7 +34,6 @@ bootlogd -p /run/bootlogd.pid
 
 run_hook sysinit_start
 
-HWCLOCK_PARAMS="--systz"
 case $HARDWARECLOCK in
 	"") ;;
 	UTC) HWCLOCK_PARAMS+=" --utc --noadjfile";;
@@ -54,7 +53,7 @@ if [[ $HWCLOCK_PARAMS ]]; then
 
 	[[ $TIMEZONE ]] && export TZ=$TIMEZONE
 
-	hwclock $HWCLOCK_PARAMS && stat_done || stat_fail
+	{ hwclock $HWCLOCK_PARAMS && hwclock --systz; } && stat_done || stat_fail
 
 	unset TZ
 fi
