--- Makefile~	2006-06-24 01:35:00.000000000 +0200
+++ Makefile	2006-08-28 13:29:15.000000000 +0200
@@ -93,7 +93,7 @@
 	IEEE80211_API := $(shell [[ "$(IEEE80211_VER)" == 1.1.14* ]] && \
 		{ echo "2" ; } || \
 		{ ver=$(IEEE80211_VER); ver=$${ver\#*.} ; echo $${ver%.*} ; })
-	EXTRA_CFLAGS += -DIEEE80211_API_VERSION=$(IEEE80211_API)
+	EXTRA_CFLAGS += -DIEEE80211_API_VERSION=2
 endif
 
 ifeq ($(CONFIG_IPW3945_DEBUG),y)
