diff -r 172f29597bd2 plugins/sudoers/sudoers.c
--- a/plugins/sudoers/sudoers.c	Sun Mar 06 15:38:02 2011 -0500
+++ b/plugins/sudoers/sudoers.c	Sun Mar 06 15:52:06 2011 -0500
@@ -546,8 +546,10 @@
     if (def_group_plugin)
 	group_plugin_unload();
 
-    if (ISSET(sudo_mode, (MODE_VALIDATE|MODE_CHECK|MODE_LIST)))
+    if (ISSET(sudo_mode, (MODE_VALIDATE|MODE_CHECK|MODE_LIST))) {
+	rval = TRUE;
 	goto done;
+    }
 
     /*
      * Set umask based on sudoers.
