--- branches/KDE/4.5/kdeplasma-addons/libs/lancelot-datamodels/BaseModel.cpp	2010/07/29 11:58:11	1156621
+++ branches/KDE/4.5/kdeplasma-addons/libs/lancelot-datamodels/BaseModel.cpp	2010/07/29 11:58:49	1156622
@@ -165,7 +165,7 @@
 {
     const KService::Ptr service = KService::serviceByStorageId(serviceName);
 
-    qDebug() << "BaseModel::addService Applic:" << service->isValid() << " " << serviceName;
+    // qDebug() << "BaseModel::addService Applic:" << service->isValid() << " " << serviceName;
 
     return addService(service);
 }

