From d2f57865d2b80ab301aba35278e71b85a1a141cd Mon Sep 17 00:00:00 2001
From: Stef Walter <stef@memberwebs.com>
Date: Thu, 10 Jun 2010 14:43:48 +0000
Subject: Remove erroneus egg-dbus dependency.

---
diff --git a/configure.in b/configure.in
index 16a5807..3906f9e 100644
--- a/configure.in
+++ b/configure.in
@@ -85,7 +85,7 @@ AC_CHECK_FUNCS(mlock)
 #
 
 DBUS_REQUIRED=1.0
-PKG_CHECK_MODULES([DBUS], dbus-1 >= 1.0 eggdbus-1 >= 0.4)
+PKG_CHECK_MODULES([DBUS], dbus-1 >= 1.0)
 
 DBUS_VERSION=`$PKG_CONFIG --modversion dbus-1`
 DBUS_MAJOR_VERSION=`echo $DBUS_VERSION | cut -d. -f1`
--
cgit v0.8.3.1
