diff --git i/server/meson.build w/server/meson.build
index bb7f315..845aaac 100644
--- i/server/meson.build
+++ w/server/meson.build
@@ -8,7 +8,8 @@ server_service = executable(
         server_dep,
     ],
     install: true,
-    install_dir: join_paths(get_option('prefix'), get_option('libexecdir'))
+    install_dir: join_paths(get_option('prefix'), get_option('libexecdir')),
+    install_rpath: join_paths(get_option('prefix'), get_option('libdir'), 'dleyna-server')
 )
 
 dbus_conf = configuration_data()
