--- apc_cache.c	2010-11-30 11:18:31.000000000 +0100
+++ apc_cache.c	2010-11-30 12:20:09.781164212 +0100
@@ -868,12 +868,12 @@
             fileinfo = apc_php_malloc(sizeof(apc_fileinfo_t) TSRMLS_CC);
 
             if (apc_search_paths(filename, include_path, fileinfo TSRMLS_CC) != 0) {
-                apc_warning("apc failed to locate %s - bailing" TSRMLS_CC, filename);
+                apc_debug("apc failed to locate %s - bailing" TSRMLS_CC, filename);
                 goto cleanup;
             }
 
             if(!VCWD_REALPATH(fileinfo->fullpath, APCG(canon_path))) {
-                apc_warning("realpath failed to canonicalize %s - bailing" TSRMLS_CC, filename);
+                apc_debug("realpath failed to canonicalize %s - bailing" TSRMLS_CC, filename);
                 goto cleanup;
             }
 
