diff -aur old/help/Makefile.in new/help/Makefile.in
--- old/help/Makefile.in	2011-01-11 17:04:09.000000000 +0100
+++ new/help/Makefile.in	2011-01-13 21:46:53.250005822 +0100
@@ -1066,7 +1066,7 @@
 @GNOME_SUPPORT_TRUE@stardict-install-html:
 @GNOME_SUPPORT_FALSE@stardict-install-html:
 @GNOME_SUPPORT_FALSE@	for lc in $(STARDICT_ALL_LINGUAS); do \
-@GNOME_SUPPORT_FALSE@		install_dir=$(STARDICT_HTML_INSTALL_DIR)/$${lc}; \
+@GNOME_SUPPORT_FALSE@		install_dir=$(DESTDIR)$(STARDICT_HTML_INSTALL_DIR)/$${lc}; \
 @GNOME_SUPPORT_FALSE@		htmldir=$(srcdir)/$${lc}/html; \
 @GNOME_SUPPORT_FALSE@		for file in `find $${htmldir} -type f -print`; do \
 @GNOME_SUPPORT_FALSE@			relfile=$${file#$${htmldir}/}; \
@@ -1085,7 +1085,7 @@
 @GNOME_SUPPORT_TRUE@stardict-uninstall-html:
 @GNOME_SUPPORT_FALSE@stardict-uninstall-html:
 @GNOME_SUPPORT_FALSE@	for lc in $(STARDICT_ALL_LINGUAS); do \
-@GNOME_SUPPORT_FALSE@		install_dir=$(STARDICT_HTML_INSTALL_DIR)/$${lc}; \
+@GNOME_SUPPORT_FALSE@		install_dir=$(DESTDIR)$(STARDICT_HTML_INSTALL_DIR)/$${lc}; \
 @GNOME_SUPPORT_FALSE@		rm -rf $${install_dir}; \
 @GNOME_SUPPORT_FALSE@	done;
 
