diff --git a/src/kalziumschemetype.cpp b/src/kalziumschemetype.cpp
index 91a5cbf0..e2935916 100644
--- a/src/kalziumschemetype.cpp
+++ b/src/kalziumschemetype.cpp
@@ -23,7 +23,7 @@
 #include "prefs.h"
 
 #ifdef HAVE_OPENBABEL2
-#include <openbabel/mol.h>
+#include <openbabel/elements.h>
 #endif
 
 #include <QBrush>
@@ -406,8 +407,9 @@ QBrush KalziumColorSchemeType::elementBrush(int el) const
 {
     QColor c;
 
-    std::vector<double> color = OpenBabel::etab.GetRGB(el);
-    c.setRgbF(color[0], color[1], color[2]);
+    double red, green, blue;
+    OpenBabel::OBElements::GetRGB(el, &red, &green, &blue);
+    c.setRgbF(red, green, blue);
 
     return QBrush(c);
 }
