diff --git a/kcms/touchpad/src/backends/x11/xcbatom.cpp b/kcms/touchpad/src/backends/x11/xcbatom.cpp
--- a/kcms/touchpad/src/backends/x11/xcbatom.cpp
+++ b/kcms/touchpad/src/backends/x11/xcbatom.cpp
@@ -46,6 +46,9 @@
 {
     if (!m_fetched) {
         m_fetched = true;
+        if(!m_connection) {
+            return 0;
+        }
         m_reply = xcb_intern_atom_reply(m_connection, m_cookie, 0);
     }
     if (m_reply) {

