diff --git a/CMakeLists.txt b/CMakeLists.txt
index 64e57148..bd37d0ed 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -73,8 +73,8 @@ if(WIN32)
    set(CMAKE_REQUIRED_INCLUDES  ${KDEWIN32_INCLUDES})
 endif()
 
-find_package(Sqlite)
-set_package_properties(Sqlite PROPERTIES
+find_package(SQLite3)
+set_package_properties(SQLite3 PROPERTIES
     DESCRIPTION "SQLite is a Binary-Database"
     TYPE RECOMMENDED
     PURPOSE "Needed for the SQLite-Backend of the KGet-History"
@@ -161,7 +161,7 @@ include_directories(
     ${CMAKE_CURRENT_BINARY_DIR}/core/
 )
 
-if(SQLITE_FOUND)
+if(SQLite3_FOUND)
     add_definitions(-DHAVE_SQLITE)
 endif()
 
