--- parprouted-0.7/parprouted.orig	2008-09-25 21:08:41.000000000 +0300
+++ parprouted-0.7/parprouted.c	2008-09-25 21:15:51.000000000 +0300
@@ -106,7 +106,7 @@ int route_remove(ARPTAB_ENTRY* cur_entry
     int success = 1;
     
     if (snprintf(routecmd_str, ROUTE_CMD_LEN-1, 
-	    "/sbin/ip route del %s/32 metric 50 dev %s scope link",
+	    "/usr/sbin/ip route del %s/32 metric 50 dev %s scope link",
 	    inet_ntoa(cur_entry->ipaddr_ia), cur_entry->ifname) > ROUTE_CMD_LEN-1) 
     {
 	syslog(LOG_INFO, "ip route command too large to fit in buffer!");
@@ -136,7 +136,7 @@ int route_add(ARPTAB_ENTRY* cur_entry)
     int success = 1;
 
     if (snprintf(routecmd_str, ROUTE_CMD_LEN-1, 
-	    "/sbin/ip route add %s/32 metric 50 dev %s scope link",
+	    "/usr/sbin/ip route add %s/32 metric 50 dev %s scope link",
 	    inet_ntoa(cur_entry->ipaddr_ia), cur_entry->ifname) > ROUTE_CMD_LEN-1) 
     {
 	syslog(LOG_INFO, "ip route command too large to fit in buffer!");
