--- makefile.unix	2010-03-30 17:26:26.000000000 +0200
+++ makefile.unix	2011-01-05 01:27:11.092105262 +0200
@@ -133,7 +133,7 @@ sfx:	$(OBJECTS)
 lib:	WHAT=RARDLL
 lib:	$(OBJECTS) $(LIB_OBJ)
 	@rm -f libunrar.so
-	$(LINK) -shared -o libunrar.so $(LDFLAGS) $(OBJECTS) $(LIB_OBJ)
+	$(LINK) -shared -Wl,-soname,libunrar.so.$(libversion) -o libunrar.so.$(libversion) $(LDFLAGS) $(OBJECTS) $(LIB_OBJ)
 
 install-unrar:
 			install -D unrar $(DESTDIR)/bin/unrar
