--- a/makefile	2013-04-29 16:27:05.000000000 +0200
+++ b/makefile	2013-10-01 22:07:29.611485989 +0200
@@ -133,7 +133,7 @@
 lib:	CXXFLAGS+=$(LIBFLAGS)
 lib:	clean $(OBJECTS) $(LIB_OBJ)
 	@rm -f libunrar.so
-	$(LINK) -shared -o libunrar.so $(LDFLAGS) $(OBJECTS) $(LIB_OBJ)
+	$(LINK) -shared -Wl,-soname,libunrar.so.$(libversion) -o libunrar.so.$(libversion) $(LDFLAGS) $(OBJECTS) $(LIB_OBJ)
 
 install-unrar:
 			install -D unrar $(DESTDIR)/bin/unrar
