--- match.cpp	2011-01-04 14:28:47.000000000 +0200
+++ match.cpp	2011-01-05 01:26:05.813980804 +0200
@@ -59,7 +59,7 @@ bool CmpName(const char *Wildcard,const
 
     char Path1[NM],Path2[NM];
     GetFilePath(Wildcard,Path1,ASIZE(Path1));
-    GetFilePath(Name,Path2,ASIZE(Path1));
+    GetFilePath(Name,Path2,ASIZE(Path2));
 
     if ((CmpMode==MATCH_EXACT || CmpMode==MATCH_EXACTPATH) &&
         mstricompc(Path1,Path2,ForceCase)!=0)
