--- kdesdk-4.5.80/kcachegrind/converters/hotshot2calltree~	2010-11-24 11:53:38.586666671 +0100
+++ kdesdk-4.5.80/kcachegrind/converters/hotshot2calltree	2010-11-24 11:53:38.623333337 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # _*_ coding: latin1 _*_
 
 #
--- kdesdk-4.5.80/lokalize/scripts/msgmerge.py~	2010-11-24 11:22:42.120000002 +0100
+++ kdesdk-4.5.80/lokalize/scripts/msgmerge.py	2010-11-24 11:22:42.146666670 +0100
@@ -114,7 +114,7 @@
         print >>sys.stderr, "Execution failed:", e
 
     cmd='%s/odf/xliffmerge.py -i %s -t %s -o %s' % (ourPath,xliffpathname,xlifftemplatepathname,xliffpathname)
-    if os.name!='nt': cmd='python '+cmd
+    if os.name!='nt': cmd='python2 '+cmd
     else: cmd=cmd.replace('/','\\')
     os.system(cmd)

--- kdesdk-4.5.80/lokalize/scripts/xliff2odf.py~	2010-11-24 11:24:10.853333336 +0100
+++ kdesdk-4.5.80/lokalize/scripts/xliff2odf.py	2010-11-24 11:24:10.883333336 +0100
@@ -42,7 +42,7 @@
     xliff2odf.convertxliff(xliffinput, translatedodfpathname, odf)
 
     ourpath=([p for p in sys.path if os.path.exists(p+'/xliff2odf.py')]+[''])[0]
-    os.system('python "'+ourpath+'/xliff2odf-standalone.py" "%s" "%s" &'%(translatedodfpathname, Editor.currentEntryId()))
+    os.system('python2 "'+ourpath+'/xliff2odf-standalone.py" "%s" "%s" &'%(translatedodfpathname, Editor.currentEntryId()))
 
 try: convert()
 except: print 'error occured'
--- kdesdk-4.5.80/scripts/rename_source_files~	2010-11-24 11:45:41.040000004 +0100
+++ kdesdk-4.5.80/scripts/rename_source_files	2010-11-24 11:45:41.093333336 +0100
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2
 #
 # Copyright David Faure <faure@kde.org>, License LGPL v2
 #
--- kdesdk-4.5.80/scripts/svn2log.sh~	2010-11-24 11:46:24.863333337 +0100
+++ kdesdk-4.5.80/scripts/svn2log.sh	2010-11-24 11:46:24.896666669 +0100
@@ -17,6 +17,6 @@
 svn cat svn://anonsvn.kde.org/home/kde/trunk/kde-common/accounts > /tmp/accounts.$PPID
 
 echo "Creating changelog...";
-svn log -v --xml $1 | python $CURRENT/svn2log.py --users=/tmp/accounts.$PPID --users-charset=UTF8
+svn log -v --xml $1 | python2 $CURRENT/svn2log.py --users=/tmp/accounts.$PPID --users-charset=UTF8
 
 rm /tmp/accounts.$PPID
--- kdesdk-4.5.80/scripts/kde_generate_export_header~	2010-11-24 11:48:49.696666669 +0100
+++ kdesdk-4.5.80/scripts/kde_generate_export_header	2010-11-24 11:48:49.753333338 +0100
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2
 
 import os, sys, string
 
--- kdesdk-4.5.80/scripts/reviewboarddiff~	2010-11-24 11:49:37.686666670 +0100
+++ kdesdk-4.5.80/scripts/reviewboarddiff	2010-11-24 11:49:37.740000003 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # encoding: utf-8
 #
 # Generates reviewboard compatible diffs from git-svn repositories.
