From 902917eb4f66f69496e977b3c72e0fa178775f89 Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <jan.steffens@gmail.com>
Date: Wed, 27 Nov 2019 22:29:48 +0000
Subject: [PATCH] filechooser: Don't try to free the method name; it's static

---
 libportal/filechooser.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/libportal/filechooser.c b/libportal/filechooser.c
index 2b665b8..135c269 100644
--- a/libportal/filechooser.c
+++ b/libportal/filechooser.c
@@ -78,7 +78,6 @@ file_call_free (FileCall *call)
   g_object_unref (call->portal);
   g_object_unref (call->task);
 
-  g_free (call->method);
   g_free (call->title);
   g_free (call->current_name);
   g_free (call->current_folder);
-- 
2.24.0

