--- src/billard3d.c	2010-02-27 04:01:40.000000000 -0600
+++ src/billard3d.c	2010-02-27 04:02:59.000000000 -0600
@@ -1219,6 +1219,8 @@
     printf("tournament_state_setup_next_match 7\n");
     queue_view=player[act_player].queue_view;
     printf("tournament_state_setup_next_match 8\n");
+    snooker_state.to_play = SN_PLAY_RED;
+    printf("tournament_state_setup_next_match 9\n");
 }
 
 void create_players_text()
