diff -Nura kdelibs-4.3.0.orig/kded/applications.menu kdelibs-4.3.0/kded/applications.menu
--- kdelibs-4.3.0.orig/kded/applications.menu	2009-07-29 13:09:44.000000000 +0200
+++ kdelibs-4.3.0/kded/applications.menu	2009-07-30 13:18:36.000000000 +0200
@@ -30,6 +30,14 @@
 				<And>
 					<Category>Core</Category>
 					<Not><Category>KDE</Category></Not>
+		<Name>Archlinux</Name>                                     
+		<Directory>Archlinux.directory</Directory>                 
+		<Include>                                                  
+		<Category>Archlinux</Category>                     
+		</Include>                                                 
+	</Menu>
+
+	<Menu>
 				</And>
 				<!-- Don't list SUSE's YaST in here -->
 				<Category>X-SuSE-YaST</Category>
