--- branches/KDE/3.5/kdelibs/kate/part/katesupercursor.cpp	2008/02/08 14:17:43	772352
+++ branches/KDE/3.5/kdelibs/kate/part/katesupercursor.cpp	2008/02/20 11:02:18	777286
@@ -186,7 +186,7 @@
 {
   if (newLine)
   {
-    if (m_line > int(line) || (m_line == int(line) && m_col > int(col)))
+    if (m_line > int(line) || (m_line == int(line) && m_col >= int(col)))
     {
       if(m_line == int(line))
          m_col -= col;
