Index: kio/kfile/kicondialog.cpp
===================================================================
--- kio/kfile/kicondialog.cpp	(revision 495706)
+++ kio/kfile/kicondialog.cpp	(working copy)
@@ -331,6 +331,13 @@
     mpCombo->insertItem(i18n("Devices"));
     mpCombo->insertItem(i18n("Filesystems"));
     mpCombo->insertItem(i18n("Mimetypes"));
+    mpCombo->insertItem(i18n("Animations"));
+    mpCombo->insertItem(i18n("Categories"));
+    mpCombo->insertItem(i18n("Emblems"));
+    mpCombo->insertItem(i18n("Emotes"));
+    mpCombo->insertItem(i18n("International"));
+    mpCombo->insertItem(i18n("Places"));
+    mpCombo->insertItem(i18n("Status"));
     mpCombo->setFixedSize(mpCombo->sizeHint());
     mpBrowseBut->setFixedWidth(mpCombo->width());
 
Index: kdecore/kicontheme.h
===================================================================
--- kdecore/kicontheme.h	(revision 495706)
+++ kdecore/kicontheme.h	(working copy)
@@ -51,7 +51,14 @@
       Application, ///< An icon that represents an application.
       Device, ///< An icon that represents a device.
       FileSystem, ///< An icon that represents a file system.
-      MimeType ///< An icon that represents a mime type (or file type).
+      MimeType, ///< An icon that represents a mime type (or file type).
+      Animation, ///< An icon that is animated.
+      Category, ///< An icon that represents a category.
+      Emblem, ///< An icon that adds information to an existing icon.
+      Emote, ///< An icon that expresses an emotion.
+      International, ///< An icon that represents a country's flag.
+      Place, ///< An icon that represents a location (e.g. 'home', 'trash').
+      StatusIcon ///< An icon that represents an event.
     };
 
     /**
Index: kdecore/kicontheme.cpp
===================================================================
--- kdecore/kicontheme.cpp	(revision 495706)
+++ kdecore/kicontheme.cpp	(working copy)
@@ -517,6 +517,20 @@
         mContext = KIcon::Application;
     else if (tmp == "Actions")
         mContext = KIcon::Action;
+    else if (tmp == "Animations")
+        mContext = KIcon::Animation;
+    else if (tmp == "Categories")
+        mContext = KIcon::Category;
+    else if (tmp == "Emblems")
+        mContext = KIcon::Emblem;
+    else if (tmp == "Emotes")
+        mContext = KIcon::Emote;
+    else if (tmp == "International")
+        mContext = KIcon::International;
+    else if (tmp == "Places")
+        mContext = KIcon::Place;
+    else if (tmp == "Status")
+        mContext = KIcon::StatusIcon;
     else {
         kdDebug(264) << "Invalid Context= line for icon theme: " << mDir << "\n";
         return;
