diff -Nur LVM2.2.02.106.orig/scripts/lvm2_pvscan_systemd_red_hat@.service.in LVM2.2.02.106/scripts/lvm2_pvscan_systemd_red_hat@.service.in
--- LVM2.2.02.106.orig/scripts/lvm2_pvscan_systemd_red_hat@.service.in	2014-04-10 17:38:46.000000000 +0200
+++ LVM2.2.02.106/scripts/lvm2_pvscan_systemd_red_hat@.service.in	2014-04-26 14:14:27.472679661 +0200
@@ -11,5 +11,5 @@
 [Service]
 Type=oneshot
 RemainAfterExit=yes
-ExecStart=@sbindir@/pvscan --cache --activate ay %i
-ExecStop=@sbindir@/pvscan --cache %i
+ExecStart=@sbindir@/lvm pvscan --cache --activate ay %i
+ExecStop=@sbindir@/lvm pvscan --cache %i
