diff -Naur squid-2.6.STABLE5-orig/helpers/basic_auth/SMB/Makefile.in squid-2.6.STABLE5/helpers/basic_auth/SMB/Makefile.in
--- squid-2.6.STABLE5-orig/helpers/basic_auth/SMB/Makefile.in	2006-10-29 21:21:19.000000000 -0500
+++ squid-2.6.STABLE5/helpers/basic_auth/SMB/Makefile.in	2006-11-07 10:57:36.000000000 -0500
@@ -262,7 +262,7 @@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
 SMB_AUTH_HELPER = smb_auth.sh
-SAMBAPREFIX = /usr/local/samba
+SAMBAPREFIX = /usr
 SMB_AUTH_HELPER_PATH = $(libexecdir)/$(SMB_AUTH_HELPER)
 libexec_SCRIPTS = $(SMB_AUTH_HELPER)
 smb_auth_CFLAGS = -DSAMBAPREFIX=\"$(SAMBAPREFIX)\" -DHELPERSCRIPT=\"$(SMB_AUTH_HELPER_PATH)\"
diff -Naur squid-2.6.STABLE5-orig/src/Makefile.in squid-2.6.STABLE5/src/Makefile.in
--- squid-2.6.STABLE5-orig/src/Makefile.in	2006-10-29 21:21:34.000000000 -0500
+++ squid-2.6.STABLE5/src/Makefile.in	2006-11-07 11:00:21.000000000 -0500
@@ -608,12 +608,12 @@
 DEFAULT_CONFIG_FILE = $(sysconfdir)/squid.conf
 DEFAULT_MIME_TABLE = $(sysconfdir)/mime.conf
 DEFAULT_DNSSERVER = $(libexecdir)/`echo dnsserver | sed '$(transform);s/$$/$(EXEEXT)/'`
-DEFAULT_LOG_PREFIX = $(localstatedir)/logs
+DEFAULT_LOG_PREFIX = $(localstatedir)/log/squid
 DEFAULT_CACHE_LOG = $(DEFAULT_LOG_PREFIX)/cache.log
 DEFAULT_ACCESS_LOG = $(DEFAULT_LOG_PREFIX)/access.log
 DEFAULT_STORE_LOG = $(DEFAULT_LOG_PREFIX)/store.log
-DEFAULT_PID_FILE = $(DEFAULT_LOG_PREFIX)/squid.pid
-DEFAULT_SWAP_DIR = $(localstatedir)/cache
+DEFAULT_PID_FILE = $(localstatedir)/run/squid.pid
+DEFAULT_SWAP_DIR = $(localstatedir)/cache/squid
 DEFAULT_PINGER = $(libexecdir)/`echo pinger | sed '$(transform);s/$$/$(EXEEXT)/'`
 DEFAULT_UNLINKD = $(libexecdir)/`echo unlinkd | sed '$(transform);s/$$/$(EXEEXT)/'`
 DEFAULT_DISKD = $(libexecdir)/`echo diskd-daemon | sed '$(transform);s/$$/$(EXEEXT)/'`
diff -Naur squid-2.6.STABLE5-orig/src/cf.data.pre squid-2.6.STABLE5/src/cf.data.pre
--- squid-2.6.STABLE5-orig/src/cf.data.pre	2006-10-12 16:48:48.000000000 -0400
+++ squid-2.6.STABLE5/src/cf.data.pre	2006-11-07 11:02:05.000000000 -0500
@@ -3280,7 +3280,7 @@
 
 NAME: cache_effective_user
 TYPE: string
-DEFAULT: nobody
+DEFAULT: proxy
 LOC: Config.effectiveUser
 DOC_START
 	If you start Squid as root, it will change its effective/real
@@ -3295,7 +3295,7 @@
 
 NAME: cache_effective_group
 TYPE: string
-DEFAULT: none
+DEFAULT: proxy
 LOC: Config.effectiveGroup
 DOC_START
 	If you want Squid to run with a specific GID regardless of
