diff --git a/data/org.freedesktop.impl.portal.desktop.kde.cmake.in b/data/org.freedesktop.impl.portal.desktop.kde.cmake.in
index df7b099..8343477 100644
--- a/data/org.freedesktop.impl.portal.desktop.kde.cmake.in
+++ b/data/org.freedesktop.impl.portal.desktop.kde.cmake.in
@@ -1,3 +1,3 @@
 [D-BUS Service]
 Name=org.freedesktop.impl.portal.desktop.kde
-Exec=@CMAKE_INSTALL_LIBEXECDIR@/xdg-desktop-portal-kde
+Exec=@CMAKE_INSTALL_FULL_LIBEXECDIR@/xdg-desktop-portal-kde
