diff -Nur cryptsetup-1.0.6.orig/lib/libdevmapper.c cryptsetup-1.0.6/lib/libdevmapper.c
--- cryptsetup-1.0.6.orig/lib/libdevmapper.c	2007-10-25 09:11:19.000000000 +0200
+++ cryptsetup-1.0.6/lib/libdevmapper.c	2008-09-12 18:27:44.000000000 +0200
@@ -18,7 +18,7 @@
 
 #define	CRYPT_TARGET	"crypt"
 
-#define UDEVSETTLE	"/sbin/udevsettle"
+#define UDEVSETTLE	"/sbin/udevadm settle"
 
 static void run_udevsettle(void)
 {
