--- /etc/bash_completion	2006-05-25 11:59:31.000000000 -0400
+++ bash_completion	2007-09-17 23:39:01.000000000 -0400
@@ -29,13 +29,17 @@
 
 # Alter the following to reflect the location of this file.
 #
+(readonly -p | grep " BASH_COMPLETION=" 2>&1 >/dev/null) || 
 {
-  # These declarations must go within braces in order to be able to silence
-  # readonly variable errors.
   BASH_COMPLETION="${BASH_COMPLETION:-/etc/bash_completion}"
+  readonly BASH_COMPLETION
+}
+
+(readonly -p | grep " BASH_COMPLETION_DIR=" 2>&1 >/dev/null) ||
+{
   BASH_COMPLETION_DIR="${BASH_COMPLETION_DIR:=/etc/bash_completion.d}"
-} 2>/dev/null || :
-readonly BASH_COMPLETION BASH_COMPLETION_DIR
+  readonly BASH_COMPLETION_DIR
+}
 
 # Set a couple of useful vars
 #
