diff -Nrup xf86-video-sis-0.9.1/src/sis_hwmc.c patched/src/sis_hwmc.c
--- xf86-video-sis-0.9.1/src/sis_hwmc.c	2008-03-04 14:31:10.000000000 -0300
+++ patched/src/sis_hwmc.c	2010-04-20 17:10:21.073690167 -0300
@@ -43,7 +43,9 @@ THE USE OR OTHER DEALINGS IN THE SOFTWAR
 
 #include "xf86.h"
 #include "xf86_OSproc.h"
+#ifndef GET_ABI_MAJOR(ABI_VIDEODRV_VERSION) < 6
 #include "xf86Resources.h"
+#endif
 #include "compiler.h"
 #include "xf86PciInfo.h"
 #include "xf86Pci.h"
