--- coredrv/coredrv.c~	2004-07-09 16:34:29.000000000 +0200
+++ coredrv/coredrv.c	2005-03-29 12:36:24.000000000 +0200
@@ -292,9 +292,6 @@
         case PM_RESUME:
 			open();
             break;
-        case PM_SAVE_STATE:
-            printk(KERN_WARNING"Saving power state is not implemented\n");
-            break;
         default:
             printk(KERN_WARNING"Unknown power mode request to 537 modem driver\n");
     }
@@ -859,9 +856,6 @@
 
 asmlinkage void kScheduleDPC(void)
 {
-#ifdef CONFIG_PM
-    pm_access(power_dev);
-#endif
    #if defined(TARGET_CATAWBA)
       dpc_function( 0 );
    #else
