--- config_check~	2010-02-27 16:18:10.000000000 +0100
+++ config_check	2010-02-27 16:18:10.000000000 +0100
@@ -36,7 +36,7 @@
 echo "   Current running kernel is: $KERNVER"
 
 
-if [ -a /lib/modules/$KERNVER/build/include/linux/autoconf.h ]; then
+if [ -a /lib/modules/$KERNVER/build/include/generated/autoconf.h ]; then
    echo "   /lib/modules...   autoconf.h exists"
 else
    echo "   /lib/modules...   autoconf.h does not exist"
@@ -44,7 +44,7 @@
    exit 1
 fi
 
-if [ "`diff /boot/vmlinuz.autoconf.h /lib/modules/$KERNVER/build/include/linux/autoconf.h`" \
+if [ "`diff /boot/vmlinuz.autoconf.h /lib/modules/$KERNVER/build/include/generated/autoconf.h`" \
     == "" ]; then
    echo "   autoconf.h matches running kernel"
 else
@@ -61,7 +61,7 @@
    echo  configurations to create a functional driver for this kernel.
    echo  call \"make config_sync\" and this script will
    echo  copy the file /boot/vmlinuz.autoconf.h to 
-   echo  /lib/modules/$KERNVER/build/include/linux/autoconf.h
+   echo  /lib/modules/$KERNVER/build/include/generated/autoconf.h
    echo  and backup the original autoconf.h as autoconf.bak
    echo  The same will be done for version.h
    echo  Then repeat the build and installation procedure over again.
--- coredrv/hamdefs.h~	2010-02-27 16:20:58.000000000 +0100
+++ coredrv/hamdefs.h	2010-02-27 16:20:58.000000000 +0100
@@ -51,7 +51,7 @@
    #define EXPORT_SYMTAB
    #include<linux/version.h>
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,18)
-   #include<linux/autoconf.h>
+   #include<generated/autoconf.h>
 #else
    #include<linux/config.h>
 #endif
