--- coredrv/softserial_ioctl.c~	2008-12-26 23:35:56.000000000 +0000
+++ coredrv/softserial_ioctl.c	2008-12-26 23:35:56.000000000 +0000
@@ -109,7 +109,7 @@
    case TIOCGSOFTCAR:
    case TIOCSSOFTCAR:
 //      return (-ENOIOCTLCMD);
-      return(n_tty_ioctl(ptty,pfile,ioctl,ioctl_parameter));
+      return(n_tty_ioctl_helper(ptty,pfile,ioctl,ioctl_parameter));
 //CASES FOR ONLY 2.1 KERNEL, IGNORE, RETURN AS DEFALUT
    case TCSBRK:
    case TCSBRKP:
