--- coredrv/coredrv.c~	2007-01-08 22:46:01.000000000 +0100
+++ coredrv/coredrv.c	2007-01-08 22:49:52.000000000 +0100
@@ -767,8 +767,8 @@
       timeout1 = wait_event_interruptible_timeout(persistReadQ, 0, timeout1);
    }
 }
-DECLARE_WORK(wait_wq2, WakeTask2, NULL);
-DECLARE_WORK(wait_wq, WakeTask, NULL);
+DECLARE_WORK(wait_wq2, WakeTask2);
+DECLARE_WORK(wait_wq, WakeTask);
 #endif
 
 asmlinkage void wake_up_interruptible_persistWriteQ(void)
--- coredrv/rts.c~	2005-07-26 17:59:33.000000000 +0200
+++ coredrv/rts.c	2007-01-08 22:53:13.000000000 +0100
@@ -77,7 +77,7 @@
 };
 #else
 //DECLARE_TASKLET(rs_tasklet, RTS_Task, 0);
-static DECLARE_WORK(rs_wq, RTS_Task, 0);
+static DECLARE_WORK(rs_wq, RTS_Task);
 
 #endif
 
--- coredrv/softserial.c~	2005-07-26 17:59:33.000000000 +0200
+++ coredrv/softserial.c	2007-01-08 22:55:30.000000000 +0100
@@ -188,8 +188,7 @@
 static struct async_struct softserial_async;
 #if LINUX_VERSION_CODE > KERNEL_VERSION(2,5,0)
 DECLARE_WORK(softserial_bg_event_work,
-             softserial_background_event_handler,
-             &softserial_async);
+             softserial_background_event_handler);
 #endif
 static struct serial_state state = {0,BASE_BAUD,-1,0,ASYNC_SPD_VHI | ASYNC_SKIP_TEST};
 int softserial_open(struct tty_struct* ptty, struct file* pfile)
