--- intel-536EP-2.56.76.0/coredrv/locks.c~	2008-10-11 13:22:47.000000000 +0000
+++ intel-536EP-2.56.76.0/coredrv/locks.c	2008-10-11 13:22:47.000000000 +0000
@@ -43,7 +43,7 @@
 #include<linux/interrupt.h>
 #include<linux/irq.h>
 #include<linux/spinlock.h>
-#include<asm/semaphore.h>
+#include<linux/semaphore.h>
 #define LOCK_DEFINED
 #include"lock_lin.h"
 
--- intel-536EP-2.56.76.0/coredrv/softserial.c~	2008-10-11 14:29:23.000000000 +0000
+++ intel-536EP-2.56.76.0/coredrv/softserial.c	2008-10-11 14:29:23.000000000 +0000
@@ -341,21 +341,7 @@
 
    ptty = p_async_struct->tty;
 
-   if(test_and_clear_bit(RS_EVENT_WRITE_WAKEUP, &p_async_struct->event))
-   {
-      if( (ptty->flags & (1 << TTY_DO_WRITE_WAKEUP)) &&
-          (ptty->ldisc.write_wakeup != NULL))
-      {
-#if defined (DEBUG_LINUX)
-	 printk("ss_bg_eh:ldisc wakeup\n");
-#endif
-         (ptty->ldisc.write_wakeup)(ptty);
-      }
-#if defined (DEBUG_LINUX)
-      printk("ss_bg_eh: wakeup write_wait\n");
-#endif
-      wake_up_interruptible(&ptty->write_wait);
-   }
+   tty_wakeup(ptty);
 }
 
 //=============================================================================
