--- coredrv/softserial.c~	2009-09-11 20:57:32.000000000 +0000
+++ coredrv/softserial.c	2009-09-11 20:57:32.000000000 +0000
@@ -362,7 +362,7 @@
    {
       if( (ptty->flags & (1 << TTY_DO_WRITE_WAKEUP)) &&
 #if LINUX_VERSION_CODE > KERNEL_VERSION(2,6,26)
-          (ptty->ldisc.ops->write_wakeup != NULL))
+          (ptty->ldisc->ops->write_wakeup != NULL))
 #else
           (ptty->ldisc.write_wakeup != NULL))
 #endif
@@ -371,7 +371,7 @@
 	 printk("ss_bg_eh:ldisc wakeup\n");
 #endif
 #if LINUX_VERSION_CODE > KERNEL_VERSION(2,6,26)
-         (ptty->ldisc.ops->write_wakeup)(ptty);
+         (ptty->ldisc->ops->write_wakeup)(ptty);
 #else
          (ptty->ldisc.write_wakeup)(ptty);
 #endif
