diff -ur scheme48-1.8/Makefile.in scheme48-1.8.sic/Makefile.in
--- scheme48-1.8/Makefile.in	2008-02-10 12:30:42.000000000 +0100
+++ scheme48-1.8.sic/Makefile.in	2008-01-26 16:06:27.000000000 +0100
@@ -416,8 +416,8 @@
 	    -e 's,@LD\@,$(LD),g'					\
 	    -e 's,@DYNAMIC_EXTERNALS_CFLAGS\@,$(DYNAMIC_EXTERNALS_CFLAGS),g' \
 	    -e 's,@DYNAMIC_EXTERNALS_LDFLAGS\@,$(DYNAMIC_EXTERNALS_LDFLAGS),g' \
-	< $(srcdir)/build/$(CONFIG_SCRIPT).in > $$config_script &&	\
-	chmod +x $$config_script
+	< $(srcdir)/build/$(CONFIG_SCRIPT).in > $(DESTDIR)/$$config_script &&	\
+	chmod +x $(DESTDIR)/$$config_script
 
 inst-externals:
 	$(INSTALL_PROGRAM) $(POSIX_EXTERNAL) $(SRFI_27_EXTERNAL) $(DESTDIR)$(LIB)/
