From 7b202653f352a6c1bd18066bc71c93b1a5b33c13 Mon Sep 17 00:00:00 2001
From: Petr Mladek <pmladek@suse.cz>
Date: Mon, 18 Jul 2011 16:48:49 +0200
Subject: [PATCH 2/3] distro-install-desktop-integration: do not pack libreoffice-base.desktop twice

---
 bin/distro-install-desktop-integration |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/bin/distro-install-desktop-integration b/bin/distro-install-desktop-integration
index 600c54e..9e3b499 100755
--- a/bin/distro-install-desktop-integration
+++ b/bin/distro-install-desktop-integration
@@ -148,7 +148,7 @@ add_wrapper lomath         soffice "--math"       "libreoffice" "libreoffice-mat
 add_wrapper loimpress      soffice "--impress"    "libreoffice" "libreoffice-impress.desktop"     "gid_Module_Brand_Prg_Impress"
 add_wrapper loweb          soffice "--web"        "libreoffice" ""                                "gid_Module_Brand_Prg_Wrt"
 add_wrapper lowriter       soffice "--writer"     "libreoffice" "libreoffice-writer.desktop"      "gid_Module_Brand_Prg_Wrt"
-add_wrapper lofromtemplate soffice ".uno:NewDoc" "libreoffice" "libreoffice-base.desktop"        "gid_Module_Root_Brand"
+add_wrapper lofromtemplate soffice ".uno:NewDoc"  "libreoffice" ""                                "gid_Module_Root_Brand"
 add_wrapper libreoffice    soffice ""            "libreoffice" "libreoffice-startcenter.desktop" "gid_Module_Root_Brand"
 add_wrapper loffice        soffice ""            "libreoffice" ""                                "gid_Module_Root_Brand"
 add_wrapper unopkg         unopkg  ""            "unopkg"      ""                                "gid_Module_Root_Brand"
-- 
1.7.3.4

