Index: gnome-games-2.30.2/libgames-support/games-stock.c
===================================================================
--- gnome-games-2.30.2.orig/libgames-support/games-stock.c	2010-07-04 11:33:16.880062363 +0200
+++ gnome-games-2.30.2/libgames-support/games-stock.c	2010-07-04 12:23:59.035567621 +0200
@@ -282,15 +282,14 @@ games_stock_init (void)
 #ifdef HAVE_GTK_ICON_FACTORY_ADD_ALIAS
     { GAMES_STOCK_REDO_MOVE,        GTK_STOCK_REDO },
     { GAMES_STOCK_UNDO_MOVE,        GTK_STOCK_UNDO },
+    { GAMES_STOCK_RESUME_GAME,      GTK_STOCK_MEDIA_PLAY },
 #endif
 #ifndef HAVE_HILDON
     { GAMES_STOCK_NETWORK_GAME,     GTK_STOCK_NETWORK },
     { GAMES_STOCK_NETWORK_LEAVE,    GTK_STOCK_STOP },
     { GAMES_STOCK_PLAYER_LIST,      GTK_STOCK_INFO },
 
-    { GAMES_STOCK_PAUSE_GAME,       "stock_timer_stopped" },
-    { GAMES_STOCK_RESUME_GAME,      "stock_timer" },
-    { GAMES_STOCK_SCORES,           "stock_scores" },
+    { GAMES_STOCK_PAUSE_GAME,       GTK_STOCK_MEDIA_PAUSE },
 #endif /* !HAVE_HILDON */
   };
 
@@ -298,6 +297,7 @@ games_stock_init (void)
   const char *stock_icon_aliases_bidi[][3] = {
     { GAMES_STOCK_REDO_MOVE, GTK_STOCK_REDO "-ltr", GTK_STOCK_REDO "-rtl" },
     { GAMES_STOCK_UNDO_MOVE, GTK_STOCK_UNDO "-ltr", GTK_STOCK_UNDO "-rtl" },
+    { GAMES_STOCK_RESUME_GAME, GTK_STOCK_MEDIA_PLAY "-ltr", GTK_STOCK_MEDIA_PLAY "-rtl" },
   };
 #endif
 
@@ -306,6 +306,7 @@ games_stock_init (void)
 #ifndef HAVE_HILDON
     { GAMES_STOCK_TELEPORT, "teleport" },
     { GAMES_STOCK_RTELEPORT, "teleport-random" },
+    { GAMES_STOCK_SCORES, "scores" },
 #endif /* !HAVE_HILDON */
     { GAMES_STOCK_DEAL_CARDS, "cards-deal" }
   };
