--- libavogadro/src/python/sip.cpp~	2010-05-02 18:31:22.000000000 +0200
+++ libavogadro/src/python/sip.cpp	2010-09-14 19:13:31.000000000 +0200
@@ -204,7 +204,11 @@
     sipWrapper *wrapper = reinterpret_cast<sipWrapper*>(obj_ptr);
 #endif
     // return the C++ pointer
+#if SIP_API_MAJOR_NR >= 8
+    return wrapper->data;
+#else
     return wrapper->u.cppPtr;
+#endif
   }
     
   QClass_converters()
