--- a/interface/tk_c.c
+++ b/interface/tk_c.c
@@ -65,6 +65,9 @@
 #define TKPROGPATH PKGLIBDIR "/tkmidity.tcl"
 #endif /* TKPROGPATH */
 
+#if (TCL_MAJOR_VERSION < 8)
+#define Tcl_GetStringResult(interp) (interp->result)
+#endif
 
 static void ctl_refresh(void);
 static void ctl_total_time(int tt);
@@ -117,7 +120,7 @@
 
 static void start_panel(void);
 
-#define MAX_TK_MIDI_CHANNELS	16
+#define MAX_TK_MIDI_CHANNELS	32
 
 typedef struct {
 	int reset_panel;
@@ -933,7 +936,7 @@
 }
 
 /* evaluate Tcl script */
-static char *v_eval(char *fmt, ...)
+static const char *v_eval(char *fmt, ...)
 {
 	char buf[256];
 	va_list ap;
@@ -941,7 +944,7 @@
 	vsnprintf(buf, sizeof(buf), fmt, ap);
 	Tcl_Eval(my_interp, buf);
 	va_end(ap);
-	return my_interp->result;
+	return Tcl_GetStringResult(my_interp);
 }
 
 static const char *v_get2(const char *v1, const char *v2)
