--- trunk/KDE/kdeadmin/system-config-printer-kde/system-config-printer-kde.py	2009/10/13 13:30:13	1034768
+++ trunk/KDE/kdeadmin/system-config-printer-kde/system-config-printer-kde.py	2009/10/13 13:30:30	1034769
@@ -48,7 +48,7 @@
 
 #use _() to keep code the same as gnome system-config-printer
 def _(string):
-    return unicode(i18n(string), "utf-8")
+    return unicode(i18n(string))
 
 def translate(self, prop):
     """reimplement method from uic to change it to use gettext"""
