--- trunk/KDE/kdeadmin/system-config-printer-kde/options.py	2010/01/25 07:05:28	1079837
+++ trunk/KDE/kdeadmin/system-config-printer-kde/options.py	2010/03/04 20:36:46	1098974
@@ -133,7 +133,7 @@
                 self.widget.clear()
                 if type(supported) == list:
                     for each in supported:
-                        self.widget.addItem(self.combobox_dict[each])
+                        self.widget.addItem(each)
                 else:
                     self.widget.addItem(self.combobox_dict[supported])
             elif (type(self.widget) == QComboBox and
