From d1282b5ddeadd1d93f61cceb122b942c56e629dc Mon Sep 17 00:00:00 2001
From: Reuben Hawkins <reubenhwk@gmail.com>
Date: Tue, 29 Jul 2014 20:31:18 -0700
Subject: [PATCH] send.c: rdnssinfo length fix

---
 send.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/send.c b/send.c
index ec6b93b..a2dded6 100644
--- a/send.c
+++ b/send.c
@@ -329,7 +329,7 @@ static void add_rdnss(struct safe_buffer * sb, struct AdvRDNSS const *rdnss, int
 		memcpy(&rdnssinfo.nd_opt_rdnssi_addr2, &rdnss->AdvRDNSSAddr2, sizeof(struct in6_addr));
 		memcpy(&rdnssinfo.nd_opt_rdnssi_addr3, &rdnss->AdvRDNSSAddr3, sizeof(struct in6_addr));
 
-		safe_buffer_append(sb, &rdnssinfo, sizeof(rdnssinfo));
+		safe_buffer_append(sb, &rdnssinfo, sizeof(rdnssinfo) - (3 - rdnss->AdvRDNSSNumber) * sizeof(struct in6_addr));
 
 		rdnss = rdnss->next;
 	}
-- 
2.0.3

