diff --git a/src/term.c b/src/term.c
--- a/src/term.c
+++ b/src/term.c
@@ -1996,6 +1996,7 @@
 #  define HMT_DEC	4
 #  define HMT_JSBTERM	8
 #  define HMT_PTERM	16
+#  define HMT_URXVT	32
 static int has_mouse_termcode = 0;
 # endif
 
@@ -2031,6 +2032,11 @@
 	has_mouse_termcode |= HMT_PTERM;
     else
 #   endif
+#   ifdef FEAT_MOUSE_URXVT
+    if (n == KS_URXVT_MOUSE)
+	has_mouse_termcode |= HMT_URXVT;
+    else
+#   endif
 	has_mouse_termcode |= HMT_NORMAL;
 #  endif
 }
@@ -2068,6 +2074,11 @@
 	has_mouse_termcode &= ~HMT_PTERM;
     else
 #   endif
+#   ifdef FEAT_MOUSE_URXVT
+    if (n == KS_URXVT_MOUSE)
+	has_mouse_termcode &= ~HMT_URXVT;
+    else
+#   endif
 	has_mouse_termcode &= ~HMT_NORMAL;
 #  endif
 }
