Index: twisted/words/protocols/jabber/jstrports.py
===================================================================
--- twisted/words/protocols/jabber/jstrports.py (revision 30346)
+++ twisted/words/protocols/jabber/jstrports.py (working copy)
@@ -6,7 +6,7 @@
 """ A temporary placeholder for client-capable strports, until we
 sufficient use cases get identified """
 
-from twisted.application import strports
+from twisted.internet.endpoints import _parse
 
 def _parseTCPSSL(factory, domain, port):
     """ For the moment, parse TCP or SSL connections the same """
@@ -22,7 +22,7 @@
 
 
 def parse(description, factory):
-    args, kw = strports._parse(description)
+    args, kw = _parse(description)
     return (args[0].upper(),) + _funcs[args[0]](factory, *args[1:], **kw)
 
 def client(description, factory):
