--- src/mred/Makefile.in	2007-04-09 08:34:00.000000000 +0000
+++ src/mred/Makefile.in.new	2007-07-03 19:30:42.000000000 +0000
@@ -318,8 +318,8 @@
 
 install-wx_xt-3m:
 	$(MAKE) @MRLIBINSTALL@-3m-wx_xt
-	cd ..; $(ICP) mred/mred@MMM@ "$(bindir)/mred@MMM_INSTALLED@"
-	$(MZSCHEME) -mvqu "$(srcdir)/../mzscheme/collects-path.ss" "$(bindir)/mred@MMM_INSTALLED@@EXE_SUFFIX@" @COLLECTS_PATH@
+	cd ..; $(ICP) mred/mred@MMM@ "$(DESTDIR)$(bindir)/mred@MMM_INSTALLED@"
+	$(MZSCHEME) -mvqu "$(srcdir)/../mzscheme/collects-path.ss" "$(DESTDIR)$(bindir)/mred@MMM_INSTALLED@@EXE_SUFFIX@" @COLLECTS_PATH@
 
 install-wx_xt-3m-final:
 	cd .
