From b73782a28894e25ed146271f9d6c6775a6836199 Mon Sep 17 00:00:00 2001
From: Behdad Esfahbod <behdad@behdad.org>
Date: Fri, 04 Jun 2010 18:36:45 +0000
Subject: Bug 601926 - Don't hardcode meta to alt

---
diff --git a/src/keymap.h b/src/keymap.h
index 3a4cefe..243e22e 100644
--- a/src/keymap.h
+++ b/src/keymap.h
@@ -27,7 +27,7 @@
 
 G_BEGIN_DECLS
 
-#define VTE_META_MASK		GDK_MOD1_MASK
+#define VTE_META_MASK		GDK_META_MASK
 #define VTE_NUMLOCK_MASK	GDK_MOD2_MASK
 
 /* Map the specified keyval/modifier setup, dependent on the mode, to either
--
cgit v0.9.2
