--- a/polkit-kde-authentication-agent-1.desktop.in
+++ b/polkit-kde-authentication-agent-1.desktop.in
@@ -15,7 +15,7 @@
 Comment[sv]=Policykit behörighetskontrollverktyg
 Comment[uk]=Агент розпізнавання PolicyKit
 Comment[x-test]=xxPolicyKit Authentication Agentxx
-Exec=${CMAKE_INSTALL_PREFIX}/${LIBEXEC_INSTALL_DIR}/polkit-kde-authentication-agent-1
+Exec=${CMAKE_INSTALL_FULL_LIBEXECDIR}/polkit-kde-authentication-agent-1
 Terminal=false
 Type=Application
 Categories=

