diff -ur redland-1.0.10.orig/src/Makefile.am redland-1.0.10/src/Makefile.am
--- redland-1.0.10.orig/src/Makefile.am	2009-11-23 08:38:32.000000000 +0200
+++ redland-1.0.10/src/Makefile.am	2009-12-18 15:29:10.000000000 +0200
@@ -152,7 +152,7 @@
 plugin_LTLIBRARIES += librdf_storage_virtuoso.la 
 librdf_storage_virtuoso_la_SOURCES = rdf_storage_virtuoso.c rdf_storage_virtuoso_internal.h
 librdf_storage_virtuoso_la_CPPFLAGS = @ODBC_CFLAGS@
-librdf_storage_virtuoso_la_LIBADD = @ODBC_LIBS@
+librdf_storage_virtuoso_la_LIBADD = @ODBC_LIBS@ librdf.la
 librdf_storage_virtuoso_la_LDFLAGS = -module -avoid-version
 endif
 
