diff -upr xfce4-smartbookmark-plugin-0.4.4.orig/src/smartbookmark.c xfce4-smartbookmark-plugin-0.4.4/src/smartbookmark.c
--- xfce4-smartbookmark-plugin-0.4.4.orig/src/smartbookmark.c	2012-11-19 22:24:16.000000000 +0200
+++ xfce4-smartbookmark-plugin-0.4.4/src/smartbookmark.c	2012-11-19 22:24:42.000000000 +0200
@@ -365,6 +365,7 @@ static void search_create_options(XfcePa
     search_apply_options_cb(search);
     gtk_widget_destroy(search->opt_dialog);
     xfce_panel_plugin_unblock_menu(plugin);
+    search_write_config(plugin, search);
 }
 
 
