diff -Naur ddcxinfo-arch-0.8/Makefile ddcxinfo-arch-0.8-new/Makefile
--- ddcxinfo-arch-0.8/Makefile	2004-12-01 01:29:20.000000000 -0500
+++ ddcxinfo-arch-0.8-new/Makefile	2009-03-02 00:01:28.000000000 -0500
@@ -1,4 +1,5 @@
 CFLAGS=-Wall -O2 -g #-DDEBUG #-DORIGINAL_LRMI_CODE_THAT_GOT_IFDEFED_OUT
+LDFLAGS= -L/usr/lib -lx86
 
 TARGETS=ddcprobe svgamodes ddcxinfo ddcxinfo-knoppix ddcxinfo-arch modetest libvbe.a
 
@@ -15,19 +16,19 @@
 install: $(TARGETS)
 	cp -a ddcprobe $(DESTDIR)/usr/sbin/ddcprobe
 
-ddcprobe: lrmi.o vesamode.o vbe.o ddcprobe.o
+ddcprobe: vesamode.o vbe.o ddcprobe.o
 
-svgamodes: lrmi.o vesamode.o vbe.o svgamodes.o
+svgamodes: vesamode.o vbe.o svgamodes.o
 
-ddcxinfo: lrmi.o vesamode.o vbe.o ddcxinfo.o
+ddcxinfo: vesamode.o vbe.o ddcxinfo.o
 
-ddcxinfo-knoppix: lrmi.o vesamode.o vbe.o ddcxinfo-knoppix.o
+ddcxinfo-knoppix: vesamode.o vbe.o ddcxinfo-knoppix.o
 
-ddcxinfo-arch: lrmi.o vesamode.o vbe.o ddcxinfo-arch.o
+ddcxinfo-arch: vesamode.o vbe.o ddcxinfo-arch.o
 
-modetest: lrmi.o vesamode.o vbe.o modetest.o
+modetest: vesamode.o vbe.o modetest.o
 
-libvbe.a: lrmi.o vesamode.o vbe.o
+libvbe.a: vesamode.o vbe.o
 	$(AR) cru $@ $^
 
 install-lib: $(prefix)/include/vbe.h $(prefix)/lib/libvbe.a
diff -Naur ddcxinfo-arch-0.8/bioscall.c ddcxinfo-arch-0.8-new/bioscall.c
--- ddcxinfo-arch-0.8/bioscall.c	2004-12-01 01:29:20.000000000 -0500
+++ ddcxinfo-arch-0.8-new/bioscall.c	2009-03-02 00:01:28.000000000 -0500
@@ -1,7 +1,7 @@
 #include <sys/types.h>
 #include <sys/io.h>
 #include <sys/stat.h>
-#include <sys/vm86.h>
+//#include <sys/vm86.h>
 #include <sys/syscall.h>
 #include <sys/mman.h>
 #include <ctype.h>
diff -Naur ddcxinfo-arch-0.8/ddcprobe.c ddcxinfo-arch-0.8-new/ddcprobe.c
--- ddcxinfo-arch-0.8/ddcprobe.c	2004-12-01 01:29:20.000000000 -0500
+++ ddcxinfo-arch-0.8-new/ddcprobe.c	2009-03-02 00:02:29.000000000 -0500
@@ -2,7 +2,7 @@
 #include <sys/types.h>
 #include <sys/io.h>
 #include <sys/stat.h>
-#include <sys/vm86.h>
+//#include <sys/vm86.h>
 #include <sys/syscall.h>
 #include <sys/mman.h>
 #include <assert.h>
@@ -16,7 +16,7 @@
 #include <netinet/in.h>
 #include "vbe.h"
 #include "vesamode.h"
-#include "lrmi.h"
+#include <libx86.h>
 //#ident "$Id: ddcprobe.c,v 1.10 2000/08/09 15:13:37 notting Exp $"
 
 char *snip(char *string)
diff -Naur ddcxinfo-arch-0.8/ddcxinfo-arch.c ddcxinfo-arch-0.8-new/ddcxinfo-arch.c
--- ddcxinfo-arch-0.8/ddcxinfo-arch.c	2004-12-01 01:29:20.000000000 -0500
+++ ddcxinfo-arch-0.8-new/ddcxinfo-arch.c	2009-03-02 00:01:28.000000000 -0500
@@ -2,7 +2,7 @@
 #include <stdlib.h>
 #include <string.h>
 #include "vbe.h"
-#include "lrmi.h"
+#include <libx86.h>
 /************* This is derived from ddcxinfo written by *****************\
 #ident "$Id: ddcxinfo.c,v 1.7 1999/08/24 01:08:47 nalin Exp $"
 modified by Klaus Knopper <knoppix@knopper.net> for KNOPPIX Feb. 2003
diff -Naur ddcxinfo-arch-0.8/svgamodes.c ddcxinfo-arch-0.8-new/svgamodes.c
--- ddcxinfo-arch-0.8/svgamodes.c	2004-12-01 01:29:20.000000000 -0500
+++ ddcxinfo-arch-0.8-new/svgamodes.c	2009-03-02 00:03:45.000000000 -0500
@@ -1,7 +1,7 @@
 #include <sys/types.h>
 #include <sys/io.h>
 #include <sys/stat.h>
-#include <sys/vm86.h>
+//#include <sys/vm86.h>
 #include <sys/syscall.h>
 #include <sys/mman.h>
 #include <assert.h>
@@ -15,7 +15,7 @@
 #include <netinet/in.h>
 #include "vbe.h"
 #include "vesamode.h"
-#include "lrmi.h"
+#include <libx86.h>
 #ident "$Id: svgamodes.c,v 1.5 1999/08/24 01:08:47 nalin Exp $"
 
 /*
diff -Naur ddcxinfo-arch-0.8/vbe.c ddcxinfo-arch-0.8-new/vbe.c
--- ddcxinfo-arch-0.8/vbe.c	2004-12-01 01:29:20.000000000 -0500
+++ ddcxinfo-arch-0.8-new/vbe.c	2009-03-02 00:01:28.000000000 -0500
@@ -8,7 +8,7 @@
 #include <assert.h>
 #include <limits.h>
 #include <ctype.h>
-#include "lrmi.h"
+#include <libx86.h>
 #include "vesamode.h"
 #include "vbe.h"
 #ident "$Id: vbe.c,v 1.9 2002/01/04 02:29:30 notting Exp $"
