diff -urN poppler-0.5.4~/configure.ac poppler-0.5.4/configure.ac
--- poppler-0.5.4~/configure.ac	2006-09-22 01:22:02.000000000 +0200
+++ poppler-0.5.4/configure.ac	2006-09-22 15:13:48.000000000 +0200
@@ -183,7 +183,7 @@
               enable_poppler_glib=$enableval,
               enable_poppler_glib="try")
 if test x$enable_poppler_glib = xyes; then
-  PKG_CHECK_MODULES(POPPLER_GLIB, gdk-2.0 >= 2.4.0 glib >= 2.6)
+  PKG_CHECK_MODULES(POPPLER_GLIB, gdk-2.0 >= 2.4.0 glib-2.0 >= 2.6)
 elif test x$enable_poppler_glib = xtry; then
   PKG_CHECK_MODULES(POPPLER_GLIB, gdk-2.0 >= 2.4.0,
                     [enable_poppler_glib="yes"],
