diff --exclude-from=/home/dang/.diffrc -u -ruN poppler-0.5.2.orig/glib/Makefile.am poppler-0.5.2/glib/Makefile.am
--- poppler-0.5.2.orig/glib/Makefile.am	2006-05-19 18:09:00.000000000 -0400
+++ poppler-0.5.2/glib/Makefile.am	2006-05-22 19:55:31.000000000 -0400
@@ -81,7 +81,7 @@
 	poppler-private.h
 
 libpoppler_glib_la_LIBADD =				\
-	$(top_builddir)/poppler/libpoppler.la		\
+	-lpoppler					\
 	$(POPPLER_GLIB_LIBS)				\
 	$(FREETYPE_LIBS)				\
 	$(cairo_libs)
@@ -94,7 +94,7 @@
        test-poppler-glib.c
 
 test_poppler_glib_LDADD =			\
-	$(top_builddir)/poppler/libpoppler.la	\
+	-lpoppler				\
 	libpoppler-glib.la			\
 	$(POPPLER_GLIB_LIBS)			\
 	$(FREETYPE_LIBS)			\
diff --exclude-from=/home/dang/.diffrc -u -ruN poppler-0.5.2.orig/Makefile.am poppler-0.5.2/Makefile.am
--- poppler-0.5.2.orig/Makefile.am	2006-01-10 11:09:33.000000000 -0500
+++ poppler-0.5.2/Makefile.am	2006-05-22 19:55:31.000000000 -0400
@@ -29,7 +29,7 @@
 utils_subdir = utils
 endif
 
-SUBDIRS = goo fofi $(splash_subdir) poppler $(utils_subdir) $(glib_subdir) $(qt_subdir) test $(qt4_subdir)
+SUBDIRS = $(glib_subdir) $(qt_subdir) $(qt4_subdir)
 
 EXTRA_DIST =					\
 	README-XPDF				\
@@ -40,9 +40,7 @@
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA =				\
-	poppler.pc				\
 	$(cairo_pc_file)			\
-	$(splash_pc_file)			\
 	$(glib_pc_file)				\
 	$(qt_pc_file)                           \
 	$(qt4_pc_file)
diff --exclude-from=/home/dang/.diffrc -u -ruN poppler-0.5.2.orig/qt/Makefile.am poppler-0.5.2/qt/Makefile.am
--- poppler-0.5.2.orig/qt/Makefile.am	2006-02-28 17:51:10.000000000 -0500
+++ poppler-0.5.2/qt/Makefile.am	2006-05-22 19:55:31.000000000 -0400
@@ -22,7 +22,7 @@
 	poppler-private.h
 
 libpoppler_qt_la_LIBADD = 			\
-	$(top_builddir)/poppler/libpoppler.la	\
+	-lpoppler				\
 	$(POPPLER_QT_LIBS)			\
 	$(FREETYPE_LIBS)
 
@@ -38,7 +38,7 @@
        test-poppler-qt.cpp
 
 test_poppler_qt_LDADD =				\
-	$(top_builddir)/poppler/libpoppler.la	\
+	-lpoppler				\
 	libpoppler-qt.la			\
 	$(POPPLER_QT_LIBS)			\
 	$(FREETYPE_LIBS)
diff --exclude-from=/home/dang/.diffrc -u -ruN poppler-0.5.2.orig/qt4/src/Makefile.am poppler-0.5.2/qt4/src/Makefile.am
--- poppler-0.5.2.orig/qt4/src/Makefile.am	2006-05-19 13:44:23.000000000 -0400
+++ poppler-0.5.2/qt4/src/Makefile.am	2006-05-22 19:56:44.000000000 -0400
@@ -27,7 +27,7 @@
 	poppler-private.h
 
 libpoppler_qt4_la_LIBADD = 			\
-	$(top_builddir)/poppler/libpoppler.la	\
+	-lpoppler			\
 	$(top_builddir)/poppler/libpoppler-arthur.la	\
 	$(POPPLER_QT4_LIBS)
 
diff --exclude-from=/home/dang/.diffrc -u -ruN poppler-0.5.2.orig/qt4/tests/Makefile.am poppler-0.5.2/qt4/tests/Makefile.am
--- poppler-0.5.2.orig/qt4/tests/Makefile.am	2006-01-23 09:43:39.000000000 -0500
+++ poppler-0.5.2/qt4/tests/Makefile.am	2006-05-22 19:55:31.000000000 -0400
@@ -7,7 +7,7 @@
 	$(POPPLER_QTTEST_CXXFLAGS)
 
 LDADDS =					\
-	$(top_builddir)/poppler/libpoppler.la	\
+	-lpoppler				\
 	$(top_builddir)/qt4/src/libpoppler-qt4.la	\
 	$(POPPLER_QT4_LIBS)
 
