diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9939167a..0fa5cd13 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -166,6 +166,7 @@ externalproject_add(qtwebengine
     BUILD_ALWAYS TRUE
     CMAKE_ARGS  ${libsCmakeArgs}
     BUILD_COMMAND ${CMAKE_COMMAND} -E echo "Starting $<CONFIG> build"
+    INSTALL_COMMAND ""
     ${coinBug699}
     COMMAND ${CMAKE_COMMAND} -E echo "$<CONFIG> build complete"
     # TODO: this is aworkaround for broken qt_lib_webengine*.pri files,
