diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8a3ff75c..14ca95a4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -149,6 +149,7 @@ externalproject_add(libs
     # and this leftover file gets sucked instead of install or src/mkspecs,
     # simply remove it, we need to implment a dry run in qt base to fix it.
     COMMAND ${CMAKE_COMMAND} -E remove_directory ../mkspecs
+    INSTALL_COMMAND ""
 )
 
 # Ensure the libs project is reconfigured if the root project is reconfigured, so that the
diff --git a/src/core/CMakeLists.txt b/src/core/CMakeLists.txt
index fd935fe1..f37ad3f6 100644
--- a/src/core/CMakeLists.txt
+++ b/src/core/CMakeLists.txt
@@ -674,28 +674,11 @@ if(QT_FEATURE_framework)
     )
 else()
     install(FILES ${localeFiles}
-        DESTINATION translations/qtwebengine_locales
+        DESTINATION ${INSTALL_DATADIR}/translations/qtwebengine_locales
         CONFIGURATIONS ${config}
     )
     install(FILES ${resourceFiles}
-        DESTINATION resources
-        CONFIGURATIONS ${config}
-    )
-    install(FILES ${icuFiles}
-        DESTINATION resources
-        CONFIGURATIONS ${config}
-    )
-# install (again) for superbuild
-    install(FILES ${localeFiles}
-        DESTINATION ${WEBENGINE_ROOT_BUILD_DIR}/translations/qtwebengine_locales
-        CONFIGURATIONS ${config}
-    )
-    install(FILES ${resourceFiles}
-        DESTINATION ${WEBENGINE_ROOT_BUILD_DIR}/resources
-        CONFIGURATIONS ${config}
-    )
-    install(FILES ${icuFiles}
-        DESTINATION ${WEBENGINE_ROOT_BUILD_DIR}/resources
+        DESTINATION ${INSTALL_DATADIR}/resources
         CONFIGURATIONS ${config}
     )
     if(NOT QT_WILL_INSTALL)
