diff -Nur davfs2-1.2.2.orig/configure davfs2-1.2.2/configure
--- davfs2-1.2.2.orig/configure	2007-07-07 11:28:15.000000000 +0200
+++ davfs2-1.2.2/configure	2007-08-25 11:56:54.000000000 +0200
@@ -5546,7 +5546,7 @@
 echo "${ECHO_T}$ne_cv_lib_neon" >&6; }
     if test "$ne_cv_lib_neon" = "yes"; then
        ne_cv_lib_neonver=no
-       for v in 24 25 26; do
+       for v in 24 25 26 27; do
           case $ne_libver in
           0.$v.*) ne_cv_lib_neonver=yes ;;
           esac
diff -Nur davfs2-1.2.2.orig/src/Makefile.in davfs2-1.2.2/src/Makefile.in
--- davfs2-1.2.2.orig/src/Makefile.in	2007-07-07 18:19:59.000000000 +0200
+++ davfs2-1.2.2/src/Makefile.in	2007-08-25 12:02:18.000000000 +0200
@@ -496,8 +496,8 @@
 	chmod u+s $(DESTDIR)$(sbindir)/mount.davfs; \
 	if test "$(sbindir)" != "$(ssbindir)"; then \
 	  $(mkinstalldirs) $(DESTDIR)$(ssbindir); \
-	  $(LN_S) -f $(DESTDIR)$(sbindir)/mount.davfs $(DESTDIR)$(ssbindir)/mount.davfs; \
-	  $(LN_S) -f $(DESTDIR)$(sbindir)/umount.davfs $(DESTDIR)$(ssbindir)/umount.davfs; \
+	  $(LN_S) -f $(sbindir)/mount.davfs $(DESTDIR)$(ssbindir)/mount.davfs; \
+	  $(LN_S) -f $(sbindir)/umount.davfs $(DESTDIR)$(ssbindir)/umount.davfs; \
 	fi
 
 uninstall-hook:
