--- cups-filters/Makefile.am	2020-11-17 21:25:27.000000000 +0100
+++ cups-filters/Makefile.am.new	2020-11-17 21:30:00.696488598 +0100
@@ -772,7 +772,8 @@
 
 bannertopdf_DEPENDENCIES = \
 	$(GETLINE) \
-	libfontembed.la
+	libfontembed.la \
+	libcupsfilters.la
 
 commandtoescpx_SOURCES = \
 	cupsfilters/driver.h \
@@ -947,7 +947,9 @@
 	$(STRCASESTR) \
 	$(CUPS_LIBS) \
 	$(LIBICONV)
-texttotext_DEPENDENCIES = $(STRCASESTR)
+texttotext_DEPENDENCIES = \
+	$(STRCASESTR) \
+	libppd.la
 
 pdftops_SOURCES = \
 	filter/pdftops.c
