unchanged:
--- klibc-1.5.orig/usr/klibc/arch/i386/MCONFIG	2007-03-04 02:52:10.000000000 +0100
+++ klibc-1.5/usr/klibc/arch/i386/MCONFIG	2008-01-27 16:06:08.000000000 +0100
@@ -31,3 +31,4 @@
 # calls, and work on the memory models for this architecture
 # 96 MB - normal binaries start at 128 MB
 KLIBCSHAREDFLAGS	= -Ttext 0x06000200
+KLIBCASMARCH = x86
unchanged:
--- klibc-1.5.orig/usr/klibc/arch/x86_64/MCONFIG	2007-03-04 02:52:10.000000000 +0100
+++ klibc-1.5/usr/klibc/arch/x86_64/MCONFIG	2008-01-27 16:05:43.000000000 +0100
@@ -36,4 +36,4 @@
 KLIBCSHAREDFLAGS     = -Ttext 0x00200200 -z max-page-size=0x100000
 
 # Additional asm- directories needed during installation
-ASMARCH = asm-i386
+KLIBCASMARCH = x86
only in patch2:
unchanged:
--- a/scripts/Kbuild.install	2007-03-04 02:52:10.000000000 +0100
+++ b/scripts/Kbuild.install    2007-11-04 15:43:28.000000000 +0100
@@ -84,6 +84,13 @@
 # 1) Create directories, install headers and man pages
 # 2) Tell that we now install binaries
 # 3) Install binaries by descending
+
+# Arch specific definitions for klibc
+include $(KLIBCSRC)/arch/$(KLIBCARCHDIR)/MCONFIG
+
+# include/asm-* architecture
+KLIBCASMARCH      ?= $(KLIBCARCH)
+
 .PHONY: header footer descend
 header:
 	$(Q)echo "  INSTALL headers + man pages to $(INSTALLROOT)$(INSTALLDIR)"
@@ -95,7 +102,7 @@
 	$(Q)mkdir -p $(INSTALLROOT)$(INSTALLDIR)/$(KCROSS)include
 	$(Q)mkdir -p $(INSTALLROOT)$(INSTALLDIR)/$(KCROSS)lib
 	$(Q)mkdir -p $(INSTALLROOT)$(INSTALLDIR)/$(KCROSS)bin
-	$(Q)set -e ; for d in linux scsi asm-$(KLIBCARCH) asm-generic $(ASMKLIBCARCH); do \
+	$(Q)set -e ; for d in linux scsi asm-$(KLIBCASMARCH) asm-generic $(ASMKLIBCARCH); do \
 	  mkdir -p $(INSTALLROOT)$(INSTALLDIR)/$(KLIBCCROSS)include/$$d ; \
 	  for r in $(KLIBCKERNELSRC)/include $(KLIBCKERNELOBJ)/include \
 	           $(KLIBCKERNELOBJ)/include2 ; do \
