--- startkde.cmake.old	2010-08-11 19:45:21.546666823 +0200
+++ startkde.cmake	2010-08-11 19:45:34.563333482 +0200
@@ -36,18 +36,6 @@
 # we have to unset this for Darwin since it will screw up KDE's dynamic-loading
 unset DYLD_FORCE_FLAT_NAMESPACE
 
-# Enable lightweight memory corruption checker if not already set
-# -- this is for trunk only, we remove it for releases
-if [ "x$MALLOC_CHECK_" = "x" ] && [ -x /lib/libc.so.6 ]; then
-    # Extract the first two components of the version from the output.
-    glibc_version=$(LC_ALL=C /lib/libc.so.6 | sed -e 's/[^0-9]*\([0-9]\.[0-9]\+\).*/\1/;s/\.\([0-9]\)$/.0\1/;q')
-
-    MALLOC_CHECK_=2 # Default to 2 unless glibc 2.9 or higher.
-    test $glibc_version \> 2.08 && MALLOC_CHECK_=3
-
-    export MALLOC_CHECK_
-fi
-
 # Boot sequence:
 #
 # kdeinit is used to fork off processes which improves memory usage
