Author: James R. Van Zandt <jrv@debian.org>
Description: Two of the chkrootkit messages have unnecessary backslashes (#498063)
--- a/chkrootkit
+++ b/chkrootkit
@@ -672,7 +672,7 @@ etc/ld.so.hash sbin/init.zk usr/lib/in.h
    if [ "`find ${LIBS} -name libproc.a 2> /dev/null`" != "" -a \
        "$SYSTEM" != "FreeBSD" ]
    then
-      echo "Possible t0rn v8 \(or variation\) rootkit installed"
+      echo "Possible t0rn v8 (or variation) rootkit installed"
    else
       if [ "${QUIET}" != "t" ]; then echo "nothing found"; fi
    fi
@@ -726,7 +726,7 @@ etc/ld.so.hash sbin/init.zk usr/lib/in.h
 
    if [ -d ${ROOTDIR}dev/ptyxx -o -r "${ROOTDIR}usr/lib/.ark?" -o \
         -d ${ROOTDIR}usr/doc/"... " ]; then
-      echo "Possible Ambient's rootkit \(ark\) installed"
+      echo "Possible Ambient's rootkit (ark) installed"
    else
       if [ "${QUIET}" != "t" ]; then echo "nothing found"; fi
    fi
