diff -u -r gnome-todo-3.20.0/src/Makefile.am gnome-todo-3.20.0-buildfix/src/Makefile.am
--- gnome-todo-3.20.0/src/Makefile.am	2016-03-14 23:07:22.000000000 +0100
+++ gnome-todo-3.20.0-buildfix/src/Makefile.am	2016-03-30 22:08:24.161223740 +0200
@@ -98,8 +98,8 @@
 	$(GNOME_TODO_WARN_LDFLAGS)
 
 gnome_todo_LDADD = \
-	$(GNOME_TODO_LIBS) \
-	$(top_builddir)/plugins/eds/libeds.a
+	$(top_builddir)/plugins/eds/libeds.a \
+	$(GNOME_TODO_LIBS)
 
 gnome_todo_DEPENDENCIES = \
 	$(top_builddir)/plugins/eds/libeds.a
