diff -up gnome-todo-3.26.2/src/gtd-task.c.libical-3.0 gnome-todo-3.26.2/src/gtd-task.c
--- gnome-todo-3.26.2/src/gtd-task.c.libical-3.0	2017-11-08 14:53:53.484294926 +0100
+++ gnome-todo-3.26.2/src/gtd-task.c	2017-11-08 14:53:57.208294874 +0100
@@ -778,7 +778,7 @@ gtd_task_set_complete (GtdTask  *task,
           dt->minute = g_date_time_get_minute (now);
           dt->second = g_date_time_get_seconds (now);
           dt->is_date = 0;
-          dt->is_utc = 1;
+          dt->zone = icaltimezone_get_utc_timezone ();
 
           /* convert timezone
            *
