From de2daf8d361ce2387514c846fe3d741f3d7572ab Mon Sep 17 00:00:00 2001
From: Peter Hutterer <peter.hutterer@who-t.net>
Date: Fri, 25 Mar 2022 14:37:28 +1000
Subject: [PATCH] meson.build: install the mouse-dpi-tool man page

Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>
---
 meson.build | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 63504ac..33365c9 100644
--- a/meson.build
+++ b/meson.build
@@ -130,7 +130,8 @@ executable('libevdev-tweak-device',
 	   dependencies: dep_libevdev,
 	   install: true)
 install_man('tools/libevdev-tweak-device.1',
-            'tools/touchpad-edge-detector.1')
+	    'tools/touchpad-edge-detector.1',
+	    'tools/mouse-dpi-tool.1')
 
 # tests
 dep_check = dependency('check', version: '>= 0.9.9',
-- 
GitLab

