--- config/gnu-flags~	2008-03-09 14:43:45.000000000 +0000
+++ config/gnu-flags	2008-03-09 14:45:11.000000000 +0000
@@ -157,6 +157,10 @@
         # later versions of gcc will fix this bug... - QAK - 2003/10/20
         PROD_CFLAGS="-O"
         ;;
+      gcc-4.3*)
+        # Make check segfaults in the long long test with -O and -O2
+        PROD_CFLAGS=""
+        ;;
       gcc-4*)
         # Be optimistic about future versions of gcc.. :-) - QAK - 2003/10/20
         PROD_CFLAGS="-O3"
@@ -166,7 +170,6 @@
         ;;
     esac
 
-    PROD_CFLAGS="$PROD_CFLAGS -fomit-frame-pointer -finline-functions"
     PROD_CPPFLAGS=
 
     # Debug
@@ -191,7 +194,7 @@
 # Closer to the gcc 4.2 release, we should check for additional flags to
 # include and break it out into it's own section, like the other versions
 # below. -QAK
-  gcc-4.[12]*)
+  gcc-4.[123]*)
     # Replace -ansi flag with -std=c99 flag
     H5_CFLAGS="`echo $H5_CFLAGS | sed -e 's/-ansi/-std=c99/g'`"
 
--- c++/src/H5PropList.cpp.orig	2008-03-08 23:17:27.000000000 +0000
+++ c++/src/H5PropList.cpp	2008-03-08 23:17:52.000000000 +0000
@@ -426,7 +426,7 @@
    if (temp_str != NULL)
    {
       H5std_string class_name = H5std_string(temp_str);
-      free(temp_str);
+      delete(temp_str);
       return(class_name);
    }
    else
