From b3e4387351c835766f96796a20d94971afea7d3b Mon Sep 17 00:00:00 2001
From: Tomasz Torcz <tomek@pipebreaker.pl>
Date: Tue, 25 Feb 2014 12:43:55 +0100
Subject: [PATCH] fix typo in iDRAC network interface name: irdac->idrac

---
 hwdb/20-net-ifname.hwdb | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hwdb/20-net-ifname.hwdb b/hwdb/20-net-ifname.hwdb
index 29d2633..2408dc1 100644
--- a/hwdb/20-net-ifname.hwdb
+++ b/hwdb/20-net-ifname.hwdb
@@ -2,4 +2,4 @@
 
 # Dell iDRAC Virtual USB NIC
 usb:v413CpA102*
- ID_NET_NAME_FROM_DATABASE=irdac
+ ID_NET_NAME_FROM_DATABASE=idrac
-- 
1.9.0

