diff -rupN ardour-2.8.11.old/libs/ardour/enums.cc ardour-2.8.11/libs/ardour/enums.cc
--- ardour-2.8.11.old/libs/ardour/enums.cc	2010-04-20 13:28:37.000000000 -0400
+++ ardour-2.8.11/libs/ardour/enums.cc	2011-02-13 11:23:02.000000000 -0500
@@ -31,7 +31,8 @@
 
 using namespace std;
 using namespace PBD;
-using namespace ARDOUR;
+namespace ARDOUR
+{
 
 void
 setup_enum_writer ()
@@ -362,3 +363,5 @@ setup_enum_writer ()
 	REGISTER (_Track_FreezeState);
 	
 }
+
+}
diff -rupN ardour-2.8.11.old/libs/rubberband/rubberband/RubberBandStretcher.h ardour-2.8.11/libs/rubberband/rubberband/RubberBandStretcher.h
--- ardour-2.8.11.old/libs/rubberband/rubberband/RubberBandStretcher.h	2009-03-26 10:25:40.000000000 -0400
+++ ardour-2.8.11/libs/rubberband/rubberband/RubberBandStretcher.h	2011-02-09 21:41:04.000000000 -0500
@@ -19,6 +19,7 @@
 #define RUBBERBAND_API_MAJOR_VERSION 2
 #define RUBBERBAND_API_MINOR_VERSION 0
 
+#include <cstddef>
 #include <vector>
 
 /**
diff -rupN ardour-2.8.11.old/libs/surfaces/wiimote/wiimote.cc ardour-2.8.11/libs/surfaces/wiimote/wiimote.cc
--- ardour-2.8.11.old/libs/surfaces/wiimote/wiimote.cc	2009-02-24 07:38:19.000000000 -0500
+++ ardour-2.8.11/libs/surfaces/wiimote/wiimote.cc	2011-02-13 11:23:24.000000000 -0500
@@ -177,7 +177,7 @@ wiimote_discovery:
 	std::cerr << "Wiimote: discovering, press 1+2" << std::endl;
 
  	while (!wiimote_handle && !main_thread_quit) {
-		bdaddr = *BDADDR_ANY;
+		bdaddr = (bdaddr_t) {{0, 0, 0, 0, 0, 0}};
 		callback_thread_registered_for_ardour = false;
 		wiimote_handle = cwiid_open(&bdaddr, 0);
 
