diff -Nur dosfstools-2.10/dosfsck/boot.c dosfstools-2.10-fixed/dosfsck/boot.c
--- dosfstools-2.10/dosfsck/boot.c	2003-06-14 22:03:20.000000000 +0200
+++ dosfstools-2.10-fixed/dosfsck/boot.c	2004-02-11 21:15:48.000000000 +0100
@@ -14,6 +14,9 @@
 #include "io.h"
 #include "boot.h"
 
+#ifndef MSDOS_FAT12
+#  define MSDOS_FAT12 4084 /* taken from 2.4 linux/msdos_fs.h */
+#endif
 
 #define ROUND_TO_MULTIPLE(n,m) ((n) && (m) ? (n)+(m)-1-((n)-1)%(m) : 0)
     /* don't divide by zero */
diff -Nur dosfstools-2.10/dosfsck/common.h dosfstools-2.10-fixed/dosfsck/common.h
--- dosfstools-2.10/dosfsck/common.h	1998-04-28 13:14:25.000000000 +0200
+++ dosfstools-2.10-fixed/dosfsck/common.h	2004-02-11 21:15:01.000000000 +0100
@@ -6,6 +6,9 @@
 #ifndef _COMMON_H
 #define _COMMON_H
 
+#include <asm/types.h>
+#include <asm/byteorder.h>
+
 void die(char *msg,...) __attribute((noreturn));
 
 /* Displays a prinf-style message and terminates the program. */
diff -Nur dosfstools-2.10/dosfsck/file.c dosfstools-2.10-fixed/dosfsck/file.c
--- dosfstools-2.10/dosfsck/file.c	2003-06-14 22:03:20.000000000 +0200
+++ dosfstools-2.10-fixed/dosfsck/file.c	2004-02-11 21:13:53.000000000 +0100
@@ -15,9 +15,9 @@
 #define _LINUX_STAT_H		/* hack to avoid inclusion of <linux/stat.h> */
 #define _LINUX_STRING_H_	/* hack to avoid inclusion of <linux/string.h>*/
 #define _LINUX_FS_H             /* hack to avoid inclusion of <linux/fs.h> */
+#include "common.h"
 #include <linux/msdos_fs.h>
 
-#include "common.h"
 #include "file.h"
 
 
diff -Nur dosfstools-2.10/dosfsck/io.c dosfstools-2.10-fixed/dosfsck/io.c
--- dosfstools-2.10/dosfsck/io.c	2003-06-14 22:03:20.000000000 +0200
+++ dosfstools-2.10-fixed/dosfsck/io.c	2004-02-11 21:14:24.000000000 +0100
@@ -21,8 +21,8 @@
 #include <fcntl.h>
 #include <linux/fd.h>
 
-#include "dosfsck.h"
 #include "common.h"
+#include "dosfsck.h"
 #include "io.h"
 
 
diff -Nur dosfstools-2.10/mkdosfs/mkdosfs.c dosfstools-2.10-fixed/mkdosfs/mkdosfs.c
--- dosfstools-2.10/mkdosfs/mkdosfs.c	2003-06-14 22:07:08.000000000 +0200
+++ dosfstools-2.10-fixed/mkdosfs/mkdosfs.c	2004-02-11 21:13:09.000000000 +0100
@@ -62,6 +62,7 @@
 #include <sys/ioctl.h>
 #include <sys/stat.h>
 #include <sys/types.h>
+#include <asm/types.h>
 #include <unistd.h>
 #include <time.h>
 #include <errno.h>
