From 4d34b379411f781a494b4775260d550ff61e22e4 Mon Sep 17 00:00:00 2001
From: Tom Gundersen <teg@jklm.no>
Date: Fri, 16 Mar 2012 01:34:10 +0100
Subject: [PATCH] README: point to the correct location of the binary

---
 README   |    2 +-
 1 file changed, 1 insertions(+), 1 deletions(-)

diff --git a/README b/README
index fa1d730..1906abb 100644
--- a/README
+++ b/README
@@ -6,7 +6,7 @@ Bootchart - a 'startup' graphing tool
 Bootchart is a tool, usually run at system startup, that collects and graphs
 the CPU and disk load of the system as it works. The output of bootchart is
 an SVG graph. Normally, bootchart is invoked as `bootchartd` by the kernel
-by passing "init=/sbin/bootchartd" to the kernel. Bootchart will then fork
+by passing "init=/usr/bin/bootchartd" to the kernel. Bootchart will then fork
 init off to resume normal system startup, while monitoring and logging
 startup information in the background.
 
-- 
1.7.9.5

