diff --git a/src/Makefile.am b/src/Makefile.am
index d6178df..66f377b 100755
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -86,6 +86,7 @@ libp2edrvstd_la_SOURCES=			\
         $(sample_drivers_src)                   
 libp2edrvstd_la_LIBADD=-L. libpstoedit.la
 libp2edrvstd_la_LDFLAGS=-no-undefined -module
+libp2edrvstd_la_DEPENDENCIES = libpstoedit.la
 
 #libp2edrvplugins_la_SOURCES=drvmif.cpp drvcgm.cpp drvrtf.cpp drvwwmf.cpp
 #libp2edrvplugins_la_LIBADD=
@@ -95,15 +96,18 @@ libp2edrvstd_la_LDFLAGS=-no-undefined -module
 libp2edrvlplot_la_SOURCES=drvlplot.cpp drvlplot.h initlibrary.cpp
 libp2edrvlplot_la_LIBADD=$(LIBPLOTTER_LDFLAGS) -L. libpstoedit.la
 libp2edrvlplot_la_LDFLAGS=-no-undefined -module
+libp2edrvlplot_la_DEPENDENCIES = libpstoedit.la
 
 libp2edrvswf_la_SOURCES=drvswf.cpp drvswf.h initlibrary.cpp
 libp2edrvswf_la_LIBADD=$(LIBMING_LDFLAGS) -L. libpstoedit.la
 libp2edrvswf_la_LDFLAGS=-no-undefined -module
+libp2edrvswf_la_DEPENDENCIES = libpstoedit.la
 
 # note the __ stand for ++ - automake treats all no alphanums chars as _
 libp2edrvmagick___la_SOURCES=drvmagick++.cpp drvmagick++.h initlibrary.cpp
 libp2edrvmagick___la_LIBADD=-L. libpstoedit.la
 libp2edrvmagick___la_LDFLAGS=-no-undefined ${LIBMAGICK_LDFLAGS}  -module
+libp2edrvmagick___la_DEPENDENCIES = libpstoedit.la
 # moved to configure.ac because needed only for g++ libp2edrvmagick___la_CPPFLAGS=$(LIBMAGICK_CFLAGS) $(AM_CPPFLAGS) -Wno-long-long
 
 if USE_EMFBYSOURCE
@@ -113,6 +117,7 @@ else
 libp2edrvwmf_la_SOURCES=drvwmf.cpp drvwmf.h initlibrary.cpp
 libp2edrvwmf_la_LDFLAGS=-no-undefined $(LIBEMF_LDFLAGS) -L. libpstoedit.la -module
 endif
+libp2edrvwmf_la_DEPENDENCIES = libpstoedit.la
 
 libpstoedit_la_SOURCES =			\
 	dynload.cpp dynload.h			\
@@ -135,6 +140,7 @@ libpstoedit_la_LDFLAGS = -no-undefined
 
 pstoedit_SOURCES = cmdmain.cpp
 pstoedit_LDFLAGS = -no-undefined -L. libpstoedit.la ${LIBLD_LDFLAGS}
+pstoedit_DEPENDENCIES = libpstoedit.la
 
 pkginclude_HEADERS =				\
 	pstoedit.h				\
