diff -ur libtunepimp-0.5.3.orig/plugins/mp4/mp4.cpp libtunepimp-0.5.3/plugins/mp4/mp4.cpp
--- libtunepimp-0.5.3.orig/plugins/mp4/mp4.cpp	2009-07-10 15:50:20.000000000 +0300
+++ libtunepimp-0.5.3/plugins/mp4/mp4.cpp	2009-07-10 15:51:18.000000000 +0300
@@ -28,7 +28,7 @@
 #include <string.h>
 #include <stdio.h>
 #include <stdlib.h>
-#include <mp4.h>
+#include <mp4v2/mp4v2.h>
 #include "metadata.h"
 #include "plugin.h"
 #ifndef WIN32
@@ -232,7 +232,7 @@
   
   strcpy(mdata->fileFormat, fileName + strlen(fileName) - 3); 
   
-  if (!MP4Close(mp4file))
+  MP4Close(mp4file);
     return 0;
   
   return 1;
@@ -316,7 +316,7 @@
   sprintf(temp, "%d", mdata->nonAlbum);  
   MP4SetMetadataFreeForm(mp4file, "MusicBrainz Non-Album", (u_int8_t *)temp, strlen(temp) + 1);
   
-  if (!MP4Close(mp4file))
+  MP4Close(mp4file);
     return 0;
 
 #ifndef WIN32
