diff --git a/modules/misc/notify/notify.c b/modules/misc/notify/notify.c
index 1e97c9e..b663caa 100644
--- a/modules/misc/notify/notify.c
+++ b/modules/misc/notify/notify.c
@@ -38,6 +38,10 @@
 #include <gdk-pixbuf/gdk-pixbuf.h>
 #include <libnotify/notify.h>
 
+#ifndef NOTIFY_CHECK_VERSION
+#define NOTIFY_CHECK_VERSION(x,y,z) 0
+#endif
+
 /*****************************************************************************
  * Module descriptor
  ****************************************************************************/
@@ -312,7 +316,12 @@ static int Notify( vlc_object_t *p_this, const char *psz_temp, GdkPixbuf *pix,
     }
 
     notification = notify_notification_new( _("Now Playing"),
-                                            psz_temp, NULL, NULL );
+                                            psz_temp, NULL
+#if NOTIFY_CHECK_VERSION (0, 7, 0)
+					    );
+#else
+					    , NULL );
+#endif
     notify_notification_set_timeout( notification,
                                 var_InheritInteger(p_this, "notify-timeout") );
     notify_notification_set_urgency( notification, NOTIFY_URGENCY_LOW );
