
diff -up antlrworks-1.4.3/src/aw/org/antlr/works/grammar/decisiondfa/DecisionDFAEngine.java.orig antlrworks-1.4.3/src/aw/org/antlr/works/grammar/decisiondfa/DecisionDFAEngine.java
--- antlrworks-1.4.3/src/aw/org/antlr/works/grammar/decisiondfa/DecisionDFAEngine.java.orig	2011-08-08 11:53:50.000000000 +0200
+++ antlrworks-1.4.3/src/aw/org/antlr/works/grammar/decisiondfa/DecisionDFAEngine.java	2012-02-02 13:53:50.536902161 +0100
@@ -106,13 +106,13 @@ public class DecisionDFAEngine {
         if(g == null) return;
 
         if(g.decisionsWhoseDFAsUsesSemPreds != null) {
-            for(DFA dfa : g.decisionsWhoseDFAsUsesSemPreds) {
+            for(DFA dfa : (List<DFA>) g.decisionsWhoseDFAsUsesSemPreds) {
                 usesSemPreds.add(dfa.getDecisionNumber());
             }
         }
 
         if(g.decisionsWhoseDFAsUsesSynPreds != null) {
-            for(DFA dfa : g.decisionsWhoseDFAsUsesSynPreds) {
+            for(DFA dfa : (List<DFA>) g.decisionsWhoseDFAsUsesSynPreds) {
                 usesSynPreds.add(dfa.getDecisionNumber());
             }
         }
diff -up antlrworks-1.4.3/src/aw/org/antlr/works/grammar/engine/GrammarPropertiesImpl.java.orig antlrworks-1.4.3/src/aw/org/antlr/works/grammar/engine/GrammarPropertiesImpl.java
--- antlrworks-1.4.3/src/aw/org/antlr/works/grammar/engine/GrammarPropertiesImpl.java.orig	2011-08-08 11:53:50.000000000 +0200
+++ antlrworks-1.4.3/src/aw/org/antlr/works/grammar/engine/GrammarPropertiesImpl.java	2012-02-02 13:54:24.610451213 +0100
@@ -400,7 +400,7 @@ public class GrammarPropertiesImpl imple
         Grammar g = antlrEngine.getDefaultGrammar();
         if(g != null) {
             names.add(g.getRecognizerName());
-            for(Grammar gd : g.getDelegates()) {
+            for(Grammar gd : (List<Grammar>) g.getDelegates()) {
                 names.add(gd.getRecognizerName());
             }
         }
@@ -408,7 +408,7 @@ public class GrammarPropertiesImpl imple
         Grammar lexer = antlrEngine.getLexerGrammar();
         if(lexer != null) {
             names.add(lexer.getRecognizerName());
-            for(Grammar gd : lexer.getDelegates()) {
+            for(Grammar gd : (List<Grammar>) lexer.getDelegates()) {
                 names.add(gd.getRecognizerName());
             }
         }
