--- PyQt5_gpl-5.11.2/configure.py.orig	2018-07-08 16:21:57.730616329 +0000
+++ PyQt5_gpl-5.11.2/configure.py	2018-07-08 16:22:23.584016854 +0000
@@ -1736,6 +1736,7 @@
         # The command to run to generate the .dist-info directory.
         distinfo_dir = os.path.join(target_config.pyqt_module_dir,
                 'PyQt5-' + PYQT_VERSION_STR + '.dist-info')
+        distinfo_dir = '$(DESTDIR)' + distinfo_dir
         mk_distinfo = sys.executable + ' mk_distinfo.py $(INSTALL_ROOT)' + distinfo_dir + ' installed.txt'
 
         out_f.write('''
