--- a/tests/test-enhanced-mixer.cpp
+++ b/tests/test-enhanced-mixer.cpp
@@ -77,7 +77,7 @@
         exit(0);
     }
 
-    int errno = 0;
+    extern int errno;
     char* tail;
     int node_id = strtol( argv[1], &tail, 0 );
     int fb_id   = strtol( argv[2], &tail, 0 );
--- a/tests/test-mixer.cpp
+++ b/tests/test-mixer.cpp
@@ -227,7 +227,7 @@
         exit(0);
     }
 
-    int errno = 0;
+    extern int errno;
     char* tail;
     int port = strtol( argv[1], &tail, 0 );
     int node_id = strtol( argv[2], &tail, 0 );
--- a/tests/test-pan.cpp
+++ b/tests/test-pan.cpp
@@ -131,7 +131,7 @@
         exit(0);
     }
 
-    int errno = 0;
+    extern int errno;
     char* tail;
     int node_id = strtol( argv[1], &tail, 0 );
     int fb_id   = strtol( argv[2], &tail, 0 );
--- a/tests/test-volume.cpp
+++ b/tests/test-volume.cpp
@@ -127,7 +127,7 @@
         exit(0);
     }
 
-    int errno = 0;
+    extern int errno;
     char* tail;
     int node_id = strtol( argv[1], &tail, 0 );
     int fb_id   = strtol( argv[2], &tail, 0 );
