From fba7e03c4466f54424a196f646f865909ceb5fc2 Mon Sep 17 00:00:00 2001
From: Christian Hesse <mail@eworm.de>
Date: Wed, 19 Jul 2017 14:38:43 +0200
Subject: [PATCH 2/3] include config.h
---
 src/libmpd-strfsong.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/libmpd-strfsong.c b/src/libmpd-strfsong.c
index 7d47bed..76fa3ff 100644
--- a/src/libmpd-strfsong.c
+++ b/src/libmpd-strfsong.c
@@ -28,6 +28,7 @@
 #include <unistd.h>
 #include <string.h>
 #include <glib.h>
+#include <config.h>
 #include "libmpd.h"
 #include "libmpd-internal.h"
 
