From 69c2d24a0a71b1c50a56beda90febdae97aa3128 Mon Sep 17 00:00:00 2001
From: Aleix Pol <aleixpol@kde.org>
Date: Tue, 9 Jan 2018 18:21:44 +0100
Subject: No need to manage these pointers anymore

---
 libdiscover/backends/PackageKitBackend/PackageKitNotifier.cpp | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/libdiscover/backends/PackageKitBackend/PackageKitNotifier.cpp b/libdiscover/backends/PackageKitBackend/PackageKitNotifier.cpp
index fdabd81..0113a06 100644
--- a/libdiscover/backends/PackageKitBackend/PackageKitNotifier.cpp
+++ b/libdiscover/backends/PackageKitBackend/PackageKitNotifier.cpp
@@ -225,7 +225,6 @@ void PackageKitNotifier::refreshDatabase()
         m_refresher = PackageKit::Daemon::refreshCache(false);
         connect(m_refresher.data(), &PackageKit::Transaction::finished, this, [this]() {
             recheckSystemUpdateNeeded();
-            delete m_refresher;
         });
     }
 
