diff -Naur old/src/decoder/ffmpeg_decoder_plugin.c new/src/decoder/ffmpeg_decoder_plugin.c
--- old/src/decoder/ffmpeg_decoder_plugin.c	2013-01-07 11:39:40.000000000 +1100
+++ new/src/decoder/ffmpeg_decoder_plugin.c	2013-03-05 21:54:27.116585462 +1100
@@ -395,6 +395,11 @@
 #endif
 		return SAMPLE_FORMAT_S32;
 
+#if LIBAVUTIL_VERSION_INT >= AV_VERSION_INT(51,17,0)
+	case AV_SAMPLE_FMT_FLTP:
+		return SAMPLE_FORMAT_FLOAT;
+#endif
+
 	default:
 		break;
 	}
