--- src/anjuta-app.c~	2012-06-01 03:14:23.000000000 -0500
+++ src/anjuta-app.c	2012-07-25 08:15:37.289630966 -0500
@@ -501,7 +501,7 @@
     g_list_free(plugins_dirs);
 
     /* Preferences */
-    app->preferences = anjuta_preferences_new(app->plugin_manager);
+    app->preferences = anjuta_preferences_new(app->plugin_manager, PREF_SCHEMA);
     g_object_add_weak_pointer(G_OBJECT (app->preferences), (gpointer) &app->preferences);
 
     g_signal_connect(app->settings, "changed::" GDL_STYLE, G_CALLBACK(on_gdl_style_changed), app);
