commit 512319738fcf9819f375f20441f09a7b1dff5168
Author: Adriano dos Santos Fernandes <adrianosf@gmail.com>
Date:   Wed May 26 10:41:25 2021 -0300

    Replace removed RE2 method.

diff --git a/src/common/SimilarToRegex.cpp b/src/common/SimilarToRegex.cpp
index 3a62b245e3..0b6e533018 100644
--- a/src/common/SimilarToRegex.cpp
+++ b/src/common/SimilarToRegex.cpp
@@ -108,7 +108,7 @@ namespace
 			options.set_log_errors(false);
 			options.set_dot_nl(true);
 			options.set_case_sensitive(!(flags & COMP_FLAG_CASE_INSENSITIVE));
-			options.set_utf8(!(flags & COMP_FLAG_LATIN));
+			options.set_encoding(flags & COMP_FLAG_LATIN ? RE2::Options::EncodingLatin1 : RE2::Options::EncodingUTF8);
 
 			re2::StringPiece sp((const char*) re2PatternStr.c_str(), re2PatternStr.length());
 			regexp = FB_NEW_POOL(pool) RE2(sp, options);
@@ -759,7 +759,7 @@ namespace
 			options.set_log_errors(false);
 			options.set_dot_nl(true);
 			options.set_case_sensitive(!(flags & COMP_FLAG_CASE_INSENSITIVE));
-			options.set_utf8(!(flags & COMP_FLAG_LATIN));
+			options.set_encoding(flags & COMP_FLAG_LATIN ? RE2::Options::EncodingLatin1 : RE2::Options::EncodingUTF8);
 
 			re2::StringPiece sp((const char*) finalRe2Pattern.c_str(), finalRe2Pattern.length());
 			regexp = FB_NEW_POOL(pool) RE2(sp, options);
