diff -up asymptote-2.32/runlabel.in.eps2write asymptote-2.32/runlabel.in
--- asymptote-2.32/runlabel.in.eps2write	2015-01-06 12:56:28.774952176 -0500
+++ asymptote-2.32/runlabel.in	2015-01-06 12:57:34.174504920 -0500
@@ -111,7 +111,7 @@ array *readpath(const string& psname, bo
 #else
   const string null="/dev/null";
 #endif
-  cmd.push_back("-sDEVICE=epswrite");
+  cmd.push_back("-sDEVICE=eps2write");
   cmd.push_back("-sOutputFile="+null);
   cmd.push_back(stripDir(psname));
   iopipestream gs(cmd,"gs","Ghostscript");
@@ -309,7 +309,7 @@ patharray2 *_texpath(stringarray *s, pen
       cmd.push_back("-dNOPAUSE");
       cmd.push_back("-dBATCH");
       if(safe) cmd.push_back("-dSAFER");
-      cmd.push_back("-sDEVICE=epswrite");
+      cmd.push_back("-sDEVICE=eps2write");
       cmd.push_back("-sOutputFile=-");
       cmd.push_back(pdfname);
       iopipestream gs(cmd,"gs","Ghostscript");
@@ -398,7 +398,7 @@ patharray2 *textpath(stringarray *s, pen
   cmd2.push_back("-dBATCH");
   cmd2.push_back("-P");
   if(safe) cmd2.push_back("-dSAFER");
-  cmd2.push_back("-sDEVICE=epswrite");
+  cmd2.push_back("-sDEVICE=eps2write");
   cmd2.push_back("-sOutputFile=-");
   cmd2.push_back("-");
   iopipestream gs(cmd2,"gs","Ghostscript");
