From ebd9de96878c224145c0b1d7bf21c7151c0563bf Mon Sep 17 00:00:00 2001
From: Eugen Dedu <Eugen.Dedu@pu-pm.univ-fcomte.fr>
Date: Tue, 19 Jul 2011 08:15:52 +0000
Subject: Fix generation of po translation files

Bug introduced by commit 2bfaf29.
---
diff --git a/configure.ac b/configure.ac
index 63d8685..eefae22 100644
--- a/configure.ac
+++ b/configure.ac
@@ -770,7 +770,9 @@ AM_CONDITIONAL(DEBUG, test "x$has_debug" = "xyes")
 dnl #########################################################################
 dnl  Support for internationalization
 dnl ########################################################################
-AC_DEFINE(GETTEXT_PACKAGE, "[AC_PACKAGE_NAME]",[fix])
+GETTEXT_PACKAGE=AC_PACKAGE_NAME
+AC_SUBST(GETTEXT_PACKAGE)  # needed by gettext in po directory
+AC_DEFINE_UNQUOTED(GETTEXT_PACKAGE, "$GETTEXT_PACKAGE",[fix])
 AM_GLIB_GNU_GETTEXT
 
 
--
cgit v0.9
