diff --git i/docs/reference/libtracker-sparql/meson.build w/docs/reference/libtracker-sparql/meson.build
index b6dad9c9a..f027ce697 100644
--- i/docs/reference/libtracker-sparql/meson.build
+++ w/docs/reference/libtracker-sparql/meson.build
@@ -69,7 +69,10 @@ hotdoc.generate_doc(docs_name,
   index: 'index.md',
   gi_index: 'gi-index.md',
   gi_smart_index: true,
-  gi_c_source_roots: [sparqlinc],
+  gi_c_source_roots: [
+    join_paths(meson.project_build_root(), 'src/libtracker-sparql'),
+    join_paths(meson.project_source_root(), 'src/libtracker-sparql'),
+  ],
   dependencies: [tracker_sparql_dep, base_ontology_docs, nepomuk_ontology_docs],
   extra_assets: [join_paths(meson.current_source_dir(), 'images')],
   html_theme: 'https://github.com/hotdoc/hotdoc_lumen_theme/releases/download/0.13.2/hotdoc_lumen_theme-0.13.2.tar.xz?sha256=5721189b7e985f27381ee20137f4a9003049a70a75ab1221a69fd04d27e752bc',
