diff --git i/tests/test-common.c w/tests/test-common.c
index a6339415..3b8880e3 100644
--- i/tests/test-common.c
+++ w/tests/test-common.c
@@ -225,23 +225,23 @@ attribute_from_string (const char *string)
 {
   char *s, *p;
   PangoAttribute *attr;
-  long start, end;
+  long long start, end;
   GEnumClass *class;
   int i;
   PangoColor color;
 
   s = string;
   g_assert (*s == '[');
 
   s++;
-  start = strtol (s, &p, 10);
+  start = strtoll (s, &p, 10);
   g_assert (p > s);
   g_assert (*p == ',');
   s = p + 1;
 
   g_assert (start >= 0);
 
-  end = strtol (s, &p, 10);
+  end = strtoll (s, &p, 10);
   g_assert (p > s);
   g_assert (*p == ']');
   s = p + 1;
diff --git i/tests/test-font.c w/tests/test-font.c
index e2147485..34689d5f 100644
--- i/tests/test-font.c
+++ w/tests/test-font.c
@@ -346,9 +346,16 @@ test_font_models (void)
           g_assert_true (pango_font_face_get_family (PANGO_FONT_FACE (obj2)) == (PangoFontFamily *)obj);
 
           pango_font_face_list_sizes (PANGO_FONT_FACE (obj2), &sizes, &n_sizes);
-          /* no more bitmap fonts */
-          g_assert_null (sizes);
-          g_assert_cmpint (n_sizes, ==, 0);
+
+          if (n_sizes == 0)
+            {
+              g_assert_null (sizes);
+            }
+          else
+            {
+              g_assert_nonnull (sizes);
+              g_free (sizes);
+            }
 
           g_object_unref (obj2);
         }
