--- officecfg/registry/data/org/openoffice/Office/Common.xcu	2010-03-16 11:45:04.000000000 +0000
+++ officecfg/registry/data/org/openoffice/Office/Common.xcu.new	2010-03-16 12:45:06.711728908 +0000
@@ -568,7 +568,7 @@
   </node>
   <node oor:name="Misc">
     <prop oor:name="UseSystemFileDialog">
-      <value install:module="UseOOoFileDialogs">false</value>
+      <value install:module="UseOOoFileDialogs">true</value>
       <value install:module="gconflockdown" oor:external=
          "com.sun.star.configuration.backend.GconfBackend UseSystemFileDialog"/>
     </prop>
