--- obex-data-server-0.3.4.old/src/ods-bluez.c	2008-06-27 16:21:14.000000000 +0100
+++ obex-data-server-0.3.4/src/ods-bluez.c	2008-09-12 18:17:27.000000000 +0100
@@ -121,11 +121,11 @@ ods_bluez_init (OdsBluez *bluez)
 	
 	bluez->priv->manager_proxy = dbus_g_proxy_new_for_name (klass->connection,
 															"org.bluez", 
-															"/org/bluez", 
+															"/", 
 															"org.bluez.Manager");
 	if (!dbus_g_proxy_call (bluez->priv->manager_proxy, "DefaultAdapter", &error, 
 								G_TYPE_INVALID,
-								G_TYPE_STRING, &adapter_object, 
+								DBUS_TYPE_G_OBJECT_PATH, &adapter_object,
 								G_TYPE_INVALID)) {
 		g_warning("Unable to connect to dbus: %s", error->message);
 		g_clear_error (&error);
@@ -314,7 +314,7 @@ get_remote_service_record_cb (DBusGProxy
 		goto err;
 	}
 	
-	sdp_record = sdp_extract_pdu ((uint8_t *)record_array->data, &scanned);
+	sdp_record = sdp_extract_pdu ((uint8_t *)record_array->data, record_array->len, &scanned);
 	
 	/* get channel for this service */
 	if (sdp_get_access_protos (sdp_record, &protos) != 0) {
