--- mail-notification-5.4/jbsrc/lib/src/extras/jb-evolution-plugin.c.orig	2009-10-12 23:15:03.000000000 +0200
+++ mail-notification-5.4/jbsrc/lib/src/extras/jb-evolution-plugin.c	2009-10-12 23:17:26.000000000 +0200
@@ -41,7 +41,7 @@
   if (! minversion)
     minversion = "2.12";
 
-  packages = g_strdup_printf("evolution-plugin >= %s", minversion);
+  packages = g_strdup_printf("evolution-plugin >= %s gtkhtml-editor", minversion);
   result = jb_check_packages("Evolution", "evolution-plugin", packages);
   g_free(packages);
 
