--- ksystemlog-15.12.3/cmake/FindJournald.cmake.orig	2016-03-25 18:16:30.817009024 +0000
+++ ksystemlog-15.12.3/cmake/FindJournald.cmake	2016-03-25 18:16:45.643603099 +0000
@@ -15,7 +15,7 @@
 # use pkg-config to get the directories and then use these values
 # in the FIND_PATH() and FIND_LIBRARY() calls
 find_package(PkgConfig)
-pkg_check_modules(PC_JOURNALD QUIET libsystemd-journal)
+pkg_check_modules(PC_JOURNALD QUIET libsystemd)
 
 set(JOURNALD_FOUND ${PC_JOURNALD_FOUND})
 set(JOURNALD_DEFINITIONS ${PC_JOURNALD_CFLAGS_OTHER})
