--- /lib/tty/tty-internal.h
+++ /lib/tty/tty-internal.h
@@ -8,4 +8,15 @@
 
 #include "lib/global.h"   /* include <glib.h> */
+
+/* Taken from S-Lang's slutty.c */
+#ifdef ultrix   /* Ultrix gets _POSIX_VDISABLE wrong! */
+# define NULL_VALUE -1
+#else
+# ifdef _POSIX_VDISABLE
+#  define NULL_VALUE _POSIX_VDISABLE
+# else
+#  define NULL_VALUE 255
+# endif
+#endif
 
 /* If true lines are shown by spaces */
--- /lib/tty/tty-ncurses.c
+++ /lib/tty/tty-ncurses.c
@@ -63,4 +63,5 @@
 
 /*** file scope macro definitions **************************************/
+
 #if defined(_AIX) && !defined(CTRL)
 #   define CTRL(x) ((x) & 0x1f)
@@ -159,4 +160,6 @@
     /* use Ctrl-g to generate SIGINT */
     cur_term->Nttyb.c_cc[VINTR] = CTRL ('g'); /* ^g */
+    /* disable SIGQUIT to allow use Ctrl-\ key */
+    cur_term->Nttyb.c_cc[VQUIT] = NULL_VALUE;
     tcsetattr (cur_term->Filedes, TCSANOW, &cur_term->Nttyb);

--- /misc/syntax/Syntax
+++ /misc/syntax/Syntax
@@ -140,5 +140,5 @@
 include spec.syntax
 
-file ..\*\\.repo$ YUM\sRepo File
+file ..\*\\.repo$ YUM\sRepo\sFile
 include yum-repo.syntax

