From cae311cf6453c888a262b435648a194517ce76f3 Mon Sep 17 00:00:00 2001
From: Mike Massonnet <mmassonnet@gmail.com>
Date: Mon, 26 Jul 2010 17:58:06 +0000
Subject: Use right comment character in Xft.xrdb (bug #6314).

---
diff --git a/scripts/Xft.xrdb b/scripts/Xft.xrdb
index af45361..cd26513 100644
--- a/scripts/Xft.xrdb
+++ b/scripts/Xft.xrdb
@@ -1,7 +1,7 @@
-# Those are fallback settings, use the ui plugin to change it
-# or add your overrides to ~/.Xresources
-# Xft.hintstyle: hintnone/hintslight/hintmedium/hintfull
-# Xft hinting: 1/0
+! Those are fallback settings, use the ui plugin to change it
+! or add your overrides to ~/.Xresources
+! Xft.hintstyle: hintnone/hintslight/hintmedium/hintfull
+! Xft hinting: 1/0
 
 Xft.hinting: 1
 Xft.hintstyle: hintmedium
--
cgit v0.8.3.4
