--- branches/KDE/4.1/kdepim/kmail/kmfoldermaildir.cpp	2008/10/30 18:32:47	877937
+++ branches/KDE/4.1/kdepim/kmail/kmfoldermaildir.cpp	2008/10/30 18:33:09	877938
@@ -717,6 +717,7 @@
       dateStr = dateStr.trimmed();
       if (!dateStr.isEmpty())
         mi->setDate(dateStr.constData());
+      uidStr = uidStr.trimmed();
       if ( !uidStr.isEmpty() )
          mi->setUID( uidStr.toULong() );
       mi->setDirty(false);
