--- branches/KDE/3.5/kdepim/certmanager/lib/cryptplug.h	2005/07/26 18:54:59	438982
+++ branches/KDE/3.5/kdepim/certmanager/lib/cryptplug.h	2008/02/21 19:15:38	777849
@@ -408,7 +408,7 @@
    \note This function <b>must</b> be implemented by each plug-in using
    this API specification.
 */
-bool hasFeature( Feature );
+bool hasFeature( ::Feature );
 
 /*! \ingroup groupSignCryptAct
    \brief Information record returned by signing and by encrypting
--- branches/KDE/3.5/kdepim/certmanager/lib/cryptplugwrapper.h	2008/01/25 17:28:51	766368
+++ branches/KDE/3.5/kdepim/certmanager/lib/cryptplugwrapper.h	2008/02/21 19:15:38	777849
@@ -718,7 +718,7 @@
 
         \return  whether the relative feature is implemented or not
     */
-    bool hasFeature( Feature );
+    bool hasFeature( ::Feature );
 
 
     /* \ingroup groupSignAct
 
--- branches/KDE/3.5/kdepim/korganizer/kodaymatrix.cpp	2006/06/09 16:31:27	549733
+++ branches/KDE/3.5/kdepim/korganizer/kodaymatrix.cpp	2008/02/24 14:14:09	778744
@@ -54,6 +54,7 @@
 #include <qcursor.h>
 #include <kpopupmenu.h>
 #include <X11/Xlib.h>
+#undef FocusIn
 #undef KeyPress
 #undef None
 #undef Status
--- branches/KDE/3.5/kdepim/kdgantt/KDGanttViewTaskItem.h	2005/07/26 18:54:59	438982
+++ branches/KDE/3.5/kdepim/kdgantt/KDGanttViewTaskItem.h	2008/02/24 13:32:04	778732
@@ -60,8 +60,10 @@
     void setStartTime( const QDateTime& start );
     void setEndTime( const QDateTime& end );
 
-private:
+protected:
     void showItem( bool show = true, int coordY = 0 );
+
+private:
     void initItem();
     void hideMe();
     QBrush myBrush, undefinedBrush;
--- branches/KDE/3.5/kdepim/korganizer/timelineitem.cpp	2008/02/23 16:28:17	778447
+++ branches/KDE/3.5/kdepim/korganizer/timelineitem.cpp	2008/02/23 16:32:02	778448
@@ -119,13 +119,13 @@
     y = coordY;
   else
     y = getCoordY();
-  int startX = myGanttView->myTimeHeader->getCoordX(myStartTime);
-  int endX = myGanttView->myTimeHeader->getCoordX(myEndTime);
+  int startX = myGanttView->timeHeaderWidget()->getCoordX(myStartTime);
+  int endX = myGanttView->timeHeaderWidget()->getCoordX(myEndTime);
 
   const int mw = QMAX( 1, QMIN( 4, endX - startX ) );
   if ( !mLeft || mw != mMarkerWidth ) {
     if ( !mLeft ) {
-      mLeft = new KDCanvasPolygon( myGanttView->myTimeTable, this, Type_is_KDGanttViewItem );
+      mLeft = new KDCanvasPolygon( myGanttView->timeTableWidget(), this, Type_is_KDGanttViewItem );
       mLeft->setBrush( Qt::black );
     }
     QPointArray a = QPointArray( 4 );
@@ -137,7 +137,7 @@
   }
   if ( !mRight || mw != mMarkerWidth ) {
     if ( !mRight ) {
-      mRight = new KDCanvasPolygon( myGanttView->myTimeTable, this, Type_is_KDGanttViewItem );
+      mRight = new KDCanvasPolygon( myGanttView->timeTableWidget(), this, Type_is_KDGanttViewItem );
       mRight->setBrush( Qt::black );
     }
     QPointArray a = QPointArray( 4 );
