Description: Fix parsing of /proc/cpuinfo
Author: Plamen Aleksandrov <plamen@aomeda.com>
Bug-Debian: https://bugs.debian.org/751228
Last-Update: 2014-07-07
Forwarded: Yes
Applied-Upstream: 2.3.1

Index: libffado/SConstruct
===================================================================
--- libffado.orig/SConstruct
+++ libffado/SConstruct
@@ -558,7 +558,7 @@ def cpuinfo_kv():
     for line in f:
         line = line.strip()
         if line:
-            k,v = line.split(':')
+            k,v = line.split(':', 1)
             yield (k.strip(), v.strip())
     f.close()
 
