diff -Naur libffado-2.0.1-old/src/bebob/bebob_dl_mgr.cpp libffado-2.0.1/src/bebob/bebob_dl_mgr.cpp
--- libffado-2.0.1-old/src/bebob/bebob_dl_mgr.cpp	2009-05-17 23:42:38.000000000 +1000
+++ libffado-2.0.1/src/bebob/bebob_dl_mgr.cpp	2012-04-29 15:26:29.011454798 +1000
@@ -35,6 +35,7 @@
 
 #include <cstdio>
 #include <cstring>
+#include <unistd.h>
 
 namespace BeBoB {
     enum {
diff -Naur libffado-2.0.1-old/src/ffadodevice.cpp libffado-2.0.1/src/ffadodevice.cpp
--- libffado-2.0.1-old/src/ffadodevice.cpp	2008-11-23 21:25:32.000000000 +1000
+++ libffado-2.0.1/src/ffadodevice.cpp	2012-04-29 15:26:54.883998052 +1000
@@ -34,6 +34,7 @@
 
 #include <iostream>
 #include <sstream>
+#include <unistd.h>
 
 #include <assert.h>
 
diff -Naur libffado-2.0.1-old/src/fireworks/fireworks_device.cpp libffado-2.0.1/src/fireworks/fireworks_device.cpp
--- libffado-2.0.1-old/src/fireworks/fireworks_device.cpp	2009-11-23 00:40:08.000000000 +1000
+++ libffado-2.0.1/src/fireworks/fireworks_device.cpp	2012-04-29 15:27:24.123105334 +1000
@@ -44,6 +44,7 @@
 #define FIREWORKS_MIN_FIRMWARE_VERSION 0x04080000
 
 #include <sstream>
+#include <unistd.h>
 using namespace std;
 
 // FireWorks is the platform used and developed by ECHO AUDIO
diff -Naur libffado-2.0.1-old/src/libieee1394/IsoHandlerManager.cpp libffado-2.0.1/src/libieee1394/IsoHandlerManager.cpp
--- libffado-2.0.1-old/src/libieee1394/IsoHandlerManager.cpp	2010-04-18 05:15:54.000000000 +1000
+++ libffado-2.0.1/src/libieee1394/IsoHandlerManager.cpp	2012-04-29 15:27:44.515816147 +1000
@@ -34,6 +34,7 @@
 #include "libutil/Configuration.h"
 
 #include <cstring>
+#include <unistd.h>
 #include <assert.h>
 
 IMPL_DEBUG_MODULE( IsoHandlerManager, IsoHandlerManager, DEBUG_LEVEL_NORMAL );
diff -Naur libffado-2.0.1-old/support/dbus/test-dbus.cpp libffado-2.0.1/support/dbus/test-dbus.cpp
--- libffado-2.0.1-old/support/dbus/test-dbus.cpp	2008-04-27 03:30:41.000000000 +1000
+++ libffado-2.0.1/support/dbus/test-dbus.cpp	2012-04-29 15:28:31.531047923 +1000
@@ -24,6 +24,7 @@
 #include <argp.h>
 #include <stdlib.h>
 #include <iostream>
+#include <unistd.h>
 #include <signal.h>
 
 #include "controlclient.h"
