diff -u -r freeglut-3.2.0/CMakeLists.txt freeglut-3.2.0-glut/CMakeLists.txt
--- freeglut-3.2.0/CMakeLists.txt	2019-03-26 18:55:42.000000000 +0000
+++ freeglut-3.2.0-glut/CMakeLists.txt	2019-09-18 22:02:21.093590716 +0000
@@ -73,7 +73,7 @@
     include/GL/freeglut_std.h
 )
 IF(FREEGLUT_REPLACE_GLUT)
-    LIST(APPEND FREEGLUT_SRCS
+    LIST(APPEND FREEGLUT_HEADERS
         include/GL/glut.h
     )
 ENDIF()
@@ -622,7 +622,7 @@
     "${CMAKE_CURRENT_BINARY_DIR}/FreeGLUT/FreeGLUTConfig.cmake"
     @ONLY
 )
-SET(ConfigPackageLocation lib/cmake/FreeGLUT)
+SET(ConfigPackageLocation ${CMAKE_INSTALL_LIBDIR}/cmake/FreeGLUT)
 INSTALL(EXPORT FreeGLUTTargets
     FILE FreeGLUTTargets.cmake
     NAMESPACE FreeGLUT::
