--- fcpcmcia_cs.c~	2011-01-26 09:53:11.543340898 +0100
+++ fcpcmcia_cs.c	2011-01-26 09:53:11.566675577 +0100
@@ -26,7 +26,6 @@
 #include <asm/io.h>
 #include <asm/system.h>
 
-#include <pcmcia/cs.h>
 #include <pcmcia/cistpl.h>
 #include <pcmcia/ciscode.h>
 #include <pcmcia/ds.h>
@@ -156,7 +155,7 @@
 #endif
     
     /* General socket configuration */
-    p_dev->conf.Attributes = CONF_ENABLE_IRQ;
+    p_dev->config_flags |= CONF_ENABLE_IRQ;
     p_dev->conf.IntType = INT_MEMORY_AND_IO;
     p_dev->conf.ConfigIndex = 1;
     p_dev->conf.Present = PRESENT_OPTION;
@@ -228,7 +227,7 @@
 	/*
          * configure the PCMCIA socket
 	  */
-	i = pcmcia_request_configuration(link, &link->conf);
+	i = pcmcia_enable_device(link);
 	if (i != 0) {
             pcmcia_disable_device(link);
 	    break;
--- fcpcmcia_cs.c~	2011-01-26 10:05:17.426675764 +0100
+++ fcpcmcia_cs.c	2011-01-26 10:05:17.436675570 +0100
@@ -156,9 +156,8 @@
     
     /* General socket configuration */
     p_dev->config_flags |= CONF_ENABLE_IRQ;
-    p_dev->conf.IntType = INT_MEMORY_AND_IO;
-    p_dev->conf.ConfigIndex = 1;
-    p_dev->conf.Present = PRESENT_OPTION;
+    p_dev->config_index = 1;
+    p_dev->config_regs = PRESENT_OPTION;
 
     return cs_config(p_dev);
 
