--- opal-3.10.11/plugins/video/common/mpi.cxx.orig	2017-07-12 19:15:10.396689138 +0000
+++ opal-3.10.11/plugins/video/common/mpi.cxx	2017-07-12 19:18:02.036496397 +0000
@@ -132,8 +132,8 @@
   // to the desired one or matches it
   for (i=0; i < MPIs.size(); i++) {
     // we square the value in order to get absolute distances
-    distance = ( abs(MPIs[i].width  - desiredWidth ) *
-                 abs(MPIs[i].height - desiredHeight) );
+    distance = ( abs((int)(MPIs[i].width  - desiredWidth) ) *
+                 abs((int)(MPIs[i].height - desiredHeight) ) );
 
     if (distance < minDistance) {
       minDistance = distance;
