diff -ur src.orig/motif/server/Config.x86 src/motif/server/Config.x86
--- src.orig/motif/server/Config.x86	2004-04-26 14:07:36.000000000 +0200
+++ src/motif/server/Config.x86	2006-01-04 18:12:39.000000000 +0100
@@ -1,6 +1,6 @@
-CFLAGS = -O2 -I/usr/X11R6/include -I. -I$(VPATH)
-LDFLAGS = -L/usr/X11R6/lib
-LIBS = -static -lXm -dynamic -lXt -lXext -lX11 -lSM -lICE -lXp -ldl -lpthread
+CFLAGS = -O2 -I/usr/include -I. -I$(VPATH)
+LDFLAGS = -L/usr/lib
+LIBS = -lXau -lXdmcp -lXm -dynamic -lXt -lXext -lX11 -lSM -lICE -lXp -ldl -lpthread
 # This def assumes you are building in the same or parallel
 # tree to the CVS souce layout. Sites may need to customize
 # this path.
