diff -ur poppler-0.5.9.orig/glib/Makefile.am poppler-0.5.9/glib/Makefile.am
--- poppler-0.5.9.orig/glib/Makefile.am	2007-05-31 12:38:10.000000000 +0200
+++ poppler-0.5.9/glib/Makefile.am	2007-05-31 12:38:17.000000000 +0200
@@ -82,7 +82,7 @@
 	poppler-private.h
 
 libpoppler_glib_la_LIBADD =				\
-	$(top_builddir)/poppler/libpoppler.la		\
+	-lpoppler					\
 	$(POPPLER_GLIB_LIBS)				\
 	$(FREETYPE_LIBS)				\
 	$(FONTCONFIG_LIBS)				\
@@ -96,7 +96,7 @@
        test-poppler-glib.c
 
 test_poppler_glib_LDADD =			\
-	$(top_builddir)/poppler/libpoppler.la	\
+	-lpoppler				\
 	libpoppler-glib.la			\
 	$(POPPLER_GLIB_LIBS)			\
 	$(FREETYPE_LIBS)			\
Nur in poppler-0.5.9/glib: Makefile.am.orig.
diff -ur poppler-0.5.9.orig/Makefile.am poppler-0.5.9/Makefile.am
--- poppler-0.5.9.orig/Makefile.am	2007-05-31 12:38:10.000000000 +0200
+++ poppler-0.5.9/Makefile.am	2007-05-31 12:38:17.000000000 +0200
@@ -29,7 +29,7 @@
 utils_subdir = utils
 endif
 
-SUBDIRS = goo fofi $(splash_subdir) poppler $(utils_subdir) $(glib_subdir) $(qt_subdir) test $(qt4_subdir)
+SUBDIRS = $(glib_subdir) $(qt_subdir) $(qt4_subdir)
 
 EXTRA_DIST =					\
 	README-XPDF				\
@@ -40,9 +40,7 @@
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA =				\
-	poppler.pc				\
 	$(cairo_pc_file)			\
-	$(splash_pc_file)			\
 	$(glib_pc_file)				\
 	$(qt_pc_file)                           \
 	$(qt4_pc_file)
diff -ur poppler-0.5.9.orig/qt/Makefile.am poppler-0.5.9/qt/Makefile.am
--- poppler-0.5.9.orig/qt/Makefile.am	2007-05-31 12:38:10.000000000 +0200
+++ poppler-0.5.9/qt/Makefile.am	2007-05-31 12:38:17.000000000 +0200
@@ -24,7 +24,7 @@
 	poppler-private.h
 
 libpoppler_qt_la_LIBADD = 			\
-	$(top_builddir)/poppler/libpoppler.la	\
+	-lpoppler				\
 	$(POPPLER_QT_LIBS)			\
 	$(FREETYPE_LIBS)			\
 	$(FONTCONFIG_LIBS)
@@ -41,7 +41,7 @@
        test-poppler-qt.cpp
 
 test_poppler_qt_LDADD =				\
-	$(top_builddir)/poppler/libpoppler.la	\
+	-lpoppler				\
 	libpoppler-qt.la			\
 	$(POPPLER_QT_LIBS)			\
 	$(FREETYPE_LIBS)
Nur in poppler-0.5.9/qt: Makefile.am.orig.
diff -ur poppler-0.5.9.orig/qt4/src/Makefile.am poppler-0.5.9/qt4/src/Makefile.am
--- poppler-0.5.9.orig/qt4/src/Makefile.am	2007-05-31 12:38:10.000000000 +0200
+++ poppler-0.5.9/qt4/src/Makefile.am	2007-05-31 12:38:55.000000000 +0200
@@ -35,7 +35,7 @@
 	poppler-private.h
 
 libpoppler_qt4_la_LIBADD = 			\
-	$(top_builddir)/poppler/libpoppler.la	\
+	-lpoppler				\
 	$(FONTCONFIG_LIBS)				\
 	$(POPPLER_QT4_LIBS)
 
Nur in poppler-0.5.9/qt4/src: Makefile.am.orig.
Nur in poppler-0.5.9/qt4/src: Makefile.am.rej.
diff -ur poppler-0.5.9.orig/qt4/tests/Makefile.am poppler-0.5.9/qt4/tests/Makefile.am
--- poppler-0.5.9.orig/qt4/tests/Makefile.am	2007-05-31 12:38:10.000000000 +0200
+++ poppler-0.5.9/qt4/tests/Makefile.am	2007-05-31 12:38:17.000000000 +0200
@@ -7,7 +7,7 @@
 	$(POPPLER_QT4_CFLAGS)
 
 LDADDS =					\
-	$(top_builddir)/poppler/libpoppler.la	\
+	-lpoppler				\
 	$(top_builddir)/qt4/src/libpoppler-qt4.la	\
 	$(FONTCONFIG_LIBS)				\
 	$(POPPLER_QT4_LIBS)
Nur in poppler-0.5.9/qt4/tests: Makefile.am.orig.
