 src/task-manager-linux.c |    7 +++++++
 1 files changed, 7 insertions(+), 0 deletions(-)

diff --git a/src/task-manager-linux.c b/src/task-manager-linux.c
index 938cee8..f329629 100644
--- a/src/task-manager-linux.c
+++ b/src/task-manager-linux.c
@@ -30,6 +30,13 @@ get_memory_usage (guint64 *memory_total, guint64 *memory_free, guint64 *memory_c
 	gchar *filename = "/proc/meminfo";
 	gushort found = 0;
 
+	*memory_total = 0;
+	*memory_free = 0;
+	*memory_cache = 0;
+	*memory_buffers = 0;
+	*swap_total = 0;
+	*swap_free = 0;
+
 	if ((file = fopen (filename, "r")) == NULL)
 		return FALSE;
 
-- 
1.7.0.4


