From 9c6451833b7550b594c4a3f8082275dfcc81e5ae Mon Sep 17 00:00:00 2001
Message-Id: <9c6451833b7550b594c4a3f8082275dfcc81e5ae.1505468187.git.jan.steffens@gmail.com>
From: "Jan Alexander Steffens (heftig)" <jan.steffens@gmail.com>
Date: Fri, 15 Sep 2017 07:36:15 +0200
Subject: [PATCH] meson: Ensure paths written to .pc are absolute

---
 meson.build | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/meson.build b/meson.build
index a357a9798f89da3c..38d37019908376fa 100644
--- a/meson.build
+++ b/meson.build
@@ -53,8 +53,8 @@ import('gnome')
 pkg_conf = configuration_data()
 pkg_conf.set('prefix', get_option('prefix'))
 pkg_conf.set('exec_prefix', get_option('prefix'))
-pkg_conf.set('libdir', get_option('libdir'))
-pkg_conf.set('includedir', get_option('includedir'))
+pkg_conf.set('libdir', join_paths(get_option('prefix'), get_option('libdir')))
+pkg_conf.set('includedir', join_paths(get_option('prefix'), get_option('includedir')))
 pkg_conf.set('VERSION', meson.project_version())
 
 configure_file(input: 'atk-bridge-2.0.pc.in',
-- 
2.14.1

