From ab1ed993badc1a6978b95c9d17a5e8b56fff4598 Mon Sep 17 00:00:00 2001
From: David Schweikert <david@schweikert.ch>
Date: Sat, 5 Mar 2022 16:58:47 +0100
Subject: [PATCH] fix socket4 vs socket6 typo (#249, thanks @sfan5)

---
 src/fping.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/fping.c b/src/fping.c
index 0bd2b70..af16061 100644
--- a/src/fping.c
+++ b/src/fping.c
@@ -1108,7 +1108,7 @@ int main(int argc, char** argv)
         socket_set_src_addr_ipv4(socket4, &src_addr, (socktype4 == SOCK_DGRAM) ? &ident4 : NULL);
     }
 #ifdef IPV6
-    if (socket6 >= 0 && (src_addr6_set || socktype4 == SOCK_DGRAM)) {
+    if (socket6 >= 0 && (src_addr6_set || socktype6 == SOCK_DGRAM)) {
         socket_set_src_addr_ipv6(socket6, &src_addr6, (socktype6 == SOCK_DGRAM) ? &ident6 : NULL);
     }
 #endif
