diff --git a/tests/testKeys.sh b/tests/testKeys.sh
index 8424e8d6d..af3ee8a01 100755
--- a/tests/testKeys.sh
+++ b/tests/testKeys.sh
@@ -22,11 +22,8 @@ echo "--- LTDL_LIBRARY_PATH=$LTDL_LIBRARY_PATH" >> $logfile
 mkdir -p $crypto_config
 rm -rf $crypto_config/*
 
-# remove old keys file and copy NSS DB files if needed
+# remove old keys file
 rm -rf $keysfile
-if [ "z$crypto" = "znss" ] ; then
-    cp -f $nssdbfolder/*.db $crypto_config
-fi
 
 ##########################################################################
 ##########################################################################
diff --git a/tests/testrun.sh b/tests/testrun.sh
index ed79e608c..05ed22974 100755
--- a/tests/testrun.sh
+++ b/tests/testrun.sh
@@ -37,7 +37,6 @@ else
     curlogfile=$TMPFOLDER/$testname.$timestamp-$$.cur.log
     failedlogfile=$TMPFOLDER/$testname.$timestamp-$$.failed.log
 fi
-nssdbfolder=$topfolder/nssdb
 
 #
 # Valgrind
