From 41449761b1200749b34e034f9b4b3c0bdf299013 Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <jan.steffens@gmail.com>
Date: Sun, 14 May 2017 18:09:31 +0200
Subject: [PATCH 3/3] Enable long PCF family names

---
 include/freetype/config/ftoption.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/freetype/config/ftoption.h b/include/freetype/config/ftoption.h
index 1d4469a99..4ff4e7821 100644
--- a/include/freetype/config/ftoption.h
+++ b/include/freetype/config/ftoption.h
@@ -847,7 +847,7 @@ FT_BEGIN_HEADER
    * If this option is activated, it can be controlled with the
    * `no-long-family-names` property of the 'pcf' driver module.
    */
-/* #define PCF_CONFIG_OPTION_LONG_FAMILY_NAMES */
+#define PCF_CONFIG_OPTION_LONG_FAMILY_NAMES
 
 
   /*************************************************************************/
-- 
2.22.0

