From a82d9cfe03679b82ab4d9586c93ad377e9e27149 Mon Sep 17 00:00:00 2001
From: Eric Leblond <eric@regit.org>
Date: Fri, 5 Feb 2016 11:55:26 +0100
Subject: [PATCH 2/2] ulogd: add missing newline in log message

---
 src/ulogd.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/ulogd.c b/src/ulogd.c
index c8d9376..5b9a586 100644
--- a/src/ulogd.c
+++ b/src/ulogd.c
@@ -1515,7 +1515,7 @@ int main(int argc, char* argv[])
 
 	if (daemonize){
 		if (daemon(0, 0) < 0) {
-			ulogd_log(ULOGD_FATAL, "can't daemonize: %s (%d)",
+			ulogd_log(ULOGD_FATAL, "can't daemonize: %s (%d)\n",
 				  errno, strerror(errno));
 			warn_and_exit(daemonize);
 		}
-- 
2.7.1

