diff -Nur -x '*.orig' -x '*~' gnome-settings-daemon-2.26.0/gnome-settings-daemon/Makefile.am gnome-settings-daemon-2.26.0.new/gnome-settings-daemon/Makefile.am
--- gnome-settings-daemon-2.26.0/gnome-settings-daemon/Makefile.am	2009-03-16 15:35:39.000000000 +0100
+++ gnome-settings-daemon-2.26.0.new/gnome-settings-daemon/Makefile.am	2009-03-18 15:34:43.000000000 +0100
@@ -11,28 +11,27 @@
 	$(NULL)
 
 noinst_LTLIBRARIES = 			\
-	libgsd-plugins.la		\
+	libgsd-profile.la		\
 	$(NULL)
 
-libgsd_plugins_la_SOURCES =		\
-	gnome-settings-plugin.c		\
-	gnome-settings-plugin.h		\
+libgsd_profile_la_SOURCES =		\
 	gnome-settings-profile.c	\
 	gnome-settings-profile.h	\
 	$(NULL)
 
-libgsd_plugins_la_CPPFLAGS = 		\
+libgsd_profile_la_CPPFLAGS = 		\
 	$(DISABLE_DEPRECATED_CFLAGS)	\
 	$(NULL)
 
-libgsd_plugins_la_CFLAGS =		\
+libgsd_profile_la_CFLAGS =		\
+	$(SIMPLE_GREETER_CFLAGS)	\
 	$(NULL)
 
-libgsd_plugins_la_LIBADD =		\
+libgsd_profile_la_LIBADD =		\
 	$(NULL)
 
-libgsd_plugins_la_LDFLAGS = 		\
-	-export-dynamic 		\
+libgsd_profile_la_LDFLAGS = 	\
+	-export-dynamic 	\
 	$(NULL)
 
 gsddir = $(libexecdir)
@@ -60,6 +59,8 @@
 	main.c				\
 	gnome-settings-manager.c	\
 	gnome-settings-manager.h	\
+	gnome-settings-plugin.c		\
+	gnome-settings-plugin.h		\
 	gnome-settings-plugin-info.c	\
 	gnome-settings-plugin-info.h	\
 	gnome-settings-module.c		\
@@ -76,7 +77,7 @@
 	$(AM_LDFLAGS)
 
 gnome_settings_daemon_LDADD = 		\
-	libgsd-plugins.la		\
+	libgsd-profile.la		\
 	$(SETTINGS_DAEMON_LIBS)
 
 EXTRA_DIST = 				\
diff -Nur -x '*.orig' -x '*~' gnome-settings-daemon-2.26.0/plugins/a11y-keyboard/Makefile.am gnome-settings-daemon-2.26.0.new/plugins/a11y-keyboard/Makefile.am
--- gnome-settings-daemon-2.26.0/plugins/a11y-keyboard/Makefile.am	2009-03-16 15:35:39.000000000 +0100
+++ gnome-settings-daemon-2.26.0.new/plugins/a11y-keyboard/Makefile.am	2009-03-18 15:34:43.000000000 +0100
@@ -60,7 +60,6 @@
 	$(NULL)
 
 liba11y_keyboard_la_LIBADD  = 		\
-	$(top_builddir)/gnome-settings-daemon/libgsd-plugins.la \
 	$(SETTINGS_PLUGIN_LIBS)		\
 	$(XF86MISC_LIBS)		\
 	$(LIBNOTIFY_LIBS)		\
diff -Nur -x '*.orig' -x '*~' gnome-settings-daemon-2.26.0/plugins/background/Makefile.am gnome-settings-daemon-2.26.0.new/plugins/background/Makefile.am
--- gnome-settings-daemon-2.26.0/plugins/background/Makefile.am	2009-03-16 15:35:38.000000000 +0100
+++ gnome-settings-daemon-2.26.0.new/plugins/background/Makefile.am	2009-03-18 15:34:43.000000000 +0100
@@ -20,7 +20,7 @@
 	$(AM_CFLAGS)
 
 test_background_LDADD =		\
-	$(top_builddir)/gnome-settings-daemon/libgsd-plugins.la	\
+	$(top_builddir)/gnome-settings-daemon/libgsd-profile.la	\
 	$(SETTINGS_PLUGIN_LIBS)	\
 	$(NULL)
 
@@ -50,7 +50,6 @@
 	$(NULL)
 
 libbackground_la_LIBADD  = 		\
-	$(top_builddir)/gnome-settings-daemon/libgsd-plugins.la	\
 	$(SETTINGS_PLUGIN_LIBS)		\
 	$(NULL)
 
diff -Nur -x '*.orig' -x '*~' gnome-settings-daemon-2.26.0/plugins/clipboard/Makefile.am gnome-settings-daemon-2.26.0.new/plugins/clipboard/Makefile.am
--- gnome-settings-daemon-2.26.0/plugins/clipboard/Makefile.am	2009-03-16 15:35:39.000000000 +0100
+++ gnome-settings-daemon-2.26.0.new/plugins/clipboard/Makefile.am	2009-03-18 15:34:43.000000000 +0100
@@ -29,7 +29,6 @@
 	$(NULL)
 
 libclipboard_la_LIBADD  = 	\
-	$(top_builddir)/gnome-settings-daemon/libgsd-plugins.la	\
 	$(SETTINGS_PLUGIN_LIBS)	\
 	$(NULL)
 
diff -Nur -x '*.orig' -x '*~' gnome-settings-daemon-2.26.0/plugins/dummy/Makefile.am gnome-settings-daemon-2.26.0.new/plugins/dummy/Makefile.am
--- gnome-settings-daemon-2.26.0/plugins/dummy/Makefile.am	2009-03-16 15:35:39.000000000 +0100
+++ gnome-settings-daemon-2.26.0.new/plugins/dummy/Makefile.am	2009-03-18 15:34:43.000000000 +0100
@@ -20,7 +20,6 @@
 	$(GSD_PLUGIN_LDFLAGS)
 
 libdummy_la_LIBADD  = 		\
-	$(top_builddir)/gnome-settings-daemon/libgsd-plugins.la	\
 	$(SETTINGS_PLUGIN_LIBS)
 
 plugin_in_files = 		\
diff -Nur -x '*.orig' -x '*~' gnome-settings-daemon-2.26.0/plugins/font/Makefile.am gnome-settings-daemon-2.26.0.new/plugins/font/Makefile.am
--- gnome-settings-daemon-2.26.0/plugins/font/Makefile.am	2009-03-16 15:35:39.000000000 +0100
+++ gnome-settings-daemon-2.26.0.new/plugins/font/Makefile.am	2009-03-18 15:34:43.000000000 +0100
@@ -23,7 +23,6 @@
 	$(AM_CFLAGS)
 
 libfont_la_LDFLAGS = 		\
-	$(top_builddir)/gnome-settings-daemon/libgsd-plugins.la	\
 	$(GSD_PLUGIN_LDFLAGS)	\
 	$(NULL)
 
diff -Nur -x '*.orig' -x '*~' gnome-settings-daemon-2.26.0/plugins/housekeeping/Makefile.am gnome-settings-daemon-2.26.0.new/plugins/housekeeping/Makefile.am
--- gnome-settings-daemon-2.26.0/plugins/housekeeping/Makefile.am	2009-03-16 15:35:38.000000000 +0100
+++ gnome-settings-daemon-2.26.0.new/plugins/housekeeping/Makefile.am	2009-03-18 15:34:43.000000000 +0100
@@ -21,11 +21,7 @@
 
 libhousekeeping_la_LDFLAGS = $(GSD_PLUGIN_LDFLAGS)
 
-libhousekeeping_la_LIBADD = 	\
-	$(top_builddir)/gnome-settings-daemon/libgsd-plugins.la \
-	$(SETTINGS_PLUGIN_LIBS)	\
-	$(GIOUNIX_LIBS)		\
-	$(LIBNOTIFY_LIBS)
+libhousekeeping_la_LIBADD = $(SETTINGS_PLUGIN_LIBS) $(GIOUNIX_LIBS) $(LIBNOTIFY_LIBS)
 
 plugin_in_files = housekeeping.gnome-settings-plugin.in
 
diff -Nur -x '*.orig' -x '*~' gnome-settings-daemon-2.26.0/plugins/keybindings/Makefile.am gnome-settings-daemon-2.26.0.new/plugins/keybindings/Makefile.am
--- gnome-settings-daemon-2.26.0/plugins/keybindings/Makefile.am	2009-03-16 15:35:39.000000000 +0100
+++ gnome-settings-daemon-2.26.0.new/plugins/keybindings/Makefile.am	2009-03-18 15:34:43.000000000 +0100
@@ -26,7 +26,6 @@
 	$(NULL)
 
 libkeybindings_la_LIBADD  = 		\
-	$(top_builddir)/gnome-settings-daemon/libgsd-plugins.la	\
 	$(top_builddir)/plugins/common/libcommon.la	\
 	$(SETTINGS_PLUGIN_LIBS)		\
 	$(NULL)
diff -Nur -x '*.orig' -x '*~' gnome-settings-daemon-2.26.0/plugins/keyboard/Makefile.am gnome-settings-daemon-2.26.0.new/plugins/keyboard/Makefile.am
--- gnome-settings-daemon-2.26.0/plugins/keyboard/Makefile.am	2009-03-16 15:35:39.000000000 +0100
+++ gnome-settings-daemon-2.26.0.new/plugins/keyboard/Makefile.am	2009-03-18 15:34:43.000000000 +0100
@@ -36,7 +36,6 @@
 	$(NULL)
 
 libkeyboard_la_LIBADD  = 	\
-	$(top_builddir)/gnome-settings-daemon/libgsd-plugins.la	\
 	$(SETTINGS_PLUGIN_LIBS)	\
 	$(XF86MISC_LIBS)	\
 	$(LIBGNOMEKBD_LIBS)	\
diff -Nur -x '*.orig' -x '*~' gnome-settings-daemon-2.26.0/plugins/media-keys/Makefile.am gnome-settings-daemon-2.26.0.new/plugins/media-keys/Makefile.am
--- gnome-settings-daemon-2.26.0/plugins/media-keys/Makefile.am	2009-03-16 15:35:38.000000000 +0100
+++ gnome-settings-daemon-2.26.0.new/plugins/media-keys/Makefile.am	2009-03-18 15:34:43.000000000 +0100
@@ -53,7 +53,6 @@
 	$(GSD_PLUGIN_LDFLAGS)
 
 libmedia_keys_la_LIBADD  = 		\
-	$(top_builddir)/gnome-settings-daemon/libgsd-plugins.la		\
 	$(top_builddir)/plugins/common/libcommon.la			\
 	$(top_builddir)/plugins/media-keys/cut-n-paste/libgvc.la	\
 	$(SETTINGS_PLUGIN_LIBS)						\
@@ -116,7 +115,7 @@
 	$(AM_CFLAGS)
 
 test_media_keys_LDADD = \
-	$(top_builddir)/gnome-settings-daemon/libgsd-plugins.la		\
+	$(top_builddir)/gnome-settings-daemon/libgsd-profile.la		\
 	$(top_builddir)/plugins/common/libcommon.la			\
 	$(SETTINGS_DAEMON_LIBS)			\
 	$(SETTINGS_PLUGIN_LIBS)			\
diff -Nur -x '*.orig' -x '*~' gnome-settings-daemon-2.26.0/plugins/mouse/Makefile.am gnome-settings-daemon-2.26.0.new/plugins/mouse/Makefile.am
--- gnome-settings-daemon-2.26.0/plugins/mouse/Makefile.am	2009-03-16 15:35:38.000000000 +0100
+++ gnome-settings-daemon-2.26.0.new/plugins/mouse/Makefile.am	2009-03-18 15:34:43.000000000 +0100
@@ -23,7 +23,6 @@
 	$(GSD_PLUGIN_LDFLAGS)
 
 libmouse_la_LIBADD  = 		\
-	$(top_builddir)/gnome-settings-daemon/libgsd-plugins.la	\
 	$(SETTINGS_PLUGIN_LIBS)	\
 	$(XINPUT_LIBS)
 
diff -Nur -x '*.orig' -x '*~' gnome-settings-daemon-2.26.0/plugins/screensaver/Makefile.am gnome-settings-daemon-2.26.0.new/plugins/screensaver/Makefile.am
--- gnome-settings-daemon-2.26.0/plugins/screensaver/Makefile.am	2009-03-16 15:35:38.000000000 +0100
+++ gnome-settings-daemon-2.26.0.new/plugins/screensaver/Makefile.am	2009-03-18 15:34:43.000000000 +0100
@@ -25,7 +25,6 @@
 	$(NULL)
 
 libscreensaver_la_LIBADD  = 		\
-	$(top_builddir)/gnome-settings-daemon/libgsd-plugins.la	\
 	$(SETTINGS_PLUGIN_LIBS)	\
 	$(NULL)
 
diff -Nur -x '*.orig' -x '*~' gnome-settings-daemon-2.26.0/plugins/sound/Makefile.am gnome-settings-daemon-2.26.0.new/plugins/sound/Makefile.am
--- gnome-settings-daemon-2.26.0/plugins/sound/Makefile.am	2009-03-16 15:35:38.000000000 +0100
+++ gnome-settings-daemon-2.26.0.new/plugins/sound/Makefile.am	2009-03-18 15:34:43.000000000 +0100
@@ -22,7 +22,6 @@
 	$(GSD_PLUGIN_LDFLAGS)
 
 libsound_la_LIBADD = \
-	$(top_builddir)/gnome-settings-daemon/libgsd-plugins.la	\
 	$(SETTINGS_PLUGIN_LIBS)	\
 	$(PULSE_LIBS)
 
diff -Nur -x '*.orig' -x '*~' gnome-settings-daemon-2.26.0/plugins/typing-break/Makefile.am gnome-settings-daemon-2.26.0.new/plugins/typing-break/Makefile.am
--- gnome-settings-daemon-2.26.0/plugins/typing-break/Makefile.am	2009-03-16 15:35:39.000000000 +0100
+++ gnome-settings-daemon-2.26.0.new/plugins/typing-break/Makefile.am	2009-03-18 15:34:43.000000000 +0100
@@ -25,7 +25,6 @@
 	$(NULL)
 
 libtyping_break_la_LIBADD  = 		\
-	$(top_builddir)/gnome-settings-daemon/libgsd-plugins.la	\
 	$(SETTINGS_PLUGIN_LIBS)	\
 	$(NULL)
 
diff -Nur -x '*.orig' -x '*~' gnome-settings-daemon-2.26.0/plugins/xrandr/Makefile.am gnome-settings-daemon-2.26.0.new/plugins/xrandr/Makefile.am
--- gnome-settings-daemon-2.26.0/plugins/xrandr/Makefile.am	2009-03-16 15:35:39.000000000 +0100
+++ gnome-settings-daemon-2.26.0.new/plugins/xrandr/Makefile.am	2009-03-18 15:34:43.000000000 +0100
@@ -59,7 +59,6 @@
 	$(GSD_PLUGIN_LDFLAGS)
 
 libxrandr_la_LIBADD  =			\
-	$(top_builddir)/gnome-settings-daemon/libgsd-plugins.la	\
 	$(SETTINGS_PLUGIN_LIBS)		\
 	$(LIBNOTIFY_LIBS)		\
 	$(RANDR_LIBS)
diff -Nur -x '*.orig' -x '*~' gnome-settings-daemon-2.26.0/plugins/xrdb/Makefile.am gnome-settings-daemon-2.26.0.new/plugins/xrdb/Makefile.am
--- gnome-settings-daemon-2.26.0/plugins/xrdb/Makefile.am	2009-03-16 15:35:39.000000000 +0100
+++ gnome-settings-daemon-2.26.0.new/plugins/xrdb/Makefile.am	2009-03-18 15:34:43.000000000 +0100
@@ -30,7 +30,6 @@
 	$(NULL)
 
 libxrdb_la_LIBADD  = 		\
-	$(top_builddir)/gnome-settings-daemon/libgsd-plugins.la	\
 	$(SETTINGS_PLUGIN_LIBS)	\
 	$(NULL)
 
diff -Nur -x '*.orig' -x '*~' gnome-settings-daemon-2.26.0/plugins/xsettings/Makefile.am gnome-settings-daemon-2.26.0.new/plugins/xsettings/Makefile.am
--- gnome-settings-daemon-2.26.0/plugins/xsettings/Makefile.am	2009-03-16 15:35:38.000000000 +0100
+++ gnome-settings-daemon-2.26.0.new/plugins/xsettings/Makefile.am	2009-03-18 15:34:43.000000000 +0100
@@ -29,7 +29,6 @@
 	$(NULL)
 
 libxsettings_la_LIBADD  = 	\
-	$(top_builddir)/gnome-settings-daemon/libgsd-plugins.la	\
 	$(SETTINGS_PLUGIN_LIBS)	\
 	$(NULL)
 
