diff -Nurp xserver-xorg-video-intel-2.4.1-patched/src/i830_quirks.c xserver-xorg-video-intel-2.4.1-working/src/i830_quirks.c
--- xserver-xorg-video-intel-2.4.1-patched/src/i830_quirks.c	2008-09-08 20:12:28.000000000 -0700
+++ xserver-xorg-video-intel-2.4.1-working/src/i830_quirks.c	2008-09-08 21:02:53.000000000 -0700
@@ -223,7 +223,10 @@ static i830_quirk i830_quirk_list[] = {
     /* Apple Mac mini has no lvds, but macbook pro does */
     { PCI_CHIP_I945_GM, 0x8086, 0x7270, quirk_mac_mini },
 
-    /* Transtec Senyo 610 mini pc */
+    /* Asus Eee PC B202 (See LP: #233787) */
+    { PCI_CHIP_I945_GME, 0x1043, 0x1252, quirk_ignore_lvds },
+
+    /* Transtec Senyo 610 mini pc (See LP: #233787) */
     { PCI_CHIP_I965_GM, 0x1509, 0x2f15, quirk_ignore_lvds },
 
     /* Clevo M720R has no tv output */
@@ -243,6 +246,8 @@ static i830_quirk i830_quirk_list[] = {
     { PCI_CHIP_I965_GM, 0x1028, 0x0286, quirk_ignore_tv },
     /* Dell Vostro A840 (LP: #235155) */
     { PCI_CHIP_I965_GM, 0x1028, 0x0298, quirk_ignore_tv },
+    /* Dell Studio Hybrid (LP: #267945) */
+    { PCI_CHIP_I965_GM, 0x1028, 0x0279, quirk_ignore_lvds },
 
     /* Lenovo Napa TV (use dmi)*/
     { PCI_CHIP_I945_GM, 0x17aa, SUBSYS_ANY, quirk_lenovo_tv_dmi },
@@ -312,6 +317,9 @@ static i830_quirk i830_quirk_list[] = {
     /* Sony vaio VGN-SZ4MN (See LP: #212163) */
     { PCI_CHIP_I830_M, 0x104d, 0x81e6, quirk_pipea_force },
 
+    /* Quanta Gigabyte W251U (See LP: #244242) */
+    { PCI_CHIP_I945_GM, 0x152d, 0x0755, quirk_pipea_force },
+
     /* Ordi Enduro UW31 (See LP: #152416) */
     { PCI_CHIP_I945_GM, 0x1584, 0x9900, quirk_ignore_tv },
 
