diff -Nur -x '*.orig' -x '*~' system-config-printer-1.4.2+20130920/scp-dbus-service.py system-config-printer-1.4.2+20130920.new/scp-dbus-service.py
--- system-config-printer-1.4.2+20130920/scp-dbus-service.py	2013-09-20 22:26:27.000000000 +0200
+++ system-config-printer-1.4.2+20130920.new/scp-dbus-service.py	2013-10-24 13:12:44.998717234 +0200
@@ -367,7 +367,7 @@
         pass
 
     @dbus.service.signal(dbus_interface=CONFIG_NEWPRINTERDIALOG_IFACE,
-                         signature='a{s}')
+                         signature='a(s)')
     def DriverDownloadChecked(self, installed_files):
         pass
 
