From 691116b22355bdaac211bd66e8c90fdecd9b2077 Mon Sep 17 00:00:00 2001
From: Max Kellermann <max@musicpd.org>
Date: Wed, 10 Oct 2018 15:47:45 +0200
Subject: [PATCH] conf: remove trailing whitespace from option names

Closes #39
---
 NEWS         | 1 +
 src/conf.cxx | 2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/conf.cxx b/src/conf.cxx
index d4c94a7..da8c163 100644
--- a/src/conf.cxx
+++ b/src/conf.cxx
@@ -236,7 +236,7 @@ separate_value(char *p)
 		return nullptr;
 	}
 
-	*value++ = 0;
+	*StripRight(p, value++) = '\0';
 
 	return Strip(value);
 }
