---
 embed/mozilla/ContentHandler.cpp |    6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

Index: epiphany-browser-2.22.1.1/embed/mozilla/ContentHandler.cpp
===================================================================
--- epiphany-browser-2.22.1.1.orig/embed/mozilla/ContentHandler.cpp
+++ epiphany-browser-2.22.1.1/embed/mozilla/ContentHandler.cpp
@@ -124,34 +124,36 @@
 }
 
 /* nsILocalFile promptForSaveToFile (in nsISupports aWindowContext, in wstring aDefaultFile, in wstring aSuggestedFileExtension); */
 NS_IMETHODIMP GContentHandler::PromptForSaveToFile(
 				    nsIHelperAppLauncher *aLauncher,			    
 				    nsISupports *aWindowContext,
 				    const PRUnichar *aDefaultFile,
 				    const PRUnichar *aSuggestedFileExtension,
+				    PRBool aForcePrompt,
 				    nsILocalFile **_retval)
 {
 	EphyFileChooser *dialog;
 	int response;
 	char *filename = NULL;
 	nsCString defaultFile;
 
 	NS_UTF16ToCString (nsString (aDefaultFile),
 			   NS_CSTRING_ENCODING_UTF8, defaultFile);
 
-	if (mAction != CONTENT_ACTION_SAVEAS)
+	if (!aForcePrompt &&
+	    mAction != CONTENT_ACTION_SAVEAS)
 	{
 		return BuildDownloadPath (defaultFile.get(), _retval);
 	}
 	nsCOMPtr<nsIDOMWindow> parentDOMWindow (do_GetInterface (aWindowContext));
 
         AutoModalDialog modalDialog (parentDOMWindow, PR_FALSE);
-        if (!modalDialog.ShouldShow ())
+        if (!aForcePrompt && !modalDialog.ShouldShow ())
           return NS_ERROR_FAILURE;
 
 	GtkWindow *parentWindow = modalDialog.GetParent ();
 
 	dialog = ephy_file_chooser_new (_("Save"), GTK_WIDGET (parentWindow),
 					GTK_FILE_CHOOSER_ACTION_SAVE,
 					CONF_STATE_SAVE_DIR,
 					EPHY_FILE_FILTER_ALL);
