@@ -, +, @@ 
 settings app. (Bug #8799)
---
 settings/main.c |   14 ++++++++++++++
 1 file changed, 14 insertions(+)
--- a/settings/main.c	
+++ a/settings/main.c	
@@ -688,6 +688,7 @@ xfdesktop_settings_dialog_populate_image_list(AppearancePanel *panel)
         xfconf_channel_set_string(panel->channel, prop_image, image_file);
         xfconf_channel_set_string(panel->channel, prop_last, image_file);
 
+        /* Add all backdrops in xfce4/backdrops/ */
         backdrop_dirs = xfce_resource_lookup_all(XFCE_RESOURCE_DATA,
                                                  "xfce4/backdrops/");
         for(i = 0; backdrop_dirs[i]; ++i) {
@@ -696,6 +697,19 @@ xfdesktop_settings_dialog_populate_image_list(AppearancePanel *panel)
             if(tmp)
                 image_file_iter = tmp;
         }
+        g_strfreev(backdrop_dirs);
+
+        /* Add all backdrops in backgrounds/xfce/ */
+        backdrop_dirs = xfce_resource_lookup_all(XFCE_RESOURCE_DATA,
+                                                 "backgrounds/xfce/");
+        for(i = 0; backdrop_dirs[i]; ++i) {
+            tmp = xfdesktop_image_list_add_dir(ls, backdrop_dirs[i],
+                                               image_file);
+            if(tmp)
+                image_file_iter = tmp;
+        }
+        g_strfreev(backdrop_dirs);
+
 
         if(!image_file_iter)
             image_file_iter = xfdesktop_settings_image_treeview_add(GTK_TREE_MODEL(ls), image_file);
-- 
