diff -Nur gnome-settings-daemon-2.32.1.orig/plugins/xrandr/gsd-xrandr-manager.c gnome-settings-daemon-2.32.1/plugins/xrandr/gsd-xrandr-manager.c
--- gnome-settings-daemon-2.32.1.orig/plugins/xrandr/gsd-xrandr-manager.c	2010-11-15 05:11:00.000000000 -0800
+++ gnome-settings-daemon-2.32.1/plugins/xrandr/gsd-xrandr-manager.c	2011-03-17 11:52:57.516694024 -0700
@@ -2530,6 +2530,10 @@
 
         manager->priv->current_fn_f7_config = -1;
         manager->priv->fn_f7_configs = NULL;
+
+#ifdef HAVE_LIBNOTIFY
+        notify_init ("gnome-settings-daemon");
+#endif /* HAVE_LIBNOTIFY */
 }
 
 static void
