Description: Prevent executable stack
Author: Andreas Metzler <ametzler@debian.org>

--- gnutls28-3.0.4.orig/lib/accelerated/x86/asm/padlock-common.s
+++ gnutls28-3.0.4/lib/accelerated/x86/asm/padlock-common.s
@@ -38,3 +38,7 @@ is_padlock_nano:
 	xorl	%eax,%eax
         ret
 .size	is_padlock_nano,.-is_padlock_nano
+
+#if defined(__linux__) && defined(__ELF__)
+.section .note.GNU-stack,"",%progbits
+#endif
