diff --git a/src/articlelistview.cpp b/src/articlelistview.cpp
index 3bc6ae1b..92a891f1 100644
--- a/src/articlelistview.cpp
+++ b/src/articlelistview.cpp
@@ -125,9 +125,9 @@ void ArticleListView::setArticleModel(ArticleModel *model)
     }
 
     m_proxy = new SortColorizeProxyModel(model);
-    m_proxy->setSourceModel(model);
     m_proxy->setSortRole(ArticleModel::SortRole);
     m_proxy->setFilters(m_matchers);
+    m_proxy->setSourceModel(model);
     auto *const proxy2 = new FilterDeletedProxyModel(model);
     proxy2->setSortRole(ArticleModel::SortRole);
     proxy2->setSourceModel(m_proxy);
@@ -137,11 +137,11 @@ void ArticleListView::setArticleModel(ArticleModel *model)
 
     auto *const columnsProxy = new FilterColumnsProxyModel(model);
     columnsProxy->setSortRole(ArticleModel::SortRole);
-    columnsProxy->setSourceModel(proxy2);
     columnsProxy->setColumnEnabled(ArticleModel::ItemTitleColumn);
     columnsProxy->setColumnEnabled(ArticleModel::FeedTitleColumn);
     columnsProxy->setColumnEnabled(ArticleModel::DateColumn);
     columnsProxy->setColumnEnabled(ArticleModel::AuthorColumn);
+    columnsProxy->setSourceModel(proxy2);
 
     setModel(columnsProxy);
     header()->setContextMenuPolicy(Qt::CustomContextMenu);
