diff -Naur netpbm-10.34-orig/converter/other/Makefile netpbm-10.34/converter/other/Makefile
--- netpbm-10.34-orig/converter/other/Makefile	2006-05-27 19:16:09.000000000 -0400
+++ netpbm-10.34/converter/other/Makefile	2006-09-18 13:03:46.000000000 -0400
@@ -73,7 +73,7 @@
 PORTBINARIES =  bmptopnm fitstopnm \
 		gemtopnm giftopnm hdifftopam infotopam \
 		pamtodjvurle pamtofits \
-		pamtohdiff pamtohtmltbl pamtopfm pamtopnm pamtouil \
+		pamtohdiff pamtohtmltbl pamtopfm          pamtouil \
 		pamtoxvmini \
 		pbmtopgm pfmtopam \
 	        pgmtopbm pgmtoppm ppmtopgm pnmtoddif \
@@ -211,9 +211,6 @@
 # In May 2002, pamtouil replaced ppmtouil
 	cd $(PKGDIR)/bin ; \
 	$(SYMLINK) pamtouil$(EXE) ppmtouil
-# In March 2005, we realized that pamtopnm obviates pnmtopnm
-	cd $(PKGDIR)/bin ; \
-	$(SYMLINK) pamtopnm$(EXE) pnmtopnm
 # In October 2005, pamtofits replaced pnmtofits
 	cd $(PKGDIR)/bin ; \
 	$(SYMLINK) pamtofits$(EXE) pnmtofits
diff -Naur netpbm-10.34-orig/other/Makefile netpbm-10.34/other/Makefile
--- netpbm-10.34-orig/other/Makefile	2006-03-26 03:53:15.000000000 -0500
+++ netpbm-10.34/other/Makefile	2006-09-18 13:03:57.000000000 -0400
@@ -7,7 +7,7 @@
 
 include $(BUILDDIR)/Makefile.config
 
-SUBDIRS = pamx
+SUBDIRS =
 
 ifneq ($(LINUXSVGALIB),NONE)
   ifneq ($(LINUXSVGAHDR_DIR),)
