diff -ur audacious-plugins-fedora-1.5.1-orig/src/mtp_up/mtp.c audacious-plugins-fedora-1.5.1/src/mtp_up/mtp.c
--- audacious-plugins-fedora-1.5.1-orig/src/mtp_up/mtp.c	2008-08-17 21:29:48.000000000 +0200
+++ audacious-plugins-fedora-1.5.1/src/mtp_up/mtp.c	2008-08-17 22:35:10.000000000 +0200
@@ -192,12 +192,12 @@
 
     g_free(from_path);
     g_free(tmp);
-    parent_id = mtp_device->default_music_folder;
+    gentrack->parent_id = mtp_device->default_music_folder;
 
 #if DEBUG
     g_print("Uploading track '%s'\n",filename);
 #endif
-    ret = LIBMTP_Send_Track_From_File(mtp_device, filename , gentrack, NULL , NULL, parent_id);
+    ret = LIBMTP_Send_Track_From_File(mtp_device, filename , gentrack, NULL , NULL);
     LIBMTP_destroy_track_t(gentrack);
     if (ret == 0) 
         g_print("Track upload finished!\n");
