--- aaa/configure.ac	2022-12-13 15:57:38.070520834 +0100
+++ bbb/configure.ac	2022-12-13 15:58:06.100487616 +0100
@@ -538,11 +538,11 @@
 	dnl and try to guess if we need -lpthread or -lthread
 	DB_TYPE=db
 	DB_EXT=.db
-	AC_LIB_LINKFLAGS([db])
+	AC_LIB_LINKFLAGS([db-5.3])
 	AC_MSG_CHECKING(db and dependent libraries)
 	saveLIBS="$LIBS"
 	havelib=0
-	for dblibs in "" "$LIBDB" ; do
+	for dblibs in "" "$LIBDB_5_3" ; do
 	    for addlibs in "" -lpthread -lthread ; do
 		LIBDB="$dblibs $addlibs"
 		export LIBDB
