Index: glade/glade-build.h
===================================================================
--- glade/glade-build.h	(revision 546)
+++ glade/glade-build.h	(working copy)
@@ -26,10 +26,7 @@
 #include <glib-object.h>
 #include <gmodule.h>
 #include <glade/glade-xml.h>
-#include <gtk/gtkwidget.h>
-#include <gtk/gtkwindow.h>
-#include <gtk/gtkaccelgroup.h>
-#include <gtk/gtkadjustment.h>
+#include <gtk/gtk.h>
 
 #include <glade/glade-parser.h>
 
Index: glade/glade-xml.h
===================================================================
--- glade/glade-xml.h	(revision 546)
+++ glade/glade-xml.h	(working copy)
@@ -23,8 +23,7 @@
 #define GLADE_XML_H
 
 #include <glib.h>
-#include <gtk/gtkwidget.h>
-#include <gtk/gtktooltips.h>
+#include <gtk/gtk.h>
 
 G_BEGIN_DECLS
 
Index: glade/glade-xml.c
===================================================================
--- glade/glade-xml.c	(revision 546)
+++ glade/glade-xml.c	(working copy)
@@ -30,11 +30,7 @@
 #include <glade/glade-build.h>
 #include <glade/glade-private.h>
 #include <gmodule.h>
-#include <gtk/gtkobject.h>
-#include <gtk/gtksignal.h>
-#include <gtk/gtkwidget.h>
-#include <gtk/gtkcontainer.h>
-#include <gtk/gtklabel.h>
+#include <gtk/gtk.h>
 #include <atk/atk.h>
 
 #ifdef ENABLE_NLS
