=== modified file 'NEWS'
--- NEWS	2012-07-11 19:48:23 +0000
+++ NEWS	2012-07-19 14:02:32 +0000
@@ -1,6 +1,11 @@
-NEWS - OpenPrinting CUPS Filters v1.0.19 - 2012-07-11
+NEWS - OpenPrinting CUPS Filters v1.0.20 - 2012-07-19
 -----------------------------------------------------
 
+CHANGES IN V1.0.20
+
+	- Made the Poppler-based filters pdftopdf and pdftoopvp build with
+	  both Poppler 0.18.x and 0.20.x (Bug #1055).
+
 CHANGES IN V1.0.19
 
 	- Fixes according to Coverity scan results (Bug #1054).

=== modified file 'configure.ac'
--- configure.ac	2012-07-11 19:48:23 +0000
+++ configure.ac	2012-07-19 14:02:32 +0000
@@ -202,6 +202,7 @@
 # ================
 POPPLER_INCLUDEDIR=`pkg-config poppler --variable includedir`
 AC_CHECK_HEADER([UGooString.h], [AC_DEFINE([HAVE_UGOOSTRING_H],,[Have UGooString.h])], [])
+AC_CHECK_HEADER([poppler/cpp/poppler-version.h], [AC_DEFINE([HAVE_CPP_POPPLER_VERSION_H],,[Define if you have Poppler's "cpp/poppler-version.h" header file.])], [])
 # FIXME:
 # Use proper c compile check not greps and then just link again poppler with all the features
 # instead of this and all the libs.

=== modified file 'filter/PDFError.h'
--- filter/PDFError.h	2012-03-29 14:06:07 +0000
+++ filter/PDFError.h	2012-07-19 14:02:32 +0000
@@ -31,6 +31,9 @@
 #include <config.h>
 #include <stdarg.h>
 #include <Error.h>
+#ifdef HAVE_CPP_POPPLER_VERSION_H
+#include "cpp/poppler-version.h"
+#endif
 
 #if POPPLER_VERSION_MAJOR > 0 || POPPLER_VERSION_MINOR >= 19
 #define pdfError(pos,...) error(errInternal,pos,__VA_ARGS__)

=== modified file 'filter/pdftoijs.cxx'
--- filter/pdftoijs.cxx	2012-06-04 16:46:44 +0000
+++ filter/pdftoijs.cxx	2012-07-19 14:02:32 +0000
@@ -30,6 +30,9 @@
 #include <config.h>
 #include <stdio.h>
 #include <stdlib.h>
+#ifdef HAVE_CPP_POPPLER_VERSION_H
+#include "cpp/poppler-version.h"
+#endif
 #include <goo/GooString.h>
 #include <goo/gmem.h>
 #include <Object.h>

=== modified file 'filter/pdftoopvp/OPVPError.h'
--- filter/pdftoopvp/OPVPError.h	2012-03-29 14:06:07 +0000
+++ filter/pdftoopvp/OPVPError.h	2012-07-19 14:02:32 +0000
@@ -31,6 +31,9 @@
 #include <config.h>
 #include <stdarg.h>
 #include <Error.h>
+#ifdef HAVE_CPP_POPPLER_VERSION_H
+#include "cpp/poppler-version.h"
+#endif
 
 #if POPPLER_VERSION_MAJOR > 0 || POPPLER_VERSION_MINOR >= 19
 #define opvpError(pos,...) error(errInternal,pos,__VA_ARGS__)

=== modified file 'filter/pdftoopvp/OPVPOutputDev.cxx'
--- filter/pdftoopvp/OPVPOutputDev.cxx	2012-03-29 09:42:35 +0000
+++ filter/pdftoopvp/OPVPOutputDev.cxx	2012-07-19 14:02:32 +0000
@@ -8,6 +8,9 @@
 //========================================================================
 
 #include <config.h>
+#ifdef HAVE_CPP_POPPLER_VERSION_H
+#include "cpp/poppler-version.h"
+#endif
 
 #ifdef USE_GCC_PRAGMAS
 #pragma implementation

=== modified file 'filter/pdftoopvp/oprs/OPVPSplash.cxx'
--- filter/pdftoopvp/oprs/OPVPSplash.cxx	2012-03-29 09:42:35 +0000
+++ filter/pdftoopvp/oprs/OPVPSplash.cxx	2012-07-19 14:02:32 +0000
@@ -5,6 +5,9 @@
 //========================================================================
 
 #include <config.h>
+#ifdef HAVE_CPP_POPPLER_VERSION_H
+#include "cpp/poppler-version.h"
+#endif
 
 #ifdef USE_GCC_PRAGMAS
 #pragma implementation

=== modified file 'filter/pdftoopvp/oprs/OPVPSplashState.cxx'
--- filter/pdftoopvp/oprs/OPVPSplashState.cxx	2012-03-29 09:42:35 +0000
+++ filter/pdftoopvp/oprs/OPVPSplashState.cxx	2012-07-19 14:02:32 +0000
@@ -11,6 +11,9 @@
 #endif
 
 #include <string.h>
+#ifdef HAVE_CPP_POPPLER_VERSION_H
+#include "cpp/poppler-version.h"
+#endif
 #include "goo/gmem.h"
 #include "splash/SplashPattern.h"
 #include "splash/SplashScreen.h"

=== modified file 'filter/pdftoopvp/oprs/OPVPSplashState.h'
--- filter/pdftoopvp/oprs/OPVPSplashState.h	2012-03-29 09:42:35 +0000
+++ filter/pdftoopvp/oprs/OPVPSplashState.h	2012-07-19 14:02:32 +0000
@@ -11,6 +11,9 @@
 #pragma interface
 #endif
 
+#ifdef HAVE_CPP_POPPLER_VERSION_H
+#include "cpp/poppler-version.h"
+#endif
 #include "splash/SplashTypes.h"
 #include "splash/SplashState.h"
 #include "splash/Splash.h"

=== modified file 'filter/pdftoopvp/oprs/OPVPSplashXPath.cxx'
--- filter/pdftoopvp/oprs/OPVPSplashXPath.cxx	2012-03-29 09:42:35 +0000
+++ filter/pdftoopvp/oprs/OPVPSplashXPath.cxx	2012-07-19 14:02:32 +0000
@@ -1,5 +1,8 @@
 #include <config.h>
 #include <stdio.h>
+#ifdef HAVE_CPP_POPPLER_VERSION_H
+#include "cpp/poppler-version.h"
+#endif
 #include "splash/Splash.h"
 #include "splash/SplashMath.h"
 #include "OPVPSplashClip.h"

=== modified file 'filter/pdftoopvp/oprs/OPVPSplashXPath.h'
--- filter/pdftoopvp/oprs/OPVPSplashXPath.h	2012-03-29 09:42:35 +0000
+++ filter/pdftoopvp/oprs/OPVPSplashXPath.h	2012-07-19 14:02:32 +0000
@@ -2,6 +2,9 @@
 #define OPVPSPLASHXPATH_H
 
 #include <config.h>
+#ifdef HAVE_CPP_POPPLER_VERSION_H
+#include "cpp/poppler-version.h"
+#endif
 #include "splash/SplashXPath.h"
 #include "OPVPSplashPath.h"
 #include "OPVPSplashState.h"

=== modified file 'filter/pdftoopvp/pdftoopvp.cxx'
--- filter/pdftoopvp/pdftoopvp.cxx	2012-06-04 16:46:44 +0000
+++ filter/pdftoopvp/pdftoopvp.cxx	2012-07-19 14:02:32 +0000
@@ -13,6 +13,9 @@
 #include <unistd.h>
 #include <ctype.h>
 #include <math.h>
+#ifdef HAVE_CPP_POPPLER_VERSION_H
+#include "cpp/poppler-version.h"
+#endif
 #include "goo/GooString.h"
 #define GSTRING_H // for old GooString.h
 #include "goo/gmem.h"

=== modified file 'filter/pdftopdf/P2PError.h'
--- filter/pdftopdf/P2PError.h	2012-03-29 14:06:07 +0000
+++ filter/pdftopdf/P2PError.h	2012-07-19 14:02:32 +0000
@@ -31,6 +31,9 @@
 #include <config.h>
 #include <stdarg.h>
 #include <Error.h>
+#ifdef HAVE_CPP_POPPLER_VERSION_H
+#include "cpp/poppler-version.h"
+#endif
 
 #if POPPLER_VERSION_MAJOR > 0 || POPPLER_VERSION_MINOR >= 19
 #define p2pError(pos,...) error(errInternal,pos,__VA_ARGS__)

=== modified file 'filter/pdftopdf/P2PFont.cxx'
--- filter/pdftopdf/P2PFont.cxx	2012-03-29 09:42:35 +0000
+++ filter/pdftopdf/P2PFont.cxx	2012-07-19 14:02:32 +0000
@@ -32,6 +32,9 @@
 #include <sys/types.h>
 #include <unistd.h>
 #include <fcntl.h>
+#ifdef HAVE_CPP_POPPLER_VERSION_H
+#include "cpp/poppler-version.h"
+#endif
 #include "goo/gmem.h"
 #include "P2PFont.h"
 #include "GfxFont.h"

=== modified file 'filter/pdftopdf/pdftopdf.cxx'
--- filter/pdftopdf/pdftopdf.cxx	2012-06-04 16:46:44 +0000
+++ filter/pdftopdf/pdftopdf.cxx	2012-07-19 14:02:32 +0000
@@ -29,6 +29,9 @@
 #include <config.h>
 #include <stdio.h>
 #include <stdlib.h>
+#ifdef HAVE_CPP_POPPLER_VERSION_H
+#include "cpp/poppler-version.h"
+#endif
 #include "goo/GooString.h"
 #include "goo/gmem.h"
 #include "Object.h"

=== modified file 'filter/pdftoraster.cxx'
--- filter/pdftoraster.cxx	2012-06-04 16:46:44 +0000
+++ filter/pdftoraster.cxx	2012-07-19 14:02:32 +0000
@@ -31,6 +31,9 @@
 #include <config.h>
 #include <stdio.h>
 #include <stdlib.h>
+#ifdef HAVE_CPP_POPPLER_VERSION_H
+#include "cpp/poppler-version.h"
+#endif
 #include "goo/GooString.h"
 #include "goo/gmem.h"
 #include "Object.h"

