--- src/cairo-xlib-surface.c
+++ src/cairo-xlib-surface.c
@@ -2023,7 +2023,7 @@ _cairo_xlib_surface_create_internal (Dis
 
     surface->buggy_repeat = FALSE;
     if (strstr (ServerVendor (dpy), "X.Org") != NULL) {
-	if (VendorRelease (dpy) <= 60802000)
+	if (VendorRelease (dpy) >= 60700000 && VendorRelease (dpy) <= 60802000)
 	    surface->buggy_repeat = TRUE;
     } else if (strstr (ServerVendor (dpy), "XFree86") != NULL) {
 	if (VendorRelease (dpy) <= 40500000)
