diff -up wrk/dao/CdrDriver.cc.wrk wrk/dao/CdrDriver.cc
--- wrk/dao/CdrDriver.cc.wrk	2016-02-16 14:06:23.158392345 +0100
+++ wrk/dao/CdrDriver.cc	2016-02-16 14:07:11.654597254 +0100
@@ -495,7 +495,7 @@ unsigned char CdrDriver::syncPattern[12]
   0, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0
 };
 
-char CdrDriver::REMOTE_MSG_SYNC_[4] = { 0xff, 0x00, 0xff, 0x00 };
+unsigned char CdrDriver::REMOTE_MSG_SYNC_[4] = { 0xff, 0x00, 0xff, 0x00 };
 
 
 /* Maps a string to the corresponding driver option value 
diff -up wrk/dao/CdrDriver.h.wrk wrk/dao/CdrDriver.h
--- wrk/dao/CdrDriver.h.wrk	2016-02-16 14:21:18.454174199 +0100
+++ wrk/dao/CdrDriver.h	2016-02-16 14:21:42.381275268 +0100
@@ -440,7 +440,7 @@ protected:
   bool fullBurn_;
 
   static unsigned char syncPattern[12];
-  static char REMOTE_MSG_SYNC_[4];
+  static unsigned char REMOTE_MSG_SYNC_[4];
 
   static int speed2Mult(int);
   static int mult2Speed(int);
