*** hplip-3.15.7_original/prnt/hpcups/CommonDefinitions.h	2015-07-13 13:18:53.000000000 +0530
--- hplip-3.15.7_latest/prnt/hpcups/CommonDefinitions.h	2015-07-28 20:07:24.018247373 +0530
***************
*** 441,447 ****
  {
      int                media_source;
      int                color_mode;
-     int                HPSPDClass;
      DUPLEXMODE         e_duplex_mode;
      int                print_borderless;
      int                krgb_mode;
--- 441,446 ----
***************
*** 466,471 ****
--- 465,471 ----
      int                integer_values[16];
      int                printer_platform_version;
      int                pre_process_raster;
+     int                HPSPDClass;
  } JobAttributes;
  
  #endif // COMMON_DEFINITIONS_H
