--- bash_completion	2006-03-01 10:20:18.000000000 -0600
+++ bash_completion	2008-01-09 22:05:55.000000000 -0600
@@ -29,13 +29,17 @@
 
 # Alter the following to reflect the location of this file.
 #
+(readonly -p | grep " BASH_COMPLETION=" 2>&1 >/dev/null) || 
 {
-  # These declarations must go within braces in order to be able to silence
-  # readonly variable errors.
   BASH_COMPLETION="${BASH_COMPLETION:-/etc/bash_completion}"
+  readonly BASH_COMPLETION
+}
+
+(readonly -p | grep " BASH_COMPLETION_DIR=" 2>&1 >/dev/null) ||
+{
   BASH_COMPLETION_DIR="${BASH_COMPLETION_DIR:=/etc/bash_completion.d}"
-} 2>/dev/null || :
-readonly BASH_COMPLETION BASH_COMPLETION_DIR
+  readonly BASH_COMPLETION_DIR
+}
 
 # Set a couple of useful vars
 #
@@ -94,7 +98,7 @@
 complete -f -X '!*.texi*' makeinfo texi2html
 complete -f -X '!*.@(?(la)tex|?(LA)TEX|texi|TEXI|dtx|DTX|ins|INS)' tex latex slitex jadetex pdfjadetex pdftex pdflatex texi2dvi
 complete -f -X '!*.@(mp3|MP3)' mpg123 mpg321 madplay
-complete -f -X '!*.@(mp?(e)g|MP?(E)G|wma|avi|AVI|asf|vob|VOB|bin|dat|vcd|ps|pes|fli|viv|rm|ram|yuv|mov|MOV|qt|QT|wmv|mp3|MP3|ogg|OGG|ogm|OGM|mp4|MP4|wav|WAV|asx|ASX|mng|MNG)' xine aaxine fbxine kaffeine
+complete -f -X '!*.@(mp?(e)g|MP?(E)G|wma|avi|AVI|asf|vob|VOB|mkv|MKV|bin|dat|vcd|ps|pes|fli|viv|rm|ram|yuv|mov|MOV|qt|QT|wmv|mp3|MP3|ogg|OGG|ogm|OGM|mp4|MP4|wav|WAV|asx|ASX|mng|MNG)' xine aaxine fbxine kaffeine gxine oxine
 complete -f -X '!*.@(avi|asf|wmv)' aviplay
 complete -f -X '!*.@(rm?(j)|ra?(m)|smi?(l))' realplay
 complete -f -X '!*.@(mpg|mpeg|avi|mov|qt)' xanim
@@ -103,7 +107,7 @@
 complete -f -X '!*.fig' xfig
 complete -f -X '!*.@(mid?(i)|MID?(I))' playmidi
 complete -f -X '!*.@(mid?(i)|MID?(I)|rmi|RMI|rcp|RCP|[gr]36|[GR]36|g18|G18|mod|MOD|xm|XM|it|IT|x3m|X3M)' timidity
-complete -f -X '*.@(o|so|so.!(conf)|a|t@(ar?(.@(Z|gz|bz?(2)))|gz|bz?(2))|rpm|zip|ZIP|gif|GIF|jp?(e)g|JP?(E)G|mp3|MP3|mp?(e)g|MPG|avi|AVI|asf|ASF|ogg|OGG|class|CLASS)' vi vim gvim rvim view rview rgvim rgview gview
+complete -f -X '*.@(o|so|so.!(conf)|a|rpm|zip|ZIP|gif|GIF|jp?(e)g|JP?(E)G|mp3|MP3|mp?(e)g|MPG|avi|AVI|asf|ASF|ogg|OGG|class|CLASS)' vi vim gvim rvim view rview rgvim rgview gview
 complete -f -X '*.@(o|so|so.!(conf)|a|rpm|gif|GIF|jp?(e)g|JP?(E)G|mp3|MP3|mp?(e)g|MPG|avi|AVI|asf|ASF|ogg|OGG|class|CLASS)' emacs
 complete -f -X '!*.@(exe|EXE|com|COM|scr|SCR|exe.so)' wine
 complete -f -X '!*.@(zip|ZIP|z|Z|gz|GZ|tgz|TGZ)' bzme
@@ -698,7 +702,7 @@
 	COMPREPLY=()
 	cur=${COMP_WORDS[COMP_CWORD]}
 
-	COMPREPLY=( $( /sbin/lsmod | \
+	COMPREPLY=( $( /bin/lsmod | \
 		  awk '{if (NR != 1 && $1 ~ /^'$cur'/) print $1}' 2>/dev/null ))
 	return 0
 }
@@ -719,7 +723,7 @@
 	# behave like lsmod for modprobe -r
 	if [ $1 = "modprobe" ] &&
 	   [ "${COMP_WORDS[1]}" = "-r" ]; then
-		COMPREPLY=( $( /sbin/lsmod | \
+		COMPREPLY=( $( /bin/lsmod | \
 				awk '{if (NR != 1 && $1 ~ /^'$cur'/) print $1}' ) )
 		return 0
 	fi
