--- NetworkManager-0.6.6/src/NetworkManager.conf.orig	2008-04-10 20:35:03.000000000 +0000
+++ NetworkManager-0.6.6/src/NetworkManager.conf	2008-04-10 20:35:38.000000000 +0000
@@ -8,7 +8,11 @@
                 <allow send_destination="org.freedesktop.NetworkManager"/>
                 <allow send_interface="org.freedesktop.NetworkManager"/>
         </policy>
-        <policy at_console="true">
+        <policy user="hal">
+                <allow send_destination="org.freedesktop.NetworkManager"/>
+                <allow send_interface="org.freedesktop.NetworkManager"/>
+        </policy>
+        <policy group="network">
                 <allow send_destination="org.freedesktop.NetworkManager"/>
                 <allow send_interface="org.freedesktop.NetworkManager"/>
         </policy>
