diff --git i/libgweather/test_libgweather.c w/libgweather/test_libgweather.c
index b10c1f16..d09aaed6 100644
--- i/libgweather/test_libgweather.c
+++ w/libgweather/test_libgweather.c
@@ -899,7 +899,6 @@ main (int argc, char *argv[])
 	g_test_add_func ("/weather/no-code-serialize", test_no_code_serialize);
 	g_test_add_func ("/weather/timezones", test_timezones);
 	g_test_add_func ("/weather/airport_distance_sanity", test_airport_distance_sanity);
-	g_test_add_func ("/weather/metar_weather_stations", test_metar_weather_stations);
 	g_test_add_func ("/weather/utc_sunset", test_utc_sunset);
 	g_test_add_func ("/weather/weather-loop-use-after-free", test_weather_loop_use_after_free);
 	/* Modifies environment, so needs to run last */
