diff -r --unified kdiff3-0.9.95/doc/de/index.docbook kdiff3-0.9.95-neu/doc/de/index.docbook
--- kdiff3-0.9.95/doc/de/index.docbook	2010-10-11 23:51:27.575286877 +0200
+++ kdiff3-0.9.95-neu/doc/de/index.docbook	2010-10-11 23:52:40.138829992 +0200
@@ -1,5 +1,5 @@
 <?xml version="1.0" ?>
-<!DOCTYPE book PUBLIC "-//KDE//DTD DocBook XML V4.1.2-Based Variant V1.1//EN" "dtd/kdex.dtd" [
+<!DOCTYPE book PUBLIC "-//KDE//DTD DocBook XML V4.2-Based Variant V1.1//EN" "dtd/kdex.dtd" [
   <!ENTITY kdiff3 "<application
 >KDiff3</application
 >">
diff -r --unified kdiff3-0.9.95/doc/fr/index.docbook kdiff3-0.9.95-neu/doc/fr/index.docbook
--- kdiff3-0.9.95/doc/fr/index.docbook	2010-10-11 23:51:27.610931550 +0200
+++ kdiff3-0.9.95-neu/doc/fr/index.docbook	2010-10-11 23:52:23.739728901 +0200
@@ -1,5 +1,5 @@
 <?xml version="1.0" ?>
-<!DOCTYPE book PUBLIC "-//KDE//DTD DocBook XML V4.1.2-Based Variant V1.1//EN" "dtd/kdex.dtd" [
+<!DOCTYPE book PUBLIC "-//KDE//DTD DocBook XML V4.2-Based Variant V1.1//EN" "dtd/kdex.dtd" [
   <!ENTITY kdiff3 "<application
 >KDiff3</application
 >">
diff -r --unified kdiff3-0.9.95/doc/it/index.docbook kdiff3-0.9.95-neu/doc/it/index.docbook
--- kdiff3-0.9.95/doc/it/index.docbook	2010-10-11 23:51:27.580378903 +0200
+++ kdiff3-0.9.95-neu/doc/it/index.docbook	2010-10-11 23:52:32.462829209 +0200
@@ -1,5 +1,5 @@
 <?xml version="1.0" ?>
-<!DOCTYPE book PUBLIC "-//KDE//DTD DocBook XML V4.1.2-Based Variant V1.1//EN" "dtd/kdex.dtd" [
+<!DOCTYPE book PUBLIC "-//KDE//DTD DocBook XML V4.2-Based Variant V1.1//EN" "dtd/kdex.dtd" [
   <!ENTITY kdiff3 "<application
 >KDiff3</application
 >">
diff -r --unified kdiff3-0.9.95/doc/nl/index.docbook kdiff3-0.9.95-neu/doc/nl/index.docbook
--- kdiff3-0.9.95/doc/nl/index.docbook	2010-10-11 23:51:27.575286877 +0200
+++ kdiff3-0.9.95-neu/doc/nl/index.docbook	2010-10-11 23:52:45.371829602 +0200
@@ -1,5 +1,5 @@
 <?xml version="1.0" ?>
-<!DOCTYPE book PUBLIC "-//KDE//DTD DocBook XML V4.1.2-Based Variant V1.1//EN" "dtd/kdex.dtd" [
+<!DOCTYPE book PUBLIC "-//KDE//DTD DocBook XML V4.2-Based Variant V1.1//EN" "dtd/kdex.dtd" [
   <!ENTITY kdiff3 "<application
 >KDiff3</application
 >">
