--- gtkspell-2.0.16/gtkspell/gtkspell.c.orig	2009-10-09 19:01:47.000000000 +0000
+++ gtkspell-2.0.16/gtkspell/gtkspell.c	2017-11-22 12:02:07.638601094 +0000
@@ -277,7 +277,7 @@
 	get_word_extents_from_mark(spell->buffer, &start, &end, spell->mark_click);
 	word = gtk_text_buffer_get_text(spell->buffer, &start, &end, FALSE);
 	
-	enchant_dict_add_to_pwl( spell->speller, word, strlen(word));
+	enchant_dict_add( spell->speller, word, strlen(word));
 
 	gtkspell_recheck_all(spell);
 
