--- configure.ac.orig	2007-07-07 12:05:37.000000000 +0000
+++ configure.ac	2007-07-07 12:12:35.000000000 +0000
@@ -71,36 +71,18 @@
 	    PRINT=$withval,
             PRINT="auto")
 
-PKG_CHECK_MODULES(GNOME_PRINT, 
-		  libgnomeprintui-2.2 >= $LIBGNOMEPRINTUI_REQUIRED,
-		  enable_gnome_print=yes, enable_gnome_print=no)
 PKG_CHECK_MODULES(GTK_PRINT,
 		  gtk+-unix-print-2.0 >= $GTK_PRINT_REQUIRED,
 		  enable_gtk_print=yes, enable_gtk_print=no)
 enable_print=no
-if test x$PRINT = xgnome -a x$enable_gnome_print = xyes; then
-   enable_print=gnome
+if test x$PRINT = xgtk -a x$enable_gtk_print = xyes; then
+  enable_print=gtk
 else
-   if test x$PRINT = xgtk -a x$enable_gtk_print = xyes; then
+  if test x$PRINT = xauto; then
+    if test x$enable_gtk_print = xyes; then
       enable_print=gtk
-   else
-      if test x$PRINT = xauto; then
-	 if test x$enable_gtk_print = xyes; then
-	    enable_print=gtk
-	 else
-	    if test x$enable_gnome_print = xyes; then
-	       enable_print=gnome
-	    fi
-	 fi
-      fi
-   fi
-fi
-
-if test x$enable_print = xgnome; then
-   AC_SUBST(GNOME_PRINT_CFLAGS)
-   AC_SUBST(GNOME_PRINT_LIBS)
-   AC_DEFINE([WITH_GNOME_PRINT],[1],[Enable GNOME Print Support.])
-   AC_DEFINE([WITH_PRINT],[1],[Enable Print Support.])
+    fi
+  fi
 fi
 
 if test x$enable_print = xgtk; then   
@@ -109,7 +91,7 @@
    AC_DEFINE([WITH_GTK_PRINT],[1],[Enable GTK Print Support.])
    AC_DEFINE([WITH_PRINT],[1],[Enable Print Support.])
 fi
-   
+ 
 AM_CONDITIONAL(WITH_GNOME_PRINT, test x$enable_print = xgnome)
 AM_CONDITIONAL(WITH_GTK_PRINT, test x$enable_print = xgtk)
 
