Index: faad2/frontend/main.c
===================================================================
--- faad2.orig/frontend/main.c
+++ faad2/frontend/main.c
@@ -916,6 +916,8 @@ int decodeMP4file(char *mp4file, char *s
                 sample_count = frameInfo.samples;
             } else {
                 sample_count = (unsigned int)(dur * frameInfo.channels);
+                if (sample_count > frameInfo.samples)
+                    sample_count = frameInfo.samples;
 
                 if (!useAacLength && !initial && (sampleId < numSamples/2) && (sample_count != frameInfo.samples))
                 {
