Index: faad2/libfaad/Makefile.am
===================================================================
--- faad2.orig/libfaad/Makefile.am
+++ faad2/libfaad/Makefile.am
@@ -3,7 +3,8 @@ lib_LTLIBRARIES = libfaad.la
 include_HEADERS = $(top_srcdir)/include/faad.h \
                   $(top_srcdir)/include/neaacdec.h
 
-libfaad_la_LDFLAGS = -lm
+libfaad_la_LDFLAGS = -version-info 1:0:0
+libfaad_la_LIBADD = -lm
 
 libfaad_la_SOURCES = bits.c cfft.c decoder.c drc.c \
 drm_dec.c error.c filtbank.c \
