diff --git a/configure.ac b/configure.ac
index 1eb9d65..e1fa4d6 100644
--- a/configure.ac
+++ b/configure.ac
@@ -93,7 +93,7 @@ AC_ARG_ENABLE([libav],
       AS_HELP_STRING([--disable-libav], [Do not build plugins depending on libav/ffmpeg (a52,lavrate...)]))
 
 if test "x$enable_libav" != "xno"; then
-  PKG_CHECK_MODULES(LIBAV, [libavcodec libavutil libavresample], [HAVE_LIBAV=yes], [HAVE_LIBAV=no])
+  PKG_CHECK_MODULES(LIBAV, [libavcodec libavutil], [HAVE_LIBAV=yes], [HAVE_LIBAV=no])
 fi
 
 if test "x$HAVE_LIBAV" = "xno"; then
@@ -121,6 +121,12 @@ AM_CONDITIONAL(HAVE_LIBAV, test x$HAVE_LIBAV = xyes)
 AC_SUBST(LIBAV_CFLAGS)
 AC_SUBST(LIBAV_LIBS)
 AC_SUBST(LIBAV_CODEC_LIBS)
+
+if test "x$enable_libav" != "xno"; then
+  PKG_CHECK_MODULES(LIBAVRESAMPLE, [libavresample], [HAVE_LIBAVRESAMPLE=yes], [HAVE_LIBAVRESAMPLE=no])
+fi
+
+AM_CONDITIONAL(HAVE_LIBAV, test x$HAVE_LIBAVRESAMLE = xyes)
 AC_SUBST(LIBAV_RESAMPLE_LIBS)
 
 AC_ARG_ENABLE([a52],
@@ -134,7 +140,7 @@ AM_CONDITIONAL(HAVE_A52, test x$HAVE_A52 = xyes)
 AC_ARG_ENABLE([lavrate],
       AS_HELP_STRING([--disable-lavrate], [Disable building of libav/ffmpeg rate plugin]))
 
-if test "x$enable_lavrate" != "xno" -a "$HAVE_LIBAV" = "yes"; then
+if test "x$enable_lavrate" != "xno" -a "$HAVE_LIBAVRESAMPLE" = "yes"; then
   HAVE_LAVRATE=yes
 fi
 AM_CONDITIONAL(HAVE_LAVRATE, test x$HAVE_LAVRATE = xyes)
