diff --git i/src/meson.build w/src/meson.build
index 40c730a0..56f4e47d 100644
--- i/src/meson.build
+++ w/src/meson.build
@@ -1,18 +1,18 @@
 fs = import('fs')
 
 if fs.exists('../.git')
-  vcs_tag(input: 'version.h.in', output: 'version.h')
+  version_h = vcs_tag(input: 'version.h.in', output: 'version.h')[0]
 else
   config = configuration_data()
   config.set ('VCS_TAG', glade_version)
-  configure_file(
+  version_h = configure_file(
     input: 'version.h.in',
     output: 'version.h',
     configuration: config
   )
 endif
 
-sources = files(
+sources = [version_h] + files(
   'glade-intro.c',
   'glade-preferences.c',
   'glade-settings.c',
