--- brltty-3.7.2/Programs/usb_linux.c.orig	2006-10-01 11:09:13.000000000 +0000
+++ brltty-3.7.2/Programs/usb_linux.c	2006-10-01 11:09:19.000000000 +0000
@@ -30,7 +30,6 @@
 #include <sys/stat.h>
 #include <sys/vfs.h>
 #include <sys/ioctl.h>
-#include <linux/compiler.h>
 #include <linux/usbdevice_fs.h>
 
 #ifndef USBDEVFS_DISCONNECT
--- brltty-3.7.2/Programs/api_client.c.orig	2006-10-01 11:09:37.000000000 +0000
+++ brltty-3.7.2/Programs/api_client.c	2006-10-01 11:09:55.000000000 +0000
@@ -74,7 +74,7 @@
 
 #ifdef linux
 #include <linux/major.h>
-#include <linux/tty.h>
+#include <linux/vt.h>
 #define MAXIMUM_VIRTUAL_CONSOLE MAX_NR_CONSOLES
 #endif /* linux */
 
