From b67326a9b271332c5bef23115d9c4ac7e7b2e6d9 Mon Sep 17 00:00:00 2001
From: Sergey V. Udaltsov<svu@gnome.org>
Date: Wed, 26  Nov  2008  23:19:41  +0000
Subject: fixed abnt2 keycodes, b.fd.o#18712

---
diff --git a/keycodes/evdev b/keycodes/evdev
index 760e2d0..9da2cd5 100644
--- a/keycodes/evdev
+++ b/keycodes/evdev
@@ -311,6 +311,7 @@ xkb_keycodes "jp106" {
 // For brazilian ABNT2 keyboard.
 xkb_keycodes "abnt2" {
     include "evdev(evdev)"
+    <KPPT> = 129;
 };
 
 // PC98
--
cgit v0.8.1-24-ge5fb
