--- thunar-file-ORIGINAL.c	2011-05-27 00:11:27.000000000 +0200
+++ thunar-1.2.2/thunar/thunar-file.c	2011-08-22 08:00:24.095066077 +0200
@@ -754,7 +754,10 @@ thunar_file_load (ThunarFile   *file,
           target_uri = 
             g_file_info_get_attribute_string (file->info,
                                               G_FILE_ATTRIBUTE_STANDARD_TARGET_URI);
-          file->is_mounted = (target_uri != NULL);
+          file->is_mounted = (target_uri != NULL)
+			? !g_file_info_get_attribute_boolean (file->info,
+                                                  G_FILE_ATTRIBUTE_MOUNTABLE_CAN_MOUNT)
+            : FALSE;
         }
     }
   else
