diff --git a/sources/pyside6/libpyside/feature_select.cpp b/sources/pyside6/libpyside/feature_select.cpp
index 9459e1ca9..c9f166107 100644
--- a/sources/pyside6/libpyside/feature_select.cpp
+++ b/sources/pyside6/libpyside/feature_select.cpp
@@ -44,7 +44,7 @@
 
 #include <shiboken.h>
 #include <sbkfeature_base.h>
-#include <signature_p.h>
+#include <signature/signature_p.h>
 
 //////////////////////////////////////////////////////////////////////////////
 //
