diff --git i/data/org.gnome.Music.appdata.xml.in.in w/data/org.gnome.Music.appdata.xml.in.in
index 633732fd..5b03d990 100644
--- i/data/org.gnome.Music.appdata.xml.in.in
+++ w/data/org.gnome.Music.appdata.xml.in.in
@@ -37,7 +37,6 @@
         <p>This release also updates translations.</p>
       </description>
     </release>
-  <releases>
     <release version="40.0" date="2021-03-21">
       <description>
         <p>
