From 0f007b28a8c3d59ff227483359a37ac29aa08648 Mon Sep 17 00:00:00 2001
From: fulalas <rcd99@zipmail.com.br>
Date: Fri, 18 Feb 2022 20:19:48 +0100
Subject: [PATCH] Fixed some window buttons not appearing in the panel (Issue
 #188)

---
 plugins/tasklist/tasklist-widget.c | 16 ----------------
 1 file changed, 16 deletions(-)

diff --git a/plugins/tasklist/tasklist-widget.c b/plugins/tasklist/tasklist-widget.c
index 83d1f09d..eb78f882 100644
--- a/plugins/tasklist/tasklist-widget.c
+++ b/plugins/tasklist/tasklist-widget.c
@@ -2872,20 +2872,6 @@ xfce_tasklist_button_geometry_changed2 (WnckWindow        *window,
 
 
 
-static gboolean
-xfce_tasklist_button_size_allocate (GtkWidget         *widget,
-                                    GdkRectangle      *allocation,
-                                    gpointer           user_data)
-{
-  XfceTasklistChild *child = user_data;
-  /* Make sure the icons have the correct size */
-  xfce_tasklist_button_icon_changed (child->window, child);
-
-  return TRUE;
-}
-
-
-
 #ifdef GDK_WINDOWING_X11
 static void
 xfce_tasklist_button_geometry_changed (WnckWindow        *window,
@@ -3561,8 +3547,6 @@ xfce_tasklist_button_new (WnckWindow   *window,
       G_CALLBACK (xfce_tasklist_button_button_release_event), child);
 
   /* monitor window changes */
-  g_signal_connect (G_OBJECT (child->button), "size-allocate",
-      G_CALLBACK (xfce_tasklist_button_size_allocate), child);
   g_signal_connect (G_OBJECT (window), "icon-changed",
       G_CALLBACK (xfce_tasklist_button_icon_changed), child);
   g_signal_connect (G_OBJECT (window), "name-changed",
