diff --git i/docs/reference/shell/meson.build w/docs/reference/shell/meson.build
index 1469ed422..bf99c9446 100644
--- i/docs/reference/shell/meson.build
+++ w/docs/reference/shell/meson.build
@@ -45,6 +45,7 @@ configure_file(
 
 gnome.gtkdoc('shell',
   main_sgml: 'shell-docs.sgml',
+  dependencies: [libshell_dep, libshell_menu_dep],
   src_dir: [
     join_paths(meson.source_root(), 'src'),
     join_paths(meson.build_root(), 'src')
diff --git i/docs/reference/st/meson.build w/docs/reference/st/meson.build
index fe89b53d4..8fdd4f4b5 100644
--- i/docs/reference/st/meson.build
+++ w/docs/reference/st/meson.build
@@ -11,6 +11,7 @@ configure_file(
 
 gnome.gtkdoc('st',
   main_sgml: 'st-docs.sgml',
+  dependencies: libst_dep,
   src_dir: [
     join_paths(meson.source_root(), 'src', 'st'),
     join_paths(meson.build_root(), 'src', 'st')
diff --git i/src/st/meson.build w/src/st/meson.build
index 97ce6134d..216700fb7 100644
--- i/src/st/meson.build
+++ w/src/st/meson.build
@@ -122,6 +122,7 @@ libst = shared_library('st-1.0',
   sources: st_gir_sources + st_non_gir_sources,
   c_args: st_cflags,
   dependencies: [clutter_dep, gtk_dep, croco_dep, x11_dep, m_dep],
+  build_rpath: mutter_typelibdir,
   install_rpath: mutter_typelibdir,
   install_dir: pkglibdir,
   install: true
