diff --git a/wscript b/wscript
index 9cedbbf..4432a9a 100644
--- a/wscript
+++ b/wscript
@@ -158,7 +158,7 @@ def test(ctx):
     cmd = os.path.join(Utils.g_module.blddir, 'talloc_testsuite')
     ret = samba_utils.RUN_COMMAND(cmd)
     print("testsuite returned %d" % ret)
-    magic_cmd = os.path.join(srcdir, 'lib', 'talloc',
+    magic_cmd = os.path.join(srcdir,
                              'test_magic_differs.sh')
 
     magic_ret = samba_utils.RUN_COMMAND(magic_cmd)
