diff --git a/exo-mount/exo-mount-hal.c b/exo-mount/exo-mount-hal.c
index bbf9ca3..d09d0db 100644
--- a/exo-mount/exo-mount-hal.c
+++ b/exo-mount/exo-mount-hal.c
@@ -741,7 +741,8 @@ exo_mount_hal_device_mount (ExoMountHalDevice *device,
         {
           option = NULL;
 
-          if (strcmp (device->fsoptions[m], "sync") == 0)
+          if (strcmp (device->fsoptions[m], "sync") == 0
+              && device->volume == NULL)
             {
               /* non-pollable drive... */
               option = g_strdup ("sync");
