From 0ac70b958c575996f760fdd987718649398fd51b Mon Sep 17 00:00:00 2001
From: Carlos Garcia Campos <carlosgc@gnome.org>
Date: Thu, 29 Apr 2010 09:53:26 +0000
Subject: Make sure dot_dir exists before creating last_settings file

Fixes bug #617154.
---
diff --git a/shell/ev-window.c b/shell/ev-window.c
index fb40f70..4c4c460 100644
--- a/shell/ev-window.c
+++ b/shell/ev-window.c
@@ -6107,7 +6107,7 @@ ev_window_create_last_settings_metadata (EvWindow *window)
 	GFile *file;
 	gchar *path;
 
-	path = g_build_filename (ev_application_get_dot_dir (EV_APP, FALSE), "last_settings", NULL);
+	path = g_build_filename (ev_application_get_dot_dir (EV_APP, TRUE), "last_settings", NULL);
 	file = g_file_new_for_path (path);
 	g_free (path);
 
--
cgit v0.8.3.1
