--- xfce4-screenshooter-1.7.9/lib/screenshooter-utils.c.orig	2010-01-02 12:42:24.000000000 +0100
+++ xfce4-screenshooter-1.7.9/lib/screenshooter-utils.c	2010-05-20 22:40:38.347861374 +0200
@@ -94,11 +94,11 @@
           g_free (title);
           title =
             g_strdup (xfce_rc_read_entry (rc, "title", _("Screenshot")));
-        }
 
-      TRACE ("Close the rc file");
+          TRACE ("Close the rc file");
 
-      xfce_rc_close (rc);
+          xfce_rc_close (rc);
+        }
     }
 
   /* And set the sd values */
