--- file-roller-2.26.2/src/fr-enum-types.c~	2009-04-30 09:03:57.000000000 +0000
+++ file-roller-2.26.2/src/fr-enum-types.c	2009-05-18 22:11:30.958610986 +0000
@@ -3,8 +3,7 @@
 
 #include <glib-object.h>
 
-/* enumerations from "/home/paolo/workspace/file-roller/src/typedefs.h" */
-#include "/home/paolo/workspace/file-roller/src/typedefs.h"
+#include "typedefs.h"
 GType
 fr_compression_get_type (void)
 {
