diff --git i/src/builder-context.c w/src/builder-context.c
index ea4d7fc9..aa094479 100644
--- i/src/builder-context.c
+++ w/src/builder-context.c
@@ -792,7 +792,7 @@ static char *rofiles_unmount_path = NULL;
 static void
 rofiles_umount_handler (int signum)
 {
-  char *argv[] = { "fusermount", "-uz", NULL,
+  char *argv[] = { "fusermount3", "-uz", NULL,
                      NULL };
 
   argv[2] = rofiles_unmount_path;
@@ -929,7 +929,7 @@ gboolean
 builder_context_disable_rofiles (BuilderContext *self,
                                  GError        **error)
 {
-  char *argv[] = { "fusermount", "-u", NULL,
+  char *argv[] = { "fusermount3", "-u", NULL,
                      NULL };
 
   if (!self->use_rofiles)
diff --git i/tests/libtest.sh w/tests/libtest.sh
index 0384c499..07ef5f7c 100644
--- i/tests/libtest.sh
+++ w/tests/libtest.sh
@@ -296,7 +296,7 @@ run_sh () {
 # fuse support is needed (and the kernel module needs to be loaded) for several
 # flatpak-builder tests
 skip_without_fuse () {
-    if [ ! -w /dev/fuse ] || ! command -v fusermount >/dev/null; then
+    if [ ! -w /dev/fuse ] || ! command -v fusermount3 >/dev/null; then
         echo "1..0 # SKIP this test requires fuse support"
         exit 0
     fi
