--- include/openbabel/shared_ptr.h.gcc5	2015-02-04 20:04:57.000000000 +0100
+++ include/openbabel/shared_ptr.h	2015-02-07 22:10:37.433163644 +0100
@@ -21,7 +21,7 @@ GNU General Public License for more deta
   #define sharedptr boost::shared_ptr
 #else
   #include <memory>
-  #if __GNUC__ == 4  //&& __GNUC_MINOR__ < 3  removed at the suggestion of Konstantin Tokarev
+  #if __GNUC__ >= 4  //&& __GNUC_MINOR__ < 3  removed at the suggestion of Konstantin Tokarev
     #include <tr1/memory>
   #endif
   using std::tr1::shared_ptr;

