diff -Nur comm-1.9.2.orig//mozilla/gfx/ots/src/os2.cc comm-1.9.2/mozilla/gfx/ots/src/os2.cc
--- comm-1.9.2.orig//mozilla/gfx/ots/src/os2.cc	2011-04-14 11:37:21.000000000 -0700
+++ comm-1.9.2/mozilla/gfx/ots/src/os2.cc	2011-04-29 09:10:23.635807433 -0700
@@ -2,6 +2,8 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+#include <cstddef>
+
 #include "os2.h"
 
 #include "head.h"
