diff -Nur comm-release.orig/mozilla/ipc/chromium/src/base/file_util.cc comm-release/mozilla/ipc/chromium/src/base/file_util.cc
--- comm-release.orig/mozilla/ipc/chromium/src/base/file_util.cc	2012-04-03 18:38:11.474783536 +0000
+++ comm-release/mozilla/ipc/chromium/src/base/file_util.cc	2012-04-03 18:43:29.218995578 +0000
@@ -8,7 +8,7 @@
 #include <io.h>
 #endif
 #include <stdio.h>
-#if defined(ANDROID)
+#if defined(ANDROID) || defined(OS_POSIX)
 #include <unistd.h>
 #endif
 
diff -Nur comm-release.orig/mozilla/ipc/chromium/src/base/file_util_linux.cc comm-release/mozilla/ipc/chromium/src/base/file_util_linux.cc
--- comm-release.orig/mozilla/ipc/chromium/src/base/file_util_linux.cc	2012-04-03 18:38:11.381450924 +0000
+++ comm-release/mozilla/ipc/chromium/src/base/file_util_linux.cc	2012-04-03 18:43:29.218995578 +0000
@@ -5,6 +5,9 @@
 #include "base/file_util.h"
 
 #include <fcntl.h>
+#if defined(ANDROID) || defined(OS_POSIX)
+#include <unistd.h>
+#endif
 
 #include <string>
 #include <vector>
diff -Nur comm-release.orig/mozilla/ipc/chromium/src/base/message_pump_libevent.cc comm-release/mozilla/ipc/chromium/src/base/message_pump_libevent.cc
--- comm-release.orig/mozilla/ipc/chromium/src/base/message_pump_libevent.cc	2012-04-03 18:38:11.438117153 +0000
+++ comm-release/mozilla/ipc/chromium/src/base/message_pump_libevent.cc	2012-04-03 18:43:29.218995578 +0000
@@ -6,6 +6,9 @@
 
 #include <errno.h>
 #include <fcntl.h>
+#if defined(ANDROID) || defined(OS_POSIX)
+#include <unistd.h>
+#endif
 
 #include "eintr_wrapper.h"
 #include "base/logging.h"
diff -Nur comm-release.orig/mozilla/ipc/chromium/src/base/time_posix.cc comm-release/mozilla/ipc/chromium/src/base/time_posix.cc
--- comm-release.orig/mozilla/ipc/chromium/src/base/time_posix.cc	2012-04-03 18:38:11.418117308 +0000
+++ comm-release/mozilla/ipc/chromium/src/base/time_posix.cc	2012-04-03 18:43:29.218995578 +0000
@@ -14,6 +14,8 @@
 #include <time.h>
 #endif
 
+#include <unistd.h>
+
 #include <limits>
 
 #include "base/basictypes.h"
diff -Nur comm-release.orig/mozilla/toolkit/crashreporter/client/Makefile.in comm-release/mozilla/toolkit/crashreporter/client/Makefile.in
--- comm-release.orig/mozilla/toolkit/crashreporter/client/Makefile.in	2012-04-03 18:38:56.101105510 +0000
+++ comm-release/mozilla/toolkit/crashreporter/client/Makefile.in	2012-04-03 18:43:29.218995578 +0000
@@ -69,6 +69,9 @@
   $(STDCXX_COMPAT) \
   $(NULL)
 
+# Needed for moz_free
+LIBS += -L$(LIBXUL_DIST)/lib -lmozalloc
+
 ifeq ($(OS_ARCH),WINNT)
 CPPSRCS += crashreporter_win.cpp
 LIBS += \
