From 3bba8eb773a96339fb841c76b04f2d4fd3317814 Mon Sep 17 00:00:00 2001
From: Panu Matilainen <pmatilai@laiskiainen.org>
Date: Wed, 23 Mar 2011 21:13:40 +0000
Subject: Fix GDM_LANG usage to be compatible with GDM3 (bug #7400).

---
diff --git a/xfce4-session/main.c b/xfce4-session/main.c
index ff6e324..fcf69f8 100644
--- a/xfce4-session/main.c
+++ b/xfce4-session/main.c
@@ -103,7 +103,7 @@ setup_environment (void)
 
   /* this is for compatibility with the GNOME Display Manager */
   lang = g_getenv ("GDM_LANG");
-  if (lang != NULL)
+  if (lang != NULL && strlen (lang) > 0)
     {
       xfce_setenv ("LANG", lang, TRUE);
       xfce_unsetenv ("GDM_LANG");
--
cgit 
