--- config_usb_printer.py	2013-05-19 02:23:17.506716304 +0300
+++ config_usb_printer.py.new	2013-05-19 02:23:30.140049848 +0300
@@ -257,7 +257,7 @@
        # ******************************* RUNNING FIRMWARE DOWNLOAD TO DEVICE FOR SUPPORTED PRINTER'S
        fw_download_req = mq.get('fw-download', False)
        if fw_download_req:
-           fw_cmd = utlis.which('hp-firmware', True)
+           fw_cmd = utils.which('hp-firmware', True)
            if fw_cmd:
                 fw_cmd += " -y3 -s %s"%param
                 log.debug(fw_cmd)
