From 62dd1f24ef08e2f3e4bc9b15d9a48acc7c88d650 Mon Sep 17 00:00:00 2001
From: Reuben Thomas <rrt@sc3d.org>
Date: Fri, 8 Dec 2017 15:34:21 +0000
Subject: [PATCH] Fix #170: potential null pointer dereference

---
 providers/enchant_hspell.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/providers/enchant_hspell.c b/providers/enchant_hspell.c
index c7028f1..0e5bbac 100644
--- a/providers/enchant_hspell.c
+++ b/providers/enchant_hspell.c
@@ -102,6 +102,13 @@ hspell_dict_suggest (EnchantDict * me, const char *const word,
 	gsize length;
 	char *iso_word = g_convert (word, len, "iso8859-8", "utf-8", NULL, &length, NULL);
 	
+	/* check we got a result */
+	if (iso_word == NULL)
+		{
+			g_free (iso_word);
+			return NULL;
+		}
+
 	/* get suggestions */
 	struct corlist cl;
 	corlist_init (&cl);
