diff -up k3b-2.0.2/src/CMakeLists.txt.no-webkit k3b-2.0.2/src/CMakeLists.txt
--- k3b-2.0.2/src/CMakeLists.txt.no-webkit	2012-05-04 12:30:46.634241618 +0200
+++ k3b-2.0.2/src/CMakeLists.txt	2012-05-04 12:30:56.980516383 +0200
@@ -287,7 +287,6 @@ target_link_libraries(k3b_bin
   ${KDE4_KUTILS_LIBS}
   ${KDE4_KDE3SUPPORT_LIBS}
   ${KDE4_SOLID_LIBS}
-  ${QT_QTWEBKIT_LIBRARY}
   ${X11_LIBRARIES}
   )
 
diff -up k3b-2.0.2/src/k3bdiskinfoview.cpp.no-webkit k3b-2.0.2/src/k3bdiskinfoview.cpp
--- k3b-2.0.2/src/k3bdiskinfoview.cpp.no-webkit	2012-05-04 12:29:13.994781781 +0200
+++ k3b-2.0.2/src/k3bdiskinfoview.cpp	2012-05-04 13:10:07.748401670 +0200
@@ -34,7 +34,6 @@
 #include <qpalette.h>
 #include <qpixmap.h>
 #include <QtGui/QTextBrowser>
-#include <QtWebKit/QWebView>
 
 #include <klocale.h>
 #include <kstandarddirs.h>
@@ -61,7 +60,7 @@ K3b::DiskInfoView::DiskInfoView( QWidget
                               Device::STATE_ALL|Device::STATE_NO_MEDIA|Device::STATE_UNKNOWN,
                               parent )
 {
-    m_infoView = new QWebView( this );
+    m_infoView = new QTextBrowser( this );
     setMainWidget( m_infoView );
 }
 
diff -up k3b-2.0.2/src/k3bdiskinfoview.h.no-webkit k3b-2.0.2/src/k3bdiskinfoview.h
--- k3b-2.0.2/src/k3bdiskinfoview.h.no-webkit	2012-05-04 12:50:25.753604831 +0200
+++ k3b-2.0.2/src/k3bdiskinfoview.h	2012-05-04 13:09:15.232985545 +0200
@@ -19,7 +19,7 @@
 
 #include "k3bmediacontentsview.h"
 
-class QWebView;
+class QTextBrowser;
 
 namespace K3b {
 class DiskInfoView : public MediaContentsView
@@ -38,7 +38,7 @@ private:
     QString createIso9660InfoItems( const Iso9660SimplePrimaryDescriptor& iso );
     QString createTrackItems( const Medium& medium );
 
-    QWebView* m_infoView;
+    QTextBrowser* m_infoView;
 };
 }
 
