diff --git a/src/mltconnection.cpp b/src/mltconnection.cpp
index 13d0e8d8c..2820c8f7c 100644
--- a/src/mltconnection.cpp
+++ b/src/mltconnection.cpp
@@ -136,14 +136,14 @@ void MltConnection::locateMeltAndProfilesPath(const QString &mltPath)
 #endif
     QString meltPath;
     if (qEnvironmentVariableIsSet("MLT_PREFIX")) {
-        meltPath = qgetenv("MLT_PREFIX") + QStringLiteral("/bin/melt") + exeSuffix;
+        meltPath = qgetenv("MLT_PREFIX") + QStringLiteral("/bin/melt6") + exeSuffix;
     } else {
         meltPath = KdenliveSettings::rendererpath();
     }
     if (!QFile::exists(meltPath)) {
-        meltPath = QDir::cleanPath(profilePath + QStringLiteral("/../../../bin/melt")) + exeSuffix;
+        meltPath = QDir::cleanPath(profilePath + QStringLiteral("/../../../bin/melt6")) + exeSuffix;
         if (!QFile::exists(meltPath)) {
-            meltPath = QStandardPaths::findExecutable("melt");
+            meltPath = QStandardPaths::findExecutable("melt6");
             if (meltPath.isEmpty()) {
                 meltPath = QStandardPaths::findExecutable("mlt-melt");
             }
