--- lirc-old/drivers/lirc_dev/lirc_dev.c	2006-07-24 01:24:35.000000000 +0200
+++ lirc/drivers/lirc_dev/lirc_dev.c	2006-10-17 18:22:24.000000000 +0200
@@ -32,7 +32,6 @@
 #error "**********************************************************"
 #endif
 
-#include <linux/config.h>
 #include <linux/module.h>
 #include <linux/kernel.h>
 #include <linux/sched.h>
--- lirc-old/drivers/lirc_sir/lirc_sir.c	2006-01-21 18:43:36.000000000 +0100
+++ lirc/drivers/lirc_sir/lirc_sir.c	2006-10-17 18:22:24.000000000 +0200
@@ -46,7 +46,6 @@
 # include <config.h>
 #endif
  
-#include <linux/config.h>
 
 #if !defined(LIRC_ON_SA1100) && !defined(CONFIG_SERIAL_MODULE)
 #warning "******************************************"
--- lirc-old/drivers/lirc_serial/lirc_serial.c	2006-01-21 18:43:36.000000000 +0100
+++ lirc/drivers/lirc_serial/lirc_serial.c	2006-10-17 18:22:24.000000000 +0200
@@ -61,7 +61,6 @@
 #error "**********************************************************"
 #endif
 
-#include <linux/config.h>
 
 #if defined(CONFIG_SERIAL) || defined(CONFIG_SERIAL_8250)
 #warning "******************************************"
--- lirc-old/drivers/lirc_bt829/lirc_bt829.c	2005-11-27 12:03:01.000000000 +0100
+++ lirc/drivers/lirc_bt829/lirc_bt829.c	2006-10-17 18:22:24.000000000 +0200
@@ -23,7 +23,6 @@
 #error "This driver needs kernel version 2.4.0 or higher"
 #endif
 
-#include <linux/config.h>
 #include <linux/kernel.h>
 #include <linux/module.h>
 #include <linux/threads.h>
--- lirc-old/drivers/lirc_cmdir/lirc_cmdir.c	2005-09-25 14:03:39.000000000 +0200
+++ lirc/drivers/lirc_cmdir/lirc_cmdir.c	2006-10-17 18:22:24.000000000 +0200
@@ -35,7 +35,6 @@
 #error "**********************************************************"
 #endif
 
-#include <linux/config.h>
 #include <linux/module.h>
 #include <linux/errno.h>
 #include <linux/signal.h>
--- lirc-old/drivers/lirc_sasem/lirc_sasem.c	2006-07-24 01:24:35.000000000 +0200
+++ lirc/drivers/lirc_sasem/lirc_sasem.c	2006-10-17 18:22:24.000000000 +0200
@@ -58,7 +58,6 @@
 #error "*** Sorry, this driver requires kernel version 2.4.22 or higher"
 #endif
 
-#include <linux/config.h>
 
 #include <linux/errno.h>
 #include <linux/init.h>
--- lirc-old/drivers/lirc_igorplugusb/lirc_igorplugusb.c	2006-03-04 23:36:38.000000000 +0100
+++ lirc/drivers/lirc_igorplugusb/lirc_igorplugusb.c	2006-10-17 18:22:24.000000000 +0200
@@ -47,7 +47,6 @@
 #error "*******************************************************"
 #endif
 
-#include <linux/config.h>
 #include <linux/module.h>
 #include <linux/kernel.h>
 #include <linux/kmod.h>
--- lirc-old/drivers/lirc_imon/lirc_imon.c	2006-07-24 01:24:35.000000000 +0200
+++ lirc/drivers/lirc_imon/lirc_imon.c	2006-10-17 18:22:24.000000000 +0200
@@ -49,7 +49,6 @@
 #error "*** Sorry, this driver requires kernel version 2.4.22 or higher"
 #endif
 
-#include <linux/config.h>
 
 #include <linux/errno.h>
 #include <linux/init.h>
--- lirc-old/drivers/lirc_it87/lirc_it87.c	2006-01-21 18:43:36.000000000 +0100
+++ lirc/drivers/lirc_it87/lirc_it87.c	2006-10-17 18:22:25.000000000 +0200
@@ -38,7 +38,6 @@
 # include <config.h>
 #endif
  
-#include <linux/config.h>
 
 
 #include <linux/sched.h>
--- lirc-old/drivers/lirc_streamzap/lirc_streamzap.c	2006-03-04 23:36:39.000000000 +0100
+++ lirc/drivers/lirc_streamzap/lirc_streamzap.c	2006-10-17 18:22:25.000000000 +0200
@@ -38,7 +38,6 @@
 #error "*******************************************************"
 #endif
 
-#include <linux/config.h>
 #include <linux/kernel.h>
 #include <linux/errno.h>
 #include <linux/init.h>
--- lirc-old/drivers/lirc_mceusb2/lirc_mceusb2.c	2006-07-25 14:56:30.000000000 +0200
+++ lirc/drivers/lirc_mceusb2/lirc_mceusb2.c	2006-10-17 18:22:25.000000000 +0200
@@ -37,7 +37,6 @@
 #error "*******************************************************"
 #endif
 
-#include <linux/config.h>
 
 #include <linux/kernel.h>
 #include <linux/errno.h>
--- lirc-old/drivers/lirc_atiusb/lirc_atiusb.c	2006-03-04 23:36:38.000000000 +0100
+++ lirc/drivers/lirc_atiusb/lirc_atiusb.c	2006-10-17 18:22:25.000000000 +0200
@@ -39,7 +39,6 @@
 #error "*******************************************************"
 #endif
 
-#include <linux/config.h>
 
 #include <linux/kernel.h>
 #include <linux/errno.h>
--- lirc-old/drivers/lirc_parallel/lirc_parallel.c	2005-09-22 22:10:37.000000000 +0200
+++ lirc/drivers/lirc_parallel/lirc_parallel.c	2006-10-17 18:22:25.000000000 +0200
@@ -40,7 +40,6 @@
 #error "**********************************************************"
 #endif
 
-#include <linux/config.h>
 
 #ifdef CONFIG_SMP
 #error "--- Sorry, this driver is not SMP safe. ---"
@@ -50,7 +49,6 @@
 #include <linux/sched.h>
 #include <linux/errno.h>
 #include <linux/signal.h>
-#include <linux/config.h>
 #include <linux/fs.h>
 #include <linux/kernel.h>
 #include <linux/ioport.h>
--- lirc-old/drivers/lirc_mceusb/lirc_mceusb.c	2006-03-04 23:36:39.000000000 +0100
+++ lirc/drivers/lirc_mceusb/lirc_mceusb.c	2006-10-17 18:22:25.000000000 +0200
@@ -88,7 +88,6 @@
  *
  */
 
-#include <linux/config.h>
 #include <linux/kernel.h>
 #include <linux/errno.h>
 #include <linux/init.h>
