diff -u -r kernel/nvidia-uvm/uvm_migrate_pageable.c kernel-5.16/nvidia-uvm/uvm_migrate_pageable.c
--- kernel/nvidia-uvm/uvm_migrate_pageable.c	2021-10-27 17:14:51.000000000 +0000
+++ kernel-5.16/nvidia-uvm/uvm_migrate_pageable.c	2022-01-10 21:03:33.937906621 +0000
@@ -406,7 +406,11 @@
         uvm_push_set_flag(&push, UVM_PUSH_FLAG_CE_NEXT_MEMBAR_NONE);
         copying_gpu->parent->ce_hal->memset_8(&push, dst_address, 0, PAGE_SIZE);
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 16, 0)
         dst[i] = migrate_pfn(page_to_pfn(dst_page)) | MIGRATE_PFN_LOCKED;
+#else
+        dst[i] = migrate_pfn(page_to_pfn(dst_page));
+#endif
     }
 
     if (copying_gpu) {
@@ -490,7 +494,11 @@
         uvm_push_set_flag(&push, UVM_PUSH_FLAG_CE_NEXT_MEMBAR_NONE);
         copying_gpu->parent->ce_hal->memcopy(&push, dst_address, src_address, PAGE_SIZE);
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 16, 0)
         dst[i] = migrate_pfn(page_to_pfn(dst_page)) | MIGRATE_PFN_LOCKED;
+#else
+        dst[i] = migrate_pfn(page_to_pfn(dst_page));
+#endif
     }
 
     // TODO: Bug 1766424: If the destination is a GPU and the copy was done by
