From d7541bca9533fc12cbf1d63fea122c50aac501b8 Mon Sep 17 00:00:00 2001
From: Simon Steinbeiss <simon.steinbeiss@elfenbeinturm.at>
Date: Wed, 5 May 2021 09:11:13 +0200
Subject: [PATCH] keyboard: Correctly initialize setting (Fixes #310)

---
 dialogs/keyboard-settings/xfce-keyboard-settings.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/dialogs/keyboard-settings/xfce-keyboard-settings.c b/dialogs/keyboard-settings/xfce-keyboard-settings.c
index e031b374..69f5d9f0 100644
--- a/dialogs/keyboard-settings/xfce-keyboard-settings.c
+++ b/dialogs/keyboard-settings/xfce-keyboard-settings.c
@@ -352,6 +352,7 @@ xfce_keyboard_settings_constructed (GObject *object)
   GObject              *xkb_layout_up_button;
   GObject              *xkb_layout_down_button;
 #endif /* HAVE_LIBXKLAVIER */
+  gboolean              xkb_disable_setting;
 
   /* XKB settings */
   xkb_key_repeat_check = gtk_builder_get_object (GTK_BUILDER (settings), "xkb_key_repeat_check");
@@ -454,6 +455,8 @@ xfce_keyboard_settings_constructed (GObject *object)
                     "state-set",
                     G_CALLBACK (xfce_keyboard_settings_system_default_cb),
                     settings);
+  xkb_disable_setting = xfconf_channel_get_bool (settings->priv->keyboard_layout_channel, "/Default/XkbDisable", TRUE);
+  gtk_switch_set_state (GTK_SWITCH (xkb_use_system_default_switch), xkb_disable_setting);
   xfce_keyboard_settings_update_sensitive (GTK_SWITCH (xkb_use_system_default_switch), settings);
 
   /* Keyboard model combo */
-- 
GitLab

