diff -rupN lightdm-1.16.2.orig/src/session-child.c lightdm-1.16.2/src/session-child.c
--- lightdm-1.16.2.orig/src/session-child.c	2015-09-23 11:21:10.110684089 +0200
+++ lightdm-1.16.2/src/session-child.c	2015-09-23 11:21:53.502937957 +0200
@@ -417,7 +417,7 @@ session_child_run (int argc, char **argv
 
             updwtmpx ("/var/log/btmp", &ut);
 
-            audit_event (AUDIT_USER_LOGIN, username, -1, remote_host_name, tty, FALSE);
+            //audit_event (AUDIT_USER_LOGIN, username, -1, remote_host_name, tty, FALSE);
         }
 
         /* Check account is valid */
@@ -734,7 +734,7 @@ session_child_run (int argc, char **argv
             endutxent ();
             updwtmpx ("/var/log/wtmp", &ut);
 
-            audit_event (AUDIT_USER_LOGIN, username, uid, remote_host_name, tty, TRUE);
+            //audit_event (AUDIT_USER_LOGIN, username, uid, remote_host_name, tty, TRUE);
         }
 
         waitpid (child_pid, &return_code, 0);
@@ -772,7 +772,7 @@ session_child_run (int argc, char **argv
             endutxent ();
             updwtmpx ("/var/log/wtmp", &ut);
 
-            audit_event (AUDIT_USER_LOGOUT, username, uid, remote_host_name, tty, TRUE);
+            //audit_event (AUDIT_USER_LOGOUT, username, uid, remote_host_name, tty, TRUE);
         }
     }
 
