--- widget/src/gtk2/nsWindow.cpp	2006-01-09 05:48:56.000000000 +0000
+++ widget/src/gtk2/nsWindow.cpp	2006-08-09 18:51:41.000000000 +0000
@@ -3512,15 +3512,19 @@
         gdkcursor = gdk_cursor_new(GDK_FLEUR);
         break;
     case eCursor_help:
+        gdkcursor = gdk_cursor_new_from_name(gdk_display_get_default(), "help");
         newType = MOZ_CURSOR_QUESTION_ARROW;
         break;
     case eCursor_copy: // CSS3
+        gdkcursor = gdk_cursor_new_from_name(gdk_display_get_default(), "copy");
         newType = MOZ_CURSOR_COPY;
         break;
     case eCursor_alias:
+        gdkcursor = gdk_cursor_new_from_name(gdk_display_get_default(), "alias");
         newType = MOZ_CURSOR_ALIAS;
         break;
     case eCursor_context_menu:
+        gdkcursor = gdk_cursor_new_from_name(gdk_display_get_default(), "context-menu");
         newType = MOZ_CURSOR_CONTEXT_MENU;
         break;
     case eCursor_cell:
@@ -3533,6 +3537,7 @@
         newType = MOZ_CURSOR_HAND_GRABBING;
         break;
     case eCursor_spinning:
+        gdkcursor = gdk_cursor_new_from_name(gdk_display_get_default(), "wait"); // or maybe "progress" ?
         newType = MOZ_CURSOR_SPINNING;
         break;
     case eCursor_zoom_in:
@@ -3542,25 +3547,34 @@
         newType = MOZ_CURSOR_ZOOM_OUT;
         break;
     case eCursor_not_allowed:
+        gdkcursor = gdk_cursor_new_from_name(gdk_display_get_default(), "not-allowed");
+        newType = MOZ_CURSOR_NOT_ALLOWED;
+        break;
     case eCursor_no_drop:
+        gdkcursor = gdk_cursor_new_from_name(gdk_display_get_default(), "no-drop");
         newType = MOZ_CURSOR_NOT_ALLOWED;
         break;
     case eCursor_col_resize:
+        gdkcursor = gdk_cursor_new_from_name(gdk_display_get_default(), "col-resize");
         newType = MOZ_CURSOR_COL_RESIZE;
         break;
     case eCursor_row_resize:
+        gdkcursor = gdk_cursor_new_from_name(gdk_display_get_default(), "row-resize");
         newType = MOZ_CURSOR_ROW_RESIZE;
         break;
     case eCursor_vertical_text:
+        gdkcursor = gdk_cursor_new_from_name(gdk_display_get_default(), "vertical-text");
         newType = MOZ_CURSOR_VERTICAL_TEXT;
         break;
     case eCursor_all_scroll:
         gdkcursor = gdk_cursor_new(GDK_FLEUR);
         break;
     case eCursor_nesw_resize:
+        gdkcursor = gdk_cursor_new_from_name(gdk_display_get_default(), "nesw-resize");
         newType = MOZ_CURSOR_NESW_RESIZE;
         break;
     case eCursor_nwse_resize:
+        gdkcursor = gdk_cursor_new_from_name(gdk_display_get_default(), "nwse-resize");
         newType = MOZ_CURSOR_NWSE_RESIZE;
         break;
     case eCursor_ns_resize:
@@ -3577,7 +3591,7 @@
 
     // if by now we dont have a xcursor, this means we have to make a
     // custom one
-    if (newType != 0xff) {
+    if (!gdkcursor && newType != 0xff) {
         gdk_color_parse("#000000", &fg);
         gdk_color_parse("#ffffff", &bg);
 
