Index: libgweather-3.5.1/configure.ac
===================================================================
--- libgweather-3.5.1.orig/configure.ac	2012-07-05 10:33:10.277373254 +1200
+++ libgweather-3.5.1/configure.ac	2012-07-05 10:33:44.957372053 +1200
@@ -97,9 +97,6 @@
 dnl gettext
 IT_PROG_INTLTOOL([0.50.0])
 
-AM_GNU_GETTEXT_VERSION([0.18])
-AM_GNU_GETTEXT([external])
-
 GETTEXT_PACKAGE=libgweather-3.0
 AC_SUBST(GETTEXT_PACKAGE)
 AC_DEFINE_UNQUOTED(GETTEXT_PACKAGE, "$GETTEXT_PACKAGE", [Gettext Package])
