diff -Nur NVIDIA-Linux-x86_64-325.15-no-compat32.orig/kernel/nv-drm.c NVIDIA-Linux-x86_64-325.15-no-compat32/kernel/nv-drm.c
--- NVIDIA-Linux-x86_64-325.15-no-compat32.orig/kernel/nv-drm.c	2013-08-01 02:49:26.000000000 +0200
+++ NVIDIA-Linux-x86_64-325.15-no-compat32/kernel/nv-drm.c	2013-11-06 01:16:40.547079457 +0100
@@ -106,7 +106,6 @@
     .unlocked_ioctl = drm_ioctl,
     .mmap = drm_gem_mmap,
     .poll = drm_poll,
-    .fasync = drm_fasync,
     .read = drm_read,
     .llseek = noop_llseek,
 };
@@ -202,12 +201,7 @@
     memset(&nv_obj->base, 0, sizeof(nv_obj->base));
     nv_obj->pages = pages;
 
-    ret = drm_gem_private_object_init(nvl->drm, &nv_obj->base, size);
-    if (ret)
-    {
-        status = RM_ERR_OPERATING_SYSTEM;
-        goto done;
-    }
+    drm_gem_private_object_init(nvl->drm, &nv_obj->base, size);
 
     ret = drm_gem_handle_create(file_priv, &nv_obj->base, handle);
     if (ret)
diff -Nur NVIDIA-Linux-x86_64-325.15-no-compat32.orig/kernel/nv-linux.h NVIDIA-Linux-x86_64-325.15-no-compat32/kernel/nv-linux.h
--- NVIDIA-Linux-x86_64-325.15-no-compat32.orig/kernel/nv-linux.h	2013-08-01 02:49:26.000000000 +0200
+++ NVIDIA-Linux-x86_64-325.15-no-compat32/kernel/nv-linux.h	2013-11-06 01:15:41.838303440 +0100
@@ -958,7 +958,11 @@
 #endif
 
 #if !defined(NV_VMWARE)
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3, 11, 0)
 #define NV_NUM_PHYSPAGES                num_physpages
+#else
+#define NV_NUM_PHYSPAGES                get_num_physpages()
+#endif
 #define NV_GET_CURRENT_PROCESS()        current->tgid
 #define NV_IN_ATOMIC()                  in_atomic()
 #define NV_LOCAL_BH_DISABLE()           local_bh_disable()
