diff -ur unison-2.9.1/ubase/uprintf.ml unison-2.9.1.patched/ubase/uprintf.ml
--- unison-2.9.1/ubase/uprintf.ml	2002-04-11 07:13:23.000000000 +0200
+++ unison-2.9.1.patched/ubase/uprintf.ml	2004-08-08 14:10:09.784776851 +0200
@@ -10,8 +10,8 @@
 (*                                                                     *)
 (***********************************************************************)
 
-external format_int: string -> int -> string = "format_int"
-external format_float: string -> float -> string = "format_float"
+external format_int: string -> int -> string = "caml_format_int"
+external format_float: string -> float -> string = "caml_format_float"
 
 let fprintf outchan doafter format =
   let format = (Obj.magic format : string) in
