diff --git i/docs/buildsystem-and-repos.md w/docs/buildsystem-and-repos.md
index e265ee7a..2e9250d5 100644
--- i/docs/buildsystem-and-repos.md
+++ w/docs/buildsystem-and-repos.md
@@ -151,7 +151,7 @@ rofiles-fuse exampleos-build mnt
 # Now run global "triggers", generate cache files:
 ldconfig -r mnt
   (Insert other programs here)
-fusermount -u mnt
+fusermount3 -u mnt
 ostree --repo=build-repo commit -b exampleos/x86_64/standard --link-checkout-speedup exampleos-build
 ```
 
diff --git i/man/rofiles-fuse.xml w/man/rofiles-fuse.xml
index f4d23739..e8569130 100644
--- i/man/rofiles-fuse.xml
+++ w/man/rofiles-fuse.xml
@@ -89,7 +89,7 @@ $ mkdir mnt/anewdir
 $ rm mnt/someoriginalcontent -rf
 
 # Commit and cleanup
-$ fusermount -u mnt
+$ fusermount3 -u mnt
 $ ostree --repo=repo commit --link-checkout-speedup -b somebranch -s 'Commit new content' --tree=dir=branch-checkout
 $ rm mnt branch-checkout -rf
 	</programlisting>
diff --git i/tests/libtest.sh w/tests/libtest.sh
index 264094b1..cd1e0986 100755
--- i/tests/libtest.sh
+++ w/tests/libtest.sh
@@ -683,7 +683,7 @@ skip_one_without_strace_fault_injection() {
 }
 
 skip_without_fuse () {
-    fusermount --version >/dev/null 2>&1 || skip "no fusermount"
+    fusermount3 --version >/dev/null 2>&1 || skip "no fusermount3"
 
     capsh --print | grep -q 'Bounding set.*[^a-z]cap_sys_admin' || \
         skip "No cap_sys_admin in bounding set, can't use FUSE"
diff --git i/tests/test-demo-buildsystem.sh w/tests/test-demo-buildsystem.sh
index 2dc78dc3..50995b87 100755
--- i/tests/test-demo-buildsystem.sh
+++ w/tests/test-demo-buildsystem.sh
@@ -55,7 +55,7 @@ exampleos_recompose() {
     # Now that we have our rootfs, run triggers
     rofiles-fuse exampleos-build mnt
     demo_triggers mnt/
-    fusermount -u mnt
+    fusermount3 -u mnt
     
     # Then we commit it, using --link-checkout-speedup to effectively
     # only re-checksum the ldconfig file.  We also have dummy commit
diff --git i/tests/test-rofiles-fuse.sh w/tests/test-rofiles-fuse.sh
index a56a76c6..c61945d6 100755
--- i/tests/test-rofiles-fuse.sh
+++ w/tests/test-rofiles-fuse.sh
@@ -37,7 +37,7 @@ $OSTREE checkout -H test2 checkout-test2
 
 rofiles-fuse checkout-test2 mnt
 cleanup_fuse() {
-    fusermount -u ${test_tmpdir}/mnt || true
+    fusermount3 -u ${test_tmpdir}/mnt || true
 }
 libtest_exit_cmds+=(cleanup_fuse)
 assert_file_has_content mnt/firstfile first
@@ -133,7 +133,7 @@ echo "ok flock"
 
 copyup_reset() {
     cd ${test_tmpdir}
-    fusermount -u mnt
+    fusermount3 -u mnt
     rm checkout-test2 -rf
     $OSTREE checkout -H test2 checkout-test2
     rofiles-fuse --copyup checkout-test2 mnt
