From 03ff202f818fe3f075a6c864ab688b57b1bb1cb4 Mon Sep 17 00:00:00 2001
From: Nick Schermer <nick@xfce.org>
Date: Thu, 27 Jan 2011 21:18:45 +0000
Subject: Fix segfault when creating plugin in 4.8 panel.

Used invalid function type to construct the plugin.
---
diff --git a/src/xfcetimer.c b/src/xfcetimer.c
index c27d37a..de7bd2b 100644
--- a/src/xfcetimer.c
+++ b/src/xfcetimer.c
@@ -42,7 +42,7 @@
 
 #include "xfcetimer.h"
 
-static gboolean create_plugin_control (XfcePanelPlugin *plugin);
+static void create_plugin_control (XfcePanelPlugin *plugin);
 XFCE_PANEL_PLUGIN_REGISTER_EXTERNAL(create_plugin_control);
 
 void make_menu(plugin_data *pd);
@@ -1366,7 +1366,7 @@ static void plugin_create_options (XfcePanelPlugin *plugin,plugin_data *pd) {
  *
  * Returns %TRUE on success, %FALSE on failure.
  **/
-static gboolean
+static void
 create_plugin_control (XfcePanelPlugin *plugin)
 {
 
@@ -1461,7 +1461,6 @@ create_plugin_control (XfcePanelPlugin *plugin)
   xfce_panel_plugin_menu_show_configure (plugin);
   g_signal_connect (plugin, "configure-plugin",
                       G_CALLBACK (plugin_create_options), pd);
-  return(TRUE);
 }
 
 
--
cgit v0.8.3.4
