diff --git a/phonon/pulsesupport.cpp b/phonon/pulsesupport.cpp
index b7009b11..d3240840 100644
--- a/phonon/pulsesupport.cpp
+++ b/phonon/pulsesupport.cpp
@@ -840,7 +840,8 @@ PulseSupport::PulseSupport()
     }
 
     // We require a glib event loop
-    if (!QByteArray(QAbstractEventDispatcher::instance()->metaObject()->className()).contains("EventDispatcherGlib")) {
+    if (!QByteArray(QAbstractEventDispatcher::instance()->metaObject()->className()).contains("EventDispatcherGlib") &&
+        !QByteArray(QAbstractEventDispatcher::instance()->metaObject()->className()).contains("GlibEventDispatcher")) {
         qWarning("WARNING: Disabling PulseAudio integration for lack of GLib event loop.");
         return;
     }
