diff -urNad imlib2-1.2.0.orig/configure.in imlib2-1.2.0/configure.in
--- imlib2-1.2.0.orig/configure.in	2005-01-05 10:39:08.000000000 +0100
+++ imlib2-1.2.0/configure.in	2005-03-29 20:03:07.000000000 +0200
@@ -86,6 +86,9 @@
 packagesrcdir=`cd $srcdir && pwd`
 AC_DEFINE_UNQUOTED(PACKAGE_SOURCE_DIR, "${packagesrcdir}", [Source code directory])
 
+# FIXME: fill in requirements for pc.in - but only ft2 has a pc file!
+requirements="freetype2"
+
 mmx=no
 AC_ARG_ENABLE(mmx,[  --disable-mmx           attempt compiling using mmx assembly [default=yes]],
 [
@@ -354,7 +357,9 @@
 fi
 AM_CONDITIONAL(BUILD_BZ2_LOADER, test "$bz2_ok" = yes)
 AC_SUBST(BZ2LIBS)
-  
+
+AC_SUBST(requirements)
+
 AC_OUTPUT([
 Makefile
 src/Makefile
