diff -p -up gtk+-2.12.9/modules/printbackends/cups/gtkprintbackendcups.c~ gtk+-2.12.9/modules/printbackends/cups/gtkprintbackendcups.c
--- gtk+-2.12.9/modules/printbackends/cups/gtkprintbackendcups.c~	2008-03-12 05:17:03.000000000 +0100
+++ gtk+-2.12.9/modules/printbackends/cups/gtkprintbackendcups.c	2008-04-02 11:30:12.000000000 +0200
@@ -36,6 +36,7 @@
 #include <cairo-ps.h>
 
 #include <glib/gi18n-lib.h>
+#include <glib/gstdio.h>
 #include <gmodule.h>
 
 #include <gtk/gtkprintoperation.h>
