From 59e326f5893c865dcfa03322e2c894a07243a9ef Mon Sep 17 00:00:00 2001
From: sfan5 <sfan5@live.de>
Date: Thu, 24 Dec 2020 18:29:34 +0100
Subject: [PATCH] events: Protect against error from XGetWMHints()

---
 src/events.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/events.c b/src/events.c
index 97879169a..9494696b0 100644
--- a/src/events.c
+++ b/src/events.c
@@ -1709,7 +1709,10 @@ handlePropertyNotify (DisplayInfo *display_info, XPropertyEvent * ev)
                 XFree (c->wmhints);
             }
 
+            myDisplayErrorTrapPush (display_info);
             c->wmhints = XGetWMHints (display_info->dpy, c->window);
+            myDisplayErrorTrapPopIgnored (display_info);
+
             if (c->wmhints)
             {
                 if (c->wmhints->flags & WindowGroupHint)
-- 
GitLab

