diff --git a/src/assetimport/qssgassetimportmanager.cpp b/src/assetimport/qssgassetimportmanager.cpp
index aec55978..15529f28 100644
--- a/src/assetimport/qssgassetimportmanager.cpp
+++ b/src/assetimport/qssgassetimportmanager.cpp
@@ -43,10 +43,12 @@ QSSGAssetImportManager::QSSGAssetImportManager(QObject *parent) : QObject(parent
     const QStringList keys = QSSGAssetImporterFactory::keys();
     for (const auto &key : keys) {
         auto importer = QSSGAssetImporterFactory::create(key, QStringList());
-        m_assetImporters.append(importer);
-        // Add to extension map
-        for (const auto &extension : importer->inputExtensions()) {
-            m_extensionsMap.insert(extension, importer);
+        if (importer) {
+            m_assetImporters.append(importer);
+            // Add to extension map
+            for (const auto &extension : importer->inputExtensions()) {
+                m_extensionsMap.insert(extension, importer);
+            }
         }
     }
 }
