From f243fe19623b69ae628ae55cd2eabb8182fcfcef Mon Sep 17 00:00:00 2001
From: Paolo Bacchilega <paobac@src.gnome.org>
Date: Sun, 17 May 2015 07:35:44 +0200
Subject: restore the original condition


diff --git a/src/fr-command-7z.c b/src/fr-command-7z.c
index 62a8fbe..fc33d4d 100644
--- a/src/fr-command-7z.c
+++ b/src/fr-command-7z.c
@@ -619,7 +619,7 @@ fr_command_7z_get_capabilities (FrArchive  *archive,
 			/* give priority to rar and unrar that supports RAR files better. */
 			if (!_g_program_is_available ("rar", check_command)
 			    && !_g_program_is_available ("unrar", check_command)
-			    && g_file_test ("/usr/lib/p7zip/Codecs/Rar29.so", G_FILE_TEST_EXISTS))
+			    && (! check_command || g_file_test ("/usr/lib/p7zip/Codecs/Rar29.so", G_FILE_TEST_EXISTS)))
 				capabilities |= FR_ARCHIVE_CAN_READ;
 		}
 		else
-- 
cgit v0.10.2

