From dd20c1f4d6419238bbb37fb93a48986f83c6e6c2 Mon Sep 17 00:00:00 2001
From: Chris Wilson <chris@chris-wilson.co.uk>
Date: Wed, 09 Jan 2013 12:38:09 +0000
Subject: xlib: map-to-image requires an extents

Signed-off-by: Chris Wilson <chris@chris-wilson.co.uk>
---
diff --git a/src/cairo-xlib-source.c b/src/cairo-xlib-source.c
index d08052a..0689d82 100644
--- a/src/cairo-xlib-source.c
+++ b/src/cairo-xlib-source.c
@@ -968,8 +968,9 @@ surface_source (cairo_xlib_surface_t *dst,
 						 0, 0);
     } else {
 	cairo_image_surface_t *image;
+	cairo_rectangle_int_t map_extents = { 0,0, upload.width,upload.height };
 
-	image = _cairo_surface_map_to_image (&xsrc->base, NULL);
+	image = _cairo_surface_map_to_image (&xsrc->base, &map_extents);
 
 	_cairo_pattern_init_for_surface (&local_pattern, pattern->surface);
 	cairo_matrix_init_translate (&local_pattern.base.matrix,
--
cgit v0.9.0.2-2-gbebe
