diff -u -r seed-3.8.1/modules/readline/seed-readline.c seed-3.8.1-rl63/modules/readline/seed-readline.c
--- seed-3.8.1/modules/readline/seed-readline.c	2013-04-16 09:58:20.000000000 +0200
+++ seed-3.8.1-rl63/modules/readline/seed-readline.c	2014-03-05 10:29:53.393385674 +0100
@@ -77,7 +77,7 @@
   key = seed_value_to_string(ctx, arguments[0], exception);
   c = seed_make_rl_closure((SeedObject) arguments[1]);
 
-  rl_bind_key(*key, (Function *) c);
+  rl_bind_key(*key, (rl_command_func_t *) c);
 
   g_free(key);
 
