From a88795c64123e4084044451d8861838ea904abd9 Mon Sep 17 00:00:00 2001
From: Chris Wilson <chris@chris-wilson.co.uk>
Date: Tue, 29 Jul 2014 21:11:09 +0100
Subject: sna: Clear cursor image after rotation

Bugzilla: https://bugs.freedesktop.org/show_bug.cgi?id=81886
Signed-off-by: Chris Wilson <chris@chris-wilson.co.uk>

diff --git a/src/sna/sna_display.c b/src/sna/sna_display.c
index 8964e77..eed68d3 100644
--- a/src/sna/sna_display.c
+++ b/src/sna/sna_display.c
@@ -4294,7 +4294,7 @@ static struct sna_cursor *__sna_get_cursor(struct sna *sna, xf86CrtcPtr crtc)
 		image = sna->cursor.scratch;
 		cursor->last_width = cursor->last_height = size;
 	}
-	if (width < cursor->last_width || height < cursor->last_height)
+	if (width < cursor->last_width || height < cursor->last_height || rotation != cursor->rotation)
 		memset(image, 0, 4*size*size);
 	if (rotation == RR_Rotate_0) {
 		if (argb == NULL) {
-- 
cgit v0.10.2

