From: Nick Gasson <nick@nickg.me.uk>
Date: Sat, 27 Feb 2021 14:27:54 +0800
Subject: Switch default server to gnudb.org

Forwarded: no
---
 include/cddb/cddb_ni.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/cddb/cddb_ni.h b/include/cddb/cddb_ni.h
index d32fdc1..4156bdb 100644
--- a/include/cddb/cddb_ni.h
+++ b/include/cddb/cddb_ni.h
@@ -78,8 +78,8 @@
 
 #define DEFAULT_USER        "anonymous"
 #define DEFAULT_HOST        "localhost"
-#define DEFAULT_SERVER      "freedb.org"
-#define DEFAULT_PORT        888
+#define DEFAULT_SERVER      "gnudb.gnudb.org"
+#define DEFAULT_PORT        8880
 #define DEFAULT_TIMEOUT     10
 #define DEFAULT_PATH_QUERY  "/~cddb/cddb.cgi"
 #define DEFAULT_PATH_SUBMIT "/~cddb/submit.cgi"
