diff -ruN totem-2.20.1/Makefile.am totem-2.20.1.plparse/Makefile.am
--- totem-2.20.1/Makefile.am	2007-10-05 14:17:27.000000000 +0000
+++ totem-2.20.1.plparse/Makefile.am	2007-12-27 14:42:06.000000000 +0000
@@ -14,16 +14,12 @@
 	totem.spec.in		\
 	totem.spec		\
 	omf.make		\
-	totem-plparser.pc.in	\
 	xmldocs.make		\
 	gnome-doc-utils.make
 
-CLEANFILES = totem.spec totem-plparser.pc
+CLEANFILES = totem.spec
 
 DISTCLEANFILES = intltool-extract intltool-merge intltool-update \
 		 gnome-doc-utils.make
 
-pkgconfigdir = $(libdir)/pkgconfig
-pkgconfig_DATA = totem-plparser.pc
-
 DISTCHECK_CONFIGURE_FLAGS = --disable-scrollkeeper
diff -ruN totem-2.20.1/bindings/python/totem.override totem-2.20.1.plparse/bindings/python/totem.override
--- totem-2.20.1/bindings/python/totem.override	2007-10-05 14:17:24.000000000 +0000
+++ totem-2.20.1.plparse/bindings/python/totem.override	2007-12-27 14:41:05.000000000 +0000
@@ -9,7 +9,7 @@
 
 #include "override_common.h"
 #include "totem.h"
-#include "plparse/totem-disc.h"
+#include "totem-disc.h"
 #include "plugins/totem-plugin.h"
 
 void pytotem_register_classes (PyObject *d);
diff -ruN totem-2.20.1/browser-plugin/Makefile.am totem-2.20.1.plparse/browser-plugin/Makefile.am
--- totem-2.20.1/browser-plugin/Makefile.am	2007-10-05 14:17:20.000000000 +0000
+++ totem-2.20.1.plparse/browser-plugin/Makefile.am	2007-12-27 14:33:50.000000000 +0000
@@ -89,10 +89,8 @@
 	-I$(top_srcdir)					\
 	-I$(top_srcdir)/lib				\
 	-I$(top_srcdir)/src				\
-	-I$(top_srcdir)/src/plparse			\
 	-I$(top_srcdir)/src/backend			\
 	-I$(top_builddir)/src				\
-	-I$(top_builddir)/src/plparse			\
 	-I$(top_builddir)/src/backend			\
 	-DGNOMELOCALEDIR="\"$(datadir)/locale\""	\
 	-DGCONF_PREFIX="\"/apps/totem\""		\
@@ -114,7 +112,6 @@
 totem_plugin_viewer_LDADD =						\
 	$(top_builddir)/src/backend/libbaconvideowidget.la		\
 	$(top_builddir)/src/libtotem_player.la				\
-	$(top_builddir)/src/plparse/libtotem-plparser.la		\
 	$(top_builddir)/lib/libtotemscrsaver.la				\
 	libtotempluginviewer_helper.la					\
 	$(EXTRA_GNOME_LIBS)	\
@@ -154,10 +151,8 @@
 libtotem_basic_plugin_la_CPPFLAGS = \
 	-I$(top_srcdir)/src				\
 	-I$(top_srcdir)/src/backend			\
-	-I$(top_srcdir)/src/plparse			\
 	-I$(top_builddir)/src				\
 	-I$(top_builddir)/src/backend			\
-	-I$(top_builddir)/src/plparse			\
 	-Iidl						\
 	$(addprefix -I$(MOZILLA_INCLUDE_ROOT)/,$(gecko_include_subdirs))	\
 	-DTOTEM_BASIC_PLUGIN				\
@@ -182,7 +177,6 @@
 	$(AM_CXXFLAGS)
 
 libtotem_basic_plugin_la_LIBADD =					\
-	$(top_builddir)/src/plparse/libtotem-plparser-mini.la		\
 	libtotempluginviewer_helper.la					\
 	$(DBUS_LIBS)	\
 	$(MOZILLA_LIBS)	\
@@ -223,10 +217,8 @@
 libtotem_gmp_plugin_la_CPPFLAGS = \
 	-I$(top_srcdir)/src				\
 	-I$(top_srcdir)/src/backend			\
-	-I$(top_srcdir)/src/plparse			\
 	-I$(top_builddir)/src				\
 	-I$(top_builddir)/src/backend			\
-	-I$(top_builddir)/src/plparse			\
 	-Iidl						\
 	$(addprefix -I$(MOZILLA_INCLUDE_ROOT)/,$(gecko_include_subdirs))	\
 	-DTOTEM_GMP_PLUGIN				\
@@ -251,7 +243,6 @@
 	$(AM_CXXFLAGS)
 
 libtotem_gmp_plugin_la_LIBADD =						\
-	$(top_builddir)/src/plparse/libtotem-plparser-mini.la		\
 	libtotempluginviewer_helper.la	\
 	$(DBUS_LIBS)	\
 	$(MOZILLA_LIBS)	\
@@ -286,10 +277,8 @@
 libtotem_complex_plugin_la_CPPFLAGS = \
 	-I$(top_srcdir)/src				\
 	-I$(top_srcdir)/src/backend			\
-	-I$(top_srcdir)/src/plparse			\
 	-I$(top_builddir)/src				\
 	-I$(top_builddir)/src/backend			\
-	-I$(top_builddir)/src/plparse			\
 	-Iidl						\
 	$(addprefix -I$(MOZILLA_INCLUDE_ROOT)/,$(gecko_include_subdirs))	\
 	-DTOTEM_COMPLEX_PLUGIN				\
@@ -314,7 +303,6 @@
 	$(AM_CXXFLAGS)
 
 libtotem_complex_plugin_la_LIBADD =					\
-	$(top_builddir)/src/plparse/libtotem-plparser-mini.la		\
 	libtotempluginviewer_helper.la					\
 	$(DBUS_LIBS)	\
 	$(MOZILLA_LIBS)	\
@@ -349,10 +337,8 @@
 libtotem_narrowspace_plugin_la_CPPFLAGS = \
 	-I$(top_srcdir)/src				\
 	-I$(top_srcdir)/src/backend			\
-	-I$(top_srcdir)/src/plparse			\
 	-I$(top_builddir)/src				\
 	-I$(top_builddir)/src/backend			\
-	-I$(top_builddir)/src/plparse			\
 	-Iidl						\
 	$(addprefix -I$(MOZILLA_INCLUDE_ROOT)/,$(gecko_include_subdirs))	\
 	-DTOTEM_NARROWSPACE_PLUGIN			\
@@ -377,7 +363,6 @@
 	$(AM_CXXFLAGS)
 
 libtotem_narrowspace_plugin_la_LIBADD =					\
-	$(top_builddir)/src/plparse/libtotem-plparser-mini.la		\
 	libtotempluginviewer_helper.la					\
 	$(DBUS_LIBS)	\
 	$(MOZILLA_LIBS)	\
@@ -412,10 +397,8 @@
 libtotem_mully_plugin_la_CPPFLAGS = \
 	-I$(top_srcdir)/src				\
 	-I$(top_srcdir)/src/backend			\
-	-I$(top_srcdir)/src/plparse			\
 	-I$(top_builddir)/src				\
 	-I$(top_builddir)/src/backend			\
-	-I$(top_builddir)/src/plparse			\
 	-Iidl						\
 	$(addprefix -I$(MOZILLA_INCLUDE_ROOT)/,$(gecko_include_subdirs))	\
 	-DTOTEM_MULLY_PLUGIN				\
@@ -440,7 +423,6 @@
 	$(AM_CXXFLAGS)
 
 libtotem_mully_plugin_la_LIBADD =					\
-	$(top_builddir)/src/plparse/libtotem-plparser-mini.la		\
 	libtotempluginviewer_helper.la					\
 	$(DBUS_LIBS)	\
 	$(MOZILLA_LIBS)	\
diff -ruN totem-2.20.1/configure.in totem-2.20.1.plparse/configure.in
--- totem-2.20.1/configure.in	2007-10-17 13:09:22.000000000 +0000
+++ totem-2.20.1.plparse/configure.in	2007-12-27 14:38:07.000000000 +0000
@@ -34,34 +34,19 @@
 GLIB_REQS=2.13.4
 GTK_REQS=2.12.1
 GNOMEVFS_REQS=2.16.0
+TOTEM_PLPARSER_REQS=2.21.6
 GNOMEUI_REQS=2.3.3
 LIBGNOME_REQS=2.14.0
 GNOMEICON_REQS=2.15.90
 DBUS_REQS=0.61
 VALA_REQS=0.0.8
 
-# Before making a release, the PLPARSER_LT_VERSION string should be modified.
-# The string is of the form C:R:A.
-# - If interfaces have been changed or added, but binary compatibility has
-#   been preserved, change to C+1:0:A+1
-# - If binary compatibility has been broken (eg removed or changed interfaces)
-#   change to C+1:0:0
-# - If the interface is the same as the previous version, change to C:R+1:A
-PLPARSER_LT_VERSION=8:1:1
-AC_SUBST(PLPARSER_LT_VERSION)
-
 TOTEM_VERSION_MAJOR=totem_version_major
 TOTEM_VERSION_MINOR=totem_version_minor
 TOTEM_VERSION_MICRO=totem_version_micro
 AC_SUBST(TOTEM_VERSION_MAJOR)
 AC_SUBST(TOTEM_VERSION_MINOR)
 AC_SUBST(TOTEM_VERSION_MICRO)
-TOTEM_PL_PARSER_VERSION_MAJOR=totem_version_major
-TOTEM_PL_PARSER_VERSION_MINOR=totem_version_minor
-TOTEM_PL_PARSER_VERSION_MICRO=totem_version_micro
-AC_SUBST(TOTEM_PL_PARSER_VERSION_MAJOR)
-AC_SUBST(TOTEM_PL_PARSER_VERSION_MINOR)
-AC_SUBST(TOTEM_PL_PARSER_VERSION_MICRO)
 
 # The full list of plugins
 allowed_plugins="screensaver ontop galago gromit lirc media-player-keys properties sidebar-test skipto sample-python sample-vala bemused"
@@ -249,11 +234,11 @@
 dnl Also check if we're going to be using GTK+ only
 if test x$ENABLE_GTK = "xno" ; then
 # FIXME: use gmodule-no-export instead!
-	PKG_CHECK_MODULES(EXTRA_GNOME, glib-2.0 >= $GLIB_REQS gtk+-2.0 >= $GTK_REQS libgnomeui-2.0 >= $GNOMEUI_REQS gnome-vfs-2.0 >= $GNOMEVFS_REQS gnome-vfs-module-2.0 >= $GNOMEVFS_REQS libgnome-2.0 >= $LIBGNOME_REQS  gnome-desktop-2.0 >= 2.1.5 gnome-icon-theme >= $GNOMEICON_REQS gmodule-2.0 $ISO_CODES $MM)
+	PKG_CHECK_MODULES(EXTRA_GNOME, glib-2.0 >= $GLIB_REQS gtk+-2.0 >= $GTK_REQS libgnomeui-2.0 >= $GNOMEUI_REQS gnome-vfs-2.0 >= $GNOMEVFS_REQS gnome-vfs-module-2.0 >= $GNOMEVFS_REQS libgnome-2.0 >= $LIBGNOME_REQS  gnome-desktop-2.0 >= 2.1.5 gnome-icon-theme >= $GNOMEICON_REQS gmodule-2.0 totem-plparser >= $TOTEM_PLPARSER_REQS $ISO_CODES $MM)
 	HAVE_GNOME=yes
 else
 # FIXME: use gmodule-no-export instead!
-	PKG_CHECK_MODULES(EXTRA_GNOME, glib-2.0 >= $GLIB_REQS gtk+-2.0 >= $GTK_REQS gnome-vfs-2.0 >= $GNOMEVFS_REQS gnome-vfs-module-2.0 >= $GNOMEVFS_REQS gnome-icon-theme >= $GNOMEICON_REQS gmodule-2.0 $ISO_CODES $MM)
+	PKG_CHECK_MODULES(EXTRA_GNOME, glib-2.0 >= $GLIB_REQS gtk+-2.0 >= $GTK_REQS gnome-vfs-2.0 >= $GNOMEVFS_REQS gnome-vfs-module-2.0 >= $GNOMEVFS_REQS gnome-icon-theme >= $GNOMEICON_REQS gmodule-2.0 totem-plparser >= $TOTEM_PLPARSER_REQS $ISO_CODES $MM)
 	AC_DEFINE(USE_STABLE_LIBGNOMEUI, 1, [defined if we don't have GNOME])
 	AC_DEFINE(HAVE_GTK_ONLY, 1, [defined if GNOME isn't used])
 	HAVE_GNOME=no
@@ -466,7 +451,8 @@
 		[glib-2.0
 		 gnome-vfs-2.0 >= $GNOMEVFS_REQS
 		 gnome-vfs-module-2.0 >= $GNOMEVFS_REQS
-		 gthread-2.0],
+		 gthread-2.0
+		 totem-plparser-mini >= $TOTEM_PLPARSER_REQS],
 		[],[enable_browser_plugins=no])
 
 	BROWSER_PLUGIN_CFLAGS="$MOZILLA_NOT_LINKED_CFLAGS $BROWSER_PLUGIN_CFLAGS"
@@ -792,13 +778,6 @@
 AC_PATH_PROG(GCONFTOOL, gconftool-2)
 AM_GCONF_SOURCE_2
 
-dnl Check for packages for building libtotem-plparser.la
-PKG_CHECK_MODULES(TOTEM_PLPARSER, [gtk+-2.0 libxml-2.0 gnome-vfs-2.0
-                                   gnome-vfs-module-2.0])
-
-PKG_CHECK_MODULES([TOTEM_PLPARSER_MINI],
-	[gnome-vfs-2.0 gnome-vfs-module-2.0])
-
 GNOME_COMMON_INIT
 GNOME_DEBUG_CHECK
 GNOME_COMPILE_WARNINGS([maximum])
@@ -910,7 +889,8 @@
 	PYGTK_REQUIRED=2.8.0
 
 	PKG_CHECK_MODULES([PYGTK], [
-		pygtk-2.0 >= $PYGTK_REQUIRED],
+		pygtk-2.0 >= $PYGTK_REQUIRED
+		totem-plparser >= $TOTEM_PLPARSER_REQS],
 		[],[have_python=no])
 
 	AC_SUBST([PYGTK_CFLAGS])
@@ -990,8 +970,6 @@
 AC_OUTPUT([
 Makefile
 totem.spec
-totem-plparser.pc
-totem-plparser-uninstalled.pc
 lib/Makefile
 src/Makefile
 src/plugins/Makefile
@@ -1008,8 +986,6 @@
 src/plugins/sample-python/Makefile
 src/plugins/sample-vala/Makefile
 src/backend/Makefile
-src/plparse/Makefile
-src/plparse/totem-pl-parser-features.h
 browser-plugin/Makefile
 browser-plugin/idl/Makefile
 data/Makefile
diff -ruN totem-2.20.1/po/POTFILES.in totem-2.20.1.plparse/po/POTFILES.in
--- totem-2.20.1/po/POTFILES.in	2007-10-05 14:17:27.000000000 +0000
+++ totem-2.20.1.plparse/po/POTFILES.in	2007-12-27 14:42:32.000000000 +0000
@@ -29,12 +29,6 @@
 src/backend/bacon-video-widget-gst-0.10.c
 src/backend/bacon-video-widget-xine.c
 src/backend/video-utils.c
-src/plparse/totem-disc.c
-src/plparse/totem-pl-parser.c
-src/plparse/totem-pl-parser-lines.c
-src/plparse/totem-pl-parser-pla.c
-src/plparse/totem-pl-parser-pls.c
-src/plparse/totem-pl-parser-xspf.c
 src/plugins/totem-plugin-manager.c
 src/plugins/totem-plugins-engine.c
 [type: gettext/ini]src/plugins/bemused/bemused.totem-plugin.in
diff -ruN totem-2.20.1/src/Makefile.am totem-2.20.1.plparse/src/Makefile.am
--- totem-2.20.1/src/Makefile.am	2007-10-05 14:17:22.000000000 +0000
+++ totem-2.20.1.plparse/src/Makefile.am	2007-12-27 14:47:25.000000000 +0000
@@ -1,4 +1,4 @@
-SUBDIRS = plugins backend plparse
+SUBDIRS = plugins backend
 
 bin_PROGRAMS = totem totem-video-thumbnailer totem-video-indexer
 libexec_PROGRAMS =
@@ -62,8 +62,6 @@
 
 libtotem_player_la_CPPFLAGS = \
 	-I$(srcdir)/backend	\
-	-I$(srcdir)/plparse	\
-	-I$(top_builddir)/src/plparse	\
 	$(common_defines)	\
 	$(AM_CPPFLAGS)
 
@@ -107,11 +105,9 @@
 	totem-subtitle-encoding.h
 
 totem_CPPFLAGS = \
-	-I$(srcdir)/plparse		\
 	-I$(srcdir)/backend		\
 	-I$(srcdir)/plugins		\
 	-I$(top_builddir)/data		\
-	-I$(top_builddir)/src/plparse	\
 	$(common_defines)		\
 	$(AM_CPPFLAGS)
 
@@ -127,7 +123,6 @@
 	$(AM_LDFLAGS)
 
 totem_LDADD = \
-	plparse/libtotem-plparser.la	\
 	backend/libbaconvideowidget.la	\
 	plugins/libtotemmodule.la	\
 	libbaconmessageconnection.la	\
@@ -154,7 +149,6 @@
 	totem-resources.h
 
 totem_video_thumbnailer_CPPFLAGS = \
-	-I$(srcdir)/plparse	\
 	-I$(srcdir)/backend	\
 	$(common_defines)	\
 	$(AM_CPPFLAGS)
@@ -191,7 +185,6 @@
 
 libtotem_properties_page_la_CPPFLAGS =		\
 	-I$(srcdir)/backend			\
-	-I$(srcdir)/plparse			\
 	-I$(srcdir)/plugins/properties		\
 	-I$(top_builddir)/data			\
 	$(common_defines)			\
@@ -229,7 +222,6 @@
 
 test_properties_page_CPPFLAGS = 		\
 	-I$(srcdir)/backend			\
-	-I$(srcdir)/plparse			\
 	-I$(top_builddir)/data			\
 	-I$(srcdir)/plugins/properties		\
 	$(common_defines)			\
@@ -266,7 +258,6 @@
 	totem-resources.h
 
 totem_video_indexer_CPPFLAGS = \
-	-I$(srcdir)/plparse	\
 	-I$(srcdir)/backend	\
 	-I$(top_builddir)/data	\
 	$(common_defines)	\
@@ -296,7 +287,6 @@
 	disc-test.c
 
 disc_test_CPPFLAGS = \
-	-I$(srcdir)/plparse	\
 	$(common_defines)	\
 	$(AM_CPPFLAGS)
 
@@ -310,7 +300,6 @@
 	$(AM_LDFLAGS)
 
 disc_test_LDADD = \
-	plparse/libtotem-plparser.la	\
 	$(GTK_LIBS)			\
 	$(EXTRA_GNOME_LIBS)
 
diff -ruN totem-2.20.1/src/backend/bacon-video-widget.h totem-2.20.1.plparse/src/backend/bacon-video-widget.h
--- totem-2.20.1/src/backend/bacon-video-widget.h	2007-10-05 14:17:21.000000000 +0000
+++ totem-2.20.1.plparse/src/backend/bacon-video-widget.h	2007-12-27 14:38:25.000000000 +0000
@@ -31,7 +31,7 @@
 #include <gtk/gtkeventbox.h>
 
 /* for optical disc enumeration type */
-#include "totem-disc.h"
+#include <totem-disc.h>
 
 G_BEGIN_DECLS
 
diff -ruN totem-2.20.1/src/totem-playlist.h totem-2.20.1.plparse/src/totem-playlist.h
--- totem-2.20.1/src/totem-playlist.h	2007-10-05 14:17:22.000000000 +0000
+++ totem-2.20.1.plparse/src/totem-playlist.h	2007-12-27 14:40:17.000000000 +0000
@@ -26,7 +26,7 @@
 #include <gtk/gtkvbox.h>
 #include <libgnomevfs/gnome-vfs-volume.h>
 
-#include "totem-pl-parser.h"
+#include <totem-pl-parser.h>
 
 G_BEGIN_DECLS
 
diff -ruN totem-2.20.1/src/totem.c totem-2.20.1.plparse/src/totem.c
--- totem-2.20.1/src/totem.c	2007-10-05 14:17:22.000000000 +0000
+++ totem-2.20.1.plparse/src/totem.c	2007-12-27 14:39:59.000000000 +0000
@@ -31,6 +31,7 @@
 #include <glib/gi18n.h>
 #include <gtk/gtk.h>
 #include <gdk/gdkkeysyms.h>
+#include <totem-disc.h>
 #include <stdlib.h>
 #include <math.h>
 
@@ -65,7 +66,6 @@
 #include "totem.h"
 #include "totem-private.h"
 #include "totem-preferences.h"
-#include "totem-disc.h"
 
 #include "debug.h"
 
diff -ruN totem-2.20.1/src/totem.h totem-2.20.1.plparse/src/totem.h
--- totem-2.20.1/src/totem.h	2007-10-05 14:17:22.000000000 +0000
+++ totem-2.20.1.plparse/src/totem.h	2007-12-27 14:40:33.000000000 +0000
@@ -30,7 +30,7 @@
 
 #include <glib-object.h>
 #include <gtk/gtk.h>
-#include "plparse/totem-disc.h"
+#include <totem-disc.h>
 
 #define TOTEM_GCONF_PREFIX "/apps/totem"
 
diff -ruN totem-2.20.1/totem-plparser-uninstalled.pc.in totem-2.20.1.plparse/totem-plparser-uninstalled.pc.in
--- totem-2.20.1/totem-plparser-uninstalled.pc.in	2007-10-05 14:17:27.000000000 +0000
+++ totem-2.20.1.plparse/totem-plparser-uninstalled.pc.in	1970-01-01 00:00:00.000000000 +0000
@@ -1,14 +0,0 @@
-prefix=@prefix@
-exec_prefix=@exec_prefix@
-libdir=@libdir@
-includedir=@includedir@
-gconf_serverdir=@libexecdir@
-
-
-Name: totem-plparser
-Description: Totem Playlist Parser library
-Version: @VERSION@
-Requires: gtk+-2.0
-Requires.private: libxml-2.0 gnome-vfs-2.0 gnome-vfs-module-2.0
-Libs: ${pc_top_builddir}/${pcfiledir}/src/plparse/libtotem-plparser.la
-Cflags: -I${pc_top_builddir}/${pcfiledir}/src/plparse
diff -ruN totem-2.20.1/totem-plparser.pc.in totem-2.20.1.plparse/totem-plparser.pc.in
--- totem-2.20.1/totem-plparser.pc.in	2007-10-05 14:17:27.000000000 +0000
+++ totem-2.20.1.plparse/totem-plparser.pc.in	1970-01-01 00:00:00.000000000 +0000
@@ -1,14 +0,0 @@
-prefix=@prefix@
-exec_prefix=@exec_prefix@
-libdir=@libdir@
-includedir=@includedir@
-gconf_serverdir=@libexecdir@
-
-
-Name: totem-plparser
-Description: Totem Playlist Parser library
-Version: @VERSION@
-Requires: gtk+-2.0
-Requires.private: libxml-2.0 gnome-vfs-2.0 gnome-vfs-module-2.0
-Libs: -L${libdir} -ltotem-plparser
-Cflags: -I${includedir}/totem/1/plparser
--- totem-2.20.1/browser-plugin/Makefile.am.orig	2007-12-27 16:01:50.000000000 +0100
+++ totem-2.20.1/browser-plugin/Makefile.am	2007-12-27 16:04:01.000000000 +0100
@@ -179,7 +179,7 @@
 libtotem_basic_plugin_la_LIBADD =					\
 	libtotempluginviewer_helper.la					\
 	$(DBUS_LIBS)	\
-	$(MOZILLA_LIBS)	\
+	$(BROWSER_PLUGIN_LIBS)	\
 	$(LIBXPCOMGLUE_S)
 
 libtotem_basic_plugin_la_LDFLAGS = \
@@ -245,7 +245,7 @@
 libtotem_gmp_plugin_la_LIBADD =						\
 	libtotempluginviewer_helper.la	\
 	$(DBUS_LIBS)	\
-	$(MOZILLA_LIBS)	\
+	$(BROWSER_PLUGIN_LIBS)	\
 	$(LIBXPCOMGLUE_S)
 
 libtotem_gmp_plugin_la_LDFLAGS = \
@@ -305,7 +305,7 @@
 libtotem_complex_plugin_la_LIBADD =					\
 	libtotempluginviewer_helper.la					\
 	$(DBUS_LIBS)	\
-	$(MOZILLA_LIBS)	\
+	$(BROWSER_PLUGIN_LIBS)	\
 	$(LIBXPCOMGLUE_S)
 
 libtotem_complex_plugin_la_LDFLAGS = \
@@ -365,7 +365,7 @@
 libtotem_narrowspace_plugin_la_LIBADD =					\
 	libtotempluginviewer_helper.la					\
 	$(DBUS_LIBS)	\
-	$(MOZILLA_LIBS)	\
+	$(BROWSER_PLUGIN_LIBS)	\
 	$(LIBXPCOMGLUE_S)
 
 libtotem_narrowspace_plugin_la_LDFLAGS = \
@@ -425,7 +425,7 @@
 libtotem_mully_plugin_la_LIBADD =					\
 	libtotempluginviewer_helper.la					\
 	$(DBUS_LIBS)	\
-	$(MOZILLA_LIBS)	\
+	$(BROWSER_PLUGIN_LIBS)	\
 	$(LIBXPCOMGLUE_S)
 
 libtotem_mully_plugin_la_LDFLAGS = \
