From 9cfd52b8b5bd1321e2b6f497a41914c5bd06ded4 Mon Sep 17 00:00:00 2001
From: Igor <f2404@yandex.ru>
Date: Thu, 15 Mar 2018 10:23:26 -0400
Subject: Update window actions and active tab indication on reordering tabs

See bug #14278
---
 terminal/terminal-window.c | 8 +++-----
 1 file changed, 3 insertions(+), 5 deletions(-)

diff --git a/terminal/terminal-window.c b/terminal/terminal-window.c
index 68daa03..9f30716 100644
--- a/terminal/terminal-window.c
+++ b/terminal/terminal-window.c
@@ -1062,12 +1062,9 @@ terminal_window_notebook_page_switched (GtkNotebook     *notebook,
                                         guint            page_num,
                                         TerminalWindow  *window)
 {
-  TerminalScreen *active;
+  TerminalScreen *active = TERMINAL_SCREEN (page);
   const gchar    *encoding;
 
-  /* get the new active page */
-  active = TERMINAL_SCREEN (page);
-
   terminal_return_if_fail (TERMINAL_IS_WINDOW (window));
   terminal_return_if_fail (active == NULL || TERMINAL_IS_SCREEN (active));
 
@@ -1105,8 +1102,9 @@ terminal_window_notebook_page_reordered (GtkNotebook     *notebook,
                                          guint            page_num,
                                          TerminalWindow  *window)
 {
-  /* Regenerate the "Go" menu */
+  /* Regenerate the "Go" menu and update actions */
   terminal_window_rebuild_tabs_menu (window);
+  terminal_window_update_actions (window);
 }
 
 
-- 
cgit v1.1

