Index: src/where.c
==================================================================
--- src/sqlite3.c
+++ src/sqlite3.c
@@ -876,11 +876,14 @@
   while( pScan->iEquiv<=pScan->nEquiv ){
     iCur = pScan->aEquiv[pScan->iEquiv-2];
     iColumn = pScan->aEquiv[pScan->iEquiv-1];
     while( (pWC = pScan->pWC)!=0 ){
       for(pTerm=pWC->a+k; k<pWC->nTerm; k++, pTerm++){
-        if( pTerm->leftCursor==iCur && pTerm->u.leftColumn==iColumn ){
+        if( pTerm->leftCursor==iCur
+         && pTerm->u.leftColumn==iColumn
+         && (pScan->iEquiv<=2 || !ExprHasProperty(pTerm->pExpr, EP_FromJoin))
+        ){
           if( (pTerm->eOperator & WO_EQUIV)!=0
            && pScan->nEquiv<ArraySize(pScan->aEquiv)
           ){
             int j;
             pX = sqlite3ExprSkipCollate(pTerm->pExpr->pRight);


