diff --git a/dm.c b/dm.c
index abe4453..8186424 100644
--- a/dm.c
+++ b/dm.c
@@ -75,6 +75,16 @@ from The Open Group.
 # endif
 #endif
 
+#if defined(HAVE_OPENLOG) && defined(HAVE_SYSLOG_H)
+# define USE_SYSLOG
+# include <syslog.h>
+# ifndef LOG_AUTHPRIV
+#  define LOG_AUTHPRIV LOG_AUTH
+# endif
+# ifndef LOG_PID
+#  define LOG_PID 0
+# endif
+#endif
 
 #if defined(SVR4) && !defined(sun)
 extern FILE    *fdopen();
@@ -749,6 +759,9 @@ StartDisplay (struct display *d)
 	    CleanUpChild ();
 	    (void) Signal (SIGPIPE, SIG_IGN);
 	}
+#ifdef USE_SYSLOG
+	openlog("xdm", LOG_PID, LOG_AUTHPRIV);
+#endif
 	LoadSessionResources (d);
 	SetAuthorization (d);
 	if (!WaitForServer (d))
diff --git a/greeter/greet.c b/greeter/greet.c
index 17d21c8..61c7a52 100644
--- a/greeter/greet.c
+++ b/greeter/greet.c
@@ -489,9 +489,6 @@ greet_user_rtn GreetUser(
 	LogError ("Cannot reopen display %s for greet window\n", d->name);
 	exit (RESERVER_DISPLAY);
     }
-#ifdef USE_SYSLOG
-    openlog("xdm", LOG_ODELAY|LOG_PID, LOG_AUTHPRIV);
-#endif
 
     for (;;) {
 #ifdef USE_PAM

