diff -dur xfce4-sensors-plugin-1.0.0.orig/lib/hddtemp.c xfce4-sensors-plugin-1.0.0/lib/hddtemp.c
--- xfce4-sensors-plugin-1.0.0.orig/lib/hddtemp.c	2010-03-27 16:11:48.000000000 +0100
+++ xfce4-sensors-plugin-1.0.0/lib/hddtemp.c	2010-11-06 20:50:11.717022222 +0100
@@ -102,7 +102,7 @@
     if (!notify_is_initted())
         notify_init(PACKAGE); /* NOTIFY_APPNAME */
 
-    nn = notify_notification_new (summary, body, icon, NULL);
+    nn = notify_notification_new (summary, body, icon);
     /* FIXME: Use channels or propagate private object or use static global variable */
     //notify_notification_add_action (nn,
                             //"confirmed",
diff -dur xfce4-sensors-plugin-1.0.0.orig/lib/sensors-interface.c xfce4-sensors-plugin-1.0.0/lib/sensors-interface.c
--- xfce4-sensors-plugin-1.0.0.orig/lib/sensors-interface.c	2010-03-27 16:11:48.000000000 +0100
+++ xfce4-sensors-plugin-1.0.0/lib/sensors-interface.c	2010-11-06 20:50:24.360007993 +0100
@@ -105,7 +105,7 @@
                 if (!notify_is_initted())
                     notify_init(PACKAGE); /* NOTIFY_APPNAME */
 
-                nn = notify_notification_new(summary, body, icon, NULL);
+                nn = notify_notification_new(summary, body, icon);
                 notify_notification_show(nn, &error);
                 #else
                 DBG("%s\n%s", summary, body);

