--- xorg-server-1.2.0/hw/xfree86/os-support/xf86_libc.h.jx	2007-01-22 22:13:15.000000000 -0500
+++ xorg-server-1.2.0/hw/xfree86/os-support/xf86_libc.h	2007-01-29 18:13:11.000000000 -0500
@@ -428,7 +428,7 @@
 #define toupper(i)		xf86toupper(i)
 #undef ungetc
 #define ungetc(i,FP)		xf86ungetc(i,FP)
-#undef vfprinf
+#undef vfprintf
 #define vfprintf(p,f,a)		xf86vfprintf(p,f,a)
 #undef vsnprintf
 #define vsnprintf(s,n,f,a)	xf86vsnprintf(s,n,f,a)
