diff -aur nmap-4.53.old/zenmap/setup.py nmap-4.53/zenmap/setup.py
--- nmap-4.53.old/zenmap/setup.py	2008-02-26 22:53:56.000000000 -0600
+++ nmap-4.53/zenmap/setup.py	2008-02-26 22:56:43.000000000 -0600
@@ -37,7 +37,7 @@
 # Directories for POSIX operating systems
 # These are created after a "install" or "py2exe" command
 # These directories are relative to the installation or dist directory
-pixmaps_dir = os.path.join('share', 'pixmaps')
+pixmaps_dir = os.path.join('share', 'pixmaps', APP_NAME)
 icons_dir = os.path.join('share', 'icons')
 locale_dir = os.path.join('share', APP_NAME, 'locale')
 config_dir = os.path.join('share', APP_NAME, 'config')
diff -aur nmap-4.53.old/zenmap/zenmapCore/BasePaths.py nmap-4.53/zenmap/zenmapCore/BasePaths.py
--- nmap-4.53.old/zenmap/zenmapCore/BasePaths.py	2008-02-26 22:53:56.000000000 -0600
+++ nmap-4.53/zenmap/zenmapCore/BasePaths.py	2008-02-26 22:55:36.000000000 -0600
@@ -42,7 +42,7 @@
                   target_list = 'target_list.txt',
                   wizard = 'wizard.xml',
                   options = 'options.xml',
-                  pixmaps_dir = os.path.join('share', 'pixmaps'),
+                  pixmaps_dir = os.path.join('share', 'pixmaps', APP_NAME),
                   i18n_dir = os.path.join('share','locale'),
                   i18n_message_file = APP_NAME + '.mo',
                   scan_results_extension = 'usr',  # comes from umit scan result
@@ -50,7 +50,7 @@
                   user_home = HOME,
                   basic_search_sequence = [HOME, CURRENT_DIR],
                   config_search_sequence = [HOME, CURRENT_DIR],
-                  pixmaps_search_sequence = [os.path.join(CURRENT_DIR, 'share', 'pixmaps'),
+                  pixmaps_search_sequence = [os.path.join(CURRENT_DIR, 'share', 'pixmaps', APP_NAME),
                                              HOME],
                   i18n_search_sequence = [os.path.join(CURRENT_DIR, 'share', 'locale'), HOME],
                   db = APP_NAME + ".db",
@@ -69,8 +69,9 @@
                                                      CURRENT_DIR, '/etc'],
                            pixmaps_search_sequence = [os.path.join(CURRENT_DIR,
                                                                    'share',
-                                                                   'pixmaps'),
-                                                      '/usr/share/pixmaps',
+                                                                   'pixmaps',
+																   APP_NAME),
+                                                      '/usr/share/pixmaps/zenmap',
                                                       '/opt/' + APP_NAME, HOME],
                            i18n_search_sequence = [os.path.join(CURRENT_DIR, 'share', 'locale'),
                                                    '/usr/share/locale',
diff -aur nmap-4.53.old/zenmap/zenmapCore/Paths.py nmap-4.53/zenmap/zenmapCore/Paths.py
--- nmap-4.53.old/zenmap/zenmapCore/Paths.py	2008-02-26 22:53:56.000000000 -0600
+++ nmap-4.53/zenmap/zenmapCore/Paths.py	2008-02-26 22:56:11.000000000 -0600
@@ -45,7 +45,7 @@
 LOCALE_DIR = join(main_dir, "share", APP_NAME, "locale")
 MISC_DIR = join(main_dir, "share", APP_NAME, "misc")
 ICONS_DIR = join(main_dir, "share", "icons")
-PIXMAPS_DIR = join(main_dir, "share", "pixmaps")
+PIXMAPS_DIR = join(main_dir, "share", "pixmaps", APP_NAME)
 DOCS_DIR = join(main_dir, "share", APP_NAME, "docs")
 
 
