--- 54-gphoto.rules~	2007-11-09 08:06:38.000000000 +0100
+++ 54-gphoto.rules	2007-11-19 07:42:53.000000000 +0100
@@ -3,8 +3,11 @@
 #   libgphoto2      2.4.1          gcc, ltdl, EXIF
 #   libgphoto2_port 0.8.0          gcc, ltdl, USB, serial without locking
 #
-SUBSYSTEM!="usb_device", GOTO="libgphoto2_rules_end"
 ACTION!="add", GOTO="libgphoto2_rules_end"
+SUBSYSTEM=="usb", GOTO="libgphoto2_rules_begin"
+SUBSYSTEM=="usb_device", GOTO="libgphoto2_rules_begin"
+SUBSYSTEM!="usb_device", GOTO="libgphoto2_rules_end"
+LABEL="libgphoto2_rules_begin"
 
 ATTRS{idVendor}=="0553", ATTRS{idProduct}=="0202", MODE="0660", GROUP="camera"
 ATTRS{idVendor}=="06bd", ATTRS{idProduct}=="0403", MODE="0660", GROUP="camera"
