From cd5a1e5cf95a165c66e7a097530058864dc3a38f Mon Sep 17 00:00:00 2001
From: Cosimo Cecchi <cosimoc@gnome.org>
Date: Thu, 28 Apr 2011 18:05:07 +0000
Subject: places-sidebar: make sure to return if we click inside empty space

https://bugzilla.gnome.org/show_bug.cgi?id=648872
---
diff --git a/src/nautilus-places-sidebar.c b/src/nautilus-places-sidebar.c
index 6e49a4c..56216ab 100644
--- a/src/nautilus-places-sidebar.c
+++ b/src/nautilus-places-sidebar.c
@@ -2685,6 +2685,7 @@ bookmarks_button_release_event_cb (GtkWidget *widget,
 	GtkTreeIter iter;
 	GtkTreeModel *model;
 	GtkTreeView *tree_view;
+	gboolean res;
 
 	path = NULL;
 
@@ -2708,8 +2709,12 @@ bookmarks_button_release_event_cb (GtkWidget *widget,
 			return FALSE;
 		}
 
-		gtk_tree_view_get_path_at_pos (tree_view, (int) event->x, (int) event->y,
-					       &path, NULL, NULL, NULL);
+		res = gtk_tree_view_get_path_at_pos (tree_view, (int) event->x, (int) event->y,
+						     &path, NULL, NULL, NULL);
+
+		if (!res) {
+			return FALSE;
+		}
 
 		gtk_tree_model_get_iter (model, &iter, path);
 
--
cgit v0.9
