diff -Naur showimg-0.9.5-orig/showimg/kexi/kexidb/drivers/pqxx/pqxxconnection.cpp showimg-0.9.5/showimg/kexi/kexidb/drivers/pqxx/pqxxconnection.cpp
--- showimg-0.9.5-orig/showimg/kexi/kexidb/drivers/pqxx/pqxxconnection.cpp	2006-01-07 03:47:46.000000000 -0500
+++ showimg-0.9.5/showimg/kexi/kexidb/drivers/pqxx/pqxxconnection.cpp	2008-07-20 14:30:44.000000000 -0400
@@ -248,7 +248,7 @@
 		//        m_trans = new pqxx::nontransaction(*m_pqxxsql);
 
         //Create a result object through the transaction
-        m_res = new pqxx::result(m_trans->data->exec(statement.utf8()));
+        m_res = new pqxx::result(m_trans->data->exec(statement.utf8().data()));
 
 		//Commit the transaction
 		if (implicityStarted) {
diff -Naur showimg-0.9.5-orig/showimg/kexi/kexidb/drivers/pqxx/pqxxconnection.h showimg-0.9.5/showimg/kexi/kexidb/drivers/pqxx/pqxxconnection.h
--- showimg-0.9.5-orig/showimg/kexi/kexidb/drivers/pqxx/pqxxconnection.h	2006-01-07 03:47:46.000000000 -0500
+++ showimg-0.9.5/showimg/kexi/kexidb/drivers/pqxx/pqxxconnection.h	2008-07-20 14:29:59.000000000 -0400
@@ -16,7 +16,7 @@
 
 #include <kexidb/connection.h>
 #include "pqxxcursor.h"
-#include <pqxx/all.h>
+#include <pqxx/pqxx>
 
 
 namespace KexiDB
diff -Naur showimg-0.9.5-orig/showimg/kexi/kexidb/drivers/pqxx/pqxxcursor.cpp showimg-0.9.5/showimg/kexi/kexidb/drivers/pqxx/pqxxcursor.cpp
--- showimg-0.9.5-orig/showimg/kexi/kexidb/drivers/pqxx/pqxxcursor.cpp	2006-01-07 03:47:46.000000000 -0500
+++ showimg-0.9.5/showimg/kexi/kexidb/drivers/pqxx/pqxxcursor.cpp	2008-07-20 14:30:52.000000000 -0400
@@ -84,7 +84,7 @@
 			m_implicityStarted = true;
 		}
 
-		m_res = new pqxx::result(((pqxxSqlConnection*)connection())->m_trans->data->exec(statement.utf8()));
+		m_res = new pqxx::result(((pqxxSqlConnection*)connection())->m_trans->data->exec(statement.utf8().data()));
 		((pqxxSqlConnection*)connection())
 			->drv_commitTransaction(((pqxxSqlConnection*)connection())->m_trans);
 //		my_conn->m_trans->commit();
diff -Naur showimg-0.9.5-orig/showimg/kexi/kexidb/drivers/pqxx/pqxxcursor.h showimg-0.9.5/showimg/kexi/kexidb/drivers/pqxx/pqxxcursor.h
--- showimg-0.9.5-orig/showimg/kexi/kexidb/drivers/pqxx/pqxxcursor.h	2006-01-07 03:47:46.000000000 -0500
+++ showimg-0.9.5/showimg/kexi/kexidb/drivers/pqxx/pqxxcursor.h	2008-07-20 14:29:51.000000000 -0400
@@ -15,7 +15,7 @@
 #include <kexidb/cursor.h>
 #include <kexidb/connection.h>
 
-#include <pqxx/all.h>
+#include <pqxx/pqxx>
 
 namespace KexiDB {
 
diff -Naur showimg-0.9.5-orig/showimg/kexi/kexidb/drivers/pqxx/pqxxdriver.cpp showimg-0.9.5/showimg/kexi/kexidb/drivers/pqxx/pqxxdriver.cpp
--- showimg-0.9.5-orig/showimg/kexi/kexidb/drivers/pqxx/pqxxdriver.cpp	2006-01-07 03:47:46.000000000 -0500
+++ showimg-0.9.5/showimg/kexi/kexidb/drivers/pqxx/pqxxdriver.cpp	2008-07-20 14:30:24.000000000 -0400
@@ -132,14 +132,14 @@
 //
 QString pqxxSqlDriver::escapeString( const QString& str) const
 {
-    return QString(pqxx::Quote(str.ascii()).c_str());
+     return "'"+QString(pqxx::sqlesc(str.ascii()).c_str())+"'";
 }
 
 //==================================================================================
 //
 QCString pqxxSqlDriver::escapeString( const QCString& str) const
 {
-    return QCString(pqxx::Quote(QString(str).ascii()).c_str());
+    return "'"+QCString(pqxx::sqlesc(QString(str).ascii()).c_str())+"'";
 }
 
 //==================================================================================
