diff -upr kitinerary-21.04.3.orig/src/lib/pdf/pdfdocument.cpp kitinerary-21.04.3/src/lib/pdf/pdfdocument.cpp
--- kitinerary-21.04.3.orig/src/lib/pdf/pdfdocument.cpp	2021-07-05 22:24:08.000000000 +0300
+++ kitinerary-21.04.3/src/lib/pdf/pdfdocument.cpp	2021-08-01 22:44:05.114172443 +0300
@@ -201,7 +201,14 @@ int PdfDocument::fileSize() const
 }
 
 #ifdef HAVE_POPPLER
-static QDateTime parsePdfDateTime(const char *str)
+
+#if KPOPPLER_VERSION >= QT_VERSION_CHECK(21, 8, 0)
+#define KPOPPLER_DATE_STRING_TYPE GooString
+#else
+#define KPOPPLER_DATE_STRING_TYPE char
+#endif
+
+static QDateTime parsePdfDateTime(const KPOPPLER_DATE_STRING_TYPE *str)
 {
     int year, month, day, hour, min, sec, tzHours, tzMins;
     char tz;
@@ -233,7 +240,9 @@ QDateTime PdfDocument::creationTime() co
     if (!dt) {
         return {};
     }
-#if KPOPPLER_VERSION >= QT_VERSION_CHECK(0, 72, 0)
+#if KPOPPLER_VERSION >= QT_VERSION_CHECK(21, 8, 0)
+    return parsePdfDateTime(dt.get());
+#elif KPOPPLER_VERSION >= QT_VERSION_CHECK(0, 72, 0)
     return parsePdfDateTime(dt->c_str());
 #else
     return parsePdfDateTime(dt->getCString());
@@ -250,7 +259,9 @@ QDateTime PdfDocument::modificationTime(
     if (!dt) {
         return {};
     }
-#if KPOPPLER_VERSION >= QT_VERSION_CHECK(0, 72, 0)
+#if KPOPPLER_VERSION >= QT_VERSION_CHECK(21, 8, 0)
+    return parsePdfDateTime(dt.get());
+#elif KPOPPLER_VERSION >= QT_VERSION_CHECK(0, 72, 0)
     return parsePdfDateTime(dt->c_str());
 #else
     return parsePdfDateTime(dt->getCString());
