diff -Naur cinepaint-orig/plug-ins/icc_examin/icc_examin/icc_modell_beobachter.cpp cinepaint/plug-ins/icc_examin/icc_examin/icc_modell_beobachter.cpp
--- cinepaint-orig/plug-ins/icc_examin/icc_examin/icc_modell_beobachter.cpp	2012-04-20 18:51:10.625503826 -0400
+++ cinepaint/plug-ins/icc_examin/icc_examin/icc_modell_beobachter.cpp	2012-04-20 18:52:04.478835055 -0400
@@ -32,6 +32,7 @@
 
 #include <list>
 #include <string>
+#include <unistd.h>
 
 #include "icc_modell_beobachter.h"
 
