--- a/configure.in	2014-04-12 17:46:28.459612102 +0200
+++ b/configure.in	2014-04-12 17:46:33.926347877 +0200
@@ -17,11 +17,6 @@
 dnl incase it is broken for example.
 AC_ARG_ENABLE(shm, [  --enable-shm            support shared memory if available [default=yes]], echo $enable_shm, enable_shm="yes")
 
-AM_PATH_GTK(1.2.1,[
-  GDK_IMLIB="gdk_imlib utils"],[
-  GDK_IMLIB="" 
-  AC_MSG_WARN([*** gdk_imlib will not be built ***])])
-
 AC_MSG_CHECKING(whether to build gmodulized imlib)
 
 AC_ARG_ENABLE(modules, [  --disable-modules       Disables dynamic module loading],[
