commit 328cfa5e2f9d39906096142af9c7468c066ea802
Author: Cedric Bellegarde <gnumdk@gmail.com>
Date:   Fri Nov 23 14:51:51 2012 +0100

    BUG:310553 Fix building akonadi support in plasma data engines

diff --git a/plasma/generic/dataengines/CMakeLists.txt b/plasma/generic/dataengines/CMakeLists.txt
index 0280482..44bc2d0 100644
--- a/plasma/generic/dataengines/CMakeLists.txt
+++ b/plasma/generic/dataengines/CMakeLists.txt
@@ -26,13 +26,13 @@ if (NepomukCore_FOUND AND Soprano_FOUND)
     add_subdirectory(metadata)
 endif ()
 
-if(Boost_FOUND AND KDEPIMLIBS_FOUND)
-   if(AKONADI_FOUND)
+if(Boost_FOUND AND KdepimLibs_FOUND)
+   if(Akonadi_FOUND)
       add_subdirectory(akonadi)
-   endif(AKONADI_FOUND)
+   endif(Akonadi_FOUND)
    add_subdirectory(calendar)
    add_subdirectory(rss)
-endif(Boost_FOUND AND KDEPIMLIBS_FOUND)
+endif(Boost_FOUND AND KdepimLibs_FOUND)
 
 if(NOT WIN32)
    add_subdirectory(mouse)
diff --git a/plasma/generic/dataengines/calendar/CMakeLists.txt b/plasma/generic/dataengines/calendar/CMakeLists.txt
index e05cbe7..5539f99 100644
--- a/plasma/generic/dataengines/calendar/CMakeLists.txt
+++ b/plasma/generic/dataengines/calendar/CMakeLists.txt
@@ -11,7 +11,7 @@ set(calendar_engine_srcs
     calendarengine.cpp
 )
 
-if(AKONADI_FOUND)
+if(Akonadi_FOUND)
     ADD_DEFINITIONS(-DAKONADI_FOUND)
     set(calendar_engine_srcs ${calendar_engine_srcs}
         eventdatacontainer.cpp
@@ -23,7 +23,7 @@ if(AKONADI_FOUND)
         akonadi/blockalarmsattribute.cpp
         akonadi/collectionselection.cpp
     )
-endif(AKONADI_FOUND)
+endif(Akonadi_FOUND)
 
 kde4_add_plugin(plasma_engine_calendar ${calendar_engine_srcs})
 
@@ -37,13 +37,13 @@ target_link_libraries(
     ${KDE4_KCALUTILS_LIBS}
 )
 
-if(AKONADI_FOUND)
+if(Akonadi_FOUND)
     target_link_libraries(
         plasma_engine_calendar
         ${KDE4_AKONADI_LIBS}
         ${KDEPIMLIBS_AKONADI_KCAL_LIBS}
     )
-endif(AKONADI_FOUND)
+endif(Akonadi_FOUND)
 
 install(TARGETS plasma_engine_calendar DESTINATION ${PLUGIN_INSTALL_DIR})
 install(FILES plasma-dataengine-calendar.desktop DESTINATION ${SERVICES_INSTALL_DIR})
