diff -up Python-2.7.2/Lib/whichdb.py.gdbm-1.9-magic Python-2.7.2/Lib/whichdb.py
--- Python-2.7.2/Lib/whichdb.py.gdbm-1.9-magic	2011-06-11 11:46:26.000000000 -0400
+++ Python-2.7.2/Lib/whichdb.py	2011-09-30 15:45:21.778872290 -0400
@@ -91,7 +91,7 @@ def whichdb(filename):
         return ""
 
     # Check for GNU dbm
-    if magic == 0x13579ace:
+    if magic in (0x13579ace, 0x13579acd, 0x13579acf):
         return "gdbm"
 
     # Check for old Berkeley db hash file format v2
diff -up Python-2.7.2/Misc/NEWS.gdbm-1.9-magic Python-2.7.2/Misc/NEWS
