diff -rupN a/src/gfxoutputdrv/gifdrv.c b/src/gfxoutputdrv/gifdrv.c
--- a/src/gfxoutputdrv/gifdrv.c	2012-07-25 23:46:05.000000000 +0000
+++ b/src/gfxoutputdrv/gifdrv.c	2014-06-01 11:48:40.493722851 +0000
@@ -114,7 +114,7 @@ static int gifdrv_open(screenshot_t *scr
   if (EGifPutScreenDesc(sdata->fd, screenshot->width, screenshot->height, 8, 0, gif_colors) == GIF_ERROR ||
       EGifPutImageDesc(sdata->fd, 0, 0, screenshot->width, screenshot->height, 0, NULL) == GIF_ERROR)
   {
-    EGifCloseFile(sdata->fd);
+    EGifCloseFile(sdata->fd, NULL);
     VICE_FreeMapObject(gif_colors);
     lib_free(sdata->data);
     lib_free(sdata->ext_filename);
@@ -145,7 +145,7 @@ static int gifdrv_close(screenshot_t *sc
 
     sdata = screenshot->gfxoutputdrv_data;
 
-    EGifCloseFile(sdata->fd);
+    EGifCloseFile(sdata->fd, NULL);
     VICE_FreeMapObject(gif_colors);
 
     /* for some reason giflib will create a file with unexpected
@@ -184,7 +184,7 @@ static char *gifdrv_memmap_ext_filename;
 
 static int gifdrv_close_memmap(void)
 {
-  EGifCloseFile(gifdrv_memmap_fd);
+  EGifCloseFile(gifdrv_memmap_fd, NULL);
   VICE_FreeMapObject(gif_colors);
   lib_free(gifdrv_memmap_ext_filename);
 
@@ -231,7 +231,7 @@ static int gifdrv_open_memmap(const char
   if (EGifPutScreenDesc(gifdrv_memmap_fd, x_size, y_size, 8, 0, gif_colors) == GIF_ERROR ||
       EGifPutImageDesc(gifdrv_memmap_fd, 0, 0, x_size, y_size, 0, NULL) == GIF_ERROR)
   {
-    EGifCloseFile(gifdrv_memmap_fd);
+    EGifCloseFile(gifdrv_memmap_fd, NULL);
     VICE_FreeMapObject(gif_colors);
     lib_free(gifdrv_memmap_ext_filename);
     return -1;
