diff --git i/data/gnome-session-manager@.service.in w/data/gnome-session-manager@.service.in
index dd10cc9b..b06dbeb3 100644
--- i/data/gnome-session-manager@.service.in
+++ w/data/gnome-session-manager@.service.in
@@ -15,5 +15,5 @@ Before=gnome-session-manager.target
 
 [Service]
 Type=notify
-ExecStart=@libexecdir@/gnome-session-binary --debug --systemd-service --session=%i
+ExecStart=@libexecdir@/gnome-session-binary --systemd-service --session=%i
 ExecStopPost=-@libexecdir@/gnome-session-ctl --shutdown
diff --git i/gnome-session/gdm-log.c w/gnome-session/gdm-log.c
index c0dca411..687f7e8e 100644
--- i/gnome-session/gdm-log.c
+++ w/gnome-session/gdm-log.c
@@ -187,9 +187,6 @@ gdm_log_init (void)
         prg_name = g_get_prgname ();
 
         options = LOG_PID;
-#ifdef LOG_PERROR
-        options |= LOG_PERROR;
-#endif
 
         openlog (prg_name, options, LOG_DAEMON);
 
