diff --git i/gdk/wayland/gdkwindow-wayland.c w/gdk/wayland/gdkwindow-wayland.c
index 77a6e978f8463ccd..c87d6d9ce7e79bef 100644
--- i/gdk/wayland/gdkwindow-wayland.c
+++ w/gdk/wayland/gdkwindow-wayland.c
@@ -3004,7 +3004,7 @@ gtk_surface_configure_edges (void                *data,
           new_state |= GDK_WINDOW_STATE_TOP_RESIZABLE;
           break;
         case GTK_SURFACE1_EDGE_CONSTRAINT_RESIZABLE_RIGHT:
-          new_state |= GDK_WINDOW_STATE_TOP_TILED;
+          new_state |= GDK_WINDOW_STATE_RIGHT_RESIZABLE;
           break;
         case GTK_SURFACE1_EDGE_CONSTRAINT_RESIZABLE_BOTTOM:
           new_state |= GDK_WINDOW_STATE_BOTTOM_RESIZABLE;
diff --git i/gtk/gtkwindow.c w/gtk/gtkwindow.c
index ffc841871fde0d61..a4ba0982240d8b6d 100644
--- i/gtk/gtkwindow.c
+++ w/gtk/gtkwindow.c
@@ -6855,7 +6855,7 @@ update_corner_windows (GtkWindow *window,
                                        region, 0, 0);
       cairo_region_destroy (region);
 
-      gdk_window_hide (priv->border_window[GDK_WINDOW_EDGE_NORTH_EAST]);
+      gdk_window_show_unraised (priv->border_window[GDK_WINDOW_EDGE_NORTH_EAST]);
     }
   else
     {
@@ -7065,57 +7065,50 @@ update_border_windows (GtkWindow *window)
 
   if (resize_n)
     {
-      y += window_border.top;
+      y += window_border.top + handle_v;
       h -= window_border.top + handle_v;
     }
 
   if (resize_w)
     {
-      x += window_border.left;
+      x += window_border.left + handle_h;
       w -= window_border.left + handle_h;
     }
 
   if (resize_s)
     h -= window_border.bottom + handle_v;
 
   if (resize_e)
     w -= window_border.right + handle_h;
 
   /* North */
   if (resize_n)
     {
-      gdk_window_move_resize (priv->border_window[GDK_WINDOW_EDGE_SOUTH],
-                              x, window_border.top + height,
-                              w, border.bottom);
+      gdk_window_move_resize (priv->border_window[GDK_WINDOW_EDGE_NORTH],
+                              x, window_border.top - border.top,
+                              w, border.top);
 
       gdk_window_show_unraised (priv->border_window[GDK_WINDOW_EDGE_NORTH]);
     }
   else
     {
       gdk_window_hide (priv->border_window[GDK_WINDOW_EDGE_NORTH]);
     }
 
   /* South */
-  if (resize_n)
+  if (resize_s)
     {
       gdk_window_move_resize (priv->border_window[GDK_WINDOW_EDGE_SOUTH],
                               x, window_border.top + height,
                               w, border.bottom);
 
       gdk_window_show_unraised (priv->border_window[GDK_WINDOW_EDGE_SOUTH]);
     }
   else
     {
       gdk_window_hide (priv->border_window[GDK_WINDOW_EDGE_SOUTH]);
     }
 
-  /*
-   * Horizontal edges
-   */
-
-  y = (resize_n || resize_s) ? window_border.top + handle_v : 0;
-  h = (resize_n || resize_s) ? height - 2 * handle_v : height + window_border.top + window_border.bottom;
-
   /* East */
   if (resize_e)
     {
