diff -up ConsoleKit-0.3.1/src/ck-manager.c.small-fixes ConsoleKit-0.3.1/src/ck-manager.c
--- ConsoleKit-0.3.1/src/ck-manager.c.small-fixes	2009-07-31 22:32:32.382848777 -0400
+++ ConsoleKit-0.3.1/src/ck-manager.c	2009-07-31 22:34:55.159597733 -0400
@@ -810,12 +810,9 @@ check_polkit_permissions (CkManager     
                           AuthorizedCallback     callback)
 {
         const char    *sender;
-        GError        *error;
         PolkitSubject *subject;
         AuthorizedCallbackData *data;
 
-        error = NULL;
-
         g_debug ("constructing polkit data");
 
         /* Check that caller is privileged */
@@ -837,6 +834,7 @@ check_polkit_permissions (CkManager     
                                               NULL,
                                               (GAsyncReadyCallback)auth_ready_callback,
                                               data);
+        g_object_unref (subject);
 }
 
 static void
@@ -873,14 +871,12 @@ get_polkit_permissions (CkManager   *man
 {
         const char    *sender;
         PolkitSubject *subject;
-        GError *error;
 
         g_debug ("get permissions for action %s", action);
 
         sender = dbus_g_method_get_sender (context);
         subject = polkit_system_bus_name_new (sender);
 
-        error = NULL;
         polkit_authority_check_authorization (manager->priv->pol_ctx,
                                               subject,
                                               action,
@@ -890,7 +886,6 @@ get_polkit_permissions (CkManager   *man
                                               (GAsyncReadyCallback) ready_cb,
                                               context);
         g_object_unref (subject);
-
 }
 #endif
 
