From a8f5d8cec87ef1586897e94e49bdf54db64bf97b Mon Sep 17 00:00:00 2001
From: Kalev Lember <kalevlember@gmail.com>
Date: Thu, 11 Oct 2012 10:59:16 +0200
Subject: [PATCH] Disable the context pane plugin when webkit isn't available

---
 plugins/Makefile.am | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/plugins/Makefile.am b/plugins/Makefile.am
index e4f6e69..3a9ea03 100644
--- a/plugins/Makefile.am
+++ b/plugins/Makefile.am
@@ -14,7 +14,6 @@ SUBDIRS =						\
 if ENABLE_PYTHON
 SUBDIRS +=						\
 	artsearch					\
-	context						\
 	im-status					\
 	lyrics						\
 	magnatune					\
@@ -24,8 +23,12 @@ SUBDIRS +=						\
 	sendto						\
 	rb
 
+if WITH_WEBKIT
+SUBDIRS += context
 endif
 
+endif # ENABLE_PYTHON
+
 if WITH_LIRC
 SUBDIRS += lirc
 endif
-- 
1.7.12.1

