--- src/tools.c.orig	2009-11-02 10:54:49.000000000 -0500
+++ src/tools.c	2009-11-02 10:55:05.000000000 -0500
@@ -44,9 +44,9 @@
 #if HAVE_SYS_CAPABILITY_H
 #undef _POSIX_SOURCE
 /* Ugly glue to get around linux header madness colliding with glibc */
-#define _LINUX_TYPES_H
+/*#define _LINUX_TYPES_H*/
 #define _LINUX_FS_H
-typedef uint32_t __u32;
+/*typedef uint32_t __u32;*/
 #include <sys/capability.h>
 #endif
 #endif
