diff -up squid-3.1.11/helpers/external_acl/ldap_group/squid_ldap_group.c.unused squid-3.1.11/helpers/external_acl/ldap_group/squid_ldap_group.c
--- squid-3.1.11/helpers/external_acl/ldap_group/squid_ldap_group.c.unused	2011-02-10 10:04:50.422205003 +0100
+++ squid-3.1.11/helpers/external_acl/ldap_group/squid_ldap_group.c	2011-02-10 10:05:48.020204424 +0100
@@ -219,7 +219,6 @@ main(int argc, char **argv)
     int use_extension_dn = 0;
     int strip_nt_domain = 0;
     int strip_kerberos_realm = 0;
-    int err = 0;
 
     setbuf(stdout, NULL);
 
@@ -602,7 +601,6 @@ error:
                 tryagain = 1;
             }
         }
-        err = 0;
     }
     if (ld)
         ldap_unbind(ld);
diff -up squid-3.1.11/helpers/ntlm_auth/fakeauth/fakeauth_auth.c.unused squid-3.1.11/helpers/ntlm_auth/fakeauth/fakeauth_auth.c
--- squid-3.1.11/helpers/ntlm_auth/fakeauth/fakeauth_auth.c.unused	2011-02-10 09:57:56.170205002 +0100
+++ squid-3.1.11/helpers/ntlm_auth/fakeauth/fakeauth_auth.c	2011-02-10 09:58:39.039205002 +0100
@@ -141,7 +141,6 @@ ntlmMakeChallenge(struct ntlm_challenge 
 {
     static unsigned hash;
     int r;
-    char *d;
     int i;
 
     debug("ntlmMakeChallenge: flg %08x\n", flags);
@@ -158,7 +157,6 @@ ntlmMakeChallenge(struct ntlm_challenge 
     chal->flags = flags;
     chal->unknown[6] = htole16(0x003a);
 
-    d = (char *) chal + 48;
     i = 0;
 
     if (authenticate_ntlm_domain != NULL)
diff -up squid-3.1.11/src/auth/negotiate/auth_negotiate.cc.unused squid-3.1.11/src/auth/negotiate/auth_negotiate.cc
--- squid-3.1.11/src/auth/negotiate/auth_negotiate.cc.unused	2011-02-10 07:57:04.293205769 +0100
+++ squid-3.1.11/src/auth/negotiate/auth_negotiate.cc	2011-02-10 07:58:22.643205463 +0100
@@ -544,7 +544,6 @@ AuthNegotiateUserRequest::module_start(R
 {
     authenticateStateData *r = NULL;
     static char buf[MAX_AUTHTOKEN_LEN];
-    negotiate_user_t *negotiate_user;
     AuthUser *auth_user = user();
 
     assert(data);
@@ -552,8 +551,6 @@ AuthNegotiateUserRequest::module_start(R
     assert(auth_user);
     assert(auth_user->auth_type == AUTH_NEGOTIATE);
 
-    negotiate_user = dynamic_cast<negotiate_user_t *>(user());
-
     debugs(29, 8, "AuthNegotiateUserRequest::module_start: auth state is '" << auth_state << "'");
 
     if (negotiateConfig.authenticate == NULL) {
@@ -655,12 +652,10 @@ AuthNegotiateUserRequest::authenticate(H
 
     /** \todo rename this!! */
     AuthUser *local_auth_user;
-    negotiate_user_t *negotiate_user;
 
     local_auth_user = user();
     assert(local_auth_user);
     assert(local_auth_user->auth_type == AUTH_NEGOTIATE);
-    negotiate_user = dynamic_cast<negotiate_user_t *>(local_auth_user);
     assert (this);
 
     /** Check that we are in the client side, where we can generate
diff -up squid-3.1.11/src/auth/ntlm/auth_ntlm.cc.unused squid-3.1.11/src/auth/ntlm/auth_ntlm.cc
--- squid-3.1.11/src/auth/ntlm/auth_ntlm.cc.unused	2011-02-10 08:07:21.181205066 +0100
+++ squid-3.1.11/src/auth/ntlm/auth_ntlm.cc	2011-02-10 08:08:52.188204812 +0100
@@ -460,7 +460,6 @@ AuthNTLMUserRequest::module_start(RH * h
 {
     authenticateStateData *r = NULL;
     static char buf[8192];
-    ntlm_user_t *ntlm_user;
     AuthUser *auth_user = user();
 
     assert(data);
@@ -468,8 +467,6 @@ AuthNTLMUserRequest::module_start(RH * h
     assert(auth_user);
     assert(auth_user->auth_type == AUTH_NTLM);
 
-    ntlm_user = dynamic_cast<ntlm_user_t *>(user());
-
     debugs(29, 8, "AuthNTLMUserRequest::module_start: auth state is '" << auth_state << "'");
 
     if (ntlmConfig.authenticate == NULL) {
@@ -572,12 +569,10 @@ AuthNTLMUserRequest::authenticate(HttpRe
 
     /* TODO: rename this!! */
     AuthUser *local_auth_user;
-    ntlm_user_t *ntlm_user;
 
     local_auth_user = user();
     assert(local_auth_user);
     assert(local_auth_user->auth_type == AUTH_NTLM);
-    ntlm_user = dynamic_cast<ntlm_user_t *>(local_auth_user);
     assert (this);
 
     /* Check that we are in the client side, where we can generate
diff -up squid-3.1.11/src/auth/User.cc.unused squid-3.1.11/src/auth/User.cc
--- squid-3.1.11/src/auth/User.cc.unused	2011-02-10 08:14:08.623207125 +0100
+++ squid-3.1.11/src/auth/User.cc	2011-02-10 08:14:52.305205340 +0100
@@ -158,13 +158,11 @@ AuthUser::CachedACLsReset()
      */
     AuthUserHashPointer *usernamehash;
     AuthUser *auth_user;
-    char const *username = NULL;
     debugs(29, 3, "AuthUser::CachedACLsReset: Flushing the ACL caches for all users.");
     hash_first(proxy_auth_username_cache);
 
     while ((usernamehash = ((AuthUserHashPointer *) hash_next(proxy_auth_username_cache)))) {
         auth_user = usernamehash->user();
-        username = auth_user->username();
         /* free cached acl results */
         aclCacheMatchFlush(&auth_user->proxy_match_cache);
 
diff -up squid-3.1.11/src/client_side_reply.cc.unused squid-3.1.11/src/client_side_reply.cc
--- squid-3.1.11/src/client_side_reply.cc.unused	2011-02-10 08:28:52.861205280 +0100
+++ squid-3.1.11/src/client_side_reply.cc	2011-02-10 08:37:22.530205334 +0100
@@ -1056,14 +1056,6 @@ clientReplyContext::storeNotOKTransferDo
         /* haven't found end of headers yet */
         return 0;
 
-    int sending = SENDING_BODY;
-
-    if (curReply->sline.status == HTTP_NO_CONTENT ||
-            curReply->sline.status == HTTP_NOT_MODIFIED ||
-            curReply->sline.status < HTTP_OK ||
-            http->request->method == METHOD_HEAD)
-        sending = SENDING_HDRSONLY;
-
     /*
      * Figure out how much data we are supposed to send.
      * If we are sending a body and we don't have a content-length,
@@ -2028,13 +2020,10 @@ clientReplyContext::sendMoreData (StoreI
 
     char *buf = next()->readBuffer.data;
 
-    char *body_buf = buf;
-
     if (buf != result.data) {
         /* we've got to copy some data */
         assert(result.length <= next()->readBuffer.length);
         xmemcpy(buf, result.data, result.length);
-        body_buf = buf;
     }
 
 #if USE_ZPH_QOS
diff -up squid-3.1.11/src/comm.cc.unused squid-3.1.11/src/comm.cc
--- squid-3.1.11/src/comm.cc.unused	2011-02-10 08:20:37.942204904 +0100
+++ squid-3.1.11/src/comm.cc	2011-02-10 08:21:43.940205003 +0100
@@ -706,7 +706,6 @@ comm_openex(int sock_type,
 {
     int new_socket;
     fde *F = NULL;
-    int tos = 0;
     struct addrinfo *AI = NULL;
 
     PROF_start(comm_open);
@@ -756,11 +755,6 @@ comm_openex(int sock_type,
 
     debugs(50, 3, "comm_openex: Opened socket FD " << new_socket << " : family=" << AI->ai_family << ", type=" << AI->ai_socktype << ", protocol=" << AI->ai_protocol );
 
-    /* set TOS if needed */
-    if (TOS && comm_set_tos(new_socket, TOS) ) {
-        tos = TOS;
-    }
-
     if ( Ip::EnableIpv6&IPV6_SPECIAL_SPLITSTACK && addr.IsIPv6() )
         comm_set_v6only(new_socket, 1);
 
diff -up squid-3.1.11/src/DiskIO/AIO/AIODiskIOStrategy.cc.unused squid-3.1.11/src/DiskIO/AIO/AIODiskIOStrategy.cc
--- squid-3.1.11/src/DiskIO/AIO/AIODiskIOStrategy.cc.unused	2011-02-10 09:42:07.029205002 +0100
+++ squid-3.1.11/src/DiskIO/AIO/AIODiskIOStrategy.cc	2011-02-10 09:42:37.171205594 +0100
@@ -123,7 +123,6 @@ AIODiskIOStrategy::callback()
     void *cbdata;
     int callback_valid;
     void *buf;
-    int filedescriptor;
     async_queue_entry_t *aqe;
     async_queue_entry_type_t type;
 
@@ -148,7 +147,6 @@ AIODiskIOStrategy::callback()
                 /* Get the callback parameters */
                 freefunc = aqe->aq_e_free;
                 buf = aqe->aq_e_buf;
-                filedescriptor = aqe->aq_e_fd;
                 type = aqe->aq_e_type;
                 callback_valid = cbdataReferenceValidDone(aqe->aq_e_callback_data, &cbdata);
                 AIODiskFile * theFile = NULL;
diff -up squid-3.1.11/src/dns_internal.cc.unused squid-3.1.11/src/dns_internal.cc
--- squid-3.1.11/src/dns_internal.cc.unused	2011-02-10 09:15:50.096204636 +0100
+++ squid-3.1.11/src/dns_internal.cc	2011-02-10 09:16:39.951205440 +0100
@@ -1360,8 +1360,6 @@ idnsInit(void)
     CBDATA_INIT_TYPE(idns_query);
 
     if (DnsSocketA < 0 && DnsSocketB < 0) {
-        int port;
-
         IpAddress addrA; // since we don't want to alter Config.Addrs.udp_* and dont have one of our own.
 
         if (!Config.Addrs.udp_outgoing.IsNoAddr())
@@ -1397,12 +1395,10 @@ idnsInit(void)
          * statement. Doing so messes up the internal Debug::level
          */
         if (DnsSocketB >= 0) {
-            port = comm_local_port(DnsSocketB);
             debugs(78, 1, "DNS Socket created at " << addrB << ", FD " << DnsSocketB);
             commSetSelect(DnsSocketB, COMM_SELECT_READ, idnsRead, NULL, 0);
         }
         if (DnsSocketA >= 0) {
-            port = comm_local_port(DnsSocketA);
             debugs(78, 1, "DNS Socket created at " << addrA << ", FD " << DnsSocketA);
             commSetSelect(DnsSocketA, COMM_SELECT_READ, idnsRead, NULL, 0);
         }
diff -up squid-3.1.11/src/errorpage.cc.unused squid-3.1.11/src/errorpage.cc
--- squid-3.1.11/src/errorpage.cc.unused	2011-02-10 08:42:44.965205002 +0100
+++ squid-3.1.11/src/errorpage.cc	2011-02-10 08:43:50.597205002 +0100
@@ -515,7 +515,6 @@ int
 ErrorState::Dump(MemBuf * mb)
 {
     MemBuf str;
-    const char *p = NULL;	/* takes priority over mb if set */
     char ntoabuf[MAX_IPSTRLEN];
 
     str.reset();
@@ -570,10 +569,6 @@ ErrorState::Dump(MemBuf * mb)
         packerToMemInit(&pck, &str);
         request->header.packInto(&pck);
         packerClean(&pck);
-    } else if (request_hdrs) {
-        p = request_hdrs;
-    } else {
-        p = "[none]";
     }
 
     str.Printf("\r\n");
diff -up squid-3.1.11/src/fqdncache.cc.unused squid-3.1.11/src/fqdncache.cc
--- squid-3.1.11/src/fqdncache.cc.unused	2011-02-10 09:29:50.426205631 +0100
+++ squid-3.1.11/src/fqdncache.cc	2011-02-10 09:30:57.407204865 +0100
@@ -497,10 +497,8 @@ fqdncacheHandleReply(void *data, char *r
 fqdncacheHandleReply(void *data, rfc1035_rr * answers, int na, const char *error_message)
 #endif
 {
-    int n;
     fqdncache_entry *f;
     static_cast<generic_cbdata *>(data)->unwrap(&f);
-    n = ++FqdncacheStats.replies;
     const int age = f->age();
     statHistCount(&statCounter.dns.svc_time, age);
 #if USE_DNSSERVERS
diff -up squid-3.1.11/src/ftp.cc.unused squid-3.1.11/src/ftp.cc
--- squid-3.1.11/src/ftp.cc.unused	2011-02-10 09:22:52.118205001 +0100
+++ squid-3.1.11/src/ftp.cc	2011-02-10 09:23:56.324205001 +0100
@@ -2453,7 +2453,6 @@ ftpReadEPSV(FtpStateData* ftpState)
 {
     int code = ftpState->ctrl.replycode;
     char h1, h2, h3, h4;
-    int n;
     u_short port;
     IpAddress ipa_remote;
     int fd = ftpState->data.fd;
@@ -2521,7 +2520,7 @@ ftpReadEPSV(FtpStateData* ftpState)
 
     buf = ftpState->ctrl.last_reply + strcspn(ftpState->ctrl.last_reply, "(");
 
-    n = sscanf(buf, "(%c%c%c%hu%c)", &h1, &h2, &h3, &port, &h4);
+    sscanf(buf, "(%c%c%c%hu%c)", &h1, &h2, &h3, &port, &h4);
 
     if (h1 != h2 || h1 != h3 || h1 != h4) {
         debugs(9, DBG_IMPORTANT, "Invalid EPSV reply from " <<
diff -up squid-3.1.11/src/neighbors.cc.unused squid-3.1.11/src/neighbors.cc
--- squid-3.1.11/src/neighbors.cc.unused	2011-02-10 09:36:24.432205185 +0100
+++ squid-3.1.11/src/neighbors.cc	2011-02-10 09:36:51.359204616 +0100
@@ -827,7 +827,6 @@ neighborsDigestSelect(HttpRequest * requ
     peer *best_p = NULL;
 #if USE_CACHE_DIGESTS
 
-    const cache_key *key;
     int best_rtt = 0;
     int choice_count = 0;
     int ichoice_count = 0;
@@ -838,8 +837,6 @@ neighborsDigestSelect(HttpRequest * requ
     if (!request->flags.hierarchical)
         return NULL;
 
-    key = storeKeyPublicByRequest(request);
-
     for (i = 0, p = first_ping; i++ < Config.npeers; p = p->next) {
         lookup_t lookup;
 
