--- librep-0.17.2/configure.in~ 2008-12-27 12:10:42.000000000 -0700
+++ librep-0.17.2/configure.in  2008-12-27 12:11:26.000000000 -0700
@@ -22,6 +22,7 @@ AC_REVISION($Revision: 1.146 $)
 
 AC_INIT(src/rep_subrs.h)
 AC_CONFIG_HEADER(config.h src/rep_config.h)
+AC_CONFIG_MACRO_DIR([m4])
 
 dnl Release versioning info
 version="0.17.2"
--- librep-0.17.2/autogen.sh~   2008-08-14 09:30:42.000000000 -0600
+++ librep-0.17.2/autogen.sh    2008-12-27 12:32:13.000000000 -0700
@@ -9,7 +9,7 @@ if [ -f configure.in ]; then
   fi
   if grep "A[MC]_PROG_LIBTOOL" configure.in >/dev/null; then
     echo "Running libtoolize"
-    libtoolize --force --copy || exit 1
+    libtoolize --install --force --copy || exit 1
   fi
   echo "Running aclocal $ACLOCAL_FLAGS"
   aclocal $ACLOCAL_FLAGS || exit 1
