--- abiword-plugins/wp/impexp/pdf/xp/ie_imp_PDF.cpp.orig	2006-12-23 19:32:10.000000000 +0000
+++ abiword-plugins/wp/impexp/pdf/xp/ie_imp_PDF.cpp	2006-12-23 19:32:32.000000000 +0000
@@ -75,7 +75,7 @@
       UT_tmpnam(temporary_file);
 
       TextOutputDev *textOut = new TextOutputDev(temporary_file, gFalse, gFalse, gFalse);
-      doc->displayPages(textOut, 1, doc->getNumPages(), 72, 72, 0, gTrue, gFalse);
+      doc->displayPages(textOut, 1, doc->getNumPages(), 72, 72, 0, gTrue, gFalse, gFalse);
       delete textOut;
 
       IE_Imp_Text txt_imp(getDoc(), "UTF-8");      
--- abiword-plugins/wp/impexp/pdf/xp/ie_imp_PDF.cpp	2007-09-07 21:45:24.000000000 +0000
+++ abiword-plugins/wp/impexp/pdf/xp/ie_imp_PDF.cpp	2007-09-07 21:49:02.000000000 +0000
@@ -63,7 +63,7 @@
   
     PDFDoc * doc = new PDFDoc(fileName);
 
-    globalParams = new GlobalParams("");
+    globalParams = new GlobalParams();
     globalParams->setTextEncoding("UTF-8");
     globalParams->setTextEOL("\n");
     globalParams->setTextPageBreaks(gFalse);
