From 3d181030a6ba2205a57ef4665e53aed0251ccaa6 Mon Sep 17 00:00:00 2001
From: Dave Airlie <airlied@gmail.com>
Date: Wed, 18 Jul 2012 08:52:38 +0000
Subject: savage: move compat-api.h down to proper place.

This screws up when DRI isn't enabled.

Signed-off-by: Dave Airlie <airlied@redhat.com>
---
diff --git a/src/savage_driver.h b/src/savage_driver.h
index b604292..ec4b1ea 100644
--- a/src/savage_driver.h
+++ b/src/savage_driver.h
@@ -85,8 +85,6 @@
 #include "GL/glxint.h"
 #include "xf86drm.h"
 
-#include "compat-api.h"
-
 /* Totals 2 Mbytes which equals 2^16 32-byte vertices divided among up
  * to 32 clients. */
 #define SAVAGE_NUM_BUFFERS 32
@@ -153,6 +151,8 @@ typedef struct _server{
 
 #endif
 
+#include "compat-api.h"
+
 typedef enum {
     MT_NONE,
     MT_CRT,
--
cgit v0.9.0.2-2-gbebe
From ca9718a887ecbc59b76869af673791ab591b849b Mon Sep 17 00:00:00 2001
From: Dave Airlie <airlied@gmail.com>
Date: Sat, 22 Dec 2012 10:00:00 +0000
Subject: savage: fix default accel arch when built with no XAA

Signed-off-by: Dave Airlie <airlied@redhat.com>
---
diff --git a/src/savage_driver.c b/src/savage_driver.c
index 026bf5c..5b49e60 100644
--- a/src/savage_driver.c
+++ b/src/savage_driver.c
@@ -1267,6 +1267,7 @@ static Bool SavagePreInit(ScrnInfoPtr pScrn, int flags)
         from = X_DEFAULT;
 	char *strptr;
         if((strptr = (char *)xf86GetOptValString(psav->Options, OPTION_ACCELMETHOD))) {
+#ifdef HAVE_XAA_H
 	    if(!xf86NameCmp(strptr,"XAA")) {
 	        from = X_CONFIG;
 	        psav->useEXA = FALSE;
@@ -1274,6 +1275,9 @@ static Bool SavagePreInit(ScrnInfoPtr pScrn, int flags)
 	       from = X_CONFIG;
 	       psav->useEXA = TRUE;
 	    }
+#else
+	    psav->useEXA = TRUE;
+#endif
        }
        xf86DrvMsg(pScrn->scrnIndex, from, "Using %s acceleration architecture\n",
 		psav->useEXA ? "EXA" : "XAA");
--
cgit v0.9.0.2-2-gbebe
