diff --git i/indentbackends/c/gca-c-backend.vala w/indentbackends/c/gca-c-backend.vala
index f44d431409bf9e40..61ea32cb7d4de026 100644
--- i/indentbackends/c/gca-c-backend.vala
+++ w/indentbackends/c/gca-c-backend.vala
@@ -193,7 +193,7 @@ class Backend : Object, Gca.IndentBackend
 
 		if (document.iter_has_context_class(iter, "comment"))
 		{
-			if (!document.iter_backward_to_context_class_toggle(ref iter, "comment"))
+			if (!document.iter_backward_to_context_class_toggle(iter, "comment"))
 			{
 				return amount;
 			}
