diff -Nur libvisual-plugins-0.4.0/plugins/actor/bumpscope/Makefile.am libvisual-plugins-0.4.0.new/plugins/actor/bumpscope/Makefile.am
--- libvisual-plugins-0.4.0/plugins/actor/bumpscope/Makefile.am	2006-02-23 20:03:24.000000000 +1100
+++ libvisual-plugins-0.4.0.new/plugins/actor/bumpscope/Makefile.am	2007-07-04 18:19:43.000000000 +1000
@@ -2,13 +2,12 @@
 
 actor_plugin_LTLIBRARIES = actor_bumpscope.la
 
-LIBS += $(LIBVISUAL_LIBS)
-
 AM_CFLAGS = $(LIBVISUAL_CFLAGS)
 
 INCLUDES = $(all_includes) -I$(top_srcdir) 
 
 actor_bumpscope_la_LDFLAGS = -module -avoid-version 
+actor_bumpscope_la_LIBADD = $(LIBVISUAL_LIBS)
 
 actor_bumpscope_la_SOURCES = actor_bumpscope.c \
 			  actor_bumpscope.h \
diff -Nur libvisual-plugins-0.4.0/plugins/actor/corona/Makefile.am libvisual-plugins-0.4.0.new/plugins/actor/corona/Makefile.am
--- libvisual-plugins-0.4.0/plugins/actor/corona/Makefile.am	2006-02-23 20:03:24.000000000 +1100
+++ libvisual-plugins-0.4.0.new/plugins/actor/corona/Makefile.am	2007-07-04 18:19:43.000000000 +1000
@@ -2,13 +2,12 @@
 
 actor_plugin_LTLIBRARIES = actor_corona.la
 
-LIBS += $(LIBVISUAL_LIBS)
-
 AM_CXXFLAGS = $(LIBVISUAL_CFLAGS)
 
 INCLUDES = $(all_includes) -I$(top_srcdir)
 
 actor_corona_la_LDFLAGS = -module -avoid-version
+actor_corona_la_LIBADD = $(LIBVISUAL_LIBS)
 
 actor_corona_la_SOURCES = actor_corona.cpp \
 			  autopal.h \
diff -Nur libvisual-plugins-0.4.0/plugins/actor/gdkpixbuf/Makefile.am libvisual-plugins-0.4.0.new/plugins/actor/gdkpixbuf/Makefile.am
--- libvisual-plugins-0.4.0/plugins/actor/gdkpixbuf/Makefile.am	2006-02-24 00:34:48.000000000 +1100
+++ libvisual-plugins-0.4.0.new/plugins/actor/gdkpixbuf/Makefile.am	2007-07-04 18:19:43.000000000 +1000
@@ -2,13 +2,12 @@
 
 actor_plugin_LTLIBRARIES = actor_gdkpixbuf.la
 
-LIBS += $(LIBVISUAL_LIBS) $(GTK_LIBS)
-
 AM_CFLAGS = $(LIBVISUAL_CFLAGS) $(GTK_CFLAGS)
 
 INCLUDES = $(all_includes) -I$(top_srcdir) 
 
 actor_gdkpixbuf_la_LDFLAGS = -module -avoid-version 
+actor_gdkpixbuf_la_LIBADD = $(LIBVISUAL_LIBS) $(GTK_LIBS)
 
 actor_gdkpixbuf_la_SOURCES = actor_gdkpixbuf.c
 
diff -Nur libvisual-plugins-0.4.0/plugins/actor/gstreamer/Makefile.am libvisual-plugins-0.4.0.new/plugins/actor/gstreamer/Makefile.am
--- libvisual-plugins-0.4.0/plugins/actor/gstreamer/Makefile.am	2006-02-23 20:03:27.000000000 +1100
+++ libvisual-plugins-0.4.0.new/plugins/actor/gstreamer/Makefile.am	2007-07-04 18:19:43.000000000 +1000
@@ -2,14 +2,13 @@
 
 actor_plugin_LTLIBRARIES = actor_gstreamer.la
 
-LIBS += $(LIBVISUAL_LIBS) $(GSTREAMER_LIBS)
-
 AM_CFLAGS = $(LIBVISUAL_CFLAGS) $(GSTREAMER_CFLAGS)
 
 INCLUDES = $(all_includes) \
 	-I$(top_srcdir) 
 
 actor_gstreamer_la_LDFLAGS = -module -avoid-version 
+actor_gstreamer_la_LIBADD = $(LIBVISUAL_LIBS) $(GSTREAMER_LIBS)
 
 actor_gstreamer_la_SOURCES = actor_gstreamer.c
 
diff -Nur libvisual-plugins-0.4.0/plugins/actor/infinite/Makefile.am libvisual-plugins-0.4.0.new/plugins/actor/infinite/Makefile.am
--- libvisual-plugins-0.4.0/plugins/actor/infinite/Makefile.am	2006-02-23 20:03:27.000000000 +1100
+++ libvisual-plugins-0.4.0.new/plugins/actor/infinite/Makefile.am	2007-07-04 18:19:43.000000000 +1000
@@ -2,13 +2,12 @@
 
 actor_plugin_LTLIBRARIES = actor_infinite.la
 
-LIBS += $(LIBVISUAL_LIBS)
-
 AM_CFLAGS = $(LIBVISUAL_CFLAGS)
 
 INCLUDES = $(all_includes) -I$(top_srcdir) 
 
 actor_infinite_la_LDFLAGS = -module -avoid-version 
+actor_infinite_la_LIBADD = $(LIBVISUAL_LIBS)
 
 actor_infinite_la_SOURCES = compute.c compute.h \
 			    display.c display.h \
diff -Nur libvisual-plugins-0.4.0/plugins/actor/jakdaw/Makefile.am libvisual-plugins-0.4.0.new/plugins/actor/jakdaw/Makefile.am
--- libvisual-plugins-0.4.0/plugins/actor/jakdaw/Makefile.am	2006-02-23 20:03:28.000000000 +1100
+++ libvisual-plugins-0.4.0.new/plugins/actor/jakdaw/Makefile.am	2007-07-04 18:19:43.000000000 +1000
@@ -2,13 +2,12 @@
 
 actor_plugin_LTLIBRARIES = actor_jakdaw.la
 
-LIBS += $(LIBVISUAL_LIBS)
-
 AM_CFLAGS = $(LIBVISUAL_CFLAGS)
 
 INCLUDES = $(all_includes) -I$(top_srcdir) 
 
 actor_jakdaw_la_LDFLAGS = -module -avoid-version 
+actor_jakdaw_la_LIBADD = $(LIBVISUAL_LIBS)
 
 actor_jakdaw_la_SOURCES = actor_jakdaw.c \
 			  actor_jakdaw.h \
diff -Nur libvisual-plugins-0.4.0/plugins/actor/JESS/Makefile.am libvisual-plugins-0.4.0.new/plugins/actor/JESS/Makefile.am
--- libvisual-plugins-0.4.0/plugins/actor/JESS/Makefile.am	2006-02-23 20:03:24.000000000 +1100
+++ libvisual-plugins-0.4.0.new/plugins/actor/JESS/Makefile.am	2007-07-04 18:19:43.000000000 +1000
@@ -2,13 +2,12 @@
 
 actor_plugin_LTLIBRARIES = actor_JESS.la
 
-LIBS += $(LIBVISUAL_LIBS)
-
 AM_CFLAGS = $(LIBVISUAL_CFLAGS)
 
 INCLUDES = $(all_includes) -I$(top_srcdir) 
 
 actor_JESS_la_LDFLAGS = -module -avoid-version
+actor_JESS_la_LIBADD = $(LIBVISUAL_LIBS)
 
 actor_JESS_la_SOURCES = analyser.c analyser.h analyser_struct.h\
 			distorsion.c distorsion.h\
diff -Nur libvisual-plugins-0.4.0/plugins/actor/lv_analyzer/Makefile.am libvisual-plugins-0.4.0.new/plugins/actor/lv_analyzer/Makefile.am
--- libvisual-plugins-0.4.0/plugins/actor/lv_analyzer/Makefile.am	2006-02-23 20:03:28.000000000 +1100
+++ libvisual-plugins-0.4.0.new/plugins/actor/lv_analyzer/Makefile.am	2007-07-04 18:19:43.000000000 +1000
@@ -2,14 +2,13 @@
 
 actor_plugin_LTLIBRARIES = actor_lv_analyzer.la
 
-LIBS += $(LIBVISUAL_LIBS)
-
 AM_CFLAGS = $(LIBVISUAL_CFLAGS)
 
 INCLUDES = $(all_includes) \
 	-I$(top_srcdir) 
 
 actor_lv_analyzer_la_LDFLAGS = -module -avoid-version 
+actor_lv_analyzer_la_LIBADD = $(LIBVISUAL_LIBS)
 
 actor_lv_analyzer_la_SOURCES = actor_lv_analyzer.c
 
diff -Nur libvisual-plugins-0.4.0/plugins/actor/lv_gltest/Makefile.am libvisual-plugins-0.4.0.new/plugins/actor/lv_gltest/Makefile.am
--- libvisual-plugins-0.4.0/plugins/actor/lv_gltest/Makefile.am	2006-02-24 22:24:49.000000000 +1100
+++ libvisual-plugins-0.4.0.new/plugins/actor/lv_gltest/Makefile.am	2007-07-04 18:19:43.000000000 +1000
@@ -4,5 +4,5 @@
 
 actor_lv_gltest_la_CFLAGS = -I$(top_srcdir) $(LIBVISUAL_CFLAGS) $(X_CFLAGS)
 actor_lv_gltest_la_LDFLAGS = -module -avoid-version 
-actor_lv_gltest_la_LIBS = $(LIBVISUAL_LIBS) $(X_LIBS) -lGL -lGLU 
+actor_lv_gltest_la_LIBADD = $(LIBVISUAL_LIBS) $(X_LIBS) -lGL -lGLU 
 actor_lv_gltest_la_SOURCES = actor_lv_gltest.c
diff -Nur libvisual-plugins-0.4.0/plugins/actor/lv_scope/Makefile.am libvisual-plugins-0.4.0.new/plugins/actor/lv_scope/Makefile.am
--- libvisual-plugins-0.4.0/plugins/actor/lv_scope/Makefile.am	2006-02-23 20:03:29.000000000 +1100
+++ libvisual-plugins-0.4.0.new/plugins/actor/lv_scope/Makefile.am	2007-07-04 18:19:43.000000000 +1000
@@ -2,13 +2,12 @@
 
 actor_plugin_LTLIBRARIES = actor_lv_scope.la
 
-LIBS += $(LIBVISUAL_LIBS)
-
 AM_CFLAGS = $(LIBVISUAL_CFLAGS)
 
 INCLUDES = $(all_includes) -I$(top_srcdir) 
 
 actor_lv_scope_la_LDFLAGS = -module -avoid-version 
+actor_lv_scope_la_LIBADD = $(LIBVISUAL_LIBS)
 
 actor_lv_scope_la_SOURCES = actor_lv_scope.c
 
diff -Nur libvisual-plugins-0.4.0/plugins/actor/madspin/Makefile.am libvisual-plugins-0.4.0.new/plugins/actor/madspin/Makefile.am
--- libvisual-plugins-0.4.0/plugins/actor/madspin/Makefile.am	2006-02-24 22:24:49.000000000 +1100
+++ libvisual-plugins-0.4.0.new/plugins/actor/madspin/Makefile.am	2007-07-04 18:19:43.000000000 +1000
@@ -10,5 +10,5 @@
 
 actor_madspin_la_CFLAGS = -I$(top_srcdir) $(LIBVISUAL_CFLAGS) -DSTAR_DIR=\""$(imagesdir)/"\"
 actor_madspin_la_LDFLAGS = -module -avoid-version 
-actor_madspin_la_LIBS = $(LIBVISUAL_LIBS) $(X_LIBS) -lGL 
+actor_madspin_la_LIBADD = $(LIBVISUAL_LIBS) $(X_LIBS) -lGL 
 actor_madspin_la_SOURCES = madspin.c
diff -Nur libvisual-plugins-0.4.0/plugins/actor/nastyfft/Makefile.am libvisual-plugins-0.4.0.new/plugins/actor/nastyfft/Makefile.am
--- libvisual-plugins-0.4.0/plugins/actor/nastyfft/Makefile.am	2006-02-24 22:24:50.000000000 +1100
+++ libvisual-plugins-0.4.0.new/plugins/actor/nastyfft/Makefile.am	2007-07-04 18:19:43.000000000 +1000
@@ -4,5 +4,5 @@
 
 actor_nastyfft_la_CFLAGS = -I$(top_srcdir) $(LIBVISUAL_CFLAGS) $(X_CFLAGS)
 actor_nastyfft_la_LDFLAGS = -module -avoid-version 
-actor_nastyfft_la_LIBS = $(LIBVISUAL_LIBS) $(X_LIBS) -lGL -lGLU
+actor_nastyfft_la_LIBADD = $(LIBVISUAL_LIBS) $(X_LIBS) -lGL -lGLU
 actor_nastyfft_la_SOURCES = actor_nastyfft.c
diff -Nur libvisual-plugins-0.4.0/plugins/actor/oinksie/Makefile.am libvisual-plugins-0.4.0.new/plugins/actor/oinksie/Makefile.am
--- libvisual-plugins-0.4.0/plugins/actor/oinksie/Makefile.am	2006-02-23 20:03:31.000000000 +1100
+++ libvisual-plugins-0.4.0.new/plugins/actor/oinksie/Makefile.am	2007-07-04 18:19:43.000000000 +1000
@@ -2,13 +2,12 @@
 
 actor_plugin_LTLIBRARIES = actor_oinksie.la
 
-LIBS += $(LIBVISUAL_LIBS)
-
 AM_CFLAGS = $(LIBVISUAL_CFLAGS)
 
 INCLUDES = $(all_includes) -I$(top_srcdir) 
 
 actor_oinksie_la_LDFLAGS = -module -avoid-version 
+actor_oinksie_la_LIBADD = $(LIBVISUAL_LIBS)
 
 actor_oinksie_la_SOURCES = actor_oinksie.c\
 			   oinksie.c oinksie.h\
diff -Nur libvisual-plugins-0.4.0/plugins/actor/pseudotoad_flower/Makefile.am libvisual-plugins-0.4.0.new/plugins/actor/pseudotoad_flower/Makefile.am
--- libvisual-plugins-0.4.0/plugins/actor/pseudotoad_flower/Makefile.am	2006-03-09 00:06:58.000000000 +1100
+++ libvisual-plugins-0.4.0.new/plugins/actor/pseudotoad_flower/Makefile.am	2007-07-04 18:19:43.000000000 +1000
@@ -4,7 +4,7 @@
 
 actor_flower_la_LDFLAGS = -module -avoid-version 
 actor_flower_la_CFLAGS = -I$(top_srcdir) $(LIBVISUAL_CFLAGS) $(X_CFLAGS)
-actor_flower_la_LIBS = $(LIBVISUAL_LIBS) $(X_LIBS) -lGL -lGLU 
+actor_flower_la_LIBADD = $(LIBVISUAL_LIBS) $(X_LIBS) -lGL -lGLU 
 
 actor_flower_la_SOURCES = \
 	actor_flower.c \
diff -Nur libvisual-plugins-0.4.0/plugins/input/alsa/Makefile.am libvisual-plugins-0.4.0.new/plugins/input/alsa/Makefile.am
--- libvisual-plugins-0.4.0/plugins/input/alsa/Makefile.am	2006-02-23 20:03:35.000000000 +1100
+++ libvisual-plugins-0.4.0.new/plugins/input/alsa/Makefile.am	2007-07-04 18:19:43.000000000 +1000
@@ -2,12 +2,11 @@
 
 input_plugin_LTLIBRARIES = input_alsa.la
 
-LIBS += $(ALSA_LIBS) $(LIBVISUAL_LIBS)
-
 AM_CFLAGS = $(ALSA_CFLAGS) $(LIBVISUAL_CFLAGS)
 
 INCLUDES = $(all_includes) -I$(top_srcdir) -I$(top_builddir)
 
 input_alsa_la_LDFLAGS = -module -avoid-version 
+input_alsa_la_LIBADD = $(ALSA_LIBS) $(LIBVISUAL_LIBS)
 
 input_alsa_la_SOURCES = input_alsa.c
diff -Nur libvisual-plugins-0.4.0/plugins/input/debug/Makefile.am libvisual-plugins-0.4.0.new/plugins/input/debug/Makefile.am
--- libvisual-plugins-0.4.0/plugins/input/debug/Makefile.am	2006-02-23 20:03:36.000000000 +1100
+++ libvisual-plugins-0.4.0.new/plugins/input/debug/Makefile.am	2007-07-04 18:19:43.000000000 +1000
@@ -2,11 +2,10 @@
 
 input_plugin_LTLIBRARIES = input_debug.la
 
-LIBS += $(LIBVISUAL_LIBS)
-
 AM_CFLAGS = $(LIBVISUAL_CFLAGS)
 
 INCLUDES = $(all_includes) -I$(top_srcdir) -I$(top_builddir)
 
 input_debug_la_LDFLAGS = -module -avoid-version 
+input_debug_la_LIBADD = $(LIBVISUAL_LIBS)
 input_debug_la_SOURCES = input_debug.c
diff -Nur libvisual-plugins-0.4.0/plugins/input/esd/Makefile.am libvisual-plugins-0.4.0.new/plugins/input/esd/Makefile.am
--- libvisual-plugins-0.4.0/plugins/input/esd/Makefile.am	2006-02-23 20:03:37.000000000 +1100
+++ libvisual-plugins-0.4.0.new/plugins/input/esd/Makefile.am	2007-07-04 18:19:43.000000000 +1000
@@ -2,11 +2,10 @@
 
 input_plugin_LTLIBRARIES = input_esd.la
 
-LIBS += $(LIBESD_LIBS) $(LIBVISUAL_LIBS)
-
 AM_CFLAGS = $(LIBESD_CFLAGS) $(LIBVISUAL_CFLAGS)
 
 INCLUDES = $(all_includes) -I$(top_srcdir)
 
 input_esd_la_LDFLAGS = -module -avoid-version 
+input_esd_la_LIBADD = $(LIBESD_LIBS) $(LIBVISUAL_LIBS)
 input_esd_la_SOURCES = input_esd.c
diff -Nur libvisual-plugins-0.4.0/plugins/input/jack/Makefile.am libvisual-plugins-0.4.0.new/plugins/input/jack/Makefile.am
--- libvisual-plugins-0.4.0/plugins/input/jack/Makefile.am	2006-02-23 20:03:37.000000000 +1100
+++ libvisual-plugins-0.4.0.new/plugins/input/jack/Makefile.am	2007-07-04 18:19:43.000000000 +1000
@@ -2,13 +2,12 @@
 
 input_plugin_LTLIBRARIES = input_jack.la
 
-LIBS += $(LIBJACK_LIBS) $(LIBVISUAL_LIBS)
-
 AM_CFLAGS = $(LIBJACK_CFLAGS) $(LIBVISUAL_CFLAGS)
 
 INCLUDES = $(all_includes) -I$(top_srcdir) 
 
 input_jack_la_LDFLAGS = -module -avoid-version 
+input_jack_la_LIBADD = $(LIBJACK_LIBS) $(LIBVISUAL_LIBS)
 
 input_jack_la_SOURCES = input_jack.c
 
diff -Nur libvisual-plugins-0.4.0/plugins/input/mplayer/Makefile.am libvisual-plugins-0.4.0.new/plugins/input/mplayer/Makefile.am
--- libvisual-plugins-0.4.0/plugins/input/mplayer/Makefile.am	2006-02-23 20:03:38.000000000 +1100
+++ libvisual-plugins-0.4.0.new/plugins/input/mplayer/Makefile.am	2007-07-04 18:19:43.000000000 +1000
@@ -2,12 +2,11 @@
 
 input_plugin_LTLIBRARIES = input_mplayer.la
 
-LIBS += $(LIBVISUAL_LIBS)
-
 AM_CFLAGS = $(LIBVISUAL_CFLAGS)
 
 INCLUDES = $(all_includes) -I$(top_srcdir) 
 
 input_mplayer_la_LDFLAGS = -module -avoid-version 
+input_mplayer_la_LIBADD = $(LIBVISUAL_LIBS)
 input_mplayer_la_SOURCES = input_mplayer.c
 
diff -Nur libvisual-plugins-0.4.0/plugins/morph/alphablend/Makefile.am libvisual-plugins-0.4.0.new/plugins/morph/alphablend/Makefile.am
--- libvisual-plugins-0.4.0/plugins/morph/alphablend/Makefile.am	2006-02-23 20:03:38.000000000 +1100
+++ libvisual-plugins-0.4.0.new/plugins/morph/alphablend/Makefile.am	2007-07-04 18:19:43.000000000 +1000
@@ -2,11 +2,10 @@
 
 morph_plugin_LTLIBRARIES = morph_alphablend.la
 
-LIBS += $(LIBVISUAL_LIBS)
-
 AM_CFLAGS = $(LIBVISUAL_CFLAGS)
 
 INCLUDES = $(all_includes) -I$(top_srcdir) 
 
 morph_alphablend_la_LDFLAGS = -module -avoid-version 
+morph_alphablend_la_LIBADD = $(LIBVISUAL_LIBS)
 morph_alphablend_la_SOURCES = morph_alphablend.c
diff -Nur libvisual-plugins-0.4.0/plugins/morph/flash/Makefile.am libvisual-plugins-0.4.0.new/plugins/morph/flash/Makefile.am
--- libvisual-plugins-0.4.0/plugins/morph/flash/Makefile.am	2006-02-23 20:03:39.000000000 +1100
+++ libvisual-plugins-0.4.0.new/plugins/morph/flash/Makefile.am	2007-07-04 18:19:43.000000000 +1000
@@ -2,11 +2,10 @@
 
 morph_plugin_LTLIBRARIES = morph_flash.la
 
-LIBS += $(LIBVISUAL_LIBS)
-
 AM_CFLAGS = $(LIBVISUAL_CFLAGS)
 
 INCLUDES = $(all_includes) -I$(top_srcdir) 
 
 morph_flash_la_LDFLAGS = -module -avoid-version 
+morph_flash_la_LIBADD = $(LIBVISUAL_LIBS)
 morph_flash_la_SOURCES = morph_flash.c
diff -Nur libvisual-plugins-0.4.0/plugins/morph/slide/Makefile.am libvisual-plugins-0.4.0.new/plugins/morph/slide/Makefile.am
--- libvisual-plugins-0.4.0/plugins/morph/slide/Makefile.am	2006-02-23 20:03:39.000000000 +1100
+++ libvisual-plugins-0.4.0.new/plugins/morph/slide/Makefile.am	2007-07-04 18:19:43.000000000 +1000
@@ -2,11 +2,10 @@
 
 morph_plugin_LTLIBRARIES = morph_slide.la
 
-LIBS += $(LIBVISUAL_LIBS)
-
 AM_CFLAGS = $(LIBVISUAL_CFLAGS)
 
 INCLUDES = $(all_includes) -I$(top_srcdir) 
 
 morph_slide_la_LDFLAGS = -module -avoid-version 
+morph_slide_la_LIBADD = $(LIBVISUAL_LIBS)
 morph_slide_la_SOURCES = morph_slide.c
diff -Nur libvisual-plugins-0.4.0/plugins/morph/tentacle/Makefile.am libvisual-plugins-0.4.0.new/plugins/morph/tentacle/Makefile.am
--- libvisual-plugins-0.4.0/plugins/morph/tentacle/Makefile.am	2006-02-23 20:03:39.000000000 +1100
+++ libvisual-plugins-0.4.0.new/plugins/morph/tentacle/Makefile.am	2007-07-04 18:19:43.000000000 +1000
@@ -2,11 +2,10 @@
 
 morph_plugin_LTLIBRARIES = morph_tentacle.la
 
-LIBS += $(LIBVISUAL_LIBS)
-
 AM_CFLAGS = $(LIBVISUAL_CFLAGS)
 
 INCLUDES = $(all_includes) -I$(top_srcdir) 
 
 morph_tentacle_la_LDFLAGS = -module -avoid-version 
+morph_tentacle_la_LIBADD = $(LIBVISUAL_LIBS)
 morph_tentacle_la_SOURCES = morph_tentacle.c
