diff -up gst-plugins-bad-0.10.9/ext/x264/gstx264enc.c.x264 gst-plugins-bad-0.10.9/ext/x264/gstx264enc.c
--- gst-plugins-bad-0.10.9/ext/x264/gstx264enc.c.x264	2008-10-11 01:13:43.000000000 +0200
+++ gst-plugins-bad-0.10.9/ext/x264/gstx264enc.c	2008-12-14 22:26:13.000000000 +0100
@@ -541,7 +541,7 @@ gst_x264_enc_init_encoder (GstX264Enc * 
   encoder->x264param.i_frame_reference = encoder->ref;
   encoder->x264param.i_bframe = encoder->bframes;
   encoder->x264param.b_bframe_pyramid = encoder->b_pyramid;
-  encoder->x264param.b_bframe_adaptive = encoder->b_adapt;
+  encoder->x264param.i_bframe_adaptive = encoder->b_adapt;
   encoder->x264param.b_interlaced = encoder->interlaced;
   encoder->x264param.b_deblocking_filter = 1;
   encoder->x264param.i_deblocking_filter_alphac0 = 0;
