diff -urN gweled-0.7~/src/main.c gweled-0.7/src/main.c
--- gweled-0.7~/src/main.c	2005-05-25 23:44:04.000000000 +0200
+++ gweled-0.7/src/main.c	2006-12-05 18:29:49.000000000 +0100
@@ -254,7 +254,11 @@
 				PACKAGE_DATA_DIR, NULL);
 
     /* register all the drivers */
-    MikMod_RegisterAllDrivers();
+    MikMod_RegisterDriver(&drv_AF);
+    MikMod_RegisterDriver(&drv_esd);
+    MikMod_RegisterDriver(&drv_alsa);
+    MikMod_RegisterDriver(&drv_oss);
+    MikMod_RegisterDriver(&drv_nos);
 
     /* register all the module loaders */
     MikMod_RegisterAllLoaders();
