diff -ur xf86-video-neomagic-1.2.7.orig/src/neo_driver.c xf86-video-neomagic-1.2.7/src/neo_driver.c
--- xf86-video-neomagic-1.2.7.orig/src/neo_driver.c	2012-12-26 22:14:47.876393607 -0800
+++ xf86-video-neomagic-1.2.7/src/neo_driver.c	2012-12-26 22:17:10.011933399 -0800
@@ -1084,9 +1084,13 @@
 	xf86DrvMsg(pScrn->scrnIndex,X_CONFIG,
 		   "Show chache for debugging\n");
 
+    /*
+     * Seems modules shadow and/or shadowfb causes black screen and dead
+     * keyboard with xorg 1.13. Just disable it for now.
+     */
     if (!xf86LoadSubModule(pScrn, "xaa")) {
 	xf86DrvMsg(pScrn->scrnIndex, X_INFO, "Falling back to shadow\n");
-	nPtr->shadowFB = 1;
+	nPtr->shadowFB = 0;
     }
 
     if (nPtr->shadowFB) {
