Index: modules/libpr0n/decoders/icon/nsIconURI.cpp
===================================================================
RCS file: /cvsroot/mozilla/modules/libpr0n/decoders/icon/nsIconURI.cpp,v
retrieving revision 1.19
diff -p -u -u -p -U10 -r1.19 nsIconURI.cpp
--- modules/libpr0n/decoders/icon/nsIconURI.cpp	4 Feb 2005 23:22:57 -0000	1.19
+++ modules/libpr0n/decoders/icon/nsIconURI.cpp	12 Oct 2005 22:48:03 -0000
@@ -427,27 +427,27 @@ nsMozIconURI::SchemeIs(const char *i_Sch
   NS_ENSURE_ARG_POINTER(o_Equals);
   if (!i_Scheme) return NS_ERROR_INVALID_ARG;
   
   *o_Equals = PL_strcasecmp("moz-icon", i_Scheme) ? PR_FALSE : PR_TRUE;
   return NS_OK;
 }
 
 NS_IMETHODIMP
 nsMozIconURI::Clone(nsIURI **result)
 {
-  return NS_OK;
+  return NS_ERROR_NOT_IMPLEMENTED;
 }
 
 NS_IMETHODIMP
 nsMozIconURI::Resolve(const nsACString &relativePath, nsACString &result)
 {
-  return NS_OK;
+  return NS_ERROR_NOT_IMPLEMENTED;
 }
 
 NS_IMETHODIMP
 nsMozIconURI::GetAsciiSpec(nsACString &aSpecA)
 {
   return GetSpec(aSpecA);
 }
 
 NS_IMETHODIMP
 nsMozIconURI::GetAsciiHost(nsACString &aHostA)
