From a0f23e1706b34bca6a65183040d1f1498cce2a50 Mon Sep 17 00:00:00 2001
From: Matthias Clasen <mclasen@redhat.com>
Date: Thu, 25 Mar 2010 04:55:15 +0000
Subject: Fix a 64bit issue

Thank you libpng, for typedef unsigned long png_uint_32.
---
diff --git a/gdk-pixbuf/io-png.c b/gdk-pixbuf/io-png.c
index c0374ca..43db70a 100644
--- a/gdk-pixbuf/io-png.c
+++ b/gdk-pixbuf/io-png.c
@@ -261,7 +261,7 @@ gdk_pixbuf__png_image_load (FILE *f, GError **error)
         gchar *icc_profile_base64;
         const gchar *icc_profile_title;
         const gchar *icc_profile;
-        guint icc_profile_size;
+        gulong icc_profile_size;
         guint32 retval;
         gint compression_type;
 
@@ -607,7 +607,7 @@ png_info_callback   (png_structp png_read_ptr,
         gchar *icc_profile_base64;
         const gchar *icc_profile_title;
         const gchar *icc_profile;
-        guint icc_profile_size;
+        gulong icc_profile_size;
         guint32 retval;
         gint compression_type;
 
--
cgit v0.8.3.1
