diff -urN tetex-src-2.0.2/texk/web2c/configure tetex-src-2.0.2-new/texk/web2c/configure
--- tetex-src-2.0.2/texk/web2c/configure	2003-02-21 06:49:42.000000000 -0800
+++ tetex-src-2.0.2-new/texk/web2c/configure	2004-03-21 22:20:21.000000000 -0800
@@ -1095,6 +1095,10 @@
 
 fi
 
+case $LEX in
+  *flex) LEX="$LEX -l" ;;
+esac
+
 # Extract the first word of "ranlib", so it can be a program name with args.
 set dummy ranlib; ac_word=$2
 echo $ac_n "checking for $ac_word""... $ac_c" 1>&6
diff -urN tetex-src-2.0.2/texk/web2c/configure.in tetex-src-2.0.2-new/texk/web2c/configure.in
--- tetex-src-2.0.2/texk/web2c/configure.in	2003-02-21 06:49:33.000000000 -0800
+++ tetex-src-2.0.2-new/texk/web2c/configure.in	2004-03-21 22:17:24.000000000 -0800
@@ -30,6 +30,9 @@
 
 AC_PROG_YACC
 AC_PROG_LEX
+case $LEX in
+  *flex) LEX="$LEX -l" ;;
+esac
 AC_PROG_RANLIB
 AC_CHECK_HEADERS(sys/time.h sys/timeb.h)
 AC_CHECK_FUNCS(strerror gettimeofday ftime mkstemp mktemp)
