--- tools/90-libgpod.rules.in~	2011-07-24 06:03:29.000000000 -0500
+++ tools/90-libgpod.rules.in	2019-10-15 15:50:28.279770507 -0500
@@ -16,10 +16,10 @@
 
 
 # "iPods" using the afc protocol (iPhone, iPod Touch, ...)
-ACTION=="add|change", ENV{USBMUX_SUPPORTED}=="1", IMPORT{program}+="@udevdir@/iphone-set-info", GOTO="libgpod_end"
+ACTION=="add|change", ENV{USBMUX_SUPPORTED}=="1", IMPORT{program}=="@udevdir@/iphone-set-info", GOTO="libgpod_end"
 # with a new enough usbmuxd, the rule below is not useful since it will
 # take care of setting USBMUX_SUPPORTED for all supported devices. Keep
 # this rule for now until distros ship a new enough usbmuxd
-ACTION=="add|change", SUBSYSTEM=="usb", ATTR{idVendor}=="05ac", ATTR{idProduct}=="129[0-9a]", IMPORT{program}+="@udevdir@/iphone-set-info"
+ACTION=="add|change", SUBSYSTEM=="usb", ATTR{idVendor}=="05ac", ATTR{idProduct}=="129[0-9a]", IMPORT{program}=="@udevdir@/iphone-set-info"
 
 LABEL="libgpod_end"
