--- src/fe-text/fe-text.c.orig	2007-04-10 19:00:26.000000000 +0200
+++ src/fe-text/fe-text.c	2007-04-10 19:00:42.000000000 +0200
@@ -856,6 +856,6 @@
 }
 void fe_tray_set_flash (const char *filename1, const char *filename2, int timeout){}
 void fe_tray_set_file (const char *filename){}
-void fe_tray_set_icon (int icon){}
+void fe_tray_set_icon (feicon icon){}
 void fe_tray_set_tooltip (const char *text){}
 void fe_tray_set_balloon (const char *title, const char *text){}
