diff -urp xchat-2.8.6.orig/src/fe-gtk/gtkutil.c xchat-2.8.6/src/fe-gtk/gtkutil.c
--- xchat-2.8.6.orig/src/fe-gtk/gtkutil.c	2009-07-18 11:25:02.000000000 +0100
+++ xchat-2.8.6/src/fe-gtk/gtkutil.c	2009-07-18 11:27:01.000000000 +0100
@@ -376,6 +376,7 @@ gtkutil_button (GtkWidget *box, char *st
 	{
 		gtk_button_set_label (GTK_BUTTON (wid), labeltext);
 		gtk_button_set_image (GTK_BUTTON (wid), gtk_image_new_from_stock (stock, GTK_ICON_SIZE_MENU));
+		gtk_button_set_use_underline (GTK_BUTTON (wid), TRUE);
 		if (box)
 			gtk_container_add (GTK_CONTAINER (box), wid);
 	}
