--- kded/applications.menu	2008-05-21 11:07:34.000000000 +0000
+++ kded/applications.menu.new	2008-09-07 11:11:22.000000000 +0000
@@ -21,6 +21,14 @@
 	</Layout>
 
 	<Menu>
+		<Name>Archlinux</Name>
+		<Directory>Archlinux.directory</Directory>
+		<Include>
+			<Category>Archlinux</Category>
+		</Include>
+	</Menu>
+
+	<Menu>
 		<Name>Applications</Name>
 		<Directory>kde-unknown.directory</Directory>
 		<OnlyUnallocated/>
