Index: kdelibs/kdecore/kernel/kcmdlineargs.cpp
===================================================================
--- kdelibs/kdecore/kernel/kcmdlineargs.cpp	(Revision 938656)
+++ kdelibs/kdecore/kernel/kcmdlineargs.cpp	(Revision 938657)
@@ -1407,7 +1407,7 @@
       value = d->parsedOptionList->value(opt);
    }
    if (!value.isEmpty())
-      return value;
+       return QString::fromLocal8Bit(value);
 
    // Look up the default.
    QByteArray opt_name;
