--- khtml/imload/decoders/jpegloader.cpp	2009/07/17 22:28:35	998544
+++ khtml/imload/decoders/jpegloader.cpp	2009/07/17 22:31:07	998545
@@ -324,7 +324,7 @@
                 scaleDown *= 2;
             }
             
-            cinfo.scale_denom = scaleDown;
+            cinfo.scale_denom *= scaleDown;
             
             if (scaleDown > 8) {
                 // Still didn't fit... Abort.
