From 900baff9ff400a6eba034089ed69362c2c2d61e4 Mon Sep 17 00:00:00 2001
From: Petr Mladek <pmladek@suse.cz>
Date: Wed, 22 Jun 2011 12:05:19 +0200
Subject: [PATCH] fix configure without the optional translations module

---
 set_soenv.in |    8 +++++---
 1 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/set_soenv.in b/set_soenv.in
index f015bab..8e39314 100755
--- a/set_soenv.in
+++ b/set_soenv.in
@@ -1068,9 +1068,11 @@ $XCLASSPATH           = '$JAVA_HOME'.$ds.'jre'.$LIB.$ds."rt.jar".$wps.'.';
 $L10N_MODULE = PathFormat($SRC_ROOT."/translations");
 
 # Check for poor help localizations, i.e. no help translation at all...
-opendir(DIR,$L10N_MODULE . "/source");
-@languages = readdir(DIR);
-closedir(DIR);
+@languages=();
+if (opendir(DIR,$L10N_MODULE . "/source"))
+{  @languages = readdir(DIR);
+   closedir(DIR);
+}
 $WITH_POOR_HELP_LOCALIZATIONS = "";
 
 foreach $language (@languages)
-- 
1.7.3.4

