Index: runlabel.in
===================================================================
--- runlabel.in	(revision 5685)
+++ runlabel.in	(working copy)
@@ -111,7 +111,7 @@
 #else
   const string null="/dev/null";
 #endif
-  cmd.push_back("-sDEVICE=epswrite");
+  cmd.push_back("-sDEVICE=eps2write");
   cmd.push_back("-sOutputFile="+null);
   cmd.push_back(stripDir(psname));
   iopipestream gs(cmd,"gs","Ghostscript");
@@ -309,7 +309,7 @@
       cmd.push_back("-dNOPAUSE");
       cmd.push_back("-dBATCH");
       if(safe) cmd.push_back("-dSAFER");
-      cmd.push_back("-sDEVICE=epswrite");
+      cmd.push_back("-sDEVICE=eps2write");
       cmd.push_back("-sOutputFile=-");
       cmd.push_back(pdfname);
       iopipestream gs(cmd,"gs","Ghostscript");
@@ -398,7 +398,7 @@
   cmd2.push_back("-dBATCH");
   cmd2.push_back("-P");
   if(safe) cmd2.push_back("-dSAFER");
-  cmd2.push_back("-sDEVICE=epswrite");
+  cmd2.push_back("-sDEVICE=eps2write");
   cmd2.push_back("-sOutputFile=-");
   cmd2.push_back("-");
   iopipestream gs(cmd2,"gs","Ghostscript");
