--- qca-tls.cpp.orig	2005-12-27 22:22:32.000000000 +0000
+++ qca-tls.cpp
@@ -454,7 +454,11 @@ public:
 			if(!r) {
 				// try this other public function, for whatever reason
 				p = (void *)in;
+#if OPENSSL_VERSION_NUMBER >= 0x00908000
+				r = d2i_RSA_PUBKEY(NULL, (const unsigned char **)&p, len);
+#else
 				r = d2i_RSA_PUBKEY(NULL, (unsigned char **)&p, len);
+#endif
 			}
 			if(r) {
 				if(pub) {
@@ -798,7 +802,11 @@ public:
 
 	bool createFromDER(const char *in, unsigned int len)
 	{
+#if OPENSSL_VERSION_NUMBER >= 0x00908000
+		const unsigned char *p = (const unsigned char *)in;
+#else
 		unsigned char *p = (unsigned char *)in;
+#endif
 		X509 *t = d2i_X509(NULL, &p, len);
 		if(!t)
 			return false;
