diff --git a/data/ontology2code b/data/ontology2code
index 63af897d..b05f7412 100755
--- a/data/ontology2code
+++ b/data/ontology2code
@@ -195,7 +195,7 @@ class Utils:
 	def replace_items(item_set, item_map):
 		if not item_set:
 			return
-		for item, value in item_map.iteritems():
+		for item, value in item_map.items():
 			try:
 				item_set.remove(item)
 			except KeyError:
@@ -277,7 +277,7 @@ class GenericSerializer:
 class PythonSerializer(GenericSerializer):
 
 	def dump(self):
-		for symbol in sorted(self.symbols.itervalues()):
+		for symbol in sorted(self.symbols.values()):
 			parents = set((symbol.uri for symbol in symbol.parents))
 			Utils.replace_items(parents, {
 				str(NIENS['InformationElement']): 'Interpretation',
@@ -435,7 +435,7 @@ class OntologyCodeGenerator:
 if __name__ == "__main__":
 	parser = argparse.ArgumentParser()
 	parser.add_argument('--vala', nargs=4, metavar=('URI_TEMPLATE', 'SYMBOLS_TEMPLATE', 'URI_DESTINATION', 'SYMBOLS_DESTINATION'))
-	parser.add_argument('--dump-python', action='store_false')
+	parser.add_argument('--dump-python', action='store_true')
 
 	args = parser.parse_args()
 
