--- aaa/wscript	2022-12-13 17:18:29.405894137 +0100
+++ bbb/wscript	2022-12-13 17:18:06.552552506 +0100
@@ -212,8 +212,8 @@
     db = opt.add_auto_option(
             'db',
             help='Use Berkeley DB (metadata)')
-    db.check(header_name='db.h')
-    db.check(lib='db')
+    db.check(header_name='db5.3/db.h')
+    db.check(lib='db-5.3')
     zalsa = opt.add_auto_option(
             'zalsa',
             help='Build internal zita-a2j/j2a client')
