--- openjdk/jdk/make/common/shared/Defs-utils.gmk.old	2008-04-13 13:33:23.000000000 +0300
+++ openjdk/jdk/make/common/shared/Defs-utils.gmk	2008-04-14 15:37:34.000000000 +0300
@@ -173,7 +173,7 @@
   BASENAME=$(firstword $(wildcard $(UTILS_COMMAND_PATH)basename) \
                        $(wildcard $(UTILS_USR_BIN_PATH)basename))
 
-  NAWK           = $(USRBIN_PATH)gawk
+  NAWK           = $(UTILS_COMMAND_PATH)gawk
   # Intrinsic unix command, with backslash-escaped character interpretation
   ECHO           = /bin/echo -e
   # These are really in UTILS_USR_BIN_PATH on Linux (only sccs is not)
