--- coredrv/locks.c~	2011-01-26 12:12:00.856675530 +0000
+++ coredrv/locks.c	2011-01-26 12:12:00.873342198 +0000
@@ -141,15 +141,15 @@
 #endif
 } 
 
-DECLARE_MUTEX(sem0);
-DECLARE_MUTEX(sem1);
-DECLARE_MUTEX(sem2);
-DECLARE_MUTEX(sem3);
-DECLARE_MUTEX(sem4);
-DECLARE_MUTEX(sem5);
-DECLARE_MUTEX(sem6);
-DECLARE_MUTEX(sem7);
-DECLARE_MUTEX(sem8);
+DEFINE_SEMAPHORE(sem0);
+DEFINE_SEMAPHORE(sem1);
+DEFINE_SEMAPHORE(sem2);
+DEFINE_SEMAPHORE(sem3);
+DEFINE_SEMAPHORE(sem4);
+DEFINE_SEMAPHORE(sem5);
+DEFINE_SEMAPHORE(sem6);
+DEFINE_SEMAPHORE(sem7);
+DEFINE_SEMAPHORE(sem8);
 
 asmlinkage void linux_RSAGetExclusive(int Resource, unsigned char *Indicator) 
 {
--- coredrv/softserial_io.c~	2011-01-26 12:16:55.753342197 +0000
+++ coredrv/softserial_io.c	2011-01-26 12:16:55.763342199 +0000
@@ -65,7 +65,7 @@
 }
 
 //=============================================================================
-static DECLARE_MUTEX(softserial_write_sem);
+static DEFINE_SEMAPHORE(softserial_write_sem);
 int softserial_write(struct tty_struct* ptty,
                       //int from_user_space,
                       const unsigned char* input_buffer,
--- coredrv/coredrv.c~	2011-01-26 12:15:54.713342038 +0000
+++ coredrv/coredrv.c	2011-01-26 12:15:54.726675572 +0000
@@ -765,7 +765,7 @@
 extern void do_schedule(void);
 #endif
 
-   DECLARE_MUTEX(exec_reg_sem);
+   DEFINE_SEMAPHORE(exec_reg_sem);
    DECLARE_WAIT_QUEUE_HEAD(persistWriteQ);
    DECLARE_WAIT_QUEUE_HEAD(persistReadQ);
    DECLARE_WAIT_QUEUE_HEAD(persistShutdownQ);
