--- coredrv/coredrv.c~	2007-11-02 08:35:56.000000000 +0000
+++ coredrv/coredrv.c	2007-11-02 08:36:47.000000000 +0000
@@ -392,7 +392,7 @@
    printk(KERN_INFO"open: request_irq CALL\n");
    retval = request_irq(G.irq,
                interrupt_handler,
-               SA_SHIRQ,
+               IRQF_SHARED,
                MDM_DEVICE_NAME,
                &(G.irq_share));
    
--- coredrv/Makefile~	2005-07-26 19:59:56.000000000 +0200
+++ coredrv/Makefile	2007-11-07 20:33:49.000000000 +0100
@@ -8,7 +8,7 @@
 ifeq ($(KERNEL_SOURCE_PATH),)
      INCLUDES = -I /lib/modules/`uname -r`/build/include -I ../inc
 endif
-CFLAGS += $(PSTN_DEF) -Wall -O $(INCLUDES) -fno-common -DTARGET_CATAWBA
+KBUILD_CFLAGS += $(PSTN_DEF) -Wall -O $(INCLUDES) -fno-common -DTARGET_CATAWBA
 
 DRIVER_OBJS = coredrv.o clmmain.o rts.o task.o uart.o wwh_dflt.o locks.o \
 		softserial_io.o softserial_ioctl.o softserial.o
