From de5da56d10a47ff8fdf55c0818e9e776dcfa01f9 Mon Sep 17 00:00:00 2001
From: Johannes Huber <johu@gentoo.org>
Date: Thu, 5 Jan 2012 10:06:05 +0100
Subject: [PATCH] Fix FindXine.cmake module with >=xine-lib-1.2.

---
 cmake/modules/FindXine.cmake |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/cmake/modules/FindXine.cmake b/cmake/modules/FindXine.cmake
index 37c58c6..d71969a 100644
--- a/cmake/modules/FindXine.cmake
+++ b/cmake/modules/FindXine.cmake
@@ -36,13 +36,13 @@ find_library(XINE_LIBRARY NAMES xine
   ${PC_LIBXINE_LIBRARY_DIRS}
  )
 
-find_program(XINECONFIG_EXECUTABLE NAMES xine-config 
+find_program(XINECONFIG_EXECUTABLE NAMES pkg-config
    HINTS
    ${PC_LIBXINE_PREFIX}/bin
 )
 
 if (XINE_INCLUDE_DIR AND XINE_LIBRARY AND XINECONFIG_EXECUTABLE)
-   exec_program(${XINECONFIG_EXECUTABLE} ARGS --version RETURN_VALUE _return_VALUE OUTPUT_VARIABLE XINE_VERSION ERROR_QUIET)
+   exec_program(${XINECONFIG_EXECUTABLE} ARGS --modversion libxine RETURN_VALUE _return_VALUE OUTPUT_VARIABLE XINE_VERSION ERROR_QUIET)
    if("${XINE_VERSION}" VERSION_GREATER "1.1.0")   #if (... VERSION_GREATER) is new since cmake 2.6.2
       set(XINE_VERSION_OK TRUE)
       string(REGEX REPLACE "[0-9]\\.[0-9]\\." "" XINE_BUGFIX_VERSION ${XINE_VERSION})
-- 
1.7.8.2

