From b41b65612085a145622d23207b810e9cbb40e3cb Mon Sep 17 00:00:00 2001
From: Theo Linkspfeifer <lastonestanding@tutanota.com>
Date: Thu, 11 Jun 2020 06:02:02 -0400
Subject: [PATCH] Allow resizing the xfce4-settings window smaller (fixes #19)

Signed-off-by: Sean Davis <smd.seandavis@gmail.com>
---
 exo/exo-icon-view.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/exo/exo-icon-view.c b/exo/exo-icon-view.c
index 939658fb..deba7f5d 100644
--- a/exo/exo-icon-view.c
+++ b/exo/exo-icon-view.c
@@ -1669,7 +1669,9 @@ exo_icon_view_get_preferred_width (GtkWidget *widget,
   GList                    *lp;
 
   /* well, this is easy */
-  *minimal_width = *natural_width = priv->width;
+  if (priv->item_width < 0)
+    *minimal_width = priv->width;
+  *natural_width = priv->width;
 
   /* handle the child widgets */
   for (lp = priv->children; lp != NULL; lp = lp->next)
@@ -1691,7 +1693,8 @@ exo_icon_view_get_preferred_height (GtkWidget *widget,
   GList                    *lp;
 
   /* well, this is easy */
-  *natural_height = *minimal_height = priv->height;
+  *minimal_height = priv->height;
+  *natural_height = priv->height;
 
   /* handle the child widgets */
   for (lp = priv->children; lp != NULL; lp = lp->next)
-- 
2.26.2

