Index: python/pykde4/sip/akonadi/addressattribute.sip
===================================================================
--- python/pykde4/sip/akonadi/addressattribute.sip	(revision 1150375)
+++ python/pykde4/sip/akonadi/addressattribute.sip	(revision 1150562)
@@ -57,6 +57,10 @@
     else if (dynamic_cast<Akonadi::MessageThreadingAttribute*>(sipCpp))
         sipType = sipType_Akonadi_MessageThreadingAttribute;
 %End
+private:
+//force
+Akonadi::AddressAttribute& operator = (const Akonadi::AddressAttribute&);
+//end
 };
 };
 %ModuleHeaderCode
