--- a/src/crypto/tls_openssl.c	2016-10-02 19:51:11.000000000 +0100
+++ b/src/crypto/tls_openssl.c	2017-06-02 11:17:37.303222333 +0100
@@ -2779,6 +2779,8 @@
 	} else
 		passwd = NULL;
 
+	SSL_set_default_passwd_cb(conn->ssl, tls_passwd_cb);
+	SSL_set_default_passwd_cb_userdata(conn->ssl, passwd);
 	SSL_CTX_set_default_passwd_cb(ssl_ctx, tls_passwd_cb);
 	SSL_CTX_set_default_passwd_cb_userdata(ssl_ctx, passwd);
 
@@ -2869,6 +2871,7 @@
 		return -1;
 	}
 	ERR_clear_error();
+	SSL_set_default_passwd_cb(conn->ssl, NULL);
 	SSL_CTX_set_default_passwd_cb(ssl_ctx, NULL);
 	os_free(passwd);
 
