--- PyQt-x11-gpl-4.8.2/pyuic/uic/uiparser.py~	2010-12-27 14:01:31.500057960 -0800
+++ PyQt-x11-gpl-4.8.2/pyuic/uic/uiparser.py	2010-12-27 14:01:44.460058033 -0800
@@ -158,7 +158,7 @@
         # Ignore the parent if it is a container.
         parent = self.stack.topwidget
         if isinstance(parent, (QtGui.QDockWidget, QtGui.QMdiArea,
-                               QtGui.QScrollArea, QtGui.StackedWidget,
+                               QtGui.QScrollArea, QtGui.QStackedWidget,
                                QtGui.QToolBox, QtGui.QTabWidget,
                                QtGui.QWizard)):
             parent = None
