--- scripts/e_gen_menu~	2006-03-29 22:09:17.000000000 +0200
+++ scripts/e_gen_menu	2006-08-15 15:11:18.000000000 +0200
@@ -25,7 +25,7 @@
 ##############################################################################
 
 # Likely  prefixes
-$Prefixes = "/usr/local:/usr:/opt:/opt/kde:$ENV{'KDEDIR'}";
+$Prefixes = "/usr/local:/usr:/opt:/opt/kde:/opt/gnome";
 $Prefixes = "$Prefixes:/opt/kde3:/opt/gnome";	# SUSE
 $Prefixes = RemoveDuplcates($Prefixes);
 
