diff -u -r libgda-5.2.4/configure.ac libgda-5.2.4-vala/configure.ac
--- libgda-5.2.4/configure.ac	2015-06-13 11:07:23.000000000 +0200
+++ libgda-5.2.4-vala/configure.ac	2015-06-14 19:11:36.700984704 +0200
@@ -547,13 +547,13 @@
 vala_api="No Vala bindings will be generated"
 if test "x$enable_vala" = "xyes"
 then
-	AM_PROG_VALAC([0.26.0])
+	AM_PROG_VALAC([0.30.0])
 	if test "x$VALA_API_VERSION" = "x"
 	then
-		VAPIGEN_CHECK([0.26.0],[0.26],,no)
-		vala_api="0.26 (Default)"
+		VAPIGEN_CHECK([0.30.0],[0.30],,no)
+		vala_api="0.30 (Default)"
 	else
-		VAPIGEN_CHECK([0.26.0],$VALA_API_VERSION,,no)
+		VAPIGEN_CHECK([0.30.0],$VALA_API_VERSION,,no)
 		vala_api=$VALA_API_VERSION
 	fi
 else
