--- QScintilla-gpl-2.7.1/Python/configure.py~	2013-03-07 08:09:00.678704222 +0000
+++ QScintilla-gpl-2.7.1/Python/configure.py	2013-03-07 08:11:48.860880852 +0000
@@ -29,6 +29,8 @@
 import glob
 import optparse
 
+src_dir = os.path.dirname(os.path.abspath(__file__))
+
 try:
     import sysconfig
 except ImportError:
@@ -455,6 +457,9 @@
 
     # Get the QScintilla version string.
     sciversstr = read_define(sciglobal, 'QSCINTILLA_VERSION_STR')
+
+    return # Debian: do not check for the installed version, we're good this way.
+
     if sciversstr is None:
         error(
                 "The QScintilla version number could not be determined by "
