Index: Makefile.in
===================================================================
RCS file: /sqlite/sqlite/Makefile.in,v
retrieving revision 1.186
diff -u -3 -p -r1.186 Makefile.in
--- Makefile.in	23 Nov 2007 15:12:44 -0000	1.186
+++ Makefile.in	23 Nov 2007 19:27:56 -0000
@@ -114,6 +114,23 @@ LTINSTALL = $(LIBTOOL) --mode=install $(
 # nawk compatible awk.
 NAWK = @AWK@
 
+# Need these defined unconditionally, as non-fts builds with
+# fts object files without using -DSQLITE_ENABLE_FTS[123]
+# will not work without it.
+#
+TCC += -DSQLITE_CORE
+TCC += -DSQLITE_ENABLE_BROKEN_FTS1
+TCC += -DSQLITE_ENABLE_BROKEN_FTS2
+
+# uncomment to build fts1 as part of library
+# TCC += -DSQLITE_ENABLE_FTS1=1
+
+# uncomment to build fts2 as part of library
+# TCC += -DSQLITE_ENABLE_FTS2=1
+
+# uncomment to build fts3 as part of library
+# TCC += -DSQLITE_ENABLE_FTS3=1
+
 # You should not have to change anything below this line
 ###############################################################################
 TCC += -DSQLITE_OMIT_LOAD_EXTENSION=1
@@ -132,6 +149,17 @@ LIBOBJ = alter.lo analyze.lo attach.lo a
          vdbe.lo vdbeapi.lo vdbeaux.lo vdbeblob.lo vdbefifo.lo vdbemem.lo \
          where.lo utf.lo legacy.lo vtab.lo
 
+# Object files for FTS
+#
+LIBOBJ += \
+  fts1.lo fts1_hash.lo fts1_porter.lo fts1_tokenizer1.lo
+
+LIBOBJ += \
+  fts2.lo fts2_hash.lo fts2_porter.lo fts2_tokenizer.lo fts2_tokenizer1.lo
+
+LIBOBJ += \
+  fts3.lo fts3_hash.lo fts3_porter.lo fts3_tokenizer.lo fts3_tokenizer1.lo
+
 # All of the source code files.
 #
 SRC = \
@@ -209,6 +237,23 @@ SRC += \
   $(TOP)/ext/fts1/fts1_tokenizer.h \
   $(TOP)/ext/fts1/fts1_tokenizer1.c
 
+SRC += \
+  $(TOP)/ext/fts2/fts2.c \
+  $(TOP)/ext/fts2/fts2.h \
+  $(TOP)/ext/fts2/fts2_hash.c \
+  $(TOP)/ext/fts2/fts2_hash.h \
+  $(TOP)/ext/fts2/fts2_porter.c \
+  $(TOP)/ext/fts2/fts2_tokenizer.h \
+  $(TOP)/ext/fts2/fts2_tokenizer1.c
+
+SRC += \
+  $(TOP)/ext/fts3/fts3.c \
+  $(TOP)/ext/fts3/fts3.h \
+  $(TOP)/ext/fts3/fts3_hash.c \
+  $(TOP)/ext/fts3/fts3_hash.h \
+  $(TOP)/ext/fts3/fts3_porter.c \
+  $(TOP)/ext/fts3/fts3_tokenizer.h \
+  $(TOP)/ext/fts3/fts3_tokenizer1.c
 
 # Source code to the test files.
 #
@@ -285,6 +330,16 @@ HDR += \
   $(TOP)/ext/fts1/fts1_hash.h \
   $(TOP)/ext/fts1/fts1_tokenizer.h
 
+HDR += \
+  $(TOP)/ext/fts2/fts2.h \
+  $(TOP)/ext/fts2/fts2_hash.h \
+  $(TOP)/ext/fts2/fts2_tokenizer.h
+
+HDR += \
+  $(TOP)/ext/fts3/fts3.h \
+  $(TOP)/ext/fts3/fts3_hash.h \
+  $(TOP)/ext/fts3/fts3_tokenizer.h
+
 # Header files used by the VDBE submodule
 #
 VDBEHDR = \
@@ -616,3 +671,48 @@ sqlite3.def: $(REAL_LIBOBJ)
 sqlite3.dll: $(REAL_LIBOBJ) sqlite3.def
 	$(TCC) -shared -o sqlite3.dll sqlite3.def \
 		-Wl,"--strip-all" $(REAL_LIBOBJ)
+
+#
+# FTS
+#
+fts1.lo: $(TOP)/ext/fts1/fts1.c $(HDR)
+	$(LTCOMPILE) -c $(TOP)/ext/fts1/fts1.c
+
+fts1_hash.lo: $(TOP)/ext/fts1/fts1_hash.c $(HDR)
+	$(LTCOMPILE) -c $(TOP)/ext/fts1/fts1_hash.c
+
+fts1_porter.lo: $(TOP)/ext/fts1/fts1_porter.c $(HDR)
+	$(LTCOMPILE) -c $(TOP)/ext/fts1/fts1_porter.c
+
+fts1_tokenizer1.lo: $(TOP)/ext/fts1/fts1_tokenizer1.c $(HDR)
+	$(LTCOMPILE) -c $(TOP)/ext/fts1/fts1_tokenizer1.c
+
+fts2.lo: $(TOP)/ext/fts2/fts2.c $(HDR)
+	$(LTCOMPILE) -c $(TOP)/ext/fts2/fts2.c
+
+fts2_hash.lo: $(TOP)/ext/fts2/fts2_hash.c $(HDR)
+	$(LTCOMPILE) -c $(TOP)/ext/fts2/fts2_hash.c
+
+fts2_porter.lo: $(TOP)/ext/fts2/fts2_porter.c $(HDR)
+	$(LTCOMPILE) -c $(TOP)/ext/fts2/fts2_porter.c
+
+fts2_tokenizer.lo: $(TOP)/ext/fts2/fts2_tokenizer.c $(HDR)
+	$(LTCOMPILE) -c $(TOP)/ext/fts2/fts2_tokenizer.c
+
+fts2_tokenizer1.lo: $(TOP)/ext/fts2/fts2_tokenizer1.c $(HDR)
+	$(LTCOMPILE) -c $(TOP)/ext/fts2/fts2_tokenizer1.c
+
+fts3.lo: $(TOP)/ext/fts3/fts3.c $(HDR)
+	$(LTCOMPILE) -c $(TOP)/ext/fts3/fts3.c
+
+fts3_hash.lo: $(TOP)/ext/fts3/fts3_hash.c $(HDR)
+	$(LTCOMPILE) -c $(TOP)/ext/fts3/fts3_hash.c
+
+fts3_porter.lo: $(TOP)/ext/fts3/fts3_porter.c $(HDR)
+	$(LTCOMPILE) -c $(TOP)/ext/fts3/fts3_porter.c
+
+fts3_tokenizer.lo: $(TOP)/ext/fts3/fts3_tokenizer.c $(HDR)
+	$(LTCOMPILE) -c $(TOP)/ext/fts3/fts3_tokenizer.c
+
+fts3_tokenizer1.lo: $(TOP)/ext/fts3/fts3_tokenizer1.c $(HDR)
+	$(LTCOMPILE) -c $(TOP)/ext/fts3/fts3_tokenizer1.c
