Index: ext/standard/php_crypt_r.c
===================================================================
--- ext/standard/php_crypt_r.c	(Revision 315217)
+++ ext/standard/php_crypt_r.c	(Revision 315218)
@@ -382,7 +382,7 @@
 	/* Now make the output string */
 	memcpy(passwd, MD5_MAGIC, MD5_MAGIC_LEN);
 	strlcpy(passwd + MD5_MAGIC_LEN, sp, sl + 1);
-	strlcat(passwd, "$", 1);
+	strcat(passwd, "$");
 
 	PHP_MD5Final(final, &ctx);
 
