--- ext/mysqlnd/mysqlnd.c	2009-11-19 14:02:01.934775092 +0100
+++ ext/mysqlnd/mysqlnd.c	2009-11-19 14:08:04.598072581 +0100
@@ -547,7 +547,7 @@
 #ifndef PHP_WIN32
 	if (host_len == sizeof("localhost") - 1 && !strncasecmp(host, "localhost", host_len)) {
 		if (!socket) {
-			socket = "/tmp/mysql.sock";
+			socket = "/var/run/mysqld/mysqld.sock";
 		}
 		transport_len = spprintf(&transport, 0, "unix://%s", socket);
 		unix_socket = TRUE;
