--- capture_freaklabs_zigbee/Makefile.in	2019-10-10 10:49:00.494256269 +0200
+++ capture_freaklabs_zigbee/Makefile.in	2019-10-10 10:52:49.374152190 +0200
@@ -10,7 +10,7 @@
 	$(PYTHON) ./setup.py bdist

 install:
-	$(PYTHON) ./setup.py install
+	$(PYTHON) ./setup.py install --root="$(DESTDIR)" -O1
 	# $(INSTALL) -o $(INSTUSR) -g $(INSTGRP) $(MONITOR_BIN) $(BIN)/$(MONITOR_BIN)

 clean:
--- capture_sdr_rtl433/Makefile.in	2019-10-10 10:49:00.507590313 +0200
+++ capture_sdr_rtl433/Makefile.in	2019-10-10 10:52:28.069331801 +0200
@@ -11,7 +11,7 @@
 	$(PYTHON) ./setup.py bdist

 install:
-	$(PYTHON) ./setup.py install
+	$(PYTHON) ./setup.py install --root="$(DESTDIR)" -O1

 clean:
 	@-$(PYTHON) ./setup.py clean
--- capture_sdr_rtladsb/Makefile.in	2019-10-10 10:49:00.510923825 +0200
+++ capture_sdr_rtladsb/Makefile.in	2019-10-10 10:52:32.542260533 +0200
@@ -11,7 +11,7 @@
 	$(PYTHON) ./setup.py bdist

 install:
-	$(PYTHON) ./setup.py install
+	$(PYTHON) ./setup.py install --root="$(DESTDIR)" -O1
 	# These are now part of the setup.py install
 	# $(INSTALL) -o $(INSTUSR) -g $(INSTGRP) aircraft_db.csv $(ETC)/kismet_aircraft_db.csv
 	# $(INSTALL) -o $(INSTUSR) -g $(INSTGRP) $(MONITOR_BIN) $(BIN)/$(MONITOR_BIN)
--- capture_sdr_rtlamr/Makefile.in	2019-10-10 10:49:00.514257335 +0200
+++ capture_sdr_rtlamr/Makefile.in	2019-10-10 10:52:35.978622527 +0200
@@ -11,7 +11,7 @@
 	$(PYTHON) ./setup.py bdist

 install:
-	$(PYTHON) ./setup.py install
+	$(PYTHON) ./setup.py install --root="$(DESTDIR)" -O1
 
 protobuf:
 	$(PROTOCBIN) -I ../protobuf_definitions --python_out=./KismetCaptureRtlamr ../protobuf_definitions/*.proto

