diff -up freenx-server-0.7.3/nxsetup~ freenx-server-0.7.3/nxsetup
--- freenx-server-0.7.3/nxsetup~	2011-12-30 23:04:53.515194711 +0200
+++ freenx-server-0.7.3/nxsetup	2011-12-30 23:05:55.499954085 +0200
@@ -366,11 +366,10 @@ EOF
 
 	echo "done"
 
-	if [ -d "$CUPS_BACKEND" ]
+	if [ -d "$CUPS_BACKEND" ] && [ ! -f "$CUPS_IPP_BACKEND" ]
 	then
 		echo -n "Setting up cups nxipp backend ..."
-		cp -af "$CUPS_BACKEND/ipp" "$CUPS_IPP_BACKEND"
-		chmod 755 "$CUPS_IPP_BACKEND"
+		ln -sf "$CUPS_BACKEND/ipp" "$CUPS_IPP_BACKEND"
 		echo "done"
 	fi
 }
