--- nxdialog.old	2007-10-15 00:25:14.000000000 +0200
+++ nxdialog	2007-10-31 18:38:57.000000000 +0100
@@ -26,6 +26,7 @@
 	done
 }
 
+PARAMS=( "$@" )
 INPUTS=$(requote "$@")	#JJK: Save input parameter line...
 
 TEMP=`getopt -a -o d: --long local,noautokill,dialog:,caption:,message:,display:,printer:,parent: -n $(basename $0) -- "$@"`
@@ -71,10 +72,13 @@
 # client does not like large databases like used when ENABLE_FOOMATIC=1.
 #
 # This seems to be because the used sorting algorithm scales in O(n^2).
+#
+# This is now fixed in NXClient 3.0.0, but still people sometimes use
+# older clients.
 
 NXCLIENT="/usr/NX/bin/nxclient"
 [ -x "$NXCLIENT" -a "$DIALOG_TYPE" != "printer" -a "$(file -bi $NXCLIENT)" != 'application/x-shellscript' ] \
-	&& exec ${NXCLIENT} "$@"
+	&& exec ${NXCLIENT} "${PARAMS[@]}"
 
 # FIXME: This should be COMMAND_XDIALOG, ...
 if [ -x /usr/bin/Xdialog ] 
@@ -444,3 +448,4 @@
 	[ $RC -eq 2 ] && kill -TERM $DIALOG_PARENT
 	[ $RC -eq 3 ] && kill -HUP $DIALOG_PARENT
 exit 0
+ 
