--- nxserver~	2007-10-15 00:25:14.000000000 +0200
+++ nxserver	2007-10-22 07:17:46.000000000 +0200
@@ -267,7 +267,7 @@
 		egrep -q "^userName=$1$" $i && let SESSION_COUNT_USER=$SESSION_COUNT_USER+1
 	done
 	
-	if [ "$ENABLE_DESKTOP_SHARING" = "1" ]
+	if [ "$4" = "vnc" -a "$ENABLE_DESKTOP_SHARING" = "1" ]
 	then
 		export DESKTOP_SHARING_IDS=""
 		for i in $(LC_ALL=C netstat -ln --protocol=unix | egrep 'X11-unix/X[0-9]$' | sed 's/.*X\(.*\)/\1/g')
