--- src/celestia/Makefile.am	2011-06-07 19:15:51.502000147 +0200
+++ src/celestia/Makefile.am.new	2011-06-07 19:15:47.208000302 +0200
@@ -15,6 +15,7 @@
 if ENABLE_GTK
 SUBDIRS += gtk
 celestiaGTKLIBS = $(GTK_LIBS) gtk/libgtkgui.a
+LIBS += $(GTK_LIBS)
 endif
 
 COMMONSOURCES = \

Index: src/celengine/frametree.h
===================================================================
--- src/celengine/frametree.h	(revision 5163)
+++ src/celengine/frametree.h	(arbetskopia)
@@ -14,6 +14,7 @@
 #define _CELENGINE_FRAMETREE_H_
 
 #include <vector>
+#include <cstddef>
 
 class Star;
 class Body;
