From 0787a8972be2b95fac26c502a82427299e7e48c0 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C5=82=20G=C3=B3rny?= <mgorny@gentoo.org>
Date: Sat, 21 Oct 2017 00:14:32 +0200
Subject: Support <sys/sysmacros.h> include required for glibc-2.25+ (Fixes
 #13940)

---
 configure.ac.in        | 1 +
 panel-plugin/devperf.c | 3 +++
 2 files changed, 4 insertions(+)

diff --git a/configure.ac b/configure.ac
index 5458dba..fd2d3eb 100644
--- a/configure.ac
+++ b/configure.ac
@@ -46,6 +46,7 @@ dnl **********************************
 dnl *** Check for standard headers ***
 dnl **********************************
 AC_HEADER_STDC()
+AC_CHECK_HEADERS([sys/sysmacros.h])
 AC_CHECK_LIBM
 AC_SUBST(LIBM)
 
diff --git a/panel-plugin/devperf.c b/panel-plugin/devperf.c
index 175fd44..29bcb87 100644
--- a/panel-plugin/devperf.c
+++ b/panel-plugin/devperf.c
@@ -35,6 +35,9 @@
 #include <sys/time.h>
 /* for major() and minor() */
 #define _BSD_SOURCE
+#ifdef HAVE_SYS_SYSMACROS_H
+#include <sys/sysmacros.h>
+#endif
 #include <sys/types.h>
 
 
-- 
cgit v1.2.1

