--- src/ocaml-4.00.1/configure	2012-06-26 08:33:50.000000000 -0700
+++ p/src/ocaml-4.00.1/configure	2013-01-05 17:06:54.877900143 -0800
@@ -804,6 +804,9 @@
 echo "ARCMD=ar" >> Makefile
 
 
+bytecccompopts="$bytecccompopts -DUSE_INTERP_RESULT"
+nativecccompopts="$nativecccompopts -DUSE_INTERP_RESULT"
+
 # Do #! scripts work?
 
 if (SHELL=/bin/sh; export SHELL; (./sharpbang || ./sharpbang2) >/dev/null); then
@@ -1450,6 +1453,7 @@
   if test -n "$tcl_version" && test "x$tcl_version" != "xnone"; then
     echo "tcl.h and tk.h version $tcl_version found with \"$tk_defs\"."
     case $tcl_version in
+    8.6) tclmaj=8 tclmin=6 tkmaj=8 tkmin=6 ;;
     8.5) tclmaj=8 tclmin=5 tkmaj=8 tkmin=5 ;;
     8.4) tclmaj=8 tclmin=4 tkmaj=8 tkmin=4 ;;
     8.3) tclmaj=8 tclmin=3 tkmaj=8 tkmin=3 ;;
