diff -ruN libavg-0.6.0.orig/src/player/FFMpegDecoder.cpp libavg-0.6.0/src/player/FFMpegDecoder.cpp
--- libavg-0.6.0.orig/src/player/FFMpegDecoder.cpp	2007-02-03 21:12:07.000000000 +0100
+++ libavg-0.6.0/src/player/FFMpegDecoder.cpp	2007-02-03 21:12:18.000000000 +0100
@@ -100,7 +100,6 @@
 
     AVG_TRACE(Logger::PROFILE, "Opening " << sFilename);
     memset(&params, 0, sizeof(params));
-    params.image_format = 0;
 
     err = av_open_input_file(&m_pFormatContext, sFilename.c_str(), 
             0, 0, &params);
