diff -r 4f76df2f1438 launcher/launchers.in
--- a/launcher/launchers.in	Fri Aug 15 09:49:16 2014 +0200
+++ b/launcher/launchers.in	Sat Aug 30 18:01:25 2014 +0200
@@ -7,7 +7,6 @@
 BINARY_LOCATION=@BIN_LOCATION@
 SPLASH_LOCATION=@JAVAWS_SPLASH_LOCATION@
 PROGRAM_NAME=@PROGRAM_NAME@
-CP=@JRE@/lib/rt.jar
 
 CONFIG_HOME=$XDG_CONFIG_HOME
 if [ "x$CONFIG_HOME" = "x" ] ; then
@@ -25,11 +24,10 @@
   CUSTOM_JRE=`grep "$CUSTOM_JRE_REGEX" /etc/.java/.deploy/deployment.properties 2>/dev/null |  sed "s/$CUSTOM_JRE_REGEX//g"`
 fi;
 if [ "x$CUSTOM_JRE" != "x" ] ; then
-  if [ -e  "$CUSTOM_JRE" -a -e "$CUSTOM_JRE/bin/java" -a -e "$CUSTOM_JRE/lib/rt.jar" ] ; then
+  if [ -e  "$CUSTOM_JRE" -a -e "$CUSTOM_JRE/bin/java" ] ; then
     JAVA=$CUSTOM_JRE/bin/java
-    CP=$CUSTOM_JRE/lib/rt.jar
   else
-    echo "Your custom JRE $CUSTOM_JRE read from deployment.properties under key $PROPERTY_NAME as $CUSTOM_JRE is not valid. Using default ($JAVA, $CP) in attempt to start. Please fix this."
+    echo "Your custom JRE $CUSTOM_JRE read from deployment.properties under key $PROPERTY_NAME as $CUSTOM_JRE is not valid. Using default ($JAVA) in attempt to start. Please fix this."
   fi
 fi;
 
@@ -78,10 +76,6 @@
   i=$((i+1))
   k=$((k+1))
 done
-COMMAND[k]="-classpath"
-k=$((k+1))
-COMMAND[k]="${CP}"
-k=$((k+1))
 COMMAND[k]="-Dicedtea-web.bin.name=${PROGRAM_NAME}"
 k=$((k+1))
 COMMAND[k]="-Dicedtea-web.bin.location=${BINARY_LOCATION}"
