diff -ur facile-1.1.old/src/facile.mli facile-1.1/src/facile.mli
--- facile-1.1.old/src/facile.mli	2004-09-08 10:51:02.000000000 +0100
+++ facile-1.1/src/facile.mli	2012-06-10 21:06:41.553916954 +0100
@@ -127,7 +127,7 @@
       module Hashtbl :
 	  sig
 	    type ('a, 'b) t
-	    val create : int -> ('a, 'b) t
+	    val create : ?random:bool -> int -> ('a, 'b) t
 	    val get : ('a, 'b) t -> ('a, 'b) Hashtbl.t
 	    val add : ('a, 'b) t -> 'a -> 'b -> unit
 	    val find : ('a, 'b) t -> 'a -> 'b
Binary files facile-1.1.old/src/fcl_data.cmi and facile-1.1/src/fcl_data.cmi differ
diff -ur facile-1.1.old/src/fcl_data.mli facile-1.1/src/fcl_data.mli
--- facile-1.1.old/src/fcl_data.mli	2004-09-08 10:51:02.000000000 +0100
+++ facile-1.1/src/fcl_data.mli	2012-06-10 21:06:25.841590828 +0100
@@ -23,7 +23,7 @@
 
 module Hashtbl : sig
   type ('a, 'b) t
-  val create : int -> ('a, 'b) t
+  val create : ?random:bool -> int -> ('a, 'b) t
   val get : ('a, 'b) t -> ('a, 'b) Hashtbl.t
   val add : ('a, 'b) t -> 'a -> 'b -> unit
   val find : ('a, 'b) t -> 'a -> 'b
