--- rartypes.hpp
+++ rartypes.hpp
@@ -1,14 +1,15 @@
 #ifndef _RAR_TYPES_
 #define _RAR_TYPES_
+#include<stdint.h>
 
 typedef unsigned char    byte;   // unsigned 8 bits
-typedef unsigned short   ushort; // preferably 16 bits, but can be more
+typedef uint16_t         ushort; //preferably 16 bits, but can be more
 typedef unsigned int     uint;   // 32 bits or more
 
 #define PRESENT_INT32 // undefine if signed 32 bits is not available
 
-typedef unsigned int     uint32; // 32 bits exactly
-typedef   signed int     int32;  // signed 32 bits exactly
+typedef uint32_t         uint32; //32 bits exactly
+typedef int32_t          int32; //signed 32 bits exactly
 
 // If compiler does not support 64 bit variables, we can define
 // uint64 and int64 as 32 bit, but it will limit the maximum processed
