From bdeff0efd3342b36e56830fd208103c4453ad0d4 Mon Sep 17 00:00:00 2001
From: Lionel Le Folgoc <mrpouit@ubuntu.com>
Date: Mon, 25 Jan 2010 21:36:32 +0000
Subject: Port to libxklavier 5.0 API.

---
diff --git a/configure.ac.in b/configure.ac.in
index 59e93b2..acd432e 100644
--- a/configure.ac
+++ b/configure.ac
@@ -137,6 +137,8 @@ XDT_CHECK_OPTIONAL_PACKAGE([LIBXKLAVIER], [libxklavier], [0.3.0],
                            [libxklavier], [Keyboard layout selection])
 XDT_CHECK_OPTIONAL_PACKAGE([LIBXKLAVIER4], [libxklavier], [4.0],
                            [libxklavier], [Keyboard layout selection])
+XDT_CHECK_OPTIONAL_PACKAGE([LIBXKLAVIER5], [libxklavier], [5.0],
+                           [libxklavier], [Keyboard layout selection])
 
 dnl ************************************
 dnl *** Xsettings daemon is optional ***
diff --git a/dialogs/keyboard-settings/xfce-keyboard-settings.c b/dialogs/keyboard-settings/xfce-keyboard-settings.c
index 631dc69..5ff7ae3 100644
--- a/dialogs/keyboard-settings/xfce-keyboard-settings.c
+++ b/dialogs/keyboard-settings/xfce-keyboard-settings.c
@@ -395,7 +395,11 @@ xfce_keyboard_settings_finalize (GObject *object)
 
 #ifdef HAVE_LIBXKLAVIER
   /* Stop xklavier engine */
+#ifdef HAVE_LIBXKLAVIER5
+  xkl_engine_stop_listen (settings->priv->xkl_engine, XKLL_TRACK_KEYBOARD_STATE);
+#else
   xkl_engine_stop_listen (settings->priv->xkl_engine);
+#endif /* HAVE_LIBXKLAVIER5 */
 #endif /* HAVE_LIBXKLAVIER */
 
   g_object_unref (G_OBJECT (settings->priv->provider));
--
cgit v0.8.2.1
