--- wrlib/load_gif.c.orig	2013-09-12 22:43:54.000000000 +0100
+++ wrlib/load_gif.c	2014-05-16 10:10:04.000000000 +0100
@@ -81,7 +81,7 @@ RImage *RLoadGIF(const char *file, int i
 	}
 
 	if (gif->SWidth < 1 || gif->SHeight < 1) {
-		DGifCloseFile(gif);
+		DGifCloseFile(gif, NULL);
 		RErrorCode = RERR_BADIMAGEFILE;
 		return NULL;
 	}
@@ -216,7 +216,7 @@ RImage *RLoadGIF(const char *file, int i
 		free(buffer);
 
 	if (gif)
-		DGifCloseFile(gif);
+		DGifCloseFile(gif, NULL);
 
 	return image;
 }
