Index: dhcp.c
==================================================================
--- dhcp.c
+++ dhcp.c
@@ -2701,11 +2701,10 @@
 	if (state == NULL)
 		goto eexit;
 	ifp->options = calloc(1, sizeof(*ifp->options));
 	if (ifp->options == NULL)
 		goto eexit;
-	strlcpy(ifp->name, ifname, sizeof(ifp->name));
 	snprintf(state->leasefile, sizeof(state->leasefile),
 	    LEASEFILE, ifp->name);
 	strlcpy(ifp->options->script, SCRIPT, sizeof(ifp->options->script));
 	state->new = read_lease(ifp);
 	if (state->new == NULL && errno == ENOENT) {

