--- driver/XScreenSaver.ad.in-orig	2009-10-17 02:19:06.000000000 -0400
+++ driver/XScreenSaver.ad.in	2009-10-17 02:21:16.000000000 -0400
@@ -347,7 +347,8 @@
 -				lcdscrub -root				    \n\
 @GL_KLUDGE@ GL: 				photopile -root				    \n\
 @GL_KLUDGE@ GL: 				skytentacles -root			    \n\
-@GL_KLUDGE@ GL: 				rubikblocks -root			    \n
+@GL_KLUDGE@ GL: 				rubikblocks -root			    \n\
+				electricsheep --root 1				    \n
 
 
 
