diff --git i/server/meson.build w/server/meson.build
index 6c78742..c1b482e 100644
--- i/server/meson.build
+++ w/server/meson.build
@@ -8,7 +8,8 @@ renderer_service = executable(
         renderer_dep,
     ],
     install: true,
-    install_dir: join_paths(get_option('prefix'), get_option('libexecdir'))
+    install_dir: join_paths(get_option('prefix'), get_option('libexecdir')),
+    install_rpath: join_paths(get_option('prefix'), get_option('libdir'), 'dleyna-renderer')
 )
 
 configure_file(
