--- branches/stable/l10n-kde4/nl/docs/kdenetwork/kppp/global-settings.docbook	2011/06/30 08:41:03	1238851
+++ branches/stable/l10n-kde4/nl/docs/kdenetwork/kppp/global-settings.docbook	2011/07/02 19:24:33	1239106
@@ -3,7 +3,7 @@
 >Globale instellingen van &kppp;</title>
 
 <para
->De hier gemaakte wijzigingen beïnvloeden alle accounts die u in &kpp; hebt ingesteld</para>
+>De hier gemaakte wijzigingen beïnvloeden alle accounts die u in &kppp; hebt ingesteld</para>
 
 <sect1 id="global-accounts">
 <title
