--- src/third_party/WebKit/WebCore/platform/image-encoders/skia/orig/PNGImageEncoder.cpp	2009-11-19 13:39:07.000000000 -0800
+++ src/third_party/WebKit/WebCore/platform/image-encoders/skia/PNGImageEncoder.cpp	2009-11-19 13:39:25.000000000 -0800
@@ -130,9 +130,9 @@
     }
 
     png_struct* pngPtr = png_create_write_struct(PNG_LIBPNG_VER_STRING,
-                                                 png_voidp_NULL,
-                                                 png_error_ptr_NULL,
-                                                 png_error_ptr_NULL);
+                                                 NULL,
+                                                 NULL,
+                                                 NULL);
     if (!pngPtr)
         return false;
 
