From 1f386a0e357a8916713fe26acd96206be2159157 Mon Sep 17 00:00:00 2001
From: Dave Reisner <dreisner@archlinux.org>
Date: Wed, 1 Feb 2012 20:09:27 -0500
Subject: [PATCH 5/8] continue after module insert fail

---
 tools/kmod-modprobe.c |    2 --
 1 files changed, 0 insertions(+), 2 deletions(-)

diff --git a/tools/kmod-modprobe.c b/tools/kmod-modprobe.c
index bd991a5..e70bf3d 100644
--- a/tools/kmod-modprobe.c
+++ b/tools/kmod-modprobe.c
@@ -825,8 +825,6 @@ static int insmod_alias(struct kmod_ctx *ctx, const char *alias,
 			array_free_array(&recursion);
 		}
 		kmod_module_unref(mod);
-		if (err < 0)
-			break;
 	}
 
 	kmod_module_unref_list(list);
-- 
1.7.9

