diff -aur old/Makefile new/Makefile
--- old/Makefile	2012-02-18 21:55:09.000000000 +1100
+++ new/Makefile	2012-02-20 18:28:08.716130584 +1100
@@ -135,8 +135,8 @@
 
 systemd:
 ifdef SYSTEMDSYSTEMUNITDIR
-	sed 's|@sbindir@|$(BINDIR)|' init/bitlbee.service.in > init/bitlbee.service
-	sed 's|@sbindir@|$(BINDIR)|' init/bitlbee@.service.in > init/bitlbee@.service
+	sed 's|@sbindir@|$(SBINDIR)|' init/bitlbee.service.in > init/bitlbee.service
+	sed 's|@sbindir@|$(SBINDIR)|' init/bitlbee@.service.in > init/bitlbee@.service
 endif
 
 install-systemd:
