diff -u -r anjuta-3.8.0/libanjuta/anjuta-token.c anjuta-3.8.0-fix/libanjuta/anjuta-token.c
--- anjuta-3.8.0/libanjuta/anjuta-token.c	2013-01-14 22:45:09.000000000 +0100
+++ anjuta-3.8.0-fix/libanjuta/anjuta-token.c	2013-03-31 13:56:11.722332230 +0200
@@ -329,7 +329,7 @@
 	if (string == NULL)
 	{
 		/* Value doesn't contain a newline */
-		fprintf (stderr, "(%lu)", length);
+		fprintf (stderr, "(%lu)", (long unsigned)length);
 	}
 	else
 	{
diff -u -r anjuta-3.8.0/plugins/dir-project/dir-project.c anjuta-3.8.0-fix/plugins/dir-project/dir-project.c
--- anjuta-3.8.0/plugins/dir-project/dir-project.c	2012-11-19 00:34:57.000000000 +0100
+++ anjuta-3.8.0-fix/plugins/dir-project/dir-project.c	2013-03-31 13:55:35.922610895 +0200
@@ -268,7 +268,7 @@
 			{
 				ptr++;
 			} while  (*ptr == '?');
-			g_string_append_printf (regex, "(.{%d})", ptr - next);
+			g_string_append_printf (regex, "(.{%ld})", (long)(ptr - next));
 		}
 		else if (*ptr == '\\')
 		{
