diff -ruN gcin-1.4.6/configure gcin-1.4.6-new/configure
--- gcin-1.4.6/configure	2009-12-17 17:29:46.000000000 +0800
+++ gcin-1.4.6-new/configure	2009-12-28 16:46:25.000000000 +0800
@@ -164,7 +164,7 @@
 
 
 QT4_IM='N'
-QT4_IM_DIR=/usr/$LIB/qt4/plugins/inputmethods
+QT4_IM_DIR=/usr/$LIB/qt/plugins/inputmethods
 echo "testing if you have qt4 support"
 if [ -d  $QT4_IM_DIR ]; then
   QT4_IM='Y'
diff -ruN gcin-1.4.6/qt4-im/Makefile gcin-1.4.6-new/qt4-im/Makefile
--- gcin-1.4.6/qt4-im/Makefile	2009-12-10 07:09:17.000000000 +0800
+++ gcin-1.4.6-new/qt4-im/Makefile	2009-12-28 16:47:59.000000000 +0800
@@ -1,5 +1,5 @@
 include ../config.mak
-QT=qt4
+QT=qt
 QTIM=$(QT)/plugins/inputmethods
 IMMODULES=$(libdir)/$(QTIM)
 
@@ -15,7 +15,7 @@
 	$(CXX) -E $(CFLAGS) $(INCS) $< > $@
 
 moc_gcin-qt.cpp:	gcin-qt.h
-	/usr/lib/$(QT)/bin/moc $< -o moc_gcin-qt.cpp
+	/usr/bin/moc $< -o moc_gcin-qt.cpp
 
 im-gcin.so: $(OBJS)
 	export LD_RUN_PATH=$(gcinlibdir) ;\
