diff -aur old/configure new/configure
--- old/configure	2011-12-21 00:48:03.000000000 +1100
+++ new/configure	2011-12-25 04:06:14.424317251 +1100
@@ -188,7 +188,7 @@
 
 
 QT4_IM='N'
-QT4_IM_DIR=/usr/$LIB/qt4/plugins/inputmethods
+QT4_IM_DIR=/usr/$LIB/qt/plugins/inputmethods
 echo "testing if you have qt4 support"
 pkg-config --libs QtCore QtGui > /dev/null 2> /dev/null
 if [ $? = 0 ]; then
@@ -245,7 +245,7 @@
   mv gcin.spec gcin.spec.tmp
   grep -v qt4/plugins/inputmethods gcin.spec.tmp > gcin.spec
 else
-  for i in /usr/share/qt4/bin/moc /usr/lib/qt4/bin/moc /usr/lib64/qt4/bin/moc
+  for i in /usr/share/qt4/bin/moc /usr/lib/qt4/bin/moc /usr/lib64/qt4/bin/moc /usr/bin/moc
   do
 	[ -x $i ] && QT4_MOC=$i
   done
diff -aur old/qt4-im/Makefile new/qt4-im/Makefile
--- old/qt4-im/Makefile	2011-12-19 14:52:46.000000000 +1100
+++ new/qt4-im/Makefile	2011-12-25 04:06:14.424317251 +1100
@@ -1,5 +1,5 @@
 include ../config.mak
-QT=qt4
+QT=qt
 QTIM=$(QT)/plugins/inputmethods
 IMMODULES=$(libdir)/$(QTIM)
 
