--- kdeutils-4.4.5/printer-applet/printer-applet.py~	2010-05-28 11:42:32.000000000 +0200
+++ kdeutils-4.4.5/printer-applet/printer-applet.py	2010-07-29 13:17:29.715164501 +0200
@@ -237,10 +237,10 @@
             APPDIR = file.left(file.lastIndexOf("/"))
 
         self.mainWindow = MainWindow()
-        uic.loadUi(APPDIR + "/" + "printer-applet.ui", self.mainWindow)
+        uic.loadUi(unicode(APPDIR + "/" + "printer-applet.ui"), self.mainWindow)
 
         self.printersWindow = PrintersWindow(self)
-        uic.loadUi(APPDIR + "/" + "printer-applet-printers.ui", self.printersWindow)
+        uic.loadUi(unicode(APPDIR + "/" + "printer-applet-printers.ui"), self.printersWindow)
 
         self.sysTray = QSystemTrayIcon(KIcon("printer"), self.mainWindow)
         #self.sysTray.show()
