From 4c64cb990fc1e32e269571b34065d7aa8d92c79e Mon Sep 17 00:00:00 2001
From: Dominique Leuenberger <dimstar@opensuse.org>
Date: Fri, 15 Aug 2014 16:47:27 +0200
Subject: Build against either webkit2gtk-3.0 or webkit2gtk-4.0

The API changes were minor and none affected gthumb.

diff --git a/configure.ac b/configure.ac
index 6f9f03e..da246f9 100644
--- a/configure.ac
+++ b/configure.ac
@@ -508,10 +508,14 @@ AC_ARG_ENABLE([webkit2],
 
 if test x$enable_webkit2 = xyes ; then
 	PKG_CHECK_MODULES(WEBKIT2,
-			  [webkit2gtk-3.0 >= $WEBKIT2_REQUIRED],
+			  [webkit2gtk-4.0 >= $WEBKIT2_REQUIRED],
 			  [AC_DEFINE(HAVE_WEBKIT2, 1, [Have webkit2gtk])
 			   enable_webkit2=yes],
-			  [enable_webkit2=no])
+			  [PKG_CHECK_MODULES(WEBKIT2,
+					     [webkit2gtk-3.0 >= $WEBKIT2_REQUIRED],
+					     [AC_DEFINE(HAVE_WEBKIT2, 1, [Have webkit2gtk])
+					      enable_webkit2=yes],
+					     [enable_webkit2=no])])
 fi
 AC_SUBST(WEBKIT2_CFLAGS)
 AC_SUBST(WEBKIT2_LIBS)
-- 
cgit v0.10.1

