diff --git a/src/libs/clangsupport/CMakeLists.txt b/src/libs/clangsupport/CMakeLists.txt
index dcaa464710..01cf7580b7 100644
--- a/src/libs/clangsupport/CMakeLists.txt
+++ b/src/libs/clangsupport/CMakeLists.txt
@@ -5,8 +5,8 @@ add_qtc_library(ClangSupport
   PUBLIC_DEPENDS Utils Sqlite Qt5::Core Qt5::Network
   PUBLIC_DEFINES
     CLANG_VERSION="${CLANG_VERSION}"
-    CLANG_INCLUDE_DIR="${IDE_LIBEXEC_PATH}/clang/lib/clang/${CLANG_VERSION}/include"
-    CLANG_BINDIR="${IDE_LIBEXEC_PATH}/clang/bin"
+    CLANG_INCLUDE_DIR="/usr/lib/clang/${CLANG_VERSION}/include"
+    CLANG_BINDIR="/usr/bin"
   DEFINES CLANGSUPPORT_BUILD_LIB
   PUBLIC_INCLUDES
     "${CMAKE_CURRENT_LIST_DIR}"
