diff -ruN f-spot-0.4.2.orig/configure.in f-spot-0.4.2/configure.in
--- f-spot-0.4.2.orig/configure.in	2008-03-24 13:52:04.000000000 +0000
+++ f-spot-0.4.2/configure.in	2008-03-24 13:57:38.000000000 +0000
@@ -74,7 +74,7 @@
 AC_SUBST(F_LIBS)
 
 PKG_CHECK_MODULES(GNOME_SHARP, gnome-sharp-2.0 >= 2.8)
-if pkg-config --atleast-version=2.19.90 gnome-sharp; then
+if pkg-config --atleast-version=2.19.90 gnome-sharp-2.0; then
 	PKG_CHECK_MODULES(GTKHTML_SHARP, gtkhtml-sharp-3.14 >= 2.19.90)
 	GTKHTML_SHARP_PKG="-pkg:gtkhtml-sharp-3.14"
 else
diff -ruN f-spot-0.4.2.orig/src/InfoDisplay.cs f-spot-0.4.2/src/InfoDisplay.cs
--- f-spot-0.4.2.orig/src/InfoDisplay.cs	2008-03-24 13:52:04.000000000 +0000
+++ f-spot-0.4.2/src/InfoDisplay.cs	2008-03-24 13:56:16.000000000 +0000
@@ -148,7 +148,7 @@
 
 		private void Update ()
 		{
-			Gtk.HTMLStream stream = this.Begin (null, "text/html; charset=utf-8", Gtk.HTMLBeginFlags.Scroll);
+			Gtk.HTMLStream stream = this.Begin (null, "text/html; charset=utf-8", Gtk.HTMLBeginFlags.KeepScroll);
 			
 			string bg = Color (this.Style.Background (Gtk.StateType.Active));
 			string fg = Color (this.Style.Foreground (Gtk.StateType.Active));
diff -ruN f-spot-0.4.2.orig/src/Makefile.am f-spot-0.4.2/src/Makefile.am
--- f-spot-0.4.2.orig/src/Makefile.am	2008-03-24 13:52:04.000000000 +0000
+++ f-spot-0.4.2/src/Makefile.am	2008-03-24 13:55:36.000000000 +0000
@@ -316,7 +316,8 @@
 	$(GCONF_PKG)				\
 	$(GTKHTML_SHARP_PKG)			\
 	-pkg:glade-sharp-2.0			\
-	-pkg:gnome-vfs-sharp-2.0
+	-pkg:gnome-vfs-sharp-2.0		\
+	-pkg:gnome-sharp-2.0
 
 F_SPOT_DISTRESOURCES =			\
 	$(srcdir)/dces.rdf		\
