--- Makefile.in.orig	2008-06-01 15:48:02.000000000 +0000
+++ Makefile.in	2008-06-01 15:48:50.000000000 +0000
@@ -892,14 +892,14 @@
 install-data-hook: 
 	$(UPDATE_DESKTOP)
 	if [ "x$(DIR_GIO)" != "x" ] ; then \
-		$(install_sh_DATA) $(top_builddir)/gio-sharp/gio/gio-sharp.dll $(libdir)/f-spot/; \
-		$(install_sh_DATA) $(top_builddir)/gio-sharp/gio/gio-sharp.dll.config $(libdir)/f-spot/; \
+		$(install_sh_DATA) $(top_builddir)/gio-sharp/gio/gio-sharp.dll $(DESTDIR)$(libdir)/f-spot/; \
+		$(install_sh_DATA) $(top_builddir)/gio-sharp/gio/gio-sharp.dll.config $(DESTDIR)$(libdir)/f-spot/; \
 	fi
 
 uninstall-hook: 
 	$(UPDATE_DESKTOP)
 	if [ "x$(DIR_GIO)" != "x" ] ; then \
-		rm -f $(libdir)/f-spot/gio-sharp.dll $(libdir)/f-spot/gio-sharp.dll.config; \
+		rm -f $(DESTDIR)$(libdir)/f-spot/gio-sharp.dll $(DESTDIR)$(libdir)/f-spot/gio-sharp.dll.config; \
 	fi
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
