diff -Naur koffice-1.6.3-orig/filters/kword/wordperfect/import/TableStyle.cxx koffice-1.6.3/filters/kword/wordperfect/import/TableStyle.cxx
--- koffice-1.6.3-orig/filters/kword/wordperfect/import/TableStyle.cxx	2008-04-12 21:18:16.000000000 -0400
+++ koffice-1.6.3/filters/kword/wordperfect/import/TableStyle.cxx	2008-04-12 21:25:04.000000000 -0400
@@ -28,6 +28,7 @@
  * Corel Corporation or Corel Corporation Limited."
  */
 #include <math.h>
+#include <cstring>
 #include "FilterInternal.hxx"
 #include "TableStyle.hxx"
 #include "DocumentElement.hxx"
diff -Naur koffice-1.6.3-orig/filters/kword/wordperfect/import/TextRunStyle.cxx koffice-1.6.3/filters/kword/wordperfect/import/TextRunStyle.cxx
--- koffice-1.6.3-orig/filters/kword/wordperfect/import/TextRunStyle.cxx	2008-04-12 21:18:16.000000000 -0400
+++ koffice-1.6.3/filters/kword/wordperfect/import/TextRunStyle.cxx	2008-04-12 21:24:42.000000000 -0400
@@ -28,6 +28,7 @@
 /* "This product is not manufactured, approved, or supported by
  * Corel Corporation or Corel Corporation Limited."
  */
+#include <cstring>
 #include "FilterInternal.hxx"
 #include "TextRunStyle.hxx"
 #include "WriterProperties.hxx"
diff -Naur koffice-1.6.3-orig/kexi/kexidb/drivers/pqxx/pqxxcursor.cpp koffice-1.6.3/kexi/kexidb/drivers/pqxx/pqxxcursor.cpp
--- koffice-1.6.3-orig/kexi/kexidb/drivers/pqxx/pqxxcursor.cpp	2008-04-12 21:18:16.000000000 -0400
+++ koffice-1.6.3/kexi/kexidb/drivers/pqxx/pqxxcursor.cpp	2008-04-12 21:24:04.000000000 -0400
@@ -26,6 +26,7 @@
 
 #include <klocale.h>
 #include <kdebug.h>
+#include <cstdlib>
 
 using namespace KexiDB;
 
diff -Naur koffice-1.6.3-orig/krita/core/kis_perspective_math.cpp koffice-1.6.3/krita/core/kis_perspective_math.cpp
--- koffice-1.6.3-orig/krita/core/kis_perspective_math.cpp	2008-04-12 21:18:16.000000000 -0400
+++ koffice-1.6.3/krita/core/kis_perspective_math.cpp	2008-04-12 21:20:27.000000000 -0400
@@ -24,7 +24,7 @@
 
 #if 1
 
-#include <iostream.h>
+#include <iostream>
 #include <stdlib.h>
 #include <math.h>
 //#define NDEBUG // uncomment to remove checking of assert()
@@ -57,9 +57,9 @@
         void  rprint()const;  //print entries on a single line
         void resize(int n);
         int operator==(const vector<ElType>& v)const;
-        friend   vector<ElType> operator*(ElType c,vector<ElType>& v );
-        friend   vector<ElType> operator*(vector<ElType>& v,ElType c );
-        friend ostream& operator<<(ostream& s,vector<ElType>& v);
+        friend vector<ElType> operator* (ElType c,vector<ElType>& v );
+        friend vector<ElType> operator*(vector<ElType>& v,ElType c );
+        friend std::ostream& operator<<(std::ostream& s,vector<ElType>& v);
 };
 template <class ElType>
         void vector<ElType>::zero()
@@ -129,10 +129,10 @@
         void  vector<ElType>::rprint()const  //print entries on a single line
 {
     int i;
-    cout << "VECTOR: ";
-    cout << "(";
-    for(i=0;i<len-1;i++) cout << data[i] << ",";
-    cout << data[len-1] << ")" << endl;
+    std::cout << "VECTOR: ";
+    std::cout << "(";
+    for(i=0;i<len-1;i++) std::cout << data[i] << ",";
+    std::cout << data[len-1] << ")" << std::endl;
     return;
 }
 template <class ElType>
@@ -164,7 +164,7 @@
     return ans;
 }
 template <class ElType>
-        ostream& operator<<(ostream& s,vector<ElType>& v)
+        std::ostream& operator<<(std::ostream& s,vector<ElType>& v)
 {
     s << "(";
     for(int i=0;i<v.len-1;i++) s << v.data[i] << ", ";
@@ -192,7 +192,7 @@
         matrix<ElType> operator-(const matrix<ElType>& a);
         matrix<ElType> transpose();
     //matrix<ElType> inverse();
-        friend ostream& operator<<(ostream& s,matrix<ElType>& m);
+        friend std::ostream& operator<<(std::ostream& s,matrix<ElType>& m);
         friend void ludcmp(matrix<ElType>& a,vector<int>& indx,double &d);
         friend void lubksb(matrix<ElType>&a,vector<int>& indx,vector<ElType>&b);
 };
@@ -357,7 +357,7 @@
     return ans;
 }
 template <class ElType>
-        ostream& operator<<(ostream& s,matrix<ElType>& m)
+        std::ostream& operator<<(std::ostream& s,matrix<ElType>& m)
 {
     for(int i=0; i<m.rows;i++) s << m[i];
     return s;
@@ -381,7 +381,7 @@
 //         kdDebug() << "new search" << endl;
         for (j=0;j<n;j++) { if ((temp=fabs(a[i][j])) > big) big=temp;
 /*            kdDebug() << temp << " " << fabs(a[i][j]) << " "<< big <<endl; */}
-            if (big == 0.0) { cerr << "Singular matrix in routine LUDCMP" << endl; big = TINY;}
+            if (big == 0.0) { std::cerr << "Singular matrix in routine LUDCMP" << std::endl; big = TINY;}
             vv[i]=1.0/big;
     }
     for (j=0;j<n;j++)
diff -Naur koffice-1.6.3-orig/kspread/plugins/scripting/kspreadcore/krs_cell.cpp koffice-1.6.3/kspread/plugins/scripting/kspreadcore/krs_cell.cpp
--- koffice-1.6.3-orig/kspread/plugins/scripting/kspreadcore/krs_cell.cpp	2008-04-12 21:18:17.000000000 -0400
+++ koffice-1.6.3/kspread/plugins/scripting/kspreadcore/krs_cell.cpp	2008-04-12 21:22:45.000000000 -0400
@@ -174,12 +174,8 @@
 	return m_cell->text();
 }
 
-bool Cell::setText(const QString& text, bool asString) {
-
-	//FIXME: there is some problem with asString parameter, when it's set
-	//to true KSpread says: ASSERT: "f" in Dependencies.cpp (621)
-	//kspread: Cell at row 6, col 1 marked as formula, but formula is NULL
 
+    bool Cell::setText(const QString& text) {
 	KSpread::ProtectedCheck prot;
 	prot.setSheet (m_sheet);
 	prot.add (QPoint (m_col, m_row));
@@ -189,7 +185,7 @@
 	KSpread::DataManipulator *dm = new KSpread::DataManipulator ();
 	dm->setSheet (m_sheet);
 	dm->setValue (text);
-	dm->setParsing (!asString);
+	dm->setParsing (true);
 	dm->add (QPoint (m_col, m_row));
 	dm->execute ();
 
diff -Naur koffice-1.6.3-orig/kspread/plugins/scripting/kspreadcore/krs_cell.h koffice-1.6.3/kspread/plugins/scripting/kspreadcore/krs_cell.h
--- koffice-1.6.3-orig/kspread/plugins/scripting/kspreadcore/krs_cell.h	2008-04-12 21:18:17.000000000 -0400
+++ koffice-1.6.3/kspread/plugins/scripting/kspreadcore/krs_cell.h	2008-04-12 21:22:55.000000000 -0400
@@ -131,11 +131,10 @@
 		*/
 		const QString text() const;
 		/**
-		* Set the text of the cell. If asString is true, the text
-		* will be handled as string else we try to parse the
-		* string to the expected value.
+		* Set the text of the cell. the text
+		* will be handled as string
 		*/
-		bool setText(const QString& text, bool asString = false);
+		bool setText(const QString& text);
 
 		/**
 		* Return the textcolor as RGB-value in the format "#RRGGBB".
