--- kexi/plugins/forms/Makefile.am.orig
+++ kexi/plugins/forms/Makefile.am
@@ -23,9 +23,10 @@
 	kexidataprovider.cpp kexiformscrollview.cpp kexiformeventhandler.cpp \
 	kexidbtextwidgetinterface.cpp kexiactionselectiondialog.cpp kexiformmanager.cpp \
 	kexidatasourcepage.cpp kexiformpart.cpp kexiformview.cpp
-libkexiformutils_la_LDFLAGS = $(all_libraries) $(VER_INFO) -Wnounresolved
+libkexiformutils_la_LDFLAGS = $(all_libraries) $(VER_INFO)
 libkexiformutils_la_LIBADD = $(top_builddir)/kexi/core/libkexicore.la \
 	$(top_builddir)/kexi/widget/libkexiextendedwidgets.la \
+	$(top_builddir)/kexi/formeditor/libkformdesigner.la \
 	$(top_builddir)/kexi/plugins/forms/widgets/libkexiformutilswidgets.la
 
 kformdesignerservicesdir=$(kde_servicesdir)/kformdesigner
--- kexi/plugins/reports/Makefile.am.orig
+++ kexi/plugins/reports/Makefile.am
@@ -5,24 +5,29 @@
 kexihandler_report_la_SOURCES = kexireports.cpp
 kexihandler_report_la_LDFLAGS = $(all_libraries) $(KDE_PLUGIN) $(VER_INFO) -module -no-undefined
 kexihandler_report_la_LIBADD = ../../core/libkexicore.la \
-  ../../widget/utils/libkexiguiutils.la \
-  ../../widget/libkexiextendedwidgets.la \
-  ../../formeditor/libkformdesigner.la \
-  ../forms/libkexiformutils.la \
+  $(top_builddir)/kexi/widget/utils/libkexiguiutils.la \
+  $(top_builddir)/kexi/widget/libkexiextendedwidgets.la \
+	$(top_builddir)/kexi/formeditor/libkformdesigner.la \
+  $(top_builddir)/kexi/plugins/forms/libkexiformutils.la \
   ./libkexireportutils.la
 
 kexireportwidgets_la_LDFLAGS =  $(all_libraries) $(KDE_PLUGIN) $(VER_INFO) -module
 kexireportwidgets_la_SOURCES =   reportwidgets.cpp kexireportfactory.cpp
 kexireportwidgets_la_LIBADD = $(top_builddir)/kexi/formeditor/libkformdesigner.la \
+	$(top_builddir)/kexi/plugins/forms/libkexiformutils.la \
+	$(top_builddir)/kexi/plugins/forms/widgets/libkexiformutilswidgets.la \
+	$(top_builddir)/kexi/widget/libkexiextendedwidgets.la \
 	./libkexireportutils.la
 
 lib_LTLIBRARIES = libkexireportutils.la
 libkexireportutils_la_SOURCES = \
 	kexireportpart.cpp kexireportview.cpp kexireportform.cpp
-libkexireportutils_la_LDFLAGS = $(all_libraries) $(VER_INFO) -Wnounresolved
+libkexireportutils_la_LDFLAGS = $(all_libraries) $(VER_INFO)
 libkexireportutils_la_LIBADD = $(top_builddir)/kexi/core/libkexicore.la \
+	$(top_builddir)/kexi/formeditor/libkformdesigner.la \
+	$(top_builddir)/kexi/plugins/forms/widgets/libkexiformutilswidgets.la \
 	$(top_builddir)/kexi/widget/libkexiextendedwidgets.la \
-	$(top_builddir)/kexi/plugins/forms/widgets/libkexiformutilswidgets.la
+	$(top_builddir)/kexi/plugins/forms/libkexiformutils.la
 
 kformdesignerservicesdir=$(kde_servicesdir)/kformdesigner
 kformdesignerservices_DATA = kformdesigner_kexireportfactory.desktop
