diff --git a/kwin/tabbox/clientmodel.cpp b/kwin/tabbox/clientmodel.cpp
index 9591bab..9790245 100644
--- a/kwin/tabbox/clientmodel.cpp
+++ b/kwin/tabbox/clientmodel.cpp
@@ -195,6 +195,9 @@ void ClientModel::createClientList(int desktop, bool partialReset)
             }
         }
         TabBoxClient* stop = c;
+        // test whether the Client marked as stop will be included at all in the list
+        // see BUG: 306260
+        bool stopInFocusChain = !tabBox->clientToAddToList(c, desktop).isNull();
         do {
             QWeakPointer<TabBoxClient> add = tabBox->clientToAddToList(c, desktop);
             if (!add.isNull()) {
@@ -204,6 +207,10 @@ void ClientModel::createClientList(int desktop, bool partialReset)
                 }
             }
             c = tabBox->nextClientFocusChain(c).data();
+            if (!stopInFocusChain) {
+                stop = c;
+                stopInFocusChain = true;
+            }
         } while (c && c != stop);
         break;
     }
