--- kdebase-workspace-4.6.0/kwin/effects/configs_builtins.cpp~	2011-02-09 23:11:21.970007079 -0800
+++ kdebase-workspace-4.6.0/kwin/effects/configs_builtins.cpp	2011-02-09 23:10:54.123340156 -0800
@@ -33,6 +33,7 @@
 #include "shadow/shadow_config.h"
 #include "showfps/showfps_config.h"
 #include "thumbnailaside/thumbnailaside_config.h"
+#include "windowgeometry/windowgeometry_config.h"
 #include "zoom/zoom_config.h"
 
 #ifdef KWIN_HAVE_OPENGL_COMPOSITING
@@ -71,6 +72,7 @@
     KWIN_EFFECT_CONFIG_SINGLE( shadow, ShadowEffectConfig )
     KWIN_EFFECT_CONFIG_SINGLE( showfps, ShowFpsEffectConfig )
     KWIN_EFFECT_CONFIG_SINGLE( thumbnailaside, ThumbnailAsideEffectConfig )
+    KWIN_EFFECT_CONFIG_SINGLE( windowgeometry, WindowGeometryConfig )
     KWIN_EFFECT_CONFIG_SINGLE( zoom, ZoomEffectConfig )
 
 #ifdef KWIN_HAVE_OPENGL_COMPOSITING
