diff -up kdebase-workspace-4.6.1/solid/networkmanager-0.7/manager.cpp.foo kdebase-workspace-4.6.1/solid/networkmanager-0.7/manager.cpp
--- kdebase-workspace-4.6.1/solid/networkmanager-0.7/manager.cpp.foo	2011-03-22 20:02:37.298879915 -0500
+++ kdebase-workspace-4.6.1/solid/networkmanager-0.7/manager.cpp	2011-03-22 20:03:40.423090754 -0500
@@ -118,16 +118,16 @@ QObject *NMNetworkManager::createNetwork
     uint deviceType = devIface.deviceType();
     NMNetworkInterface * createdInterface = 0;
     switch ( deviceType ) {
-        case DEVICE_TYPE_802_3_ETHERNET:
+        case NM_DEVICE_TYPE_ETHERNET:
             createdInterface = new NMWiredNetworkInterface(uni, this, 0); // these are deleted by the frontend manager
             break;
-        case DEVICE_TYPE_802_11_WIRELESS:
+        case NM_DEVICE_TYPE_WIFI:
             createdInterface = new NMWirelessNetworkInterface(uni, this, 0);
             break;
-        case DEVICE_TYPE_GSM:
+        case 3:
             createdInterface = new NMGsmNetworkInterface(uni, this, 0);
             break;
-        case DEVICE_TYPE_CDMA:
+        case 4:
             createdInterface = new NMCdmaNetworkInterface(uni, this, 0);
             break;
             /*
