--- a/drivers/gpu/drm/drm_sysfs.c~	2009-12-04 09:24:31.000000000 +0100
+++ b/drivers/gpu/drm/drm_sysfs.c	2009-12-04 09:24:31.000000000 +0100
@@ -15,6 +15,7 @@
 #include <linux/device.h>
 #include <linux/kdev_t.h>
 #include <linux/err.h>
+#include <linux/version.h>
 
 #include "drm_sysfs.h"
 #include "drm_core.h"
@@ -77,7 +78,11 @@
 		       CORE_MINOR, CORE_PATCHLEVEL, CORE_DATE);
 }
 
-static char *drm_nodename(struct device *dev)
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(2,6,32))
+  static char *drm_nodename(struct device *dev)
+#else
+  static char *drm_devnode(struct device *dev, mode_t *mode)
+#endif /* (LINUX_VERSION_CODE < KERNEL_VERSION(2,6,32)) */
 {
 	return kasprintf(GFP_KERNEL, "dri/%s", dev_name(dev));
 }
@@ -112,8 +117,11 @@
 	err = class_create_file(class, &class_attr_version);
 	if (err)
 		goto err_out_class;
-
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(2,6,32))
 	class->nodename = drm_nodename;
+#else
+	class->devnode = drm_devnode;
+#endif /* (LINUX_VERSION_CODE < KERNEL_VERSION(2,6,32)) */
 
 	return class;
 
