From c82fc3131802f4fe1f069facb9aae833cc6cdc5f Mon Sep 17 00:00:00 2001
From: Yanko Kaneti <yaneti@declera.com>
Date: Wed, 13 Feb 2013 21:15:13 +0200
Subject: [PATCH] Fix NumberBox envents configuration with latest pygobject

---
 src/lib/number_box.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/lib/number_box.py b/src/lib/number_box.py
index 4087de1..60b255a 100644
--- a/src/lib/number_box.py
+++ b/src/lib/number_box.py
@@ -130,7 +130,7 @@ class NumberBox (Gtk.DrawingArea):
         self.bottom_note_list = []
         self.tinfo = tracker_info.TrackerInfo()
         self.set_property('can-focus', True)
-        self.set_property('events', Gdk.EventMask.ALL_EVENTS_MASK)
+        self.add_events(Gdk.EventMask.ALL_EVENTS_MASK)
         self.connect('button-press-event', self.button_press_cb)
         self.connect('key-release-event', self.key_press_cb)
         self.connect('enter-notify-event', self.pointer_enter_cb)
-- 
1.8.1.2

