From 48063198a77c0274ae7d9ba2f50dc4c25ddba99f Mon Sep 17 00:00:00 2001
From: Thomas Dziedzic <gostrc@gmail.com>
Date: Fri, 18 Apr 2014 09:40:01 -0700
Subject: [PATCH] relax upper bound on network

---
 HTTP.cabal | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/HTTP.cabal b/HTTP.cabal
index 594e22d..a1b5578 100644
--- a/HTTP.cabal
+++ b/HTTP.cabal
@@ -85,7 +85,7 @@ Library
                  Network.HTTP.Utils
                  Paths_HTTP
   GHC-options: -fwarn-missing-signatures -Wall
-  Build-depends: base >= 2 && < 4.8, network < 2.5, parsec
+  Build-depends: base >= 2 && < 4.8, network < 2.6, parsec
   Extensions: FlexibleInstances
   if flag(old-base)
     Build-depends: base < 3
-- 
1.9.2

