diff -Nru libofx-0.9.4/debian/patches/fix-ftbfs-gcc4.7.diff libofx-0.9.4/debian/patches/fix-ftbfs-gcc4.7.diff
--- libofx-0.9.4.orig/ofxconnect/ofxpartner.cpp	2011-03-30 22:30:50.000000000 +0000
+++ libofx-0.9.4/ofxconnect/ofxpartner.cpp	2012-04-17 06:08:38.711940892 +0000
@@ -37,6 +37,7 @@
 #include <vector>
 #include <algorithm>
 #include <string.h>
+#include <unistd.h>
 
 using std::string;
 using std::vector;
