--- krita/ui/kis_png_converter.cpp.old	2010-01-20 17:48:11.890898029 +0100
+++ krita/ui/kis_png_converter.cpp	2010-01-20 17:50:13.477553307 +0100
@@ -404,7 +404,7 @@
 
     png_byte signature[8];
     iod->peek((char*)signature, 8);
-    if (!png_check_sig(signature, 8)) {
+    if (!png_sig_cmp(signature, 0, 8) == 0) {
         iod->close();
         return (KisImageBuilder_RESULT_BAD_FETCH);
     }
