Index: 2.8/src/gtk/gsockgtk.cpp
===================================================================
--- 2.8/src/gtk/gsockgtk.cpp (revision 60599)
+++ 2.8/src/gtk/gsockgtk.cpp (working copy)
@@ -15,8 +15,13 @@
 #include <stdlib.h>
 #include <stdio.h>

+// newer versions of glib define its own GSocket but we unfortunately use this
+// name in our own (semi-)public header and so can't change it -- rename glib
+// one instead
+#define GSocket GlibGSocket
 #include <gdk/gdk.h>
 #include <glib.h>
+#undef GSocket

 #include "wx/gsocket.h"
 #include "wx/unix/gsockunx.h"
