diff -up gnome-disk-utility-0.4/src/gdu/gdu.pc.in.drop-polkit gnome-disk-utility-0.4/src/gdu/gdu.pc.in
--- gnome-disk-utility-0.4/src/gdu/gdu.pc.in.drop-polkit	2009-07-27 01:13:07.886996591 -0400
+++ gnome-disk-utility-0.4/src/gdu/gdu.pc.in	2009-07-27 01:13:19.967999834 -0400
@@ -6,6 +6,6 @@ includedir=@includedir@
 Name: gdu
 Description: GObject based Disk Utility Library
 Version: @VERSION@
-Requires: gobject-2.0 gio-2.0 polkit
+Requires: gobject-2.0 gio-2.0
 Libs: -L${libdir} -lgdu
 Cflags: -I${includedir}/gnome-disk-utility
diff -up gnome-disk-utility-0.4/src/gdu-gtk/gdu-gtk.pc.in.drop-polkit gnome-disk-utility-0.4/src/gdu-gtk/gdu-gtk.pc.in
--- gnome-disk-utility-0.4/src/gdu-gtk/gdu-gtk.pc.in.drop-polkit	2009-07-27 01:13:39.889005506 -0400
+++ gnome-disk-utility-0.4/src/gdu-gtk/gdu-gtk.pc.in	2009-07-27 01:13:48.500002504 -0400
@@ -6,6 +6,6 @@ includedir=@includedir@
 Name: gdu-gtk
 Description: GTK+ library for libgdu
 Version: @VERSION@
-Requires: gdu polkit-gnome gtk+-2.0
+Requires: gdu gtk+-2.0
 Libs: -L${libdir} -lgdu-gtk
 Cflags: -I${includedir}/gnome-disk-utility
