From ad094a827e37219b76cf6b6bf3d6fab49a32e913 Mon Sep 17 00:00:00 2001
From: Alec Leamas <leamas.alec@gmail.com>
Date: Fri, 9 Jan 2015 12:46:36 +0100
Subject: [PATCH 04/13] lirc-lsplugins: Bugfix.

---
 tools/lirc-lsplugins.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/lirc-lsplugins.c b/tools/lirc-lsplugins.c
index 10403a7..9eceb3b 100644
--- a/tools/lirc-lsplugins.c
+++ b/tools/lirc-lsplugins.c
@@ -212,7 +212,7 @@ static void line_print_long(const line_t* line)
 	}
 
 	printf("Plugin path:\t%s\n", line->path);
-	printf("Driver name:\t%s\n", line->path ? line->name : "-");
+	printf("Driver name:\t%s\n", line->name ? line->name : "-");
 	printf("Load state:\t%s\n", loadstate);
 	printf("Timing info:\t%s\n", handles_timing);
 	printf("Can send:\t%s\n", can_send);
-- 
2.2.2

