--- cpu/fetchdecode.cc.orig	2010-04-16 15:52:44.000000000 -0400
+++ cpu/fetchdecode.cc	2010-10-07 13:23:06.000000000 -0400
@@ -1,5 +1,5 @@
 /////////////////////////////////////////////////////////////////////////
-// $Id: fetchdecode.cc,v 1.266 2010/04/16 19:52:44 sshwarts Exp $
+// $Id: fetchdecode.cc,v 1.267 2010/04/29 21:04:23 sshwarts Exp $
 /////////////////////////////////////////////////////////////////////////
 //
 //  Copyright (C) 2001-2009  The Bochs Project
@@ -474,7 +474,7 @@
   /* 0F 0D /wr */ { 0, BX_IA_PREFETCHW },       // 3DNow! PREFETCH on AMD, NOP on Intel
   /* 0F 0E /wr */ { 0, BX_IA_FEMMS },           // 3DNow! FEMMS
 #if BX_SUPPORT_3DNOW
-  /* 0F 0F /wr */ { BxImmediate_Ib, BX_IA_ERROR, Bx3DNowOpcodeInfo },
+  /* 0F 0F /wr */ { BxImmediate_Ib, BX_IA_ERROR },
 #else
   /* 0F 0F /wr */ { 0, BX_IA_ERROR },
 #endif
@@ -1025,7 +1025,7 @@
   /* 0F 0D /dr */ { 0, BX_IA_PREFETCHW },       // 3DNow! PREFETCH on AMD, NOP on Intel
   /* 0F 0E /dr */ { 0, BX_IA_FEMMS },           // 3DNow! FEMMS
 #if BX_SUPPORT_3DNOW
-  /* 0F 0F /dr */ { BxImmediate_Ib, BX_IA_ERROR, Bx3DNowOpcodeInfo },
+  /* 0F 0F /dr */ { BxImmediate_Ib, BX_IA_ERROR },
 #else
   /* 0F 0F /dr */ { 0, BX_IA_ERROR },
 #endif
@@ -1583,7 +1583,7 @@
   /* 0F 0D /wm */ { 0, BX_IA_PREFETCHW },       // 3DNow! PREFETCH on AMD, NOP on Intel
   /* 0F 0E /wm */ { 0, BX_IA_FEMMS },           // 3DNow! FEMMS
 #if BX_SUPPORT_3DNOW
-  /* 0F 0F /wm */ { BxImmediate_Ib, BX_IA_ERROR, Bx3DNowOpcodeInfo },
+  /* 0F 0F /wm */ { BxImmediate_Ib, BX_IA_ERROR },
 #else
   /* 0F 0F /wm */ { 0, BX_IA_ERROR },
 #endif
@@ -2134,7 +2134,7 @@
   /* 0F 0D /dm */ { 0, BX_IA_PREFETCHW },       // 3DNow! PREFETCH on AMD, NOP on Intel
   /* 0F 0E /dm */ { 0, BX_IA_FEMMS },           // 3DNow! FEMMS
 #if BX_SUPPORT_3DNOW
-  /* 0F 0F /dm */ { BxImmediate_Ib, BX_IA_ERROR, Bx3DNowOpcodeInfo },
+  /* 0F 0F /dm */ { BxImmediate_Ib, BX_IA_ERROR },
 #else
   /* 0F 0F /dm */ { 0, BX_IA_ERROR },
 #endif
