From 5b9daa4f892c32a4c5d5199acc2b62142d85b382 Mon Sep 17 00:00:00 2001
From: Andre Miranda <andreldm@xfce.org>
Date: Fri, 11 Mar 2022 19:07:48 -0300
Subject: [PATCH] Do not quit after dismissing dialog if invoked from plugin
 (#71)

Fixes #71
---
 lib/screenshooter-dialogs.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/lib/screenshooter-dialogs.c b/lib/screenshooter-dialogs.c
index d56f88b..371563a 100644
--- a/lib/screenshooter-dialogs.c
+++ b/lib/screenshooter-dialogs.c
@@ -630,7 +630,8 @@ cb_dialog_response (GtkWidget *dialog, gint response, ScreenshotData *sd)
   else
     {
       gtk_widget_destroy (dialog);
-      gtk_main_quit ();
+      if (!sd->plugin)
+        gtk_main_quit ();
     }
 }
 
