diff --git a/cmake/FindIMobileDevice.cmake b/cmake/FindIMobileDevice.cmake
index 5168fc4..d86fd95 100644
--- a/cmake/FindIMobileDevice.cmake
+++ b/cmake/FindIMobileDevice.cmake
@@ -33,7 +33,7 @@ find_package(PkgConfig QUIET)
 pkg_check_modules(PC_libimobiledevice QUIET libimobiledevice-1.0)
 
 find_path(IMobileDevice_INCLUDE_DIRS NAMES libimobiledevice/libimobiledevice.h HINTS ${PC_libimobiledevice_INCLUDE_DIRS})
-find_library(IMobileDevice_LIBRARIES NAMES imobiledevice HINTS ${PC_libimobiledevice_LIBRARY_DIRS})
+find_library(IMobileDevice_LIBRARIES NAMES imobiledevice-1.0 imobiledevice HINTS ${PC_libimobiledevice_LIBRARY_DIRS})
 
 set(IMobileDevice_VERSION ${PC_libimobiledevice_VERSION})
 
diff --git a/cmake/FindPList.cmake b/cmake/FindPList.cmake
index 363981e..e13db6b 100644
--- a/cmake/FindPList.cmake
+++ b/cmake/FindPList.cmake
@@ -33,7 +33,7 @@ find_package(PkgConfig QUIET)
 pkg_search_module(PC_libplist QUIET libplist-2.0 libplist)
 
 find_path(PList_INCLUDE_DIRS NAMES plist/plist.h HINTS ${PC_libplist_INCLUDE_DIRS})
-find_library(PList_LIBRARIES NAMES plist HINTS ${PC_libplist_LIBRARY_DIRS})
+find_library(PList_LIBRARIES NAMES plist-2.0 plist HINTS ${PC_libplist_LIBRARY_DIRS})
 
 set(PList_VERSION ${PC_libplist_VERSION})
 
