--- avogadro-1.0.3/libavogadro/src/CMakeLists.txt~	2011-11-29 18:33:01.777291627 +0000
+++ avogadro-1.0.3/libavogadro/src/CMakeLists.txt	2011-11-29 18:33:23.480814612 +0000
@@ -149,7 +149,7 @@
   endforeach(P_ITEM ${PYTHON_SRCS})
 endif(NOT ENABLE_PYTHON OR NOT ALL_PYTHON_FOUND)
 
-QT4_WRAP_CPP(libavogadro_MOC_SRCS ${libavogadro_MOC_HDRS})
+QT4_WRAP_CPP(libavogadro_MOC_SRCS ${libavogadro_MOC_HDRS} OPTIONS -DBOOST_TT_HAS_OPERATOR_HPP_INCLUDED)
 
 # you have to add link_directories before you add the target
 if(ENABLE_PYTHON AND ALL_PYTHON_FOUND)
--- avogadro-1.0.3/libavogadro/src/python/CMakeLists.txt~	2011-11-29 18:37:15.449506548 +0000
+++ avogadro-1.0.3/libavogadro/src/python/CMakeLists.txt	2011-11-29 18:37:52.729831744 +0000
@@ -11,7 +11,7 @@
 # use all cpp files in this directory
 FILE(GLOB wrapper_SRCS "*.cpp")
 
-QT4_WRAP_CPP(MOC_SRCS moleculelist.h)
+QT4_WRAP_CPP(MOC_SRCS moleculelist.h OPTIONS -DBOOST_TT_HAS_OPERATOR_HPP_INCLUDED)
 
 ADD_LIBRARY(python-module MODULE ${wrapper_SRCS} ${MOC_SRCS})
 SET_TARGET_PROPERTIES(python-module PROPERTIES OUTPUT_NAME Avogadro)
--- avogadro-1.0.3/libavogadro/CMakeLists.txt~	2011-11-29 19:41:34.066938217 +0000
+++ avogadro-1.0.3/libavogadro/CMakeLists.txt	2011-11-29 19:42:53.084320315 +0000
@@ -18,7 +18,7 @@
   # Assume all MOC stuff is in the headers, replace .cpp and use qt4_wrap_cpp
   # We should probably scan the header to verify the Q_OBJECT macro is used
   string(REPLACE ".cpp" ".h" hdr_list "${src_list}")
-  qt4_wrap_cpp(moc_files ${hdr_list})
+  qt4_wrap_cpp(moc_files ${hdr_list} OPTIONS -DBOOST_TT_HAS_OPERATOR_HPP_INCLUDED)
   # Now sort out the ui and qrc files, process them as appropriate
   set(ui_plugin_files)
   set(qrc_plugin_files)
