--- keytouch-acpid/Makefile.am
+++ keytouch-acpid/Makefile.am
@@ -3,7 +3,8 @@
 
 AM_CPPFLAGS =	-I. -I../string_to_keycode/ -I../mxml \
 		-DSYSCONF_DIR=\""$(sysconfdir)"\" \
-		-DPACKAGE_DATA_DIR=\""$(datadir)"\"
+		-DPACKAGE_DATA_DIR=\""$(datadir)"\" \
+		-D_GNU_SOURCE # struct ucred
 
 AM_CFLAGS=@CFLAGS@ @X_CFLAGS@ -Wall
 
--- keytouch-keyboard/src/check_syntax.c
+++ keytouch-keyboard/src/check_syntax.c
@@ -29,10 +29,10 @@
     You can contact me at: marvinr(at)users(dot)sf(dot)net
     (replace (at) by @ and (dot) by .)
 -----------------------------------------------------------------------------------*/
+#define _XOPEN_SOURCE /* strptime */
 #include <stdlib.h>
 #include <stdio.h>
 #include <string.h>
-#define _XOPEN_SOURCE /* glibc2 needs this */
 #include <time.h>
 
 #include <mxml.h>
--- keytouch-keyboard/src/config.c
+++ keytouch-keyboard/src/config.c
@@ -37,6 +37,7 @@
     You can contact me at: marvinr(at)users(dot)sf(dot)net
     (replace (at) by @ and (dot) by .)
 -----------------------------------------------------------------------------------*/
+#define _GNU_SOURCE /* strptime, scandir */
 #include <stdlib.h>
 #include <dirent.h>
 #include <dlfcn.h>
@@ -45,7 +46,6 @@
 #include <unistd.h>
 #include <string.h>
 #include <stdio.h>
-#define _GNU_SOURCE /* glibc2 needs this */
 #include <time.h>
 
 #include <mxml.h>
