diff -ur vamp-plugin-sdk-2.0.orig/examples/FixedTempoEstimator.cpp vamp-plugin-sdk-2.0/examples/FixedTempoEstimator.cpp
--- vamp-plugin-sdk-2.0.orig/examples/FixedTempoEstimator.cpp	2008-12-01 13:48:41.000000000 +0200
+++ vamp-plugin-sdk-2.0/examples/FixedTempoEstimator.cpp	2009-06-06 00:57:58.000000000 +0300
@@ -43,9 +43,9 @@
 
 using Vamp::RealTime;
 
+#include <cstdio>
 #include <cmath>
 
-
 class FixedTempoEstimator::D
 // this class just avoids us having to declare any data members in the header
 {
