diff -Naur bluez-utils-2.14-orig/scripts/bluepin bluez-utils-2.14/scripts/bluepin
--- bluez-utils-2.14-orig/scripts/bluepin	2004-03-05 15:14:00.000000000 -0800
+++ bluez-utils-2.14/scripts/bluepin	2005-01-26 11:19:35.000000000 -0800
@@ -28,7 +28,7 @@
 			break
 
 	os.environ['DISPLAY']    = disp 
-	os.environ['XAUTHORITY'] = auth
+#	os.environ['XAUTHORITY'] = auth
 
 # Set X display before initializing GTK
 set_display()
