From f401d649318e0e01ccefeb99a398a6f32fd820a8 Mon Sep 17 00:00:00 2001
From: Toms Bauģis <toms.baugis@gmail.com>
Date: Thu, 04 Nov 2010 00:46:53 +0000
Subject: not sure how it was working before, really.  fixes bug 633103

---
diff --git a/src/hamster/db.py b/src/hamster/db.py
index 5412b58..77e73c3 100644
--- a/src/hamster/db.py
+++ b/src/hamster/db.py
@@ -937,7 +937,7 @@ class Storage(storage.Storage):
     def end_transaction(self):
         self.__con.commit()
         self.__cur.close()
-        self.__con = None
+        self.__con, self.__cur = None, None
         self.register_modification()
 
     def run_fixtures(self):
--
cgit v0.8.3.1
