--- libgphoto2_port/configure.ac.orig	2007-02-07 09:23:51.000000000 +0000
+++ libgphoto2_port/configure.ac	2007-02-07 09:24:17.000000000 +0000
@@ -154,12 +154,6 @@
 
 dnl Apple Bonjour / mDNSResponder multicast DNS queries.
 MDNS_LIBS=
-AC_CHECK_HEADER([dns_sd.h],[
-		AC_CHECK_LIB(dns_sd,DNSServiceResolve,[
-			MDNS_LIBS="-ldns_sd"
-			AC_DEFINE(HAVE_MDNS_BONJOUR,1,[Define if you have Bonjour/mDNSResponder.])
-		])
-])
 
 dnl Check for mnt_mountp in struct mnttab
 AC_MSG_CHECKING([for mnt_mountp in struct mnttab])
