Index: phonon/platform_kde/kdeplatformplugin.cpp
===================================================================
--- phonon/platform_kde/kdeplatformplugin.cpp	(revision 1196218)
+++ phonon/platform_kde/kdeplatformplugin.cpp	(revision 1196219)
@@ -43,9 +43,11 @@
 
 typedef QPair<QByteArray, QString> PhononDeviceAccess;
 typedef QList<PhononDeviceAccess> PhononDeviceAccessList;
+#if (PHONON_VERSION < PHONON_VERSION_CHECK(4, 4, 3))
 #ifndef KDE_USE_FINAL
 Q_DECLARE_METATYPE(PhononDeviceAccessList)
 #endif
+#endif
 namespace Phonon
 {
 
Index: phonon/platform_kde/devicelisting.cpp
===================================================================
--- phonon/platform_kde/devicelisting.cpp	(revision 1196218)
+++ phonon/platform_kde/devicelisting.cpp	(revision 1196219)
@@ -19,6 +19,7 @@
 */
 
 #include "devicelisting.h"
+#include "phononnamespace.h"
 
 #include <QtCore/QFile>
 #include <QtDBus/QDBusReply>
@@ -35,7 +36,9 @@
 #endif // HAVE_ALSA_ASOUNDLIB_H
 
 typedef QList<QPair<QByteArray, QString> > PhononDeviceAccessList;
+#if (PHONON_VERSION < PHONON_VERSION_CHECK(4, 4, 3))
 Q_DECLARE_METATYPE(PhononDeviceAccessList)
+#endif
 
 static void installAlsaPhononDeviceHandle()
 {
Index: phonon/kded-module/phononserver.cpp
===================================================================
--- phonon/kded-module/phononserver.cpp	(revision 1196218)
+++ phonon/kded-module/phononserver.cpp	(revision 1196219)
@@ -18,6 +18,7 @@
 
 */
 
+#include "phononnamespace.h"
 #include "phononserver.h"
 #include "audiodevice.h"
 
@@ -56,7 +57,9 @@
 K_EXPORT_PLUGIN(PhononServerFactory("phononserver"))
 
 typedef QList<QPair<QByteArray, QString> > PhononDeviceAccessList;
+#if (PHONON_VERSION < PHONON_VERSION_CHECK(4, 4, 3))
 Q_DECLARE_METATYPE(PhononDeviceAccessList)
+#endif
 
 PhononServer::PhononServer(QObject *parent, const QList<QVariant> &)
     : KDEDModule(parent),
Index: phonon/platform_kde/devicelisting.cpp
===================================================================
--- phonon/platform_kde/devicelisting.cpp	(revision 1197038)
+++ phonon/platform_kde/devicelisting.cpp	(revision 1197039)
@@ -19,7 +19,6 @@
 */
 
 #include "devicelisting.h"
-#include "phononnamespace.h"
 
 #include <QtCore/QFile>
 #include <QtDBus/QDBusReply>
@@ -29,6 +28,7 @@
 #include <kdebug.h>
 #include <klocale.h>
 #include <ksharedconfig.h>
+#include <phonon/phononnamespace.h>
 
 #include <../config-alsa.h>
 #ifdef HAVE_ALSA_ASOUNDLIB_H
Index: phonon/kded-module/phononserver.cpp
===================================================================
--- phonon/kded-module/phononserver.cpp	(revision 1197038)
+++ phonon/kded-module/phononserver.cpp	(revision 1197039)
@@ -18,7 +18,6 @@
 
 */
 
-#include "phononnamespace.h"
 #include "phononserver.h"
 #include "audiodevice.h"
 
@@ -45,6 +44,7 @@
 #include <Solid/GenericInterface>
 #include <Solid/Device>
 #include <Solid/DeviceNotifier>
+#include <phonon/phononnamespace.h>
 
 #include <../config-alsa.h>
 #ifdef HAVE_LIBASOUND2
