Index: mjpegtools-1.8.0/mpeg2enc/Makefile.am
===================================================================
--- mjpegtools-1.8.0.orig/mpeg2enc/Makefile.am
+++ mjpegtools-1.8.0/mpeg2enc/Makefile.am
@@ -110,7 +110,7 @@ libmpeg2encpp_la_LIBADD = $(LIBMJPEGUTIL
 
 mpeg2enc_DEPENDENCIES = \
 	$(LIBMJPEGUTILS) \
-	$(top_builddir)/mpeg2enc/libmpeg2encpp.la
+	libmpeg2encpp.la
 
 mpeg2enc_LDADD = \
 	$(top_builddir)/mpeg2enc/libmpeg2encpp.la \
Index: mjpegtools-1.8.0/mpeg2enc/Makefile.in
===================================================================
--- mjpegtools-1.8.0.orig/mpeg2enc/Makefile.in
+++ mjpegtools-1.8.0/mpeg2enc/Makefile.in
@@ -374,7 +374,7 @@ libmpeg2encpp_la_DEPENDENCIES = $(LIBMJP
 libmpeg2encpp_la_LIBADD = $(LIBMJPEGUTILS)
 mpeg2enc_DEPENDENCIES = \
 	$(LIBMJPEGUTILS) \
-	$(top_builddir)/mpeg2enc/libmpeg2encpp.la
+	libmpeg2encpp.la
 
 mpeg2enc_LDADD = \
 	$(top_builddir)/mpeg2enc/libmpeg2encpp.la \
Index: mjpegtools-1.8.0/mplex/Makefile.am
===================================================================
--- mjpegtools-1.8.0.orig/mplex/Makefile.am
+++ mjpegtools-1.8.0/mplex/Makefile.am
@@ -79,5 +79,5 @@ libmplex2_la_CXXFLAGS = $(ZALPHA_FLAGS)
 mplex_SOURCES = main.cpp 
 
 mplex_LDADD = \
-	$(top_builddir)/mplex/libmplex2.la \
+	libmplex2.la \
 	@LIBGETOPT_LIB@ $(LIBM_LIBS) 
Index: mjpegtools-1.8.0/mplex/Makefile.in
===================================================================
--- mjpegtools-1.8.0.orig/mplex/Makefile.in
+++ mjpegtools-1.8.0/mplex/Makefile.in
@@ -95,8 +95,7 @@ PROGRAMS = $(bin_PROGRAMS)
 am_mplex_OBJECTS = main.$(OBJEXT)
 mplex_OBJECTS = $(am_mplex_OBJECTS)
 am__DEPENDENCIES_2 =
-mplex_DEPENDENCIES = $(top_builddir)/mplex/libmplex2.la \
-	$(am__DEPENDENCIES_2)
+mplex_DEPENDENCIES = libmplex2.la $(am__DEPENDENCIES_2)
 DEFAULT_INCLUDES = -I. -I$(srcdir) -I$(top_builddir)
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__depfiles_maybe = depfiles
@@ -340,7 +339,7 @@ libmplex2_la_LIBADD = $(top_builddir)/ut
 libmplex2_la_CXXFLAGS = $(ZALPHA_FLAGS)
 mplex_SOURCES = main.cpp 
 mplex_LDADD = \
-	$(top_builddir)/mplex/libmplex2.la \
+	libmplex2.la \
 	@LIBGETOPT_LIB@ $(LIBM_LIBS) 
 
 all: all-am
