Index: drscheme-4.0/src/Makefile.in
===================================================================
--- drscheme-4.0.orig/src/Makefile.in
+++ drscheme-4.0/src/Makefile.in
@@ -81,7 +81,8 @@ no-run:
 	cd .
 
 copytree-run:
-	mzscheme/mzscheme@CGC@ -u \
+	LD_LIBRARY_PATH="${LD_LIBRARY_PATH}${LD_LIBRARY_PATH:+:}$(DESTDIR)$(libdir)" \
+          mzscheme/mzscheme@CGC@ -u \
           "$(srcdir)/../collects/setup/unixstyle-install.ss" \
           make-install-copytree "$(srcdir)/.." \
           $(ALLDIRINFO) "@INSTALL_ORIG_TREE@"
@@ -94,6 +95,7 @@ install-no-post-collects:
 
 fix-paths:
 	if [ "$(DESTDIR)" != "" ]; then \
+		LD_LIBRARY_PATH="${LD_LIBRARY_PATH}${LD_LIBRARY_PATH:+:}$(DESTDIR)$(libdir)" \
           mzscheme/mzscheme@CGC@ -u \
             "$(srcdir)/../collects/setup/unixstyle-install.ss" \
             make-install-destdir-fix "$(srcdir)/.." \
@@ -110,7 +112,8 @@ install-3m-common:
 
 install-3m:
 	$(MAKE) install-3m-common
-	mzscheme/mzscheme@MMM@ $(SETUP_ARGS)
+	LD_LIBRARY_PATH="${LD_LIBRARY_PATH}${LD_LIBRARY_PATH:+:}$(DESTDIR)$(libdir)" \
+		mzscheme/mzscheme@MMM@ $(SETUP_ARGS)
 	$(MAKE) install-common-last
 
 plain-install-3m:
@@ -133,7 +136,8 @@ install-cgc-common:
 
 install-cgc:
 	$(MAKE) install-cgc-common
-	mzscheme/mzscheme@CGC@ $(SETUP_ARGS)
+	LD_LIBRARY_PATH="${LD_LIBRARY_PATH}${LD_LIBRARY_PATH:+:}$(DESTDIR)$(libdir)" \
+		mzscheme/mzscheme@CGC@ $(SETUP_ARGS)
 	$(MAKE) install-common-last
 
 plain-install-cgc:
@@ -156,7 +160,8 @@ install-both-common:
 
 install-both:
 	$(MAKE) install-both-common
-	mzscheme/mzscheme@MAIN_VARIANT@ $(SETUP_ARGS)
+	LD_LIBRARY_PATH="${LD_LIBRARY_PATH}${LD_LIBRARY_PATH:+:}$(DESTDIR)$(libdir)" \
+		mzscheme/mzscheme@MAIN_VARIANT@ $(SETUP_ARGS)
 	$(MAKE) install-common-last
 
 plain-install-both:
