--- tools/extra/unittests/clang-tidy/ReadabilityModuleTest.cpp.orig	2022-04-24 02:31:36.544185904 +0000
+++ tools/extra/unittests/clang-tidy/ReadabilityModuleTest.cpp	2022-04-24 02:32:20.390088782 +0000
@@ -1,4 +1,4 @@
-#include "../../clang/unittests/ASTMatchers/ASTMatchersTest.h"
+#include "../../../../unittests/ASTMatchers/ASTMatchersTest.h"
 #include "ClangTidyTest.h"
 #include "readability/BracesAroundStatementsCheck.h"
 #include "readability/NamespaceCommentCheck.h"
