diff -Naur libcap-1.10.orig/Make.Rules libcap-1.10/Make.Rules
--- libcap-1.10.orig/Make.Rules	2005-05-30 16:10:07.000000000 +0000
+++ libcap-1.10/Make.Rules	2005-05-30 16:11:18.000000000 +0000
@@ -42,8 +42,8 @@
 # Compilation specifics
 
 CC=gcc
-COPTFLAGS=-O2
-DEBUG=-g #-DDEBUG
+COPTFLAGS=-O2 -fPIC
+DEBUG=
 WARNINGS=-ansi -D_POSIX_SOURCE -Wall -Wwrite-strings \
         -Wpointer-arith -Wcast-qual -Wcast-align \
         -Wtraditional -Wstrict-prototypes -Wmissing-prototypes \
@@ -54,7 +54,7 @@
 IPATH=-I$(topdir)/libcap/include
 INCS=$(topdir)/libcap/include/sys/capability.h
 LIBS=-L$(topdir)/libcap -lcap
-CFLAGS=-Dlinux $(WARNINGS) $(DEBUG) $(COPTFLAG) $(IPATH)
+CFLAGS+=-Dlinux $(WARNINGS) $(DEBUG) $(COPTFLAG) $(IPATH)
 
 # Global cleanup stuff