 src/select.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git c/src/select.c i/src/select.c
index beb5c1e3b..00b709652 100644
--- c/src/select.c
+++ i/src/select.c
@@ -4003,6 +4003,9 @@ static int flattenSubquery(
   ** queries.
   */
   if( pSub->pPrior ){
+    if( pSrc->nSrc!=1 ){
+      return 0; /* workaround https://www.sqlite.org/forum/forumpost/1061e627d0 */
+    }
     if( pSub->pOrderBy ){
       return 0;  /* Restriction (20) */
     }
