diff -ruN kded.old/Makefile.am kded/Makefile.am
--- kded.old/Makefile.am	2005-05-23 12:17:00.000000000 +0000
+++ kded/Makefile.am	2005-09-13 07:20:32.000000000 +0000
@@ -60,7 +60,7 @@
 servicetype_DATA = kdedmodule.desktop
 servicetypedir = $(kde_servicetypesdir)
 
-xdg_menu_DATA = applications.menu
+xdg_menu_DATA = kde-applications.menu
 
 update_DATA = kded.upd
 updatedir = $(kde_datadir)/kconf_update
diff -ruN kded.old/Makefile.in kded/Makefile.in
--- kded.old/Makefile.in	2005-07-20 21:20:08.000000000 +0000
+++ kded/Makefile.in	2005-09-13 07:20:51.000000000 +0000
@@ -603,7 +603,7 @@
 
 servicetype_DATA = kdedmodule.desktop
 servicetypedir = $(kde_servicetypesdir)
-xdg_menu_DATA = applications.menu
+xdg_menu_DATA = kde-applications.menu
 update_DATA = kded.upd
 updatedir = $(kde_datadir)/kconf_update
 #>- all: all-am
@@ -1143,7 +1143,7 @@
 	-rm -f  vfolder_menu.moc kded.moc kdedmodule.moc kbuildsycoca.moc
 
 #>+ 2
-KDE_DIST=DESIGN kded.upd Makefile.in HOWTO vfolder_menu.h applications.menu kded.h kdedmodule.desktop README.kded kctimefactory.h Makefile.am 
+KDE_DIST=DESIGN kded.upd Makefile.in HOWTO vfolder_menu.h kde-applications.menu kded.h kdedmodule.desktop README.kded kctimefactory.h Makefile.am 
 
 #>+ 2
 docs-am:
diff -ruN kded.old/kde-applications.menu kded/kde-applications.menu
--- kded.old/kde-applications.menu	1970-01-01 00:00:00.000000000 +0000
+++ kded/kde-applications.menu	2005-09-13 07:18:55.000000000 +0000
@@ -0,0 +1,388 @@
+ <!DOCTYPE Menu PUBLIC "-//freedesktop//DTD Menu 1.0//EN"
+  "http://www.freedesktop.org/standards/menu-spec/1.0/menu.dtd">
+
+<Menu>
+	<Name>Applications</Name>
+	<Directory>kde-main.directory</Directory>
+	<!-- Search the default locations -->
+	<KDELegacyDirs/>
+	<DefaultAppDirs/>
+	<DefaultDirectoryDirs/>
+	<DefaultLayout>
+		<Merge type="menus"/>
+		<Merge type="files"/>
+		<Separator/>
+		<Menuname>More</Menuname>
+	</DefaultLayout>
+	<Layout>
+		<Merge type="menus"/>
+		<Menuname>Applications</Menuname>
+		<Merge type="files"/>
+	</Layout>
+
+	<Menu>
+		<Name>Applications</Name>
+		<Directory>kde-unknown.directory</Directory>
+		<OnlyUnallocated/>
+		<Include>
+			<Not>
+				<!-- Don't list non-KDE core applications -->
+				<And>
+					<Category>Core</Category>
+					<Not><Category>KDE</Category></Not>
+				</And>
+				<!-- Don't list SUSE's YaST in here -->
+				<Category>X-SuSE-YaST</Category>
+			</Not>
+		</Include>
+	</Menu>
+       <Menu>
+                <Name>Development</Name>
+                <Directory>kde-development.directory</Directory>
+                <Menu>
+                        <Name>X-KDE-KDevelopIDE</Name>
+                        <Directory>kde-development-kdevelop.directory</Directory>
+                        <Include>
+                                <And>
+                                <Category>Development</Category>
+                                <Category>X-KDE-KDevelopIDE</Category>
+                                </And>
+                        </Include>
+                </Menu>
+                <Include>
+                        <And>
+                        <Category>Development</Category>
+                        <Not>
+                        <Category>X-KDE-KDevelopIDE</Category>
+                        </Not>
+                        </And>
+                </Include>
+        </Menu>
+	<Menu>
+		<Name>Science</Name>
+		<Directory>kde-science.directory</Directory>
+		<Include>
+			<And><!-- Include /any/ Science app which is not an Education app -->
+				<Or>
+					<Category>Astronomy</Category>
+					<Category>Biology</Category>
+					<Category>Chemistry</Category>
+					<Category>Geology</Category>
+					<Category>MedicalSoftware</Category>
+					<Category>Physics</Category>
+					<Category>Math</Category>
+					<Category>Science</Category>
+				</Or>
+				<Not><Category>Education</Category></Not>
+			</And>
+		</Include>
+	</Menu>
+	<Menu>
+		<Name>Editors</Name>
+		<Directory>kde-editors.directory</Directory>
+		<Include>
+			<Category>TextEditor</Category>
+		</Include>
+	</Menu>
+	<Menu>
+		<Name>Edutainment</Name>
+		<Directory>kde-edutainment.directory</Directory>
+		<Menu>
+			<Name>Languages</Name>
+			<Directory>kde-edu-languages.directory</Directory>
+			<Include>
+				<And>
+				<Category>Education</Category>
+				<Or>
+				<Category>Languages</Category>
+				<Category>X-KDE-Edu-Language</Category>
+				</Or>
+				</And>
+			</Include>
+		</Menu>
+		<Menu>
+			<Name>Mathematics</Name>
+			<Directory>kde-edu-mathematics.directory</Directory>
+			<Include>
+				<And>
+				<Category>Education</Category>
+				<Category>Math</Category>
+				</And>
+			</Include>
+		</Menu>
+		<Menu>
+			<Name>Miscellaneous</Name>
+			<Directory>kde-edu-miscellaneous.directory</Directory>
+			<Include>
+				<And>
+				<Category>Education</Category>
+				<Not>
+				<Category>Languages</Category>
+				<Category>X-KDE-Edu-Language</Category>
+				<Category>Math</Category>
+				<Category>Science</Category>
+				<Category>Teaching</Category>
+				<Category>X-KDE-Edu-Teaching</Category>
+				</Not>
+				</And>
+			</Include>
+		</Menu>
+		<Menu>
+			<Name>Science</Name>
+			<Directory>kde-edu-science.directory</Directory>
+			<Include>
+				<And>
+				<Category>Education</Category>
+				<Category>Science</Category>
+				</And>
+			</Include>
+		</Menu>
+		<Menu>
+			<Name>Tools</Name>
+			<Directory>kde-edu-tools.directory</Directory>
+			<Include>
+				<And>
+				<Category>Education</Category>
+				<Or>
+				<Category>Teaching</Category>
+				<Category>X-KDE-Edu-Teaching</Category>
+				</Or>
+				</And>
+			</Include>
+		</Menu>
+	</Menu>
+	<Menu>
+		<Name>Games</Name>
+		<Directory>kde-games.directory</Directory>
+		<Menu>
+			<Name>Arcade</Name>
+			<Directory>kde-games-arcade.directory</Directory>
+			<Include>
+				<And>
+				<Category>Game</Category>
+				<Category>ArcadeGame</Category>
+				</And>
+			</Include>
+		</Menu>
+		<Menu>
+			<Name>Board</Name>
+			<Directory>kde-games-board.directory</Directory>
+			<Include>
+				<And>
+				<Category>Game</Category>
+				<Category>BoardGame</Category>
+				</And>
+			</Include>
+		</Menu>
+		<Menu>
+			<Name>Card</Name>
+			<Directory>kde-games-card.directory</Directory>
+			<Include>
+				<And>
+				<Category>Game</Category>
+				<Category>CardGame</Category>
+				</And>
+			</Include>
+		</Menu>
+		<Menu>
+			<Name>Kidsgames</Name>
+			<Directory>kde-games-kids.directory</Directory>
+			<Include>
+				<And>
+				<Category>Game</Category>
+				<Or>
+				<Category>X-KDE-KidsGame</Category>
+				<Category>KidsGame</Category>
+				</Or>
+				</And>
+			</Include>
+		</Menu>
+		<Menu>
+			<Name>TacticStrategy</Name>
+			<Directory>kde-games-strategy.directory</Directory>
+			<Include>
+				<And>
+				<Category>Game</Category>
+				<Category>StrategyGame</Category>
+				</And>
+			</Include>
+		</Menu>
+		<Include>
+			<And>
+			<Category>Game</Category>
+			<Not>
+			<Category>ArcadeGame</Category>
+			<Category>BoardGame</Category>
+			<Category>CardGame</Category>
+			<Category>X-KDE-KidsGame</Category>
+			<Category>KidsGame</Category>
+			<Category>StrategyGame</Category>
+			</Not>
+			</And>
+		</Include>
+	</Menu>
+	<Menu>
+		<Name>Graphics</Name>
+		<Directory>kde-graphics.directory</Directory>
+		<Include>
+			<And>
+			<Category>Graphics</Category>
+			<Not><Category>X-KDE-More</Category></Not>
+			</And>
+		</Include>
+		<Menu>
+			<Name>More</Name>
+			<Directory>kde-more.directory</Directory>
+			<Include>
+				<And>
+				<Category>Graphics</Category>
+				<Category>X-KDE-More</Category>
+				</And>
+			</Include>
+		</Menu>
+	</Menu>
+	<Menu>
+		<Name>Internet</Name>
+		<Directory>kde-internet.directory</Directory>
+		<Include>
+			<And>
+			<Category>Network</Category>
+			<Not><Category>X-KDE-More</Category></Not>
+			</And>
+		</Include>
+		<Menu>
+		  <Name>Terminal</Name>
+		  <Directory>kde-internet-terminal.directory</Directory>
+		</Menu>
+		<Menu>
+			<Name>More</Name>
+			<Directory>kde-more.directory</Directory>
+			<Include>
+				<And>
+				<Category>Network</Category>
+				<Category>X-KDE-More</Category>
+				</And>
+			</Include>
+		</Menu>
+	</Menu>
+	<Menu>
+		<Name>Multimedia</Name>
+		<Directory>kde-multimedia.directory</Directory>
+		<Include>
+			<And>
+			<Category>AudioVideo</Category>
+			<Not><Category>X-KDE-More</Category></Not>
+			</And>
+		</Include>
+		<Menu>
+			<Name>More</Name>
+			<Directory>kde-more.directory</Directory>
+			<Include>
+				<And>
+				<Category>AudioVideo</Category>
+				<Category>X-KDE-More</Category>
+				</And>
+			</Include>
+		</Menu>
+	</Menu>
+	<Menu>
+		<Name>Office</Name>
+		<Directory>kde-office.directory</Directory>
+		<Layout>
+			<Merge type="menus"/>
+			<Filename>kde-koshell.desktop</Filename>
+			<Filename>kde-Kontact.desktop</Filename>
+			<Separator/>
+			<Filename>kde-kword.desktop</Filename>
+			<Filename>kde-kspread.desktop</Filename>
+			<Filename>kde-kpresenter.desktop</Filename>
+			<Merge type="files"/>
+			<Separator/>
+			<Menuname>More</Menuname>
+		</Layout>
+		<Include>
+			<Category>Office</Category>
+		</Include>
+	</Menu>
+	<Menu>
+		<Name>Settingsmenu</Name>
+		<Directory>kde-settingsmenu.directory</Directory>
+		<Include>
+			<Category>Settings</Category>
+		</Include>
+	</Menu>
+	<Menu>
+		<Name>System</Name>
+		<Directory>kde-system.directory</Directory>
+		<Include>
+			<And>
+			<Category>System</Category>
+			<Not><Category>X-KDE-More</Category></Not>
+			</And>
+		</Include>
+		<Menu>
+			<Name>More</Name>
+			<Directory>kde-more.directory</Directory>
+			<Include>
+				<And>
+				<Category>System</Category>
+				<Category>X-KDE-More</Category>
+				</And>
+			</Include>
+		</Menu>
+		<Menu>
+			<Name>ScreenSavers</Name>
+			<Directory>kde-system-screensavers.directory</Directory>
+		</Menu>
+		<Menu>
+			<Name>Terminal</Name>
+			<Directory>kde-system-terminal.directory</Directory>
+		</Menu>
+	</Menu>
+	<Menu>
+		<Name>Toys</Name>
+		<Directory>kde-toys.directory</Directory>
+		<Include>
+			<Category>Amusement</Category>
+		</Include>
+	</Menu>
+	<Menu>
+		<Name>Utilities</Name>
+		<Directory>kde-utilities.directory</Directory>
+		<Include>
+			<And>
+			<Category>Utility</Category>
+			<Not><Category>X-KDE-More</Category></Not>
+			</And>
+		</Include>
+		<Menu>
+			<Name>XUtilities</Name>
+			<Directory>kde-utilities-xutils.directory</Directory>
+		</Menu>
+		<Menu>
+			<Name>More</Name>
+			<Directory>kde-more.directory</Directory>
+			<Include>
+				<And>
+				<Category>Utility</Category>
+				<Category>X-KDE-More</Category>
+				</And>
+			</Include>
+		</Menu>
+	</Menu>
+	<Menu>
+		<Name>Archlinux</Name>
+		<Directory>Archlinux.directory</Directory>
+		<Include>
+			<Category>Archlinux</Category>
+		</Include>
+	</Menu>
+	<Include>
+		<And>
+		<Category>KDE</Category>
+		<Category>Core</Category>
+		</And>
+	</Include>
+	<DefaultMergeDirs/>
+	<MergeFile>applications-kmenuedit.menu</MergeFile>
+</Menu>
