--- branches/KDE/4.2/kdelibs/plasma/popupapplet.cpp	2009/04/09 04:18:10	951366
+++ branches/KDE/4.2/kdelibs/plasma/popupapplet.cpp	2009/04/09 04:21:08	951367
@@ -275,8 +275,7 @@
                 //stuff out of your Dialog (extenders). Monitor WindowDeactivate events so we can
                 //emulate the same kind of behavior as Qt::Popup (close when you click somewhere
                 //else.
-                //Use Qt::Tool otherwise dialogs get shown over screensaver (bug #179924).
-                Qt::WindowFlags wflags = Qt::Tool | Qt::FramelessWindowHint | Qt::WindowStaysOnTopHint;
+                Qt::WindowFlags wflags = Qt::FramelessWindowHint | Qt::WindowStaysOnTopHint;
 
                 if (passive) {
                     wflags |= Qt::X11BypassWindowManagerHint;
