Index: kdeui/shortcuts/kdedglobalaccel.cpp
===================================================================
--- kdeui/shortcuts/kdedglobalaccel.cpp	(Revision 866735)
+++ kdeui/shortcuts/kdedglobalaccel.cpp	(Revision 866736)
@@ -394,8 +394,9 @@
 
 void KdedGlobalAccel::unRegister(const QStringList &actionId)
 {
+    kDebug(125) << actionId;
+
     Q_ASSERT(actionId.size()==4);
-
     if (actionId.size() < 4) {
         return;
     }
@@ -661,7 +662,14 @@
 
             foreach (int key, ad->keys) {
                 if (key != 0) {
-                    d->keyToAction.insert(key, ad);
+                    if (d->keyToAction.contains(key)) {
+                        // The shortcut is already used. The config file is
+                        // broken. Ignore the request.
+                        ad->keys.removeAll(key);
+                        kWarning() << "Shortcut found twice in kglobalshortcutsrc.";
+                    } else {
+                        d->keyToAction.insert(key, ad);
+                    }
                 }
             }
         }
