diff -aur old/lib/isc/unix/socket.c new/lib/isc/unix/socket.c
--- old/lib/isc/unix/socket.c	2010-03-12 04:25:20.000000000 +0100
+++ new/lib/isc/unix/socket.c	2011-01-22 21:07:52.410000038 +0100
@@ -681,6 +681,8 @@
 	isc_sockstatscounter_fdwatchrecvfail
 };
 
+#undef SO_BSDCOMPAT
+
 static void
 manager_log(isc__socketmgr_t *sockmgr,
 	    isc_logcategory_t *category, isc_logmodule_t *module, int level,
Only in new/lib/isc/unix: socket.c.orig
