diff -ur Newt-1.08.orig/newtlib/listbox.c Newt-1.08/newtlib/listbox.c
--- Newt-1.08.orig/newtlib/listbox.c	1998-10-16 18:51:31.000000000 +0200
+++ Newt-1.08/newtlib/listbox.c	2006-04-08 01:55:30.000000000 +0200
@@ -630,7 +630,7 @@
 	    er.result = ER_SWALLOWED;
 	    break;
 	  default:
-	    /* keeps gcc quiet */
+	    ; /* keeps gcc quiet */
 	}
 	break;
 	
diff -ur Newt-1.08.orig/newtlib/whiptcl.c Newt-1.08/newtlib/whiptcl.c
--- Newt-1.08.orig/newtlib/whiptcl.c	1997-08-18 23:12:02.000000000 +0200
+++ Newt-1.08/newtlib/whiptcl.c	2006-04-08 01:55:30.000000000 +0200
@@ -228,6 +228,7 @@
 	break;
 
       case MODE_NONE:
+	;
 	/* this can't happen */
     }
 
