diff --git a/smb/CMakeLists.txt b/smb/CMakeLists.txt
index 45cd1f11..b4d59160 100644
--- a/smb/CMakeLists.txt
+++ b/smb/CMakeLists.txt
@@ -3,6 +3,7 @@ find_package(KDSoapWSDiscoveryClient QUIET)
 set(INTERNAL_WSDCLIENT ${BUILD_KDSoapWSDiscoveryClient})
 if(KDSoapWSDiscoveryClient_FOUND)
     set(INTERNAL_WSDCLIENT OFF)
+    find_package(KDSoap)
 endif()
 if(INTERNAL_WSDCLIENT)
     # Special internal version, mangled to be a STATIC lib.
