diff -upr kitinerary-21.12.2.orig/src/lib/pdf/pdfdocument.cpp kitinerary-21.12.2/src/lib/pdf/pdfdocument.cpp
--- kitinerary-21.12.2.orig/src/lib/pdf/pdfdocument.cpp	2022-01-29 01:31:44.000000000 +0200
+++ kitinerary-21.12.2/src/lib/pdf/pdfdocument.cpp	2022-03-02 11:01:34.112359075 +0200
@@ -301,7 +301,11 @@ PdfDocument* PdfDocument::fromData(const
     obj.initNull();
     auto stream = new MemStream(const_cast<char*>(doc->d->m_pdfData.constData()), 0, doc->d->m_pdfData.size(), &obj);
 #endif
+#if KPOPPLER_VERSION >= QT_VERSION_CHECK(22, 3, 0)
+    std::unique_ptr<PDFDoc> popplerDoc(new PDFDoc(stream));
+#else
     std::unique_ptr<PDFDoc> popplerDoc(new PDFDoc(stream, nullptr, nullptr));
+#endif
     if (!popplerDoc->isOk()) {
         qCWarning(Log) << "Got invalid PDF document!" << popplerDoc->getErrorCode();
         return nullptr;
