--- grub-1.99.orig/conf/Makefile.common
+++ grub-1.99/conf/Makefile.common
@@ -11,10 +11,10 @@ if COND_i386_pc
   CFLAGS_PLATFORM += -mrtd -mregparm=3
 endif
 if COND_i386_efi
-  LDFLAGS_PLATFORM = -melf_i386
+  LDFLAGS_PLATFORM = -Wl,-melf_i386
 endif
 if COND_x86_64_efi
-  LDFLAGS_PLATFORM = -melf_x86_64
+  LDFLAGS_PLATFORM = -Wl,-melf_x86_64
 endif
 if COND_i386_qemu
   CFLAGS_PLATFORM += -mrtd -mregparm=3
