--- discover-5.9.1/discover/org.kde.discover.desktop.orig	2017-02-10 10:41:44.439510995 +0000
+++ discover-5.9.1/discover/org.kde.discover.desktop	2017-02-10 10:42:05.339476223 +0000
@@ -36,7 +36,7 @@
 Name[x-test]=xxDiscoverxx
 Name[zh_CN]=发现者
 Name[zh_TW]=Discover
-MimeType=x-scheme-handler/appstream;application/vnd.debian.binary-package;application/x-rpm;
+MimeType=x-scheme-handler/appstream;
 Exec=plasma-discover %U
 Icon=plasmadiscover
 Type=Application
