Index: gconf/gconftool.c
===================================================================
--- gconf/gconftool.c	(révision 2396)
+++ gconf/gconftool.c	(copie de travail)
@@ -543,6 +543,9 @@ main (int argc, char** argv)
   setlocale (LC_ALL, "");
   bindtextdomain (GETTEXT_PACKAGE,GCONF_LOCALE_DIR);
   textdomain (GETTEXT_PACKAGE);
+
+  g_thread_init (NULL);
+
   _gconf_init_i18n ();
   
   context = g_option_context_new (_("- Tool to manipulate a GConf configuration"));
Index: gconf/gconf-sanity-check.c
===================================================================
--- gconf/gconf-sanity-check.c	(révision 2396)
+++ gconf/gconf-sanity-check.c	(copie de travail)
@@ -42,7 +42,9 @@ main (int argc, char** argv)
 {
   GOptionContext *context;
   GError *error;
-  
+
+  g_thread_init (NULL);
+
   context = g_option_context_new (_("- Sanity checks for GConf"));
   g_option_context_add_group (context, gtk_get_option_group (TRUE));
 
