--- lapack-3.1.1/INSTALL/make.inc.gfortran.BAD	2007-05-25 15:34:55.000000000 -0500
+++ lapack-3.1.1/INSTALL/make.inc.gfortran	2007-05-25 15:35:51.000000000 -0500
@@ -8,7 +8,7 @@ SHELL = /bin/sh
 #
 #  The machine (platform) identifier to append to the library names
 #
-PLAT = _LINUX
+PLAT = 
 #  
 #  Modify the FORTRAN and OPTS definitions to refer to the
 #  compiler and desired compiler options for your machine.  NOOPT
@@ -17,9 +17,9 @@ PLAT = _LINUX
 #  desired load options for your machine.
 #
 FORTRAN  = gfortran 
-OPTS     = -O2
+#OPTS     = -O2
 DRVOPTS  = $(OPTS)
-NOOPT    = -O0
+NOOPT    = 
 LOADER   = gfortran
 LOADOPTS =
 #
