From d733314ef8293a447568ef48af8b0ddd83a4ad42 Mon Sep 17 00:00:00 2001
From: Inando <inand0@users.sf.net>
Date: Mon, 29 Dec 2014 14:45:08 +0100
Subject: [PATCH 02/13] lirc.pc: Fix bad variable reference/ordering (#86)

---
 lirc.pc.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lirc.pc.in b/lirc.pc.in
index 43f2b9b..432f51a 100644
--- a/lirc.pc.in
+++ b/lirc.pc.in
@@ -2,8 +2,8 @@ Name: lirc
 Description: LIRC Linux Infrared Control client libs.
 Version: @VERSION@
 
-datarootdir=@datarootdir@
 prefix=@prefix@
+datarootdir=@datarootdir@
 exec_prefix=@exec_prefix@
 libdir=@libdir@
 includedir=@includedir@
-- 
2.2.2

