diff -Nurp rosegarden-16.02.orig/src/gui/dialogs/PitchBendSequenceDialog.cpp rosegarden-16.02.mod/src/gui/dialogs/PitchBendSequenceDialog.cpp
--- rosegarden-16.02.orig/src/gui/dialogs/PitchBendSequenceDialog.cpp	2016-02-29 13:18:01.000000000 +0100
+++ rosegarden-16.02.mod/src/gui/dialogs/PitchBendSequenceDialog.cpp	2016-04-25 07:30:49.578374291 +0200
@@ -919,7 +919,7 @@ PitchBendSequenceDialog::addStepwiseEven
         {
             const int rawStepSize = spinboxToControlDelta(m_stepSize);
             if (rawStepSize == 0) { return; }
-            numSteps = abs(float(valueChange) / float(rawStepSize) + 0.5);
+            numSteps = fabs(float(valueChange) / float(rawStepSize) + 0.5);
             break;
         }
     }

