diff -Naur librep-0.16-orig/man/Makefile.in librep-0.16/man/Makefile.in
--- librep-0.16-orig/man/Makefile.in	2002-04-14 13:01:38.000000000 -0700
+++ librep-0.16/man/Makefile.in	2002-08-24 20:46:08.000000000 -0700
@@ -24,7 +24,7 @@
 
 SRCS = librep.texi lang.texi repl.texi news.texi interface.texi
 
-all : librep.info ../NEWS
+all : ../NEWS
 
 ../NEWS : news.texi
 	$(MAKEINFO) $(MAKEINFOFLAGS) --no-headers $< -o ../NEWS
@@ -44,14 +44,9 @@
 librep.pdf : $(SRCS)
 	$(TEXI2PDF) -I $(srcdir) $<
 
-install : librep.info installdirs
-	for f in librep.info*; do \
-	  $(INSTALL_DATA) $$f $(DESTDIR)$(infodir); \
-	done
-	-install-info --info-dir=$(DESTDIR)$(infodir) librep.info
+install : installdirs
 
 installdirs : mkinstalldirs
-	$(SHELL) $< $(DESTDIR)$(infodir)
 
 uninstall :
 	rm -f $(DESTDIR)$(infodir)/librep.info*
