diff -Nurp xserver-xorg-video-intel-2.2.0.90.patched/src/i830_exa.c xserver-xorg-video-intel-2.2.0.90.working/src/i830_exa.c
--- xserver-xorg-video-intel-2.2.0.90.patched/src/i830_exa.c	2008-02-19 10:59:47.000000000 -0800
+++ xserver-xorg-video-intel-2.2.0.90.working/src/i830_exa.c	2008-02-19 10:57:54.000000000 -0800
@@ -541,6 +541,12 @@ I830EXAInit(ScreenPtr pScreen)
 	pI830->EXADriverPtr->pixmapPitchAlign = 16;
 	pI830->EXADriverPtr->maxX = 8192;
 	pI830->EXADriverPtr->maxY = 8192;
+#if defined(EXA_MIGRATION_GREEDY)
+        /* Force use of Option "MigrationHeuristic" "greedy" here for 965 with EXA (LP: #177492) */
+        pI830->EXADriverPtr->flags |= EXA_MIGRATION_GREEDY;
+        xf86DrvMsg(pScrn->scrnIndex, X_WARNING,
+                   "EXA greedy mode enabled for i965 migration.\n");
+#endif
     } else {
 	pI830->EXADriverPtr->pixmapOffsetAlign = 4;
 	pI830->EXADriverPtr->pixmapPitchAlign = 16;
