diff -Nurp xserver-xorg-video-intel-2.2.0.90-prepatch/src/i830_quirks.c xserver-xorg-video-intel-2.2.0.90-patched/src/i830_quirks.c
--- xserver-xorg-video-intel-2.2.0.90-prepatch/src/i830_quirks.c	2008-02-29 13:18:07.000000000 -0800
+++ xserver-xorg-video-intel-2.2.0.90-patched/src/i830_quirks.c	2008-02-29 13:22:58.000000000 -0800
@@ -104,6 +104,9 @@ static i830_quirk i830_quirk_list[] = {
     /* Dell Inspiron 510m needs pipe A force quirk */
     { PCI_CHIP_I855_GM, 0x1028, 0x0164, quirk_pipea_force },
 
+    /* Intel 945GM hardware (See LP: #152416) */
+    { PCI_CHIP_I945_GM, 0x1584, 0x9900, quirk_ignore_tv },
+
     { 0, 0, 0, NULL },
 };
 
