diff -Nurp xserver-xorg-video-intel-2.3.2-1ubuntu2-patched/src/i830_quirks.c xserver-xorg-video-intel-2.3.2-1ubuntu2-working/src/i830_quirks.c
--- xserver-xorg-video-intel-2.3.2-1ubuntu2-patched/src/i830_quirks.c	2008-06-26 20:21:14.000000000 -0700
+++ xserver-xorg-video-intel-2.3.2-1ubuntu2-working/src/i830_quirks.c	2008-06-26 20:30:23.000000000 -0700
@@ -242,6 +242,8 @@ static i830_quirk i830_quirk_list[] = {
 
     /* Lenovo Napa TV (use dmi)*/
     { PCI_CHIP_I945_GM, 0x17aa, SUBSYS_ANY, quirk_lenovo_tv_dmi },
+    /* Lenovo T60/R60 needs pipe A force quirk (See LP: #243405) */
+    { PCI_CHIP_I945_GM, 0x17aa, 0x201a, quirk_pipea_force },
     /* Lenovo T61 has no TV output */
     { PCI_CHIP_I965_GM, 0x17aa, 0x20b5, quirk_ignore_tv },
     /* Lenovo 3000 v200 */
