diff -Naur xmoto-0.5.2-orig/src/image/tim_png.cpp xmoto-0.5.2/src/image/tim_png.cpp
--- xmoto-0.5.2-orig/src/image/tim_png.cpp	2010-01-22 02:50:26.000000000 -0500
+++ xmoto-0.5.2/src/image/tim_png.cpp	2010-01-22 02:52:04.000000000 -0500
@@ -166,7 +166,7 @@
 			png_set_palette_to_rgb(PngPtr);
 			
 		if(nColorType==PNG_COLOR_TYPE_GRAY && nBitDepth<8)
-			png_set_gray_1_2_4_to_8(PngPtr);
+			png_set_expand_gray_1_2_4_to_8(PngPtr);
 
 		if(png_get_valid(PngPtr,InfoPtr,PNG_INFO_tRNS))
 			png_set_tRNS_to_alpha(PngPtr);
