--- archive_files/common/lib/modules/fglrx/build_mod/firegl_public.c	2009-02-11 17:40:35.000000000 +0000
+++ archive_files/common/lib/modules/fglrx/build_mod/firegl_public.c.new	2009-02-21 15:22:11.632031227 +0000
@@ -2256,7 +2256,7 @@
  * kernel <  2.6.27, on_each_cpu has 4 parameters.
  * kernel >= 2.6.27, on_each_cpu has 3 parameters (removed the "retry" parameter)
  */
-#if defined(__x86_64__) && defined(__SMP__) && (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,25)) 
+#if defined(__x86_64__) && (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,25)) 
 #   if (LINUX_VERSION_CODE < KERNEL_VERSION(2,6,27))
         on_each_cpu(KCL_flush_tlb_one, &va, 1, 1);
 #   else
