--- connectivity/source/parse/sqlbison.y	2011-05-20 13:21:02.000000000 -0600
+++ connectivity/source/parse/sqlbison.y	2011-05-20 13:22:11.000000000 -0600
@@ -4338,7 +4338,7 @@
 }
 
 // -------------------------------------------------------------------------
-void OSQLParser::error(sal_Char *fmt)
+void OSQLParser::error(const sal_Char *fmt)
 {
 	if(!m_sErrorMessage.getLength())
 	{
--- connectivity/inc/connectivity/sqlparse.hxx	2011-05-20 13:23:58.000000000 -0600
+++ connectivity/inc/connectivity/sqlparse.hxx	2011-05-20 13:23:46.000000000 -0600
@@ -233,7 +233,7 @@
         // returns the type for a parameter in a given function name
         static sal_Int32 getFunctionParameterType(sal_uInt32 _nTokenId,sal_uInt32 _nPos);
 
-        void error(sal_Char *fmt);
+        void error(const sal_Char *fmt);
         int SQLlex();
 #ifdef YYBISON
         void setParseTree(OSQLParseNode * pNewParseTree);

