From 52c5eb20fd96d63e17cb9cd18912030131bcc787 Mon Sep 17 00:00:00 2001
From: Stephan Bergmann <sbergman@redhat.com>
Date: Fri, 08 Mar 2013 09:59:05 +0000
Subject: -Werror=unused-local-typedefs (GCC 4.8)

Change-Id: Ib1dedbe8e273463fa47573486d67dc9d509953db
---
diff --git a/jvmfwk/plugins/sunmajor/pluginlib/sunjavaplugin.cxx b/jvmfwk/plugins/sunmajor/pluginlib/sunjavaplugin.cxx
index f8267c6..810470e 100644
--- a/jvmfwk/plugins/sunmajor/pluginlib/sunjavaplugin.cxx
+++ b/jvmfwk/plugins/sunmajor/pluginlib/sunjavaplugin.cxx
@@ -608,7 +608,6 @@ javaPluginError jfw_plugin_startJavaVirtualMachine(
     putenv(strdup(osJavaHome.getStr()));
 #endif
 
-    typedef jint JNICALL JNI_InitArgs_Type(void *);
     typedef jint JNICALL JNI_CreateVM_Type(JavaVM **, JNIEnv **, void *);
     rtl::OUString sSymbolCreateJava(
             RTL_CONSTASCII_USTRINGPARAM("JNI_CreateJavaVM"));
diff --git a/sd/source/ui/dlg/RemoteDialogClientBox.cxx b/sd/source/ui/dlg/RemoteDialogClientBox.cxx
index 8c4f129..2694218 100644
--- a/sd/source/ui/dlg/RemoteDialogClientBox.cxx
+++ b/sd/source/ui/dlg/RemoteDialogClientBox.cxx
@@ -201,8 +201,6 @@ void ClientBox::DeleteRemoved()
 
     if ( ! m_vRemovedEntries.empty() )
     {
-        typedef std::vector< TClientBoxEntry >::iterator ITER;
-
         m_vRemovedEntries.clear();
     }
 
diff --git a/sd/source/ui/toolpanel/controls/MasterPageObserver.cxx b/sd/source/ui/toolpanel/controls/MasterPageObserver.cxx
index cd7df65..1eb94c9 100644
--- a/sd/source/ui/toolpanel/controls/MasterPageObserver.cxx
+++ b/sd/source/ui/toolpanel/controls/MasterPageObserver.cxx
@@ -225,7 +225,6 @@ void MasterPageObserver::Implementation::AddEventListener (
 
         // Tell the new listener about all the master pages that are
         // currently in use.
-        typedef ::std::vector<String> StringList;
         MasterPageContainer::iterator aDocumentIterator;
         for (aDocumentIterator=maUsedMasterPages.begin();
              aDocumentIterator!=maUsedMasterPages.end();
--
cgit v0.9.0.2-2-gbebe
