--- blinken/src/blinken.cpp
+++ blinken/src/blinken.cpp
@@ -28,7 +28,7 @@
 
 #include "button.h"
 #include "counter.h"
-#include "libkdeedu/kedufontutils.h"
+#include <kedufontutils.h>
 #include "number.h"
 #include "highscoredialog.h"
 #include "settings.h"
--- blinken/src/highscoredialog.cpp
+++ blinken/src/highscoredialog.cpp
@@ -19,7 +19,7 @@
 #include <ktabwidget.h>
 
 #include "counter.h"
-#include "libkdeedu/kedufontutils.h"
+#include <kedufontutils.h>
 #include "settings.h"
 
 static const int margin = 15;
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -2,6 +2,11 @@
 
 set(CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake/modules )
 
+include_directories(
+   ${CMAKE_SOURCE_DIR}/libkdeedu/kdeeduui
+   ${CMAKE_SOURCE_DIR}/libkdeedu/keduvocdocument
+)
+
 # search packages used by KDE
 find_package(KDE4 4.3.80 REQUIRED)
 if(WIN32)
--- kanagram/src/CMakeLists.txt
+++ kanagram/src/CMakeLists.txt
@@ -1,7 +1,8 @@
 
-find_path(LIBKEDUVOCDOCUMENT_INCLUDE_DIR keduvocdocument.h PATH_SUFFIXES libkdeedu)
-include_directories( ${LIBKEDUVOCDOCUMENT_INCLUDE_DIR} )
-
+include_directories(
+   ${CMAKE_SOURCE_DIR}/libkdeedu/kdeeduui
+   ${CMAKE_SOURCE_DIR}/libkdeedu/keduvocdocument
+)
 
 ########### next target ###############
 
--- kanagram/src/kanagram.cpp
+++ kanagram/src/kanagram.cpp
@@ -51,7 +51,7 @@
 #include "kanagramsettings.h"
 #include "mainsettings.h"
 #include "vocabsettings.h"
-#include "libkdeedu/kedufontutils.h"
+#include <kedufontutils.h>
 
 static const char* m_textRevealWord = I18N_NOOP("reveal word");
 static const char* m_textHint = I18N_NOOP("hint");
--- khangman/src/CMakeLists.txt
+++ khangman/src/CMakeLists.txt
@@ -1,6 +1,6 @@
 
 find_path(LIBKEDUVOCDOCUMENT_INCLUDE_DIR keduvocdocument.h PATH_SUFFIXES libkdeedu)
-include_directories( ${LIBKEDUVOCDOCUMENT_INCLUDE_DIR} )
+#include_directories( ${LIBKEDUVOCDOCUMENT_INCLUDE_DIR} )
 
 ########### next target ###############
 
--- kwordquiz/src/CMakeLists.txt
+++ kwordquiz/src/CMakeLists.txt
@@ -4,7 +4,6 @@
 add_subdirectory( examples )
 
 find_path(LIBKEDUVOCDOCUMENT_INCLUDE_DIR keduvocdocument.h PATH_SUFFIXES libkdeedu)
-include_directories( ${LIBKEDUVOCDOCUMENT_INCLUDE_DIR} )
 
 
 ########### next target ###############
--- libkdeedu/LibKdeEduConfig.cmake.in
+++ libkdeedu/LibKdeEduConfig.cmake.in
@@ -28,7 +28,7 @@
 endif(NOT WIN32)
 
 if( NOT TARGET LibKdeEduImport__keduvocdocument )
-    include("${_LIBKDEEDU_CURRENT_DIR}/LibKdeEduTargets.cmake")
+   # include("${_LIBKDEEDU_CURRENT_DIR}/LibKdeEduTargets.cmake")
 endif( NOT TARGET LibKdeEduImport__keduvocdocument )
 
 macro( _libkdeedu_set_lib_vars _prefix _lib )
--- parley/plasmoid/CMakeLists.txt
+++ parley/plasmoid/CMakeLists.txt
@@ -11,7 +11,7 @@
 kde4_add_plugin(plasma_applet_parley ${parley_plasma_SRCS})
 target_link_libraries(plasma_applet_parley
     plasma ${KDE4_KIO_LIBS}
-    ${LIBKDEEDU_KEDUVOCDOCUMENT_LIBRARIES}
+    keduvocdocument
     )
 
 install(
--- parley/plasmoid/engine/CMakeLists.txt
+++ parley/plasmoid/engine/CMakeLists.txt
@@ -7,7 +7,7 @@
 kde4_add_plugin(plasma_engine_parley ${parley_engine_SRCS})
 target_link_libraries(plasma_engine_parley ${KDE4_KDECORE_LIBS}
   plasma
-  ${LIBKDEEDU_KEDUVOCDOCUMENT_LIBRARIES}
+  keduvocdocument
 )
 
 install(TARGETS plasma_engine_parley DESTINATION ${PLUGIN_INSTALL_DIR})
--- parley/src/CMakeLists.txt
+++ parley/src/CMakeLists.txt
@@ -164,7 +164,7 @@
 
 target_link_libraries(
     parley
-    ${LIBKDEEDU_KEDUVOCDOCUMENT_LIBRARIES}
+    keduvocdocument
     ${KDE4_KHTML_LIBS}
     ${KDE4_KNEWSTUFF3_LIBS}
     ${KDE4_PHONON_LIBS}