--- ui/prefs.glade.old	2007-06-18 08:43:03.000000000 -0400
+++ ui/prefs.glade	2007-06-18 08:44:16.000000000 -0400
@@ -2570,7 +2570,7 @@
 			  <property name="column_spacing">12</property>
 
 			  <child>
-			    <widget class="GtkCheckButton" id="checkbutton1">
+			    <widget class="GtkCheckButton" id="prefs_hex_signed">
 			      <property name="visible">True</property>
 			      <property name="can_focus">True</property>
 			      <property name="label" translatable="yes">Signed</property>
@@ -2593,7 +2593,7 @@
 			  </child>
 
 			  <child>
-			    <widget class="GtkSpinButton" id="spinbutton1">
+			    <widget class="GtkSpinButton" id="prefs_hex_bits">
 			      <property name="visible">True</property>
 			      <property name="can_focus">True</property>
 			      <property name="climb_rate">1</property>
