diff --git i/js/src/build/Makefile.in w/js/src/build/Makefile.in
index 0d5e980e8254..96c70b5432d4 100644
--- i/js/src/build/Makefile.in
+++ w/js/src/build/Makefile.in
@@ -79,6 +79,8 @@ ifneq (,$(REAL_LIBRARY))
 endif
 ifneq (,$(SHARED_LIBRARY))
 	$(SYSINSTALL) $(SHARED_LIBRARY) $(DESTDIR)$(libdir)
+	mv -f $(DESTDIR)$(libdir)/$(SHARED_LIBRARY) $(DESTDIR)$(libdir)/$(SHARED_LIBRARY).0
+	ln -s $(SHARED_LIBRARY).0 $(DESTDIR)$(libdir)/$(SHARED_LIBRARY)
 ifeq ($(OS_ARCH),Darwin)
 	install_name_tool -id $(abspath $(libdir)/$(SHARED_LIBRARY)) $(DESTDIR)$(libdir)/$(SHARED_LIBRARY)
 endif
diff --git i/js/src/build/moz.build w/js/src/build/moz.build
index 9b7ad87afea8..dbe3b6b89996 100644
--- i/js/src/build/moz.build
+++ w/js/src/build/moz.build
@@ -25,6 +25,7 @@ if not CONFIG['JS_STANDALONE']:
 if CONFIG['JS_SHARED_LIBRARY']:
     GeckoSharedLibrary('js', linkage=None)
     SHARED_LIBRARY_NAME = CONFIG['JS_LIBRARY_NAME']
+    LDFLAGS += ['-Wl,-soname,lib{}.so.0'.format(SHARED_LIBRARY_NAME)]
 
     # Ensure symbol versions of shared library on Linux do not conflict
     # with those in libxul.
