diff -aur PyXML-0.8.4.orig//xml/parsers/xmlproc/xmlutils.py PyXML-0.8.4/xml/parsers/xmlproc/xmlutils.py
--- PyXML-0.8.4.orig//xml/parsers/xmlproc/xmlutils.py	2010-11-06 17:27:53.936666669 +0800
+++ PyXML-0.8.4/xml/parsers/xmlproc/xmlutils.py	2010-11-06 17:30:39.230000003 +0800
@@ -720,6 +720,7 @@
             # to the recoding.
             try:
                 self.data = self.charset_converter(self.data)
+		self.datasize = len(self.data)
             except UnicodeError, e:
                 self._handle_decoding_error(self.data, e)
             self.input_encoding = enc1
