diff -uNr ../truecrypt-5.1a-source-orig/Common/BootEncryption.cpp truecrypt-5.1a-source/Common/BootEncryption.cpp
--- ../truecrypt-5.1a-source-orig/Common/BootEncryption.cpp	2008-04-14 12:39:50.000000000 +0100
+++ truecrypt-5.1a-source/Common/BootEncryption.cpp	2008-04-14 14:29:11.000000000 +0100
@@ -6,6 +6,7 @@
 distribution packages.
 */
 
+#include <cstring>
 #include "Tcdefs.h"
 #include "Platform/Finally.h"
 #include "Platform/ForEach.h"
diff -uNr ../truecrypt-5.1a-source-orig/Core/FatFormatter.cpp truecrypt-5.1a-source/Core/FatFormatter.cpp
--- ../truecrypt-5.1a-source-orig/Core/FatFormatter.cpp	2008-04-14 12:39:50.000000000 +0100
+++ truecrypt-5.1a-source/Core/FatFormatter.cpp	2008-04-14 14:28:22.000000000 +0100
@@ -9,6 +9,7 @@
  file License.txt included in TrueCrypt binary and source code distribution
  packages. */
 
+#include <cstring>
 #include <time.h>
 #include "Common/Tcdefs.h"
 #include "Platform/Platform.h"
diff -uNr ../truecrypt-5.1a-source-orig/Platform/Memory.cpp truecrypt-5.1a-source/Platform/Memory.cpp
--- ../truecrypt-5.1a-source-orig/Platform/Memory.cpp	2008-04-14 12:39:50.000000000 +0100
+++ truecrypt-5.1a-source/Platform/Memory.cpp	2008-04-14 14:27:26.000000000 +0100
@@ -6,6 +6,7 @@
  distribution packages.
 */
 
+#include <cstring>
 #include "Memory.h"
 #include "Exception.h"
 
diff -uNr ../truecrypt-5.1a-source-orig/Volume/EncryptionTest.cpp truecrypt-5.1a-source/Volume/EncryptionTest.cpp
--- ../truecrypt-5.1a-source-orig/Volume/EncryptionTest.cpp	2008-04-14 12:39:50.000000000 +0100
+++ truecrypt-5.1a-source/Volume/EncryptionTest.cpp	2008-04-14 14:27:26.000000000 +0100
@@ -6,6 +6,7 @@
  distribution packages.
 */
 
+#include <cstring>
 #include "Cipher.h"
 #include "Common/Crc.h"
 #include "EncryptionAlgorithm.h"
