$ diff -urN Dm-target.c.asli Dm-target.c
--- Dm-target.c.asli    2007-04-24 23:32:06 +0700
+++ Dm-target.c 2007-10-05 03:41:00 +0700
@@ -656,7 +656,11 @@
 		goto err;
 	}

+#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,23)
 	bio_ctx_cache = kmem_cache_create ("truecrypt-bioctx", sizeof (struct bio_ctx), 0, 0, NULL, NULL);
+#else
+	bio_ctx_cache = kmem_cache_create ("truecrypt-bioctx", sizeof (struct bio_ctx), 0, 0, NULL);
+#endif
 	if (!bio_ctx_cache)
 	{
 		error ("kmem_cache_create failed"); 
