diff -Nru --exclude changelog python-visual-3.2.9/include/platlinux.h python-visual-3.2.9/include/platlinux.h
--- python-visual-3.2.9/include/platlinux.h	2005-06-25 17:08:57.000000000 +0530
+++ python-visual-3.2.9/include/platlinux.h	2008-03-23 15:13:48.000000000 +0530
@@ -24,7 +24,7 @@
 	pthread_mutex_t mtx;
   
  public:
-	typedef lock<mutex> lock;
+	typedef visual::lock<mutex> lock;
 
 	mutex(int spincount=0, int count=1);
 	~mutex();
