--- QtCurve-KDE4-1.0.0/style/qtcurve.cpp	2010-01-04 19:19:24.000000000 +0100
+++ QtCurve-KDE4-1.0.0/style/qtcurve.cpp	2010-01-06 14:58:34.977943636 +0100
@@ -1113,8 +1113,6 @@
     // NOTE: This call will never actually happen, its only here so that the qtcurve.so
     // contains a kio link so that this is not removed by some 'optimisation' of the
     // link process.
-    if((int)this==(int)itsHoverWidget)
-        (void)KFileDialog::getSaveFileName();
 #endif
 }
 
