diff --git i/src/tracker-control/meson.build w/src/tracker-control/meson.build
index 85fb19573..6cbb823f8 100644
--- i/src/tracker-control/meson.build
+++ w/src/tracker-control/meson.build
@@ -35,6 +35,7 @@ executable('tracker-miner-fs-control-@0@'.format(tracker_api_major),
     c_args: tracker_c_args,
     install: true,
     install_dir: get_option('libexecdir'),
+    install_rpath: tracker_internal_libs_dir,
 )
 
 # This file allows starting the service with D-Bus autoactivation.
