diff --git a/configure.ac b/configure.ac
index 5cfc1d1..1c66051 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1263,7 +1263,7 @@ esac], [sys_lua=false])
 dnl There's wide divergence on what the pkg-config file for Lua is called
 dnl See http://lua-users.org/lists/lua-l/2008-09/msg00184.html
 if test "x$sys_lua" = "xtrue" || test "x$sys_lua" = "xtest" ; then
-  PKG_CHECK_MODULES([LUA], [lua5.2], [sys_lua=true],
+  PKG_CHECK_MODULES([LUA], [lua52], [sys_lua=true],
     [PKG_CHECK_MODULES([LUA], [lua-5.2], [sys_lua=true],
       [PKG_CHECK_MODULES([LUA], [lua >= 5.2 lua < 5.3], [sys_lua=true],
 [if test "x$sys_lua" = "xtrue" ; then
