From 78b9eac0fca36e27c83c7678b18962ba286f041b Mon Sep 17 00:00:00 2001
From: Vincent Untz <vuntz@gnome.org>
Date: Fri, 01 Oct 2010 16:36:23 +0000
Subject: introspection: Fix build by telling g-ir-scanner what the prefix is

https://bugzilla.gnome.org/show_bug.cgi?id=631119
---
diff --git a/src/Makefile.am b/src/Makefile.am
index e9d3715..acbf8ff 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -350,6 +350,7 @@ Epiphany-$(EPIPHANY_API_VERSION).gir: $(INTROSPECTION_SCANNER) $(EPHY_GIR_H_FILE
 		--include=WebKit-1.0 \
 		--program=./epiphany \
 		--output $@ \
+		--strip-prefix=Ephy \
 		--pkg epiphany-$(EPIPHANY_API_VERSION) \
 		-DEPIPHANY_COMPILATION \
 		-I$(top_srcdir)/embed \
--
cgit v0.8.3.1
