--- modutils.py	2013-06-28 15:00:05.169823983 -0300
+++ modutils.py.new	2013-06-28 15:00:15.733156861 -0300
@@ -613,7 +613,7 @@
                 return _search_zip(modpath, pic)[:2]
             raise
         else:
-            if checkeggs:
+            if checkeggs and mp_filename:
                 fullabspath = [abspath(x) for x in _path]
                 try:
                     pathindex = fullabspath.index(dirname(abspath(mp_filename)))
