--- CMakeLists.txt~	2011-01-27 07:53:19.783350535 -0800
+++ CMakeLists.txt	2011-01-27 08:02:52.133350536 -0800
@@ -1,6 +1,11 @@
 project(konq-plugins)
 
-include_directories (${KDEBASE_APPS_SOURCE_DIR}/lib/konq)
+find_package(KDE4 REQUIRED)
+include(KDE4Defaults)
+
+add_definitions (${QT_DEFINITIONS} ${KDE4_DEFINITIONS})
+
+include_directories (${KDEBASE_APPS_SOURCE_DIR}/lib/konq ${KDE4_INCLUDES})
 
 add_subdirectory( khtmlsettingsplugin )
 add_subdirectory( kimgalleryplugin )
