diff -up mozilla-1.9.2/config/autoconf.mk.in.pk mozilla-1.9.2/config/autoconf.mk.in
--- mozilla-1.9.2/config/autoconf.mk.in.pk	2009-11-09 01:33:58.000000000 +0100
+++ mozilla-1.9.2/config/autoconf.mk.in	2009-11-12 12:49:28.000000000 +0100
@@ -58,14 +58,14 @@ MOZ_PKG_SPECIAL = @MOZ_PKG_SPECIAL@
 prefix		= @prefix@
 exec_prefix	= @exec_prefix@
 bindir		= @bindir@
-includedir	= @includedir@/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
+includedir	= @includedir@/$(MOZ_APP_NAME)-sdk-$(MOZ_APP_VERSION)
 libdir		= @libdir@
 datadir		= @datadir@
 mandir		= @mandir@
-idldir		= $(datadir)/idl/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
+idldir		= $(datadir)/idl/$(MOZ_APP_NAME)-sdk-$(MOZ_APP_VERSION)
 
 installdir	= $(libdir)/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
-sdkdir		= $(libdir)/$(MOZ_APP_NAME)-devel-$(MOZ_APP_VERSION)
+sdkdir		= $(libdir)/$(MOZ_APP_NAME)-sdk-$(MOZ_APP_VERSION)
 
 DIST		= $(DEPTH)/dist
 LIBXUL_SDK      = @LIBXUL_SDK@
diff -up mozilla-1.9.2/xulrunner/installer/libxul-embedding.pc.in.pk mozilla-1.9.2/xulrunner/installer/libxul-embedding.pc.in
--- mozilla-1.9.2/xulrunner/installer/libxul-embedding.pc.in.pk	2009-11-09 01:34:25.000000000 +0100
+++ mozilla-1.9.2/xulrunner/installer/libxul-embedding.pc.in	2009-11-12 13:09:18.000000000 +0100
@@ -6,5 +6,6 @@ idldir=%idldir%
 Name: libxul-embedding
 Description: Static library for version-independent embedding of the Mozilla runtime
 Version: %MOZILLA_VERSION%
+Requires: %NSPR_NAME% >= %NSPR_VERSION%
 Libs: -L${sdkdir}/lib -lxpcomglue
 Cflags: -DXPCOM_GLUE -I${includedir} %WCHAR_CFLAGS%
diff -up mozilla-1.9.2/xulrunner/installer/libxul.pc.in.pk mozilla-1.9.2/xulrunner/installer/libxul.pc.in
--- mozilla-1.9.2/xulrunner/installer/libxul.pc.in.pk	2009-11-09 01:34:25.000000000 +0100
+++ mozilla-1.9.2/xulrunner/installer/libxul.pc.in	2009-11-12 13:10:03.000000000 +0100
@@ -2,6 +2,7 @@ prefix=%prefix%
 sdkdir=%sdkdir%
 includedir=%includedir%
 idldir=%idldir%
+libdir=%libdir%
 
 Name: libxul
 Description: The Mozilla Runtime and Embedding Engine
diff -up mozilla-1.9.2/xulrunner/installer/Makefile.in.pk mozilla-1.9.2/xulrunner/installer/Makefile.in
--- mozilla-1.9.2/xulrunner/installer/Makefile.in.pk	2009-11-12 12:49:28.000000000 +0100
+++ mozilla-1.9.2/xulrunner/installer/Makefile.in	2009-11-12 12:49:28.000000000 +0100
@@ -123,6 +123,7 @@ endif
 	-e "s|%includedir%|$(includedir)|" \
 	-e "s|%idldir%|$(idldir)|" \
 	-e "s|%sdkdir%|$(sdkdir)|" \
+	-e "s|%libdir%|$(installdir)|" \
 	-e "s|%MOZ_APP_NAME%|$(MOZ_APP_NAME)|" \
 	-e "s|%MOZILLA_VERSION%|$(MOZ_APP_VERSION)|" \
 	-e "s|%WCHAR_CFLAGS%|$(WCHAR_CFLAGS)|" \
diff -up mozilla-1.9.2/xulrunner/installer/mozilla-gtkmozembed-embedding.pc.in.pk mozilla-1.9.2/xulrunner/installer/mozilla-gtkmozembed-embedding.pc.in
--- mozilla-1.9.2/xulrunner/installer/mozilla-gtkmozembed-embedding.pc.in.pk	2009-11-09 01:34:25.000000000 +0100
+++ mozilla-1.9.2/xulrunner/installer/mozilla-gtkmozembed-embedding.pc.in	2009-11-12 12:49:28.000000000 +0100
@@ -1,5 +1,6 @@
 prefix=%prefix%
 sdkdir=%sdkdir%
+libdir=%libdir%
 includedir=%includedir%
 
 Name: mozilla-gtkembedmoz
diff -up mozilla-1.9.2/xulrunner/installer/mozilla-gtkmozembed.pc.in.pk mozilla-1.9.2/xulrunner/installer/mozilla-gtkmozembed.pc.in
--- mozilla-1.9.2/xulrunner/installer/mozilla-gtkmozembed.pc.in.pk	2009-11-09 01:34:25.000000000 +0100
+++ mozilla-1.9.2/xulrunner/installer/mozilla-gtkmozembed.pc.in	2009-11-12 12:49:28.000000000 +0100
@@ -1,5 +1,6 @@
 prefix=%prefix%
 sdkdir=%sdkdir%
+libdir=%libdir%
 includedir=%includedir%
 
 Name: mozilla-gtkembedmoz
