diff -u -r wpebackend-fdo-1.4.0/src/view-backend-exportable-fdo-egl.cpp wpebackend-fdo-1.4.0-glvnd/src/view-backend-exportable-fdo-egl.cpp
--- wpebackend-fdo-1.4.0/src/view-backend-exportable-fdo-egl.cpp	2019-09-18 12:32:18.000000000 +0000
+++ wpebackend-fdo-1.4.0-glvnd/src/view-backend-exportable-fdo-egl.cpp	2019-11-23 17:35:13.156702854 +0000
@@ -28,6 +28,7 @@
 #include "ws.h"
 #include <EGL/egl.h>
 #include <EGL/eglext.h>
+#include <EGL/eglmesaext.h>
 #include <cassert>
 #include <list>
 #include <wpe-fdo/view-backend-exportable-egl.h>
diff -u -r wpebackend-fdo-1.4.0/src/ws.cpp wpebackend-fdo-1.4.0-glvnd/src/ws.cpp
--- wpebackend-fdo-1.4.0/src/ws.cpp	2019-09-18 12:32:18.000000000 +0000
+++ wpebackend-fdo-1.4.0-glvnd/src/ws.cpp	2019-11-23 17:34:59.746513647 +0000
@@ -27,6 +27,7 @@
 
 #include <EGL/egl.h>
 #include <EGL/eglext.h>
+#include <EGL/eglmesaext.h>
 #include "linux-dmabuf/linux-dmabuf.h"
 #include "bridge/wpe-bridge-server-protocol.h"
 #include <cassert>
