From 6219ae62e7b6ce7361e7ec9301171ee1e8ba6056 Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <jan.steffens@gmail.com>
Date: Tue, 10 Sep 2019 20:27:44 +0000
Subject: [PATCH 7/8] meson: Add missing prefix to system path defines

---
 p11-kit/meson.build | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/p11-kit/meson.build b/p11-kit/meson.build
index 48a3c75..b0d965b 100644
--- a/p11-kit/meson.build
+++ b/p11-kit/meson.build
@@ -24,12 +24,12 @@ libp11_kit_internal_sources += custom_target('generate virtual-fixed-generated.h
 libp11_kit_internal_sources += 'virtual.c'
 
 libp11_kit_internal_c_args = [
-  '-DP11_SYSTEM_CONFIG_FILE="@0@"'.format(p11_system_config_file),
-  '-DP11_SYSTEM_CONFIG_MODULES="@0@"'.format(p11_system_config_modules),
-  '-DP11_PACKAGE_CONFIG_MODULES="@0@"'.format(p11_package_config_modules),
+  '-DP11_SYSTEM_CONFIG_FILE="@0@"'.format(prefix / p11_system_config_file),
+  '-DP11_SYSTEM_CONFIG_MODULES="@0@"'.format(prefix / p11_system_config_modules),
+  '-DP11_PACKAGE_CONFIG_MODULES="@0@"'.format(prefix / p11_package_config_modules),
   '-DP11_USER_CONFIG_FILE="@0@"'.format(p11_user_config_file),
   '-DP11_USER_CONFIG_MODULES="@0@"'.format(p11_user_config_modules),
-  '-DP11_MODULE_PATH="@0@"'.format(p11_module_path)
+  '-DP11_MODULE_PATH="@0@"'.format(prefix / p11_module_path)
 ]
 
 libp11_kit_internal = static_library('p11-kit-internal',
-- 
2.23.0

