diff --git a/libs/plasmagenericshell/panelshadows.cpp b/libs/plasmagenericshell/panelshadows.cpp
index ee55571..2f45153 100644
--- a/libs/plasmagenericshell/panelshadows.cpp
+++ b/libs/plasmagenericshell/panelshadows.cpp
@@ -122,7 +122,7 @@ void PanelShadows::Private::initPixmap(const QString &element)
 {
 #ifdef Q_WS_X11
     QPixmap pix = q->pixmap(element);
-    if (pix.handle() == 0) {
+    if (!pix.isNull() && pix.handle() == 0) {
         Pixmap xPix = XCreatePixmap(QX11Info::display(), QX11Info::appRootWindow(), pix.width(), pix.height(), 32);
         QPixmap tempPix = QPixmap::fromX11Pixmap(xPix, QPixmap::ExplicitlyShared);
         tempPix.fill(Qt::transparent);
