--- branches/KDE/4.4/kdebase/workspace/powerdevil/daemon/PowerDevilDaemon.cpp	2010/04/15 18:15:29	1115242
+++ branches/KDE/4.4/kdebase/workspace/powerdevil/daemon/PowerDevilDaemon.cpp	2010/04/15 18:18:39	1115243
@@ -985,10 +985,6 @@
 void PowerDevilDaemon::emitNotification(const QString &evid, const QString &message,
                                         const char *slot, const QString &iconname)
 {
-    if (slot) {
-        QTimer::singleShot(0, this, slot);
-    }
-
     if (!slot) {
         KNotification::event(evid, message, KIcon(iconname).pixmap(20, 20),
                              0, KNotification::CloseOnTimeout, d->applicationData);
