--- kdebase-workspace-4.6.3/solid/CMakeLists.txt~	2011-04-29 03:43:20.321994578 -0700
+++ kdebase-workspace-4.6.3/solid/CMakeLists.txt	2011-04-29 03:43:33.271995023 -0700
@@ -17,7 +17,7 @@
     add_subdirectory(networkmanager-0.6)
   endif (NM_0_6)
 
-  macro_ensure_version_range("0.7.0" ${NETWORKMANAGER_VERSION} "0.9.0" NM_0_7)
+  macro_ensure_version("0.7.0" ${NETWORKMANAGER_VERSION} NM_0_7)
   if (NM_0_7)
     message(STATUS "Will build networkmanager-0.7")
     set(HAVE_NETWORKMANAGER TRUE)
--- kdebase-workspace-4.6.3/solid/networkmanager-0.7/manager.cpp~	2011-04-29 03:32:33.838644845 -0700
+++ kdebase-workspace-4.6.3/solid/networkmanager-0.7/manager.cpp	2011-04-29 03:40:52.308657544 -0700
@@ -118,16 +118,16 @@
     uint deviceType = devIface.deviceType();
     NMNetworkInterface * createdInterface = 0;
     switch ( deviceType ) {
-        case DEVICE_TYPE_802_3_ETHERNET:
+        case NM_DEVICE_TYPE_ETHERNET:
             createdInterface = new NMWiredNetworkInterface(uni, this, 0); // these are deleted by the frontend manager
             break;
-        case DEVICE_TYPE_802_11_WIRELESS:
+        case NM_DEVICE_TYPE_WIFI:
             createdInterface = new NMWirelessNetworkInterface(uni, this, 0);
             break;
-        case DEVICE_TYPE_GSM:
+        case 3:
             createdInterface = new NMGsmNetworkInterface(uni, this, 0);
             break;
-        case DEVICE_TYPE_CDMA:
+        case 4:
             createdInterface = new NMCdmaNetworkInterface(uni, this, 0);
             break;
             /*
