commit 6878e972b87e9ef34617c6280eda8ac5d7113a29
Author: Eike Hein <hein@kde.org>
Date:   Sun Aug 25 16:38:15 2013 +0200

    Don't attempt to publish geometry for launcher items.
    
    CCBUG:322283

diff --git a/plasma/desktop/applets/tasks/package/contents/code/tools.js b/plasma/desktop/applets/tasks/package/contents/code/tools.js
index 53e5563..028dd2c 100644
--- a/plasma/desktop/applets/tasks/package/contents/code/tools.js
+++ b/plasma/desktop/applets/tasks/package/contents/code/tools.js
@@ -100,7 +100,7 @@ function publishIconGeometries(taskItems) {
             for (j = 0; j < taskIdList.length; ++j) {
                 tasks.itemGeometryChanged(taskIdList[j].itemId, task.x, task.y, task.width, task.height);
             }
-        } else {
+        } else if (!task.isLauncher) {
             tasks.itemGeometryChanged(task.itemId, task.x, task.y, task.width, task.height);
         }
     }
