--- kdebase-workspace-4.4.3/powerdevil/daemon/PowerDevilDaemon.cpp.orig	2010-04-29 21:54:40.000000000 +0200
+++ kdebase-workspace-4.4.3/powerdevil/daemon/PowerDevilDaemon.cpp	2010-05-09 17:06:54.000000000 +0200
@@ -571,7 +571,7 @@
 
 void PowerDevilDaemon::buttonPressed(int but)
 {
-    if (!checkIfCurrentSessionActive() || d->screenSaverIface->GetActive()) {
+    if (!checkIfCurrentSessionActive()) {
         return;
     }
 

