diff --git a/filter/foomatic-rip/options.c b/filter/foomatic-rip/options.c
index 14eddc75..0f46855a 100644
--- a/filter/foomatic-rip/options.c
+++ b/filter/foomatic-rip/options.c
@@ -1031,12 +1031,10 @@ int option_set_value(option_t *opt, int optionset, const char *value)
         /* TODO only set the changed option, not all of them */
         choice = option_find_choice(fromopt, 
                                     option_get_value(fromopt, optionset));
-
         composite_set_values(fromopt, optionset, choice->command);
-    }
-    else {
+	free(newvalue);
+    } else
         val->value = newvalue;
-    }
 
     if (option_is_composite(opt)) {
         /* set dependent values */
@@ -1044,7 +1042,6 @@ int option_set_value(option_t *opt, int optionset, const char *value)
         if (choice && !isempty(choice->command))
             composite_set_values(opt, optionset, choice->command);
     }
-    free(newvalue);
     return 1;
 }
 

