diff -upr slim-1.4.0.orig/slim.conf slim-1.4.0/slim.conf
--- slim-1.4.0.orig/slim.conf	2023-01-14 13:47:23.000000000 +0200
+++ slim-1.4.0/slim.conf	2023-05-14 11:09:38.095795009 +0300
@@ -1,7 +1,7 @@
 # Path, X server and arguments (if needed)
 # Note: -xauth $authfile is automatically appended, vt07 appended if no
 # vtxx argument given.
-default_path        /bin:/usr/bin:/usr/local/bin
+default_path        /usr/local/sbin:/usr/local/bin:/usr/bin
 default_xserver     /usr/bin/X
 #xserver_arguments   -dpi 75
 xserver_arguments   -nolisten tcp -deferglyphs 16
