diff -upr slim-1.4.0.orig/slim.conf slim-1.4.0/slim.conf
--- slim-1.4.0.orig/slim.conf	2023-05-14 11:57:33.114337665 +0300
+++ slim-1.4.0/slim.conf	2023-05-14 11:58:16.718957988 +0300
@@ -43,8 +43,6 @@ login_cmd           exec /bin/bash -logi
 #
 # sessionstart_cmd	some command
 # sessionstop_cmd	some command
-sessionstart_cmd	/usr/bin/sessreg -a -l "$DISPLAY" %user
-sessionstop_cmd     /usr/bin/sessreg -d -l "$DISPLAY" %user
 
 # Start in daemon mode. Valid values: yes | no
 # Note that this can be overridden by the command line options "-d" and "-n"
