Package:     hylafax
Subject:     Make hylafax compile using gcc-7
Author:      Giuseppe Sacco <giuseppe at eppesuigoccas.homedns.org>
Bug-Debian:  https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=853448
Forwarded:   yes
Last-Update: 2017-08-28

Index: hylafax-6.0.6/faxd/ModemConfig.c++
===================================================================
--- hylafax-6.0.6.orig/faxd/ModemConfig.c++
+++ hylafax-6.0.6/faxd/ModemConfig.c++
@@ -817,4 +817,4 @@ ModemConfig::setConfigItem(const char* t
 }
 #undef N
 
-fxIMPLEMENT_ObjArray(IDConfArray, id_config);
+fxIMPLEMENT_ObjArray(IDConfArray, id_config)
Index: hylafax-6.0.6/libhylafax/FaxRecvInfo.c++
===================================================================
--- hylafax-6.0.6.orig/libhylafax/FaxRecvInfo.c++
+++ hylafax-6.0.6/libhylafax/FaxRecvInfo.c++
@@ -112,7 +112,7 @@ FaxRecvInfo::decode(const char* cp)
     if (cp == NULL || cp[1] != ',' || cp[2] != '"')
 	return (false);
     u_int i = 0;
-    while (cp+2 != '\0') {
+    while (*(cp+2) != '\0') {
 	callid[i] = cp+3;		// +1 for "/+1 for ,/+1 for "
 	if (*cp == '\"') break;
 	callid[i].resize(callid[i].next(0,'"'));

