--- src/kdegraphics-3.4.91/kamera/kcontrol/Makefile.old	2005-09-14 14:05:22.000000000 +0200
+++ src/kdegraphics-3.4.91/kamera/kcontrol/Makefile	2005-09-14 14:06:58.000000000 +0200
@@ -165,7 +165,7 @@
 GLLIB = -lGLU -lGL $(LIB_X11)
 GMSGFMT = /usr/bin/msgfmt
 GPHOTO2_INCS = -I/usr/include/gphoto2
-GPHOTO2_LIBS = -L/usr/lib/ -lgphoto2
+GPHOTO2_LIBS = -L/usr/lib/ -lgphoto2 -lgphoto2_port
 GREP = /bin/grep
 HAVE_GCC_VISIBILITY = 0
 IMLIB_CFLAGS =  
--- src/kdegraphics-3.4.91/kamera/Makefile.old	2005-09-14 14:11:37.000000000 +0200
+++ src/kdegraphics-3.4.91/kamera/Makefile	2005-09-14 14:11:53.000000000 +0200
@@ -130,7 +130,7 @@
 GLLIB = -lGLU -lGL $(LIB_X11)
 GMSGFMT = /usr/bin/msgfmt
 GPHOTO2_INCS = -I/usr/include/gphoto2
-GPHOTO2_LIBS = -L/usr/lib/ -lgphoto2
+GPHOTO2_LIBS = -L/usr/lib/ -lgphoto2 -lgphoto_port
 GREP = /bin/grep
 HAVE_GCC_VISIBILITY = 0
 IMLIB_CFLAGS =  
--- src/kdegraphics-3.4.91/kamera/kioslave/Makefile.old	2005-09-14 14:12:29.000000000 +0200
+++ src/kdegraphics-3.4.91/kamera/kioslave/Makefile	2005-09-14 14:12:46.000000000 +0200
@@ -162,7 +162,7 @@
 GLLIB = -lGLU -lGL $(LIB_X11)
 GMSGFMT = /usr/bin/msgfmt
 GPHOTO2_INCS = -I/usr/include/gphoto2
-GPHOTO2_LIBS = -L/usr/lib/ -lgphoto2
+GPHOTO2_LIBS = -L/usr/lib/ -lgphoto2 -lgphoto2_port
 GREP = /bin/grep
 HAVE_GCC_VISIBILITY = 0
 IMLIB_CFLAGS =  
