From b12040aeab5fbaf02677571db1d8bf1995bd5ee0 Mon Sep 17 00:00:00 2001
From: Nikos Mavrogiannopoulos <nmav@gnutls.org>
Date: Sun, 2 Jun 2013 12:10:06 +0200
Subject: [PATCH] Avoid comparing the expiration date to prevent false positive error in 32-bit systems.

---
 tests/cert-tests/pem-decoding |    6 ++++--
 1 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/tests/cert-tests/pem-decoding b/tests/cert-tests/pem-decoding
index fe769ec..f8c6372 100755
--- a/tests/cert-tests/pem-decoding
+++ b/tests/cert-tests/pem-decoding
@@ -61,7 +61,9 @@ if test "$rc" != "0"; then
   exit $rc
 fi
 
-diff $srcdir/complex-cert.pem tmp-pem.pem
+cat $srcdir/complex-cert.pem |grep -v "Not After:" >tmp1
+cat $srcdir/tmp-pem.pem |grep -v "Not After:" >tmp2
+diff tmp1 tmp2
 rc=$?
 
 if test "$rc" != "0"; then
@@ -69,6 +71,6 @@ if test "$rc" != "0"; then
   exit $rc
 fi
 
-rm -f tmp-pem.pem
+rm -f tmp-pem.pem tmp1 tmp2
 
 exit 0
-- 
1.7.1
