diff --git a/lib/includes/gnutls/openpgp.h b/lib/includes/gnutls/openpgp.h
index 4b0d853..6dfa786 100644
--- a/lib/includes/gnutls/openpgp.h
+++ b/lib/includes/gnutls/openpgp.h
@@ -53,7 +53,8 @@ extern "C"
     GNUTLS_OPENPGP_FMT_BASE64
   } gnutls_openpgp_crt_fmt_t;
 
-  typedef unsigned char gnutls_openpgp_keyid_t[8];
+#define GNUTLS_OPENPGP_KEYID_SIZE 8
+  typedef unsigned char gnutls_openpgp_keyid_t[GNUTLS_OPENPGP_KEYID_SIZE];
 
 /* gnutls_openpgp_cert_t should be defined in gnutls.h
  */
diff --git a/lib/openpgp/gnutls_openpgp.c b/lib/openpgp/gnutls_openpgp.c
index ba7cd27..5917447 100644
--- a/lib/openpgp/gnutls_openpgp.c
+++ b/lib/openpgp/gnutls_openpgp.c
@@ -321,7 +321,7 @@ gnutls_certificate_set_openpgp_key_file (gnutls_certificate_credentials_t res,
 static int
 get_keyid (gnutls_openpgp_keyid_t keyid, const char *str)
 {
-  size_t keyid_size = sizeof (keyid);
+  size_t keyid_size = GNUTLS_OPENPGP_KEYID_SIZE;
 
   if (strlen (str) != 16)
     {
@@ -744,7 +744,7 @@ _gnutls_openpgp_crt_to_gcert (gnutls_cert * gcert, gnutls_openpgp_crt_t cert)
 
       _gnutls_debug_log
         ("Importing Openpgp cert and using openpgp sub key: %s\n",
-         _gnutls_bin2hex (keyid, sizeof (keyid), err_buf, sizeof (err_buf),
+         _gnutls_bin2hex (keyid, GNUTLS_OPENPGP_KEYID_SIZE, err_buf, sizeof (err_buf),
                           NULL));
 
       KEYID_IMPORT (kid32, keyid);
@@ -762,7 +762,7 @@ _gnutls_openpgp_crt_to_gcert (gnutls_cert * gcert, gnutls_openpgp_crt_t cert)
       gnutls_openpgp_crt_get_subkey_usage (cert, idx, &gcert->key_usage);
       gcert->use_subkey = 1;
 
-      memcpy (gcert->subkey_id, keyid, sizeof (keyid));
+      memcpy (gcert->subkey_id, keyid, GNUTLS_OPENPGP_KEYID_SIZE);
 
       ret =
         _gnutls_openpgp_crt_get_mpis (cert, kid32, gcert->params,
diff --git a/lib/openpgp/pgp.c b/lib/openpgp/pgp.c
index 229b69d..77a931d 100644
--- a/lib/openpgp/pgp.c
+++ b/lib/openpgp/pgp.c
@@ -1568,7 +1568,7 @@ gnutls_openpgp_crt_get_preferred_key_id (gnutls_openpgp_crt_t key,
       return GNUTLS_E_INVALID_REQUEST;
     }
 
-  memcpy (keyid, key->preferred_keyid, sizeof (gnutls_openpgp_keyid_t));
+  memcpy (keyid, key->preferred_keyid, GNUTLS_OPENPGP_KEYID_SIZE);
 
   return 0;
 }
@@ -1606,7 +1606,7 @@ gnutls_openpgp_crt_set_preferred_key_id (gnutls_openpgp_crt_t key,
     }
 
   key->preferred_set = 1;
-  memcpy (key->preferred_keyid, keyid, sizeof (gnutls_openpgp_keyid_t));
+  memcpy (key->preferred_keyid, keyid, GNUTLS_OPENPGP_KEYID_SIZE);
 
   return 0;
 }
diff --git a/lib/openpgp/privkey.c b/lib/openpgp/privkey.c
index 4b26a8c..7fef5ad 100644
--- a/lib/openpgp/privkey.c
+++ b/lib/openpgp/privkey.c
@@ -1186,7 +1186,7 @@ gnutls_openpgp_privkey_get_preferred_key_id (gnutls_openpgp_privkey_t key,
       return GNUTLS_E_INVALID_REQUEST;
     }
 
-  memcpy (keyid, key->preferred_keyid, sizeof (gnutls_openpgp_keyid_t));
+  memcpy (keyid, key->preferred_keyid, GNUTLS_OPENPGP_KEYID_SIZE);
 
   return 0;
 }
@@ -1224,7 +1224,7 @@ gnutls_openpgp_privkey_set_preferred_key_id (gnutls_openpgp_privkey_t key,
     }
 
   key->preferred_set = 1;
-  memcpy (key->preferred_keyid, keyid, sizeof (gnutls_openpgp_keyid_t));
+  memcpy (key->preferred_keyid, keyid, GNUTLS_OPENPGP_KEYID_SIZE);
 
   return 0;
 }

