Index: qscintilla2-2.8.3+dfsg/Python/configure.py
===================================================================
--- qscintilla2-2.8.3+dfsg.orig/Python/configure.py	2014-07-05 10:05:48.614827688 -0400
+++ qscintilla2-2.8.3+dfsg/Python/configure.py	2014-07-05 10:05:48.610827688 -0400
@@ -1434,7 +1434,10 @@
 
     libs = qmake_config.get('LIBS')
     if libs:
-        pro.write('LIBS += %s\n' % libs)
+        if target_config.pyqt_package == 'PyQt5':
+            pro.write('LIBS += %s -lqt5scintilla2\n' % libs)
+        else:
+            pro.write('LIBS += %s -lqscintilla2\n' % libs)
 
     if not opts.static:
         pro.write('''
