diff -Nur zabbix-4.0.0-orig/configure.ac zabbix-4.0.0/configure.ac
--- zabbix-4.0.0-orig/configure.ac	2018-10-01 10:36:02.000000000 +0200
+++ zabbix-4.0.0/configure.ac	2018-10-02 11:44:59.818887035 +0200
@@ -1714,25 +1714,6 @@
 dnl *                                                               *
 dnl *****************************************************************
 
-if test -d tests; then
-	AC_CONFIG_FILES(
-	tests/Makefile
-	tests/libs/Makefile
-	tests/libs/zbxcommon/Makefile
-	tests/libs/zbxconf/Makefile
-	tests/libs/zbxdbcache/Makefile
-	tests/libs/zbxdbhigh/Makefile
-	tests/libs/zbxhistory/Makefile
-	tests/libs/zbxjson/Makefile
-	tests/libs/zbxsysinfo/Makefile
-	tests/libs/zbxsysinfo/linux/Makefile
-	tests/libs/zbxsysinfo/common/Makefile
-	tests/libs/zbxcommshigh/Makefile
-	tests/libs/zbxalgo/Makefile
-	)
-	AC_DEFINE([HAVE_TESTS], [1], ["Define to 1 if tests directory is present"])
-fi
-
 AC_OUTPUT([
 	Makefile
 	database/Makefile
