Index: src/cheese-window.c
===================================================================
--- src/cheese-window.c	(.../trunk)	(revision 580)
+++ src/cheese-window.c	(.../branches/gnome-2-22)	(working copy)
@@ -1045,6 +1045,7 @@
   gtk_label_set_text_with_mnemonic (GTK_LABEL (cheese_window->label_take_photo), str);
   g_free (str);
   gtk_label_set_use_markup (GTK_LABEL (cheese_window->label_take_photo), TRUE);
+  gtk_widget_set_sensitive (GTK_WIDGET (cheese_window->take_picture), FALSE);
 
   cheese_window->thumb_view = cheese_thumb_view_new ();
   gtk_container_add (GTK_CONTAINER (cheese_window->thumb_scrollwindow), cheese_window->thumb_view);
@@ -1177,6 +1178,8 @@
   cheese_window->webcam = cheese_webcam_new (cheese_window->screen, webcam_device);
   g_free (webcam_device);
 
+  cheese_webcam_setup (cheese_window->webcam);
+
   g_signal_connect (cheese_window->webcam, "photo-saved",
                     G_CALLBACK (cheese_window_photo_saved_cb), cheese_window);
   g_signal_connect (cheese_window->webcam, "video-saved",
@@ -1190,6 +1193,7 @@
   gtk_notebook_set_current_page (GTK_NOTEBOOK(cheese_window->notebook), 0);
   ephy_spinner_stop (EPHY_SPINNER (cheese_window->throbber));
   gdk_threads_leave ();
+  gtk_widget_set_sensitive (GTK_WIDGET (cheese_window->take_picture), TRUE);
 }
 
 void
Index: src/cheese-webcam.c
===================================================================
--- src/cheese-webcam.c	(.../trunk)	(revision 580)
+++ src/cheese-webcam.c	(.../branches/gnome-2-22)	(working copy)
@@ -483,9 +483,9 @@
     pipeline = gst_parse_launch (pipeline_desc, &err);
     if ((pipeline != NULL) && (err == NULL))
     {
-      /* Start the pipeline and wait for max. 3 seconds for it to start up */
+      /* Start the pipeline and wait for max. 10 seconds for it to start up */
       gst_element_set_state (pipeline, GST_STATE_PLAYING);
-      ret = gst_element_get_state (pipeline, NULL, NULL, 3 * GST_SECOND);
+      ret = gst_element_get_state (pipeline, NULL, NULL, 10 * GST_SECOND);
 
       /* Check if any error messages were posted on the bus */
       bus = gst_element_get_bus (pipeline);
@@ -562,11 +562,10 @@
   CheeseWebcamPrivate* priv = CHEESE_WEBCAM_GET_PRIVATE (webcam);
   GError *err = NULL;
   char *webcam_input;
-  
+
   if (priv->num_webcam_devices == 0)
   {
-    priv->webcam_source_bin = gst_parse_bin_from_description ("videotestsrc name=video_source",
-                                                              TRUE, &err);
+    goto fallback;
   }
   else
   {
@@ -579,7 +578,7 @@
     selected_device = 0;
     for (i = 1; i < priv->num_webcam_devices ; i++)
     {
-	if (strcmp (priv->webcam_devices[i].video_device, priv->device_name) == 0)
+	if (g_strcmp0 (priv->webcam_devices[i].video_device, priv->device_name) == 0)
           selected_device = i;
     }
     CheeseWebcamDevice *selected_webcam = &(priv->webcam_devices[selected_device]);
@@ -588,12 +587,16 @@
     format = &(g_array_index (selected_webcam->video_formats, CheeseVideoFormat, 0));
     for (i = 1; i < selected_webcam->num_video_formats; i++)
     {
-      
+
       if (g_array_index (selected_webcam->video_formats, CheeseVideoFormat, i).width >  format->width)
       {
         format = &(g_array_index (selected_webcam->video_formats, CheeseVideoFormat, i));
       }
     }
+
+    if (format == NULL)
+      goto fallback;
+
     /* Select the highest framerate up to 30 Hz*/
     framerate_numerator = 1;
     framerate_denominator = 1;
@@ -620,15 +623,29 @@
 
     priv->webcam_source_bin = gst_parse_bin_from_description (webcam_input,
                                                               TRUE, &err);
-    g_free (webcam_input);    
+    g_free (webcam_input);
+
+    if ( priv->webcam_source_bin == NULL)
+      goto fallback;
   }
+
+  priv->video_source = gst_bin_get_by_name (GST_BIN (priv->webcam_source_bin), "video_source");
+  return TRUE;
+
+fallback:
   if (err != NULL)
   {
     g_error_free (err);
+    err = NULL;
+  }
+
+  priv->webcam_source_bin = gst_parse_bin_from_description ("videotestsrc name=video_source",
+                                                              TRUE, &err);
+  if (err != NULL)
+  {
+    g_error_free (err);
     return FALSE;
   }
-
-  priv->video_source = gst_bin_get_by_name (GST_BIN (priv->webcam_source_bin), "video_source");
   return TRUE;
 }
 
@@ -1081,14 +1098,20 @@
 cheese_webcam_init (CheeseWebcam *webcam)
 {
   CheeseWebcamPrivate* priv = CHEESE_WEBCAM_GET_PRIVATE (webcam);
-  gboolean ok;
 
   priv->is_recording = FALSE;
   priv->pipeline_is_playing = FALSE;
   priv->photo_filename = NULL;
   priv->webcam_devices = NULL;
   priv->device_name = NULL;
+}
 
+void
+cheese_webcam_setup (CheeseWebcam *webcam)
+{
+  CheeseWebcamPrivate* priv = CHEESE_WEBCAM_GET_PRIVATE (webcam);
+  gboolean ok = TRUE;
+
   cheese_webcam_detect_webcam_devices (webcam);
   cheese_webcam_create_video_display_bin (webcam);
   cheese_webcam_create_photo_save_bin (webcam);
@@ -1115,6 +1138,7 @@
   gdk_threads_leave();
 }
 
+
 CheeseWebcam*
 cheese_webcam_new (GtkWidget* video_window, char *webcam_device_name)
 {
@@ -1122,7 +1146,7 @@
   if (webcam_device_name)
   {
     webcam = g_object_new (CHEESE_TYPE_WEBCAM, "video-window", video_window, 
-                           "device_name", webcam_device_name, NULL);
+                           "device-name", webcam_device_name, NULL);
   }
   else
   {
Index: src/cheese-webcam.h
===================================================================
--- src/cheese-webcam.h	(.../trunk)	(revision 580)
+++ src/cheese-webcam.h	(.../branches/gnome-2-22)	(working copy)
@@ -67,6 +67,7 @@
 
 GType		 cheese_webcam_get_type			(void);
 CheeseWebcam 	*cheese_webcam_new 			(GtkWidget *video_window, char *webcam_device_name);
+void 		 cheese_webcam_setup 			(CheeseWebcam *webcam);
 void 		 cheese_webcam_play 			(CheeseWebcam *webcam);
 void 		 cheese_webcam_stop 			(CheeseWebcam *webcam);
 void 		 cheese_webcam_set_effect 		(CheeseWebcam *webcam, CheeseWebcamEffect effect);
Index: ChangeLog
===================================================================
--- ChangeLog	(.../trunk)	(revision 580)
+++ ChangeLog	(.../branches/gnome-2-22)	(working copy)
@@ -1,3 +1,47 @@
+2008-03-20  daniel g. siegel <dgsiegel@gmail.com>
+
+	* data/cheese.schemas.in: 
+	  show the countdown widget by default
+
+2008-03-19  daniel g. siegel <dgsiegel@gmail.com>
+
+	* src/cheese-window.c: 
+	  do not allow to take pictures, while the webcam is initialising, fixes bug #523121
+
+2008-03-17  daniel g. siegel <dgsiegel@gmail.com>
+
+	* src/cheese-webcam.c: 
+	  set the max waiting time for detecting a camera to 10 seconds, fixes bug #520394
+
+2008-03-13  daniel g. siegel <dgsiegel@gmail.com>
+
+	* src/cheese-webcam.c: 
+	  cheese crashes if getting information from a webcam fails, courtesy of Sjoerd Simons, fixes bug #522197
+
+2008-03-13  daniel g. siegel <dgsiegel@gmail.com>
+
+	* src/cheese-webcam.c,
+	  src/cheese-webcam.h,
+	  src/cheese-window.c: 
+	  cheese doesn't open the webcam device set in gconf, courtesy of Sjoerd Simons, fixes bug #522198
+
+2008-03-13  daniel g. siegel <dgsiegel@gmail.com>
+
+	* configure.ac,
+	  README: 
+	  cheese does not list its gnome-vfs dependency, fixes bug #522152, courtesy of Alex Rostovtsev
+
+2008-03-13  daniel g. siegel <dgsiegel@gmail.com>
+
+	* src/cheese-webcam.c: 
+	  cheese crashes while initializing webcam, does strcmp on NULL, fixes bug #522143, courtesy of Alex Rostovtsev
+
+2008-03-11  daniel g. siegel <dgsiegel@gmail.com>
+
+	* configure.ac,
+	  wscript: 
+	  post release version bump
+
 2008-03-10  daniel g. siegel <dgsiegel@gmail.com>
 
 	* help/Makefile.am: 
Index: help/ChangeLog
===================================================================
--- help/ChangeLog	(.../trunk)	(revision 580)
+++ help/ChangeLog	(.../branches/gnome-2-22)	(working copy)
@@ -1,3 +1,7 @@
+2008-03-12  Jorge Gonzalez  <jorgegonz@svn.gnome.org>
+
+	* es/es.po: Updated Spanish translation
+
 2008-03-08  Claude Paroz  <claude@2xlibre.net>
 
 	* fr/fr.po:
Index: help/es/es.po
===================================================================
--- help/es/es.po	(.../trunk)	(revision 580)
+++ help/es/es.po	(.../branches/gnome-2-22)	(working copy)
@@ -1,10 +1,10 @@
-# translation of cheese.help.HEAD.po to Español
+# translation of help.HEAD.po to Español
 # Jorge González <jorgegonz@svn.gnome.org>, 2007, 2008.
 msgid ""
 msgstr ""
-"Project-Id-Version: cheese.help.HEAD\n"
-"POT-Creation-Date: 2008-03-03 23:48+0000\n"
-"PO-Revision-Date: 2008-03-04 00:47+0100\n"
+"Project-Id-Version: help.HEAD\n"
+"POT-Creation-Date: 2008-03-11 04:15+0000\n"
+"PO-Revision-Date: 2008-03-11 07:41+0100\n"
 "Last-Translator: Jorge González <jorgegonz@svn.gnome.org>\n"
 "Language-Team: Español <gnome-es-list@gnome.org>\n"
 "MIME-Version: 1.0\n"
@@ -18,20 +18,20 @@
 #: C/cheese.xml:164(None)
 msgid ""
 "@@image: 'figures/cheese-screenshot.jpg'; "
-"md5=ca4906636e8e73be8c9d4e0cd419595c"
+"md5=e24f99c6208e8556528e347fcf730554"
 msgstr ""
 "@@image: 'figures/cheese-screenshot.jpg'; "
-"md5=ca4906636e8e73be8c9d4e0cd419595c"
+"md5=e24f99c6208e8556528e347fcf730554"
 
 #. When image changes, this message will be marked fuzzy or untranslated for you.
 #. It doesn't matter what you translate it to: it's not used at all.
 #: C/cheese.xml:186(None)
 msgid ""
-"@@image: 'figures/cheese-effects-screenshot.png'; "
-"md5=013a7121d57060807de05e77b2e67a36"
+"@@image: 'figures/cheese-effects-screenshot.jpg'; "
+"md5=0a773e6a421f02d6dbf53259c63a34b0"
 msgstr ""
-"@@image: 'figures/cheese-effects-screenshot.png'; "
-"md5=013a7121d57060807de05e77b2e67a36"
+"@@image: 'figures/cheese-effects-screenshot.jpg'; "
+"md5=0a773e6a421f02d6dbf53259c63a34b0"
 
 #: C/cheese.xml:25(title)
 msgid "<application>Cheese</application> Manual"
@@ -265,8 +265,7 @@
 
 #: C/cheese.xml:155(para)
 msgid "The main screen of Cheese looks like the screen shot below"
-msgstr ""
-"La pantalla principal de Cheese es la que se presenta en una captura debajo"
+msgstr "La pantalla principal de Cheese es la que se presenta en una captura debajo"
 
 #: C/cheese.xml:160(title)
 msgid "Cheese screenshot"
@@ -373,8 +372,7 @@
 
 #: C/cheese.xml:238(para)
 msgid "To take a photo in Cheese, be sure to be in “Photo” mode."
-msgstr ""
-"Para sacar una fotografía en Cheese, asegúrese de estar en el modo «Foto»."
+msgstr "Para sacar una fotografía en Cheese, asegúrese de estar en el modo «Foto»."
 
 #: C/cheese.xml:242(para)
 msgid ""
@@ -406,10 +404,8 @@
 msgstr "Grabar un vídeo"
 
 #: C/cheese.xml:259(para)
-msgid ""
-"To take a video in Cheese, make sure that the “Video” button has been pressed"
-msgstr ""
-"Para grabar un vídeo en Cheese, asegúrese de haber pulsado el botón «Vídeo»"
+msgid "To take a video in Cheese, make sure that the “Video” button has been pressed"
+msgstr "Para grabar un vídeo en Cheese, asegúrese de haber pulsado el botón «Vídeo»"
 
 #: C/cheese.xml:263(para)
 msgid ""
@@ -603,10 +599,8 @@
 "de vídeo y cambie los ajustes apropiados."
 
 #: C/cheese.xml:397(title)
-msgid ""
-"I have a Mac with iSight and a ATI graphics card, and the colors are weird."
-msgstr ""
-"Tengo un Mac con iSight y una tarjeta gráfica ATI y los colores se ven raros."
+msgid "I have a Mac with iSight and a ATI graphics card, and the colors are weird."
+msgstr "Tengo un Mac con iSight y una tarjeta gráfica ATI y los colores se ven raros."
 
 #: C/cheese.xml:398(para)
 msgid ""
@@ -689,10 +683,8 @@
 "información."
 
 #: C/cheese.xml:442(title)
-msgid ""
-"My Quickcam Express doesn't work with <application>Cheese</application>..."
-msgstr ""
-"Mi QuickCam Express no funciona con <application>Cheese</application>..."
+msgid "My Quickcam Express doesn't work with <application>Cheese</application>..."
+msgstr "Mi QuickCam Express no funciona con <application>Cheese</application>..."
 
 #: C/cheese.xml:443(para)
 msgid ""
@@ -778,3 +770,4 @@
 #: C/cheese.xml:0(None)
 msgid "translator-credits"
 msgstr "Jorge González <jorgegonz@svn.gnome.org>, 2007-2008."
+
Index: README
===================================================================
--- README	(.../trunk)	(revision 580)
+++ README	(.../branches/gnome-2-22)	(working copy)
@@ -6,7 +6,8 @@
 gstreamer-backend.
 
 Cheese currently requires:
-  - glib-2.0 >= 2.15.4
+  - glib-2.0 >= 2.15.5
+  - gio-2.0 >= 2.15.5
   - gobject-2.0
   - gtk+-2.0 >= 2.10.0
   - gdk-2.0
Index: configure.ac
===================================================================
--- configure.ac	(.../trunk)	(revision 580)
+++ configure.ac	(.../branches/gnome-2-22)	(working copy)
@@ -1,6 +1,6 @@
 # Process this file with autoconf to produce a configure script.
 
-AC_INIT(cheese, 2.22.0)
+AC_INIT(cheese, 2.22.1)
 
 AM_INIT_AUTOMAKE(AC_PACKAGE_NAME, AC_PACKAGE_VERSION)
 AM_CONFIG_HEADER(cheese-config.h)
@@ -63,6 +63,7 @@
 GCONF_REQUIRED=2.16.0
 GSTREAMER_REQUIRED=0.10.15
 CAIRO_REQUIRED=1.2.4
+GNOME_VFS_REQUIRED=2.18.0
 LIBEBOOK_REQUIRED=1.12.0
 DBUS_REQUIRED=1.0
 HAL_REQUIRED=0.5.9
@@ -78,6 +79,7 @@
   gconf-2.0 >= $GCONF_REQUIRED \
   gstreamer-0.10 >= $GSTREAMER_REQUIRED \
   gstreamer-plugins-base-0.10 >= $GSTREAMER_REQUIRED \
+  gnome-vfs-2.0 >= $GNOME_VFS_REQUIRED \
   libebook-1.2 >= $LIBEBOOK_REQUIRED \
   cairo >= $CAIRO_REQUIRED \
   dbus-1 >= $DBUS_REQUIRED \
Index: data/cheese.schemas.in
===================================================================
--- data/cheese.schemas.in	(.../trunk)	(revision 580)
+++ data/cheese.schemas.in	(.../branches/gnome-2-22)	(working copy)
@@ -5,7 +5,7 @@
 			<applyto>/apps/cheese/countdown</applyto>
 			<owner>cheese</owner>
 			<type>bool</type>
-			<default>0</default>
+			<default>1</default>
 			<locale name="C">
 				<short>Use a countdown</short>
 				<long>Whether a countdown should be used, when taking a photo</long>
Index: po/et.po
===================================================================
--- po/et.po	(.../trunk)	(revision 580)
+++ po/et.po	(.../branches/gnome-2-22)	(working copy)
@@ -5,14 +5,15 @@
 # This file is distributed under the same license as the Cheese package.
 #
 # Ivar Smolin <okul@linux.ee>, 2008.
+# Mattias Põldaru <mahfiaz@gmail.com>, 2008.
 #
 msgid ""
 msgstr ""
-"Project-Id-Version: cheese HEAD\n"
+"Project-Id-Version: cheese 2.22\n"
 "Report-Msgid-Bugs-To: \n"
 "POT-Creation-Date: 2008-02-03 12:42+0000\n"
-"PO-Revision-Date: 2008-02-04 07:25+0200\n"
-"Last-Translator: Ivar Smolin <okul@linux.ee>\n"
+"PO-Revision-Date: 2008-03-12 09:32+0300\n"
+"Last-Translator: Mattias Põldaru <mahfiaz@gmail.com>\n"
 "Language-Team: Estonian <gnome-et@linux.ee>\n"
 "MIME-Version: 1.0\n"
 "Content-Type: text/plain; charset=utf-8\n"
@@ -20,7 +21,7 @@
 
 #: ../data/cheese.desktop.in.in.h:1 ../src/cheese-window.c:576
 msgid "A cheesy program to take photos and videos from your webcam"
-msgstr "Vilets programm sinu veebikaamerast fotode ja videote salvestamiseks"
+msgstr "Programm veebikaamerast pärit fotode ja videote salvestamiseks"
 
 #: ../data/cheese.desktop.in.in.h:2 ../data/cheese.glade.h:1
 #: ../src/cheese.c:84
@@ -180,7 +181,7 @@
 
 #: ../src/cheese-window.c:461
 msgid "Could not set the Account Photo"
-msgstr "Konto pilti pole võimalik seada"
+msgstr "Kontopilti pole võimalik seada"
 
 #: ../src/cheese-window.c:536
 msgid "Unable to open help file for Cheese"
@@ -188,7 +189,9 @@
 
 #: ../src/cheese-window.c:553
 msgid "translator-credits"
-msgstr "Ivar Smolin <okul@linux.ee>, 2008."
+msgstr ""
+"Ivar Smolin <okul@linux.ee>, 2008.\n"
+"Mattias Põldaru <mahfiaz@gmail.com>, 2008."
 
 #: ../src/cheese-window.c:556
 msgid ""
@@ -280,7 +283,7 @@
 
 #: ../src/cheese-window.c:869
 msgid "_Set As Account Photo"
-msgstr "Määra konto _pildiks"
+msgstr "Määra konto_pildiks"
 
 #: ../src/cheese-window.c:873
 msgid "Send by _Mail"
@@ -288,11 +291,11 @@
 
 #: ../src/cheese-window.c:877
 msgid "Export to F-_Spot"
-msgstr "Ekspordi F-_Spot'i saiti"
+msgstr "Ekspordi F-_Spot'i"
 
 #: ../src/cheese-window.c:881
 msgid "Export to _Flickr"
-msgstr "Ekspordi _Filckr'i saiti"
+msgstr "Ekspordi _Flickr'i saiti"
 
 #: ../src/cheese.c:71
 msgid "Be verbose"
@@ -301,6 +304,3 @@
 #: ../src/cheese.c:86
 msgid "- Take photos and videos from your webcam"
 msgstr "- Veebikaamerast fotode ja videote salvestamine"
-
-#~ msgid "Failed to show url %s"
-#~ msgstr "Tõrge URL-i %s näitamisel"
Index: po/ChangeLog
===================================================================
--- po/ChangeLog	(.../trunk)	(revision 580)
+++ po/ChangeLog	(.../branches/gnome-2-22)	(working copy)
@@ -1,3 +1,11 @@
+2008-03-20  Priit Laes  <plaes at svn dot gnome dot org>
+
+	* et.po: Translation updated by Mattias Põldaru
+
+2008-03-11  Priit Laes  <plaes at svn dot gnome dot org>
+
+	* et.po: Translation updated by Ivar Smolin
+
 2008-03-10  Luca Ferretti  <elle.uca@libero.it>
 
 	* it.po: Updated Italian translation by Alessandro Falappa.
