diff -r e303235cefa8 Makefile.am
--- a/Makefile.am	Thu Jan 21 16:11:56 2010 +0000
+++ b/Makefile.am	Fri Jan 22 16:21:38 2010 -0500
@@ -1528,13 +1528,13 @@
 	  -DPLUGIN_VERSION="\"$$plugin_version\"" \
 	  -DMOZILLA_VERSION_COLLAPSED="$(MOZILLA_VERSION_COLLAPSED)" \
 	  $(GTK_CFLAGS) \
-	  $(XULRUNNER_CFLAGS) \
+	  $(MOZILLA_CFLAGS) \
 	  -fPIC -c -o $@ $<
 IcedTeaPlugin.so: IcedTeaPlugin.o
 	$(CXX) $(CXXFLAGS) \
 	  $< \
 	  $(GTK_LIBS) \
-	  $(XULRUNNER_LIBS) \
+	  $(MOZILLA_LIBS) \
 	  -shared -o $@
 
 clean-IcedTeaPlugin:
diff -r e303235cefa8 configure.ac
--- a/configure.ac	Thu Jan 21 16:11:56 2010 +0000
+++ b/configure.ac	Fri Jan 22 16:21:38 2010 -0500
@@ -429,7 +429,7 @@
 dnl FIXME: use unstable
 if test "x${enable_plugin}" = "xyes"
   then
-    PKG_CHECK_MODULES(XULRUNNER, \
+    PKG_CHECK_MODULES(MOZILLA, \
       nspr mozilla-js mozilla-plugin libxul-unstable >= 1.9)
     PKG_CHECK_MODULES(GTK, gtk+-2.0)
     PKG_CHECK_MODULES(GLIB, glib-2.0)
@@ -445,7 +445,7 @@
 if test "x${enable_npplugin}" = "xyes"
   then
     PKG_CHECK_MODULES(MOZILLA, \
-      mozilla-plugin libxul)
+      mozilla-plugin libxul libxul-unstable)
     PKG_CHECK_MODULES(GTK, gtk+-2.0)
     PKG_CHECK_MODULES(GLIB, glib-2.0)
 
@@ -461,7 +461,7 @@
 then
   AC_LANG_PUSH([C++])
   OLDCPPFLAGS="$CPPFLAGS"
-  CPPFLAGS="$CPPFLAGS $XULRUNNER_CFLAGS"
+  CPPFLAGS="$CPPFLAGS $MOZILLA_CFLAGS"
 
   AC_CACHE_CHECK([for xulrunner version], [xulrunner_cv_collapsed_version],
       [AC_RUN_IFELSE(
diff -r e303235cefa8 plugin/icedteanp/IcedTeaNPPlugin.cc
--- a/plugin/icedteanp/IcedTeaNPPlugin.cc	Thu Jan 21 16:11:56 2010 +0000
+++ b/plugin/icedteanp/IcedTeaNPPlugin.cc	Fri Jan 22 16:21:38 2010 -0500
@@ -1336,7 +1336,7 @@
 
   // if proxy info is available, extract it
   nsCString phost;
-  PRint32_t pport;
+  PRInt32 pport;
   nsCString ptype;
 
   info->GetHost(phost);
