--- openjdk/jdk/src/share/native/sun/awt/splashscreen/splashscreen_png.c	2010-01-22 16:40:55.984689900 +0000
+++ openjdk/jdk/src/share/native/sun/awt/splashscreen/splashscreen_png.c.new	2010-01-22 16:59:25.450412461 +0000
@@ -182,7 +182,7 @@
     int success = 0;
 
     stream->read(stream, sig, SIG_BYTES);
-    if (!png_check_sig(sig, SIG_BYTES)) {
+    if (png_sig_cmp(sig, 0, SIG_BYTES)) {
         goto done;
     }
     success = SplashDecodePng(splash, my_png_read_stream, stream);
