From 27fdcb786666e19ea147180820f49acbc17bd694 Mon Sep 17 00:00:00 2001
From: Kohei Yoshida <kohei.yoshida@gmail.com>
Date: Tue, 7 Jul 2015 22:20:34 -0400
Subject: [PATCH] Use the master branch of mdds.

---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 1b09236..aafdc51 100644
--- a/configure.ac
+++ b/configure.ac
@@ -188,7 +188,7 @@ AS_IF([test "x$with_tools" != "xno"], [
 # ============
 # mdds support
 # ============
-PKG_CHECK_MODULES([MDDS],[mdds >= 0.11.0])
+PKG_CHECK_MODULES([MDDS],[mdds-1.0 >= 0.13.0])
 CXXFLAGS="$CXXFLAGS -DMDDS_HASH_CONTAINER_BOOST $MDDS_CFLAGS"
 
 # =================
--
libgit2 0.23.3

