diff -Nur libproxy-0.3.1.orig//configure.ac libproxy-0.3.1/configure.ac
--- libproxy-0.3.1.orig//configure.ac	2009-10-05 21:42:34.000000000 +0300
+++ libproxy-0.3.1/configure.ac	2010-05-05 18:45:29.569107696 +0300
@@ -7,6 +7,7 @@
 ### Checks for programs.
 AC_PROG_CC
 AC_PROG_CPP
+AM_PROG_CC_C_O
 AC_PROG_CXX
 AC_PROG_INSTALL
 AC_PROG_LIBTOOL
@@ -34,8 +35,9 @@
 		[PKG_CHECK_MODULES(mozjs, firefox-js, have_mozjs=yes,
 		[PKG_CHECK_MODULES(mozjs, mozilla-js, have_mozjs=yes, 
 		[PKG_CHECK_MODULES(mozjs, seamonkey-js, have_mozjs=yes,have_mozjs=no)])])])
-PKG_CHECK_MODULES(NetworkManager, NetworkManager dbus-1, 
-					have_networkmanager=yes, have_networkmanager=no)
+PKG_CHECK_MODULES(NetworkManager, NetworkManager,
+                [PKG_CHECK_MODULES(DBUS, dbus-1, have_networkmanager=yes, have_networkmanager=no)],
+		have_networkmanager=no)
 PKG_CHECK_MODULES(QtCore, QtCore, have_qtcore=yes, have_qtcore=no)
 
 # Direct
@@ -124,12 +126,12 @@
             [test x$have_networkmanager = xyes && with_networkmanager=yes])
 if test x$with_networkmanager = xyes; then
 	if test x$have_networkmanager = xyes; then
-		NETWORKMANAGER_CFLAGS="$NetworkManager_CFLAGS"
-		NETWORKMANAGER_LIBS="$NetworkManager_LIBS"
+	        NETWORKMANAGER_CFLAGS="$NetworkManager_CFLAGS $DBUS_CFLAGS"
+	        NETWORKMANAGER_LIBS="$NetworkManager_LIBS $DBUS_LIBS"
 		AC_SUBST(NETWORKMANAGER_CFLAGS)
 		AC_SUBST(NETWORKMANAGER_LIBS)
 	else
-		echo "NetworkManager module requires: NetworkManager!"
+		echo "NetworkManager module requires: NetworkManager and DBus!"
 		exit 1
 	fi
 else
