diff --git a/org.kde.artikulate.desktop b/org.kde.artikulate.desktop
index 50bd00a..62eeb9e 100755
--- a/org.kde.artikulate.desktop
+++ b/org.kde.artikulate.desktop
@@ -4,7 +4,7 @@
 # SPDX-License-Identifier: BSD-2-Clause
 [Desktop Entry]
 Type=Application
-Exec=artikulate -qwindowtitle "%c" %i
+Exec=artikulate -qwindowtitle "%c"
 Icon=artikulate
 X-DocPath=artikulate/index.html
 GenericName=Artikulate Pronunciation Trainer
