diff -Naur xmoto-0.4.2/src/xmscene/BikePlayer.cpp 
xmoto-0.4.2-patched/src/xmscene/BikePlayer.cpp
--- xmoto-0.4.2/src/xmscene/BikePlayer.cpp	2008-07-29 
19:03:01.000000000 +0200
+++ xmoto-0.4.2-patched/src/xmscene/BikePlayer.cpp	2008-07-29 
19:02:37.000000000 +0200
@@ -146,6 +146,7 @@
   m_bFirstPhysicsUpdate = true;
 
   /* Setup ODE */
+  dInitODE();
   m_WorldID = dWorldCreate();
   dWorldSetERP(m_WorldID,PHYS_WORLD_ERP);
   dWorldSetCFM(m_WorldID,PHYS_WORLD_CFM);

