diff -Naur src.old/XMArgs.cpp src/XMArgs.cpp
--- src.old/XMArgs.cpp	2009-05-14 14:20:54.603960980 +0200
+++ src/XMArgs.cpp	2009-05-14 14:21:07.584007699 +0200
@@ -24,6 +24,7 @@
 #include "VFileIO.h"
 #include <sstream>
 #include <stdlib.h>
+#include <cstdio>
 
 XMArguments::XMArguments() {
   m_opt_pack   	       = false;
diff -Naur src.old/helpers/Log.cpp src/helpers/Log.cpp
--- src.old/helpers/Log.cpp	2009-05-14 14:20:54.613925407 +0200
+++ src/helpers/Log.cpp	2009-05-14 14:23:14.092429804 +0200
@@ -23,6 +23,7 @@
 #include "../VFileIO.h"
 #include "assert.h"
 #include <stdarg.h>
+#include <cstdio>
 
 bool  Logger::m_isInitialized = false;
 bool  Logger::m_activ         = true;
diff -Naur src.old/helpers/Text.cpp src/helpers/Text.cpp
--- src.old/helpers/Text.cpp	2009-05-14 14:20:54.613925407 +0200
+++ src/helpers/Text.cpp	2009-05-14 14:21:31.804817862 +0200
@@ -19,6 +19,7 @@
 =============================================================================*/
 
 #include "Text.h"
+#include <cstdio>
 
 std::string txtToLower(const std::string& i_str) {
     std::string v_res;
