diff -aur old/gtk3-im/gtkimcontextgcin.c new/gtk3-im/gtkimcontextgcin.c
--- old/gtk3-im/gtkimcontextgcin.c	2011-09-07 01:15:55.789537810 -0700
+++ new/gtk3-im/gtkimcontextgcin.c	2011-09-07 01:16:14.822829130 -0700
@@ -137,7 +137,7 @@
     return;
 
   if (!context_xim->gcin_ch) {
-    if (!(context_xim->gcin_ch = gcin_im_client_open(GDK_DISPLAY())))
+    if (!(context_xim->gcin_ch = gcin_im_client_open(GDK_DISPLAY_XDISPLAY(gdk_display_get_default()))))
       perror("cannot open gcin_ch");
 #if 1
     context_xim->timeout_handle = 0;
