diff -aur old/gen-dist new/gen-dist
--- old/gen-dist	2011-09-07 10:51:59.214132579 -0700
+++ new/gen-dist	2011-09-07 10:53:27.534010656 -0700
@@ -1,5 +1,5 @@
 GCIN_VERSION="gcin-`cat VERSION.gcin`"
-sh gen-gtk3
+#sh gen-gtk3
 sed -i '/^$/d' Changelog.html
 rm -f tags core.*
 cd ..
diff -aur old/gtk3-im/gtkimcontextgcin.c new/gtk3-im/gtkimcontextgcin.c
--- old/gtk3-im/gtkimcontextgcin.c	2011-09-07 10:51:59.224132565 -0700
+++ new/gtk3-im/gtkimcontextgcin.c	2011-09-07 10:53:09.984034884 -0700
@@ -128,7 +128,7 @@
   GdkWindow *client_window = context_xim->client_window;
   if (!client_window)
     return;
-  GdkScreen *screen = gdk_drawable_get_screen (client_window);
+  GdkScreen *screen = gdk_window_get_screen (client_window);
   if (!screen)
     return;
 
@@ -137,7 +137,7 @@
     return;
 
   if (!context_xim->gcin_ch) {
-    if (!(context_xim->gcin_ch = gcin_im_client_open(GDK_DISPLAY())))
+    if (!(context_xim->gcin_ch = gcin_im_client_open(GDK_DISPLAY_XDISPLAY(gdk_display_get_default()))))
       perror("cannot open gcin_ch");
 #if 1
     context_xim->timeout_handle = 0;
