diff --git i/src/libostree/ostree-sysroot-deploy.c w/src/libostree/ostree-sysroot-deploy.c
index 2dc9f58b..b565129a 100644
--- i/src/libostree/ostree-sysroot-deploy.c
+++ w/src/libostree/ostree-sysroot-deploy.c
@@ -23,7 +23,6 @@
 #include <gio/gunixoutputstream.h>
 #include <glib-unix.h>
 #include <stdint.h>
-#include <sys/mount.h>
 #include <sys/statvfs.h>
 #include <sys/socket.h>
 #include <sys/ioctl.h>
@@ -3504,7 +3503,7 @@ _ostree_sysroot_finalize_staged (OstreeSysroot *self,
     {
       g_autoptr(GError) writing_error = NULL;
       g_assert_cmpint (self->boot_fd, !=, -1);
-      if (!glnx_file_replace_contents_at (self->boot_fd, _OSTREE_FINALIZE_STAGED_FAILURE_PATH, 
+      if (!glnx_file_replace_contents_at (self->boot_fd, _OSTREE_FINALIZE_STAGED_FAILURE_PATH,
                                            (guint8*)finalization_error->message, -1,
                                            0, cancellable, &writing_error))
         {
diff --git i/src/ostree/ot-main.c w/src/ostree/ot-main.c
index b7b50d67..7a4405a5 100644
--- i/src/ostree/ot-main.c
+++ w/src/ostree/ot-main.c
@@ -28,7 +28,6 @@
 #include <string.h>
 #include <sys/statvfs.h>
 #include <sys/mount.h>
-#include <linux/fs.h>
 
 #include "ot-main.h"
 #include "ostree.h"
