From 945e44ffe2296bc37dc58363e5beaf5b4eca73fa Mon Sep 17 00:00:00 2001
From: Alec Leamas <leamas.alec@gmail.com>
Date: Thu, 11 Dec 2014 05:34:26 +0100
Subject: [PATCH 01/13] doc: Add Doxyfile to tarball.

---
 doc/Makefile.am | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/doc/Makefile.am b/doc/Makefile.am
index 8a05a7d..1b6730e 100644
--- a/doc/Makefile.am
+++ b/doc/Makefile.am
@@ -3,6 +3,8 @@ ACLOCAL_AMFLAGS         = -I m4
 
 DISTCLEANFILES          = man/* html/* man-html/*
 
+EXTRA_DIST              = Doxyfile
+
 LIB_SRC                 = $(srcdir)/../lib
 API_SRC                 = $(LIB_SRC)/*.c \
                           $(LIB_SRC)/*.h \
-- 
2.2.2

