Index: ooo/file_ooo.scp
===================================================================
RCS file: /cvs/installation/scp2/source/ooo/file_ooo.scp,v
retrieving revision 1.248.8.1
diff -u -u -r1.248.8.1 file_ooo.scp
--- scp2/source/ooo/file_ooo.scp	30 Jun 2008 17:07:17 -0000	1.248.8.1
+++ scp2/source/ooo/file_ooo.scp	2 Jul 2008 23:14:23 -0000
@@ -2923,6 +2923,8 @@
     Name = "/registry/data/org/openoffice/Office/Linguistic-lingucomponent-thesaurus.xcu";
 End
 
+#ifndef WITHOUT_MYSPELL_DICTS
+
 // Dictionary extensions
 
 File gid_File_Extension_Dictionary_Af
@@ -3078,3 +3080,5 @@
 	Styles = (PACKED);
 	UnixRights = 444;
 End
+
+#endif
Index: ooo/makefile.mk
===================================================================
RCS file: /cvs/installation/scp2/source/ooo/makefile.mk,v
retrieving revision 1.69
diff -u -u -r1.69 makefile.mk
--- scp2/source/ooo/makefile.mk	9 Jun 2008 12:51:56 -0000	1.69
+++ scp2/source/ooo/makefile.mk	2 Jul 2008 23:14:23 -0000
@@ -119,6 +119,10 @@
 SCPDEFS+=-DWITHOUT_MOZILLA
 .ENDIF
 
+.IF "$(WITH_MYSPELL_DICTS)" == "NO"
+SCPDEFS+=-DWITHOUT_MYSPELL_DICTS
+.ENDIF
+
 .IF "$(SYSTEM_MOZILLA)" == "YES"
 SCPDEFS+=-DSYSTEM_MOZILLA
 .ENDIF



