--- munin-1.4.4/Makefile.old	2010-05-13 12:17:36.000000000 -0500
+++ munin-1.4.4/Makefile	2010-05-13 12:19:06.000000000 -0500
@@ -225,9 +225,9 @@
 
 build-common-pre: common/Build
 	cd common && $(PERL) Build code
-	rm -f common/blib/lib/Munin/Common/Defaults.pm
 
 common/blib/lib/Munin/Common/Defaults.pm: common/lib/Munin/Common/Defaults.pm build-common-pre
+	rm -f common/blib/lib/Munin/Common/Defaults.pm
 	$(PERL) -pe 's{(PREFIX     \s+=\s).*}{\1q{$(PREFIX)};}x;   \
                   s{(CONFDIR    \s+=\s).*}{\1q{$(CONFDIR)};}x;     \
                   s{(BINDIR     \s+=\s).*}{\1q{$(BINDIR)};}x;      \
@@ -383,9 +383,6 @@
 build-%: %/Build
 	cd $* && $(PERL) Build
 
-build-common: common/Build
-	cd common && $(PERL) Build && rm -f common/blib/lib/Munin/Common/Defaults.pm && true
-
 
 # BUG: the Build script writes files under PWD when it does "install"
 # can't seem to find a way to persuade it to write otherwhere.
