diff -Naur munin-1.4.5-orig//common/lib/Munin/Common/Defaults.pm munin-1.4.5/common/lib/Munin/Common/Defaults.pm
--- munin-1.4.5-orig//common/lib/Munin/Common/Defaults.pm	2009-11-10 14:13:54.000000000 -0600
+++ munin-1.4.5/common/lib/Munin/Common/Defaults.pm	2011-04-09 15:09:13.215722026 -0500
@@ -30,6 +30,7 @@
 our $MUNIN_MANDIR     = '';
 our $MUNIN_LOGDIR     = "$COMPONENT_ROOT/log/";
 our $MUNIN_STATEDIR   = ''; 
+our $MUNIN_LOCKDIR    = ''; 
 our $MUNIN_USER       = getpwuid $UID;
 our $MUNIN_GROUP      = getgrgid $GID;
 our $MUNIN_PLUGINUSER = getpwuid $UID;
diff -Naur munin-1.4.5-orig//Makefile munin-1.4.5/Makefile
--- munin-1.4.5-orig//Makefile	2009-12-29 10:03:49.000000000 -0600
+++ munin-1.4.5/Makefile	2011-04-09 15:20:44.201098440 -0500
@@ -199,6 +199,7 @@
              -e 's|@@HTMLDIR@@|$(HTMLDIR)|g'                    \
              -e 's|@@DBDIR@@|$(DBDIR)|g'                        \
              -e 's|@@STATEDIR@@|$(STATEDIR)|g'                  \
+             -e 's|@@LOCKDIR@@|$(LOCKDIR)|g'                    \
              -e 's|@@PERL@@|$(PERL)|g'                          \
              -e 's|@@PERLLIB@@|$(PERLLIB)|g'                    \
              -e 's|@@PYTHON@@|$(PYTHON)|g'                      \
@@ -239,6 +240,7 @@
                   s{(HTMLDIR	\s+=\s).*}{\1q{$(HTMLDIR)};}x;     \
                   s{(DBDIR	\s+=\s).*}{\1q{$(DBDIR)};}x;       \
                   s{(STATEDIR	\s+=\s).*}{\1q{$(STATEDIR)};}x;    \
+                  s{(LOCKDIR	\s+=\s).*}{\1q{$(LOCKDIR)};}x;    \
                   s{(PERL	\s+=\s).*}{\1q{$(PERL)};}x;        \
                   s{(PERLLIB	\s+=\s).*}{\1q{$(PERLLIB)};}x;     \
                   s{(PYTHON	\s+=\s).*}{\1q{$(PYTHON)};}x;      \
diff -Naur munin-1.4.5-orig//Makefile.config munin-1.4.5/Makefile.config
--- munin-1.4.5-orig//Makefile.config	2009-11-06 08:35:44.000000000 -0600
+++ munin-1.4.5/Makefile.config	2011-04-09 15:11:15.669416316 -0500
@@ -61,10 +61,13 @@
 # Where Munin should place its logs.
 LOGDIR     = $(PREFIX)/log/munin
 
-# Location of PID files and other statefiles. On the server, must be
-# writable by the user "munin".
+# Location of PID files and other statefiles.
 STATEDIR   = $(DESTDIR)/var/run/munin
 
+# Location of server lock files. On the server, must be writable by the user
+# "munin".
+LOCKDIR    = $(DESTDIR)/var/lock/munin
+
 # The perl interpreter to use
 PERL       := $(shell which perl)
 
diff -upr munin-1.4.5.orig/master/lib/Munin/Master/Config.pm munin-1.4.5/master/lib/Munin/Master/Config.pm
--- munin-1.4.5.orig/master/lib/Munin/Master/Config.pm	2011-06-20 12:08:45.000000000 +0300
+++ munin-1.4.5/master/lib/Munin/Master/Config.pm	2011-06-20 12:09:29.000000000 +0300
@@ -148,7 +148,7 @@ my %booleans = map {$_ => 1} qw(
 		local_address    => 0,
 		logdir           => $Munin::Common::Defaults::MUNIN_LOGDIR,
 		max_processes    => $MAXINT,
-		rundir           => $Munin::Common::Defaults::MUNIN_STATEDIR,
+		rundir           => $Munin::Common::Defaults::MUNIN_LOCKDIR,
 		timeout          => 180,
 		tls              => 'disabled',
 		tls_ca_certificate => "$Munin::Common::Defaults::MUNIN_CONFDIR/cacert.pem",
diff -Naur munin-1.4.5-orig//master/lib/Munin/Master/Utils.pm munin-1.4.5/master/lib/Munin/Master/Utils.pm
--- munin-1.4.5-orig//master/lib/Munin/Master/Utils.pm	2009-12-18 00:25:03.000000000 -0600
+++ munin-1.4.5/master/lib/Munin/Master/Utils.pm	2011-04-09 15:05:57.965152601 -0500
@@ -270,7 +270,7 @@
 
     # Some important defaults before we return...
     $config->{'dropdownlimit'} ||= $Munin::Common::Defaults::DROPDOWNLIMIT;
-    $config->{'rundir'}        ||= $Munin::Common::Defaults::MUNIN_STATEDIR;
+    $config->{'rundir'}        ||= $Munin::Common::Defaults::MUNIN_LOCKDIR;
     $config->{'dbdir'}         ||= $Munin::Common::Defaults::MUNIN_DBDIR;
     $config->{'logdir'}        ||= $Munin::Common::Defaults::MUNIN_LOGDIR;
     $config->{'tmpldir'}       ||= "$Munin::Common::Defaults::MUNIN_CONFDIR/templates/";
diff -Naur munin-1.4.5-orig//master/munin.conf.in munin-1.4.5/master/munin.conf.in
--- munin-1.4.5-orig//master/munin.conf.in	2009-11-23 06:26:36.000000000 -0600
+++ munin-1.4.5/master/munin.conf.in	2011-04-09 15:18:04.650625378 -0500
@@ -8,7 +8,7 @@
 # dbdir	@@DBDIR@@
 # htmldir @@HTMLDIR@@
 # logdir @@LOGDIR@@
-# rundir  @@STATEDIR@@
+# rundir  @@LOCKDIR@@
 #
 # Where to look for the HTML templates
 # tmpldir	@@CONFDIR@@/templates
