diff -u -r Master.orig/HTMLOld.pm Master/HTMLOld.pm
--- Master.orig/HTMLOld.pm	2010-01-14 13:13:57.000000000 +0100
+++ Master/HTMLOld.pm	2010-01-25 12:39:55.000000000 +0100
@@ -169,7 +169,7 @@
     };
 	
 
-    if (defined $groups->{"name"} and $groups->{"name"} eq "root") {
+    if (defined $groups->{"name"} and $groups->{"name"} eq "__root__") {
         $groups = $groups->{"groups"};    # root->groups
     }
 
@@ -509,7 +509,7 @@
         my $peername = munin_get_node_name($peer);
         next
             if $peername eq "contact"
-                and munin_get_node_name($parent) eq "root";
+                and munin_get_node_name($parent) eq "__root__";
         if ($peername eq $me) {
             unshift @$ret, {"name" => $peername, "link" => undef};
         }
diff -u -r Master.orig/LimitsOld.pm Master/LimitsOld.pm
--- Master.orig/LimitsOld.pm	2010-01-14 13:13:57.000000000 +0100
+++ Master/LimitsOld.pm	2010-01-25 12:39:56.000000000 +0100
@@ -250,7 +250,7 @@
     my $parent     = munin_get_parent($group);
     my $name       = get_notify_name($group);
 
-    if (defined $parent and munin_get_node_name($parent) ne "root") {
+    if (defined $parent and munin_get_node_name($parent) ne "__root__") {
 	return (get_full_group_path($parent) . " :: " . $name);
     } else {
 	return $name;
diff -u -r Master.orig/Utils.pm Master/Utils.pm
--- Master.orig/Utils.pm	2010-01-14 13:13:57.000000000 +0100
+++ Master/Utils.pm	2010-01-25 12:39:57.000000000 +0100
@@ -277,7 +277,7 @@
     $config->{'htmldir'}       ||= $Munin::Common::Defaults::MUNIN_HTMLDIR;
     $config->{'spooldir'}      ||= $Munin::Common::Defaults::MUNIN_SSPOOLDIR;
     $config->{'#%#parent'}     = undef;
-    $config->{'#%#name'}       = "root";
+    $config->{'#%#name'}       = "__root__";
 
     return ($config);
 }
@@ -920,7 +920,7 @@
 	# Not sure when a #%#name node can go missing
 	my $name = $i->{'#%#name'} || '*BUG*';
 	goto gotoparent if $name eq '*BUG*';
-	last if $name eq 'root';
+	last if $name eq '__root__';
 	if ($host) {
 	    # Into group land now
 	    unshift(@group,$name);
