--- appl/kx/rxterm.in	2004-02-21 21:49:23.893934354 +0000
+++ appl/kx/rxterm.in	2004-02-21 21:49:34.617299076 +0000
@@ -2,6 +2,6 @@
 #
 usage="Usage: $0 [-l username] [-k] [-f] [-r rsh_args] [-x xterm_args] [-K kx_args] [-w term_emulator] [-b rsh_binary][-v] [-h | --help] [--version] host"
-binary=rsh
+binary=krsh
 term=xterm
 while true
 do
--- appl/kx/rxtelnet.in	2005-06-16 16:29:10.000000000 +0000
+++ appl/kx/rxtelnet.in	2005-07-11 20:33:10.000000000 +0000
@@ -2,7 +2,7 @@
 #
 usage="Usage: $0 [-l username] [-k] [-fF] [-t args_to_telnet] [-x args_to_xterm] [-K args_to_kx] [-w term_emulator] [-b telnet_binary] [-n] [-v] [-h | --help] [--version] host [port]"
-binary=telnet
+binary=ktelnet
 term=
 kx_args=-P
