--- jks-keystore.hook	2010-04-11 20:47:48.000000000 +0200
+++ jks-keystore.hook	2010-10-24 14:52:38.837234542 +0200
@@ -28,14 +28,6 @@
 export JAVA_HOME=/usr/lib/jvm/$jvm
 PATH=$JAVA_HOME/bin:$PATH
 
-temp_jvm_cfg=
-if [ ! -f /etc/$jvm/jvm.cfg ]; then
-    # the jre is not yet configured, but jvm.cfg is needed to run it
-    temp_jvm_cfg=/etc/$jvm/jvm.cfg
-    mkdir -p /etc/$jvm
-    printf -- "-server KNOWN\n" > $temp_jvm_cfg
-fi
-
 # read lines of the form: [+-]/etc/ssl/certs/*.pem
 
 echo "updating keystore $KEYSTORE..."
@@ -62,7 +54,7 @@
 	  elif LANG=C LC_ALL=C keytool -importcert -trustcacerts \
 		-keystore $KEYSTORE -noprompt -storepass "$storepass" \
 	        -providerClass sun.security.pkcs11.SunPKCS11 \
-	        -providerArg '${java.home}/lib/security/nss.cfg' \
+	        -providerArg '/usr/lib/jvm/java-6-openjdk/jre/lib/security/nss.cfg' \
 		-alias "$alias" -file "$pem" > $log 2>&1
 	  then
 	      echo "  added: ${line#+*} (using NSS provider)"
@@ -85,7 +77,7 @@
 	    elif LANG=C LC_ALL=C keytool -delete -keystore $KEYSTORE \
 		-noprompt -storepass "$storepass" \
 	        -providerClass sun.security.pkcs11.SunPKCS11 \
-	        -providerArg '${java.home}/lib/security/nss.cfg' \
+	        -providerArg '/usr/lib/jvm/java-6-openjdk/jre/lib/security/nss.cfg' \
 		-alias "$alias"
 	    then
 		echo "  removed ${line#-*} (using NSS provider)"
@@ -103,8 +95,6 @@
 done
 rm -f $log
 
-[ -z "$temp_jvm_cfg" ] || rm -f $temp_jvm_cfg
-
 if [ $errors -gt 0 ]; then
     echo >&2 "failed (VM used: $jvm)."
     exit 1
