diff -ru xfig.3.2.4/u_fonts.h xfig.3.2.4.new/u_fonts.h
--- xfig.3.2.4/u_fonts.h	2002-03-27 01:18:23.000000000 +0100
+++ xfig.3.2.4.new/u_fonts.h	2005-12-03 15:21:02.054250392 +0100
@@ -30,12 +30,6 @@
 /* font number for the "nil" font (when user wants tiny text) */
 #define NILL_FONT NUM_FONTS
 
-extern int		psfontnum();
-extern int		latexfontnum();
-extern struct _xfstruct	x_fontinfo[], x_backup_fontinfo[];
-extern struct _fstruct	ps_fontinfo[];
-extern struct _fstruct	latex_fontinfo[];
-
 /* element of linked list for each font
    The head of list is for the different font NAMES,
    and the elements of this list are for each different
@@ -61,5 +55,11 @@
 				 * sizes */
 };
 
+extern int		psfontnum();
+extern int		latexfontnum();
+extern struct _xfstruct	x_fontinfo[], x_backup_fontinfo[];
+extern struct _fstruct	ps_fontinfo[];
+extern struct _fstruct	latex_fontinfo[];
+
 int		x_fontnum();
 #endif /* U_FONTS_H */
