--- xfig.3.2.4/app-defaults/Fig	2002-12-19 01:05:41.000000000 +0100
+++ xfig.3.2.4/app-defaults/Fig	2003-05-06 11:45:15.000000000 +0200
@@ -34,7 +34,7 @@
 ! Spell check program - put your favorite spelling check program here.
 !		It must write the misspelled words to standard output.
 
-Fig.spellcheckcommand:		spell %f
+Fig.spellcheckcommand:		aspell %f
 
 ! delay before spinner automatically counts when pressed (milliseconds)
 Fig.spinner_delay:		1000
