diff -up x2goserver-4.0.1.10/x2goserver-fmbindings/share/applications/x2gofm.desktop.mimetype 
x2goserver-4.0.1.10/x2goserver-fmbindings/share/applications/x2gofm.desktop
--- x2goserver-4.0.1.10/x2goserver-fmbindings/share/applications/x2gofm.desktop.mimetype	2014-01-03 10:27:31.167601207 -0700
+++ x2goserver-4.0.1.10/x2goserver-fmbindings/share/applications/x2gofm.desktop	2014-01-03 10:28:15.184321072 -0700
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Encoding=UTF-8
 Name=x2gofm
-MimeType=application/sshfsdisk-x2go;application/sshfscdrom-x2go;application/sshfsremovable-x2go
+MimeType=application/sshfsdisk-x2go;application/sshfscdrom-x2go;application/sshfsremovable-x2go;
 Exec=x2gofm %f
 Type=Application
 Terminal=false

