diff --git a/src/assets/view/widgets/curves/bezier/beziersplineeditor.cpp b/src/assets/view/widgets/curves/bezier/beziersplineeditor.cpp
index 91641c60c..cea14bc9f 100644
--- a/src/assets/view/widgets/curves/bezier/beziersplineeditor.cpp
+++ b/src/assets/view/widgets/curves/bezier/beziersplineeditor.cpp
@@ -24,6 +24,7 @@
 
 #include <QMouseEvent>
 #include <QPainter>
+#include <QPainterPath>
 
 BezierSplineEditor::BezierSplineEditor(QWidget *parent)
     : AbstractCurveWidget(parent)
diff --git a/src/bin/abstractprojectitem.cpp b/src/bin/abstractprojectitem.cpp
index f37591be3..e36ab5d3a 100644
--- a/src/bin/abstractprojectitem.cpp
+++ b/src/bin/abstractprojectitem.cpp
@@ -32,6 +32,7 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #include "jobs/thumbjob.hpp"
 
 #include <QPainter>
+#include <QPainterPath>
 #include <QVariant>
 #include <utility>
 AbstractProjectItem::AbstractProjectItem(PROJECTITEMTYPE type, QString id, const std::shared_ptr<ProjectItemModel> &model, bool isRoot)
diff --git a/src/titler/gradientwidget.cpp b/src/titler/gradientwidget.cpp
index 6ad1e8373..a259f5fad 100644
--- a/src/titler/gradientwidget.cpp
+++ b/src/titler/gradientwidget.cpp
@@ -23,6 +23,7 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 #include <QLinearGradient>
 #include <QPainter>
+#include <QPainterPath>
 #include <QPixmap>
 #include <QtMath>
 
