--- ggz-gtk-client-0.0.14.1/ggz-gtk/props.c	2008-01-05 20:03:36.000000000 +0100
+++ ggz-gtk-client-0.0.14.1.PATCHED/ggz-gtk/props.c	2010-08-04 01:53:37.573425091 +0200
@@ -1203,7 +1203,7 @@
   gtk_box_pack_start (GTK_BOX (vbox9), color_table, TRUE, TRUE, 0);
   gtk_container_set_border_width (GTK_CONTAINER (color_table), 3);
 
-  h_spin_adj = gtk_adjustment_new (1, 0, 15, 1, 1, 1);
+  h_spin_adj = gtk_adjustment_new (1, 0, 15, 1, 1, 0);
   h_spin = gtk_spin_button_new (GTK_ADJUSTMENT (h_spin_adj), 1, 0);
   gtk_widget_ref (h_spin);
   g_object_set_data_full(G_OBJECT (dlg_props), "h_spin", h_spin,
@@ -1213,7 +1213,7 @@
                     (GtkAttachOptions) (0), 0, 0);
   gtk_spin_button_set_numeric (GTK_SPIN_BUTTON (h_spin), TRUE);
 
-  n_spin_adj = gtk_adjustment_new (1, 0, 15, 1, 1, 1);
+  n_spin_adj = gtk_adjustment_new (1, 0, 15, 1, 1, 0);
   n_spin = gtk_spin_button_new (GTK_ADJUSTMENT (n_spin_adj), 1, 0);
   gtk_widget_ref (n_spin);
   g_object_set_data_full(G_OBJECT (dlg_props), "n_spin", n_spin,
@@ -1223,7 +1223,7 @@
                     (GtkAttachOptions) (0), 0, 0);
   gtk_spin_button_set_numeric (GTK_SPIN_BUTTON (n_spin), TRUE);
 
-  f_spin_adj = gtk_adjustment_new (1, 0, 15, 1, 1, 1);
+  f_spin_adj = gtk_adjustment_new (1, 0, 15, 1, 1, 0);
   f_spin = gtk_spin_button_new (GTK_ADJUSTMENT (f_spin_adj), 1, 0);
   gtk_widget_ref (f_spin);
   g_object_set_data_full(G_OBJECT (dlg_props), "f_spin", f_spin,
--- ggz-gtk-client-0.0.14.1/ggz-gtk/xtext.c	2008-01-05 20:03:36.000000000 +0100
+++ ggz-gtk-client-0.0.14.1.PATCHED/ggz-gtk/xtext.c	2010-08-04 01:52:36.446764446 +0200
@@ -720,7 +720,7 @@
 	xtext->overdraw = FALSE;
 	xtext->tint_red = xtext->tint_green = xtext->tint_blue = TINT_VALUE;
 
-	xtext->adj = (GtkAdjustment *) gtk_adjustment_new (0, 0, 1, 1, 1, 1);
+	xtext->adj = (GtkAdjustment *) gtk_adjustment_new (0, 0, 1, 1, 1, 0);
 	g_object_ref (G_OBJECT (xtext->adj));
 	gtk_object_sink ((GtkObject *) xtext->adj);
 
