--- configure-orig	2006-04-02 16:29:42.000000000 +0200
+++ configure	2008-02-11 18:16:47.000000000 +0100
@@ -3117,6 +3117,7 @@
 #include<sys/socket.h>
 #include<net/ethernet.h>
 #include<linux/if.h>
+#include<linux/in.h>
 
 
 #include <$ac_header>
@@ -5646,6 +5647,7 @@
 #include <sys/socket.h>
 #include <net/ethernet.h>
 #include <linux/if.h>
+#include <linux/in.h>
 #include <linux/if_pppox.h>
 int main()
 {
@@ -5693,7 +5695,7 @@
 fi
 
 if test "$GCC" = yes; then
-	CFLAGS="$CFLAGS -Wall -Wstrict-prototypes -ansi"
+	CFLAGS="$CFLAGS -Wall -Wstrict-prototypes"
 fi
 
 if test "$PPPD" = "NOTFOUND"; then

--- plugin.c_orig	2006-04-02 16:29:42.000000000 +0200
+++ plugin.c	2008-02-11 19:11:23.000000000 +0100
@@ -52,7 +52,6 @@
 #include <net/ethernet.h>
 #include <net/if_arp.h>
 #include <linux/ppp_defs.h>
-#include <linux/if_ppp.h>
 #include <linux/if_pppox.h>
 
 #ifndef _ROOT_PATH
