diff -urNad amarok-1.4.7/amarok/src/playlist.cpp amarok-1.4.8/amarok/src/playlist.cpp
--- amarok-1.4.7/amarok/src/playlist.cpp	2007-08-07 18:48:28.000000000 +0100
+++ amarok-1.4.8/amarok/src/playlist.cpp	2007-11-18 15:00:25.000000000 +0100
@@ -1227,7 +1227,10 @@
     }
     else
     {
-        if ( !AmarokConfig::randomMode() || m_prevTracks.count() <= 1 )
+        if( dynamicMode() )
+        {
+        }
+        else if( !AmarokConfig::randomMode() || m_prevTracks.count() <= 1 )
         {
             if( item )
             {
