diff -Nur mozilla-2.0.orig//browser/branding/unofficial/pref/firefox-branding.js mozilla-2.0/browser/branding/unofficial/pref/firefox-branding.js
--- mozilla-2.0.orig//browser/branding/unofficial/pref/firefox-branding.js	2011-03-03 14:11:19.000000000 -0800
+++ mozilla-2.0/browser/branding/unofficial/pref/firefox-branding.js	2011-03-13 10:02:02.950027389 -0700
@@ -1,5 +1,6 @@
-pref("startup.homepage_override_url","http://www.mozilla.org/projects/%APP%/%VERSION%/whatsnew/");
-pref("startup.homepage_welcome_url","http://www.mozilla.org/projects/%APP%/%VERSION%/firstrun/");
+pref("startup.homepage_override_url","http://www.mozilla.com/%APP%/%VERSION%/whatsnew/");
+
+pref("startup.homepage_welcome_url","http://www.mozilla.com/%APP%/%VERSION%/firstrun/");
 pref("browser.panorama.welcome_url", "http://www.mozilla.com/firefox/panorama/");
 // The time interval between checks for a new version (in seconds)
 // nightly=8 hours, official=24 hours
@@ -15,7 +16,7 @@
 pref("app.update.url.details", "http://www.mozilla.org/projects/%APP%/");
 
 // Release notes and vendor URLs
-pref("app.releaseNotesURL", "http://www.mozilla.org/projects/%APP%/%VERSION%/releasenotes/");
+pref("app.releaseNotesURL", "http://www.mozilla.com/%APP%/%VERSION%/releasenotes/");
 pref("app.vendorURL", "http://www.mozilla.org/projects/%APP%/");
 
 // Search codes belong only in builds with official branding
