diff --git i/Makefile.in w/Makefile.in
index 626edae551d3..781e41269723 100644
--- i/Makefile.in
+++ w/Makefile.in
@@ -239,7 +239,7 @@ prepsymbolsarchive:
 	echo packing symbols
 	$(NSINSTALL) -D $(DIST)/$(PKG_PATH)
 
-ifndef MOZ_AUTOMATION
+ifndef MOZ_PROFILE_USE
 prepsymbolsarchive: recurse_syms
 endif
 
diff --git i/config/recurse.mk w/config/recurse.mk
index ed2e187061aa..379b9042a8e4 100644
--- i/config/recurse.mk
+++ w/config/recurse.mk
@@ -76,11 +76,9 @@ $(compile_targets) $(syms_targets):
 $(syms_targets): %/syms: %/target
 
 # Only hook symbols targets into the main compile graph in automation.
-ifdef MOZ_AUTOMATION
-ifeq (1,$(MOZ_AUTOMATION_BUILD_SYMBOLS))
+ifdef MOZ_PROFILE_USE
 recurse_compile: $(syms_targets)
 endif
-endif
 
 # Create a separate rule that depends on every 'syms' target so that
 # symbols can be dumped on demand locally.
diff --git i/config/rules.mk w/config/rules.mk
index fefe2805fb92..b5641eae15ba 100644
--- i/config/rules.mk
+++ w/config/rules.mk
@@ -750,11 +750,9 @@ DUMP_SYMS_TARGETS := $(SHARED_LIBRARY) $(PROGRAM) $(SIMPLE_PROGRAMS)
 endif
 endif
 
-ifdef MOZ_AUTOMATION
-ifeq (,$(filter 1,$(MOZ_AUTOMATION_BUILD_SYMBOLS)))
+ifdef MOZ_PROFILE_GENERATE
 DUMP_SYMS_TARGETS :=
 endif
-endif
 
 ifdef MOZ_COPY_PDBS
 MAIN_PDB_FILES = $(addsuffix .pdb,$(basename $(DUMP_SYMS_TARGETS)))
