diff -Nur mozilla-release.orig/configure.in mozilla-release/configure.in
--- mozilla-release.orig/configure.in	2012-02-05 16:01:35.722024142 +0000
+++ mozilla-release/configure.in	2012-02-05 16:03:34.871064547 +0000
@@ -5629,7 +5629,7 @@
             dnl v0.9.6 one to check for.
             AC_TRY_COMPILE([
                 #include <vpx/vpx_decoder.h>
-                #if !defined(VPX_CODEC_USE_INPUT_PARTITION)
+                #if !defined(VPX_CODEC_USE_INPUT_FRAGMENTS)
                     #error "test failed."
                 #endif
                 ],
