From fe72f63c91a1458d97d455ab0078d042464366e7 Mon Sep 17 00:00:00 2001
From: Enrico Minack <enrico-minack@gmx.de>
Date: Mon, 03 Jan 2011 19:58:13 +0000
Subject: gweather: fixes NetworkManager support to compile, see bug 636217

---
diff --git a/gweather/Makefile.am b/gweather/Makefile.am
index ea6ac96..165efda 100644
--- a/gweather/Makefile.am
+++ b/gweather/Makefile.am
@@ -6,6 +6,7 @@ INCLUDES =					\
 	-I$(srcdir)				\
 	-I$(top_srcdir)				\
 	-DGWEATHER_MENU_UI_DIR=\""$(uidir)"\"	\
+	$(DBUS_CFLAGS)				\
 	$(GNOME_APPLETS3_CFLAGS)		\
 	$(LIBNOTIFY_CFLAGS)			\
 	$(LIBGWEATHER_CFLAGS)			\
@@ -22,6 +23,7 @@ gweather_applet_2_SOURCES = \
 	gweather-applet.c gweather-applet.h
 
 gweather_applet_2_LDADD = 	\
+	$(DBUS_LIBS)		\
 	$(LIBNOTIFY_LIBS) 	\
 	$(GNOME_APPLETS3_LIBS)	\
 	$(GNOME_LIBS2_LIBS)	\
--
cgit v0.8.3.4
