--- openbox-3.4.8/data/xsession/openbox-gnome-session.in~	2009-12-13 18:46:58.480826774 +0100
+++ openbox-3.4.8/data/xsession/openbox-gnome-session.in	2009-12-13 18:49:15.776973708 +0100
@@ -7,6 +7,9 @@
   exit
 fi
 
+# Clean up after GDM
+xprop -root -remove _NET_NUMBER_OF_DESKTOPS -remove _NET_DESKTOP_NAMES -remove _NET_CURRENT_DESKTOP 2> /dev/null
+
 VER=$(gnome-session --version 2>/dev/null | \
       sed -e 's/[^0-9.]*\([0-9.]\+\)/\1/')
 
--- openbox-3.4.8/data/xsession/openbox-kde-session.in~	2009-12-13 18:47:07.676987674 +0100
+++ openbox-3.4.8/data/xsession/openbox-kde-session.in	2009-12-13 18:48:48.140310531 +0100
@@ -7,6 +7,9 @@
   exit
 fi
 
+# Clean up after GDM
+xprop -root -remove _NET_NUMBER_OF_DESKTOPS -remove _NET_DESKTOP_NAMES -remove _NET_CURRENT_DESKTOP 2> /dev/null
+
 # Run KDE with Openbox as its window manager
 export KDEWM="@bindir@/openbox"
 exec startkde "$@"
--- openbox-3.4.8/data/xsession/openbox-session.in~	2009-12-13 18:47:13.173649966 +0100
+++ openbox-3.4.8/data/xsession/openbox-session.in	2009-12-13 18:48:17.696977111 +0100
@@ -7,6 +7,9 @@
   exit
 fi
 
+# Clean up after GDM
+xprop -root -remove _NET_NUMBER_OF_DESKTOPS -remove _NET_DESKTOP_NAMES -remove _NET_CURRENT_DESKTOP 2> /dev/null
+
 AUTOSTART="${XDG_CONFIG_HOME:-"$HOME/.config"}/openbox/autostart.sh"
 GLOBALAUTOSTART="@configdir@/openbox/autostart.sh"
 
