--- openbox-3.4.7.2/data/xsession/openbox-gnome-session.in~	2008-02-29 22:18:12.000000000 +0100
+++ openbox-3.4.7.2/data/xsession/openbox-gnome-session.in	2008-06-10 18:37:10.000000000 +0200
@@ -7,6 +7,9 @@ if test -n "$1"; then
   exit
 fi
 
+# Clean up after GDM
+xprop -root -remove _NET_NUMBER_OF_DESKTOPS -remove _NET_DESKTOP_NAMES -remove _NET_CURRENT_DESKTOP 2> /dev/null
+
 # Run GNOME with Openbox as its window manager
 export WINDOW_MANAGER="@bindir@/openbox"
 exec gnome-session --choose-session=openbox-session "$@"
--- openbox-3.4.7.2/data/xsession/openbox-kde-session.in~	2008-02-29 22:18:12.000000000 +0100
+++ openbox-3.4.7.2/data/xsession/openbox-kde-session.in	2008-06-10 18:37:26.000000000 +0200
@@ -7,6 +7,9 @@ if test -n "$1"; then
   exit
 fi
 
+# Clean up after GDM
+xprop -root -remove _NET_NUMBER_OF_DESKTOPS -remove _NET_DESKTOP_NAMES -remove _NET_CURRENT_DESKTOP 2> /dev/null
+
 # Run KDE with Openbox as its window manager
 export KDEWM="@bindir@/openbox"
 exec startkde "$@"
--- openbox-3.4.7.2/data/xsession/openbox-session.in~	2008-02-29 22:18:12.000000000 +0100
+++ openbox-3.4.7.2/data/xsession/openbox-session.in	2008-06-10 18:37:54.000000000 +0200
@@ -7,6 +7,9 @@ if test -n "$1"; then
   exit
 fi
 
+# Clean up after GDM
+xprop -root -remove _NET_NUMBER_OF_DESKTOPS -remove _NET_DESKTOP_NAMES -remove _NET_CURRENT_DESKTOP 2> /dev/null
+
 AUTOSTART="$HOME/.config/openbox/autostart.sh"
 GLOBALAUTOSTART="@configdir@/openbox/autostart.sh"
 
