Index: kioslave/media/mediamanager/halbackend.cpp
===================================================================
--- kioslave/media/mediamanager/halbackend.cpp	(revision 724099)
+++ kioslave/media/mediamanager/halbackend.cpp	(revision 724100)
@@ -74,7 +74,10 @@
         const QPtrList<Medium> medlist = m_mediaList.list();
         QPtrListIterator<Medium> it (medlist);
         for ( const Medium *current_medium = it.current(); current_medium; current_medium = ++it)
-	    unmount(current_medium->id());
+        {
+            if( !current_medium->id().startsWith( "/org/kde" ))
+                unmount(current_medium->id());
+        }
 
 
         /* Remove all the registered media first */
