diff -aur old/src/linux.c new/src/linux.c
--- old/src/linux.c	2010-10-05 23:29:36.000000000 +0200
+++ new/src/linux.c	2010-12-06 22:18:59.346667183 +0100
@@ -1671,6 +1671,8 @@
  			the tradition! */
  			else if (strncmp(buf, "POWER_SUPPLY_CURRENT_NOW=", 25) == 0)
  				sscanf(buf, "POWER_SUPPLY_CURRENT_NOW=%d", &present_rate);
+ 			else if (strncmp(buf, "POWER_SUPPLY_POWER_NOW=", 23) == 0)
+ 				sscanf(buf, "POWER_SUPPLY_POWER_NOW=%d", &present_rate);
  			else if (strncmp(buf, "POWER_SUPPLY_ENERGY_NOW=", 24) == 0)
  				sscanf(buf, "POWER_SUPPLY_ENERGY_NOW=%d", &remaining_capacity);
  			else if (strncmp(buf, "POWER_SUPPLY_ENERGY_FULL=", 25) == 0)
