This patch fixes a silly looping oversight
--- lsdvd.c	2005-04-13 16:34:34.000000000 -0500
+++ lsdvd.c	2006-01-11 00:32:55.000000000 -0600
@@ -320,7 +320,7 @@
 		// PALETTE
 		if (opt_P) {
 			dvd_info.titles[j].palette = malloc(16 * sizeof(int));
-        	        for (i=1; i < 18; i++) { dvd_info.titles[j].palette[i] = pgc->palette[i]; }
+        	        for (i=0; i < 16; i++) { dvd_info.titles[j].palette[i] = pgc->palette[i]; }
 		} else {
 			dvd_info.titles[j].palette = NULL;
 		}
