diff -up sqlite-3.5.8/sqlite3.pc.in.release sqlite-3.5.8/sqlite3.pc.in
--- sqlite-3.5.8/sqlite3.pc.in.release	2008-04-23 09:11:38.000000000 +0300
+++ sqlite-3.5.8/sqlite3.pc.in	2008-04-23 09:11:48.000000000 +0300
@@ -7,7 +7,7 @@ includedir=@includedir@
 
 Name: SQLite
 Description: SQL database engine
-Version: @VERSION@
+Version: @RELEASE@
 Libs: -L${libdir} -lsqlite3
 Libs.private: @LIBS@
 Cflags: -I${includedir}
