From 4a0915c25815647c165e7ca6047f3da84ece04ea Mon Sep 17 00:00:00 2001
From: Debarshi Ray <debarshir@gnome.org>
Date: Mon, 22 Oct 2012 13:07:04 +0000
Subject: gdaemonmount: Check for a NULL GMountOperation when unmounting

https://bugzilla.gnome.org/show_bug.cgi?id=686637
---
diff --git a/client/gdaemonmount.c b/client/gdaemonmount.c
index 53f22b5..ca77648 100644
--- a/client/gdaemonmount.c
+++ b/client/gdaemonmount.c
@@ -330,8 +330,9 @@ g_daemon_mount_unmount_with_operation (GMount *mount,
   data = g_new0 (AsyncProxyCreate, 1);
   data->mount = g_object_ref (mount);
   data->mount_info = daemon_mount->mount_info;
-  data->mount_operation = g_object_ref (mount_operation);
   data->flags = flags;
+  if (mount_operation)
+    data->mount_operation = g_object_ref (mount_operation);
   if (cancellable)
     data->cancellable = g_object_ref (cancellable);
 
--
cgit v0.9.0.2
