diff -up gok-2.27.2/gok/Makefile.in.lm gok-2.27.2/gok/Makefile.in
--- gok-2.27.2/gok/Makefile.in.lm	2009-06-13 00:59:29.162580117 -0400
+++ gok-2.27.2/gok/Makefile.in	2009-06-13 00:59:11.583583888 -0400
@@ -374,7 +374,7 @@ INCLUDES = -DGOK_LOCALEDIR=\""$(datadir)
 	-DDATADIR=\""$(datadir)"\" -DLIBDIR=\""$(libdir)"\" \
 	-I$(top_srcdir) -I$(top_builddir) $(GOK_CFLAGS) \
 	$(am__append_1)
-LDADD = $(GOK_LIBS) $(X_LIBS) @LT_VERSION_INFO@ $(am__append_2)
+LDADD = $(GOK_LIBS) $(X_LIBS) -lm @LT_VERSION_INFO@ $(am__append_2)
 gok_LDFLAGS = -export-dynamic
 gok_SOURCES = callbacks.c callbacks.h gok-branchback-stack.c \
 	gok-branchback-stack.h gok-button.c gok-button.h gok-chunker.c \
