--- trunk/KDE/kdeadmin/system-config-printer-kde/system-config-printer-kde.py	2010/07/28 12:28:36	1156005
+++ trunk/KDE/kdeadmin/system-config-printer-kde/system-config-printer-kde.py	2010/07/28 12:30:01	1156006
@@ -101,7 +101,7 @@
     def __init__(self, component_data, parent, gui):
         KCModule.__init__(self, component_data, parent)
 
-        uic.loadUi(APPDIR + "/" + "system-config-printer.ui", self)
+        uic.loadUi(unicode(APPDIR + "/" + "system-config-printer.ui"), self)
         self.setButtons(KCModule.Apply)
         self.gui = gui
         
@@ -2046,7 +2046,7 @@
         self.WaitWindow = QMessageBox(self.mainapp)
         self.WaitWindow.setStandardButtons(QMessageBox.NoButton)
 
-        uic.loadUi(APPDIR + "/" + "new-printer.ui", self)
+        uic.loadUi(unicode(APPDIR + "/" + "new-printer.ui"), self)
 
         self.btnNPBack.setIcon(KIcon("go-previous"))
         self.btnNPForward.setIcon(KIcon("go-next"))
