diff -Naur NVIDIA.orig/usr/src/nv/nv.c NVIDIA-Linux-x86_64-100.14.11-pkg2/usr/src/nv/nv.c
--- NVIDIA.orig/usr/src/nv/nv.c	2007-06-14 02:18:38.000000000 +0200
+++ NVIDIA-Linux-x86_64-100.14.11-pkg2/usr/src/nv/nv.c	2007-07-24 20:46:40.000000000 +0200
@@ -1566,8 +1566,7 @@
         if (apm_nv_dev[i] != NULL) pm_unregister(apm_nv_dev[i]);
 #endif
 
-    if (unregister_chrdev(nv_major, "nvidia") < 0)
-        nv_printf(NV_DBG_ERRORS, "NVRM: unregister nv chrdev failed\n");
+    unregister_chrdev(nv_major, "nvidia");
 
     for (i = 0; i < num_nv_devices; i++)
     {
@@ -1598,8 +1597,7 @@
 
     nv_printf(NV_DBG_INFO, "NVRM: nvidia_exit_module\n");
 
-    if (unregister_chrdev(nv_major, "nvidia") < 0)
-        nv_printf(NV_DBG_ERRORS, "NVRM: unregister nv chrdev failed\n");
+    unregister_chrdev(nv_major, "nvidia"); 
 
     for (i = 0; i < num_nv_devices; i++)
     {
diff -Naur NVIDIA.orig/usr/src/nv/nv-linux.h NVIDIA-Linux-x86_64-100.14.11-pkg2/usr/src/nv/nv-linux.h
--- NVIDIA.orig/usr/src/nv/nv-linux.h	2007-06-14 02:18:38.000000000 +0200
+++ NVIDIA-Linux-x86_64-100.14.11-pkg2/usr/src/nv/nv-linux.h	2007-07-24 20:46:40.000000000 +0200
@@ -533,7 +533,7 @@
 #define NV_KMEM_CACHE_CREATE(kmem_cache, name, type)            \
     {                                                           \
         kmem_cache = kmem_cache_create(name, sizeof(type),      \
-                        0, 0, NULL, NULL);                      \
+                        0, 0, NULL);                            \
     } 
 
 #define NV_KMEM_CACHE_DESTROY(kmem_cache)                       \
