--- hunspell-1.1.5/src/hunspell/hunspell.hxx.orig	2007-03-20 10:44:20.000000000 +0000
+++ hunspell-1.1.5/src/hunspell/hunspell.hxx	2007-03-20 10:44:27.000000000 +0000
@@ -1,12 +1,8 @@
-#include "license.hunspell"
-#include "license.myspell"
-
 #include "hashmgr.hxx"
 #include "affixmgr.hxx"
 #include "suggestmgr.hxx"
 #include "csutil.hxx"
 #include "langnum.hxx"
-#include "config.h"
 
 #define  SPELL_COMPOUND  (1 << 0)
 #define  SPELL_FORBIDDEN (1 << 1)
