diff --git a/Makefile.PL.orig b/Makefile.PL
index fc8ef4a..26313e2 100644
--- a/Makefile.PL.orig
+++ b/Makefile.PL
@@ -26,7 +26,6 @@ my %WriteMakefileArgs = (
     "IO::Socket::SSL" => "1.54",
     "LWP::Protocol::http" => 0,
     "LWP::UserAgent" => "6.06",
-    "Mozilla::CA" => 20180117,
     "Net::HTTPS" => 6,
     "base" => 0,
     "strict" => 0
diff --git a/lib/LWP/Protocol/https.pm.orig b/lib/LWP/Protocol/https.pm
index ed4d832..b8667cf 100644
--- a/lib/LWP/Protocol/https.pm.orig
+++ b/lib/LWP/Protocol/https.pm
@@ -25,25 +25,7 @@ sub _extra_sock_opts
     }
     if ($ssl_opts{SSL_verify_mode}) {
 	unless (exists $ssl_opts{SSL_ca_file} || exists $ssl_opts{SSL_ca_path}) {
-	    eval {
-		require Mozilla::CA;
-	    };
-	    if ($@) {
-		if ($@ =~ /^Can't locate Mozilla\/CA\.pm/) {
-		    $@ = <<'EOT';
-Can't verify SSL peers without knowing which Certificate Authorities to trust
-
-This problem can be fixed by either setting the PERL_LWP_SSL_CA_FILE
-environment variable or by installing the Mozilla::CA module.
-
-To disable verification of SSL peers set the PERL_LWP_SSL_VERIFY_HOSTNAME
-environment variable to 0.  If you do this you can't be sure that you
-communicate with the expected peer.
-EOT
-		}
-		die $@;
-	    }
-	    $ssl_opts{SSL_ca_file} = Mozilla::CA::SSL_ca_file();
+	    $ssl_opts{SSL_ca_file} = '/etc/ssl/certs/ca-certificates.crt';
 	}
     }
     $self->{ssl_opts} = \%ssl_opts;
