diff --git a/find-modules/FindPythonModuleGeneration.cmake b/find-modules/FindPythonModuleGeneration.cmake
index 31cc52d..469471b 100644
--- a/find-modules/FindPythonModuleGeneration.cmake
+++ b/find-modules/FindPythonModuleGeneration.cmake
@@ -174,8 +174,8 @@ if (NOT GBP_SIP_COMMAND)
   _report_NOT_FOUND("The sip executable must be available to use ${CMAKE_FIND_PACKAGE_NAME}.")
 endif()
 
-if (NOT GPB_PYTHON2_COMMAND)
-  _report_NOT_FOUND("The python2 executable is required by clang-python for the ${CMAKE_FIND_PACKAGE_NAME} Module.")
+if (NOT GPB_PYTHON3_COMMAND)
+  _report_NOT_FOUND("The python3 executable is required by clang-python for the ${CMAKE_FIND_PACKAGE_NAME} Module.")
 endif()
 
 if (NOT libclang_LIBRARY)
@@ -211,7 +211,7 @@ if (NOT libclang_LIBRARY)
 endif()
 
 execute_process(
-  COMMAND ${GPB_PYTHON2_COMMAND} ${CMAKE_CURRENT_LIST_DIR}/sip_generator.py --self-check ${libclang_LIBRARY}
+  COMMAND ${GPB_PYTHON3_COMMAND} ${CMAKE_CURRENT_LIST_DIR}/sip_generator.py --self-check ${libclang_LIBRARY}
   RESULT_VARIABLE selfCheckErrors
 )
 
@@ -352,7 +352,7 @@ function(ecm_generate_python_binding
         set(comp_flags "$<JOIN:$<TARGET_PROPERTY:${target_value},INTERFACE_COMPILE_OPTIONS>;${stdFlag},;>")
 
         add_custom_command(OUTPUT ${sip_file}
-            COMMAND ${GPB_PYTHON2_COMMAND} ${GPB_MODULE_DIR}/sip_generator.py
+            COMMAND ${GPB_PYTHON3_COMMAND} ${GPB_MODULE_DIR}/sip_generator.py
               --flags " ${inc_dirs};${sys_inc_dirs};${comp_defs};${comp_flags}"
               --include_filename "${hdr_filename}"
               ${libclang_LIBRARY}
@@ -400,7 +400,7 @@ headers = sipAPI${modulename_value}
 
     add_custom_command(OUTPUT
       "${CMAKE_CURRENT_BINARY_DIR}/pybuild/${pythonnamespace_value}/${modulename_value}/unified${modulename_value}.cpp"
-      COMMAND ${GPB_PYTHON2_COMMAND} "${GPB_MODULE_DIR}/run-sip.py" --sip ${GBP_SIP_COMMAND}
+      COMMAND ${GPB_PYTHON3_COMMAND} "${GPB_MODULE_DIR}/run-sip.py" --sip ${GBP_SIP_COMMAND}
        --unify "${CMAKE_CURRENT_BINARY_DIR}/pybuild/${pythonnamespace_value}/${modulename_value}/unified${modulename_value}.cpp"
        --module-name "${modulename_value}"
        -c "${CMAKE_CURRENT_BINARY_DIR}/pybuild/${pythonnamespace_value}/${modulename_value}"
diff --git a/find-modules/rules_engine.py b/find-modules/rules_engine.py
index 8ec5ac5..d4f2bac 100755
--- a/find-modules/rules_engine.py
+++ b/find-modules/rules_engine.py
@@ -113,7 +113,7 @@ class Rule(object):
             logger.debug(_("Rule {} suppressed {}, {}").format(self, fqn, original))
         else:
             delta = False
-            for k, v in original.iteritems():
+            for k, v in iter(original.items()):
                 if v != modified[k]:
                     delta = True
                     break
@@ -620,7 +620,7 @@ class AbstractCompiledCodeDb(object):
             logger.debug(_("Rule {} suppressed {}, {}").format(self, fqn, original))
         else:
             delta = False
-            for k, v in original.iteritems():
+            for k, v in iter(original.items()):
                 if v != modified[k]:
                     delta = True
                     break
