diff --git a/find-modules/FindPythonModuleGeneration.cmake b/find-modules/FindPythonModuleGeneration.cmake
index 9a34220..6d4f97e 100644
--- a/find-modules/FindPythonModuleGeneration.cmake
+++ b/find-modules/FindPythonModuleGeneration.cmake
@@ -309,6 +309,7 @@ get_filename_component(libclang_file "${libclang_file}" REALPATH)
 
 find_file(SIP_Qt5Core_Mod_FILE
   NAMES QtCoremod.sip
+  HINTS ${KDE_INSTALL_PYTHON3DIR}/PyQt5/bindings/QtCore
   PATH_SUFFIXES share/sip/PyQt5/QtCore
 )
 
