diff --git i/src/app/currentLocationController.js w/src/app/currentLocationController.js
index 67538d2..4cad182 100644
--- i/src/app/currentLocationController.js
+++ w/src/app/currentLocationController.js
@@ -87,10 +87,8 @@ var CurrentLocationController = class CurrentLocationController {
     _onLocationUpdated(simple) {
         let geoclueLocation = simple.get_location();
 
-        this.currentLocation = GWeather.Location.new_detached(geoclueLocation.description,
-                                                              null,
-                                                              geoclueLocation.latitude,
-                                                              geoclueLocation.longitude);
+        this.currentLocation = this._world._world.find_nearest_city(geoclueLocation.latitude,
+                                                                    geoclueLocation.longitude);
         this._world.currentLocationChanged(this.currentLocation);
     }
 
