--- extra-cmake-modules-5.23.0/modules/ECMGeneratePriFile.cmake.orig	2016-06-23 07:40:57.240619651 +0000
+++ extra-cmake-modules-5.23.0/modules/ECMGeneratePriFile.cmake	2016-06-23 07:41:26.010499925 +0000
@@ -84,6 +84,8 @@
 # (To distribute this file outside of extra-cmake-modules, substitute the full
 #  License text for the above reference.)
 
+include(KDEInstallDirs)
+
 if(KDE_INSTALL_USE_QT_SYS_PATHS)
   include(ECMQueryQmake)
   query_qmake(qt_host_data_dir QT_HOST_DATA)
