diff --git a/scripts/buildversion.py b/scripts/buildversion.py
index 8875497..a0eb646 100755
--- a/scripts/buildversion.py
+++ b/scripts/buildversion.py
@@ -124,10 +124,6 @@ def main():
         cleanbuild = cleanbuild and ver and options.extra != ""
         if not ver:
             ver = "?"
-    if not cleanbuild:
-        btime = time.strftime("%Y%m%d_%H%M%S")
-        hostname = socket.gethostname()
-        ver = "%s-%s-%s" % (ver, btime, hostname)
     write_version(outfile, ver + options.extra, toolstr)
 
 if __name__ == '__main__':
