--- meld-3.14.1-orig/meld/meldbuffer.py 2015-10-02 22:10:11.000000000 -0400
+++ meld-3.14.1/meld/meldbuffer.py      2015-11-11 20:37:09.412543281 -0500
@@ -76,6 +76,9 @@
         self.data.filename = filename
         self.data.label = label
 
+    def get_iter_at_line(self, line):
+        return super(MeldBuffer, self).get_iter_at_line(line)[1]
+
     def get_iter_at_line_or_eof(self, line):
         """Return a Gtk.TextIter at the given line, or the end of the buffer.
