From 4cc4801e096e7054a0e14fb2cc3465c06385bb6f Mon Sep 17 00:00:00 2001
From: Natanael Arndt <arndtn@gmail.com>
Date: Thu, 28 Jan 2021 18:56:27 +0100
Subject: [PATCH] Update SPARQL tests since the DBpedia was updated

---
 test/test_sparql_service.py | 28 +++++++++++++++-------------
 1 file changed, 15 insertions(+), 13 deletions(-)

diff --git a/test/test_sparql_service.py b/test/test_sparql_service.py
index ea1af4134..7bfe8e4c0 100644
--- a/test/test_sparql_service.py
+++ b/test/test_sparql_service.py
@@ -7,14 +7,14 @@
 
 def test_service():
     g = Graph()
-    q = '''select ?dbpHypernym ?dbpComment
+    q = '''select ?sameAs ?dbpComment
     where
     { service <http://DBpedia.org/sparql>
     { select ?dbpHypernym ?dbpComment
     where
     {
     <http://dbpedia.org/resource/John_Lilburne>
-        <http://purl.org/linguistics/gold/hypernym> ?dbpHypernym ;
+        <http://www.w3.org/2002/07/owl#sameAs> ?sameAs ;
         <http://www.w3.org/2000/01/rdf-schema#comment> ?dbpComment .
 
     } }  } limit 2'''
@@ -27,17 +29,17 @@ def test_service():
 
 def test_service_with_bind():
     g = Graph()
-    q = '''select ?dbpHypernym ?dbpComment ?dbpDeathPlace
+    q = '''select ?sameAs ?dbpComment ?subject
     where
-    { bind (<http://dbpedia.org/resource/Eltham> as ?dbpDeathPlace)
+    { bind (<http://dbpedia.org/resource/Category:1614_births> as ?subject)
       service <http://DBpedia.org/sparql>
-    { select ?dbpHypernym ?dbpComment ?dbpDeathPlace
+    { select ?sameAs ?dbpComment ?subject
     where
     {
     <http://dbpedia.org/resource/John_Lilburne>
-        <http://purl.org/linguistics/gold/hypernym> ?dbpHypernym ;
+        <http://www.w3.org/2002/07/owl#sameAs> ?sameAs ;
         <http://www.w3.org/2000/01/rdf-schema#comment> ?dbpComment ;
-        <http://dbpedia.org/ontology/deathPlace> ?dbpDeathPlace .
+        <http://purl.org/dc/terms/subject> ?subject .
 
     } }  } limit 2'''
     results = helper.query_with_retry(g, q)
@@ -49,17 +51,17 @@ def test_service_with_bind():
 
 def test_service_with_values():
     g = Graph()
-    q = '''select ?dbpHypernym ?dbpComment ?dbpDeathPlace
+    q = '''select ?sameAs ?dbpComment ?subject
     where
-    { values (?dbpHypernym ?dbpDeathPlace) {(<http://dbpedia.org/resource/Leveller> <http://dbpedia.org/resource/London>) (<http://dbpedia.org/resource/Leveller> <http://dbpedia.org/resource/Eltham>)}
+    { values (?sameAs ?subject) {(<http://de.dbpedia.org/resource/John_Lilburne> <http://dbpedia.org/resource/Category:1614_births>) (<https://global.dbpedia.org/id/4t6Fk> <http://dbpedia.org/resource/Category:Levellers>)}
       service <http://DBpedia.org/sparql>
-    { select ?dbpHypernym ?dbpComment ?dbpDeathPlace
+    { select ?sameAs ?dbpComment ?subject
     where
     {
     <http://dbpedia.org/resource/John_Lilburne>
-        <http://purl.org/linguistics/gold/hypernym> ?dbpHypernym ;
+        <http://www.w3.org/2002/07/owl#sameAs> ?sameAs ;
         <http://www.w3.org/2000/01/rdf-schema#comment> ?dbpComment ;
-        <http://dbpedia.org/ontology/deathPlace> ?dbpDeathPlace .
+        <http://purl.org/dc/terms/subject> ?subject .
 
     } }  } limit 2'''
     results = helper.query_with_retry(g, q)
