--- distro-configs/ArchLinux.conf.in	2009-04-19 09:29:40.000000000 +0000
+++ distro-configs/ArchLinux.conf.in.new	2009-04-19 09:35:54.000000000 +0000
@@ -0,0 +1,41 @@
+--with-vendor=\"the ArchLinux Linux team\"
+--disable-binfilter
+--disable-fontooo
+--disable-gnome-vfs
+--enable-gio
+--enable-gtk
+--enable-cups
+--enable-dbus
+--enable-vba
+--with-system-db
+--with-system-curl
+--with-system-expat
+--with-system-freetype
+--with-system-hunspell
+--with-system-icu
+--with-system-libwpd
+--with-system-libxml
+--with-system-libxslt
+--with-system-mozilla=libxul
+--with-system-neon
+--with-system-python
+--with-system-redland
+--with-system-sane-header
+--with-system-sndfile
+--with-system-stdlibs
+--with-system-xrender-headers
+--with-system-zlib
+--with-system-openssl
+--with-system-odbc-headers
+--with-jdk-home=${JAVA_HOME}
+--with-system-xerces
+--with-xerces-jar=/usr/share/java/xercesImpl.jar
+--with-system-xalan
+--with-xalan-jar=/usr/share/java/xalan.jar
+--with-system-xml-apis
+--with-xml-apis-jar=/usr/share/java/xml-apis.jar
+--with-system-hsqldb
+--with-hsqldb-jar=/usr/share/java/hsqldb.jar
+--with-serializer-jar=/usr/share/java/serializer.jar
+--with-use-shell=bash
+--disable-dependency-tracking
--- distro-configs/Makefile.in	2009-04-30 16:26:20.000000000 +0000
+++ distro-configs/Makefile.in.new	2009-05-01 10:02:40.000000000 +0000
@@ -30,7 +30,7 @@
 PRE_UNINSTALL = :
 POST_UNINSTALL = :
 subdir = distro-configs
-DIST_COMMON = $(srcdir)/Ark.conf.in $(srcdir)/Common.conf.in \
+DIST_COMMON = $(srcdir)/ArchLinux.conf.in $(srcdir)/Ark.conf.in $(srcdir)/Common.conf.in \
 	$(srcdir)/CommonLinux.conf.in $(srcdir)/CommonMac.conf.in \
 	$(srcdir)/CommonWin32.conf.in $(srcdir)/Debian.conf.in \
 	$(srcdir)/DebianEtch.conf.in $(srcdir)/DebianSarge.conf.in \
@@ -65,7 +65,7 @@
 am__configure_deps = $(am__aclocal_m4_deps) $(CONFIGURE_DEPENDENCIES) \
 	$(ACLOCAL_M4)
 mkinstalldirs = $(install_sh) -d
-CONFIG_CLEAN_FILES = Ark.conf Common.conf CommonLinux.conf \
+CONFIG_CLEAN_FILES = ArchLinux.conf Ark.conf Common.conf CommonLinux.conf \
 	CommonWin32.conf CommonMac.conf GoOoWin32.conf \
 	DroplineGNOME.conf DroplineGNOME64.conf Debian.conf \
 	DebianEtch.conf DebianSarge.conf FSFhuLinux.conf \
@@ -300,6 +300,8 @@
 	cd $(top_builddir) && $(MAKE) $(AM_MAKEFLAGS) am--refresh
 $(ACLOCAL_M4): @MAINTAINER_MODE_TRUE@ $(am__aclocal_m4_deps)
 	cd $(top_builddir) && $(MAKE) $(AM_MAKEFLAGS) am--refresh
+ArchLinux.conf: $(top_builddir)/config.status $(srcdir)/ArchLinux.conf.in
+	cd $(top_builddir) && $(SHELL) ./config.status $(subdir)/$@
 Ark.conf: $(top_builddir)/config.status $(srcdir)/Ark.conf.in
 	cd $(top_builddir) && $(SHELL) ./config.status $(subdir)/$@
 Common.conf: $(top_builddir)/config.status $(srcdir)/Common.conf.in
--- configure.in	2009-04-30 16:26:17.000000000 +0000
+++ configure.in.new	2009-05-01 10:10:08.000000000 +0000
@@ -592,6 +592,7 @@
 AC_MSG_CHECKING([for vendor name based on the distribution name])
 VENDORNAME=OpenOffice
 case $DISTRO in
+        ArchLinux*) VENDORNAME="ArchLinux" ;;
         Ark*) VENDORNAME="Ark" ;;
         Debian*|Ubuntu*) VENDORNAME="Debian" ;;
         Dropline*) VENDORNAME="DroplineGNOME" ;;
@@ -1378,6 +1379,7 @@
 desktop/mimetypes/32x32/Makefile
 desktop/mimetypes/48x48/Makefile
 distro-configs/Makefile
+distro-configs/ArchLinux.conf
 distro-configs/Ark.conf
 distro-configs/Common.conf
 distro-configs/CommonLinux.conf
--- download.in	2009-04-28 09:39:18.000000000 +0000
+++ download.in.new	2009-05-01 10:12:15.000000000 +0000
@@ -383,7 +383,8 @@
 }
 
 my $distro = '@DISTRO@';
-if ($distro eq 'Ark' ||
+if ($distro eq 'ArchLinux' ||
+    $distro eq 'Ark' ||
     $distro eq 'DroplineGNOME' ||
     $distro eq 'Frugalware' || $distro eq 'Frugalware64' ||
     $distro eq 'Gentoo.conf' ||
--- patches/dev300/apply	2009-04-30 14:55:12.000000000 +0000
+++ patches/dev300/apply.new	2009-05-01 10:15:16.000000000 +0000
@@ -105,6 +105,8 @@
 rPathCommon: LinuxCommon, NotDebian, Lockdown, BerkeleyDB43, CairoFonts
 rpl2: rPathCommon
 rpl2-64: rPathCommon
+# ArchLinux
+ArchLinux: LinuxCommon
 # Win32
 Win32 : Win32Common
 # translate.org.za
