diff -Nur wesnoth-1.8.orig/src/tools/exploder_utils.cpp wesnoth-1.8/src/tools/exploder_utils.cpp
--- wesnoth-1.8.orig/src/tools/exploder_utils.cpp	2010-01-01 15:16:49.000000000 +0200
+++ wesnoth-1.8/src/tools/exploder_utils.cpp	2010-04-08 17:38:03.066201123 +0300
@@ -174,8 +174,8 @@
 	//TODO: review whether providing NULL error handlers is something
 	//sensible
 	png_struct* png_ptr = png_create_write_struct
-		(PNG_LIBPNG_VER_STRING, reinterpret_cast<png_voidp>(png_voidp_NULL),
-		 png_error_ptr_NULL, png_error_ptr_NULL);
+		(PNG_LIBPNG_VER_STRING, reinterpret_cast<png_voidp>(NULL),
+		 NULL, NULL);
 	if(!png_ptr)
 		throw exploder_failure("Unable to initialize the png write structure");
 
