Index: krdc/CMakeLists.txt
===================================================================
--- krdc/CMakeLists.txt	(revision 1203716)
+++ krdc/CMakeLists.txt	(revision 1203717)
@@ -27,9 +27,6 @@
     add_definitions(${QT_DEFINITIONS} ${QT_QTDBUS_DEFINITIONS} ${KDE4_DEFINITIONS})
     include_directories(${CMAKE_SOURCE_DIR} ${CMAKE_BINARY_DIR} ${KDE4_INCLUDES})
 
-    macro_optional_find_package(LibVNCServer)
-    macro_log_feature(LIBVNCSERVER_FOUND "libvncserver" "VNC Server library" "http://libvncserver.sourceforge.net/" FALSE "0.9" "Needed to build Krfb and VNC support in Krdc")
-
     # NX support is not ready for KDE 4.2; disabled (uwolfer)
     # macro_optional_find_package(LibNXCL)
     # macro_log_feature(LIBNXCL_FOUND "libnxcl" "NX X compression client library" "http://svn.berlios.de/svnroot/repos/freenx/trunk/freenx-client/nxcl/" FALSE "1.0" "Needed to build Krdc with NX support")
@@ -48,6 +45,9 @@
 
 endif(NOT INSIDE_KDENETWORK)
 
+macro_optional_find_package(LibVNCServer)
+macro_log_feature(LIBVNCSERVER_FOUND "libvncserver" "VNC Server library" "http://libvncserver.sourceforge.net/" FALSE "0.9" "Needed to build Krfb and VNC support in Krdc")
+
 include_directories(
     ${CMAKE_CURRENT_SOURCE_DIR}
     ${CMAKE_CURRENT_SOURCE_DIR}/core/
