Index: kdenetwork/kget/transfer-plugins/bittorrent/scandlg.cpp
===================================================================
--- kdenetwork/kget/transfer-plugins/bittorrent/scandlg.cpp	(revision 1138806)
+++ kdenetwork/kget/transfer-plugins/bittorrent/scandlg.cpp	(revision 1138807)
@@ -81,6 +81,11 @@
 		}
 	}
 
+	void ScanDlg::error(const QString &err)
+	{
+		KMessageBox::error(0,i18n("Error scanning data: %1",err));
+	}
+
 	void ScanDlg::execute(bt::TorrentInterface* tc,bool silently)
 	{
 		m_torrent_label->setText(i18n("Scanning data of <b>%1</b> :",tc->getStats().torrent_name));
Index: kdenetwork/kget/transfer-plugins/bittorrent/scandlg.h
===================================================================
--- kdenetwork/kget/transfer-plugins/bittorrent/scandlg.h	(revision 1138806)
+++ kdenetwork/kget/transfer-plugins/bittorrent/scandlg.h	(revision 1138807)
@@ -59,6 +59,9 @@
 		/// Handle the close event
 		virtual void closeEvent(QCloseEvent* e);
 
+		/// Report an error
+		virtual void error(const QString &err);
+
 	protected slots:
 		virtual void reject();
 		virtual void accept();
