--- capi4hylafax-01.03.00/src/standard/aStdLib.cpp.old	2013-05-21 16:55:37.031658830 +0200
+++ capi4hylafax-01.03.00/src/standard/aStdLib.cpp	2013-05-21 16:55:42.314905957 +0200
@@ -233,7 +233,7 @@
 
 tUInt a_pointer2string (void *pointer, tWiChar *string) {
     tULong    p   = (tULong) pointer;
-    tWiChar  *fmt = (sizeof (p) > 4) ? L"    .    :    .    " : L"    :    "; // must be symmetric!
+    const tWiChar  *fmt = (sizeof (p) > 4) ? L"    .    :    .    " : L"    :    "; // must be symmetric!
     unsigned i  = s_strlen (fmt);
 
     string[i] = '\0';
