https://bugzilla.mozilla.org/show_bug.cgi?id=436870

diff --git a/security/manager/ssl/src/nsCertOverrideService.h b/security/manager/ssl/src/nsCertOverrideService.h
--- a/security/manager/ssl/src/nsCertOverrideService.h
+++ b/security/manager/ssl/src/nsCertOverrideService.h
@@ -109,6 +109,7 @@
     nsCertOverrideEntry(const nsCertOverrideEntry& toCopy)
     {
       mSettings = toCopy.mSettings;
+      mHostWithPort = toCopy.mHostWithPort;
     }
 
     ~nsCertOverrideEntry()

