diff -up comm-1.9.2/mozilla/toolkit/system/gnome/nsAlertsIconListener.cpp.libnotify comm-1.9.2/mozilla/toolkit/system/gnome/nsAlertsIconListener.cpp
--- comm-1.9.2/mozilla/toolkit/system/gnome/nsAlertsIconListener.cpp.libnotify	2010-10-27 09:05:36.000000000 +0200
+++ comm-1.9.2/mozilla/toolkit/system/gnome/nsAlertsIconListener.cpp	2010-11-08 13:28:04.564002379 +0100
@@ -204,7 +204,7 @@ nsAlertsIconListener::ShowAlert(GdkPixbu
 {
   NotifyNotification* notify = notify_notification_new(mAlertTitle.get(),
                                                        mAlertText.get(),
-                                                       NULL, NULL);
+                                                       NULL);
   if (!notify)
     return NS_ERROR_OUT_OF_MEMORY;
 
