diff -ur gav-0.8.0/Makefile gav-0.8.0-new/Makefile
--- gav-0.8.0/Makefile	2003-02-11 06:47:40.000000000 -0500
+++ gav-0.8.0-new/Makefile	2004-04-12 07:26:57.000000000 -0400
@@ -55,8 +55,8 @@
 
 install: all
 	install gav $(ROOT)/usr/bin
-	install -d $(ROOT)/usr/share/games/gav/themes
-	cp -r themes/* $(ROOT)/usr/share/games/gav/themes
+	install -d $(ROOT)/usr/share/gav/themes
+	cp -r themes/* $(ROOT)/usr/share/gav/themes
 
 depend:
 	for i in $(SUBDIRS) ; do \
diff -ur gav-0.8.0/Theme.h gav-0.8.0-new/Theme.h
--- gav-0.8.0/Theme.h	2003-08-20 20:20:20.000000000 -0400
+++ gav-0.8.0-new/Theme.h	2004-04-12 07:26:19.000000000 -0400
@@ -81,7 +81,7 @@
 #ifndef WIN32
       DIR *dir;
       if ((dir = opendir(ThemeDir.c_str())) == NULL) {
-	ThemeDir = "/usr/share/games/gav/" + ThemeDir;
+	ThemeDir = "/usr/share/gav/" + ThemeDir;
 	if ((dir = opendir(ThemeDir.c_str())) == NULL) {
 	  std::cerr << "Cannot find themes directory\n";
 	  exit(0);
