diff -r -u libgda-1.2.0.old/providers/bdb/gda-bdb-recordset.c libgda-1.2.0/providers/bdb/gda-bdb-recordset.c
--- libgda-1.2.0.old/providers/bdb/gda-bdb-recordset.c	2003-12-17 11:27:02.000000000 +0000
+++ libgda-1.2.0/providers/bdb/gda-bdb-recordset.c	2005-02-19 15:43:12.000000000 +0000
@@ -227,10 +227,14 @@
 
 	/* get the number of records in the database */
 	ret = dbp->stat (dbp,
+#if BDB_VERSION > 40300
+			 NULL,
+#endif
 			 &statp,
 #if BDB_VERSION < 40000
 			 NULL,
 #endif
+
 			 0);
 	if (ret != 0) {
 		gda_connection_add_error (cnc, gda_bdb_make_error (ret));
