diff --git i/deluge/log.py w/deluge/log.py
index 75e8308b5..fbe206ad9 100644
--- i/deluge/log.py
+++ w/deluge/log.py
@@ -86,25 +86,6 @@ def critical(self, msg, *args, **kwargs):
     def exception(self, msg, *args, **kwargs):
         yield LoggingLoggerClass.exception(self, msg, *args, **kwargs)
 
-    def findCaller(self, stack_info=False):  # NOQA: N802
-        f = logging.currentframe().f_back
-        rv = '(unknown file)', 0, '(unknown function)'
-        while hasattr(f, 'f_code'):
-            co = f.f_code
-            filename = os.path.normcase(co.co_filename)
-            if filename in (
-                __file__.replace('.pyc', '.py'),
-                defer.__file__.replace('.pyc', '.py'),
-            ):
-                f = f.f_back
-                continue
-            if common.PY2:
-                rv = (filename, f.f_lineno, co.co_name)
-            else:
-                rv = (filename, f.f_lineno, co.co_name, None)
-            break
-        return rv
-
 
 levels = {
     'info': logging.INFO,
