--- component.py	2008-11-23 10:28:01.000000000 +0530
+++ component-patched.py	2008-12-02 02:25:29.619655571 +0530
@@ -169,7 +169,8 @@
         for component in self.components.keys():
             log.debug("Shutting down component %s..", component)
             try:
-                self.components[component].shutdown()
+                if self.components[component].get_state() != COMPONENT_STATE.index("Stopped"):
+                    self.components[component].shutdown()
             except Exception, e:
                 log.debug("Unable to call shutdown(): %s", e)
 

