--- makefile.unix.orig	2006-03-02 20:33:13.000000000 -0500
+++ makefile.unix	2006-03-02 20:47:27.000000000 -0500
@@ -127,7 +127,7 @@
 ###########################################################################
 
 # GCC on x86
-CFLAGS = -O2
+#CFLAGS = -O2
 
 # GCC on x86 with some optimizations
 # CFLAGS = -O2 -mtune=i686 -fomit-frame-pointer -fstrength-reduce -ffast-math
@@ -468,9 +468,9 @@
 
 # OSes that don't have an install command should use the provided 
 # install.sh script.
-INSTALL =              /usr/bin/install
+# INSTALL =              /usr/bin/install
 # INSTALL =            install-sh
-# INSTALL =            /bin/install
+INSTALL =            /bin/install
 # INSTALL =            install      # e.g., for BeOS 
 
 # The user and group to be used for installation.
@@ -658,7 +658,7 @@
 # JOY_PS2 = 1
 
 # SDL joystick support.
-# JOY_SDL = 1
+JOY_SDL = 1
 
 # This will enable the use of the Happ Controls UGCI(tm) USB devices'
 # Coin/Start inputs using the libugci library available from
@@ -701,10 +701,10 @@
 # use SDL.
 
 # X11 (including Glide & OpenGL if enabled below)
-DISPLAY_METHOD = x11
+# DISPLAY_METHOD = x11
 
 # SDL library 
-# DISPLAY_METHOD = SDL
+DISPLAY_METHOD = SDL
 
 # SVGALib, only supported under Linux
 # DISPLAY_METHOD = svgalib
@@ -752,12 +752,12 @@
 # Choose the location of your X headers and libraries.
 
 # Standard location for X11 
-# X11INC		= -I/usr/include/X11
-# X11LIB		= -L/usr/lib/X11
+X11INC		= -I/usr/include/X11
+X11LIB		= -L/usr/lib/X11
 
 # Standard location for XFree86
-X11INC		= -I/usr/X11R6/include
-X11LIB		= -L/usr/X11R6/lib
+# X11INC		= -I/usr/X11R6/include
+# X11LIB		= -L/usr/X11R6/lib
 
 # Standard location for Sun systems
 # X11INC		= -I/usr/openwin/include
