From d906981b20d41466d040e6cc2570099c8d665f7f Mon Sep 17 00:00:00 2001
From: Dave Reisner <dreisner@archlinux.org>
Date: Sat, 28 Jan 2012 08:05:02 -0500
Subject: [PATCH] include headers for tcp-keepalives

---
 src/tool_cb_skt.c |    8 ++++++++
 1 files changed, 8 insertions(+), 0 deletions(-)

diff --git a/src/tool_cb_skt.c b/src/tool_cb_skt.c
index 156c110..e74d5e4 100644
--- a/src/tool_cb_skt.c
+++ b/src/tool_cb_skt.c
@@ -27,6 +27,14 @@
 #  include <sys/socket.h>
 #endif
 
+#ifdef HAVE_NETINET_TCP_H
+#  include <netinet/tcp.h>
+#endif
+
+#ifdef HAVE_NETINET_IN_H
+#  include <netinet/in.h>
+#endif
+
 #define ENABLE_CURLX_PRINTF
 /* use our own printf() functions */
 #include "curlx.h"
-- 
1.7.8.4

