diff -u -r btrfs-progs-v4.20.1/utils.c btrfs-progs-v4.20.1-issue162/utils.c
--- btrfs-progs-v4.20.1/utils.c	2019-01-23 12:56:53.000000000 +0000
+++ btrfs-progs-v4.20.1-issue162/utils.c	2019-02-02 22:56:06.320029310 +0000
@@ -2079,7 +2079,7 @@
 	while ((ent = getmntent(mnttab))) {
 		len = strlen(ent->mnt_dir);
 		if (strncmp(ent->mnt_dir, path, len) == 0 &&
-		    (path[len] == '/' || path[len] == '\0')) {
+		    (len == 1 || path[len] == '/' || path[len] == '\0')) {
 			/* match found and use the latest match */
 			if (longest_matchlen <= len) {
 				free(longest_match);
