--- ext/amrnb/amrnbdec.h.orig	2012-03-12 16:49:14.417405139 +0000
+++ ext/amrnb/amrnbdec.h	2012-03-12 16:49:35.393899419 +0000
@@ -22,7 +22,7 @@
 
 #include <gst/gst.h>
 #include <gst/audio/gstaudiodecoder.h>
-#include <interf_dec.h>
+#include <opencore-amrnb/interf_dec.h>
 
 G_BEGIN_DECLS
 
--- ext/amrnb/amrnbenc.h.orig	2012-03-12 16:49:23.833996062 +0000
+++ ext/amrnb/amrnbenc.h	2012-03-12 16:49:46.597141876 +0000
@@ -21,7 +21,7 @@
 #define __GST_AMRNBENC_H__
 
 #include <gst/gst.h>
-#include <interf_enc.h>
+#include <opencore-amrnb/interf_enc.h>
 #include <gst/audio/gstaudioencoder.h>
 
 G_BEGIN_DECLS
--- ext/amrwbdec/amrwbdec.h.orig	2012-03-12 16:47:48.658100992 +0000
+++ ext/amrwbdec/amrwbdec.h	2012-03-12 16:48:12.177910733 +0000
@@ -22,8 +22,8 @@
 
 #include <gst/gst.h>
 #include <gst/audio/gstaudiodecoder.h>
-#include <dec_if.h>
-#include <if_rom.h>
+#include <opencore-amrwb/dec_if.h>
+#include <opencore-amrwb/if_rom.h>
 
 G_BEGIN_DECLS
 
