--- nototools/unicode_data.py.orig	2015-10-20 19:21:32.611686276 +0000
+++ nototools/unicode_data.py	2015-10-20 19:22:40.021351354 +0000
@@ -329,8 +329,7 @@
     return frozenset(_script_code_to_long_name.keys())
 
 
-_DATA_DIR_PATH = path.join(path.abspath(path.dirname(__file__)),
-                           os.pardir, "third_party", "ucd")
+_DATA_DIR_PATH = "/usr/share/nototools/third_party/ucd"
 
 
 def open_unicode_data_file(data_file_name):
