diff -Nur libimobiledevice-1.0.3.old/configure libimobiledevice-1.0.3/configure
--- libimobiledevice-1.0.3.old/configure	2010-10-04 07:38:37.000000000 -0700
+++ libimobiledevice-1.0.3/configure	2010-10-08 05:43:05.466686656 -0700
@@ -3477,9 +3477,7 @@
                         if test -z "$available_patch" ; then
                                 available_patch=0
                         fi
-                        if test $available_major -ne $required_major \
-                                -o $available_minor -ne $required_minor \
-                                -o $available_patch -lt $required_patch ; then
+                        if test $available_major -lt $required_major ; then
                                 { $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: SWIG version >= 1.3.21 is required.  You have $swig_version.  You should look at http://www.swig.org" >&5
 $as_echo "$as_me: WARNING: SWIG version >= 1.3.21 is required.  You have $swig_version.  You should look at http://www.swig.org" >&2;}
                                 SWIG=false
