diff -u -r pip-18.1/src/pip/_internal/utils/ui.py pip-18.1-p/src/pip/_internal/utils/ui.py
--- pip-18.1/src/pip/_internal/utils/ui.py	2018-10-05 09:07:26.000000000 +0000
+++ pip-18.1-p/src/pip/_internal/utils/ui.py	2019-03-08 08:51:47.324230770 +0000
@@ -12,7 +12,7 @@
     Bar, ChargingBar, FillingCirclesBar, FillingSquaresBar, IncrementalBar,
     ShadyBar,
 )
-from pip._vendor.progress.helpers import HIDE_CURSOR, SHOW_CURSOR, WritelnMixin
+from pip._vendor.progress import HIDE_CURSOR, SHOW_CURSOR
 from pip._vendor.progress.spinner import Spinner
 
 from pip._internal.utils.compat import WINDOWS
@@ -241,7 +241,7 @@
 
 
 class DownloadProgressSpinner(WindowsMixin, InterruptibleMixin,
-                              DownloadProgressMixin, WritelnMixin, Spinner):
+                              DownloadProgressMixin, Spinner):
 
     file = sys.stdout
     suffix = "%(downloaded)s %(download_speed)s"
