--- trunk/msn/message.cpp	2009/03/18 21:57:57	95
+++ trunk/msn/message.cpp	2009/04/03 18:04:25	96
@@ -59,10 +59,13 @@
         if (formatHeader.empty())
             return formatInfo;
         
-        std::vector<std::string> parameters = splitString(formatHeader, "; ");
-        std::vector<std::string>::const_iterator i = parameters.begin();
+        std::vector<std::string> parameters = splitString(formatHeader, ";");
+        std::vector<std::string>::iterator i = parameters.begin();
         for (; i != parameters.end(); i++)
         {
+	    if (i->at(0) == ' ')
+	        i->erase(0, 1);
+
             std::vector<std::string> pair = splitString(*i, "=");
             if (pair.size() == 2)
                 formatInfo[decodeURL(pair[0])] = decodeURL(pair[1]);
