--- gcc/treelang/parse.y	2007-07-18 08:19:29.000000000 +0000
+++ gcc/treelang/parse.y.new	2008-11-20 06:40:00.013275486 +0000
@@ -369,7 +369,6 @@
   current_function = NULL;
 }
 ;
-
 variable_defs_opt:
 /* Nil.  */ {
   $$ = 0;
@@ -685,9 +684,9 @@
 
 expressions_with_commas_opt:
 /* Nil.  */ {
-$$ = 0
+$$ = 0;
 }
-|expressions_with_commas { $$ = $1 }
+|expressions_with_commas { $$ = $1; }
 ;
 
 expressions_with_commas:
