diff -Nur mrxvt-0.5.4.old/configure.ac mrxvt-0.5.4/configure.ac
--- mrxvt-0.5.4.old/configure.ac	2008-05-12 20:21:56.000000000 +0300
+++ mrxvt-0.5.4/configure.ac	2010-01-30 16:51:04.000000000 +0200
@@ -952,7 +952,7 @@
 if test "x$support_png" = "xyes"; then
     AC_CHECK_LIB(
         png,
-        png_check_sig,
+        png_sig_cmp,
         [],
         [support_png=no],
         [-lz -lm]
diff -Nur mrxvt-0.5.4.old/src/mpng.c mrxvt-0.5.4/src/mpng.c
--- mrxvt-0.5.4.old/src/mpng.c	2008-02-18 00:04:50.000000000 +0200
+++ mrxvt-0.5.4/src/mpng.c	2010-01-30 16:55:28.000000000 +0200
@@ -83,7 +83,7 @@
     display_depth = XDefaultDepth(display,XDefaultScreen(display));
 
     fread(sig, 1, 8, ifile);
-    if (!png_check_sig(sig, 8)){
+    if (png_sig_cmp(sig, 0, 8)){
       fclose(ifile);
       return -1;
     }
@@ -113,7 +113,7 @@
 
     if (png_depth < 8){
       if (png_color_type == PNG_COLOR_TYPE_GRAY ){
-        png_set_gray_1_2_4_to_8(png_ptr);
+        png_set_expand_gray_1_2_4_to_8(png_ptr);
         png_row_bytes = png_width;
       }else{
         png_set_expand(png_ptr);
