--- a/src/chmfile.h	2009-05-17 01:28:11.000000000 -0700
+++ b/src/chmfile.h	2009-05-17 01:28:11.000000000 -0700
@@ -24,6 +24,7 @@
 #define __CHMFILE_H_
 
 #include <config.h>
+#include <stdint.h>
 #include <chm_lib.h>
 #include <wx/filefn.h>
 #include <wx/string.h>
--- a/src/bitfiddle.inl	2009-05-17 01:29:06.000000000 -0700
+++ b/src/bitfiddle.inl	2009-05-17 01:29:06.000000000 -0700
@@ -23,7 +23,7 @@
 
 
 #include <wx/string.h>
-
+#include <stdint.h>
 
 #define FIXENDIAN16(x) (x = wxUINT16_SWAP_ON_BE(x))
 #define FIXENDIAN32(x) (x = wxUINT32_SWAP_ON_BE(x))
