diff -ruN mozilla/xulrunner/installer/libxul-embedding.pc.in mozilla/xulrunner/installer/libxul-embedding.pc.in
--- mozilla/xulrunner/installer/libxul-embedding.pc.in	2008-07-16 20:27:31.000000000 +0200
+++ mozilla/xulrunner/installer/libxul-embedding.pc.in	2008-07-16 21:43:47.000000000 +0200
@@ -1,7 +1,7 @@
 prefix=%prefix%
 sdkdir=%sdkdir%
 includedir=%includedir%
-idldir=%idldir%
+idldir=%idldir%/stable
 includetype=stable
 
 # Note: the default pkg-config cflags will only list include directories
diff -ruN mozilla/xulrunner/installer/libxul-embedding-unstable.pc.in mozilla/xulrunner/installer/libxul-embedding-unstable.pc.in
--- mozilla/xulrunner/installer/libxul-embedding-unstable.pc.in	2008-07-16 20:27:31.000000000 +0200
+++ mozilla/xulrunner/installer/libxul-embedding-unstable.pc.in	2008-07-16 21:44:06.000000000 +0200
@@ -1,7 +1,7 @@
 prefix=%prefix%
 sdkdir=%sdkdir%
 includedir=%includedir%
-idldir=%idldir%
+idldir=%idldir%/unstable
 includetype=unstable
 
 Name: libxul-embedding-unstable
diff -ruN mozilla/xulrunner/installer/libxul.pc.in mozilla/xulrunner/installer/libxul.pc.in
--- mozilla/xulrunner/installer/libxul.pc.in	2007-08-03 16:46:28.000000000 +0200
+++ mozilla/xulrunner/installer/libxul.pc.in	2008-07-16 21:44:33.000000000 +0200
@@ -1,7 +1,8 @@
 prefix=%prefix%
 sdkdir=%sdkdir%
+libdir=%libdir%
 includedir=%includedir%
-idldir=%idldir%
+idldir=%idldir%/stable
 includetype=stable
 
 # Note: the default pkg-config includes will only list include directories
@@ -13,4 +14,4 @@
 Version: %MOZILLA_VERSION%
 Requires: %NSPR_NAME% >= %NSPR_VERSION%
 Libs: -L${sdkdir}/lib -lxpcomglue_s -lxul -lxpcom
-Cflags: -I${includedir}/${includetype} -fshort-wchar
\ No newline at end of file
+Cflags: -I${includedir}/${includetype} -fshort-wchar
diff -ruN mozilla/xulrunner/installer/libxul-unstable.pc.in mozilla/xulrunner/installer/libxul-unstable.pc.in
--- mozilla/xulrunner/installer/libxul-unstable.pc.in	2008-01-05 07:50:34.000000000 +0100
+++ mozilla/xulrunner/installer/libxul-unstable.pc.in	2008-07-16 21:45:11.000000000 +0200
@@ -1,7 +1,8 @@
 prefix=%prefix%
 sdkdir=%sdkdir%
+libdir=%libdir%
 includedir=%includedir%
-idldir=%idldir%
+idldir=%idldir%/unstable
 includetype=unstable
 
 Name: libxul
diff -ruN mozilla/xulrunner/installer/Makefile.in mozilla/xulrunner/installer/Makefile.in
--- mozilla/xulrunner/installer/Makefile.in	2008-07-16 20:27:31.000000000 +0200
+++ mozilla/xulrunner/installer/Makefile.in	2008-07-16 21:45:51.000000000 +0200
@@ -125,6 +125,7 @@
 	-e "s|%includedir%|$(includedir)|" \
 	-e "s|%idldir%|$(idldir)|" \
 	-e "s|%sdkdir%|$(sdkdir)|" \
+	-e "s|%libdir%|$(installdir)|" \
 	-e "s|%MOZ_APP_NAME%|$(MOZ_APP_NAME)|" \
 	-e "s|%MOZILLA_VERSION%|$(MOZ_APP_VERSION)|" \
 	-e "s|%FULL_NSPR_LIBS%|$(FULL_NSPR_LIBS)|" \
diff -ruN mozilla/xulrunner/installer/mozilla-gtkmozembed-embedding.pc.in mozilla/xulrunner/installer/mozilla-gtkmozembed-embedding.pc.in
--- mozilla/xulrunner/installer/mozilla-gtkmozembed-embedding.pc.in	2008-01-10 10:17:49.000000000 +0100
+++ mozilla/xulrunner/installer/mozilla-gtkmozembed-embedding.pc.in	2008-07-16 21:46:14.000000000 +0200
@@ -1,5 +1,6 @@
 prefix=%prefix%
 sdkdir=%sdkdir%
+libdir=%libdir%
 includedir=%includedir%
 
 Name: mozilla-gtkembedmoz
diff -ruN mozilla/xulrunner/installer/mozilla-gtkmozembed.pc.in mozilla/xulrunner/installer/mozilla-gtkmozembed.pc.in
--- mozilla/xulrunner/installer/mozilla-gtkmozembed.pc.in	2008-01-10 10:17:49.000000000 +0100
+++ mozilla/xulrunner/installer/mozilla-gtkmozembed.pc.in	2008-07-16 21:46:28.000000000 +0200
@@ -1,5 +1,6 @@
 prefix=%prefix%
 sdkdir=%sdkdir%
+libdir=%libdir%
 includedir=%includedir%
 
 Name: mozilla-gtkembedmoz
