--- webkit-1.2.5/JavaScriptCore/wtf/Platform.h	2010-09-10 15:20:23.000000000 +0200
+++ webkit-1.2.5/JavaScriptCore/wtf/Platform.h.new	2010-12-06 10:49:06.000000000 +0100
@@ -775,7 +775,7 @@
 /* ENABLE macro defaults */
 
 #if PLATFORM(QT)
-// We musn't customize the global operator new and delete for the Qt port.
+/* We musn't customize the global operator new and delete for the Qt port. */
 #define ENABLE_GLOBAL_FASTMALLOC_NEW 0
 #endif
 
