diff -Naur pekwm-0.1.11-orig/src/PImageLoaderPng.cc pekwm-0.1.11/src/PImageLoaderPng.cc
--- pekwm-0.1.11-orig/src/PImageLoaderPng.cc	2010-01-22 01:57:08.000000000 -0500
+++ pekwm-0.1.11/src/PImageLoaderPng.cc	2010-01-22 02:00:02.000000000 -0500
@@ -108,7 +108,7 @@
     
     // gray -> 8 bit gray
     if (color_type == PNG_COLOR_TYPE_GRAY && (bpp < 8)) {
-        png_set_gray_1_2_4_to_8(png_ptr);
+        png_set_expand_gray_1_2_4_to_8(png_ptr);
     }
 
     if (png_get_valid(png_ptr, info_ptr, PNG_INFO_tRNS)) {
@@ -179,7 +179,7 @@
 
     status = fread(sig, 1, PImageLoaderPng::PNG_SIG_BYTES, fp);
     if (status == PImageLoaderPng::PNG_SIG_BYTES) {
-        return (png_check_sig(sig, PImageLoaderPng::PNG_SIG_BYTES) != 0);
+      return (png_sig_cmp(sig, 0, PImageLoaderPng::PNG_SIG_BYTES) == 0);
     }
     return false;
 }
