diff -Naur mplayerplug-in.orig/configure.in mplayerplug-in/configure.in
--- mplayerplug-in.orig/configure.in	2007-12-02 04:53:04.000000000 +0530
+++ mplayerplug-in/configure.in	2008-04-11 00:30:26.000000000 +0530
@@ -54,9 +54,10 @@
 
 ],[
 	AC_MSG_NOTICE([Determining mozilla/firefox packages to build against])
-	PKG_CHECK_MODULES(MOZPLUG,[mozilla-plugin],[AC_SUBST(MOZPLUG_CFLAGS) MOZPLUG_CFLAGS="${MOZPLUG_CFLAGS} -I`$PKG_CONFIG --variable=includedir mozilla-plugin`"
-MOZILLA_HOME=`$PKG_CONFIG --variable=libdir mozilla-plugin`
-],[AC_MSG_WARN([mozilla-plugin not found])])
+	# There's no www-client/mozilla anymore, and this interferes with =net-libs/xulrunner-1.9* now
+#	PKG_CHECK_MODULES(MOZPLUG,[mozilla-plugin],[AC_SUBST(MOZPLUG_CFLAGS) MOZPLUG_CFLAGS="${MOZPLUG_CFLAGS} -I`$PKG_CONFIG --variable=includedir mozilla-plugin`"
+#MOZILLA_HOME=`$PKG_CONFIG --variable=libdir mozilla-plugin`
+#],[AC_MSG_WARN([mozilla-plugin not found])])
 
 	if test "$MOZPLUG_CFLAGS" = ""; then
 		PKG_CHECK_MODULES(MOZPLUG,[firefox-plugin firefox-xpcom],[AC_SUBST(MOZPLUG_CFLAGS) MOZPLUG_CFLAGS="${MOZPLUG_CFLAGS} -I`$PKG_CONFIG --variable=includedir firefox-plugin`"
