diff -Naur showimg-0.9.5-orig/showimg/kexi/kexidb/drivers/pqxx/pqxxcursor.cpp showimg-0.9.5/showimg/kexi/kexidb/drivers/pqxx/pqxxcursor.cpp
--- showimg-0.9.5-orig/showimg/kexi/kexidb/drivers/pqxx/pqxxcursor.cpp	2006-01-07 03:47:46.000000000 -0500
+++ showimg-0.9.5/showimg/kexi/kexidb/drivers/pqxx/pqxxcursor.cpp	2008-07-20 14:27:27.000000000 -0400
@@ -17,6 +17,7 @@
 
 #include <klocale.h>
 #include <kdebug.h>
+#include <cstdlib>
 
 using namespace KexiDB;
 
diff -Naur showimg-0.9.5-orig/showimg/kexi/kexidb/parser/sqlparser.cpp showimg-0.9.5/showimg/kexi/kexidb/parser/sqlparser.cpp
--- showimg-0.9.5-orig/showimg/kexi/kexidb/parser/sqlparser.cpp	2006-01-07 03:47:45.000000000 -0500
+++ showimg-0.9.5/showimg/kexi/kexidb/parser/sqlparser.cpp	2008-07-20 14:26:37.000000000 -0400
@@ -359,6 +359,7 @@
 #include <iostream>
 #include <assert.h>
 #include <limits.h>
+#include <cstdlib>
 //TODO OK?
 #ifdef Q_WS_WIN
 //workaround for bug on msvc
diff -Naur showimg-0.9.5-orig/showimg/kexi/kexidb/parser/sqlparser.y showimg-0.9.5/showimg/kexi/kexidb/parser/sqlparser.y
--- showimg-0.9.5-orig/showimg/kexi/kexidb/parser/sqlparser.y	2006-01-07 03:47:45.000000000 -0500
+++ showimg-0.9.5/showimg/kexi/kexidb/parser/sqlparser.y	2008-07-20 14:26:37.000000000 -0400
@@ -440,6 +440,8 @@
 #include <iostream>
 #include <assert.h>
 #include <limits.h>
+#include <alloca.h>
+#include <cstdlib>
 //TODO OK?
 #ifdef Q_WS_WIN
 //workaround for bug on msvc
diff -Naur showimg-0.9.5-orig/showimg/showimg/imagelistview.cpp showimg-0.9.5/showimg/showimg/imagelistview.cpp
--- showimg-0.9.5-orig/showimg/showimg/imagelistview.cpp	2006-01-07 03:48:04.000000000 -0500
+++ showimg-0.9.5/showimg/showimg/imagelistview.cpp	2008-07-20 14:26:37.000000000 -0400
@@ -49,6 +49,9 @@
 #include "imageviewer.h"
 #include "imagemetainfo.h"
 
+#include <cstdlib>
+#include <climits>
+
 #ifdef HAVE_KIPI
 #include "kipiplugins/kipipluginmanager.h"
 #endif /* HAVE_KIPI */
