--- hplip-3.19.11/pcard/pcardext/pcardext.c	2019-11-04 11:31:08.000000000 +0100
+++ hplip-3.19.11/pcard/pcardext/pcardext.c.new	2019-12-11 12:13:20.710959557 +0100
@@ -80,7 +80,7 @@
     {
         result = PyObject_CallFunction( writesectorFunc, "iis#", sector, nsector, buf, size );
         
-        return PyInt_AS_LONG( result );
+        return PyLong_AS_LONG( result );
     }
 
     return 1;
