diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 255c9f1..10f71cc 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -411,5 +411,12 @@ install(EXPORT KPim${KF_MAJOR_VERSION}TextEditTargets
     FILE KPim${KF_MAJOR_VERSION}TextEditTargets.cmake
     NAMESPACE KF5::
 )
+if (BUILD_QCH)
+    install(FILES
+        "${CMAKE_CURRENT_BINARY_DIR}/KPim${KF_MAJOR_VERSION}TextEditQchTargets.cmake"
+        DESTINATION "${CMAKECONFIG_INSTALL_DIR_KF5}"
+        COMPONENT Devel
+    )
+endif()
 endif()
 endif()
