diff --git i/configure.ac w/configure.ac
index 7bbcab84..15b00485 100644
--- i/configure.ac
+++ w/configure.ac
@@ -163,32 +163,28 @@ GLIB_GENMARSHAL="`$PKG_CONFIG --variable=glib_genmarshal glib-2.0`"
 AC_SUBST(GLIB_GENMARSHAL)
 
 
-AX_REQUIRE_DEFINED([GTK_DOC_CHECK])
-GTK_DOC_CHECK([1.13],[--flavour no-tmpl])
-
 AC_CONFIG_FILES([
 Makefile
 data/dtd/catalog
 data/xslt/db2html.xsl
 data/xslt/info2html.xsl
 data/xslt/links2html.xsl
 data/xslt/mal2html.xsl
 data/xslt/man2html.xsl
 data/xslt/yelp-common.xsl
 po/Makefile.in
 ])
 
 AC_OUTPUT
 
 dnl ==========================================================================
 echo "
 yelp-$VERSION:
 
 	prefix:                   ${prefix}
 	source code location:	  ${srcdir}
 	compiler:		  ${CC}
 
-	Documentation:            ${enable_gtk_doc}
 	Debug enabled:            ${enable_debug}
 	Enable BZ2:               ${enable_bz2}
 	Enable LZMA:              ${enable_lzma}
