From 835d5dee8aaf078960b16ebd9018da095b1bcfbd Mon Sep 17 00:00:00 2001
From: Matthew Barnes <mbarnes@redhat.com>
Date: Mon, 21 Sep 2009 14:37:33 +0000
Subject: Fix an editing crash.

Apparently G_VALUE_HOLDS() doesn't like NULL GValues.
---
diff --git a/components/editor/gtkhtml-editor.c b/components/editor/gtkhtml-editor.c
index aca15f9..58ae08f 100644
--- a/components/editor/gtkhtml-editor.c
+++ b/components/editor/gtkhtml-editor.c
@@ -384,7 +384,7 @@ editor_method_event (GtkHTML *html,
 	guint signal_id;
 
 	/* GtkHTML event arguments are either NULL or a single string. */
-	if (G_VALUE_HOLDS (args, G_TYPE_STRING))
+	if (args != NULL && G_VALUE_HOLDS (args, G_TYPE_STRING))
 		string = g_value_get_string (args);
 
 	switch (event) {
--
cgit v0.8.2
