Index: transcode-1.0.6/configure.in
===================================================================
--- transcode-1.0.6.orig/configure.in
+++ transcode-1.0.6/configure.in
@@ -1247,7 +1247,7 @@ dnl
 IMAGEMAGICK_EXTRA_LIBS="$IMAGEMAGICK_EXTRA_LIBS -lz -lm"
 dnl MagickWand libraries/headers always (?) are a superset of MagickCore ones.
 TC_PKG_CHECK(imagemagick, no, IMAGEMAGICK, Wand-config, [magick/api.h],
- Magick, InitializeMagick, Wand, [http://www.imagemagick.org/])
+ MagickWand, InitializeMagick, Wand, [http://www.imagemagick.org/])
 TC_PKG_HAVE(imagemagick, IMAGEMAGICK)
 if test x"$have_imagemagick" = x"yes" ; then
   AC_CHECK_HEADER([wand/magick-wand.h],
