diff -p -up tcsh-6.15.00/tw.color.c.ls-colors tcsh-6.15.00/tw.color.c
--- tcsh-6.15.00/tw.color.c.ls-colors	2006-03-02 19:46:45.000000000 +0100
+++ tcsh-6.15.00/tw.color.c	2008-09-15 15:08:19.000000000 +0200
@@ -85,6 +85,9 @@ static Variable variables[] = {
     VAR(NOS, "tw", ""),		/* Sticky and other writable dir (+t,o+w) */
     VAR(NOS, "ow", ""),		/* Other writable dir (o+w) but not sticky */
     VAR(NOS, "st", ""),		/* Sticky dir (+t) but not other writable */
+    VAR(NOS, "rs", ""),		/* Reset to "normal" color */
+    VAR(NOS, "hl", ""),		/* Regular file with more than one link */
+    VAR(NOS, "ca", ""),		/* File with capability */
 };
 
 enum FileType {
