diff -up amarok-2.8.0/src/context/applets/CMakeLists.txt.webkit amarok-2.8.0/src/context/applets/CMakeLists.txt
diff -up amarok-2.8.0/src/context/applets/info/CMakeLists.txt.webkit amarok-2.8.0/src/context/applets/info/CMakeLists.txt
--- amarok-2.8.0/src/context/applets/info/CMakeLists.txt.webkit	2013-08-14 15:22:52.000000000 -0500
+++ amarok-2.8.0/src/context/applets/info/CMakeLists.txt	2014-06-18 13:35:31.496217418 -0500
@@ -16,7 +16,6 @@ target_link_libraries(amarok_context_app
     amarokcore
     amaroklib
     ${KDE4_PLASMA_LIBS}
-    ${KDE4_KDEWEBKIT_LIBS}
     ${QT_QTWEBKIT_LIBRARY}
     )
 
diff -up amarok-2.8.0/src/context/applets/info/InfoApplet.cpp.webkit amarok-2.8.0/src/context/applets/info/InfoApplet.cpp
--- amarok-2.8.0/src/context/applets/info/InfoApplet.cpp.webkit	2013-08-14 15:22:52.000000000 -0500
+++ amarok-2.8.0/src/context/applets/info/InfoApplet.cpp	2014-06-18 14:10:15.337284616 -0500
@@ -26,7 +26,7 @@
 #include "PaletteHandler.h"
 #include "playlist/PlaylistController.h"
 
-#include <KGraphicsWebView>
+#include <QGraphicsWebView>
 #include <KStandardDirs>
 
 #include <QPainter>
@@ -63,7 +63,7 @@ void InfoApplet::init()
 
     dataEngine( "amarok-info" )->connectSource( "info", this );
 
-    m_webView = new KGraphicsWebView( this );
+    m_webView = new QGraphicsWebView( this );
 
     QPalette p = m_webView->palette();
     p.setColor( QPalette::Dark, QColor( 255, 255, 255, 0)  );
diff -up amarok-2.8.0/src/context/applets/info/InfoApplet.h.webkit amarok-2.8.0/src/context/applets/info/InfoApplet.h
--- amarok-2.8.0/src/context/applets/info/InfoApplet.h.webkit	2013-08-14 15:22:52.000000000 -0500
+++ amarok-2.8.0/src/context/applets/info/InfoApplet.h	2014-06-18 13:35:31.496217418 -0500
@@ -27,7 +27,7 @@
 
 #include <QGraphicsProxyWidget>
 
-class KGraphicsWebView;
+class QGraphicsWebView;
 
 class InfoApplet : public Context::Applet
 {
@@ -47,7 +47,7 @@ private slots:
     void linkClicked( const QUrl & url );
 
 private:
-    KGraphicsWebView *m_webView;
+    QGraphicsWebView *m_webView;
     bool m_initialized;
 
     static QString s_defaultHtml;
diff -up amarok-2.8.0/src/context/applets/wikipedia/CMakeLists.txt.webkit amarok-2.8.0/src/context/applets/wikipedia/CMakeLists.txt
--- amarok-2.8.0/src/context/applets/wikipedia/CMakeLists.txt.webkit	2014-06-18 14:07:24.355163151 -0500
+++ amarok-2.8.0/src/context/applets/wikipedia/CMakeLists.txt	2014-06-18 14:07:34.347695400 -0500
@@ -18,7 +18,6 @@ target_link_libraries(amarok_context_app
     amaroklib
     ${KDE4_PLASMA_LIBS}
     ${KDE4_KIO_LIBS}
-    ${KDE4_KDEWEBKIT_LIBS}
     ${QT_QTWEBKIT_LIBRARY}
 )
 
diff -up amarok-2.8.0/src/context/applets/wikipedia/WikipediaApplet_p.h.webkit amarok-2.8.0/src/context/applets/wikipedia/WikipediaApplet_p.h
--- amarok-2.8.0/src/context/applets/wikipedia/WikipediaApplet_p.h.webkit	2013-08-14 15:22:52.000000000 -0500
+++ amarok-2.8.0/src/context/applets/wikipedia/WikipediaApplet_p.h	2014-06-18 13:56:15.446760436 -0500
@@ -23,7 +23,7 @@
 #include "ui_wikipediaGeneralSettings.h"
 #include "ui_wikipediaLanguageSettings.h"
 
-#include <KGraphicsWebView>
+#include <QGraphicsWebView>
 #include <KLineEdit>
 #include <KUrl>
 
@@ -180,13 +180,13 @@ protected:
     }
 };
 
-class WikipediaWebView : public KGraphicsWebView
+class WikipediaWebView : public QGraphicsWebView
 {
     Q_OBJECT
 
 public:
     WikipediaWebView( QGraphicsItem *parent = 0 )
-        : KGraphicsWebView( parent )
+        : QGraphicsWebView( parent )
     {
         m_lineEdit = new WikipediaSearchLineEdit( this );
         m_lineEdit->setContentsMargins( 0, 0, 0, 0 );
@@ -242,7 +242,7 @@ protected:
             }
             return false;
         }
-        return KGraphicsWebView::eventFilter( obj, event );
+        return QGraphicsWebView::eventFilter( obj, event );
     }
 
     void keyPressEvent( QKeyEvent *event )
@@ -258,12 +258,12 @@ protected:
             event->accept();
         }
         else
-            KGraphicsWebView::keyPressEvent( event );
+            QGraphicsWebView::keyPressEvent( event );
     }
 
     void resizeEvent( QGraphicsSceneResizeEvent *event )
     {
-        KGraphicsWebView::resizeEvent( event );
+        QGraphicsWebView::resizeEvent( event );
         if( m_topBorder )
         {
             m_topBorder->resize( event->newSize().width(), m_topBorder->size().height() );
