--- a/src/context/engines/photos/PhotosEngine.cpp
+++ b/src/context/engines/photos/PhotosEngine.cpp
@@ -156,7 +156,7 @@ void PhotosEngine::update()
         // Show the information
         if( !m_artist.isEmpty() )
         {
-            setData( "photos", "message", i18n( "Fetching" ) );
+            setData( "photos", "message", "Fetching");
             setData( "photos", "artist", m_artist );
         }
         else