diff -Naur old/fontforge/python.c new/fontforge/python.c
--- old/fontforge/python.c	2015-06-12 10:48:13.000000000 -1000
+++ new/fontforge/python.c	2015-09-19 08:44:21.523111860 -1000
@@ -18175,7 +18175,7 @@
     mdef->runtime.pymod_def.m_doc = mdef->docstring;
     mdef->runtime.pymod_def.m_methods = mdef->methods;
     mdef->runtime.pymod_def.m_size = -1;
-    mdef->runtime.pymod_def.m_reload = NULL;
+    mdef->runtime.pymod_def.m_slots = NULL;
     mdef->runtime.pymod_def.m_traverse = NULL;
     mdef->runtime.pymod_def.m_clear = NULL;
     mdef->runtime.pymod_def.m_free = NULL;
