--- xine-lib-1.1.15.orig/src/libxinevdec/image.c
+++ xine-lib-1.1.15/src/libxinevdec/image.c
@@ -110,7 +110,7 @@
     width = MagickGetImageWidth(wand) & ~1; /* must be even for init_yuv_planes */
     height = MagickGetImageHeight(wand);
     img_buf = malloc(width * height * 3);
-    MagickGetImagePixels(wand, 0, 0, width, height, "RGB", CharPixel, img_buf);
+    MagickGetAuthenticPixels(wand, 0, 0, width, height, "RGB", CharPixel, img_buf);
     DestroyMagickWand(wand);
 
     _x_stream_info_set(this->stream, XINE_STREAM_INFO_VIDEO_WIDTH, width);
