--- etc/rc.d/avahi-daemon	2010-02-26 21:10:30.000000000 -0600
+++ etc/rc.d/avahi-daemon	2010-02-26 21:09:44.000000000 -0600
@@ -35,8 +35,11 @@
 
 case "$1" in
     start)
-        stat_busy "Starting $DESC"
-        $DAEMON -D > /dev/null 2>&1
+	if ck_daemon dbus; then
+		/etc/rc.d/dbus start
+	fi
+    	stat_busy "Starting $DESC"
+        modprobe capability  > /dev/null 2>&1 ;  $DAEMON -D > /dev/null 2>&1
         if [ $? -gt 0 ]; then
             stat_fail
         else
