diff -up openjpeg-1.5.1/doc/Doxyfile.dox.cmake.in.doxygen_timestamp openjpeg-1.5.1/doc/Doxyfile.dox.cmake.in
--- openjpeg-1.5.1/doc/Doxyfile.dox.cmake.in.doxygen_timestamp	2012-09-13 02:58:39.000000000 -0500
+++ openjpeg-1.5.1/doc/Doxyfile.dox.cmake.in	2012-12-06 15:23:35.079838524 -0600
@@ -148,7 +148,7 @@ HTML_STYLESHEET        =
 HTML_COLORSTYLE_HUE    = 220
 HTML_COLORSTYLE_SAT    = 100
 HTML_COLORSTYLE_GAMMA  = 80
-HTML_TIMESTAMP         = YES
+HTML_TIMESTAMP         = NO 
 HTML_ALIGN_MEMBERS     = YES
 HTML_DYNAMIC_SECTIONS  = NO
 GENERATE_DOCSET        = NO
diff -up openjpeg-1.5.1/doc/Doxyfile.dox.doxygen_timestamp openjpeg-1.5.1/doc/Doxyfile.dox
--- openjpeg-1.5.1/doc/Doxyfile.dox.doxygen_timestamp	2012-09-13 02:58:39.000000000 -0500
+++ openjpeg-1.5.1/doc/Doxyfile.dox	2012-12-06 15:23:37.177813275 -0600
@@ -147,7 +147,7 @@ HTML_STYLESHEET        =
 HTML_COLORSTYLE_HUE    = 220
 HTML_COLORSTYLE_SAT    = 100
 HTML_COLORSTYLE_GAMMA  = 80
-HTML_TIMESTAMP         = YES
+HTML_TIMESTAMP         = NO 
 HTML_ALIGN_MEMBERS     = YES
 HTML_DYNAMIC_SECTIONS  = NO
 GENERATE_DOCSET        = NO
