From 2877c3a41b7e8feeb6eb6d6f718f4556ffb0161c Mon Sep 17 00:00:00 2001
From: Dominique Leuenberger <dimstar@opensuse.org>
Date: Mon, 05 Dec 2011 08:10:59 +0000
Subject: Bug 665309: Adapt to gnome-bluetooth 3.3 API

---
diff --git a/configure.in b/configure.in
index 101a77f..d0a7651 100644
--- a/configure.in
+++ b/configure.in
@@ -13,7 +13,7 @@ dnl Initialize libtool
 AM_PROG_LIBTOOL
 
 GNOKII_REQS="gnokii >= 0.6.28"
-GNOME_BLUETOOTH_REQS="gnome-bluetooth-1.0 >= 3.0"
+GNOME_BLUETOOTH_REQS="gnome-bluetooth-1.0 >= 3.3"
 
 AC_PROG_CC
 AC_PROG_INSTALL
diff --git a/gnome-bluetooth/phonemgr.c b/gnome-bluetooth/phonemgr.c
index 9ebac12..fecc367 100644
--- a/gnome-bluetooth/phonemgr.c
+++ b/gnome-bluetooth/phonemgr.c
@@ -44,7 +44,7 @@ get_type (const char *address)
 	gboolean cont;
 
 	client = bluetooth_client_new (); 
-	model = bluetooth_client_get_device_model (client, NULL);
+	model = bluetooth_client_get_device_model (client);
 	if (model == NULL) {
 		g_object_unref (client);
 		return type;
--
cgit v0.9.0.2
