diff -wbBur celestia-1.4.1/configure celestia-1.4.1.my/configure
--- celestia-1.4.1/configure	2006-02-16 00:53:09.000000000 +0300
+++ celestia-1.4.1.my/configure	2006-06-03 13:40:37.000000000 +0400
@@ -39509,7 +39509,7 @@
 echo $ECHO_N "checking whether to enable Celestia Extension Language... $ECHO_C" >&6
 if (test "$enable_lua" != "no"); then
 	CXXFLAGS="$CXXFLAGS -DCELX"
-	LDFLAGS="$LDFLAGS -llualib -llua"
+	LDFLAGS="$LDFLAGS -llua"
 
 	if (test "$lua_includes" != "no"); then
 		CXXFLAGS="$CXXFLAGS -I$lua_includes"
diff -wbBur celestia-1.4.1/configure.in celestia-1.4.1.my/configure.in
--- celestia-1.4.1/configure.in	2006-02-03 21:09:54.000000000 +0300
+++ celestia-1.4.1.my/configure.in	2006-06-03 13:40:28.000000000 +0400
@@ -383,7 +383,7 @@
 AC_MSG_CHECKING([whether to enable Celestia Extension Language])
 if (test "$enable_lua" != "no"); then
 	CXXFLAGS="$CXXFLAGS -DCELX"
-	LDFLAGS="$LDFLAGS -llualib -llua"
+	LDFLAGS="$LDFLAGS -llua"
 	
 	if (test "$lua_includes" != "no"); then
 		CXXFLAGS="$CXXFLAGS -I$lua_includes"
diff -wbBur celestia-1.4.1/src/celestia/celx.cpp celestia-1.4.1.my/src/celestia/celx.cpp
--- celestia-1.4.1/src/celestia/celx.cpp	2006-01-07 03:01:51.000000000 +0300
+++ celestia-1.4.1.my/src/celestia/celx.cpp	2006-06-03 13:41:36.000000000 +0400
@@ -32,9 +32,7 @@
 
 #include "celx.h"
 #include "celestiacore.h"
-extern "C" {
-#include "lualib.h"
-}
+#include "lua.hpp"
 
 using namespace std;
 
@@ -598,7 +596,7 @@
         int stackTop = lua_gettop(costate);
         if (strcmp(c_p, "y") == 0)
         {
-            lua_iolibopen(costate);
+            luaL_openlibs(costate);
             ioMode = IOAllowed;
         }
         else
@@ -4577,10 +4575,7 @@
     initMaps();
 
     // Import the base and math libraries
-    lua_baselibopen(state);
-    lua_mathlibopen(state);
-    lua_tablibopen(state);
-    lua_strlibopen(state);
+    luaL_openlibs(state);
 
     // Add an easy to use wait function, so that script writers can
     // live in ignorance of coroutines.  There will probably be a significant
