diff --git a/docs/libcurl/libcurl.m4 b/docs/libcurl/libcurl.m4
index ce8e476..50dc20d 100644
--- a/docs/libcurl/libcurl.m4
+++ b/docs/libcurl/libcurl.m4
@@ -198,7 +198,7 @@ x=CURLOPT_VERBOSE;
 
 	      # We don't have --protocols, so just assume that all
 	      # protocols are available
-	      _libcurl_protocols="HTTP FTP FILE TELNET LDAP DICT TFTP
+	      _libcurl_protocols="HTTP FTP FILE TELNET LDAP DICT TFTP"
 
 	      if test x$libcurl_feature_SSL = xyes ; then
 	         _libcurl_protocols="$_libcurl_protocols HTTPS"
