
diff -r 0377a1af6110 -r 1fc6ad6e7c7a src/foreign/vigra/vigra/multi_iterator.hxx
--- a/src/foreign/vigra/vigra/multi_iterator.hxx	Mon Jun 25 16:36:21 2007 +0000
+++ b/src/foreign/vigra/vigra/multi_iterator.hxx	Tue Jan 25 21:14:13 2011 +0000
@@ -41,6 +41,7 @@
 #define VIGRA_MULTI_ITERATOR_HXX
 
 #include <sys/types.h>
+#include <stddef.h>
 #include "tinyvector.hxx"
 #include "iteratortags.hxx"
 




