diff -p -up avidemux_2.5.0/avidemux/ADM_core/src/ADM_fileio.cpp~ avidemux_2.5.0/avidemux/ADM_core/src/ADM_fileio.cpp
--- avidemux_2.5.0/avidemux/ADM_core/src/ADM_fileio.cpp~	2009-03-09 21:10:03.000000000 +0100
+++ avidemux_2.5.0/avidemux/ADM_core/src/ADM_fileio.cpp	2009-07-09 21:34:15.000000000 +0200
@@ -685,7 +685,7 @@ const char *ADM_GetFileName(const char *
 	char *filename2;
 
 #ifndef __WIN32
-	filename = strrchr(str, '/');
+	filename = strrchr((char*) str, '/');
 #else
 	filename = strrchr(str, '\\');
 	filename2 = strrchr(str, '/');
diff -p -up avidemux_2.5.0/avidemux/ADM_inputs/ADM_mpegdemuxer/dmx_io.cpp~ avidemux_2.5.0/avidemux/ADM_inputs/ADM_mpegdemuxer/dmx_io.cpp
--- avidemux_2.5.0/avidemux/ADM_inputs/ADM_mpegdemuxer/dmx_io.cpp~	2008-09-13 18:27:58.000000000 +0200
+++ avidemux_2.5.0/avidemux/ADM_inputs/ADM_mpegdemuxer/dmx_io.cpp	2009-07-09 21:48:17.000000000 +0200
@@ -87,7 +87,7 @@ uint8_t fileParser::open( const char *fi
 
         
         // find the last dot
-        dot = strrchr( filename, '.' );
+        dot = strrchr( (char *) filename, '.' );
 
         // count the decimals before the dot
         decimals = 1;
diff -p -up avidemux_2.5.0/avidemux/ADM_libraries/ADM_libmpeg2enc/yuv4mpeg_ratio.cc~ avidemux_2.5.0/avidemux/ADM_libraries/ADM_libmpeg2enc/yuv4mpeg_ratio.cc
--- avidemux_2.5.0/avidemux/ADM_libraries/ADM_libmpeg2enc/yuv4mpeg_ratio.cc~	2004-01-30 20:32:00.000000000 +0100
+++ avidemux_2.5.0/avidemux/ADM_libraries/ADM_libmpeg2enc/yuv4mpeg_ratio.cc	2009-07-09 21:49:28.000000000 +0200
@@ -104,7 +104,7 @@ void y4m_ratio_reduce(y4m_ratio_t *r)
 
 int y4m_parse_ratio(y4m_ratio_t *r, const char *s)
 {
-  char *t = strchr(s, ':');
+  char *t = strchr((char *) s, ':');
   if (t == NULL) return Y4M_ERR_RANGE;
   r->n = atoi(s);
   r->d = atoi(t+1);
diff -p -up avidemux_2.5.0/avidemux/ADM_libraries/ADM_mplex/lpcmstrm_in.cpp~ avidemux_2.5.0/avidemux/ADM_libraries/ADM_mplex/lpcmstrm_in.cpp
--- avidemux_2.5.0/avidemux/ADM_libraries/ADM_mplex/lpcmstrm_in.cpp~	2007-08-03 02:04:40.000000000 +0200
+++ avidemux_2.5.0/avidemux/ADM_libraries/ADM_mplex/lpcmstrm_in.cpp	2009-07-09 21:50:06.000000000 +0200
@@ -53,7 +53,7 @@ LPCMStream::LPCMStream(IBitStream &ibs, 
 
 bool LPCMStream::Probe(IBitStream &bs )
 {
-    char *last_dot = strrchr( bs.StreamName(), '.' );
+    char *last_dot = strrchr( (char *) bs.StreamName(), '.' );
     return 
         last_dot != NULL 
         && strcmp( last_dot+1, "lpcm") == 0;
--- avidemux_2.5.0/plugins/ADM_videoEncoder/ADM_vidEnc_x264/guiHelper.cpp~	2009-07-10 13:39:52.745385459 +0200
+++ avidemux_2.5.0/plugins/ADM_videoEncoder/ADM_vidEnc_x264/guiHelper.cpp	2009-07-10 14:00:03.432445620 +0200
@@ -67,7 +67,7 @@
 	char *nextvar;
 	int i;
 
-	p = strstr(buf, name);
+	p = strstr((char *) buf, name);
 
 	if (!p)
 	{
