diff -u -r Vulkan-Loader-1.2.174/cmake/FindVulkanHeaders.cmake Vulkan-Loader-1.2.174-verfix/cmake/FindVulkanHeaders.cmake
--- Vulkan-Loader-1.2.174/cmake/FindVulkanHeaders.cmake	2021-03-29 20:05:47.000000000 +0000
+++ Vulkan-Loader-1.2.174-verfix/cmake/FindVulkanHeaders.cmake	2021-04-16 14:08:29.582548243 +0000
@@ -114,19 +114,19 @@
 file(STRINGS
         ${VulkanHeaders_main_header}
         VulkanHeaders_lines
-        REGEX "^#define (VK_API_VERSION.*VK_MAKE_VERSION|VK_HEADER_VERSION)")
+        REGEX "^#define (VK_API_VERSION.*VK_MAKE_API_VERSION|VK_HEADER_VERSION)")
 
 foreach(VulkanHeaders_line ${VulkanHeaders_lines})
 
     # First, handle the case where we have a major/minor version
     #   Format is:
-    #        #define VK_API_VERSION_X_Y VK_MAKE_VERSION(X, Y, 0)
+    #        #define VK_API_VERSION_X_Y VK_MAKE_API_VERSION(0, X, Y, 0)
     #   We grab the major version (X) and minor version (Y) out of the parentheses
-    string(REGEX MATCH "VK_MAKE_VERSION\\(.*\\)" VulkanHeaders_out ${VulkanHeaders_line})
+    string(REGEX MATCH "VK_MAKE_API_VERSION\\(.*\\)" VulkanHeaders_out ${VulkanHeaders_line})
     string(REGEX MATCHALL "[0-9]+" VulkanHeaders_MAJOR_MINOR "${VulkanHeaders_out}")
     if (VulkanHeaders_MAJOR_MINOR)
-        list (GET VulkanHeaders_MAJOR_MINOR 0 VulkanHeaders_cur_major)
-        list (GET VulkanHeaders_MAJOR_MINOR 1 VulkanHeaders_cur_minor)
+        list (GET VulkanHeaders_MAJOR_MINOR 1 VulkanHeaders_cur_major)
+        list (GET VulkanHeaders_MAJOR_MINOR 2 VulkanHeaders_cur_minor)
         if (${VulkanHeaders_cur_major} GREATER ${VulkanHeaders_VERSION_MAJOR})
             set(VulkanHeaders_VERSION_MAJOR ${VulkanHeaders_cur_major})
             set(VulkanHeaders_VERSION_MINOR ${VulkanHeaders_cur_minor})
