diff --git a/client/scripts/linux b/client/scripts/linux
index a174cb2..40d4e65 100755
--- a/client/scripts/linux
+++ b/client/scripts/linux
@@ -106,8 +106,8 @@ set_hostname() {
 # Must be used on exit. Invokes the local dhcp client exit hooks, if any.
 exit_with_hooks() {
   exit_status=$1
-  if [ -f /etc/dhcp/dhclient-exit-hooks ]; then
-    . /etc/dhcp/dhclient-exit-hooks
+  if [ -f /etc/dhclient-exit-hooks ]; then
+    . /etc/dhclient-exit-hooks
   fi
   exit $exit_status
 }
@@ -124,9 +124,9 @@ fi
 # The action starts here
 
 # Invoke the local dhcp client enter hooks, if they exist.
-if [ -f /etc/dhcp/dhclient-enter-hooks ]; then
+if [ -f /etc/dhclient-enter-hooks ]; then
   exit_status=0
-  . /etc/dhcp/dhclient-enter-hooks
+  . /etc/dhclient-enter-hooks
   if [ $exit_status -ne 0 ]; then
     exit $exit_status
   fi
