diff -up xf86-input-vmmouse-12.6.5/src/vmmouse.c.jx xf86-input-vmmouse-12.6.5/src/vmmouse.c
--- xf86-input-vmmouse-12.6.5/src/vmmouse.c.jx	2009-08-06 13:23:09.000000000 -0400
+++ xf86-input-vmmouse-12.6.5/src/vmmouse.c	2009-08-27 14:26:54.000000000 -0400
@@ -1175,8 +1175,6 @@ VMMousePlug(pointer	module,
 {
    static Bool Initialised = FALSE;
 
-   xf86LoaderReqSymLists(reqSymbols, NULL);
-
    if (!Initialised)
       Initialised = TRUE;
 
