diff --git i/deluge/core/core.py w/deluge/core/core.py
index 8498ff9a8..fbef1cbd7 100644
--- i/deluge/core/core.py
+++ w/deluge/core/core.py
@@ -120,7 +120,7 @@ def __init__(
         component.Component.__init__(self, 'Core')
 
         # Start the libtorrent session.
-        user_agent = 'Deluge/{} libtorrent/{}'.format(DELUGE_VER, LT_VERSION)
+        user_agent = 'Deluge/{} libtorrent/{}'.format('2.0.3', LT_VERSION)
         peer_id = self._create_peer_id(DELUGE_VER)
         log.debug('Starting session (peer_id: %s, user_agent: %s)', peer_id, user_agent)
         settings_pack = {
