--- xfig.3.2.5/Imakefile.Xaw3d	2007-04-16 14:56:58.000000000 +0200
+++ xfig.3.2.5/Imakefile	2007-04-16 14:57:26.000000000 +0200
@@ -50,7 +50,7 @@
 XCOMM Uncomment the following definition for XAW3D if you want to use
 XCOMM the 3d Athena Widget Set (highly recommended!)
 
-#define XAW3D
+XCOMM #define XAW3D
 
 XCOMM Uncomment the following if you have David Hawkey's Xaw3D version 1.5E which has
 XCOMM some new features, including "Tips", which replace xfig's "help balloons"
