diff --unified --recursive --text a/lib/parseppd.c b/lib/parseppd.c
--- a/lib/parseppd.c	2021-09-04 07:07:55.604714271 +0000
+++ b/lib/parseppd.c	2021-09-04 07:13:38.701566084 +0000
@@ -1707,7 +1707,7 @@
 void
 yyerror (const char *msg)
 {
-  error_at_line (1, 0, ppdfilename, ppdlineno, msg);
+  error_at_line (1, 0, ppdfilename, ppdlineno, "%s", msg);
 }
 
 /*
diff --unified --recursive --text a/src/parsessh.c b/src/parsessh.c
--- a/src/parsessh.c	2021-09-04 07:07:55.591380931 +0000
+++ b/src/parsessh.c	2021-09-04 07:08:33.754734927 +0000
@@ -2639,7 +2639,7 @@
 void
 yyerror (const char *msg)
 {
-  error_at_line (1, 0, sshfilename, sshlineno, msg);
+  error_at_line (1, 0, sshfilename, sshlineno, "%s", msg);
 }
 
 /*
