--- autoconf/emacsbrand.orig	2015-12-21 05:31:11.000000000 -0700
+++ autoconf/emacsbrand	2017-02-11 21:08:11.214135100 -0700
@@ -35,6 +35,7 @@ done
 #*    The autodetection ELisp expression                               */
 #*---------------------------------------------------------------------*/
 expr="(cond \
+	((string-match \"GNU Emacs 25\" (emacs-version)) (kill-emacs 25)) \
         ((string-match \"GNU Emacs 24\" (emacs-version)) (kill-emacs 24)) \
         ((string-match \"GNU Emacs 23\" (emacs-version)) (kill-emacs 23)) \
         ((string-match \"GNU Emacs 22\" (emacs-version)) (kill-emacs 22)) \
@@ -63,7 +64,11 @@ else
           if [ "$version" = "24" ]; then
             echo "emacs24"
           else
-            echo "generic";
+            if [ "$version" = "25" ]; then
+              echo "emacs25"
+            else
+              echo "generic";
+            fi
           fi
         fi
       fi
--- bmacs/Makefile.orig	2015-12-21 05:31:11.000000000 -0700
+++ bmacs/Makefile	2017-02-11 20:51:48.555481223 -0700
@@ -115,7 +115,7 @@ POPULATION	= Makefile README $(EMACS_SOU
 #*---------------------------------------------------------------------*/
 all: $(EMACSBRAND)
 
-emacs22 emacs23 emacs24:
+emacs22 emacs23 emacs24 emacs25:
 	(expr=load-path; \
           for p in . $(LOADPATH); do \
              expr="(cons \"../$$p\" (cons \"$$p\" $$expr))"; \
@@ -179,6 +179,7 @@ install.emacs21: doinstall
 install.emacs22: doinstall
 install.emacs23: doinstall
 install.emacs24: doinstall
+install.emacs25: doinstall
 install.xemacs: doinstall
 
 install.generic:
