diff --git a/inc/libcmis/session.hxx b/inc/libcmis/session.hxx
index 0a768a8..ec95ab4 100644
--- a/inc/libcmis/session.hxx
+++ b/inc/libcmis/session.hxx
@@ -95,6 +95,8 @@ namespace libcmis
                 certificate exception feature available on common web browser.
               */
             virtual void setNoSSLCertificateCheck( bool noCheck ) = 0;
+
+            virtual std::string getRefreshToken() { return ""; };
     };
 }
 
