diff -up lyx-2.1.3/autogen.sh.automake15 lyx-2.1.3/autogen.sh
--- lyx-2.1.3/autogen.sh.automake15	2015-05-18 12:54:19.896333451 -0500
+++ lyx-2.1.3/autogen.sh	2015-05-18 12:45:52.637730166 -0500
@@ -16,12 +16,12 @@ test "$automake_version" != "" && {
 }
 
 case $automake_version in
-    *' '1.[8-9]*|*' '1.1[01234]*)
+    *' '1.[8-9]*|*' '1.1[012345]*)
 	;;
     *)
 
 	echo "This automake version is not supported by LyX."
-	echo "LyX only supports automake 1.8 to 1.14."
+	echo "LyX only supports automake 1.8 to 1.15."
 	exit 1
 	;;
 esac
diff -up lyx-2.1.3/configure.ac.automake15 lyx-2.1.3/configure.ac
--- lyx-2.1.3/configure.ac.automake15	2015-02-07 09:25:07.000000000 -0600
+++ lyx-2.1.3/configure.ac	2015-05-18 12:53:48.408046691 -0500
@@ -27,7 +27,7 @@ fi
 AM_MAINTAINER_MODE
 
 save_PACKAGE=$PACKAGE
-AM_INIT_AUTOMAKE([foreign dist-bzip2 no-define 1.8])
+AM_INIT_AUTOMAKE([foreign dist-bzip2 no-define 1.8 subdir-objects])
 m4_ifdef([AM_SILENT_RULES],[AM_SILENT_RULES([yes])])
 PACKAGE=$save_PACKAGE
 
