--- src/IDocumentLink.h.orig	2008-04-12 15:24:01.000000000 +0000
+++ src/IDocumentLink.h	2008-04-12 15:25:32.000000000 +0000
@@ -33,7 +33,7 @@
     class IDocumentLink
     {
         public:
-            IDocumentLink (gdouble x1, gdouble y1, gdouble x2, gdouble y1);
+            IDocumentLink (gdouble x1, gdouble y1, gdouble x2, gdouble y2);
             virtual ~IDocumentLink (void);
 
             virtual void activate (IDocument *document) = 0;
