diff -up xfig.3.2.5/Imakefile.no-Xaw3d xfig.3.2.5/Imakefile
--- xfig.3.2.5/Imakefile.no-Xaw3d	2007-11-16 10:38:18.000000000 +0100
+++ xfig.3.2.5/Imakefile	2007-11-16 10:38:49.000000000 +0100
@@ -50,16 +50,16 @@ XAPPLOADDIR = /usr/share/X11/app-default
 XCOMM Uncomment the following definition for XAW3D if you want to use
 XCOMM the 3d Athena Widget Set (highly recommended!)
 
-XCOMM #define XAW3D
+#define XAW3D
 
 XCOMM Uncomment the following if you have David Hawkey's Xaw3D version 1.5E which has
 XCOMM some new features, including "Tips", which replace xfig's "help balloons"
 
-XCOMM #define XAW3D1_5E
+#define XAW3D1_5E
 
 #ifdef XAW3D1_5E
     DUSEXAW3D = -DXAW3D -DXAW3D1_5E
-    XAWLIB = -lXaw3d15e
+    XAWLIB = -lXaw3d
 #else
     XAW_SRC = w_menuentry.c SmeCascade.c SmeBSB.c SimpleMenu.c
     XAW_OBJ = w_menuentry.o SmeCascade.o SmeBSB.o SimpleMenu.o
diff -up xfig.3.2.5/w_library.c.no-Xaw3d xfig.3.2.5/w_library.c
--- xfig.3.2.5/w_library.c.no-Xaw3d	2006-04-03 18:27:30.000000000 +0200
+++ xfig.3.2.5/w_library.c	2007-11-16 10:38:18.000000000 +0100
@@ -1029,6 +1029,7 @@ make_library_menu(Widget parent, char *n
     Widget	     menu, entry;
 #ifndef XAW3D1_5E
     Widget	     submenu;
+#else
     char	     submenu_name[200];
 #endif /* XAW3D1_5E */
     char	     menu_name[200];
