diff -urNad trunk~/pdns/tcpreceiver.cc trunk/pdns/tcpreceiver.cc
--- trunk~/pdns/tcpreceiver.cc	2005-11-01 13:09:52.000000000 +0100
+++ trunk/pdns/tcpreceiver.cc	2006-10-29 11:00:39.402048787 +0100
@@ -321,7 +321,7 @@
   soa.qname=target;
   soa.qtype=QType::SOA;
   soa.content=DNSPacket::serializeSOAData(sd);
-  soa.ttl=sd.default_ttl;
+  soa.ttl=sd.ttl;
   soa.domain_id=sd.domain_id;
   soa.d_place=DNSResourceRecord::ANSWER;
     
