--- org/gudy/azureus2/core3/security/impl/SESecurityManagerImpl.java.orig	2007-08-26 11:52:57.000000000 +0000
+++ org/gudy/azureus2/core3/security/impl/SESecurityManagerImpl.java	2007-08-26 11:53:29.000000000 +0000
@@ -130,7 +130,7 @@
 		installAuthenticator();
 		
 	
-		String[]	providers = { "com.sun.net.ssl.internal.ssl.Provider", "org.metastatic.jessie.provider.Jessie" };
+		String[]	providers = { "com.sun.net.ssl.internal.ssl.Provider", "gnu.javax.net.ssl.provider.Jessie" };
 			
 		String	provider = null;
 		
