diff -Nur dhcp-3.0.6.orig/client/scripts/linux dhcp-3.0.6/client/scripts/linux
--- dhcp-3.0.6.orig/client/scripts/linux	2008-03-29 18:59:39.000000000 +0100
+++ dhcp-3.0.6/client/scripts/linux	2008-03-29 19:00:16.000000000 +0100
@@ -140,7 +140,7 @@
   if [ x$old_ip_address != x ] && [ x$old_ip_address != x$new_ip_address ]; then
     # IP address changed. Bringing down the interface will delete all routes,
     # and clear the ARP cache.
-    ifconfig $interface inet 0 down
+    ifconfig $interface inet 0
   fi
  fi
 
@@ -185,7 +185,7 @@
   fi
   if [ x$old_ip_address != x ]; then
     # Shut down interface, which will delete routes and clear arp cache.
-    ifconfig $interface inet 0 down
+    ifconfig $interface inet 0
   fi
   if [ x$alias_ip_address != x ]; then
     ifconfig $interface:0 inet $alias_ip_address $alias_subnet_arg
@@ -228,7 +228,7 @@
     exit_with_hooks 0
   fi
  if [ -z "${dhc_dbus}" ] || (( ( dhc_dbus & 2 ) != 2 )); then
-  ifconfig $interface inet 0 down
+  ifconfig $interface inet 0
  fi
   exit_with_hooks 1
 fi
