--- nxnode~	2010-04-30 15:42:46.000000000 +0200
+++ nxnode	2010-04-30 15:42:46.000000000 +0200
@@ -328,6 +328,9 @@
 		NODE_WM_PID=$!
 	fi
 
+	# Patch - delay to allow nxagent to complete launching
+	sleep 4
+
 	echo "Xft.dpi: 96" | DISPLAY=:$display xrdb -merge >>"$USER_FAKE_HOME/.nx/C-$sess_id/session" 2>&1
 	[ -d /etc/X11/Xresources ] && xrdb -display :$display -merge /etc/X11/Xresources/* >>"$USER_FAKE_HOME/.nx/C-$sess_id/session" 2>&1
 
