diff -uprw avidemux_2.5.6.orig/avidemux/ADM_core/include/ADM_mangle.h avidemux_2.5.6/avidemux/ADM_core/include/ADM_mangle.h
--- avidemux_2.5.6.orig/avidemux/ADM_core/include/ADM_mangle.h	2011-12-29 02:32:37.000000000 +0200
+++ avidemux_2.5.6/avidemux/ADM_core/include/ADM_mangle.h	2015-06-19 07:35:59.554911423 +0300
@@ -35,7 +35,7 @@
 #        define FUNNY_MANGLE_ARRAY(x, y) __attribute__((used)) x[y] asm(#x)
 #    else
 #        define MANGLE(a) #a
-#        define FUNNY_MANGLE(x) x asm(MANGLE(x))
+#        define FUNNY_MANGLE(x) __attribute__((used)) x asm(MANGLE(x))
 #        define FUNNY_MANGLE_ARRAY(x, y) x[y] asm(MANGLE(x))
 #    endif
 #endif
