From c257b0d7b4dd1c3f9b63c6a27b68a27bc246acea Mon Sep 17 00:00:00 2001
From: Dave Reisner <dreisner@archlinux.org>
Date: Tue, 12 Jun 2012 19:18:18 -0400
Subject: [PATCH 2/2] shutdown: don't alert udev on disable VGs

Fixes FS#30271.

Signed-off-by: Dave Reisner <dreisner@archlinux.org>
---
 shutdown |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/shutdown b/shutdown
index ce7de23..fd88105 100644
--- a/shutdown
+++ b/shutdown
@@ -25,7 +25,7 @@ stop_device() {
             read devname <"$1/dm/name"
             lvm lvdisplay -c "/dev/mapper/$devname" | {
                 IFS=: read _ vgname _
-                lvm vgchange -an "$vgname"
+                lvm vgchange --noudevsync -an "$vgname"
             }
             ;;
         raid*)
-- 
1.7.10.4

