diff --git i/src/CMakeLists.txt w/src/CMakeLists.txt
index 4a02dbe..4d731c9 100644
--- i/src/CMakeLists.txt
+++ w/src/CMakeLists.txt
@@ -188,7 +188,7 @@ target_include_directories(checkShared
 )
 
 if(NOT THIS_IS_SUBPROJECT)
-  install(TARGETS check checkShared
+  install(TARGETS checkShared
     EXPORT check-targets
     ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
     LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
