Index: kioslave/imap4/imap4.cpp
===================================================================
--- kioslave/imap4/imap4.cpp	(revision 1216162)
+++ kioslave/imap4/imap4.cpp	(revision 1216163)
@@ -2576,7 +2576,7 @@
     ssize_t copyLen = (len < readBufferLen) ? len : readBufferLen;
     memcpy(data, readBuffer, copyLen);
     readBufferLen -= copyLen;
-    if (readBufferLen) memcpy(readBuffer, &readBuffer[copyLen], readBufferLen);
+    if (readBufferLen) memmove(readBuffer, &readBuffer[copyLen], readBufferLen);
     return copyLen;
   }
   if (!isConnected()) return 0;
