From 04e4b3aebf592abf24c07c5e56bcbb18ba04c30b Mon Sep 17 00:00:00 2001
Message-Id: <04e4b3aebf592abf24c07c5e56bcbb18ba04c30b.1534424269.git.jan.steffens@gmail.com>
From: "Jan Alexander Steffens (heftig)" <jan.steffens@gmail.com>
Date: Thu, 16 Aug 2018 13:17:58 +0200
Subject: [PATCH] =?UTF-8?q?x264enc:=20Set=20bit=20depth=20for=20x264=20?=
 =?UTF-8?q?=E2=89=A5=20153?=
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

---
 ext/x264/gstx264enc.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/ext/x264/gstx264enc.c b/ext/x264/gstx264enc.c
index f88cfcb7..9e7986ab 100644
--- a/ext/x264/gstx264enc.c
+++ b/ext/x264/gstx264enc.c
@@ -1578,6 +1578,9 @@ gst_x264_enc_init_encoder (GstX264Enc * encoder)
   }
 
   /* set up encoder parameters */
+#if X264_BUILD >= 153
+  encoder->x264param.i_bitdepth = GST_VIDEO_INFO_COMP_DEPTH (info, 0);
+#endif
   encoder->x264param.i_csp =
       gst_x264_enc_gst_to_x264_video_format (info->finfo->format, NULL);
   if (info->fps_d == 0 || info->fps_n == 0) {
-- 
2.18.0

