diff -up OpenCV-2.2.0/modules/highgui/src/cap_v4l.cpp.nov4l1 OpenCV-2.2.0/modules/highgui/src/cap_v4l.cpp
--- OpenCV-2.2.0/modules/highgui/src/cap_v4l.cpp.nov4l1	2010-12-05 11:35:25.000000000 +0800
+++ OpenCV-2.2.0/modules/highgui/src/cap_v4l.cpp	2011-04-18 10:40:01.984950054 +0800
@@ -214,7 +214,7 @@
 #include <sys/types.h>
 #include <sys/mman.h>
 
-#include <linux/videodev.h>
+#include <libv4l1-videodev.h>
 
 #include <string.h>
 #include <stdlib.h>
diff -up OpenCV-2.2.0/modules/highgui/src/cap_v4l.cpp.nov4l1 OpenCV-2.2.0/modules/highgui/src/cap_libv4l.cpp.cpp
--- OpenCV-2.2.0/modules/highgui/src/cap_libv4l.cpp.nov4l1	2010-12-05 11:35:25.000000000 +0800
+++ OpenCV-2.2.0/modules/highgui/src/cap_libv4l.cpp	2011-04-18 10:39:07.568578185 +0800
@@ -224,7 +224,7 @@
 #include "highgui.h"
 #include "precomp.hpp"
 
-#if !defined WIN32 && defined HAVE_CAMV4L && defined HAVE_CAMV4L2
+#if !defined WIN32 && defined HAVE_CAMV4L2
 
 #define CLEAR(x) memset (&(x), 0, sizeof (x))
 
@@ -241,7 +241,7 @@
 #include <sys/stat.h>
 #include <sys/ioctl.h>
 
-#include <linux/videodev.h>
+#include <libv4l1-videodev.h>
 #include <linux/videodev2.h>
 
 #include <libv4l1.h>
