diff --git a/src/lib/marble/MergedLayerDecorator.cpp b/src/lib/marble/MergedLayerDecorator.cpp
index d2e454e3d..eb3d22e42 100644
--- a/src/lib/marble/MergedLayerDecorator.cpp
+++ b/src/lib/marble/MergedLayerDecorator.cpp
@@ -36,6 +36,7 @@
 
 #include <QPointer>
 #include <QPainter>
+#include <QPainterPath>
 
 using namespace Marble;
 
diff --git a/src/lib/marble/VisiblePlacemark.cpp b/src/lib/marble/VisiblePlacemark.cpp
index cfe08af97..1a84006b7 100644
--- a/src/lib/marble/VisiblePlacemark.cpp
+++ b/src/lib/marble/VisiblePlacemark.cpp
@@ -22,6 +22,7 @@
 
 #include <QApplication>
 #include <QPainter>
+#include <QPainterPath>
 #include <QPalette>
 #include <QPixmapCache>
 
diff --git a/src/lib/marble/graphicsview/FrameGraphicsItem.cpp b/src/lib/marble/graphicsview/FrameGraphicsItem.cpp
index 8a90526fb..065f2452c 100644
--- a/src/lib/marble/graphicsview/FrameGraphicsItem.cpp
+++ b/src/lib/marble/graphicsview/FrameGraphicsItem.cpp
@@ -18,6 +18,7 @@
 // Qt
 #include <QSizeF>
 #include <QPainter>
+#include <QPainterPath>
 #include <QPixmapCache>
 #include <QMargins>
 #include <qdrawutil.h>
diff --git a/src/lib/marble/layers/GroundLayer.h b/src/lib/marble/layers/GroundLayer.h
index b18b160cf..ad8d717b8 100644
--- a/src/lib/marble/layers/GroundLayer.h
+++ b/src/lib/marble/layers/GroundLayer.h
@@ -14,6 +14,7 @@
 #include "LayerInterface.h"
 
 #include <QColor>
+#include <QPainterPath>
 
 namespace Marble
 {
diff --git a/src/plugins/render/compass/CompassFloatItem.cpp b/src/plugins/render/compass/CompassFloatItem.cpp
index 09e4c7655..9f222dcd9 100644
--- a/src/plugins/render/compass/CompassFloatItem.cpp
+++ b/src/plugins/render/compass/CompassFloatItem.cpp
@@ -19,6 +19,7 @@
 #include <QRect>
 #include <QColor>
 #include <QPainter>
+#include <QPainterPath>
 #include <QPushButton>
 #include <QSvgRenderer>
 
diff --git a/src/plugins/render/elevationprofilefloatitem/ElevationProfileFloatItem.cpp b/src/plugins/render/elevationprofilefloatitem/ElevationProfileFloatItem.cpp
index 154264e67..c63235958 100644
--- a/src/plugins/render/elevationprofilefloatitem/ElevationProfileFloatItem.cpp
+++ b/src/plugins/render/elevationprofilefloatitem/ElevationProfileFloatItem.cpp
@@ -31,6 +31,7 @@
 #include <QContextMenuEvent>
 #include <QRect>
 #include <QPainter>
+#include <QPainterPath>
 #include <QPushButton>
 #include <QMenu>
 #include <QMouseEvent>
diff --git a/src/plugins/render/foursquare/FoursquareItem.cpp b/src/plugins/render/foursquare/FoursquareItem.cpp
index d9e1f57d8..0bf68face 100644
--- a/src/plugins/render/foursquare/FoursquareItem.cpp
+++ b/src/plugins/render/foursquare/FoursquareItem.cpp
@@ -13,6 +13,7 @@
 #include "ViewportParams.h"
 
 #include <QFontMetrics>
+#include <QPainterPath>
  
 namespace Marble
 {
diff --git a/src/plugins/render/license/License.cpp b/src/plugins/render/license/License.cpp
index 6faaa3e08..45682c3b8 100644
--- a/src/plugins/render/license/License.cpp
+++ b/src/plugins/render/license/License.cpp
@@ -23,6 +23,7 @@
 #include <QCommonStyle>
 #include <QContextMenuEvent>
 #include <QPainter>
+#include <QPainterPath>
 #include <QLabel>
 #include <QMenu>
 #include <QMouseEvent>
diff --git a/src/plugins/render/notes/NotesItem.cpp b/src/plugins/render/notes/NotesItem.cpp
index 9d0961fcd..0c9fe992b 100644
--- a/src/plugins/render/notes/NotesItem.cpp
+++ b/src/plugins/render/notes/NotesItem.cpp
@@ -9,6 +9,7 @@
 #include "MarbleDirs.h"
 
 #include <QPainter>
+#include <QPainterPath>
 #include <QRect>
 
 using namespace Marble;
diff --git a/src/plugins/render/postalcode/PostalCodeItem.cpp b/src/plugins/render/postalcode/PostalCodeItem.cpp
index 772c0b269..0ff7957d2 100644
--- a/src/plugins/render/postalcode/PostalCodeItem.cpp
+++ b/src/plugins/render/postalcode/PostalCodeItem.cpp
@@ -17,6 +17,7 @@
 // Qt
 #include <QFontMetrics>
 #include <QPainter>
+#include <QPainterPath>
 
 using namespace Marble;
 
diff --git a/src/plugins/render/progress/ProgressFloatItem.cpp b/src/plugins/render/progress/ProgressFloatItem.cpp
index a4658577c..6e2452c5f 100644
--- a/src/plugins/render/progress/ProgressFloatItem.cpp
+++ b/src/plugins/render/progress/ProgressFloatItem.cpp
@@ -22,6 +22,7 @@
 #include <QColor>
 #include <QPaintDevice>
 #include <QPainter>
+#include <QPainterPath>
 
 namespace Marble
 {
diff --git a/src/plugins/render/stars/StarsPlugin.cpp b/src/plugins/render/stars/StarsPlugin.cpp
index ba1d2e349..5c83b8088 100644
--- a/src/plugins/render/stars/StarsPlugin.cpp
+++ b/src/plugins/render/stars/StarsPlugin.cpp
@@ -19,6 +19,7 @@
 #include <QContextMenuEvent>
 #include <QMenu>
 #include <QColorDialog>
+#include <QPainterPath>
 #include <qmath.h>
 
 #include "MarbleClock.h"
