diff -Naur coreutils-7.3-old/configure.ac coreutils-7.3/configure.ac
--- coreutils-7.3-old/configure.ac	2009-04-24 22:50:28.000000000 +1000
+++ coreutils-7.3/configure.ac	2009-05-03 01:30:17.000000000 +1000
@@ -32,8 +32,7 @@
 AC_CONFIG_AUX_DIR([build-aux])
 AC_CONFIG_HEADERS([lib/config.h:lib/config.hin])
 
-AM_INIT_AUTOMAKE([1.10b dist-xz color-tests parallel-tests])
-AM_SILENT_RULES([yes]) # make --enable-silent-rules the default.
+AM_INIT_AUTOMAKE([1.10.2 dist-lzma])
 
 AC_PROG_CC_STDC
 AM_PROG_CC_C_O
