From 961e4526925b7b1e1d3582f2fc9fb38035e2b5fb Mon Sep 17 00:00:00 2001
From: Lennart Poettering <lennart@poettering.net>
Date: Fri, 15 Mar 2013 16:41:13 +0100
Subject: [PATCH] strv: fix STRV_FOREACH_PAIR macro definition

---
 src/shared/strv.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/shared/strv.h b/src/shared/strv.h
index da9fae6..49058f8 100644
--- a/src/shared/strv.h
+++ b/src/shared/strv.h
@@ -77,7 +77,7 @@ bool strv_overlap(char **a, char **b);
         for (; (l) && ((s) >= (l)); (s)--)
 
 #define STRV_FOREACH_PAIR(x, y, l)               \
-        for ((x) = (l), (y) = (x+1); (x) && *(x) && *(y); (x) += 2)
+        for ((x) = (l), (y) = (x+1); (x) && *(x) && *(y); (x) += 2, (y) = (x + 1))
 
 
 char **strv_sort(char **l);
-- 
1.8.2

