diff -Naur old/ChangeLog new/ChangeLog
--- old/ChangeLog	2013-03-09 13:19:31.000000000 -1000
+++ new/ChangeLog	2013-11-22 06:33:59.189976814 -1000
@@ -1,3 +1,10 @@
+2013-10-16  Glenn Randers-Pehrson  <glennrp@simple.dallas.tx.us>
+
+	* Added calls to png_set_benign_errors() to allow benign errors
+	to be handled as warnings.  In particular, GM builds with libpng-1.6.x
+	will not crash while copying a PNG with a "known incorrect ICC
+	profile".
+
 2013-03-09  Bob Friesenhahn  <bfriesen@simple.dallas.tx.us>
 
 	* version.sh, www/index.rst: Prepare for 1.3.18 release.
diff -Naur old/coders/png.c new/coders/png.c
--- old/coders/png.c	2013-03-09 13:19:31.000000000 -1000
+++ new/coders/png.c	2013-11-22 06:33:25.423156308 -1000
@@ -1689,6 +1689,12 @@
         }
       return(image);
     }
+
+#ifdef PNG_BENIGN_ERRORS_SUPPORTED
+  /* Allow benign errors */
+  png_set_benign_errors(ping, 1);
+#endif
+
   /*
     Prepare PNG for reading.
   */
@@ -6453,6 +6459,12 @@
 #endif
       return(MagickFail);
     }
+
+#ifdef PNG_BENIGN_ERRORS_SUPPORTED
+  /* Allow benign errors */
+  png_set_benign_errors(ping, 1);
+#endif
+
   /*
     Prepare PNG for writing.
   */
