diff -aur ecasound-2.9.0.orig/configure ecasound-2.9.0/configure
--- ecasound-2.9.0.orig/configure	2013-03-26 17:14:51.759045636 +0800
+++ ecasound-2.9.0/configure	2013-03-26 17:15:50.502060731 +0800
@@ -6535,7 +6535,7 @@
   rubyecasound_support=no
   ECA_S_RUBY_SITEDIR=""
 else
-  ECA_S_RUBY_SITEDIR="`ruby -e 'require "rbconfig"; include Config; print CONFIG["sitedir"] + "/" + CONFIG["MAJOR"] + "." + CONFIG["MINOR"]'`"
+  ECA_S_RUBY_SITEDIR="`ruby -e 'include RbConfig; print CONFIG["vendordir"] + "/" + CONFIG["MAJOR"] + "." + CONFIG["MINOR"]'`"
 fi
 
 
