Correct path to checkpassword.

diff -Naur pkg-orig/etc/xinetd.d/bincimap pkg/etc/xinetd.d/bincimap
--- pkg-orig/etc/xinetd.d/bincimap	2006-03-27 15:21:30.000000000 -0500
+++ pkg/etc/xinetd.d/bincimap	2006-03-27 15:44:52.000000000 -0500
@@ -21,6 +21,6 @@
 	group		= root
 	instances	= UNLIMITED
 	server		= /usr/bin/bincimap-up
-	server_args	= --conf=/etc/bincimap/bincimap.conf --logtype=syslog -- /bin/checkpassword /usr/bin/bincimapd
+	server_args	= --conf=/etc/bincimap/bincimap.conf --logtype=syslog -- /usr/bin/checkpassword /usr/bin/bincimapd
 	log_type	= SYSLOG daemon info
 }
diff -Naur pkg-orig/etc/xinetd.d/bincimaps pkg/etc/xinetd.d/bincimaps
--- pkg-orig/etc/xinetd.d/bincimaps	2006-03-27 15:21:30.000000000 -0500
+++ pkg/etc/xinetd.d/bincimaps	2006-03-27 15:45:02.000000000 -0500
@@ -21,6 +21,6 @@
 	group		= root
 	instances	= UNLIMITED
 	server		= /usr/bin/bincimap-up
-	server_args	= --ssl --conf=/etc/bincimap/bincimap.conf --logtype=syslog -- /bin/checkpassword /usr/bin/bincimapd
+	server_args	= --ssl --conf=/etc/bincimap/bincimap.conf --logtype=syslog -- /usr/bin/checkpassword /usr/bin/bincimapd
 	log_type	= SYSLOG daemon info
 }
diff -Naur pkg-orig/var/service/bincimap/run pkg/var/service/bincimap/run
--- pkg-orig/var/service/bincimap/run	2006-03-27 15:21:29.000000000 -0500
+++ pkg/var/service/bincimap/run	2006-03-27 15:45:35.000000000 -0500
@@ -10,7 +10,7 @@
     /usr/bin/bincimap-up			\
     --logtype=multilog				\
     --conf=/etc/bincimap/bincimap.conf --	\
-    /bin/checkpassword				\
+    /usr/bin/checkpassword			\
     /usr/bin/bincimapd
 
 
diff -Naur pkg-orig/var/service/bincimaps/run pkg/var/service/bincimaps/run
--- pkg-orig/var/service/bincimaps/run	2006-03-27 15:21:29.000000000 -0500
+++ pkg/var/service/bincimaps/run	2006-03-27 15:45:49.000000000 -0500
@@ -10,5 +10,5 @@
     /usr/bin/bincimap-up				\
     --logtype=multilog					\
     --conf=/etc/bincimap/bincimap.conf --ssl --     	\
-    /bin/checkpassword					\
+    /usr/bin/checkpassword				\
     /usr/bin/bincimapd
