diff -Naur pkg.orig/etc/xinetd.d/bincimap pkg/etc/xinetd.d/bincimap
--- pkg.orig/etc/xinetd.d/bincimap	2004-02-09 13:50:44.000000000 +0200
+++ pkg/etc/xinetd.d/bincimap	2004-02-09 15:35:04.000000000 +0200
@@ -13,7 +13,7 @@
 # description: Binc IMAP
 service imap
 {
-	disable		= no
+	disable		= yes
 	flags		= REUSE
 	socket_type	= stream        
 	wait		= no
diff -Naur pkg.orig/etc/xinetd.d/bincimaps pkg/etc/xinetd.d/bincimaps
--- pkg.orig/etc/xinetd.d/bincimaps	2004-02-09 13:50:44.000000000 +0200
+++ pkg/etc/xinetd.d/bincimaps	2004-02-09 15:35:14.000000000 +0200
@@ -13,7 +13,7 @@
 # description: Binc IMAP, SSL
 service imaps
 {
-	disable		= no
+	disable		= yes
 	flags		= REUSE
 	socket_type	= stream        
 	wait		= no
