diff -ub src/emacs-23.1/configure src/emacs-23.1.fixed/configure
--- src/emacs-23.1/configure	2009-07-29 17:12:00.000000000 +0200
+++ src/emacs-23.1.fixed/configure	2010-04-29 23:13:41.493641756 +0200
@@ -25742,7 +25742,7 @@
   sed -e '1,/start of cpp stuff/d'\
       -e 's,/\*\*/#\(.*\)$,/* \1 */,' \
       < Makefile.c > junk.c
-  $CPP $cpp_undefs -I. -I$srcdir/src $CPPFLAGS junk.c | \
+  $CPP -P $cpp_undefs -I. -I$srcdir/src $CPPFLAGS junk.c | \
       sed -e 's/^ /	/' -e '/^#/d' -e '/^[ 	]*$/d' > junk2.c
   cat junk1.c junk2.c > Makefile.new
   rm -f junk.c junk1.c junk2.c
@@ -25758,7 +25758,7 @@
   sed -e '1,/start of cpp stuff/d'\
       -e 's,/\*\*/#\(.*\)$,/* \1 */,' \
       < Makefile.c > junk.c
-  $CPP $cpp_undefs -I. -I$srcdir/src $CPPFLAGS junk.c | \
+  $CPP -P $cpp_undefs -I. -I$srcdir/src $CPPFLAGS junk.c | \
       sed -e 's/^ /	/' -e '/^#/d' -e '/^[ 	]*$/d' > junk2.c
   cat junk1.c junk2.c > Makefile.new
   rm -f junk.c junk1.c junk2.c
Common subdirectories: src/emacs-23.1/doc and src/emacs-23.1.fixed/doc
Common subdirectories: src/emacs-23.1/etc and src/emacs-23.1.fixed/etc
Common subdirectories: src/emacs-23.1/info and src/emacs-23.1.fixed/info
Common subdirectories: src/emacs-23.1/leim and src/emacs-23.1.fixed/leim
Common subdirectories: src/emacs-23.1/lib-src and src/emacs-23.1.fixed/lib-src
Common subdirectories: src/emacs-23.1/lisp and src/emacs-23.1.fixed/lisp
Common subdirectories: src/emacs-23.1/lwlib and src/emacs-23.1.fixed/lwlib
Common subdirectories: src/emacs-23.1/m4 and src/emacs-23.1.fixed/m4
Common subdirectories: src/emacs-23.1/msdos and src/emacs-23.1.fixed/msdos
Common subdirectories: src/emacs-23.1/nextstep and src/emacs-23.1.fixed/nextstep
Common subdirectories: src/emacs-23.1/nt and src/emacs-23.1.fixed/nt
Common subdirectories: src/emacs-23.1/oldXMenu and src/emacs-23.1.fixed/oldXMenu
Common subdirectories: src/emacs-23.1/site-lisp and src/emacs-23.1.fixed/site-lisp
Common subdirectories: src/emacs-23.1/src and src/emacs-23.1.fixed/src
