diff -Naur pan-0.133-orig/pan/general/log.h pan-0.133/pan/general/log.h
--- pan-0.133-orig/pan/general/log.h	2007-08-01 11:00:01.000000000 -0600
+++ pan-0.133/pan/general/log.h	2009-05-15 16:03:53.699935401 -0600
@@ -20,6 +20,7 @@
 #ifndef __Log_h__
 #define __Log_h__
 
+#include <cstdio>
 #include <ctime>
 #include <set>
 #include <string>
diff -Naur pan-0.133-orig/pan/general/progress.cc pan-0.133/pan/general/progress.cc
--- pan-0.133-orig/pan/general/progress.cc	2007-08-01 11:00:01.000000000 -0600
+++ pan-0.133/pan/general/progress.cc	2009-05-15 16:00:06.843924198 -0600
@@ -19,6 +19,7 @@
 
 #include <config.h>
 #include <cstdarg>
+#include <cstdio>
 #include "progress.h"
 #include "string-view.h"
 
diff -Naur pan-0.133-orig/pan/usenet-utils/numbers.cc pan-0.133/pan/usenet-utils/numbers.cc
--- pan-0.133-orig/pan/usenet-utils/numbers.cc	2008-07-04 09:01:53.000000000 -0600
+++ pan-0.133/pan/usenet-utils/numbers.cc	2009-05-15 16:04:48.354310199 -0600
@@ -22,6 +22,7 @@
 #include <pan/general/string-view.h>
 #include "numbers.h"
 #include <algorithm>
+#include <cstdio>
 
 using namespace pan;
 
