From 03ba7f6a58dc2ca0ef59a689c06ab1cf18f57ec4 Mon Sep 17 00:00:00 2001
From: Robert Ancell <robert.ancell@canonical.com>
Date: Tue, 22 May 2012 04:26:33 +0000
Subject: Fix data directories pointing to old locations

Modified by Arch Linux for gnome-games 3.6

---
diff --git a/glines/Makefile.am b/glines/Makefile.am
index 794122f..270b0b5 100644
--- a/glines/Makefile.am
+++ b/glines/Makefile.am
@@ -1,4 +1,4 @@
-pixmapdir = $(datadir)/gnome-games/glines/pixmaps
+pixmapdir = $(datadir)/glines/themes
 pixmap_DATA = classic.png dots.png gumball.png pulse.png
 
 EXTRA_DIST = $(pixmap_DATA)
diff --git a/gnobots2/Makefile.am b/gnobots2/Makefile.am
index fc70f09..d4917ba 100644
--- a/gnobots2/Makefile.am
+++ b/gnobots2/Makefile.am
@@ -1,4 +1,4 @@
-themedir = $(datadir)/gnome-games/gnobots2/themes
+themedir = $(datadir)/gnobots2/themes
 theme_DATA = anthills.png cows.png eggs.png mice.png spills.png stormclouds.png toasters.png
 
 EXTRA_DIST = $(theme_DATA)
diff --git a/iagno/Makefile.am b/iagno/Makefile.am
index f311272..8bf2d26 100644
--- a/iagno/Makefile.am
+++ b/iagno/Makefile.am
@@ -1,4 +1,4 @@
-pixmapdir = $(datadir)/gnome-games/iagno/pixmaps
+pixmapdir = $(datadir)/iagno/themes
 pixmap_DATA = blend.png flipblend.png metal.png metal2.png \
 	      shrink.png sunmoon.png 
 
diff --git a/mahjongg/Makefile.am b/mahjongg/Makefile.am
index 5ad3cfa..c712738 100644
--- a/mahjongg/Makefile.am
+++ b/mahjongg/Makefile.am
@@ -1,4 +1,4 @@
-pixmapdir = $(datadir)/gnome-games/mahjongg/pixmaps
+pixmapdir = $(datadir)/gnome-mahjongg/themes
 pixmap_DATA = default.png maya.png default2.png seta.png greek-letters.png
 
 EXTRA_DIST = $(pixmap_DATA)
--
cgit v0.9.0.2
