diff -Nur NVIDIA-Linux-x86_64-304.117-no-compat32.orig/kernel/nv-acpi.c NVIDIA-Linux-x86_64-304.117-no-compat32/kernel/nv-acpi.c
--- NVIDIA-Linux-x86_64-304.117-no-compat32.orig/kernel/nv-acpi.c	2013-11-27 06:25:42.000000000 +0100
+++ NVIDIA-Linux-x86_64-304.117-no-compat32/kernel/nv-acpi.c	2014-01-26 12:25:06.443610357 +0100
@@ -15,6 +15,10 @@
 #include "nv-linux.h"
 #include "nv-reg.h"
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 13, 0) && defined(ACPI_HANDLE)
+#define DEVICE_ACPI_HANDLE(a) ACPI_HANDLE(a)
+#endif
+
 #if defined(NV_LINUX_ACPI_EVENTS_SUPPORTED)
 
 static int         nv_acpi_add             (struct acpi_device *);
@@ -299,7 +303,10 @@
 
     if (pNvAcpiObject->notify_handler_installed)
     {
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3, 13, 0)
+ /* beginning with 3.13, acpi_remove_notify_handler() waits for events to finish */
         NV_ACPI_OS_WAIT_EVENTS_COMPLETE();
+#endif
 
         // remove event notifier
         status = acpi_remove_notify_handler(device->handle, ACPI_DEVICE_NOTIFY, nv_acpi_event);
