From 01501939d5f608541c1ea98fe2da84d841f9c9bc Mon Sep 17 00:00:00 2001
From: Tom Gundersen <teg@jklm.no>
Date: Mon, 2 Jun 2014 15:05:05 +0200
Subject: [PATCH] tmpfiles: systemd.conf - fix ownership of network directories

---
 tmpfiles.d/systemd.conf | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/tmpfiles.d/systemd.conf b/tmpfiles.d/systemd.conf
index 7efab0e..1fce514 100644
--- a/tmpfiles.d/systemd.conf
+++ b/tmpfiles.d/systemd.conf
@@ -21,9 +21,9 @@ d /run/systemd/sessions 0755 root root -
 d /run/systemd/users 0755 root root -
 d /run/systemd/machines 0755 root root -
 d /run/systemd/shutdown 0755 root root -
-d /run/systemd/network 0755 root root -
-d /run/systemd/network/links 0755 root root -
-d /run/systemd/network/leases 0755 root root -
+d /run/systemd/network 0755 systemd-network systemd-network -
+d /run/systemd/network/links 0755 systemd-network systemd-network -
+d /run/systemd/network/leases 0755 systemd-network systemd-network -
 
 m /var/log/journal 2755 root systemd-journal - -
 Z /var/log/journal/%m 2755 root systemd-journal - -
-- 
2.0.0

