commit 5407327bc6fff77dabb6dd55150832373ba163a7
Author: David Teigland <teigland@redhat.com>
Date:   Tue Sep 19 09:19:24 2017 -0500

    toollib: fix parentheses in assignment + comparison

diff --git a/tools/toollib.c b/tools/toollib.c
index 1799bc91c..bf3e9d643 100644
--- a/tools/toollib.c
+++ b/tools/toollib.c
@@ -4464,7 +4464,7 @@ int process_each_pv(struct cmd_context *cmd,
 	 * from all VGs are processed first, removing them from all_devices.  Then
 	 * any devs remaining in all_devices are processed.
 	 */
-	if ((ret = _get_all_devices(cmd, &all_devices) != ECMD_PROCESSED)) {
+	if ((ret = _get_all_devices(cmd, &all_devices)) != ECMD_PROCESSED) {
 		ret_max = ret;
 		goto_out;
 	}
