diff --git a/src/i830.h b/src/i830.h
index 5b78236..89f19d2 100644
--- a/src/i830.h
+++ b/src/i830.h
@@ -950,6 +950,13 @@ static inline int i830_fb_compression_supported(I830Ptr pI830)
      */
     if (!pI830->tiling || (IS_I965G(pI830) && pI830->accel <= ACCEL_XAA))
 	return FALSE;
+    /* We have not gotten FBC to work consistently on 965GM. Our best
+     * working theory right now is that FBC simply isn't reliable on
+     * that device. See this bug report for more details:
+     * https://bugs.freedesktop.org/show_bug.cgi?id=16257
+     */
+    if (IS_I965GM(pI830))
+	return FALSE;
     return TRUE;
 }
 
