--- a/conf/50-synaptics.conf	2012-05-11 01:01:10.000000000 +0200
+++ b/conf/50-synaptics.conf.new	2013-01-05 14:20:21.959670353 +0100
@@ -11,10 +11,13 @@
         Identifier "touchpad catchall"
         Driver "synaptics"
         MatchIsTouchpad "on"
+        Option "TapButton1" "1"
+        Option "TapButton2" "2"
+        Option "TapButton3" "3"
 # This option is recommend on all Linux systems using evdev, but cannot be
 # enabled by default. See the following link for details:
 # http://who-t.blogspot.com/2010/11/how-to-ignore-configuration-errors.html
-#       MatchDevicePath "/dev/input/event*"
+        MatchDevicePath "/dev/input/event*"
 EndSection
 
 Section "InputClass"
