--- certdata2pem.py	2014-08-24 15:16:24.927192958 +0200
+++ certdata2pem.py.loudness	2014-08-24 15:17:30.193535402 +0200
@@ -104,9 +104,7 @@
         trust[obj['CKA_LABEL']] = True
     elif obj['CKA_TRUST_SERVER_AUTH'] in ('CKT_NETSCAPE_UNTRUSTED',
                                           'CKT_NSS_NOT_TRUSTED'):
-        print '!'*74
-        print "UNTRUSTED BUT NOT BLACKLISTED CERTIFICATE FOUND: %s" % obj['CKA_LABEL']
-        print '!'*74
+        print "Certificate %s untrusted, ignoring." % obj['CKA_LABEL']
     else:
         print "Ignoring certificate %s.  SAUTH=%s, EPROT=%s" % \
               (obj['CKA_LABEL'], obj['CKA_TRUST_SERVER_AUTH'],
