diff -urNp coreutils-6.11-orig/tests/touch/no-create-missing coreutils-6.11/tests/touch/no-create-missing
--- coreutils-6.11-orig/tests/touch/no-create-missing	2008-04-19 23:34:23.000000000 +0200
+++ coreutils-6.11/tests/touch/no-create-missing	2008-04-24 12:34:52.000000000 +0200
@@ -36,7 +36,7 @@ test="$abs_top_builddir/src/test"
 # This test is ineffective unless /dev/stdout also works.
 if "$test" -w /dev/stdout >/dev/null &&
    "$test" ! -w /dev/stdout >&-; then
-  touch -c - >&- 2> /dev/null || fail=1
+  strace touch -c - >&- || fail=1
   touch -cm - >&- 2> /dev/null || fail=1
   touch -ca - >&- 2> /dev/null || fail=1
 fi
diff -urNp coreutils-6.11-orig/tests/mkdir/selinux coreutils-6.11/tests/mkdir/selinux
--- coreutils-6.11-orig/tests/mkdir/selinux	2008-04-19 23:34:23.000000000 +0200
+++ coreutils-6.11/tests/mkdir/selinux	2008-04-22 13:23:50.000000000 +0200
@@ -30,6 +30,7 @@ fi
 . $srcdir/../envvar-check
 . $srcdir/../lang-default
 . $srcdir/../test-lib.sh
+require_selinux_
 
 c=invalid-selinux-context
 msg="failed to set default file creation context to \`$c':"
diff -urNp coreutils-6.11-orig/tests/test-lib.sh coreutils-6.11/tests/test-lib.sh
--- coreutils-6.11-orig/tests/test-lib.sh	2008-04-19 23:34:23.000000000 +0200
+++ coreutils-6.11/tests/test-lib.sh	2008-04-24 14:18:59.000000000 +0200
@@ -97,8 +97,8 @@ skip_if_()
 
 require_selinux_()
 {
-  case `ls -Zd .` in
-    '? .'|'unlabeled .')
+  case `ls --scontext -d . | cut -f1 -d" "` in
+    '?'|'unlabeled')
       skip_test_ "this system (or maybe just" \
         "the current file system) lacks SELinux support"
     ;;
diff -urp coreutils-6.11-orig/gnulib-tests/test-getaddrinfo.c coreutils-6.11/gnulib-tests/test-getaddrinfo.c
--- coreutils-6.11-orig/gnulib-tests/test-getaddrinfo.c
+++ coreutils-6.11/gnulib-tests/test-getaddrinfo.c
@@ -70,6 +70,10 @@ int simple (char *host, char *service)
       if (res == EAI_NODATA)
 	return 0;
 
+      /* Do not fail this test for temporary name resolution errors. */
+      if (res == EAI_AGAIN)
+         return 0;
+
       return 1;
     }
 
