diff -u -r plasma-workspace-5.4.1/startkde/startkde.cmake plasma-workspace-5.4.1-dbus/startkde/startkde.cmake
--- plasma-workspace-5.4.1/startkde/startkde.cmake	2015-09-25 19:12:09.630988076 +0200
+++ plasma-workspace-5.4.1-dbus/startkde/startkde.cmake	2015-09-25 19:15:07.320598595 +0200
@@ -286,6 +286,8 @@
 # D-Bus autolaunch is broken
 if test -z "$DBUS_SESSION_BUS_ADDRESS" ; then
     eval `dbus-launch --sh-syntax --exit-with-session`
+elif which dbus-update-activation-environment >/dev/null 2>/dev/null ; then
+    dbus-update-activation-environment --systemd --all
 fi
 if $qdbus >/dev/null 2>/dev/null; then
     : # ok
diff -u -r plasma-workspace-5.4.1/startkde/startplasmacompositor.cmake plasma-workspace-5.4.1-dbus/startkde/startplasmacompositor.cmake
--- plasma-workspace-5.4.1/startkde/startplasmacompositor.cmake	2015-09-03 12:26:33.000000000 +0200
+++ plasma-workspace-5.4.1-dbus/startkde/startplasmacompositor.cmake	2015-09-25 19:15:17.150577050 +0200
@@ -163,6 +163,8 @@
 # D-Bus autolaunch is broken
 if test -z "$DBUS_SESSION_BUS_ADDRESS" ; then
     eval `dbus-launch --sh-syntax --exit-with-session`
+elif which dbus-update-activation-environment >/dev/null 2>/dev/null ; then
+    dbus-update-activation-environment --systemd --all
 fi
 if $qdbus >/dev/null 2>/dev/null; then
     : # ok
