--- hw/xfree86/common/xf86Configure.c.orig	2006-03-04 10:19:27.000000000 +0000
+++ hw/xfree86/common/xf86Configure.c	2006-03-04 10:19:32.000000000 +0000
@@ -91,7 +91,7 @@
 static char *DFLT_MOUSE_DEV = "/dev/sysmouse";
 static char *DFLT_MOUSE_PROTO = "auto";
 #else
-static char *DFLT_MOUSE_DEV = "/dev/mouse";
+static char *DFLT_MOUSE_DEV = "/dev/input/mice";
 static char *DFLT_MOUSE_PROTO = "auto";
 #endif
 
