From 50dc5fecabc2c3434d7a652b4f5642556b4e87bf Mon Sep 17 00:00:00 2001
From: Dan Williams <dcbw@redhat.com>
Date: Mon, 25 Apr 2016 15:39:50 -0500
Subject: dhcp/systemd: actually hook up information-only mode

Fixes: 89d1e466157839096b446068a780cb2563424a5a
Tested-by: Celti on IRC
(cherry picked from commit 8c8a577a30acc0b5f1d4765493f6547f275477b0)

diff --git a/src/dhcp-manager/nm-dhcp-systemd.c b/src/dhcp-manager/nm-dhcp-systemd.c
index 5d49c27..1674341 100644
--- a/src/dhcp-manager/nm-dhcp-systemd.c
+++ b/src/dhcp-manager/nm-dhcp-systemd.c
@@ -857,6 +857,7 @@ dhcp6_event_cb (sd_dhcp6_client *client, int event, gpointer user_data)
 		nm_dhcp_client_set_state (NM_DHCP_CLIENT (user_data), NM_DHCP_STATE_FAIL, NULL, NULL);
 		break;
 	case SD_DHCP6_CLIENT_EVENT_IP_ACQUIRE:
+	case SD_DHCP6_CLIENT_EVENT_INFORMATION_REQUEST:
 		bound6_handle (self);
 		break;
 	default:
@@ -893,6 +894,9 @@ ip6_start (NMDhcpClient *client,
 		return FALSE;
 	}
 
+	if (info_only)
+	    sd_dhcp6_client_set_information_request (priv->client6, 1);
+
 	/* NM stores the entire DUID which includes the uint16 "type", while systemd
 	 * wants the type passed separately from the following data.
 	 */
-- 
cgit v0.10.2

