diff -Nur NetworkManager-0.8.1.orig/src/nm-policy.c NetworkManager-0.8.1/src/nm-policy.c
--- NetworkManager-0.8.1.orig/src/nm-policy.c	2010-07-21 18:48:36.000000000 -0500
+++ NetworkManager-0.8.1/src/nm-policy.c	2010-08-01 19:48:57.474426178 -0500
@@ -286,7 +286,7 @@
 		g_free (configured_hostname);
 		return;
 	}
-
+#if 0
 	/* Try automatically determined hostname from the best device's IP config */
 	if (!best)
 		best = get_best_ip4_device (policy->manager, &best_req);
@@ -318,6 +318,7 @@
 		}
 	}
 
+#endif
 	/* If no automatically-configured hostname, try using the hostname from
 	 * when NM started up.
 	 */
