--- file-5.04/magic/Magdir/archive	2010-09-27 17:52:19.242245630 +0530
+++ file-5.04/magic/Magdir/archive	2010-09-27 17:52:12.015252938 +0530
@@ -581,6 +581,8 @@
 !:mime	application/zip
 >>4	byte		0x14		Zip archive data, at least v2.0 to extract
 !:mime	application/zip
+>>4     byte            0x2d            Zip64 archive data, at least v3.0 to extract
+!:mime  application/zip
 
 # OpenOffice.org / KOffice / StarOffice documents
 # Listed here because they ARE zip files
