diff -Nur gdm-3.3.92.orig/daemon/gdm-server.c gdm-3.3.92/daemon/gdm-server.c
--- gdm-3.3.92.orig/daemon/gdm-server.c	2012-03-19 22:30:05.311490006 +0000
+++ gdm-3.3.92/daemon/gdm-server.c	2012-03-19 22:30:17.004732251 +0000
@@ -139,7 +139,7 @@
         g_return_val_if_fail (GDM_IS_SERVER (server), NULL);
 
         error = NULL;
-        command = g_strdup_printf (LIBEXECDIR "/ck-get-x11-display-device --display %s",
+        command = g_strdup_printf ("/usr/lib/ConsoleKit/ck-get-x11-display-device --display %s",
                                    server->priv->display_name);
 
         g_debug ("GdmServer: Running helper %s", command);
diff -Nur gdm-3.3.92.orig/data/gdm-shell.session.in gdm-3.3.92/data/gdm-shell.session.in
--- gdm-3.3.92.orig/data/gdm-shell.session.in	2012-03-19 22:30:05.311490006 +0000
+++ gdm-3.3.92/data/gdm-shell.session.in	2012-03-19 22:30:17.004732251 +0000
@@ -1,5 +1,5 @@
 [GNOME Session]
 Name=Display Manager
 RequiredComponents=gnome-shell;gnome-settings-daemon;
-IsRunnableHelper=bash -c 'gnome-shell --help | grep -q gdm-mode && @libexecdir@/gnome-session-check-accelerated'
+IsRunnableHelper=bash -c 'gnome-shell --help | grep -q gdm-mode && /usr/lib/gnome-session/gnome-session-check-accelerated'
 FallbackSession=gdm-fallback
diff -Nur gdm-3.3.92.orig/gui/simple-chooser/gdm-chooser-session.c gdm-3.3.92/gui/simple-chooser/gdm-chooser-session.c
--- gdm-3.3.92.orig/gui/simple-chooser/gdm-chooser-session.c	2012-03-19 22:30:05.304823391 +0000
+++ gdm-3.3.92/gui/simple-chooser/gdm-chooser-session.c	2012-03-19 22:30:17.004732251 +0000
@@ -131,7 +131,7 @@
         ret = FALSE;
 
         error = NULL;
-        g_spawn_command_line_async (LIBEXECDIR "/gnome-settings-daemon", &error);
+        g_spawn_command_line_async ("/usr/lib/gnome-settings-daemon/gnome-settings-daemon", &error);
         if (error != NULL) {
                 g_warning ("Error starting settings daemon: %s", error->message);
                 g_error_free (error);
