--- prnt/filters/hpps	2016-08-26 12:04:29.000000000 +0200
+++ prnt/filters/hpps.new	2016-08-31 21:19:57.868640372 +0200
@@ -191,11 +191,11 @@
             msg("Setting Pin to default")
     else:
         szKeyInitials = ['HPFIDigit', 'HPSEDigit', 'HPTHDigit', 'HPFTDigit']                                 
-	for x in szKeyInitials:
+    for x in szKeyInitials:
            try:
-	        secpin += opts[x]
+            secpin += opts[x]
            except KeyError:
-	        secpin += '0'
+            secpin += '0'
 
     os.write(output_fd, to_bytes_utf8('@PJL SET HOLDKEY="%s"\x0a' % secpin))
 
