From 0411c37409102c71e8ba980f9afb2d4a8c64189d Mon Sep 17 00:00:00 2001
From: Alex Deucher <alexdeucher@gmail.com>
Date: Wed, 15 Jul 2009 18:40:41 +0000
Subject: Add R600_SCRATCH_REG_OFFSET to radeon_drm.h

This lets us libdrm from master for r6xx/r7xx 3d
---
diff --git a/shared-core/radeon_drm.h b/shared-core/radeon_drm.h
index e9f2afa..3fc417a 100644
--- a/shared-core/radeon_drm.h
+++ b/shared-core/radeon_drm.h
@@ -303,6 +303,7 @@ typedef union {
 #define RADEON_INDEX_PRIM_OFFSET	20
 
 #define RADEON_SCRATCH_REG_OFFSET	32
+#define R600_SCRATCH_REG_OFFSET         256
 
 #define RADEON_NR_SAREA_CLIPRECTS	12
 
--
cgit v0.8.2
From 9aed44beeac4f250a58c792d64a4dee1dde3d086 Mon Sep 17 00:00:00 2001
From: Alex Deucher <alexdeucher@gmail.com>
Date: Wed, 15 Jul 2009 18:51:38 +0000
Subject: Add RADEON_INIT_R600_CP too

this rounds out the r6xx/r7xx stuff in radeon_drm.h
---
diff --git a/shared-core/radeon_drm.h b/shared-core/radeon_drm.h
index 3fc417a..85aa025 100644
--- a/shared-core/radeon_drm.h
+++ b/shared-core/radeon_drm.h
@@ -548,7 +548,8 @@ typedef struct drm_radeon_init {
 		RADEON_INIT_CP = 0x01,
 		RADEON_CLEANUP_CP = 0x02,
 		RADEON_INIT_R200_CP = 0x03,
-		RADEON_INIT_R300_CP = 0x04
+		RADEON_INIT_R300_CP = 0x04,
+		RADEON_INIT_R600_CP = 0x05
 	} func;
 	unsigned long sarea_priv_offset;
 	int is_pci; /* for overriding only */
--
cgit v0.8.2
