--- util/grub.d/10_linux.in.old	2013-07-28 08:55:47.825946239 +0200
+++ util/grub.d/10_linux.in	2013-07-28 08:56:14.082201512 +0200
@@ -138,7 +138,7 @@
   message="$(gettext_printf "Loading Linux %s ..." "${version}")"
   sed "s/^/$submenu_indentation/" << EOF
 	echo	'$(echo "$message" | grub_quote)'
-	linux	${rel_dirname}/${basename} root=${linux_root_device_thisversion} ro ${args}
+	linux	${rel_dirname}/${basename} root=${linux_root_device_thisversion} rw ${args}
 EOF
   if test -n "${initrd}" ; then
     # TRANSLATORS: ramdisk isn't identifier. Should be translated.
