--- gtk+-2.11.6/gtk/gtkclipboard.c.orig	2007-08-18 13:48:22.000000000 +0000
+++ gtk+-2.11.6/gtk/gtkclipboard.c	2007-08-18 13:49:08.000000000 +0000
@@ -287,6 +287,8 @@
 gtk_clipboard_get_for_display (GdkDisplay *display, 
 			       GdkAtom     selection)
 {
+  if (display == NULL)
+	  return NULL;
   g_return_val_if_fail (GDK_IS_DISPLAY (display), NULL);
   g_return_val_if_fail (!display->closed, NULL);
 
