--- io/hpmud/hp-mkuri.c.orig	2013-03-17 09:15:15.360624061 -0400
+++ io/hpmud/hp-mkuri.c	2013-03-17 09:15:30.924316847 -0400
@@ -266,7 +266,7 @@
 
     /* Bypass glib build dependencies by loading libnotify manually. */  
 
-    if ((handle = load_library("libnotify.so.1")) == NULL)
+    if ((handle = load_library("libnotify.so.4")) == NULL)
     {
        BUG("failed to open libnotify: %m\n");
        goto bugout;
