diff -Nur iptables-1.4.12.2/iptables/ip6tables-restore.c iptables-1.4.12.2-fixrestore/iptables/ip6tables-restore.c
--- iptables-1.4.12.2/iptables/ip6tables-restore.c	2012-01-03 02:19:09.000000000 +0900
+++ iptables-1.4.12.2-fixrestore/iptables/ip6tables-restore.c	2012-03-01 10:56:10.000000000 +0900
@@ -380,9 +380,9 @@
 			quote_open = 0;
 			escaped = 0;
 			param_len = 0;
+			char param_buffer[1024];
 
 			for (curchar = parsestart; *curchar; curchar++) {
-				char param_buffer[1024];
 
 				if (quote_open) {
 					if (escaped) {
diff -Nur iptables-1.4.12.2/iptables/iptables-restore.c iptables-1.4.12.2-fixrestore/iptables/iptables-restore.c
--- iptables-1.4.12.2/iptables/iptables-restore.c	2012-01-03 02:19:09.000000000 +0900
+++ iptables-1.4.12.2-fixrestore/iptables/iptables-restore.c	2012-03-01 10:56:00.000000000 +0900
@@ -377,9 +377,9 @@
 			quote_open = 0;
 			escaped = 0;
 			param_len = 0;
+			char param_buffer[1024];
 
 			for (curchar = parsestart; *curchar; curchar++) {
-				char param_buffer[1024];
 
 				if (quote_open) {
 					if (escaped) {
