--- panel-plugin/systemload.c	2007-01-17 18:01:09.000000000 +0000
+++ panel-plugin/systemload.c.new	2009-05-14 17:17:24.001076632 +0000
@@ -315,11 +315,18 @@
 {
     int count;
     t_global_monitor *global;
+    GtkSettings *settings;
 
     tooltips = gtk_tooltips_new ();
     g_object_ref (tooltips);
     gtk_object_sink (GTK_OBJECT (tooltips));
     
+    /* reduce the default tooltip timeout to be smaller than the update interval otherwise
+     * we won't see tooltips on GTK 2.16 or newer */
+    	settings = gtk_settings_get_default();
+    if (g_object_class_find_property(G_OBJECT_GET_CLASS(settings), "gtk-tooltip-timeout"))
+        g_object_set(settings, "gtk-tooltip-timeout", UPDATE_TIMEOUT - 10, NULL);
+
     global = g_new(t_global_monitor, 1);
     global->plugin = plugin;
     global->timeout_id = 0;
