--- sbin/update-ca-certificates~	2008-06-02 01:51:30.000000000 +0200
+++ sbin/update-ca-certificates	2008-06-02 01:51:30.000000000 +0200
@@ -88,15 +88,6 @@
     c_rehash .
   fi
   echo "done."
-
-  HOOKSDIR=/etc/ca-certificates/update.d
-  echo -n "Running hooks in $HOOKSDIR...."
-  VERBOSE_ARG=
-  [ "$verbose" = 0 ] || VERBOSE_ARG=--verbose
-  eval run-parts $VERB_ARG --test -- $HOOKSDIR | while read hook; do
-  printf -- "${removed:+$removed\n}${added:+$added\n}" | eval $hook
-  done
-  echo "done."
 else
   echo "done."
 fi
