--- config.c	2011-06-21 04:12:02.000000000 -0400
+++ config.c	2011-07-12 13:47:36.274319050 -0400
@@ -41,39 +41,6 @@
 #include "asprintf.c"
 #endif
 
-#if !defined(asprintf)
-#include <stdarg.h>
-
-int asprintf(char **string_ptr, const char *format, ...)
-{
-	va_list arg;
-	char *str;
-	int size;
-	int rv;
-
-	va_start(arg, format);
-	size = vsnprintf(NULL, 0, format, arg);
-	size++;
-	va_start(arg, format);
-	str = malloc(size);
-	if (str == NULL) {
-		va_end(arg);
-		/*
-		 * Strictly speaking, GNU asprintf doesn't do this,
-		 * but the caller isn't checking the return value.
-		 */
-		fprintf(stderr, "failed to allocate memory\\n");
-		exit(1);
-	}
-	rv = vsnprintf(str, size, format, arg);
-	va_end(arg);
-
-	*string_ptr = str;
-	return (rv);
-}
-
-#endif
-
 #if !defined(strndup)
 char *strndup(const char *s, size_t n)
 {
--- logrotate.h	2011-06-21 04:12:02.000000000 -0400
+++ logrotate.h	2011-07-12 13:47:38.949285608 -0400
@@ -66,8 +66,5 @@ extern int numLogs;
 extern int debug;
 
 int readAllConfigPaths(const char **paths);
-#if !defined(asprintf)
-int asprintf(char **string_ptr, const char *format, ...);
-#endif
 
 #endif
