diff --git a/modules/nvidia-kernel/nv-i2c.c b/modules/nvidia-kernel/nv-i2c.c
index a702506..ece9c3c 100644
--- a/modules/nvidia-kernel/nv-i2c.c
+++ b/modules/nvidia-kernel/nv-i2c.c
@@ -324,13 +324,10 @@ BOOL NV_API_CALL nv_i2c_del_adapter(nv_state_t *nv, void *data)
     if (!pI2cAdapter) return FALSE;
 
     // attempt release with the OS
-    osstatus = i2c_del_adapter(pI2cAdapter);
+    i2c_del_adapter(pI2cAdapter);
 
-    if (!osstatus)
-    {
-        os_free_mem(pI2cAdapter);
-        wasReleased = TRUE;
-    }
+	os_free_mem(pI2cAdapter);
+	wasReleased = TRUE;
 
     return wasReleased;
 }
diff --git a/modules/nvidia-kernel/nv-procfs.c b/modules/nvidia-kernel/nv-procfs.c
index 3e05ef5..75fc7e0 100644
--- a/modules/nvidia-kernel/nv-procfs.c
+++ b/modules/nvidia-kernel/nv-procfs.c
@@ -14,6 +14,8 @@
 #include "os-interface.h"
 #include "nv-linux.h"
 
+#undef CONFIG_PROC_FS
+
 #if defined(CONFIG_PROC_FS)
 
 #include "nv_compiler.h"