diff -up patch-2.6.1/src/patch.c.get-arg patch-2.6.1/src/patch.c
--- patch-2.6.1/src/patch.c.get-arg	2010-07-27 18:05:26.217150510 +0100
+++ patch-2.6.1/src/patch.c	2010-07-27 18:05:47.464150892 +0100
@@ -558,7 +558,7 @@ static struct option const longopts[] =
   {"remove-empty-files", no_argument, NULL, 'E'},
   {"force", no_argument, NULL, 'f'},
   {"fuzz", required_argument, NULL, 'F'},
-  {"get", no_argument, NULL, 'g'},
+  {"get", required_argument, NULL, 'g'},
   {"input", required_argument, NULL, 'i'},
   {"ignore-whitespace", no_argument, NULL, 'l'},
 #ifdef ENABLE_MERGE
