--- a/embed/ephy-embed-shell.c
+++ b/embed/ephy-embed-shell.c
@@ -671,6 +671,7 @@ ephy_embed_shell_create_web_context (EphyEmbedShell *shell)
   g_free (cache_dir);
 
   priv->web_context = webkit_web_context_new_with_website_data_manager (manager);
+  webkit_web_context_set_additional_plugins_directory (priv->web_context, EPHY_PLUGINS_DIR);
   g_object_unref (manager);
 }
 
--- a/meson.build	2017-09-18 11:49:28.745287692 +0000
+++ b/meson.build	2017-09-18 11:52:53.432417200 +0000
@@ -25,6 +25,7 @@
 conf.set_quoted('BUILD_ROOT', meson.build_root())
 conf.set_quoted('DISTRIBUTOR_NAME', get_option('distributor_name'))
 conf.set_quoted('EPHY_WEB_EXTENSIONS_DIR', webextensionsdir)
+conf.set_quoted('EPHY_PLUGINS_DIR', join_paths(pkglibdir, 'plugins'))
 conf.set_quoted('GETTEXT_PACKAGE', meson.project_name())
 conf.set_quoted('ISO_CODES_PREFIX', iso_codes_prefix)
 conf.set_quoted('LOCALEDIR', localedir)
