--- a/embed/ephy-embed-shell.c
+++ b/embed/ephy-embed-shell.c
@@ -671,6 +671,7 @@ ephy_embed_shell_create_web_context (EphyEmbedShell *shell)
   g_free (cache_dir);
 
   priv->web_context = webkit_web_context_new_with_website_data_manager (manager);
+  webkit_web_context_set_additional_plugins_directory (priv->web_context, EPHY_PLUGINS_DIR);
   g_object_unref (manager);
 }
 
--- a/embed/Makefile.am.orig	2017-04-08 12:22:50.192861523 +0000
+++ b/embed/Makefile.am	2017-04-08 12:23:13.926290709 +0000
@@ -61,6 +61,7 @@
 	-I$(top_srcdir)/lib/widgets		\
 	-I$(top_srcdir)/libgd			\
 	-DPKGDATADIR=\"$(pkgdatadir)\"		\
+	-DEPHY_PLUGINS_DIR=\""$(libdir)/epiphany/plugins"\" \
 	-DEPHY_WEB_EXTENSIONS_DIR=\"$(pkglibdir)/web-extensions\"	\
 	$(GDK_CFLAGS)				\
 	$(GIO_CFLAGS)				\
