--- QScintilla_gpl-2.9.4/Python/configure.py.orig	2017-02-03 21:30:08.943203061 +0100
+++ QScintilla_gpl-2.9.4/Python/configure.py	2017-02-03 21:29:50.836480389 +0100
@@ -1318,8 +1318,6 @@
     sip_file = module_config.get_sip_file(target_config)
 
     head, tail = os.path.split(sip_file)
-    while head:
-        head, tail = os.path.split(head)
 
     if tail != sip_file:
         argv.append('-I')
