diff -up QScintilla-gpl-2.9.1/Qt4Qt5/features/qscintilla2.prf.qt5 QScintilla-gpl-2.9.1/Qt4Qt5/features/qscintilla2.prf
--- QScintilla-gpl-2.9.1/Qt4Qt5/features/qscintilla2.prf.qt5	2015-10-24 08:00:37.000000000 -0500
+++ QScintilla-gpl-2.9.1/Qt4Qt5/features/qscintilla2.prf	2015-10-24 10:44:02.960900703 -0500
@@ -17,9 +17,17 @@ CONFIG(debug, debug|release) {
         win32: {
             LIBS += -lqscintilla2d
         } else {
+        greaterThan(QT_MAJOR_VERSION, 4) {
+            LIBS += -lqscintilla2-qt$${QT_MAJOR_VERSION}
+        } else {
             LIBS += -lqscintilla2
         }
+        }
     }
 } else {
+greaterThan(QT_MAJOR_VERSION, 4) {
+    LIBS += -lqscintilla2-qt$${QT_MAJOR_VERSION}
+} else {
     LIBS += -lqscintilla2
 }
+}
diff -up QScintilla-gpl-2.9.1/Qt4Qt5/qscintilla.pro.qt5 QScintilla-gpl-2.9.1/Qt4Qt5/qscintilla.pro
--- QScintilla-gpl-2.9.1/Qt4Qt5/qscintilla.pro.qt5	2015-10-24 10:44:02.961900732 -0500
+++ QScintilla-gpl-2.9.1/Qt4Qt5/qscintilla.pro	2015-10-24 10:45:16.445068859 -0500
@@ -23,7 +23,12 @@
 !win32:VERSION = 12.0.1
 
 TEMPLATE = lib
+greaterThan(QT_MAJOR_VERSION, 4) {
+TARGET = qscintilla2-qt$${QT_MAJOR_VERSION}
+} else {
 TARGET = qscintilla2
+}
+
 CONFIG += qt warn_off release thread exceptions
 INCLUDEPATH += . ../include ../lexlib ../src
 
