--- dhcpcd-hooks/10-mtu.orig	2011-11-04 07:45:46.477712667 +0100
+++ dhcpcd-hooks/10-mtu	2011-11-04 07:46:38.964379550 +0100
@@ -7,7 +7,7 @@
 elif [ -n "$new_interface_mtu" ] && $if_up; then
 	# The smalled MTU dhcpcd can work with is 576
 	if [ "$new_interface_mtu" -ge 576 ]; then
-		if ifconfig "$interface" mtu "$new_interface_mtu"; then
+		if ip link set "$interface" mtu "$new_interface_mtu"; then
 			syslog info "$interface: MTU set to $new_interface_mtu"
 			# Save the MTU so we can restore it later
 			if [ ! -e "$mtu_dir/$interface" ]; then
@@ -21,7 +21,7 @@
 		# No MTU in this state, so restore the prior MTU
 		mtu=$(cat "$mtu_dir/$interface")
 		syslog info "$interface: MTU restored to $mtu"
-		ifconfig "$interface" mtu "$mtu"
+		ip link set "$interface" mtu "$new_interface_mtu"
 		rm "$mtu_dir/$interface"
 	fi
 fi
