--- libtool	2011-02-20 10:46:47.000000000 +0100
+++ libtool.new	2011-02-20 10:52:58.285764909 +0100
@@ -324,7 +324,7 @@
 
 # Flag to hardcode $libdir into a binary during linking.
 # This must work even if $libdir does not exist
-hardcode_libdir_flag_spec="\${wl}-rpath \${wl}\$libdir"
+hardcode_libdir_flag_spec=" "
 
 # If ld is used when linking, flag to hardcode $libdir into a binary
 # during linking.  This must work even if $libdir does not exist.
