diff --git a/tests/hostname-check.c b/tests/hostname-check.c
index 120f3b9..670248a 100644
--- a/tests/hostname-check.c
+++ b/tests/hostname-check.c
@@ -1147,6 +1147,7 @@ void doit(void)
 	if (ret)
 		fail("%d: Hostname incorrectly matches (%d)\n", __LINE__, ret);
 
+#ifdef HAVE_LIBIDN
 	ret = gnutls_x509_crt_check_hostname(x509, "www.teχ.gr");
 	if (!ret)
 		fail("%d: Hostname incorrectly does not match (%d)\n", __LINE__, ret);
@@ -1166,6 +1167,7 @@ void doit(void)
 	ret = gnutls_x509_crt_check_hostname(x509, "ΤΈΣΤ.gr");
 	if (!ret)
 		fail("%d: Hostname incorrectly does not match (%d)\n", __LINE__, ret);
+#endif
 
 #ifdef ENABLE_OPENPGP
 	if (debug)

