diff -u -r gtk+-3.4.2/gdk/x11/gdkdevicemanager-xi2.c gtk+-3.4.2-wacom/gdk/x11/gdkdevicemanager-xi2.c
--- gtk+-3.4.2/gdk/x11/gdkdevicemanager-xi2.c	2012-05-02 14:45:04.000000000 +0200
+++ gtk+-3.4.2-wacom/gdk/x11/gdkdevicemanager-xi2.c	2012-05-03 18:34:18.034807644 +0200
@@ -331,6 +331,8 @@
         input_source = GDK_SOURCE_ERASER;
       else if (strstr (tmp_name, "cursor"))
         input_source = GDK_SOURCE_CURSOR;
+      else if (strstr (tmp_name, "finger"))
+        input_source = GDK_SOURCE_TOUCHSCREEN;
       else if (strstr (tmp_name, "wacom") ||
                strstr (tmp_name, "pen"))
         input_source = GDK_SOURCE_PEN;
