Index: ce_parser.y
===================================================================
--- gcalctool/ce_parser.y	(revision 2044)
+++ gcalctool/ce_parser.y	(working copy)
@@ -97,10 +97,12 @@
 %start statement
 %left '+' '-'
 %left '*' '/'
-%right '^'
 %left MED
 %left NEG
 %left POS
+%right '^'
+%right '!'
+%right '%'
 %left HIGH
 
 %%
@@ -200,12 +202,12 @@
 | rcl {cp($1, $$);}
 | term '/' term {mpdiv($1, $3, $$);}
 | term '*' term {mpmul($1, $3, $$);}
-| term '^' term {calc_xpowy($1, $3, $$);}
 | 'e' '^' term {calc_epowy($3, $$);} 
 | term '!' {do_factorial($1 ,$$);}
 | term '%' {calc_percent($1, $$);}
 | '-' term %prec NEG {mpneg($2, $$);}
 | '+' term %prec POS {cp($2, $$);}
+| term '^' term {calc_xpowy($1, $3, $$);}
 
 | func {cp($1, $$);}
 | reg {cp($1, $$);}
