diff -Naur pan-0.132-orig/pan/data-impl/my-tree.cc pan-0.132/pan/data-impl/my-tree.cc
--- pan-0.132-orig/pan/data-impl/my-tree.cc	2008-04-30 00:04:44.000000000 -0400
+++ pan-0.132/pan/data-impl/my-tree.cc	2008-04-30 00:06:53.000000000 -0400
@@ -19,7 +19,7 @@
 
 #include <config.h>
 #include <cassert>
-#include <glib/gmessages.h> // for g_assert
+#include <glib.h> // for g_assert
 #include <pan/general/debug.h>
 #include <pan/general/foreach.h>
 #include <pan/general/quark.h>
diff -Naur pan-0.132-orig/pan/general/file-util.cc pan-0.132/pan/general/file-util.cc
--- pan-0.132-orig/pan/general/file-util.cc	2008-04-30 00:04:44.000000000 -0400
+++ pan-0.132/pan/general/file-util.cc	2008-04-30 00:10:06.000000000 -0400
@@ -19,6 +19,7 @@
 
 #include <config.h>
 
+#include <algorithm>
 #include <cassert>
 #include <cerrno>
 #include <cctype>
diff -Naur pan-0.132-orig/pan/general/line-reader.cc pan-0.132/pan/general/line-reader.cc
--- pan-0.132-orig/pan/general/line-reader.cc	2008-04-30 00:04:44.000000000 -0400
+++ pan-0.132/pan/general/line-reader.cc	2008-04-30 00:10:06.000000000 -0400
@@ -1,5 +1,6 @@
 #include <config.h>
 #include <cstdio>
+#include <cstdlib>
 extern "C"{ 
   #include <glib/gi18n.h>
 }
diff -Naur pan-0.132-orig/pan/general/string-view.h pan-0.132/pan/general/string-view.h
--- pan-0.132-orig/pan/general/string-view.h	2008-04-30 00:04:44.000000000 -0400
+++ pan-0.132/pan/general/string-view.h	2008-04-30 00:10:06.000000000 -0400
@@ -22,6 +22,7 @@
 
 #include <iosfwd>
 #include <string>
+#include <cstring>
 
 namespace pan
 {
diff -Naur pan-0.132-orig/pan/gui/prefs.cc pan-0.132/pan/gui/prefs.cc
--- pan-0.132-orig/pan/gui/prefs.cc	2008-04-30 00:04:44.000000000 -0400
+++ pan-0.132/pan/gui/prefs.cc	2008-04-30 00:06:18.000000000 -0400
@@ -22,6 +22,7 @@
 #include <fstream>
 #include <sstream>
 #include <string>
+#include <cstdlib>
 extern "C" {
   #include <glib.h>
   #include <glib/gi18n.h>
diff -Naur pan-0.132-orig/pan/tasks/socket-impl-scripted.cc pan-0.132/pan/tasks/socket-impl-scripted.cc
--- pan-0.132-orig/pan/tasks/socket-impl-scripted.cc	2008-04-30 00:04:44.000000000 -0400
+++ pan-0.132/pan/tasks/socket-impl-scripted.cc	2008-04-30 00:07:33.000000000 -0400
@@ -19,6 +19,7 @@
 
 #include <config.h>
 #include <cassert>
+#include <cstdlib>
 #include <pan/general/debug.h>
 #include <pan/general/messages.h>
 #include <pan/general/string-view.h>
diff -Naur pan-0.132-orig/pan/usenet-utils/gnksa-test.cc pan-0.132/pan/usenet-utils/gnksa-test.cc
--- pan-0.132-orig/pan/usenet-utils/gnksa-test.cc	2008-04-30 00:04:44.000000000 -0400
+++ pan-0.132/pan/usenet-utils/gnksa-test.cc	2008-04-30 00:08:53.000000000 -0400
@@ -1,6 +1,7 @@
 #include <config.h>
 #include <iostream>
 #include <string>
+#include <climits>
 #include <pan/general/string-view.h>
 #include <pan/general/test.h>
 #include "gnksa.h"
diff -Naur pan-0.132-orig/pan/usenet-utils/numbers.cc pan-0.132/pan/usenet-utils/numbers.cc
--- pan-0.132-orig/pan/usenet-utils/numbers.cc	2008-04-30 00:04:44.000000000 -0400
+++ pan-0.132/pan/usenet-utils/numbers.cc	2008-04-30 00:08:53.000000000 -0400
@@ -19,6 +19,7 @@
 
 #include <config.h>
 #include <cctype>
+#include <algorithm>
 #include <pan/general/string-view.h>
 #include "numbers.h"
 
