--- Makefile	2003-06-04 21:01:45.000000000 -0400
+++ Makefile.fixed	2004-08-13 08:24:50.000000000 -0400
@@ -11,15 +11,14 @@
 # assignment operators....)
 # ----------------------------------------------------------------------
 
-PREFIX = /usr/local
-SHARE = /share
+PREFIX = #PREFIX#
 
 # This is where the 'host' executable will be referenced.
 BINDIR = ${PREFIX}/bin
 
 # This is where manual pages will be referenced.
-MANDIR = ${PREFIX}${SHARE}/man/man1
-CATMANDIR = ${PREFIX}${SHARE}/man/cat1
+MANDIR = ${PREFIX}/man/man1
+CATMANDIR = ${PREFIX}/man/cat1
 CATMANSUF = .1
 
 # This is where the rblookup, etc. config file(s) will be referenced.
@@ -139,15 +138,13 @@
  -Wswitch \
  -Wcomment \
  -Wcast-qual \
- -Wid-clash-30 \
  -Wpointer-arith \
  -Wshadow
 #endif
 
 #if $(__GNULD__) >= 1
 GNULDWARNFLAGS = -W \
- -Wall \
- -Wid-clash-30
+ -Wall 
 #endif
 
 #if $(__GNUC__) >= 2
@@ -177,7 +174,7 @@
 #else f defined(ultrix)
 #CC = cc -Olimit 1000
 #else if gcc != cc
-#CC = gcc
+CC = gcc
 #endif
 
 # ----------------------------------------------------------------------
@@ -216,7 +213,7 @@
 #RES_LIB = -lsocket
 #endif
 #if defined(NEED_LIBRESOLV) || (sunos5.x) || defined(__LINUX__) || defined(__GLIBC__)
-#RES_LIB = -lresolv
+RES_LIB = -lresolv
 #endif
 #if (RedHat-8.x) && defined(__GLIBC__)
 #RES_LIB = -static -I/usr/lib/debug -lresolv
@@ -294,7 +291,7 @@
 # ----------------------------------------------------------------------
 
 BINOWN = root
-BINGRP = staff
+BINGRP = root
 BINMODE = 755
 #STRIPFLAG = -s
 
@@ -384,3 +381,4 @@
 # Keep it simple....  it's not that big a program!
 #
 $(OBJS): $(HDRS)
+
