Index: kget/transfer-plugins/bittorrent/bttransferfactory.cpp
===================================================================
--- kget/transfer-plugins/bittorrent/bttransferfactory.cpp	(revision 1209130)
+++ kget/transfer-plugins/bittorrent/bttransferfactory.cpp	(revision 1209131)
@@ -28,12 +28,13 @@
 BTTransferFactory::BTTransferFactory(QObject *parent, const QVariantList &args)
   : TransferFactory(parent, args)
 {
+#if LIBKTORRENT_VERSION >= 0x010100
     if (!bt::InitLibKTorrent())
     {
         kError(5001) << "Failed to initialize libktorrent";
         KGet::showNotification(0, "error", i18n("Cannot initialize libktorrent. Torrent support might not work."));
     }
-#if LIBKTORRENT_VERSION >= 0x010100
+
     bt::Job::setJobTracker(new BTJobTracker(this));
 #endif
 }
