diff -Nur comm-1.9.2.orig/mail/app/profile/all-thunderbird.js comm-1.9.2/mail/app/profile/all-thunderbird.js
--- comm-1.9.2.orig/mail/app/profile/all-thunderbird.js	2010-06-23 20:43:08.000000000 +0300
+++ comm-1.9.2/mail/app/profile/all-thunderbird.js	2010-06-25 00:37:36.518929684 +0300
@@ -125,7 +125,7 @@
 pref("app.update.showInstalledUI", false);
 
 // Release notes URL
-pref("app.releaseNotesURL", "http://live.mozillamessaging.com/%APP%/releasenotes?locale=%LOCALE%&version=%VERSION%&os=%OS%&buildid=%APPBUILDID%");
+pref("app.releaseNotesURL", "http://www.mozillamessaging.com/%APP%/%VERSION%/releasenotes?locale=%LOCALE%&version=%VERSION%&os=%OS%&buildid=%APPBUILDID%");
 
 // Base URL for web-based support pages.
 pref("app.support.baseURL", "http://support.live.mozillamessaging.com/%LOCALE%/%APP%/%APPBUILDID%/");
diff -Nur comm-1.9.2.orig/mail/branding/nightly/thunderbird-branding.js comm-1.9.2/mail/branding/nightly/thunderbird-branding.js
--- comm-1.9.2.orig/mail/branding/nightly/thunderbird-branding.js	2010-06-23 20:43:08.000000000 +0300
+++ comm-1.9.2/mail/branding/nightly/thunderbird-branding.js	2010-06-25 00:36:10.605569121 +0300
@@ -1,11 +1,11 @@
 // Default start page
-pref("mailnews.start_page.url","http://live.mozillamessaging.com/%APP%/start?locale=%LOCALE%&version=%VERSION%&os=%OS%&buildid=%APPBUILDID%");
+pref("mailnews.start_page.url","http://www.mozillamessaging.com/%APP%/%VERSION%/start?locale=%LOCALE%&version=%VERSION%&os=%OS%&buildid=%APPBUILDID%");
 
 // first launch welcome page
-pref("mailnews.start_page.welcome_url","http://live.mozillamessaging.com/%APP%/firstrun?locale=%LOCALE%&version=%VERSION%&os=%OS%&buildid=%APPBUILDID%");
+pref("mailnews.start_page.welcome_url","http://www.mozillamessaging.com/%APP%/%VERSION%/firstrun?locale=%LOCALE%&version=%VERSION%&os=%OS%&buildid=%APPBUILDID%");
 
 // start page override to load after an update
-pref("mailnews.start_page.override_url","http://live.mozillamessaging.com/%APP%/whatsnew?locale=%LOCALE%&version=%VERSION%&os=%OS%&buildid=%APPBUILDID%");
+pref("mailnews.start_page.override_url","http://www.mozillamessaging.com/%APP%/%VERSION%/whatsnew?locale=%LOCALE%&version=%VERSION%&os=%OS%&buildid=%APPBUILDID%");
 
 // Interval: Time between checks for a new version (in seconds)
 // nightly=8 hours, official=24 hours
