diff -up comm-central/mail/installer/Makefile.in.shared-error comm-central/mail/installer/Makefile.in
--- comm-central/mail/installer/Makefile.in.shared-error	2009-09-16 13:47:43.000000000 +0200
+++ comm-central/mail/installer/Makefile.in	2009-09-16 13:48:25.000000000 +0200
@@ -95,9 +95,11 @@ endif
 # mozconfig instead.
 ifndef MAIL_PKG_SHARED
 ifndef BUILD_STATIC_LIBS
+ifeq (BUILD_STATIC_LIBS, 1)
 $(error you need an "--enable-static" build to package a build)
 endif
 endif
+endif
 
 # On mozilla-central, packager.mk uses core_abspath to redefine DIST
 # As the comm-central config.mk doesn't set it, do it ourselves instead
