From 96d87da379879f20e2c162b468525aaa1a0a82dd Mon Sep 17 00:00:00 2001
From: Matthias Clasen <mclasen@redhat.com>
Date: Sat, 04 Sep 2010 00:30:54 +0000
Subject: Remove redundant header inclusions

---
diff --git a/glib/gregex.c b/glib/gregex.c
index 576632d..f21fbe8 100644
--- a/glib/gregex.c
+++ b/glib/gregex.c
@@ -23,16 +23,16 @@
 
 #include <string.h>
 
-#include "glib.h"
-#include "glibintl.h"
-#include "gregex.h"
-
 #ifdef USE_SYSTEM_PCRE
 #include <pcre.h>
 #else
 #include "pcre/pcre.h"
 #endif
 
+#include "gtypes.h"
+#include "gregex.h"
+#include "glibintl.h"
+
 /**
  * SECTION:gregex
  * @title: Perl-compatible regular expressions
--
cgit v0.8.3.1
