Index: khtml/khtml_part.cpp
===================================================================
--- khtml/khtml_part.cpp	(Revision 1032786)
+++ khtml/khtml_part.cpp	(Revision 1032787)
@@ -609,8 +609,7 @@
 
   if ( d->m_view )
   {
-    d->m_view->hide();
-    d->m_view->viewport()->hide();
+    widget()->hide();
     d->m_view->m_part = 0;
   }
 
Index: khtml/khtml_part.cpp
===================================================================
--- khtml/khtml_part.cpp	(Revision 1033983)
+++ khtml/khtml_part.cpp	(Revision 1033984)
@@ -609,7 +609,8 @@
 
   if ( d->m_view )
   {
-    widget()->hide();
+    if (widget())
+        widget()->hide();
     d->m_view->m_part = 0;
   }
 
