diff -u -r gnome-nettool-3.2.0/src/lookup.c gnome-nettool-3.2.0-ldns/src/lookup.c
--- gnome-nettool-3.2.0/src/lookup.c	2012-01-03 10:38:09.000000000 +0100
+++ gnome-nettool-3.2.0-ldns/src/lookup.c	2013-03-13 12:32:31.551213914 +0100
@@ -98,7 +98,7 @@
 
 	parent = gtk_widget_get_toplevel (netinfo->output);
 	
-	program = util_find_program_dialog ("dig", parent);
+	program = util_find_program_dialog ("drill", parent);
 
 	if (program != NULL) {
 		use_reverse_lookup = pattern_match (host, address_regular_expression);
@@ -116,7 +116,7 @@
 		command_line = g_new (gchar *, num_terms + 1);
 		i = 0;
 		command_line[i++] = g_strdup (program);
-		command_line[i++] = g_strdup ("dig");
+		command_line[i++] = g_strdup ("drill");
 		if (command_options != NULL) {
 			for (j = 0; command_options[j] != NULL; j++)
 				command_line[i++] = g_strdup (command_options[j]);
