diff --git i/tools/extra_defs_gen/meson.build w/tools/extra_defs_gen/meson.build
index 900943c9..239a543a 100644
--- i/tools/extra_defs_gen/meson.build
+++ w/tools/extra_defs_gen/meson.build
@@ -13,6 +13,8 @@ glibmm_generate_extra_defs_library = library(
   install: true,
 )
 
+install_headers('generate_extra_defs.h', subdir: glibmm_pcname / 'glibmm_generate_extra_defs')
+
 executable('generate_defs_glib', 'generate_defs_glib.cc',
   dependencies: glibmm_build_dep,
   link_with: glibmm_generate_extra_defs_library,
