--- battstat/battstat_applet.c
+++ battstat/battstat_applet.c
@@ -42,6 +42,9 @@
 
 #ifdef HAVE_LIBNOTIFY
 #include <libnotify/notify.h>
+#ifndef NOTIFY_CHECK_VERSION
+#define NOTIFY_CHECK_VERSION(x,y,z) 0
+#endif
 #endif
 
 #include "battstat.h"
@@ -435,7 +438,12 @@
 			GTK_ICON_LOOKUP_USE_BUILTIN,
 			NULL);
 	
-	NotifyNotification *n = notify_notification_new (_("Your battery is now fully recharged"), "", /* "battery" */ NULL, applet);
+	NotifyNotification *n = notify_notification_new (_("Your battery is now fully recharged"), "", /* "battery" */ NULL
+#if NOTIFY_CHECK_VERSION (0, 7, 0)
+	);
+#else
+	, applet);
+#endif
 
 	/* XXX: it would be nice to pass this as a named icon */
 	notify_notification_set_icon_from_pixbuf (n, icon);
--- gweather/gweather-applet.c
+++ gweather/gweather-applet.c
@@ -30,6 +30,9 @@
 #ifdef HAVE_LIBNOTIFY
 #include <libnotify/notify.h>
 #include <libnotify/notification.h>
+#ifndef NOTIFY_CHECK_VERSION
+#define NOTIFY_CHECK_VERSION(x,y,z) 0
+#endif
 #endif
 
 #define GWEATHER_I_KNOW_THIS_IS_UNSTABLE
@@ -471,8 +474,12 @@
 			 if (icon == NULL)
 				 icon = "stock-unknown";
 	           	 
-			 n = notify_notification_new (message, detail, icon,
-					 gw_applet->container);
+			 n = notify_notification_new (message, detail, icon
+#if NOTIFY_CHECK_VERSION (0, 7, 0)
+			 );
+#else
+			 , gw_applet->container);
+#endif
 	
 		   	 notify_notification_show (n, &error);
 			 if (error)
