From 390c5b3aeb3d07ff6de3594fd08370e43b43c652 Mon Sep 17 00:00:00 2001
From: Richard Hughes <richard@hughsie.com>
Date: Wed, 21 Oct 2009 08:26:48 +0000
Subject: Fix a trivial copy-paste error that broke hibernate for a lot of people

---
diff --git a/src/dkp-daemon.c b/src/dkp-daemon.c
index 20e65f8..eb16a63 100644
--- a/src/dkp-daemon.c
+++ b/src/dkp-daemon.c
@@ -563,7 +563,7 @@ dkp_daemon_hibernate (DkpDaemon *daemon, DBusGMethodInvocation *context)
 	}
 
 	/* encrypted swap? */
-	if (!daemon->priv->hibernate_has_encrypted_swap) {
+	if (daemon->priv->hibernate_has_encrypted_swap) {
 		error = g_error_new (DKP_DAEMON_ERROR,
 				     DKP_DAEMON_ERROR_GENERAL,
 				     "Swap space is encrypted");
--
cgit v0.8.2
