--- src/pebrot-0.8.7/pypebrot/pebrot.orig	2005-07-25 07:40:43.000000000 -0400
+++ src/pebrot-0.8.7/pypebrot/pebrot.py	2005-07-25 07:40:55.000000000 -0400
@@ -1877,7 +1877,7 @@
             print _('I can\'t find %s, so I will copy the supplied pebrotrc there.') \
                 % CONFIG_FILE
             raw_input( _('Press Enter to continue...') )
-            shutil.copyfile( sys.prefix + '/share/doc/pebrot/pebrotrc', CONFIG_FILE )
+            shutil.copyfile( '/etc/pebrotrc', CONFIG_FILE )
             #os.chmod( CONFIG_FILE, 0600 )
             print _('Please edit %s and specify your login and password.') \
                 % CONFIG_FILE
