diff --git a/src/mesa/drivers/dri/i965/brw_context.c b/src/mesa/drivers/dri/i965/brw_context.c
index e113654..5790680 100644
--- a/src/mesa/drivers/dri/i965/brw_context.c
+++ b/src/mesa/drivers/dri/i965/brw_context.c
@@ -645,7 +645,7 @@ brwCreateContext(gl_api api,
 
    brw_init_driver_functions(brw, &functions);
 
-   if (notify_reset)
+   if (notify_reset && brw->gen >= 6)
       functions.GetGraphicsResetStatus = brw_get_graphics_reset_status;
 
    struct gl_context *ctx = &brw->ctx;


