--- src/smartbookmark.desktop.in.in.orig	2009-07-04 00:28:14.000000000 +0200
+++ src/smartbookmark.desktop.in.in	2009-07-04 00:34:09.000000000 +0200
@@ -3,6 +3,7 @@
 Encoding=UTF-8
 _Name=SmartBookmark
 _Comment=Query websites from the Xfce panel
-Icon=gtk-preferences
-X-XFCE-Module=@INTERNAL_PLUGIN_PATH@/libsmartbookmark.so
+Icon=gtk-find
+X-XFCE-Module=smartbookmark
+X-XFCE-Module-Path=@INTERNAL_PLUGIN_PATH@
 X-XFCE-Unique=false
