diff -upr xfce4-smartbookmark-plugin-0.5.2.orig/src/smartbookmark.c xfce4-smartbookmark-plugin-0.5.2/src/smartbookmark.c
--- xfce4-smartbookmark-plugin-0.5.2.orig/src/smartbookmark.c	2020-12-24 18:01:51.000000000 +0200
+++ xfce4-smartbookmark-plugin-0.5.2/src/smartbookmark.c	2020-12-24 18:26:20.132328085 +0200
@@ -173,8 +173,8 @@ static t_search *search_new(XfcePanelPlu
     gtk_widget_set_valign(GTK_WIDGET(search->box), GTK_ALIGN_CENTER);
 
     /* default options */
-    search->url = "https://bugs.debian.org/";
-    search->label_text = "BTS";
+    search->url = "https://bugs.archlinux.org/";
+    search->label_text = "ArchBug #";
     search->size = 5;
     search->hide_label = FALSE;
     /* read config file options */
@@ -209,8 +209,8 @@ static void search_read_config(t_search
     if( (rcfile = xfce_rc_simple_open(filename, TRUE) ))
     {
         xfce_rc_set_group(rcfile, NULL);
-        search->url = g_strdup(xfce_rc_read_entry(rcfile,"url","https://bugs.debian.org/"));
-        search->label_text = g_strdup(xfce_rc_read_entry(rcfile,"value","DBS"));
+        search->url = g_strdup(xfce_rc_read_entry(rcfile,"url","https://bugs.archlinux.org/"));
+        search->label_text = g_strdup(xfce_rc_read_entry(rcfile,"value","ArchBug #"));
         search->size = xfce_rc_read_int_entry(rcfile, "size", 5);
         search->hide_label = xfce_rc_read_bool_entry(rcfile, "hidelabel", FALSE);
     }
