diff -aur old/dict/src/lib/compositelookup.cpp new/dict/src/lib/compositelookup.cpp
--- old/dict/src/lib/compositelookup.cpp	2011-07-02 23:58:40.000000000 -0700
+++ new/dict/src/lib/compositelookup.cpp	2011-07-18 07:40:01.795710505 -0700
@@ -50,7 +50,7 @@
 void CompositeLookup::send_net_dict_request(const std::string& dict_id, const std::string& key)
 {
 	NetDictRequest request(dict_id, key);
-	g_assert(NetDictRequests.end() == std::find(NetDictRequests.begin(), NetDictRequests.end(), request));
+	std::find(NetDictRequests.begin(), NetDictRequests.end(), request);
 	NetDictRequests.push_back(request);
 }
 
