diff -Naur old/lib/src/libcommon.cpp new/lib/src/libcommon.cpp
--- old/lib/src/libcommon.cpp	2012-07-12 11:33:30.343333334 +0000
+++ new/lib/src/libcommon.cpp	2012-07-12 11:34:05.006568754 +0000
@@ -614,7 +614,7 @@
 		return EXIT_FAILURE;
 	}
 	while(true) {
-		len = gzread(get_impl(in), buf, buffer_size);
+		len = gzread((gzFile)get_impl(in), buf, buffer_size);
 		if(len < 0) {
 			g_critical(read_file_err, arch_file_name, "");
 			return EXIT_FAILURE;
@@ -871,3 +871,8 @@
 		return res;
 	}
 }
+
+int gzclose_compat(void * file)
+{
+    return gzclose ((gzFile)file);
+}
diff -Naur old/lib/src/libcommon.h new/lib/src/libcommon.h
--- old/lib/src/libcommon.h	2012-07-12 11:33:30.343333334 +0000
+++ new/lib/src/libcommon.h	2012-07-12 11:34:05.006568754 +0000
@@ -187,8 +187,9 @@
 	typedef ResourceWrapper<FILE, FILE*, int, fclose> File;
 }
 
+extern int gzclose_compat(void * file);
 namespace zip {
-typedef ResourceWrapper<void, void*, int, gzclose> gzFile;
+typedef ResourceWrapper<void, void*, int, gzclose_compat> gzFile;
 }
 
 /* Create a new temporary file. Return file name in file name encoding.
