diff -Nur klibc-1.5.15.orig/usr/kinit/nfsmount/mount.c klibc-1.5.15/usr/kinit/nfsmount/mount.c
--- klibc-1.5.15.orig/usr/kinit/nfsmount/mount.c	2009-01-04 20:28:03.000000000 +0100
+++ klibc-1.5.15/usr/kinit/nfsmount/mount.c	2009-10-21 18:39:21.587772034 +0200
@@ -12,6 +12,12 @@
 #include "nfsmount.h"
 #include "sunrpc.h"
 
+
+/* These have been removed from the kernel header
+ * As a workaround, add them here */
+#define MNTPROC_MNT             1
+#define MNTPROC_UMNT            3
+
 static uint32_t mount_port;
 
 struct mount_call {
