diff -aur old/src/keyevents.c new/src/keyevents.c
--- old/src/keyevents.c	2011-01-22 11:54:26.000000000 +0100
+++ new/src/keyevents.c	2011-01-26 11:13:20.366666674 +0100
@@ -303,7 +303,7 @@
 	for (i = 0; i < 3; i++) {
 		if (
 				(key->keysyms[i] == sym) && 
-				((key->keystates[i] | 0x1) == (state | 0x1)))
+				((key->keystates[i] | 0x11) == (state | 0x11)))
 			return 1;
 		else if (key->keysyms[i] == 0)
 			return 0;
