--- fcpcmcia_cs.c~	2010-08-04 13:02:02.150001261 +0200
+++ fcpcmcia_cs.c	2010-08-04 13:02:02.176667927 +0200
@@ -121,33 +121,6 @@
 
 static void cs_detach(struct pcmcia_device *p_dev);
 
-
-/*
-   A linked list of "instances" of the skeleton device.  Each actual
-   PCMCIA card corresponds to one device instance, and is described
-   by one struct pcmcia_device structure (defined in ds.h).
-
-   You may not want to use a linked list for this -- for example, the
-   memory card driver uses an array of struct pcmcia_device pointers, where minor
-   device numbers are used to derive the corresponding array index.
-*/
-
-
-/*
-   A driver needs to provide a dev_node_t structure for each device
-   on a card.  In some cases, there is only one device per card (for
-   example, ethernet cards, modems).  In other cases, there may be
-   many actual or logical devices (SCSI adapters, memory cards with
-   multiple partitions).  The dev_node_t structures need to be kept
-   in a linked list starting at the 'dev' field of a struct pcmcia_device
-   structure.  We allocate them in the card's private data structure,
-   because they generally can't be allocated dynamically.
-*/
-   
-typedef struct local_info_t {
-    dev_node_t	node;
-} local_info_t;
-
 /*======================================================================
 
     cs_attach() creates an "instance" of the driver, allocating
@@ -162,7 +135,6 @@
 
 static int cs_probe(struct pcmcia_device *p_dev)
 {
-    local_info_t *local;
 #if defined (PCMCIA_IRQ_INFO2)
     int i;
 #endif
@@ -194,11 +166,6 @@
     p_dev->conf.ConfigIndex = 1;
     p_dev->conf.Present = PRESENT_OPTION;
 
-    /* Allocate space for private device-specific data */
-    local = kzalloc(sizeof(local_info_t), GFP_KERNEL);
-    if (!local)
-        goto err_kfree;
-    p_dev->priv = local;
     return cs_config(p_dev);
 
  err_kfree:
@@ -249,10 +216,8 @@
 
 static int cs_config(struct pcmcia_device *link)
 {
-    local_info_t *dev;
     int i;
     char devname[128];
-    dev = link->priv;
 
     devname[0] = 0;
     if (link->prod_id[1])
@@ -283,14 +248,6 @@
 
     } while (0);
 
-    /* At this point, the dev_node_t structure(s) should be
-       initialized and arranged in a linked list at link->dev. */
-
-    strcpy(dev->node.dev_name, "A1");
-    dev->node.major = 64;
-    dev->node.minor = 0;
-
-    link->dev_node = &dev->node;
     /* If any step failed, release any partially configured state */
     if (i != 0) {
         ERROR("Failed to setup controller, releasing link...\n");
@@ -309,7 +266,7 @@
 	cs_release(link);
 	return -ENODEV;
     }
-    dev->node.minor = i;
+    link->priv = (void *) (unsigned long) i;
 
     NOTE(
 	"Card driver for '%s' has been set up: i/o %x, irq %d\n", 
--- fcpcmcia_cs.c~	2010-08-04 13:13:33.610000837 +0200
+++ fcpcmcia_cs.c	2010-08-04 13:13:33.633334616 +0200
@@ -165,8 +165,6 @@
 
     return cs_config(p_dev);
 
- err_kfree:
-    return -ENOMEM;
 } /* cs_attach */
 
 /*======================================================================
--- fcpcmcia_cs.c~	2010-08-04 22:01:27.123336123 +0200
+++ fcpcmcia_cs.c	2010-08-04 22:01:27.150002183 +0200
@@ -146,9 +146,6 @@
     p_dev->io.Attributes1 = IO_DATA_PATH_WIDTH_8;
     p_dev->io.NumPorts2 = 0;
 
-    /* Interrupt setup */
-    p_dev->irq.Attributes = IRQ_TYPE_DYNAMIC_SHARING;
-
 #if defined (PCMCIA_IRQ_INFO2)
     p_dev->irq.IRQInfo1 = IRQ_INFO2_VALID|IRQ_LEVEL_ID;
     if (irq_list_count > 0) {
@@ -214,7 +211,7 @@
 
 static int cs_config(struct pcmcia_device *link)
 {
-    int i;
+    int i = -1;
     char devname[128];
 
     devname[0] = 0;
@@ -228,8 +225,7 @@
 	/*
 	 * allocate an interrupt line
 	 */
-	i = pcmcia_request_irq(link, &link->irq);
-	if (i != 0) {
+	if (!link->irq) {
             /* undo */
             pcmcia_disable_device(link);
 	    break;
@@ -259,7 +255,7 @@
 		"Failed to add AVM-%s-Controller at i/o %x, irq %d\n",
 		dev->node.dev_name, 
 		link->io.BasePort1, 
-		link->irq.AssignedIRQ
+		link->irq
 	);
 	cs_release(link);
 	return -ENODEV;
@@ -270,7 +266,7 @@
 	"Card driver for '%s' has been set up: i/o %x, irq %d\n", 
 	devname, 
 	link->io.BasePort1, 
-	link->irq.AssignedIRQ
+	link->irq
     );
     return 0;
 } /* cs_config */
--- fcpcmcia_cs.c~	2010-08-04 22:04:40.550002169 +0200
+++ fcpcmcia_cs.c	2010-08-04 22:04:40.573335565 +0200
@@ -250,7 +250,7 @@
     }
     NOTE("Ready to call card driver for '%s'...\n", devname);
 
-    if ((i = fcpcmcia_addcard(link->io.BasePort1, link->irq.AssignedIRQ)) < 0) {
+    if ((i = fcpcmcia_addcard(link->io.BasePort1, link->irq)) < 0) {
         ERROR(
 		"Failed to add AVM-%s-Controller at i/o %x, irq %d\n",
 		dev->node.dev_name, 
@@ -281,7 +281,7 @@
 
 static void cs_release(struct pcmcia_device *link)
 {
-    fcpcmcia_delcard(link->io.BasePort1, link->irq.AssignedIRQ);
+    fcpcmcia_delcard(link->io.BasePort1, link->irq);
 
     pcmcia_disable_device(link);
 } /* cs_release */
--- fcpcmcia_cs.c~	2010-08-04 22:22:29.090002845 +0200
+++ fcpcmcia_cs.c	2010-08-04 22:22:29.113335966 +0200
@@ -253,7 +253,6 @@
     if ((i = fcpcmcia_addcard(link->io.BasePort1, link->irq)) < 0) {
         ERROR(
 		"Failed to add AVM-%s-Controller at i/o %x, irq %d\n",
-		dev->node.dev_name, 
 		link->io.BasePort1, 
 		link->irq
 	);
