Index: sddm-0.1.0+git20140718/data/CMakeLists.txt
===================================================================
--- sddm-0.1.0+git20140718.orig/data/CMakeLists.txt	2014-07-18 11:36:58.974892509 +0200
+++ sddm-0.1.0+git20140718/data/CMakeLists.txt	2014-07-18 11:36:58.970892509 +0200
@@ -1,7 +1,7 @@
 install(DIRECTORY   "faces"                             DESTINATION "${DATA_INSTALL_DIR}")
 install(DIRECTORY   "flags"                             DESTINATION "${DATA_INSTALL_DIR}")
 
-install(FILES "org.freedesktop.DisplayManager.conf"       DESTINATION "${DBUS_CONFIG_DIR}")
+install(FILES "org.freedesktop.DisplayManager.conf"       DESTINATION "${DBUS_CONFIG_DIR}" RENAME sddm_org.freedesktop.DisplayManager.conf)
 
 install(FILES "scripts/Xsession" "scripts/Xsetup" DESTINATION "${DATA_INSTALL_DIR}/scripts"
         PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE
