diff -ur qt-orig/tools/qdoc3/qdoc3.pro qt-4.5.2/tools/qdoc3/qdoc3.pro
--- qt-orig/tools/qdoc3/qdoc3.pro	2009-06-20 06:57:59.000000000 +0200
+++ qt-4.5.2/tools/qdoc3/qdoc3.pro	2009-06-25 20:00:12.000000000 +0200
@@ -99,3 +99,6 @@
            webxmlgenerator.cpp \
 	   yyindent.cpp
 
+TARGET= qdoc3
+target.path = $$[QT_INSTALL_BINS]
+INSTALLS += target
diff -ur qt-orig/tools/tools.pro qt-4.5.2/tools/tools.pro
--- qt-orig/tools/tools.pro	2009-06-20 06:57:59.000000000 +0200
+++ qt-4.5.2/tools/tools.pro	2009-06-25 19:59:19.000000000 +0200
@@ -13,6 +13,7 @@
          SUBDIRS += designer
      }
      SUBDIRS     += linguist
+     SUBDIRS     += qdoc3
      wince*: SUBDIRS = qtestlib designer
      unix:!mac:!embedded:contains(QT_CONFIG, qt3support):SUBDIRS += qtconfig
      win32:!wince*:SUBDIRS += activeqt
