--- sip-5.0.0/sipbuild/installable.py.orig      2019-12-13 19:12:02.163301429 +0000
+++ sip-5.0.0/sipbuild/installable.py   2019-12-13 19:12:33.600398673 +0000
@@ -58,7 +58,8 @@
         """
 
         target_dir = self.get_full_target_dir(target_dir)
-        os.makedirs(target_dir, exist_ok=True)
+        if do_install:
+            os.makedirs(target_dir, exist_ok=True)
 
         for fn in self.files:
             t_path = os.path.join(target_dir, os.path.basename(fn))
--- sip-5.0.0/sipbuild/distinfo/distinfo.py.orig	2019-12-13 18:28:50.195055008 +0000
+++ sip-5.0.0/sipbuild/distinfo/distinfo.py	2019-12-13 18:30:22.169531691 +0000
@@ -72,20 +72,20 @@
 
     # Make sure we have an empty dist-info directory.  Handle exceptions as the
     # user may be trying something silly with a system directory.
-    if os.path.exists(distinfo_dir):
+    if os.path.exists(real_distinfo_dir):
         try:
-            shutil.rmtree(distinfo_dir)
+            shutil.rmtree(real_distinfo_dir)
         except Exception as e:
             raise UserException(
                     "unable remove old dist-info directory '{}'".format(
-                            distinfo_dir),
+                            real_distinfo_dir),
                     str(e))
 
     try:
-        os.mkdir(distinfo_dir)
+        os.mkdir(real_distinfo_dir)
     except Exception as e:
         raise UserException(
-                "unable create dist-info directory '{}'".format(distinfo_dir),
+                "unable create dist-info directory '{}'".format(real_distinfo_dir),
                 str(e))
 
     # Reproducable builds.
 
