Description: deal with changes in Net::DNS >= 0.75
Origin: CPAN RT
Bug: https://rt.cpan.org/Public/Bug/Display.html?id=95901
Bug-Debian: https://bugs.debian.org/752988
Author: Willen <NLNETLABS@cpan.org>
Reviewed-by: gregor herrmann <gregoa@debian.org>
Last-Update: 2014-06-29

--- a/lib/Net/DNS/Resolver/Programmable.pm
+++ b/lib/Net/DNS/Resolver/Programmable.pm
@@ -204,8 +204,10 @@
     if (defined(my $resolver_code = $self->{resolver_code})) {
         ($result, $aa, @answer_rrs) = $resolver_code->($domain, $rr_type, $class);
     }
-    
-    if (not defined($result) or defined($Net::DNS::rcodesbyname{$result})) {
+
+    if (not defined($result)
+         or defined($Net::DNS::rcodesbyname{$result})
+         or defined($Net::DNS::Parameters::rcodebyname{$result})) {
         # Valid RCODE, return a packet:
         
         $aa     = TRUE      if not defined($aa);
