--- kde-runtime-4.7.2/activitymanager/CMakeLists.txt~	2011-10-29 15:46:55.003710230 +0000
+++ kde-runtime-4.7.2/activitymanager/CMakeLists.txt	2011-10-29 15:47:28.910716093 +0000
@@ -21,7 +21,7 @@
     )
 endif(Nepomuk_FOUND)
 
-# Checking for QtZeitgeist
+# Checking for QZeitgeist
 macro_optional_find_package(QZeitgeist)
 
 if (QZEITGEIST_INCLUDE_DIR)
@@ -30,8 +30,8 @@
 
 macro_log_feature(
         QZeitgeist_FOUND
-        "QtZeitgeist" "Qt bindings for Zeitgeist" "http://gitorious.org/kde-zeitgeist/libqzeitgeist/" FALSE ""
-        "RECOMMENDED: Zeitgeist and QtZeitgeist is needed for resource tracking")
+        "QZeitgeist" "Qt bindings for Zeitgeist" "http://gitorious.org/kde-zeitgeist/libqzeitgeist/" FALSE ""
+        "RECOMMENDED: Zeitgeist and QZeitgeist is needed for resource tracking")
 if (QZeitgeist_FOUND)
     set(HAVE_QZEITGEIST 1)
     include_directories(${QZEITGEIST_INCLUDE_DIR})
--- kde-runtime-4.7.2/activitymanager/ZeitgeistEventBackend.cpp~	2011-10-29 15:47:40.960836591 +0000
+++ kde-runtime-4.7.2/activitymanager/ZeitgeistEventBackend.cpp	2011-10-29 15:48:57.034930073 +0000
@@ -21,31 +21,31 @@
 
 #ifndef HAVE_QZEITGEIST
 #ifdef __GNUC__
-    #warning "No QtZeitgeist, disabling desktop events processing"
+    #warning "No QZeitgeist, disabling desktop events processing"
 #endif
 
 #else // HAVE_QZEITGEIST
 
 #include "ZeitgeistEventBackend.h"
 
-#include <QtZeitgeist/QtZeitgeist>
-#include <QtZeitgeist/Interpretation>
-#include <QtZeitgeist/Manifestation>
+#include <QZeitgeist/QZeitgeist>
+#include <QZeitgeist/Interpretation>
+#include <QZeitgeist/Manifestation>
 
 static QString eventInterpretation(Event::Type type)
 {
     switch (type) {
         case Event::Accessed:
-            return QtZeitgeist::Interpretation::Event::ZGAccessEvent;
+            return QZeitgeist::Interpretation::Event::ZGAccessEvent;
 
         case Event::Opened:
-            return QtZeitgeist::Interpretation::Event::ZGAccessEvent;
+            return QZeitgeist::Interpretation::Event::ZGAccessEvent;
 
         case Event::Modified:
-            return QtZeitgeist::Interpretation::Event::ZGModifyEvent;
+            return QZeitgeist::Interpretation::Event::ZGModifyEvent;
 
         case Event::Closed:
-            return QtZeitgeist::Interpretation::Event::ZGLeaveEvent;
+            return QZeitgeist::Interpretation::Event::ZGLeaveEvent;
 
     }
 
@@ -57,23 +57,23 @@
 {
     switch (reason) {
         case Event::User:
-            return QtZeitgeist::Manifestation::Event::ZGUserActivity;
+            return QZeitgeist::Manifestation::Event::ZGUserActivity;
 
         case Event::Heuristic:
-            return QtZeitgeist::Manifestation::Event::ZGHeuristicActivity;
+            return QZeitgeist::Manifestation::Event::ZGHeuristicActivity;
 
         case Event::Scheduled:
-            return QtZeitgeist::Manifestation::Event::ZGScheduledActivity;
+            return QZeitgeist::Manifestation::Event::ZGScheduledActivity;
 
         case Event::System:
-            return QtZeitgeist::Manifestation::Event::ZGSystemNotification;
+            return QZeitgeist::Manifestation::Event::ZGSystemNotification;
 
         case Event::World:
-            return QtZeitgeist::Manifestation::Event::ZGWorldActivity;
+            return QZeitgeist::Manifestation::Event::ZGWorldActivity;
     }
 
     // shut up GCC
-    return QtZeitgeist::Manifestation::Event::ZGUserActivity;
+    return QZeitgeist::Manifestation::Event::ZGUserActivity;
 }
 
 static QString applicationUri(const QString & application)
@@ -84,7 +84,6 @@
 
 ZeitgeistEventBackend::ZeitgeistEventBackend()
 {
-    QtZeitgeist::init();
 }
 
 void ZeitgeistEventBackend::addEvents(const EventList & events)
