diff --git i/src/truetype/ttobjs.c w/src/truetype/ttobjs.c
index 46f1332..af50a1d 100644
--- i/src/truetype/ttobjs.c
+++ w/src/truetype/ttobjs.c
@@ -1264,10 +1264,11 @@
       return FT_THROW( Could_Not_Find_Context );
 
 #ifdef TT_CONFIG_OPTION_SUBPIXEL_HINTING
-    driver->interpreter_version = TT_INTERPRETER_VERSION_38;
-#else
-    driver->interpreter_version = TT_INTERPRETER_VERSION_35;
+    if ( !getenv( "FT2_NO_SUBPIXEL_HINTING" ) )
+      driver->interpreter_version = TT_INTERPRETER_VERSION_38;
+    else
 #endif
+      driver->interpreter_version = TT_INTERPRETER_VERSION_35;
 
 #else /* !TT_USE_BYTECODE_INTERPRETER */
 
