diff -ruN seahorse-plugins-2.28.0/plugins/nautilus/seahorse-tool.c seahorse-plugins-2.28.0.patched/plugins/nautilus/seahorse-tool.c
--- seahorse-plugins-2.28.0/plugins/nautilus/seahorse-tool.c	2009-09-03 13:20:15.000000000 +0200
+++ seahorse-plugins-2.28.0.patched/plugins/nautilus/seahorse-tool.c	2009-10-13 09:03:34.000000000 +0200
@@ -153,6 +153,7 @@
 prompt_recipients (gpgme_key_t *signkey)
 {
     gpgme_error_t gerr = GPG_OK;
+    gpgme_protocol_t proto = GPGME_PROTOCOL_OpenPGP;
     CryptUIKeyset *keyset;
     gpgme_ctx_t ctx;
     gpgme_key_t key;
@@ -171,6 +172,9 @@
         
         if (recips) {
             
+            gerr = gpgme_engine_check_version (proto);
+            g_return_val_if_fail (GPG_IS_OK (gerr), NULL);
+
             gerr = gpgme_new (&ctx);
             g_return_val_if_fail (GPG_IS_OK (gerr), NULL);
         
