diff -up scim-pinyin-0.5.91/src/scim_phrase.cpp.gcc43 scim-pinyin-0.5.91/src/scim_phrase.cpp
--- scim-pinyin-0.5.91/src/scim_phrase.cpp.gcc43	2008-02-25 16:27:05.000000000 +0800
+++ scim-pinyin-0.5.91/src/scim_phrase.cpp	2008-02-25 16:26:41.000000000 +0800
@@ -28,6 +28,7 @@
 #define SCIM_PHRASE_MAX_RELATION 1000
 
 #include <scim.h>
+#include <cstring>
 #include "scim_pinyin_private.h"
 #include "scim_phrase.h"
 
diff -up scim-pinyin-0.5.91/src/scim_special_table.cpp.gcc43 scim-pinyin-0.5.91/src/scim_special_table.cpp
--- scim-pinyin-0.5.91/src/scim_special_table.cpp.gcc43	2008-02-25 16:27:55.000000000 +0800
+++ scim-pinyin-0.5.91/src/scim_special_table.cpp	2008-02-25 16:28:08.000000000 +0800
@@ -22,6 +22,7 @@
 
 #include <time.h>
 #include <scim.h>
+#include <cstring>
 #include "scim_pinyin_private.h"
 #include "scim_special_table.h"
 
diff -up scim-pinyin-0.5.91/src/scim_pinyin.cpp.gcc43 scim-pinyin-0.5.91/src/scim_pinyin.cpp
--- scim-pinyin-0.5.91/src/scim_pinyin.cpp.gcc43	2008-02-25 16:26:10.000000000 +0800
+++ scim-pinyin-0.5.91/src/scim_pinyin.cpp	2008-02-25 16:26:26.000000000 +0800
@@ -29,6 +29,7 @@
 #define Uses_SCIM_LOOKUP_TABLE
 
 #include <scim.h>
+#include <cstring>
 #include "scim_pinyin.h"
 
 // Internal functions
diff -up scim-pinyin-0.5.91/src/scim_pinyin_phrase.cpp.gcc43 scim-pinyin-0.5.91/src/scim_pinyin_phrase.cpp
--- scim-pinyin-0.5.91/src/scim_pinyin_phrase.cpp.gcc43	2008-02-25 16:27:15.000000000 +0800
+++ scim-pinyin-0.5.91/src/scim_pinyin_phrase.cpp	2008-02-25 16:27:28.000000000 +0800
@@ -27,6 +27,7 @@
 #define Uses_SCIM_LOOKUP_TABLE
 
 #include <scim.h>
+#include <cstring>
 #include "scim_pinyin_private.h"
 #include "scim_phrase.h"
 #include "scim_pinyin.h"
