From 9d6e8052b188c2b1cb310ef05842da74a730a39f Mon Sep 17 00:00:00 2001
From: Marek Habersack <grendel@twistedcode.net>
Date: Wed, 15 Sep 2010 21:21:52 +0000
Subject: PixbufLoader from gtk-sharp/master does not have a Close () overload which takes a boolean parameter

https://bugzilla.gnome.org/show_bug.cgi?id=629224
---
diff --git a/src/Clients/MainApp/FSpot.Loaders/GdkImageLoader.cs b/src/Clients/MainApp/FSpot.Loaders/GdkImageLoader.cs
index 2eea50e..9c9c07a 100644
--- a/src/Clients/MainApp/FSpot.Loaders/GdkImageLoader.cs
+++ b/src/Clients/MainApp/FSpot.Loaders/GdkImageLoader.cs
@@ -132,7 +132,7 @@ namespace FSpot.Loaders {
 		public new bool Close ()
 		{
 			lock (sync_handle) {
-				return base.Close (true);
+				return base.Close ();
 			}
 		}
 #endregion
--
cgit v0.8.3.1
