diff -up xdg-user-dirs-gtk-0.10/user-dirs-update-gtk.desktop.in.mate xdg-user-dirs-gtk-0.10/user-dirs-update-gtk.desktop.in
--- xdg-user-dirs-gtk-0.10/user-dirs-update-gtk.desktop.in.mate	2014-01-27 19:09:55.954616086 -0500
+++ xdg-user-dirs-gtk-0.10/user-dirs-update-gtk.desktop.in	2014-01-27 19:10:02.965602005 -0500
@@ -4,7 +4,7 @@ Exec=xdg-user-dirs-gtk-update
 _Name=User folders update
 _Comment=Update common folders names to match current locale
 Terminal=false
-OnlyShowIn=GNOME;LXDE;Unity;
+OnlyShowIn=GNOME;LXDE;Unity;MATE;
 Type=Application
 StartupNotify=false
 X-KDE-autostart-after=panel
