diff -ruN cups-1.4.7.orig//config-scripts/cups-ssl.m4 cups-1.4.7/config-scripts/cups-ssl.m4
--- cups-1.4.7.orig//config-scripts/cups-ssl.m4	2011-01-11 08:05:58.000000000 +0100
+++ cups-1.4.7/config-scripts/cups-ssl.m4	2011-08-02 10:44:26.011047900 +0200
@@ -74,7 +74,6 @@
     dnl Then look for GNU TLS...
     if test $have_ssl = 0 -a "x$enable_gnutls" != "xno" -a "x$PKGCONFIG" != x; then
     	AC_PATH_PROG(LIBGNUTLSCONFIG,libgnutls-config)
-    	AC_PATH_PROG(LIBGCRYPTCONFIG,libgcrypt-config)
 	if $PKGCONFIG --exists gnutls; then
 	    if test "x$have_pthread" = xyes; then
 		AC_MSG_WARN([The current version of GNU TLS cannot be made thread-safe.])
@@ -96,16 +95,6 @@
 	        AC_DEFINE(HAVE_GNUTLS)
 	    fi
 	fi
-
-	if test $have_ssl = 1; then
-            if $PKGCONFIG --exists gcrypt; then
-	        SSLLIBS="$SSLLIBS `$PKGCONFIG --libs gcrypt`"
-	        SSLFLAGS="$SSLFLAGS `$PKGCONFIG --cflags gcrypt`"
-	    elif test "x$LIBGCRYPTCONFIG" != x; then
-	        SSLLIBS="$SSLLIBS `$LIBGCRYPTCONFIG --libs`"
-	        SSLFLAGS="$SSLFLAGS `$LIBGCRYPTCONFIG --cflags`"
-	    fi
-	fi
     fi
 
     dnl Check for the OpenSSL library last...
diff -ruN cups-1.4.7.orig//cups/http-private.h cups-1.4.7/cups/http-private.h
--- cups-1.4.7.orig//cups/http-private.h	2011-01-22 01:07:22.000000000 +0100
+++ cups-1.4.7/cups/http-private.h	2011-08-02 10:42:43.341604107 +0200
@@ -98,7 +98,6 @@
  * The GNU TLS library is more of a "bare metal" SSL/TLS library...
  */
 #    include <gnutls/gnutls.h>
-#    include <gcrypt.h>
 
 typedef struct
 {
