diff -upr OpenGTL-0.9.17.orig/CMakeLists.txt OpenGTL-0.9.17/CMakeLists.txt
--- OpenGTL-0.9.17.orig/CMakeLists.txt	2012-06-04 17:51:43.000000000 +0300
+++ OpenGTL-0.9.17/CMakeLists.txt	2012-12-25 03:00:30.000000000 +0200
@@ -47,8 +47,8 @@ configure_file(config-endian.h.cmake ${C
 
 find_package(LLVM REQUIRED)
 
-if( NOT MSVC AND NOT (LLVM_VERSION STREQUAL "3.1" OR LLVM_VERSION STREQUAL "3.0" ) ) # There is no way with MSVC to know the llvm version
-  message(FATAL_ERROR "LLVM 3.0 or 3.1 is required.")
+if( NOT MSVC AND NOT LLVM_VERSION STREQUAL "3.2" ) # There is no way with MSVC to know the llvm version
+  message(FATAL_ERROR "LLVM 3.2 is required.")
 endif()
 
 if(MSVC)
diff -upr OpenGTL-0.9.17.orig/OpenCTL/OpenCTL/Program.cpp OpenGTL-0.9.17/OpenCTL/OpenCTL/Program.cpp
--- OpenGTL-0.9.17.orig/OpenCTL/OpenCTL/Program.cpp	2012-06-04 17:51:43.000000000 +0300
+++ OpenGTL-0.9.17/OpenCTL/OpenCTL/Program.cpp	2012-12-25 02:59:17.000000000 +0200
@@ -29,7 +29,7 @@
 #include <llvm/PassManager.h>
 #include <llvm/Analysis/LoopPass.h>
 #include <llvm/Analysis/Verifier.h>
-#include <llvm/Target/TargetData.h>
+#include <llvm/DataLayout.h>
 #include <llvm/GlobalVariable.h>
 
 // GTLCore
diff -upr OpenGTL-0.9.17.orig/OpenGTL/GTLCore/ModuleData_p.cpp OpenGTL-0.9.17/OpenGTL/GTLCore/ModuleData_p.cpp
--- OpenGTL-0.9.17.orig/OpenGTL/GTLCore/ModuleData_p.cpp	2012-06-04 17:51:43.000000000 +0300
+++ OpenGTL-0.9.17/OpenGTL/GTLCore/ModuleData_p.cpp	2012-12-25 02:59:34.000000000 +0200
@@ -23,7 +23,7 @@
 
 #include <llvm/Module.h>
 #include <llvm/PassManager.h>
-#include <llvm/Target/TargetData.h>
+#include <llvm/DataLayout.h>
 #include <llvm/Target/TargetMachine.h>
 #include <llvm/Analysis/Verifier.h>
 #include <llvm/Transforms/Utils/Cloning.h>
