diff --git a/COPYING b/COPYING
index 7f33cbf..05bcb74 100644
--- a/COPYING
+++ b/COPYING
@@ -1,12 +1,91 @@
-This is a stub file.  This package has not yet had its complete licensing
-information compiled.  Please see the individual source files for details on
-your rights to use and modify this software.
+Copyright 2007 George Sapountzis
 
-Please submit updated COPYING files to the Xorg bugzilla:
+Permission is hereby granted, free of charge, to any person obtaining a
+copy of this software and associated documentation files (the "Software"),
+to deal in the Software without restriction, including without limitation
+the rights to use, copy, modify, merge, publish, distribute, sublicense,
+and/or sell copies of the Software, and to permit persons to whom the
+Software is furnished to do so, subject to the following conditions:
 
-https://bugs.freedesktop.org/enter_bug.cgi?product=xorg
+The above copyright notice and this permission notice (including the next
+paragraph) shall be included in all copies or substantial portions of the
+Software.
 
-All licensing questions regarding this software should be directed at the
-Xorg mailing list:
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL
+THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+
+Copyright 1999, 2000 ATI Technologies Inc., Markham, Ontario,
+                     Precision Insight, Inc., Cedar Park, Texas, and
+                     VA Linux Systems Inc., Fremont, California.
+
+All Rights Reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining
+a copy of this software and associated documentation files (the
+"Software"), to deal in the Software without restriction, including
+without limitation on the rights to use, copy, modify, merge,
+publish, distribute, sublicense, and/or sell copies of the Software,
+and to permit persons to whom the Software is furnished to do so,
+subject to the following conditions:
+
+The above copyright notice and this permission notice (including the
+next paragraph) shall be included in all copies or substantial
+portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+NON-INFRINGEMENT.  IN NO EVENT SHALL ATI, PRECISION INSIGHT, VA LINUX
+SYSTEMS AND/OR THEIR SUPPLIERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+OTHER DEALINGS IN THE SOFTWARE.
+
+Copyright 1999, 2000 Precision Insight, Inc., Cedar Park, Texas.
+Copyright 2002 Tungsten Graphics, Inc., Cedar Park, Texas.
+All Rights Reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a
+copy of this software and associated documentation files (the "Software"),
+to deal in the Software without restriction, including without limitation
+the rights to use, copy, modify, merge, publish, distribute, sublicense,
+and/or sell copies of the Software, and to permit persons to whom the
+Software is furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice (including the next
+paragraph) shall be included in all copies or substantial portions of the
+Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL
+PRECISION INSIGHT AND/OR ITS SUPPLIERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+DEALINGS IN THE SOFTWARE.
+
+Copyright 2000 through 2004 by Marc Aurele La France (TSI @ UQV), tsi@xfree86.org
+
+Permission to use, copy, modify, distribute, and sell this software and its
+documentation for any purpose is hereby granted without fee, provided that
+the above copyright notice appear in all copies and that both that copyright
+notice and this permission notice appear in supporting documentation, and
+that the name of Marc Aurele La France not be used in advertising or
+publicity pertaining to distribution of the software without specific,
+written prior permission.  Marc Aurele La France makes no representations
+about the suitability of this software for any purpose.  It is provided
+"as-is" without express or implied warranty.
+
+MARC AURELE LA FRANCE DISCLAIMS ALL WARRANTIES WITH REGARD TO THIS SOFTWARE,
+INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS.  IN NO
+EVENT SHALL MARC AURELE LA FRANCE BE LIABLE FOR ANY SPECIAL, INDIRECT OR
+CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE,
+DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER
+TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
+PERFORMANCE OF THIS SOFTWARE.
 
-http://lists.freedesktop.org/mailman/listinfo/xorg
diff --git a/Makefile.am b/Makefile.am
index 1f4c3c4..4c278ba 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -18,21 +18,15 @@
 #  IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
 #  CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 
-AUTOMAKE_OPTIONS = foreign
 SUBDIRS = src man
+MAINTAINERCLEANFILES = ChangeLog INSTALL
 
-if BUILD_LINUXDOC
-README.r128: README.r128.sgml
-	$(MAKE_TEXT) README.r128.sgml && mv README.r128.txt README.r128
+.PHONY: ChangeLog INSTALL
 
-endif
-
-EXTRA_DIST = README.r128 README.r128.sgml ChangeLog
-CLEANFILES = ChangeLog
-
-.PHONY: ChangeLog
+INSTALL:
+	$(INSTALL_CMD)
 
 ChangeLog:
 	$(CHANGELOG_CMD)
 
-dist-hook: ChangeLog
+dist-hook: ChangeLog INSTALL
diff --git a/README b/README
index cb61870..5c6f59d 100644
--- a/README
+++ b/README
@@ -1,4 +1,161 @@
-xf86-video-r128 - ATI Rage 128 video driver for the Xorg X server
+  Information for ATI Rage 128 Users
+  Precision Insight, Inc., SuSE GmbH
+  13 June 2000
+  ____________________________________________________________
+
+  Table of Contents
+
+
+  1. Supported Hardware
+  2. Features
+  3. Technical Notes
+  4. Reported Working Video Cards
+  5. Configuration
+  6. Driver Options
+  7. Known Limitations
+  8. Authors
+
+
+  ______________________________________________________________________
+
+  1.  Supported Hardware
+
+
+  o  ATI Rage 128 based cards
+
+
+
+  2.  Features
+
+
+  o  Full support (including hardware accelerated 2D drawing) for 8, 15,
+     16, 24 bit pixel depths.
+
+  o  Hardware cursor support to reduce sprite flicker.
+
+  o  Support for high resolution video modes up to 1800x1440 @ 70Hz.
+
+  o  Support for doublescan video modes (e.g., 320x200 and 320x240).
+
+  o  Support for gamma correction at all pixel depths.
+
+  o  Fully programmable clock supported.
+
+  o  Robust text mode restore for VT switching.
+
+
+
+  3.  Technical Notes
+
+
+  o  None
+
+
+
+  4.  Reported Working Video Cards
+
+
+  o  Rage Fury AGP 32MB
+
+  o  XPERT 128 AGP 16MB
+
+  o  XPERT 99 AGP 8MB
+
+
+
+  5.  Configuration
+
+  The driver auto-detects all device information necessary to initialize
+  the card. The only lines you need in the "Device" section of your
+  xorg.conf file are:
+
+         Section "Device"
+             Identifier "Rage 128"
+             Driver     "r128"
+         EndSection
+
+
+  or let xorgconfig do this for you.
+
+  However, if you have problems with auto-detection, you can specify:
+
+  o  VideoRam - in kilobytes
+
+  o  MemBase - physical address of the linear framebuffer
+
+  o  IOBase - physical address of the memory mapped IO registers
+
+  o  ChipID - PCI DEVICE ID
+
+
+
+  6.  Driver Options
+
+
+  o  "hw_cursor" - request hardware cursor (default)
+
+  o  "sw_cursor" - software cursor only
+
+  o  "no_accel" - software rendering only
+
+  o  "dac_8_bit" - use color weight 888 in 8 bpp mode (default)
+
+  o  "dac_6_bit" - use color weight 666 in 8 bpp mode (VGA emulation)
+
+
+
+  7.  Known Limitations
+
+
+  o  None
+
+
+
+  8.  Authors
+
+  The X11R7.5 driver was originally part of XFree86 4.4 rc2.
+
+  The XFree86 4 driver was ported from XFree86 3.3.x and enhanced by:
+
+  o  Rickard E. (Rik) Faith  <mailto:faith@precisioninsight.com>
+
+  o  Kevin E. Martin  <mailto:kevin@precisioninsight.com>
+
+  The XFree86 4 driver was funded by ATI and was donated to The XFree86
+  Project by:
+
+      Precision Insight, Inc.
+      Cedar Park, TX
+      USA
+
+
+  The XFree86 3.3.x driver used for the port was written by:
+
+  o  Rickard E. (Rik) Faith  <mailto:faith@precisioninsight.com>
+
+  o  Kevin E. Martin  <mailto:kevin@precisioninsight.com>
+
+     The XFree86 3.3.x driver was funded by ATI and was donated to The
+     XFree86 Project by Precision Insight, Inc. It was based in part on
+     an earlier driver that was written by:
+
+  o  Alan Hourihane  <mailto:alanh@fairlite.demon.co.uk>
+
+  o  Dirk Hohndel  <mailto:hohndel@suse.de>
+
+  This early driver was funded and donated to The XFree86 Project by:
+
+      SuSE GmbH
+      Schanzaekerstr. 10
+      90443 Nuernberg
+      Germany
+
+
+
+  http://www.precisioninsight.com
+
+  http://www.suse.com
+
 
 Please submit bugs & patches to the Xorg bugzilla:
 
diff --git a/README.r128 b/README.r128
deleted file mode 100644
index dcc2715..0000000
--- a/README.r128
+++ /dev/null
@@ -1,160 +0,0 @@
-  Information for ATI Rage 128 Users
-  Precision Insight, Inc., SuSE GmbH
-  13 June 2000
-  ____________________________________________________________
-
-  Table of Contents
-
-
-  1. Supported Hardware
-  2. Features
-  3. Technical Notes
-  4. Reported Working Video Cards
-  5. Configuration
-  6. Driver Options
-  7. Known Limitations
-  8. Authors
-
-
-  ______________________________________________________________________
-
-  [1m1.  Supported Hardware[0m
-
-
-  +o  ATI Rage 128 based cards
-
-
-
-  [1m2.  Features[0m
-
-
-  +o  Full support (including hardware accelerated 2D drawing) for 8, 15,
-     16, 24 bit pixel depths.
-
-  +o  Hardware cursor support to reduce sprite flicker.
-
-  +o  Support for high resolution video modes up to 1800x1440 @ 70Hz.
-
-  +o  Support for doublescan video modes (e.g., 320x200 and 320x240).
-
-  +o  Support for gamma correction at all pixel depths.
-
-  +o  Fully programmable clock supported.
-
-  +o  Robust text mode restore for VT switching.
-
-
-
-  [1m3.  Technical Notes[0m
-
-
-  +o  None
-
-
-
-  [1m4.  Reported Working Video Cards[0m
-
-
-  +o  Rage Fury AGP 32MB
-
-  +o  XPERT 128 AGP 16MB
-
-  +o  XPERT 99 AGP 8MB
-
-
-
-  [1m5.  Configuration[0m
-
-  The driver auto-detects all device information necessary to initialize
-  the card.  The only lines you need in the "Device" section of your
-  xorg.conf file are:
-
-         Section "Device"
-             Identifier "Rage 128"
-             Driver     "r128"
-         EndSection
-
-
-  or let xorgconfig do this for you.
-
-  However, if you have problems with auto-detection, you can specify:
-
-  +o  VideoRam - in kilobytes
-
-  +o  MemBase  - physical address of the linear framebuffer
-
-  +o  IOBase   - physical address of the memory mapped IO registers
-
-  +o  ChipID   - PCI DEVICE ID
-
-
-
-  [1m6.  Driver Options[0m
-
-
-  +o  "hw_cursor" - request hardware cursor (default)
-
-  +o  "sw_cursor" - software cursor only
-
-  +o  "no_accel"  - software rendering only
-
-  +o  "dac_8_bit" - use color weight 888 in 8 bpp mode (default)
-
-  +o  "dac_6_bit" - use color weight 666 in 8 bpp mode (VGA emulation)
-
-
-
-  [1m7.  Known Limitations[0m
-
-
-  +o  None
-
-
-
-  [1m8.  Authors[0m
-
-  The X11R6.8 driver was originally part of XFree86 4.4 rc2.
-
-  The XFree86 4 driver was ported from XFree86 3.3.x and enhanced by:
-
-  +o  Rickard E. (Rik) Faith  <faith@precisioninsight.com>
-
-  +o  Kevin E. Martin  <kevin@precisioninsight.com>
-
-  The XFree86 4 driver was funded by ATI and was donated to The XFree86
-  Project by:
-
-      Precision Insight, Inc.
-      Cedar Park, TX
-      USA
-
-
-  The XFree86 3.3.x driver used for the port was written by:
-
-  +o  Rickard E. (Rik) Faith  <faith@precisioninsight.com>
-
-  +o  Kevin E. Martin  <kevin@precisioninsight.com>
-
-     The XFree86 3.3.x driver was funded by ATI and was donated to The
-     XFree86 Project by Precision Insight, Inc.  It was based in part on
-     an earlier driver that was written by:
-
-  +o  Alan Hourihane  <alanh@fairlite.demon.co.uk>
-
-  +o  Dirk Hohndel  <hohndel@suse.de>
-
-  This early driver was funded and donated to The XFree86 Project by:
-
-      SuSE GmbH
-      Schanzaekerstr. 10
-      90443 Nuernberg
-      Germany
-
-
-
-  http://www.precisioninsight.com
-
-  http://www.suse.com
-
-
-
diff --git a/README.r128.sgml b/README.r128.sgml
deleted file mode 100644
index 8d7f448..0000000
--- a/README.r128.sgml
+++ /dev/null
@@ -1,138 +0,0 @@
-<!DOCTYPE linuxdoc PUBLIC "-//Xorg//DTD linuxdoc//EN"[
-<!ENTITY % defs SYSTEM "defs.ent"> %defs;
-]>
-
-<article>
-<title>Information for ATI Rage 128 Users
-<author>Precision Insight, Inc., SuSE GmbH
-<date>13 June 2000
-
-<ident>
-</ident>
-
-<toc>
-
-<sect>Supported Hardware
-<p>
-<itemize>
-  <item>ATI Rage 128 based cards
-</itemize>
-
-
-<sect>Features
-<p>
-<itemize>
-  <item>Full support (including hardware accelerated 2D drawing) for 8, 15,
-        16, 24 bit pixel depths.
-  <item>Hardware cursor support to reduce sprite flicker.
-  <item>Support for high resolution video modes up to 1800x1440 @ 70Hz.
-  <item>Support for doublescan video modes (e.g., 320x200 and 320x240).
-  <item>Support for gamma correction at all pixel depths.
-  <item>Fully programmable clock supported.
-  <item>Robust text mode restore for VT switching.
-</itemize>
-
-
-<sect>Technical Notes
-<p>
-<itemize>
-  <item>None
-</itemize>
-
-
-<sect>Reported Working Video Cards
-<p>
-<itemize>
-  <item>Rage Fury AGP 32MB
-  <item>XPERT 128 AGP 16MB
-  <item>XPERT 99 AGP 8MB
-</itemize>
-
-
-<sect>Configuration
-<p>
-The driver auto-detects all device information necessary to
-initialize the card.  The only lines you need in the "Device"
-section of your xorg.conf file are:
-<verb>
-       Section "Device"
-           Identifier "Rage 128"
-           Driver     "r128"
-       EndSection
-</verb>
-or let <tt>xorgconfig</tt> do this for you.
-
-However, if you have problems with auto-detection, you can specify:
-<itemize>
-  <item>VideoRam - in kilobytes
-  <item>MemBase  - physical address of the linear framebuffer
-  <item>IOBase   - physical address of the memory mapped IO registers
-  <item>ChipID   - PCI DEVICE ID
-</itemize>
-
-
-<sect>Driver Options
-<p>
-<itemize>
-  <item>"hw_cursor" - request hardware cursor (default)
-  <item>"sw_cursor" - software cursor only
-  <item>"no_accel"  - software rendering only
-  <item>"dac_8_bit" - use color weight 888 in 8 bpp mode (default)
-  <item>"dac_6_bit" - use color weight 666 in 8 bpp mode (VGA emulation)
-</itemize>
-
-
-<sect>Known Limitations
-<p>
-<itemize>
-  <item>None
-</itemize>
-
-
-<sect>Authors
-<p>
-The X11R&relvers; driver was originally part of XFree86 4.4 rc2.
-
-The XFree86 4 driver was ported from XFree86 3.3.x and enhanced by:
-<itemize>
-   <item>Rickard E. (Rik) Faith <email>faith@precisioninsight.com</email>
-   <item>Kevin E. Martin <email>kevin@precisioninsight.com</email>
-</itemize>
-<p>
-The XFree86 4 driver was funded by ATI and was donated to The XFree86
-Project by:
-<verb>
-    Precision Insight, Inc.
-    Cedar Park, TX
-    USA
-</verb>
-<p>
-The XFree86 3.3.x driver used for the port was written by:
-<itemize>
-   <item>Rickard E. (Rik) Faith <email>faith@precisioninsight.com</email>
-   <item>Kevin E. Martin <email>kevin@precisioninsight.com</email>
-</itemize>
-The XFree86 3.3.x driver was funded by ATI and was donated to The XFree86
-Project by Precision Insight, Inc.  It was based in part on an earlier
-driver that was written by:
-<itemize>
-   <item>Alan Hourihane <email>alanh@fairlite.demon.co.uk</email>
-   <item>Dirk Hohndel <email>hohndel@suse.de</email>
-</itemize>
-<p>This early driver was funded and donated to The XFree86 Project by:
-<verb>
-    SuSE GmbH
-    Schanzaekerstr. 10
-    90443 Nuernberg
-    Germany
-</verb>
-
-<p>
-<htmlurl name="http://www.precisioninsight.com"
-          url="http://www.precisioninsight.com">
-<p>
-<htmlurl name="http://www.suse.com"
-          url="http://www.suse.com">
-
-
-</article>
diff --git a/configure.ac b/configure.ac
index 4e0f463..d5a0821 100644
--- a/configure.ac
+++ b/configure.ac
@@ -20,28 +20,30 @@
 #
 # Process this file with autoconf to produce a configure script
 
-AC_PREREQ(2.57)
+# Initialize Autoconf
+AC_PREREQ([2.60])
 AC_INIT([xf86-video-r128],
-        6.8.1,
+        [6.8.1],
         [https://bugs.freedesktop.org/enter_bug.cgi?product=xorg],
-        xf86-video-r128)
-
+        [xf86-video-r128])
 AC_CONFIG_SRCDIR([Makefile.am])
-AM_CONFIG_HEADER([config.h])
+AC_CONFIG_HEADERS([config.h])
 AC_CONFIG_AUX_DIR(.)
 
-AM_INIT_AUTOMAKE([dist-bzip2])
+# Initialize Automake
+AM_INIT_AUTOMAKE([foreign dist-bzip2])
 
-# Require xorg-macros: XORG_CWARNFLAGS, XORG_CHANGELOG
-m4_ifndef([XORG_MACROS_VERSION], [AC_FATAL([must install xorg-macros 1.2 or later before running autoconf/autogen])])
-XORG_MACROS_VERSION(1.2)
+# Require X.Org macros 1.8 or later for MAN_SUBSTS set by XORG_MANPAGE_SECTIONS
+m4_ifndef([XORG_MACROS_VERSION],
+          [m4_fatal([must install xorg-macros 1.8 or later before running autoconf/autogen])])
+XORG_MACROS_VERSION(1.8)
+XORG_DEFAULT_OPTIONS
 
 AM_MAINTAINER_MODE
 
-# Checks for programs.
+# Initialize libtool
 AC_DISABLE_STATIC
 AC_PROG_LIBTOOL
-AC_PROG_CC
 
 if test "x$GCC" = "xyes"; then
 	CPPFLAGS="$CPPFLAGS -Wall"
@@ -49,38 +51,35 @@ fi
 
 AH_TOP([#include "xorg-server.h"])
 
+# Define a configure option for an alternate module directory
 AC_ARG_WITH(xorg-module-dir,
-            AC_HELP_STRING([--with-xorg-module-dir=DIR],
+            AS_HELP_STRING([--with-xorg-module-dir=DIR],
                            [Default xorg module directory [[default=$libdir/xorg/modules]]]),
             [moduledir="$withval"],
             [moduledir="$libdir/xorg/modules"])
 
-AC_ARG_ENABLE(dri, AC_HELP_STRING([--disable-dri],
+AC_ARG_ENABLE(dri, AS_HELP_STRING([--disable-dri],
                                   [Disable DRI support [[default=auto]]]),
               [DRI="$enableval"],
               [DRI=auto])
 
-# Checks for extensions
-XORG_DRIVER_CHECK_EXT(XINERAMA, xineramaproto)
+# Store the list of server defined optional extensions in REQUIRED_MODULES
 XORG_DRIVER_CHECK_EXT(RANDR, randrproto)
 XORG_DRIVER_CHECK_EXT(RENDER, renderproto)
 XORG_DRIVER_CHECK_EXT(XV, videoproto)
 XORG_DRIVER_CHECK_EXT(XF86MISC, xf86miscproto)
 XORG_DRIVER_CHECK_EXT(DPMSExtension, xextproto)
 
-# Checks for pkg-config packages
+# Obtain compiler/linker options for the driver dependencies
 PKG_CHECK_MODULES(XORG, [xorg-server >= 1.2 xproto fontsproto $REQUIRED_MODULES])
 PKG_CHECK_MODULES(XEXT, [xextproto >= 7.0.99.1],
                   HAVE_XEXTPROTO_71="yes"; AC_DEFINE(HAVE_XEXTPROTO_71, 1, [xextproto 7.1 available]),
                   HAVE_XEXTPROTO_71="no")
 AM_CONDITIONAL(HAVE_XEXTPROTO_71, [ test "$HAVE_XEXTPROTO_71" = "yes" ])
-sdkdir=$(pkg-config --variable=sdkdir xorg-server)
+sdkdir=`$PKG_CONFIG --variable=sdkdir xorg-server`
 
 # Checks for libraries.
 
-# Checks for header files.
-AC_HEADER_STDC
-
 if test "$DRI" != no; then
 	AC_CHECK_FILE([${sdkdir}/dri.h],
                       [have_dri_h="yes"], [have_dri_h="no"])
@@ -126,26 +125,20 @@ if test "x$XSERVER_LIBPCIACCESS" = xyes; then
 fi
 AM_CONDITIONAL(XSERVER_LIBPCIACCESS, test "x$XSERVER_LIBPCIACCESS" = xyes)
 
-AC_SUBST([XORG_CFLAGS])
-AC_SUBST([DRI_CFLAGS])
 AC_SUBST([moduledir])
 
 DRIVER_NAME=r128
 AC_SUBST([DRIVER_NAME])
 
-XORG_MANPAGE_SECTIONS
-XORG_RELEASE_VERSION
-XORG_CHECK_LINUXDOC
-XORG_CHANGELOG
-
 AC_MSG_NOTICE(
 [Please change the Driver line in xorg.conf from "ati" to "r128" or install]
 [the ati wrapper as well:]
 [    git://anongit.freedesktop.org/git/xorg/driver/xf86-video-ati]
 )
 
-AC_OUTPUT([
-	Makefile
-	src/Makefile
-	man/Makefile
+AC_CONFIG_FILES([
+                Makefile
+                src/Makefile
+                man/Makefile
 ])
+AC_OUTPUT
diff --git a/man/Makefile.am b/man/Makefile.am
index 858ba62..4e22d61 100644
--- a/man/Makefile.am
+++ b/man/Makefile.am
@@ -1,27 +1,24 @@
 #
 # Copyright 2005 Sun Microsystems, Inc.  All rights reserved.
-# 
-# Permission to use, copy, modify, distribute, and sell this software and its
-# documentation for any purpose is hereby granted without fee, provided that
-# the above copyright notice appear in all copies and that both that
-# copyright notice and this permission notice appear in supporting
-# documentation.
-# 
-# The above copyright notice and this permission notice shall be included
-# in all copies or substantial portions of the Software.
-# 
-# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
-# OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
-# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
-# IN NO EVENT SHALL THE OPEN GROUP BE LIABLE FOR ANY CLAIM, DAMAGES OR
-# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
-# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
-# OTHER DEALINGS IN THE SOFTWARE.
-# 
-# Except as contained in this notice, the name of the copyright holders shall
-# not be used in advertising or otherwise to promote the sale, use or
-# other dealings in this Software without prior written authorization
-# from the copyright holders.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice (including the next
+# paragraph) shall be included in all copies or substantial portions of the
+# Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+# DEALINGS IN THE SOFTWARE.
 # 
 
 drivermandir = $(DRIVER_MAN_DIR)
@@ -34,25 +31,11 @@ EXTRA_DIST = r128.man
 
 CLEANFILES = $(driverman_DATA)
 
-SED = sed
 
-# Strings to replace in man pages
-XORGRELSTRING = @PACKAGE_STRING@
-  XORGMANNAME = X Version 11
+# String replacements in MAN_SUBSTS now come from xorg-macros.m4 via configure
 
-MAN_SUBSTS = \
-	-e 's|__vendorversion__|"$(XORGRELSTRING)" "$(XORGMANNAME)"|' \
-	-e 's|__xorgversion__|"$(XORGRELSTRING)" "$(XORGMANNAME)"|' \
-	-e 's|__xservername__|Xorg|g' \
-	-e 's|__xconfigfile__|xorg.conf|g' \
-	-e 's|__projectroot__|$(prefix)|g' \
-	-e 's|__appmansuffix__|$(APP_MAN_SUFFIX)|g' \
-	-e 's|__drivermansuffix__|$(DRIVER_MAN_SUFFIX)|g' \
-	-e 's|__adminmansuffix__|$(ADMIN_MAN_SUFFIX)|g' \
-	-e 's|__miscmansuffix__|$(MISC_MAN_SUFFIX)|g' \
-	-e 's|__filemansuffix__|$(FILE_MAN_SUFFIX)|g'
 
 SUFFIXES = .$(DRIVER_MAN_SUFFIX) .man
 
 .man.$(DRIVER_MAN_SUFFIX):
-	sed $(MAN_SUBSTS) < $< > $@
+	$(AM_V_GEN)$(SED) $(MAN_SUBSTS) < $< > $@
diff --git a/man/r128.man b/man/r128.man
index 5c7c4fe..4ba933d 100644
--- a/man/r128.man
+++ b/man/r128.man
@@ -1,4 +1,3 @@
-.\" $XFree86: xc/programs/Xserver/hw/xfree86/drivers/ati/r128.man,v 1.3 2001/06/01 02:10:05 dawes Exp $
 .\" shorthand for double quote that works everywhere.
 .ds q \N'34'
 .TH R128 __drivermansuffix__ __vendorversion__
diff --git a/src/r128_driver.c b/src/r128_driver.c
index d6448b6..9d08d77 100644
--- a/src/r128_driver.c
+++ b/src/r128_driver.c
@@ -938,14 +938,6 @@ static Bool R128PreInitConfig(ScrnInfoPtr pScrn)
 				/* BIOS */
     from              = X_PROBED;
     info->BIOSAddr    = info->PciInfo->biosBase & 0xfffe0000;
-    if (dev->BiosBase) {
-	xf86DrvMsg(pScrn->scrnIndex, X_INFO,
-		   "BIOS address override, using 0x%08lx instead of 0x%08lx\n",
-		   dev->BiosBase,
-		   info->BIOSAddr);
-	info->BIOSAddr = dev->BiosBase;
-	from           = X_CONFIG;
-    }
     if (info->BIOSAddr) {
 	xf86DrvMsg(pScrn->scrnIndex, from,
 		   "BIOS at 0x%08lx\n", info->BIOSAddr);
