--- branches/KDE/4.5/kdenetwork/kopete/protocols/oscar/icq/icqaccount.cpp	2010/11/16 15:01:13	1197797
+++ branches/KDE/4.5/kdenetwork/kopete/protocols/oscar/icq/icqaccount.cpp	2010/11/16 15:04:32	1197798
@@ -136,7 +136,7 @@
 	// Create actions
 	mEditInfoAction = new KAction( KIcon("user-properties"), i18n( "Edit User Info..." ), this );
 	QObject::connect( mEditInfoAction, SIGNAL(triggered(bool)), this, SLOT(slotUserInfo()) );
-	
+
 	mActionInvisible = new KToggleAction( i18n( "In&visible" ), this );
 	QObject::connect( mActionInvisible, SIGNAL(triggered(bool)), this, SLOT(slotToggleInvisible()) );
 
@@ -188,7 +188,7 @@
 	*/
 
 	KActionMenu *xtrazStatusMenu = new KActionMenu( i18n( "Set Xtraz Status" ), actionMenu );
-	
+
 	KAction* xtrazStatusSetAction = new KAction( i18n( "Set Status..." ), xtrazStatusMenu );
 	QObject::connect( xtrazStatusSetAction, SIGNAL(triggered(bool)), this, SLOT(setXtrazStatus()) );
 	xtrazStatusMenu->addAction( xtrazStatusSetAction );
@@ -236,7 +236,7 @@
 		myself()->setOnlineStatus( protocol()->statusManager()->connectingStatus() );
 		QString icqNumber = accountId();
 		kDebug(14153) << "Logging in as " << icqNumber;
-		QString server = configGroup()->readEntry( "Server", QString::fromLatin1( "login.oscar.aol.com" ) );
+		QString server = configGroup()->readEntry( "Server", QString::fromLatin1( "login.icq.com" ) );
 		uint port = configGroup()->readEntry( "Port", 5190 );
 
 		//set up the settings for the account
--- branches/KDE/4.5/kdenetwork/kopete/protocols/oscar/icq/ui/icqeditaccountwidget.cpp	2010/11/16 15:01:13	1197797
+++ branches/KDE/4.5/kdenetwork/kopete/protocols/oscar/icq/ui/icqeditaccountwidget.cpp	2010/11/16 15:04:32	1197798
@@ -65,7 +65,7 @@
 
 	mProtocol->fillComboFromTable( mAccountSettings->encodingCombo, mProtocol->encodings() );
 
-	//Setup the edtAccountId 
+	//Setup the edtAccountId
 	QRegExp rx("[0-9]{9}");
 	QValidator* validator = new QRegExpValidator( rx, this );
 	mAccountSettings->edtAccountId->setValidator(validator);
@@ -80,9 +80,9 @@
 		mAccountSettings->mPasswordWidget->load(&mAccount->password());
 		mAccountSettings->chkAutoLogin->setChecked(mAccount->excludeConnect());
 
-		QString serverEntry = mAccount->configGroup()->readEntry("Server", "login.oscar.aol.com");
+		QString serverEntry = mAccount->configGroup()->readEntry("Server", "login.icq.com");
 		int portEntry = mAccount->configGroup()->readEntry("Port", 5190);
-		mAccountSettings->optionOverrideServer->setChecked( serverEntry != "login.oscar.aol.com" || ( portEntry != 5190) );
+		mAccountSettings->optionOverrideServer->setChecked( serverEntry != "login.icq.com" || ( portEntry != 5190) );
 
 		mAccountSettings->edtServerAddress->setText( serverEntry );
 		mAccountSettings->edtServerPort->setValue( portEntry );
@@ -181,7 +181,7 @@
 			default:
 				encodingId=4;
 		}
-		
+
 		mProtocol->setComboFromTable( mAccountSettings->encodingCombo,
 		                              mProtocol->encodings(),
 		                              encodingId );
@@ -253,7 +253,7 @@
 	}
 	else
 	{
-		mAccount->setServerAddress("login.oscar.aol.com");
+		mAccount->setServerAddress("login.icq.com");
 		mAccount->setServerPort(5190);
 	}
 
@@ -286,17 +286,17 @@
 	{
 		if ( m_visibleEngine )
 			m_visibleEngine->storeChanges();
-		
+
 		if ( m_invisibleEngine )
 			m_invisibleEngine->storeChanges();
-		
+
 		if ( m_ignoreEngine )
 			m_ignoreEngine->storeChanges();
 
 		//Update Oscar settings
 		static_cast<ICQMyselfContact*>( mAccount->myself() )->fetchShortInfo();
 	}
-	
+
 	return mAccount;
 }
 
@@ -306,7 +306,7 @@
 	bool bOk;
 	QString userId = mAccountSettings->edtAccountId->text();
 	qulonglong uid = userId.toULongLong( &bOk );
-	
+
 	if( !bOk || uid == 0 || userId.isEmpty() )
 	{	KMessageBox::queuedMessageBox(this, KMessageBox::Sorry,
 	 	                              i18n("<qt>You must enter a valid ICQ No.</qt>"), i18n("ICQ"));
--- branches/KDE/4.5/kdenetwork/kopete/kopete/kconf_update/CMakeLists.txt	2010/11/19 19:32:07	1198790
+++ branches/KDE/4.5/kdenetwork/kopete/kopete/kconf_update/CMakeLists.txt	2010/11/19 19:33:39	1198791
@@ -4,9 +4,9 @@
 endif(NOT WIN32)
 
 
-install( FILES kopete-pluginloader.upd  kopete-nameTracking.upd  kopete-initialstatus.upd  kopete-gaim_to_pidgin_style.upd kopete-update_yahoo_server.upd  DESTINATION ${KCONF_UPDATE_INSTALL_DIR})
+install( FILES kopete-pluginloader.upd  kopete-nameTracking.upd  kopete-initialstatus.upd  kopete-gaim_to_pidgin_style.upd kopete-update_yahoo_server.upd kopete-update_icq_server.upd DESTINATION ${KCONF_UPDATE_INSTALL_DIR})
 
-install( PROGRAMS kopete-pluginloader.pl kopete-update_yahoo_server.pl
+install( PROGRAMS kopete-pluginloader.pl kopete-update_yahoo_server.pl kopete-update_icq_server.pl
 	kopete-account-0.10.pl kopete-initialstatus.pl kopete-gaim_to_pidgin_style.pl
   DESTINATION ${KCONF_UPDATE_INSTALL_DIR})
 
--- /dev/null	2010-11-23 14:10:08.536666668 +0100
+++ branches/KDE/4.5/kdenetwork/kopete/kopete/kconf_update/kopete-update_icq_server.pl	2010-11-23 15:36:38.626671763 +0100
@@ -0,0 +1,20 @@
+#!/usr/bin/perl
+
+# AOL sold ICQ; change the default server to login.icq.com
+
+my $inICQ = 0;
+foreach (<>) {
+    $inICQ = 1 if (/^\[Account_ICQProtocol_.*$/);
+    if ($inICQ) {
+        if (/^Server\=(.*)/) {
+            my $oldServer = $1;
+            if ($oldServer =~ m/\.aol\.com$/) {
+                print "Server=login.icq.com\n";
+                $inICQ = 0;
+                next;
+            }
+        }
+    }
+
+    print $_;
+}
--- /dev/null	2010-11-23 14:10:08.536666668 +0100
+++ branches/KDE/4.5/kdenetwork/kopete/kopete/kconf_update/kopete-update_icq_server.upd	2010-11-23 15:38:08.133338428 +0100
@@ -0,0 +1,4 @@
+Id=kopete-update-icq-server/r1
+File=kopeterc
+Options=overwrite
+Script=kopete-update_icq_server.pl,perl
