Index: subversion/libsvn_auth_kwallet/kwallet.cpp
===================================================================
--- subversion/libsvn_auth_kwallet/kwallet.cpp	(revision 1421011)
+++ subversion/libsvn_auth_kwallet/kwallet.cpp	(working copy)
@@ -253,7 +253,7 @@ kwallet_password_get(svn_boolean_t *done,
         }
     }
 
-  apr_pool_cleanup_register(pool, parameters, kwallet_terminate, NULL);
+  apr_pool_cleanup_register(pool, parameters, kwallet_terminate, apr_pool_cleanup_null);
 
   return SVN_NO_ERROR;
 }
@@ -327,7 +327,7 @@ kwallet_password_set(svn_boolean_t *done,
         }
     }
 
-  apr_pool_cleanup_register(pool, parameters, kwallet_terminate, NULL);
+  apr_pool_cleanup_register(pool, parameters, kwallet_terminate, apr_pool_cleanup_null);
 
   return SVN_NO_ERROR;
 }