--- kdepim-4.10.0/kleopatra/models/keylistmodel.cpp~	2013-01-31 09:40:34.270651208 +0000
+++ kdepim-4.10.0/kleopatra/models/keylistmodel.cpp	2013-01-31 09:41:26.570259714 +0000
@@ -60,9 +60,9 @@
 #ifdef _WIN32_WCE
 #define BOOST_ALL_NO_LIB
 #endif
-#include <boost/graph/adjacency_list.hpp>
 
 #ifndef Q_MOC_RUN // QTBUG-22829
+#include <boost/graph/adjacency_list.hpp>
 #include <boost/graph/topological_sort.hpp>
 #endif
 
