--- kdepim-4.9.0/kleopatra/CMakeLists.txt~	2012-07-30 08:06:15.031181647 +0000
+++ kdepim-4.9.0/kleopatra/CMakeLists.txt	2012-07-30 08:07:13.594082584 +0000
@@ -365,7 +365,7 @@
 endif(KDEPIM_MOBILE_UI)
 
 add_definitions ( -DKDE_DEFAULT_DEBUG_AREA=5151 )
-
+add_definitions ( -DBOOST_TT_HAS_OPERATOR_HPP_INCLUDED )
 
 kde4_add_app_icon(_kleopatra_mainwindow_SRCS "hi*-app-kleopatra.png")
 kde4_add_executable(kleopatra_bin ${_kleopatra_SRCS} ${_kleopatra_mainwindow_SRCS} ${_kleopatra_uiserver_SRCS} ${_kleopatra_libkdepim_SRCS} )
