Index: Makefile.am
===================================================================
--- korganizer/Makefile.am	(revision 593026)
+++ korganizer/Makefile.am	(working copy)
@@ -171,7 +171,7 @@ messages: rc.cpp
 	$(XGETTEXT) `find . -name "*.cpp" -o -name "*.h"` -o $(podir)/korganizer.pot
 	rm -f tips.cpp
 
-xdg_apps_DATA = korganizer.desktop
+xdg_apps_DATA = korganizer.desktop korganizer-import.desktop
 
 kde_kcfg_DATA = korganizer.kcfg
 
Index: korganizer.desktop
===================================================================
--- korganizer/korganizer.desktop	(revision 593026)
+++ korganizer/korganizer.desktop	(working copy)
@@ -1,6 +1,5 @@
 [Desktop Entry]
 Encoding=UTF-8
-MimeType=text/calendar;text/x-vcalendar;
 Comment=Calendar and Scheduling Program
 Comment[af]=Kalender en Skedulering Program
 Comment[ar]=برنامج الجدولة والتقويم
@@ -61,7 +60,7 @@ Comment[xh]=Ikhalenda no Dweliso lwenkqu
 Comment[zh_CN]=日历和日程安排程序
 Comment[zh_TW]=行事曆與排程軟體
 Comment[zu]=Ikhalenda kanye Neprogramu Yokugcina isikhathi
-Exec=korganizer --import %u
+Exec=korganizer
 Icon=korganizer
 Path=
 DocPath=korganizer/index.html
Index: korganizer-import.desktop
===================================================================
--- korganizer/korganizer-import.desktop	(revision 0)
+++ korganizer/korganizer-import.desktop	(revision 0)
@@ -0,0 +1,138 @@
+[Desktop Entry]
+Encoding=UTF-8
+MimeType=text/calendar;text/x-vcalendar;
+Comment=Calendar and Scheduling Program
+Comment[af]=Kalender en Skedulering Program
+Comment[ar]=برنامج الجدولة والتقويم
+Comment[bg]=Програма за календар и разписание
+Comment[bs]=Kalendar i rokovnik
+Comment[ca]=Un programa de calendari i planificació
+Comment[cs]=Kalendářový a plánovací program
+Comment[cy]=Rhaglen Galendr a Drefnlennu
+Comment[da]=Kalender- og planlægningsprogram
+Comment[de]=Ein Kalender und Zeitplaner
+Comment[el]=Πρόγραμμα ημερολογίου και προγραμματισμού
+Comment[eo]=Kalendara kaj plana programo
+Comment[es]=Calendario y planificador
+Comment[et]=Kalendri ja ajakava haldamise rakendus
+Comment[eu]=Egutegi eta antolaketa progrmaa
+Comment[fa]=تقویم و برنامۀ زمان‌بندی
+Comment[fi]=Kalenteri ja ajanhallintaohjelma
+Comment[fr]=Calendrier et agenda personnel
+Comment[gl]=Programa de Calendario e Axenda
+Comment[he]=תוכניות לוח שנה ותזמון משימות
+Comment[hi]=कैलेन्डर तथा समय-सारणी प्रोग्राम
+Comment[hr]=Kalendar i rokovnik
+Comment[hu]=Határidőnapló és eseményszervező
+Comment[is]=Dagbók og skipulag
+Comment[it]=Programma di calendario e di agenda
+Comment[ja]=カレンダーとスケジュール管理プログラム
+Comment[km]=កម្មវិធី​ប្រតិទិន និង កាលវិភាគ
+Comment[lt]=Kalendoriaus ir planavimo programa
+Comment[lv]=Kalendāra un Plānošanas Programma
+Comment[mk]=Календар и роковник
+Comment[ms]=Kalendar dan Program Penjadualan 
+Comment[mt]=Programm b' kalendarju w skeda
+Comment[nb]=Et kalender- og tidsplanleggingsprogram
+Comment[nds]=Kalenner un Tietplaner
+Comment[nl]=Agenda- en afsprakenprogramma
+Comment[nn]=Kalender- og planleggingsprogram
+Comment[nso]=Lenaneo la Peakanyo ya Tshupamabaka
+Comment[pl]=Kalendarz i terminarz
+Comment[pt]=Calendário e Programa de Escalonamento
+Comment[pt_BR]=Programa de Calendário e Agenda
+Comment[ro]=Program de planificare şi calendar
+Comment[ru]=Календарь и личное расписание
+Comment[se]=Kaleandar- ja plánenprográmma
+Comment[sk]=Kalendár a plánovací program
+Comment[sl]=Program za koledar in razporejanje
+Comment[sr]=Календарски и планерски програм
+Comment[sr@Latn]=Kalendarski i planerski program
+Comment[sv]=Kalender- och schemaläggningsprogram
+Comment[ta]=நாள்காட்டி மற்றும் திட்ட நிரல்
+Comment[tg]=Тақвимот ва ҷадвали шахсӣ
+Comment[th]=โปรแกรมจัดการปฏิทินและตารางการนัดหมาย
+Comment[tr]=Takvim ve Zamanlama Programı
+Comment[uk]=Програма календаря та розкладу
+Comment[uz]=Календар ва режалаштириш дастури
+Comment[ven]=Khalenda na mbekanyamushumo ya u shedula
+Comment[vi]=Chương trình lịch và kế hoạch
+Comment[xh]=Ikhalenda no Dweliso lwenkqubo Yokucwangcisa
+Comment[zh_CN]=日历和日程安排程序
+Comment[zh_TW]=行事曆與排程軟體
+Comment[zu]=Ikhalenda kanye Neprogramu Yokugcina isikhathi
+Exec=korganizer --import %u
+Icon=korganizer
+Path=
+DocPath=korganizer/index.html
+Type=Application
+Terminal=false
+Name=KOrganizer
+Name[af]=Korganizer
+Name[ar]=منظم كيدي
+Name[be]=K Арганізатар
+Name[cy]=KTrefnydd
+Name[eo]=Organizilo
+Name[hi]=के-आर्गेनाइज़र
+Name[lv]=KOrganaizers
+Name[mk]=Организатор
+Name[nso]=KMokopanyi
+Name[pl]=Organizator
+Name[sv]=Korganizer
+Name[ta]=கேஅமைப்பாளர்
+Name[ven]=Mulugisi wa K
+Name[zh_TW]=KOrganizer 行事曆
+GenericName=Personal Organizer
+GenericName[be]=Пэрсанальны арганізатар
+GenericName[bg]=Организатор
+GenericName[bs]=Lični organizer
+GenericName[ca]=Organitzador personal
+GenericName[cs]=Osobní organizér
+GenericName[cy]=Trefnydd Personol
+GenericName[da]=Personlig organisering
+GenericName[de]=Persönliche Daten organisieren
+GenericName[el]=Προσωπικός οργανωτής
+GenericName[en_GB]=Personal Organiser
+GenericName[es]=Organizador personal
+GenericName[et]=Personaalne ajaarvestus
+GenericName[eu]=Antolatzaile pertsonala
+GenericName[fa]=سازمان‌دهندۀ شخصی‌
+GenericName[fi]=Henkilökohtainen ajanhallintaohjelma
+GenericName[fr]=Organiseur personnel
+GenericName[gl]=Organizador Persoal
+GenericName[he]=מנהל זמן אישי
+GenericName[hi]=निजी प्रबंधक
+GenericName[hu]=Határidőnapló
+GenericName[is]=Persónuleg skipulagsbók
+GenericName[it]=Organizzatore personale
+GenericName[ja]=個人向けスケジュール管理
+GenericName[km]=កម្មវិធី​រៀបចំ​ផ្ទាល់​ខ្លួន
+GenericName[lt]=Asmeninės informacijos tvarkyklė
+GenericName[ms]=Penyusun Peribadi
+GenericName[nb]=Personlig planlegger
+GenericName[nds]=Persöönlich Mötenkalenner
+GenericName[nl]=Persoonlijke organizer
+GenericName[nn]=Personleg organiserar
+GenericName[pl]=Osobisty organizator
+GenericName[pt]=Organizador Pessoal
+GenericName[pt_BR]=Organizador Pessoal
+GenericName[ro]=Organizator personal
+GenericName[ru]=Персональный органайзер
+GenericName[sk]=Osobný plánovač
+GenericName[sl]=Osebni organizator
+GenericName[sr]=Лични планер
+GenericName[sr@Latn]=Lični planer
+GenericName[sv]=Filofax
+GenericName[ta]=தனிப்பயன் அமைப்பாளர்
+GenericName[tg]=Органайзери инфиродӣ
+GenericName[tr]=Kişisel Bilgi Yöneticisi
+GenericName[uk]=Персональний тижневик
+GenericName[uz]=Шахсий органайзер
+GenericName[zh_CN]=个人日程安排
+GenericName[zh_TW]=個人行程組織軟體
+ServiceTypes=Browser/View,DCOP/Organizer
+X-KDE-Library=libkorganizerpart
+X-KDE-StartupNotify=true
+X-DCOP-ServiceType=Unique
+X-DCOP-ServiceName=korganizer
+Hidden=true
--- kmail/kmfoldercachedimap.cpp	2006/08/14 10:07:38	572899
+++ kmail/kmfoldercachedimap.cpp	2006/10/22 19:56:30	598190
@@ -75,6 +75,7 @@
 #include <globalsettings.h>
 
 #define UIDCACHE_VERSION 1
+#define MAIL_LOSS_DEBUGGING 0
 
 static QString incidencesForToString( KMFolderCachedImap::IncidencesFor r ) {
   switch (r) {
@@ -150,10 +151,22 @@
     mFolderRemoved( false ),
     /*mHoldSyncs( false ),*/ mRecurse( true ),
     mStatusChangedLocally( false ), mAnnotationFolderTypeChanged( false ),
-    mIncidencesForChanged( false ), mPersonalNamespacesCheckDone( true )
+    mIncidencesForChanged( false ), mPersonalNamespacesCheckDone( true ),
+    mFoundAnIMAPDigest( false )
 {
   setUidValidity("");
-  readUidCache();
+  // if we fail to read a uid file but there is one, nuke it
+  if ( readUidCache() == -1 ) {
+    if ( QFile::exists( uidCacheLocation() ) ) {
+        KMessageBox::error( 0,
+        i18n( "The UID cache file for folder %1 could not be read. There "
+              "could be a problem with file system permission, or it is corrupted."
+              ).arg( folder->prettyURL() ) );
+        // try to unlink it, in case it was corruped. If it couldn't be read 
+        // because of permissions, this will fail, which is fine
+        unlink( QFile::encodeName( uidCacheLocation() ) );
+    }
+  }
 
   mProgress = 0;
 }
@@ -306,7 +319,7 @@
   if( uidValidity().isEmpty() || uidValidity() == "INVALID" ) {
     // No info from the server yet, remove the file.
     if( QFile::exists( uidCacheLocation() ) )
-      unlink( QFile::encodeName( uidCacheLocation() ) );
+      return unlink( QFile::encodeName( uidCacheLocation() ) );
     return 0;
   }
 
@@ -317,17 +330,23 @@
     str << uidValidity() << endl;
     str << lastUid() << endl;
     uidcache.flush();
-    fsync( uidcache.handle() ); /* this is probably overkill */
-    uidcache.close();
-    return 0;
-  } else {
-    return errno; /* does QFile set errno? */
+    if ( uidcache.status() == IO_Ok ) {
+      fsync( uidcache.handle() ); /* this is probably overkill */
+      uidcache.close();
+      if ( uidcache.status() == IO_Ok )
+        return 0;
+    }
   }
+  KMessageBox::error( 0,
+        i18n( "The UID cache file for folder %1 could not be written. There "
+              "could be a problem with file system permission." ).arg( folder()->prettyURL() ) );
+
+  return -1;
 }
 
 void KMFolderCachedImap::reloadUidMap()
 {
-  kdDebug(5006) << "Reloading Uid Map " << endl;
+  //kdDebug(5006) << "Reloading Uid Map " << endl;
   uidMap.clear();
   open();
   for( int i = 0; i < count(); ++i ) {
@@ -448,7 +467,8 @@
 {
   killTimer( uidWriteTimer );
   uidWriteTimer = -1;
-  writeUidCache();
+  if ( writeUidCache() == -1 )
+    unlink( QFile::encodeName( uidCacheLocation() ) );
 }
 
 ulong KMFolderCachedImap::lastUid()
@@ -467,10 +487,22 @@
   QMap<ulong,int>::Iterator it = uidMap.find( uid );
   if( it != uidMap.end() ) {
     KMMsgBase *msg = getMsgBase( *it );
+#if MAIL_LOSS_DEBUGGING
+    kdDebug(5006) << "UID " << uid << " is supposed to be in the map" << endl;
+    kdDebug(5006) << "UID's index is to be " << *it << endl;
+    kdDebug(5006) << "There is a message there? " << (msg != 0) << endl;
+    if ( msg ) {
+      kdDebug(5006) << "Its UID is: " << msg->UID() << endl;
+    }
+#endif
+
     if( msg && msg->UID() == uid )
       return msg;
+    kdDebug(5006) << "########## Didn't find uid: " << uid << "in cache athough it's supposed to be there!" << endl;
   } else {
+#if MAIL_LOSS_DEBUGGING
     kdDebug(5006) << "Didn't find uid: " << uid << "in cache!" << endl;
+#endif
   }
   // Not found by now
  // if( mapReloaded )
@@ -482,8 +514,10 @@
   if( it != uidMap.end() )
     // Since the uid map is just rebuilt, no need for the sanity check
     return getMsgBase( *it );
+#if MAIL_LOSS_DEBUGGING
   else
     kdDebug(5006) << "Reloaded, but stil didn't find uid: " << uid << endl;
+#endif
   // Then it's not here
   return 0;
 }
@@ -841,9 +875,14 @@
            to be deleted on the server has been deleted, adjust our local notion of the
            highes uid seen thus far. */
         slotUpdateLastUid();
-        if( mLastUid == 0 && uidWriteTimer == -1 )
+        if( mLastUid == 0 && uidWriteTimer == -1 ) {
           // This is probably a new and empty folder. Write the UID cache
-          writeUidCache();
+          if ( writeUidCache() == -1 ) {
+            resetSyncState();
+            emit folderComplete( this, false );
+            return;
+          }
+        }
       }
     }
 
@@ -1209,9 +1248,10 @@
 void KMFolderCachedImap::slotImapStatusChanged(KMFolder* folder, const QString&, bool cont)
 {
   if ( mSyncState == SYNC_STATE_INITIAL ){
-      kdDebug(5006) << "IMAP status changed but reset " << endl;
+      //kdDebug(5006) << "IMAP status changed but reset " << endl;
       return; // we were reset
   }
+  //kdDebug(5006) << "IMAP status changed for folder: " << folder->prettyURL() << endl;
   if ( folder->storage() == this ) {
     --mStatusFlagsJobs;
     if ( mStatusFlagsJobs == 0 || !cont ) // done or aborting
@@ -1220,6 +1260,7 @@
     if ( mStatusFlagsJobs == 0 && cont ) {
       mProgress += 5;
       serverSyncInternal();
+      //kdDebug(5006) << "Proceeding with mailcheck." << endl;
     }
   }
 }
@@ -1288,15 +1329,24 @@
   // them one by one because the index list can get resized under
   // us. So use msg pointers instead
 
+  QStringList uids;
   QMap<ulong,int>::const_iterator it = uidMap.constBegin();
   for( ; it != uidMap.end(); it++ ) {
     ulong uid ( it.key() );
-    if( uid!=0 && !uidsOnServer.find( uid ) )
+    if( uid!=0 && !uidsOnServer.find( uid ) ) {
+      uids << QString::number( uid );
       msgsForDeletion.append( getMsg( *it ) );
+    }
   }
 
   if( !msgsForDeletion.isEmpty() ) {
-    removeMsg( msgsForDeletion );
+#if MAIL_LOSS_DEBUGGING
+      if ( KMessageBox::warningYesNo(
+             0, i18n( "<qt><p>Mails on the server in folder <b>%1</b> were deleted. "
+                 "Do you want to delete them locally?<br>UIDs: %2</p></qt>" )
+             .arg( folder()->prettyURL() ).arg( uids.join(",") ) ) == KMessageBox::Yes )
+#endif
+        removeMsg( msgsForDeletion );
   }
 
   /* Delete messages from the server that we dont have anymore */
@@ -1370,6 +1420,8 @@
   uidsForDeletionOnServer.clear();
   mMsgsForDownload.clear();
   mUidsForDownload.clear();
+  // listing is only considered successful if saw a syntactically correct imapdigest
+  mFoundAnIMAPDigest = false;
 
   CachedImapJob* job = new CachedImapJob( FolderJob::tListMessages, this );
   connect( job, SIGNAL( result(KMail::FolderJob *) ),
@@ -1415,6 +1467,7 @@
       setReadOnly( access == "Read only" );
     }
     (*it).cdata.remove(0, pos);
+    mFoundAnIMAPDigest = true;
   }
   pos = (*it).cdata.find("\r\n--IMAPDIGEST", 1);
   // Start with something largish when rebuilding the cache
@@ -1432,7 +1485,7 @@
       if( uid != 0 ) {
         if ( uidsOnServer.count() == uidsOnServer.size() ) {
           uidsOnServer.resize( KMail::nextPrime( uidsOnServer.size() * 2 ) );
-          kdDebug( 5006 ) << "Resizing to: " << uidsOnServer.size() << endl;
+          //kdDebug( 5006 ) << "Resizing to: " << uidsOnServer.size() << endl;
         }
         uidsOnServer.insert( uid, &v );
       }
@@ -1451,7 +1504,9 @@
         KMMsgBase *existingMessage = findByUID(uid);
         if( !existingMessage ) {
           if ( mUserRights <= 0 || ( mUserRights & KMail::ACLJobs::Delete ) ) {
-            // kdDebug(5006) << "message with uid " << uid << " is gone from local cache. Must be deleted on server!!!" << endl;
+#if MAIL_LOSS_DEBUGGING
+            kdDebug(5006) << "message with uid " << uid << " is gone from local cache. Must be deleted on server!!!" << endl;
+#endif
             uidsForDeletionOnServer << uid;
           } else {
             redownload = true;
@@ -1490,6 +1545,13 @@
 void KMFolderCachedImap::getMessagesResult( KMail::FolderJob *job, bool lastSet )
 {
   mProgress += 10;
+  if ( !job->error() && !mFoundAnIMAPDigest ) {
+      kdWarning(5006) << "######## Folderlisting did not complete, but there was no error! "
+          "Aborting sync of folder: " << folder()->prettyURL() << endl;
+#if MAIL_LOSS_DEBUGGING
+      kmkernel->emergencyExit( i18n("Folder listing failed in interesting ways." ) );
+#endif
+  }
   if( job->error() ) { // error listing messages but the user chose to continue
     mContentState = imapNoInformation;
     mSyncState = SYNC_STATE_HANDLE_INBOX; // be sure not to continue in this folder
@@ -1741,7 +1803,7 @@
   KMFolderNode *node;
   bool root = ( this == mAccount->rootFolder() );
   if ( root && !mAccount->hasInbox() ) {
-    kdDebug(5006) << "check INBOX" << endl;
+    //kdDebug(5006) << "check INBOX" << endl;
     // create the INBOX
     for (node = folder()->child()->first(); node; node = folder()->child()->next())
       if (!node->isDir() && node->name() == "INBOX") break;
@@ -2216,7 +2278,7 @@
 void
 KMFolderCachedImap::slotAnnotationChanged( const QString& entry, const QString& attribute, const QString& value )
 {
-  kdDebug(5006) << k_funcinfo << entry << " " << attribute << " " << value << endl;
+  //kdDebug(5006) << k_funcinfo << entry << " " << attribute << " " << value << endl;
   if ( entry == KOLAB_FOLDERTYPE )
     mAnnotationFolderTypeChanged = false;
   else if ( entry == KOLAB_INCIDENCESFOR ) {
--- kmail/kmfoldercachedimap.h	2005/09/29 22:46:35	465494
+++ kmail/kmfoldercachedimap.h	2006/10/21 09:34:00	597659
@@ -445,6 +445,11 @@
       mLastUid. See above for details. */
   ulong mTentativeHighestUid;
 
+  /** Used to determine whether listing messages yielded a sensible result.
+   * Only then is the deletion o messages (which relies on succesful
+   * listing) attempted, during the sync.  */
+  bool mFoundAnIMAPDigest;
+
   int mUserRights;
   ACLList mACLList;
 

