--- libtorrent-rasterbar-0.16.10/include/libtorrent/disk_buffer_pool.hpp~	2013-07-11 07:37:03.057798742 +0000
+++ libtorrent-rasterbar-0.16.10/include/libtorrent/disk_buffer_pool.hpp	2013-07-11 07:37:27.110960260 +0000
@@ -33,6 +33,8 @@
 #ifndef TORRENT_DISK_BUFFER_POOL
 #define TORRENT_DISK_BUFFER_POOL
 
+#include <boost/noncopyable.hpp>
+
 #include "libtorrent/config.hpp"
 #include "libtorrent/thread.hpp"
 #include "libtorrent/session_settings.hpp"
