--- kdepim-runtime-4.7.2/plugins/akonadi_serializer_addressee.desktop~	2011-10-12 10:05:55.432618174 +0000
+++ kdepim-runtime-4.7.2/plugins/akonadi_serializer_addressee.desktop	2011-10-12 10:06:06.586046531 +0000
@@ -89,7 +89,7 @@
 Comment[zh_TW]=地址物件的 Akonadi 序列器外掛程式
 
 [Plugin]
-Type=text/vcard,text/directory
+Type=text/vcard
 X-Akonadi-Class=legacy;default;KABC::Addressee;
 X-KDE-Library=akonadi_serializer_addressee
 X-KDE-ClassName=Akonadi::SerializerPluginAddressee
