diff -Nbaur eric5-5.1.7.orig/eric/eric5.desktop eric5-5.1.7/eric/eric5.desktop
--- eric5-5.1.7.orig/eric/eric5.desktop	2011-12-06 06:42:40.196800129 +0000
+++ eric5-5.1.7/eric/eric5.desktop	1970-01-01 00:00:00.000000000 +0000
@@ -1,14 +0,0 @@
-[Desktop Entry]
-Type=Application
-Exec=eric5
-MimeType=application/x-eric5;
-Icon=eric
-Terminal=false
-Name=eric5
-Name[de]=eric5
-Comment=Integrated Development Environment for Python
-Comment[de]=Integrierte Entwicklungsumgebung für Python
-GenericName=Integrated Development Environment for Python
-GenericName[de]=Integrierte Entwicklungsumgebung für Python
-Categories=Qt;Python;Development;IDE;
-StartupNotify=true
diff -Nbaur eric5-5.1.7.orig/eric/eric.desktop eric5-5.1.7/eric/eric.desktop
--- eric5-5.1.7.orig/eric/eric.desktop	1970-01-01 00:00:00.000000000 +0000
+++ eric5-5.1.7/eric/eric.desktop	2011-12-06 06:53:24.102176503 +0000
@@ -0,0 +1,10 @@
+[Desktop Entry]
+Type=Application
+Exec=eric5
+MimeType=application/x-eric5;
+Icon=eric
+Terminal=false
+Name=Eric
+Comment=Python 3.x & Ruby IDE
+Categories=Qt;Python;Development;IDE;
+StartupNotify=true
diff -Nbaur eric5-5.1.7.orig/install.py eric5-5.1.7/install.py
--- eric5-5.1.7.orig/install.py	2011-12-06 06:42:40.253467010 +0000
+++ eric5-5.1.7/install.py	2011-12-06 06:58:31.401420942 +0000
@@ -276,7 +276,7 @@
     # Remove the menu entry for Linux systems
     if sys.platform.startswith("linux"):
         for name in ["/usr/share/pixmaps/eric.png", 
-                     "/usr/share/applications/eric5.desktop"]:
+                     "/usr/share/applications/eric.desktop"]:
             if os.path.exists(name):
                 os.remove(name)
     
@@ -519,11 +519,11 @@
             dst = os.path.normpath(os.path.join(distDir, "usr/share/applications"))
             if not os.path.exists(dst):
                 os.makedirs(dst)
-            shutil.copy(os.path.join(sourceDir, "eric5.desktop"), dst)
+            shutil.copy(os.path.join(sourceDir, "eric.desktop"), dst)
         else:
             shutil.copy(os.path.join(sourceDir, "icons", "default", "eric.png"), 
                 "/usr/share/pixmaps")
-            shutil.copy(os.path.join(sourceDir, "eric5.desktop"), 
+            shutil.copy(os.path.join(sourceDir, "eric.desktop"),
                 "/usr/share/applications")
     
     # Create a Mac application bundle
