diff -urNp ndesk-dbus-0.6.0/src/TypeImplementer.cs mine/src/TypeImplementer.cs
--- ndesk-dbus-0.6.0/src/TypeImplementer.cs	2007-10-11 16:01:11.000000000 -0400
+++ mine/src/TypeImplementer.cs	2010-01-19 01:37:31.317786525 -0500
@@ -34,7 +34,7 @@ namespace NDesk.DBus
 
 			InitHack ();
 
-			TypeBuilder typeB = modB.DefineType (declType.Name + "Proxy", TypeAttributes.Class | TypeAttributes.Public, typeof (BusObject));
+			TypeBuilder typeB = modB.DefineType (declType.FullName + "Proxy", TypeAttributes.Class | TypeAttributes.Public, typeof (BusObject));
 
 			Implement (typeB, declType);
 
