--- kdebase/kdesu/configure.in.in	2006/10/04 23:00:23	592501
+++ kdebase/kdesu/configure.in.in	2008/04/13 14:06:43	796383
@@ -1,6 +1,15 @@
 dnl Check for "struct ucred"
 AC_MSG_CHECKING("struct ucred")
-AC_EGREP_HEADER([struct ucred], [sys/socket.h], have_ucred=yes)
+AC_TRY_COMPILE(
+[
+  #define _GNU_SOURCE 1
+  #include <sys/socket.h>
+],
+[
+  struct ucred red;
+],  have_ucred=yes
+  , have_ucred=no
+)
 if test "$have_ucred" = "yes"; then
     AC_DEFINE(HAVE_STRUCT_UCRED, 1, [Define if you have the struct ucred])
     AC_MSG_RESULT(yes)
