Index: kdebase/apps/konsole/src/ViewContainer.cpp
===================================================================
--- kdebase/apps/konsole/src/ViewContainer.cpp	(Revision 937338)
+++ kdebase/apps/konsole/src/ViewContainer.cpp	(Revision 937339)
@@ -784,6 +784,7 @@
     searchBar()->setParent(_containerWidget);
     layout->addWidget(searchBar());
     layout->addWidget(_stackWidget);
+    layout->setMargin(0);
 }
 StackedViewContainer::~StackedViewContainer()
 {
@@ -829,6 +830,7 @@
     searchBar()->setParent(contentArea);
     layout->addWidget(_stackWidget);
     layout->addWidget(searchBar());
+    layout->setMargin(0);
 
     // elide left is used because the most informative part of the session name is often
     // the rightmost part
