--- connectivity/source/drivers/evoab2/NStatement.cxx
+++ connectivity/source/drivers/evoab2/NStatement.cxx
@@ -403,7 +403,7 @@
 			OSQLParseNode *pNodeForTableName = pAllTableNames->getChild( 0 )->getChild( 0 );
 			if( m_aSQLIterator.isTableNode( pNodeForTableName ) )
 			{
-                aTableName = getTableRange(pAllTableNames->getChild( 0 ));
+                                aTableName = OSQLParseNode::getTableRange(pAllTableNames->getChild( 0 ));
 				if( !aTableName.getLength() )
         			OSQLParseNode::getTableComponents( pNodeForTableName, aCatalog, aSchema, aTableName);
 			}
