From 0ca6cc5cfeb040a2eda486c62195fd4ca935dc99 Mon Sep 17 00:00:00 2001
From: Dave Reisner <dreisner@archlinux.org>
Date: Thu, 24 Jan 2013 16:26:43 -0500
Subject: [mkinitcpio] [PATCH] symlink /lib64 -> usr/lib in image

Signed-off-by: Dave Reisner <dreisner@archlinux.org>
---
 functions | 1 +
 1 file changed, 1 insertion(+)

diff --git a/functions b/functions
index 2286ac1..19b56ef 100644
--- a/functions
+++ b/functions
@@ -608,6 +608,7 @@ initialize_buildroot() {
     # base directory structure
     install -dm755 "$workdir/root"/{new_root,proc,sys,dev,run,tmp,etc,usr/{local,lib,bin}}
     ln -s "usr/lib" "$workdir/root/lib"
+    ln -s "usr/lib" "$workdir/root/lib64"
     ln -s "../lib"  "$workdir/root/usr/local/lib"
     ln -s "bin"     "$workdir/root/usr/sbin"
     ln -s "usr/bin" "$workdir/root/bin"
-- 
1.8.1.1

