--- data/epdfview.desktop	2008-10-15 23:20:01.000000000 +0200
+++ data/epdfview.desktop	2008-10-15 23:20:31.000000000 +0200
@@ -1,17 +1,16 @@
 [Desktop Entry]
-Categories=Viewer;Office;GTK
+Categories=Viewer;Office;GTK;
 Comment=Lightweight PDF document viewer
 Comment[ca]=Visor de documents PDF lleuger
 Comment[es]=Visor de documentos PDF ligero
-Encoding=UTF-8
 Exec=epdfview %f
 GenericName=PDF Viewer
 GenericName[ca]=Visor PDF
 GenericName[es]=Visor PDF
-Icon=postscript-viewer.png
+Icon=epdfview
 Name=ePDFViewer
 Name[ca]=ePDFViewer
 Name[es]=ePDFViewer
-MimeType=application/pdf
+MimeType=application/pdf;
 Terminal=false
 Type=Application
