diff --git a/lxde-common/startlxde.in b/lxde-common/startlxde.in
index 8372d3a..9391864 100755
--- a/lxde-common/startlxde.in
+++ b/lxde-common/startlxde.in
@@ -24,5 +24,13 @@ xprop -root -remove _NET_NUMBER_OF_DESKTOPS -remove _NET_DESKTOP_NAMES -remove _
 # Enable GTK+2 integration for OpenOffice.org, if available.
 export SAL_USE_VCLPLUGIN=gtk
 
+# Migrate old config files
+OLD_CONFIG=${XDG_CONFIG_HOME}/lxsession/LXDE/config
+NEW_CONFIG=${XDG_CONFIG_HOME}/lxsession/LXDE/desktop.conf
+if [ ! -f ${NEW_CONFIG} -a -f ${OLD_CONFIG} ]; then 
+    echo "Warning: Migrating config file: ${OLD_CONFIG} -> ${NEW_CONFIG}" >&2;
+    mv ${OLD_CONFIG} ${NEW_CONFIG}
+fi
+
 # Start the LXDE session
 exec @prefix@/bin/lxsession -s LXDE -e LXDE
