diff -r df1bb67fb168 plugins/sudoers/sudo_nss.c
--- a/plugins/sudoers/sudo_nss.c	Wed Mar 02 12:39:11 2011 -0500
+++ b/plugins/sudoers/sudo_nss.c	Thu Mar 03 09:37:48 2011 -0500
@@ -309,6 +309,7 @@
 /*
  * Check user_cmnd against sudoers and print the matching entry if the
  * command is allowed.
+ * Returns TRUE if the command is allowed, else FALSE.
  */
 int
 display_cmnd(struct sudo_nss_list *snl, struct passwd *pw)
@@ -320,7 +321,7 @@
 
     tq_foreach_fwd(snl, nss) {
 	if (nss->display_cmnd(nss, pw) == 0)
-	    return 0;
+	    return TRUE;
     }
-    return 1;
+    return FALSE;
 }
