--- configure.ac	2010-07-08 11:51:15.000000000 +0300
+++ configure.ac~	2010-08-06 20:45:35.998833001 +0300
@@ -179,11 +179,11 @@
                           ${VA_MICRO_VERSION:-0} "*" 100     + \
                           ${VA_SDS_VERSION:-0}`
 VA_DRIVER_INIT_FUNC="__vaDriverInit_${VA_MAJOR_VERSION}_${VA_MINOR_VERSION}"
-if test $va_full_version_int -ge 00310005; then
-    VA_DRIVER_INIT_FUNC="${VA_DRIVER_INIT_FUNC}_${VA_MICRO_VERSION}_sds${VA_SDS_VERSION}"
-else
-    VA_DRIVER_INIT_FUNC="${VA_DRIVER_INIT_FUNC}_sds"
-fi
+#if test $va_full_version_int -ge 00310005; then
+#    VA_DRIVER_INIT_FUNC="${VA_DRIVER_INIT_FUNC}_${VA_MICRO_VERSION}_sds${VA_SDS_VERSION}"
+#else
+#    VA_DRIVER_INIT_FUNC="${VA_DRIVER_INIT_FUNC}_sds"
+#fi
 if test "$ac_cv_libva_sds_extensions" = "yes"; then
     AC_DEFINE_UNQUOTED([VA_DRIVER_INIT_FUNC], [$VA_DRIVER_INIT_FUNC], [Define driver entry-point])
 else
