diff -Naur old/dirmngr/dns-stuff.c new/dirmngr/dns-stuff.c
--- old/dirmngr/dns-stuff.c	2016-12-20 00:19:05.000000000 -1000
+++ new/dirmngr/dns-stuff.c	2016-12-22 21:14:41.123388787 -1000
@@ -181,7 +181,9 @@
 enable_recursive_resolver (int yes)
 {
   recursive_resolver = yes;
+#ifdef USE_LIBDNS
   libdns_reinit_pending = 1;
+#endif /*USE_LIBDNS*/
 }
 
 
@@ -251,8 +253,10 @@
   strncpy (tor_nameserver, ipaddr? ipaddr : DEFAULT_NAMESERVER,
            sizeof tor_nameserver -1);
   tor_nameserver[sizeof tor_nameserver -1] = 0;
+#ifdef USE_LIBDNS
   libdns_reinit_pending = 1;
   libdns_tor_port = 0;  /* Start again with the default port.  */
+#endif /*USE_LIBDNS*/
 }
 
 
@@ -534,15 +538,15 @@
 void
 reload_dns_stuff (int force)
 {
+#ifdef USE_LIBDNS
   if (force)
     {
-#ifdef USE_LIBDNS
       libdns_deinit ();
-#endif
       libdns_reinit_pending = 0;
     }
   else
     libdns_reinit_pending = 1;
+#endif /*USE_LIBDNS*/
 }
 
 
@@ -1745,7 +1749,6 @@
 
   if (err)
     {
-      if (gpg_err_code (err) == GPG_ERR_NO_NAME)
         err = 0;
       goto leave;
     }
