--- sawfish-1.3.1/Makefile.in.orig	2007-10-08 11:35:16.000000000 -0500
+++ sawfish-1.3.1/Makefile.in	2007-10-08 11:35:48.000000000 -0500
@@ -29,7 +29,7 @@
 
 OLD_BINARIES = sawmill sawmill-client sawmill-capplet sawmill-ui sawmill-themer
 
-all : build.h DOC NEWS FAQ
+all : build.h DOC 
 	for dir in $(SUBDIRS); do \
 	  [ -d $$dir ] && ( cd $$dir && $(MAKE) $@ ) || exit 1; \
  	done
--- sawfish-1.3.1/configure.in.orig	2007-10-08 11:39:32.000000000 -0500
+++ sawfish-1.3.1/configure.in	2007-10-08 11:39:40.000000000 -0500
@@ -33,7 +33,7 @@
  man/Makefile po/Makefile Makedefs Makefile sawfish.spec\
  lisp/sawfish/gtk/widgets/font.jl"
 
-SUBDIRS="src lisp scripts themes sounds man po"
+SUBDIRS="src lisp scripts themes sounds po"
 AC_SUBST(SUBDIRS)
 
 add_subdir () {
