diff -u -r libpeas-1.8.0/loaders/gjs/peas-extension-gjs.c libpeas-1.8.0-js17/loaders/gjs/peas-extension-gjs.c
--- libpeas-1.8.0/loaders/gjs/peas-extension-gjs.c	2013-01-05 20:32:04.000000000 +0100
+++ libpeas-1.8.0-js17/loaders/gjs/peas-extension-gjs.c	2013-04-03 00:22:56.726029598 +0200
@@ -362,7 +362,7 @@
 
           if (!JS_GetElement (gexten->js_context, JSVAL_TO_OBJECT (js_retval),
                               nth_out_arg++, &js_value) ||
-              js_value == JSVAL_VOID)
+              js_value.asBits == JSVAL_VOID.asBits)
             {
               g_warning ("Error failed to get out argument %i", nth_out_arg);
               return FALSE;
diff -u -r libpeas-1.8.0/loaders/gjs/peas-plugin-loader-gjs.c libpeas-1.8.0-js17/loaders/gjs/peas-plugin-loader-gjs.c
--- libpeas-1.8.0/loaders/gjs/peas-plugin-loader-gjs.c	2012-08-15 12:40:31.000000000 +0200
+++ libpeas-1.8.0-js17/loaders/gjs/peas-plugin-loader-gjs.c	2013-04-03 00:19:12.884403305 +0200
@@ -279,7 +279,7 @@
 
       if (!JS_GetPropertyById (js_context, ginfo->extensions,
                                prop_name_id, &prop_extension_ctor) ||
-          prop_extension_ctor != extension_ctor)
+          prop_extension_ctor.asBits != extension_ctor.asBits)
         continue;
 
       if (!JS_IdToValue (js_context, prop_name_id, &prop_name_val) ||
