Index: dhcpcd.c
===================================================================
--- a/dhcpcd.c
+++ b/dhcpcd.c
@@ -404,4 +404,5 @@
         if (iface->addr.s_addr == 0) {
                 /* We failed to reboot, so enter discovery. */
+                iface->state->lease.addr.s_addr = 0;
                 start_discover(iface);
                 return;
@@ -1167,4 +1168,6 @@
                             now.tv_sec - st.st_mtime)
                         {
+                                syslog(LOG_DEBUG,
+                                    "%s: discarding expired lease", iface->name);
                                 free(iface->state->offer);
                                 iface->state->offer = NULL;
