From 07cbf21a0c729000db28da7cb9ba08e5b79f7674 Mon Sep 17 00:00:00 2001
From: Dieter Verfaillie <dieterv@optionexplicit.be>
Date: Tue, 23 Aug 2011 10:45:50 +0200
Subject: [PATCH] Fix documentation installation directory

---
 docs/Makefile.am |    5 +++--
 1 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/docs/Makefile.am b/docs/Makefile.am
index 775e1e9..b9086a9 100644
--- a/docs/Makefile.am
+++ b/docs/Makefile.am
@@ -1,3 +1,4 @@
+PLATFORM_VERSION = 3.0
 REF_VERSION = $(PYGOBJECT_MAJOR_VERSION).$(PYGOBJECT_MINOR_VERSION)
 FULL_VERSION = $(REF_VERSION).$(PYGOBJECT_MICRO_VERSION)
 
@@ -52,7 +53,7 @@ XSL_FILES =			\
 # so the date is newer then the files in HTML_FILES
 FIXXREF = xsl/fixxref.py
 
-TARGET_DIR = $(datadir)/gtk-doc/html/pygobject
+TARGET_DIR = $(datadir)/gtk-doc/html/pygobject-$(PLATFORM_VERSION)
 CSSdir = $(TARGET_DIR)
 CSS_DATA = $(CSS_FILES)
 
@@ -75,7 +76,7 @@ uninstall-local:
 	rm -f $(DESTDIR)$(TARGET_DIR)/*
 
 
-XSLdir = $(datadir)/pygobject/xsl
+XSLdir = $(datadir)/pygobject-$(PLATFORM_VERSION)/xsl
 XSL_DATA = $(XSL_FILES) $(FIXXREF)
 
 EXTRA_DIST = 		\
-- 
1.7.6.msysgit.0

