From d489b1d29f51f2d7d4caacaaca5232660c1da970 Mon Sep 17 00:00:00 2001
From: Christian Kellner <gicmo@gnome.org>
Date: Tue, 23 Feb 2010 11:58:53 +0000
Subject: Fix with-dbus-service-dir configure switch

Got broken by the commit 9936956378f295a12b05b5194af2397be2a9c94f
---
diff --git a/configure.ac b/configure.ac
index 0340b4e..8425190 100644
--- a/configure.ac
+++ b/configure.ac
@@ -48,8 +48,8 @@ PKG_CHECK_MODULES(GLIB, glib-2.0 >= 2.23.4 gthread-2.0 gobject-2.0 gmodule-no-ex
 PKG_CHECK_MODULES(DBUS, dbus-1)
 
 AC_ARG_WITH(dbus_service_dir,
-		AS_HELP_STRING([--with-dbus-service-dir=PATH],[choose directory for dbus service files]),
-		[default=PREFIX/share/dbus-1/services]], with_dbus_service_dir="$withval", with_dbus_service_dir=$datadir/dbus-1/services)
+		AS_HELP_STRING([--with-dbus-service-dir=PATH],[choose directory for dbus service files, [default=PREFIX/share/dbus-1/services]]),
+            with_dbus_service_dir="$withval", with_dbus_service_dir=$datadir/dbus-1/services)
 DBUS_SERVICE_DIR=$with_dbus_service_dir
 AC_SUBST(DBUS_SERVICE_DIR)
 
--
cgit v0.8.3.1
