diff -u -Nr liferea-1.7.4/src/notification/libnotify.c liferea-1.7.4-libnotify/src/notification/libnotify.c
--- liferea-1.7.4/src/notification/libnotify.c	2010-01-16 12:02:12.000000000 +0100
+++ liferea-1.7.4-libnotify/src/notification/libnotify.c	2011-03-07 00:27:39.906668939 +0100
@@ -158,7 +158,7 @@
 //		notify_notification_update ( n, node_get_title(node_p), labelText_now_p, NULL);
 //		notify_notification_clear_actions(n);
 
-		n = notify_notification_new (node_get_title (node_p), labelText_now_p, NULL, NULL);
+		n = notify_notification_new (node_get_title (node_p), labelText_now_p, NULL);
 
 		notify_notification_set_icon_from_pixbuf (n, node_get_icon (node_p));
 		notify_notification_set_category (n, "feed");
@@ -173,8 +173,6 @@
 							node_p->id, NULL);
 		}
 
-		notify_notification_attach_to_status_icon (n, ui_tray_get_status_icon ());
-
 		if (!notify_notification_show (n, NULL)) {
 			g_warning ("libnotify.c - failed to update notification via libnotify\n");
 		}
@@ -249,7 +247,7 @@
 
 	labelSummary_p = g_strdup_printf (ngettext ("%s has %d new / updated headline\n", "%s has %d new / updated headlines\n", item_count), 
 	                                  node_get_title (node), item_count);
-	n = notify_notification_new (_("Feed Update"), labelSummary_p, "liferea", NULL);
+	n = notify_notification_new (_("Feed Update"), labelSummary_p, "liferea");
 	g_free (labelSummary_p);
 
  	if (supports_append) {
@@ -272,8 +270,6 @@
 	}
 	notify_notification_set_category (n, "feed");
 
-	notify_notification_attach_to_status_icon (n, ui_tray_get_status_icon ());
-
 	if (!notify_notification_show (n, NULL))
 		g_warning ("notif_libnotify.c - failed to send notification via libnotify");
 }
