From d231fba5392972aea61030db708d8dcdf9087963 Mon Sep 17 00:00:00 2001
From: Martyn Russell <martyn@lanedo.com>
Date: Tue, 31 Mar 2015 23:00:13 +0100
Subject: tracker: Fixed execution bit of tracker-compat, which is broken by
 sed

https://bugzilla.gnome.org/show_bug.cgi?id=746939

diff --git a/src/tracker/Makefile.am b/src/tracker/Makefile.am
index 5815a7c..d487276 100644
--- a/src/tracker/Makefile.am
+++ b/src/tracker/Makefile.am
@@ -81,6 +81,7 @@ completion_DATA = tracker-prompt.sh
 
 tracker-compat: tracker-compat.in
 	$(AM_V_at)sed -e "s|@bindir[@]|${bindir}|" $< > $@
+	$(AM_V_at)chmod +x $@
 
 EXTRA_DIST = $(completion_DATA) tracker-compat.in
 CLEANFILES = tracker-compat
-- 
cgit v0.10.2

