diff -urNp farstream-0.1.1.orig/gst/fsrtpconference/default-codec-preferences farstream-0.1.1/gst/fsrtpconference/default-codec-preferences
--- farstream-0.1.1.orig/gst/fsrtpconference/default-codec-preferences	2012-03-20 12:12:22.644324485 +0100
+++ farstream-0.1.1/gst/fsrtpconference/default-codec-preferences	2012-03-21 09:32:04.468634455 +0100
@@ -41,6 +41,8 @@ feedback:nack/pli=
 [video/H263]
 feedback:nack/pli=
 
+[video/VP8-DRAFT-IETF-01]
+
 [video/THEORA]
 #feedback:tfrc=
 feedback:nack/pli=
