--- configure_old	2009-10-01 20:27:13.000000000 +0200
+++ configure	2010-03-25 12:31:10.000000000 +0100
@@ -19988,7 +19988,7 @@
 
 		{ echo "$as_me:$LINENO: checking Mozilla XPCOM > 1.8" >&5
 echo $ECHO_N "checking Mozilla XPCOM > 1.8... $ECHO_C" >&6; }
-		mozilla_xpcom_pcs="libxul-embedding-unstable"
+		mozilla_xpcom_pcs="libxul-embedding"
 		for pc in $mozilla_xpcom_pcs; do
 			if $PKG_CONFIG --exists $pc; then
 				{ echo "$as_me:$LINENO: result: $pc" >&5
