--- src/display.c	2003/08/19 05:34:31	1.45
+++ src/display.c	2005/01/04 15:34:25	1.46
@@ -44,7 +44,7 @@
 # include <netdb.h>
 #endif
 
-char *prog_name, *visual_name;
+char *visual_name;
 Display *dpy;
 int screen_num, screen_width, screen_height;
 Window root_window, no_focus_window;
@@ -286,7 +286,7 @@
 bool
 sys_init(char *program_name)
 {
-    char *display_name = 0;
+    char *display_name = 0, *prog_name = 0;
     repv opt;
 
 #ifdef HAVE_UNIX

--- src/sawmill_subrs.h	2003/08/19 05:34:31	1.108
+++ src/sawmill_subrs.h	2005/01/04 15:34:25	1.109
@@ -51,7 +51,6 @@
 extern void commands_init (void);
 
 /* from display.c */
-extern char *prog_name;
 extern Display *dpy;
 extern int screen_num, screen_width, screen_height, preferred_depth;
 extern Visual *preferred_visual;

--- src/main.c	2003/01/12 20:35:23	1.48
+++ src/main.c	2005/01/04 15:34:25	1.49
@@ -41,6 +41,7 @@
 #include "build.h"
 #include <string.h>
 #include <limits.h>
+#include <unistd.h>
 #include <X11/Xlocale.h>		/* for setlocale () */
 #ifdef HAVE_GDK_PIXBUF
 #include <glib-object.h>
@@ -49,9 +50,6 @@
 /* one of the ec_ values */
 int exit_code = ec_no_exit;
 
-/* Saved value of argv[0] */
-static char *prog_name;
-
 DEFSYM(sawfish_directory, "sawfish-directory");
 DEFSYM(sawfish_lisp_lib_directory, "sawfish-lisp-lib-directory");
 DEFSYM(sawfish_site_lisp_directory, "sawfish-site-lisp-directory");
@@ -336,6 +334,7 @@
     char **old_argv;
     int old_argc;
     char *lang;
+    char *prog_name;
 
     /* This will fork multiple copies, if necessary */
     multihead_init (&argc, &argv);

