diff -ur mpd-0.16.3.orig/src/decoder/ffmpeg_decoder_plugin.c mpd-0.16.3/src/decoder/ffmpeg_decoder_plugin.c
--- mpd-0.16.3.orig/src/decoder/ffmpeg_decoder_plugin.c	2011-06-04 17:36:33.000000000 +0300
+++ mpd-0.16.3/src/decoder/ffmpeg_decoder_plugin.c	2011-06-24 18:18:51.840218251 +0300
@@ -321,7 +321,7 @@
 	}
 
 	//ffmpeg works with ours "fileops" helper
-	AVFormatContext *format_context;
+	AVFormatContext *format_context = NULL;
 	if (av_open_input_stream(&format_context, stream->io, input->uri,
 				 input_format, NULL) != 0) {
 		g_warning("Open failed\n");
@@ -470,7 +470,7 @@
 	if (stream == NULL)
 		return NULL;
 
-	AVFormatContext *f;
+	AVFormatContext *f = NULL;
 	if (av_open_input_stream(&f, stream->io, is->uri,
 				 input_format, NULL) != 0) {
 		mpd_ffmpeg_stream_close(stream);
