diff -Nur telepathy-idle-0.1.7.orig/configure.ac telepathy-idle-0.1.7/configure.ac
--- telepathy-idle-0.1.7.orig/configure.ac	2010-12-07 03:52:29.000000000 -0800
+++ telepathy-idle-0.1.7/configure.ac	2010-12-07 09:41:21.179715253 -0800
@@ -103,7 +103,7 @@
   AC_MSG_ERROR([xsltproc (from the libxslt source package) is required])
 fi
 PYTHON=
-AC_CHECK_PROGS([PYTHON], [python2.3 python2.4 python2.5 python])
+AC_CHECK_PROGS([PYTHON], [python2.3 python2.4 python2.5 python2.7 python])
 if test -z "$PYTHON"; then
   AC_MSG_ERROR([Python is required to compile this package])
 fi
@@ -111,7 +111,7 @@
 
 dnl check for a version of python that can run the twisted tests
 AC_MSG_CHECKING([for Python with Twisted and IRC protocol support])
-for TEST_PYTHON in python2.5 python2.6 python; do
+for TEST_PYTHON in python2.5 python2.6 python2.7 python; do
   if $TEST_PYTHON -c "from sys import version_info; import dbus, dbus.mainloop.glib; raise SystemExit(version_info < (2, 5, 0, 'final', 0))" >/dev/null 2>&1; then
     if $TEST_PYTHON -c "import twisted.words.protocols.irc, twisted.internet.reactor" >/dev/null 2>&1; then
       AM_CONDITIONAL([WANT_TWISTED_TESTS], true)
