Fix a FTBFS with libpoppler 0.12: use getPDFMinorVersion instead of getPDFVersion
Index: texlive-bin-2007.dfsg.2/build/source/texk/web2c/pdftexdir/pdftoepdf.cc
===================================================================
--- texlive-bin-2007.dfsg.2.orig/build/source/texk/web2c/pdftexdir/pdftoepdf.cc	2009-09-28 10:25:18.000000000 +0000
+++ texlive-bin-2007.dfsg.2/build/source/texk/web2c/pdftexdir/pdftoepdf.cc	2009-09-28 10:59:33.000000000 +0000
@@ -673,6 +673,7 @@
     Page *page;
     int rotate;
     PDFRectangle *pagebox;
+    int minor_pdf_version_found, major_pdf_version_found;
     float pdf_version_found, pdf_version_wanted;
     // initialize
     if (!isInit) {
@@ -688,8 +689,11 @@
     // this works only for PDF 1.x -- but since any versions of PDF newer
     // than 1.x will not be backwards compatible to PDF 1.x, pdfTeX will
     // then have to changed drastically anyway.
-    pdf_version_found = pdf_doc->doc->getPDFVersion();
+    minor_pdf_version_found = pdf_doc->doc->getPDFMinorVersion();
+    major_pdf_version_found = pdf_doc->doc->getPDFMajorVersion();
+    pdf_version_found = major_pdf_version_found + (minor_pdf_version_found * 0.1);
     pdf_version_wanted = 1 + (minor_pdf_version_wanted * 0.1);
+
     if (pdf_version_found > pdf_version_wanted) {
         char msg[] =
             "PDF inclusion: found PDF version <%.1f>, but at most version <%.1f> allowed";
