--- QtWebKit-2.2.0/Source/JavaScriptCore/DerivedSources.pro~	2011-10-20 22:39:56.044519075 +0000
+++ QtWebKit-2.2.0/Source/JavaScriptCore/DerivedSources.pro	2011-10-20 22:41:21.302047082 +0000
@@ -88,5 +88,5 @@
 retgen.output = $$JSC_GENERATED_SOURCES_DIR/RegExpJitTables.h
 retgen.wkScript = $$PWD/create_regex_tables 
 retgen.input = retgen.wkScript
-retgen.commands = python $$retgen.wkScript > ${QMAKE_FILE_OUT}
+retgen.commands = python2 $$retgen.wkScript > ${QMAKE_FILE_OUT}
 addExtraCompiler(retgen)
--- QtWebKit-2.2.0/Source/WebCore/CodeGenerators.pri~	2011-10-20 22:44:23.827225580 +0000
+++ QtWebKit-2.2.0/Source/WebCore/CodeGenerators.pri	2011-10-20 22:44:38.550707819 +0000
@@ -640,7 +640,7 @@
 inspectorJSON.output = $${WC_GENERATED_SOURCES_DIR}/Inspector.idl
 inspectorJSON.input = INSPECTOR_JSON
 inspectorJSON.wkScript = $$PWD/inspector/generate-inspector-idl
-inspectorJSON.commands = python $$inspectorJSON.wkScript -o $${WC_GENERATED_SOURCES_DIR}/Inspector.idl $$PWD/inspector/Inspector.json
+inspectorJSON.commands = python2 $$inspectorJSON.wkScript -o $${WC_GENERATED_SOURCES_DIR}/Inspector.idl $$PWD/inspector/Inspector.json
 inspectorJSON.depends = $$PWD/inspector/generate-inspector-idl
 inspectorJSON.wkAddOutputToSources = false
 addExtraCompiler(inspectorJSON)
@@ -723,7 +723,7 @@
 entities.output = $${WC_GENERATED_SOURCES_DIR}/HTMLEntityTable.cpp
 entities.input = HTML_ENTITIES
 entities.wkScript = $$PWD/html/parser/create-html-entity-table
-entities.commands = python $$entities.wkScript -o $${WC_GENERATED_SOURCES_DIR}/HTMLEntityTable.cpp $$HTML_ENTITIES
+entities.commands = python2 $$entities.wkScript -o $${WC_GENERATED_SOURCES_DIR}/HTMLEntityTable.cpp $$HTML_ENTITIES
 entities.clean = ${QMAKE_FILE_OUT}
 entities.depends = $$PWD/html/parser/create-html-entity-table
 addExtraCompiler(entities)
