diff --git a/pixman/pixman-compose.c b/pixman/pixman-compose.c
index 1e91864..027394a 100644
--- a/pixman/pixman-compose.c
+++ b/pixman/pixman-compose.c
@@ -3880,7 +3880,7 @@ static void fbFetchTransformed(bits_image_t * pict, int x, int y, int width, uin
 			    idistx = 256 - distx;
 			    idisty = 256 - disty;
 			    
-			    b = bits + (y1)*stride;
+			    b = bits + (signed)((y1)*stride);
 			    x_off = x1;
 			    
 			    x1_out = (x1 < box.x1) | (x1 >= box.x2);
@@ -3946,7 +3946,7 @@ static void fbFetchTransformed(bits_image_t * pict, int x, int y, int width, uin
 			    idistx = 256 - distx;
 			    idisty = 256 - disty;
 			    
-			    b = bits + (y1)*stride;
+			    b = bits + (signed)((y1)*stride);
 			    x_off = x1;
 			    
 			    tl = pixman_region_contains_point(pict->common.src_clip, x1, y1, &box)
