diff --git a/refind/main.c b/refind/main.c
index beac3bf..1a2cf58 100644
--- a/refind/main.c
+++ b/refind/main.c
@@ -62,5 +62,5 @@
 #define SHELL_NAMES             L"\\EFI\\tools\\shell.efi,\\EFI\\tools\\shellia32.efi,\\shellia32.efi"
 #define DRIVER_DIRS             L"drivers,drivers_ia32"
 #else
-#define SHELL_NAMES             L"\\EFI\\tools\\shell.efi"
+#define SHELL_NAMES             L"\\EFI\\tools\\shell.efi,\\shell.efi"
 #define DRIVER_DIRS             L"drivers"
 #endif
 
