--- prog/pwm/fancontrol	2012-11-07 03:26:37.000000000 +0200
+++ prog/pwm/fancontrol	2013-03-22 18:15:00.566696011 +0200
@@ -297,7 +297,7 @@
 cd $DIR
 
 # Check for configuration change
-if [ "$DIR" != "/" ] && [ -z "$DEVPATH" -o -z "$DEVNAME" ]
+if [[ "$DIR" != "/"  && ( -z "$DEVPATH"  ||  -z "$DEVNAME" ) ]]
 then
 	echo "Configuration is too old, please run pwmconfig again" >&2
 	exit 1
@@ -337,7 +337,7 @@
 	# It didn't work, try pwmN_enable=1 pwmN=255
 	echo 1 > $ENABLE 2> /dev/null
 	echo $MAX > $1
-	if [ `cat $ENABLE` -eq 1 -a `cat $1` -ge 190 ]
+	if [ `cat $ENABLE` -eq 1 ] && [ `cat $1` -ge 190 ]
 	then
 		# Success
 		return 0
@@ -469,7 +469,7 @@
 		else
 		  # calculate the new value from temperature and settings
 		  pwmval="(${tval}-${mint})*(${maxpwm}-${minso})/(${maxt}-${mint})+${minso}"
-		  if [ $pwmpval -eq 0 -o $min_fanval -eq 0 ]
+		  if [ $pwmpval -eq 0 ] || [ $min_fanval -eq 0 ]
 		  then # if fan was stopped start it using a safe value
 		  	echo $minsa > $pwmo
 			# Sleep while still handling signals
