diff -ur kdegraphics-4.2.2/kolourpaint/pixmapfx/kpPixmapFX_MaskOps.cpp kdegraphics-4.2.2-kolourpaint-kde#183850-workaround/kolourpaint/pixmapfx/kpPixmapFX_MaskOps.cpp
--- kdegraphics-4.2.2/kolourpaint/pixmapfx/kpPixmapFX_MaskOps.cpp	2008-12-21 21:19:06.000000000 +0100
+++ kdegraphics-4.2.2-kolourpaint-kde#183850-workaround/kolourpaint/pixmapfx/kpPixmapFX_MaskOps.cpp	2009-04-03 15:23:49.000000000 +0200
@@ -252,7 +252,7 @@
         // Copy RGB layer.
         *destPixmapPtr = QPixmap (oldPixmap.width (), oldPixmap.height ());
         QPainter p (destPixmapPtr);
-        p.drawPixmap (QPoint (0, 0), oldPixmap);
+        p.drawImage (QPoint (0, 0), oldPixmap.toImage());
         p.end ();
 
         // Copy mask layer (if any).
