--- branches/KDE/4.1/kdegraphics/okular/generators/poppler/generator_pdf.cpp	2008/08/20 14:09:12	849950
+++ branches/KDE/4.1/kdegraphics/okular/generators/poppler/generator_pdf.cpp	2008/08/20 14:09:26	849951
@@ -868,6 +868,9 @@
     {
         userMutex()->unlock();
         delete psConverter;
+#ifdef HAVE_POPPLER_0_7
+        tf.close();
+#endif
         int ret = Okular::FilePrinter::printFile( printer, tempfilename,
                                                   Okular::FilePrinter::SystemDeletesFiles,
                                                   Okular::FilePrinter::ApplicationSelectsPages,
