--- kdegraphics-4.4.3/gwenview/lib/jpegcontent.cpp~	2010-05-05 00:34:00.190335453 +0200
+++ kdegraphics-4.4.3/gwenview/lib/jpegcontent.cpp	2010-05-05 00:35:32.947818779 +0200
@@ -455,9 +455,8 @@
 
 	// Init transformation
 	jpeg_transform_info transformoption;
+	memset(&transformoption, 0, sizeof(jpeg_transform_info));
 	transformoption.transform = findJxform(d->mTransformMatrix);
-	transformoption.force_grayscale = false;
-	transformoption.trim = false;
 	jtransform_request_workspace(&srcinfo, &transformoption);
 
 	/* Read source file as DCT coefficients */
