--- driver.c.orig	2011-05-20 11:59:24.000000000 +0200
+++ driver.c	2011-05-20 12:03:07.439120213 +0200
@@ -101,7 +101,7 @@
 static unsigned long		crit_flags;
 static atomic_t			scheduler_enabled	= ATOMIC_INIT (0);
 static atomic_t			scheduler_id		= ATOMIC_INIT (-1);
-static spinlock_t		stack_lock		= SPIN_LOCK_UNLOCKED;
+static DEFINE_SPINLOCK(stack_lock);
 #if !defined (__fcclassic__)
 static int			card_id			= 0;
 #endif
