--- branches/gnome-2-18/bonobo-activation/bonobo-activation-activate.c	2007/03/12 13:54:46	3353
+++ branches/gnome-2-18/bonobo-activation/bonobo-activation-activate.c	2007/04/10 11:47:54	3364
@@ -817,8 +817,7 @@
 	activation_environment._release = TRUE;
 
 	for (i = 0; i < activation_environment._length - 1; i++)
-		Bonobo_ActivationEnvValue_copy (
-			&activation_environment._buffer [i], &old_buffer [i]);
+		activation_environment._buffer[i] = old_buffer[i];
 
 	Bonobo_ActivationEnvValue_set (&activation_environment._buffer [i], name, value);
 
