--- trunk/activation-server/activation-context-corba.c	2007/09/18 16:51:55	3403
+++ trunk/activation-server/activation-context-corba.c	2007/09/18 16:54:14	3404
@@ -324,6 +324,8 @@
                 CORBA_Object_release (l2->data, &ev);
                 CORBA_exception_free (&ev);
         }
+
+	g_slist_free (notify);
 }
 
 gboolean
--- trunk/bonobo-activation/bonobo-activation-activate.c	2007/09/17 12:08:18	3402
+++ trunk/bonobo-activation/bonobo-activation-activate.c	2007/09/18 16:51:55	3403
@@ -797,7 +797,7 @@
 {
 	int                        i;
 
-	g_return_if_fail (name != NULL);
+	g_return_val_if_fail (name != NULL, NULL);
 
 	for (i = 0; i < activation_environment->_length; i++) {
 		if (strcmp (activation_environment->_buffer [i].name, name) == 0) {
--- trunk/activation-server/object-directory-load.c	2007/09/17 12:08:18	3402
+++ trunk/activation-server/object-directory-load.c	2007/09/18 16:51:55	3403
@@ -483,7 +483,8 @@
                         info->cur_prop->v._u.value_stringv._buffer =
                                 CORBA_sequence_CORBA_string_allocbuf (len);
                         
-                        for (i = 0, p = g_list_reverse (info->cur_items); p; p = p->next, i++)
+			info->cur_items = g_list_reverse (info->cur_items);
+                        for (i = 0, p = info->cur_items; p; p = p->next, i++)
                                 info->cur_prop->v._u.
                                         value_stringv._buffer[i] = p->data;
                         g_list_free (info->cur_items);
@@ -509,7 +510,8 @@
                         info->cur_server->props._buffer =
                                 CORBA_sequence_Bonobo_ActivationProperty_allocbuf (len);
 
-                        for (i = 0, p = g_list_reverse (info->cur_props); p; p = p->next, i++) {
+			info->cur_props = g_list_reverse (info->cur_props);
+                        for (i = 0, p = info->cur_props; p; p = p->next, i++) {
                                 Bonobo_ActivationProperty_copy (&info->cur_server->props._buffer[i],
                                                                 (Bonobo_ActivationProperty *) p->data);
                                 CORBA_free (p->data);
