--- cxx/ncvalues.cpp.orig	2008-03-08 22:09:19.000000000 +0000
+++ cxx/ncvalues.cpp	2008-03-08 22:08:30.000000000 +0000
@@ -9,7 +9,7 @@
 
 #include <config.h>
 #include <iostream>
-#include <string>
+#include <string.h>
 
 #include "ncvalues.h"
 
--- examples/CXX/sfc_pres_temp_rd.cpp.orig	2008-03-08 22:12:05.000000000 +0000
+++ examples/CXX/sfc_pres_temp_rd.cpp	2008-03-08 22:13:41.000000000 +0000
@@ -18,6 +18,7 @@
 
 #include <iostream>
 #include <netcdfcpp.h>
+#include <string.h>
 
 using namespace std;
 
