diff -ur kdesdk-3.5.5/kompare/kompare_shell.cpp kdesdk-3.5.5-patch/kompare/kompare_shell.cpp
--- kdesdk-3.5.5/kompare/kompare_shell.cpp	2005-09-10 10:21:05.000000000 +0200
+++ kdesdk-3.5.5-patch/kompare/kompare_shell.cpp	2006-10-18 15:36:24.000000000 +0200
@@ -192,7 +192,6 @@
 {
 	kdDebug(8102) << "Url = " << url.prettyURL() << endl;
 	m_diffURL = url;
-	m_viewPart->setEncoding( "Default" );
 	m_viewPart->openDiff( url );
 }
 
@@ -207,7 +206,6 @@
 
 	file.close();
 
-	m_viewPart->setEncoding( "Default" );
 	m_viewPart->openDiff( diff );
 
 }
@@ -217,7 +215,6 @@
 	m_sourceURL = source;
 	m_destinationURL = destination;
 
-	m_viewPart->setEncoding( "Default" );
 	m_viewPart->compare( source, destination );
 }
 
@@ -226,7 +223,6 @@
 	m_sourceURL = url1;
 	m_destinationURL = diff;
 
-	m_viewPart->setEncoding( "Default" );
 	m_viewPart->openDirAndDiff( url1, diff );
 }
