=== modified file 'data/gnome-screensaver'
--- data/gnome-screensaver
+++ data/gnome-screensaver
@@ -1,13 +1,16 @@
 #%PAM-1.0
 
 # Fedora Core
-auth       include	system-auth
-account    include	system-auth
-password   include	system-auth
-session    include	system-auth
+#auth       include	system-auth
+#account    include	system-auth
+#password   include	system-auth
+#session    include	system-auth
 
 # SuSE/Novell
 #auth       include      common-auth
 #account    include      common-account
 #password   include      common-password
 #session    include      common-session
+
+# Arch Linux
+auth        required     pam_unix_auth.so

