diff --git a/sources/pyside6/plugins/designer/CMakeLists.txt b/sources/pyside6/plugins/designer/CMakeLists.txt
index 50074be51..5d6064606 100644
--- a/sources/pyside6/plugins/designer/CMakeLists.txt
+++ b/sources/pyside6/plugins/designer/CMakeLists.txt
@@ -53,4 +53,4 @@ target_link_libraries(PySidePlugin PRIVATE
                       Qt::Widgets
                       ${SHIBOKEN_PYTHON_LIBRARIES})
 
-install(TARGETS PySidePlugin LIBRARY DESTINATION "plugins/designer")
+install(TARGETS PySidePlugin LIBRARY DESTINATION "lib/qt6/plugins/designer")
