From cdfdb7290abb270f5f98c93089c3b6e3c35859bc Mon Sep 17 00:00:00 2001
From: Thaddaeus Tintenfisch <thad.fisch@gmail.com>
Date: Sun, 22 Feb 2015 23:34:45 +0100
Subject: [PATCH] Fallback to non-cached version of source file (bug #9142)

---
 thunar/thunar-file.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/thunar/thunar-file.c b/thunar/thunar-file.c
index 9b5aa9b..9dedf47 100644
--- a/thunar/thunar-file.c
+++ b/thunar/thunar-file.c
@@ -1978,6 +1978,10 @@ thunar_file_accepts_drop (ThunarFile     *file,
               /* determine the cached version of the source file */
               ofile = thunar_file_cache_lookup (lp->data);
 
+              /* fallback to non-cached version */
+              if (ofile == NULL)
+                ofile = thunar_file_get (lp->data, NULL);
+
               /* we have only move if we know the source and both the source and the target
                * are on the same disk, and the source file is owned by the current user.
                */
-- 
2.3.4

