Index: src/freeglut_window.c
===================================================================
--- src/freeglut_window.c	(Revision 832)
+++ src/freeglut_window.c	(Arbeitskopie)
@@ -594,10 +594,9 @@
 #if TARGET_HOST_POSIX_X11
     if ( window )
     {
-        glXMakeContextCurrent(
+        glXMakeCurrent(
             fgDisplay.Display,
             window->Window.Handle,
-            window->Window.Handle,
             window->Window.Context
         );
 
@@ -932,10 +932,9 @@
     XSetWMProtocols( fgDisplay.Display, window->Window.Handle,
                      &fgDisplay.DeleteWindow, 1 );
 
-    glXMakeContextCurrent(
+    glXMakeCurrent(
         fgDisplay.Display,
         window->Window.Handle,
-        window->Window.Handle,
         window->Window.Context
     );
 
