diff -ur avidemux_2.4.3.orig/avidemux/ADM_codecs/ADM_x264.cpp avidemux_2.4.3/avidemux/ADM_codecs/ADM_x264.cpp
--- avidemux_2.4.3.orig/avidemux/ADM_codecs/ADM_x264.cpp	2008-10-09 01:45:06.048940024 -0700
+++ avidemux_2.4.3/avidemux/ADM_codecs/ADM_x264.cpp	2008-10-09 01:49:07.606940126 -0700
@@ -109,8 +109,14 @@
   MKPARAM(i_bframe,MaxBFrame);
   MKPARAM(i_bframe_bias,Bias);
   MKPARAM( b_bframe_pyramid,BasReference );
+#if X264_BUILD < 65
   MKPARAM(analyse. b_bidir_me,BidirME );
+#endif
+#if X264_BUILD >= 63
+  MKPARAM( i_bframe_adaptive, Adaptative);
+#else
   MKPARAM( b_bframe_adaptive, Adaptative);
+#endif
   MKPARAM( analyse.b_weighted_bipred, Weighted);
   MKPARAM( b_cabac , CABAC);
   MKPARAM( analyse.i_trellis, Trellis);
@@ -118,12 +124,16 @@
 #define MIN_RDO 6
   if(zparam->PartitionDecision+1>=MIN_RDO)
   {
-      int rank,parity;
-      rank=((zparam->PartitionDecision+1-MIN_RDO)>>1)+MIN_RDO;
-      parity=(zparam->PartitionDecision+1-MIN_RDO)&1;
+#if X264_BUILD >= 65
+    param.analyse.i_subpel_refine = zparam->PartitionDecision + 1;
+#else
+    int rank,parity;
+    rank=((zparam->PartitionDecision+1-MIN_RDO)>>1)+MIN_RDO;
+    parity=(zparam->PartitionDecision+1-MIN_RDO)&1;
       
-      param.analyse.i_subpel_refine=rank;
-      param.analyse.b_bframe_rdo=parity;
+    param.analyse.i_subpel_refine=rank;
+    param.analyse.b_bframe_rdo=parity;
+#endif
   }
   MKPARAM(analyse.b_chroma_me,ChromaME);
   MKPARAM(b_deblocking_filter,DeblockingFilter);
