--- avidemux/ADM_colorspace/ADM_rgb.h
+++ avidemux/ADM_colorspace/ADM_rgb.h
@@ -19,6 +19,8 @@
  #ifndef COLYUVRGB_H
  #define COLYUVRGB_H
 
+#include <inttypes.h>
+
 #define ADM_COLOR_IS_YUV 0x1000
  typedef enum 
  {
--- avidemux/ADM_inputs/ADM_mpegdemuxer/dmx_io.cpp
+++ avidemux/ADM_inputs/ADM_mpegdemuxer/dmx_io.cpp
@@ -71,7 +71,7 @@
 */
 uint8_t fileParser::open( const char *filename,FP_TYPE *multi )
 {
-        char *dot = NULL;                   // pointer to the last dot in filename
+        const char *dot = NULL;                   // pointer to the last dot in filename
         uint8_t decimals = 0;               // number of decimals
         char *left = NULL, *number = NULL, *right = NULL; // parts of filename (after splitting)
 
--- avidemux/ADM_libraries/ADM_libmpeg2enc/yuv4mpeg_ratio.cc
+++ avidemux/ADM_libraries/ADM_libmpeg2enc/yuv4mpeg_ratio.cc
@@ -104,7 +104,7 @@
 
 int y4m_parse_ratio(y4m_ratio_t *r, const char *s)
 {
-  char *t = strchr(s, ':');
+  const char *t = strchr(s, ':');
   if (t == NULL) return Y4M_ERR_RANGE;
   r->n = atoi(s);
   r->d = atoi(t+1);
--- avidemux/ADM_libraries/ADM_mplex/lpcmstrm_in.cpp
+++ avidemux/ADM_libraries/ADM_mplex/lpcmstrm_in.cpp
@@ -53,7 +53,7 @@
 
 bool LPCMStream::Probe(IBitStream &bs )
 {
-    char *last_dot = strrchr( bs.StreamName(), '.' );
+    const char *last_dot = strrchr( bs.StreamName(), '.' );
     return 
         last_dot != NULL 
         && strcmp( last_dot+1, "lpcm") == 0;
--- avidemux/ADM_osSupport/ADM_fileio.cpp
+++ avidemux/ADM_osSupport/ADM_fileio.cpp
@@ -543,7 +543,7 @@
 */
 const char *GetFileName(const char *str)
 {
-	char *filename;
+	const char *filename;
         char *filename2;
 #ifndef ADM_WIN32		
 	filename = strrchr(str, '/');
--- avidemux/ADM_userInterfaces/ADM_QT4/ADM_dialog/DIA_color.cpp
+++ avidemux/ADM_userInterfaces/ADM_QT4/ADM_dialog/DIA_color.cpp
@@ -1,6 +1,7 @@
 #include <QtGui>
 
 #include "../ADM_toolkit/qtToolkit.h"
+#include <stdint.h>
 
 int DIA_colorSel(uint8_t *r, uint8_t *g, uint8_t *b)
 {
@@ -17,4 +18,4 @@
 	}
 
 	return 0;
-}
\ No newline at end of file
+}
--- avidemux/ADM_userInterfaces/ADM_QT4/ADM_dialog/Q_cnr2.cpp
+++ avidemux/ADM_userInterfaces/ADM_QT4/ADM_dialog/Q_cnr2.cpp
@@ -16,6 +16,7 @@
  ***************************************************************************/
 
 #include <math.h>
+#include <stdint.h>
 
 #include "config.h"
 #include "ui_cnr2.h"
--- avidemux/ADM_userInterfaces/ADM_QT4/ADM_dialog/Q_coloryuv.cpp
+++ avidemux/ADM_userInterfaces/ADM_QT4/ADM_dialog/Q_coloryuv.cpp
@@ -16,6 +16,7 @@
  ***************************************************************************/
 #include "config.h"
 
+#include <stdint.h>
 #include <QtGui/QDialog>
 
 #include "ui_coloryuv.h"
--- avidemux/ADM_userInterfaces/ADM_QT4/ADM_gui/ADM_qslider.cpp
+++ avidemux/ADM_userInterfaces/ADM_QT4/ADM_gui/ADM_qslider.cpp
@@ -13,7 +13,7 @@
 ***************************************************************************/
 
 #include "config.h"
-
+#include <cstdio>
 #include <math.h>
 #include <QPainter>
 #include <QSlider>
--- avidemux/ADM_userInterfaces/ADM_QT4/ADM_gui/ADM_qslider.h
+++ avidemux/ADM_userInterfaces/ADM_QT4/ADM_gui/ADM_qslider.h
@@ -1,6 +1,6 @@
 #ifndef ADM_Q_SLIDER_H
 #define ADM_Q_SLIDER_H
-
+#include <inttypes.h>
 class ADM_QSlider : public QSlider
 {
 protected:
--- avidemux/ADM_userInterfaces/ADM_QT4/ADM_gui/gui_none.cpp
+++ avidemux/ADM_userInterfaces/ADM_QT4/ADM_gui/gui_none.cpp
@@ -9,7 +9,7 @@
 //
 
 #include "config.h"
-
+#include <inttypes.h>
 #include <QtGui/QApplication>
 #include <QtGui/QDesktopWidget>
 
--- avidemux/ADM_video/mmx_macros.h
+++ avidemux/ADM_video/mmx_macros.h
@@ -353,7 +353,7 @@
 #define	mmx_m2r(op, mem, reg) \
 	__asm__ __volatile__ (#op " %0, %%" #reg \
 			      : /* nothing */ \
-			      : "X" (mem))
+			      : "m" (mem))
 
 #define	mmx_r2m(op, reg, mem) \
 	__asm__ __volatile__ (#op " %%" #reg ", %0" \
