--- avidemux-plugins_2.5.0/plugins/ADM_videoFilters/AvisynthResize/qt4/Q_resize.h
+++ avidemux-plugins_2.5.0/plugins/ADM_videoFilters/AvisynthResize/qt4/Q_resize.h
@@ -1,6 +1,8 @@
 #ifndef Q_resizing_h
 #define Q_resizing_h
 
+#include <stdint.h>
+
 #include "ui_resizing.h"
 
 typedef struct resParam
--- avidemux-plugins_2.5.0/plugins/ADM_videoFilters/CNR2/ADM_vidCNR2_param.h
+++ avidemux-plugins_2.5.0/plugins/ADM_videoFilters/CNR2/ADM_vidCNR2_param.h
@@ -15,6 +15,8 @@
 #ifndef _CNR2_
 #define _CNR2_
 
+#include <stdint.h>
+
 typedef struct CNR2Param
 {
         double scdthr;          // Scene change threshold in % default 10
--- avidemux-plugins_2.5.0/plugins/ADM_videoFilters/colorYUV/ADM_vidColorYuv_param.h
+++ avidemux-plugins_2.5.0/plugins/ADM_videoFilters/colorYUV/ADM_vidColorYuv_param.h
@@ -14,6 +14,9 @@
 
 #ifndef ADM_VID_COLOR_YUV_PARAM_H
 #define ADM_VID_COLOR_YUV_PARAM_H
+
+#include <stdint.h>
+
 typedef struct COLOR_YUV_PARAM
 {
     double y_contrast, y_bright, y_gamma, y_gain;
--- avidemux-plugins_2.5.0/plugins/ADM_videoFilters/MPDelogo/ADM_vidMPdelogo.h
+++ avidemux-plugins_2.5.0/plugins/ADM_videoFilters/MPDelogo/ADM_vidMPdelogo.h
@@ -12,6 +12,9 @@
  ***************************************************************************/
 #ifndef ADM_DELOGO_H
 #define ADM_DELOGO_H
+
+#include <stdint.h>
+
 typedef struct MPDELOGO_PARAM
 {
         uint32_t xoff;
--- avidemux-plugins_2.5.0/plugins/ADM_videoFilters/MPDelogo/DIA_flyMpDelogo.h
+++ avidemux-plugins_2.5.0/plugins/ADM_videoFilters/MPDelogo/DIA_flyMpDelogo.h
@@ -1,5 +1,8 @@
 #ifndef FLY_MPDELOGO_H
 #define FLY_MPDELOGO_H
+
+#include <stdint.h>
+
 class flyMpDelogo : public FLY_DIALOG_TYPE
 {
 public:
--- avidemux-plugins_2.5.0/plugins/ADM_videoFilters/MplayerResize/qt4/Q_resize.h
+++ avidemux-plugins_2.5.0/plugins/ADM_videoFilters/MplayerResize/qt4/Q_resize.h
@@ -1,6 +1,8 @@
 #ifndef Q_resizing_h
 #define Q_resizing_h
 
+#include <stdint.h>
+
 #include "ui_resizing.h"
 
 typedef struct resParam
diff -Naur avidemux-plugins_2.5.0.orig/plugins/ADM_videoEncoder/ADM_vidEnc_x264/guiHelper.cpp avidemux-plugins_2.5.0/plugins/ADM_videoEncoder/ADM_vidEnc_x264/guiHelper.cpp
--- avidemux-plugins_2.5.0.orig/plugins/ADM_videoEncoder/ADM_vidEnc_x264/guiHelper.cpp	2009-07-12 01:47:28.082355322 -0400
+++ avidemux-plugins_2.5.0/plugins/ADM_videoEncoder/ADM_vidEnc_x264/guiHelper.cpp	2009-07-12 01:49:40.403171050 -0400
@@ -62,8 +62,8 @@
 
 int x264_cqm_parse_jmlist(const char *buf, const char *name, uint8_t *cqm, uint8_t *defaultMatrix, int length)
 {
-	char *p;
-	char *nextvar;
+	const char *p;
+	const char *nextvar;
 	int i;
 
 	p = strstr(buf, name);
diff -Naur avidemux-plugins_2.5.0.orig/plugins/ADM_videoEncoder/ADM_vidEnc_x264/qt4/x264CustomMatrixDialog.h avidemux-plugins_2.5.0/plugins/ADM_videoEncoder/ADM_vidEnc_x264/qt4/x264CustomMatrixDialog.h
--- avidemux-plugins_2.5.0.orig/plugins/ADM_videoEncoder/ADM_vidEnc_x264/qt4/x264CustomMatrixDialog.h	2009-07-12 01:47:28.079020819 -0400
+++ avidemux-plugins_2.5.0/plugins/ADM_videoEncoder/ADM_vidEnc_x264/qt4/x264CustomMatrixDialog.h	2009-07-12 01:50:19.009000160 -0400
@@ -1,6 +1,7 @@
 #ifndef x264CustomMatrixDialog_h
 #define x264CustomMatrixDialog_h
 
+#include <stdint.h>
 #include "ui_x264CustomMatrixDialog.h"
 
 class x264CustomMatrixDialog : public QDialog
diff -Naur avidemux-plugins_2.5.0.orig/plugins/ADM_videoEncoder/ADM_vidEnc_x264/zoneOptions.h avidemux-plugins_2.5.0/plugins/ADM_videoEncoder/ADM_vidEnc_x264/zoneOptions.h
--- avidemux-plugins_2.5.0.orig/plugins/ADM_videoEncoder/ADM_vidEnc_x264/zoneOptions.h	2009-07-12 01:47:28.079020819 -0400
+++ avidemux-plugins_2.5.0/plugins/ADM_videoEncoder/ADM_vidEnc_x264/zoneOptions.h	2009-07-12 01:50:44.178997807 -0400
@@ -17,6 +17,8 @@
 #ifndef zoneOptions_h
 #define zoneOptions_h
 
+#include <stdint.h>
+
 extern "C"
 {
 #include "x264.h"
diff -Naur avidemux-plugins_2.5.0.orig/plugins/ADM_videoEncoder/ADM_vidEnc_xvid/options.h avidemux-plugins_2.5.0/plugins/ADM_videoEncoder/ADM_vidEnc_xvid/options.h
--- avidemux-plugins_2.5.0.orig/plugins/ADM_videoEncoder/ADM_vidEnc_xvid/options.h	2009-07-12 01:47:28.082355322 -0400
+++ avidemux-plugins_2.5.0/plugins/ADM_videoEncoder/ADM_vidEnc_xvid/options.h	2009-07-12 01:51:12.152328208 -0400
@@ -17,6 +17,7 @@
 #ifndef options_h
 #define options_h
 
+#include <stdint.h>
 #include <libxml/tree.h>
 #include "xvid.h"
 
