--- Makefile.misc.orig	2007-02-02 21:37:20.000000000 -0500
+++ Makefile.misc	2007-02-02 21:37:51.000000000 -0500
@@ -92,7 +92,7 @@
 	      $(RM) -f $(LIBDIR)/$(SUBDIR)$(LIB).$(SHAREDSUFFIX) && \
               (cd $(LIBDIR) && \
 	       $(RM) -f $(LIB).$(SHAREDSUFFIX) && \
-               ln -s $(DESTDIR)$(FILDIR)/$(SUBDIR)$(LIB).$(SHAREDSUFFIX) \
+               ln -s $(FILDIR)/$(SUBDIR)$(LIB).$(SHAREDSUFFIX) \
                 $(LIB).$(SHAREDSUFFIX) \
                || \
                cp $(DESTDIR)$(FILDIR)/$(SUBDIR)$(LIB).$(SHAREDSUFFIX) \
@@ -128,7 +128,7 @@
 	  (cd $(LIBDIR) && \
            if [ -f $(DESTDIR)$(FILDIR)/$(SUBDIR)$(LIB).$(SHAREDSUFFIX) ]; then \
 	     $(RM) -f $(LN).$(SHAREDSUFFIX) && \
-             ln -s $(DESTDIR)$(FILDIR)/$(SUBDIR)$(LIB).$(SHAREDSUFFIX) \
+             ln -s $(FILDIR)/$(SUBDIR)$(LIB).$(SHAREDSUFFIX) \
                    $(LN).$(SHAREDSUFFIX) \
              || \
              cp $(DESTDIR)$(FILDIR)/$(SUBDIR)$(LIB).$(SHAREDSUFFIX) \
