From 1f266d2e8183c703be391ee8393c56a3e5654b27 Mon Sep 17 00:00:00 2001
From: Albert Astals Cid <aacid@kde.org>
Date: Tue, 01  Jul  2008  18:16:50  +0000
Subject: Fix condition, we want fetched_thumb to be a Stream, not to be non null

Fixes bug 16579
---
diff --git a/poppler/Page.cc b/poppler/Page.cc
index 3da5976..fa2d86d 100644
--- a/poppler/Page.cc
+++ b/poppler/Page.cc
@@ -466,7 +466,7 @@ GBool Page::loadThumb(unsigned char **data_out,
 
   /* Get stream dict */
   thumb.fetch(xref, &fetched_thumb);
-  if (fetched_thumb.isNull()) {
+  if (!fetched_thumb.isStream()) {
     fetched_thumb.free();
     return gFalse;
   }
--
cgit v0.7.2-37-g538c
