--- pango-1.24.4/pango/pangofc-font.c~	2009-07-19 12:50:45.000000000 +0200
+++ pango-1.24.4/pango/pangofc-font.c	2009-07-19 12:51:01.000000000 +0200
@@ -141,8 +141,7 @@
 
   if (fcfont->fontmap)
     {
       _pango_fc_font_map_remove (PANGO_FC_FONT_MAP (fcfont->fontmap), fcfont);
-      g_object_remove_weak_pointer (G_OBJECT (fcfont->fontmap), (gpointer *) (gpointer) &fcfont->fontmap);
       fcfont->fontmap = NULL;
     }
 
