diff -u -r NetworkManager-openvpn-1.2.4/properties/tests/Makefile.am NetworkManager-openvpn-1.2.4-buildfix/properties/tests/Makefile.am
--- NetworkManager-openvpn-1.2.4/properties/tests/Makefile.am	2016-07-02 11:28:03.000000000 +0200
+++ NetworkManager-openvpn-1.2.4-buildfix/properties/tests/Makefile.am	2016-07-10 20:45:57.976832915 +0200
@@ -23,12 +23,12 @@
 	$(LIBNMA_CFLAGS)
 
 test_import_export_LDADD = \
+	$(top_builddir)/properties/libnm-vpn-plugin-openvpn-test.la \
 	$(GTHREAD_LIBS) \
 	$(GTK_LIBS) \
 	$(LIBNM_LIBS) \
 	$(LIBNMA_LIBS) \
-	-ldl \
-	$(top_builddir)/properties/libnm-vpn-plugin-openvpn-test.la
+	-ldl
 
 
 test_import_export_glib_SOURCES = \
