diff -ruN gdm-2.30.4/gui/simple-greeter/gdm-user-manager.c gdm-2.30.4.since//gui/simple-greeter/gdm-user-manager.c
--- gdm-2.30.4/gui/simple-greeter/gdm-user-manager.c	2010-06-26 00:09:31.000000000 +0200
+++ gdm-2.30.4.since//gui/simple-greeter/gdm-user-manager.c	2010-08-10 20:11:21.080000061 +0200
@@ -75,6 +75,10 @@
 #define GDM_USERNAME "gdm"
 #endif
 
+/* approximately two months */
+#define LOGIN_FREQUENCY_TIME_WINDOW_SECS (60 * 24 * 60 * 60)
+
+
 #define RELOAD_PASSWD_THROTTLE_SECS 5
 
 typedef enum {
@@ -1579,12 +1583,14 @@
 load_ck_history (GdmUserManager *manager)
 {
         char       *command;
+        char       *since;
         const char *seat_id;
         GError     *error;
         gboolean    res;
         char      **argv;
         int         standard_out;
         GIOChannel *channel;
+        GTimeVal    tv;
 
         g_assert (manager->priv->ck_history_id == 0);
 
@@ -1602,8 +1608,14 @@
                 goto out;
         }
 
-        command = g_strdup_printf ("ck-history --frequent --seat='%s' --session-type=''",
+        g_get_current_time (&tv);
+        tv.tv_sec -= LOGIN_FREQUENCY_TIME_WINDOW_SECS;
+        since = g_time_val_to_iso8601 (&tv);
+
+        command = g_strdup_printf ("ck-history --frequent --since='%s' --seat='%s' --session-type=''",
+                                   since,
                                    seat_id);
+        g_free (since);
         g_debug ("GdmUserManager: running '%s'", command);
         error = NULL;
         if (! g_shell_parse_argv (command, NULL, &argv, &error)) {
