diff -ru xf86-video-intel-2.6.3.orig/src/i830_driver.c xf86-video-intel-2.6.3/src/i830_driver.c
--- xf86-video-intel-2.6.3.orig/src/i830_driver.c	2009-03-02 19:21:02.000000000 +0000
+++ xf86-video-intel-2.6.3/src/i830_driver.c	2009-03-07 15:58:05.000000000 +0000
@@ -2951,7 +2951,6 @@
 	 batch_size = 4096;
 
       pI830->bufmgr = intel_bufmgr_gem_init(pI830->drmSubFD, batch_size);
-      intel_bufmgr_gem_enable_reuse(pI830->bufmgr);
    } else {
       assert(pI830->FbBase != NULL);
       pI830->bufmgr = intel_bufmgr_fake_init(pI830->drmSubFD,
