diff -Naur wxPython-src-2.8.11.0-orig/wxPython/setup.py wxPython-src-2.8.11.0/wxPython/setup.py
--- wxPython-src-2.8.11.0-orig/wxPython/setup.py	2010-06-04 04:15:47.000000000 -0400
+++ wxPython-src-2.8.11.0/wxPython/setup.py	2010-06-04 04:18:36.000000000 -0400
@@ -893,6 +893,7 @@
                     'wx.tools.Editra.src.eclib',
                     'wx.tools.Editra.src.ebmlib',
                     'wx.tools.Editra.src.extern',
+                    'wx.tools.Editra.src.extern.aui',
                     'wx.tools.Editra.src.syntax',
                     ]
 
