From 6eb49392fad8c76ade38fbd712509f0daa419a06 Mon Sep 17 00:00:00 2001
From: Jan de Groot <jan@jgc.homeip.net>
Date: Thu, 31 Mar 2011 12:27:37 +0200
Subject: [PATCH] Initialize exit status with EXIT_SUCCESS instead of -1. When things go wrong, the exit status is set to something else anyways. Fixes bug #646317

---
 src/terminal.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/terminal.c b/src/terminal.c
index 202b850..06de533 100644
--- a/src/terminal.c
+++ b/src/terminal.c
@@ -579,7 +579,7 @@ main (int argc, char **argv)
     data = g_new (OwnData, 1);
     data->factory_name = get_factory_name_for_display (display_name);
     data->options = options;
-    data->exit_code = -1;
+    data->exit_code = EXIT_SUCCESS;
     data->argv = argv_copy;
     data->argc = argc_copy;
 
-- 
1.7.4.2

