Index: /trunk/gtk/main.c
===================================================================
--- /trunk/gtk/main.c	(revision 12968)
+++ /trunk/gtk/main.c	(revision 13067)
@@ -73,4 +73,5 @@
     gboolean                    is_iconified;
 
+    guint                       activation_count;
     guint                       timer;
     guint                       update_model_soon_tag;
@@ -533,6 +534,14 @@
 
 static void
-on_activate( GApplication * app UNUSED, gpointer unused UNUSED )
-{
+on_activate( GApplication * app UNUSED, struct cbdata * cbdata )
+{
+    cbdata->activation_count++;
+
+    /* GApplication emits an 'activate' signal when bootstrapping the primary.
+     * Ordinarily we handle that by presenting the main window, but if the user
+     * user started Transmission minimized, ignore that initial signal... */
+    if( cbdata->is_iconified && ( cbdata->activation_count == 1 ) )
+        return;
+
     gtr_action_activate( "present-main-window" );
 }
