diff -ruN FTGL.org/include/FTTextureGlyph.h FTGL/include/FTTextureGlyph.h
--- FTGL.org/include/FTTextureGlyph.h	2006-07-11 17:49:42.000000000 +0200
+++ FTGL/include/FTTextureGlyph.h	2006-07-11 17:50:23.000000000 +0200
@@ -52,7 +52,7 @@
          * Reset the currently active texture to zero to get into a known state before
          * drawing a string. This is to get round possible threading issues.
          */
-        static void FTTextureGlyph::ResetActiveTexture(){ activeTextureID = 0;}
+        static void ResetActiveTexture(){ activeTextureID = 0;}
         
     private:
         /**
