--- decoder.c~	2011-05-08 09:28:28.077137883 +0200
+++ decoder.c	2011-05-10 21:40:48.887941968 +0200
@@ -259,6 +259,9 @@
 	for (i = 0; i < plugins_num; i++)
 		if (plugins[i].decoder->destroy)
 			plugins[i].decoder->destroy ();
+	for (i = 0; i < plugins_num; i++)
+		if (plugins[i].handle)
+                    lt_dlclose(plugins[i].handle);
 
 	if (lt_dlexit())
 		logit ("lt_exit() failed: %s", lt_dlerror());
