--- cdrdao-1.2.0/trackdb/FormatConverter.h	2005-06-30 13:35:59.000000000 +0200
+++ cdrdao-1.2.0.az/trackdb/FormatConverter.h	2005-06-30 13:34:18.000000000 +0200
@@ -95,7 +95,7 @@
 
   // Convert all files contained in a given Toc object, and update the
   // Toc accordingly. This is a big time blocking call.
-  FormatSupport::Status convert(Toc* toc);
+  FormatSupport::Status convert(class Toc* toc);
 
   // Dynamic allocator.
   FormatSupport* newConverter(const char* src);
--- cdrdao-1.2.0/trackdb/CueParser.cc	2005-06-30 13:36:09.000000000 +0200
+++ cdrdao-1.2.0.az/trackdb/CueParser.cc	2005-06-30 13:36:22.000000000 +0200
@@ -23,7 +23,7 @@
 
 #include "Cue2Toc.h"
 
-extern Toc *parseToc(const char* tocBuffer, const char *filename);
+extern class Toc *parseToc(const char* tocBuffer, const char *filename);
 
 Toc *parseCue(FILE *fp, const char *filename)
 {

