--- gnome-power-manager-2.26.1/src/gpm-session.c	2009-04-21 12:43:38.000000000 +0200
+++ gnome-power-manager-2.26.1/src/gpm-session.c	2009-05-08 21:45:43.000000000 +0200
@@ -254,6 +254,8 @@
 	DBusGConnection *connection;
 
 	g_return_val_if_fail (GPM_IS_SESSION (session), FALSE);
+	if (session->priv->proxy == NULL)
+		return FALSE;
 
 	/* find out if this change altered the inhibited state */
 	ret = dbus_g_proxy_call (session->priv->proxy, "RegisterClient", &error,
