--- gnome-session-2.18.0/gnome-session/gsm-dbus.c	2007-03-12 19:48:31.000000000 +0100
+++ gnome-session-2.16.2/gnome-session/gsm-dbus.c	2007-03-24 23:22:17.000000000 +0100
@@ -25,7 +25,6 @@
 #include <sys/wait.h>
 
 #include <errno.h>
-#include <fcntl.h>
 #include <limits.h>
 #include <signal.h>
 #include <stdlib.h>
@@ -195,9 +194,6 @@
   do
     {
       ssize_t i, result;
-      ssize_t oldbytes;
-
-      oldbytes = bytes;
 
       result = read (fd, &buf[bytes], bufsize - bytes);
       if (result < 0)
@@ -212,12 +208,11 @@
       else
         done = TRUE;
 
-      for (i = oldbytes; !discard && i < bytes; i++)
+      for (i = 0; !discard && i < bytes; i++)
         if (buf[i] == '\n')
           {
             buf[i] = '\0';
             discard = TRUE;
-	    done = TRUE;
           }
     }
   while (!done);
@@ -247,7 +242,7 @@
     {
       if (fd != STDIN_FILENO && fd != STDOUT_FILENO && fd != STDERR_FILENO &&
           fd != address_fd && fd != pid_fd)
-        fcntl (fd, F_SETFD, FD_CLOEXEC);
+        close (fd);
     }
 
   g_snprintf (address_str, sizeof (address_str), "%d", address_fd);
@@ -344,7 +339,7 @@
    * strtoul dance to check for a valid number.
    */
   errno = 0;
-  tmp_num = strtoul (pid_str, &tmp_ep, 10);
+  tmp_num = strtoul(pid_str, &tmp_ep, 10);
   if (pid_str[0] == '\0' || *tmp_ep != '\0')
     {
       g_printerr ("dbus-daemon pid invalid (not a number)\n");
