From a01e33e6db788bb5a3aaf310d3f64aa25c0b842a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Sebasti=C3=A1n=20E=2E=20Peyrott?= <pseus7@gmail.com>
Date: Tue, 15 Oct 2013 19:35:24 -0300
Subject: [PATCH] Fixed double-free bug. Fixes bug 709967.

---
 gtk/gtkicontheme.c | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/gtk/gtkicontheme.c b/gtk/gtkicontheme.c
index fad7ff4..2be0f75 100644
--- a/gtk/gtkicontheme.c
+++ b/gtk/gtkicontheme.c
@@ -3873,7 +3873,14 @@ gtk_icon_info_load_icon (GtkIconInfo *icon_info,
   if (!icon_info_ensure_scale_and_pixbuf (icon_info, FALSE))
     {
       if (icon_info->load_error)
-        g_propagate_error (error, icon_info->load_error);
+        {
+          /* g_propagate_error would be wrong here as
+           * icon_info->load_error gets freed in the
+           * finalizer.
+           */
+          if (error)
+            *error = g_error_copy (icon_info->load_error);
+        }
       else
         g_set_error_literal (error,  
                              GTK_ICON_THEME_ERROR,  
-- 
1.8.4

