--- gnumeric-1.6.1/src/wbcg-actions.c	2005-11-14 07:12:35.000000000 +0100
+++ gnumeric-1.6.1/src/wbcg-actions.c.new	2006-01-13 19:23:17.000000000 +0100
@@ -909,8 +909,10 @@
 #warning "handle translations when we generate them"
 #ifndef G_OS_WIN32
 	argv[0] = (char *)"yelp";
-	argv[1] = g_build_filename (gnm_sys_data_dir (), "doc", "C",
-			"gnumeric.xml", NULL);
+	/*argv[1] = g_build_filename (gnm_sys_data_dir (), "doc", "C",
+			"gnumeric.xml", NULL);*/
+	argv[1] = ("/usr/share/gnome/help/gnumeric/C/gnumeric.xml");
+	
 #else
 	argv[0] = (char *)"hh";
 	argv[1] = g_build_filename (gnm_sys_data_dir (), "doc", "C",
@@ -918,7 +920,7 @@
 #endif
 	g_spawn_async (NULL, argv, NULL, G_SPAWN_SEARCH_PATH,
 		NULL, NULL, NULL, &error);
-	g_free (argv[1]);
+	/*g_free (argv[1]);*/
 }
 
 static GNM_ACTION_DEF (cb_help_web)
