From: Roman Kosenko <madkite@gmail.com>
Date: Thu, 10 Mar 2011 16:23:53 +0200
Subject: [PATCH] Fix stupid bug with --sort-key

--- htop.c.old	2010-11-24 20:45:38.000000000 +0200
+++ htop.c	2011-03-10 16:32:18.473716667 +0200
@@ -282,7 +282,7 @@
             printVersionFlag();
             break;
          case 's':
-            if (strcmp(optarg, "help")) {
+            if (!strcmp(optarg, "help")) {
                for (int j = 1; j < LAST_PROCESSFIELD; j++)
                   printf ("%s\n", Process_fieldNames[j]);
                exit(0);
