--- intltool-merge.in.old	2007-09-30 17:25:14.000000000 -0500
+++ intltool-merge.in	2007-09-30 17:25:41.000000000 -0500
@@ -93,7 +93,7 @@
 
 my %po_files_by_lang = ();
 my %translations = ();
-my $iconv = $ENV{"ICONV"} || $ENV{"INTLTOOL_ICONV"} || "@INTLTOOL_ICONV@";
+my $iconv = $ENV{"ICONV"} || "iconv";
 my $devnull = ($^O eq 'MSWin32' ? 'NUL:' : '/dev/null');
 
 # Use this instead of \w for XML files to handle more possible characters.
