From 8bb278944496102a4afad46b53e1e8279425aaac Mon Sep 17 00:00:00 2001
From: Jiří Klimeš <jklimes@redhat.com>
Date: Mon, 02 Apr 2012 14:37:55 +0000
Subject: libnm-glib: initialize GError, else invalid free() crash can occur (rh #809123)

---
(limited to 'libnm-glib/nm-remote-settings.c')

diff --git a/libnm-glib/nm-remote-settings.c b/libnm-glib/nm-remote-settings.c
index 9a81a8d..68aaeb4 100644
--- a/libnm-glib/nm-remote-settings.c
+++ b/libnm-glib/nm-remote-settings.c
@@ -109,7 +109,7 @@ static void
 _nm_remote_settings_ensure_inited (NMRemoteSettings *self)
 {
 	NMRemoteSettingsPrivate *priv = NM_REMOTE_SETTINGS_GET_PRIVATE (self);
-	GError *error;
+	GError *error = NULL;
 
 	if (!priv->inited) {
 		if (!g_initable_init (G_INITABLE (self), NULL, &error)) {
--
cgit v0.9.0.2-2-gbebe
