diff -Naur kdeplasma-addons-4.6.5.bak/applets/kdeobservatory/src/commithistoryview.cpp kdeplasma-addons-4.6.5/applets/kdeobservatory/src/commithistoryview.cpp
--- kdeplasma-addons-4.6.5.bak/applets/kdeobservatory/src/commithistoryview.cpp	2011-07-15 14:04:17.323188525 +0200
+++ kdeplasma-addons-4.6.5/applets/kdeobservatory/src/commithistoryview.cpp	2011-07-15 14:04:39.186521458 +0200
@@ -124,7 +124,7 @@
         plot->setCanvasBackground(QColor(0, 0, 140));
 
         QwtPlotCurve *curve = new QwtPlotCurve;
-        curve->setData(x, y, j);
+        curve->setSamples(x, y, j);
         delete []x;
         delete []y;
 
