--- kvm-72/qemu/hw/pc.c.orig	2008-05-06 20:48:35.468799015 +0200
+++ kvm-72/qemu/hw/pc.c	2008-05-06 20:50:00.392123267 +0200
@@ -37,10 +37,10 @@
 /* output Bochs bios info messages */
 //#define DEBUG_BIOS
 
-#define BIOS_FILENAME "bios.bin"
-#define VGABIOS_FILENAME "vgabios.bin"
-#define VGABIOS_CIRRUS_FILENAME "vgabios-cirrus.bin"
-#define EXTBOOT_FILENAME "extboot.bin"
+#define BIOS_FILENAME "bios-kvm.bin"
+#define VGABIOS_FILENAME "vgabios-kvm.bin"
+#define VGABIOS_CIRRUS_FILENAME "vgabios-cirrus-kvm.bin"
+#define EXTBOOT_FILENAME "extboot-kvm.bin"
 
 #define PC_MAX_BIOS_SIZE (4 * 1024 * 1024)

