--- browser/app/mozilla.in.old	2005-01-01 14:12:53.000000000 +0000
+++ browser/app/mozilla.in	2005-01-01 14:13:19.000000000 +0000
@@ -109,6 +109,7 @@
 if [ $found = 0 ]; then
   # Check default compile-time libdir
   if [ -x "$moz_libdir/run-mozilla.sh" ]; then
+    run_moz="$moz_libdir/run-mozilla.sh"
     dist_bin="$moz_libdir"
   else 
     echo "Cannot find mozilla runtime directory. Exiting."
