--- source/texk/web2c/xetexdir/pngimage.c.orig	2010-01-20 18:50:16.000000000 -0500
+++ source/texk/web2c/xetexdir/pngimage.c	2010-01-20 18:56:45.000000000 -0500
@@ -101,7 +101,7 @@
   rewind (png_file);
   if (fread (sigbytes, 1, sizeof(sigbytes), png_file) !=
       sizeof(sigbytes) ||
-      (!png_check_sig (sigbytes, sizeof(sigbytes))))
+      (png_sig_cmp (sigbytes, 0, sizeof(sigbytes))))
     return 0;
   else
     return 1;
--- source/texk/dvipdfmx/src/pngimage.c.orig	2010-01-20 19:55:20.000000000 -0500
+++ source/texk/dvipdfmx/src/pngimage.c	2010-01-20 19:56:39.000000000 -0500
@@ -148,7 +148,7 @@
   rewind (png_file);
   if (fread (sigbytes, 1, sizeof(sigbytes), png_file) !=
       sizeof(sigbytes) ||
-      (!png_check_sig (sigbytes, sizeof(sigbytes))))
+      (png_sig_cmp (sigbytes, 0, sizeof(sigbytes))))
     return 0;
   else
     return 1;
--- source/texk/xdvipdfmx/src/pngimage.c.orig	2010-01-20 19:59:18.000000000 -0500
+++ source/texk/xdvipdfmx/src/pngimage.c	2010-01-20 19:59:53.000000000 -0500
@@ -146,7 +146,7 @@
   rewind (png_file);
   if (fread (sigbytes, 1, sizeof(sigbytes), png_file) !=
       sizeof(sigbytes) ||
-      (!png_check_sig (sigbytes, sizeof(sigbytes))))
+      (png_sig_cmp (sigbytes, 0, sizeof(sigbytes))))
     return 0;
   else
     return 1;
--- source/texk/xdvipdfmx/src/xbb.c.orig	2010-01-20 20:45:24.000000000 -0500
+++ source/texk/xdvipdfmx/src/xbb.c	2010-01-20 20:46:07.000000000 -0500
@@ -246,7 +246,7 @@
   unsigned char sigbytes[4];
   rewind (png_file);
   if (fread(sigbytes, 1, sizeof(sigbytes), png_file) != sizeof(sigbytes) ||
-      (!png_check_sig (sigbytes, sizeof(sigbytes)))) return 0;
+      (png_sig_cmp (sigbytes, 0, sizeof(sigbytes)))) return 0;
   else return 1;
 }             
 
