diff -u -r rtkit-0.11/configure.ac rtkit-0.11-sd/configure.ac
--- rtkit-0.11/configure.ac	2012-05-15 15:25:40.000000000 +0200
+++ rtkit-0.11-sd/configure.ac	2013-05-13 08:12:17.616825455 +0200
@@ -115,6 +115,7 @@
 AC_SEARCH_LIBS([cap_init], [cap])
 
 PKG_CHECK_MODULES(DBUS, dbus-1)
+PKG_CHECK_MODULES(LIBSYSTEMD_DAEMON, libsystemd-daemon)
 
 AC_ARG_WITH([systemdsystemunitdir],
         AS_HELP_STRING([--with-systemdsystemunitdir=DIR], [Directory for systemd service files]),
diff -u -r rtkit-0.11/Makefile.am rtkit-0.11-sd/Makefile.am
--- rtkit-0.11/Makefile.am	2012-05-15 15:38:05.000000000 +0200
+++ rtkit-0.11-sd/Makefile.am	2013-05-13 08:12:18.086822253 +0200
@@ -56,13 +56,14 @@
 endif
 
 rtkit_daemon_SOURCES = \
-	rtkit-daemon.c rtkit.h \
-	sd-daemon.c sd-daemon.h
+	rtkit-daemon.c rtkit.h
 rtkit_daemon_LDADD = \
-	$(DBUS_LIBS)
+	$(DBUS_LIBS) \
+	$(LIBSYSTEMD_DAEMON_LIBS)
 rtkit_daemon_CFLAGS = \
 	$(AM_CFLAGS) \
-	$(DBUS_CFLAGS)
+	$(DBUS_CFLAGS) \
+	$(LIBSYSTEMD_DAEMON_CFLAGS)
 
 rtkitctl_SOURCES = \
 	rtkitctl.c rtkit.h
@@ -93,7 +94,3 @@
 
 DISTCHECK_CONFIGURE_FLAGS = \
 	--with-systemdsystemunitdir=$$dc_install_base/$(systemdsystemunitdir)
-
-update-systemd:
-	curl http://cgit.freedesktop.org/systemd/systemd/plain/src/libsystemd-daemon/sd-daemon.c > sd-daemon.c
-	curl http://cgit.freedesktop.org/systemd/systemd/plain/src/systemd/sd-daemon.h > sd-daemon.h
diff -u -r rtkit-0.11/rtkit-daemon.c rtkit-0.11-sd/rtkit-daemon.c
--- rtkit-0.11/rtkit-daemon.c	2012-05-15 15:25:40.000000000 +0200
+++ rtkit-0.11-sd/rtkit-daemon.c	2013-05-13 08:13:07.933149359 +0200
@@ -50,9 +50,9 @@
 #include <dirent.h>
 #include <syslog.h>
 #include <grp.h>
+#include <systemd/sd-daemon.h>
 
 #include "rtkit.h"
-#include "sd-daemon.h"
 
 #ifndef __linux__
 #error "This stuff only works on Linux!"
Only in rtkit-0.11: sd-daemon.c
Only in rtkit-0.11: sd-daemon.h
