--- upnp/src/soap/soap_ctrlpt.c	2008-02-10 02:55:33.000000000 +0100
+++ upnp/src/soap/soap_ctrlpt.c	2009-01-29 21:09:41.000000000 +0100
@@ -931,7 +931,7 @@
     content_length = strlen( xml_start ) + strlen( var_name ) + strlen( xml_end );
     if (http_MakeMessage(
 	&request, 1, 1,
-	"Q" "sbc" "N" "s" "s" "Ucc" "sss",
+	"Q" "sbc" "N" "s" "sc" "Ucc" "sss",
 	SOAPMETHOD_POST, path.buf, path.length,
 	"HOST: ", host.buf, host.length,
 	content_length,

 	  	 
