diff -ru libX11-1.1.99.2.orig/src/xcb_disp.c libX11-1.1.99.2/src/xcb_disp.c
--- libX11-1.1.99.2.orig/src/xcb_disp.c	2009-02-02 11:53:49.000000000 +0000
+++ libX11-1.1.99.2/src/xcb_disp.c	2009-02-02 11:57:55.000000000 +0000
@@ -94,6 +94,9 @@
 	dpy->xcb->next_xid = xcb_generate_id(dpy->xcb->connection);
 
 	dpy->xcb->event_notify = xcondition_malloc();
+	if (!dpy->xcb->event_notify)
+		return 0;
+	xcondition_init(dpy->xcb->event_notify);
 	return !xcb_connection_has_error(c);
 }
 
