diff -upr llvm-3.5.0.src.orig/bindings/ocaml/Makefile.ocaml llvm-3.5.0.src/bindings/ocaml/Makefile.ocaml
--- llvm-3.5.0.src.orig/bindings/ocaml/Makefile.ocaml	2014-12-23 12:28:27.000000000 +0200
+++ llvm-3.5.0.src/bindings/ocaml/Makefile.ocaml	2014-12-23 12:35:36.000000000 +0200
@@ -243,8 +243,8 @@ uninstall-local:: uninstall-shared
 
 $(SharedLib): $(ObjectsO) $(OcamlDir)/.dir
 	$(Echo) "Building $(BuildMode) $(notdir $@)"
-	$(Verb) $(Link) $(SharedLinkOptions) $(OCAMLRPATH) $(LLVMLibsOptions) \
-			-o $@ $(ObjectsO)
+	$(Verb) $(Link) $(SharedLinkOptions) $(OCAMLRPATH) -o $@ $(ObjectsO) \
+			$(LLVMLibsOptions)
 
 clean-shared::
 	-$(Verb) $(RM) -f $(SharedLib)
