--- term/lua/gnuplot-tikz-444.lua	Sun Nov 13 00:57:15 2011
+++ term/lua/gnuplot-tikz.lua	Fri Nov 25 01:11:38 2011
@@ -251,6 +251,7 @@
   gp.write(string.format("\\begin{tikzpicture}[gnuplot%s]\n",global_opt))
   gp.write(string.format("%%%% generated with GNUPLOT %sp%s (%s; terminal rev. %s, script rev. %s)\n%%%% %s\n",
       term.gp_version, term.gp_patchlevel,
+      _VERSION,
       string.sub(term.lua_term_revision,7,-3),
       pgf.REVISION,os.date()))
   if font ~= "" then