--- luatex-beta-0.43.0/source/texk/web2c/luatexdir/image/pdftoepdf.cc.orig	2009-10-10 23:33:01.000000000 +0200
+++ luatex-beta-0.43.0/source/texk/web2c/luatexdir/image/pdftoepdf.cc	2009-10-10 23:34:23.000000000 +0200
@@ -672,7 +672,7 @@
     }
     for (r = encodingList; r != NULL; r = n) {
         n = r->next;
-        delete r->font;
+        //delete r->font;
         delete r;
     }
 }
@@ -712,6 +712,7 @@
     Page *page;
     int rotate;
     PDFRectangle *pagebox;
+    int minor_pdf_version_found, major_pdf_version_found;
     float pdf_version_found, pdf_version_wanted, xsize, ysize, xorig, yorig;
     assert(idict != NULL);
     assert(img_type(idict) == IMG_TYPE_PDF);
@@ -727,7 +728,9 @@
     // this works only for PDF 1.x -- but since any versions of PDF newer
     // than 1.x will not be backwards compatible to PDF 1.x, pdfTeX will
     // then have to changed drastically anyway.
-    pdf_version_found = pdf_doc->doc->getPDFVersion();
+    minor_pdf_version_found = pdf_doc->doc->getPDFMinorVersion();
+    major_pdf_version_found = pdf_doc->doc->getPDFMajorVersion();
+    pdf_version_found = major_pdf_version_found + (minor_pdf_version_found * 0.1);
     pdf_version_wanted = 1 + (minor_pdf_version_wanted * 0.1);
     if (pdf_version_found > pdf_version_wanted + 0.01) {
         char msg[] =
