--- lynx2-8-8/WWW/Library/Implementation/HTTP.c.orig	2017-03-16 09:24:30.608537810 +0000
+++ lynx2-8-8/WWW/Library/Implementation/HTTP.c	2017-03-16 09:25:21.728457895 +0000
@@ -720,7 +720,7 @@
 #elif SSLEAY_VERSION_NUMBER >= 0x0900
 #ifndef USE_NSS_COMPAT_INCL
 	if (!try_tls) {
-	    handle->options |= SSL_OP_NO_TLSv1;
+	    SSL_set_options(handle, SSL_OP_NO_TLSv1);
 #if OPENSSL_VERSION_NUMBER >= 0x0090806fL && !defined(OPENSSL_NO_TLSEXT)
 	} else {
 	    int ret = (int) SSL_set_tlsext_host_name(handle, ssl_host);
