diff -p -up meld-1.2/help/fr/GNUmakefile~ meld-1.2/help/fr/GNUmakefile
--- meld-1.2/help/fr/GNUmakefile~	2008-03-31 22:44:53.000000000 +0200
+++ meld-1.2/help/fr/GNUmakefile	2008-06-27 10:37:54.000000000 +0200
@@ -15,9 +15,8 @@ all $(OMF_NAME).install : $(OMF_NAME)
 
 .PHONY : install
 install : $(OMF_NAME).install
-	-mkdir -m 755 -p $(OMF_DIR_) $(XML_DIR_) $(XML_DIR_)/figures
+	-mkdir -m 755 -p $(OMF_DIR_) $(XML_DIR_)
 	install -m 644 meld.$(LANG).xml $(XML_DIR_)/meld.xml
-	install -m 644 figures/*.png $(XML_DIR_)/figures
 	-install -m 644 $< $(OMF_DIR_)/$(OMF_NAME)
 	-scrollkeeper-update -p $(OMF_STATE) -o $(OMF_DIR_)
 
