Index: decoder_plugins/ffmpeg/ffmpeg.c
===================================================================
--- decoder_plugins/ffmpeg/ffmpeg.c	(revision 2307)
+++ decoder_plugins/ffmpeg/ffmpeg.c	(working copy)
@@ -28,12 +28,6 @@
 #include <ffmpeg/avformat.h>
 #endif
 
-/* libavformat's API will be changing at version 53, but at present there
- * appears to be no guidance on what will replace the deprecated fields. */
-#ifndef FF_API_OLD_METADATA
-#define FF_API_OLD_METADATA            (LIBAVFORMAT_VERSION_MAJOR < 53)
-#endif
-
 /* FFmpeg also likes common names, without that, our common.h and log.h
  * would not be included. */
 #undef COMMON_H
@@ -137,7 +131,7 @@
 	av_read_play (data->ic);
 	for (i = 0; i < data->ic->nb_streams; i++) {
 		data->enc = data->ic->streams[i]->codec;
-		if (data->enc->codec_type == CODEC_TYPE_AUDIO) {
+		if (data->enc->codec_type == AVMEDIA_TYPE_AUDIO) {
 			audio_index = i;
 			break;
 		}
