diff -ur easytag-2.1.6.orig/configure easytag-2.1.6/configure
--- easytag-2.1.6.orig/configure	2008-07-12 17:57:44.000000000 +0300
+++ easytag-2.1.6/configure	2009-07-10 16:44:39.000000000 +0300
@@ -22418,12 +22418,12 @@
 /* end confdefs.h.  */
 
                 #include <stdio.h>
-                #include <mpeg4ip_config.h>
+                #include <mp4v2/mp4v2.h>
                 int main (void)
                 {
                     FILE *output;
                     output=fopen("conftest.mp4v2","w");
-                    fprintf(output,"LIBMP4V2_NAME=%s\nLIBMP4V2_VERSION=%s\n",PACKAGE_TARNAME,PACKAGE_VERSION);
+                    fprintf(output,"MP4V2_PROJECT_name=%s\nMP4V2_PROJECT_version=%s\n",PACKAGE_TARNAME,PACKAGE_VERSION);
                     fclose(output);
                     exit(0);
                 }
@@ -22471,7 +22471,7 @@
 
                 cat >conftest.$ac_ext <<_ACEOF
 
-           #include <mp4.h>
+           #include <mp4v2/mp4v2.h>
            main () {
            MP4FileHandle hFile;
            MP4GetMetadataCoverArt(hFile, NULL, NULL,0); }
diff -ur easytag-2.1.6.orig/src/mp4_header.c easytag-2.1.6/src/mp4_header.c
--- easytag-2.1.6.orig/src/mp4_header.c	2007-11-28 10:54:35.000000000 +0200
+++ easytag-2.1.6/src/mp4_header.c	2009-07-10 16:44:54.000000000 +0300
@@ -44,7 +44,7 @@
 #undef PACKAGE_STRING
 #undef PACKAGE_TARNAME
 #undef PACKAGE_VERSION
-#include <mp4.h>
+#include <mp4v2/mp4v2.h>
 
 
 /****************
diff -ur easytag-2.1.6.orig/src/mp4_tag.c easytag-2.1.6/src/mp4_tag.c
--- easytag-2.1.6.orig/src/mp4_tag.c	2007-12-11 00:08:11.000000000 +0200
+++ easytag-2.1.6/src/mp4_tag.c	2009-07-10 16:45:08.000000000 +0300
@@ -50,7 +50,7 @@
 #undef PACKAGE_STRING
 #undef PACKAGE_TARNAME
 #undef PACKAGE_VERSION
-#include <mp4.h>
+#include <mp4v2/mp4v2.h>
 
 
 /****************
