diff --git i/cert-to-efi-hash-list.c w/cert-to-efi-hash-list.c
index d4484f9b8e6535f3..399ba704eaab238d 100644
--- i/cert-to-efi-hash-list.c
+++ w/cert-to-efi-hash-list.c
@@ -135,7 +135,7 @@ main(int argc, char *argv[])
         X509 *cert = PEM_read_bio_X509(cert_bio, NULL, NULL, NULL);
 	unsigned char *cert_buf = NULL;
 
-	int cert_len = i2d_X509_CINF(cert->cert_info, &cert_buf);
+	int cert_len = i2d_re_X509_tbs(cert, &cert_buf);
 	ERR_print_errors_fp(stdout);
 
 	int len, digest_len, time_offset;
