Index: modules/highgui/src/cap.cpp
===================================================================
--- modules/highgui/src/cap.cpp	(revision 4283)
+++ modules/highgui/src/cap.cpp	(working copy)
@@ -52,10 +52,10 @@
 namespace cv
 {
 
-template<> inline void Ptr<CvCapture>::delete_obj()
+template<> void Ptr<CvCapture>::delete_obj()
 { cvReleaseCapture(&obj); }
 
-template<> inline void Ptr<CvVideoWriter>::delete_obj()
+template<> void Ptr<CvVideoWriter>::delete_obj()
 { cvReleaseVideoWriter(&obj); }
 
 }
