diff -up wafadmin/Tools/python.py.pypath wafadmin/Tools/python.py
--- wafadmin/Tools/python.py.pypath	2011-02-10 15:30:44.118506004 -0500
+++ wafadmin/Tools/python.py	2011-02-10 15:32:50.922506002 -0500
@@ -156,6 +156,9 @@ MACOSX_DEPLOYMENT_TARGET = %r
 		env.append_value('LINKFLAGS_PYEMBED',python_LDFLAGS.split())
 	result=False
 	name='python'+env['PYTHON_VERSION']
+	python_config=conf.find_program('python%s-config'%('.'.join(env['PYTHON_VERSION'].split('.')[:2])),var='PYTHON_CONFIG')
+	if not python_config:
+		python_config=conf.find_program('python-config-%s'%('.'.join(env['PYTHON_VERSION'].split('.')[:2])),var='PYTHON_CONFIG')
 	if python_LIBDIR is not None:
 		path=[python_LIBDIR]
 		conf.log.write("\n\n# Trying LIBDIR: %r\n"%path)
@@ -169,6 +172,12 @@ MACOSX_DEPLOYMENT_TARGET = %r
 		path=[os.path.join(python_prefix,"libs")]
 		name='python'+env['PYTHON_VERSION'].replace('.','')
 		result=conf.check(lib=name,uselib='PYEMBED',libpath=path)
+	if not result and python_config:
+		conf.log.write("\n\n# try again with help of python-config\n")
+		for libstr in Utils.cmd_output("%s %s --libs"%(python,python_config)).strip().split():
+			if(libstr.startswith('-lpython')):
+				name=libstr[2:]
+		result=conf.check(lib=name,uselib='PYEMBED',libpath=path)
 	if result:
 		env['LIBPATH_PYEMBED']=path
 		env.append_value('LIB_PYEMBED',name)
@@ -177,9 +186,6 @@ MACOSX_DEPLOYMENT_TARGET = %r
 	if(sys.platform=='win32'or sys.platform.startswith('os2')or sys.platform=='darwin'or Py_ENABLE_SHARED):
 		env['LIBPATH_PYEXT']=env['LIBPATH_PYEMBED']
 		env['LIB_PYEXT']=env['LIB_PYEMBED']
-	python_config=conf.find_program('python%s-config'%('.'.join(env['PYTHON_VERSION'].split('.')[:2])),var='PYTHON_CONFIG')
-	if not python_config:
-		python_config=conf.find_program('python-config-%s'%('.'.join(env['PYTHON_VERSION'].split('.')[:2])),var='PYTHON_CONFIG')
 	includes=[]
 	if python_config:
 		for incstr in Utils.cmd_output("%s %s --includes"%(python,python_config)).strip().split():
