--- akonadi-1.5.1/libs/xdgbasedirs.cpp~	2011-03-18 02:18:19.543333786 -0700
+++ akonadi-1.5.1/libs/xdgbasedirs.cpp	2011-03-18 02:19:15.893335157 -0700
@@ -313,7 +313,7 @@
 QString XdgBaseDirs::findPluginFile( const QString &relPath, const QStringList &searchPath )
 {
   if ( instance()->mPluginDirs.isEmpty() ) {
-    QStringList pluginDirs = instance()->systemPathList( "QT_PLUGIN_PATH", AKONADILIB ":" AKONADILIB "/qt4/plugins/:" AKONADILIB "/kde4/:" AKONADILIB "/kde4/plugins/:/usr/lib/qt4/plugins/" );
+    QStringList pluginDirs = instance()->systemPathList( "QT_PLUGIN_PATH", AKONADILIB ":" AKONADILIB "/qt/plugins/:" AKONADILIB "/kde4/:" AKONADILIB "/kde4/plugins/:/usr/lib/qt/plugins/" );
 
     if ( QCoreApplication::instance() != 0 ) {
       foreach ( const QString &libraryPath, QCoreApplication::instance()->libraryPaths() ) {
--- akonadi-1.5.1/qsqlite/CMakeLists.txt~	2011-03-18 02:22:50.976674875 -0700
+++ akonadi-1.5.1/qsqlite/CMakeLists.txt	2011-03-18 02:23:07.370000620 -0700
@@ -14,8 +14,8 @@
 if (INSTALL_QSQLITE_IN_QT_PREFIX)
   set(QSQLITE_INSTALL_PREFIX "${QT_PLUGINS_DIR}/sqldrivers")
 else()
-  set(QSQLITE_INSTALL_PREFIX "${LIB_INSTALL_DIR}/qt4/plugins/sqldrivers")
-  message(STATUS "Make sure that you add ${LIB_INSTALL_DIR}/qt4/plugins to your QT_PLUGIN_PATH environment variable.")
+  set(QSQLITE_INSTALL_PREFIX "${LIB_INSTALL_DIR}/qt/plugins/sqldrivers")
+  message(STATUS "Make sure that you add ${LIB_INSTALL_DIR}/qt/plugins to your QT_PLUGIN_PATH environment variable.")
 endif()
 
 add_definitions(-Wall)
