diff -upr texlive-20080816-source.orig/texk/afm2pl/afm2pl.c texlive-20080816-source/texk/afm2pl/afm2pl.c
--- texlive-20080816-source.orig/texk/afm2pl/afm2pl.c	2006-01-18 01:41:51.000000000 +0400
+++ texlive-20080816-source/texk/afm2pl/afm2pl.c	2009-05-19 23:44:11.624309901 +0500
@@ -345,7 +345,7 @@ P2C (register int, x, register int, y)
 
 /* read a line from infile into buffer and obuffer */
 int
-  getline
+  _getline
 P1H (void)
 {
   register char *p;
@@ -781,7 +781,7 @@ P1H (void)
     afmencoding->vec[i] = ".notdef";
   afmencoding->name = "Unspecified";
 
-  while (getline ()) {
+  while (_getline ()) {
     switch (interest (paramstring ())) {
     case FontName:
       fontname = paramnewstring ();
@@ -963,7 +963,7 @@ gettoken ()
 
   while (1) {
     while (param == 0 || *param == 0) {
-      if (getline () == 0)
+      if (_getline () == 0)
         error ("! premature end in encoding file");
     }
     if (param[0] == '%') {
@@ -1540,7 +1540,7 @@ extraligkerninfo ()
     encfilename =
       openin (encfilename, kpse_enc_format, ".enc");
     for (lig_it = 0; lig_it < 2; lig_it++) {
-      while (getline ()) {
+      while (_getline ()) {
         /* search for (ligkern) comment in line */
         for (p = buffer; *p; p++)
           if (*p == '%') {
@@ -1576,7 +1576,7 @@ extraligkerninfo ()
     for (i = 0; i < ligfilenames->n; i++) {
       ligfilename = openin (ligfilenames->names[i],
                             kpse_lig_format, ".lig");
-      while (getline ())
+      while (_getline ())
         checkligkern (buffer, 0);       /* 2nd param 0: lig file */
       fclose (infile);
     }
@@ -1607,7 +1607,7 @@ extraligkerninfo ()
   for (i = 0; i < Ligfilenames->n; i++) {
     ligfilename = openin (Ligfilenames->names[i],
                           kpse_lig_format, ".lig");
-    while (getline ())
+    while (_getline ())
       checkligkern (buffer, 0);       /* 2nd param 0: lig file */
     fclose (infile);
   }
diff -upr texlive-20080816-source.orig/texk/dvipsk/afm2tfm.c texlive-20080816-source/texk/dvipsk/afm2tfm.c
--- texlive-20080816-source.orig/texk/dvipsk/afm2tfm.c	2008-07-10 04:40:36.000000000 +0500
+++ texlive-20080816-source/texk/dvipsk/afm2tfm.c	2009-05-19 23:44:11.624309901 +0500
@@ -260,7 +260,7 @@ transform P2C(register int, x, register 
 }
 
 int
-getline P1H(void) {
+_getline P1H(void) {
    register char *p ;
    register int c ;
 
@@ -615,7 +615,7 @@ readadobe P1H(void) {
    ai = newchar() ;
    ai->adobenum = -1 ;
    ai->adobename = "||" ; /* boundary character name */
-   while (getline()) {
+   while (_getline()) {
       switch(interest(paramstring())) {
 case FontName:
          fontname = paramnewstring() ;
@@ -1891,7 +1891,7 @@ char *gettoken() {
 
    while (1) {
       while (param == 0 || *param == 0) {
-         if (getline() == 0)
+         if (_getline() == 0)
             error("! premature end in encoding file") ;
          for (p=buffer; *p; p++)
             if (*p == '%') {
@@ -1982,7 +1982,7 @@ struct encoding *readencoding P1C(char *
       p = gettoken() ;
       if (strcmp(p, "]"))
          error("! token 258 in encoding must be make-array (])") ;
-      while (getline()) {
+      while (_getline()) {
          for (p=buffer; *p; p++)
             if (*p == '%') {
                if (ignoreligkern == 0)
Only in texlive-20080816-source/texk/dvipsk: afm2tfm.c.orig
diff -upr texlive-20080816-source.orig/texk/web2c/cpascal.h texlive-20080816-source/texk/web2c/cpascal.h
--- texlive-20080816-source.orig/texk/web2c/cpascal.h	2008-05-25 04:56:44.000000000 +0500
+++ texlive-20080816-source/texk/web2c/cpascal.h	2009-05-19 23:44:11.624309901 +0500
@@ -14,6 +14,7 @@
 
 /* We must include this first, to resolve many C issues.  */
 #include "config.h"
+#define getline _getline
 
 /* We only use getopt in the applications, not in web2c itself.  */
 #include <kpathsea/getopt.h>
diff -upr texlive-20080816-source.orig/texk/web2c/mpware/mpto.c texlive-20080816-source/texk/web2c/mpware/mpto.c
--- texlive-20080816-source.orig/texk/web2c/mpware/mpto.c	2008-04-11 21:33:03.000000000 +0500
+++ texlive-20080816-source/texk/web2c/mpware/mpto.c	2009-05-19 23:44:45.744525169 +0500
@@ -97,7 +97,7 @@ usage(char *progn)
 }
 
 char *
-getline(void)
+_getline(void)
 {				/* returns NULL on EOF or error, otherwise buf */
     int c;
     unsigned loc = 0;
@@ -302,7 +302,7 @@ copytex(void)
     char *res = NULL;
     do {
 	if (*aa == 0)
-	  if ((aa = getline()) == NULL)
+	  if ((aa = _getline()) == NULL)
 	    err("btex section does not end");
 
 	if (getbta(aa) && *tt == 'e') {
@@ -483,7 +483,7 @@ Current maintainer: Taco Hoekwater.\n");
 	postverb = troff_postverb;
     }
     printf("%s", predoc);
-    while (getline() != NULL)
+    while (_getline() != NULL)
 	do_line();
     printf("%s", postdoc);
     exit(0);
--- texlive-20080816-source/texk/dvi2tty/dvistuff.c~	2006-01-17 16:41:51.000000000 -0500
+++ texlive-20080816-source/texk/dvi2tty/dvistuff.c	2009-06-20 17:13:57.000000000 -0400
@@ -123,7 +123,7 @@
 void            ruleaux         P3H(long, long, char);
 long            horizontalmove  P1H(long);
 int             skipnops        P1H(void);
-linetype    *   getline         P1H(void);
+linetype    *   _getline         P1H(void);
 linetype    *   findline        P1H(void);
 unsigned long   num             P1H(int);
 long            snum            P1H(int);
@@ -149,7 +149,7 @@
 void            ruleaux         (long, long, char);
 long            horizontalmove  (long);
 int             skipnops        (void);
-linetype    *   getline         (void);
+linetype    *   _getline         (void);
 linetype    *   findline        (void);
 unsigned long   num             (int);
 long            snum            (int);
@@ -174,7 +174,7 @@
 void            ruleaux         ();
 long            horizontalmove  ();
 int             skipnops        ();
-linetype    *   getline         ();
+linetype    *   _getline         ();
 linetype    *   findline        ();
 unsigned long   num             ();
 long            snum            ();
@@ -316,7 +316,7 @@
     h = 0L;  v = 0L;                        /* initialize coordinates   */
     x = 0L;  w = 0L;  y = 0L;  z = 0L;      /* initialize amounts       */
     sptr = 0;                               /* initialize stack         */
-    currentline = getline();                /* initialize list of lines */
+    currentline = _getline();                /* initialize list of lines */
     currentline->vv = 0L;
     firstline   = currentline;
     lastline    = currentline;
@@ -671,7 +671,7 @@
 
 /*----------------------------------------------------------------------------*/
 
-linetype *getline()             /* returns an initialized line-object */
+linetype *_getline()             /* returns an initialized line-object */
 {
     register int  i;
     register linetype *temp;
@@ -686,7 +686,7 @@
     temp->text[i] = '\0';
     return temp;
 
-} /* getline */
+} /* _getline */
 
 /*----------------------------------------------------------------------------*/
 
@@ -697,7 +697,7 @@
 
     if (v <= firstline->vv) {                      /* above first line */
         if (firstline->vv - v > VERTICALEPSILON) {
-            temp = getline();
+            temp = _getline();
             temp->next = firstline;
             firstline->prev = temp;
             temp->vv = v;
@@ -708,7 +708,7 @@
 
     if (v >= lastline->vv) {                       /* below last line */
         if (v - lastline->vv > VERTICALEPSILON) {
-            temp = getline();
+            temp = _getline();
             temp->prev = lastline;
             lastline->next = temp;
             temp->vv = v;
@@ -732,7 +732,7 @@
     }
 
     /* no line fits suitable, generate a new one */
-    currentline = getline();
+    currentline = _getline();
     currentline->next = temp->next;
     currentline->prev = temp;
     temp->next->prev = currentline;
--- texlive-20080816-source/texk/ttf2pk/newobj.c~	2006-01-17 16:41:51.000000000 -0500
+++ texlive-20080816-source/texk/ttf2pk/newobj.c	2009-06-20 17:55:23.000000000 -0400
@@ -128,13 +128,13 @@
 
 
 /*
- *   getline() is a wrapper function for get_line().  It returns `False' in
+ *   _getline() is a wrapper function for get_line().  It returns `False' in
  *   case of error and expects a pointer to a buffer to store the current
  *   line.   Additionally, the final newline character is stripped.
  */
 
 Boolean
-getline(char **bufferp, FILE *f)
+_getline(char **bufferp, FILE *f)
 {
   size_t l;
 
--- texlive-20080816-source/texk/ttf2pk/newobj.h~	2006-01-17 16:41:51.000000000 -0500
+++ texlive-20080816-source/texk/ttf2pk/newobj.h	2009-06-20 17:53:54.000000000 -0400
@@ -26,7 +26,7 @@
 void *myrealloc(void *oldp, size_t len);
 
 char *get_line(FILE *f);
-Boolean getline(char **bufferp, FILE *f);
+Boolean _getline(char **bufferp, FILE *f);
 char *newstring(char *s);
 
 ttfinfo *newchar(Font *fnt);
--- texlive-20080816-source/texk/ttf2pk/parse.c~	2006-01-17 16:41:51.000000000 -0500
+++ texlive-20080816-source/texk/ttf2pk/parse.c	2009-06-20 17:55:52.000000000 -0400
@@ -63,7 +63,7 @@
       if (*bufferp)
         free(*bufferp);
 
-      if (getline(bufferp, f) == False)
+      if (_getline(bufferp, f) == False)
         oops("Premature end in encoding file.");
 
       curp = *bufferp;
@@ -187,7 +187,7 @@
         "Token 258 in encoding must be make-array (]).");
     free(p);
 
-    while (getline(&buffer, enc_file))
+    while (_getline(&buffer, enc_file))
     {
       for (p = buffer; *p; p++)
         if (*p == '%')
@@ -246,7 +246,7 @@
   if (replacement_file == NULL)
     oops("Cannot open replacement file `%s'.", real_replacement_name);
 
-  while (getline(&buffer, replacement_file))
+  while (_getline(&buffer, replacement_file))
   {
     for (p = buffer; *p; p++)
       if (*p == '%')
