diff -Nur comm-1.9.1.orig/mail/installer/Makefile.in comm-1.9.1/mail/installer/Makefile.in
--- comm-1.9.1.orig/mail/installer/Makefile.in	2009-12-05 03:53:48.000000000 +0200
+++ comm-1.9.1/mail/installer/Makefile.in	2010-01-17 18:14:54.000000000 +0200
@@ -42,6 +42,8 @@
 
 include $(DEPTH)/config/autoconf.mk
 
+MOZ_APP_VERSION="3.0"
+
 NO_PKG_FILES = \
 	thunderbird-bin.elf \
 	thunderbird-config \
