From 9e00790f40ee498a7359b00e0b11a7523fdd1b3e Mon Sep 17 00:00:00 2001
From: Vincent Untz <vuntz@gnome.org>
Date: Thu, 19 Jan 2012 09:30:28 +0000
Subject: build: Support build against tracker 0.13/0.14

https://bugzilla.gnome.org/show_bug.cgi?id=667855
---
diff --git a/configure.ac b/configure.ac
index 1fd6743..71aceed 100644
--- a/configure.ac
+++ b/configure.ac
@@ -134,11 +134,17 @@ PKG_CHECK_MODULES(GMIME, gmime-2.6,
                                                    HAVE_GMIME=yes,
                                                    HAVE_GMIME=no)])
 
-PKG_CHECK_MODULES(TRACKER_SPARQL, tracker-sparql-0.12,
+PKG_CHECK_MODULES(TRACKER_SPARQL, tracker-sparql-0.14,
                                   HAVE_TRACKER_SPARQL=yes,
                                   HAVE_TRACKER_SPARQL=no)
 
 if test "x$HAVE_TRACKER_SPARQL" = "xno"; then
+   PKG_CHECK_MODULES(TRACKER_SPARQL, tracker-sparql-0.12,
+                                     HAVE_TRACKER_SPARQL=yes,
+                                     HAVE_TRACKER_SPARQL=no)
+fi
+
+if test "x$HAVE_TRACKER_SPARQL" = "xno"; then
    PKG_CHECK_MODULES(TRACKER_SPARQL, tracker-sparql-0.11,
                                      HAVE_TRACKER_SPARQL=yes,
                                      HAVE_TRACKER_SPARQL=no)
--
cgit v0.9.0.2
