--- src/smartbookmark.c.redhat	2006-12-18 02:06:21.000000000 +0100
+++ src/smartbookmark.c	2006-12-18 02:11:33.000000000 +0100
@@ -197,8 +197,8 @@
     box = gtk_vbox_new(FALSE, 0);
 
     /* default options */
-    search->url = "http://bugs.debian.org/";
-    search->label_text = "BTS";
+    search->url = "http://bugs.archlinux.org/";
+    search->label_text = "ArchBug #";
     search->size = 5;
     search->hide_label = FALSE;
     /* read config file options */
@@ -269,8 +269,8 @@
     if( (rcfile = xfce_rc_simple_open(filename, TRUE) ))
     {
         xfce_rc_set_group(rcfile, NULL);
-        search->url = g_strdup(xfce_rc_read_entry(rcfile,"url","http://bugs.debian.org/"));
-        search->label_text = g_strdup(xfce_rc_read_entry(rcfile,"value","DBS"));
+        search->url = g_strdup(xfce_rc_read_entry(rcfile,"url","http://bugs.archlinux.org/"));
+        search->label_text = g_strdup(xfce_rc_read_entry(rcfile,"value","ArchBug #"));
         search->size = xfce_rc_read_int_entry(rcfile, "size", 5);
         search->hide_label = xfce_rc_read_bool_entry(rcfile, "hidelabel", FALSE);
     }
