--- xorg-server-1.1.1/hw/xfree86/xf4bpp/Makefile.am.xf1bpp	2006-07-05 14:31:42.000000000 -0400
+++ xorg-server-1.1.1/hw/xfree86/xf4bpp/Makefile.am	2006-11-10 12:53:25.000000000 -0500
@@ -46,8 +46,6 @@
         mfbzerarc.c \
         vgamodule.c
 
-libxf4bpp_la_LIBADD = ../xf1bpp/libxf1bpp.la
-
 EXTRA_DIST = OScompiler.h ibmTrace.h ppcGCstr.h ppcSpMcro.h vgaReg.h \
              vgaVideo.h wm3.h NOTES
 
