--- xorg-server-1.1.1/hw/xfree86/os-support/linux/Makefile.am.automake-1.10	2006-11-30 15:35:09.000000000 -0500
+++ xorg-server-1.1.1/hw/xfree86/os-support/linux/Makefile.am	2006-12-01 11:00:04.000000000 -0500
@@ -41,7 +41,7 @@
 INCLUDES = $(XORG_INCS) $(PLATFORM_INCLUDES) -I/usr/include/drm # FIXME this last part is crack
 
 # FIXME: These need to be added to the build
-LNX_EXTRA_SOURCES = \
+LNX_EXTRA_SRCS = \
 	lnx_font.c \
 	lnx_jstk.c \
 	lnxResource.c
--- xorg-server-1.1.1/hw/xfree86/os-support/misc/Makefile.am.automake-1.10	2006-07-05 14:31:41.000000000 -0400
+++ xorg-server-1.1.1/hw/xfree86/os-support/misc/Makefile.am	2006-12-01 11:00:57.000000000 -0500
@@ -1,14 +1,14 @@
 # FIXME: Add the *.S files to build when applicable
-I386_SOURCES = BUSmemcpy.S IODelay.S SlowBcopy.S
+I386_SRCS = BUSmemcpy.S IODelay.S SlowBcopy.S
 OTHER_SOURCES = BUSmemcpy.c IODelay.c SlowBcopy.c
 
 ARCH_SOURCES = $(OTHER_SOURCES)
 
 # FIXME: Add to the build (NeedPortIO)
-PORTIO_SOURCES = PortIO.S
+PORTIO_SRCS = PortIO.S
 
 # FIXME: Add to the build (if HasGcc || HasGcc2)
-ILHACK_SOURCES = xf86_IlHack.c
+ILHACK_SRCS = xf86_IlHack.c
 
 noinst_LTLIBRARIES = libmisc.la
 
