diff -upr 389-ds-base-389-ds-base-2.0.10.orig/Makefile.am 389-ds-base-389-ds-base-2.0.10/Makefile.am
--- 389-ds-base-389-ds-base-2.0.10.orig/Makefile.am	2021-09-20 16:12:55.000000000 +0300
+++ 389-ds-base-389-ds-base-2.0.10/Makefile.am	2021-11-13 23:02:18.801136472 +0200
@@ -186,7 +186,7 @@ NSS_LINK = $(NSS_LIBS)
 # with recent versions of openldap - if you link with both ldap_r and ldap, the
 # shared lib _fini for one will stomp on the other, and the program will crash
 LDAPSDK_LINK_NOTHR = @openldap_lib@ -lldap@ol_libver@ @ldap_lib_ldif@ -llber@ol_libver@
-LDAPSDK_LINK = @openldap_lib@ -lldap_r@ol_libver@ @ldap_lib_ldif@ -llber@ol_libver@
+LDAPSDK_LINK = @openldap_lib@ -lldap@ol_libver@ @ldap_lib_ldif@ -llber@ol_libver@
 ldaplib = @ldaplib@
 ldaplib_defs = @ldaplib_defs@
 
