--- branches/KDE/3.5/kdenetwork/kopete/protocols/oscar/liboscar/oscartypes.h	2006/07/11 18:00:43	561016
+++ branches/KDE/3.5/kdenetwork/kopete/protocols/oscar/liboscar/oscartypes.h	2006/07/31 16:42:04	568263
@@ -237,7 +237,7 @@
 	const WORD ICQ_MAJOR 			= 0x0014;
 	const WORD ICQ_MINOR 			= 0x0034;
 	const WORD ICQ_POINT 			= 0x0000;
-	const WORD ICQ_BUILD 			= 0x0A0F;
+	const WORD ICQ_BUILD 			= 0x0BB8;
 	const char ICQ_OTHER[] 			= { 0x00, 0x00, 0x04, 0x3d };
 	const char ICQ_COUNTRY[] 		= "us";
 	const char ICQ_LANG[] 			= "en";
