diff -Nur comm-2.0.orig//suite/installer/Makefile.in comm-2.0/suite/installer/Makefile.in
--- comm-2.0.orig//suite/installer/Makefile.in	2011-06-18 19:04:41.592250768 +0000
+++ comm-2.0/suite/installer/Makefile.in	2011-06-18 19:05:40.416090761 +0000
@@ -42,6 +42,8 @@
 
 include $(DEPTH)/config/autoconf.mk
 
+MOZ_APP_VERSION="2.1"
+
 include $(topsrcdir)/config/rules.mk
 
 MOZ_PKG_REMOVALS = $(srcdir)/removed-files.in
