diff --git a/src/aticonsole.c b/src/aticonsole.c
index 1be147e..6e742d9 100644
--- a/src/aticonsole.c
+++ b/src/aticonsole.c
@@ -28,6 +28,7 @@
 #include "config.h"
 #endif
 
+#include "xorgVersion.h"
 #include "ati.h"
 #include "aticonsole.h"
 #include "atii2c.h"
@@ -689,7 +690,9 @@ ATIEnterVT
     ScreenPtr   pScreen     = pScreenInfo->pScreen;
     ATIPtr      pATI        = ATIPTR(pScreenInfo);
     PixmapPtr   pScreenPixmap;
+#if (XORG_VERSION_CURRENT < XORG_VERSION_NUMERIC(1, 9, 99, 1, 0))
     DevUnion    PixmapPrivate;
+#endif
     Bool        Entered;
 
     if (!ATIEnterGraphics(NULL, pScreenInfo, pATI))
@@ -714,19 +717,24 @@ ATIEnterVT
     }
 
     pScreenPixmap = (*pScreen->GetScreenPixmap)(pScreen);
+
+#if (XORG_VERSION_CURRENT < XORG_VERSION_NUMERIC(1, 9, 99, 1, 0))
     PixmapPrivate = pScreenPixmap->devPrivate;
     if (!PixmapPrivate.ptr)
         pScreenPixmap->devPrivate = pScreenInfo->pixmapPrivate;
+#endif
 
     /* Tell framebuffer about remapped aperture */
     Entered = (*pScreen->ModifyPixmapHeader)(pScreenPixmap,
         -1, -1, -1, -1, -1, pATI->pMemory);
 
+#if (XORG_VERSION_CURRENT < XORG_VERSION_NUMERIC(1, 9, 99, 1, 0))
     if (!PixmapPrivate.ptr)
     {
         pScreenInfo->pixmapPrivate = pScreenPixmap->devPrivate;
         pScreenPixmap->devPrivate.ptr = NULL;
     }
+#endif
 
 #ifdef XF86DRI_DEVEL
 
