--- trunk/ndiswrapper/driver/wrapndis.c	2010/09/13 09:36:56	2728
+++ trunk/ndiswrapper/driver/wrapndis.c	2011/02/14 17:00:00	2729
@@ -13,16 +13,16 @@
  *
  */
 
-#include "ndis.h"
-#include "iw_ndis.h"
-#include "pnp.h"
-#include "loader.h"
-#include "wrapndis.h"
 #include <linux/inetdevice.h>
 #include <linux/ip.h>
 #include <linux/tcp.h>
 #include <linux/udp.h>
 #include <linux/in.h>
+#include "ndis.h"
+#include "iw_ndis.h"
+#include "pnp.h"
+#include "loader.h"
+#include "wrapndis.h"
 #include "wrapper.h"
 
 /* Functions callable from the NDIS driver */
