--- bmpx-0.40.14.orig/src/uri.cc	2008-04-06 01:49:24.000000000 +0200
+++ bmpx-0.40.14/src/uri.cc	2009-02-23 19:23:56.000000000 +0100
@@ -35,6 +35,7 @@
 //  BMPx is covered by.
 
 #include <glib.h>
+#include <stdio.h>
 #include <string>
 #include <boost/algorithm/string.hpp>
 #include <glibmm.h>
--- bmpx-0.40.14.orig/widgets/popup.cc	2009-03-10 16:01:16.000000000 +0000
+++ bmpx-0.40.14/widgets/popup.cc	2009-03-10 16:01:30.000000000 +0000
@@ -308,7 +308,7 @@
 
 		if (m_location == ARROW_TOP)
 		{
-			Cairo::Matrix matrix = { 1, 0, 0, -1, 0, h };
+			Cairo::Matrix matrix(1, 0, 0, -1, 0, h);
 			cr->set_matrix (matrix);
 		}
 
--- bmpx-0.40.14.orig/src/ui-part-library.cc	2009-03-10 16:31:30.000000000 +0000
+++ bmpx-0.40.14/src/ui-part-library.cc	2009-03-10 16:32:26.000000000 +0000
@@ -509,8 +509,8 @@
       if( node != NODE_BRANCH )
       {
         // Clone this node as one child row
-        UID uid (Bmp::AlbumArtist ((*i_toplevel)[mStoreArtistCR.artist]).bmpx_album_artist_id);
         Bmp::AlbumArtist aa ((*i_toplevel)[mStoreArtistCR.artist]);
+        UID uid (aa.bmpx_album_artist_id);
 
         UidIterMapIter uid_i = mUidIterMap.find (uid);
         if( uid_i != mUidIterMap.end() )
--- bmpx-0.40.14.orig/docs/images/Makefile.am	2009-07-14 08:40:53.000000000 +0100
+++ bmpx-0.40.14/docs/images/Makefile.am	2009-07-14 09:57:46.000000000 +0100
@@ -1,4 +1,4 @@
-image_DATA=main-window.png bmp.png active-source.png playing-source.png important.png tip.png favicon.ico bmp-logo-small.png gradient.png rect1869.png favicon.ico
+image_DATA=main-window.png bmp.png active-source.png playing-source.png important.png tip.png favicon.ico bmp-logo-small.png gradient.png rect1869.png
 imagedir=$(prefix)/share/doc/@PACKAGE@/images
 
 EXTRA_DIST=$(image_DATA)
