diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5473802..1b07eb9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -57,7 +57,7 @@ find_package(KF5AkonadiSearch ${AKONADISEARCH_LIB_VERSION} CONFIG REQUIRED)
 set_package_properties(KF5AkonadiSearch PROPERTIES DESCRIPTION "The Akonadi Search libraries" URL "http://www.kde.org" TYPE REQUIRED PURPOSE "Provides search capabilities in KMail and Akonadi")
 endif()
 
-set(DISABLE_AKONADI_SEARCH FORCE_DISABLE_AKONADI_SEARCH)
+set(DISABLE_AKONADI_SEARCH ${FORCE_DISABLE_AKONADI_SEARCH})
 configure_file(config-akonadi-search.h.in ${CMAKE_CURRENT_BINARY_DIR}/src/config-akonadi-search.h)
 
 find_package(KF5DesignerPlugin ${KF5_MIN_VERSION} CONFIG)
