diff --git a/CMakeLists.txt b/CMakeLists.txt
index de5e035..b2bd4d6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -92,6 +92,15 @@ install(EXPORT KPim${KF_MAJOR_VERSION}MimeTargets
     FILE KPim${KF_MAJOR_VERSION}MimeTargets.cmake
     NAMESPACE KF5::
 )
+if (BUILD_QCH)
+    ecm_install_qch_export(
+        TARGETS KPim${KF_MAJOR_VERSION}Mime_QCH
+        FILE KPim${KF_MAJOR_VERSION}MimeQchTargets.cmake
+        DESTINATION "${CMAKECONFIG_INSTALL_DIR_KF5}"
+        COMPONENT Devel
+    )
+    set(PACKAGE_INCLUDE_QCHTARGETS "include(\"\${CMAKE_CURRENT_LIST_DIR}/KPim${KF_MAJOR_VERSION}MimeQchTargets.cmake\")")
+endif()
 endif()
 #endif()
 
@@ -108,10 +117,9 @@ endif()
 set(CMAKECONFIG_INSTALL_DIR "${KDE_INSTALL_CMAKEPACKAGEDIR}/KPim${KF_MAJOR_VERSION}Mime")
 set(KMIME_KF5_COMPAT FALSE)
 if (BUILD_QCH)
-    ecm_install_qch_export(
-        TARGETS KPim${KF_MAJOR_VERSION}Mime_QCH
-        FILE KPim${KF_MAJOR_VERSION}MimeQchTargets.cmake
-        DESTINATION "${CMAKECONFIG_INSTALL_DIR_KF5}"
+    install(FILES
+        "${CMAKE_CURRENT_BINARY_DIR}/KPim${KF_MAJOR_VERSION}MimeQchTargets.cmake"
+        DESTINATION "${CMAKECONFIG_INSTALL_DIR}"
         COMPONENT Devel
     )
     set(PACKAGE_INCLUDE_QCHTARGETS "include(\"\${CMAKE_CURRENT_LIST_DIR}/KPim${KF_MAJOR_VERSION}MimeQchTargets.cmake\")")
