diff -urp xchat-gnome-0.26.1.OLD/src/fe-gnome/fe-gnome.c xchat-gnome-0.26.1/src/fe-gnome/fe-gnome.c
--- xchat-gnome-0.26.1.OLD/src/fe-gnome/fe-gnome.c	2008-09-07 09:18:45.000000000 -0400
+++ xchat-gnome-0.26.1/src/fe-gnome/fe-gnome.c	2010-03-28 16:16:52.181405173 -0400
@@ -753,7 +753,7 @@ fe_gui_info (session * sess, int info_ty
 {
 	switch (info_type) {
 	case 0:
-		if (!GTK_WIDGET_VISIBLE (GTK_WINDOW (gui.main_window))) {
+		if (!gtk_widget_get_visible (GTK_WIDGET (gui.main_window))) {
 			return 2;
 		}
 		if (gtk_window_is_active (GTK_WINDOW (gui.main_window))) {
diff -urp xchat-gnome-0.26.1.OLD/src/fe-gnome/find-bar.c xchat-gnome-0.26.1/src/fe-gnome/find-bar.c
--- xchat-gnome-0.26.1.OLD/src/fe-gnome/find-bar.c	2007-03-09 10:42:57.000000000 -0500
+++ xchat-gnome-0.26.1/src/fe-gnome/find-bar.c	2010-03-28 16:26:24.300405158 -0400
@@ -247,7 +247,7 @@ find_bar_close (FindBar *bar)
 {
 	gint position;
 
-	if (!GTK_WIDGET_VISIBLE (bar)) {
+	if (!gtk_widget_get_visible (GTK_WIDGET (bar))) {
 		return;
 	}
 
diff -urp xchat-gnome-0.26.1.OLD/src/fe-gnome/text-entry.c xchat-gnome-0.26.1/src/fe-gnome/text-entry.c
--- xchat-gnome-0.26.1.OLD/src/fe-gnome/text-entry.c	2008-09-07 10:02:45.000000000 -0400
+++ xchat-gnome-0.26.1/src/fe-gnome/text-entry.c	2010-03-28 16:39:48.755153605 -0400
@@ -667,7 +667,7 @@ text_entry_set_current (TextEntry *entry
 	char *selection = NULL, *text = NULL;
 	int start, end;
 
-	g_return_if_fail (GTK_WIDGET_REALIZED (widget));
+	g_return_if_fail (gtk_widget_get_realized (GTK_WIDGET (widget)));
 
 	if (sess == priv->current) {
 		return;
diff -urp xchat-gnome-0.26.1.OLD/src/fe-gnome/userlist-gui.c xchat-gnome-0.26.1/src/fe-gnome/userlist-gui.c
--- xchat-gnome-0.26.1.OLD/src/fe-gnome/userlist-gui.c	2008-10-13 17:02:16.000000000 -0400
+++ xchat-gnome-0.26.1/src/fe-gnome/userlist-gui.c	2010-03-28 16:47:55.703154452 -0400
@@ -261,12 +261,12 @@ userlist_gui_show (void)
 
 	gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (gui.userlist_toggle), TRUE);
 
-	if (!GTK_WIDGET_REALIZED (gui.userlist_window)) {
+	if (!gtk_widget_get_realized (GTK_WIDGET (gui.userlist_window))) {
 		gtk_widget_realize (gui.userlist_window);
 	}
 	gtk_widget_size_request (gui.userlist, &request);
 
-	if (GTK_WIDGET_REALIZED (gui.userlist_toggle))
+	if (gtk_widget_get_realized (GTK_WIDGET (gui.userlist_toggle)))
 		anchor_widget = gui.userlist_toggle;
 	else
 		anchor_widget = gui.main_window;
diff -urp xchat-gnome-0.26.1.OLD/src/fe-gnome/xtext.c xchat-gnome-0.26.1/src/fe-gnome/xtext.c
--- xchat-gnome-0.26.1.OLD/src/fe-gnome/xtext.c	2009-01-16 08:17:31.000000000 -0500
+++ xchat-gnome-0.26.1/src/fe-gnome/xtext.c	2010-03-28 16:57:17.441402617 -0400
@@ -1086,7 +1086,7 @@ gtk_xtext_size_allocate (GtkWidget * wid
 	xtext->avoid_trans = FALSE;
 
 	widget->allocation = *allocation;
-	if (GTK_WIDGET_REALIZED (widget))
+	if (gtk_widget_get_realized (GTK_WIDGET (widget)))
 	{
 		xtext->buffer->window_width = allocation->width;
 		xtext->buffer->window_height = allocation->height;
@@ -4113,7 +4113,7 @@ gtk_xtext_set_palette (GtkXText * xtext,
 		xtext->palette[i] = palette[i].pixel;
 	}
 
-	if (GTK_WIDGET_REALIZED (xtext))
+	if (gtk_widget_get_realized (GTK_WIDGET (xtext)))
 	{
 		xtext_set_fg (xtext, xtext->fgc, XTEXT_FG);
 		xtext_set_bg (xtext, xtext->fgc, XTEXT_BG);
@@ -4211,7 +4211,7 @@ gtk_xtext_set_font (GtkXText *xtext, cha
 
 	gtk_xtext_fix_indent (xtext->buffer);
 
-	if (GTK_WIDGET_REALIZED (xtext))
+	if (gtk_widget_get_realized (GTK_WIDGET (xtext)))
 		gtk_xtext_recalc_widths (xtext->buffer, TRUE);
 
 	return TRUE;
@@ -4248,7 +4248,7 @@ gtk_xtext_set_background (GtkXText * xte
 	if (trans)
 	{
 		xtext->shaded = shaded;
-		if (GTK_WIDGET_REALIZED (xtext))
+		if (gtk_widget_get_realized (GTK_WIDGET (xtext)))
 			gtk_xtext_load_trans (xtext);
 		return;
 	}
@@ -4260,14 +4260,14 @@ gtk_xtext_set_background (GtkXText * xte
 	if (pixmap != 0)
 	{
 		g_object_ref (pixmap);
-		if (GTK_WIDGET_REALIZED (xtext))
+		if (gtk_widget_get_realized (GTK_WIDGET (xtext)))
 		{
 			gdk_gc_set_tile (xtext->bgc, pixmap);
 			gdk_gc_set_ts_origin (xtext->bgc, 0, 0);
 			xtext->ts_x = xtext->ts_y = 0;
 			gdk_gc_set_fill (xtext->bgc, GDK_TILED);
 		}
-	} else if (GTK_WIDGET_REALIZED (xtext))
+	} else if (gtk_widget_get_realized (GTK_WIDGET (xtext)))
 	{
 		g_object_unref (xtext->bgc);
 		val.subwindow_mode = GDK_INCLUDE_INFERIORS;
@@ -4516,7 +4516,7 @@ gtk_xtext_render_page (GtkXText * xtext)
 	int subline;
 	int startline = xtext->adj->value;
 
-	if(!GTK_WIDGET_REALIZED(xtext))
+	if(!gtk_widget_get_realized (GTK_WIDGET (xtext)))
 	  return;
 
 	if (xtext->buffer->indent < MARGIN)
@@ -4636,7 +4636,7 @@ gtk_xtext_render_page (GtkXText * xtext)
 void
 gtk_xtext_refresh (GtkXText * xtext, int do_trans)
 {
-	if (GTK_WIDGET_REALIZED (GTK_WIDGET (xtext)))
+	if (gtk_widget_get_realized (GTK_WIDGET (xtext)))
 	{
 #if defined(USE_XLIB) || defined(WIN32)
 		if (xtext->transparent && do_trans)
@@ -5204,7 +5204,7 @@ gtk_xtext_buffer_show (GtkXText *xtext, 
 		xtext->io_tag = 0;
 	}
 
-	if (!GTK_WIDGET_REALIZED (GTK_WIDGET (xtext)))
+	if (!gtk_widget_get_realized (GTK_WIDGET (xtext)))
 		gtk_widget_realize (GTK_WIDGET (xtext));
 
 	gdk_drawable_get_size (GTK_WIDGET (xtext)->window, &w, &h);
