--- xchat-gnome-0.17.old/plugins/tcl/tclplugin.c	2007-03-06 17:47:26.000000000 -0500
+++ xchat-gnome-0.17/plugins/tcl/tclplugin.c	2007-06-24 17:53:04.000000000 -0400
@@ -2215,8 +2215,8 @@
 void xchat_plugin_get_info(char **name, char **desc, char **version, void **reserved)
 {
    strncpy(PVERSION, &RCSID[19], 5);
-   *name = _("tclplugin");
-   *desc = _("Tcl plugin for XChat");
+   *name = "tclplugin";
+   *desc = "Tcl plugin for XChat";
    *version = PVERSION;
    if (reserved)
       *reserved = NULL;
