diff -ur slang-2.0.6.orig/autoconf/Makefile.in slang-2.0.6/autoconf/Makefile.in
--- slang-2.0.6.orig/autoconf/Makefile.in	2006-02-05 21:47:31.000000000 +0100
+++ slang-2.0.6/autoconf/Makefile.in	2006-04-09 16:17:04.000000000 +0200
@@ -8,12 +8,9 @@
 
 all:
 	cd src; $(MAKE) all
-	cd slsh; $(MAKE) all
-	cd modules; $(MAKE) all
+	cd src; $(MAKE) elf
 elf:
 	cd src; $(MAKE) elf
-	cd slsh; $(MAKE) all
-	cd modules; $(MAKE) all
 	@echo Use make install-elf to install it.
 runtests:
 	cd src; $(MAKE) all
@@ -25,17 +22,12 @@
 clean:
 	/bin/rm -f *~
 	cd src; $(MAKE) clean
-	cd modules; $(MAKE) clean
-	cd slsh; $(MAKE) clean
 	cd demo; $(MAKE) clean
 install:
 	cd src; $(MAKE) install
+	cd src; $(MAKE) install-elf
 install-elf:
 	cd src; $(MAKE) install-elf
-	@echo Now installing slsh
-	cd slsh; $(MAKE) install
-	@echo Now installing the modules
-	cd modules; $(MAKE) install
 	@echo ""
 	@echo "On some systems, e.g., linux, you may also have to run ldconfig."
 	@echo ""
@@ -46,8 +38,6 @@
 	/bin/rm -f *~ Makefile config.status config.log config.cache files.pck
 	cd src; $(MAKE) distclean
 	cd demo; $(MAKE) distclean
-	cd modules; $(MAKE) distclean
-	cd slsh; $(MAKE) distclean
 #
 configure: autoconf/aclocal.m4 autoconf/configure.ac
 	cd autoconf && autoconf && mv ./configure ..
Only in slang-2.0.6/autoconf: Makefile.in~
