diff -Nurp libgpod-0.8.3/bindings/mono/libgpod-sharp/Makefile.am libgpod-0.8.3-pkgconfig_sharp/bindings/mono/libgpod-sharp/Makefile.am
--- libgpod-0.8.3/bindings/mono/libgpod-sharp/Makefile.am	2013-07-09 09:37:38.000000000 +0000
+++ libgpod-0.8.3-pkgconfig_sharp/bindings/mono/libgpod-sharp/Makefile.am	2023-02-05 21:34:47.042376473 +0000
@@ -23,12 +23,12 @@ libgpod-sharp.dll.config: $(top_srcdir)/
 	sed "s/@DLNAME@/$(dlname)/g" $< >$@
 
 pkgconfigdir = $(libdir)/pkgconfig
-pkgconfig_DATA = libgpod-sharp.pc
 EXTRAS = \
 	libgpod-sharp.pc.in
 
 if HAVE_MONO
 include $(srcdir)/../Makefile.include
+pkgconfig_DATA = libgpod-sharp.pc
 module_SCRIPTS += libgpod-sharp.dll.config
 EXTRA_DIST += libgpod-sharp.dll.config.in
 DISTCLEANFILES += libgpod-sharp.dll.config
