diff -uNr hplip-1.7.1_old/prnt/hpijs/dj3320.cpp hplip-1.7.1_new/prnt/hpijs/dj3320.cpp
--- hplip-1.7.1_old/prnt/hpijs/dj3320.cpp	2007-01-24 15:42:03.000000000 -0800
+++ hplip-1.7.1_new/prnt/hpijs/dj3320.cpp	2007-01-31 13:33:42.000000000 -0800
@@ -3066,7 +3066,7 @@
     }
 
     // Send Sync packet
-    err = pPrinterXBow->Send (pbySync, (DWORD) sizeof (pbySync));
+    err = pPrinterXBow->Send (pbySync, SYNCSIZE);
     if(err)
     {
         return err;
--- hplip-1.7.1/levels.py~	2007-01-23 19:08:48.000000000 +0100
+++ hplip-1.7.1/levels.py	2007-02-25 12:08:49.000000000 +0100
@@ -201,8 +201,7 @@
         log.error("Error occured during interactive mode. Exiting.")
         sys.exit(1)
 
-        
-user_cfg.last_used.device_uri = d.device_uri
+
 
 try:
     d = device.Device(device_uri, printer_name)
