diff -ru QScintilla-2.11.5.dev2005091235/Python/sip/qsciabstractapis.sip QScintilla-2.11.5.dev2006091431/Python/sip/qsciabstractapis.sip
--- QScintilla-2.11.5.dev2005091235/Python/sip/qsciabstractapis.sip	2020-06-09 03:34:50.000000000 +0200
+++ QScintilla-2.11.5.dev2006091431/Python/sip/qsciabstractapis.sip	2020-06-10 03:34:31.000000000 +0200
@@ -25,7 +25,7 @@
 %End
 
 public:
-    QsciAbstractAPIs(QsciLexer *lexer /TransferThis/ = 0);
+    QsciAbstractAPIs(QsciLexer *lexer /TransferThis/);
     virtual ~QsciAbstractAPIs();
 
     QsciLexer *lexer() const;
diff -ru QScintilla-2.11.5.dev2005091235/Python/sip/qsciapis.sip QScintilla-2.11.5.dev2006091431/Python/sip/qsciapis.sip
--- QScintilla-2.11.5.dev2005091235/Python/sip/qsciapis.sip	2020-06-09 03:34:50.000000000 +0200
+++ QScintilla-2.11.5.dev2006091431/Python/sip/qsciapis.sip	2020-06-10 03:34:31.000000000 +0200
@@ -25,7 +25,7 @@
 %End
 
 public:
-    QsciAPIs(QsciLexer *lexer /TransferThis/ = 0);
+    QsciAPIs(QsciLexer *lexer /TransferThis/);
     virtual ~QsciAPIs();
 
     void add(const QString &entry);
