--- src/gpm-tray-icon.c	(révision 1934)
+++ src/gpm-tray-icon.c	(copie de travail)
@@ -447,8 +447,10 @@ 
 {
 	g_return_if_fail (GPM_IS_TRAY_ICON (icon));
 
-	gtk_status_icon_set_visible (GTK_STATUS_ICON (icon->priv->status_icon), enabled);
-	icon->priv->is_visible = enabled != FALSE;
+	if (enabled != icon->priv->is_visible) {
+		gtk_status_icon_set_visible (GTK_STATUS_ICON (icon->priv->status_icon), enabled);
+		icon->priv->is_visible = enabled != FALSE;
+	}
 }
 
 /**
