--- xf86-video-mga-1.4.6.1/src/binding.h	2006-11-08 17:16:44.000000000 +0000
+++ xf86-video-mga-1.4.6.1/src/binding.h	2004-12-08 21:03:00.000000000 +0000
@@ -210,6 +210,8 @@
 #define MGAMODEINFO_FORCE_PITCH        (1L << 5) 
 #define MGAMODEINFO_FORCE_DISPLAYORG   (1L << 6) 
 #define MGAMODEINFO_TV                 (1L << 7)
+#define MGAMODEINFO_FLATPANEL1         (1L << 8)
+#define MGAMODEINFO_FLATPANEL2         (1L << 9)
 #define MGAMODEINFO_TESTONLY           0x80000000
 
 /* Cable Type */
@@ -243,7 +245,6 @@
 ULONG MGAGetBOARDHANDLESize(void);
 ULONG MGASetTVStandard(LPBOARDHANDLE pBoard, ULONG ulTVStandard);
 ULONG MGASetTVCableType(LPBOARDHANDLE pBoard, ULONG ulCableType);
-ULONG HALSetDisplayStart(LPBOARDHANDLE pBoard, ULONG x, ULONG y, ULONG crtc);
 
 #if defined(__cplusplus)
     }
