From 037b839c84ad6dcbe9aade420cf0d3bf06f23119 Mon Sep 17 00:00:00 2001
From: Adam Jackson <ajax@redhat.com>
Date: Wed, 16 Nov 2011 20:03:25 +0000
Subject: Adapt to missing PIOOffset in videoabi 12

Signed-off-by: Adam Jackson <ajax@redhat.com>
---
diff --git a/src/smi_driver.c b/src/smi_driver.c
index 9c10e46..86644c7 100644
--- a/src/smi_driver.c
+++ b/src/smi_driver.c
@@ -442,7 +442,11 @@ SMI_PreInit(ScrnInfoPtr pScrn, int flags)
 	    LEAVE(FALSE);
 
 	hwp = VGAHWPTR(pScrn);
+#if ABI_VIDEODRV_VERSION < 12
 	pSmi->PIOBase = hwp->PIOOffset;
+#else
+	pSmi->PIOBase = 0;
+#endif
 
 	xf86ErrorFVerb(VERBLEV, "\tSMI_PreInit vgaCRIndex=%x, vgaIOBase=%x, "
 		       "MMIOBase=%p\n", hwp->IOBase + VGA_CRTC_INDEX_OFFSET,
--
cgit v0.9.0.2-2-gbebe
