diff -Nur watchdog-5.12.orig/src/wd_keepalive.c watchdog-5.12/src/wd_keepalive.c
--- watchdog-5.12.orig/src/wd_keepalive.c	2012-04-05 12:16:33.000000000 +0200
+++ watchdog-5.12/src/wd_keepalive.c	2012-05-13 16:15:36.786543263 +0200
@@ -26,6 +26,7 @@
 #include <sys/mman.h>
 #include <sys/ioctl.h>
 #include <sys/wait.h>
+#include <linux/oom.h>
 #include <linux/watchdog.h>
 #include <libgen.h>
 #include <string.h>
