Index: abcde
===================================================================
--- abcde	(revision 282)
+++ abcde	(revision 283)
@@ -835,7 +835,7 @@
 						-A "$DALBUM" -a "$TRACKARTIST" -t "$TRACKNAME" \
 						-y "$CDYEAR" -g "$GENREID" \
 						-T "${TRACKNUM:-$1}/$TRACKS" \
-						"${TPE2:+--TPE2 \"$TPE2\"}" \
+						${TPE2:+--TPE2 "$TPE2"} \
 						"$ABCDETEMPDIR/track$1.$OUTPUT"
 					;;
 			esac
