--- branches/KDE/4.5/kdeutils/printer-applet/printer-applet.py	2010/07/29 11:37:43	1156612
+++ branches/KDE/4.5/kdeutils/printer-applet/printer-applet.py	2010/07/29 11:38:09	1156613
@@ -237,10 +237,10 @@
             APPDIR = file.left(file.lastIndexOf('/'))
 
         self.mainWindow = MainWindow()
-        uic.loadUi(APPDIR + '/' + "printer-applet.ui", self.mainWindow)
+        uic.loadUi(unicode(APPDIR + '/' + "printer-applet.ui"), self.mainWindow)
 
         self.printersWindow = PrintersWindow(self)
-        uic.loadUi(APPDIR + '/' + "printer-applet-printers.ui", self.printersWindow)
+        uic.loadUi(unicode(APPDIR + '/' + "printer-applet-printers.ui"), self.printersWindow)
 
         self.sysTray = KStatusNotifierItem(self.mainWindow)
         self.sysTray.setCategory(KStatusNotifierItem.Hardware)
