diff -Naur kdeutils-3.5.7/klaptopdaemon/Makefile.am kdeutils-3.5.7-p/klaptopdaemon/Makefile.am
--- kdeutils-3.5.7/klaptopdaemon/Makefile.am	2007-05-14 09:40:45.000000000 +0200
+++ kdeutils-3.5.7-p/klaptopdaemon/Makefile.am	2007-05-26 16:27:01.000000000 +0200
@@ -9,7 +9,7 @@

 MESSAGE_SOURCES = laptop_daemon.cpp kpcmcia.cpp xautolock.cc kpcmciainfo.cpp daemondock.cpp xautolock_diy.c xautolock_engine.c
 kded_klaptopdaemon_la_SOURCES = $(MESSAGE_SOURCES) laptop_daemon.skel  
-kded_klaptopdaemon_la_LDFLAGS = $(all_libraries) -module -avoid-version  -lXtst
+kded_klaptopdaemon_la_LDFLAGS = $(all_libraries) -module -avoid-version $(LIB_XEXT) $(LIB_XSS) -lXtst
 kded_klaptopdaemon_la_LIBADD   = $(LIB_KDEUI) $(LIB_XSS) libkcmlaptop.la
 
 klaptop_acpi_helper_SOURCES = acpi_helper.cpp
