--- src/xine-engine/buffer.h	2008-07-16 23:01:56.000000000 +0200
+++ src/xine-engine/buffer.h	2008-08-24 21:14:39.000000000 +0200
@@ -676,7 +676,7 @@
 /* convert xine_waveformatex struct from little endian */
 void _x_waveformatex_le2me( xine_waveformatex *wavex ) XINE_PROTECTED;
 
-static inline _x_is_fourcc(void *ptr, void *tag) {
+static __inline _x_is_fourcc(void *ptr, void *tag) {
   return memcmp(ptr, tag, 4) == 0;
 }
 
