--- akode/plugins/xiph_decoder/flac113_decoder.cpp.orig	2007-06-12 06:59:28.000000000 +0000
+++ akode/plugins/xiph_decoder/flac113_decoder.cpp	2007-06-12 07:07:23.000000000 +0000
@@ -108,7 +108,7 @@
 static FLAC__StreamDecoderReadStatus flac_read_callback(
         const FLAC__StreamDecoder *,
         FLAC__byte buffer[],
-        unsigned *bytes,
+        size_t *bytes,
         void *client_data)
 {
     FLACDecoder::private_data *data = (FLACDecoder::private_data*)client_data;
