--- agent/CMakeLists.txt	2009-12-23 12:31:29.000000000 +0100
+++ agent/CMakeLists.txt	2010-02-10 21:26:05.183779591 +0100
@@ -18,8 +18,8 @@
 
 install(TARGETS polkit-kde-authentication-agent-1 DESTINATION ${LIBEXEC_INSTALL_DIR})
 
-# Install .desktop file to the XDG_CONFIG_DIRS if exists, to the /etc/xdg/autostart otherwise
-set (DESKTOP_INSTALL_DIR $ENV{XDG_CONFIG_DIRS})
+# Install .desktop file to the XDG_CONFIG_DIRS/autostart if exists, to the /etc/xdg/autostart otherwise
+set (DESKTOP_INSTALL_DIR $ENV{XDG_CONFIG_DIRS}/autostart)
 if (DESKTOP_INSTALL_DIR)
 else (DESKTOP_INSTALL_DIR)
   set (DESKTOP_INSTALL_DIR /etc/xdg/autostart)
