diff -aur source/texk/am/script_links.am source.new/texk/am/script_links.am
--- source/texk/am/script_links.am	2011-09-29 14:17:38.000000000 +0200
+++ source.new/texk/am/script_links.am	2011-12-04 13:14:35.480368959 +0100
@@ -52,7 +52,7 @@
 
 # We support native builds, multiplatform or not, as well as distro builds.
 install-links:
-	@REL=`$(SHELL) $(srcdir)/../../build-aux/relpath '$(DESTDIR)' '$(bindir)' '$(prefix)'`; \
+	@REL=$(datadir); \
 	if test -z "$$REL"; then \
 	  echo 'unable to compute relative path for linked $(TYPE) scripts' >&2; \
 	  exit 1; \
diff -aur source/texk/tetex/Makefile.am source.new/texk/tetex/Makefile.am
--- source/texk/tetex/Makefile.am	2011-09-29 14:14:07.000000000 +0200
+++ source.new/texk/tetex/Makefile.am	2011-12-04 13:13:32.089463768 +0100
@@ -37,7 +37,7 @@
 
 include $(srcdir)/../am/script_links.am
 
-texmfdir = ${prefix}/$(scriptsdir)
+texmfdir = $(datadir)/$(scriptsdir)
 dist_texmf_SCRIPTS = $(perl_scripts:=.pl) $(shell_scripts:=.sh)
 
 if !WIN32
diff -aur source/texk/texlive/linked_scripts/Makefile.am source.new/texk/texlive/linked_scripts/Makefile.am
--- source/texk/texlive/linked_scripts/Makefile.am	2011-12-03 17:22:19.000000000 +0100
+++ source.new/texk/texlive/linked_scripts/Makefile.am	2011-12-04 13:14:20.550940820 +0100
@@ -177,7 +177,7 @@
 ## Name the link as the basename, removing any extension,
 ## except for listings-ext.sh, which is documented as that name.
 ## Downcase for the sake of TeXcount.pl -> texcount.
-	@REL=`$(relpath) '$(DESTDIR)' '$(bindir)' '$(prefix)'`; \
+	@REL=$(datadir); \
 	if test -z "$$REL"; then \
 	  echo 'unable to compute relative path for linked scripts' >&2; \
 	  exit 1; \
