Index: ext/pitfdll/dmo_adec.c
===================================================================
RCS file: /cvsroot/pitfdll/pitfdll/ext/pitfdll/dmo_adec.c,v
retrieving revision 1.12
diff -u -p -r1.12 dmo_adec.c
--- ext/pitfdll/dmo_adec.c	15 Jun 2006 18:00:40 -0000	1.12
+++ ext/pitfdll/dmo_adec.c	10 Apr 2007 18:56:07 -0000
@@ -146,6 +146,7 @@ dmo_audiodec_init (DMOAudioDec * dec)
                                                                                 
   dec->srcpad = gst_pad_new_from_template (
       gst_element_class_get_pad_template (eklass, "src"), "src");
+  gst_pad_use_fixed_caps (dec->srcpad);
   gst_element_add_pad (GST_ELEMENT (dec), dec->srcpad);
 
   dec->ctx = NULL;
Index: ext/pitfdll/dmo_vdec.c
===================================================================
RCS file: /cvsroot/pitfdll/pitfdll/ext/pitfdll/dmo_vdec.c,v
retrieving revision 1.10
diff -u -p -r1.10 dmo_vdec.c
--- ext/pitfdll/dmo_vdec.c	15 Jun 2006 18:00:40 -0000	1.10
+++ ext/pitfdll/dmo_vdec.c	10 Apr 2007 18:56:08 -0000
@@ -139,6 +139,7 @@ dmo_videodec_init (DMOVideoDec * dec)
   
   dec->srcpad = gst_pad_new_from_template (
       gst_element_class_get_pad_template (eklass, "src"), "src");
+  gst_pad_use_fixed_caps (dec->srcpad);
   gst_element_add_pad (GST_ELEMENT (dec), dec->srcpad);
 
   dec->ctx = NULL;
Index: ext/pitfdll/dshow_vdec.c
===================================================================
RCS file: /cvsroot/pitfdll/pitfdll/ext/pitfdll/dshow_vdec.c,v
retrieving revision 1.9
diff -u -p -r1.9 dshow_vdec.c
--- ext/pitfdll/dshow_vdec.c	18 Feb 2007 00:27:13 -0000	1.9
+++ ext/pitfdll/dshow_vdec.c	10 Apr 2007 18:56:08 -0000
@@ -130,6 +130,7 @@ dshow_videodec_init (DSVideoDec * dec)
   
   dec->srcpad = gst_pad_new_from_template (
       gst_element_class_get_pad_template (eklass, "src"), "src");
+  gst_pad_use_fixed_caps (dec->srcpad);
   gst_element_add_pad (GST_ELEMENT (dec), dec->srcpad);
 
   dec->ctx = NULL;
Index: ext/pitfdll/qt_adec.c
===================================================================
RCS file: /cvsroot/pitfdll/pitfdll/ext/pitfdll/qt_adec.c,v
retrieving revision 1.6
diff -u -p -r1.6 qt_adec.c
--- ext/pitfdll/qt_adec.c	15 Jun 2006 18:00:40 -0000	1.6
+++ ext/pitfdll/qt_adec.c	10 Apr 2007 18:56:09 -0000
@@ -168,6 +168,7 @@ qt_audiodec_init (QtAudioDec * dec)
                                                                                 
   dec->srcpad = gst_pad_new_from_template (
       gst_element_class_get_pad_template (eklass, "src"), "src");
+  gst_pad_use_fixed_caps (dec->srcpad);
   gst_element_add_pad (GST_ELEMENT (dec), dec->srcpad);
 
   dec->dll = NULL;
