diff -u bash_completion.orig/bash_completion bash_completion/bash_completion
--- bash_completion.orig/bash_completion	2005-07-21 14:21:22.000000000 -0500
+++ bash_completion/bash_completion	2006-02-15 23:46:23.000000000 -0600
@@ -691,7 +691,7 @@
 	COMPREPLY=()
 	cur=${COMP_WORDS[COMP_CWORD]}
 
-	COMPREPLY=( $( /sbin/lsmod | \
+	COMPREPLY=( $( /bin/lsmod | \
 		  awk '{if (NR != 1 && $1 ~ /^'$cur'/) print $1}' 2>/dev/null ))
 	return 0
 }
@@ -712,7 +712,7 @@
 	# behave like lsmod for modprobe -r
 	if [ $1 = "modprobe" ] &&
 	   [ "${COMP_WORDS[1]}" = "-r" ]; then
-		COMPREPLY=( $( /sbin/lsmod | \
+		COMPREPLY=( $( /bin/lsmod | \
 				awk '{if (NR != 1 && $1 ~ /^'$cur'/) print $1}' ) )
 		return 0
 	fi
