=== modified file 'Makefile.am'
--- Makefile.am	2012-06-22 15:40:53 +0000
+++ Makefile.am	2012-07-19 09:36:30 +0000
@@ -329,7 +329,8 @@
 	$(FREETYPE_LIBS) \
 	$(FONTCONFIG_LIBS) \
 	$(LIBPNG_LIBS) \
-	$(POPPLER_LIBS)
+	$(POPPLER_LIBS) \
+	$(DLOPEN_LIBS)
 
 EXTRA_DIST += $(pkgfontconfig_DATA)

=== modified file 'configure.ac'
--- configure.ac	2012-07-19 14:02:32 +0000
+++ configure.ac	2012-07-19 14:39:25 +0000
@@ -110,6 +110,13 @@
 AC_DEFINE(PDFTOPDF, [], [Needed for pdftopdf filter compilation])
 AC_DEFINE_DIR(BANNERTOPDF_DATADIR, "{CUPS_DATADIR}/data", [Directory where bannertopdf finds its data files (PDF templates)])
 
+AC_SEARCH_LIBS([dlopen],
+	[dl],
+	DLOPEN_LIBS="-ldl",
+	AC_MSG_ERROR([unable to find the dlopen() function])
+)
+AC_SUBST(DLOPEN_LIBS)
+
 # ======================
 # Check system functions
 # ======================

