--- src/celengine/star.h.orig	2005-12-06 11:30:23.000000000 -0600
+++ src/celengine/star.h	2006-02-10 15:21:49.000000000 -0600
@@ -30,7 +30,7 @@
     // orbits and StarDetails objects.  Currently, it's assumed that they
     // have lifetimes that continue until program termination.
  public:
-    StarDetails::StarDetails();
+    StarDetails();
 
     inline float getRadius() const;
     inline float getTemperature() const;
--- src/celengine/command.h.orig	2004-12-22 16:07:15.000000000 -0600
+++ src/celengine/command.h	2006-02-10 18:46:25.000000000 -0600
@@ -94,10 +94,10 @@
 class CommandGotoLongLat : public InstantaneousCommand
 {
  public:
-    CommandGotoLongLat::CommandGotoLongLat(double t,
-                                           double dist,
-                                           float _longitude, float _latitude,
-                                           Vec3f _up);
+    CommandGotoLongLat(double t,
+                       double dist,
+                       float _longitude, float _latitude,
+                       Vec3f _up);
     ~CommandGotoLongLat();
     void process(ExecutionEnvironment&);
 
