diff -Naur smc-1.5.orig/src/core/game_core.cpp smc-1.5.new/src/core/game_core.cpp
--- smc-1.5.orig/src/core/game_core.cpp	2008-05-02 19:33:38.000000000 -0400
+++ smc-1.5.new/src/core/game_core.cpp	2008-09-17 23:50:21.000000000 -0400
@@ -1544,29 +1544,29 @@
 			if( fs::is_directory( *dir_itr ) )
 			{
 				// ignore hidden directories
-				if( dir_itr->leaf().find( "." ) == 0 )
+				if( dir_itr->filename().find( "." ) == 0 )
 				{
 					continue;
 				}
 
 				if( with_directories )
 				{
-					valid_files.push_back( dir + "/" + dir_itr->leaf() );
+					valid_files.push_back( dir + "/" + dir_itr->filename() );
 				}
 
 				// load all items from the sub-directory
-				vector<string> new_valid_files = Get_Directory_Files( dir + "/" + dir_itr->leaf(), file_type, with_directories );
+				vector<string> new_valid_files = Get_Directory_Files( dir + "/" + dir_itr->filename(), file_type, with_directories );
 				valid_files.insert( valid_files.end(), new_valid_files.begin(), new_valid_files.end() );
 			}
 			// valid file
-			else if( file_type.empty() || dir_itr->leaf().rfind( file_type ) != string::npos )
+			else if( file_type.empty() || dir_itr->filename().rfind( file_type ) != string::npos )
 			{
-				valid_files.push_back( dir + "/" + dir_itr->leaf() );
+				valid_files.push_back( dir + "/" + dir_itr->filename() );
 			}
 		}
 		catch( const std::exception &ex )
 		{
-			printf( "%s %s\n", dir_itr->leaf().c_str(), ex.what() );
+			printf( "%s %s\n", dir_itr->filename().c_str(), ex.what() );
 		}
 	}
 
diff -Naur smc-1.5.orig/src/overworld/worlds.cpp smc-1.5.new/src/overworld/worlds.cpp
--- smc-1.5.orig/src/overworld/worlds.cpp	2008-04-22 01:47:42.000000000 -0400
+++ smc-1.5.new/src/overworld/worlds.cpp	2008-09-17 23:51:03.000000000 -0400
@@ -111,7 +111,7 @@
 	{
 		try
 		{
-			string current_dir = dir_itr->leaf();
+			string current_dir = dir_itr->filename();
 
 			// only directories with an existing description
 			if( fs::is_directory( *dir_itr ) && File_Exists( dir + "/" + current_dir + "/description.xml" ) )
@@ -138,7 +138,7 @@
 		}
 		catch( const std::exception &ex )
 		{
-			printf( "%s %s\n", dir_itr->leaf().c_str(), ex.what() );
+			printf( "%s %s\n", dir_itr->filename().c_str(), ex.what() );
 		}
 	}
 }
