diff --git a/util/grub-mkconfig.in b/util/grub-mkconfig.in
index c408ccc..5be8f78 100644
--- a/util/grub-mkconfig.in
+++ b/util/grub-mkconfig.in
@@ -39,7 +39,9 @@ self=`basename $0`
 grub_probe=${sbindir}/`echo grub-probe | sed "${transform}"`
 grub_script_check="${bindir}/`echo grub-script-check | sed "${transform}"`"
 
-GRUB_PREFIX=`echo '/@bootdirname@/@grubdirname@' | sed "s,//*,/,g"`
+if test -z "${GRUB_PREFIX}"; then
+    GRUB_PREFIX=`echo '/@bootdirname@/@grubdirname@' | sed "s,//*,/,g"`
+fi
 
 # Usage: usage
 # Print the usage.
