--- kdelibs-4.2.85/kded/applications.menu~	2009-05-21 11:04:15.000000000 +0200
+++ kdelibs-4.2.85/kded/applications.menu	2009-05-21 11:05:30.000000000 +0200
@@ -30,6 +30,14 @@
 	</Layout>
 
 	<Menu>
+		<Name>Archlinux</Name>                                     
+		<Directory>Archlinux.directory</Directory>                 
+		<Include>                                                  
+		<Category>Archlinux</Category>                     
+		</Include>                                                 
+	</Menu>
+
+	<Menu>
 		<Name>LostFound</Name>
 		<Directory>kde-unknown.directory</Directory>
 		<OnlyUnallocated/>
