--- dcopperl/Makefile.PL.orig	2001-09-02 16:24:30.000000000 +0300
+++ dcopperl/Makefile.PL	2005-02-05 16:00:11.376798384 +0200
@@ -31,6 +31,7 @@
     XS		=> {'DCOP.xs' => 'DCOP.cpp'},
     XSOPT	=> '-C++',
     CCFLAGS	=> '-x c++',
+    INSTALLDIRS => 'vendor',
 );
 
 sub MY::xs_c {
--- dcopperl/Makefile.PL.in.orig	2003-12-28 15:43:20.000000000 +0200
+++ dcopperl/Makefile.PL.in	2005-02-05 16:00:21.002335080 +0200
@@ -17,6 +17,7 @@
     XS		=> {'DCOP.xs' => 'DCOP.cpp'},
     XSOPT	=> '-C++',
     CCFLAGS	=> '-x c++',
+    INSTALLDIRS => 'vendor',
 );
 
 sub MY::xs_c {
 
