--- kdebindings-4.4.0/python/pykde4/sip/plasma/animation.sip.org	2010-02-04 22:24:40.894406975 +0100
+++ kdebindings-4.4.0/python/pykde4/sip/plasma/animation.sip	2010-02-04 22:26:53.744810045 +0100
@@ -29,11 +29,11 @@
     int                     duration () const;
     enum Reference
     {
-        Center,
-        Up,
-        Down,
-        Left,
-        Right
+        Center = 0,
+        Up = 0x1,
+        Down = 0x2,
+        Left = 0x4,
+        Right = 0x8
     };
     explicit                Animation (QObject* parent /TransferThis/ = 0);
     virtual ~Animation ()=0;
@@ -52,14 +52,11 @@
     QGraphicsWidget*        targetWidget () const;
     enum MovementDirection
     {
-        MoveUp,
-        MoveUpRight,
-        MoveRight,
-        MoveDownRight,
-        MoveDown,
-        MoveDownLeft,
-        MoveLeft,
-        MoveUpLeft
+        MoveAny = 0,
+        MoveUp = 0x1,
+        MoveRight = 0x2,
+        MoveDown = 0x4,
+        MoveLeft = 0x8
     };
     void                    setEasingCurve (const QEasingCurve& curve);
     QEasingCurve            easingCurve () const;
