--- trunk/KDE/kdebindings/csharp/kimono/CMakeLists.txt	2010/05/19 22:20:26	1128672
+++ trunk/KDE/kdebindings/csharp/kimono/CMakeLists.txt	2010/10/09 14:30:30	1184234
@@ -22,7 +22,7 @@
                       ${KDE4_KDECORE_LIBS} ${KDE4_KIO_LIBS})
 
 # write FindPackage modules for these
-pkg_search_module(LIBMONO mono)
+pkg_search_module(LIBMONO mono-2 mono)
 pkg_search_module(GLIB2 glib-2.0)
 
 macro_log_feature(LIBMONO_FOUND "libmono" "Mono library" "http://www.mono-project.com" FALSE "" "Required to compile kimonopluginfactory")
--- trunk/KDE/kdebindings/csharp/kimono/src/kimonopluginfactory.cpp	2010/10/09 14:25:41	1184233
+++ trunk/KDE/kdebindings/csharp/kimono/src/kimonopluginfactory.cpp	2010/10/09 14:30:30	1184234
@@ -42,6 +42,7 @@
 #include <mono/metadata/assembly.h>
 #include <mono/metadata/debug-helpers.h>
 #include <mono/metadata/mono-config.h>
+#include <glib.h>
 
 #include <stdlib.h>
 #include <stdio.h>
