diff --git a/find-modules/sip_generator.py b/find-modules/sip_generator.py
--- a/find-modules/sip_generator.py
+++ b/find-modules/sip_generator.py
@@ -416,7 +416,10 @@
 
     def _enum_get(self, container, enum, level):
         pad = " " * (level * 4)
-        decl = pad + "enum {} {{\n".format(enum.displayname)
+        scoped = ""
+        if enum.is_scoped_enum():
+            scoped = "class "
+        decl = pad + "enum {}{} {{\n".format(scoped, enum.displayname)
         enumerations = []
         for enum in enum.get_children():
             #
@@ -609,7 +612,10 @@
                 typeText = text
                 typeInit = ""
 
-            prefix = parameterType.spelling.rsplit("::", 1)[0]
+            if parameterType.kind == TypeKind.ENUM and parameterType.get_declaration().is_scoped_enum():
+                prefix = parameterType.spelling
+            else:
+                prefix = parameterType.spelling.rsplit("::", 1)[0]
             if "::" in typeText:
                 typeText = typeText.rsplit("::", 1)[1]
             return prefix + "::" + typeText + typeInit

