diff -u -r libgda-5.2.4/getsp.java libgda-5.2.4-java/getsp.java
--- libgda-5.2.4/getsp.java	2014-02-05 21:28:20.000000000 +0100
+++ libgda-5.2.4-java/getsp.java	2015-06-14 20:27:35.770310228 +0200
@@ -25,7 +25,7 @@
 				while (i<=j) {
 					if (i==j || lp.charAt(i)==ps) {
 						String lib=lp.substring(k,i);
-						String suffix="/lib/amd64/server";
+						String suffix="/lib/"+System.getProperty("os.arch")+"/server";
 						k=i+1;
 						if (lib.compareTo(".")!=0)
 							r=(r==null)?(prefix+lib+suffix):(r+" "+prefix+lib+suffix);
@@ -50,7 +50,7 @@
 
 				if (r!=null) System.out.println(r);
 			} else if (args[0].compareTo("-ldpath")==0) {
-				String lp1=System.getProperty("java.home")+"/lib/amd64/server";
+				String lp1=System.getProperty("java.home")+"/lib/"+System.getProperty("os.arch")+"/server";
 				String lp2=System.getProperty("java.library.path");
 				System.out.println(lp1+":"+lp2);
 			}
diff -u -r libgda-5.2.4/m4/java.m4 libgda-5.2.4-java/m4/java.m4
--- libgda-5.2.4/m4/java.m4	2015-06-13 10:36:25.000000000 +0200
+++ libgda-5.2.4-java/m4/java.m4	2015-06-14 20:16:59.865581930 +0200
@@ -175,6 +175,11 @@
 		    JTYPE="Sun JRE 1.7"
 		    JFLAGS="-Xlint:unchecked -Xlint:deprecation"
 		    ;;
+                JRE1.8.*)
+		    try_java=true
+		    JTYPE="Sun JRE 1.8"
+		    JFLAGS="-Xlint:unchecked -Xlint:deprecation"
+		    ;;
                 JREgcj-4*)
 		    try_java=true
 		    JTYPE="GCJ"
