diff --git a/wscript b/wscript
index 9cedbbf..4432a9a 100644
--- a/wscript
+++ b/wscript
@@ -159,7 +159,7 @@ def test(ctx):
     ret = samba_utils.RUN_COMMAND(cmd)
     print("testsuite returned %d" % ret)
     magic_helper_cmd = os.path.join(Utils.g_module.blddir, 'talloc_test_magic_differs_helper')
-    magic_cmd = os.path.join(srcdir, 'lib', 'talloc',
+    magic_cmd = os.path.join(srcdir,
                              'test_magic_differs.sh')
 
     magic_ret = samba_utils.RUN_COMMAND(magic_cmd + " " +  magic_helper_cmd)
