 libgnomekbd/meson.build | 7 +++++++
 meson.build             | 1 +
 2 files changed, 8 insertions(+)

diff --git c/libgnomekbd/meson.build i/libgnomekbd/meson.build
index ca65a89..0bffa32 100644
--- c/libgnomekbd/meson.build
+++ i/libgnomekbd/meson.build
@@ -39,6 +39,7 @@ libgnomekbd_deps = [
   glib_dep,
   gobject_dep,
   gio_dep,
+  gmodule_export_dep,
   gdk_dep,
   gtk_dep,
   x11_dep,
@@ -72,6 +73,11 @@ pkg.generate(
   name: 'libgnomekbd',
   description: 'GNOME keyboard shared library',
   version: meson.project_version(),
+  requires: [
+    'gmodule-export-2.0',
+    'gdk-3.0',
+    'libxklavier',
+  ],
 )
 
 libgnomekbd_dep = declare_dependency(
@@ -137,6 +143,7 @@ pkg.generate(
   name: 'libgnomekbdui',
   description: 'GNOME keyboard shared library',
   version: meson.project_version(),
+  requires: ['libgnomekbd'],
 )
 
 libgnomekbdui_dep = declare_dependency(
diff --git c/meson.build i/meson.build
index cf8a9f5..921591c 100644
--- c/meson.build
+++ i/meson.build
@@ -28,6 +28,7 @@ libxklavier_req = '>= 5.2'
 glib_dep = dependency('glib-2.0', version: glib_req)
 gobject_dep = dependency('gobject-2.0', version: glib_req)
 gio_dep = dependency('gio-2.0', version: glib_req)
+gmodule_export_dep = dependency('gmodule-export-2.0', version: glib_req)
 gdk_dep = dependency('gdk-3.0', version: gdk_req)
 gtk_dep = dependency('gtk+-3.0', version: gtk_req)
 x11_dep = dependency('x11')
