diff -Naur TiMidity++-2.14.0-orig/timidity/jack_a.c TiMidity++-2.14.0/timidity/jack_a.c
--- TiMidity++-2.14.0-orig/timidity/jack_a.c	2014-09-26 22:08:51.531559553 -0400
+++ TiMidity++-2.14.0/timidity/jack_a.c	2014-09-26 22:11:12.864659285 -0400
@@ -508,6 +508,7 @@
 				pthread_cond_wait(&ctx->cond, &ctx->lock);
 		}
 		/* fallthrough */
+	case PM_REQ_PLAY_START:
 	case PM_REQ_DISCARD:
 		ctx->running = 0;
 		ringbuf_clear(&ctx->rbuf);
