--- vorbis-tools-1.1.1/ogg123/http_transport.c.orig	2006-12-09 16:34:44.000000000 +1100
+++ vorbis-tools-1.1.1/ogg123/http_transport.c	2006-12-09 16:39:46.000000000 +1100
@@ -116,7 +116,6 @@
   if (inputOpts.ProxyTunnel)
     curl_easy_setopt (handle, CURLOPT_HTTPPROXYTUNNEL, inputOpts.ProxyTunnel);
   */
-  curl_easy_setopt(handle, CURLOPT_MUTE, 1);
   curl_easy_setopt(handle, CURLOPT_ERRORBUFFER, private->error);
   curl_easy_setopt(handle, CURLOPT_PROGRESSFUNCTION, progress_callback);
   curl_easy_setopt(handle, CURLOPT_PROGRESSDATA, private);
