From 71d842c01992b3678de4da4773ed54f08c0ab4f6 Mon Sep 17 00:00:00 2001
From: Karel Zak <kzak@redhat.com>
Date: Mon, 18 Nov 2013 11:27:35 +0100
Subject: [PATCH] lsblk: fix -D segfault

References: https://bugzilla.redhat.com/show_bug.cgi?id=1031262
Signed-off-by: Karel Zak <kzak@redhat.com>
---
 misc-utils/lsblk.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/misc-utils/lsblk.c b/misc-utils/lsblk.c
index 9f7f1b6..9b53be3 100644
--- a/misc-utils/lsblk.c
+++ b/misc-utils/lsblk.c
@@ -903,7 +903,7 @@ static void set_tt_data(struct blkdev_cxt *cxt, int col, int id, struct tt_line
 		if (cxt->discard && p)
 			tt_line_set_data(ln, col, p);
 		else
-			tt_line_set_data(ln, col, "0");
+			tt_line_set_data(ln, col, xstrdup("0"));
 		break;
 	case COL_DGRAN:
 		if (lsblk->bytes)
@@ -936,7 +936,7 @@ static void set_tt_data(struct blkdev_cxt *cxt, int col, int id, struct tt_line
 		if (cxt->discard && p)
 			tt_line_set_data(ln, col, p);
 		else
-			tt_line_set_data(ln, col, "0");
+			tt_line_set_data(ln, col, xstrdup("0"));
 		break;
 	case COL_WSAME:
 		if (lsblk->bytes)
@@ -948,7 +948,7 @@ static void set_tt_data(struct blkdev_cxt *cxt, int col, int id, struct tt_line
 					   "queue/write_same_max_bytes", &x) == 0)
 				p = size_to_human_string(SIZE_SUFFIX_1LETTER, x);
 		}
-		tt_line_set_data(ln, col, p ? p : "0");
+		tt_line_set_data(ln, col, p ? p : xstrdup("0"));
 		break;
 	};
 }
-- 
1.8.4.2

