--- config_office/configure.in
+++ config_office/configure.in
@@ -4072,7 +4072,10 @@
             AGG_INCDIR="`echo $AGG_CFLAGS | $SED -e s/-I//`"
          fi
          # we could have used pkg-config --modversion but 2.4's libagg.pc.in still contains 2.3 :/
-         if $GREP -q "Version 2.4" `echo $AGG_INCDIR`/agg_basics.h; then
+         if echo Detected agg version is `pkg-config --modversion libagg` | grep 2.5.0
+         then AGG_VERSION=2500
+         AC_MSG_RESULT([2.5])
+         elif $GREP -q "Version 2.4" `echo $AGG_INCDIR`/agg_basics.h; then
             AC_MSG_RESULT([2.4])
             AGG_VERSION=2400
          else
