Index: bin/modules/installer/servicesfile.pm
===================================================================
RCS file: /cvs/tools/solenv/bin/modules/installer/servicesfile.pm,v
retrieving revision 1.23
diff -u -r1.23 servicesfile.pm
--- solenv/bin/modules/installer/servicesfile.pm	10 Jul 2006 18:51:00 -0000	1.23
+++ solenv/bin/modules/installer/servicesfile.pm	23 Feb 2007 16:12:39 -0000
@@ -258,7 +258,7 @@
 					$counter++;
 				}
 			
-				if ((( $counter > 0 ) && ( $counter%$installer::globals::javamaxservices == 0 )) || (( $counter > 0 ) && ( $i == $#{$javacomponents} )))	# limiting to $installer::globals::maxservices files
+				if ( $counter > 0 )
 				{
 					$filestring =~ s/\;\s*$//;
 					chdir($onesourcepath);

