--- config_office/configure.in	2006-05-29 16:01:41.000000000 +0000
+++ config_office/configure.in	2006-05-31 13:40:43.000000000 +0000
@@ -533,6 +533,25 @@ AC_ARG_WITH(dict,
 
                           Usage: --with-dict=ENGB,ENUS,ITIT
 ],,)
+AC_ARG_WITH(intro-bitmaps,
+[  --with-intro-bitmaps    Prefer the specified intro bitmaps over the
+                          the default one.  Can be more than one (separated by
+                          commas), the order means priority of fallback if the
+                          first does not exist (in the installed tree).
+
+                          Usage: --with-intro-bitmaps=/path/my_ooo_intro.bmp
+],,)
+AC_ARG_WITH(about-bitmaps,
+[  --with-about-bitmaps    Similarly to --with-intro-bitmaps, this allows
+                          specification of bitmaps for the About box.
+
+                          Usage: --with-about-bitmaps=/path/my_ooo_about.bmp
+],,)
+AC_ARG_WITH(vendor,
+[  --with-vendor           Set vendor of the build.
+
+                          Usage: --with-vendor="John the Builder"
+],,)
 AC_ARG_WITH(asm-home,
 [  --with-asm-home         For Windows users, please supply the path for the
                           ml.exe assembler.
@@ -4898,6 +4906,70 @@ else
 fi
 AC_SUBST(WITH_DICT)
 
+AC_MSG_CHECKING([for additional 'intro' bitmaps])
+INTRO_BITMAPS=
+INTRO_BITMAP_NAMES=
+if test -z "$with_intro_bitmaps" -o "$with_intro_bitmaps" = "no" ; then
+   INTRO_BITMAPS=
+   AC_MSG_RESULT([none])
+else
+   for bitmap in `echo $with_intro_bitmaps | tr ',' ' '` ; do
+      intro=`basename $bitmap`
+      case "$bitmap" in
+         *.bmp) ;;
+         *)     bitmap= ; AC_MSG_WARN([Intro bitmaps should be .bmp files!]) ;;
+      esac
+      if test -n "$bitmap" ; then
+         INTRO_BITMAPS="$INTRO_BITMAPS $bitmap"
+         if test -z "$INTRO_BITMAP_NAMES" ; then
+            INTRO_BITMAP_NAMES="$intro"
+         else
+            INTRO_BITMAP_NAMES="$INTRO_BITMAP_NAMES,$intro"
+         fi
+      fi
+   done
+   AC_MSG_RESULT([$INTRO_BITMAP_NAMES])
+fi
+AC_SUBST(INTRO_BITMAPS)
+AC_SUBST(INTRO_BITMAP_NAMES)
+
+AC_MSG_CHECKING([for additional 'about' bitmaps])
+ABOUT_BITMAPS=
+ABOUT_BITMAP_NAMES=
+if test -z "$with_about_bitmaps" -o "$with_about_bitmaps" = "no" ; then
+   ABOUT_BITMAPS=
+   AC_MSG_RESULT([none])
+else
+   for bitmap in `echo $with_about_bitmaps | tr ',' ' '` ; do
+      about=`basename $bitmap`
+      case "$bitmap" in
+         *.bmp) ;;
+         *)     bitmap= ; AC_MSG_WARN([About bitmaps should be .bmp files!]) ;;
+      esac
+      if test -n "$bitmap" ; then
+         ABOUT_BITMAPS="$ABOUT_BITMAPS $bitmap"
+         if test -z "$ABOUT_BITMAP_NAMES" ; then
+            ABOUT_BITMAP_NAMES="$about"
+         else
+            ABOUT_BITMAP_NAMES="$ABOUT_BITMAP_NAMES,$about"
+         fi
+      fi
+   done
+   AC_MSG_RESULT([$ABOUT_BITMAP_NAMES])
+fi
+AC_SUBST(ABOUT_BITMAPS)
+AC_SUBST(ABOUT_BITMAP_NAMES)
+
+OOO_VENDOR=
+AC_MSG_CHECKING([for vendor])
+if test -z "$with_vendor" -o "$with_vendor" = "no" ; then
+   AC_MSG_RESULT([not set])
+else
+   OOO_VENDOR="$with_vendor"
+   AC_MSG_RESULT([$OOO_VENDOR])
+fi
+AC_SUBST(OOO_VENDOR)
+
 AC_MSG_CHECKING([whether to statically link to Gtk])
 if test -n "$enable_static_gtk" && test "$enable_static_gtk" != "no"; then
    ENABLE_STATIC_GTK="TRUE"
--- config_office/set_soenv.in	2006-05-29 16:01:41.000000000 +0000
+++ config_office/set_soenv.in	2006-05-30 12:30:39.000000000 +0000
@@ -1439,6 +1439,11 @@ else
 # Languages
 ToFile( "WITH_LANG",         "@WITH_LANG@",      "e" );
 ToFile( "WITH_POOR_HELP_LOCALIZATIONS", "@WITH_POOR_HELP_LOCALIZATIONS@", "e" );
+ToFile( "INTRO_BITMAPS",     "@INTRO_BITMAPS@",  "e" );
+ToFile( "INTRO_BITMAP_NAMES","@INTRO_BITMAP_NAMES@","e" );
+ToFile( "ABOUT_BITMAPS",     "@ABOUT_BITMAPS@",  "e" );
+ToFile( "ABOUT_BITMAP_NAMES","@ABOUT_BITMAP_NAMES@","e" );
+ToFile( "OOO_VENDOR",        "@OOO_VENDOR@",     "e" );
 
 ToFile( "CALL_CDECL",        $CALL_CDECL,        "e" );
 ToFile( "COMMON_OUTDIR",     $COMMON_OUTDIR,     "e" );
