diff --git i/src/interface.cpp w/src/interface.cpp
index 48b42d0b..65e52682 100644
--- i/src/interface.cpp
+++ w/src/interface.cpp
@@ -698,36 +698,22 @@ create_main_window (GsmApplication *app)
     GdkDisplay *display;
     GdkMonitor *monitor;
     GdkRectangle monitor_geometry;
-    const char* session;
 
     int width, height, xpos, ypos;
 
     GtkBuilder *builder = gtk_builder_new();
     gtk_builder_add_from_resource (builder, "/org/gnome/gnome-system-monitor/data/interface.ui", NULL);
     gtk_builder_add_from_resource (builder, "/org/gnome/gnome-system-monitor/data/menus.ui", NULL);
     gtk_builder_add_from_resource (builder, "/org/gnome/gnome-system-monitor/gtk/help-overlay.ui", NULL);
 
     main_window = HDY_APPLICATION_WINDOW (gtk_builder_get_object (builder, "main_window"));
     gtk_window_set_application (GTK_WINDOW (main_window), app->gobj());
     gtk_widget_set_name (GTK_WIDGET (main_window), "gnome-system-monitor");
     app->main_window = main_window;
 
     gtk_application_window_set_help_overlay (GTK_APPLICATION_WINDOW (app->main_window),
                                              GTK_SHORTCUTS_WINDOW (gtk_builder_get_object (builder, "help_overlay")));
 
-    session = g_getenv ("XDG_CURRENT_DESKTOP");
-    if (session && !strstr (session, "GNOME")){
-        GtkBox *mainbox;
-        HdyHeaderBar *headerbar;
-
-        mainbox = GTK_BOX (gtk_builder_get_object (builder, "main_box"));
-        headerbar = HDY_HEADER_BAR (gtk_builder_get_object (builder, "header_bar"));
-        gtk_style_context_remove_class (gtk_widget_get_style_context (GTK_WIDGET (headerbar)), "titlebar");
-        gtk_window_set_titlebar (GTK_WINDOW (main_window), NULL);
-        hdy_header_bar_set_show_close_button (headerbar, FALSE);
-        gtk_box_pack_start (mainbox, GTK_WIDGET (headerbar), FALSE, FALSE, 0);
-    }
-
     g_settings_get (app->settings->gobj(), GSM_SETTING_WINDOW_STATE, "(iiii)",
                     &width, &height, &xpos, &ypos);
     
