diff --git a/thumbnail/CMakeLists.txt b/thumbnail/CMakeLists.txt
index a26f4e3f..d3c7569e 100644
--- a/thumbnail/CMakeLists.txt
+++ b/thumbnail/CMakeLists.txt
@@ -1,6 +1,9 @@
 add_definitions(-DTRANSLATION_DOMAIN=\"kio5_thumbnail\")
 
-find_package(OpenEXR)
+find_package(OpenEXR 3.0 CONFIG)
+if(NOT OpenEXR_FOUND)
+    find_package(OpenEXR)
+endif()
 set_package_properties(OpenEXR PROPERTIES DESCRIPTION "API for accessing OpenEXR formatted images"
                        URL "https://www.openexr.com"
                        TYPE OPTIONAL
@@ -142,9 +145,13 @@ if(OpenEXR_FOUND)
     target_link_libraries(exrthumbnail
             KF5::KIOCore
             KF5::KIOWidgets
-            ${OpenEXR_LIBRARIES}
     )
-    target_include_directories(exrthumbnail SYSTEM PRIVATE ${OpenEXR_INCLUDE_DIRS})
+    if(TARGET OpenEXR::OpenEXR)
+        target_link_libraries(exrthumbnail OpenEXR::OpenEXR)
+    else()
+        target_include_directories(exrthumbnail SYSTEM PRIVATE ${OpenEXR_INCLUDE_DIRS})
+        target_link_libraries(exrthumbnail OpenEXR::OpenEXR)
+    endif()
 
     # OpenEXR headers use exceptions; at least clang refuses to build the target
     # when exceptions are not enabled.
diff --git a/thumbnail/exrcreator.cpp b/thumbnail/exrcreator.cpp
index d4c490d7..dd2dae65 100644
--- a/thumbnail/exrcreator.cpp
+++ b/thumbnail/exrcreator.cpp
@@ -23,6 +23,7 @@
 #include <QImage>
 #include <QFile>
 
+#include <ImfHeader.h>
 #include <ImfInputFile.h>
 #include <ImfPreviewImage.h>
 
