diff --git a/find-modules/FindPythonModuleGeneration.cmake b/find-modules/FindPythonModuleGeneration.cmake
index 2362c45..c789e9b 100644
--- a/find-modules/FindPythonModuleGeneration.cmake
+++ b/find-modules/FindPythonModuleGeneration.cmake
@@ -251,20 +251,6 @@ if(NOT SIP_Qt5Core_Mod_FILE)
   _report_NOT_FOUND("PyQt module files not found for the ${CMAKE_FIND_PACKAGE_NAME} Module.")
 endif()
 
-file(STRINGS "${SIP_Qt5Core_Mod_FILE}" _SIP_Qt5_VERSIONS
-  REGEX "^%Timeline"
-)
-
-string(REGEX MATCHALL "Qt_5_[^ }]+" _SIP_Qt5_VERSIONS "${_SIP_Qt5_VERSIONS}")
-
-set(GPB_Qt5_Tag Qt_5_${Qt5Core_VERSION_MINOR}_${Qt5Core_VERSION_PATCH})
-
-list(FIND _SIP_Qt5_VERSIONS ${GPB_Qt5_Tag} _SIP_Qt5_Version_Index)
-
-if(_SIP_Qt5_Version_Index EQUAL -1)
-  _report_NOT_FOUND("PyQt module \"${SIP_Qt5Core_Mod_FILE}\" does not support Qt version 5.${Qt5Core_VERSION_MINOR}.${Qt5Core_VERSION_PATCH} for the ${CMAKE_FIND_PACKAGE_NAME} Module. Found available Qt5 tags: \"${_SIP_Qt5_VERSIONS}\".")
-endif()
-
 set(PythonModuleGeneration_FOUND TRUE)
 
 include(CMakeParseArguments)
@@ -395,14 +395,6 @@ headers = sipAPI${modulename_value}
       endif()
     endforeach()
 
-    if (WIN32)
-      set(GPB_WS_Tag -t WS_WIN)
-    elif(APPLE)
-      set(GPB_WS_Tag -t WS_MACX)
-    else()
-      set(GPB_WS_Tag -t WS_X11)
-    endif()
-
     add_custom_target(generate_${modulename_value}_sip_files ALL DEPENDS ${sip_files})
 
     add_custom_command(OUTPUT
@@ -404,7 +404,6 @@ headers = sipAPI${modulename_value}
        --module-name "${modulename_value}"
        -c "${CMAKE_CURRENT_BINARY_DIR}/pybuild/${pythonnamespace_value}/${modulename_value}"
        -b "${CMAKE_CURRENT_BINARY_DIR}/pybuild/${pythonnamespace_value}/${modulename_value}/module.sbf"
-       -t ${GPB_Qt5_Tag} ${GPB_WS_Tag}
 
        -x VendorID -x Py_v3
 
diff --git a/find-modules/run-sip.py b/find-modules/run-sip.py
index 74c8486..0edaff7 100755
--- a/find-modules/run-sip.py
+++ b/find-modules/run-sip.py
@@ -6,6 +6,8 @@ import subprocess
 
 sipArgs = sys.argv[1:]
 
+from PyQt5.Qt import PYQT_CONFIGURATION
+sipArgs = PYQT_CONFIGURATION["sip_flags"].split(' ') + sipArgs
 idx = sipArgs.index("--module-name")
 modname = sipArgs[idx + 1]
 del sipArgs[idx]
