diff --git a/find-modules/GeneratePythonBindingUmbrellaModule.cmake b/find-modules/GeneratePythonBindingUmbrellaModule.cmake
index de79393..144fab4 100644
--- a/find-modules/GeneratePythonBindingUmbrellaModule.cmake
+++ b/find-modules/GeneratePythonBindingUmbrellaModule.cmake
@@ -4,4 +4,4 @@ get_filename_component(PYTHON_UMBRELLA_MODULE_DIR ${PYTHON_UMBRELLA_MODULE_FILE}
 
 file(MAKE_DIRECTORY "${PYTHON_UMBRELLA_MODULE_DIR}")
 
-execute_process(COMMAND "${CMAKE_COMMAND}" -E touch "${PYTHON_UMBRELLA_MODULE_FILE}")
+#execute_process(COMMAND "${CMAKE_COMMAND}" -E touch "${PYTHON_UMBRELLA_MODULE_FILE}")
