diff -upr mod_perl-2.0.6.orig/src/modules/perl/modperl_perl.c mod_perl-2.0.6/src/modules/perl/modperl_perl.c
--- mod_perl-2.0.6.orig/src/modules/perl/modperl_perl.c	2012-05-26 21:49:27.000000000 +0300
+++ mod_perl-2.0.6/src/modules/perl/modperl_perl.c	2012-05-26 21:55:39.000000000 +0300
@@ -102,13 +102,4 @@ static void modperl_perl_init_ids(pTHX_
     sv_setiv(GvSV(gv_fetchpv("$", TRUE, SVt_PV)), ids->pid);
 
-#ifndef WIN32
-    PL_uid  = ids->uid;
-    PL_euid = ids->euid;
-    PL_gid  = ids->gid;
-    PL_egid = ids->egid;
-#endif
-#ifdef MP_MAINTAIN_PPID
-    PL_ppid = ids->ppid;
-#endif
 }
 
