--- frostwire-4.13.5.noarch/runFrostwire.sh.orig	2008-04-17 21:36:51.000000000 +0100
+++ frostwire-4.13.5.noarch/runFrostwire.sh	2008-04-17 21:37:33.000000000 +0100
@@ -67,7 +67,7 @@
       return 1
   fi
 
-  JAVA_HEADER=`${JAVA_PROGRAM_DIR}java -version 2>&1 | head -n 1`
+  JAVA_HEADER=`${JAVA_PROGRAM_DIR}java -version 2>&1 | grep version`
   JAVA_IMPL=`echo ${JAVA_HEADER} | cut -f1 -d' '`
   if [ "$JAVA_IMPL" = "java" ] ; then
     VERSION=`echo ${JAVA_HEADER} | sed "s/java version \"\(.*\)\"/\1/"`
