diff --git a/xfce4-session/main.c b/xfce4-session/main.c
index 0c27a55..925069f 100644
--- a/xfce4-session/main.c
+++ b/xfce4-session/main.c
@@ -243,8 +243,10 @@ main (int argc, char **argv)
   XfsmShutdown     *shutdown_helper;
   gboolean          succeed = TRUE;
 
+#if 0
   if (!xfsm_dbus_require_session (argc, argv))
     return EXIT_SUCCESS;
+#endif
 
   xfce_textdomain (GETTEXT_PACKAGE, PACKAGE_LOCALE_DIR, "UTF-8");
 
