--- libindi-0.7/CMakeLists.txt~	2010-11-08 12:48:00.680005642 +0100
+++ libindi-0.7/CMakeLists.txt	2010-11-08 12:48:18.430005641 +0100
@@ -56,6 +56,8 @@
 check_include_files(termios.h TERMIOS_FOUND)
 macro_bool_to_01(TERMIOS_FOUND HAVE_TERMIOS_H)
 
+configure_file(${CMAKE_CURRENT_SOURCE_DIR}/config.h.cmake ${CMAKE_CURRENT_BINARY_DIR}/config.h )
+
 include_directories( ${CMAKE_CURRENT_BINARY_DIR})
 include_directories( ${CMAKE_SOURCE_DIR})
 include_directories( ${CMAKE_SOURCE_DIR}/libs)
--- /dev/null	2010-11-08 00:05:37.466666667 +0100
+++ libindi-0.7/config.h.cmake	2010-11-08 12:48:38.693338976 +0100
@@ -0,0 +1,15 @@
+/* Define to 1 if you have the <linux/videodev2.h> header file. */
+#cmakedefine HAVE_LINUX_VIDEODEV2_H 1
+
+/* The symbol timezone is an int, not a function */
+#define TIMEZONE_IS_INT 1
+
+/* Define if you have termios.h */
+#cmakedefine   HAVE_TERMIOS_H 1
+
+/* Define if you have fitsio.h */
+#cmakedefine   HAVE_CFITSIO_H 1
+
+/* Define if you have libnova.h */
+#cmakedefine   HAVE_NOVA_H 1
+
