diff -ur qalculate-kde-0.9.7/src/qalculateeditfunctiondialog.cpp qalculate-kde-0.9.7-gcc47/src/qalculateeditfunctiondialog.cpp
--- qalculate-kde-0.9.7/src/qalculateeditfunctiondialog.cpp	2009-12-03 17:06:06.000000000 +0100
+++ qalculate-kde-0.9.7-gcc47/src/qalculateeditfunctiondialog.cpp	2012-07-31 23:50:07.000000000 +0200
@@ -377,10 +377,10 @@
 				str = defarg.printlong().c_str();
 				str2 = "";
 			}
-			QListViewItem *i = new KListViewItem(argumentsView, i_prev, str2, str);
-			i_prev = i;
-			argumentsItems[i] = arg;
-			argumentsItemsEdited[i] = false;
+			QListViewItem *item = new KListViewItem(argumentsView, i_prev, str2, str);
+			i_prev = item;
+			argumentsItems[item] = arg;
+			argumentsItemsEdited[item] = false;
 		}
 	}
 	argumentsView->setResizeMode(KListView::AllColumns);
