diff -baur libffado-2.1.0.orig/SConstruct libffado-2.1.0/SConstruct
--- libffado-2.1.0.orig/SConstruct	2012-09-19 14:18:53.827448582 +0800
+++ libffado-2.1.0/SConstruct	2012-09-19 14:20:01.670414379 +0800
@@ -1,4 +1,4 @@
-#! /usr/bin/python
+#!/usr/bin/python2
 # -*- coding: utf-8 -*-
 #
 # Copyright (C) 2007, 2008, 2010 Arnold Krille
@@ -134,7 +134,7 @@
 
 def CheckForPyModule( context, module ):
     context.Message( "Checking for the python module '" + module + "' " )
-    ret = context.TryAction( "python $SOURCE", "import %s" % module, ".py" )
+    ret = context.TryAction( "python2 $SOURCE", "import %s" % module, ".py" )
     context.Result( ret[0] )
     return ret[0]
 
@@ -351,7 +351,7 @@
 
 # PyQT checks
 build_mixer = False
-if conf.CheckForApp( 'which pyuic4' ) and conf.CheckForPyModule( 'dbus' ) and conf.CheckForPyModule( 'PyQt4' ) and conf.CheckForPyModule( 'dbus.mainloop.qt' ):
+if conf.CheckForApp( 'which python2-pyuic4' ) and conf.CheckForPyModule( 'dbus' ) and conf.CheckForPyModule( 'PyQt4' ) and conf.CheckForPyModule( 'dbus.mainloop.qt' ):
     env['PYUIC4'] = True
     build_mixer = True
 
diff -baur libffado-2.1.0.orig/admin/dbus.py libffado-2.1.0/admin/dbus.py
--- libffado-2.1.0.orig/admin/dbus.py	2012-09-19 14:18:53.857448419 +0800
+++ libffado-2.1.0/admin/dbus.py	2012-09-19 14:19:02.957399122 +0800
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 #
 # Copyright (C) 2007-2008 Arnold Krille
 # Copyright (C) 2007-2008 Pieter Palmers
diff -baur libffado-2.1.0.orig/admin/doxygen.py libffado-2.1.0/admin/doxygen.py
--- libffado-2.1.0.orig/admin/doxygen.py	2012-09-19 14:18:53.857448419 +0800
+++ libffado-2.1.0/admin/doxygen.py	2012-09-19 14:19:02.957399122 +0800
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 #
 # Copyright (C) 2007-2008 Arnold Krille
 #
diff -baur libffado-2.1.0.orig/admin/pkgconfig.py libffado-2.1.0/admin/pkgconfig.py
--- libffado-2.1.0.orig/admin/pkgconfig.py	2012-09-19 14:18:53.857448419 +0800
+++ libffado-2.1.0/admin/pkgconfig.py	2012-09-19 14:19:02.957399123 +0800
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 #
 # Copyright (C) 2007-2008 Arnold Krille
 #
diff -baur libffado-2.1.0.orig/admin/pyuic.py libffado-2.1.0/admin/pyuic.py
--- libffado-2.1.0.orig/admin/pyuic.py	2012-09-19 14:18:53.857448419 +0800
+++ libffado-2.1.0/admin/pyuic.py	2012-09-19 14:19:02.957399123 +0800
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 #
 # Copyright (C) 2007-2008 Arnold Krille
 #
diff -baur libffado-2.1.0.orig/admin/pyuic4.py libffado-2.1.0/admin/pyuic4.py
--- libffado-2.1.0.orig/admin/pyuic4.py	2012-09-19 14:18:53.857448419 +0800
+++ libffado-2.1.0/admin/pyuic4.py	2012-09-19 14:19:02.957399123 +0800
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 #
 # Copyright (C) 2007-2008 Arnold Krille
 #
@@ -21,10 +21,8 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #
 
-import imp
-
 def pyuic4_action( target, source, env ):
-	env.Execute( "pyuic4 " + str( source[0] ) + " > " + str( target[0] ) )
+	env.Execute( "python2-pyuic4 " + str( source[0] ) + " > " + str( target[0] ) )
 	return 0
 
 def pyuic4_string( target, source, env ):
@@ -34,7 +32,7 @@
 	context.Message( "Checking for pyuic4 (by checking for the python module pyqtconfig) " )
 	ret = True
 	try:
-		imp.find_module( "pyqtconfig" )
+		import PyQt4.pyqtconfig
 	except ImportError:
 		ret = False
 	context.Result( ret )
diff -baur libffado-2.1.0.orig/admin/scanreplace.py libffado-2.1.0/admin/scanreplace.py
--- libffado-2.1.0.orig/admin/scanreplace.py	2012-09-19 14:18:53.857448419 +0800
+++ libffado-2.1.0/admin/scanreplace.py	2012-09-19 14:19:02.957399123 +0800
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 #
 # Copyright (C) 2007-2008 Arnold Krille
 #
diff -baur libffado-2.1.0.orig/doc/SConscript libffado-2.1.0/doc/SConscript
--- libffado-2.1.0.orig/doc/SConscript	2012-09-19 14:18:53.857448419 +0800
+++ libffado-2.1.0/doc/SConscript	2012-09-19 14:19:02.957399123 +0800
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2
 
 Import( 'env' )
 
diff -baur libffado-2.1.0.orig/support/dbus/SConscript libffado-2.1.0/support/dbus/SConscript
--- libffado-2.1.0.orig/support/dbus/SConscript	2012-09-19 14:18:53.827448582 +0800
+++ libffado-2.1.0/support/dbus/SConscript	2012-09-19 14:19:02.957399123 +0800
@@ -1,4 +1,4 @@
-#!/bin/env python
+#!/usr/bin/python2
 #
 # Copyright (C) 2007,2008,2011 Arnold Krille
 # Copyright (C) 2007-2008 Pieter Palmers
diff -baur libffado-2.1.0.orig/support/mixer-qt4/SConscript libffado-2.1.0/support/mixer-qt4/SConscript
--- libffado-2.1.0.orig/support/mixer-qt4/SConscript	2012-09-19 14:18:53.830781898 +0800
+++ libffado-2.1.0/support/mixer-qt4/SConscript	2012-09-19 14:19:02.957399123 +0800
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 #
 # Copyright (C) 2007-2009 Arnold Krille
 # Copyright (C) 2007-2008 Pieter Palmers
diff -baur libffado-2.1.0.orig/support/mixer-qt4/ffado/ffadowindow.py libffado-2.1.0/support/mixer-qt4/ffado/ffadowindow.py
--- libffado-2.1.0.orig/support/mixer-qt4/ffado/ffadowindow.py	2012-09-19 14:18:53.830781898 +0800
+++ libffado-2.1.0/support/mixer-qt4/ffado/ffadowindow.py	2012-09-19 14:19:02.957399123 +0800
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 #
 # Copyright (C) 2005-2008 by Pieter Palmers
 #               2007-2009 by Arnold Krille
diff -baur libffado-2.1.0.orig/support/mixer-qt4/ffado-mixer-profiler.in libffado-2.1.0/support/mixer-qt4/ffado-mixer-profiler.in
--- libffado-2.1.0.orig/support/mixer-qt4/ffado-mixer-profiler.in	2012-09-19 14:18:53.830781898 +0800
+++ libffado-2.1.0/support/mixer-qt4/ffado-mixer-profiler.in	2012-09-19 14:19:02.957399123 +0800
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 #
 # Copyright (C) 2005-2009 by Pieter Palmers
 #               2007-2009 by Arnold Krille
diff -baur libffado-2.1.0.orig/support/mixer-qt4/ffado-mixer.in libffado-2.1.0/support/mixer-qt4/ffado-mixer.in
--- libffado-2.1.0.orig/support/mixer-qt4/ffado-mixer.in	2012-09-19 14:18:53.834115214 +0800
+++ libffado-2.1.0/support/mixer-qt4/ffado-mixer.in	2012-09-19 14:19:02.957399123 +0800
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 #
 # Copyright (C) 2005-2008 by Pieter Palmers
 #               2007-2009 by Arnold Krille
diff -baur libffado-2.1.0.orig/support/tools/SConscript libffado-2.1.0/support/tools/SConscript
--- libffado-2.1.0.orig/support/tools/SConscript	2012-09-19 14:18:53.830781898 +0800
+++ libffado-2.1.0/support/tools/SConscript	2012-09-19 14:20:51.970141882 +0800
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 #
 # Copyright (C) 2007-2008 Arnold Krille
 # Copyright (C) 2007-2008 Pieter Palmers
@@ -43,7 +43,7 @@
 # For the installation of the stuff
 e['pythondir'] = Template( os.path.join( e['sharedir'], 'python' ) ).safe_substitute( e )
 
-e.Command( "static_info.txt", "#/SConstruct", "python support/tools/ffado-diag-static > $TARGET" )
+e.Command( "static_info.txt", "#/SConstruct", "python2 support/tools/ffado-diag-static > $TARGET" )
 
 e.ScanReplace( "ffado-diag.in" )
 
diff -baur libffado-2.1.0.orig/support/tools/ffado-diag-static libffado-2.1.0/support/tools/ffado-diag-static
--- libffado-2.1.0.orig/support/tools/ffado-diag-static	2012-09-19 14:18:53.830781898 +0800
+++ libffado-2.1.0/support/tools/ffado-diag-static	2012-09-19 14:19:02.957399123 +0800
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 #
 
 #
@@ -74,7 +74,7 @@
     # check libraries
     print "   gcc ............... %s" % get_version_first_line('gcc --version')
     print "   g++ ............... %s" % get_version_first_line('g++ --version')
-    print "   PyQt4 (by pyuic4) . %s" % get_version_first_line('pyuic4 --version')
+    print "   PyQt4 (by pyuic4) . %s" % get_version_first_line('python2-pyuic4 --version')
     print "   jackd ............. %s" % get_version_first_line('jackd --version')
     print "     path ............ %s" % get_command_path('jackd')
     print "     flags ........... %s" % get_package_flags("jack")
diff -baur libffado-2.1.0.orig/support/tools/ffado-diag.in libffado-2.1.0/support/tools/ffado-diag.in
--- libffado-2.1.0.orig/support/tools/ffado-diag.in	2012-09-19 14:18:53.830781898 +0800
+++ libffado-2.1.0/support/tools/ffado-diag.in	2012-09-19 14:19:02.957399123 +0800
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 #
 
 #
@@ -120,7 +120,7 @@
     print " Prerequisites (dynamic at run-time)..."
     print "   gcc ............... %s" % get_version_first_line('gcc --version')
     print "   g++ ............... %s" % get_version_first_line('g++ --version')
-    print "   PyQt4 (by pyuic4) . %s" % get_version_first_line('pyuic4 --version')
+    print "   PyQt4 (by pyuic4) . %s" % get_version_first_line('python2-pyuic4 --version')
     print "   jackd ............. %s" % get_version_first_line('jackd --version')
     print "     path ............ %s" % get_command_path('jackd')
     print "     flags ........... %s" % get_package_flags("jack")
diff -baur libffado-2.1.0.orig/support/tools/ffado-sandbox-install.py libffado-2.1.0/support/tools/ffado-sandbox-install.py
--- libffado-2.1.0.orig/support/tools/ffado-sandbox-install.py	2012-09-19 14:18:53.830781898 +0800
+++ libffado-2.1.0/support/tools/ffado-sandbox-install.py	2012-09-19 14:19:02.957399123 +0800
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 #
 
 #
diff -baur libffado-2.1.0.orig/support/tools/listirqinfo.py libffado-2.1.0/support/tools/listirqinfo.py
--- libffado-2.1.0.orig/support/tools/listirqinfo.py	2012-09-19 14:18:53.830781898 +0800
+++ libffado-2.1.0/support/tools/listirqinfo.py	2012-09-19 14:19:02.957399123 +0800
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 #
 
 #
diff -baur libffado-2.1.0.orig/tests/dbus_test.py libffado-2.1.0/tests/dbus_test.py
--- libffado-2.1.0.orig/tests/dbus_test.py	2012-09-19 14:18:53.837448529 +0800
+++ libffado-2.1.0/tests/dbus_test.py	2012-09-19 14:19:02.957399123 +0800
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 #
 # Copyright (C) 2005-2007 by Pieter Palmers
 #               2007-2008 by Arnold Krille
diff -baur libffado-2.1.0.orig/tests/python/test-eap-ctrl.py libffado-2.1.0/tests/python/test-eap-ctrl.py
--- libffado-2.1.0.orig/tests/python/test-eap-ctrl.py	2012-09-19 14:18:53.837448529 +0800
+++ libffado-2.1.0/tests/python/test-eap-ctrl.py	2012-09-19 14:19:02.957399123 +0800
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 #
 # Copyright (C) 2005-2009 by Pieter Palmers
 #
