--- libffado/60-ffado.rules	
+++ libffado/60-ffado.rules	
@@ -1,35 +1,58 @@ 
 SUBSYSTEM!="firewire", GOTO="ffado_end"
 
-ATTR{vendor}=="0x000166", GROUP="audio", ENV{ID_FFADO}="1" # TC GROUP A/S
-ATTR{vendor}=="0x0001f2", GROUP="audio", ENV{ID_FFADO}="1" # Mark of the Unicorn, Inc. (aka MOTU)
-ATTR{vendor}=="0x0003db", GROUP="audio", ENV{ID_FFADO}="1" # Apogee Electronics Corp.
-ATTR{vendor}=="0x000595", GROUP="audio", ENV{ID_FFADO}="1" # Alesis Corporation
-ATTR{vendor}=="0x0007f5", GROUP="audio", ENV{ID_FFADO}="1" # Bridgeco Co AG
-ATTR{vendor}=="0x000a92", GROUP="audio", ENV{ID_FFADO}="1" # Presonus Corporation
-ATTR{vendor}=="0x000aac", GROUP="audio", ENV{ID_FFADO}="1" # TerraTec Electronic GmbH
-ATTR{vendor}=="0x000d6c", GROUP="audio", ENV{ID_FFADO}="1" # M-Audio
-ATTR{vendor}=="0x000f1b", GROUP="audio", ENV{ID_FFADO}="1" # Ego Systems Inc.
-ATTR{vendor}=="0x000ff2", GROUP="audio", ENV{ID_FFADO}="1" # Loud Technologies Inc.
-ATTR{vendor}=="0x001260", GROUP="audio", ENV{ID_FFADO}="1" # Stanton Magnetics,inc.
-ATTR{vendor}=="0x00130e", GROUP="audio", ENV{ID_FFADO}="1" # Focusrite Audio Engineering Limited
-ATTR{vendor}=="0x001486", GROUP="audio", ENV{ID_FFADO}="1" # Echo Digital Audio Corporation
-ATTR{vendor}=="0x001496", GROUP="audio", ENV{ID_FFADO}="1" # Phonic Corporation
-ATTR{vendor}=="0x001564", GROUP="audio", ENV{ID_FFADO}="1" # BEHRINGER Spezielle Studiotechnik GmbH
-ATTR{vendor}=="0x001c2d", GROUP="audio", ENV{ID_FFADO}="1" # FlexRadio Systems
-ATTR{vendor}=="0x001c6a", GROUP="audio", ENV{ID_FFADO}="1" # Weiss Engineering Ltd.
-ATTR{vendor}=="0x0040ab", GROUP="audio", ENV{ID_FFADO}="1" # ROLAND DG CORPORATION
-ATTR{vendor}=="0x000f64", GROUP="audio", ENV{ID_FFADO}="1" # DnR
+# TC GROUP A/S
+ATTR{vendor}=="0x000166", GROUP="audio", ENV{ID_FFADO}="1"
+# Mark of the Unicorn, Inc. (aka MOTU)
+ATTR{vendor}=="0x0001f2", GROUP="audio", ENV{ID_FFADO}="1"
+# Apogee Electronics Corp.
+ATTR{vendor}=="0x0003db", GROUP="audio", ENV{ID_FFADO}="1"
+# Alesis Corporation
+ATTR{vendor}=="0x000595", GROUP="audio", ENV{ID_FFADO}="1"
+# Bridgeco Co AG
+ATTR{vendor}=="0x0007f5", GROUP="audio", ENV{ID_FFADO}="1"
+# Presonus Corporation
+ATTR{vendor}=="0x000a92", GROUP="audio", ENV{ID_FFADO}="1"
+# TerraTec Electronic GmbH
+ATTR{vendor}=="0x000aac", GROUP="audio", ENV{ID_FFADO}="1"
+# M-Audio
+ATTR{vendor}=="0x000d6c", GROUP="audio", ENV{ID_FFADO}="1"
+# Ego Systems Inc.
+ATTR{vendor}=="0x000f1b", GROUP="audio", ENV{ID_FFADO}="1"
+# Loud Technologies Inc.
+ATTR{vendor}=="0x000ff2", GROUP="audio", ENV{ID_FFADO}="1"
+# Stanton Magnetics,inc.
+ATTR{vendor}=="0x001260", GROUP="audio", ENV{ID_FFADO}="1"
+# Focusrite Audio Engineering Limited
+ATTR{vendor}=="0x00130e", GROUP="audio", ENV{ID_FFADO}="1"
+# Echo Digital Audio Corporation
+ATTR{vendor}=="0x001486", GROUP="audio", ENV{ID_FFADO}="1"
+# Phonic Corporation
+ATTR{vendor}=="0x001496", GROUP="audio", ENV{ID_FFADO}="1"
+# BEHRINGER Spezielle Studiotechnik GmbH
+ATTR{vendor}=="0x001564", GROUP="audio", ENV{ID_FFADO}="1"
+# FlexRadio Systems
+ATTR{vendor}=="0x001c2d", GROUP="audio", ENV{ID_FFADO}="1"
+# Weiss Engineering Ltd.
+ATTR{vendor}=="0x001c6a", GROUP="audio", ENV{ID_FFADO}="1"
+# ROLAND DG CORPORATION
+ATTR{vendor}=="0x0040ab", GROUP="audio", ENV{ID_FFADO}="1"
+# DnR
+ATTR{vendor}=="0x000f64", GROUP="audio", ENV{ID_FFADO}="1"
 
 # The devices below are by vendors who make other firewire devices in
 # addition to their audio interfaces.  They need more specific rules to
 # ensure only audio interfaces are covered here.
 
-ATTR{vendor}=="0x00022e", ATTR{model}=="0x010067", GROUP="audio", ENV{ID_FFADO}="1" # Tascam, a subsiduary of TEAC (the OUI is TEAC's)
+# Tascam, a subsiduary of TEAC (the OUI is TEAC's)
+ATTR{vendor}=="0x00022e", ATTR{model}=="0x010067", GROUP="audio", ENV{ID_FFADO}="1"
 
 # The devices below abuse another Vendor's ID, and therefore we need more advanced rules for those.
 
-ATTR{vendor}=="0x00000a", ATTR{model}=="0x030000", ATTR{units}=="*0x00a02d:0x010001*", GROUP="audio", ENV{ID_FFADO}="1" # CME, Matrix K FW
-ATTR{vendor}=="0x00000f", ATTR{model}=="0x01006?", ATTR{units}=="*0x00a02d:0x010001*", GROUP="audio", ENV{ID_FFADO}="1" # Mackie, Onyx Firewire 
-ATTR{vendor}=="0x000a35", ATTR{units}=="0x000a35:0x00000[12]", GROUP="audio", ENV{ID_FFADO}="1" # RME
+# CME, Matrix K FW
+ATTR{vendor}=="0x00000a", ATTR{model}=="0x030000", ATTR{units}=="*0x00a02d:0x010001*", GROUP="audio", ENV{ID_FFADO}="1"
+# Mackie, Onyx Firewire 
+ATTR{vendor}=="0x00000f", ATTR{model}=="0x01006?", ATTR{units}=="*0x00a02d:0x010001*", GROUP="audio", ENV{ID_FFADO}="1"
+# RME
+ATTR{vendor}=="0x000a35", ATTR{units}=="0x000a35:0x00000[12]", GROUP="audio", ENV{ID_FFADO}="1"
 
 LABEL="ffado_end"
