--- openjdk/hotspot/make/linux/makefiles/gcc.make	2012-06-07 16:30:51.000000000 +0200
+++ openjdk/hotspot/make/linux/makefiles/gcc.make.new	2012-06-14 15:32:44.967695139 +0200
@@ -150,7 +150,7 @@
 endif
 
 # Compiler warnings are treated as errors
-WARNINGS_ARE_ERRORS = -Werror
+#WARNINGS_ARE_ERRORS = -Werror
 
 # Except for a few acceptable ones
 # Since GCC 4.3, -Wconversion has changed its meanings to warn these implicit
