diff -up openjpeg_v1_4_sources_r697/doc/Makefile.am.autoconf_DESTDIR openjpeg_v1_4_sources_r697/doc/Makefile.am
--- openjpeg_v1_4_sources_r697/doc/Makefile.am.autoconf_DESTDIR	2011-01-02 12:14:47.000000000 -0600
+++ openjpeg_v1_4_sources_r697/doc/Makefile.am	2011-01-12 12:12:20.959518549 -0600
@@ -20,30 +20,30 @@ all-local:
 	@cp man/man3/* .
 
 install-data-hook:
-	$(INSTALL) -d $(doc_dir)
-	$(INSTALL) -d ${mandir}/man1 ${mandir}/man3
+	$(INSTALL) -d $(DESTDIR)$(doc_dir)
+	$(INSTALL) -d $(DESTDIR)${mandir}/man1 $(DESTDIR)${mandir}/man3
 if with_libjpwl
-	@(cd ${mandir}/man1 && $(RM) JPWL_image_to_j2k* JPWL_j2k_to_image*)
-	@(cd ${mandir}/man3 && $(RM) JPWL_libopenjpeg* )
+	@(cd $(DESTDIR)${mandir}/man1 && $(RM) JPWL_image_to_j2k* JPWL_j2k_to_image*)
+	@(cd $(DESTDIR)${mandir}/man3 && $(RM) JPWL_libopenjpeg* )
 endif
-	@(cd ${mandir}/man1 && $(RM) image_to_j2k* j2k_to_image* j2k_dump* )
-	@(cd ${mandir}/man3 && $(RM) libopenjpeg* )
+	@(cd $(DESTDIR)${mandir}/man1 && $(RM) image_to_j2k* j2k_to_image* j2k_dump* )
+	@(cd $(DESTDIR)${mandir}/man3 && $(RM) libopenjpeg* )
 	@gzip -f image_to_j2k.1 j2k_to_image.1 j2k_dump.1 libopenjpeg.3
-	@cp image_to_j2k.1.gz j2k_to_image.1.gz j2k_dump.1.gz ${mandir}/man1
-	@cp libopenjpeg.3.gz ${mandir}/man3
+	@cp image_to_j2k.1.gz j2k_to_image.1.gz j2k_dump.1.gz $(DESTDIR)${mandir}/man1
+	@cp libopenjpeg.3.gz $(DESTDIR)${mandir}/man3
 	@$(RM) *\.gz
 if with_libjpwl
-	@(cd ${mandir}/man1 && \
+	@(cd $(DESTDIR)${mandir}/man1 && \
 	$(LN_S) image_to_j2k.1.gz JPWL_image_to_j2k.1.gz && \
 	$(LN_S) j2k_to_image.1.gz JPWL_j2k_to_image.1.gz)
 	@(cd ${mandir}/man3 && $(LN_S) libopenjpeg.3.gz JPWL_libopenjpeg.3.gz)
 endif
-	@$(INSTALL) -m 644 ../LICENSE ../CHANGES $(doc_dir)
+	@$(INSTALL) -m 644 ../LICENSE ../CHANGES $(DESTDIR)$(doc_dir)
 
 uninstall-hook:
-	@(cd ${mandir}/man1 && $(RM) JPWL_image_to_j2k* JPWL_j2k_to_image* && \
+	@(cd $(DESTDIR)${mandir}/man1 && $(RM) JPWL_image_to_j2k* JPWL_j2k_to_image* && \
 	$(RM) image_to_j2k* j2k_to_image* j2k_dump* )
-	@(cd ${mandir}/man3 && $(RM) JPWL_libopenjpeg* && \
+	@(cd $(DESTDIR)${mandir}/man3 && $(RM) JPWL_libopenjpeg* && \
 	$(RM) libopenjpeg* )
-	@(cd $(doc_dir) && $(RM) * )
-	@rmdir $(doc_dir)
+	@(cd $(DESTDIR)$(doc_dir) && $(RM) * )
+	@rmdir $(DESTDIR)$(doc_dir)
diff -up openjpeg_v1_4_sources_r697/Makefile.am.autoconf_DESTDIR openjpeg_v1_4_sources_r697/Makefile.am
--- openjpeg_v1_4_sources_r697/Makefile.am.autoconf_DESTDIR	2011-01-02 12:14:55.000000000 -0600
+++ openjpeg_v1_4_sources_r697/Makefile.am	2011-01-12 12:15:37.006321242 -0600
@@ -9,7 +9,7 @@ clean-local:
 
 distclean-local: clean-local
 
-pkgconf_dir = /usr/lib/pkgconfig
+pkgconf_dir = $(libdir)/pkgconfig
 
 install-data-hook:
 	@echo ""
@@ -17,12 +17,12 @@ install-data-hook:
 	cat $$f/.report.txt ; \
 	done
 	@echo ""
-	@echo "Installing: libopenjpeg1.pc $(pkgconf_dir)"
+	@echo "Installing: libopenjpeg1.pc $(DESTDIR)$(pkgconf_dir)"
 	@echo ""
-	@$(INSTALL) -d $(pkgconf_dir)
-	@cp libopenjpeg.pc $(pkgconf_dir)/libopenjpeg1.pc
-	@(cd $(pkgconf_dir) && $(RM) libopenjpeg.pc && \
+	@$(INSTALL) -d $(DESTDIR)$(pkgconf_dir)
+	@cp libopenjpeg.pc $(DESTDIR)$(pkgconf_dir)/libopenjpeg1.pc
+	@(cd $(DESTDIR)$(pkgconf_dir) && $(RM) libopenjpeg.pc && \
 	$(LN_S) libopenjpeg1.pc libopenjpeg.pc)
 
 uninstall-hook:
-	@(cd $(pkgconf_dir) && $(RM) libopenjpeg.pc libopenjpeg1.pc)
+	@(cd $(DESTDIR)$(pkgconf_dir) && $(RM) libopenjpeg.pc libopenjpeg1.pc)
