Index: libarchive/archive_read_extract.c
===================================================================
--- libarchive/archive_read_extract.c	(revision 3722)
+++ libarchive/archive_read_extract.c	(revision 3723)
@@ -108,7 +108,7 @@
 	if (r != ARCHIVE_OK)
 		/* If _write_header failed, copy the error. */
  		archive_copy_error(&a->archive, ad);
-	else if (archive_entry_size(entry) > 0)
+	else if (!archive_entry_size_is_set(entry) || archive_entry_size(entry) > 0)
 		/* Otherwise, pour data into the entry. */
 		r = copy_data(_a, ad);
 	r2 = archive_write_finish_entry(ad);

