--- korundum-4.12.80/CMakeLists.txt~	2014-03-08 16:02:37.343367661 +0000
+++ korundum-4.12.80/CMakeLists.txt	2014-03-08 16:13:01.699397634 +0000
@@ -19,12 +19,16 @@
         Phonon
         Plasma
         QtCore
-        Solid
+        Solid )
+
+find_package(Smoke COMPONENTS
+        Akonadi
+        Okular
+        Kate
         Soprano
         SopranoClient
         SopranoServer
         Nepomuk )
-find_package(Smoke COMPONENTS Akonadi Okular Kate)
 include(KDE4Defaults)
 include_directories( ${KDE4_INCLUDES} ${SMOKE_INCLUDE_DIR} ${QTRUBY_INCLUDE_DIR} ${SMOKE_QTCORE_INCLUDE_DIR} ${SMOKE_KDECORE_INCLUDE_DIR} )
 
