Index: akonadi/contact/contactsearchjob.cpp
===================================================================
--- akonadi/contact/contactsearchjob.cpp	(revision 1156790)
+++ akonadi/contact/contactsearchjob.cpp	(revision 1156791)
@@ -50,7 +50,7 @@
 
 void ContactSearchJob::setQuery( Criterion criterion, const QString &value )
 {
-  setQuery( criterion, value, ContainsMatch );
+  setQuery( criterion, value, ExactMatch );
 }
 
 void ContactSearchJob::setQuery( Criterion criterion, const QString &value, Match match )
