--- jvmfwk/plugins/sunmajor/pluginlib/gnujre.cxx
+++ jvmfwk/plugins/sunmajor/pluginlib/gnujre.cxx
@@ -68,6 +68,18 @@
     static char const* ar[]= {
           "/lib/" JFW_PLUGIN_ARCH "/client/libjvm.so",
           "/gcj-4.1.1/libjvm.so",
+	  "/gcj-4.2/libjvm.so",
+	  "/gcj-4.1/libjvm.so",
+	  "/gcj-4.2.0/libjvm.so",
+	  "/gcj-4.2.1/libjvm.so",
+	  "/gcj-4.2.2/libjvm.so",
+	  "/gcj-4.2.2-81/libjvm.so",
+         "/gcj-4.1.2/libjvm.so",
+	  "/libgcj.so.80",
+	  "/libgcj.so.81",
+	  "/libgcj.so.8",
+	  "/libgcj.so.71",
+	  "/libgcj.so.70",
           "/libgcj.so.7",
           "/libgcj.so.6"
     };
