Index: unx/gtk/gdi/salnativewidgets-gtk.cxx
===================================================================
RCS file: /cvs/gsl/vcl/unx/gtk/gdi/salnativewidgets-gtk.cxx,v
retrieving revision 1.39
diff -u -r1.39 salnativewidgets-gtk.cxx
--- vcl/unx/gtk/gdi/salnativewidgets-gtk.cxx	3 Jul 2007 14:06:42 -0000	1.39
+++ vcl/unx/gtk/gdi/salnativewidgets-gtk.cxx	19 Oct 2007 08:46:59 -0000
@@ -2898,9 +2899,9 @@
     aPartRect = Rectangle( aPartPos, aPartSize );
 
 	if ( pIndicatorSize )
-		g_free( pIndicatorSize );
+		gtk_requisition_free( pIndicatorSize );
 	if ( pIndicatorSpacing )
-		g_free( pIndicatorSpacing );
+		gtk_border_free( pIndicatorSpacing );
 
 	return( aPartRect );
 }
@@ -2947,9 +2948,9 @@
 		aIndicatorRect.Move( 0, 1 );
 
 	if ( pIndicatorSize )
-		g_free( pIndicatorSize );
+		gtk_requisition_free( pIndicatorSize );
 	if ( pIndicatorSpacing )
-		g_free( pIndicatorSpacing );
+		gtk_border_free( pIndicatorSpacing );
 
 	return( aIndicatorRect );
 }
