diff -rupN a/src/tracker-extract/tracker-extract-gif.c b/src/tracker-extract/tracker-extract-gif.c
--- a/src/tracker-extract/tracker-extract-gif.c	2014-05-08 15:48:22.000000000 +0200
+++ b/src/tracker-extract/tracker-extract-gif.c	2014-06-01 13:35:37.407254708 +0200
@@ -684,7 +684,7 @@ tracker_extract_get_metadata (TrackerExt
 
 	g_free (uri);
 
-	if (DGifCloseFile (gifFile) != GIF_OK) {
+	if (DGifCloseFile (gifFile, NULL) != GIF_OK) {
 #if GIFLIB_MAJOR < 5
 		print_gif_error ();
 #else  /* GIFLIB_MAJOR < 5 */
