--- kmplot-15.12.3/kmplot/view.cpp.0	2016-03-11 21:19:48.011194225 +0000
+++ kmplot-15.12.3/kmplot/view.cpp	2016-03-11 21:20:37.341056493 +0000
@@ -476,7 +476,7 @@
 	double x = pixel.x();
 	double y = pixel.y();
 	
-	if ( isnan(x) )
+	if ( std::isnan(x) )
 	{
 		xclipflg = true;
 		x = pixelIfNaN.x();
@@ -496,14 +496,14 @@
 	}
 	else
 	{
-		if ( isinf(x) == -1 )
+		if ( std::isinf(x) == -1 )
 			x = 0;
 			
-		else if ( isinf(x) == 1 )
+		else if ( std::isinf(x) == 1 )
 			x = m_clipRect.right();
 	}
 	
-	if ( isnan(y) )
+	if ( std::isnan(y) )
 	{
 		yclipflg = true;
 		y = pixelIfNaN.y();
@@ -523,10 +523,10 @@
 	}
 	else
 	{
-		if ( isinf(y) == -1 )
+		if ( std::isinf(y) == -1 )
 			y = 0;
 		
-		else if ( isinf(y) == 1 )
+		else if ( std::isinf(y) == 1 )
 			y = m_clipRect.bottom();
 	}
 	
