diff -ub lib/fcl_cstr.ml lib.sic/fcl_cstr.ml
--- lib/fcl_cstr.ml	2017-11-30 20:24:10.000000000 +0100
+++ lib.sic/fcl_cstr.ml	2023-05-18 12:20:58.691033891 +0200
@@ -50,7 +50,7 @@
   and nb_solved = Fcl_stak.ref 0 in
   let update i =
     if update i then
-      if Pervasives.not solved.(i) then begin
+      if Stdlib.not solved.(i) then begin
        	Fcl_stak.set nb_solved (Fcl_stak.get nb_solved + 1);
        	array_set_true solved i
       end in
diff -ub lib/fcl_fdArray.ml lib.sic/fcl_fdArray.ml
--- lib/fcl_fdArray.ml	2017-11-30 20:24:10.000000000 +0100
+++ lib.sic/fcl_fdArray.ml	2023-05-18 12:22:34.355125486 +0200
@@ -299,7 +299,7 @@
       let (mi, ma) =
       	Array.fold_left
 	  (fun (mi, ma) e ->
-	    (Pervasives.min mi (Fd.min e), Pervasives.max ma (Fd.max e)))
+	    (Stdlib.min mi (Fd.min e), Stdlib.max ma (Fd.max e)))
 	  (max_int, min_int) array in
       if mi = ma then Fd.int mi else
       let value = Fd.create (Fcl_domain.interval mi ma) in
diff -ub lib/fcl_misc.ml lib.sic/fcl_misc.ml
--- lib/fcl_misc.ml	2017-11-30 20:24:10.000000000 +0100
+++ lib.sic/fcl_misc.ml	2023-05-18 12:20:22.631984218 +0200
@@ -50,9 +50,9 @@
     x - y
 *)
 
-  let (+) = Pervasives.(+)
-  let (-) = Pervasives.(-)
-  let ( * ) = Pervasives.( * )
+  let (+) = Stdlib.(+)
+  let (-) = Stdlib.(-)
+  let ( * ) = Stdlib.( * )
 
   let (=+) x y = x := !x + y
   let (=+.) x y = x := !x +. y
diff -ub lib/fcl_nonlinear.ml lib.sic/fcl_nonlinear.ml
--- lib/fcl_nonlinear.ml	2017-11-30 20:24:10.000000000 +0100
+++ lib.sic/fcl_nonlinear.ml	2023-05-18 12:22:26.218697740 +0200
@@ -274,7 +274,7 @@
 let min_max_of_remainder x y =
   let r_abs_max =
     let min_y, max_y = Fd.min_max y in
-    max (Pervasives.abs min_y) (Pervasives.abs max_y) - 1 in
+    max (Stdlib.abs min_y) (Stdlib.abs max_y) - 1 in
   if Fd.min x >= 0 then (0, r_abs_max)
   else if Fd.max x <= 0 then ((0 - r_abs_max), 0)
   else ((0 - r_abs_max), r_abs_max)
diff -ub lib/fcl_sorting.ml lib.sic/fcl_sorting.ml
--- lib/fcl_sorting.ml	2017-11-30 20:24:10.000000000 +0100
+++ lib.sic/fcl_sorting.ml	2023-05-18 12:22:07.745882402 +0200
@@ -355,7 +355,7 @@
   else
     let inf, sup =
       Array.fold_left
-      	(fun (inf, sup) x -> Pervasives.min (Fd.min x) inf, Pervasives.max (Fd.max x) sup) (max_int, min_int) a in
+      	(fun (inf, sup) x -> Stdlib.min (Fd.min x) inf, Stdlib.max (Fd.max x) sup) (max_int, min_int) a in
     let d = Fd.array n inf sup
     and p = Fd.array n 0 (n - 1) in
     Fcl_cstr.post (cstr a ~p:(Some p) d);
@@ -368,7 +368,7 @@
   else
     let inf, sup =
       Array.fold_left
-      	(fun (inf, sup) x -> Pervasives.min (Fd.min x) inf, Pervasives.max (Fd.max x) sup) (max_int, min_int) a in
+      	(fun (inf, sup) x -> Stdlib.min (Fd.min x) inf, Stdlib.max (Fd.max x) sup) (max_int, min_int) a in
     let d = Fd.array n inf sup in
     Fcl_cstr.post (cstr a d);
     d;;

