diff -Nur klibc-1.5.orig/Kbuild klibc-1.5/Kbuild
--- klibc-1.5.orig/Kbuild	2007-03-04 02:52:10.000000000 +0100
+++ klibc-1.5/Kbuild	2008-05-14 12:45:02.000000000 +0200
@@ -10,9 +10,8 @@
 	$(Q)$(MAKE) $(klibc)=usr/kinit
 	$(Q)$(MAKE) $(klibc)=usr/dash
 	$(Q)$(MAKE) $(klibc)=usr/utils
-	$(Q)$(MAKE) $(klibc)=usr/gzip
 
 
 # Directories to visit during clean and install
-subdir- := scripts/basic klcc usr/klibc usr/dash usr/utils usr/gzip \
+subdir- := scripts/basic klcc usr/klibc usr/dash usr/utils \
 	   usr/kinit usr/klibc/tests
diff -Nur klibc-1.5.orig/usr/Kbuild klibc-1.5/usr/Kbuild
--- klibc-1.5.orig/usr/Kbuild	2007-03-04 02:52:10.000000000 +0100
+++ klibc-1.5/usr/Kbuild	2008-05-14 12:45:02.000000000 +0200
@@ -6,7 +6,7 @@
 
 include-subdir := include
 klibc-subdir := klibc
-usr-subdirs  := kinit utils dash gzip
+usr-subdirs  := kinit utils dash
 subdir-      := $(include-subdir) $(klibc-subdir) $(usr-subdirs)
 
 usr-subdirs  := $(addprefix _usr_,$(usr-subdirs))
diff -Nur klibc-1.5.orig/usr/utils/Kbuild klibc-1.5/usr/utils/Kbuild
--- klibc-1.5.orig/usr/utils/Kbuild	2008-05-14 12:44:47.000000000 +0200
+++ klibc-1.5/usr/utils/Kbuild	2008-05-14 12:45:20.000000000 +0200
@@ -2,9 +2,9 @@
 # Kbuild file for klib utils
 #
 
-progs := chroot dd mkdir mkfifo mknod mount pivot_root umount
+progs := chroot dd mkdir mkfifo mount pivot_root umount
 progs += true false sleep ln nuke minips cat
-progs += uname halt kill readlink cpio
+progs += uname halt readlink
 
 static-y := $(addprefix static/, $(progs))
 shared-y := $(addprefix shared/, $(progs))
