From 0d00a6ad3385e08eff0c50efe970af8d5d63c536 Mon Sep 17 00:00:00 2001
From: Robert Ancell <robert.ancell@canonical.com>
Date: Tue, 29 Mar 2011 05:50:12 +0000
Subject: editor: Load *.enums.xml as well as *.gschema.xml.  This was stopping some enums from being loaded.

Fixes: https://bugzilla.gnome.org/show_bug.cgi?id=636880
---
diff --git a/editor/dconf-schema.vala b/editor/dconf-schema.vala
index 2b34e28..c2654b5 100644
--- a/editor/dconf-schema.vala
+++ b/editor/dconf-schema.vala
@@ -371,7 +371,7 @@ public class SchemaList
         while ((info = i.next_file (null)) != null) {
             string name = info.get_name();
 
-            if (!name.has_suffix(".gschema.xml"))
+            if (!name.has_suffix(".gschema.xml") && !name.has_suffix(".enums.xml"))
                 continue;
 
             string path = Path.build_filename(dir, name, null);
--
cgit v0.9
