From 5301e7349ba4d1c3f3808829947dab8ee2ac73b0 Mon Sep 17 00:00:00 2001
From: anthraxx <levente@leventepolyak.net>
Date: Wed, 21 Mar 2018 02:12:52 +0100
Subject: [PATCH] make: unfuck paths, flags and modes

---
 Makefile.in | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/Makefile.in b/Makefile.in
index 6d6ee03..e931934 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -3,6 +3,6 @@ CC?=@CC@
 INSTALL=@INSTALL@
 prefix=@prefix@
-CFLAGS=@CFLAGS@ @DEFS@ -Wall -I. -DVERSION=\"$(VERSION)\" -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -DPREFIX=\"$(prefix)\" @DEBUG@
+CFLAGS=@CFLAGS@ @CPPFLAGS@ @DEFS@ -Wall -I. -DVERSION=\"$(VERSION)\" -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -DPREFIX=\"$(prefix)\" @DEBUG@
 LIBS=@LIBS@
 mandir=@mandir@
 sbindir=@sbindir@
@@ -15,7 +15,7 @@ OBJS=bftpdutmp.o commands.o commands_admin.o cwd.o dirlist.o list.o login.o logg
 SRCS=bftpdutmp.c commands.c commands_admin.c cwd.c dirlist.c list.c login.c logging.c main.c mystring.c options.c md5.c
 
 OBJS2LINK=$(OBJS) $(PAX)
-LDFLAGS=
+LDFLAGS?=@LDFLAGS@
 
 all: bftpd
 
@@ -26,13 +26,13 @@ bftpd: $(OBJS)
 $(OBJS): $(HEADERS) Makefile
 
 install: all
-	mkdir -p $(DESTDIR)/$(prefix)/sbin
-	$(INSTALL) -g 0 -m 700 -o 0 bftpd $(DESTDIR)/$(prefix)/sbin
+	mkdir -p $(DESTDIR)/$(sbindir)
+	$(INSTALL) -g 0 -m 755 -o 0 bftpd $(DESTDIR)/$(sbindir)
 	mkdir -p $(DESTDIR)/$(mandir)/man8
 	$(INSTALL) -g 0 -m 644 -o 0 bftpd.8 $(DESTDIR)/$(mandir)/man8
-	mkdir -p $(DESTDIR)/$(prefix)/etc
-	[ -f $(DESTDIR)/$(prefix)/etc/bftpd.conf ] || \
-		$(INSTALL) -g 0 -m 600 -o 0 bftpd.conf $(DESTDIR)/$(prefix)/etc
+	mkdir -p $(DESTDIR)/etc
+	[ -f $(DESTDIR)/etc/bftpd.conf ] || \
+		$(INSTALL) -g 0 -m 600 -o 0 bftpd.conf $(DESTDIR)/etc
 	mkdir -p $(DESTDIR)/$(prefix)/var/log
 	touch $(DESTDIR)/$(prefix)/var/log/bftpd.log
 	chmod 644 $(DESTDIR)/$(prefix)/var/log/bftpd.log
@@ -49,8 +49,8 @@ newversion: clean
 	./configure --enable-pax=pax --enable-libz --enable-pam
 
 uninstall:
-	rm -f $(DESTDIR)/$(prefix)/sbin/bftpd $(DESTDIR)/$(mandir)/man8/bftpd.8 \
-		$(DESTDIR)/$(prefix)/etc/bftpd.conf
+	rm -f $(DESTDIR)/$(sbindir)/bftpd $(DESTDIR)/$(mandir)/man8/bftpd.8 \
+		$(DESTDIR)/etc/bftpd.conf
 
 distribute: install
 	rm -rf dist
-- 
2.16.2

