diff -ruN d4x.orig/main/face/edit.cc d4x.new/main/face/edit.cc
--- d4x.orig/main/face/edit.cc	2009-05-29 02:35:26.000000000 -0700
+++ d4x.new/main/face/edit.cc	2009-05-29 02:33:45.000000000 -0700
@@ -1450,7 +1450,7 @@
 
 static void _proxy_port_changed_(GtkEntry *entry,GtkEntry *entryh){
 	const char *tmp=gtk_entry_get_text(entryh);
-	char *tmp1=index(tmp,':');
+	char *tmp1 = (char *) index(tmp,':');
 	if (tmp1){
 		*tmp1=0;
 		char *ns=sum_strings(tmp,":",gtk_entry_get_text(entry),NULL);
diff -ruN d4x.orig/main/ftp.cc d4x.new/main/ftp.cc
--- d4x.orig/main/ftp.cc	2009-05-29 02:35:02.000000000 -0700
+++ d4x.new/main/ftp.cc	2009-05-29 02:32:41.000000000 -0700
@@ -261,7 +261,7 @@
 };
 
 static void d4x_ftp_parse_pasv(const char *str,int args[]){
-	char *a=index(str,'(');
+	char *a = (char *) index(str,'(');
 	if (a==NULL) return;
 	a+=1;
 	int i=0;
diff -ruN d4x.orig/main/html.cc d4x.new/main/html.cc
--- d4x.orig/main/html.cc	2009-05-29 02:35:09.000000000 -0700
+++ d4x.new/main/html.cc	2009-05-29 02:33:55.000000000 -0700
@@ -700,7 +700,7 @@
 
 void tHtmlParser::set_content_type(const char *ct){
 	//Example: text/html; charset=koi8-r
-	char *a=index(ct,'=');
+	const char *a=index(ct,'=');
 	if (a) codepage=a+1;
 };
 
diff -ruN d4x.orig/main/locstr.cc d4x.new/main/locstr.cc
--- d4x.orig/main/locstr.cc	2009-05-29 02:35:14.000000000 -0700
+++ d4x.new/main/locstr.cc	2009-05-29 02:32:47.000000000 -0700
@@ -281,7 +281,7 @@
 	DBC_RETVAL_IF_FAIL(str!=NULL,NULL);
 	DBC_RETVAL_IF_FAIL(what!=NULL,NULL);
 	while (*str){
-		char *a=index(what,*str);
+		const char *a=index(what,*str);
 		if (a) return(str);
 		str++;
 	};
@@ -472,16 +472,16 @@
 char *escape_char(const char *where,char what,char bywhat){
 	DBC_RETVAL_IF_FAIL(where!=NULL,NULL);
 	int num=0;
-	char *tmp=index(where,what);
+	char *tmp = (char *) index(where,what);
 	while(tmp){
 		num+=1;
-		tmp=index(tmp+1,what);
+		tmp = (char *) index(tmp+1,what);
 	};
 	if (num){
 		char *rvalue=new char[strlen(where)+num*2+1];
 		*rvalue=0;
 		char *r=rvalue;
-		tmp=index(where,what);
+		tmp = (char *) index(where,what);
 		while(tmp){
 			if (tmp-where)
 				memcpy(r,where,tmp-where);
@@ -490,7 +490,7 @@
 			r[1]=what;
 			r+=2;
 			where=tmp+1;
-			tmp=index(where,what);
+			tmp = (char *) index(where,what);
 		};
 		*r=0;
 		if (*where)
@@ -962,17 +962,17 @@
 	DBC_RETVAL_IF_FAIL(a!=NULL,NULL);
 	DBC_RETVAL_IF_FAIL(b!=NULL,NULL);
 	int i=0;
-	char *temp=index(b,'/');
+	char *temp = (char *) index(b,'/');
 	while (temp){
 		while (*temp=='/') temp+=1;
 		temp=index(temp,'/');
 		i+=1;
 	};
 	i-=1;
-	temp=rindex(a,'/');
+	temp = (char *) rindex(a,'/');
 	while (temp && i>0){
 		*temp=0;
-		char *tmp=rindex(a,'/');
+		char *tmp = (char *) rindex(a,'/');
 		*temp='/';
 		temp=tmp;
 		i-=1;
