--- trunk/banshee/src/Core/Banshee.Widgets/Banshee.Widgets/VolumeButton.cs	2007/11/05 22:18:55	2682
+++ trunk/banshee/src/Core/Banshee.Widgets/Banshee.Widgets/VolumeButton.cs	2008/03/21 05:19:23	3504
@@ -35,6 +35,8 @@
  *  DEALINGS IN THE SOFTWARE.
  */
 
+#pragma warning disable 0612
+
 using System;
 using System.Runtime.InteropServices;
 
@@ -693,3 +695,5 @@
         }
     }
 }
+
+#pragma warning restore 0612
--- trunk/banshee/src/Core/Banshee.ThickClient/Banshee.Gui.Dialogs/TrackEditor.cs	2008/03/20 00:32:14	3488
+++ trunk/banshee/src/Core/Banshee.ThickClient/Banshee.Gui.Dialogs/TrackEditor.cs	2008/03/21 05:12:40	3503
@@ -26,6 +26,8 @@
 // WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 //
 
+#pragma warning disable 0612
+
 using System;
 using Gtk;
 using Glade;
@@ -634,3 +636,5 @@
         }
     }
 }
+
+#pragma warning restore 0612
--- trunk/banshee/src/Core/Banshee.ThickClient/Banshee.Gui.Dialogs/PreferencesDialog.cs	2008/01/31 05:38:19	3114
+++ trunk/banshee/src/Core/Banshee.ThickClient/Banshee.Gui.Dialogs/PreferencesDialog.cs	2008/03/21 05:12:40	3503
@@ -26,6 +26,8 @@
 // WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 //
 
+#pragma warning disable 0612
+
 using System;
 using System.IO;
 
@@ -188,3 +190,5 @@
         }
     }
 }
+
+#pragma warning restore 0612
--- trunk/banshee/src/Core/Banshee.ThickClient/Banshee.Gui.Dialogs/AboutDialog.cs	2008/03/16 19:42:55	3468
+++ trunk/banshee/src/Core/Banshee.ThickClient/Banshee.Gui.Dialogs/AboutDialog.cs	2008/03/21 05:12:40	3503
@@ -25,6 +25,8 @@
 // OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
 // WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 //
+
+#pragma warning disable 0618
  
 using System;
 using System.Text;
@@ -76,6 +78,9 @@
                 Banshee.Web.Browser.Open (link);
             });
             
+            // TODO: We should really use ProgramName in the future rather
+            // than plain Name, since it's been depreciated. We can't do that
+            // yet though since it breaks stuff for other people. 
             Name = "Banshee";
             Logo = Gdk.Pixbuf.LoadFromResource ("banshee-logo.png");
             Version = String.Format ("{0} ({1})", 
@@ -99,3 +104,5 @@
         }
     }
 }
+
+#pragma warning restore 0618
