From 1a00809a0d89b0849a57647c878d192354247a33 Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <jan.steffens@gmail.com>
Date: Tue, 1 Oct 2019 14:07:55 +0000
Subject: [PATCH 4/4] config: Fall back to a default submission nick if lookup
 fails

---
 src/gclue-config.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/gclue-config.c b/src/gclue-config.c
index 76222ff..95f78df 100644
--- a/src/gclue-config.c
+++ b/src/gclue-config.c
@@ -218,6 +218,7 @@ load_enable_source_config (GClueConfig *config,
 
 #define DEFAULT_WIFI_URL "https://location.services.mozilla.com/v1/geolocate?key=" MOZILLA_API_KEY
 #define DEFAULT_WIFI_SUBMIT_URL "https://location.services.mozilla.com/v1/submit?key=" MOZILLA_API_KEY
+#define DEFAULT_WIFI_SUBMIT_NICK "geoclue"
 
 static void
 load_wifi_config (GClueConfig *config)
@@ -269,6 +270,7 @@ load_wifi_config (GClueConfig *config)
                 g_debug ("Failed to get config \"wifi/submission-nick\": %s",
                          error->message);
                 g_error_free (error);
+                priv->wifi_submit_nick = g_strdup (DEFAULT_WIFI_SUBMIT_NICK);
         }
 }
 
-- 
2.23.0

