diff --git a/src/modules/avformat/producer_avformat.c b/src/modules/avformat/producer_avformat.c
index 9e8a63df..82a412bc 100644
--- a/src/modules/avformat/producer_avformat.c
+++ b/src/modules/avformat/producer_avformat.c
@@ -1942,14 +1942,14 @@ static int producer_get_image( mlt_frame frame, uint8_t **buffer, mlt_image_form
 					av_packet_free( &tmp );
 					pthread_cond_signal( &self->packets_cond );
 				} else {
-					// notify packets_worker that we've seen the error
-					self->packets_thread_ret = 0;
-					pthread_cond_signal( &self->packets_cond );
-
 					if ( self->packets_thread_ret == AVERROR_EOF )
 					{
 						self->pkt.stream_index = self->video_index;
 					}
+					// notify packets_worker that we've seen the error
+					self->packets_thread_ret = 0;
+					pthread_cond_signal( &self->packets_cond );
+
 					if ( !self->video_seekable && mlt_properties_get_int( properties, "reconnect" ) )
 					{
 						// Try to reconnect to live sources by closing context and codecs,
