diff -Nur gnome-shell.orig/data/org.gnome.shell.gschema.xml.in gnome-shell/data/org.gnome.shell.gschema.xml.in
--- gnome-shell.orig/data/org.gnome.shell.gschema.xml.in	2011-04-06 08:47:58.638600793 -0700
+++ gnome-shell/data/org.gnome.shell.gschema.xml.in	2011-04-06 08:49:17.035274675 -0700
@@ -30,7 +30,7 @@
       </_description>
     </key>
     <key name="favorite-apps" type="as">
-      <default>[ 'mozilla-firefox.desktop', 'evolution.desktop', 'empathy.desktop', 'rhythmbox.desktop', 'shotwell.desktop', 'openoffice.org-writer.desktop', 'nautilus.desktop' ]</default>
+      <default>[ 'firefox.desktop', 'evolution.desktop', 'empathy.desktop', 'rhythmbox.desktop', 'shotwell.desktop', 'writer.desktop', 'nautilus.desktop' ]</default>
       <_summary>List of desktop file IDs for favorite applications</_summary>
       <_description>
         The applications corresponding to these identifiers
