From 0646f34fc3eeb47627822df860986cf7bee4ebed Mon Sep 17 00:00:00 2001
From: Gisle Aas <gisle@activestate.com>
Date: Thu, 28 Mar 2013 00:41:07 -0700
Subject: [PATCH] eval { 1->method } triggers different error in 5.17.11

---
 t/02_____destroy.t |    4 +++-
 1 files changed, 3 insertions(+), 1 deletions(-)

diff --git a/t/02_____destroy.t b/t/02_____destroy.t
index 41e3aa3..adcf995 100644
--- a/t/02_____destroy.t
+++ b/t/02_____destroy.t
@@ -17,7 +17,9 @@ if (ref($set) eq '')
 {print "ok $n\n";} else {print "not ok $n\n";}
 $n++;
 eval { $set->DESTROY(); };
-if ($@ =~ /Can't call method "DESTROY" without a package or object reference/)
+if ($@ =~ /Can't call method "DESTROY" without a package or object reference/ ||
+    $@ =~ /Can't locate object method "DESTROY" via package "1"/
+)
 {print "ok $n\n";} else {print "not ok $n\n";}
 $n++;
 eval { Bit::Vector::DESTROY($set); };
-- 
1.7.0.5

