From: arvidn <arvid@libtorrent.org>
Date: Tue, 28 Nov 2017 23:57:38 +0100
Subject: [PATCH] update io_service forward declaration to be boost-1.66
 compatible

diff --git a/include/libtorrent/io_service_fwd.hpp b/include/libtorrent/io_service_fwd.hpp
index c14997d82e..398c91216f 100644
--- a/include/libtorrent/io_service_fwd.hpp
+++ b/include/libtorrent/io_service_fwd.hpp
@@ -54,7 +54,12 @@ namespace sim { namespace asio {
 #endif
 
 namespace boost { namespace asio {
+#if BOOST_VERSION < 106600
 	class io_service;
+#else
+	class io_context;
+	typedef io_context io_service;
+#endif
 }}
 
 namespace libtorrent
