--- libproxy-0.4.11/libproxy/modules/pxgsettings.cpp.orig	2013-05-30 13:18:11.597138996 +0200
+++ libproxy-0.4.11/libproxy/modules/pxgsettings.cpp	2013-05-30 13:18:19.801139269 +0200
@@ -156,10 +156,7 @@
 
 	g_main_loop_run(loop);
 
-	// Cleanup
-	while (G_IS_OBJECT(client)) {
-		g_object_unref(client);
-	}
+	g_object_unref(client);
 	g_io_channel_shutdown(inchan,  FALSE, NULL);
 	g_io_channel_shutdown(outchan, FALSE, NULL);
 	g_io_channel_unref(inchan);
