diff --git a/core/libs/threadimageio/video/videodecoder_p.h b/core/libs/threadimageio/video/videodecoder_p.h
index 251e092809..d82926b77e 100644
--- a/core/libs/threadimageio/video/videodecoder_p.h
+++ b/core/libs/threadimageio/video/videodecoder_p.h
@@ -56,7 +56,7 @@ public:
     AVFormatContext*   pFormatContext;
     AVCodecContext*    pVideoCodecContext;
     AVCodecParameters* pVideoCodecParameters;
-    AVCodec*           pVideoCodec;
+    const AVCodec*     pVideoCodec;
     AVStream*          pVideoStream;
     AVFrame*           pFrame;
     quint8*            pFrameBuffer;
