diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7d075a7223..6dc88e243a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1052,10 +1052,12 @@ list (APPEND ANDROID_EXTRA_LIBS ${LCMS2_LIBRARIES})
 ##
 ## Test for xsimd
 ##
-find_package(xsimd 8.1.0)
-if(NOT xsimd_FOUND)
-    find_package(xsimd 9)
-endif()
+foreach(xsimd_version 8.1.0 9 10)
+    if(NOT xsimd_FOUND)
+        find_package(xsimd ${xsimd_version})
+    endif()
+endforeach()
+
 set_package_properties(xsimd PROPERTIES
     DESCRIPTION "C++ wrappers for SIMD intrinsics"
     URL "https://github.com/xtensor-stack/xsimd"
