--- usbutils-002/lsusb.py	2010-12-16 01:07:09.000000000 +0100
+++ usbutils-002/lsusb.py	2011-04-09 22:43:59.043828595 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # lsusb.py
 # Displays your USB devices in reasonable form.
 # (c) Kurt Garloff <garloff@suse.de>, 2/2009, GPL v2 or v3.
@@ -16,7 +16,7 @@
 warnsort = False
 
 prefix = "/sys/bus/usb/devices/"
-usbids = "/usr/share/usb.ids"
+usbids = "/usr/share/hwdata/usb.ids"
 
 esc = chr(27)
 norm = esc + "[0;0m"
