--- ispell-3.3.02/correct.c	2005-04-28 09:46:51.000000000 -0500
+++ correct.c	2010-05-20 16:36:37.000000000 -0500
@@ -245,7 +245,7 @@
 		  struct flagent * sufent,
 		  ichar_t savearea[MAX_CAPS][INPUTWORDLEN + MAXAFFIXLEN],
 		  int * nsaved));
-static char *	getline P ((char * buf, int bufsize));
+static char *	mygetline P ((char * buf, int bufsize));
 void		askmode P ((void));
 void		copyout P ((unsigned char ** cc, int cnt));
 static void	lookharder P ((unsigned char * string));
@@ -571,7 +571,7 @@
 
 		imove (li - 1, 0);
 		(void) putchar ('!');
-		if (getline ((char *) buf, sizeof buf) == NULL)
+		if (mygetline ((char *) buf, sizeof buf) == NULL)
 		    {
 		    (void) putchar (7);
 		    ierase ();
@@ -596,7 +596,7 @@
 		    (void) printf ("%s ", CORR_C_READONLY);
 		    }
 		(void) printf (CORR_C_REPLACE_WITH);
-		if (getline ((char *) ctok, ctokl) == NULL)
+		if (mygetline ((char *) ctok, ctokl) == NULL)
 		    {
 		    (void) putchar (7);
 		    /* Put it back */
@@ -664,7 +664,7 @@
 		unsigned char	buf[100];
 		imove (li - 1, 0);
 		(void) printf (CORR_C_LOOKUP_PROMPT);
-		if (getline ((char *) buf, sizeof buf) == NULL)
+		if (mygetline ((char *) buf, sizeof buf) == NULL)
 		    {
 		    (void) putchar (7);
 		    ierase ();
@@ -1583,7 +1583,7 @@
     return;
     }
 
-static char * getline (s, len)
+static char * mygetline (s, len)
     register char *	s;
     register int	len;
     {
