diff --git a/cmd/gvedit/csettings.cpp b/cmd/gvedit/csettings.cpp
index abd321b61768126ab76c43609bf2f96f303f2a82..3c28818ef346bfdb78e46d567b0c2b968e3cf924 100644
--- a/cmd/gvedit/csettings.cpp
+++ b/cmd/gvedit/csettings.cpp
@@ -16,7 +16,7 @@
 #include "csettings.h"
 #include "qmessagebox.h"
 #include "qfiledialog.h"
-#include <QtGui>
+#include <QtWidgets>
 #include <qfile.h>
 #include "mdichild.h"
 #include "string.h"
diff --git a/cmd/gvedit/imageviewer.h b/cmd/gvedit/imageviewer.h
index 5145c6a693cea2927b5ffb05719825bdbd289f14..9786aafe6066a341f4b023d620c102a798f98eb0 100644
--- a/cmd/gvedit/imageviewer.h
+++ b/cmd/gvedit/imageviewer.h
@@ -15,7 +15,7 @@
 
 #ifndef IMAGEVIEWER_H
 #define IMAGEVIEWER_H
-#include <QtGui>
+#include <QtWidgets>
 #include <QMainWindow>
 #include <QPrinter>
 
diff --git a/cmd/gvedit/mainwindow.cpp b/cmd/gvedit/mainwindow.cpp
index c5c9dbd6d8f89335768099d5ee559bbdcb534ef5..0caab977ba04e37a23898f18930b1365d1ffceea 100644
--- a/cmd/gvedit/mainwindow.cpp
+++ b/cmd/gvedit/mainwindow.cpp
@@ -11,7 +11,7 @@
  * Contributors: See CVS logs. Details at http://www.graphviz.org/
  *************************************************************************/
 
-#include <QtGui>
+#include <QtWidgets>
 #include <qframe.h>
 #include "mainwindow.h"
 #include "mdichild.h"
diff --git a/cmd/gvedit/mdichild.cpp b/cmd/gvedit/mdichild.cpp
index f416b6532980834c4d2a5cde9e8fc6a6bae3b337..457d34b0841f50e3d60b720129a5c6e58072a8a4 100644
--- a/cmd/gvedit/mdichild.cpp
+++ b/cmd/gvedit/mdichild.cpp
@@ -12,7 +12,7 @@
  *************************************************************************/
 
 
-#include <QtGui>
+#include <QtWidgets>
 
 #include "mdichild.h"
 #include "mainwindow.h"
diff --git a/cmd/gvedit/ui_settings.h b/cmd/gvedit/ui_settings.h
index 79c73d84fed1ba6d695d9742c94125ce19ac36be..e5c274a90273e2ddbd55e0489f96bc942e3446ac 100644
--- a/cmd/gvedit/ui_settings.h
+++ b/cmd/gvedit/ui_settings.h
@@ -10,22 +10,22 @@
 #ifndef UI_SETTINGS_H
 #define UI_SETTINGS_H
 
-#include <QtCore/QVariant>
-#include <QtGui/QAction>
-#include <QtGui/QApplication>
-#include <QtGui/QButtonGroup>
-#include <QtGui/QComboBox>
-#include <QtGui/QDialog>
-#include <QtGui/QFrame>
-#include <QtGui/QHBoxLayout>
-#include <QtGui/QHeaderView>
-#include <QtGui/QLabel>
-#include <QtGui/QLineEdit>
-#include <QtGui/QPushButton>
-#include <QtGui/QSpacerItem>
-#include <QtGui/QTextEdit>
-#include <QtGui/QVBoxLayout>
-#include <QtGui/QWidget>
+#include <QVariant>
+#include <QAction>
+#include <QApplication>
+#include <QButtonGroup>
+#include <QComboBox>
+#include <QDialog>
+#include <QFrame>
+#include <QHBoxLayout>
+#include <QHeaderView>
+#include <QLabel>
+#include <QLineEdit>
+#include <QPushButton>
+#include <QSpacerItem>
+#include <QTextEdit>
+#include <QVBoxLayout>
+#include <QWidget>
 
 QT_BEGIN_NAMESPACE
 
diff --git a/configure.ac b/configure.ac
index c80fd64dcf68889b8f04fb48eb5e9f7666f8c43a..811e46dae4bc665eb85f52b7ba271f4c3a22ff9c 100644
--- a/configure.ac
+++ b/configure.ac
@@ -2332,10 +2332,10 @@ if test "x$with_qt" != "xyes"; then
   use_qt="No (disabled)"
 else
 
-  AC_CHECK_PROGS(QMAKE,qmake-qt4 qmake-qt3 qmake,false)
+  AC_CHECK_PROGS(QMAKE,qmake-qt5 qmake,false)
   if test "$QMAKE" != "false"; then
-    PKG_CHECK_MODULES(QTCORE, [QtCore],[
-      PKG_CHECK_MODULES(QTGUI, [QtGui],[
+    PKG_CHECK_MODULES(QTCORE, [Qt5Core],[
+      PKG_CHECK_MODULES(QTGUI, [Qt5Widgets Qt5PrintSupport],[
 	use_qt="Yes"
       ],[
         use_qt="No (QtGui not available)"
