--- net-tools-1.60.20110819cvs/include/mii.h	2011-08-19 10:14:00.743873299 +0200
+++ net-tools-1.60.20110819cvs/include/mii.h.new	2011-08-19 10:28:26.500524160 +0200
@@ -14,11 +14,9 @@
 /* network interface ioctl's for MII commands */
 #ifndef SIOCGMIIPHY
 #warning "SIOCGMIIPHY is not defined by your kernel source"
-#define SIOCGMIIPHY (SIOCDEVPRIVATE)	/* Read from current PHY */
-#define SIOCGMIIREG (SIOCDEVPRIVATE+1) 	/* Read any PHY register */
-#define SIOCSMIIREG (SIOCDEVPRIVATE+2) 	/* Write any PHY register */
-#define SIOCGPARAMS (SIOCDEVPRIVATE+3) 	/* Read operational parameters */
-#define SIOCSPARAMS (SIOCDEVPRIVATE+4) 	/* Set operational parameters */
+#define SIOCGMIIPHY 0x8947     /* Read from current PHY */
+#define SIOCGMIIREG 0x8948     /* Read any PHY register */
+#define SIOCSMIIREG 0x8949     /* Write any PHY register */
 #endif
 
 #include <linux/types.h>
