diff -u -r postgresql-15.1.orig/src/pl/plperl/GNUmakefile postgresql-15.1/src/pl/plperl/GNUmakefile
--- postgresql-15.1.orig/src/pl/plperl/GNUmakefile	2022-02-07 21:15:36.000000000 +0000
+++ postgresql-15.1/src/pl/plperl/GNUmakefile	2022-04-19 18:53:33.738551038 +0000
@@ -55,6 +55,9 @@
 
 SHLIB_LINK = $(perl_embed_ldflags)
 
+# Force rpath to be used even though we disable it everywhere else
+SHLIB_LINK += $(rpath)
+
 REGRESS_OPTS = --dbname=$(PL_TESTDB)
 REGRESS = plperl_setup plperl plperl_lc plperl_trigger plperl_shared \
 	plperl_elog plperl_util plperl_init plperlu plperl_array \
