diff -Naur postgresql-9.1.5.orig/src/pl/plperl/GNUmakefile postgresql-9.1.5/src/pl/plperl/GNUmakefile
--- postgresql-9.1.5.orig/src/pl/plperl/GNUmakefile	2012-08-14 18:41:04.000000000 -0400
+++ postgresql-9.1.5/src/pl/plperl/GNUmakefile	2012-08-17 11:15:09.457116708 -0400
@@ -43,6 +43,9 @@
 
 SHLIB_LINK = $(perl_embed_ldflags)
 
+# Force rpath to be used even though we disable it everywhere else
+SHLIB_LINK += $(rpath)
+
 REGRESS_OPTS = --dbname=$(PL_TESTDB) --load-extension=plperl  --load-extension=plperlu
 REGRESS = plperl plperl_lc plperl_trigger plperl_shared plperl_elog plperl_util plperl_init plperlu plperl_array
 # if Perl can support two interpreters in one backend,
