diff -ru cantor.orig/cmake/FindPythonLibs3.cmake cantor/cmake/FindPythonLibs3.cmake
--- cantor.orig/cmake/FindPythonLibs3.cmake	2015-09-20 11:54:42.206749420 +0200
+++ cantor/cmake/FindPythonLibs3.cmake	2015-09-20 11:55:00.896859349 +0200
@@ -5,7 +5,7 @@
 find_path(PYTHONLIBS3_INCLUDE_DIR Python.h
           HINTS ${PYTHONLIBS3_INCLUDEDIR} ${PYTHONLIBS3_INCLUDE_DIRS})
 
-find_library(PYTHONLIBS3_LIBRARY NAMES python3.2m python3.3m python3.4m
+find_library(PYTHONLIBS3_LIBRARY NAMES python3.2m python3.3m python3.4m python3.5m
              HINTS ${PYTHONLIBS3_LIBDIR} ${PYTHONLIBS3_LIBRARY_DIRS} )
 
 set(PYTHONLIBS3_LIBRARIES ${PYTHONLIBS3_LIBRARY} )
