diff -upr libpst-0.6.76.orig/configure libpst-0.6.76/configure
--- libpst-0.6.76.orig/configure	2021-03-28 02:08:10.000000000 +0200
+++ libpst-0.6.76/configure	2021-12-03 16:24:53.025153596 +0200
@@ -24672,7 +24672,7 @@ if test "$enable_python" = "yes"; then
 $as_echo_n "checking for python build information... " >&6; }
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: " >&5
 $as_echo "" >&6; }
-for python in python3.9 python3.8 python3.7 python3.6 python3.5 python3.3 python3.2 python3.1 python3.0 python2.7 python2.6 python2.5 python2.4 python2.3 python2.2 python2.1 python; do
+for python in python3.10 python3.9 python3.8 python3.7 python3.6 python3.5 python3.3 python3.2 python3.1 python3.0 python2.7 python2.6 python2.5 python2.4 python2.3 python2.2 python2.1 python; do
 for ac_prog in $python
 do
   # Extract the first word of "$ac_prog", so it can be a program name with args.
