--- configure.ac.orig	2013-03-13 16:09:02.355422066 +0000
+++ configure.ac	2013-03-13 16:09:26.701917153 +0000
@@ -269,8 +269,6 @@
 			 -Wchar-subscripts -Wwrite-strings \
 			 -Wdeclaration-after-statement -Wnested-externs \
 			 -Wmissing-noreturn \
-			 -Werror=missing-prototypes -Werror=nested-externs \
-			 -Werror=implicit-function-declaration \
 			 -Wmissing-declarations -Wno-pointer-sign \
 			 -Werror=format-security -Wbitwise -Wcast-to-as \
 			 -Wdefault-bitfield-sign -Wdo-while -Wparen-string \
