diff -Nur comm-beta.orig/config/autoconf.mk.in comm-beta/config/autoconf.mk.in
--- comm-beta.orig/config/autoconf.mk.in	2011-11-06 14:39:51.064983780 +0000
+++ comm-beta/config/autoconf.mk.in	2011-11-06 14:42:57.216735333 +0000
@@ -65,8 +65,8 @@
 mandir		= @mandir@
 idldir		= $(datadir)/idl/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
 
-installdir	= $(libdir)/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
-sdkdir		= $(libdir)/$(MOZ_APP_NAME)-devel-$(MOZ_APP_VERSION)
+installdir	= $(libdir)/$(MOZ_APP_NAME)
+sdkdir		= $(libdir)/$(MOZ_APP_NAME)-devel
 
 MOZDEPTH	= $(DEPTH)/mozilla
 DIST		= $(MOZDEPTH)/dist
diff -Nur comm-beta.orig/mozilla/config/autoconf.mk.in comm-beta/mozilla/config/autoconf.mk.in
--- comm-beta.orig/mozilla/config/autoconf.mk.in	2011-11-06 14:39:41.771562967 +0000
+++ comm-beta/mozilla/config/autoconf.mk.in	2011-11-06 14:44:18.404165515 +0000
@@ -68,8 +68,8 @@
 mandir		= @mandir@
 idldir		= $(datadir)/idl/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
 
-installdir	= $(libdir)/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
-sdkdir		= $(libdir)/$(MOZ_APP_NAME)-devel-$(MOZ_APP_VERSION)
+installdir	= $(libdir)/$(MOZ_APP_NAME)
+sdkdir		= $(libdir)/$(MOZ_APP_NAME)-devel
 
 DIST		= $(DEPTH)/dist
 LIBXUL_SDK      = @LIBXUL_SDK@
diff -Nur comm-beta.orig/mozilla/js/src/config/autoconf.mk.in comm-beta/mozilla/js/src/config/autoconf.mk.in
--- comm-beta.orig/mozilla/js/src/config/autoconf.mk.in	2011-11-06 14:39:48.014955070 +0000
+++ comm-beta/mozilla/js/src/config/autoconf.mk.in	2011-11-06 14:44:32.977635875 +0000
@@ -61,8 +61,8 @@
 datadir		= @datadir@
 mandir		= @mandir@
 
-installdir	= $(libdir)/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
-sdkdir		= $(libdir)/$(MOZ_APP_NAME)-devel-$(MOZ_APP_VERSION)
+installdir	= $(libdir)/$(MOZ_APP_NAME)
+sdkdir		= $(libdir)/$(MOZ_APP_NAME)-devel
 
 TOP_DIST	= @TOP_DIST@
 ifneq (,$(filter /%,$(TOP_DIST)))
