--- a/makefile	2013-04-29 16:27:05.000000000 +0200
+++ b/makefile	2013-10-01 22:07:29.611485989 +0200
@@ -146,7 +146,7 @@
 lib:	clean $(OBJECTS) $(LIB_OBJ)
 	@rm -f libunrar.so
	@rm -f libunrar.a
-	$(LINK) -shared -o libunrar.so $(LDFLAGS) $(OBJECTS) $(LIB_OBJ)
+	$(LINK) -shared -Wl,-soname,libunrar.so.$(libversion) -o libunrar.so.$(libversion) $(LDFLAGS) $(OBJECTS) $(LIB_OBJ)
 	$(AR) rcs libunrar.a $(OBJECTS) $(LIB_OBJ)

 install-unrar:
