--- xfce4-mailwatch-plugin-1.1.0/panel-plugin/Makefile.am	2008-08-26 03:42:36.000000000 +0100
+++ xfce4-mailwatch-plugin-1.1.0/panel-plugin/Makefile.am.new	2011-01-14 17:27:44.116199383 +0000
@@ -50,6 +50,7 @@
 	-I$(top_srcdir)/libmailwatch-core \
 	@GTHREAD_CFLAGS@ \
 	@LIBXFCE4PANEL_CFLAGS@ \
+	@LIBXFCEGUI4_CFLAGS@ \
 	-DDATADIR=\"$(datadir)\" \
 	-DLOCALEDIR=\"$(localedir)\"
 
@@ -60,7 +61,9 @@
 	$(top_builddir)/libmailwatch-core/libmailwatch-core.la \
 	@GTHREAD_LIBS@ \
 	@LIBXFCE4PANEL_LIBS@ \
-	@GNUTLS_LIBS@
+	@LIBXFCEGUI4_LIBS@ \
+	@GNUTLS_LIBS@ \
+	-lgcrypt
 
 if HAVE_CYGWIN
   xfce4_mailwatch_plugin_LDFLAGS += \
