diff -ur xfce4-genmon-plugin-3.2.orig/configure.ac xfce4-genmon-plugin-3.2/configure.ac
--- xfce4-genmon-plugin-3.2.orig/configure.ac	2008-03-15 23:38:12.000000000 +0200
+++ xfce4-genmon-plugin-3.2/configure.ac	2010-08-08 15:49:35.000000000 +0300
@@ -32,6 +32,7 @@
 
 dnl configure the panel plugin
 XDT_CHECK_PACKAGE([LIBXFCE4PANEL], [libxfce4panel-1.0], [4.3.22])
+XDT_CHECK_PACKAGE([LIBXFCEGUI4], [libxfcegui4-1.0], [4.3.22])
 
 dnl Translations
 XDT_I18N([ca cs en_GB eu fr pl pt_PT sq])
diff -ur xfce4-genmon-plugin-3.2.orig/panel-plugin/Makefile.am xfce4-genmon-plugin-3.2/panel-plugin/Makefile.am
--- xfce4-genmon-plugin-3.2.orig/panel-plugin/Makefile.am	2007-11-23 17:32:40.000000000 +0200
+++ xfce4-genmon-plugin-3.2/panel-plugin/Makefile.am	2010-08-08 15:50:27.000000000 +0300
@@ -3,10 +3,12 @@
 
 xfce4_genmon_plugin_CFLAGS =					\
 	-DPACKAGE_LOCALE_DIR=\"$(localedir)\"			\
-	@LIBXFCE4PANEL_CFLAGS@
+	@LIBXFCE4PANEL_CFLAGS@					\
+	@LIBXFCEGUI4_CFLAGS@
 
 xfce4_genmon_plugin_LDADD =					\
-	@LIBXFCE4PANEL_LIBS@
+	@LIBXFCE4PANEL_LIBS@					\
+	@LIBXFCEGUI4_LIBS@
 
 xfce4_genmon_plugin_SOURCES =	\
 	main.c			\
