diff -ur minicom-2.1-orig/src/window.c minicom-2.1/src/window.c
--- minicom-2.1-orig/src/window.c	2005-05-28 19:59:42.000000000 -0400
+++ minicom-2.1/src/window.c	2005-05-28 20:00:22.000000000 -0400
@@ -62,7 +62,7 @@
 static char *CS, *SF, *SR, *VB, *BL;
 static char *VE, *VI, *KS, *KE;
 static char *CD, *CL, *IC, *DC;
-static char *BC, *CR, *NL;
+static char *CR, *NL;
 #if ST_LINE
 static char *TS, *FS, *DS;
 #endif
