--- motion-3.2.11.1/picture.c~	2010-01-19 21:16:45.621459550 +0100
+++ motion-3.2.11.1/picture.c	2010-01-19 21:17:56.818117906 +0100
@@ -52,7 +52,7 @@
     dest->jpegsize = dest->bufsize - dest->pub.free_in_buffer;
 }
 
-static GLOBAL(void) jpeg_mem_dest(j_compress_ptr cinfo, JOCTET* buf, size_t bufsize)
+static GLOBAL(void) _jpeg_mem_dest(j_compress_ptr cinfo, JOCTET* buf, size_t bufsize)
 {
     mem_dest_ptr dest;
 
@@ -131,7 +131,7 @@
     jpeg_set_quality(&cinfo, quality, TRUE);
     cinfo.dct_method = JDCT_FASTEST;
 
-    jpeg_mem_dest(&cinfo, dest_image, image_size);    // data written to mem
+    _jpeg_mem_dest(&cinfo, dest_image, image_size);    // data written to mem
     
     jpeg_start_compress (&cinfo, TRUE);
 
@@ -182,7 +182,7 @@
 
     jpeg_set_quality(&cjpeg, quality, TRUE);
     cjpeg.dct_method = JDCT_FASTEST;
-    jpeg_mem_dest(&cjpeg, dest_image, image_size);  // data written to mem
+    _jpeg_mem_dest(&cjpeg, dest_image, image_size);  // data written to mem
 
     jpeg_start_compress (&cjpeg, TRUE);
 
