diff -ur mbrowse-0.3.1-orig/configure mbrowse-0.3.1/configure
--- mbrowse-0.3.1-orig/configure	2003-03-12 17:37:41.000000000 +0100
+++ mbrowse-0.3.1/configure	2006-05-29 22:37:07.000000000 +0200
@@ -4699,7 +4699,7 @@
           }
 
           #ifdef HAVE_NETSNMP
-	    version2 = strdup( NetSnmpVersionInfo );
+	    version2 = strdup( netsnmp_get_version() );
           #else
             version2 = strdup( VersionInfo);
           #endif
Only in mbrowse-0.3.1: configure~
diff -ur mbrowse-0.3.1-orig/src/snmpstuff.c mbrowse-0.3.1/src/snmpstuff.c
--- mbrowse-0.3.1-orig/src/snmpstuff.c	2002-06-21 13:45:18.000000000 +0200
+++ mbrowse-0.3.1/src/snmpstuff.c	2006-05-29 22:36:35.000000000 +0200
@@ -359,7 +359,7 @@
       case TYPE_NSAPADDRESS:
       case TYPE_NETADDR:
       case TYPE_OTHER:
-      default:
+      default:;
     }
   }
   if (type == 0) {
Only in mbrowse-0.3.1/src: snmpstuff.c~

