diff -aur old/gtk3-im/gtkimcontextgcin.c new/gtk3-im/gtkimcontextgcin.c
--- old/gtk3-im/gtkimcontextgcin.c	2011-08-31 09:14:58.000000000 -0700
+++ new/gtk3-im/gtkimcontextgcin.c	2011-09-04 13:29:58.838065420 -0700
@@ -127,6 +127,8 @@
 {
   GdkWindow *client_window = context_xim->client_window;
   GdkScreen *screen = gdk_window_get_screen (client_window);
+  if (screen == NULL)
+	  return;
   GdkDisplay *display = gdk_screen_get_display (screen);
   if (!context_xim->gcin_ch) {
     if (!(context_xim->gcin_ch = gcin_im_client_open(GDK_DISPLAY_XDISPLAY(gdk_display_get_default()))))
