diff -aur old/configure new/configure
--- old/configure	2011-06-08 16:00:37.000000000 +0200
+++ new/configure	2011-07-05 08:02:33.223601351 +0200
@@ -164,7 +164,7 @@
 
 LIB='lib'
 if [ -d /lib64 ]; then
-LIB='lib64'
+LIB='lib'
 fi
 
 QT_IM='N'
@@ -188,7 +188,7 @@
 
 
 QT4_IM='N'
-QT4_IM_DIR=/usr/$LIB/qt4/plugins/inputmethods
+QT4_IM_DIR=/usr/$LIB/qt/plugins/inputmethods
 echo "testing if you have qt4 support"
 pkg-config --libs QtCore QtGui > /dev/null 2> /dev/null
 if [ $? = 0 ]; then
diff -aur old/qt4-im/Makefile new/qt4-im/Makefile
--- old/qt4-im/Makefile	2011-06-08 15:03:50.000000000 +0200
+++ new/qt4-im/Makefile	2011-07-05 08:04:15.717731695 +0200
@@ -1,5 +1,5 @@
 include ../config.mak
-QT=qt4
+QT=qt
 QTIM=$(QT)/plugins/inputmethods
 IMMODULES=$(libdir)/$(QTIM)
 
@@ -15,7 +15,7 @@
 	$(CXX) -E $(CFLAGS) $(INCS) $< > $@
 
 moc_gcin-qt.cpp:	gcin-qt.h
-	/usr/$(LIB)/$(QT)/bin/moc $< -o moc_gcin-qt.cpp
+	/usr/bin/moc $< -o moc_gcin-qt.cpp
 
 im-gcin.so: $(OBJS)
 	export LD_RUN_PATH=$(gcinlibdir) ;\
