--- ungrab-winmodem-20080126/ungrab-winmodem.c~	2010-05-17 21:29:24.261834039 +0200
+++ ungrab-winmodem-20080126/ungrab-winmodem.c	2010-05-17 21:29:24.289770941 +0200
@@ -76,7 +76,7 @@
 {
 	struct pci_dev *dev = NULL;
 	struct device *reldev;
-        while ((dev = pci_find_device(PCI_ANY_ID, PCI_ANY_ID, dev)) != NULL) {
+        while ((dev = pci_get_device(PCI_ANY_ID, PCI_ANY_ID, dev)) != NULL) {
 		if(pci_match_id(softmodem_pci_tbl, dev) && pci_dev_driver(dev) &&
 		   (reldev = get_device(&dev->dev)) ) {
 			printk(KERN_INFO "device %04x:%04x is grabbed by driver %s: try to release\n",
--- slmodem-2.9.11-20080126/drivers/amrmo_init.c~	2010-05-17 21:31:53.029933461 +0200
+++ slmodem-2.9.11-20080126/drivers/amrmo_init.c	2010-05-17 21:31:53.076443380 +0200
@@ -749,7 +749,7 @@
 #ifdef OLD_KERNEL
 	pci_for_each_dev(dev) {
 #else
-        while ((dev = pci_find_device(PCI_ANY_ID, PCI_ANY_ID, dev)) != NULL) {
+        while ((dev = pci_get_device(PCI_ANY_ID, PCI_ANY_ID, dev)) != NULL) {
 #endif
 		if(pci_match_id(amrmo_pci_tbl, dev) &&
 		   pci_dev_driver(dev)) {
