diff -u -r totem-3.8.2/data/totem-bugreport.py totem-3.8.2-bugreport/data/totem-bugreport.py
--- totem-3.8.2/data/totem-bugreport.py	2013-03-13 16:34:44.000000000 +0100
+++ totem-3.8.2-bugreport/data/totem-bugreport.py	2013-05-14 13:54:04.243699665 +0200
@@ -4,7 +4,7 @@
 from gi.repository import Gtk
 
 # Get the GStreamer version
-if os.system ('gst-typefind-0.10 --version') == 0:
+if os.system ('gst-typefind-1.0 --version') == 0:
 	# List the formats of the last files played
 	last_visited = 0
 	recent_manager = Gtk.RecentManager.get_default ()
@@ -15,7 +15,7 @@
 				last = recent.get_uri_display ()
 
 	if last != None:
-		file_handle = os.popen ('gst-typefind-0.10 "%s"' % (last))
+		file_handle = os.popen ('gst-typefind-1.0 "%s"' % (last))
 		parts = file_handle.read ().split (' ')
 		date = datetime.fromtimestamp (last_visited)
-		print 'Listened to a "%s" file on %s' % (parts.pop ().strip (), date.isoformat ())
+		print('Listened to a "%s" file on %s' % (parts.pop ().strip (), date.isoformat ()))
