---
 src/Makefile.am |    6 ++++++
 1 file changed, 6 insertions(+)

Index: xserver-xorg-video-ati/src/Makefile.am
===================================================================
--- xserver-xorg-video-ati.orig/src/Makefile.am	2007-10-06 02:44:47.000000000 +0200
+++ xserver-xorg-video-ati/src/Makefile.am	2007-10-06 02:48:34.000000000 +0200
@@ -192,3 +192,9 @@
 	theatre_reg.h \
 	atipciids.h \
 	atipcirename.h
+
+pcidatadir = $(datadir)/xserver-xorg/pci
+pcidata_DATA = ati.ids
+
+ati.ids: atipciids.h
+	awk '/^#define.*PCI_CHIP/ {print $$3}' ${srcdir}/atipciids.h | sed -e 's/0x/1002/' > ati.ids
