From a9746a469b522dded6828851617b3f3ee8c1bf89 Mon Sep 17 00:00:00 2001
From: Peter Hutterer <peter.hutterer@who-t.net>
Date: Tue, 27 Apr 2010 13:51:17 +1000
Subject: [PATCH] conf: add WACf, FUJ02e5 and FUJ02e7 to serial identifers.

Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>
---
 conf/50-wacom.conf |    8 ++++++++
 1 files changed, 8 insertions(+), 0 deletions(-)

diff --git a/conf/50-wacom.conf b/conf/50-wacom.conf
index e1869e9..9a02b0e 100644
--- a/conf/50-wacom.conf
+++ b/conf/50-wacom.conf
@@ -15,6 +15,14 @@ Section "InputClass"
 	Option "ForceDevice" "ISDV4"
 EndSection
 
+Section "InputClass"
+        Identifier "Wacom serial class identifiers"
+        MatchProduct "WACf|FUJ02e5|FUJ02e7"
+        Driver "wacom"
+        Option "ForceDevice" "ISDV4"
+EndSection
+
+
 # N-Trig Duosense Electromagnetic Digitizer
 Section "InputClass"
 	Identifier "Wacom N-Trig class"
-- 
1.6.6.1

