--- gnash-0.8.6/libbase/GnashImagePng.cpp~	2010-01-18 00:41:35.612492833 +0100
+++ gnash-0.8.6/libbase/GnashImagePng.cpp	2010-01-18 00:43:01.102478031 +0100
@@ -165,7 +165,11 @@
     if (type == PNG_COLOR_TYPE_GRAY && bitDepth < 8)
     {
         log_debug("Setting grey bit depth(%d) to 8", bitDepth);
-        png_set_gray_1_2_4_to_8(_pngPtr);
+	#if PNG_LIBPNG_VER < 10400
+	    png_set_gray_1_2_4_to_8(_pngPtr);
+	#else
+	    png_set_expand_gray_1_2_4_to_8(_pngPtr);
+	#endif
     }
 
     // Apply the transparency block if it exists.
