--- src.orig/FilterHDF5.cc	2009-10-15 17:33:15.736674721 +0200
+++ src/FilterHDF5.cc	2009-10-15 17:23:51.386155522 +0200
@@ -76,7 +76,7 @@
 		printf(" GROUP \"%s\" \n", name);
 		numgroups++;
 
-		hid_t group = H5Gopen(loc_id, name);
+		hid_t group = H5Gopen(loc_id, name, H5P_DEFAULT);
 
 		int nrattr = H5Aget_num_attrs(group);
 		printf(" GROUP has %d attributes\n",nrattr);
@@ -129,7 +129,7 @@
 		printf("	DATASET \"%s\"\n", name);
 
 		hid_t dataset;
-		dataset = H5Dopen(loc_id, name);
+		dataset = H5Dopen(loc_id, name, H5P_DEFAULT);
 
 		numsets++;
 		data.resize(numsets);
@@ -260,7 +260,7 @@
 
 						int *matrix = new int[DY];
 						hsize_t tdims[] = {1};
-						hid_t s4_tid = H5Tarray_create(H5T_NATIVE_INT,1,tdims,NULL);
+						hid_t s4_tid = H5Tarray_create(H5T_NATIVE_INT,1,tdims);
 						hid_t s3_tid = H5Tcreate(H5T_COMPOUND, sizeof(s4_tid));
 						status = H5Tinsert(s3_tid, H5Tget_member_name (datatype,j), 0, s4_tid);
 
@@ -279,7 +279,7 @@
 
 						float *matrix = new float[DY];
 						hsize_t tdims[] = {1};
-						hid_t s4_tid = H5Tarray_create(H5T_NATIVE_FLOAT,1,tdims,NULL);
+						hid_t s4_tid = H5Tarray_create(H5T_NATIVE_FLOAT,1,tdims);
 						hid_t s3_tid = H5Tcreate(H5T_COMPOUND, sizeof(s4_tid));
 						status = H5Tinsert(s3_tid, H5Tget_member_name (datatype,j), 0, s4_tid);
 
@@ -467,7 +467,7 @@
 	hid_t file = H5Fopen(filename.latin1(), H5F_ACC_RDONLY, H5P_DEFAULT);
 
 	printf(" ROOT GROUP :\n");
-	hid_t group = H5Gopen(file, "/");
+	hid_t group = H5Gopen(file, "/", H5P_DEFAULT);
 
 	int nrattr = H5Aget_num_attrs(group);
 	attributes.resize(nrattr);
@@ -567,7 +567,8 @@
 				H5Tset_order(datatype, H5T_ORDER_BE);
 				break;
 			}
-			hid_t dataset = H5Dcreate(file,s->Title().latin1(),type,dataspace,H5P_DEFAULT);
+			hid_t dataset = H5Dcreate(file,s->Title().latin1(),type,dataspace,
+				H5P_DEFAULT, H5P_DEFAULT, H5P_DEFAULT);
 
 			hid_t status;
 			switch(datatype) {

