Index: src/gWindow.mli
===================================================================
--- src/gWindow.mli	(revision 1514)
+++ src/gWindow.mli	(revision 1515)
@@ -191,9 +191,10 @@
 
 (** Create popup windows
    @gtkdoc gtk GtkDialog *)
-class ['a] dialog_ext : [> Gtk.dialog] obj ->
+class ['a] dialog_ext : ([> Gtk.dialog] as 'b) obj ->
   object
     inherit ['a] dialog_skel
+    val obj : 'b obj
     method add_button : string -> 'a -> unit
     method add_button_stock : GtkStock.id -> 'a -> unit
   end
@@ -203,6 +204,7 @@
 class ['a] dialog : [> Gtk.dialog] obj ->
   object
     inherit ['a] dialog_ext
+    val obj : Gtk.dialog obj
     method connect : 'a dialog_signals
   end
 
Index: src/gButton.mli
===================================================================
--- src/gButton.mli	(revision 1514)
+++ src/gButton.mli	(revision 1515)
@@ -357,9 +357,10 @@
 (** @gtkdoc gtk GtkToolItem 
     @since GTK 2.4 *)
 class tool_item : 
-  [> Gtk.tool_item] obj ->
+  ([> Gtk.tool_item] as 'a) obj ->
   object
     inherit tool_item_skel
+    val obj : 'a obj
     method connect : GContainer.container_signals
   end
 
@@ -423,9 +424,10 @@
 (** @gtkdoc gtk GtkToolButton
     @since GTK 2.4 *)
 class tool_button : 
-  [> Gtk.tool_button] obj ->
+  ([> Gtk.tool_button] as 'a) obj ->
   object
     inherit tool_button_skel
+    val obj : 'a obj
     method connect : tool_button_signals
   end
 
Index: src/gAction.mli
===================================================================
--- src/gAction.mli	(revision 1514)
+++ src/gAction.mli	(revision 1515)
@@ -20,6 +20,10 @@
 (*                                                                        *)
 (**************************************************************************)
 
+(* $Id: lablgtk-2.14.1-ocaml312.patch,v 1.1 2010/08/03 20:38:29 aballier Exp $ *)
+
+open Gobject
+
 (** Action-based menus and toolbars *)
 
 (** {3 GtkAction} *)
@@ -27,9 +31,9 @@
 (** @since GTK 2.4
     @gtkdoc gtk GtkAction *)
 class action_signals :
-  ([> Gtk.action ] as 'b) Gobject.obj ->
+  ([> Gtk.action ] as 'b) obj ->
   object ('a)
-    val obj : 'b Gobject.obj
+    val obj : 'b obj
     val after : bool
     method after : < after : 'a; .. > as 'a
     method activate : callback:(unit -> unit) -> GtkSignal.id
@@ -38,10 +42,10 @@
 (** @since GTK 2.4
     @gtkdoc gtk GtkAction *)
 class action_skel :
-  ([> Gtk.action ] as 'a) Gobject.obj ->
+  ([> Gtk.action ] as 'a) obj ->
   object
-    val obj : 'a Gobject.obj
-    method as_action : Gtk.action Gobject.obj
+    val obj : 'a obj
+    method as_action : Gtk.action obj
 
     (** Properties *)
 
@@ -88,10 +92,10 @@
 (** @since GTK 2.4
     @gtkdoc gtk GtkAction *)
 class action :
-  ([> Gtk.action ] as 'a) Gobject.obj ->
+  ([> Gtk.action ] as 'a) obj ->
   object
     inherit action_skel
-    val obj : 'a Gobject.obj
+    val obj : 'a obj
     method connect : action_signals
   end
 
@@ -103,20 +107,20 @@
 (** @since GTK 2.4
     @gtkdoc gtk GtkToggleAction *)
 class toggle_action_signals :
-  ([> Gtk.toggle_action ] as 'b) Gobject.obj ->
+  ([> Gtk.toggle_action ] as 'b) obj ->
   object
     inherit action_signals
-    val obj : 'b Gobject.obj
+    val obj : 'b obj
     method toggled : callback:(unit -> unit) -> GtkSignal.id
   end
 
 (** @since GTK 2.4
     @gtkdoc gtk GtkToggleAction *)
 class toggle_action_skel :
-  ([> Gtk.toggle_action ] as 'a) Gobject.obj ->
+  ([> Gtk.toggle_action ] as 'a) obj ->
   object
     inherit action_skel
-    val obj : 'a Gobject.obj
+    val obj : 'a obj
     method draw_as_radio : bool
     method get_active : bool
     method set_active : bool -> unit
@@ -127,9 +131,10 @@
 (** @since GTK 2.4
     @gtkdoc gtk GtkToggleAction *)
 class toggle_action :
-  ([> Gtk.toggle_action ] as 'a) Gobject.obj ->
+  ([> Gtk.toggle_action ] as 'a) obj ->
   object
     inherit toggle_action_skel
+    val obj : 'a obj
     method connect : toggle_action_signals
   end
 
@@ -140,10 +145,10 @@
 (** @since GTK 2.4
     @gtkdoc gtk GtkRadioAction *)
 class radio_action_signals :
-  ([> Gtk.radio_action] as 'b) Gobject.obj ->
+  ([> Gtk.radio_action] as 'b) obj ->
   object
     inherit toggle_action_signals
-    val obj : 'b Gobject.obj
+    val obj : 'b obj
     method changed :
       callback:(int -> unit) -> GtkSignal.id
   end
@@ -151,12 +156,12 @@
 (** @since GTK 2.4
     @gtkdoc gtk GtkRadioAction *)
 class radio_action :
-  ([> Gtk.radio_action] as 'a) Gobject.obj ->
+  ([> Gtk.radio_action] as 'a) obj ->
   object
     inherit toggle_action_skel
-    val obj : 'a Gobject.obj
+    val obj : 'a obj
     method connect : radio_action_signals
-    method as_radio_action : Gtk.radio_action Gobject.obj
+    method as_radio_action : Gtk.radio_action obj
     method get_current_value : int
     method set_group : Gtk.radio_action Gtk.group -> unit
     method set_value : int -> unit
@@ -172,10 +177,10 @@
 (** @since GTK 2.4
     @gtkdoc gtk GtkActionGroup *)
 class action_group_signals :
-  ([> Gtk.action_group ] as 'b) Gobject.obj ->
+  ([> Gtk.action_group ] as 'b) obj ->
   object ('a)
     val after : bool
-    val obj : 'b Gobject.obj
+    val obj : 'b obj
     method after : 'a
     method connect_proxy : callback:(action -> GObj.widget -> unit) -> GtkSignal.id
     method disconnect_proxy : callback:(action -> GObj.widget -> unit) -> GtkSignal.id
@@ -186,10 +191,10 @@
 (** @since GTK 2.4
     @gtkdoc gtk GtkActionGroup *)
 class action_group :
-  ([> Gtk.action_group ] as 'a) Gobject.obj ->
+  ([> Gtk.action_group ] as 'a) obj ->
   object
-    val obj : 'a Gobject.obj
-    method as_group : Gtk.action_group Gobject.obj
+    val obj : 'a obj
+    method as_group : Gtk.action_group obj
     method connect : action_group_signals
     method sensitive : bool
     method set_sensitive : bool -> unit
Index: src/gnoCanvas.mli
===================================================================
--- src/gnoCanvas.mli	(revision 1514)
+++ src/gnoCanvas.mli	(revision 1515)
@@ -180,9 +180,10 @@
   end
 
 (** @gtkdoc libgnomecanvas GnomeCanvasItem *)
-class ['p] item : [> GnomeCanvas.item] Gtk.obj -> 
+class ['p] item : ([> GnomeCanvas.item] as 'a) Gtk.obj -> 
   object
     inherit base_item
+    val obj : 'a Gtk.obj
     constraint 'p = [< items_properties]
     method set : 'p list -> unit
   end
Index: src/gWindow.ml
===================================================================
--- src/gWindow.ml	(revision 1514)
+++ src/gWindow.ml	(revision 1515)
@@ -171,7 +171,7 @@
 end
 
 class ['a] dialog obj = object (self)
-  inherit ['a] dialog_ext obj
+  inherit ['a] dialog_ext (obj :> Gtk.dialog obj)
   method connect : 'a dialog_signals = new dialog_signals obj (self#decode)
 end
 
@@ -193,7 +193,7 @@
 type any_response = [GtkEnums.response | `OTHER of int]
 
 class dialog_any obj = object (self)
-  inherit [any_response] dialog_base obj
+  inherit [any_response] dialog_base (obj :> Gtk.dialog obj)
   method private encode = function
       `OTHER n -> n
     | #GtkEnums.response as v -> Dialog.std_response v
