--- avogadro-1.0.3/libavogadro/src/camera.cpp~	2012-02-27 18:35:52.109999585 +0000
+++ avogadro-1.0.3/libavogadro/src/camera.cpp	2012-02-27 18:36:27.103053453 +0000
@@ -25,6 +25,8 @@
 #include "camera.h"
 #include "glwidget.h"
 #include <avogadro/molecule.h>
+#include <Eigen/LU>
+#include <GL/glu.h>
 
 using namespace Eigen;
 
--- avogadro-1.0.3/libavogadro/src/glpainter_p.cpp~	2012-02-27 18:37:28.535894209 +0000
+++ avogadro-1.0.3/libavogadro/src/glpainter_p.cpp	2012-02-27 18:37:46.912413557 +0000
@@ -42,6 +42,7 @@
 #include <QColor>
 #include <QVarLengthArray>
 #include <Eigen/Geometry>
+#include <GL/glu.h>
 
 namespace Avogadro
 {
--- avogadro-1.0.3/libavogadro/src/glwidget.cpp~	2012-02-27 18:37:55.099014598 +0000
+++ avogadro-1.0.3/libavogadro/src/glwidget.cpp	2012-02-27 18:38:30.805395066 +0000
@@ -70,6 +70,8 @@
 
 #include <openbabel/mol.h>
 
+#include <GL/glu.h>
+
 using namespace OpenBabel;
 using namespace Eigen;
 
--- avogadro-1.0.3/libavogadro/src/extensions/povpainter.cpp~	2012-02-27 18:36:53.646173944 +0000
+++ avogadro-1.0.3/libavogadro/src/extensions/povpainter.cpp	2012-02-27 18:37:10.739370227 +0000
@@ -36,6 +36,7 @@
 #include <QFile>
 #include <QDebug>
 #include <Eigen/Geometry>
+#include <GL/glu.h>
 
 namespace Avogadro
 {
--- avogadro-1.0.3/libavogadro/src/tools/selectrotatetool.cpp~	2012-02-27 18:38:47.491928000 +0000
+++ avogadro-1.0.3/libavogadro/src/tools/selectrotatetool.cpp	2012-02-27 18:39:05.735115136 +0000
@@ -44,6 +44,8 @@
 #include <QComboBox>
 #include <QDebug>
 
+#include <GL/glu.h>
+
 using namespace std;
 using namespace OpenBabel;
 using namespace Eigen;
